<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.554">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="AJ Smit">
<meta name="dcterms.date" content="2022-08-08">
<meta name="description" content="The Biodiversity and Conservation Biology R Pages serving BCB744.">
<title>The Tangled Bank - BCB744: Introduction to R, and Biostatistics</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../BCB744/intro_r/01-RStudio.html" rel="next">
<link href="../images/favicon.ico" rel="icon">
<script src="../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-SCYH68M489"></script><script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-SCYH68M489', { 'anonymize_ip': true});
</script><script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"headline",
  "consent_type":"express",
  "palette":"dark",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  ,
"language":"en"
  });
});
</script><style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script><script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<meta property="og:title" content="The Tangled Bank - BCB744: Introduction to R, and Biostatistics">
<meta property="og:description" content="The Biodiversity and Conservation Biology R Pages serving BCB744.">
<meta property="og:image" content="https://tangledbank.netlify.app/BCB744/data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==">
<meta property="og:site_name" content="The Tangled Bank">
</head>
<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">The Tangled Bank</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-vignettes" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Vignettes</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-vignettes">
<li>
    <a class="dropdown-item" href="../vignettes/netCDF_dates.html">
 <span class="dropdown-text">Dates From netCDF Files: Two Approaches</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/prep_NOAA_OISST.html">
 <span class="dropdown-text">Downloading and Prep: ERDDAP</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_ERDDAP.html">
 <span class="dropdown-text">Retrieving Chlorophyll-a Data from ERDDAP Servers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/alt_method.html">
 <span class="dropdown-text">Downloading and Prep: Python + CDO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/gridded_data.html">
 <span class="dropdown-text">Detecting Events in Gridded Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/regridding.html">
 <span class="dropdown-text">Regridding gridded data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/MHW_MCS_horizonplots.html">
 <span class="dropdown-text">Displaying MHWs and MCSs as Horizon Plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/heatwaveR_issues.html">
 <span class="dropdown-text">heatwaveR Issues</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_sightings.html">
 <span class="dropdown-text">Plotting the Whale Sightings and Chlorophyll-<em>a</em> Concentrations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_localisation.html">
 <span class="dropdown-text">Spatial Localisation, Subsetting, and Aggregation of the Chlorophyll-<em>a</em> Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/elem_ts_methods.html">
 <span class="dropdown-text">Wavelet Analysis of Diatom Time Series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/buffer_data_extract.html">
 <span class="dropdown-text">Extracting gridded data within a buffer</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-high-performace-computing" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">High-Performace Computing</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-high-performace-computing">
<li>
    <a class="dropdown-item" href="../vignettes/README_Lengau.html">
 <span class="dropdown-text">Using Lengau</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/PBSPro_users.html">
 <span class="dropdown-text">PBSPro user and job management</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/README_tmux.html">
 <span class="dropdown-text">Using tmux</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item">
    <a class="nav-link" href="../blog.html"> 
<span class="menu-text">Tangled Bank Blog</span></a>
  </li>  
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-blogroll" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Blogroll</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-blogroll">
<li>
    <a class="dropdown-item" href="https://www.r-bloggers.com">
 <span class="dropdown-text">R-bloggers</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item compact">
    <a class="nav-link" href="../blog.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">
<li>
    <a class="dropdown-item" href="https://github.com/ajsmit/tangled_bank">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://bugs.com">
 <span class="dropdown-text">Report a Bug</span></a>
  </li>  
    </ul>
</li>
</ul>
</div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BCB744/BCB744_index.html">BCB744 (Intro R)</a></li><li class="breadcrumb-item"><a href="../BCB744/BCB744_index.html">BCB744: Introduction to R, and Biostatistics</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto"><div class="pt-lg-2 mt-2 text-end sidebar-header">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../images/hex_logo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none"></a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">BCB UNDERGRADUATE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
 <span class="menu-text">BDC223</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/BDC223_FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
 <span class="menu-text">BDC334</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/BDC334_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BDC334: Macroecology and Global Biogeography</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1: Ecological Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/02a-r_rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2a: R and RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/02b-env_dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2b: Environmental Distance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/03-biodiversity1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3: Quantifying Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/04-biodiversity2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4: Describing Biodiversity Patterns</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/Transboundary_systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transboundary systems</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">BCB HONOURS CORE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">BCB744 (Intro R)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/BCB744_index.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">BCB744: Introduction to R, and Biostatistics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/01-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. R and RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/02-working-with-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Working with data and code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/03-workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. R workflows</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/04-graphics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Graphics with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/05-faceting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Faceting figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/06-brewing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Brewing colours</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/07-mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Mapping with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/08-mapping_style.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Mapping with style</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/09-mapping_rnaturalearth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Mapping with Natural Earth and the <strong>sf</strong> package</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/10-mapping_quakes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. The Fiji Earthquake data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/11-tidy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Tidy data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/12-tidier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Tidier data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/13-tidiest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Tidiest data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/14-recap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Recap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/15-functions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15. Functions by chapter</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/16-base_r.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16. Base R primer</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/17-dates.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">17. Dates</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false">
 <span class="menu-text">BCB744 (Biostatistics)</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/BCB744_scientific_philosophy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Biostatistics and the philosophy of scientific inquiry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/01-data-in-R.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Data classes and structures in R</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/02-summarise-and-describe.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Exploring with summaries and descriptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/03-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Exploring with figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/04-distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/05-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Statistical inference and hypothesis testing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/06-assumptions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Assumptions for parametric statistics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/07-t_tests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. <em>t</em>-tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/08-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Analysis of Variance (ANOVA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/09-regressions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Simple linear regressions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/10-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/11-decision_guide.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. A guide to selecting the right parametric test</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/12-glance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Non-parametric statistical tests at a glance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/13-confidence.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Confidence intervals</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/14-transformations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Data transformations</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">BCB HONOURS ELECTIVES</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">
 <span class="menu-text">BCB743 (Quantitative Ecology)</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/BCB743_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Overview &amp; Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/review.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1-4: Review Concepts fo Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5: Correlations &amp; Associations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/dis-metrics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6: Distance Metrics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7: Intro to Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8a: Principal Components Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA_examples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8b: PCA: Additional Examples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA_SDG_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8c: PCA: WHO SDG Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/CA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9: Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCoA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10: Principal Coordinates Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/nMDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11a: Non-metric Multidimensional Scaling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/nMDS_diatoms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11b: nMDS: PERMANOVA (Diatoms) Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/unconstrained-summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12: Unconstrained Ordination Summary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/model_building.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13: Model Building</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/multiple_regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14a: Multiple Regression</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/deep_dive.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14b: Gradients Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/constrained_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15a: Distance-Based Redundancy Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/Seaweed_in_Two_Oceans_v2/two_oceans_appendices.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15b: db-RDA: Seaweeds Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/cluster_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16: Cluster Analysis</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">SUPPORT</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
 <span class="menu-text">Web Resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/general_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/ecology_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quantitative Ecology Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/spatial_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial R Resources</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#workshop-schedule" id="toc-workshop-schedule" class="nav-link active" data-scroll-target="#workshop-schedule"><span class="header-section-number">1</span> 2024 Workshop Schedule</a></li>
  <li><a href="#venue" id="toc-venue" class="nav-link" data-scroll-target="#venue"><span class="header-section-number">2</span> Venue</a></li>
  <li><a href="#honours-coordinator" id="toc-honours-coordinator" class="nav-link" data-scroll-target="#honours-coordinator"><span class="header-section-number">3</span> Honours Coordinator</a></li>
  <li><a href="#course-coordinator" id="toc-course-coordinator" class="nav-link" data-scroll-target="#course-coordinator"><span class="header-section-number">4</span> Course Coordinator</a></li>
  <li><a href="#instructors" id="toc-instructors" class="nav-link" data-scroll-target="#instructors"><span class="header-section-number">5</span> Instructors</a></li>
  <li><a href="#sec-descr" id="toc-sec-descr" class="nav-link" data-scroll-target="#sec-descr"><span class="header-section-number">6</span> Course description</a></li>
  <li>
<a href="#course-timetable-and-content" id="toc-course-timetable-and-content" class="nav-link" data-scroll-target="#course-timetable-and-content"><span class="header-section-number">7</span> Course timetable and content</a>
  <ul class="collapse">
<li><a href="#core-theoretical-content-and-philosophical-framework-of-the-intro-r-workshop" id="toc-core-theoretical-content-and-philosophical-framework-of-the-intro-r-workshop" class="nav-link" data-scroll-target="#core-theoretical-content-and-philosophical-framework-of-the-intro-r-workshop"><span class="header-section-number">7.1</span> Core theoretical content and philosophical framework of the Intro R Workshop</a></li>
  <li><a href="#core-statistical-content-of-the-biostatistics-workshop" id="toc-core-statistical-content-of-the-biostatistics-workshop" class="nav-link" data-scroll-target="#core-statistical-content-of-the-biostatistics-workshop"><span class="header-section-number">7.2</span> Core statistical content of the Biostatistics Workshop</a></li>
  </ul>
</li>
  <li><a href="#core-skills-developed-by-bcb744" id="toc-core-skills-developed-by-bcb744" class="nav-link" data-scroll-target="#core-skills-developed-by-bcb744"><span class="header-section-number">8</span> Core skills developed by BCB744</a></li>
  <li><a href="#graduate-attributes" id="toc-graduate-attributes" class="nav-link" data-scroll-target="#graduate-attributes"><span class="header-section-number">9</span> Graduate attributes</a></li>
  <li><a href="#data-used-in-support-of-the-module" id="toc-data-used-in-support-of-the-module" class="nav-link" data-scroll-target="#data-used-in-support-of-the-module"><span class="header-section-number">10</span> Data used in support of the module</a></li>
  <li><a href="#prerequisites" id="toc-prerequisites" class="nav-link" data-scroll-target="#prerequisites"><span class="header-section-number">11</span> Prerequisites</a></li>
  <li><a href="#method-of-instruction" id="toc-method-of-instruction" class="nav-link" data-scroll-target="#method-of-instruction"><span class="header-section-number">12</span> Method of instruction</a></li>
  <li><a href="#sec-collab-learning" id="toc-sec-collab-learning" class="nav-link" data-scroll-target="#sec-collab-learning"><span class="header-section-number">13</span> Learning colaboratively</a></li>
  <li><a href="#reusing-code-found-elsewhere" id="toc-reusing-code-found-elsewhere" class="nav-link" data-scroll-target="#reusing-code-found-elsewhere"><span class="header-section-number">14</span> Reusing code found elsewhere</a></li>
  <li><a href="#openai-chatgpt-and-other-ai-tools" id="toc-openai-chatgpt-and-other-ai-tools" class="nav-link" data-scroll-target="#openai-chatgpt-and-other-ai-tools"><span class="header-section-number">15</span> OpenAI ChatGPT and other AI tools</a></li>
  <li><a href="#software" id="toc-software" class="nav-link" data-scroll-target="#software"><span class="header-section-number">16</span> Software</a></li>
  <li><a href="#computers" id="toc-computers" class="nav-link" data-scroll-target="#computers"><span class="header-section-number">17</span> Computers</a></li>
  <li><a href="#attendance" id="toc-attendance" class="nav-link" data-scroll-target="#attendance"><span class="header-section-number">18</span> Attendance</a></li>
  <li>
<a href="#sec-policy" id="toc-sec-policy" class="nav-link" data-scroll-target="#sec-policy"><span class="header-section-number">19</span> Assessment policy</a>
  <ul class="collapse">
<li><a href="#daily-tasks" id="toc-daily-tasks" class="nav-link" data-scroll-target="#daily-tasks"><span class="header-section-number">19.1</span> Daily Tasks</a></li>
  <li><a href="#sec-summative" id="toc-sec-summative" class="nav-link" data-scroll-target="#sec-summative"><span class="header-section-number">19.2</span> Summative Tasks</a></li>
  <li><a href="#intro-r-assessment" id="toc-intro-r-assessment" class="nav-link" data-scroll-target="#intro-r-assessment"><span class="header-section-number">19.3</span> Intro R Assessment</a></li>
  <li><a href="#integrative-final-assessment" id="toc-integrative-final-assessment" class="nav-link" data-scroll-target="#integrative-final-assessment"><span class="header-section-number">19.4</span> Integrative (Final) Assessment</a></li>
  <li><a href="#submission-of-tasks-assessments-and-the-final-assessment" id="toc-submission-of-tasks-assessments-and-the-final-assessment" class="nav-link" data-scroll-target="#submission-of-tasks-assessments-and-the-final-assessment"><span class="header-section-number">19.5</span> Submission of Tasks, Assessments, and the Final Assessment</a></li>
  <li><a href="#late-submission-of-tasks" id="toc-late-submission-of-tasks" class="nav-link" data-scroll-target="#late-submission-of-tasks"><span class="header-section-number">19.6</span> Late submission of Tasks</a></li>
  </ul>
</li>
  <li>
<a href="#support" id="toc-support" class="nav-link" data-scroll-target="#support"><span class="header-section-number">20</span> Support</a>
  <ul class="collapse">
<li><a href="#help-via-bcb744-issues-on-github" id="toc-help-via-bcb744-issues-on-github" class="nav-link" data-scroll-target="#help-via-bcb744-issues-on-github"><span class="header-section-number">20.1</span> Help via BCB744 Issues on GitHub</a></li>
  </ul>
</li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB744/BCB744_index.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BCB744/BCB744_index.html">BCB744 (Intro R)</a></li><li class="breadcrumb-item"><a href="../BCB744/BCB744_index.html">BCB744: Introduction to R, and Biostatistics</a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">BCB744: Introduction to R, and Biostatistics</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>

<div>
  <div class="description">
    The Biodiversity and Conservation Biology R Pages serving BCB744.
  </div>
</div>

<div class="quarto-title-meta-author column-body">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://ajsmit.netlify.app">AJ Smit</a> <a href="https://orcid.org/0000-0002-3799-6126" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uwc.ac.za">
            University of the Western Cape
            </a>
          </p>
      </div>
  </div>

<div class="quarto-title-meta column-body">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">August 8, 2022</p>
    </div>
  </div>
  
    
  </div>
  


</header><p><img src="../images/PhD_comics.JPG" class="img-fluid" style="width:100.0%"></p>
<blockquote class="blockquote">
<p><em>“Most people use statistics like a drunk man uses a lamppost; more for support than illumination.”</em></p>
<p>— Andrew Lang</p>
</blockquote>
<blockquote class="blockquote">
<p><em>“If your experiment needs a statistician, you need a better experiment.”</em></p>
<p>— Ernest Rutherford</p>
</blockquote>
<p><strong>Welcome to the pages for BCB744. This page provides the syllabus and teaching policies for the module, and it serves as a starting point for accessing all the theory, instruction, and data.</strong></p>
<section id="workshop-schedule" class="level1" data-number="1"><h1 data-number="1">
<span class="header-section-number">1</span> 2024 Workshop Schedule</h1>
<ul>
<li>
<strong>Intro to R</strong>: 5th - 9th February</li>
<li>
<strong>Biostatistics</strong>: 25th - 28th March (during the mid-semester break of Semester 1)</li>
</ul></section><section id="venue" class="level1" data-number="2"><h1 data-number="2">
<span class="header-section-number">2</span> Venue</h1>
<p>5th Floor Computer Lab, BCB Department, Life Sciences Building, University of the Western Cape.</p>
</section><section id="honours-coordinator" class="level1" data-number="3"><h1 data-number="3">
<span class="header-section-number">3</span> Honours Coordinator</h1>
<p><strong>Prof.&nbsp;Bryan Maritz</strong>: Room 4.105, Department of Biodiversity &amp; Conservation Biology</p>
</section><section id="course-coordinator" class="level1" data-number="4"><h1 data-number="4">
<span class="header-section-number">4</span> Course Coordinator</h1>
<p><strong>Prof.&nbsp;AJ Smit</strong>: Room 4.103, Department of Biodiversity &amp; Conservation Biology, ajsmit@uwc.ac.za</p>
</section><section id="instructors" class="level1" data-number="5"><h1 data-number="5">
<span class="header-section-number">5</span> Instructors</h1>
<p><strong>Cayley Cammell</strong>: Department of Biodiversity &amp; Conservation Biology, 4269088@myuwc.ac.za</p>
<p><strong>Zoë-Angelique Petersen</strong>: Department of Biodiversity &amp; Conservation Biology, 4042512@myuwc.ac.za</p>
</section><section id="sec-descr" class="level1" data-number="6"><h1 data-number="6">
<span class="header-section-number">6</span> Course description</h1>
<p>Yes, the comma in this page’s title is correct: “BCB744: Introduction to R, and Biostatistics.” The module provides an introduction to the R software and language. I will also teach biostatistics.</p>
<p>This is a core module in your Honours programme. You will learn to use R for data analysis, visualisation, and statistical inference. You will also learn fundamental biostatistics concepts, such as hypothesis testing, probabilities, confidence intervals, regression analysis, Analysis of Variance, and other staples of biostatistics. I will use real-world datasets from the biological, ecological, and environmental fields that you can use to practice applying your R and biostatistics skills.</p>
<p>The approach taken in this Workshop is not dissimilar from a course in Data Science. However, in this Workshop, we won’t do data science, but we will use R do actually <strong>do</strong> science. There is a difference! Any scientist that can use R is also ideally equipped to be a data scientist, and some people who have completed this module actually do just that. The difference between the two ideas, philosophies, careers is provided in the box immediately below.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
‘Real’ Scientists and Data ‘Scientists’
</div>
</div>
<div class="callout-body-container callout-body">
<p>A <strong>Scientist</strong> able to apply their intermediate to advanced R skills is by default also a ‘Data Scientist’. The opposite is generally not true: Data Scientists are not <em>real</em> Scientists—especially after only having completed ‘traditional’ courses in data science.</p>
<p><strong>Science</strong> refers to the application of the <em>scientific method</em> of conducting research, where hypotheses are proposed, experiments are designed and conducted to test these hypotheses, and data are collected and analysed to draw conclusions. The aim of Science is to generate <em>new knowledge and understanding</em> of the natural world. A Scientist will typically be equipped to work through all of these steps.</p>
<p><strong>Data Science</strong>, on the other hand, involves the use of computational and statistical tools to extract knowledge and insights from data. These datasets typically already exist because someone (companies, industries, NGOs, etc.) collected them. Data Science focuses on analysing large and complex datasets to uncover patterns, trends, and relationships that can be used to inform decision-making. The Data Scientist is not typically involved in generating the data from <em>de novo</em>.</p>
<p>These key aspects summarise the difference between the two fields:</p>
<ul>
<li><p><strong>Approach</strong> Science is hypothesis-driven, while Data Science is data-driven. Science begins with a hypothesis that is tested through experiments, while Data Science begins with data and uses statistical and computational methods to uncover insights.</p></li>
<li><p><strong>Goals</strong> Science aims to generate new knowledge and understanding of the natural world, while Data Science aims to uncover insights and make predictions based on existing data. Scientist focus on understanding the underlying mechanisms of natural phenomena and their area of focus is the real world, while Data Scientists focus on extracting knowledge and insights from data, often in the realm of business.</p></li>
<li><p><strong>Methods</strong> Science involves making observations of the world, conducting experiments, collecting and analysing data, and drawing conclusions based on the results. Data Science <em>only</em> involves using statistical and computational tools to analyse data and uncover patterns and relationships.</p></li>
<li><p><strong>Context</strong> Science is typically focused on a specific domain, such as biology, chemistry, or physics. Data Science can be applied to any domain that involves data, including business, finance, healthcare, and social media.</p></li>
</ul>
</div>
</div>
</section><section id="course-timetable-and-content" class="level1" data-number="7"><h1 data-number="7">
<span class="header-section-number">7</span> Course timetable and content</h1>
<p>The Workshop has two sub-components. The first, the <strong>Intro R Workshop</strong> will take place at the very start of your Honours year. The second is called <strong>Biostatistics</strong> and it usually starts in late March or early April. This year’s timetable is provided below.</p>
<table class="table">
<thead><tr class="header">
<th>Wk</th>
<th>Lecture</th>
<th>Topic</th>
<th>Class Date</th>
<th>Tasks/Assessment</th>
<th>Task/Assess. due</th>
</tr></thead>
<tbody>
<tr class="odd">
<td></td>
<td></td>
<td><strong>INTRO R</strong></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Wk1</td>
<td>L1</td>
<td>Introduction</td>
<td>5 Feb ’24</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><a href="intro_r/01-RStudio.html">1. R and RStudio</a></td>
<td></td>
<td>Task A</td>
<td>6 Feb ’24</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td><a href="intro_r/02-working-with-data.html">2. Working with data and code</a></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><a href="intro_r/03-workflow.html">3. R workflows</a></td>
<td></td>
<td>Task B</td>
<td>7 Feb ’24</td>
</tr>
<tr class="even">
<td></td>
<td>L2</td>
<td><a href="intro_r/04-graphics.html">4. Graphics with ggplot2</a></td>
<td>6 Feb ’24</td>
<td>Task C</td>
<td>7 Feb ’24</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><a href="intro_r/05-faceting.html">5. Faceting figures</a></td>
<td></td>
<td>Task D</td>
<td>7 Feb ’24</td>
</tr>
<tr class="even">
<td></td>
<td>L3</td>
<td><a href="intro_r/06-brewing.html">6. Brewing colours</a></td>
<td>7 Feb ’24</td>
<td>Task E</td>
<td>8 Feb ’24</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><a href="intro_r/07-mapping.html">7. Mapping with ggplot2</a></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td><a href="intro_r/08-mapping_style.html">8. Mapping with style</a></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><a href="intro_r/09-mapping_rnaturalearth.html">9. Mapping with Natural Earth and the sf package</a></td>
<td></td>
<td>Task F</td>
<td>8 Feb ’24</td>
</tr>
<tr class="even">
<td></td>
<td>Self</td>
<td><a href="intro_r/10-mapping_quakes.html">10. The Fiji Earthquake data</a></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td></td>
<td></td>
<td>Bonus Task</td>
<td>1 Apr ’24</td>
</tr>
<tr class="even">
<td></td>
<td>L4</td>
<td><a href="intro_r/11-tidy.html">11. Tidy data</a></td>
<td>8 Feb ’24</td>
<td>Task G</td>
<td>1 Mar ’24</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><a href="intro_r/12-tidier.html">12. Tidier data</a></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td><a href="intro_r/13-tidiest.html">13. Tidiest data</a></td>
<td></td>
<td><a href="../assessments/Intro-R_CA_Summative_Tasks.html">Summative Task 1</a></td>
<td>1 &amp; 25 Mar ’24</td>
</tr>
<tr class="odd">
<td></td>
<td>L5</td>
<td>Recap</td>
<td>9 Feb ’24</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Wk2</td>
<td></td>
<td></td>
<td>16 Feb ’24</td>
<td><a href="../assessments/Intro-R_CA_Assessment.html">Intro R Assessment</a></td>
<td>6 Mar ’24</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><strong>BIOSTATISTICS</strong></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Wk10</td>
<td>L1</td>
<td><a href="../BCB744/basic_stats/BCB744_scientific_philosophy.html">Biostatistics and the philosophy of scientific inquiry</a></td>
<td>25 Mar ’24</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><a href="../BCB744/basic_stats/01-data-in-R.html">1. Data classes and structures in R</a></td>
<td></td>
<td>Task A</td>
<td>26 Mar ’24</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td><a href="../BCB744/basic_stats/02-summarise-and-describe.html">2. Exploring with summaries and descriptions</a></td>
<td></td>
<td>Task B</td>
<td>26 Mar ’24</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><a href="../BCB744/basic_stats/03-visualise.html">3. Exploring with figures</a></td>
<td></td>
<td>Task C</td>
<td>26 Mar ’24</td>
</tr>
<tr class="even">
<td></td>
<td>L2</td>
<td><a href="../BCB744/basic_stats/04-distributions.html">4. Data distributions</a></td>
<td>26 Mar ’24</td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><a href="../BCB744/basic_stats/05-inference.html">5. Statistical inference and hypothesis testing</a></td>
<td></td>
<td>Task D</td>
<td>27 Mas ’24</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td><a href="../BCB744/basic_stats/06-assumptions.html">6. Assumptions</a></td>
<td></td>
<td><a href="../assessments/BCB744_Summative_Task_2_2024.html">Summative Task 2</a></td>
<td>12 Apr ’24</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><a href="../BCB744/basic_stats/07-t_tests.html">7. Inferences about one or two populations</a></td>
<td></td>
<td>Task E</td>
<td>27 Mar ’24</td>
</tr>
<tr class="even">
<td></td>
<td>L3</td>
<td><a href="../BCB744/basic_stats/08-anova.html">8. Analysis of Variance (ANOVA)</a></td>
<td>27 Mar ’24</td>
<td>Task F</td>
<td>28 Mar ’24</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><a href="../BCB744/basic_stats/09-regressions.html">9. Simple linear regressions</a></td>
<td></td>
<td>Task G</td>
<td>28 Mar ’24</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td><a href="../BCB744/basic_stats/10-correlations.html">10. Correlations</a></td>
<td></td>
<td>Task H</td>
<td>28 Mar ’24</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><a href="../BCB744/basic_stats/11-decision_guide.html">11. A guide to selecting the right parametric test</a></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td><a href="../BCB744/basic_stats/12-glance.html">12. Non-Parametric statistics</a></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td><a href="../BCB744/basic_stats/13-confidence.html">13. Confidence intervals</a></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td><a href="../BCB744/basic_stats/14-transformations.html">14. Data transformations</a></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Wk12</td>
<td></td>
<td></td>
<td>11 Apr ’24</td>
<td><a href="../assessments/Final_Integrative_Assessment_2024.html">Final Integrative Assessment</a></td>
<td>14 Apr ’24</td>
</tr>
</tbody>
</table>
<!-- |       |           | [14. Power and sample size calculations](basic_stats/14-power.qmd)                        |               |                                                                       |                   | --><!-- Insert table here --><section id="core-theoretical-content-and-philosophical-framework-of-the-intro-r-workshop" class="level2" data-number="7.1"><h2 data-number="7.1" class="anchored" data-anchor-id="core-theoretical-content-and-philosophical-framework-of-the-intro-r-workshop">
<span class="header-section-number">7.1</span> Core theoretical content and philosophical framework of the Intro R Workshop</h2>
<p>The Intro R Workshop focuses on the functionality offered by the <a href="https://www.tidyverse.org/"><strong>tidyverse</strong></a> suite of packages. I designed the Workshop to introduce you to a powerful set of tools for data manipulation, exploration, and visualisation. The <strong>tidyverse</strong> is a collection of R packages that work together to provide a cohesive set of functions for manipulating data. This course will cover the most popular packages in the <strong>tidyverse</strong>, including <strong>tidyr</strong> for data reshaping, <strong>dplyr</strong> for data ‘wrangling’, and <strong>ggplot2</strong> for data visualisation. You will learn how to clean, transform, and visualise data, as well as how to use these tools to build reproducible and informative data analysis pipelines. With a focus on practical application and hands-on exercises, you will gain the skills and knowledge needed to effectively use the tidyverse in your own data analysis projects.</p>
<p>One of the key heuristic devices I use throughout is to focus on figures, particularly maps. Maps can be a powerful tool for teaching coding because they combine both aesthetics and information in a visually compelling format. Maps are not only beautiful, but they also provide insights and context that can be difficult to grasp through numbers or text alone. By working with code to create your own maps, you can learn programming concepts and techniques while also developing your visual literacy skills. This engaging and interactive approach to learning coding can help to demystify the subject and make it more accessible when you don’t have a background in coding.</p>
</section><section id="core-statistical-content-of-the-biostatistics-workshop" class="level2" data-number="7.2"><h2 data-number="7.2" class="anchored" data-anchor-id="core-statistical-content-of-the-biostatistics-workshop">
<span class="header-section-number">7.2</span> Core statistical content of the Biostatistics Workshop</h2>
<p>In biological and ecological sciences, statistical methods play a crucial role in analysing and interpreting data. Some of the basic statistical methods used include:</p>
<ul>
<li><p><strong>Descriptive statistics</strong> These methods are used to summarise and describe the basic features of a dataset, such as the mean, median, and standard deviation.</p></li>
<li><p><strong>Inferential statistics</strong> These allow you, the scientist, to make predictions and inferences about a population based on a sample of data. Common inferential statistical techniques include <em>t</em>-tests, ANOVA, and regression analysis.</p></li>
<li><p><strong>Non-parametric statistics</strong> Non-parametric methods are called for when the data do not meet the assumptions of parametric statistics. Examples of non-parametric techniques include Wilcoxon rank-sum test and Kruskal-Wallis test.</p></li>
</ul></section></section><section id="core-skills-developed-by-bcb744" class="level1" data-number="8"><h1 data-number="8">
<span class="header-section-number">8</span> Core skills developed by BCB744</h1>
<p>By the end of this module, you will be able to:</p>
<ul>
<li>Understand and use use R within the RStudio IDE</li>
<li>Know and understand the the <strong>tidyverse</strong> suite of functions and approach to data analysis and graphics</li>
<li>Understand the principles underlying <em>tidy data</em>
</li>
<li>Understand the types of data and data distributions that biologists and ecologists will frequently encounter</li>
<li>Understand and be able to execute the most frequently used inferential statistics</li>
<li>Use the R software and associated packages to undertake these analyses</li>
<li>Interpret the outcomes of these analyses and use it to probabilistically make inferences about the scientific enquiries</li>
<li>Communicate the findings by written and oral means</li>
</ul></section><section id="graduate-attributes" class="level1" data-number="9"><h1 data-number="9">
<span class="header-section-number">9</span> Graduate attributes</h1>
<p>The <a href="../pages/graduate_attributes.html"><strong>graduate attributes</strong></a> resulting from completion of this modules alignment with the expectations of the workspace across diverse organisations and institutions where graduates typically find employment.</p>
</section><section id="data-used-in-support-of-the-module" class="level1" data-number="10"><h1 data-number="10">
<span class="header-section-number">10</span> Data used in support of the module</h1>
<p>All the data required for BCB744 may be <a href="../data/Archive.zip">downloaded here</a>.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Built-in data
</div>
</div>
<div class="callout-body-container callout-body">
<p>R has many built-in datasets that are useful for practicing our R skills. To find out which datasets are available to you on your system, execute the following command. Help files for each of the datasets are also available:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># load the data like this:</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">data</span>()</span>
<span id="cb1-3"><a href="#cb1-3"></a></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="co"># find help, for example:</span></span>
<span id="cb1-5"><a href="#cb1-5"></a>?datasets<span class="sc">::</span>ChickWeight</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</div>
</div>
<p>It is important to use these (or any) datasets to practice your R skills on. Actively engaging with my comprehensive and detailed web pages, and practising on the included and additional other datasets will make to difference between a 60% average mark for the module, and a mark in excess of 80%.</p>
</section><section id="prerequisites" class="level1" data-number="11"><h1 data-number="11">
<span class="header-section-number">11</span> Prerequisites</h1>
<p>You should have a moderate numerical literacy, but prior programming experience is not required. In all sciences, practical problem solving skills and a tenacity for challenges are crucial for success. Scientific disciplines constantly evolve and present new and complex problems that require creative and innovative solutions. You will have to demonstrate agile and adaptive approaches to solving challenges, and you must have the ability to break down complex problems into smaller parts and approach them systematically. You must also be able to identify and overcome roadblocks, and be persistent in your efforts to find a solution. These attributes will allow you to be effective in this module.</p>
</section><section id="method-of-instruction" class="level1" data-number="12"><h1 data-number="12">
<span class="header-section-number">12</span> Method of instruction</h1>
<p>The workshop is designed to be as interactive as possible, so while you are working on exercises the tutor and I will circulate among you and engage with you to help you understand any material and the associated code you are uncomfortable with. Often this will result in discussions of novel applications and alternative approaches to the data analysis challenges you are required to solve. More challenging concepts might emerge during the Tasks and Assignments (typically these will be submitted the following day), and any such challenges will be dealt with in class prior to learning new concepts.</p>
<p>Although the module ultimately supports the application of biologically-oriented statistics, a large part of it is also about coding. It is up to you to take your coding skills to the next level and move beyond what I teach in class. Coding is a bit like learning a language, and as such programming is a skill that is best learned by doing.</p>
</section><section id="sec-collab-learning" class="level1" data-number="13"><h1 data-number="13">
<span class="header-section-number">13</span> Learning colaboratively</h1>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Also read: How to learn
</div>
</div>
<div class="callout-body-container callout-body">
<p>Please refer to my <a href="../pages/How_to_learn.html">advice about how to learn</a>.</p>
</div>
</div>
<p>Collaborative learning provides an opportunity for you to work together and learn from each other. In this way, you will develop a deeper understanding of the subject matter. Collaborating with your friends and peers allows you to explore different perspectives and ideas, which can broaden your understanding and help you to see the subject matter from new angles. This type of learning environment also fosters the development of important skills such as communication, teamwork, and leadership, which are essential for success in academic and professional careers. Collaborative learning can create a sense of community and support among your group of peers. In the end, it will enhance your university experience, drive your love for learning, and prepare you for success beyond the university.</p>
<p>Discuss the BCB744 Workshop activities with your peers as you work on them. Use the WhatsApp group set up for the module for discussion purposes (I might assist via this medium if necessary if your questions/comments have relevance to the whole class). A better option is to use <a href="../quantecol/#help-via-bcb744-and-bcb743-issues-on-github">GitHub Issues</a>. You will learn more in this module if you work with your friends than if you do not. Ask questions, answer questions, and share ideas liberally. Please identify your work partners by name on all assignments (if you decide to work in pairs).</p>
<p><strong>Cooperative learning is not a licence for plagiarism. Plagiarism is a serious offence and will be dealt with concisely. Consequences of cheating are severe—they range from a 0% for the assignment or exam up to dismissal from the course for a second offense.</strong></p>
</section><section id="reusing-code-found-elsewhere" class="level1" data-number="14"><h1 data-number="14">
<span class="header-section-number">14</span> Reusing code found elsewhere</h1>
<p>A huge volume of code is available on the web and it can be adapted to solve your own problems. You may make use of any online resources (e.g.&nbsp;form <a href="https://stackoverflow.com/">StackOverflow</a>, a thoroughly-used source of discussion about <a href="https://stackoverflow.com/questions/tagged/r">R code</a>)—but you <strong>MUST</strong> clearly indicate (cite) that your solution relies on found code, regardless to what extent you have modified it to your own needs. Reused code that is discovered via a web search and which is not explicitly cited is plagiarism and it will be treated as such. On assignments you may not directly share code with your peers in this workshop.</p>
</section><section id="openai-chatgpt-and-other-ai-tools" class="level1" data-number="15"><h1 data-number="15">
<span class="header-section-number">15</span> OpenAI ChatGPT and other AI tools</h1>
<p>The 2024 BSc (Hons) cohort will be the first to experience the use of AI tools in the BCB744 module. The use of AI tools is a new and exciting development and it is important that you are exposed to these tools. The use of AI tools will be limited to the use of the OpenAI ChatGPT tool, which may be used to generate ‘proto-code’ that will assist you in becoming familiar with the R langauge. We will explore ideas together, and the mark allocation to tasks and assignments will be adjusted accoringly.</p>
</section><section id="software" class="level1" data-number="16"><h1 data-number="16">
<span class="header-section-number">16</span> Software</h1>
<p>In this course you will rely entirely on <a href="https://cran.r-project.org/">R</a> running within the <a href="https://www.rstudio.com/">RStudio</a> IDE. The use of R is covered extensively in the <a href="https://tangledbank.netlify.app/bcb744/intro_r/01-rstudio">BCB744</a> module where the <a href="https://tangledbank.netlify.app/BCB744/intro_r/01-RStudio">installation process</a> is discussed.</p>
<p>Additionally, the very basics—i.e.&nbsp;about R, RStudio, packages, their installation, etc.—can also be found on the <a href="https://moderndive.netlify.app/1-getting-started.html">ModernDive</a> website. A slightly longer and more detailed account of the installation process and the very basics is provided on the <a href="https://www.datacamp.com/tutorial/r-studio-tutorial">datacamp</a> platform.</p>
<p><a href="https://moderndive.netlify.app">ModernDive</a> also provides a nice overview of using R for data science.</p>
<p>For more in-depth coverage of the R language, refer to R Master <a href="https://r4ds.hadley.nz/">Hadley Wickham’s</a> pages. There you will find everything you need to know in a well thought through presentation. Thoroughly working through this material, page by page, will quickly make you a R Master yourself (well, almost).</p>
</section><section id="computers" class="level1" data-number="17"><h1 data-number="17">
<span class="header-section-number">17</span> Computers</h1>
<p>You are encouraged to provide your own laptops and to install the necessary software before the module starts. Limited support can be provided if required, but in the end, the onus is on you to understand how your computer works (from the filesystem through to dealing with software installation issues). There are also computers with R and RStudio (and the necessary add-on libraries) available in the 5th floor lab in the BCB Department.</p>
</section><section id="attendance" class="level1" data-number="18"><h1 data-number="18">
<span class="header-section-number">18</span> Attendance</h1>
<p>This workshop-based, hands on course can only deliver acceptible outcomes if you attend all classes. The schedule is set and cannot be changed. Sometimes an occasional absence cannot be avoided. Please be curtious and notify myself or the tutor in advance of any absence. If you work with a partner in class, notify them too. Keep up with the reading assignments while you are away and we will all work with you to get you back up to speed on what you miss. If you do miss a class, however, the assignments must still be submitted on time (also see <a href="#late-submission-of-ca"><strong>Late submission of CA</strong></a>).</p>
<p>Since you may decide to work in collaboration with a peer on tasks and assignments, please keep this person informed at all times in case some emergency makes you unavailable for a period of time. Someone might depend on your input and contributions—do not leave someone in the lurch so that they cannot complete a task in your absence.</p>
</section><section id="sec-policy" class="level1" data-number="19"><h1 data-number="19">
<span class="header-section-number">19</span> Assessment policy</h1>
<p><strong>Continuous Assessment (CA)</strong> and a <strong>Final Assessment</strong> will provide a <strong>Final Mark</strong> for the module. These modes of assessment meet our needs as far as <a href="../pages/assessment_theory.html">formative and summative assessments</a> are concerned. The weighting of the CA and the Final Assessment is 0.6 and 0.4, respectively. All assessments are open book, so consult your code and reading material if and when you need to.</p>
<table class="table">
<colgroup>
<col style="width: 62%">
<col style="width: 13%">
<col style="width: 23%">
</colgroup>
<thead><tr class="header">
<th>Assessment Component</th>
<th>Weighting</th>
<th>Contribution (%)</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>CONTINUOUS ASSESSMENT</strong></td>
<td>(60)</td>
<td></td>
</tr>
<tr class="even">
<td><strong>Introduction to R</strong></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Daily Tasks (A-G)</td>
<td></td>
<td>7.5</td>
</tr>
<tr class="even">
<td>Summative Task 1</td>
<td></td>
<td>12.5</td>
</tr>
<tr class="odd">
<td>Intro R Assessment</td>
<td></td>
<td>30</td>
</tr>
<tr class="even">
<td><strong>Biostatistics</strong></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Daily Tasks (A-G)</td>
<td></td>
<td>7.5</td>
</tr>
<tr class="even">
<td>Summative Task 2</td>
<td></td>
<td>42.5</td>
</tr>
<tr class="odd">
<td><em>Total</em></td>
<td></td>
<td><em>100</em></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><strong>FINAL ASSESSMENT</strong></td>
<td>(40)</td>
<td></td>
</tr>
<tr class="even">
<td><em>Integrative Assessment (Intro R + Biostatistics)</em></td>
<td></td>
<td><em>100</em></td>
</tr>
</tbody>
</table>
<p>You must work alone on all these Tasks. Care must be taken that the Tasks are submitted as instructed, i.e.&nbsp;paying attention to naming conventions and the format of the files submitted—typically this will be in a Quarto document (.qmd) and the knitted output (I prefer .html).</p>
<p>Random quizzes will not form part of the CA for BCB744.</p>
<section id="daily-tasks" class="level2" data-number="19.1"><h2 data-number="19.1" class="anchored" data-anchor-id="daily-tasks">
<span class="header-section-number">19.1</span> Daily Tasks</h2>
<p>Intro R and Biostatistics require that we work with real-world datasets. To this end, a series of Daily Tasks involving real data is a required part of the BCB744. These are interwoven into the daily Chapters, and your feedback is required the following day. The Daily Tasks are part of the CA.</p>
<p>When assessing the tasks, we will pay attention to the following criteria:</p>
<ul>
<li>Content (10%):
<ul>
<li>Questions answered in order</li>
<li>Annotations (meta-data in file header, commnets about code, ideas, and approach)</li>
</ul>
</li>
<li>Code formatting and correctness (45%):
<ul>
<li>Application of <a href="http://adv-r.had.co.nz/Style.html">R code conventions</a>, e.g.&nbsp;spaces around <code>&lt;-</code>, after <code>#</code>, after <code>,</code>, etc.</li>
<li>New line for each <code>dplyr</code> function (lines end in <code>%&gt;%</code>) or <code>ggplot</code> layer (lines end in <code>+</code>)</li>
</ul>
</li>
<li>Figures (45%):
<ul>
<li>Sensible use of themes / colours</li>
<li>Publication quality</li>
<li>Informative and complete titles, axes labels, legends, etc.</li>
</ul>
</li>
</ul></section><section id="sec-summative" class="level2" data-number="19.2"><h2 data-number="19.2" class="anchored" data-anchor-id="sec-summative">
<span class="header-section-number">19.2</span> Summative Tasks</h2>
<p>At the end of Intro R and Biostatistics, the more demanding Summative Tasks 1 and 2 will be required. As the table above indicates, the weighting of the Summative Tasks is more than that of the Daily Tasks but they also differ slightly between the two parts of the module. These assessments will take place over a few days and you may work at home or on campus. Like the Daily Tasks, the Summative Tasks are also open book assessments. The Summative Tasks form part of the CA and will prepare you for the Intro R Assessment (at the end of Intro R) and the Integrative Assessment (after Biostatistics; see ‘Final Assessment’, below).</p>
<p>We will assess the Summative Tasks as per the assessment breakdown provided under the Daily Tasks.</p>
</section><section id="intro-r-assessment" class="level2" data-number="19.3"><h2 data-number="19.3" class="anchored" data-anchor-id="intro-r-assessment">
<span class="header-section-number">19.3</span> Intro R Assessment</h2>
<p>The Intro R Assessment will, as the name suggests, be about the material covered in the Introduction to R section of the work only. It forms part of the CA.</p>
<p>The Intro R Assessment will be graded with the following in expectations in mind:</p>
<ul>
<li>Content (20%):
<ul>
<li>Questions answered in order</li>
<li>A written explanation of approach included for each question</li>
<li>Appropriate formatting of text, for example, fonts not larger than necessary, headings used properly, etc. Be sensible and tasteful.</li>
</ul>
</li>
<li>Code formatting, structure, and correctness (50%):
<ul>
<li>Use <strong>Tidyverse</strong> code</li>
<li>No more than ~80 characters of code per line (pay particular attention to the comments)</li>
<li>Application of <a href="http://adv-r.had.co.nz/Style.html">R code conventions</a>, e.g.&nbsp;spaces around <code>&lt;-</code>, after <code>#</code>, after <code>,</code>, etc.</li>
<li>New line for each <code>dplyr</code> function (lines end in <code>%&gt;%</code>) or <code>ggplot</code> layer (lines end in <code>+</code>)</li>
<li>Proper indentation of pipes and <code>ggplot()</code> layers</li>
<li>All chunks labelled without spaces</li>
<li>No unwanted / commented out code left behind in the document</li>
</ul>
</li>
<li>Figures (30%):
<ul>
<li>Sensible use of themes / colours</li>
<li>Publication quality</li>
<li>Informative and complete titles, axes labels, legends, etc.</li>
<li>No redundant features or aesthetics</li>
</ul>
</li>
</ul></section><section id="integrative-final-assessment" class="level2" data-number="19.4"><h2 data-number="19.4" class="anchored" data-anchor-id="integrative-final-assessment">
<span class="header-section-number">19.4</span> Integrative (Final) Assessment</h2>
<p>The Integrative Assessment is the Final Assessment. As such, it will test your skills broadly across both Intro R <em>and</em> Biostatistics, and hence its weighting causes it to contribute more towards your final grade as it falls outside of the pool of marks that form your CA. The Integrative Assessment may be up to five days in duration. It will involve the analysis of real world data. Some of the Questions might expect (as per Question) that you write 1) statements of aims, objectives, and hypotheses; 2) the full and detailed methods followed by analyses together with all code, 3) full reporting of results in a manner suited for peer reviewed publications; 4) graphical support highlighting the patterns observed (again with the code), and 5) a discussion if and when required. The weighting of marks to these various sections is:</p>
<ol type="1">
<li>Aims, objectives, and hypotheses: 5%</li>
<li>Methods and analyses: 45%</li>
<li>Results: 20%</li>
<li>Graphs: 15%</li>
<li>Discussion: 15%</li>
</ol>
<p>Other Questions might be shorter in nature, designed to specifically test important aspects of BCB744. Such Question might be worth anything from 10 to 50 marks, depending on the nature of the Questions.</p>
<p>The Integrative Assessment is also open book. Go home. Look at the questions. Answer them at home. Submit them by the deadline.</p>
</section><section id="submission-of-tasks-assessments-and-the-final-assessment" class="level2" data-number="19.5"><h2 data-number="19.5" class="anchored" data-anchor-id="submission-of-tasks-assessments-and-the-final-assessment">
<span class="header-section-number">19.5</span> Submission of Tasks, Assessments, and the Final Assessment</h2>
<p>A statement such as the one below accompanies every assignment—pay attention, as failing to observe this instruction may result in a loss of marks (i.e.&nbsp;if an assignment remains ungraded because the owner of the material cannot be identified):</p>
<p>Submit a Quarto script wherein you provide answers to Questions by no later than 8:00 the following data (or the Monday in cases when assignments were given on Fridays). Label the script as follows (e.g.): <strong>BCB744_AJ_Smit_Task_A.qmd</strong>.</p>
</section><section id="late-submission-of-tasks" class="level2" data-number="19.6"><h2 data-number="19.6" class="anchored" data-anchor-id="late-submission-of-tasks">
<span class="header-section-number">19.6</span> Late submission of Tasks</h2>
<p>Late assignments will be penalised 10% per day and will not be accepted more than 48 hours late, unless evidence such as a doctor’s note, a death certificate, or another documented emergency can be provided. If you know in advance that a submission will be late, please discuss this and seek prior approval. This policy is based on the idea that in order to learn how to translate your human thoughts into computer language (coding) you should be working with them at multiple times each week—ideally daily. Time has been allocated in class for working on assignments and students are expected to continue to work on the assignments outside of class. Successfully completing (and passing) this module requires that you finish assignments based on what we have covered in class by the following class period. Work diligently from the onset so that even if something unexpected happens at the last minute you should already be close to done. This approach also allows rapid feedback to be provided to you, which can only be accomplished by returning assignments quickly and punctually.</p>
</section></section><section id="support" class="level1" data-number="20"><h1 data-number="20">
<span class="header-section-number">20</span> Support</h1>
<p>It’s expected that some tricky aspects of the module will take time to master, and the best way to master problematic material is to practice, practice some more, and then to ask questions. Trying for 10 minutes and then giving up is not good enough. I’ll be more sympathetic to your cause if you can demonstrate having tried for a full day before giving up and asking me. When you ask questions about some challenge, this is the way to do it—explain to me your numerous attempts at trying to solve the problem, and explain how these various attempts have failed. <em>I will not help you if you have not tried to help yourself first</em> (maybe with advice from friends). There will be time in class to do this, typically before we embark on a new topic. You are also encouraged to bring up related questions that arise in your own B.Sc. (Hons.) research project.</p>
<p>Should you require more time with me, find out when I am ‘free’ and set an appointment by sending me a calendar invitation. I am happy to have a personal meeting with you via Zoom, but I prefer face-to-face in my office.</p>
<section id="help-via-bcb744-issues-on-github" class="level2" data-number="20.1"><h2 data-number="20.1" class="anchored" data-anchor-id="help-via-bcb744-issues-on-github">
<span class="header-section-number">20.1</span> Help via BCB744 Issues on GitHub</h2>
<p>All discussion for the BCB744 and BCB743 workshops will be held in the <a href="https://github.com/ajsmit/R_courses/issues">Issues</a> of <a href="https://github.com/ajsmit/R_courses">this repository</a>. Please post all content-related questions there, and use email only for personal matters. Note that this is a public repository, so be professional in your writing here (grammar, etc.).</p>
<p>To start a new thread, create a <strong>New issue</strong>. Tag your peers using their handle—<code>@ajsmit</code>, for example—to get their attention.</p>
<p>Once a question has been answered, the issue will be closed, so lots of good answers might end up in closed issues. Don’t forget to look there when looking for answers—you can use the <strong>Search</strong> feature on this repository to find answers that might have been offered by the same or similar problem experienced by someone else in the past.</p>
<p><strong>Guidelines for posting questions:</strong></p>
<ul>
<li>First search existing issues (open or closed) for answers. If the question has already been answered, you’re done! If there is an open issue, feel free to contribute to it. Or feel free to open a closed issue if you believe the answer is not satisfactory.</li>
<li>Give your issue an informative title.
<ul>
<li>Good: “Error: could not find function”ggplot””</li>
<li>Bad: “My code does not work!” Note that you can edit an issue’s title after it’s been posted.</li>
</ul>
</li>
<li>Format your questions nicely using markdown and code formatting. Preview your issue prior to posting.</li>
<li>As I explained above, your peers and I will more sympathetic to your cause if you can show <em>all the things you have tried as you, yourself, tried to fix the issue first</em>.</li>
<li>Include code and example data so the person trying to help you have something to work with (and which results in the error, perhaps)</li>
<li>Where appropriate, provide links to specific files, or even lines within them, in the body of your issue. This will help your peers understand your question. Note that only the teaching team will have access to private repos.</li>
<li>(Optional) Tag someone or some group of people. Start by typing their GitHub username prefixed with the @ symbol. Of course this supposes that each of you have a GitHub account and username.</li>
<li>Hit <strong>Submit new issue</strong> when you’re ready to post.</li>
</ul>


<!-- -->

</section></section><div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-reuse"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a></div></div></section><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{smit2022,
  author = {Smit, AJ},
  title = {BCB744: {Introduction} to {R,} and {Biostatistics}},
  date = {2022-08-08},
  url = {https://tangledbank.netlify.app/BCB744/BCB744_index.html},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-smit2022" class="csl-entry quarto-appendix-citeas" role="listitem">
Smit A (2022) BCB744: Introduction to R, and Biostatistics. <a href="https://tangledbank.netlify.app/BCB744/BCB744_index.html">https://tangledbank.netlify.app/BCB744/BCB744_index.html.
</a>
</div></div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/tangledbank\.netlify\.app");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
          // default icon
          link.classList.add("external");
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation column-body"><div class="nav-page nav-page-previous">
  </div>
  <div class="nav-page nav-page-next">
      <a href="../BCB744/intro_r/01-RStudio.html" class="pagination-link" aria-label="1. R and RStudio">
        <span class="nav-page-text">1. R and RStudio</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb2" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb2-1"><a href="#cb2-1"></a><span class="co">---</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="an">date:</span><span class="co"> "2022-08-08"</span></span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="an">title:</span><span class="co"> "BCB744: Introduction to R, and Biostatistics"</span></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="an">description:</span><span class="co"> The Biodiversity and Conservation Biology R Pages serving BCB744.</span></span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="co">---</span></span>
<span id="cb2-6"><a href="#cb2-6"></a></span>
<span id="cb2-7"><a href="#cb2-7"></a><span class="al">![](../images/PhD_comics.JPG)</span>{width=100%}</span>
<span id="cb2-8"><a href="#cb2-8"></a></span>
<span id="cb2-9"><a href="#cb2-9"></a><span class="at">&gt; *"Most people use statistics like a drunk man uses a lamppost; more for support than illumination."*</span></span>
<span id="cb2-10"><a href="#cb2-10"></a><span class="at">&gt;</span></span>
<span id="cb2-11"><a href="#cb2-11"></a><span class="at">&gt; --- Andrew Lang</span></span>
<span id="cb2-12"><a href="#cb2-12"></a></span>
<span id="cb2-13"><a href="#cb2-13"></a><span class="at">&gt; *"If your experiment needs a statistician, you need a better experiment."*</span></span>
<span id="cb2-14"><a href="#cb2-14"></a><span class="at">&gt;</span></span>
<span id="cb2-15"><a href="#cb2-15"></a><span class="at">&gt; --- Ernest Rutherford</span></span>
<span id="cb2-16"><a href="#cb2-16"></a></span>
<span id="cb2-17"><a href="#cb2-17"></a></span>
<span id="cb2-18"><a href="#cb2-18"></a></span>
<span id="cb2-19"><a href="#cb2-19"></a>**Welcome to the pages for BCB744. This page provides the syllabus and teaching policies for the module, and it serves as a starting point for accessing all the theory, instruction, and data.**</span>
<span id="cb2-20"><a href="#cb2-20"></a></span>
<span id="cb2-21"><a href="#cb2-21"></a><span class="fu"># 2024 Workshop Schedule</span></span>
<span id="cb2-22"><a href="#cb2-22"></a></span>
<span id="cb2-23"><a href="#cb2-23"></a><span class="ss">* </span>**Intro to R**: 5th - 9th February</span>
<span id="cb2-24"><a href="#cb2-24"></a><span class="ss">* </span>**Biostatistics**: 25th - 28th March (during the mid-semester break of Semester 1)</span>
<span id="cb2-25"><a href="#cb2-25"></a></span>
<span id="cb2-26"><a href="#cb2-26"></a><span class="fu"># Venue</span></span>
<span id="cb2-27"><a href="#cb2-27"></a>5th Floor Computer Lab, BCB Department, Life Sciences Building, University of the Western Cape.</span>
<span id="cb2-28"><a href="#cb2-28"></a></span>
<span id="cb2-29"><a href="#cb2-29"></a><span class="fu"># Honours Coordinator</span></span>
<span id="cb2-30"><a href="#cb2-30"></a></span>
<span id="cb2-31"><a href="#cb2-31"></a>**Prof. Bryan Maritz**: Room 4.105, Department of Biodiversity &amp; Conservation Biology</span>
<span id="cb2-32"><a href="#cb2-32"></a></span>
<span id="cb2-33"><a href="#cb2-33"></a><span class="fu"># Course Coordinator</span></span>
<span id="cb2-34"><a href="#cb2-34"></a></span>
<span id="cb2-35"><a href="#cb2-35"></a>**Prof. AJ Smit**: Room 4.103, Department of Biodiversity &amp; Conservation Biology, ajsmit@uwc.ac.za</span>
<span id="cb2-36"><a href="#cb2-36"></a></span>
<span id="cb2-37"><a href="#cb2-37"></a><span class="fu"># Instructors</span></span>
<span id="cb2-38"><a href="#cb2-38"></a></span>
<span id="cb2-39"><a href="#cb2-39"></a>**Cayley Cammell**: Department of Biodiversity &amp; Conservation Biology, 4269088@myuwc.ac.za</span>
<span id="cb2-40"><a href="#cb2-40"></a></span>
<span id="cb2-41"><a href="#cb2-41"></a>**Zoë-Angelique Petersen**: Department of Biodiversity &amp; Conservation Biology, 4042512@myuwc.ac.za</span>
<span id="cb2-42"><a href="#cb2-42"></a></span>
<span id="cb2-43"><a href="#cb2-43"></a><span class="fu"># Course description {#sec-descr}</span></span>
<span id="cb2-44"><a href="#cb2-44"></a></span>
<span id="cb2-45"><a href="#cb2-45"></a>Yes, the comma in this page's title is correct: "BCB744: Introduction to R, and Biostatistics." The module provides an introduction to the R software and language. I will also teach biostatistics.</span>
<span id="cb2-46"><a href="#cb2-46"></a></span>
<span id="cb2-47"><a href="#cb2-47"></a>This is a core module in your Honours programme. You will learn to use R for data analysis, visualisation, and statistical inference. You will also learn fundamental biostatistics concepts, such as hypothesis testing, probabilities, confidence intervals, regression analysis, Analysis of Variance, and other staples of biostatistics. I will use real-world datasets from the biological, ecological, and environmental fields that you can use to practice applying your R and biostatistics skills.</span>
<span id="cb2-48"><a href="#cb2-48"></a></span>
<span id="cb2-49"><a href="#cb2-49"></a>The approach taken in this Workshop is not dissimilar from a course in Data Science. However, in this Workshop, we won't do data science, but we will use R do actually **do** science. There is a difference! Any scientist that can use R is also ideally equipped to be a data scientist, and some people who have completed this module actually do just that. The difference between the two ideas, philosophies, careers is provided in the box immediately below.</span>
<span id="cb2-50"><a href="#cb2-50"></a></span>
<span id="cb2-51"><a href="#cb2-51"></a>::: {.callout-note  appearance="simple"}</span>
<span id="cb2-52"><a href="#cb2-52"></a><span class="fu">## 'Real' Scientists and Data 'Scientists' {#sec-datascience}</span></span>
<span id="cb2-53"><a href="#cb2-53"></a></span>
<span id="cb2-54"><a href="#cb2-54"></a>A **Scientist** able to apply their intermediate to advanced R skills is by default also a 'Data Scientist'. The opposite is generally not true: Data Scientists are not *real* Scientists---especially after only having completed 'traditional' courses in data science.</span>
<span id="cb2-55"><a href="#cb2-55"></a></span>
<span id="cb2-56"><a href="#cb2-56"></a>**Science** refers to the application of the *scientific method* of conducting research, where hypotheses are proposed, experiments are designed and conducted to test these hypotheses, and data are collected and analysed to draw conclusions. The aim of Science is to generate *new knowledge and understanding* of the natural world. A Scientist will typically be equipped to work through all of these steps.</span>
<span id="cb2-57"><a href="#cb2-57"></a></span>
<span id="cb2-58"><a href="#cb2-58"></a>**Data Science**, on the other hand, involves the use of computational and statistical tools to extract knowledge and insights from data. These datasets typically already exist because someone (companies, industries, NGOs, etc.) collected them. Data Science focuses on analysing large and complex datasets to uncover patterns, trends, and relationships that can be used to inform decision-making. The Data Scientist is not typically involved in generating the data from *de novo*.</span>
<span id="cb2-59"><a href="#cb2-59"></a></span>
<span id="cb2-60"><a href="#cb2-60"></a>These key aspects summarise the difference between the two fields:</span>
<span id="cb2-61"><a href="#cb2-61"></a></span>
<span id="cb2-62"><a href="#cb2-62"></a><span class="ss">-   </span>**Approach** Science is hypothesis-driven, while Data Science is data-driven. Science begins with a hypothesis that is tested through experiments, while Data Science begins with data and uses statistical and computational methods to uncover insights.</span>
<span id="cb2-63"><a href="#cb2-63"></a></span>
<span id="cb2-64"><a href="#cb2-64"></a><span class="ss">-   </span>**Goals** Science aims to generate new knowledge and understanding of the natural world, while Data Science aims to uncover insights and make predictions based on existing data. Scientist focus on understanding the underlying mechanisms of natural phenomena and their area of focus is the real world, while Data Scientists focus on extracting knowledge and insights from data, often in the realm of business.</span>
<span id="cb2-65"><a href="#cb2-65"></a></span>
<span id="cb2-66"><a href="#cb2-66"></a><span class="ss">-   </span>**Methods** Science involves making observations of the world, conducting experiments, collecting and analysing data, and drawing conclusions based on the results. Data Science *only* involves using statistical and computational tools to analyse data and uncover patterns and relationships.</span>
<span id="cb2-67"><a href="#cb2-67"></a></span>
<span id="cb2-68"><a href="#cb2-68"></a><span class="ss">-   </span>**Context** Science is typically focused on a specific domain, such as biology, chemistry, or physics. Data Science can be applied to any domain that involves data, including business, finance, healthcare, and social media.</span>
<span id="cb2-69"><a href="#cb2-69"></a>:::</span>
<span id="cb2-70"><a href="#cb2-70"></a></span>
<span id="cb2-71"><a href="#cb2-71"></a><span class="fu"># Course timetable and content</span></span>
<span id="cb2-72"><a href="#cb2-72"></a></span>
<span id="cb2-73"><a href="#cb2-73"></a>The Workshop has two sub-components. The first, the **Intro R Workshop** will take place at the very start of your Honours year. The second is called **Biostatistics** and it usually starts in late March or early April. This year's timetable is provided below.</span>
<span id="cb2-74"><a href="#cb2-74"></a></span>
<span id="cb2-75"><a href="#cb2-75"></a>| Wk    | Lecture   | Topic                                                                                     | Class Date    | Tasks/Assessment                                                      | Task/Assess. due  |</span>
<span id="cb2-76"><a href="#cb2-76"></a>|-------|-----------|-------------------------------------------------------------------------------------------|---------------|-----------------------------------------------------------------------|-------------------|</span>
<span id="cb2-77"><a href="#cb2-77"></a>|       |           | **INTRO R**                                                                               |               |                                                                       |                   |</span>
<span id="cb2-78"><a href="#cb2-78"></a>| Wk1   | L1        | Introduction                                                                              | 5 Feb '24     |                                                                       |                   |</span>
<span id="cb2-79"><a href="#cb2-79"></a>|       |           | <span class="co">[</span><span class="ot">1. R and RStudio</span><span class="co">](intro_r/01-RStudio.html)</span>                                               |               | Task A                                                                | 6 Feb '24         |</span>
<span id="cb2-80"><a href="#cb2-80"></a>|       |           | <span class="co">[</span><span class="ot">2. Working with data and code</span><span class="co">](intro_r/02-working-with-data.html)</span>                        |               |                                                                       |                   |</span>
<span id="cb2-81"><a href="#cb2-81"></a>|       |           | <span class="co">[</span><span class="ot">3. R workflows</span><span class="co">](intro_r/03-workflow.html)</span>                                                |               | Task B                                                                | 7 Feb '24         |</span>
<span id="cb2-82"><a href="#cb2-82"></a>|       | L2        | <span class="co">[</span><span class="ot">4. Graphics with ggplot2</span><span class="co">](intro_r/04-graphics.html)</span>                                      | 6 Feb '24     | Task C                                                                | 7 Feb '24         |</span>
<span id="cb2-83"><a href="#cb2-83"></a>|       |           | <span class="co">[</span><span class="ot">5. Faceting figures</span><span class="co">](intro_r/05-faceting.html)</span>                                           |               | Task D                                                                | 7 Feb '24         |</span>
<span id="cb2-84"><a href="#cb2-84"></a>|       | L3        | <span class="co">[</span><span class="ot">6. Brewing colours</span><span class="co">](intro_r/06-brewing.html)</span>                                             | 7 Feb '24     | Task E                                                                | 8 Feb '24         |</span>
<span id="cb2-85"><a href="#cb2-85"></a>|       |           | <span class="co">[</span><span class="ot">7. Mapping with ggplot2</span><span class="co">](intro_r/07-mapping.html)</span>                                        |               |                                                                       |                   |</span>
<span id="cb2-86"><a href="#cb2-86"></a>|       |           | <span class="co">[</span><span class="ot">8. Mapping with style</span><span class="co">](intro_r/08-mapping_style.html)</span>                                    |               |                                                                       |                   |</span>
<span id="cb2-87"><a href="#cb2-87"></a>|       |           | <span class="co">[</span><span class="ot">9. Mapping with Natural Earth and the sf package</span><span class="co">](intro_r/09-mapping_rnaturalearth.html)</span> |               | Task F                                                                | 8 Feb '24         | </span>
<span id="cb2-88"><a href="#cb2-88"></a>|       | Self      | <span class="co">[</span><span class="ot">10. The Fiji Earthquake data</span><span class="co">](intro_r/10-mapping_quakes.html)</span>                            |               |                                                                       |                   |</span>
<span id="cb2-89"><a href="#cb2-89"></a>|       |           |                                                                                           |               | Bonus Task                                                            | 1 Apr '24         |</span>
<span id="cb2-90"><a href="#cb2-90"></a>|       | L4        | <span class="co">[</span><span class="ot">11. Tidy data</span><span class="co">](intro_r/11-tidy.html)</span>                                                     | 8 Feb '24     | Task G                                                                | 1 Mar '24         |</span>
<span id="cb2-91"><a href="#cb2-91"></a>|       |           | <span class="co">[</span><span class="ot">12. Tidier data</span><span class="co">](intro_r/12-tidier.html)</span>                                                 |               |                                                                       |                   |</span>
<span id="cb2-92"><a href="#cb2-92"></a>|       |           | <span class="co">[</span><span class="ot">13. Tidiest data</span><span class="co">](intro_r/13-tidiest.html)</span>                                               |               | <span class="co">[</span><span class="ot">Summative Task 1</span><span class="co">](../assessments/Intro-R_CA_Summative_Tasks.Rmd)</span>     | 1 &amp; 25 Mar '24    |</span>
<span id="cb2-93"><a href="#cb2-93"></a>|       | L5        | Recap                                                                                     | 9 Feb '24     |                                                                       |                   |</span>
<span id="cb2-94"><a href="#cb2-94"></a>| Wk2   |           |                                                                                           | 16 Feb '24    | <span class="co">[</span><span class="ot">Intro R Assessment</span><span class="co">](../assessments/Intro-R_CA_Assessment.html)</span>       | 6 Mar '24         |</span>
<span id="cb2-95"><a href="#cb2-95"></a>|       |           | **BIOSTATISTICS**                                                                         |               |                                                                       |                   |</span>
<span id="cb2-96"><a href="#cb2-96"></a>| Wk10  | L1        | <span class="co">[</span><span class="ot">Biostatistics and the philosophy of scientific inquiry</span><span class="co">](basic_stats/BCB744_scientific_philosophy.qmd)</span> | 25 Mar '24 |                                                             |                   |</span>
<span id="cb2-97"><a href="#cb2-97"></a>|       |           | <span class="co">[</span><span class="ot">1. Data classes and structures in R</span><span class="co">](basic_stats/01-data-in-R.qmd)</span>                       |               | Task A                                                                | 26 Mar '24        |</span>
<span id="cb2-98"><a href="#cb2-98"></a>|       |           | <span class="co">[</span><span class="ot">2. Exploring with summaries and descriptions</span><span class="co">](basic_stats/02-summarise-and-describe.qmd)</span> |               | Task B                                                                | 26 Mar '24        |</span>
<span id="cb2-99"><a href="#cb2-99"></a>|       |           | <span class="co">[</span><span class="ot">3. Exploring with figures</span><span class="co">](basic_stats/03-visualise.qmd)</span>                                 |               | Task C                                                                | 26 Mar '24        |</span>
<span id="cb2-100"><a href="#cb2-100"></a>|       | L2        | <span class="co">[</span><span class="ot">4. Data distributions</span><span class="co">](basic_stats/04-distributions.qmd)</span>                                 | 26 Mar '24    |                                                                       |                   |</span>
<span id="cb2-101"><a href="#cb2-101"></a>|       |           | <span class="co">[</span><span class="ot">5. Statistical inference and hypothesis testing</span><span class="co">](basic_stats/05-inference.qmd)</span>           |               | Task D                                                                | 27 Mas '24        |</span>
<span id="cb2-102"><a href="#cb2-102"></a>|       |           | <span class="co">[</span><span class="ot">6. Assumptions</span><span class="co">](basic_stats/06-assumptions.qmd)</span>                                          |               | <span class="co">[</span><span class="ot">Summative Task 2</span><span class="co">](../assessments/BCB744_Summative_Task_2_2024.Rmd)</span>   | 12 Apr '24        |</span>
<span id="cb2-103"><a href="#cb2-103"></a>|       |           | <span class="co">[</span><span class="ot">7. Inferences about one or two populations</span><span class="co">](basic_stats/07-t_tests.qmd)</span>                  |               | Task E                                                                | 27 Mar '24        |</span>
<span id="cb2-104"><a href="#cb2-104"></a>|       | L3        | <span class="co">[</span><span class="ot">8. Analysis of Variance (ANOVA)</span><span class="co">](basic_stats/08-anova.qmd)</span>                               | 27 Mar '24    | Task F                                                                | 28 Mar '24        |</span>
<span id="cb2-105"><a href="#cb2-105"></a>|       |           | <span class="co">[</span><span class="ot">9. Simple linear regressions</span><span class="co">](basic_stats/09-regressions.qmd)</span>                            |               | Task G                                                                | 28 Mar '24        |</span>
<span id="cb2-106"><a href="#cb2-106"></a>|       |           | <span class="co">[</span><span class="ot">10. Correlations</span><span class="co">](basic_stats/10-correlations.qmd)</span>                                       |               | Task H                                                                | 28 Mar '24        |</span>
<span id="cb2-107"><a href="#cb2-107"></a>|       |           | <span class="co">[</span><span class="ot">11. A guide to selecting the right parametric test</span><span class="co">](basic_stats/11-decision_guide.qmd)</span>   |               |                                                                       |                   |</span>
<span id="cb2-108"><a href="#cb2-108"></a>|       |           | <span class="co">[</span><span class="ot">12. Non-Parametric statistics</span><span class="co">](basic_stats/12-glance.qmd)</span>                                |               |                                                                       |                   |</span>
<span id="cb2-109"><a href="#cb2-109"></a>|       |           | <span class="co">[</span><span class="ot">13. Confidence intervals</span><span class="co">](basic_stats/13-confidence.qmd)</span>                                 |               |                                                                       |                   |</span>
<span id="cb2-110"><a href="#cb2-110"></a>|       |           | <span class="co">[</span><span class="ot">14. Data transformations</span><span class="co">](basic_stats/14-transformations.qmd)</span>                            |               |                                                                       |                   |</span>
<span id="cb2-111"><a href="#cb2-111"></a>| Wk12  |           |                                                                                           | 11 Apr '24    | <span class="co">[</span><span class="ot">Final Integrative Assessment</span><span class="co">](../assessments/Final_Integrative_Assessment_2024.qmd)</span>| 14 Apr '24           |</span>
<span id="cb2-112"><a href="#cb2-112"></a><span class="co">&lt;!-- |       |           | [14. Power and sample size calculations](basic_stats/14-power.qmd)                        |               |                                                                       |                   | --&gt;</span></span>
<span id="cb2-113"><a href="#cb2-113"></a></span>
<span id="cb2-114"><a href="#cb2-114"></a></span>
<span id="cb2-115"><a href="#cb2-115"></a><span class="co">&lt;!-- Insert table here --&gt;</span></span>
<span id="cb2-116"><a href="#cb2-116"></a></span>
<span id="cb2-117"><a href="#cb2-117"></a><span class="fu">## Core theoretical content and philosophical framework of the Intro R Workshop</span></span>
<span id="cb2-118"><a href="#cb2-118"></a>The Intro R Workshop focuses on the functionality offered by the <span class="co">[</span><span class="ot">**tidyverse**</span><span class="co">](https://www.tidyverse.org/)</span> suite of packages. I designed the Workshop to introduce you to a powerful set of tools for data manipulation, exploration, and visualisation. The **tidyverse** is a collection of R packages that work together to provide a cohesive set of functions for manipulating data. This course will cover the most popular packages in the **tidyverse**, including **tidyr** for data reshaping, **dplyr** for data 'wrangling', and **ggplot2** for data visualisation. You will learn how to clean, transform, and visualise data, as well as how to use these tools to build reproducible and informative data analysis pipelines. With a focus on practical application and hands-on exercises, you will gain the skills and knowledge needed to effectively use the tidyverse in your own data analysis projects.</span>
<span id="cb2-119"><a href="#cb2-119"></a></span>
<span id="cb2-120"><a href="#cb2-120"></a>One of the key heuristic devices I use throughout is to focus on figures, particularly maps. Maps can be a powerful tool for teaching coding because they combine both aesthetics and information in a visually compelling format. Maps are not only beautiful, but they also provide insights and context that can be difficult to grasp through numbers or text alone. By working with code to create your own maps, you can learn programming concepts and techniques while also developing your visual literacy skills. This engaging and interactive approach to learning coding can help to demystify the subject and make it more accessible when you don't have a background in coding.</span>
<span id="cb2-121"><a href="#cb2-121"></a></span>
<span id="cb2-122"><a href="#cb2-122"></a><span class="fu">## Core statistical content of the Biostatistics Workshop</span></span>
<span id="cb2-123"><a href="#cb2-123"></a>In biological and ecological sciences, statistical methods play a crucial role in analysing and interpreting data. Some of the basic statistical methods used include:</span>
<span id="cb2-124"><a href="#cb2-124"></a></span>
<span id="cb2-125"><a href="#cb2-125"></a><span class="ss">-   </span>**Descriptive statistics** These methods are used to summarise and describe the basic features of a dataset, such as the mean, median, and standard deviation.</span>
<span id="cb2-126"><a href="#cb2-126"></a></span>
<span id="cb2-127"><a href="#cb2-127"></a><span class="ss">-   </span>**Inferential statistics** These allow you, the scientist, to make predictions and inferences about a population based on a sample of data. Common inferential statistical techniques include *t*-tests, ANOVA, and regression analysis.</span>
<span id="cb2-128"><a href="#cb2-128"></a></span>
<span id="cb2-129"><a href="#cb2-129"></a><span class="ss">-   </span>**Non-parametric statistics** Non-parametric methods are called for when the data do not meet the assumptions of parametric statistics. Examples of non-parametric techniques include Wilcoxon rank-sum test and Kruskal-Wallis test.</span>
<span id="cb2-130"><a href="#cb2-130"></a></span>
<span id="cb2-131"><a href="#cb2-131"></a><span class="fu"># Core skills developed by BCB744</span></span>
<span id="cb2-132"><a href="#cb2-132"></a></span>
<span id="cb2-133"><a href="#cb2-133"></a>By the end of this module, you will be able to:</span>
<span id="cb2-134"><a href="#cb2-134"></a></span>
<span id="cb2-135"><a href="#cb2-135"></a><span class="ss">- </span>Understand and use use R within the RStudio IDE</span>
<span id="cb2-136"><a href="#cb2-136"></a><span class="ss">- </span>Know and understand the the **tidyverse** suite of functions and approach to data analysis and graphics</span>
<span id="cb2-137"><a href="#cb2-137"></a><span class="ss">- </span>Understand the principles underlying *tidy data*</span>
<span id="cb2-138"><a href="#cb2-138"></a><span class="ss">- </span>Understand the types of data and data distributions that biologists and ecologists will frequently encounter</span>
<span id="cb2-139"><a href="#cb2-139"></a><span class="ss">- </span>Understand and be able to execute the most frequently used inferential statistics</span>
<span id="cb2-140"><a href="#cb2-140"></a><span class="ss">- </span>Use the R software and associated packages to undertake these analyses</span>
<span id="cb2-141"><a href="#cb2-141"></a><span class="ss">- </span>Interpret the outcomes of these analyses and use it to probabilistically make inferences about the scientific enquiries</span>
<span id="cb2-142"><a href="#cb2-142"></a><span class="ss">- </span>Communicate the findings by written and oral means</span>
<span id="cb2-143"><a href="#cb2-143"></a></span>
<span id="cb2-144"><a href="#cb2-144"></a><span class="fu"># Graduate attributes</span></span>
<span id="cb2-145"><a href="#cb2-145"></a></span>
<span id="cb2-146"><a href="#cb2-146"></a>The <span class="co">[</span><span class="ot">**graduate attributes**</span><span class="co">](../pages/graduate_attributes.qmd)</span> resulting from completion of this modules alignment with the expectations of the workspace across diverse organisations and institutions where graduates typically find employment.</span>
<span id="cb2-147"><a href="#cb2-147"></a></span>
<span id="cb2-148"><a href="#cb2-148"></a><span class="fu"># Data used in support of the module</span></span>
<span id="cb2-149"><a href="#cb2-149"></a></span>
<span id="cb2-150"><a href="#cb2-150"></a>All the data required for BCB744 may be <span class="co">[</span><span class="ot">downloaded here</span><span class="co">](../data/Archive.zip)</span>.</span>
<span id="cb2-151"><a href="#cb2-151"></a></span>
<span id="cb2-152"><a href="#cb2-152"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb2-153"><a href="#cb2-153"></a><span class="fu">## Built-in data</span></span>
<span id="cb2-154"><a href="#cb2-154"></a></span>
<span id="cb2-155"><a href="#cb2-155"></a>R has many built-in datasets that are useful for practicing our R skills. To find out which datasets are available to you on your system, execute the following command. Help files for each of the datasets are also available:</span>
<span id="cb2-156"><a href="#cb2-156"></a></span>
<span id="cb2-159"><a href="#cb2-159"></a><span class="in">```{r}</span></span>
<span id="cb2-160"><a href="#cb2-160"></a><span class="co">#| eval: false</span></span>
<span id="cb2-161"><a href="#cb2-161"></a><span class="co"># load the data like this:</span></span>
<span id="cb2-162"><a href="#cb2-162"></a><span class="fu">data</span>()</span>
<span id="cb2-163"><a href="#cb2-163"></a></span>
<span id="cb2-164"><a href="#cb2-164"></a><span class="co"># find help, for example:</span></span>
<span id="cb2-165"><a href="#cb2-165"></a>?datasets<span class="sc">::</span>ChickWeight</span>
<span id="cb2-166"><a href="#cb2-166"></a><span class="in">```</span></span>
<span id="cb2-167"><a href="#cb2-167"></a>:::</span>
<span id="cb2-168"><a href="#cb2-168"></a></span>
<span id="cb2-169"><a href="#cb2-169"></a>It is important to use these (or any) datasets to practice your R skills on. Actively engaging with my comprehensive and detailed web pages, and practising on the included and additional other datasets will make to difference between a 60% average mark for the module, and a mark in excess of 80%.</span>
<span id="cb2-170"><a href="#cb2-170"></a></span>
<span id="cb2-171"><a href="#cb2-171"></a><span class="fu"># Prerequisites</span></span>
<span id="cb2-172"><a href="#cb2-172"></a></span>
<span id="cb2-173"><a href="#cb2-173"></a>You should have a moderate numerical literacy, but prior programming experience is not required. In all sciences, practical problem solving skills and a tenacity for challenges are crucial for success. Scientific disciplines constantly evolve and present new and complex problems that require creative and innovative solutions. You will have to demonstrate agile and adaptive approaches to solving challenges, and you must have the ability to break down complex problems into smaller parts and approach them systematically. You must also be able to identify and overcome roadblocks, and be persistent in your efforts to find a solution. These attributes will allow you to be effective in this module.</span>
<span id="cb2-174"><a href="#cb2-174"></a></span>
<span id="cb2-175"><a href="#cb2-175"></a><span class="fu"># Method of instruction</span></span>
<span id="cb2-176"><a href="#cb2-176"></a></span>
<span id="cb2-177"><a href="#cb2-177"></a>The workshop is designed to be as interactive as possible, so while you are working on exercises the tutor and I will circulate among you and engage with you to help you understand any material and the associated code you are uncomfortable with. Often this will result in discussions of novel applications and alternative approaches to the data analysis challenges you are required to solve. More challenging concepts might emerge during the Tasks and Assignments (typically these will be submitted the following day), and any such challenges will be dealt with in class prior to learning new concepts.</span>
<span id="cb2-178"><a href="#cb2-178"></a></span>
<span id="cb2-179"><a href="#cb2-179"></a>Although the module ultimately supports the application of biologically-oriented statistics, a large part of it is also about coding. It is up to you to take your coding skills to the next level and move beyond what I teach in class. Coding is a bit like learning a language, and as such programming is a skill that is best learned by doing.</span>
<span id="cb2-180"><a href="#cb2-180"></a></span>
<span id="cb2-181"><a href="#cb2-181"></a><span class="fu"># Learning colaboratively {#sec-collab-learning}</span></span>
<span id="cb2-182"><a href="#cb2-182"></a></span>
<span id="cb2-183"><a href="#cb2-183"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb2-184"><a href="#cb2-184"></a><span class="fu">## Also read: How to learn</span></span>
<span id="cb2-185"><a href="#cb2-185"></a>Please refer to my <span class="co">[</span><span class="ot">advice about how to learn</span><span class="co">](../pages/How_to_learn.qmd)</span>.</span>
<span id="cb2-186"><a href="#cb2-186"></a>:::</span>
<span id="cb2-187"><a href="#cb2-187"></a></span>
<span id="cb2-188"><a href="#cb2-188"></a>Collaborative learning provides an opportunity for you to work together and learn from each other. In this way, you will develop a deeper understanding of the subject matter. Collaborating with your friends and peers allows you to explore different perspectives and ideas, which can broaden your understanding and help you to see the subject matter from new angles. This type of learning environment also fosters the development of important skills such as communication, teamwork, and leadership, which are essential for success in academic and professional careers. Collaborative learning can create a sense of community and support among your group of peers. In the end, it will enhance your university experience, drive your love for learning, and prepare you for success beyond the university.</span>
<span id="cb2-189"><a href="#cb2-189"></a></span>
<span id="cb2-190"><a href="#cb2-190"></a>Discuss the BCB744 Workshop activities with your peers as you work on them. Use the WhatsApp group set up for the module for discussion purposes (I might assist via this medium if necessary if your questions/comments have relevance to the whole class). A better option is to use <span class="co">[</span><span class="ot">GitHub Issues</span><span class="co">](/quantecol/#help-via-bcb744-and-bcb743-issues-on-github)</span>. You will learn more in this module if you work with your friends than if you do not. Ask questions, answer questions, and share ideas liberally. Please identify your work partners by name on all assignments (if you decide to work in pairs).</span>
<span id="cb2-191"><a href="#cb2-191"></a></span>
<span id="cb2-192"><a href="#cb2-192"></a>**Cooperative learning is not a licence for plagiarism. Plagiarism is a serious offence and will be dealt with concisely. Consequences of cheating are severe---they range from a 0% for the assignment or exam up to dismissal from the course for a second offense.**</span>
<span id="cb2-193"><a href="#cb2-193"></a></span>
<span id="cb2-194"><a href="#cb2-194"></a><span class="fu"># Reusing code found elsewhere</span></span>
<span id="cb2-195"><a href="#cb2-195"></a></span>
<span id="cb2-196"><a href="#cb2-196"></a>A huge volume of code is available on the web and it can be adapted to solve your own problems. You may make use of any online resources (e.g. form <span class="co">[</span><span class="ot">StackOverflow</span><span class="co">](https://stackoverflow.com/)</span>, a thoroughly-used source of discussion about <span class="co">[</span><span class="ot">R code</span><span class="co">](https://stackoverflow.com/questions/tagged/r)</span>)---but you **MUST** clearly indicate (cite) that your solution relies on found code, regardless to what extent you have modified it to your own needs. Reused code that is discovered via a web search and which is not explicitly cited is plagiarism and it will be treated as such. On assignments you may not directly share code with your peers in this workshop.</span>
<span id="cb2-197"><a href="#cb2-197"></a></span>
<span id="cb2-198"><a href="#cb2-198"></a><span class="fu"># OpenAI ChatGPT and other AI tools</span></span>
<span id="cb2-199"><a href="#cb2-199"></a></span>
<span id="cb2-200"><a href="#cb2-200"></a>The 2024 BSc (Hons) cohort will be the first to experience the use of AI tools in the BCB744 module. The use of AI tools is a new and exciting development and it is important that you are exposed to these tools. The use of AI tools will be limited to the use of the OpenAI ChatGPT tool, which may be used to generate 'proto-code' that will assist you in becoming familiar with the R langauge. We will explore ideas together, and the mark allocation to tasks and assignments will be adjusted accoringly.</span>
<span id="cb2-201"><a href="#cb2-201"></a></span>
<span id="cb2-202"><a href="#cb2-202"></a><span class="fu"># Software</span></span>
<span id="cb2-203"><a href="#cb2-203"></a></span>
<span id="cb2-204"><a href="#cb2-204"></a>In this course you will rely entirely on <span class="co">[</span><span class="ot">R</span><span class="co">](https://cran.r-project.org/)</span> running within the <span class="co">[</span><span class="ot">RStudio</span><span class="co">](https://www.rstudio.com/)</span> IDE. The use of R is covered extensively in the <span class="co">[</span><span class="ot">BCB744</span><span class="co">](https://tangledbank.netlify.app/bcb744/intro_r/01-rstudio)</span> module where the <span class="co">[</span><span class="ot">installation process</span><span class="co">](https://tangledbank.netlify.app/BCB744/intro_r/01-RStudio)</span> is discussed.</span>
<span id="cb2-205"><a href="#cb2-205"></a></span>
<span id="cb2-206"><a href="#cb2-206"></a>Additionally, the very basics---i.e. about R, RStudio, packages, their installation, etc.---can also be found on the <span class="co">[</span><span class="ot">ModernDive</span><span class="co">](https://moderndive.netlify.app/1-getting-started.html)</span> website. A slightly longer and more detailed account of the installation process and the very basics is provided on the <span class="co">[</span><span class="ot">datacamp</span><span class="co">](https://www.datacamp.com/tutorial/r-studio-tutorial)</span> platform.</span>
<span id="cb2-207"><a href="#cb2-207"></a></span>
<span id="cb2-208"><a href="#cb2-208"></a><span class="co">[</span><span class="ot">ModernDive</span><span class="co">](https://moderndive.netlify.app)</span> also provides a nice overview of using R for data science.</span>
<span id="cb2-209"><a href="#cb2-209"></a></span>
<span id="cb2-210"><a href="#cb2-210"></a>For more in-depth coverage of the R language, refer to R Master <span class="co">[</span><span class="ot">Hadley Wickham's</span><span class="co">](https://r4ds.hadley.nz/)</span> pages. There you will find everything you need to know in a well thought through presentation. Thoroughly working through this material, page by page, will quickly make you a R Master yourself (well, almost).</span>
<span id="cb2-211"><a href="#cb2-211"></a></span>
<span id="cb2-212"><a href="#cb2-212"></a><span class="fu"># Computers</span></span>
<span id="cb2-213"><a href="#cb2-213"></a></span>
<span id="cb2-214"><a href="#cb2-214"></a>You are encouraged to provide your own laptops and to install the necessary software before the module starts. Limited support can be provided if required, but in the end, the onus is on you to understand how your computer works (from the filesystem through to dealing with software installation issues). There are also computers with R and RStudio (and the necessary add-on libraries) available in the 5th floor lab in the BCB Department.</span>
<span id="cb2-215"><a href="#cb2-215"></a></span>
<span id="cb2-216"><a href="#cb2-216"></a><span class="fu"># Attendance</span></span>
<span id="cb2-217"><a href="#cb2-217"></a></span>
<span id="cb2-218"><a href="#cb2-218"></a>This workshop-based, hands on course can only deliver acceptible outcomes</span>
<span id="cb2-219"><a href="#cb2-219"></a>if you attend all classes. The schedule is set and cannot be changed.</span>
<span id="cb2-220"><a href="#cb2-220"></a>Sometimes an occasional absence cannot be avoided. Please be curtious</span>
<span id="cb2-221"><a href="#cb2-221"></a>and notify myself or the tutor in advance of any absence. If you work</span>
<span id="cb2-222"><a href="#cb2-222"></a>with a partner in class, notify them too. Keep up with the reading</span>
<span id="cb2-223"><a href="#cb2-223"></a>assignments while you are away and we will all work with you to get you</span>
<span id="cb2-224"><a href="#cb2-224"></a>back up to speed on what you miss. If you do miss a class, however, the</span>
<span id="cb2-225"><a href="#cb2-225"></a>assignments must still be submitted on time (also see [**Late submission</span>
<span id="cb2-226"><a href="#cb2-226"></a>of CA**](#late-submission-of-ca)).</span>
<span id="cb2-227"><a href="#cb2-227"></a></span>
<span id="cb2-228"><a href="#cb2-228"></a>Since you may decide to work in collaboration with a peer on tasks and</span>
<span id="cb2-229"><a href="#cb2-229"></a>assignments, please keep this person informed at all times in case some</span>
<span id="cb2-230"><a href="#cb2-230"></a>emergency makes you unavailable for a period of time. Someone might</span>
<span id="cb2-231"><a href="#cb2-231"></a>depend on your input and contributions---do not leave someone in the</span>
<span id="cb2-232"><a href="#cb2-232"></a>lurch so that they cannot complete a task in your absence.</span>
<span id="cb2-233"><a href="#cb2-233"></a></span>
<span id="cb2-234"><a href="#cb2-234"></a><span class="fu"># Assessment policy {#sec-policy}</span></span>
<span id="cb2-235"><a href="#cb2-235"></a></span>
<span id="cb2-236"><a href="#cb2-236"></a>**Continuous Assessment (CA)** and a **Final Assessment** will provide a **Final Mark** for the module. These modes of assessment meet our needs as far as <span class="co">[</span><span class="ot">formative and summative assessments</span><span class="co">](../pages/assessment_theory.qmd)</span> are concerned. The weighting of the CA and the Final Assessment is 0.6 and 0.4, respectively. All assessments are open book, so consult your code and reading material if and when you need to.</span>
<span id="cb2-237"><a href="#cb2-237"></a></span>
<span id="cb2-238"><a href="#cb2-238"></a>| Assessment Component                              | Weighting | Contribution (%)  |</span>
<span id="cb2-239"><a href="#cb2-239"></a>|---------------------------------------------------|-----------|-------------------|</span>
<span id="cb2-240"><a href="#cb2-240"></a>| **CONTINUOUS ASSESSMENT**                         | (60)      |                   |</span>
<span id="cb2-241"><a href="#cb2-241"></a>| **Introduction to R**                             |           |                   |</span>
<span id="cb2-242"><a href="#cb2-242"></a>| Daily Tasks (A-G)                                 |           | 7.5               |</span>
<span id="cb2-243"><a href="#cb2-243"></a>| Summative Task 1                                  |           | 12.5              |</span>
<span id="cb2-244"><a href="#cb2-244"></a>| Intro R Assessment                                |           | 30                |</span>
<span id="cb2-245"><a href="#cb2-245"></a>| **Biostatistics**                                 |           |                   |</span>
<span id="cb2-246"><a href="#cb2-246"></a>| Daily Tasks (A-G)                                 |           | 7.5               |</span>
<span id="cb2-247"><a href="#cb2-247"></a>| Summative Task 2                                  |           | 42.5              |</span>
<span id="cb2-248"><a href="#cb2-248"></a>| *Total*                                           |           | *100*             |</span>
<span id="cb2-249"><a href="#cb2-249"></a>|                                                   |           |                   |</span>
<span id="cb2-250"><a href="#cb2-250"></a>| **FINAL ASSESSMENT**                              | (40)      |                   |</span>
<span id="cb2-251"><a href="#cb2-251"></a>| *Integrative Assessment (Intro R + Biostatistics)*|           | *100*             |</span>
<span id="cb2-252"><a href="#cb2-252"></a></span>
<span id="cb2-253"><a href="#cb2-253"></a>You must work alone on all these Tasks. Care must be taken that the Tasks are submitted as instructed, i.e. paying attention to naming conventions and the format of the files submitted---typically this will be in a Quarto document (.qmd) and the knitted output (I prefer .html).</span>
<span id="cb2-254"><a href="#cb2-254"></a></span>
<span id="cb2-255"><a href="#cb2-255"></a>Random quizzes will not form part of the CA for BCB744.</span>
<span id="cb2-256"><a href="#cb2-256"></a></span>
<span id="cb2-257"><a href="#cb2-257"></a><span class="fu">## Daily Tasks</span></span>
<span id="cb2-258"><a href="#cb2-258"></a></span>
<span id="cb2-259"><a href="#cb2-259"></a>Intro R and Biostatistics require that we work with real-world datasets. To this end, a series of Daily Tasks involving real data is a required part of the BCB744. These are interwoven into the daily Chapters, and your feedback is required the following day. The Daily Tasks are part of the CA.</span>
<span id="cb2-260"><a href="#cb2-260"></a></span>
<span id="cb2-261"><a href="#cb2-261"></a>When assessing the tasks, we will pay attention to the following criteria:</span>
<span id="cb2-262"><a href="#cb2-262"></a></span>
<span id="cb2-263"><a href="#cb2-263"></a><span class="ss">-   </span>Content (10%):</span>
<span id="cb2-264"><a href="#cb2-264"></a><span class="ss">    -   </span>Questions answered in order</span>
<span id="cb2-265"><a href="#cb2-265"></a><span class="ss">    -   </span>Annotations (meta-data in file header, commnets about code, ideas, and approach)</span>
<span id="cb2-266"><a href="#cb2-266"></a><span class="ss">-   </span>Code formatting and correctness (45%):</span>
<span id="cb2-267"><a href="#cb2-267"></a><span class="ss">    -   </span>Application of [R code</span>
<span id="cb2-268"><a href="#cb2-268"></a>        conventions](http://adv-r.had.co.nz/Style.html), e.g. spaces</span>
<span id="cb2-269"><a href="#cb2-269"></a>        around <span class="in">`&lt;-`</span>, after <span class="in">`#`</span>, after <span class="in">`,`</span>, etc.</span>
<span id="cb2-270"><a href="#cb2-270"></a><span class="ss">    -   </span>New line for each <span class="in">`dplyr`</span> function (lines end in <span class="in">`%&gt;%`</span>) or</span>
<span id="cb2-271"><a href="#cb2-271"></a>        <span class="in">`ggplot`</span> layer (lines end in <span class="in">`+`</span>)</span>
<span id="cb2-272"><a href="#cb2-272"></a><span class="ss">-   </span>Figures (45%):</span>
<span id="cb2-273"><a href="#cb2-273"></a><span class="ss">    -   </span>Sensible use of themes / colours</span>
<span id="cb2-274"><a href="#cb2-274"></a><span class="ss">    -   </span>Publication quality</span>
<span id="cb2-275"><a href="#cb2-275"></a><span class="ss">    -   </span>Informative and complete titles, axes labels, legends, etc.</span>
<span id="cb2-276"><a href="#cb2-276"></a></span>
<span id="cb2-277"><a href="#cb2-277"></a><span class="fu">## Summative Tasks {#sec-summative}</span></span>
<span id="cb2-278"><a href="#cb2-278"></a></span>
<span id="cb2-279"><a href="#cb2-279"></a>At the end of Intro R and Biostatistics, the more demanding Summative Tasks 1 and 2 will be required. As the table above indicates, the weighting of the Summative Tasks is more than that of the Daily Tasks but they also differ slightly between the two parts of the module. These assessments will take place over a few days and you may work at home or on campus. Like the Daily Tasks, the Summative Tasks are also open book assessments. The Summative Tasks form part of the CA and will prepare you for the Intro R Assessment (at the end of Intro R) and the Integrative Assessment (after Biostatistics; see 'Final Assessment', below).</span>
<span id="cb2-280"><a href="#cb2-280"></a></span>
<span id="cb2-281"><a href="#cb2-281"></a>We will assess the Summative Tasks as per the assessment breakdown provided under the Daily Tasks.</span>
<span id="cb2-282"><a href="#cb2-282"></a></span>
<span id="cb2-283"><a href="#cb2-283"></a><span class="fu">## Intro R Assessment</span></span>
<span id="cb2-284"><a href="#cb2-284"></a></span>
<span id="cb2-285"><a href="#cb2-285"></a>The Intro R Assessment will, as the name suggests, be about the material covered in the Introduction to R section of the work only. It forms part of the CA.</span>
<span id="cb2-286"><a href="#cb2-286"></a></span>
<span id="cb2-287"><a href="#cb2-287"></a>The Intro R Assessment will be graded with the following in expectations in mind:</span>
<span id="cb2-288"><a href="#cb2-288"></a></span>
<span id="cb2-289"><a href="#cb2-289"></a><span class="ss">-   </span>Content (20%):</span>
<span id="cb2-290"><a href="#cb2-290"></a><span class="ss">    -   </span>Questions answered in order</span>
<span id="cb2-291"><a href="#cb2-291"></a><span class="ss">    -   </span>A written explanation of approach included for each question</span>
<span id="cb2-292"><a href="#cb2-292"></a><span class="ss">    -   </span>Appropriate formatting of text, for example, fonts not larger</span>
<span id="cb2-293"><a href="#cb2-293"></a>        than necessary, headings used properly, etc. Be sensible and</span>
<span id="cb2-294"><a href="#cb2-294"></a>        tasteful.</span>
<span id="cb2-295"><a href="#cb2-295"></a><span class="ss">-   </span>Code formatting, structure, and correctness  (50%):</span>
<span id="cb2-296"><a href="#cb2-296"></a><span class="ss">    -   </span>Use **Tidyverse** code</span>
<span id="cb2-297"><a href="#cb2-297"></a><span class="ss">    -   </span>No more than \~80 characters of code per line (pay particular</span>
<span id="cb2-298"><a href="#cb2-298"></a>        attention to the comments)</span>
<span id="cb2-299"><a href="#cb2-299"></a><span class="ss">    -   </span>Application of [R code</span>
<span id="cb2-300"><a href="#cb2-300"></a>        conventions](http://adv-r.had.co.nz/Style.html), e.g. spaces</span>
<span id="cb2-301"><a href="#cb2-301"></a>        around <span class="in">`&lt;-`</span>, after <span class="in">`#`</span>, after <span class="in">`,`</span>, etc.</span>
<span id="cb2-302"><a href="#cb2-302"></a><span class="ss">    -   </span>New line for each <span class="in">`dplyr`</span> function (lines end in <span class="in">`%&gt;%`</span>) or</span>
<span id="cb2-303"><a href="#cb2-303"></a>        <span class="in">`ggplot`</span> layer (lines end in <span class="in">`+`</span>)</span>
<span id="cb2-304"><a href="#cb2-304"></a><span class="ss">    -   </span>Proper indentation of pipes and <span class="in">`ggplot()`</span> layers</span>
<span id="cb2-305"><a href="#cb2-305"></a><span class="ss">    -   </span>All chunks labelled without spaces</span>
<span id="cb2-306"><a href="#cb2-306"></a><span class="ss">    -   </span>No unwanted / commented out code left behind in the document</span>
<span id="cb2-307"><a href="#cb2-307"></a><span class="ss">-   </span>Figures (30%):</span>
<span id="cb2-308"><a href="#cb2-308"></a><span class="ss">    -   </span>Sensible use of themes / colours</span>
<span id="cb2-309"><a href="#cb2-309"></a><span class="ss">    -   </span>Publication quality</span>
<span id="cb2-310"><a href="#cb2-310"></a><span class="ss">    -   </span>Informative and complete titles, axes labels, legends, etc.</span>
<span id="cb2-311"><a href="#cb2-311"></a><span class="ss">    -   </span>No redundant features or aesthetics</span>
<span id="cb2-312"><a href="#cb2-312"></a></span>
<span id="cb2-313"><a href="#cb2-313"></a><span class="fu">## Integrative (Final) Assessment</span></span>
<span id="cb2-314"><a href="#cb2-314"></a></span>
<span id="cb2-315"><a href="#cb2-315"></a>The Integrative Assessment is the Final Assessment. As such, it will test your skills broadly across both Intro R *and* Biostatistics, and hence its weighting causes it to contribute more towards your final grade as it falls outside of the pool of marks that form your CA. The Integrative Assessment may be up to five days in duration. It will involve the analysis of real world data. Some of the Questions might expect (as per Question) that you write 1) statements of aims, objectives, and hypotheses; 2) the full and detailed methods followed by analyses together with all code, 3) full reporting of results in a manner suited for peer reviewed publications; 4) graphical support highlighting the patterns observed (again with the code), and 5) a discussion if and when required. The weighting of marks to these various sections is:</span>
<span id="cb2-316"><a href="#cb2-316"></a></span>
<span id="cb2-317"><a href="#cb2-317"></a><span class="ss">1.  </span>Aims, objectives, and hypotheses: 5%</span>
<span id="cb2-318"><a href="#cb2-318"></a><span class="ss">2.  </span>Methods and analyses: 45%</span>
<span id="cb2-319"><a href="#cb2-319"></a><span class="ss">3.  </span>Results: 20%</span>
<span id="cb2-320"><a href="#cb2-320"></a><span class="ss">4.  </span>Graphs: 15%</span>
<span id="cb2-321"><a href="#cb2-321"></a><span class="ss">5.  </span>Discussion: 15%</span>
<span id="cb2-322"><a href="#cb2-322"></a></span>
<span id="cb2-323"><a href="#cb2-323"></a>Other Questions might be shorter in nature, designed to specifically test important aspects of BCB744. Such Question might be worth anything from 10 to 50 marks, depending on the nature of the Questions.</span>
<span id="cb2-324"><a href="#cb2-324"></a></span>
<span id="cb2-325"><a href="#cb2-325"></a>The Integrative Assessment is also open book. Go home. Look at the questions. Answer them at home. Submit them by the deadline.</span>
<span id="cb2-326"><a href="#cb2-326"></a></span>
<span id="cb2-327"><a href="#cb2-327"></a><span class="fu">## Submission of Tasks, Assessments, and the Final Assessment</span></span>
<span id="cb2-328"><a href="#cb2-328"></a></span>
<span id="cb2-329"><a href="#cb2-329"></a>A statement such as the one below accompanies every assignment---pay attention, as failing to observe this instruction may result in a loss</span>
<span id="cb2-330"><a href="#cb2-330"></a>of marks (i.e. if an assignment remains ungraded because the owner of the material cannot be identified):</span>
<span id="cb2-331"><a href="#cb2-331"></a></span>
<span id="cb2-332"><a href="#cb2-332"></a>Submit a Quarto script wherein you provide answers to Questions by no later than 8:00 the following data (or the Monday in cases when assignments were given on Fridays). Label the script as follows (e.g.): **BCB744_AJ_Smit_Task_A.qmd**.</span>
<span id="cb2-333"><a href="#cb2-333"></a></span>
<span id="cb2-334"><a href="#cb2-334"></a><span class="fu">## Late submission of Tasks</span></span>
<span id="cb2-335"><a href="#cb2-335"></a></span>
<span id="cb2-336"><a href="#cb2-336"></a>Late assignments will be penalised 10% per day and will not be accepted</span>
<span id="cb2-337"><a href="#cb2-337"></a>more than 48 hours late, unless evidence such as a doctor's note, a</span>
<span id="cb2-338"><a href="#cb2-338"></a>death certificate, or another documented emergency can be provided. If</span>
<span id="cb2-339"><a href="#cb2-339"></a>you know in advance that a submission will be late, please discuss this</span>
<span id="cb2-340"><a href="#cb2-340"></a>and seek prior approval. This policy is based on the idea that in order</span>
<span id="cb2-341"><a href="#cb2-341"></a>to learn how to translate your human thoughts into computer language</span>
<span id="cb2-342"><a href="#cb2-342"></a>(coding) you should be working with them at multiple times each</span>
<span id="cb2-343"><a href="#cb2-343"></a>week---ideally daily. Time has been allocated in class for working on</span>
<span id="cb2-344"><a href="#cb2-344"></a>assignments and students are expected to continue to work on the</span>
<span id="cb2-345"><a href="#cb2-345"></a>assignments outside of class. Successfully completing (and passing) this</span>
<span id="cb2-346"><a href="#cb2-346"></a>module requires that you finish assignments based on what we have</span>
<span id="cb2-347"><a href="#cb2-347"></a>covered in class by the following class period. Work diligently from the</span>
<span id="cb2-348"><a href="#cb2-348"></a>onset so that even if something unexpected happens at the last minute</span>
<span id="cb2-349"><a href="#cb2-349"></a>you should already be close to done. This approach also allows rapid</span>
<span id="cb2-350"><a href="#cb2-350"></a>feedback to be provided to you, which can only be accomplished by</span>
<span id="cb2-351"><a href="#cb2-351"></a>returning assignments quickly and punctually.</span>
<span id="cb2-352"><a href="#cb2-352"></a></span>
<span id="cb2-353"><a href="#cb2-353"></a><span class="fu"># Support</span></span>
<span id="cb2-354"><a href="#cb2-354"></a></span>
<span id="cb2-355"><a href="#cb2-355"></a>It's expected that some tricky aspects of the module will take time to</span>
<span id="cb2-356"><a href="#cb2-356"></a>master, and the best way to master problematic material is to practice,</span>
<span id="cb2-357"><a href="#cb2-357"></a>practice some more, and then to ask questions. Trying for 10 minutes and</span>
<span id="cb2-358"><a href="#cb2-358"></a>then giving up is not good enough. I'll be more sympathetic to your</span>
<span id="cb2-359"><a href="#cb2-359"></a>cause if you can demonstrate having tried for a full day before giving</span>
<span id="cb2-360"><a href="#cb2-360"></a>up and asking me. When you ask questions about some challenge, this is</span>
<span id="cb2-361"><a href="#cb2-361"></a>the way to do it---explain to me your numerous attempts at trying to</span>
<span id="cb2-362"><a href="#cb2-362"></a>solve the problem, and explain how these various attempts have failed.</span>
<span id="cb2-363"><a href="#cb2-363"></a>*I will not help you if you have not tried to help yourself first*</span>
<span id="cb2-364"><a href="#cb2-364"></a>(maybe with advice from friends). There will be time in class to do</span>
<span id="cb2-365"><a href="#cb2-365"></a>this, typically before we embark on a new topic. You are also encouraged</span>
<span id="cb2-366"><a href="#cb2-366"></a>to bring up related questions that arise in your own B.Sc. (Hons.)</span>
<span id="cb2-367"><a href="#cb2-367"></a>research project.</span>
<span id="cb2-368"><a href="#cb2-368"></a></span>
<span id="cb2-369"><a href="#cb2-369"></a>Should you require more time with me, find out when I am 'free' and set</span>
<span id="cb2-370"><a href="#cb2-370"></a>an appointment by sending me a calendar invitation. I am happy to have a</span>
<span id="cb2-371"><a href="#cb2-371"></a>personal meeting with you via Zoom, but I prefer face-to-face in my</span>
<span id="cb2-372"><a href="#cb2-372"></a>office.</span>
<span id="cb2-373"><a href="#cb2-373"></a></span>
<span id="cb2-374"><a href="#cb2-374"></a><span class="fu">## Help via BCB744 Issues on GitHub</span></span>
<span id="cb2-375"><a href="#cb2-375"></a></span>
<span id="cb2-376"><a href="#cb2-376"></a>All discussion for the BCB744 and BCB743 workshops will be held in the</span>
<span id="cb2-377"><a href="#cb2-377"></a><span class="co">[</span><span class="ot">Issues</span><span class="co">](https://github.com/ajsmit/R_courses/issues)</span> of [this</span>
<span id="cb2-378"><a href="#cb2-378"></a>repository](https://github.com/ajsmit/R_courses). Please post all</span>
<span id="cb2-379"><a href="#cb2-379"></a>content-related questions there, and use email only for personal</span>
<span id="cb2-380"><a href="#cb2-380"></a>matters. Note that this is a public repository, so be professional in</span>
<span id="cb2-381"><a href="#cb2-381"></a>your writing here (grammar, etc.).</span>
<span id="cb2-382"><a href="#cb2-382"></a></span>
<span id="cb2-383"><a href="#cb2-383"></a>To start a new thread, create a **New issue**. Tag your peers using</span>
<span id="cb2-384"><a href="#cb2-384"></a>their handle---<span class="in">`@ajsmit`</span>, for example---to get their attention.</span>
<span id="cb2-385"><a href="#cb2-385"></a></span>
<span id="cb2-386"><a href="#cb2-386"></a>Once a question has been answered, the issue will be closed, so lots of</span>
<span id="cb2-387"><a href="#cb2-387"></a>good answers might end up in closed issues. Don't forget to look there</span>
<span id="cb2-388"><a href="#cb2-388"></a>when looking for answers---you can use the **Search** feature on this</span>
<span id="cb2-389"><a href="#cb2-389"></a>repository to find answers that might have been offered by the same or</span>
<span id="cb2-390"><a href="#cb2-390"></a>similar problem experienced by someone else in the past.</span>
<span id="cb2-391"><a href="#cb2-391"></a></span>
<span id="cb2-392"><a href="#cb2-392"></a>**Guidelines for posting questions:**</span>
<span id="cb2-393"><a href="#cb2-393"></a></span>
<span id="cb2-394"><a href="#cb2-394"></a><span class="ss">-   </span>First search existing issues (open or closed) for answers. If the</span>
<span id="cb2-395"><a href="#cb2-395"></a>    question has already been answered, you're done! If there is an open</span>
<span id="cb2-396"><a href="#cb2-396"></a>    issue, feel free to contribute to it. Or feel free to open a closed</span>
<span id="cb2-397"><a href="#cb2-397"></a>    issue if you believe the answer is not satisfactory.</span>
<span id="cb2-398"><a href="#cb2-398"></a><span class="ss">-   </span>Give your issue an informative title.</span>
<span id="cb2-399"><a href="#cb2-399"></a><span class="ss">    -   </span>Good: "Error: could not find function "ggplot""</span>
<span id="cb2-400"><a href="#cb2-400"></a><span class="ss">    -   </span>Bad: "My code does not work!" Note that you can edit an issue's</span>
<span id="cb2-401"><a href="#cb2-401"></a>        title after it's been posted.</span>
<span id="cb2-402"><a href="#cb2-402"></a><span class="ss">-   </span>Format your questions nicely using markdown and code formatting.</span>
<span id="cb2-403"><a href="#cb2-403"></a>    Preview your issue prior to posting.</span>
<span id="cb2-404"><a href="#cb2-404"></a><span class="ss">-   </span>As I explained above, your peers and I will more sympathetic to your</span>
<span id="cb2-405"><a href="#cb2-405"></a>    cause if you can show *all the things you have tried as you,</span>
<span id="cb2-406"><a href="#cb2-406"></a>    yourself, tried to fix the issue first*.</span>
<span id="cb2-407"><a href="#cb2-407"></a><span class="ss">-   </span>Include code and example data so the person trying to help you have</span>
<span id="cb2-408"><a href="#cb2-408"></a>    something to work with (and which results in the error, perhaps)</span>
<span id="cb2-409"><a href="#cb2-409"></a><span class="ss">-   </span>Where appropriate, provide links to specific files, or even lines</span>
<span id="cb2-410"><a href="#cb2-410"></a>    within them, in the body of your issue. This will help your peers</span>
<span id="cb2-411"><a href="#cb2-411"></a>    understand your question. Note that only the teaching team will have</span>
<span id="cb2-412"><a href="#cb2-412"></a>    access to private repos.</span>
<span id="cb2-413"><a href="#cb2-413"></a><span class="ss">-   </span>(Optional) Tag someone or some group of people. Start by typing</span>
<span id="cb2-414"><a href="#cb2-414"></a>    their GitHub username prefixed with the \@ symbol. Of course this</span>
<span id="cb2-415"><a href="#cb2-415"></a>    supposes that each of you have a GitHub account and username.</span>
<span id="cb2-416"><a href="#cb2-416"></a><span class="ss">-   </span>Hit **Submit new issue** when you're ready to post.</span>
<span id="cb2-417"><a href="#cb2-417"></a></span>
<span id="cb2-418"><a href="#cb2-418"></a></span>
<span id="cb2-419"><a href="#cb2-419"></a></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>© 2024, AJ Smit</p>
</div>   
    <div class="nav-footer-center">

<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB744/BCB744_index.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
<p>This page is built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>


<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>