<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.361">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="AJ Smit">
<meta name="dcterms.date" content="2021-01-01">
<title>The Tangled Bank - 4. Describing Biodiversity Patterns</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../pages/Transboundary_systems.html" rel="next">
<link href="../BDC334/03-biodiversity1.html" rel="prev">
<link href="../images/favicon.ico" rel="icon">
<script src="../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-SCYH68M489"></script><script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-SCYH68M489', { 'anonymize_ip': true});
</script><script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"simple",
  "consent_type":"implied",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary","functionality","tracking","targeting"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  });
});
</script><style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><link rel="stylesheet" href="../styles.css">
</head>
<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">The Tangled Bank</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-vignettes" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Vignettes</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-vignettes">
<li>
    <a class="dropdown-item" href="../vignettes/chl_ERDDAP.html" rel="" target="">
 <span class="dropdown-text">Retrieving Chlorophyll-a Data from ERDDAP Servers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_sightings.html" rel="" target="">
 <span class="dropdown-text">Plotting the Whale Sightings and Chlorophyll-<em>a</em> Concentrations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_localisation.html" rel="" target="">
 <span class="dropdown-text">Spatial Localisation, Subsetting, and Aggregation of the Chlorophyll-<em>a</em> Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/prep_NOAA_OISST.html" rel="" target="">
 <span class="dropdown-text">Downloading and Prep: ERDDAP</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/regridding.html" rel="" target="">
 <span class="dropdown-text">Regridding gridded data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/MHW_basic_detection.html" rel="" target="">
 <span class="dropdown-text">Basic Detection and Visualisation of Events</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/gridded_data.html" rel="" target="">
 <span class="dropdown-text">Detecting Events in Gridded Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/alt_method.html" rel="" target="">
 <span class="dropdown-text">Downloading and Prep: Python + CDO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/elem_ts_methods.html" rel="" target="">
 <span class="dropdown-text">Wavelet Analysis of Diatom Time Series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/MHW_MCS_horizonplots.html" rel="" target="">
 <span class="dropdown-text">Displaying MHWs and MCSs as horizon plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/heatwaveR_issues.html" rel="" target="">
 <span class="dropdown-text">heatwaveR issues</span></a>
  </li>  
    </ul>
</li>
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/ajsmit" rel="" target=""><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">
<li>
    <a class="dropdown-item" href="https://github.com/ajsmit/tangled_bank" rel="" target="">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://bugs.com" rel="" target="">
 <span class="dropdown-text">Report a Bug</span></a>
  </li>  
    </ul>
</li>
</ul>
<div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BDC334/BDC334_index.html">BDC334</a></li><li class="breadcrumb-item"><a href="../BDC334/04-biodiversity2.html">Lab 4: Describing Biodiversity Patterns</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-center sidebar-header">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../images/hex_logo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none"></a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
 <span class="menu-text">☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">UNDERGRADUATE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
 <span class="menu-text">BDC223</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/BDC223_FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">BDC334</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/BDC334_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BDC334: Macroecology and Global Biogeography</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1: Ecological Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/02a-r_rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2a: R and RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/02b-env_dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2b: Environmental Distance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/03-biodiversity1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3: Quantifying Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/04-biodiversity2.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Lab 4: Describing Biodiversity Patterns</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/Transboundary_systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transboundary systems</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS CORE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
 <span class="menu-text">BCB744 (Intro R)</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/BCB744_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744: Introduction to R, and Biostatistics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/01-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. R and RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/02-working-with-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Working with data and code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/03-workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. R workflows</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/04-graphics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Graphics with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/05-faceting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Faceting figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/06-brewing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Brewing colours</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/07-mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Mapping with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/08-mapping_style.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Mapping with style</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/09-mapping_rnaturalearth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Mapping with Natural Earth and the <strong>sf</strong> package</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/10-mapping_quakes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. The Fiji Earthquake Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/11-tidy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Tidy data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/12-tidier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Tidier data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/13-tidiest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Tidiest data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/14-recap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Recap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/15-functions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15. Functions by chapter</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/16-base_r.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16. Base R primer</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/17-dates.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">17. Dates</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false">
 <span class="menu-text">BCB744 (Biostatistics)</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/01-data-in-R.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Data classes and structures in R</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/02-summarise-and-describe.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Exploring with summaries and descriptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/03-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Exploring with figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/04-distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/05-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Statistical inference and hypothesis testing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/06-assumptions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Assumptions for parametric statistics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/07-t_tests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. <em>t</em>-tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/08-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Analysis of Variance (ANOVA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/09-regressions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Simple linear regressions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/10-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/11-glance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Non-parametric statistical tests at a glance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/12-confidence.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Confidence intervals</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS ELECTIVES</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">
 <span class="menu-text">BCB743</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/BCB743_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB743: Quantitative Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/00-review.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1-4. Review Biogeography and Global Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/05-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Correlations and Associations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/06-deep_dive.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Deep Dive into Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/07-ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/08-PCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8a. Principal Component Analysis (PCA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/08-PCA_examples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8b. PCA Additional Examples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/09-CA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Correspondence Analysis (CA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/10-PCoA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Principal Coordinate Analysis (PCoA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/11-nMDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11a. non-Metric Multidimensional Scaling (nMDS)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/11-nMDS_diatoms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11b. nMDS of Mayombo’s Diatom Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/12-constrained_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Constrained Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/13-cluster_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Cluster Analysis</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">SUPPORT</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
 <span class="menu-text">Web Resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/general_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/ecology_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quantitative Ecology Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/spatial_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial R Resources</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘︎</span>
  </li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li>
<a href="#other-macroecological-patterns" id="toc-other-macroecological-patterns" class="nav-link active" data-scroll-target="#other-macroecological-patterns"><span class="header-section-number">1</span> Other macroecological patterns</a>
  <ul class="collapse">
<li><a href="#species-abundance-distribution" id="toc-species-abundance-distribution" class="nav-link" data-scroll-target="#species-abundance-distribution"><span class="header-section-number">1.1</span> Species-abundance distribution</a></li>
  <li><a href="#occupancy-abundance-curves" id="toc-occupancy-abundance-curves" class="nav-link" data-scroll-target="#occupancy-abundance-curves"><span class="header-section-number">1.2</span> Occupancy-abundance curves</a></li>
  <li><a href="#species-area-accumulation" id="toc-species-area-accumulation" class="nav-link" data-scroll-target="#species-area-accumulation"><span class="header-section-number">1.3</span> Species-area (accumulation)</a></li>
  <li><a href="#rarefaction-curves" id="toc-rarefaction-curves" class="nav-link" data-scroll-target="#rarefaction-curves"><span class="header-section-number">1.4</span> Rarefaction curves</a></li>
  <li><a href="#distance-decay-curves" id="toc-distance-decay-curves" class="nav-link" data-scroll-target="#distance-decay-curves"><span class="header-section-number">1.5</span> Distance-decay curves</a></li>
  <li><a href="#elevation-and-other-gradients" id="toc-elevation-and-other-gradients" class="nav-link" data-scroll-target="#elevation-and-other-gradients"><span class="header-section-number">1.6</span> Elevation and other gradients</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references"><span class="header-section-number">1.7</span> References</a></li>
  </ul>
</li>
  </ul><div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/ajsmit/tangled_bank/edit/main/BDC334/04-biodiversity2.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">4. Describing Biodiversity Patterns</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
<p class="subtitle lead">Yet More Things we can do with Species Data</p>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://ajsmit.netlify.app">AJ Smit</a> <a href="https://orcid.org/0000-0002-3799-6126" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
    <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uwc.ac.za">
            University of the Western Cape
            </a>
          </p>
      </div>
    </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 1, 2021</p>
    </div>
  </div>
  
    
  </div>
  

</header><div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
BCB743
</div>
</div>
<div class="callout-body-container callout-body">
<p><strong>This material must be reviewed by BCB743 students in Week 1 of Quantitative Ecology.</strong></p>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Data for this Lab
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li><p>The Barro Colorado Island Tree Counts data <span class="citation" data-cites="condit2002beta">(<a href="#ref-condit2002beta" role="doc-biblioref">Condit et al. 2002</a>)</span> – load <strong>vegan</strong> and load the data with <code>data(BCI)</code></p></li>
<li><p>The Oribatid mite data <span class="citation" data-cites="borcard1992partialling borcard1994environmental">(<a href="#ref-borcard1992partialling" role="doc-biblioref">Borcard et al. 1992</a>; <a href="#ref-borcard1994environmental" role="doc-biblioref">Borcard and Legendre 1994</a>)</span> – load <strong>vegan</strong> and load the data with <code>data(mite)</code></p></li>
<li><p>The seaweed species data <span class="citation" data-cites="smit2017seaweeds">(<a href="#ref-smit2017seaweeds" role="doc-biblioref">Smit et al. 2017</a>)</span> – <a href="../data/seaweed/SeaweedSpp.csv"><code>SeaweedSpp.csv</code></a></p></li>
<li><p>The Doubs River species data <span class="citation" data-cites="verneaux1973cours borcard2011numerical">(<a href="#ref-verneaux1973cours" role="doc-biblioref">Verneaux 1973</a>; <a href="#ref-borcard2011numerical" role="doc-biblioref">Borcard et al. 2011</a>)</span> – <a href="../data/DoubsSpe.csv"><code>DoubsSpe.csv</code></a></p></li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Reading required for this Lab
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Shade et al.&nbsp;(2018) <a href="../docs/Shade_et_al_2018.pdf"><code>Shade_et_al_2018.pdf</code></a>
</li>
</ul>
</div>
</div>
<section id="other-macroecological-patterns" class="level1" data-number="1"><h1 data-number="1">
<span class="header-section-number">1</span> Other macroecological patterns</h1>
<p>Univariate diversity measures such as Simpson and Shannon diversity have already been prepared from species tables, and we have also calculated measures of <span class="math inline">\(\beta\)</span>-diversity that looked at pairwise comparisons and offered insight into community structure across a landscape and hinted at the processes that might have resulted in these patterns.</p>
<p>Let’s shine the spotlight to additional views on ecological patterns and the ecological processes that structure the communities—sometimes we will see reference to ‘community or species assembly processes’ to offer mechanistic views on how species come to be arranged into communities (the aforementioned turnover and nestedness-resultant <span class="math inline">\(\beta\)</span>-diversity are examples of other assembly processes). Let’s develop views that are based on all the information contained in the species tables, i.e.&nbsp;abundance, the number of sites, and the diversity of the biota. This deeper view is not necessarily captured if we limit our toolkit to the various univariate and pairwise descriptors of biodiversity.</p>
<div id="figure-shade" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="../images/Shade_et_al_2018_Fig_1.png" class="img-fluid figure-img" width="600"></p>
<figcaption class="figure-caption">Species distribution curves by <span class="citation" data-cites="shade2018macroecology">Shade et al. (<a href="#ref-shade2018macroecology" role="doc-biblioref">2018</a>)</span>.</figcaption></figure>
</div>
<p>You will already be familiar with the paper by <span class="citation" data-cites="shade2018macroecology">Shade et al. (<a href="#ref-shade2018macroecology" role="doc-biblioref">2018</a>)</span>. Several kinds of ecological patterns are mentioned in the paper, and they can be derived from a species table with abundance data (but <em>not</em> presence-absence data!). Note that Figure 1 in <span class="citation" data-cites="shade2018macroecology">Shade et al. (<a href="#ref-shade2018macroecology" role="doc-biblioref">2018</a>)</span> starts with a species table where the species are arranged down the rows and the sites along the variables (columns). I, and also the <strong>vegan</strong> package, require that the <strong>species are along the variables and the sites down the rows</strong>. This is the convention that will be used throughout this module.</p>
<!-- ```{r} -->
<!-- library(tidyverse) -->
<!-- library(vegan) -->
<!-- data(mite) # data contained within vegan -->
<!-- # make a head-tail function -->
<!-- ht <- function(d) rbind(head(d, 7), tail(d, 7)) -->
<!-- # Lets look at a portion of the data: -->
<!-- ht(mite[, 1:7]) -->
<!-- ``` -->
<p>The patterns that can be derived from such a table include:</p>
<ul>
<li>species-abundance distribution;</li>
<li>occupancy-abundance curves;</li>
<li>species-area curves;</li>
<li>rarefaction curves;</li>
<li>distance-decay curves;</li>
<li>elevation gradients.</li>
</ul>
<p>In this Chapter we will replicate the species distribution patterns seen in <span class="citation" data-cites="shade2018macroecology">Shade et al. (<a href="#ref-shade2018macroecology" role="doc-biblioref">2018</a>)</span>.</p>
<p>We will calculate each for the <a href="https://repository.si.edu/handle/10088/20925">Barro Colorado Island Tree Counts data</a> that come with <strong>vegan</strong>. See <code><a href="https://rdrr.io/pkg/vegan/man/BCI.html">?vegan::BCI</a></code> for a description of the data contained with the package, as well as a selection of publications relevant to the data and analyses. The primary publication of interest is <span class="citation" data-cites="condit2002beta">Condit et al. (<a href="#ref-condit2002beta" role="doc-biblioref">2002</a>)</span>.</p>
<div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vegandevs/vegan">vegan</a></span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#library(vegan) # already loaded</span></span>
<span><span class="co">#library(tidyverse) # already loaded</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">BCI</span><span class="op">)</span> <span class="co"># data contained within vegan</span></span>
<span></span>
<span><span class="co"># make a head-tail function</span></span>
<span><span class="va">ht</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">d</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">d</span>, <span class="fl">7</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span><span class="op">(</span><span class="va">d</span>, <span class="fl">7</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Lets look at a portion of the data:</span></span>
<span><span class="fu">ht</span><span class="op">(</span><span class="va">BCI</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">7</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Abarema.macradenia Vachellia.melanoceras Acalypha.diversifolia
1                  0                     0                     0
2                  0                     0                     0
3                  0                     0                     0
4                  0                     0                     0
5                  0                     0                     0
6                  0                     0                     0
7                  0                     0                     0
  Acalypha.macrostachya Adelia.triloba Aegiphila.panamensis
1                     0              0                    0
2                     0              0                    0
3                     0              0                    0
4                     0              3                    0
5                     0              1                    1
6                     0              0                    0
7                     0              0                    1
  Alchornea.costaricensis
1                       2
2                       1
3                       2
4                      18
5                       3
6                       2
7                       0</code></pre>
</div>
</div>
<section id="species-abundance-distribution" class="level2" data-number="1.1"><h2 data-number="1.1" class="anchored" data-anchor-id="species-abundance-distribution">
<span class="header-section-number">1.1</span> Species-abundance distribution</h2>
<p>The species abundance distribution (SAD) is a fundamental pattern in ecology. Typical communities have a few species that are very abundant, whereas most of them are quite rare; indeed—this is perhaps a universal law in ecology. SAD represents this relationship graphically by plotting the abundance rank on the <span class="math inline">\(x\)</span>-axis and the number of species (or some other taxonomic level) along <span class="math inline">\(y\)</span>, as was first done by <span class="citation" data-cites="fisher1943relation">Fisher et al. (<a href="#ref-fisher1943relation" role="doc-biblioref">1943</a>)</span>. He then fitted the data by log series that ideally capture situations where most of the species are quite rare with only a few very abundant ones—called <strong>Fisher’s log series distribution</strong>—and is implemented in <strong>vegan</strong> by the <code><a href="https://rdrr.io/pkg/vegan/man/fisherfit.html">fisherfit()</a></code> function (<a href="#fig-fishers-log">Figure&nbsp;1</a>). The curve in Fisher’s logarithmic series shows the expected number of species <span class="math inline">\(f\)</span> with <span class="math inline">\(n\)</span> observed individuals. In fact, the interpretation of the curve is the same for all species-abundance models shown below, and it is only the math and rationale that differ.</p>
<div class="cell">
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># take one random sample of a row (site):</span></span>
<span><span class="co"># for this website's purpose, this function ensure the same random</span></span>
<span><span class="co"># sample is drawn each time the web page is recreated</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">13</span><span class="op">)</span> </span>
<span><span class="va">k</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">BCI</span><span class="op">)</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">fish</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/fisherfit.html">fisherfit</a></span><span class="op">(</span><span class="va">BCI</span><span class="op">[</span><span class="va">k</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="va">fish</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Fisher log series model
No. of species: 95 
Fisher alpha:   39.87658 </code></pre>
</div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">fish</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-fishers-log" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="04-biodiversity2_files/figure-html/fig-fishers-log-1.svg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;1: Fisher’s log series distribution calculated for the Barro Colorado Island Tree Counts data.</figcaption></figure>
</div>
</div>
</div>
<p><span class="citation" data-cites="preston1948commonness">Preston (<a href="#ref-preston1948commonness" role="doc-biblioref">1948</a>)</span> showed that when data from a thoroughly sampled population are transformed into octaves along the <span class="math inline">\(x\)</span>-axis (number of species binned into intervals of 1, 2, 4, 8, 16, 32 etc.), the SAD that results is approximated by a symmetric Gaussian distribution. This is because more thorough sampling makes species that occur with a high frequency more common and those that occur only once or are very rare become either less common will remain completely absent. This SAD is called <strong>Preston’s log-normal distribution</strong>. In the <strong>vegan</strong> package there is an updated version of Preston’s approach with a mathematical improvement to better handle ties. It is called <code><a href="https://rdrr.io/pkg/vegan/man/fisherfit.html">prestondistr()</a></code> (<a href="#fig-preston">Figure&nbsp;2</a>):</p>
<div class="cell">
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">pres</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/fisherfit.html">prestondistr</a></span><span class="op">(</span><span class="va">BCI</span><span class="op">[</span><span class="va">k</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="va">pres</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Preston lognormal model
Method: maximized likelihood to log2 abundances 
No. of species: 95 

      mode      width         S0 
 0.9234918  1.6267630 26.4300640 

Frequencies by Octave
                0        1        2        3        4        5         6
Observed 19.00000 27.00000 21.50000 17.00000 7.000000 2.500000 1.0000000
Fitted   22.49669 26.40085 21.23279 11.70269 4.420327 1.144228 0.2029835</code></pre>
</div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">pres</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-preston" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="04-biodiversity2_files/figure-html/fig-preston-1.svg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;2: Preston’s log-normal distribution demonstrated for the BCI data.</figcaption></figure>
</div>
</div>
</div>
<p><span class="citation" data-cites="whittaker1965dominance">Whittaker (<a href="#ref-whittaker1965dominance" role="doc-biblioref">1965</a>)</span> introduced <strong>rank abundance distribution curves</strong> (RAD; sometimes called a dominance-diversity curve or Whittaker plots). Here the <span class="math inline">\(x\)</span>-axis has species ranked according to their relative abundance, with the most abundant species at the left and rarest at the right. The <span class="math inline">\(y\)</span>-axis represents relative species abundances (sometimes log-transformed). The shape of the profile as—influenced by the steepness and the length of the tail—indicates the relative proportion of abundant and scarce species in the community. In <strong>vegan</strong> we can accomplish fitting this type of SAD with the <code><a href="https://rdrr.io/pkg/vegan/man/radfit.html">radfit()</a></code> function. The default plot is somewhat more complicated as it shows broken-stick, preemption, log-Normal, Zipf and Zipf-Mandelbrot models fitted to the ranked species abundance data (<a href="#fig-rad">Figure&nbsp;3</a>):</p>
<div class="cell">
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">rad</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/radfit.html">radfit</a></span><span class="op">(</span><span class="va">BCI</span><span class="op">[</span><span class="va">k</span>,<span class="op">]</span><span class="op">)</span></span>
<span><span class="va">rad</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
RAD models, family poisson 
No. of species 95, total abundance 392

           par1      par2     par3    Deviance AIC      BIC     
Null                                   56.3132 324.6477 324.6477
Preemption  0.042685                   55.8621 326.1966 328.7504
Lognormal   0.84069   1.0912           16.1740 288.5085 293.6162
Zipf        0.12791  -0.80986          21.0817 293.4161 298.5239
Mandelbrot  0.66461  -1.2374   4.1886   6.6132 280.9476 288.6093</code></pre>
</div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">rad</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-rad" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="04-biodiversity2_files/figure-html/fig-rad-1.svg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;3: Whittaker’s rank abundance distribution curves demonstrated for the BCI data.</figcaption></figure>
</div>
</div>
</div>
<p>We can also fit the rank abundance distribution curves to several sites at once (previously we have done so on only one site) (<a href="#fig-rad2">Figure&nbsp;4</a>):</p>
<div class="cell">
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">BCI</span><span class="op">)</span>, <span class="fl">6</span><span class="op">)</span></span>
<span><span class="va">rad2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/radfit.html">radfit</a></span><span class="op">(</span><span class="va">BCI</span><span class="op">[</span><span class="va">m</span>, <span class="op">]</span><span class="op">)</span></span>
<span><span class="va">rad2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Deviance for RAD models:

                  3       37       10       13        6       22
Null        86.1127  93.5952  77.2737  52.6207  72.1627 114.1747
Preemption  58.9295 104.0978  62.7210  57.7372  54.7709 110.5156
Lognormal   29.2719  19.0653  20.4770  15.8218  19.5788  26.2510
Zipf        50.1262  11.3048  39.7066  22.8006  32.4630  15.5222
Mandelbrot   5.7342   8.9107   9.8353  12.1701   5.5973   9.6047</code></pre>
</div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">rad2</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-rad2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="04-biodiversity2_files/figure-html/fig-rad2-1.svg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;4: Rank abundance distribution curves fitted to several sites.</figcaption></figure>
</div>
</div>
</div>
<p>Above, we see that the model selected for capturing the shape of the SAD is the Mandelbrot, and it is plotted individually for each of the randomly selected sites. Model selection works through Akaike’s or Schwartz’s Bayesian information criteria (AIC or BIC; AIC is the default—select the model with the lowest AIC).</p>
<p><a href="https://github.com/cran/BiodiversityR"><strong>BiodiversityR</strong></a> (and <a href="http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf">here</a> and <a href="https://rpubs.com/Roeland-KINDT">here</a>) also offers options for rank abundance distribution curves; see <code><a href="https://rdrr.io/pkg/BiodiversityR/man/rankabundance.html">rankabundance()</a></code> (<a href="#fig-rac">Figure&nbsp;5</a>):</p>
<div class="cell">
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://www.worldagroforestry.org/output/tree-diversity-analysis">BiodiversityR</a></span><span class="op">)</span></span>
<span><span class="va">rankabund</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiodiversityR/man/rankabundance.html">rankabundance</a></span><span class="op">(</span><span class="va">BCI</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiodiversityR/man/rankabundance.html">rankabunplot</a></span><span class="op">(</span><span class="va">rankabund</span>, cex <span class="op">=</span> <span class="fl">0.8</span>, pch <span class="op">=</span> <span class="fl">0.8</span>, col <span class="op">=</span> <span class="st">"indianred4"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-rac" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="04-biodiversity2_files/figure-html/fig-rac-1.svg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;5: Rank-abundance curves for the BCI data.</figcaption></figure>
</div>
</div>
</div>
<p>Refer to the help files for the respective functions to see their differences.</p>
</section><section id="occupancy-abundance-curves" class="level2" data-number="1.2"><h2 data-number="1.2" class="anchored" data-anchor-id="occupancy-abundance-curves">
<span class="header-section-number">1.2</span> Occupancy-abundance curves</h2>
<p>Occupancy refers to the number or proportion of sites in which a species is detected. Occupancy-abundance relationships are used to infer niche specialisation patterns in the sampling region. The hypothesis (almost a theory) is that species that tend to have high local abundance within one site also tend to occupy many other sites (<a href="#fig-occupancy">Figure&nbsp;6</a>).</p>
<div class="cell">
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/">ggpubr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># A function for counts:</span></span>
<span><span class="co"># count number of non-zero elements per column</span></span>
<span><span class="va">count_fun</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="va">x</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">BCI_OA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>occ <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">BCI</span>, MARGIN <span class="op">=</span> <span class="fl">2</span>, <span class="va">count_fun</span><span class="op">)</span>,</span>
<span>                     ab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">BCI</span>, MARGIN <span class="op">=</span> <span class="fl">2</span>, <span class="va">mean</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">BCI_OA</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">ab</span>, y <span class="op">=</span> <span class="va">occ</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">occ</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"indianred3"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="co"># scale_y_log10() +</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Barro Colorado Island Tree Counts"</span>,</span>
<span>     x <span class="op">=</span> <span class="st">"Log (abundance)"</span>, y <span class="op">=</span> <span class="st">"Occupancy"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_linedraw</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-occupancy" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="04-biodiversity2_files/figure-html/fig-occupancy-1.svg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;6: Occupancy-abundance relationships seen in the BCI data.</figcaption></figure>
</div>
</div>
</div>
</section><section id="species-area-accumulation" class="level2" data-number="1.3"><h2 data-number="1.3" class="anchored" data-anchor-id="species-area-accumulation">
<span class="header-section-number">1.3</span> Species-area (accumulation)</h2>
<p>Species accumulation curves (species area relationships, SAR) try and estimate the number of unseen species. These curves can be used to predict and compare changes in diversity over increasing spatial extent. Within an ecosystem type, one would expect that more and more species would be added (accumulates) as the number of sampled sites increases (i.e.&nbsp;extent increases). This continues to a point where no more new species are added as the number of sampled sites continues to increase (i.e.&nbsp;the curve plateaus). Species accumulation curves, as the name suggests, accomplishes this by adding (accumulation or collecting) more and more sites and counting the average number of species along <span class="math inline">\(y\)</span> each time a new site is added. See Roeland Kindt’s description of <a href="http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf">how species accumulation curves work</a> (on p.&nbsp;41). In the community matrix (the sites × species table), we can do this by successively adding more rows to the curve (seen along the <span class="math inline">\(x\)</span>-axis). The <code><a href="https://rdrr.io/pkg/vegan/man/specaccum.html">specaccum()</a></code> function has many different ways of adding the new sites to the curve, but the default ‘exact’ seems to be a sensible choice. <strong>BiodiversityR</strong> has the <code><a href="https://rdrr.io/pkg/BiodiversityR/man/accumresult.html">accumresult()</a></code> function that does nearly the same. Let’s demonstrate using <strong>vegan</strong>’s function (<a href="#fig-accum">Figure&nbsp;7</a>, <a href="#fig-accum2">Figure&nbsp;8</a>, and <a href="#fig-accum3">Figure&nbsp;9</a>):</p>
<div class="cell">
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sp1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/specaccum.html">specaccum</a></span><span class="op">(</span><span class="va">BCI</span><span class="op">)</span></span>
<span><span class="va">sp2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/specaccum.html">specaccum</a></span><span class="op">(</span><span class="va">BCI</span>, <span class="st">"random"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># par(mfrow = c(2,2), mar = c(4,2,2,1))</span></span>
<span><span class="co"># par(mfrow = c(1,2))</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">sp1</span>, ci.type <span class="op">=</span> <span class="st">"polygon"</span>, col <span class="op">=</span> <span class="st">"indianred4"</span>, lwd <span class="op">=</span> <span class="fl">2</span>, ci.lty <span class="op">=</span> <span class="fl">0</span>,</span>
<span>     ci.col <span class="op">=</span> <span class="st">"steelblue2"</span>, main <span class="op">=</span> <span class="st">"Default: exact"</span>,</span>
<span>     ylab <span class="op">=</span> <span class="st">"No. of species"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-accum" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="04-biodiversity2_files/figure-html/fig-accum-1.svg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;7: Species-area accumulation curves seen in the BCI data.</figcaption></figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/specaccum.html">fitspecaccum</a></span><span class="op">(</span><span class="va">sp2</span>, <span class="st">"arrh"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">mods</span>, col <span class="op">=</span> <span class="st">"indianred"</span>, ylab <span class="op">=</span> <span class="st">"No. of species"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span><span class="op">(</span><span class="va">sp2</span>, col <span class="op">=</span> <span class="st">"yellow"</span>, border <span class="op">=</span> <span class="st">"steelblue2"</span>, lty <span class="op">=</span> <span class="fl">1</span>, cex <span class="op">=</span> <span class="fl">0.3</span>, add <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">mods</span><span class="op">$</span><span class="va">models</span>, <span class="va">AIC</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  [1] 311.4642 303.7835 346.3668 320.0786 338.7978 320.2538 325.6968 346.2671
  [9] 320.3900 343.8570 318.2509 369.8303 335.9936 350.8711 327.9831 348.1287
 [17] 328.2393 347.8133 324.3837 314.8555 333.1390 340.5678 332.6836 360.5208
 [25] 335.3660 325.3150 347.4324 336.7498 336.6374 276.1878 349.9283 295.0268
 [33] 308.4656 315.8304 303.0776 329.8425 356.2393 368.4302 318.0514 359.5975
 [41] 327.4228 335.7604 259.8340 318.0063 335.7753 285.8790 323.5174 300.3546
 [49] 327.1448 355.2747 288.2583 366.5995 287.4120 327.5877 362.6487 323.5904
 [57] 339.5650 321.2264 336.6331 353.1295 317.9578 311.6528 336.3613 337.8327
 [65] 328.4787 311.6842 345.8035 367.5620 319.0269 305.6546 338.7805 321.8859
 [73] 330.6029 326.7097 345.8923 338.4755 352.8710 355.8038 307.7327 329.2355
 [81] 341.6628 340.1687 333.4771 348.3144 321.4417 317.4331 339.2211 313.1990
 [89] 305.3069 342.4581 318.0308 299.7067 294.7851 324.3237 333.5849 349.2749
 [97] 369.8287 323.0041 332.6820 329.3875</code></pre>
</div>
<div class="cell-output-display">
<div id="fig-accum2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="04-biodiversity2_files/figure-html/fig-accum2-1.svg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;8: Fit Arrhenius models to all random accumulations</figcaption></figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">accum</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiodiversityR/man/accumresult.html">accumresult</a></span><span class="op">(</span><span class="va">BCI</span>, method <span class="op">=</span> <span class="st">"exact"</span>, permutations <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiodiversityR/man/accumresult.html">accumplot</a></span><span class="op">(</span><span class="va">accum</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-accum3" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="04-biodiversity2_files/figure-html/fig-accum3-1.svg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;9: A species accumulation curve.</figcaption></figure>
</div>
</div>
</div>
<p>Species accumulation curves can also be calculated with the <code><a href="https://rdrr.io/pkg/BAT/man/alpha.accum.html">alpha.accum()</a></code> function of the <strong>BAT</strong> package (<a href="#fig-accum4">Figure&nbsp;10</a>). In addition, the <strong>BAT</strong> package can also apply various diversity and species distribution assessments to <strong>phylogenetic</strong> and <strong>functional</strong> diversity. See the examples provided by <span class="citation" data-cites="cardoso2015bat">Cardoso et al. (<a href="#ref-cardoso2015bat" role="doc-biblioref">2015</a>)</span>.</p>
<div class="cell">
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BAT</span><span class="op">)</span></span>
<span><span class="va">BCI.acc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BAT/man/alpha.accum.html">alpha.accum</a></span><span class="op">(</span><span class="va">BCI</span>, prog <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">BCI.acc</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>, <span class="va">BCI.acc</span><span class="op">[</span>,<span class="fl">17</span><span class="op">]</span>, col <span class="op">=</span> <span class="st">"indianred"</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"Individuals"</span>, ylab <span class="op">=</span> <span class="st">"Chao1P"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">BCI.acc</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/BAT/man/slope.html">slope</a></span><span class="op">(</span><span class="va">BCI.acc</span><span class="op">)</span><span class="op">[</span>,<span class="fl">17</span><span class="op">]</span>, col <span class="op">=</span> <span class="st">"indianred"</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"Individuals"</span>, ylab <span class="op">=</span> <span class="st">"Slope"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-accum4" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="04-biodiversity2_files/figure-html/fig-accum4-1.svg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;10: A species accumulation curve made with the <code><a href="https://rdrr.io/pkg/BAT/man/alpha.accum.html">alpha.accum()</a></code> function of <strong>BAT</strong>.</figcaption></figure>
</div>
</div>
</div>
</section><section id="rarefaction-curves" class="level2" data-number="1.4"><h2 data-number="1.4" class="anchored" data-anchor-id="rarefaction-curves">
<span class="header-section-number">1.4</span> Rarefaction curves</h2>
<p>Like species accumulation curves, rarefaction curves also try to estimate the number of unseen species. Rarefaction, meaning to scale down <span class="citation" data-cites="heck1975explicit">(<a href="#ref-heck1975explicit" role="doc-biblioref">Heck Jr et al. 1975</a>)</span>, is a statistical technique used by ecologists to assess species richness (represented as <em>S</em>, or diversity indices such as Shannon diversity, <span class="math inline">\(H'\)</span>, or Simpson’s diversity, <span class="math inline">\(\lambda\)</span>) from data on species samples, such as that which we may find in site × species tables. Rarefaction can be used to determine whether a habitat, community, or ecosystem has been sufficiently sampled to fully capture the full complement of species present.</p>
<p>Rarefaction curves may seem similar to species accumulation curves, but there is a difference as I will note below. Species richness, <em>S</em>, accumulates with sample size <em>or</em> with the number of individuals sampled (across all species). The first way that rarefaction curves are presented is to show species richness as a function of number of individuals sampled. Here the principle demonstrated is that when only a few individuals are sampled, those individuals may belong to only a few species; however, when more individuals are present more species will be represented. The second approach to rarefaction is to plot the number of samples along <span class="math inline">\(x\)</span> and the species richness along the <span class="math inline">\(y\)</span>-axis (as in SADs too). So, rarefaction shows how richness accumulates with the number of individuals counted or with the number of samples taken. Rarefaction curves rise rapidly at the start when few species have been sampled and the most common species have been found; the slope then decreases and eventually plateaus suggesting that the rarest species remain to be sampled.</p>
<p>But what really distinguishes rarefaction curves from SADs is that rarefaction randomly re-samples the pool of <span class="math inline">\(N\)</span> samples (that is equal or less than the total community size) a number of times, <span class="math inline">\(n\)</span>, and plots the average number of species found in each resample (1,2, …, <span class="math inline">\(n\)</span>) as a function of individuals or samples. The <code><a href="https://rdrr.io/pkg/vegan/man/rarefy.html">rarecurve()</a></code> function draws a rarefaction curve for each row of the species data table. All these plots are made with base R graphics <a href="#fig-rarefy">Figure&nbsp;11</a>, but it will be a trivial exercise to reproduce them with <strong>ggplot2</strong>.</p>
<div class="cell">
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Example provided in ?vegan::rarefy</span></span>
<span><span class="co"># observed number of species per row (site)</span></span>
<span><span class="va">S</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/diversity.html">specnumber</a></span><span class="op">(</span><span class="va">BCI</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># calculate total no. individuals sampled per row, and find the minimum</span></span>
<span><span class="op">(</span><span class="va">raremax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">BCI</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 340</code></pre>
</div>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">Srare</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vegan/man/rarefy.html">rarefy</a></span><span class="op">(</span><span class="va">BCI</span>, <span class="va">raremax</span>, se <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">S</span>, <span class="va">Srare</span>, col <span class="op">=</span> <span class="st">"indianred3"</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"Sample size\n(observed no. of individuals)"</span>, ylab <span class="op">=</span> <span class="st">"No. species found"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/vegan/man/rarefy.html">rarecurve</a></span><span class="op">(</span><span class="va">BCI</span>, step <span class="op">=</span> <span class="fl">20</span>, sample <span class="op">=</span> <span class="va">raremax</span>, col <span class="op">=</span> <span class="st">"indianred3"</span>, cex <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>          xlab <span class="op">=</span> <span class="st">"Sample size\n(observed no. of individuals)"</span>, ylab <span class="op">=</span> <span class="st">"No. species found"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-rarefy" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="04-biodiversity2_files/figure-html/fig-rarefy-1.svg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;11: Rarefaction curves for the BCI data.</figcaption></figure>
</div>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p><strong>iNEXT</strong></p>
<p>We can also use the <a href="https://github.com/JohnsonHsieh/iNEXT"><strong>iNEXT</strong></a> package for rarefaction curves. From the package’s <a href="https://cran.r-project.org/web/packages/iNEXT/vignettes/Introduction.html">Introduction Vignette</a>:</p>
<p>iNEXT focuses on three measures of Hill numbers of order q: species richness (q = 0), Shannon diversity (q = 1, the exponential of Shannon entropy) and Simpson diversity (q = 2, the inverse of Simpson concentration). For each diversity measure, iNEXT uses the observed sample of abundance or incidence data (called the “reference sample”) to compute diversity estimates and the associated 95% confidence intervals for the following two types of rarefaction and extrapolation (R/E):</p>
<ol type="1">
<li><p>Sample‐size‐based R/E sampling curves: iNEXT computes diversity estimates for rarefied and extrapolated samples up to an appropriate size. This type of sampling curve plots the diversity estimates with respect to sample size.</p></li>
<li><p>Coverage‐based R/E sampling curves: iNEXT computes diversity estimates for rarefied and extrapolated samples with sample completeness (as measured by sample coverage) up to an appropriate coverage. This type of sampling curve plots the diversity estimates with respect to sample coverage.</p></li>
</ol>
<p>iNEXT also plots the above two types of sampling curves and a sample completeness curve. The sample completeness curve provides a bridge between these two types of curves.</p>
<p>For information about Hill numbers see David Zelený’s <a href="https://www.davidzeleny.net/anadat-r/doku.php/en:diversity_analysis">Analysis of community data in R</a> and Jari Oksanen’s coverage of <a href="https://cran.r-project.org/web/packages/vegan/vignettes/diversity-vegan.pdf">diversity measures in <strong>vegan</strong></a>.</p>
<p>There are four datasets distributed with iNEXT and numerous examples are provided in the <a href="https://cran.r-project.org/web/packages/iNEXT/vignettes/Introduction.html">Introduction Vignette</a>. iNEXT has an ‘odd’ data format that might seem foreign to <strong>vegan</strong> users. To use iNEXT with dataset suitable for analysis in vegan, we first need to convert BCI data to a species × site matrix (<a href="#fig-iNEXT">Figure&nbsp;12</a>):</p>
<div class="cell">
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://chao.stat.nthu.edu.tw/wordpress/software_download/">iNEXT</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># transpose the BCI data: </span></span>
<span><span class="va">BCI_t</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>BCI <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">BCI</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">BCI_t</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>List of 1
 $ BCI: int [1:225, 1:50] 0 0 0 0 0 0 2 0 0 0 ...
  ..- attr(*, "dimnames")=List of 2
  .. ..$ : chr [1:225] "Abarema.macradenia" "Vachellia.melanoceras" "Acalypha.diversifolia" "Acalypha.macrostachya" ...
  .. ..$ : chr [1:50] "1" "2" "3" "4" ...</code></pre>
</div>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">BCI_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iNEXT/man/iNEXT.html">iNEXT</a></span><span class="op">(</span><span class="va">BCI_t</span>, q <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>, datatype <span class="op">=</span> <span class="st">"incidence_raw"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/iNEXT/man/ggiNEXT.html">ggiNEXT</a></span><span class="op">(</span><span class="va">BCI_out</span>, type <span class="op">=</span> <span class="fl">1</span>, color.var <span class="op">=</span> <span class="st">"Order.q"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-iNEXT" class="quarto-figure quarto-figure-center anchored">
<figure class="figure"><p><img src="04-biodiversity2_files/figure-html/fig-iNEXT-1.svg" class="img-fluid figure-img"></p>
<figcaption class="figure-caption">Figure&nbsp;12: Demonstration of iNEXT capabilities.</figcaption></figure>
</div>
</div>
</div>
<p>The warning is produced because the function expects incidence data (presence-absence), but I’m feeding it abundance (count) data. Nothing serious, as the function converts the abundance data to incidences.</p>
</div>
</div>
</section><section id="distance-decay-curves" class="level2" data-number="1.5"><h2 data-number="1.5" class="anchored" data-anchor-id="distance-decay-curves">
<span class="header-section-number">1.5</span> Distance-decay curves</h2>
<p>The principles of distance decay relationships are clearly captured in analyses of <a href="04-biodiversity.html#contemporary-definitions-β-diversity"><span class="math inline">\(\beta\)</span>-diversity</a>—see specifically <strong>turnover</strong>, <span class="math inline">\(\beta_\text{sim}\)</span>. Distance decay is the primary explanation for the spatial pattern of <span class="math inline">\(\beta\)</span>-diversity along the South African coast in <span class="citation" data-cites="smit2017seaweeds">Smit et al. (<a href="#ref-smit2017seaweeds" role="doc-biblioref">2017</a>)</span>. A deeper dive into distance decay calculation can be seen in <a href="../BCB743/06-deep_dive.html">Deep Dive into Gradients</a>.</p>
</section><section id="elevation-and-other-gradients" class="level2" data-number="1.6"><h2 data-number="1.6" class="anchored" data-anchor-id="elevation-and-other-gradients">
<span class="header-section-number">1.6</span> Elevation and other gradients</h2>
<p>In once sense, an elevation gradient can be seen as specific case of distance decay. The Doubs River dataset offer a nice example of data collected along an elevation gradient. Elevation gradients have many similarities with depth gradients (e.g.&nbsp;down the ocean depths) and latitudinal gradients.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Lab 4
</div>
</div>
<div class="callout-body-container callout-body">
<p>(To be reviewed by BCB743 student but not for marks)</p>
<ol type="1">
<li>
<p>Produce the following figures for the species data indicated in [square brackets]:</p>
<ol type="a">
<li>species-abundance distribution [mite];</li>
<li>occupancy-abundance curves [mite];</li>
<li>species-area curves [seaweed]—note, do not use the <strong>BAT</strong> package’s <code><a href="https://rdrr.io/pkg/BAT/man/alpha.accum.html">alpha.accum()</a></code> function as your computer might fall over;</li>
<li>rarefaction curves [mite].</li>
</ol>
<p>Answer each under its own heading. For each, also explain briefly what the purpose of the analysis is (i.e.&nbsp;what ecological insights might be provided), and describe the findings of your own analysis as well as any ecological implications that you might be able to detect.</p>
</li>
<li><p>Using the <strong>biodiversityR</strong> package, find the most dominant species in the Doubs River dataset.</p></li>
<li><p>Discuss how elevation, depth, or latitudinal gradients are similar in many aspects to distance decay relationships.</p></li>
</ol>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Submission instructions
</div>
</div>
<div class="callout-body-container callout-body">
<p>The Lab 4 assignment on Species Data was discussed on Thursday 25 August and is due at <strong>07:00 on Thursday 1 September 2022</strong>.</p>
<p>Provide a <strong>neat and thoroughly annotated</strong> R file which can recreate all the graphs and all calculations. Written answers must be typed in the same file as comments.</p>
<p>Please label the R file as follows:</p>
<ul>
<li><code>BDC334_&lt;first_name&gt;_&lt;last_name&gt;_Lab_4.R</code></li>
</ul>
<p>(the <code>&lt;</code> and <code>&gt;</code> must be omitted as they are used in the example as field indicators only).</p>
<p>Submit your appropriately named R documents on iKamva when ready.</p>
<p>Failing to follow these instructions carefully, precisely, and thoroughly will cause you to lose marks, which could cause a significant drop in your score as formatting counts for 15% of the final mark (out of 100%).</p>
</div>
</div>
</section><section id="references" class="level2" data-number="1.7"><h2 data-number="1.7" class="anchored" data-anchor-id="references">
<span class="header-section-number">1.7</span> References</h2>
<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-borcard1994environmental" class="csl-entry" role="listitem">
Borcard D, Legendre P (1994) Environmental control and spatial structure in ecological communities: An example using oribatid mites (acari, oribatei). Environmental and Ecological statistics 1:37–61.
</div>
<div id="ref-borcard1992partialling" class="csl-entry" role="listitem">
Borcard D, Legendre P, Drapeau P (1992) Partialling out the spatial component of ecological variation. Ecology 73:1045–1055.
</div>
<div id="ref-borcard2011numerical" class="csl-entry" role="listitem">
Borcard D, Gillet F, Legendre P, others (2011) Numerical ecology with <span>R</span>. Springer
</div>
<div id="ref-cardoso2015bat" class="csl-entry" role="listitem">
Cardoso P, Rigal F, Carvalho JC (2015) BAT–biodiversity assessment tools, an r package for the measurement and estimation of alpha and beta taxon, phylogenetic and functional diversity. Methods in Ecology and Evolution 6:232–236.
</div>
<div id="ref-condit2002beta" class="csl-entry" role="listitem">
Condit R, Pitman N, Leigh Jr EG, Chave J, Terborgh J, Foster RB, Núnez P, Aguilar S, Valencia R, Villa G, others (2002) Beta-diversity in tropical forest trees. Science 295:666–669.
</div>
<div id="ref-fisher1943relation" class="csl-entry" role="listitem">
Fisher RA, Corbet AS, Williams CB (1943) The relation between the number of species and the number of individuals in a random sample of an animal population. <span>The Journal of Animal Ecology</span> 42–58.
</div>
<div id="ref-heck1975explicit" class="csl-entry" role="listitem">
Heck Jr KL, Belle G van, Simberloff D (1975) Explicit calculation of the rarefaction diversity measurement and the determination of sufficient sample size. Ecology 56:1459–1461.
</div>
<div id="ref-preston1948commonness" class="csl-entry" role="listitem">
Preston FW (1948) The commonness, and rarity, of species. Ecology 29:254–283.
</div>
<div id="ref-shade2018macroecology" class="csl-entry" role="listitem">
Shade A, Dunn RR, Blowes SA, Keil P, Bohannan BJ, Herrmann M, Küsel K, Lennon JT, Sanders NJ, Storch D, others (2018) Macroecology to unite all life, large and small. <span>Trends in Ecology &amp; Evolution</span> 33:731–744.
</div>
<div id="ref-smit2017seaweeds" class="csl-entry" role="listitem">
Smit AJ, Bolton JJ, Anderson RJ (2017) Seaweeds in two oceans: Beta-diversity. <span>Frontiers in Marine Science</span> 4:404.
</div>
<div id="ref-verneaux1973cours" class="csl-entry" role="listitem">
Verneaux J (1973) Cours d’eau de <span>F</span>ranche-<span>C</span>omt<span>é</span> (<span>M</span>assif du <span>J</span>ura). Recherches <span class="nocase">é</span>cologiques sur le r<span>é</span>seau hydrographique du <span>D</span>oubs.
</div>
<div id="ref-whittaker1965dominance" class="csl-entry" role="listitem">
Whittaker RH (1965) <span class="nocase">Dominance and Diversity in Land Plant Communities: Numerical relations of species express the importance of competition in community function and evolution.</span> Science 147:250–260.
</div>
</div>


<!-- -->

</section></section><div id="quarto-appendix" class="default"><section class="quarto-appendix-contents"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div id="quarto-reuse" class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by/4.0/">https://creativecommons.org/licenses/by/4.0/</a></div></div></section><section class="quarto-appendix-contents"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{smit2021,
  author = {Smit, AJ},
  title = {4. {Describing} {Biodiversity} {Patterns}},
  date = {2021-01-01},
  url = {http://tangledbank.netlify.app/BDC334/04-biodiversity2.html},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-smit2021" class="csl-entry quarto-appendix-citeas" role="listitem">
Smit A (2021) 4. Describing Biodiversity Patterns. <a href="http://tangledbank.netlify.app/BDC334/04-biodiversity2.html">http://tangledbank.netlify.app/BDC334/04-biodiversity2.html.
</a>
</div></div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><script src="https://utteranc.es/client.js" repo="ajsmit/tangled_bank" issue-term="pathname" theme="github-light" crossorigin="anonymous" async="">
</script><nav class="page-navigation column-body"><div class="nav-page nav-page-previous">
      <a href="../BDC334/03-biodiversity1.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Lab 3: Quantifying Biodiversity</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../pages/Transboundary_systems.html" class="pagination-link">
        <span class="nav-page-text">Transboundary systems</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb29" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2021-01-01"</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "4. Describing Biodiversity Patterns"</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Yet More Things we can do with Species Data"</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="fu">## BCB743</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>**This material must be reviewed by BCB743 students in Week 1 of Quantitative Ecology.**</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>::: callout-tip</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a><span class="fu">## Data for this Lab</span></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The Barro Colorado Island Tree Counts data <span class="co">[</span><span class="ot">@condit2002beta</span><span class="co">]</span> -- load **vegan** and load the data with <span class="in">`data(BCI)`</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The Oribatid mite data <span class="co">[</span><span class="ot">@borcard1992partialling; @borcard1994environmental</span><span class="co">]</span> -- load **vegan** and load the data with <span class="in">`data(mite)`</span></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The seaweed species data <span class="co">[</span><span class="ot">@smit2017seaweeds</span><span class="co">]</span> -- <span class="co">[</span><span class="ot">`SeaweedSpp.csv`</span><span class="co">](../data/seaweed/SeaweedSpp.csv)</span></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The Doubs River species data <span class="co">[</span><span class="ot">@verneaux1973cours; @borcard2011numerical</span><span class="co">]</span> -- <span class="co">[</span><span class="ot">`DoubsSpe.csv`</span><span class="co">](../data/DoubsSpe.csv)</span></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>::: callout-tip</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a><span class="fu">## Reading required for this Lab</span></span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Shade et al. (2018) <span class="co">[</span><span class="ot">`Shade_et_al_2018.pdf`</span><span class="co">](../docs/Shade_et_al_2018.pdf)</span></span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a><span class="fu"># Other macroecological patterns</span></span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a>Univariate diversity measures such as Simpson and Shannon diversity have already been prepared from species tables, and we have also calculated measures of $\beta$-diversity that looked at pairwise comparisons and offered insight into community structure across a landscape and hinted at the processes that might have resulted in these patterns.</span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a>Let's shine the spotlight to additional views on ecological patterns and the ecological processes that structure the communities---sometimes we will see reference to 'community or species assembly processes' to offer mechanistic views on how species come to be arranged into communities (the aforementioned turnover and nestedness-resultant $\beta$-diversity are examples of other assembly processes). Let's develop views that are based on all the information contained in the species tables, i.e. abundance, the number of sites, and the diversity of the biota. This deeper view is not necessarily captured if we limit our toolkit to the various univariate and pairwise descriptors of biodiversity.</span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a><span class="al">![Species distribution curves by @shade2018macroecology.](../images/Shade_et_al_2018_Fig_1.png)</span>{#figure-shade fig-align="center" width="600"}</span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a>You will already be familiar with the paper by @shade2018macroecology. Several kinds of ecological patterns are mentioned in the paper, and they can be derived from a species table with abundance data (but *not* presence-absence data!). Note that Figure 1 in @shade2018macroecology starts with a species table where the species are arranged down the rows and the sites along the variables (columns). I, and also the **vegan** package, require that the **species are along the variables and the sites down the rows**. This is the convention that will be used throughout this module.</span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- ```{r} --&gt;</span></span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- library(tidyverse) --&gt;</span></span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- library(vegan) --&gt;</span></span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- data(mite) # data contained within vegan --&gt;</span></span>
<span id="cb29-45"><a href="#cb29-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-46"><a href="#cb29-46" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- # make a head-tail function --&gt;</span></span>
<span id="cb29-47"><a href="#cb29-47" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- ht &lt;- function(d) rbind(head(d, 7), tail(d, 7)) --&gt;</span></span>
<span id="cb29-48"><a href="#cb29-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-49"><a href="#cb29-49" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- # Lets look at a portion of the data: --&gt;</span></span>
<span id="cb29-50"><a href="#cb29-50" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- ht(mite[, 1:7]) --&gt;</span></span>
<span id="cb29-51"><a href="#cb29-51" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- ``` --&gt;</span></span>
<span id="cb29-52"><a href="#cb29-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-53"><a href="#cb29-53" aria-hidden="true" tabindex="-1"></a>The patterns that can be derived from such a table include:</span>
<span id="cb29-54"><a href="#cb29-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-55"><a href="#cb29-55" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>species-abundance distribution;</span>
<span id="cb29-56"><a href="#cb29-56" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>occupancy-abundance curves;</span>
<span id="cb29-57"><a href="#cb29-57" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>species-area curves;</span>
<span id="cb29-58"><a href="#cb29-58" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>rarefaction curves;</span>
<span id="cb29-59"><a href="#cb29-59" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>distance-decay curves;</span>
<span id="cb29-60"><a href="#cb29-60" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>elevation gradients.</span>
<span id="cb29-61"><a href="#cb29-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-62"><a href="#cb29-62" aria-hidden="true" tabindex="-1"></a>In this Chapter we will replicate the species distribution patterns seen in @shade2018macroecology.</span>
<span id="cb29-63"><a href="#cb29-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-64"><a href="#cb29-64" aria-hidden="true" tabindex="-1"></a>We will calculate each for the <span class="co">[</span><span class="ot">Barro Colorado Island Tree Counts data</span><span class="co">](https://repository.si.edu/handle/10088/20925)</span> that come with **vegan**. See <span class="in">`?vegan::BCI`</span> for a description of the data contained with the package, as well as a selection of publications relevant to the data and analyses. The primary publication of interest is @condit2002beta.</span>
<span id="cb29-65"><a href="#cb29-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-68"><a href="#cb29-68" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-69"><a href="#cb29-69" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb29-70"><a href="#cb29-70" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb29-71"><a href="#cb29-71" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-72"><a href="#cb29-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-75"><a href="#cb29-75" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-76"><a href="#cb29-76" aria-hidden="true" tabindex="-1"></a><span class="co">#library(vegan) # already loaded</span></span>
<span id="cb29-77"><a href="#cb29-77" aria-hidden="true" tabindex="-1"></a><span class="co">#library(tidyverse) # already loaded</span></span>
<span id="cb29-78"><a href="#cb29-78" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(BCI) <span class="co"># data contained within vegan</span></span>
<span id="cb29-79"><a href="#cb29-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-80"><a href="#cb29-80" aria-hidden="true" tabindex="-1"></a><span class="co"># make a head-tail function</span></span>
<span id="cb29-81"><a href="#cb29-81" aria-hidden="true" tabindex="-1"></a>ht <span class="ot">&lt;-</span> <span class="cf">function</span>(d) <span class="fu">rbind</span>(<span class="fu">head</span>(d, <span class="dv">7</span>), <span class="fu">tail</span>(d, <span class="dv">7</span>))</span>
<span id="cb29-82"><a href="#cb29-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-83"><a href="#cb29-83" aria-hidden="true" tabindex="-1"></a><span class="co"># Lets look at a portion of the data:</span></span>
<span id="cb29-84"><a href="#cb29-84" aria-hidden="true" tabindex="-1"></a><span class="fu">ht</span>(BCI)[<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb29-85"><a href="#cb29-85" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-86"><a href="#cb29-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-87"><a href="#cb29-87" aria-hidden="true" tabindex="-1"></a><span class="fu">## Species-abundance distribution</span></span>
<span id="cb29-88"><a href="#cb29-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-89"><a href="#cb29-89" aria-hidden="true" tabindex="-1"></a>The species abundance distribution (SAD) is a fundamental pattern in ecology. Typical communities have a few species that are very abundant, whereas most of them are quite rare; indeed---this is perhaps a universal law in ecology. SAD represents this relationship graphically by plotting the abundance rank on the $x$-axis and the number of species (or some other taxonomic level) along $y$, as was first done by @fisher1943relation. He then fitted the data by log series that ideally capture situations where most of the species are quite rare with only a few very abundant ones---called **Fisher's log series distribution**---and is implemented in **vegan** by the <span class="in">`fisherfit()`</span> function (@fig-fishers-log). The curve in Fisher's logarithmic series shows the expected number of species $f$ with $n$ observed individuals. In fact, the interpretation of the curve is the same for all species-abundance models shown below, and it is only the math and rationale that differ.</span>
<span id="cb29-90"><a href="#cb29-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-93"><a href="#cb29-93" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-94"><a href="#cb29-94" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Fisher's log series distribution calculated for the Barro Colorado Island Tree Counts data."</span></span>
<span id="cb29-95"><a href="#cb29-95" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-fishers-log</span></span>
<span id="cb29-96"><a href="#cb29-96" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb29-97"><a href="#cb29-97" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb29-98"><a href="#cb29-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-99"><a href="#cb29-99" aria-hidden="true" tabindex="-1"></a><span class="co"># take one random sample of a row (site):</span></span>
<span id="cb29-100"><a href="#cb29-100" aria-hidden="true" tabindex="-1"></a><span class="co"># for this website's purpose, this function ensure the same random</span></span>
<span id="cb29-101"><a href="#cb29-101" aria-hidden="true" tabindex="-1"></a><span class="co"># sample is drawn each time the web page is recreated</span></span>
<span id="cb29-102"><a href="#cb29-102" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">13</span>) </span>
<span id="cb29-103"><a href="#cb29-103" aria-hidden="true" tabindex="-1"></a>k <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">nrow</span>(BCI), <span class="dv">1</span>)</span>
<span id="cb29-104"><a href="#cb29-104" aria-hidden="true" tabindex="-1"></a>fish <span class="ot">&lt;-</span> <span class="fu">fisherfit</span>(BCI[k,])</span>
<span id="cb29-105"><a href="#cb29-105" aria-hidden="true" tabindex="-1"></a>fish</span>
<span id="cb29-106"><a href="#cb29-106" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(fish)</span>
<span id="cb29-107"><a href="#cb29-107" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-108"><a href="#cb29-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-109"><a href="#cb29-109" aria-hidden="true" tabindex="-1"></a>@preston1948commonness showed that when data from a thoroughly sampled population are transformed into octaves along the $x$-axis (number of species binned into intervals of 1, 2, 4, 8, 16, 32 etc.), the SAD that results is approximated by a symmetric Gaussian distribution. This is because more thorough sampling makes species that occur with a high frequency more common and those that occur only once or are very rare become either less common will remain completely absent. This SAD is called **Preston's log-normal distribution**. In the **vegan** package there is an updated version of Preston's approach with a mathematical improvement to better handle ties. It is called <span class="in">`prestondistr()`</span> (@fig-preston):</span>
<span id="cb29-110"><a href="#cb29-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-113"><a href="#cb29-113" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-114"><a href="#cb29-114" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Preston's log-normal distribution demonstrated for the BCI data."</span></span>
<span id="cb29-115"><a href="#cb29-115" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-preston</span></span>
<span id="cb29-116"><a href="#cb29-116" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb29-117"><a href="#cb29-117" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb29-118"><a href="#cb29-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-119"><a href="#cb29-119" aria-hidden="true" tabindex="-1"></a>pres <span class="ot">&lt;-</span> <span class="fu">prestondistr</span>(BCI[k,])</span>
<span id="cb29-120"><a href="#cb29-120" aria-hidden="true" tabindex="-1"></a>pres</span>
<span id="cb29-121"><a href="#cb29-121" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pres)</span>
<span id="cb29-122"><a href="#cb29-122" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-123"><a href="#cb29-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-124"><a href="#cb29-124" aria-hidden="true" tabindex="-1"></a>@whittaker1965dominance introduced **rank abundance distribution curves** (RAD; sometimes called a dominance-diversity curve or Whittaker plots). Here the $x$-axis has species ranked according to their relative abundance, with the most abundant species at the left and rarest at the right. The $y$-axis represents relative species abundances (sometimes log-transformed). The shape of the profile as---influenced by the steepness and the length of the tail---indicates the relative proportion of abundant and scarce species in the community. In **vegan** we can accomplish fitting this type of SAD with the <span class="in">`radfit()`</span> function. The default plot is somewhat more complicated as it shows broken-stick, preemption, log-Normal, Zipf and Zipf-Mandelbrot models fitted to the ranked species abundance data (@fig-rad):</span>
<span id="cb29-125"><a href="#cb29-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-128"><a href="#cb29-128" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-129"><a href="#cb29-129" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Whittaker's rank abundance distribution curves demonstrated for the BCI data."</span></span>
<span id="cb29-130"><a href="#cb29-130" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-rad</span></span>
<span id="cb29-131"><a href="#cb29-131" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb29-132"><a href="#cb29-132" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb29-133"><a href="#cb29-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-134"><a href="#cb29-134" aria-hidden="true" tabindex="-1"></a>rad <span class="ot">&lt;-</span> <span class="fu">radfit</span>(BCI[k,])</span>
<span id="cb29-135"><a href="#cb29-135" aria-hidden="true" tabindex="-1"></a>rad</span>
<span id="cb29-136"><a href="#cb29-136" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(rad)</span>
<span id="cb29-137"><a href="#cb29-137" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-138"><a href="#cb29-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-139"><a href="#cb29-139" aria-hidden="true" tabindex="-1"></a>We can also fit the rank abundance distribution curves to several sites at once (previously we have done so on only one site) (@fig-rad2):</span>
<span id="cb29-140"><a href="#cb29-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-143"><a href="#cb29-143" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-144"><a href="#cb29-144" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Rank abundance distribution curves fitted to several sites."</span></span>
<span id="cb29-145"><a href="#cb29-145" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-rad2</span></span>
<span id="cb29-146"><a href="#cb29-146" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb29-147"><a href="#cb29-147" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb29-148"><a href="#cb29-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-149"><a href="#cb29-149" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">nrow</span>(BCI), <span class="dv">6</span>)</span>
<span id="cb29-150"><a href="#cb29-150" aria-hidden="true" tabindex="-1"></a>rad2 <span class="ot">&lt;-</span> <span class="fu">radfit</span>(BCI[m, ])</span>
<span id="cb29-151"><a href="#cb29-151" aria-hidden="true" tabindex="-1"></a>rad2</span>
<span id="cb29-152"><a href="#cb29-152" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(rad2)</span>
<span id="cb29-153"><a href="#cb29-153" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-154"><a href="#cb29-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-155"><a href="#cb29-155" aria-hidden="true" tabindex="-1"></a>Above, we see that the model selected for capturing the shape of the SAD is the Mandelbrot, and it is plotted individually for each of the randomly selected sites. Model selection works through Akaike’s or Schwartz’s Bayesian information criteria (AIC or BIC; AIC is the default---select the model with the lowest AIC).</span>
<span id="cb29-156"><a href="#cb29-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-157"><a href="#cb29-157" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">**BiodiversityR**</span><span class="co">](https://github.com/cran/BiodiversityR)</span> (and <span class="co">[</span><span class="ot">here</span><span class="co">](http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf)</span> and <span class="co">[</span><span class="ot">here</span><span class="co">](https://rpubs.com/Roeland-KINDT)</span>) also offers options for rank abundance distribution curves; see <span class="in">`rankabundance()`</span> (@fig-rac):</span>
<span id="cb29-158"><a href="#cb29-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-161"><a href="#cb29-161" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-162"><a href="#cb29-162" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Rank-abundance curves for the BCI data."</span></span>
<span id="cb29-163"><a href="#cb29-163" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-rac</span></span>
<span id="cb29-164"><a href="#cb29-164" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb29-165"><a href="#cb29-165" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb29-166"><a href="#cb29-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-167"><a href="#cb29-167" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BiodiversityR)</span>
<span id="cb29-168"><a href="#cb29-168" aria-hidden="true" tabindex="-1"></a>rankabund <span class="ot">&lt;-</span> <span class="fu">rankabundance</span>(BCI)</span>
<span id="cb29-169"><a href="#cb29-169" aria-hidden="true" tabindex="-1"></a><span class="fu">rankabunplot</span>(rankabund, <span class="at">cex =</span> <span class="fl">0.8</span>, <span class="at">pch =</span> <span class="fl">0.8</span>, <span class="at">col =</span> <span class="st">"indianred4"</span>)</span>
<span id="cb29-170"><a href="#cb29-170" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-171"><a href="#cb29-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-172"><a href="#cb29-172" aria-hidden="true" tabindex="-1"></a>Refer to the help files for the respective functions to see their differences.</span>
<span id="cb29-173"><a href="#cb29-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-174"><a href="#cb29-174" aria-hidden="true" tabindex="-1"></a><span class="fu">## Occupancy-abundance curves</span></span>
<span id="cb29-175"><a href="#cb29-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-176"><a href="#cb29-176" aria-hidden="true" tabindex="-1"></a>Occupancy refers to the number or proportion of sites in which a species is detected. Occupancy-abundance relationships are used to infer niche specialisation patterns in the sampling region. The hypothesis (almost a theory) is that species that tend to have high local abundance within one site also tend to occupy many other sites (@fig-occupancy).</span>
<span id="cb29-177"><a href="#cb29-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-180"><a href="#cb29-180" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-181"><a href="#cb29-181" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Occupancy-abundance relationships seen in the BCI data."</span></span>
<span id="cb29-182"><a href="#cb29-182" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-occupancy</span></span>
<span id="cb29-183"><a href="#cb29-183" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb29-184"><a href="#cb29-184" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 3</span></span>
<span id="cb29-185"><a href="#cb29-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-186"><a href="#cb29-186" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb29-187"><a href="#cb29-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-188"><a href="#cb29-188" aria-hidden="true" tabindex="-1"></a><span class="co"># A function for counts:</span></span>
<span id="cb29-189"><a href="#cb29-189" aria-hidden="true" tabindex="-1"></a><span class="co"># count number of non-zero elements per column</span></span>
<span id="cb29-190"><a href="#cb29-190" aria-hidden="true" tabindex="-1"></a>count_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb29-191"><a href="#cb29-191" aria-hidden="true" tabindex="-1"></a>  <span class="fu">length</span>(x[x <span class="sc">&gt;</span> <span class="dv">0</span>])</span>
<span id="cb29-192"><a href="#cb29-192" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-193"><a href="#cb29-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-194"><a href="#cb29-194" aria-hidden="true" tabindex="-1"></a>BCI_OA <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">occ =</span> <span class="fu">apply</span>(BCI, <span class="at">MARGIN =</span> <span class="dv">2</span>, count_fun),</span>
<span id="cb29-195"><a href="#cb29-195" aria-hidden="true" tabindex="-1"></a>                     <span class="at">ab =</span> <span class="fu">apply</span>(BCI, <span class="at">MARGIN =</span> <span class="dv">2</span>, mean))</span>
<span id="cb29-196"><a href="#cb29-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-197"><a href="#cb29-197" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(BCI_OA, <span class="fu">aes</span>(<span class="at">x =</span> ab, <span class="at">y =</span> occ<span class="sc">/</span><span class="fu">max</span>(occ))) <span class="sc">+</span></span>
<span id="cb29-198"><a href="#cb29-198" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">colour =</span> <span class="st">"indianred3"</span>) <span class="sc">+</span></span>
<span id="cb29-199"><a href="#cb29-199" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb29-200"><a href="#cb29-200" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_y_log10() +</span></span>
<span id="cb29-201"><a href="#cb29-201" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Barro Colorado Island Tree Counts"</span>,</span>
<span id="cb29-202"><a href="#cb29-202" aria-hidden="true" tabindex="-1"></a>     <span class="at">x =</span> <span class="st">"Log (abundance)"</span>, <span class="at">y =</span> <span class="st">"Occupancy"</span>) <span class="sc">+</span></span>
<span id="cb29-203"><a href="#cb29-203" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_linedraw</span>()</span>
<span id="cb29-204"><a href="#cb29-204" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-205"><a href="#cb29-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-206"><a href="#cb29-206" aria-hidden="true" tabindex="-1"></a><span class="fu">## Species-area (accumulation)</span></span>
<span id="cb29-207"><a href="#cb29-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-208"><a href="#cb29-208" aria-hidden="true" tabindex="-1"></a>Species accumulation curves (species area relationships, SAR) try and estimate the number of unseen species. These curves can be used to predict and compare changes in diversity over increasing spatial extent. Within an ecosystem type, one would expect that more and more species would be added (accumulates) as the number of sampled sites increases (i.e. extent increases). This continues to a point where no more new species are added as the number of sampled sites continues to increase (i.e. the curve plateaus). Species accumulation curves, as the name suggests, accomplishes this by adding (accumulation or collecting) more and more sites and counting the average number of species along $y$ each time a new site is added. See Roeland Kindt's description of <span class="co">[</span><span class="ot">how species accumulation curves work</span><span class="co">](http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf)</span> (on p. 41). In the community matrix (the sites × species table), we can do this by successively adding more rows to the curve (seen along the $x$-axis). The <span class="in">`specaccum()`</span> function has many different ways of adding the new sites to the curve, but the default 'exact' seems to be a sensible choice. **BiodiversityR** has the `accumresult()` function that does nearly the same. Let's demonstrate using **vegan**'s function (@fig-accum, @fig-accum2, and @fig-accum3):</span>
<span id="cb29-209"><a href="#cb29-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-212"><a href="#cb29-212" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-213"><a href="#cb29-213" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Species-area accumulation curves seen in the BCI data."</span></span>
<span id="cb29-214"><a href="#cb29-214" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-accum</span></span>
<span id="cb29-215"><a href="#cb29-215" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb29-216"><a href="#cb29-216" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb29-217"><a href="#cb29-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-218"><a href="#cb29-218" aria-hidden="true" tabindex="-1"></a>sp1 <span class="ot">&lt;-</span> <span class="fu">specaccum</span>(BCI)</span>
<span id="cb29-219"><a href="#cb29-219" aria-hidden="true" tabindex="-1"></a>sp2 <span class="ot">&lt;-</span> <span class="fu">specaccum</span>(BCI, <span class="st">"random"</span>)</span>
<span id="cb29-220"><a href="#cb29-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-221"><a href="#cb29-221" aria-hidden="true" tabindex="-1"></a><span class="co"># par(mfrow = c(2,2), mar = c(4,2,2,1))</span></span>
<span id="cb29-222"><a href="#cb29-222" aria-hidden="true" tabindex="-1"></a><span class="co"># par(mfrow = c(1,2))</span></span>
<span id="cb29-223"><a href="#cb29-223" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sp1, <span class="at">ci.type =</span> <span class="st">"polygon"</span>, <span class="at">col =</span> <span class="st">"indianred4"</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">ci.lty =</span> <span class="dv">0</span>,</span>
<span id="cb29-224"><a href="#cb29-224" aria-hidden="true" tabindex="-1"></a>     <span class="at">ci.col =</span> <span class="st">"steelblue2"</span>, <span class="at">main =</span> <span class="st">"Default: exact"</span>,</span>
<span id="cb29-225"><a href="#cb29-225" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"No. of species"</span>)</span>
<span id="cb29-226"><a href="#cb29-226" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-227"><a href="#cb29-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-230"><a href="#cb29-230" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-231"><a href="#cb29-231" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Fit Arrhenius models to all random accumulations"</span></span>
<span id="cb29-232"><a href="#cb29-232" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-accum2</span></span>
<span id="cb29-233"><a href="#cb29-233" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb29-234"><a href="#cb29-234" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb29-235"><a href="#cb29-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-236"><a href="#cb29-236" aria-hidden="true" tabindex="-1"></a>mods <span class="ot">&lt;-</span> <span class="fu">fitspecaccum</span>(sp2, <span class="st">"arrh"</span>)</span>
<span id="cb29-237"><a href="#cb29-237" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mods, <span class="at">col =</span> <span class="st">"indianred"</span>, <span class="at">ylab =</span> <span class="st">"No. of species"</span>)</span>
<span id="cb29-238"><a href="#cb29-238" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(sp2, <span class="at">col =</span> <span class="st">"yellow"</span>, <span class="at">border =</span> <span class="st">"steelblue2"</span>, <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">cex =</span> <span class="fl">0.3</span>, <span class="at">add =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-239"><a href="#cb29-239" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(mods<span class="sc">$</span>models, AIC)</span>
<span id="cb29-240"><a href="#cb29-240" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-241"><a href="#cb29-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-244"><a href="#cb29-244" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-245"><a href="#cb29-245" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "A species accumulation curve."</span></span>
<span id="cb29-246"><a href="#cb29-246" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-accum3</span></span>
<span id="cb29-247"><a href="#cb29-247" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb29-248"><a href="#cb29-248" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb29-249"><a href="#cb29-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-250"><a href="#cb29-250" aria-hidden="true" tabindex="-1"></a>accum <span class="ot">&lt;-</span> <span class="fu">accumresult</span>(BCI, <span class="at">method =</span> <span class="st">"exact"</span>, <span class="at">permutations =</span> <span class="dv">100</span>)</span>
<span id="cb29-251"><a href="#cb29-251" aria-hidden="true" tabindex="-1"></a><span class="fu">accumplot</span>(accum)</span>
<span id="cb29-252"><a href="#cb29-252" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-253"><a href="#cb29-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-254"><a href="#cb29-254" aria-hidden="true" tabindex="-1"></a>Species accumulation curves can also be calculated with the <span class="in">`alpha.accum()`</span> function of the **BAT** package (@fig-accum4). In addition, the **BAT** package can also apply various diversity and species distribution assessments to **phylogenetic** and **functional** diversity. See the examples provided by @cardoso2015bat.</span>
<span id="cb29-255"><a href="#cb29-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-258"><a href="#cb29-258" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-259"><a href="#cb29-259" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "A species accumulation curve made with the `alpha.accum()` function of **BAT**."</span></span>
<span id="cb29-260"><a href="#cb29-260" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-accum4</span></span>
<span id="cb29-261"><a href="#cb29-261" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb29-262"><a href="#cb29-262" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb29-263"><a href="#cb29-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-264"><a href="#cb29-264" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BAT)</span>
<span id="cb29-265"><a href="#cb29-265" aria-hidden="true" tabindex="-1"></a>BCI.acc <span class="ot">&lt;-</span> <span class="fu">alpha.accum</span>(BCI, <span class="at">prog =</span> <span class="cn">FALSE</span>)</span>
<span id="cb29-266"><a href="#cb29-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-267"><a href="#cb29-267" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb29-268"><a href="#cb29-268" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(BCI.acc[,<span class="dv">2</span>], BCI.acc[,<span class="dv">17</span>], <span class="at">col =</span> <span class="st">"indianred"</span>,</span>
<span id="cb29-269"><a href="#cb29-269" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Individuals"</span>, <span class="at">ylab =</span> <span class="st">"Chao1P"</span>)</span>
<span id="cb29-270"><a href="#cb29-270" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(BCI.acc[,<span class="dv">2</span>], <span class="fu">slope</span>(BCI.acc)[,<span class="dv">17</span>], <span class="at">col =</span> <span class="st">"indianred"</span>,</span>
<span id="cb29-271"><a href="#cb29-271" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Individuals"</span>, <span class="at">ylab =</span> <span class="st">"Slope"</span>)</span>
<span id="cb29-272"><a href="#cb29-272" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-273"><a href="#cb29-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-274"><a href="#cb29-274" aria-hidden="true" tabindex="-1"></a><span class="fu">## Rarefaction curves</span></span>
<span id="cb29-275"><a href="#cb29-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-276"><a href="#cb29-276" aria-hidden="true" tabindex="-1"></a>Like species accumulation curves, rarefaction curves also try to estimate the number of unseen species. Rarefaction, meaning to scale down <span class="co">[</span><span class="ot">@heck1975explicit</span><span class="co">]</span>, is a statistical technique used by ecologists to assess species richness (represented as *S*, or diversity indices such as Shannon diversity, $H'$, or Simpson's diversity, $\lambda$) from data on species samples, such as that which we may find in site × species tables. Rarefaction can be used to determine whether a habitat, community, or ecosystem has been sufficiently sampled to fully capture the full complement of species present.</span>
<span id="cb29-277"><a href="#cb29-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-278"><a href="#cb29-278" aria-hidden="true" tabindex="-1"></a>Rarefaction curves may seem similar to species accumulation curves, but there is a difference as I will note below. Species richness, *S*, accumulates with sample size *or* with the number of individuals sampled (across all species). The first way that rarefaction curves are presented is to show species richness as a function of number of individuals sampled. Here the principle demonstrated is that when only a few individuals are sampled, those individuals may belong to only a few species; however, when more individuals are present more species will be represented. The second approach to rarefaction is to plot the number of samples along $x$ and the species richness along the $y$-axis (as in SADs too). So, rarefaction shows how richness accumulates with the number of individuals counted or with the number of samples taken. Rarefaction curves rise rapidly at the start when few species have been sampled and the most common species have been found; the slope then decreases and eventually plateaus suggesting that the rarest species remain to be sampled.</span>
<span id="cb29-279"><a href="#cb29-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-280"><a href="#cb29-280" aria-hidden="true" tabindex="-1"></a>But what really distinguishes rarefaction curves from SADs is that rarefaction randomly re-samples the pool of $N$ samples (that is equal or less than the total community size) a number of times, $n$, and plots the average number of species found in each resample (1,2, ..., $n$) as a function of individuals or samples. The <span class="in">`rarecurve()`</span> function draws a rarefaction curve for each row of the species data table. All these plots are made with base R graphics <span class="co">[</span><span class="ot">@fig-rarefy</span><span class="co">]</span>, but it will be a trivial exercise to reproduce them with **ggplot2**.</span>
<span id="cb29-281"><a href="#cb29-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-284"><a href="#cb29-284" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-285"><a href="#cb29-285" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Rarefaction curves for the BCI data."</span></span>
<span id="cb29-286"><a href="#cb29-286" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-rarefy</span></span>
<span id="cb29-287"><a href="#cb29-287" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb29-288"><a href="#cb29-288" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb29-289"><a href="#cb29-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-290"><a href="#cb29-290" aria-hidden="true" tabindex="-1"></a><span class="co"># Example provided in ?vegan::rarefy</span></span>
<span id="cb29-291"><a href="#cb29-291" aria-hidden="true" tabindex="-1"></a><span class="co"># observed number of species per row (site)</span></span>
<span id="cb29-292"><a href="#cb29-292" aria-hidden="true" tabindex="-1"></a>S <span class="ot">&lt;-</span> <span class="fu">specnumber</span>(BCI) </span>
<span id="cb29-293"><a href="#cb29-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-294"><a href="#cb29-294" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate total no. individuals sampled per row, and find the minimum</span></span>
<span id="cb29-295"><a href="#cb29-295" aria-hidden="true" tabindex="-1"></a>(raremax <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">rowSums</span>(BCI)))</span>
<span id="cb29-296"><a href="#cb29-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-297"><a href="#cb29-297" aria-hidden="true" tabindex="-1"></a>Srare <span class="ot">&lt;-</span> <span class="fu">rarefy</span>(BCI, raremax, <span class="at">se =</span> <span class="cn">FALSE</span>)</span>
<span id="cb29-298"><a href="#cb29-298" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb29-299"><a href="#cb29-299" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(S, Srare, <span class="at">col =</span> <span class="st">"indianred3"</span>,</span>
<span id="cb29-300"><a href="#cb29-300" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Sample size</span><span class="sc">\n</span><span class="st">(observed no. of individuals)"</span>, <span class="at">ylab =</span> <span class="st">"No. species found"</span>)</span>
<span id="cb29-301"><a href="#cb29-301" aria-hidden="true" tabindex="-1"></a><span class="fu">rarecurve</span>(BCI, <span class="at">step =</span> <span class="dv">20</span>, <span class="at">sample =</span> raremax, <span class="at">col =</span> <span class="st">"indianred3"</span>, <span class="at">cex =</span> <span class="fl">0.6</span>,</span>
<span id="cb29-302"><a href="#cb29-302" aria-hidden="true" tabindex="-1"></a>          <span class="at">xlab =</span> <span class="st">"Sample size</span><span class="sc">\n</span><span class="st">(observed no. of individuals)"</span>, <span class="at">ylab =</span> <span class="st">"No. species found"</span>)</span>
<span id="cb29-303"><a href="#cb29-303" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-304"><a href="#cb29-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-305"><a href="#cb29-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-306"><a href="#cb29-306" aria-hidden="true" tabindex="-1"></a>:::{.callout-note}</span>
<span id="cb29-307"><a href="#cb29-307" aria-hidden="true" tabindex="-1"></a>**iNEXT**</span>
<span id="cb29-308"><a href="#cb29-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-309"><a href="#cb29-309" aria-hidden="true" tabindex="-1"></a>We can also use the <span class="co">[</span><span class="ot">**iNEXT**</span><span class="co">](https://github.com/JohnsonHsieh/iNEXT)</span> package for rarefaction curves. From the package's <span class="co">[</span><span class="ot">Introduction Vignette</span><span class="co">](https://cran.r-project.org/web/packages/iNEXT/vignettes/Introduction.html)</span>:</span>
<span id="cb29-310"><a href="#cb29-310" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb29-311"><a href="#cb29-311" aria-hidden="true" tabindex="-1"></a>iNEXT focuses on three measures of Hill numbers of order q: species richness (q = 0), Shannon diversity (q = 1, the exponential of Shannon entropy) and Simpson diversity (q = 2, the inverse of Simpson concentration). For each diversity measure, iNEXT uses the observed sample of abundance or incidence data (called the “reference sample”) to compute diversity estimates and the associated 95% confidence intervals for the following two types of rarefaction and extrapolation (R/E):</span>
<span id="cb29-312"><a href="#cb29-312" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb29-313"><a href="#cb29-313" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Sample‐size‐based R/E sampling curves: iNEXT computes diversity estimates for rarefied and extrapolated samples up to an appropriate size. This type of sampling curve plots the diversity estimates with respect to sample size.</span>
<span id="cb29-314"><a href="#cb29-314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-315"><a href="#cb29-315" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>Coverage‐based R/E sampling curves: iNEXT computes diversity estimates for rarefied and extrapolated samples with sample completeness (as measured by sample coverage) up to an appropriate coverage. This type of sampling curve plots the diversity estimates with respect to sample coverage.</span>
<span id="cb29-316"><a href="#cb29-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-317"><a href="#cb29-317" aria-hidden="true" tabindex="-1"></a>iNEXT also plots the above two types of sampling curves and a sample completeness curve. The sample completeness curve provides a bridge between these two types of curves.</span>
<span id="cb29-318"><a href="#cb29-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-319"><a href="#cb29-319" aria-hidden="true" tabindex="-1"></a>For information about Hill numbers see David Zelený's <span class="co">[</span><span class="ot">Analysis of community data in R</span><span class="co">](https://www.davidzeleny.net/anadat-r/doku.php/en:diversity_analysis)</span> and Jari Oksanen's coverage of <span class="co">[</span><span class="ot">diversity measures in **vegan**</span><span class="co">](https://cran.r-project.org/web/packages/vegan/vignettes/diversity-vegan.pdf)</span>.</span>
<span id="cb29-320"><a href="#cb29-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-321"><a href="#cb29-321" aria-hidden="true" tabindex="-1"></a>There are four datasets distributed with iNEXT and numerous examples are provided in the <span class="co">[</span><span class="ot">Introduction Vignette</span><span class="co">](https://cran.r-project.org/web/packages/iNEXT/vignettes/Introduction.html)</span>. iNEXT has an 'odd' data format that might seem foreign to **vegan** users. To use iNEXT with dataset suitable for analysis in vegan, we first need to convert BCI data to a species × site matrix (@fig-iNEXT):</span>
<span id="cb29-322"><a href="#cb29-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-325"><a href="#cb29-325" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-326"><a href="#cb29-326" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Demonstration of iNEXT capabilities."</span></span>
<span id="cb29-327"><a href="#cb29-327" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-iNEXT</span></span>
<span id="cb29-328"><a href="#cb29-328" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb29-329"><a href="#cb29-329" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb29-330"><a href="#cb29-330" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(iNEXT)</span>
<span id="cb29-331"><a href="#cb29-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-332"><a href="#cb29-332" aria-hidden="true" tabindex="-1"></a><span class="co"># transpose the BCI data: </span></span>
<span id="cb29-333"><a href="#cb29-333" aria-hidden="true" tabindex="-1"></a>BCI_t <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">BCI =</span> <span class="fu">t</span>(BCI))</span>
<span id="cb29-334"><a href="#cb29-334" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(BCI_t)</span>
<span id="cb29-335"><a href="#cb29-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-336"><a href="#cb29-336" aria-hidden="true" tabindex="-1"></a>BCI_out <span class="ot">&lt;-</span> <span class="fu">iNEXT</span>(BCI_t, <span class="at">q =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>), <span class="at">datatype =</span> <span class="st">"incidence_raw"</span>)</span>
<span id="cb29-337"><a href="#cb29-337" aria-hidden="true" tabindex="-1"></a><span class="fu">ggiNEXT</span>(BCI_out, <span class="at">type =</span> <span class="dv">1</span>, <span class="at">color.var =</span> <span class="st">"Order.q"</span>)</span>
<span id="cb29-338"><a href="#cb29-338" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-339"><a href="#cb29-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-340"><a href="#cb29-340" aria-hidden="true" tabindex="-1"></a>The warning is produced because the function expects incidence data (presence-absence), but I'm feeding it abundance (count) data. Nothing serious, as the function converts the abundance data to incidences.</span>
<span id="cb29-341"><a href="#cb29-341" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-342"><a href="#cb29-342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-343"><a href="#cb29-343" aria-hidden="true" tabindex="-1"></a><span class="fu">## Distance-decay curves</span></span>
<span id="cb29-344"><a href="#cb29-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-345"><a href="#cb29-345" aria-hidden="true" tabindex="-1"></a>The principles of distance decay relationships are clearly captured in analyses of  <span class="co">[</span><span class="ot">$\beta$-diversity</span><span class="co">](04-biodiversity.html#contemporary-definitions-%CE%B2-diversity)</span>---see specifically **turnover**, $\beta_\text{sim}$. Distance decay is the primary explanation for the spatial pattern of $\beta$-diversity along the South African coast in @smit2017seaweeds. A deeper dive into distance decay calculation can be seen in <span class="co">[</span><span class="ot">Deep Dive into Gradients</span><span class="co">](../BCB743/06-deep_dive.qmd)</span>.</span>
<span id="cb29-346"><a href="#cb29-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-347"><a href="#cb29-347" aria-hidden="true" tabindex="-1"></a><span class="fu">## Elevation and other gradients</span></span>
<span id="cb29-348"><a href="#cb29-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-349"><a href="#cb29-349" aria-hidden="true" tabindex="-1"></a>In once sense, an elevation gradient can be seen as specific case of distance decay. The Doubs River dataset offer a nice example of data collected along an elevation gradient. Elevation gradients have many similarities with depth gradients (e.g. down the ocean depths) and latitudinal gradients.</span>
<span id="cb29-350"><a href="#cb29-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-351"><a href="#cb29-351" aria-hidden="true" tabindex="-1"></a>::: callout-important</span>
<span id="cb29-352"><a href="#cb29-352" aria-hidden="true" tabindex="-1"></a><span class="fu">## Lab 4</span></span>
<span id="cb29-353"><a href="#cb29-353" aria-hidden="true" tabindex="-1"></a>(To be reviewed by BCB743 student but not for marks)</span>
<span id="cb29-354"><a href="#cb29-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-355"><a href="#cb29-355" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Produce the following figures for the species data indicated in <span class="co">[</span><span class="ot">square brackets</span><span class="co">]</span>:</span>
<span id="cb29-356"><a href="#cb29-356" aria-hidden="true" tabindex="-1"></a>    a. species-abundance distribution <span class="co">[</span><span class="ot">mite</span><span class="co">]</span>;</span>
<span id="cb29-357"><a href="#cb29-357" aria-hidden="true" tabindex="-1"></a>    b. occupancy-abundance curves <span class="co">[</span><span class="ot">mite</span><span class="co">]</span>;</span>
<span id="cb29-358"><a href="#cb29-358" aria-hidden="true" tabindex="-1"></a>    c. species-area curves <span class="co">[</span><span class="ot">seaweed</span><span class="co">]</span>---note, do not use the **BAT** package's <span class="in">`alpha.accum()`</span> function as your computer might fall over;</span>
<span id="cb29-359"><a href="#cb29-359" aria-hidden="true" tabindex="-1"></a>    d. rarefaction curves <span class="co">[</span><span class="ot">mite</span><span class="co">]</span>.</span>
<span id="cb29-360"><a href="#cb29-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-361"><a href="#cb29-361" aria-hidden="true" tabindex="-1"></a>    Answer each under its own heading. For each, also explain briefly what the purpose of the analysis is (i.e. what ecological insights might be provided), and describe the findings of your own analysis as well as any ecological implications that you might be able to detect.</span>
<span id="cb29-362"><a href="#cb29-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-363"><a href="#cb29-363" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>Using the **biodiversityR** package, find the most dominant species in the Doubs River dataset.</span>
<span id="cb29-364"><a href="#cb29-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-365"><a href="#cb29-365" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>Discuss how elevation, depth, or latitudinal gradients are similar in many aspects to distance decay relationships.</span>
<span id="cb29-366"><a href="#cb29-366" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-367"><a href="#cb29-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-368"><a href="#cb29-368" aria-hidden="true" tabindex="-1"></a>:::{.callout-important}</span>
<span id="cb29-369"><a href="#cb29-369" aria-hidden="true" tabindex="-1"></a><span class="fu">## Submission instructions</span></span>
<span id="cb29-370"><a href="#cb29-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-371"><a href="#cb29-371" aria-hidden="true" tabindex="-1"></a>The Lab 4 assignment on Species Data was discussed on Thursday 25 August</span>
<span id="cb29-372"><a href="#cb29-372" aria-hidden="true" tabindex="-1"></a>and is due at **07:00 on Thursday 1 September 2022**.</span>
<span id="cb29-373"><a href="#cb29-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-374"><a href="#cb29-374" aria-hidden="true" tabindex="-1"></a>Provide a **neat and thoroughly annotated** R file which can recreate all the</span>
<span id="cb29-375"><a href="#cb29-375" aria-hidden="true" tabindex="-1"></a>graphs and all calculations. Written answers must be typed in the same file as comments.</span>
<span id="cb29-376"><a href="#cb29-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-377"><a href="#cb29-377" aria-hidden="true" tabindex="-1"></a>Please label the R file as follows:</span>
<span id="cb29-378"><a href="#cb29-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-379"><a href="#cb29-379" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`BDC334_&lt;first_name&gt;_&lt;last_name&gt;_Lab_4.R`</span></span>
<span id="cb29-380"><a href="#cb29-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-381"><a href="#cb29-381" aria-hidden="true" tabindex="-1"></a>(the <span class="in">`&lt;`</span> and <span class="in">`&gt;`</span> must be omitted as they are used in the example as</span>
<span id="cb29-382"><a href="#cb29-382" aria-hidden="true" tabindex="-1"></a>field indicators only).</span>
<span id="cb29-383"><a href="#cb29-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-384"><a href="#cb29-384" aria-hidden="true" tabindex="-1"></a>Submit your appropriately named R documents on iKamva when ready.</span>
<span id="cb29-385"><a href="#cb29-385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-386"><a href="#cb29-386" aria-hidden="true" tabindex="-1"></a>Failing to follow these instructions carefully, precisely, and</span>
<span id="cb29-387"><a href="#cb29-387" aria-hidden="true" tabindex="-1"></a>thoroughly will cause you to lose marks, which could cause a significant</span>
<span id="cb29-388"><a href="#cb29-388" aria-hidden="true" tabindex="-1"></a>drop in your score as formatting counts for 15% of the final mark (out</span>
<span id="cb29-389"><a href="#cb29-389" aria-hidden="true" tabindex="-1"></a>of 100%).</span>
<span id="cb29-390"><a href="#cb29-390" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-391"><a href="#cb29-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-392"><a href="#cb29-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-393"><a href="#cb29-393" aria-hidden="true" tabindex="-1"></a><span class="fu">##     References</span></span>
<span id="cb29-394"><a href="#cb29-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-395"><a href="#cb29-395" aria-hidden="true" tabindex="-1"></a>::: {#refs}</span>
<span id="cb29-396"><a href="#cb29-396" aria-hidden="true" tabindex="-1"></a>:::</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">© 2023, AJ Smit</div>   
    <div class="nav-footer-center"><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">This page is built with <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>


<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>