<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.33">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Smit, A. J.">
<meta name="dcterms.date" content="2021-01-01">
<title>Lab 4. Species Distribution Patterns – The Tangled Bank</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../BDC334/BDC334_AI_Tutor.html" rel="next">
<link href="../BDC334/Lab-03-biodiversity.html" rel="prev">
<link href="../images/favicon.ico" rel="icon">
<script src="../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-59898bd1c6b9d2bb783127feaa000c76.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-7116f089cfdb46b66375c744b53d3c7f.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-SCYH68M489"></script><script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-SCYH68M489', { 'anonymize_ip': true});
</script><script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"headline",
  "consent_type":"express",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  ,
"language":"en"
  });
});
</script><style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script><script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script><script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script><link rel="stylesheet" href="../LesCodex2.css">
<meta property="og:title" content="Lab 4. Species Distribution Patterns – The Tangled Bank">
<meta property="og:description" content="UWC’s Biodiversity &amp; Conservation Biology Department Pages For BDC334, BCB743, and BCB744.">
<meta property="og:image" content="http://tangledbank.netlify.app/BDC334/data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==">
<meta property="og:site_name" content="The Tangled Bank">
</head>
<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">The Tangled Bank</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-vignettes" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Vignettes</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-vignettes">
<li>
    <a class="dropdown-item" href="../vignettes/netCDF_dates.html">
 <span class="dropdown-text">Dates From netCDF Files: Two Approaches</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/prep_NOAA_OISST.html">
 <span class="dropdown-text">Downloading and Prep: ERDDAP</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_ERDDAP.html">
 <span class="dropdown-text">Retrieving Chlorophyll-a Data from ERDDAP Servers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/alt_method.html">
 <span class="dropdown-text">Downloading and Prep: Python + CDO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/gridded_data.html">
 <span class="dropdown-text">Detecting Events in Gridded Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/regridding.html">
 <span class="dropdown-text">Regridding gridded data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/MHW_MCS_horizonplots.html">
 <span class="dropdown-text">Displaying MHWs and MCSs as Horizon Plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/heatwaveR_issues.html">
 <span class="dropdown-text">heatwaveR Issues</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_sightings.html">
 <span class="dropdown-text">Plotting the Whale Sightings and Chlorophyll-<em>a</em> Concentrations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_localisation.html">
 <span class="dropdown-text">Spatial Localisation, Subsetting, and Aggregation of the Chlorophyll-<em>a</em> Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/elem_ts_methods.html">
 <span class="dropdown-text">Wavelet Analysis of Diatom Time Series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/buffer_data_extract.html">
 <span class="dropdown-text">Extracting gridded data within a buffer</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-high-performace-computing" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">High-Performace Computing</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-high-performace-computing">
<li>
    <a class="dropdown-item" href="../vignettes/README_Lengau.html">
 <span class="dropdown-text">Using Lengau</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/PBSPro_users.html">
 <span class="dropdown-text">PBSPro user and job management</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/README_tmux.html">
 <span class="dropdown-text">Using tmux</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item">
    <a class="nav-link" href="../blog.html"> 
<span class="menu-text">Tangled Bank Blog</span></a>
  </li>  
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item">
    <a class="nav-link" href="../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-blogroll" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Blogroll</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-blogroll">
<li>
    <a class="dropdown-item" href="https://www.r-bloggers.com">
 <span class="dropdown-text">R-bloggers</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item compact">
    <a class="nav-link" href="../blog.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">
<li>
    <a class="dropdown-item" href="https://github.com/ajsmit/tangled_bank">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://bugs.com">
 <span class="dropdown-text">Report a Bug</span></a>
  </li>  
    </ul>
</li>
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BDC334/BDC334_syllabus.html">BDC334: Biogeography &amp; Global Ecology</a></li><li class="breadcrumb-item"><a href="../BDC334/Lab-04-biodiversity.html">Lab 4. Species Distribution Patterns</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../images/my_tree_hex.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none"></a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">UNDERGRADUATE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC223: Plant Ecophysiology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/BDC223_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L00-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 0: Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L01-worldmapper.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: Worldmapper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L02-SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2: Surface Area to Volume (SA/V) Ratios in Biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L03-plant_stress.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3: Plant Stress</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L04-carbon_cycle.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Perturbations to the Carbon Cycle</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L05-light.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 5: Light</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L06a-pigments_photosynthesis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6a: Pigments and Photosynthesis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L06b-jassby_platt.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6b: PI Curves – Jassby and Platt</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L07-chromatic_adaptation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 7: Chromatic Adaptation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L08a-nutrient_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8a: Nutrient Uptake</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L08b-nutrients_michaelis_menten.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8b: Uptake Kinetics – Michaelis-Menten</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab1_SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1: Surface Area to Volume (SA/V) Ratios in Biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab2_misc_calcs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2: Miscellaneous Calculations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab3_PI_curves.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3: PI Curves – Jassby and Platt</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab4_nitrogen_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4: Uptake Kinetics – Michaelis-Menten</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/BDC223_FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">BDC334: Biogeography &amp; Global Ecology</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/BDC334_syllabus.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/BDC334-Lecture-Transcripts.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">BCB334 Lecture Transcript</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: Introductory Lecture</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-02-ecosystems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2. Overview of Ecosystems</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-03-gradients.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3. Ecological Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Biodiversity Concepts</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 5: Multivariate Data</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-06-unified-ecology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6: Unified Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 7: Threats to Biodiversity</span>
  </li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 8: Nature's Contribution to People</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../docs/IKS_Conservation.pdf" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 9: Indigenous Knowledge Systems and Conservation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1. Ecological Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-02a-r_rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2a. Introduction `R` &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-02b-env_dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2b. Environmental Distance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-03-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3. Quantifying Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-04-biodiversity.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Lab 4. Species Distribution Patterns</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/BDC334_AI_Tutor.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">BCB334 AI Tutor</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/Transboundary_systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transboundary systems</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS CORE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">BCB744: Introduction to R, &amp; Biostatistics</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/BCB744_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/01-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. R &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/02-working-with-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Working With Data &amp; Code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/03-data-in-R.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Data Classes &amp; Structures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/04-workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. R Workflows</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/05-graphics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Graphics with <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/06-faceting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Faceting Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/07-brewing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Brewing Colours</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/08-mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Mapping With <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/09-mapping_style.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Mapping With Style</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/10-mapping_rnaturalearth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Mapping With Natural Earth</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/11-mapping_quakes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. The Fiji Earthquake Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/12-tidy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Tidy Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/13-tidier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Tidier Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/14-tidiest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Tidiest Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/15-recap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15. Recap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/BCB744_Intro_R_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Intro R Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/examples/BCB744_Intro_R_Example_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 Intro R Example 1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/examples/BCB744_Intro_R_Example_2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 Intro R Example 2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../self-assessment_v1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Self-Assessment Form</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/BCB744_Intro_R_Presentations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Presentations</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/BCB744_Intro_R_Test_2025.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Intro R Test</span></span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/01-scientific-philosophy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Philosophy of Science</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/02-summarise-and-describe.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Data Summaries &amp; Descriptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/03-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Statistical Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/04-distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data Distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/05-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Statistical Inference</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/06-assumptions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Assumptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/07-t_tests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. <em>t</em>-Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/08-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. ANOVA</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/09-regressions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Simple Linear Regressions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/10-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/11-decision_guide.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/12-glance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Non-Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/13-confidence.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Confidence Intervals</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/14-transformations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Data Transformations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/BCB744_Biostatistics_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Biostatistics Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/examples/BCB744_BioStats_Example_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 BioStats 2025 Example 1</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS ELECTIVES</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">BCB743: Quantitative Ecology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/BCB743_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/review.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1-4: Review Concepts fo Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5: Correlations &amp; Associations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7: Intro to Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8a: Principal Components Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA_examples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8b: PCA: Additional Examples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA_SDG_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8c: PCA: WHO SDG Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/CA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9a: Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/DCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9b: Detrended Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCoA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10: Principal Coordinates Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/nMDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11a: Non-metric Multidimensional Scaling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/nMDS_diatoms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11b: nMDS: PERMANOVA (Diatoms) Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/unconstrained-summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12: Unconstrained Ordination Summary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/constrained_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13a: Distance-Based Redundancy Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/two_oceans_appendices.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13b: db-RDA: Seaweeds Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/cluster_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14: Cluster Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/model_building.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15: Model Building</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/deep_dive.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16a: Deep Dive Into Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/multiple_regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16b: Multiple Regression</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../docs/Combined Faculty AI Policy 15082025.pdf" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Faculty of Natural Sciences AI Policy</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">SUPPORT</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Web Resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/general_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/ecology_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quantitative Ecology Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/spatial_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial R Resources</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#the-data" id="toc-the-data" class="nav-link active" data-scroll-target="#the-data"><span class="header-section-number">1</span> The Data</a></li>
  <li>
<a href="#species-abundance-distribution" id="toc-species-abundance-distribution" class="nav-link" data-scroll-target="#species-abundance-distribution"><span class="header-section-number">2</span> Species-Abundance Distribution</a>
  <ul class="collapse">
<li><a href="#calculating-a-sad-from-a-site-species-table" id="toc-calculating-a-sad-from-a-site-species-table" class="nav-link" data-scroll-target="#calculating-a-sad-from-a-site-species-table"><span class="header-section-number">2.1</span> Calculating a SAD From a Site × Species Table</a></li>
  </ul>
</li>
  <li>
<a href="#occupancy-abundance-relationship" id="toc-occupancy-abundance-relationship" class="nav-link" data-scroll-target="#occupancy-abundance-relationship"><span class="header-section-number">3</span> Occupancy-Abundance Relationship</a>
  <ul class="collapse">
<li><a href="#calculate-the-oar-from-a-site-species-table" id="toc-calculate-the-oar-from-a-site-species-table" class="nav-link" data-scroll-target="#calculate-the-oar-from-a-site-species-table"><span class="header-section-number">3.1</span> Calculate the OAR from a Site × Species Table</a></li>
  </ul>
</li>
  <li>
<a href="#species-area-accumulation" id="toc-species-area-accumulation" class="nav-link" data-scroll-target="#species-area-accumulation"><span class="header-section-number">4</span> Species-Area (Accumulation)</a>
  <ul class="collapse">
<li><a href="#calculating-a-species-accumulation-curve-from-scratch" id="toc-calculating-a-species-accumulation-curve-from-scratch" class="nav-link" data-scroll-target="#calculating-a-species-accumulation-curve-from-scratch"><span class="header-section-number">4.1</span> Calculating a Species Accumulation Curve From Scratch</a></li>
  </ul>
</li>
  <li>
<a href="#rarefaction-curves" id="toc-rarefaction-curves" class="nav-link" data-scroll-target="#rarefaction-curves"><span class="header-section-number">5</span> Rarefaction Curves</a>
  <ul class="collapse">
<li><a href="#recipe-for-individual-based-rarefaction" id="toc-recipe-for-individual-based-rarefaction" class="nav-link" data-scroll-target="#recipe-for-individual-based-rarefaction"><span class="header-section-number">5.1</span> “Recipe” For Individual-Based Rarefaction</a></li>
  <li><a href="#recipe-for-sample-based-rarefaction" id="toc-recipe-for-sample-based-rarefaction" class="nav-link" data-scroll-target="#recipe-for-sample-based-rarefaction"><span class="header-section-number">5.2</span> “Recipe” For Sample-Based Rarefaction</a></li>
  </ul>
</li>
  <li><a href="#distance-decay-curves-and-elevation-and-other-gradients" id="toc-distance-decay-curves-and-elevation-and-other-gradients" class="nav-link" data-scroll-target="#distance-decay-curves-and-elevation-and-other-gradients"><span class="header-section-number">6</span> Distance-Decay Curves and Elevation (and Other) Gradients</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BDC334/Lab-04-biodiversity.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="Lab-04-biodiversity.pdf"><i class="bi bi-file-pdf"></i>Typst</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BDC334/BDC334_syllabus.html">BDC334: Biogeography &amp; Global Ecology</a></li><li class="breadcrumb-item"><a href="../BDC334/Lab-04-biodiversity.html">Lab 4. Species Distribution Patterns</a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Lab 4. Species Distribution Patterns</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://tangledbank.netlify.app">Smit, A. J.</a> <a href="https://orcid.org/0000-0002-3799-6126" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uwc.ac.za">
            University of the Western Cape
            </a>
          </p>
      </div>
  </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 1, 2021</p>
    </div>
  </div>
  
    
  </div>
  


</header><div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
BCB743
</div>
</div>
<div class="callout-body-container callout-body">
<p><strong>This material must be reviewed by BCB743 students in Week 1 of Quantitative Ecology.</strong></p>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
This Lab Accompanies the Following Lecture
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li><a href="../BDC334/Lec-06-unified-ecology.html">Lecture 6: Unified Ecology</a></li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Data For This Lab
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>The Barro Colorado Island Tree Counts data <span class="citation" data-cites="condit2002beta">(<a href="#ref-condit2002beta" role="doc-biblioref">Condit et al. 2002</a>)</span> – load <strong>vegan</strong> and load the data with <code>data(BCI)</code>
</li>
<li>The Oribatid mite data <span class="citation" data-cites="borcard1992partialling borcard1994environmental">(<a href="#ref-borcard1992partialling" role="doc-biblioref">Borcard et al. 1992</a>; <a href="#ref-borcard1994environmental" role="doc-biblioref">Borcard and Legendre 1994</a>)</span> – load <strong>vegan</strong> and load the data with <code>data(mite)</code>
</li>
<li>The seaweed species data <span class="citation" data-cites="smit2017seaweeds">(<a href="#ref-smit2017seaweeds" role="doc-biblioref">Smit et al. 2017</a>)</span> – <a href="../data/seaweed/SeaweedSpp.csv"><code>SeaweedSpp.csv</code></a>
</li>
<li>The Doubs River species data <span class="citation" data-cites="verneaux1973cours borcard2011numerical">(<a href="#ref-verneaux1973cours" role="doc-biblioref">Verneaux 1973</a>; <a href="#ref-borcard2011numerical" role="doc-biblioref">Borcard et al. 2011</a>)</span> – <a href="../data/DoubsSpe.csv"><code>DoubsSpe.csv</code></a>
</li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Reading Required For This Lab
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li><a href="../docs/Matthews_and_Whittaker_2015.pdf">Matthews and Whittaker (2015)</a></li>
<li><a href="../docs/Shade_et_al_2018.pdf">Shade et al.&nbsp;(2018)</a></li>
</ul>
</div>
</div>
<p>In this Lab, we will calculate the various species distribution patterns included in the paper by <a href="../docs/Shade_et_al_2018.pdf">Shade et al.&nbsp;(2018)</a>.</p>
<section id="the-data" class="level2" data-number="1"><h2 data-number="1" class="anchored" data-anchor-id="the-data">
<span class="header-section-number">1</span> The Data</h2>
<p>We will calculate each for the <a href="https://repository.si.edu/handle/10088/20925">Barro Colorado Island Tree Counts data</a> that come with <strong>vegan</strong>. See <code><a href="https://vegandevs.github.io/vegan/reference/BCI.html">?vegan::BCI</a></code> for a description of the data contained with the package, as well as a selection of publications relevant to the data and analyses. The primary publication of interest is <span class="citation" data-cites="condit2002beta">Condit et al. (<a href="#ref-condit2002beta" role="doc-biblioref">2002</a>)</span>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(vegan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="co">#library(vegan) # already loaded</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="co">#library(tidyverse) # already loaded</span></span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="fu">data</span>(BCI) <span class="co"># data contained within vegan</span></span>
<span id="cb2-4"><a href="#cb2-4"></a></span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="co"># make a head-tail function</span></span>
<span id="cb2-6"><a href="#cb2-6"></a>ht <span class="ot">&lt;-</span> <span class="cf">function</span>(d) <span class="fu">rbind</span>(<span class="fu">head</span>(d, <span class="dv">7</span>), <span class="fu">tail</span>(d, <span class="dv">7</span>))</span>
<span id="cb2-7"><a href="#cb2-7"></a></span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="co"># Lets look at a portion of the data:</span></span>
<span id="cb2-9"><a href="#cb2-9"></a><span class="fu">ht</span>(BCI)[<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Abarema.macradenia Vachellia.melanoceras Acalypha.diversifolia
1                  0                     0                     0
2                  0                     0                     0
3                  0                     0                     0
4                  0                     0                     0
5                  0                     0                     0
6                  0                     0                     0
7                  0                     0                     0
  Acalypha.macrostachya Adelia.triloba Aegiphila.panamensis
1                     0              0                    0
2                     0              0                    0
3                     0              0                    0
4                     0              3                    0
5                     0              1                    1
6                     0              0                    0
7                     0              0                    1
  Alchornea.costaricensis
1                       2
2                       1
3                       2
4                      18
5                       3
6                       2
7                       0</code></pre>
</div>
</div>
</section><section id="species-abundance-distribution" class="level2" data-number="2"><h2 data-number="2" class="anchored" data-anchor-id="species-abundance-distribution">
<span class="header-section-number">2</span> Species-Abundance Distribution</h2>
<p>The species abundance distribution (SAD) expresses one of ecology’s most persistent regularities, <em>viz.</em>, in almost any community, a few species dominate in abundance, while the majority are comparatively rare. This skewed pattern is often summarised as “few common, many rare” and has been recognised since the earliest quantitative analyses of community structure. It has become a central object of ecological theory. Graphically, SADs plot species abundance against either species identity or rank to provide a means of comparing empirical assemblages with theoretical expectations of community organisation.</p>
<p>The first formulation of this pattern was developed by <span class="citation" data-cites="fisher1943relation">Fisher et al. (<a href="#ref-fisher1943relation" role="doc-biblioref">1943</a>)</span>, who observed that species frequencies often conform to what he termed a <strong>logarithmic series distribution</strong>. In this representation, <em>the expected number of species with a given number of individuals declines rapidly as abundance increases</em>, showing communities where most species occur at low frequencies and only a small fraction reach very high densities. The curve of Fisher’s log-series distribution thus shows the expected number of species <span class="math inline">\(f\)</span> with <span class="math inline">\(n\)</span> observed individuals, and the interpretation of this curve is conceptually similar across all SAD models—only the underlying mathematics and rationale differ. Fisher’s model remains widely used as a conceptual benchmark and as a statistical model. In <strong>R</strong>, the <strong>vegan</strong> package implements this procedure through the <code>fisherfit()</code> function, which estimates the log-series parameter and compares it to observed community data (<a href="#fig-fishers-log" class="quarto-xref">Figure&nbsp;1</a>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># take one random sample of a row (site):</span></span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="co"># for this website's purpose, this function ensure the same random</span></span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="co"># sample is drawn each time the web page is recreated</span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="fu">set.seed</span>(<span class="dv">13</span>) </span>
<span id="cb4-5"><a href="#cb4-5"></a>k <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">nrow</span>(BCI), <span class="dv">1</span>)</span>
<span id="cb4-6"><a href="#cb4-6"></a>fish <span class="ot">&lt;-</span> <span class="fu">fisherfit</span>(BCI[k,])</span>
<span id="cb4-7"><a href="#cb4-7"></a>fish</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Fisher log series model
No. of species: 95 
Fisher alpha:   39.87659 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="fu">plot</span>(fish)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-fishers-log" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-fishers-log-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-04-biodiversity_files/figure-html/fig-fishers-log-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: Fisher’s log series distribution calculated for the Barro Colorado Island Tree Counts data."><img src="Lab-04-biodiversity_files/figure-html/fig-fishers-log-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-fishers-log-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Fisher’s log series distribution calculated for the Barro Colorado Island Tree Counts data.
</figcaption></figure>
</div>
</div>
</div>
<p>Building on this foundation, <span class="citation" data-cites="preston1948commonness">Preston (<a href="#ref-preston1948commonness" role="doc-biblioref">1948</a>)</span> introduced an alternative formulation: the <strong>log-normal distribution</strong> of species abundances. Instead of ranking species, he grouped them into “octaves”, <em>i.e.</em>, abundance classes that double in size (1, 2, 4, 8, 16, 32, …), and showed that, <em>in sufficiently well-sampled communities</em>, the frequency of species within these classes approximates a bell-shaped Gaussian curve when plotted on a logarithmic scale. Preston also emphasised the effect of incomplete sampling: rare species falling below a “veil line” remain undetected, giving the distribution its characteristic truncated form. This method captures the idea that rarity and commonness are not absolute properties but functions of sampling intensity. In <strong>vegan</strong>, Preston’s method is implemented with mathematical modifications to handle ties more effectively via the <code>prestondistr()</code> function (<a href="#fig-preston" class="quarto-xref">Figure&nbsp;2</a>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>pres <span class="ot">&lt;-</span> <span class="fu">prestondistr</span>(BCI[k,])</span>
<span id="cb7-2"><a href="#cb7-2"></a>pres</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Preston lognormal model
Method: maximized likelihood to log2 abundances 
No. of species: 95 

      mode      width         S0 
 0.9234918  1.6267630 26.4300640 

Frequencies by Octave
                0        1        2        3        4        5         6
Observed 19.00000 27.00000 21.50000 17.00000 7.000000 2.500000 1.0000000
Fitted   22.49669 26.40085 21.23279 11.70269 4.420327 1.144228 0.2029835</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="fu">plot</span>(pres)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-preston" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-preston-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-04-biodiversity_files/figure-html/fig-preston-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;2: Preston’s log-normal distribution demonstrated for the BCI data."><img src="Lab-04-biodiversity_files/figure-html/fig-preston-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-preston-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Preston’s log-normal distribution demonstrated for the BCI data.
</figcaption></figure>
</div>
</div>
</div>
<p>A further step was taken by <span class="citation" data-cites="whittaker1965dominance">Whittaker (<a href="#ref-whittaker1965dominance" role="doc-biblioref">1965</a>)</span>, who reformulated SADs as <strong>rank-abundance distributions</strong> (RADs), also known as dominance–diversity curves or Whittaker plots. Here, species are ordered from most to least abundant along the <span class="math inline">\(x\)</span>-axis, and their relative abundances (often log-transformed) are plotted on the <span class="math inline">\(y\)</span>-axis. The slope and curvature of the resulting profile reveal both evenness and the extent of dominance: steep curves indicate communities dominated by a few species, whereas long, shallow tails point to greater species equity. Unlike Fisher’s or Preston’s frequency-based models, Whittaker’s approach highlights the shape of the assemblage as a whole and facilitates comparison across communities with different richness levels. In <strong>vegan</strong>, this is achieved via the <code>radfit()</code> function, which overlays multiple fitted models—broken-stick, preemption, log-normal, Zipf, and Zipf–Mandelbrot—onto the observed ranked data (<a href="#fig-rad" class="quarto-xref">Figure&nbsp;3</a>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>rad <span class="ot">&lt;-</span> <span class="fu">radfit</span>(BCI[k,])</span>
<span id="cb10-2"><a href="#cb10-2"></a>rad</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
RAD models, family poisson 
No. of species 95, total abundance 392

           par1      par2     par3    Deviance AIC      BIC     
Null                                   56.3132 324.6477 324.6477
Preemption  0.042685                   55.8621 326.1966 328.7504
Lognormal   0.84069   1.0912           16.1740 288.5085 293.6162
Zipf        0.12791  -0.80986          21.0817 293.4161 298.5239
Mandelbrot  0.66461  -1.2374   4.1886   6.6132 280.9476 288.6093</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="fu">plot</span>(rad)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-rad" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-rad-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-04-biodiversity_files/figure-html/fig-rad-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;3: Whittaker’s rank abundance distribution curves demonstrated for the BCI data."><img src="Lab-04-biodiversity_files/figure-html/fig-rad-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rad-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Whittaker’s rank abundance distribution curves demonstrated for the BCI data.
</figcaption></figure>
</div>
</div>
</div>
<p>We can also fit the rank abundance distribution curves to several sites at once (previously we have done so on only one site) (<a href="#fig-rad2" class="quarto-xref">Figure&nbsp;4</a>):</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>m <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">nrow</span>(BCI), <span class="dv">6</span>)</span>
<span id="cb13-2"><a href="#cb13-2"></a>rad2 <span class="ot">&lt;-</span> <span class="fu">radfit</span>(BCI[m, ])</span>
<span id="cb13-3"><a href="#cb13-3"></a>rad2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Deviance for RAD models:

                  3       37       10       13        6       22
Null        86.1127  93.5952  77.2737  52.6207  72.1627 114.1747
Preemption  58.9295 104.0978  62.7210  57.7372  54.7709 110.5156
Lognormal   29.2719  19.0653  20.4770  15.8218  19.5788  26.2510
Zipf        50.1262  11.3048  39.7066  22.8006  32.4630  15.5222
Mandelbrot   5.7342   8.9107   9.8353  12.1701   5.5973   9.6047</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="fu">plot</span>(rad2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-rad2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-rad2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-04-biodiversity_files/figure-html/fig-rad2-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Figure&nbsp;4: Rank abundance distribution curves fitted to several sites."><img src="Lab-04-biodiversity_files/figure-html/fig-rad2-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rad2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Rank abundance distribution curves fitted to several sites.
</figcaption></figure>
</div>
</div>
</div>
<p>Above, we see that the model selected for capturing the shape of the SAD is the Mandelbrot, and it is plotted individually for each of the randomly selected sites. Model selection works through Akaike’s or Schwartz’s Bayesian information criteria (AIC or BIC; AIC is the default—select the model with the lowest AIC).</p>
<p><a href="https://github.com/cran/BiodiversityR"><strong>BiodiversityR</strong></a> (and <a href="http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf">here</a> and <a href="https://rpubs.com/Roeland-KINDT">here</a>) also offers options for rank abundance distribution curves; see <code>rankabundance()</code> (<a href="#fig-rac" class="quarto-xref">Figure&nbsp;5</a>):</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="fu">library</span>(BiodiversityR)</span>
<span id="cb16-2"><a href="#cb16-2"></a>rankabund <span class="ot">&lt;-</span> <span class="fu">rankabundance</span>(BCI)</span>
<span id="cb16-3"><a href="#cb16-3"></a><span class="fu">rankabunplot</span>(rankabund, <span class="at">cex =</span> <span class="fl">0.8</span>, <span class="at">pch =</span> <span class="fl">0.8</span>, <span class="at">col =</span> <span class="st">"indianred4"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-rac" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-rac-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-04-biodiversity_files/figure-html/fig-rac-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Figure&nbsp;5: Rank-abundance curves for the BCI data."><img src="Lab-04-biodiversity_files/figure-html/fig-rac-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rac-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Rank-abundance curves for the BCI data.
</figcaption></figure>
</div>
</div>
</div>
<p>Refer to the help files for the respective functions to see their differences.</p>
<section id="calculating-a-sad-from-a-site-species-table" class="level3" data-number="2.1"><h3 data-number="2.1" class="anchored" data-anchor-id="calculating-a-sad-from-a-site-species-table">
<span class="header-section-number">2.1</span> Calculating a SAD From a Site × Species Table</h3>
<p>Here’s how to manually derive an SAD from a typical community dataset. Let’s assume you have a site × species table, where rows are sampling sites and columns are species, with the cells containing counts (abundances).</p>
<ol type="1">
<li><p><strong>Get Total Species Abundances</strong>: Your first step is to get the total abundance for each species across all your sites. You do this by <strong>summing each column</strong>. This collapses your site × species matrix into a single list of numberswhich we call an “<strong>abundance vector</strong>;” here each number is the total count for one species.</p></li>
<li><p><strong>Create a Frequency Table</strong>: Now, you’ll process this abundance vector. Count <em>how many</em> species have an abundance of 1, how many have an abundance of 2, and so on, for all abundance values present in your data. The result is a frequency table.</p></li>
</ol>
<p>For example, if your abundance vector is <code>(1, 1, 5, 2, 8, 1, 2, 5)</code>, your frequency table would be:</p>
<ul>
<li>Abundance Class 1: 3 species</li>
<li>Abundance Class 2: 2 species</li>
<li>Abundance Class 5: 2 species</li>
<li>Abundance Class 8: 1 species</li>
</ul>
<ol start="3" type="1">
<li><p><strong>Plot the SAD</strong>: This frequency table is your SAD. To visualise it as a frequency distribution (like for Fisher’s or Preston’s models), you plot the <strong>Abundance Class on the <em>x</em>-axis</strong> and the <strong>Number of Species (frequency) on the <em>y</em>-axis</strong>. You will typically see that classic hollow curve.</p></li>
<li><p><strong>Plot the RAD (Whittaker Plot)</strong>: To create a Rank Abundance Distribution from the same data, you take your original abundance vector <code>(1, 1, 5, 2, 8, 1, 2, 5)</code>, sort it in descending order <code>(8, 5, 5, 2, 2, 1, 1, 1)</code>, and then plot it. The <strong><em>x</em>-axis is the rank</strong> <code>(1, 2, 3, 4, 5, 6, 7, 8)</code>, and the <strong><em>y</em>-axis is the corresponding abundance value</strong> <code>(8, 5, 5, 2, 2, 1, 1, 1)</code>.</p></li>
</ol></section></section><section id="occupancy-abundance-relationship" class="level2" data-number="3"><h2 data-number="3" class="anchored" data-anchor-id="occupancy-abundance-relationship">
<span class="header-section-number">3</span> Occupancy-Abundance Relationship</h2>
<p><strong>Occupancy</strong> refers to the number (or proportion) of sites in which a species is recorded as present. When linked with measures of abundance, it provides a way to examine the spatial distribution of species across a landscape. <strong>Occupancy–abundance relationships</strong> (OARs) reveal that species that are locally abundant at a given site tend also to be widespread across many sites in the region; conversely, species of low local abundance are more often restricted in their wider distribution. This empirical pattern has been so consistently observed that it has been described as verging on an ecological “law;” it tell us about the underlying processes of niche breadth, dispersal capacity, and environmental tolerance. OARs are therefore often used to infer degrees of niche specialisation, with generalists expected to show both high occupancy and high abundance, and specialists clustering toward low values of both measures (<a href="#fig-occupancy" class="quarto-xref">Figure&nbsp;6</a>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb17-2"><a href="#cb17-2"></a></span>
<span id="cb17-3"><a href="#cb17-3"></a><span class="co"># A function for counts:</span></span>
<span id="cb17-4"><a href="#cb17-4"></a><span class="co"># count number of non-zero elements per column</span></span>
<span id="cb17-5"><a href="#cb17-5"></a>count_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb17-6"><a href="#cb17-6"></a>  <span class="fu">length</span>(x[x <span class="sc">&gt;</span> <span class="dv">0</span>])</span>
<span id="cb17-7"><a href="#cb17-7"></a>}</span>
<span id="cb17-8"><a href="#cb17-8"></a></span>
<span id="cb17-9"><a href="#cb17-9"></a>BCI_OA <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">occ =</span> <span class="fu">apply</span>(BCI, <span class="at">MARGIN =</span> <span class="dv">2</span>, count_fun),</span>
<span id="cb17-10"><a href="#cb17-10"></a>                     <span class="at">ab =</span> <span class="fu">apply</span>(BCI, <span class="at">MARGIN =</span> <span class="dv">2</span>, mean))</span>
<span id="cb17-11"><a href="#cb17-11"></a></span>
<span id="cb17-12"><a href="#cb17-12"></a><span class="fu">ggplot</span>(BCI_OA, <span class="fu">aes</span>(<span class="at">x =</span> ab, <span class="at">y =</span> occ<span class="sc">/</span><span class="fu">max</span>(occ))) <span class="sc">+</span></span>
<span id="cb17-13"><a href="#cb17-13"></a>  <span class="fu">geom_point</span>(<span class="at">colour =</span> <span class="st">"indianred3"</span>) <span class="sc">+</span></span>
<span id="cb17-14"><a href="#cb17-14"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb17-15"><a href="#cb17-15"></a>  <span class="co"># scale_y_log10() +</span></span>
<span id="cb17-16"><a href="#cb17-16"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Barro Colorado Island Tree Counts"</span>,</span>
<span id="cb17-17"><a href="#cb17-17"></a>     <span class="at">x =</span> <span class="st">"Log (abundance)"</span>, <span class="at">y =</span> <span class="st">"Occupancy"</span>) <span class="sc">+</span></span>
<span id="cb17-18"><a href="#cb17-18"></a>  <span class="fu">theme_linedraw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-occupancy" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-occupancy-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-04-biodiversity_files/figure-html/fig-occupancy-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-6" title="Figure&nbsp;6: Occupancy-abundance relationships seen in the BCI data."><img src="Lab-04-biodiversity_files/figure-html/fig-occupancy-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-occupancy-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: Occupancy-abundance relationships seen in the BCI data.
</figcaption></figure>
</div>
</div>
</div>
<section id="calculate-the-oar-from-a-site-species-table" class="level3" data-number="3.1"><h3 data-number="3.1" class="anchored" data-anchor-id="calculate-the-oar-from-a-site-species-table">
<span class="header-section-number">3.1</span> Calculate the OAR from a Site × Species Table</h3>
<p>Calculating and plotting this relationship is easy. Let’s start with your site × species table with cells containing abundance counts.</p>
<ol type="1">
<li><p><strong>Calculate Occupancy for Each Species</strong>: Occupancy is simply <em>the number or proportion of sites where a species is present</em>. To calculate this, you look down each species’ column and count how many of the cells have a value greater than zero. You’ll end up with a vector of numbers (integers; counts), one number for each species, representing its occupancy. For example, if a species is found at 5 out of 20 surveyed sites, its occupancy count is 5 (or its proportional occupancy is <span class="math inline">\(5/20=0.25\)</span>).</p></li>
<li><p><strong>Calculate Mean Local Abundance for Each Species</strong>: This step is important. You need the <strong>average abundance</strong> of a species <em>only in the sites where it actually occurs</em>. Including sites where it’s absent (with an abundance of zero) would artificially deflate the average and obscure the true relationship. Then, for each species, <strong>sum all the abundance values in its column</strong> (the total abundance) and divide this by the occupancy count you calculated in Step 1. This gives you the <strong>mean local abundance</strong> for that species.</p></li>
<li><p><strong>Plot the Relationship</strong>: You now have two corresponding vectors of values for all your species: one for occupancy and one for mean local abundance. To visualise the OAR, you create a scatter plot:</p></li>
</ol>
<ul>
<li>
<em>x</em>-axis: Occupancy (the number or proportion of sites occupied).</li>
<li>
<em>y</em>-axis: Mean Local Abundance. Because abundance values often span several orders of magnitude, this axis is almost always log-transformed (<em>e.g.</em>, using <span class="math inline">\(log_{10}\)</span> or <span class="math inline">\(log_{e}\)</span>) to better bring out the pattern and meet the assumptions of statistical tests.</li>
</ul></section></section><section id="species-area-accumulation" class="level2" data-number="4"><h2 data-number="4" class="anchored" data-anchor-id="species-area-accumulation">
<span class="header-section-number">4</span> Species-Area (Accumulation)</h2>
<p>Species accumulation curves—often presented as sample-based species–area relationships (SAR)—address the question: given what has already been sampled, how many species are yet unseen, and how does observed richness grow as sampling effort expands? Within a single ecosystem type, one ordinarily expects richness to rise with the number of distinct sampling units (plots, quadrats, trawls), then bend toward a plateau as further additions yield few or no new species. The curve is thus a record of discovery under a specified sampling design. Conceptually, the <span class="math inline">\(x\)</span>-axis represents cumulative sampling effort (number of sites, or cumulative area if sites are areally commensurate), while the <span class="math inline">\(y\)</span>-axis records the expected number of species discovered at that effort. As <a href="http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf">Roeland Kindt</a> explains (see p.&nbsp;41 in the cited note), the curve is generated by “adding” sites and tallying the average richness at each step—an averaging that already hints at a statistical question: in what order are sites added, and how is that order handled?</p>
<p>The <code>specaccum()</code> function has many different ways of adding the new sites to the curve, but the default ‘exact’ seems to be a sensible choice. <strong>BiodiversityR</strong> has the <code>accumresult()</code> function that does nearly the same. Let’s demonstrate using <strong>vegan</strong>’s function (<a href="#fig-accum" class="quarto-xref">Figure&nbsp;7</a>, <a href="#fig-accum2" class="quarto-xref">Figure&nbsp;8</a>, and <a href="#fig-accum3" class="quarto-xref">Figure&nbsp;9</a>):</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>sp1 <span class="ot">&lt;-</span> <span class="fu">specaccum</span>(BCI)</span>
<span id="cb18-2"><a href="#cb18-2"></a>sp2 <span class="ot">&lt;-</span> <span class="fu">specaccum</span>(BCI, <span class="st">"random"</span>)</span>
<span id="cb18-3"><a href="#cb18-3"></a></span>
<span id="cb18-4"><a href="#cb18-4"></a><span class="co"># par(mfrow = c(2,2), mar = c(4,2,2,1))</span></span>
<span id="cb18-5"><a href="#cb18-5"></a><span class="co"># par(mfrow = c(1,2))</span></span>
<span id="cb18-6"><a href="#cb18-6"></a><span class="fu">plot</span>(sp1, <span class="at">ci.type =</span> <span class="st">"polygon"</span>, <span class="at">col =</span> <span class="st">"indianred4"</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">ci.lty =</span> <span class="dv">0</span>,</span>
<span id="cb18-7"><a href="#cb18-7"></a>     <span class="at">ci.col =</span> <span class="st">"steelblue2"</span>, <span class="at">main =</span> <span class="st">"Default: exact"</span>,</span>
<span id="cb18-8"><a href="#cb18-8"></a>     <span class="at">ylab =</span> <span class="st">"No. of species"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-accum" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-accum-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-04-biodiversity_files/figure-html/fig-accum-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-7" title="Figure&nbsp;7: Species-area accumulation curves seen in the BCI data."><img src="Lab-04-biodiversity_files/figure-html/fig-accum-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-accum-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: Species-area accumulation curves seen in the BCI data.
</figcaption></figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>mods <span class="ot">&lt;-</span> <span class="fu">fitspecaccum</span>(sp2, <span class="st">"arrh"</span>)</span>
<span id="cb19-2"><a href="#cb19-2"></a><span class="fu">plot</span>(mods, <span class="at">col =</span> <span class="st">"indianred"</span>, <span class="at">ylab =</span> <span class="st">"No. of species"</span>)</span>
<span id="cb19-3"><a href="#cb19-3"></a><span class="fu">boxplot</span>(sp2, <span class="at">col =</span> <span class="st">"yellow"</span>, <span class="at">border =</span> <span class="st">"steelblue2"</span>, <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">cex =</span> <span class="fl">0.3</span>, <span class="at">add =</span> <span class="cn">TRUE</span>)</span>
<span id="cb19-4"><a href="#cb19-4"></a><span class="fu">sapply</span>(mods<span class="sc">$</span>models, AIC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  [1] 311.4642 303.7835 346.3668 320.0786 338.7978 320.2538 325.6968 346.2671
  [9] 320.3900 343.8570 318.2509 369.8303 335.9936 350.8711 327.9831 348.1287
 [17] 328.2393 347.8133 324.3837 314.8555 333.1390 340.5678 332.6836 360.5208
 [25] 335.3660 325.3150 347.4324 336.7498 336.6374 276.1878 349.9283 295.0268
 [33] 308.4656 315.8304 303.0776 329.8425 356.2393 368.4302 318.0514 359.5975
 [41] 327.4228 335.7604 259.8340 318.0063 335.7753 285.8790 323.5174 300.3546
 [49] 327.1448 355.2747 288.2583 366.5995 287.4120 327.5877 362.6487 323.5904
 [57] 339.5650 321.2264 336.6331 353.1295 317.9578 311.6528 336.3613 337.8327
 [65] 328.4787 311.6842 345.8035 367.5620 319.0269 305.6546 338.7805 321.8859
 [73] 330.6029 326.7097 345.8923 338.4755 352.8710 355.8038 307.7327 329.2355
 [81] 341.6628 340.1687 333.4771 348.3144 321.4417 317.4331 339.2211 313.1990
 [89] 305.3069 342.4581 318.0308 299.7067 294.7851 324.3237 333.5849 349.2749
 [97] 369.8287 323.0041 332.6820 329.3875</code></pre>
</div>
<div class="cell-output-display">
<div id="fig-accum2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-accum2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-04-biodiversity_files/figure-html/fig-accum2-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-8" title="Figure&nbsp;8: Fit Arrhenius models to all random accumulations"><img src="Lab-04-biodiversity_files/figure-html/fig-accum2-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-accum2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8: Fit Arrhenius models to all random accumulations
</figcaption></figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a>accum <span class="ot">&lt;-</span> <span class="fu">accumresult</span>(BCI, <span class="at">method =</span> <span class="st">"exact"</span>, <span class="at">permutations =</span> <span class="dv">100</span>)</span>
<span id="cb21-2"><a href="#cb21-2"></a><span class="fu">accumplot</span>(accum)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-accum3" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-accum3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-04-biodiversity_files/figure-html/fig-accum3-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-9" title="Figure&nbsp;9: A species accumulation curve."><img src="Lab-04-biodiversity_files/figure-html/fig-accum3-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-accum3-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;9: A species accumulation curve.
</figcaption></figure>
</div>
</div>
</div>
<p>Species accumulation curves can also be calculated with the <code>alpha.accum()</code> function of the <strong>BAT</strong> package (<a href="#fig-accum4" class="quarto-xref">Figure&nbsp;10</a>). In addition, the <strong>BAT</strong> package can also apply various diversity and species distribution assessments to <strong>phylogenetic</strong> and <strong>functional</strong> diversity. See the examples provided by <span class="citation" data-cites="cardoso2015bat">Cardoso et al. (<a href="#ref-cardoso2015bat" role="doc-biblioref">2015</a>)</span>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a><span class="fu">library</span>(BAT)</span>
<span id="cb22-2"><a href="#cb22-2"></a>BCI.acc <span class="ot">&lt;-</span> <span class="fu">alpha.accum</span>(BCI, <span class="at">prog =</span> <span class="cn">FALSE</span>)</span>
<span id="cb22-3"><a href="#cb22-3"></a></span>
<span id="cb22-4"><a href="#cb22-4"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb22-5"><a href="#cb22-5"></a><span class="fu">plot</span>(BCI.acc[,<span class="dv">2</span>], BCI.acc[,<span class="dv">17</span>], <span class="at">col =</span> <span class="st">"indianred"</span>,</span>
<span id="cb22-6"><a href="#cb22-6"></a>     <span class="at">xlab =</span> <span class="st">"Individuals"</span>, <span class="at">ylab =</span> <span class="st">"Chao1P"</span>)</span>
<span id="cb22-7"><a href="#cb22-7"></a><span class="fu">plot</span>(BCI.acc[,<span class="dv">2</span>], <span class="fu">slope</span>(BCI.acc)[,<span class="dv">17</span>], <span class="at">col =</span> <span class="st">"indianred"</span>,</span>
<span id="cb22-8"><a href="#cb22-8"></a>     <span class="at">xlab =</span> <span class="st">"Individuals"</span>, <span class="at">ylab =</span> <span class="st">"Slope"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-accum4" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-accum4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-04-biodiversity_files/figure-html/fig-accum4-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-10" title="Figure&nbsp;10: A species accumulation curve made with the alpha.accum() function of BAT."><img src="Lab-04-biodiversity_files/figure-html/fig-accum4-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-accum4-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10: A species accumulation curve made with the <code>alpha.accum()</code> function of <strong>BAT</strong>.
</figcaption></figure>
</div>
</div>
</div>
<section id="calculating-a-species-accumulation-curve-from-scratch" class="level3" data-number="4.1"><h3 data-number="4.1" class="anchored" data-anchor-id="calculating-a-species-accumulation-curve-from-scratch">
<span class="header-section-number">4.1</span> Calculating a Species Accumulation Curve From Scratch</h3>
<p>Calculating a proper SAC from a site × species table involves more than just adding up samples in their original order. The order in which you “collect” your samples can drastically change the shape of the curve. To solve this, the process relies on repeated randomisation to create a smooth, average curve.</p>
<ol type="1">
<li><strong>Randomise the Order of Samples</strong></li>
</ol>
<p>The first and most important step is to randomise the order of your samples (the rows in your site × species table). This ensures the resulting curve isn’t biased by happening to have a particularly species-rich or species-poor site at the beginning of the sequence.</p>
<ol start="2" type="1">
<li><strong>Accumulate Species in Sequence</strong></li>
</ol>
<p>Once the sample order is shuffled, you build the curve step-by-step:</p>
<ul>
<li>For the 1st sample: The cumulative species count is simply the number of species in that first sample.</li>
<li>For the 2nd sample: You pool the species from the first and second samples and count the total number of unique species found across both.</li>
<li>For the 3rd sample: You pool the species from the first three samples and count the total number of unique species found across all three.</li>
<li>You continue this process until all samples in your shuffled list have been added. This single pass creates one possible accumulation curve.</li>
</ul>
<ol start="3" type="1">
<li><strong>Repeat and Average</strong></li>
</ol>
<p>A curve from a single random order is not representative. Therefore, you must repeat steps 1 and 2 many times (typically 100 to 1,000 times), creating a new random permutation of your samples each time. This gives you a large collection of possible accumulation curves.</p>
<ol start="4" type="1">
<li><strong>Plot the Final Curve</strong></li>
</ol>
<p>The final species accumulation curve is not a single line but rather the average of all the randomised curves you generated.</p>
<ul>
<li>
<em>x</em>-axis: Number of Samples Accumulated (<em>e.g.</em>, 1, 2, 3, … up to your total number of sites).</li>
<li>
<em>y</em>-axis: The average cumulative number of species found at that level of sampling effort.</li>
</ul>
<p>It’s also standard practice to plot the standard deviation or confidence intervals around this mean curve. This creates a shaded region or error bars, showing the amount of variation from the different randomisations. A narrow confidence interval indicates a more robust and reliable curve.</p>
</section></section><section id="rarefaction-curves" class="level2" data-number="5"><h2 data-number="5" class="anchored" data-anchor-id="rarefaction-curves">
<span class="header-section-number">5</span> Rarefaction Curves</h2>
<p>Like <strong>species accumulation curves</strong>, <strong>rarefaction curves</strong> also address the question of unseen species, but they do so in a different way. Rarefaction – literally “scaling down” <span class="citation" data-cites="heck1975explicit">(<a href="#ref-heck1975explicit" role="doc-biblioref">Heck Jr et al. 1975</a>)</span> – is a statistical technique that estimates species richness (often denoted S, or sometimes diversity indices such as Shannon diversity, <span class="math inline">\(H’\)</span>, or Simpson’s diversity, <span class="math inline">\(\lambda\)</span>) from community data, typically organised as a site × species table. The main goal is to ask: given a smaller sample of individuals or sites than were actually collected, how many species would we expect to detect? Rarefaction thus provides a way of standardising richness estimates across samples of unequal size and of judging whether sampling effort has been sufficient to capture the community’s complement of species.</p>
<p>At first sight, rarefaction curves resemble species accumulation curves, since both plot species richness against increasing effort. The difference lies in how “effort” is defined and how the curves are generated. <strong>Species accumulation curves</strong> build up by successively adding new sites or samples (rows of the community matrix) and tallying the union of species observed. By contrast, <strong>rarefaction curves</strong> are constructed by randomly sub-sampling the existing pool of <span class="math inline">\(N\)</span> individuals or samples. Repeated resampling at size <span class="math inline">\(n\)</span> (where <span class="math inline">\(1 \leq n \leq N\)</span>) yields an expected number of species at that level of effort, and the curve plots this expectation as a function of <span class="math inline">\(n\)</span>.</p>
<p>Two standard rarefaction approaches exist. In <strong>individual-based rarefaction</strong>, the <span class="math inline">\(x\)</span>-axis represents the number of individuals sampled across all species, with richness S as the response. The principle is intuitive: when only a few individuals are drawn, they are likely to belong to a small set of species, but as more individuals are added, the chance of encountering new species increases—at a decelerating rate as common species are already represented. In <strong>sample-based rarefaction</strong>, the <span class="math inline">\(x\)</span>-axis represents the number of samples (sites, quadrats, trawls), and species richness is again tallied as a function of sample number. In both cases, the resulting curves rise steeply at first, then level off toward a plateau, reflecting the declining probability of detecting the rarest species as sampling effort grows.</p>
<p>This resampling distinction is what sets rarefaction apart from <strong>species accumulation curves</strong> and <strong>species abundance distributions (SADs)</strong>. While SADs describe how species are distributed in terms of abundance and accumulation curves describe how observed richness grows as sampling units are added, rarefaction explicitly <strong>re-samples the observed data at smaller effort levels</strong> to estimate expected richness. In <strong>R</strong>, the function <code>rarecurve()</code> in <strong>vegan</strong> draws rarefaction curves for each row (site) in a community table, producing a family of curves showing how species richness scales with the number of individuals sampled. These are drawn using base graphics <a href="#fig-rarefy" class="quarto-xref">Figure&nbsp;11</a>, but can be easily reproduced with <strong>ggplot2</strong> for more flexible visualisation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="co"># Example provided in ?vegan::rarefy</span></span>
<span id="cb23-2"><a href="#cb23-2"></a><span class="co"># observed number of species per row (site)</span></span>
<span id="cb23-3"><a href="#cb23-3"></a>S <span class="ot">&lt;-</span> <span class="fu">specnumber</span>(BCI) </span>
<span id="cb23-4"><a href="#cb23-4"></a></span>
<span id="cb23-5"><a href="#cb23-5"></a><span class="co"># calculate total no. individuals sampled per row, and find the minimum</span></span>
<span id="cb23-6"><a href="#cb23-6"></a>(raremax <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">rowSums</span>(BCI)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 340</code></pre>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a>Srare <span class="ot">&lt;-</span> <span class="fu">rarefy</span>(BCI, raremax, <span class="at">se =</span> <span class="cn">FALSE</span>)</span>
<span id="cb25-2"><a href="#cb25-2"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb25-3"><a href="#cb25-3"></a><span class="fu">plot</span>(S, Srare, <span class="at">col =</span> <span class="st">"indianred3"</span>,</span>
<span id="cb25-4"><a href="#cb25-4"></a>     <span class="at">xlab =</span> <span class="st">"Sample size</span><span class="sc">\n</span><span class="st">(observed no. of individuals)"</span>, <span class="at">ylab =</span> <span class="st">"No. species found"</span>)</span>
<span id="cb25-5"><a href="#cb25-5"></a><span class="fu">rarecurve</span>(BCI, <span class="at">step =</span> <span class="dv">20</span>, <span class="at">sample =</span> raremax, <span class="at">col =</span> <span class="st">"indianred3"</span>, <span class="at">cex =</span> <span class="fl">0.6</span>,</span>
<span id="cb25-6"><a href="#cb25-6"></a>          <span class="at">xlab =</span> <span class="st">"Sample size</span><span class="sc">\n</span><span class="st">(observed no. of individuals)"</span>, <span class="at">ylab =</span> <span class="st">"No. species found"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-rarefy" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-rarefy-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-04-biodiversity_files/figure-html/fig-rarefy-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-11" title="Figure&nbsp;11: Rarefaction curves for the BCI data."><img src="Lab-04-biodiversity_files/figure-html/fig-rarefy-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-rarefy-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;11: Rarefaction curves for the BCI data.
</figcaption></figure>
</div>
</div>
</div>
<p>We can also use the <a href="https://github.com/JohnsonHsieh/iNEXT"><strong>iNEXT</strong></a> package for rarefaction curves. From the package’s <a href="https://cran.r-project.org/web/packages/iNEXT/vignettes/Introduction.html">Introduction Vignette</a>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a><span class="fu">library</span>(iNEXT)</span>
<span id="cb26-2"><a href="#cb26-2"></a></span>
<span id="cb26-3"><a href="#cb26-3"></a><span class="co"># transpose the BCI data: </span></span>
<span id="cb26-4"><a href="#cb26-4"></a>BCI_t <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">BCI =</span> <span class="fu">t</span>(BCI))</span>
<span id="cb26-5"><a href="#cb26-5"></a></span>
<span id="cb26-6"><a href="#cb26-6"></a>BCI_out <span class="ot">&lt;-</span> <span class="fu">iNEXT</span>(BCI_t, <span class="at">q =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>), <span class="at">datatype =</span> <span class="st">"incidence_raw"</span>)</span>
<span id="cb26-7"><a href="#cb26-7"></a><span class="fu">ggiNEXT</span>(BCI_out, <span class="at">type =</span> <span class="dv">1</span>, <span class="at">color.var =</span> <span class="st">"Order.q"</span>)</span>
<span id="cb26-8"><a href="#cb26-8"></a><span class="do">## A warning is produced because the function expects incidence data</span></span>
<span id="cb26-9"><a href="#cb26-9"></a><span class="do">## (presence-absence), but I'm feeding it abundance (count) data.</span></span>
<span id="cb26-10"><a href="#cb26-10"></a><span class="do">## Nothing serious, as the function converts the abundance data to</span></span>
<span id="cb26-11"><a href="#cb26-11"></a><span class="do">## incidences.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-iNEXT" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-iNEXT-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-04-biodiversity_files/figure-html/fig-iNEXT-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-12" title="Figure&nbsp;12: Demonstration of iNEXT capabilities."><img src="Lab-04-biodiversity_files/figure-html/fig-iNEXT-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-iNEXT-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12: Demonstration of iNEXT capabilities.
</figcaption></figure>
</div>
</div>
</div>
<p>iNEXT focuses on three measures of Hill numbers of order q: species richness (q = 0), Shannon diversity (q = 1, the exponential of Shannon entropy) and Simpson diversity (q = 2, the inverse of Simpson concentration). For each diversity measure, iNEXT uses the observed sample of abundance or incidence data (called the “reference sample”) to compute diversity estimates and the associated 95% confidence intervals for the following two types of rarefaction and extrapolation (R/E):</p>
<ol type="1">
<li><p>Sample‐size‐based R/E sampling curves: iNEXT computes diversity estimates for rarefied and extrapolated samples up to an appropriate size. This type of sampling curve plots the diversity estimates with respect to sample size.</p></li>
<li><p>Coverage‐based R/E sampling curves: iNEXT computes diversity estimates for rarefied and extrapolated samples with sample completeness (as measured by sample coverage) up to an appropriate coverage. This type of sampling curve plots the diversity estimates with respect to sample coverage.</p></li>
</ol>
<p>iNEXT also plots the above two types of sampling curves and a sample completeness curve. The sample completeness curve provides a bridge between these two types of curves.</p>
<p>For information about Hill numbers see David Zelený’s <a href="https://www.davidzeleny.net/anadat-r/doku.php/en:diversity_analysis">Analysis of community data in R</a> and Jari Oksanen’s coverage of <a href="https://cran.r-project.org/web/packages/vegan/vignettes/diversity-vegan.pdf">diversity measures in <strong>vegan</strong></a>.</p>
<p>There are four datasets distributed with iNEXT and numerous examples are provided in the <a href="https://cran.r-project.org/web/packages/iNEXT/vignettes/Introduction.html">Introduction Vignette</a>. iNEXT has an ‘odd’ data format that might seem foreign to <strong>vegan</strong> users. To use iNEXT with dataset suitable for analysis in vegan, we first need to convert BCI data to a species × site matrix (<a href="#fig-iNEXT" class="quarto-xref">Figure&nbsp;12</a>):</p>
<section id="recipe-for-individual-based-rarefaction" class="level3" data-number="5.1"><h3 data-number="5.1" class="anchored" data-anchor-id="recipe-for-individual-based-rarefaction">
<span class="header-section-number">5.1</span> “Recipe” For Individual-Based Rarefaction</h3>
<p>This method estimates the expected number of species for a given number of individuals sampled. The R function <code>rarecurve()</code> in the <strong>vegan</strong> package automates this process for each site, but here is the recipe for generating one such curve for a single site (one row from your table):</p>
<ol type="1">
<li><p><strong>Create the Individual Pool</strong>: From your chosen site’s row, create a conceptual “pool” of all individuals. For example, if the site has 3 individuals of Species A, 0 of Species B, and 5 of Species C, your pool is a list containing <code>(A, A, A, C, C, C, C, C)</code>. The total number of individuals in this pool is <span class="math inline">\(N\)</span>.</p></li>
<li><p><strong>Set the Subsample Size (<span class="math inline">\(n\)</span>)</strong>: Choose a subsample size <span class="math inline">\(n\)</span>, where <span class="math inline">\(n\)</span> is an integer from 1 to <span class="math inline">\(N\)</span>. You will repeat the following steps for many different values of n to build the curve’s <em>x</em>-axis.</p></li>
<li><p><strong>Randomly Subsample Individuals</strong>: From your pool of N individuals, randomly draw <span class="math inline">\(n\)</span> individuals without replacement.</p></li>
<li><p><strong>Count Unique Species</strong>: Count the number of unique species in your random subsample. For instance, if you drew <code>(A, C, C)</code>, your species count for this draw is 2.</p></li>
<li><p><strong>Repeat and Average</strong>: A single draw isn’t enough. Repeat steps 3 and 4 many times (<em>e.g.</em>, 100 or 1,000 times) for the same subsample size <span class="math inline">\(n\)</span>. Then, calculate the average species count across all those repetitions. This gives you the expected species richness, <span class="math inline">\(E(S_n)\)</span>, for that specific sample size.</p></li>
<li><p><strong>Plot the Curve</strong>: Repeat the entire process (steps 3-5) for a range of subsample sizes from <span class="math inline">\(n=1\)</span> up to <span class="math inline">\(N\)</span>. Plot the subsample size (<span class="math inline">\(n\)</span>) on the <em>x</em>-axis and the expected species richness (<span class="math inline">\(E(S_n)\)</span>) on the <em>y</em>-axis. The result is a smooth curve that shows how many species you can expect to find for any given number of individuals sampled from that site.</p></li>
</ol></section><section id="recipe-for-sample-based-rarefaction" class="level3" data-number="5.2"><h3 data-number="5.2" class="anchored" data-anchor-id="recipe-for-sample-based-rarefaction">
<span class="header-section-number">5.2</span> “Recipe” For Sample-Based Rarefaction</h3>
<p>This method, which is very similar to a species accumulation curve, estimates the expected number of species for a given number of samples (<em>e.g.</em>, sites, quadrats). Here’s the recipe, which uses your entire site × species table:</p>
<ol type="1">
<li><p><strong>Define the Sample Pool</strong>: Your pool is the complete set of samples (all the rows in your site × species table). Let’s say you have <span class="math inline">\(K\)</span> total samples.</p></li>
<li><p><strong>Set the Subsample Size (<span class="math inline">\(k\)</span>)</strong>: Choose a number of samples <span class="math inline">\(k\)</span> to draw, where <span class="math inline">\(k\)</span> is an integer from 1 to <span class="math inline">\(K\)</span>. This will be the value on your <em>x</em>-axis.</p></li>
<li><p><strong>Randomly Subsample Samples</strong>: From your pool of <span class="math inline">\(K\)</span> samples, randomly select <span class="math inline">\(k\)</span> of them without replacement.</p></li>
<li><p><strong>Count Unique Species</strong>: Combine the species lists from the <span class="math inline">\(k\)</span> chosen samples and count the total number of unique species found across all of them.</p></li>
<li><p><strong>Repeat and Average</strong>: Just as before, repeat steps 3 and 4 many times for the same subsample size <span class="math inline">\(k\)</span>. Calculate the average species count from all these repetitions to get the expected species richness for that level of sampling effort.</p></li>
<li><p><strong>Plot the Curve</strong>: Repeat the process for all subsample sizes from <span class="math inline">\(k=1\)</span> up to <span class="math inline">\(K\)</span>. Plot the number of samples (<span class="math inline">\(k\)</span>) on the <em>x</em>-axis and the corresponding expected species richness on the <em>y</em>-axis to create the sample-based rarefaction curve.</p></li>
</ol></section></section><section id="distance-decay-curves-and-elevation-and-other-gradients" class="level2" data-number="6"><h2 data-number="6" class="anchored" data-anchor-id="distance-decay-curves-and-elevation-and-other-gradients">
<span class="header-section-number">6</span> Distance-Decay Curves and Elevation (and Other) Gradients</h2>
<p>The principles of distance decay relationships are clearly captured in analyses of <a href="04-biodiversity.html#contemporary-definitions-β-diversity"><span class="math inline">\(\beta\)</span>-diversity</a>—see specifically <strong>turnover</strong>, <span class="math inline">\(\beta_\text{sim}\)</span>. Distance decay is the primary explanation for the spatial pattern of <span class="math inline">\(\beta\)</span>-diversity along the South African coast in <span class="citation" data-cites="smit2017seaweeds">Smit et al. (<a href="#ref-smit2017seaweeds" role="doc-biblioref">2017</a>)</span>. A deeper dive into distance decay calculation can be seen in <a href="../BCB743/deep_dive.html">Deep Dive into Gradients</a>.</p>
<p>In once sense, an elevation gradient can be seen as specific case of distance decay. The Doubs River dataset offers a nice example of data collected along an elevation gradient (which also happens to be a gradient along a river). Elevation gradients have many similarities with depth gradients (<em>e.g.</em>, down the ocean depths) and latitudinal gradients.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Two things to think deeply about
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="A">
<li><p>When we look at gradients such as elevation (and some others), please think about what the species are <em>actually</em> responding to. Are they responding to elevation directly?</p></li>
<li><p>A recipe for the analysis of distance decay relationships can be found in our earlier lectures on the topic. However, there are many ways to approach these kinds of problems, and, for the sake of doing well in tests and exams, I suggest you deeply think about them.</p></li>
</ol>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Lab 4
</div>
</div>
<div class="callout-body-container callout-body">
<p>(To be reviewed by BCB743 student but not for marks)</p>
<ol type="1">
<li>
<p>Produce the following figures for the species data indicated in [square brackets]:</p>
<ol type="a">
<li>species-abundance distribution [mite];</li>
<li>occupancy-abundance curves [mite];</li>
<li>species-area curves [seaweed]—note, do not use the <strong>BAT</strong> package’s <code>alpha.accum()</code> function as your computer might fall over;</li>
<li>rarefaction curves [mite].</li>
</ol>
<p>Answer each under its own heading. For each, also explain briefly what the purpose of the analysis is (i.e.&nbsp;what ecological insights might be provided), and describe the findings of your own analysis as well as any ecological implications that you might be able to detect.</p>
</li>
<li><p>Find the most dominant species in the Doubs River dataset.</p></li>
<li><p>Discuss how elevation, depth, or latitudinal gradients are similar in many aspects to distance decay relationships.</p></li>
</ol>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Submission Instructions
</div>
</div>
<div class="callout-body-container callout-body">
<p>The Lab 4 assignment will not be assessed.</p>
</div>
</div>


<!-- -->


</section><div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-borcard1994environmental" class="csl-entry" role="listitem">
Borcard D, Legendre P (1994) Environmental control and spatial structure in ecological communities: An example using oribatid mites (acari, oribatei). Environmental and Ecological statistics 1:37–61.
</div>
<div id="ref-borcard1992partialling" class="csl-entry" role="listitem">
Borcard D, Legendre P, Drapeau P (1992) Partialling out the spatial component of ecological variation. Ecology 73:1045–1055.
</div>
<div id="ref-borcard2011numerical" class="csl-entry" role="listitem">
Borcard D, Gillet F, Legendre P, others (2011) Numerical ecology with <span>R</span>. Springer
</div>
<div id="ref-cardoso2015bat" class="csl-entry" role="listitem">
Cardoso P, Rigal F, Carvalho JC (2015) BAT–biodiversity assessment tools, an r package for the measurement and estimation of alpha and beta taxon, phylogenetic and functional diversity. Methods in Ecology and Evolution 6:232–236.
</div>
<div id="ref-condit2002beta" class="csl-entry" role="listitem">
Condit R, Pitman N, Leigh Jr EG, Chave J, Terborgh J, Foster RB, Núnez P, Aguilar S, Valencia R, Villa G, others (2002) Beta-diversity in tropical forest trees. Science 295:666–669.
</div>
<div id="ref-fisher1943relation" class="csl-entry" role="listitem">
Fisher RA, Corbet AS, Williams CB (1943) The relation between the number of species and the number of individuals in a random sample of an animal population. <span>The Journal of Animal Ecology</span> 42–58.
</div>
<div id="ref-heck1975explicit" class="csl-entry" role="listitem">
Heck Jr KL, Belle G van, Simberloff D (1975) Explicit calculation of the rarefaction diversity measurement and the determination of sufficient sample size. Ecology 56:1459–1461.
</div>
<div id="ref-preston1948commonness" class="csl-entry" role="listitem">
Preston FW (1948) The commonness, and rarity, of species. Ecology 29:254–283.
</div>
<div id="ref-smit2017seaweeds" class="csl-entry" role="listitem">
Smit AJ, Bolton JJ, Anderson RJ (2017) Seaweeds in two oceans: Beta-diversity. <span>Frontiers in Marine Science</span> 4:404.
</div>
<div id="ref-verneaux1973cours" class="csl-entry" role="listitem">
Verneaux J (1973) Cours d’eau de <span>F</span>ranche-<span>C</span>omt<span>é</span> (<span>M</span>assif du <span>J</span>ura). Recherches <span class="nocase">é</span>cologiques sur le r<span>é</span>seau hydrographique du <span>D</span>oubs.
</div>
<div id="ref-whittaker1965dominance" class="csl-entry" role="listitem">
Whittaker RH (1965) <span class="nocase">Dominance and Diversity in Land Plant Communities: Numerical relations of species express the importance of competition in community function and evolution.</span> Science 147:250–260.
</div>
</div></section><section class="quarto-appendix-contents" id="quarto-reuse"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a></div></div></section><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{smit,_a._j.2021,
  author = {Smit, A. J.,},
  title = {Lab 4. {Species} {Distribution} {Patterns}},
  date = {2021-01-01},
  url = {http://tangledbank.netlify.app/BDC334/Lab-04-biodiversity.html},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-smit,_a._j.2021" class="csl-entry quarto-appendix-citeas" role="listitem">
Smit, A. J. (2021) Lab 4. Species Distribution Patterns. <a href="http://tangledbank.netlify.app/BDC334/Lab-04-biodiversity.html">http://tangledbank.netlify.app/BDC334/Lab-04-biodiversity.html.
</a>
</div></div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("http:\/\/tangledbank\.netlify\.app");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
            // default icon
            link.classList.add("external");
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="../BDC334/Lab-03-biodiversity.html" class="pagination-link" aria-label="Lab 3. Quantifying Biodiversity">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Lab 3. Quantifying Biodiversity</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../BDC334/BDC334_AI_Tutor.html" class="pagination-link" aria-label="[BCB334 AI Tutor]{.my-highlight}">
        <span class="nav-page-text"><span class="my-highlight">BCB334 AI Tutor</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb27" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb27-1"><a href="#cb27-1"></a><span class="co">---</span></span>
<span id="cb27-2"><a href="#cb27-2"></a><span class="an">date:</span><span class="co"> "2021-01-01"</span></span>
<span id="cb27-3"><a href="#cb27-3"></a><span class="an">title:</span><span class="co"> "Lab 4. Species Distribution Patterns"</span></span>
<span id="cb27-4"><a href="#cb27-4"></a><span class="an">format:</span></span>
<span id="cb27-5"><a href="#cb27-5"></a><span class="co">  html:</span></span>
<span id="cb27-6"><a href="#cb27-6"></a><span class="co">    anchor-sections: true</span></span>
<span id="cb27-7"><a href="#cb27-7"></a><span class="co">    number-sections: true</span></span>
<span id="cb27-8"><a href="#cb27-8"></a><span class="co">    page-layout: article</span></span>
<span id="cb27-9"><a href="#cb27-9"></a><span class="co">  typst: default</span></span>
<span id="cb27-10"><a href="#cb27-10"></a><span class="co">---</span></span>
<span id="cb27-11"><a href="#cb27-11"></a></span>
<span id="cb27-12"><a href="#cb27-12"></a>::: callout-note</span>
<span id="cb27-13"><a href="#cb27-13"></a><span class="fu">## BCB743</span></span>
<span id="cb27-14"><a href="#cb27-14"></a></span>
<span id="cb27-15"><a href="#cb27-15"></a>**This material must be reviewed by BCB743 students in Week 1 of Quantitative Ecology.**</span>
<span id="cb27-16"><a href="#cb27-16"></a>:::</span>
<span id="cb27-17"><a href="#cb27-17"></a></span>
<span id="cb27-18"><a href="#cb27-18"></a>::: callout-tip</span>
<span id="cb27-19"><a href="#cb27-19"></a><span class="fu">## This Lab Accompanies the Following Lecture</span></span>
<span id="cb27-20"><a href="#cb27-20"></a></span>
<span id="cb27-21"><a href="#cb27-21"></a><span class="ss">- </span><span class="co">[</span><span class="ot">Lecture 6: Unified Ecology</span><span class="co">](Lec-06-unified-ecology.qmd)</span></span>
<span id="cb27-22"><a href="#cb27-22"></a>:::</span>
<span id="cb27-23"><a href="#cb27-23"></a></span>
<span id="cb27-24"><a href="#cb27-24"></a>::: callout-tip</span>
<span id="cb27-25"><a href="#cb27-25"></a><span class="fu">## Data For This Lab</span></span>
<span id="cb27-26"><a href="#cb27-26"></a><span class="ss">- </span>The Barro Colorado Island Tree Counts data <span class="co">[</span><span class="ot">@condit2002beta</span><span class="co">]</span> -- load **vegan** and load the data with <span class="in">`data(BCI)`</span></span>
<span id="cb27-27"><a href="#cb27-27"></a><span class="ss">- </span>The Oribatid mite data <span class="co">[</span><span class="ot">@borcard1992partialling; @borcard1994environmental</span><span class="co">]</span> -- load **vegan** and load the data with <span class="in">`data(mite)`</span></span>
<span id="cb27-28"><a href="#cb27-28"></a><span class="ss">- </span>The seaweed species data <span class="co">[</span><span class="ot">@smit2017seaweeds</span><span class="co">]</span> -- <span class="co">[</span><span class="ot">`SeaweedSpp.csv`</span><span class="co">](../data/seaweed/SeaweedSpp.csv)</span></span>
<span id="cb27-29"><a href="#cb27-29"></a><span class="ss">- </span>The Doubs River species data <span class="co">[</span><span class="ot">@verneaux1973cours; @borcard2011numerical</span><span class="co">]</span> -- <span class="co">[</span><span class="ot">`DoubsSpe.csv`</span><span class="co">](../data/DoubsSpe.csv)</span></span>
<span id="cb27-30"><a href="#cb27-30"></a>:::</span>
<span id="cb27-31"><a href="#cb27-31"></a></span>
<span id="cb27-32"><a href="#cb27-32"></a>::: callout-tip</span>
<span id="cb27-33"><a href="#cb27-33"></a><span class="fu">## Reading Required For This Lab</span></span>
<span id="cb27-34"><a href="#cb27-34"></a></span>
<span id="cb27-35"><a href="#cb27-35"></a><span class="ss">- </span><span class="co">[</span><span class="ot">Matthews and Whittaker (2015)</span><span class="co">](../docs/Matthews_and_Whittaker_2015.pdf)</span></span>
<span id="cb27-36"><a href="#cb27-36"></a><span class="ss">- </span><span class="co">[</span><span class="ot">Shade et al. (2018)</span><span class="co">](../docs/Shade_et_al_2018.pdf)</span></span>
<span id="cb27-37"><a href="#cb27-37"></a>:::</span>
<span id="cb27-38"><a href="#cb27-38"></a></span>
<span id="cb27-39"><a href="#cb27-39"></a></span>
<span id="cb27-40"><a href="#cb27-40"></a>In this Lab, we will calculate the various species distribution patterns included in the paper by <span class="co">[</span><span class="ot">Shade et al. (2018)</span><span class="co">](../docs/Shade_et_al_2018.pdf)</span>.</span>
<span id="cb27-41"><a href="#cb27-41"></a></span>
<span id="cb27-42"><a href="#cb27-42"></a><span class="fu">## The Data</span></span>
<span id="cb27-43"><a href="#cb27-43"></a></span>
<span id="cb27-44"><a href="#cb27-44"></a>We will calculate each for the <span class="co">[</span><span class="ot">Barro Colorado Island Tree Counts data</span><span class="co">](https://repository.si.edu/handle/10088/20925)</span> that come with **vegan**. See <span class="in">`?vegan::BCI`</span> for a description of the data contained with the package, as well as a selection of publications relevant to the data and analyses. The primary publication of interest is @condit2002beta.</span>
<span id="cb27-45"><a href="#cb27-45"></a></span>
<span id="cb27-48"><a href="#cb27-48"></a><span class="in">```{r}</span></span>
<span id="cb27-49"><a href="#cb27-49"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb27-50"><a href="#cb27-50"></a><span class="fu">library</span>(vegan)</span>
<span id="cb27-51"><a href="#cb27-51"></a><span class="in">```</span></span>
<span id="cb27-52"><a href="#cb27-52"></a></span>
<span id="cb27-55"><a href="#cb27-55"></a><span class="in">```{r}</span></span>
<span id="cb27-56"><a href="#cb27-56"></a><span class="co">#library(vegan) # already loaded</span></span>
<span id="cb27-57"><a href="#cb27-57"></a><span class="co">#library(tidyverse) # already loaded</span></span>
<span id="cb27-58"><a href="#cb27-58"></a><span class="fu">data</span>(BCI) <span class="co"># data contained within vegan</span></span>
<span id="cb27-59"><a href="#cb27-59"></a></span>
<span id="cb27-60"><a href="#cb27-60"></a><span class="co"># make a head-tail function</span></span>
<span id="cb27-61"><a href="#cb27-61"></a>ht <span class="ot">&lt;-</span> <span class="cf">function</span>(d) <span class="fu">rbind</span>(<span class="fu">head</span>(d, <span class="dv">7</span>), <span class="fu">tail</span>(d, <span class="dv">7</span>))</span>
<span id="cb27-62"><a href="#cb27-62"></a></span>
<span id="cb27-63"><a href="#cb27-63"></a><span class="co"># Lets look at a portion of the data:</span></span>
<span id="cb27-64"><a href="#cb27-64"></a><span class="fu">ht</span>(BCI)[<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>]</span>
<span id="cb27-65"><a href="#cb27-65"></a><span class="in">```</span></span>
<span id="cb27-66"><a href="#cb27-66"></a></span>
<span id="cb27-67"><a href="#cb27-67"></a><span class="fu">## Species-Abundance Distribution {#species-abundance-distribution}</span></span>
<span id="cb27-68"><a href="#cb27-68"></a></span>
<span id="cb27-69"><a href="#cb27-69"></a>The species abundance distribution (SAD) expresses one of ecology’s most persistent regularities, *viz.*, in almost any community, a few species dominate in abundance, while the majority are comparatively rare. This skewed pattern is often summarised as “few common, many rare” and has been recognised since the earliest quantitative analyses of community structure. It has become a central object of ecological theory. Graphically, SADs plot species abundance against either species identity or rank to provide a means of comparing empirical assemblages with theoretical expectations of community organisation.</span>
<span id="cb27-70"><a href="#cb27-70"></a></span>
<span id="cb27-71"><a href="#cb27-71"></a>The first formulation of this pattern was developed by @fisher1943relation, who observed that species frequencies often conform to what he termed a **logarithmic series distribution**. In this representation, *the expected number of species with a given number of individuals declines rapidly as abundance increases*, showing communities where most species occur at low frequencies and only a small fraction reach very high densities. The curve of Fisher’s log-series distribution thus shows the expected number of species $f$ with $n$ observed individuals, and the interpretation of this curve is conceptually similar across all SAD models—only the underlying mathematics and rationale differ. Fisher’s model remains widely used as a conceptual benchmark and as a statistical model. In **R**, the **vegan** package implements this procedure through the <span class="in">`fisherfit()`</span> function, which estimates the log-series parameter and compares it to observed community data (@fig-fishers-log).</span>
<span id="cb27-72"><a href="#cb27-72"></a></span>
<span id="cb27-75"><a href="#cb27-75"></a><span class="in">```{r}</span></span>
<span id="cb27-76"><a href="#cb27-76"></a><span class="co">#| fig-cap: "Fisher's log series distribution calculated for the Barro Colorado Island Tree Counts data."</span></span>
<span id="cb27-77"><a href="#cb27-77"></a><span class="co">#| label: fig-fishers-log</span></span>
<span id="cb27-78"><a href="#cb27-78"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb27-79"><a href="#cb27-79"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb27-80"><a href="#cb27-80"></a></span>
<span id="cb27-81"><a href="#cb27-81"></a><span class="co"># take one random sample of a row (site):</span></span>
<span id="cb27-82"><a href="#cb27-82"></a><span class="co"># for this website's purpose, this function ensure the same random</span></span>
<span id="cb27-83"><a href="#cb27-83"></a><span class="co"># sample is drawn each time the web page is recreated</span></span>
<span id="cb27-84"><a href="#cb27-84"></a><span class="fu">set.seed</span>(<span class="dv">13</span>) </span>
<span id="cb27-85"><a href="#cb27-85"></a>k <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">nrow</span>(BCI), <span class="dv">1</span>)</span>
<span id="cb27-86"><a href="#cb27-86"></a>fish <span class="ot">&lt;-</span> <span class="fu">fisherfit</span>(BCI[k,])</span>
<span id="cb27-87"><a href="#cb27-87"></a>fish</span>
<span id="cb27-88"><a href="#cb27-88"></a><span class="fu">plot</span>(fish)</span>
<span id="cb27-89"><a href="#cb27-89"></a><span class="in">```</span></span>
<span id="cb27-90"><a href="#cb27-90"></a></span>
<span id="cb27-91"><a href="#cb27-91"></a>Building on this foundation, @preston1948commonness introduced an alternative formulation: the **log-normal distribution** of species abundances. Instead of ranking species, he grouped them into “octaves”, *i.e.*, abundance classes that double in size (1, 2, 4, 8, 16, 32, …), and showed that, *in sufficiently well-sampled communities*, the frequency of species within these classes approximates a bell-shaped Gaussian curve when plotted on a logarithmic scale. Preston also emphasised the effect of incomplete sampling: rare species falling below a “veil line” remain undetected, giving the distribution its characteristic truncated form. This method captures the idea that rarity and commonness are not absolute properties but functions of sampling intensity. In **vegan**, Preston’s method is implemented with mathematical modifications to handle ties more effectively via the <span class="in">`prestondistr()`</span> function (@fig-preston).</span>
<span id="cb27-92"><a href="#cb27-92"></a></span>
<span id="cb27-95"><a href="#cb27-95"></a><span class="in">```{r}</span></span>
<span id="cb27-96"><a href="#cb27-96"></a><span class="co">#| fig-cap: "Preston's log-normal distribution demonstrated for the BCI data."</span></span>
<span id="cb27-97"><a href="#cb27-97"></a><span class="co">#| label: fig-preston</span></span>
<span id="cb27-98"><a href="#cb27-98"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb27-99"><a href="#cb27-99"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb27-100"><a href="#cb27-100"></a></span>
<span id="cb27-101"><a href="#cb27-101"></a>pres <span class="ot">&lt;-</span> <span class="fu">prestondistr</span>(BCI[k,])</span>
<span id="cb27-102"><a href="#cb27-102"></a>pres</span>
<span id="cb27-103"><a href="#cb27-103"></a><span class="fu">plot</span>(pres)</span>
<span id="cb27-104"><a href="#cb27-104"></a><span class="in">```</span></span>
<span id="cb27-105"><a href="#cb27-105"></a></span>
<span id="cb27-106"><a href="#cb27-106"></a>A further step was taken by @whittaker1965dominance, who reformulated SADs as **rank-abundance distributions** (RADs), also known as dominance–diversity curves or Whittaker plots. Here, species are ordered from most to least abundant along the $x$-axis, and their relative abundances (often log-transformed) are plotted on the $y$-axis. The slope and curvature of the resulting profile reveal both evenness and the extent of dominance: steep curves indicate communities dominated by a few species, whereas long, shallow tails point to greater species equity. Unlike Fisher’s or Preston’s frequency-based models, Whittaker’s approach highlights the shape of the assemblage as a whole and facilitates comparison across communities with different richness levels. In **vegan**, this is achieved via the <span class="in">`radfit()`</span> function, which overlays multiple fitted models—broken-stick, preemption, log-normal, Zipf, and Zipf–Mandelbrot—onto the observed ranked data (@fig-rad).</span>
<span id="cb27-107"><a href="#cb27-107"></a></span>
<span id="cb27-110"><a href="#cb27-110"></a><span class="in">```{r}</span></span>
<span id="cb27-111"><a href="#cb27-111"></a><span class="co">#| fig-cap: "Whittaker's rank abundance distribution curves demonstrated for the BCI data."</span></span>
<span id="cb27-112"><a href="#cb27-112"></a><span class="co">#| label: fig-rad</span></span>
<span id="cb27-113"><a href="#cb27-113"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb27-114"><a href="#cb27-114"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb27-115"><a href="#cb27-115"></a></span>
<span id="cb27-116"><a href="#cb27-116"></a>rad <span class="ot">&lt;-</span> <span class="fu">radfit</span>(BCI[k,])</span>
<span id="cb27-117"><a href="#cb27-117"></a>rad</span>
<span id="cb27-118"><a href="#cb27-118"></a><span class="fu">plot</span>(rad)</span>
<span id="cb27-119"><a href="#cb27-119"></a><span class="in">```</span></span>
<span id="cb27-120"><a href="#cb27-120"></a></span>
<span id="cb27-121"><a href="#cb27-121"></a>We can also fit the rank abundance distribution curves to several sites at once (previously we have done so on only one site) (@fig-rad2):</span>
<span id="cb27-122"><a href="#cb27-122"></a></span>
<span id="cb27-125"><a href="#cb27-125"></a><span class="in">```{r}</span></span>
<span id="cb27-126"><a href="#cb27-126"></a><span class="co">#| fig-cap: "Rank abundance distribution curves fitted to several sites."</span></span>
<span id="cb27-127"><a href="#cb27-127"></a><span class="co">#| label: fig-rad2</span></span>
<span id="cb27-128"><a href="#cb27-128"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb27-129"><a href="#cb27-129"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb27-130"><a href="#cb27-130"></a></span>
<span id="cb27-131"><a href="#cb27-131"></a>m <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">nrow</span>(BCI), <span class="dv">6</span>)</span>
<span id="cb27-132"><a href="#cb27-132"></a>rad2 <span class="ot">&lt;-</span> <span class="fu">radfit</span>(BCI[m, ])</span>
<span id="cb27-133"><a href="#cb27-133"></a>rad2</span>
<span id="cb27-134"><a href="#cb27-134"></a><span class="fu">plot</span>(rad2)</span>
<span id="cb27-135"><a href="#cb27-135"></a><span class="in">```</span></span>
<span id="cb27-136"><a href="#cb27-136"></a></span>
<span id="cb27-137"><a href="#cb27-137"></a>Above, we see that the model selected for capturing the shape of the SAD is the Mandelbrot, and it is plotted individually for each of the randomly selected sites. Model selection works through Akaike’s or Schwartz’s Bayesian information criteria (AIC or BIC; AIC is the default---select the model with the lowest AIC).</span>
<span id="cb27-138"><a href="#cb27-138"></a></span>
<span id="cb27-139"><a href="#cb27-139"></a><span class="co">[</span><span class="ot">**BiodiversityR**</span><span class="co">](https://github.com/cran/BiodiversityR)</span> (and <span class="co">[</span><span class="ot">here</span><span class="co">](http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf)</span> and <span class="co">[</span><span class="ot">here</span><span class="co">](https://rpubs.com/Roeland-KINDT)</span>) also offers options for rank abundance distribution curves; see <span class="in">`rankabundance()`</span> (@fig-rac):</span>
<span id="cb27-140"><a href="#cb27-140"></a></span>
<span id="cb27-143"><a href="#cb27-143"></a><span class="in">```{r}</span></span>
<span id="cb27-144"><a href="#cb27-144"></a><span class="co">#| fig-cap: "Rank-abundance curves for the BCI data."</span></span>
<span id="cb27-145"><a href="#cb27-145"></a><span class="co">#| label: fig-rac</span></span>
<span id="cb27-146"><a href="#cb27-146"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb27-147"><a href="#cb27-147"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb27-148"><a href="#cb27-148"></a></span>
<span id="cb27-149"><a href="#cb27-149"></a><span class="fu">library</span>(BiodiversityR)</span>
<span id="cb27-150"><a href="#cb27-150"></a>rankabund <span class="ot">&lt;-</span> <span class="fu">rankabundance</span>(BCI)</span>
<span id="cb27-151"><a href="#cb27-151"></a><span class="fu">rankabunplot</span>(rankabund, <span class="at">cex =</span> <span class="fl">0.8</span>, <span class="at">pch =</span> <span class="fl">0.8</span>, <span class="at">col =</span> <span class="st">"indianred4"</span>)</span>
<span id="cb27-152"><a href="#cb27-152"></a><span class="in">```</span></span>
<span id="cb27-153"><a href="#cb27-153"></a></span>
<span id="cb27-154"><a href="#cb27-154"></a>Refer to the help files for the respective functions to see their differences.</span>
<span id="cb27-155"><a href="#cb27-155"></a></span>
<span id="cb27-156"><a href="#cb27-156"></a><span class="fu">### Calculating a SAD From a Site × Species Table</span></span>
<span id="cb27-157"><a href="#cb27-157"></a></span>
<span id="cb27-158"><a href="#cb27-158"></a>Here’s how to manually derive an SAD from a typical community dataset. Let's assume you have a site × species table, where rows are sampling sites and columns are species, with the cells containing counts (abundances).</span>
<span id="cb27-159"><a href="#cb27-159"></a></span>
<span id="cb27-160"><a href="#cb27-160"></a><span class="ss">1. </span>**Get Total Species Abundances**: Your first step is to get the total abundance for each species across all your sites. You do this by **summing each column**. This collapses your site × species matrix into a single list of numberswhich we call an "**abundance vector**;" here each number is the total count for one species.</span>
<span id="cb27-161"><a href="#cb27-161"></a></span>
<span id="cb27-162"><a href="#cb27-162"></a><span class="ss">2. </span>**Create a Frequency Table**: Now, you'll process this abundance vector. Count *how many* species have an abundance of 1, how many have an abundance of 2, and so on, for all abundance values present in your data. The result is a frequency table.</span>
<span id="cb27-163"><a href="#cb27-163"></a></span>
<span id="cb27-164"><a href="#cb27-164"></a>For example, if your abundance vector is <span class="in">`(1, 1, 5, 2, 8, 1, 2, 5)`</span>, your frequency table would be:</span>
<span id="cb27-165"><a href="#cb27-165"></a></span>
<span id="cb27-166"><a href="#cb27-166"></a><span class="ss">  * </span>Abundance Class 1: 3 species</span>
<span id="cb27-167"><a href="#cb27-167"></a><span class="ss">  * </span>Abundance Class 2: 2 species</span>
<span id="cb27-168"><a href="#cb27-168"></a><span class="ss">  * </span>Abundance Class 5: 2 species</span>
<span id="cb27-169"><a href="#cb27-169"></a><span class="ss">  * </span>Abundance Class 8: 1 species</span>
<span id="cb27-170"><a href="#cb27-170"></a></span>
<span id="cb27-171"><a href="#cb27-171"></a><span class="ss">3. </span>**Plot the SAD**: This frequency table is your SAD. To visualise it as a frequency distribution (like for Fisher's or Preston's models), you plot the **Abundance Class on the *x*-axis** and the **Number of Species (frequency) on the *y*-axis**. You will typically see that classic hollow curve.</span>
<span id="cb27-172"><a href="#cb27-172"></a></span>
<span id="cb27-173"><a href="#cb27-173"></a><span class="ss">4. </span>**Plot the RAD (Whittaker Plot)**: To create a Rank Abundance Distribution from the same data, you take your original abundance vector `(1, 1, 5, 2, 8, 1, 2, 5)`, sort it in descending order `(8, 5, 5, 2, 2, 1, 1, 1)`, and then plot it. The ***x*-axis is the rank** `(1, 2, 3, 4, 5, 6, 7, 8)`, and the ***y*-axis is the corresponding abundance value** <span class="in">`(8, 5, 5, 2, 2, 1, 1, 1)`</span>.</span>
<span id="cb27-174"><a href="#cb27-174"></a></span>
<span id="cb27-175"><a href="#cb27-175"></a><span class="fu">## Occupancy-Abundance Relationship</span></span>
<span id="cb27-176"><a href="#cb27-176"></a></span>
<span id="cb27-177"><a href="#cb27-177"></a>**Occupancy** refers to the number (or proportion) of sites in which a species is recorded as present. When linked with measures of abundance, it provides a way to examine the spatial distribution of species across a landscape. **Occupancy–abundance relationships** (OARs) reveal that species that are locally abundant at a given site tend also to be widespread across many sites in the region; conversely, species of low local abundance are more often restricted in their wider distribution. This empirical pattern has been so consistently observed that it has been described as verging on an ecological “law;” it tell us about the underlying processes of niche breadth, dispersal capacity, and environmental tolerance. OARs are therefore often used to infer degrees of niche specialisation, with generalists expected to show both high occupancy and high abundance, and specialists clustering toward low values of both measures (@fig-occupancy).</span>
<span id="cb27-178"><a href="#cb27-178"></a></span>
<span id="cb27-181"><a href="#cb27-181"></a><span class="in">```{r}</span></span>
<span id="cb27-182"><a href="#cb27-182"></a><span class="co">#| fig-cap: "Occupancy-abundance relationships seen in the BCI data."</span></span>
<span id="cb27-183"><a href="#cb27-183"></a><span class="co">#| label: fig-occupancy</span></span>
<span id="cb27-184"><a href="#cb27-184"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb27-185"><a href="#cb27-185"></a><span class="co">#| fig-height: 3</span></span>
<span id="cb27-186"><a href="#cb27-186"></a></span>
<span id="cb27-187"><a href="#cb27-187"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb27-188"><a href="#cb27-188"></a></span>
<span id="cb27-189"><a href="#cb27-189"></a><span class="co"># A function for counts:</span></span>
<span id="cb27-190"><a href="#cb27-190"></a><span class="co"># count number of non-zero elements per column</span></span>
<span id="cb27-191"><a href="#cb27-191"></a>count_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb27-192"><a href="#cb27-192"></a>  <span class="fu">length</span>(x[x <span class="sc">&gt;</span> <span class="dv">0</span>])</span>
<span id="cb27-193"><a href="#cb27-193"></a>}</span>
<span id="cb27-194"><a href="#cb27-194"></a></span>
<span id="cb27-195"><a href="#cb27-195"></a>BCI_OA <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">occ =</span> <span class="fu">apply</span>(BCI, <span class="at">MARGIN =</span> <span class="dv">2</span>, count_fun),</span>
<span id="cb27-196"><a href="#cb27-196"></a>                     <span class="at">ab =</span> <span class="fu">apply</span>(BCI, <span class="at">MARGIN =</span> <span class="dv">2</span>, mean))</span>
<span id="cb27-197"><a href="#cb27-197"></a></span>
<span id="cb27-198"><a href="#cb27-198"></a><span class="fu">ggplot</span>(BCI_OA, <span class="fu">aes</span>(<span class="at">x =</span> ab, <span class="at">y =</span> occ<span class="sc">/</span><span class="fu">max</span>(occ))) <span class="sc">+</span></span>
<span id="cb27-199"><a href="#cb27-199"></a>  <span class="fu">geom_point</span>(<span class="at">colour =</span> <span class="st">"indianred3"</span>) <span class="sc">+</span></span>
<span id="cb27-200"><a href="#cb27-200"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb27-201"><a href="#cb27-201"></a>  <span class="co"># scale_y_log10() +</span></span>
<span id="cb27-202"><a href="#cb27-202"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Barro Colorado Island Tree Counts"</span>,</span>
<span id="cb27-203"><a href="#cb27-203"></a>     <span class="at">x =</span> <span class="st">"Log (abundance)"</span>, <span class="at">y =</span> <span class="st">"Occupancy"</span>) <span class="sc">+</span></span>
<span id="cb27-204"><a href="#cb27-204"></a>  <span class="fu">theme_linedraw</span>()</span>
<span id="cb27-205"><a href="#cb27-205"></a><span class="in">```</span></span>
<span id="cb27-206"><a href="#cb27-206"></a></span>
<span id="cb27-207"><a href="#cb27-207"></a><span class="fu">### Calculate the OAR from a Site × Species Table</span></span>
<span id="cb27-208"><a href="#cb27-208"></a></span>
<span id="cb27-209"><a href="#cb27-209"></a>Calculating and plotting this relationship is easy. Let's start with your site × species table with cells containing abundance counts.</span>
<span id="cb27-210"><a href="#cb27-210"></a></span>
<span id="cb27-211"><a href="#cb27-211"></a><span class="ss">1. </span>**Calculate Occupancy for Each Species**: Occupancy is simply *the number or proportion of sites where a species is present*. To calculate this, you look down each species' column and count how many of the cells have a value greater than zero. You'll end up with a vector of numbers (integers; counts), one number for each species, representing its occupancy. For example, if a species is found at 5 out of 20 surveyed sites, its occupancy count is 5 (or its proportional occupancy is $5/20=0.25$).</span>
<span id="cb27-212"><a href="#cb27-212"></a></span>
<span id="cb27-213"><a href="#cb27-213"></a><span class="ss">2. </span>**Calculate Mean Local Abundance for Each Species**: This step is important. You need the **average abundance** of a species *only in the sites where it actually occurs*. Including sites where it's absent (with an abundance of zero) would artificially deflate the average and obscure the true relationship. Then, for each species, **sum all the abundance values in its column** (the total abundance) and divide this by the occupancy count you calculated in Step 1. This gives you the **mean local abundance** for that species.</span>
<span id="cb27-214"><a href="#cb27-214"></a></span>
<span id="cb27-215"><a href="#cb27-215"></a><span class="ss">3. </span>**Plot the Relationship**: You now have two corresponding vectors of values for all your species: one for occupancy and one for mean local abundance. To visualise the OAR, you create a scatter plot:</span>
<span id="cb27-216"><a href="#cb27-216"></a></span>
<span id="cb27-217"><a href="#cb27-217"></a><span class="ss">  * </span>*x*-axis: Occupancy (the number or proportion of sites occupied).</span>
<span id="cb27-218"><a href="#cb27-218"></a><span class="ss">  * </span>*y*-axis: Mean Local Abundance. Because abundance values often span several orders of magnitude, this axis is almost always log-transformed (*e.g.*, using $log_{10}$ or $log_{e}$) to better bring out the pattern and meet the assumptions of statistical tests.</span>
<span id="cb27-219"><a href="#cb27-219"></a></span>
<span id="cb27-220"><a href="#cb27-220"></a><span class="fu">## Species-Area (Accumulation)</span></span>
<span id="cb27-221"><a href="#cb27-221"></a></span>
<span id="cb27-222"><a href="#cb27-222"></a>Species accumulation curves—often presented as sample-based species–area relationships (SAR)—address the question: given what has already been sampled, how many species are yet unseen, and how does observed richness grow as sampling effort expands? Within a single ecosystem type, one ordinarily expects richness to rise with the number of distinct sampling units (plots, quadrats, trawls), then bend toward a plateau as further additions yield few or no new species. The curve is thus a record of discovery under a specified sampling design. Conceptually, the $x$-axis represents cumulative sampling effort (number of sites, or cumulative area if sites are areally commensurate), while the $y$-axis records the expected number of species discovered at that effort. As <span class="co">[</span><span class="ot">Roeland Kindt</span><span class="co">](http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf)</span> explains (see p. 41 in the cited note), the curve is generated by “adding” sites and tallying the average richness at each step—an averaging that already hints at a statistical question: in what order are sites added, and how is that order handled?</span>
<span id="cb27-223"><a href="#cb27-223"></a></span>
<span id="cb27-224"><a href="#cb27-224"></a>The <span class="in">`specaccum()`</span> function has many different ways of adding the new sites to the curve, but the default 'exact' seems to be a sensible choice. **BiodiversityR** has the `accumresult()` function that does nearly the same. Let's demonstrate using **vegan**'s function (@fig-accum, @fig-accum2, and @fig-accum3):</span>
<span id="cb27-225"><a href="#cb27-225"></a></span>
<span id="cb27-228"><a href="#cb27-228"></a><span class="in">```{r}</span></span>
<span id="cb27-229"><a href="#cb27-229"></a><span class="co">#| fig-cap: "Species-area accumulation curves seen in the BCI data."</span></span>
<span id="cb27-230"><a href="#cb27-230"></a><span class="co">#| label: fig-accum</span></span>
<span id="cb27-231"><a href="#cb27-231"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb27-232"><a href="#cb27-232"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb27-233"><a href="#cb27-233"></a></span>
<span id="cb27-234"><a href="#cb27-234"></a>sp1 <span class="ot">&lt;-</span> <span class="fu">specaccum</span>(BCI)</span>
<span id="cb27-235"><a href="#cb27-235"></a>sp2 <span class="ot">&lt;-</span> <span class="fu">specaccum</span>(BCI, <span class="st">"random"</span>)</span>
<span id="cb27-236"><a href="#cb27-236"></a></span>
<span id="cb27-237"><a href="#cb27-237"></a><span class="co"># par(mfrow = c(2,2), mar = c(4,2,2,1))</span></span>
<span id="cb27-238"><a href="#cb27-238"></a><span class="co"># par(mfrow = c(1,2))</span></span>
<span id="cb27-239"><a href="#cb27-239"></a><span class="fu">plot</span>(sp1, <span class="at">ci.type =</span> <span class="st">"polygon"</span>, <span class="at">col =</span> <span class="st">"indianred4"</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">ci.lty =</span> <span class="dv">0</span>,</span>
<span id="cb27-240"><a href="#cb27-240"></a>     <span class="at">ci.col =</span> <span class="st">"steelblue2"</span>, <span class="at">main =</span> <span class="st">"Default: exact"</span>,</span>
<span id="cb27-241"><a href="#cb27-241"></a>     <span class="at">ylab =</span> <span class="st">"No. of species"</span>)</span>
<span id="cb27-242"><a href="#cb27-242"></a><span class="in">```</span></span>
<span id="cb27-243"><a href="#cb27-243"></a></span>
<span id="cb27-246"><a href="#cb27-246"></a><span class="in">```{r}</span></span>
<span id="cb27-247"><a href="#cb27-247"></a><span class="co">#| fig-cap: "Fit Arrhenius models to all random accumulations"</span></span>
<span id="cb27-248"><a href="#cb27-248"></a><span class="co">#| label: fig-accum2</span></span>
<span id="cb27-249"><a href="#cb27-249"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb27-250"><a href="#cb27-250"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb27-251"><a href="#cb27-251"></a></span>
<span id="cb27-252"><a href="#cb27-252"></a>mods <span class="ot">&lt;-</span> <span class="fu">fitspecaccum</span>(sp2, <span class="st">"arrh"</span>)</span>
<span id="cb27-253"><a href="#cb27-253"></a><span class="fu">plot</span>(mods, <span class="at">col =</span> <span class="st">"indianred"</span>, <span class="at">ylab =</span> <span class="st">"No. of species"</span>)</span>
<span id="cb27-254"><a href="#cb27-254"></a><span class="fu">boxplot</span>(sp2, <span class="at">col =</span> <span class="st">"yellow"</span>, <span class="at">border =</span> <span class="st">"steelblue2"</span>, <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">cex =</span> <span class="fl">0.3</span>, <span class="at">add =</span> <span class="cn">TRUE</span>)</span>
<span id="cb27-255"><a href="#cb27-255"></a><span class="fu">sapply</span>(mods<span class="sc">$</span>models, AIC)</span>
<span id="cb27-256"><a href="#cb27-256"></a><span class="in">```</span></span>
<span id="cb27-257"><a href="#cb27-257"></a></span>
<span id="cb27-260"><a href="#cb27-260"></a><span class="in">```{r}</span></span>
<span id="cb27-261"><a href="#cb27-261"></a><span class="co">#| fig-cap: "A species accumulation curve."</span></span>
<span id="cb27-262"><a href="#cb27-262"></a><span class="co">#| label: fig-accum3</span></span>
<span id="cb27-263"><a href="#cb27-263"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb27-264"><a href="#cb27-264"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb27-265"><a href="#cb27-265"></a></span>
<span id="cb27-266"><a href="#cb27-266"></a>accum <span class="ot">&lt;-</span> <span class="fu">accumresult</span>(BCI, <span class="at">method =</span> <span class="st">"exact"</span>, <span class="at">permutations =</span> <span class="dv">100</span>)</span>
<span id="cb27-267"><a href="#cb27-267"></a><span class="fu">accumplot</span>(accum)</span>
<span id="cb27-268"><a href="#cb27-268"></a><span class="in">```</span></span>
<span id="cb27-269"><a href="#cb27-269"></a></span>
<span id="cb27-270"><a href="#cb27-270"></a>Species accumulation curves can also be calculated with the <span class="in">`alpha.accum()`</span> function of the **BAT** package (@fig-accum4). In addition, the **BAT** package can also apply various diversity and species distribution assessments to **phylogenetic** and **functional** diversity. See the examples provided by @cardoso2015bat.</span>
<span id="cb27-271"><a href="#cb27-271"></a></span>
<span id="cb27-274"><a href="#cb27-274"></a><span class="in">```{r}</span></span>
<span id="cb27-275"><a href="#cb27-275"></a><span class="co">#| fig-cap: "A species accumulation curve made with the `alpha.accum()` function of **BAT**."</span></span>
<span id="cb27-276"><a href="#cb27-276"></a><span class="co">#| label: fig-accum4</span></span>
<span id="cb27-277"><a href="#cb27-277"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb27-278"><a href="#cb27-278"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb27-279"><a href="#cb27-279"></a></span>
<span id="cb27-280"><a href="#cb27-280"></a><span class="fu">library</span>(BAT)</span>
<span id="cb27-281"><a href="#cb27-281"></a>BCI.acc <span class="ot">&lt;-</span> <span class="fu">alpha.accum</span>(BCI, <span class="at">prog =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-282"><a href="#cb27-282"></a></span>
<span id="cb27-283"><a href="#cb27-283"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb27-284"><a href="#cb27-284"></a><span class="fu">plot</span>(BCI.acc[,<span class="dv">2</span>], BCI.acc[,<span class="dv">17</span>], <span class="at">col =</span> <span class="st">"indianred"</span>,</span>
<span id="cb27-285"><a href="#cb27-285"></a>     <span class="at">xlab =</span> <span class="st">"Individuals"</span>, <span class="at">ylab =</span> <span class="st">"Chao1P"</span>)</span>
<span id="cb27-286"><a href="#cb27-286"></a><span class="fu">plot</span>(BCI.acc[,<span class="dv">2</span>], <span class="fu">slope</span>(BCI.acc)[,<span class="dv">17</span>], <span class="at">col =</span> <span class="st">"indianred"</span>,</span>
<span id="cb27-287"><a href="#cb27-287"></a>     <span class="at">xlab =</span> <span class="st">"Individuals"</span>, <span class="at">ylab =</span> <span class="st">"Slope"</span>)</span>
<span id="cb27-288"><a href="#cb27-288"></a><span class="in">```</span></span>
<span id="cb27-289"><a href="#cb27-289"></a></span>
<span id="cb27-290"><a href="#cb27-290"></a><span class="fu">### Calculating a Species Accumulation Curve From Scratch</span></span>
<span id="cb27-291"><a href="#cb27-291"></a></span>
<span id="cb27-292"><a href="#cb27-292"></a>Calculating a proper SAC from a site × species table involves more than just adding up samples in their original order. The order in which you "collect" your samples can drastically change the shape of the curve. To solve this, the process relies on repeated randomisation to create a smooth, average curve.</span>
<span id="cb27-293"><a href="#cb27-293"></a></span>
<span id="cb27-294"><a href="#cb27-294"></a><span class="ss">1. </span>**Randomise the Order of Samples**</span>
<span id="cb27-295"><a href="#cb27-295"></a></span>
<span id="cb27-296"><a href="#cb27-296"></a>The first and most important step is to randomise the order of your samples (the rows in your site × species table). This ensures the resulting curve isn't biased by happening to have a particularly species-rich or species-poor site at the beginning of the sequence.</span>
<span id="cb27-297"><a href="#cb27-297"></a></span>
<span id="cb27-298"><a href="#cb27-298"></a><span class="ss">2. </span>**Accumulate Species in Sequence**</span>
<span id="cb27-299"><a href="#cb27-299"></a></span>
<span id="cb27-300"><a href="#cb27-300"></a>Once the sample order is shuffled, you build the curve step-by-step:</span>
<span id="cb27-301"><a href="#cb27-301"></a></span>
<span id="cb27-302"><a href="#cb27-302"></a><span class="ss">* </span>For the 1st sample: The cumulative species count is simply the number of species in that first sample.</span>
<span id="cb27-303"><a href="#cb27-303"></a><span class="ss">* </span>For the 2nd sample: You pool the species from the first and second samples and count the total number of unique species found across both.</span>
<span id="cb27-304"><a href="#cb27-304"></a><span class="ss">* </span>For the 3rd sample: You pool the species from the first three samples and count the total number of unique species found across all three.</span>
<span id="cb27-305"><a href="#cb27-305"></a><span class="ss">* </span>You continue this process until all samples in your shuffled list have been added. This single pass creates one possible accumulation curve.</span>
<span id="cb27-306"><a href="#cb27-306"></a></span>
<span id="cb27-307"><a href="#cb27-307"></a><span class="ss">3. </span>**Repeat and Average**</span>
<span id="cb27-308"><a href="#cb27-308"></a></span>
<span id="cb27-309"><a href="#cb27-309"></a>A curve from a single random order is not representative. Therefore, you must repeat steps 1 and 2 many times (typically 100 to 1,000 times), creating a new random permutation of your samples each time. This gives you a large collection of possible accumulation curves.</span>
<span id="cb27-310"><a href="#cb27-310"></a></span>
<span id="cb27-311"><a href="#cb27-311"></a><span class="ss">4. </span>**Plot the Final Curve**</span>
<span id="cb27-312"><a href="#cb27-312"></a></span>
<span id="cb27-313"><a href="#cb27-313"></a>The final species accumulation curve is not a single line but rather the average of all the randomised curves you generated.</span>
<span id="cb27-314"><a href="#cb27-314"></a></span>
<span id="cb27-315"><a href="#cb27-315"></a><span class="ss">* </span>*x*-axis: Number of Samples Accumulated (*e.g.*, 1, 2, 3, ... up to your total number of sites).</span>
<span id="cb27-316"><a href="#cb27-316"></a><span class="ss">* </span>*y*-axis: The average cumulative number of species found at that level of sampling effort.</span>
<span id="cb27-317"><a href="#cb27-317"></a></span>
<span id="cb27-318"><a href="#cb27-318"></a>It's also standard practice to plot the standard deviation or confidence intervals around this mean curve. This creates a shaded region or error bars, showing the amount of variation from the different randomisations. A narrow confidence interval indicates a more robust and reliable curve.</span>
<span id="cb27-319"><a href="#cb27-319"></a></span>
<span id="cb27-320"><a href="#cb27-320"></a><span class="fu">## Rarefaction Curves</span></span>
<span id="cb27-321"><a href="#cb27-321"></a></span>
<span id="cb27-322"><a href="#cb27-322"></a>Like **species accumulation curves**, **rarefaction curves** also address the question of unseen species, but they do so in a different way. Rarefaction -- literally “scaling down” <span class="co">[</span><span class="ot">@heck1975explicit</span><span class="co">]</span> -- is a statistical technique that estimates species richness (often denoted S, or sometimes diversity indices such as Shannon diversity, $H’$, or Simpson’s diversity, $\lambda$) from community data, typically organised as a site × species table. The main goal is to ask: given a smaller sample of individuals or sites than were actually collected, how many species would we expect to detect? Rarefaction thus provides a way of standardising richness estimates across samples of unequal size and of judging whether sampling effort has been sufficient to capture the community’s complement of species.</span>
<span id="cb27-323"><a href="#cb27-323"></a></span>
<span id="cb27-324"><a href="#cb27-324"></a>At first sight, rarefaction curves resemble species accumulation curves, since both plot species richness against increasing effort. The difference lies in how “effort” is defined and how the curves are generated. **Species accumulation curves** build up by successively adding new sites or samples (rows of the community matrix) and tallying the union of species observed. By contrast, **rarefaction curves** are constructed by randomly sub-sampling the existing pool of $N$ individuals or samples. Repeated resampling at size $n$ (where $1 \leq n \leq N$) yields an expected number of species at that level of effort, and the curve plots this expectation as a function of $n$.</span>
<span id="cb27-325"><a href="#cb27-325"></a></span>
<span id="cb27-326"><a href="#cb27-326"></a>Two standard rarefaction approaches exist. In **individual-based rarefaction**, the $x$-axis represents the number of individuals sampled across all species, with richness S as the response. The principle is intuitive: when only a few individuals are drawn, they are likely to belong to a small set of species, but as more individuals are added, the chance of encountering new species increases—at a decelerating rate as common species are already represented. In **sample-based rarefaction**, the $x$-axis represents the number of samples (sites, quadrats, trawls), and species richness is again tallied as a function of sample number. In both cases, the resulting curves rise steeply at first, then level off toward a plateau, reflecting the declining probability of detecting the rarest species as sampling effort grows.</span>
<span id="cb27-327"><a href="#cb27-327"></a></span>
<span id="cb27-328"><a href="#cb27-328"></a>This resampling distinction is what sets rarefaction apart from **species accumulation curves** and **species abundance distributions (SADs)**. While SADs describe how species are distributed in terms of abundance and accumulation curves describe how observed richness grows as sampling units are added, rarefaction explicitly **re-samples the observed data at smaller effort levels** to estimate expected richness. In **R**, the function `rarecurve()` in **vegan** draws rarefaction curves for each row (site) in a community table, producing a family of curves showing how species richness scales with the number of individuals sampled. These are drawn using base graphics [@fig-rarefy], but can be easily reproduced with **ggplot2** for more flexible visualisation.</span>
<span id="cb27-329"><a href="#cb27-329"></a></span>
<span id="cb27-332"><a href="#cb27-332"></a><span class="in">```{r}</span></span>
<span id="cb27-333"><a href="#cb27-333"></a><span class="co">#| fig-cap: "Rarefaction curves for the BCI data."</span></span>
<span id="cb27-334"><a href="#cb27-334"></a><span class="co">#| label: fig-rarefy</span></span>
<span id="cb27-335"><a href="#cb27-335"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb27-336"><a href="#cb27-336"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb27-337"><a href="#cb27-337"></a></span>
<span id="cb27-338"><a href="#cb27-338"></a><span class="co"># Example provided in ?vegan::rarefy</span></span>
<span id="cb27-339"><a href="#cb27-339"></a><span class="co"># observed number of species per row (site)</span></span>
<span id="cb27-340"><a href="#cb27-340"></a>S <span class="ot">&lt;-</span> <span class="fu">specnumber</span>(BCI) </span>
<span id="cb27-341"><a href="#cb27-341"></a></span>
<span id="cb27-342"><a href="#cb27-342"></a><span class="co"># calculate total no. individuals sampled per row, and find the minimum</span></span>
<span id="cb27-343"><a href="#cb27-343"></a>(raremax <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">rowSums</span>(BCI)))</span>
<span id="cb27-344"><a href="#cb27-344"></a></span>
<span id="cb27-345"><a href="#cb27-345"></a>Srare <span class="ot">&lt;-</span> <span class="fu">rarefy</span>(BCI, raremax, <span class="at">se =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-346"><a href="#cb27-346"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb27-347"><a href="#cb27-347"></a><span class="fu">plot</span>(S, Srare, <span class="at">col =</span> <span class="st">"indianred3"</span>,</span>
<span id="cb27-348"><a href="#cb27-348"></a>     <span class="at">xlab =</span> <span class="st">"Sample size</span><span class="sc">\n</span><span class="st">(observed no. of individuals)"</span>, <span class="at">ylab =</span> <span class="st">"No. species found"</span>)</span>
<span id="cb27-349"><a href="#cb27-349"></a><span class="fu">rarecurve</span>(BCI, <span class="at">step =</span> <span class="dv">20</span>, <span class="at">sample =</span> raremax, <span class="at">col =</span> <span class="st">"indianred3"</span>, <span class="at">cex =</span> <span class="fl">0.6</span>,</span>
<span id="cb27-350"><a href="#cb27-350"></a>          <span class="at">xlab =</span> <span class="st">"Sample size</span><span class="sc">\n</span><span class="st">(observed no. of individuals)"</span>, <span class="at">ylab =</span> <span class="st">"No. species found"</span>)</span>
<span id="cb27-351"><a href="#cb27-351"></a><span class="in">```</span></span>
<span id="cb27-352"><a href="#cb27-352"></a></span>
<span id="cb27-353"><a href="#cb27-353"></a>We can also use the <span class="co">[</span><span class="ot">**iNEXT**</span><span class="co">](https://github.com/JohnsonHsieh/iNEXT)</span> package for rarefaction curves. From the package's <span class="co">[</span><span class="ot">Introduction Vignette</span><span class="co">](https://cran.r-project.org/web/packages/iNEXT/vignettes/Introduction.html)</span>:</span>
<span id="cb27-354"><a href="#cb27-354"></a></span>
<span id="cb27-357"><a href="#cb27-357"></a><span class="in">```{r}</span></span>
<span id="cb27-358"><a href="#cb27-358"></a><span class="co">#| fig-cap: "Demonstration of iNEXT capabilities."</span></span>
<span id="cb27-359"><a href="#cb27-359"></a><span class="co">#| label: fig-iNEXT</span></span>
<span id="cb27-360"><a href="#cb27-360"></a><span class="co">#| fig-width: 6</span></span>
<span id="cb27-361"><a href="#cb27-361"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb27-362"><a href="#cb27-362"></a><span class="fu">library</span>(iNEXT)</span>
<span id="cb27-363"><a href="#cb27-363"></a></span>
<span id="cb27-364"><a href="#cb27-364"></a><span class="co"># transpose the BCI data: </span></span>
<span id="cb27-365"><a href="#cb27-365"></a>BCI_t <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">BCI =</span> <span class="fu">t</span>(BCI))</span>
<span id="cb27-366"><a href="#cb27-366"></a></span>
<span id="cb27-367"><a href="#cb27-367"></a>BCI_out <span class="ot">&lt;-</span> <span class="fu">iNEXT</span>(BCI_t, <span class="at">q =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>), <span class="at">datatype =</span> <span class="st">"incidence_raw"</span>)</span>
<span id="cb27-368"><a href="#cb27-368"></a><span class="fu">ggiNEXT</span>(BCI_out, <span class="at">type =</span> <span class="dv">1</span>, <span class="at">color.var =</span> <span class="st">"Order.q"</span>)</span>
<span id="cb27-369"><a href="#cb27-369"></a><span class="do">## A warning is produced because the function expects incidence data</span></span>
<span id="cb27-370"><a href="#cb27-370"></a><span class="do">## (presence-absence), but I'm feeding it abundance (count) data.</span></span>
<span id="cb27-371"><a href="#cb27-371"></a><span class="do">## Nothing serious, as the function converts the abundance data to</span></span>
<span id="cb27-372"><a href="#cb27-372"></a><span class="do">## incidences.</span></span>
<span id="cb27-373"><a href="#cb27-373"></a><span class="in">```</span></span>
<span id="cb27-374"><a href="#cb27-374"></a></span>
<span id="cb27-375"><a href="#cb27-375"></a>iNEXT focuses on three measures of Hill numbers of order q: species richness (q = 0), Shannon diversity (q = 1, the exponential of Shannon entropy) and Simpson diversity (q = 2, the inverse of Simpson concentration). For each diversity measure, iNEXT uses the observed sample of abundance or incidence data (called the “reference sample”) to compute diversity estimates and the associated 95% confidence intervals for the following two types of rarefaction and extrapolation (R/E):</span>
<span id="cb27-376"><a href="#cb27-376"></a> </span>
<span id="cb27-377"><a href="#cb27-377"></a><span class="ss">1. </span>Sample‐size‐based R/E sampling curves: iNEXT computes diversity estimates for rarefied and extrapolated samples up to an appropriate size. This type of sampling curve plots the diversity estimates with respect to sample size.</span>
<span id="cb27-378"><a href="#cb27-378"></a></span>
<span id="cb27-379"><a href="#cb27-379"></a><span class="ss">2. </span>Coverage‐based R/E sampling curves: iNEXT computes diversity estimates for rarefied and extrapolated samples with sample completeness (as measured by sample coverage) up to an appropriate coverage. This type of sampling curve plots the diversity estimates with respect to sample coverage.</span>
<span id="cb27-380"><a href="#cb27-380"></a></span>
<span id="cb27-381"><a href="#cb27-381"></a>iNEXT also plots the above two types of sampling curves and a sample completeness curve. The sample completeness curve provides a bridge between these two types of curves.</span>
<span id="cb27-382"><a href="#cb27-382"></a></span>
<span id="cb27-383"><a href="#cb27-383"></a>For information about Hill numbers see David Zelený's <span class="co">[</span><span class="ot">Analysis of community data in R</span><span class="co">](https://www.davidzeleny.net/anadat-r/doku.php/en:diversity_analysis)</span> and Jari Oksanen's coverage of <span class="co">[</span><span class="ot">diversity measures in **vegan**</span><span class="co">](https://cran.r-project.org/web/packages/vegan/vignettes/diversity-vegan.pdf)</span>.</span>
<span id="cb27-384"><a href="#cb27-384"></a></span>
<span id="cb27-385"><a href="#cb27-385"></a>There are four datasets distributed with iNEXT and numerous examples are provided in the <span class="co">[</span><span class="ot">Introduction Vignette</span><span class="co">](https://cran.r-project.org/web/packages/iNEXT/vignettes/Introduction.html)</span>. iNEXT has an 'odd' data format that might seem foreign to **vegan** users. To use iNEXT with dataset suitable for analysis in vegan, we first need to convert BCI data to a species × site matrix (@fig-iNEXT):</span>
<span id="cb27-386"><a href="#cb27-386"></a></span>
<span id="cb27-387"><a href="#cb27-387"></a><span class="fu">### "Recipe" For Individual-Based Rarefaction</span></span>
<span id="cb27-388"><a href="#cb27-388"></a></span>
<span id="cb27-389"><a href="#cb27-389"></a>This method estimates the expected number of species for a given number of individuals sampled. The R function <span class="in">`rarecurve()`</span> in the **vegan** package automates this process for each site, but here is the recipe for generating one such curve for a single site (one row from your table):</span>
<span id="cb27-390"><a href="#cb27-390"></a></span>
<span id="cb27-391"><a href="#cb27-391"></a><span class="ss">1. </span>**Create the Individual Pool**: From your chosen site's row, create a conceptual "pool" of all individuals. For example, if the site has 3 individuals of Species A, 0 of Species B, and 5 of Species C, your pool is a list containing <span class="in">`(A, A, A, C, C, C, C, C)`</span>. The total number of individuals in this pool is $N$.</span>
<span id="cb27-392"><a href="#cb27-392"></a></span>
<span id="cb27-393"><a href="#cb27-393"></a><span class="ss">2. </span>**Set the Subsample Size ($n$)**: Choose a subsample size $n$, where $n$ is an integer from 1 to $N$. You will repeat the following steps for many different values of n to build the curve's *x*-axis.</span>
<span id="cb27-394"><a href="#cb27-394"></a></span>
<span id="cb27-395"><a href="#cb27-395"></a><span class="ss">3. </span>**Randomly Subsample Individuals**: From your pool of N individuals, randomly draw $n$ individuals without replacement.</span>
<span id="cb27-396"><a href="#cb27-396"></a></span>
<span id="cb27-397"><a href="#cb27-397"></a><span class="ss">4. </span>**Count Unique Species**: Count the number of unique species in your random subsample. For instance, if you drew <span class="in">`(A, C, C)`</span>, your species count for this draw is 2.</span>
<span id="cb27-398"><a href="#cb27-398"></a></span>
<span id="cb27-399"><a href="#cb27-399"></a><span class="ss">5. </span>**Repeat and Average**: A single draw isn't enough. Repeat steps 3 and 4 many times (*e.g.*, 100 or 1,000 times) for the same subsample size $n$. Then, calculate the average species count across all those repetitions. This gives you the expected species richness, $E(S_n)$, for that specific sample size.</span>
<span id="cb27-400"><a href="#cb27-400"></a></span>
<span id="cb27-401"><a href="#cb27-401"></a><span class="ss">6. </span>**Plot the Curve**: Repeat the entire process (steps 3-5) for a range of subsample sizes from $n=1$ up to $N$. Plot the subsample size ($n$) on the *x*-axis and the expected species richness ($E(S_n)$) on the *y*-axis. The result is a smooth curve that shows how many species you can expect to find for any given number of individuals sampled from that site.</span>
<span id="cb27-402"><a href="#cb27-402"></a></span>
<span id="cb27-403"><a href="#cb27-403"></a><span class="fu">### "Recipe" For Sample-Based Rarefaction</span></span>
<span id="cb27-404"><a href="#cb27-404"></a></span>
<span id="cb27-405"><a href="#cb27-405"></a>This method, which is very similar to a species accumulation curve, estimates the expected number of species for a given number of samples (*e.g.*, sites, quadrats). Here’s the recipe, which uses your entire site × species table:</span>
<span id="cb27-406"><a href="#cb27-406"></a></span>
<span id="cb27-407"><a href="#cb27-407"></a><span class="ss">1. </span>**Define the Sample Pool**: Your pool is the complete set of samples (all the rows in your site × species table). Let's say you have $K$ total samples.</span>
<span id="cb27-408"><a href="#cb27-408"></a></span>
<span id="cb27-409"><a href="#cb27-409"></a><span class="ss">2. </span>**Set the Subsample Size ($k$)**: Choose a number of samples $k$ to draw, where $k$ is an integer from 1 to $K$. This will be the value on your *x*-axis.</span>
<span id="cb27-410"><a href="#cb27-410"></a></span>
<span id="cb27-411"><a href="#cb27-411"></a><span class="ss">3. </span>**Randomly Subsample Samples**: From your pool of $K$ samples, randomly select $k$ of them without replacement.</span>
<span id="cb27-412"><a href="#cb27-412"></a></span>
<span id="cb27-413"><a href="#cb27-413"></a><span class="ss">4. </span>**Count Unique Species**: Combine the species lists from the $k$ chosen samples and count the total number of unique species found across all of them.</span>
<span id="cb27-414"><a href="#cb27-414"></a></span>
<span id="cb27-415"><a href="#cb27-415"></a><span class="ss">5. </span>**Repeat and Average**: Just as before, repeat steps 3 and 4 many times for the same subsample size $k$. Calculate the average species count from all these repetitions to get the expected species richness for that level of sampling effort.</span>
<span id="cb27-416"><a href="#cb27-416"></a></span>
<span id="cb27-417"><a href="#cb27-417"></a><span class="ss">6. </span>**Plot the Curve**: Repeat the process for all subsample sizes from $k=1$ up to $K$. Plot the number of samples ($k$) on the *x*-axis and the corresponding expected species richness on the *y*-axis to create the sample-based rarefaction curve.</span>
<span id="cb27-418"><a href="#cb27-418"></a></span>
<span id="cb27-419"><a href="#cb27-419"></a><span class="fu">## Distance-Decay Curves and Elevation (and Other) Gradients</span></span>
<span id="cb27-420"><a href="#cb27-420"></a></span>
<span id="cb27-421"><a href="#cb27-421"></a>The principles of distance decay relationships are clearly captured in analyses of  <span class="co">[</span><span class="ot">$\beta$-diversity</span><span class="co">](04-biodiversity.html#contemporary-definitions-%CE%B2-diversity)</span>---see specifically **turnover**, $\beta_\text{sim}$. Distance decay is the primary explanation for the spatial pattern of $\beta$-diversity along the South African coast in @smit2017seaweeds. A deeper dive into distance decay calculation can be seen in <span class="co">[</span><span class="ot">Deep Dive into Gradients</span><span class="co">](../BCB743/deep_dive.qmd)</span>.</span>
<span id="cb27-422"><a href="#cb27-422"></a></span>
<span id="cb27-423"><a href="#cb27-423"></a>In once sense, an elevation gradient can be seen as specific case of distance decay. The Doubs River dataset offers a nice example of data collected along an elevation gradient (which also happens to be a gradient along a river). Elevation gradients have many similarities with depth gradients (*e.g.*, down the ocean depths) and latitudinal gradients. </span>
<span id="cb27-424"><a href="#cb27-424"></a></span>
<span id="cb27-425"><a href="#cb27-425"></a>::: callout-note</span>
<span id="cb27-426"><a href="#cb27-426"></a><span class="fu">## Two things to think deeply about</span></span>
<span id="cb27-427"><a href="#cb27-427"></a>A) When we look at gradients such as elevation (and some others), please think about what the species are *actually* responding to. Are they responding to elevation directly?</span>
<span id="cb27-428"><a href="#cb27-428"></a></span>
<span id="cb27-429"><a href="#cb27-429"></a>B) A recipe for the analysis of distance decay relationships can be found in our earlier lectures on the topic. However, there are many ways to approach these kinds of problems, and, for the sake of doing well in tests and exams, I suggest you deeply think about them.</span>
<span id="cb27-430"><a href="#cb27-430"></a>:::</span>
<span id="cb27-431"><a href="#cb27-431"></a></span>
<span id="cb27-432"><a href="#cb27-432"></a>::: callout-important</span>
<span id="cb27-433"><a href="#cb27-433"></a><span class="fu">## Lab 4</span></span>
<span id="cb27-434"><a href="#cb27-434"></a>(To be reviewed by BCB743 student but not for marks)</span>
<span id="cb27-435"><a href="#cb27-435"></a></span>
<span id="cb27-436"><a href="#cb27-436"></a><span class="ss">1. </span>Produce the following figures for the species data indicated in <span class="co">[</span><span class="ot">square brackets</span><span class="co">]</span>:</span>
<span id="cb27-437"><a href="#cb27-437"></a>    a. species-abundance distribution <span class="co">[</span><span class="ot">mite</span><span class="co">]</span>;</span>
<span id="cb27-438"><a href="#cb27-438"></a>    b. occupancy-abundance curves <span class="co">[</span><span class="ot">mite</span><span class="co">]</span>;</span>
<span id="cb27-439"><a href="#cb27-439"></a>    c. species-area curves <span class="co">[</span><span class="ot">seaweed</span><span class="co">]</span>---note, do not use the **BAT** package's <span class="in">`alpha.accum()`</span> function as your computer might fall over;</span>
<span id="cb27-440"><a href="#cb27-440"></a>    d. rarefaction curves <span class="co">[</span><span class="ot">mite</span><span class="co">]</span>.</span>
<span id="cb27-441"><a href="#cb27-441"></a></span>
<span id="cb27-442"><a href="#cb27-442"></a>    Answer each under its own heading. For each, also explain briefly what the purpose of the analysis is (i.e. what ecological insights might be provided), and describe the findings of your own analysis as well as any ecological implications that you might be able to detect.</span>
<span id="cb27-443"><a href="#cb27-443"></a></span>
<span id="cb27-444"><a href="#cb27-444"></a><span class="ss">2. </span>Find the most dominant species in the Doubs River dataset.</span>
<span id="cb27-445"><a href="#cb27-445"></a></span>
<span id="cb27-446"><a href="#cb27-446"></a><span class="ss">3. </span>Discuss how elevation, depth, or latitudinal gradients are similar in many aspects to distance decay relationships.</span>
<span id="cb27-447"><a href="#cb27-447"></a>:::</span>
<span id="cb27-448"><a href="#cb27-448"></a></span>
<span id="cb27-449"><a href="#cb27-449"></a>:::{.callout-important}</span>
<span id="cb27-450"><a href="#cb27-450"></a><span class="fu">## Submission Instructions</span></span>
<span id="cb27-451"><a href="#cb27-451"></a></span>
<span id="cb27-452"><a href="#cb27-452"></a>The Lab 4 assignment will not be assessed.</span>
<span id="cb27-453"><a href="#cb27-453"></a>:::</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>© 2025, AJ Smit</p>
</div>   
    <div class="nav-footer-center">

<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BDC334/Lab-04-biodiversity.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>


<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>