<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Smit, A. J.">
<meta name="dcterms.date" content="2021-01-01">

<title>Lab 3. Quantifying Biodiversity – The Tangled Bank</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../BDC334/Lab-04-biodiversity.html" rel="next">
<link href="../BDC334/Lab-02b-env_dist.html" rel="prev">
<link href="../images/favicon.ico" rel="icon">
<script src="../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-1b3e43c72e8be34557c75123b0b69e0d.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-52dce9c962c322af68a5bf3d175e49c8.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/quarto-contrib/fontawesome6-0.1.0/all.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/fontawesome6-0.1.0/latex-fontsize.css" rel="stylesheet">
<script src="../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-SCYH68M489"></script>

<script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-SCYH68M489', { 'anonymize_ip': true});
</script>

<script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"headline",
  "consent_type":"express",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  ,
"language":"en"
  });
});
</script> 
  
<style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../styles.css">
<meta property="og:title" content="Lab 3. Quantifying Biodiversity – The Tangled Bank">
<meta property="og:description" content="UWC’s Biodiversity &amp; Conservation Biology Department Pages Serving BDC334, BCB743, and BCB744.">
<meta property="og:image" content="http://tangledbank.netlify.app/BDC334/data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==">
<meta property="og:site_name" content="The Tangled Bank">
</head>

<body class="nav-sidebar floating nav-fixed slimcontent quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">The Tangled Bank</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-vignettes" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Vignettes</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-vignettes">    
        <li>
    <a class="dropdown-item" href="../vignettes/netCDF_dates.html">
 <span class="dropdown-text">Dates From netCDF Files: Two Approaches</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/prep_NOAA_OISST.html">
 <span class="dropdown-text">Downloading and Prep: ERDDAP</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_ERDDAP.html">
 <span class="dropdown-text">Retrieving Chlorophyll-a Data from ERDDAP Servers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/alt_method.html">
 <span class="dropdown-text">Downloading and Prep: Python + CDO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/gridded_data.html">
 <span class="dropdown-text">Detecting Events in Gridded Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/regridding.html">
 <span class="dropdown-text">Regridding gridded data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/MHW_MCS_horizonplots.html">
 <span class="dropdown-text">Displaying MHWs and MCSs as Horizon Plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/heatwaveR_issues.html">
 <span class="dropdown-text">heatwaveR Issues</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_sightings.html">
 <span class="dropdown-text">Plotting the Whale Sightings and Chlorophyll-<em>a</em> Concentrations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_localisation.html">
 <span class="dropdown-text">Spatial Localisation, Subsetting, and Aggregation of the Chlorophyll-<em>a</em> Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/elem_ts_methods.html">
 <span class="dropdown-text">Wavelet Analysis of Diatom Time Series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/buffer_data_extract.html">
 <span class="dropdown-text">Extracting gridded data within a buffer</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-high-performace-computing" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">High-Performace Computing</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-high-performace-computing">    
        <li>
    <a class="dropdown-item" href="../vignettes/README_Lengau.html">
 <span class="dropdown-text">Using Lengau</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/PBSPro_users.html">
 <span class="dropdown-text">PBSPro user and job management</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/README_tmux.html">
 <span class="dropdown-text">Using tmux</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../blog.html"> 
<span class="menu-text">Tangled Bank Blog</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-blogroll" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Blogroll</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-blogroll">    
        <li>
    <a class="dropdown-item" href="https://www.r-bloggers.com">
 <span class="dropdown-text">R-bloggers</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="../blog.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/ajsmit/tangled_bank">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://bugs.com">
 <span class="dropdown-text">Report a Bug</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BDC334/BDC334_syllabus.html">BDC334: Biogeography &amp; Global Ecology</a></li><li class="breadcrumb-item"><a href="../BDC334/Lab-03-biodiversity.html">Lab 3. Quantifying Biodiversity</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../images/hex_logo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">UNDERGRADUATE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC223: Plant Ecophysiology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/BDC223_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L00-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 0: Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L01-worldmapper.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: Worldmapper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L02-SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2: Surface Area to Volume (SA/V) Ratios in Biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L03-plant_stress.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3: Plant Stress</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L04-carbon_cycle.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Perturbations to the Carbon Cycle</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L05-light.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 5: Light</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L06a-pigments_photosynthesis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6a: Pigments and Photosynthesis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L06b-jassby_platt.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6b: PI Curves – Jassby and Platt</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L07-chromatic_adaptation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 7: Chromatic Adaptation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L08a-nutrient_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8a: Nutrient Uptake</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L08b-nutrients_michaelis_menten.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8b: Uptake Kinetics – Michaelis-Menten</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab1_SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1: Surface Area to Volume (SA/V) Ratios in Biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab2_misc_calcs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2: Miscellaneous Calculations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab3_PI_curves.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3: PI Curves – Jassby and Platt</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab4_nitrogen_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4: Uptake Kinetics – Michaelis-Menten</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/BDC223_FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">BDC334: Biogeography &amp; Global Ecology</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/BDC334_syllabus.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/BDC334-Lecture-Transcripts.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">BCB334 Lecture Transcript</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: Introductory Lecture</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-02-ecosystems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2. Overview of Ecosystems</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-03-gradients.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3. Ecological Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Biodiversity Concepts</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 5. Multivariate Data</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-06-unified-ecology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6: Unified Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 7: Threats to Biodiversity</span>
  </li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 8: Nature's Contribution to People</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1. Ecological Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-02a-r_rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2a. R &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-02b-env_dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2b. Environmental Distance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-03-biodiversity.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Lab 3. Quantifying Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4. Species Distribution Patterns</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/Transboundary_systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transboundary systems</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS CORE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">BCB744: Introduction to R, &amp; Biostatistics</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/BCB744_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/01-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. R &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/02-working-with-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Working With Data &amp; Code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/03-data-in-R.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Data Classes &amp; Structures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/04-workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. R Workflows</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/05-graphics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Graphics with <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/06-faceting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Faceting Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/07-brewing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Brewing Colours</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/08-mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Mapping With <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/09-mapping_style.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Mapping With Style</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/10-mapping_rnaturalearth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Mapping With Natural Earth</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/11-mapping_quakes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. The Fiji Earthquake Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/12-tidy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Tidy Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/13-tidier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Tidier Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/14-tidiest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Tidiest Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/15-recap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15. Recap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/BCB744_Intro_R_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Intro R Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/examples/BCB744_Intro_R_Example_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 Intro R Example 1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/examples/BCB744_Intro_R_Example_2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 Intro R Example 2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../self-assessment_v1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Self-Assessment Form</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/BCB744_Intro_R_Presentations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Presentations</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/BCB744_Intro_R_Test_2025.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Intro R Test</span></span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/01-scientific-philosophy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Philosophy of Science</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/02-summarise-and-describe.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Data Summaries &amp; Descriptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/03-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Statistical Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/04-distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data Distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/05-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Statistical Inference</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/06-assumptions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Assumptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/07-t_tests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. <em>t</em>-Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/08-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. ANOVA</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/09-regressions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Simple Linear Regressions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/10-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/11-decision_guide.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/12-glance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Non-Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/13-confidence.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Confidence Intervals</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/14-transformations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Data Transformations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/BCB744_Biostatistics_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Biostatistics Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/examples/BCB744_BioStats_Example_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 BioStats 2025 Example 1</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS ELECTIVES</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">BCB743: Quantitative Ecology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/BCB743_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/review.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1-4: Review Concepts fo Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5: Correlations &amp; Associations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7: Intro to Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8a: Principal Components Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA_examples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8b: PCA: Additional Examples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA_SDG_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8c: PCA: WHO SDG Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/CA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9a: Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/DCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9b: Detrended Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCoA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10: Principal Coordinates Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/nMDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11a: Non-metric Multidimensional Scaling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/nMDS_diatoms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11b: nMDS: PERMANOVA (Diatoms) Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/unconstrained-summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12: Unconstrained Ordination Summary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/constrained_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13a: Distance-Based Redundancy Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/two_oceans_appendices.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13b: db-RDA: Seaweeds Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/cluster_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14: Cluster Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/model_building.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15: Model Building</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/deep_dive.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16a: Deep Dive Into Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/multiple_regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16b: Multiple Regression</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../AI4AI/AI4AI.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Artificial Intelligence for Academic Integrity (AI4AI)</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">SUPPORT</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Web Resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/general_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/ecology_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quantitative Ecology Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/spatial_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial R Resources</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#preparation" id="toc-preparation" class="nav-link active" data-scroll-target="#preparation">Preparation</a>
  <ul class="collapse">
  <li><a href="#the-south-african-seaweed-data" id="toc-the-south-african-seaweed-data" class="nav-link" data-scroll-target="#the-south-african-seaweed-data">The South African Seaweed Data</a></li>
  <li><a href="#setting-up-the-analysis-environment" id="toc-setting-up-the-analysis-environment" class="nav-link" data-scroll-target="#setting-up-the-analysis-environment">Setting Up the Analysis Environment</a></li>
  <li><a href="#a-look-at-the-data" id="toc-a-look-at-the-data" class="nav-link" data-scroll-target="#a-look-at-the-data">A Look at the Data</a></li>
  </ul></li>
  <li><a href="#species-data" id="toc-species-data" class="nav-link" data-scroll-target="#species-data">Species Data</a></li>
  <li><a href="#three-measures-of-biodiversity-alpha--gamma--beta-diversity" id="toc-three-measures-of-biodiversity-alpha--gamma--beta-diversity" class="nav-link" data-scroll-target="#three-measures-of-biodiversity-alpha--gamma--beta-diversity">Three Measures of Biodiversity: <span class="math inline">\(\alpha\)</span>-, <span class="math inline">\(\gamma\)</span>-, <span class="math inline">\(\beta\)</span>-Diversity</a>
  <ul class="collapse">
  <li><a href="#alpha-diversity" id="toc-alpha-diversity" class="nav-link" data-scroll-target="#alpha-diversity">Alpha-Diversity</a></li>
  <li><a href="#gamma-diversity" id="toc-gamma-diversity" class="nav-link" data-scroll-target="#gamma-diversity">Gamma-Diversity</a></li>
  <li><a href="#beta-diversity" id="toc-beta-diversity" class="nav-link" data-scroll-target="#beta-diversity">Beta-Diversity</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BDC334/Lab-03-biodiversity.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BDC334/BDC334_syllabus.html">BDC334: Biogeography &amp; Global Ecology</a></li><li class="breadcrumb-item"><a href="../BDC334/Lab-03-biodiversity.html">Lab 3. Quantifying Biodiversity</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Lab 3. Quantifying Biodiversity</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://tangledbank.netlify.app">Smit, A. J.</a> <a href="https://orcid.org/0000-0002-3799-6126" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uwc.ac.za">
            University of the Western Cape
            </a>
          </p>
      </div>
  </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 1, 2021</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
BCB743
</div>
</div>
<div class="callout-body-container callout-body">
<p><strong>This material must be reviewed by BCB743 students in Week 1 of Quantitative Ecology.</strong></p>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
This Lab Accompanies the Following Lecture
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li><a href="../BDC334/Lec-04-biodiversity.html">Lecture 4: Biodiversity Concepts</a></li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Data For This Lab
</div>
</div>
<div class="callout-body-container callout-body">
<p>The seaweed <span class="citation" data-cites="smit2017seaweeds">(<a href="#ref-smit2017seaweeds" role="doc-biblioref">Smit et al. 2017</a>)</span> and toy data are at the links below:</p>
<ul>
<li>The seaweed species data – <a href="../data/seaweed/SeaweedSpp.csv"><code>SeaweedSpp.csv</code></a></li>
<li>The seaweed environmental data – <a href="../data/seaweed/SeaweedEnv.csv"><code>SeaweedEnv.csv</code></a></li>
<li>The seaweed coastal sections – <a href="../data/seaweed/SeaweedSites.csv"><code>SeaweedSites.csv</code></a></li>
<li>The fictitious light data <a href="../data/light_levels.csv"><code>light_levels.csv</code></a></li>
</ul>
</div>
</div>
<blockquote class="blockquote">
<p><em><strong>Biodiversity</strong> The variability among living organisms from all sources including,</em> inter alia<em>, terrestrial, marine and other aquatic ecosystems and the ecological complexes of which they are part; this includes diversity within species, between species and of ecosystems.</em></p>
<p>— International Union for the Conservation of Nature (IUCN), Convention on Biological Diversity</p>
</blockquote>
<p>The IUCN definition considers a diversity of diversity concepts. This module looks at diversity only at the species level (species diversity). However, we can also approach macroecological problems from phylogenetic and functional (and other) diversity concepts of view. Functional and phylogenetic diversity ideas will be introduced in the <a href="../BCB743/BCB743_index.html">BDC743 module Quantitative Ecology</a>.</p>
<section id="preparation" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="preparation">Preparation</h2>
<section id="the-south-african-seaweed-data" class="level3 page-columns page-full">
<h3 class="anchored" data-anchor-id="the-south-african-seaweed-data">The South African Seaweed Data</h3>
<p>In these examples, we will use the seaweed data of <span class="citation" data-cites="smit2017seaweeds">Smit et al. (<a href="#ref-smit2017seaweeds" role="doc-biblioref">2017</a>)</span>. Please make sure that you read this paper. An additional file describing the background to the data is available <a href="../docs/Smit_the_seaweed_data.pdf">here</a> (<a href="#fig-seaweed-sections" class="quarto-xref">Figure&nbsp;1</a>).</p>
<div id="fig-seaweed-sections" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-seaweed-sections-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../images/Seaweed_sections.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: The coastal sections and associated seawater temperature profile associated with the study by @smit2017seaweeds."><img src="../images/Seaweed_sections.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a>
</div>
<figcaption class="quarto-float-caption-margin quarto-float-caption quarto-float-fig margin-caption" id="fig-seaweed-sections-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: The coastal sections and associated seawater temperature profile associated with the study by <span class="citation" data-cites="smit2017seaweeds">Smit et al. (<a href="#ref-smit2017seaweeds" role="doc-biblioref">2017</a>)</span>.
</figcaption>
</figure>
</div>
<p>One of the datasets, <span class="math inline">\(Y\)</span> (in the file <code>SeaweedSpp.csv</code>), comprises updated distribution records of 847 macroalgal species within each of 58 × 50 km-long sections of the South African coast <span class="citation" data-cites="bolton2002seaweed">(<a href="#ref-bolton2002seaweed" role="doc-biblioref">Bolton and Stegenga 2002</a>)</span>. The dataset captures <em>ca</em>. 90% of the known seaweed flora of South Africa, but excludes some very small and/or very rare species for which data are insufficient. The data are from verifiable literature sources and John Bolton and Rob Anderson’s collections, assembled from information collected by teams of phycologists over three decades <span class="citation" data-cites="bolton1986marine stegenga1997seaweeds bolton2002seaweed de2005guide">(<a href="#ref-bolton1986marine" role="doc-biblioref">Bolton 1986</a>; <a href="#ref-stegenga1997seaweeds" role="doc-biblioref">Stegenga et al. 1997</a>; <a href="#ref-bolton2002seaweed" role="doc-biblioref">Bolton and Stegenga 2002</a>; <a href="#ref-de2005guide" role="doc-biblioref">De Clerck et al. 2005</a>)</span>. Another file, <span class="math inline">\(E\)</span> (in <code>env.csv</code>), is a dataset of <em>in situ</em> coastal seawater temperatures derived from daily measurements over 40 years <span class="citation" data-cites="smit2013coastal">(<a href="#ref-smit2013coastal" role="doc-biblioref">Smit et al. 2013</a>)</span>.</p>
</section>
<section id="setting-up-the-analysis-environment" class="level3">
<h3 class="anchored" data-anchor-id="setting-up-the-analysis-environment">Setting Up the Analysis Environment</h3>
<p>We will use <strong>R</strong>, so first, we must find, install and load various packages. Some packages will be available on CRAN and can be accessed and installed the usual way, but you will need to download others from <a href="https://r-forge.r-project.org/R/?group_id=195">R Forge</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(vegan)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(betapart)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(BiodiversityR) <span class="co"># this package may at times be problematic to install</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="a-look-at-the-data" class="level3">
<h3 class="anchored" data-anchor-id="a-look-at-the-data">A Look at the Data</h3>
<p>Let’s load the data and see how it is structured:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>spp <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">'../data/seaweed/SeaweedSpp.csv'</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a>spp <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(spp, <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="co"># Lets look at the data:</span></span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="fu">dim</span>(spp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  58 847</code></pre>
</div>
</div>
<p>We see that our dataset has 58 rows and 847 columns. What is in the columns and rows? Start with the first five rows and five columns:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>spp[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  ACECAL ACEMOE ACRVIR AROSP1 ANAWRI
1      0      0      0      0      0
2      0      0      0      0      0
3      0      0      0      0      0
4      0      0      0      0      0
5      0      0      0      0      0</code></pre>
</div>
</div>
<p>Now the last five rows and five columns:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>spp[(<span class="fu">nrow</span>(spp) <span class="sc">-</span> <span class="dv">5</span>)<span class="sc">:</span><span class="fu">nrow</span>(spp), (<span class="fu">ncol</span>(spp) <span class="sc">-</span> <span class="dv">5</span>)<span class="sc">:</span><span class="fu">ncol</span>(spp)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   WOMKWA WOMPAC WRAARG WRAPUR WURMIN ZONSEM
53      0      0      1      0      0      0
54      0      0      1      0      0      0
55      0      0      1      0      0      0
56      0      1      1      0      1      0
57      1      0      1      0      1      0
58      0      0      1      0      1      0</code></pre>
</div>
</div>
<p>So, each row corresponds to a site (i.e.&nbsp;each of the coastal sections), and each column contains a species. We arrange the species alphabetically and use a six-letter code to identify them.</p>
</section>
</section>
<section id="species-data" class="level2">
<h2 class="anchored" data-anchor-id="species-data">Species Data</h2>
<p>When ecologists talk about species diversity, they typically consider the characteristics of biological communities in a specific habitat, ecological community, or ecosystem. Species diversity considers three essential concepts about how species are distributed in space: their <strong>richness</strong>, <strong>abundance</strong>, and <strong>evenness</strong>. We can express each of these as biodiversity metrics that allow us to compare communities in space and time.</p>
<p>When ecologists talk about ‘biodiversity’, they might not necessarily be interested in <em>all</em> the plants and animals and things that are neither plant nor animal that occur at a particular place. Some ecologists are interested in ants and moths. Others might find fish more insightful. Some even like marine mammals! I prefer seaweed. The analysis of biodiversity data might often be constrained to some higher-level taxon, such as all angiosperms in a landscape, reptiles, etc. (but we sample all species in the higher-level taxon). Some ecological questions benefit from comparisons of diversity assessments among selected taxa (avifauna vs small mammals, for example), as this focus might address some particular ecological hypothesis. The bird vs small mammal comparison might reveal how barriers such as streams and rivers structure biodiversity patterns. In our examples, we will use such focused datasets.</p>
<p>Here we look at the various measures of biodiversity, viz.&nbsp;<span class="math inline">\(\alpha\)</span>-, <span class="math inline">\(\gamma\)</span>- and <span class="math inline">\(\beta\)</span>-diversity. David Zelený, in his <a href="https://www.davidzeleny.net/anadat-r/doku.php/en:diversity_analysis">Analysis of community data in R</a>, provides deeper analysis and compulsory reading.</p>
<!--- # Topic 2: Measures of biodiversity --->
</section>
<section id="three-measures-of-biodiversity-alpha--gamma--beta-diversity" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="three-measures-of-biodiversity-alpha--gamma--beta-diversity">Three Measures of Biodiversity: <span class="math inline">\(\alpha\)</span>-, <span class="math inline">\(\gamma\)</span>-, <span class="math inline">\(\beta\)</span>-Diversity</h2>
<p><span class="citation" data-cites="whittaker1972evolution">Whittaker (<a href="#ref-whittaker1972evolution" role="doc-biblioref">1972</a>)</span> coined three measures of biodiversity, and the concepts were ‘modernised’ by <span class="citation" data-cites="jurasinski2009inventory">Jurasinski et al. (<a href="#ref-jurasinski2009inventory" role="doc-biblioref">2009</a>)</span>. The concepts represent the measurement of biodiversity <em>across different spatial scales</em>. <span class="math inline">\(\alpha\)</span>- and <span class="math inline">\(\gamma\)</span>-diversity express the total number of species in an area. The first, <span class="math inline">\(\alpha\)</span>-diversity, represents the number of species at the small (local) scale, such as, for example, within a sampling unit like a quadrat, transect, plot, or trawl. Alternatively, maybe the research question represents the local scale by several sampling units nesting within a small patch of landscape and defines the mean species richness within this patch as local. Multiples (sampling units or patches) are nested within a larger region (or ecosystem) and serve as replicates. The complete number of species across all of these replicates indicates the diversity at a larger scale—this is called <span class="math inline">\(\gamma\)</span>-diversity. <span class="math inline">\(\beta\)</span>-diversity refers to the change in species composition among samples (sites).</p>
<p>By now, you will have received a brief <a href="../BDC334/Lab-02a-r_rstudio.html">Introduction to R</a>, and we can proceed with looking at some of the measures of biodiversity. We will start by using data on the seaweeds of South Africa to demonstrate some ideas around diversity measures. The <strong>vegan</strong><a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a> (for <em>vegetation analysis</em>) package <span class="citation" data-cites="oksanen2022vegan">(<a href="#ref-oksanen2022vegan" role="doc-biblioref">Oksanen et al. 2022</a>)</span> offers <a href="../docs/Oksanen_diversity-vegan.pdf">various functions</a> to calculate diversity indices. I will demonstrate some of these functions below.</p>
<div class="no-row-height column-margin column-container"><div id="fn1"><p><sup>1</sup>&nbsp;I am by no means an advocate for veganism.</p></div></div><section id="alpha-diversity" class="level3 page-columns page-full">
<h3 class="anchored" data-anchor-id="alpha-diversity">Alpha-Diversity</h3>
<p>We can represent <span class="math inline">\(\alpha\)</span>-diversity in three ways:</p>
<ol type="1">
<li>as <em>species richness</em>, <span class="math inline">\(S\)</span>;</li>
<li>as a <em>univariate diversity index</em>, such as the <span class="math inline">\(\alpha\)</span> parameter of <a href="04-biodiversity.html#species-abundance-distribution">Fisher’s log-series</a>, Shannon diversity, <span class="math inline">\(H'\)</span>, Simpson’s diversity, <span class="math inline">\(\lambda\)</span>; or</li>
<li>Species <em>evenness</em>, e.g.&nbsp;Pielou’s evenness, <span class="math inline">\(J\)</span>.</li>
</ol>
<p>We will work through each in turn.</p>
<section id="species-richness-s" class="level4 page-columns page-full">
<h4 class="anchored" data-anchor-id="species-richness-s">Species Richness, <span class="math inline">\(S\)</span></h4>
<p>First, is species richness, which we denote by the symbol <span class="math inline">\(S\)</span>. This is the simplest measure of <span class="math inline">\(\alpha\)</span>-diversity, counting the number of species (or another taxonomic level) present in a given community or sample. It doesn’t consider the abundance of species.</p>
<p>In the seaweed biodiversity data, I count the number of species within each of the sections. This is because we view each coastal section as the local scale (the smallest unit of sampling).</p>
<p>The preferred option for calculating species richness is the <code>specnumber()</code> function in <strong>vegan</strong>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="annotated-cell-5"><pre class="sourceCode numberSource r code-annotation-code number-lines code-with-copy code-annotated"><code class="sourceCode r"><a class="code-annotation-anchor" data-target-cell="annotated-cell-5" data-target-annotation="1" onclick="event.preventDefault();">1</a><span id="annotated-cell-5-1" class="code-annotation-target"><a href="#annotated-cell-5-1"></a><span class="fu">specnumber</span>(spp, <span class="at">MARGIN =</span> <span class="dv">1</span>)</span><div class="code-annotation-gutter-bg"></div><div class="code-annotation-gutter"></div></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-annotation">
<dl class="code-annotation-container-grid">
<dt data-target-cell="annotated-cell-5" data-target-annotation="1">1</dt>
<dd>
<span data-code-cell="annotated-cell-5" data-code-lines="1" data-code-annotation="1">The <code>MARGIN = 1</code> argument tells R to calculate the number of species within each row (site).</span>
</dd>
</dl>
</div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] 138 139 139 140 143 143 143 145 149 148 159 162 208 147 168 204 269 276 280
[20] 265 265 283 269 279 281 295 290 290 299 295 311 317 298 299 301 315 308 327
[39] 340 315 315 302 311 280 300 282 283 321 319 319 330 293 291 292 294 313 333
[58] 316</code></pre>
</div>
</div>
<p>The data output is easier to understand if we display it as a <code>tibble()</code>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>spp_richness <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">section =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">58</span>,</span>
<span id="cb9-2"><a href="#cb9-2"></a>                       <span class="at">richness =</span> <span class="fu">specnumber</span>(spp, <span class="at">MARGIN =</span> <span class="dv">1</span>))</span>
<span id="cb9-3"><a href="#cb9-3"></a><span class="fu">head</span>(spp_richness)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 2
  section richness
    &lt;int&gt;    &lt;int&gt;
1       1      138
2       2      139
3       3      139
4       4      140
5       5      143
6       6      143</code></pre>
</div>
</div>
<p>The <code>diversityresult()</code> function in the <strong>BiodiversityR</strong> package can do the same (sometimes this package is difficult to install due to various software dependencies that might be required for the package to load properly—do not be sad if this method does not work):</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>spp_richness <span class="ot">&lt;-</span> <span class="fu">diversityresult</span>(spp, <span class="at">index =</span> <span class="st">'richness'</span>,</span>
<span id="cb11-2"><a href="#cb11-2"></a>                                <span class="at">method =</span> <span class="st">'each site'</span>)</span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="fu">head</span>(spp_richness)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now we make a plot seen in <a href="#fig-species-richness" class="quarto-xref">Figure&nbsp;2</a>:</p>
<div class="cell page-columns page-full">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> spp_richness, (<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">58</span>, <span class="at">y =</span> richness))) <span class="sc">+</span></span>
<span id="cb12-2"><a href="#cb12-2"></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.2</span>, <span class="at">colour =</span> <span class="st">"indianred"</span>) <span class="sc">+</span></span>
<span id="cb12-3"><a href="#cb12-3"></a>  <span class="fu">xlab</span>(<span class="st">"Coastal section, west to east"</span>) <span class="sc">+</span></span>
<span id="cb12-4"><a href="#cb12-4"></a>  <span class="fu">ylab</span>(<span class="st">"Species richness"</span>) <span class="sc">+</span></span>
<span id="cb12-5"><a href="#cb12-5"></a>  <span class="fu">theme_linedraw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display page-columns page-full">
<div id="fig-species-richness" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-species-richness-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-03-biodiversity_files/figure-html/fig-species-richness-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;2: The seaweed species richness, S, within each of the coastal sections along the shore of South Africa."><img src="Lab-03-biodiversity_files/figure-html/fig-species-richness-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-margin quarto-float-caption quarto-float-fig margin-caption" id="fig-species-richness-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: The seaweed species richness, <span class="math inline">\(S\)</span>, within each of the coastal sections along the shore of South Africa.
</figcaption>
</figure>
</div>
</div>
</div>
<p>In other instances, it makes more sense to calculate the mean species richness of all the sampling units (e.g.&nbsp;quadrats) taken inside the ecosystem of interest. How you calculate and present species richness depend on your research question and so you will have to decide based on your data and study.</p>
<p>In the seaweed study, the mean ± SD species richness across all of the 58 coastal sections is:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="fu">round</span>(<span class="fu">mean</span>(spp_richness<span class="sc">$</span>richness), <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 259.24</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="fu">round</span>(<span class="fu">sd</span>(spp_richness<span class="sc">$</span>richness), <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 68.03</code></pre>
</div>
</div>
</section>
<section id="univariate-diversity-indices" class="level4">
<h4 class="anchored" data-anchor-id="univariate-diversity-indices">Univariate Diversity Indices</h4>
<p>The second way we can express <span class="math inline">\(\alpha\)</span>-diversity is to use one of the univariate diversity indices. The choice of which index to use should be informed by the extent to which one wants to emphasise richness or evenness. Species richness, <span class="math inline">\(S\)</span>, does not consider evenness as it is all about richness (obviously). Simpson’s <span class="math inline">\(\lambda\)</span> emphasises evenness a lot more. Shannon’s <span class="math inline">\(H'\)</span> is somewhere in the middle.</p>
<p><strong>Shannon’s <span class="math inline">\(H'\)</span></strong> is sometimes called Shannon’s diversity, the Shannon-Wiener index, the Shannon-Weaver index, or the Shannon entropy. This is a more nuanced measure that considers both species richness and evenness (how evenly individuals are distributed across different species).</p>
<p>It is calculated as:</p>
<p><span class="math display">\[H' = -\sum_{i=1}^{S} p_{i} \ln p_{i}\]</span> where <span class="math inline">\(p_{i}\)</span> is the proportion of individuals belonging to the <span class="math inline">\(i\)</span>th species, and <span class="math inline">\(S\)</span> is the species richness.</p>
<p><strong>Simpson’s <span class="math inline">\(\lambda\)</span></strong>, or simply the Simpson index, is a measure that represents the probability that two individuals randomly selected from a sample will belong to the same species. It is calculated as:</p>
<p><span class="math display">\[\displaystyle \lambda = \sum_{i=1}^{S} p_{i}^{2}\]</span> where <span class="math inline">\(S\)</span> is the species richness and <span class="math inline">\(p_{i}\)</span> is the relative abundance of the <span class="math inline">\(i\)</span>th species.</p>
<p><strong>Fisher’s <span class="math inline">\(\alpha\)</span></strong> estimates the <span class="math inline">\(\alpha\)</span> parameter of <a href="../BDC334/Lab-04-biodiversity.html#species-abundance-distribution">Fisher’s logarithmic series</a> (see functions <code>fisher.alpha()</code> and <code>fisherfit()</code>). The estimation is possible only for actual counts (i.e.&nbsp;integers) of individuals, so it will not work for per cent cover, biomass, and other measures that real numbers can express. It’s especially useful for comparing the diversity of samples with different total abundances. We will get to this function later under <a href="../BDC334/Lab-04-biodiversity.html#species-abundance-distribution">Fisher’s logarithmic series</a>.</p>
<p>Except for Fisher’s-<span class="math inline">\(\alpha\)</span>, we cannot calculate these for the seaweed data, because, in order to do so, we require abundance data—the seaweed data are presence-absence only. Let us load a fictitious dataset of the diversity of three different communities of plants, with each community corresponding to a different light environment (dim, mid, and high light):</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>light <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/light_levels.csv"</span>)</span>
<span id="cb17-2"><a href="#cb17-2"></a>light</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        Site    A    B    C    D    E    F
1  low_light 0.75 0.62 0.24 0.33 0.21 0.14
2  mid_light 0.38 0.15 0.52 0.57 0.28 0.29
3 high_light 0.08 0.15 0.18 0.52 0.54 0.56</code></pre>
</div>
</div>
<p>We can see above that instead of having data with 1s and 0s for presence-absence, here we have some values that indicate the relative number of individuals belonging to each of the species in the three light environments. We calculate species richness (as before), and also the Shannon and Simpson indices using <strong>vegan</strong>’s <code>diversity()</code> function:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>light_div <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb19-2"><a href="#cb19-2"></a>  <span class="at">site =</span> <span class="fu">c</span>(<span class="st">"low_light"</span>, <span class="st">"mid_light"</span>, <span class="st">"high_light"</span>),</span>
<span id="cb19-3"><a href="#cb19-3"></a>  <span class="at">richness =</span> <span class="fu">specnumber</span>(light[, <span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>], <span class="at">MARGIN =</span> <span class="dv">1</span>),</span>
<span id="cb19-4"><a href="#cb19-4"></a>  <span class="at">shannon =</span> <span class="fu">round</span>(<span class="fu">diversity</span>(light[, <span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>], <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">index =</span> <span class="st">"shannon"</span>), <span class="dv">2</span>),</span>
<span id="cb19-5"><a href="#cb19-5"></a>  <span class="at">simpson =</span> <span class="fu">round</span>(<span class="fu">diversity</span>(light[, <span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>], <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">index =</span> <span class="st">"simpson"</span>), <span class="dv">2</span>)</span>
<span id="cb19-6"><a href="#cb19-6"></a>)</span>
<span id="cb19-7"><a href="#cb19-7"></a>light_div</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 3 × 4
  site       richness shannon simpson
  &lt;chr&gt;         &lt;int&gt;   &lt;dbl&gt;   &lt;dbl&gt;
1 low_light         6    1.62    0.78
2 mid_light         6    1.71    0.81
3 high_light        6    1.59    0.77</code></pre>
</div>
</div>
<!-- Hill numbers, $q$, (Hill, 1973) summarise $S$, Shannon's $H'$ and Simpson's $\lambda$. Higher $q$ puts less weight on rare species and weighs abundant ones more. Hill numbers can be used to draw diversity profiles, which allow for an elegant comparison of diversity among communities considering both richness and evenness. -->
<!-- ```{r fig.height=3, fig.width=5} -->
<!-- data(BCI) -->
<!-- i <- sample(nrow(BCI), 12) -->
<!-- mod <- renyi(BCI[i,]) -->
<!-- plot(mod) -->
<!-- mod <- renyiaccum(BCI[i,]) -->
<!-- plot(mod, as.table=TRUE, col = c(1, 2, 2)) -->
<!-- persp(mod) -->
<!-- ``` -->
<p>Evenness refers to the shape of a <a href="../BDC334/Lab-04-biodiversity.html#species-abundance-distribution">species abundance distribution</a>, which suggests the relative abundance of different species.</p>
<p>One index for evenness is Pielou’s evenness, <span class="math inline">\(J\)</span>:</p>
<p><span class="math display">\[J = \frac{H^{\prime}} {log(S)}\]</span></p>
<p>where <span class="math inline">\(H'\)</span> is Shannon’s diversity index, and <span class="math inline">\(S\)</span> the number of species (i.e.&nbsp;<span class="math inline">\(S\)</span>).</p>
<p>To calculate Pielou’s evenness index for the <code>light</code> data, we can do this:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a>H <span class="ot">&lt;-</span> <span class="fu">diversity</span>(light[, <span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>], <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">index =</span> <span class="st">"shannon"</span>)</span>
<span id="cb21-2"><a href="#cb21-2"></a></span>
<span id="cb21-3"><a href="#cb21-3"></a>J <span class="ot">&lt;-</span> H<span class="sc">/</span><span class="fu">log</span>(<span class="fu">specnumber</span>(light[, <span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>]))</span>
<span id="cb21-4"><a href="#cb21-4"></a><span class="fu">round</span>(J, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.91 0.95 0.89</code></pre>
</div>
</div>
<p><strong>Berger-Parker Index</strong> indicates the proportion of the community that the most abundant species represents. It is given by the formula:</p>
<p><span class="math display">\[d = \frac{N_{max}}{N}\]</span> where <span class="math inline">\(N_{max}\)</span> is the number of individuals of the most common species and <span class="math inline">\(N\)</span> is the total number of individuals in the sample.</p>
<p><strong>Chao1 and ACE</strong> are estimators often used to predict the total species richness in a community based on the number of rare species observed in samples.</p>
</section>
</section>
<section id="gamma-diversity" class="level3">
<h3 class="anchored" data-anchor-id="gamma-diversity">Gamma-Diversity</h3>
<p>Returning to the seaweed data, <span class="math inline">\(Y\)</span>, let us now look at <span class="math inline">\(\gamma\)</span>-diversity—this would be the total number of species along the South African coastline in all 58 coastal sections. Since each column represents one species, and the dataset contains data collected at each of the 58 sites (the number of rows), we can do:</p>
<div class="cell">
<div class="sourceCode cell-code" id="annotated-cell-14"><pre class="sourceCode numberSource r code-annotation-code number-lines code-with-copy code-annotated"><code class="sourceCode r"><a class="code-annotation-anchor" data-target-cell="annotated-cell-14" data-target-annotation="1" onclick="event.preventDefault();">1</a><span id="annotated-cell-14-1" class="code-annotation-target"><a href="#annotated-cell-14-1"></a><span class="fu">ncol</span>(spp)</span><div class="code-annotation-gutter-bg"></div><div class="code-annotation-gutter"></div></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-annotation">
<dl class="code-annotation-container-grid">
<dt data-target-cell="annotated-cell-14" data-target-annotation="1">1</dt>
<dd>
<span data-code-cell="annotated-cell-14" data-code-lines="1" data-code-annotation="1">The number of columns gives the total number of species in this example.</span>
</dd>
</dl>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 847</code></pre>
</div>
</div>
<p>We can also use:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="fu">diversityresult</span>(spp, <span class="at">index =</span> <span class="st">'richness'</span>, <span class="at">method =</span> <span class="st">'pooled'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       richness
pooled      846</code></pre>
</div>
</div>
<!-- ```{r} -->
<!-- specpool() -->
<!-- ``` -->
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Lab 3
</div>
</div>
<div class="callout-body-container callout-body">
<p>(To be reviewed by BCB743 student but not for marks)</p>
<ol type="1">
<li>Why is there a difference between the two?</li>
<li>Which is correct?</li>
</ol>
</div>
</div>
<p>Think before you calculate <span class="math inline">\(\gamma\)</span>-diversity for your own data as it might not be as simple as here!</p>
</section>
<section id="beta-diversity" class="level3 page-columns page-full">
<h3 class="anchored" data-anchor-id="beta-diversity">Beta-Diversity</h3>
<section id="whittakers-beta-diversity" class="level4 page-columns page-full">
<h4 class="anchored" data-anchor-id="whittakers-beta-diversity">Whittaker’s <span class="math inline">\(\beta\)</span>-Diversity</h4>
<p>The first measure of <span class="math inline">\(\beta\)</span>-diversity comes from <span class="citation" data-cites="whittaker1960vegetation">Whittaker (<a href="#ref-whittaker1960vegetation" role="doc-biblioref">1960</a>)</span> and is called <em>true <span class="math inline">\(\beta\)</span>-diversity</em>. In this instance, divide the <span class="math inline">\(\gamma\)</span>-diversity for the region by the <span class="math inline">\(\alpha\)</span>-diversity for a specific coastal section. We can calculate it all at once for the whole dataset and make a graph (<a href="#fig-true-beta" class="quarto-xref">Figure&nbsp;3</a>):</p>
<div class="cell page-columns page-full">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a>true_beta <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb26-2"><a href="#cb26-2"></a>  <span class="at">beta =</span> <span class="fu">specnumber</span>(spp, <span class="at">MARGIN =</span> <span class="dv">1</span>) <span class="sc">/</span> <span class="fu">ncol</span>(spp),</span>
<span id="cb26-3"><a href="#cb26-3"></a>  <span class="at">section_no =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">58</span>)</span>
<span id="cb26-4"><a href="#cb26-4"></a>)</span>
<span id="cb26-5"><a href="#cb26-5"></a><span class="co"># true_beta</span></span>
<span id="cb26-6"><a href="#cb26-6"></a><span class="fu">ggplot</span>(<span class="at">data =</span> true_beta, (<span class="fu">aes</span>(<span class="at">x =</span> section_no, <span class="at">y =</span> beta))) <span class="sc">+</span></span>
<span id="cb26-7"><a href="#cb26-7"></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.2</span>, <span class="at">colour =</span> <span class="st">"indianred"</span>) <span class="sc">+</span></span>
<span id="cb26-8"><a href="#cb26-8"></a>  <span class="fu">xlab</span>(<span class="st">"Coastal section, west to east"</span>) <span class="sc">+</span></span>
<span id="cb26-9"><a href="#cb26-9"></a>  <span class="fu">ylab</span>(<span class="st">"True beta-diversity"</span>) <span class="sc">+</span></span>
<span id="cb26-10"><a href="#cb26-10"></a>  <span class="fu">theme_linedraw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display page-columns page-full">
<div id="fig-true-beta" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-true-beta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-03-biodiversity_files/figure-html/fig-true-beta-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;3: Whittaker’s true β-diversity shown in the seaweed data."><img src="Lab-03-biodiversity_files/figure-html/fig-true-beta-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-margin quarto-float-caption quarto-float-fig margin-caption" id="fig-true-beta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Whittaker’s true β-diversity shown in the seaweed data.
</figcaption>
</figure>
</div>
</div>
</div>
<p>The second measure of <span class="math inline">\(\beta\)</span>-diversity is <em>absolute species turnover</em>, and to calculate this, we subtract <span class="math inline">\(\alpha\)</span>-diversity for each section from the region’s <span class="math inline">\(\gamma\)</span>-diversity (<a href="#fig-abs-beta" class="quarto-xref">Figure&nbsp;4</a>):</p>
<div class="cell page-columns page-full">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a>abs_beta <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb27-2"><a href="#cb27-2"></a>  <span class="at">beta =</span> <span class="fu">ncol</span>(spp) <span class="sc">-</span> <span class="fu">specnumber</span>(spp, <span class="at">MARGIN =</span> <span class="dv">1</span>),</span>
<span id="cb27-3"><a href="#cb27-3"></a>  <span class="at">section_no =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">58</span>)</span>
<span id="cb27-4"><a href="#cb27-4"></a>)</span>
<span id="cb27-5"><a href="#cb27-5"></a><span class="co"># abs_beta</span></span>
<span id="cb27-6"><a href="#cb27-6"></a><span class="fu">ggplot</span>(<span class="at">data =</span> abs_beta, (<span class="fu">aes</span>(<span class="at">x =</span> section_no, <span class="at">y =</span> beta))) <span class="sc">+</span></span>
<span id="cb27-7"><a href="#cb27-7"></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.2</span>, <span class="at">colour =</span> <span class="st">"indianred"</span>) <span class="sc">+</span></span>
<span id="cb27-8"><a href="#cb27-8"></a>  <span class="fu">xlab</span>(<span class="st">"Coastal section, west to east"</span>) <span class="sc">+</span></span>
<span id="cb27-9"><a href="#cb27-9"></a>  <span class="fu">ylab</span>(<span class="st">"Absolute beta-diversity"</span>) <span class="sc">+</span></span>
<span id="cb27-10"><a href="#cb27-10"></a>  <span class="fu">theme_linedraw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display page-columns page-full">
<div id="fig-abs-beta" class="quarto-float quarto-figure quarto-figure-center anchored page-columns page-full">
<figure class="quarto-float quarto-float-fig figure page-columns page-full">
<div aria-describedby="fig-abs-beta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="Lab-03-biodiversity_files/figure-html/fig-abs-beta-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Figure&nbsp;4: Whittaker’s absolute species turnover shown in action in the seaweed data."><img src="Lab-03-biodiversity_files/figure-html/fig-abs-beta-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-margin quarto-float-caption quarto-float-fig margin-caption" id="fig-abs-beta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Whittaker’s absolute species turnover shown in action in the seaweed data.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="contemporary-definitions-beta-diversity" class="level4 page-columns page-full">
<h4 class="anchored" data-anchor-id="contemporary-definitions-beta-diversity">Contemporary Definitions <span class="math inline">\(\beta\)</span>-Diversity</h4>
<p>Contemporary definitions of <span class="math inline">\(\beta\)</span>-diversity rely on <a href="../BDC334/Lec-04-biodiversity.html#sec-dissimilarity-indices">pairwise <em>dissimilarity indices</em></a> such as Bray-Curtis, Jaccard, or Sørensen dissimilarities—see <span class="citation" data-cites="koleff2003measuring">Koleff et al. (<a href="#ref-koleff2003measuring" role="doc-biblioref">2003</a>)</span> for many more; also see <code>?vegdist</code>. However, discussing pairwise dissimilarities with <span class="math inline">\(\beta\)</span>-diversity makes more sense.</p>
<section id="dissimilarity-indices" class="level5">
<h5 class="anchored" data-anchor-id="dissimilarity-indices">Dissimilarity indices</h5>
<p>Dissimilarity indices are special cases of diversity indices that use pairwise comparisons between sampling units, habitats, or ecosystems.</p>
<p>Species dissimilarities result in pairwise matrices similar to the pairwise correlation or Euclidian distance matrices we have seen in <a href="../BDC334/Lec-01-introduction.html">Lab 1</a>. In <a href="../BDC334/Lec-04-biodiversity.html">Lab 2b</a> you will have also learned how to calculate these ecological distances in R. These dissimilarity indices are multivariate and compare between sites, sections, plots, etc., and must therefore not be confused with the univariate diversity indices.</p>
<p>We use the Bray-Curtis and Jaccard indices with abundance data and the Sørensen dissimilarity with presence-absence data. The seaweed dataset is a presence-absence dataset that necessitates using the Sørensen index. The interpretation of the resulting square (number of rows = number of columns) dissimilarity matrices is the same regardless of whether we calculate it for an abundance or presence-absence dataset. The values in the matrix range from 0 to 1. A 0 means that the pair of sites we compare is identical (all species in common) but 1 means they are completely different (no species in common). In the square dissimilarity matrix, the diagonal is 0, which essentially (and obviously) means that any site is identical to itself. Elsewhere the values will range from 0 to 1. Since this is a pairwise calculation (each site compared to every other site), our seaweed dataset will contain (58 × (58 - 1))/2 = 1653 values, each one ranging from 0 to 1.</p>
<p>The first step involves the species table, <span class="math inline">\(Y\)</span>. First, we compute the Sørensen dissimilarity index, <span class="math inline">\(\beta_{\text{sør}}\)</span>, to compare the dissimilarity of all pairs of coastal sections using presence-absence data. The dissimilarity in species composition between two sections is calculated from three parameters, <em>viz</em>., <em>b</em> and <em>c</em>, which represent the number of species unique to each of the two sites, and <em>a</em>, the number of species in common between them. It is given by:</p>
<p><span class="math display">\[\beta_\text{sør}=\frac{2a}{2a+b+c}\]</span> Where <span class="math inline">\(a\)</span> is the number of species in common between two sites, and <span class="math inline">\(b\)</span> and <span class="math inline">\(c\)</span> are the number of species unique to each site. The Sørensen dissimilarity index ranges from 0 to 1, where 0 means that the pair of sites we compare is identical (all species in common) and 1 means they are completely different (no species in common).</p>
<p>The <strong>vegan</strong> function <code>vegdist()</code> provides access to the dissimilarity indices. We calculate the Sørensen dissimilarity index:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a>sor <span class="ot">&lt;-</span> <span class="fu">vegdist</span>(spp, <span class="at">binary =</span> <span class="cn">TRUE</span>) <span class="co"># makes the lower triangle matrix</span></span>
<span id="cb28-2"><a href="#cb28-2"></a>sor_df <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">as.matrix</span>(sor), <span class="dv">4</span>)</span>
<span id="cb28-3"><a href="#cb28-3"></a><span class="fu">dim</span>(sor_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 58 58</code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a>sor_df[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>] <span class="co"># the first 10 rows and columns</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        1      2      3      4      5      6      7      8      9     10
1  0.0000 0.0036 0.0036 0.0072 0.0249 0.0391 0.0391 0.0459 0.0592 0.0629
2  0.0036 0.0000 0.0000 0.0036 0.0213 0.0355 0.0355 0.0423 0.0556 0.0592
3  0.0036 0.0000 0.0000 0.0036 0.0213 0.0355 0.0355 0.0423 0.0556 0.0592
4  0.0072 0.0036 0.0036 0.0000 0.0177 0.0318 0.0318 0.0386 0.0519 0.0556
5  0.0249 0.0213 0.0213 0.0177 0.0000 0.0140 0.0140 0.0208 0.0342 0.0378
6  0.0391 0.0355 0.0355 0.0318 0.0140 0.0000 0.0000 0.0069 0.0205 0.0241
7  0.0391 0.0355 0.0355 0.0318 0.0140 0.0000 0.0000 0.0069 0.0205 0.0241
8  0.0459 0.0423 0.0423 0.0386 0.0208 0.0069 0.0069 0.0000 0.0136 0.0171
9  0.0592 0.0556 0.0556 0.0519 0.0342 0.0205 0.0205 0.0136 0.0000 0.0034
10 0.0629 0.0592 0.0592 0.0556 0.0378 0.0241 0.0241 0.0171 0.0034 0.0000</code></pre>
</div>
</div>
<p>What we see above is a square dissimilarity matrix. The most important characteristics of the matrix are:</p>
<ol type="i">
<li>whereas the raw species data, <span class="math inline">\(Y\)</span>, is rectangular (number rows ≠ number columns), the dissimilarity matrix is square (number rows = number columns);</li>
<li>the diagonal is filled with 0;</li>
<li>the matrix is symmetrical—it is comprised of symetrical upper and lower triangles.</li>
</ol>
<p>Create a <code>data.frame</code> suitable for plotting:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a>sor_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">round</span>(<span class="fu">as.matrix</span>(sor), <span class="dv">4</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Lab 3
</div>
</div>
<div class="callout-body-container callout-body">
<p>(To be reviewed by BCB743 student but not for marks)</p>
<p>These questions concern matrices produced from species data using any of the indices available in <code>vegdist()</code>:</p>
<ol start="3" type="1">
<li>Why is the matrix square, and what determines the number of rows/columns?</li>
<li>What is the meaning of the diagonal?</li>
<li>What is the meaning of the non-diagonal elements?</li>
<li>Referring to the seaweed species data specifically, take the data in row 1 or column 1 and create a line graph showing these values as a function of the section number.</li>
<li>Provide a mechanistic (ecological) explanation for why this figure takes the shape that it does. Which community assembly process does this hint at? <!-- > Answer Q4: --></li>
</ol>
<!-- ```{r} -->
<!-- ggplot(data = sor_df, (aes(x = 1:58, y = X1))) + -->
<!--   geom_line(size = 1.2, colour = "indianred") + -->
<!--   xlab("Coastal section, west to east") + -->
<!--   ylab("Total beta-diversity") + -->
<!--   theme_linedraw() -->
<!-- ``` -->
</div>
</div>
<p>There are different interpretations linked to <span class="math inline">\(\beta\)</span>-diversity, each telling us something different about community formation processes.</p>
</section>
<section id="species-turnover-and-nestedness-resultant-beta-diversity" class="level5 page-columns page-full">
<h5 class="anchored" data-anchor-id="species-turnover-and-nestedness-resultant-beta-diversity">Species turnover and nestedness-resultant <span class="math inline">\(\beta\)</span>-diversity</h5>
<p>There are two kinds of <span class="math inline">\(\beta\)</span>-diversity: <a href="../BDC334/Lec-04-biodiversity.html#sec-beta-diversity">species turnover and nestedness-resultant <span class="math inline">\(\beta\)</span>-diversity</a>. The former is the result of species replacement between sites, whereas the latter is the result of species loss or gain between sites. The Sørensen dissimilarity index, <span class="math inline">\(\beta_\text{sør}\)</span>, can be decomposed into these two components.</p>
<p>How do we calculate the turnover and nestedness-resultant components of <span class="math inline">\(\beta\)</span>-diversity? The <strong>betapart</strong> package <span class="citation" data-cites="baselga2022betapart">(<a href="#ref-baselga2022betapart" role="doc-biblioref">Baselga et al. 2022</a>)</span> comes to the rescue. We decompose the dissimilarity into the <span class="math inline">\(\beta_\text{sim}\)</span> and <span class="math inline">\(\beta_\text{sne}\)</span> components <span class="citation" data-cites="baselga2010partitioning">(<a href="#ref-baselga2010partitioning" role="doc-biblioref">Baselga 2010</a>)</span> using the <code>betapart.core()</code> and <code>betapart.pair()</code> functions. The outcomes of this partitioning calculation are placed into the matrices <span class="math inline">\(Y1\)</span> and <span class="math inline">\(Y2\)</span>. These data can then be analysed further—e.g.&nbsp;we can apply a principal components analysis (PCA) or another multivariate analysis on <span class="math inline">\(Y\)</span> to find the major patterns in the community data—we will do this in BCB743.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a><span class="co"># Decompose total Sørensen dissimilarity into turnover and nestedness-resultant</span></span>
<span id="cb33-2"><a href="#cb33-2"></a><span class="co"># components:</span></span>
<span id="cb33-3"><a href="#cb33-3"></a>Y.core <span class="ot">&lt;-</span> <span class="fu">betapart.core</span>(spp)</span>
<span id="cb33-4"><a href="#cb33-4"></a>Y.pair <span class="ot">&lt;-</span> <span class="fu">beta.pair</span>(Y.core, <span class="at">index.family =</span> <span class="st">"sor"</span>)</span>
<span id="cb33-5"><a href="#cb33-5"></a></span>
<span id="cb33-6"><a href="#cb33-6"></a><span class="co"># Let Y1 be the turnover component (beta-sim):</span></span>
<span id="cb33-7"><a href="#cb33-7"></a>Y1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">round</span>(<span class="fu">as.matrix</span>(Y.pair<span class="sc">$</span>beta.sim), <span class="dv">3</span>))</span>
<span id="cb33-8"><a href="#cb33-8"></a></span>
<span id="cb33-9"><a href="#cb33-9"></a><span class="co"># Let Y2 be the nestedness-resultant component (beta-sne):</span></span>
<span id="cb33-10"><a href="#cb33-10"></a>Y2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">round</span>(<span class="fu">as.matrix</span>(Y.pair<span class="sc">$</span>beta.sne), <span class="dv">3</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>A portion of the turnover component matrix:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a>Y1[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>      X1    X2    X3    X4    X5    X6    X7    X8    X9   X10
1  0.000 0.000 0.000 0.000 0.007 0.022 0.022 0.022 0.022 0.029
2  0.000 0.000 0.000 0.000 0.007 0.022 0.022 0.022 0.022 0.029
3  0.000 0.000 0.000 0.000 0.007 0.022 0.022 0.022 0.022 0.029
4  0.000 0.000 0.000 0.000 0.007 0.021 0.021 0.021 0.021 0.029
5  0.007 0.007 0.007 0.007 0.000 0.014 0.014 0.014 0.014 0.021
6  0.022 0.022 0.022 0.021 0.014 0.000 0.000 0.000 0.000 0.007
7  0.022 0.022 0.022 0.021 0.014 0.000 0.000 0.000 0.000 0.007
8  0.022 0.022 0.022 0.021 0.014 0.000 0.000 0.000 0.000 0.007
9  0.022 0.022 0.022 0.021 0.014 0.000 0.000 0.000 0.000 0.000
10 0.029 0.029 0.029 0.029 0.021 0.007 0.007 0.007 0.000 0.000</code></pre>
</div>
</div>
<p>A portion of the nestedness-resultant matrix:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a>Y2[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>      X1    X2    X3    X4    X5    X6    X7    X8    X9   X10
1  0.000 0.004 0.004 0.007 0.018 0.017 0.017 0.024 0.037 0.034
2  0.004 0.000 0.000 0.004 0.014 0.014 0.014 0.021 0.034 0.030
3  0.004 0.000 0.000 0.004 0.014 0.014 0.014 0.021 0.034 0.030
4  0.007 0.004 0.004 0.000 0.011 0.010 0.010 0.017 0.030 0.027
5  0.018 0.014 0.014 0.011 0.000 0.000 0.000 0.007 0.020 0.017
6  0.017 0.014 0.014 0.010 0.000 0.000 0.000 0.007 0.021 0.017
7  0.017 0.014 0.014 0.010 0.000 0.000 0.000 0.007 0.021 0.017
8  0.024 0.021 0.021 0.017 0.007 0.007 0.007 0.000 0.014 0.010
9  0.037 0.034 0.034 0.030 0.020 0.021 0.021 0.014 0.000 0.003
10 0.034 0.030 0.030 0.027 0.017 0.017 0.017 0.010 0.003 0.000</code></pre>
</div>
</div>
<p>A portion of the nestedness-resultant matrix reformatted as a <code>tibble()</code><a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>:</p>
<div class="no-row-height column-margin column-container"><div id="fn2"><p><sup>2</sup>&nbsp;Note that the rows are no longer numbered in the tibble view, but it can easily be recreated by <code>seq(1:58)</code>.</p></div></div><div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1"></a>Y2_tib <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(Y2)</span>
<span id="cb38-2"><a href="#cb38-2"></a><span class="fu">head</span>(Y2_tib)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 58
     X1    X2    X3    X4    X5    X6    X7    X8    X9   X10   X11   X12   X13
  &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
1 0     0.004 0.004 0.007 0.018 0.017 0.017 0.024 0.037 0.034 0.069 0.078 0.196
2 0.004 0     0     0.004 0.014 0.014 0.014 0.021 0.034 0.03  0.065 0.074 0.193
3 0.004 0     0     0.004 0.014 0.014 0.014 0.021 0.034 0.03  0.065 0.074 0.193
4 0.007 0.004 0.004 0     0.011 0.01  0.01  0.017 0.03  0.027 0.062 0.071 0.19 
5 0.018 0.014 0.014 0.011 0     0     0     0.007 0.02  0.017 0.052 0.061 0.181
6 0.017 0.014 0.014 0.01  0     0     0     0.007 0.021 0.017 0.053 0.062 0.184
# ℹ 45 more variables: X14 &lt;dbl&gt;, X15 &lt;dbl&gt;, X16 &lt;dbl&gt;, X17 &lt;dbl&gt;, X18 &lt;dbl&gt;,
#   X19 &lt;dbl&gt;, X20 &lt;dbl&gt;, X21 &lt;dbl&gt;, X22 &lt;dbl&gt;, X23 &lt;dbl&gt;, X24 &lt;dbl&gt;,
#   X25 &lt;dbl&gt;, X26 &lt;dbl&gt;, X27 &lt;dbl&gt;, X28 &lt;dbl&gt;, X29 &lt;dbl&gt;, X30 &lt;dbl&gt;,
#   X31 &lt;dbl&gt;, X32 &lt;dbl&gt;, X33 &lt;dbl&gt;, X34 &lt;dbl&gt;, X35 &lt;dbl&gt;, X36 &lt;dbl&gt;,
#   X37 &lt;dbl&gt;, X38 &lt;dbl&gt;, X39 &lt;dbl&gt;, X40 &lt;dbl&gt;, X41 &lt;dbl&gt;, X42 &lt;dbl&gt;,
#   X43 &lt;dbl&gt;, X44 &lt;dbl&gt;, X45 &lt;dbl&gt;, X46 &lt;dbl&gt;, X47 &lt;dbl&gt;, X48 &lt;dbl&gt;,
#   X49 &lt;dbl&gt;, X50 &lt;dbl&gt;, X51 &lt;dbl&gt;, X52 &lt;dbl&gt;, X53 &lt;dbl&gt;, X54 &lt;dbl&gt;, …</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Lab 3 (continue)
</div>
</div>
<div class="callout-body-container callout-body">
<p>(To be reviewed by BCB743 student but not for marks)</p>
<ol start="8" type="1">
<li>Plot species turnover as a function of Section number, and provide a mechanistic explanation for the pattern observed.</li>
<li>Based on an assessment of literature on the topic, provide a discussion of nestedness-resultant <span class="math inline">\(\beta\)</span>-diversity. Use either a marine or terrestrial example to explain this mode of structuring biodiversity (i.e.&nbsp;assembly of species into a community). <!-- > Answer Q8:--></li>
</ol>
<!-- ```{r} -->
<!-- # The turnover component -->
<!-- ggplot(data = Y1, (aes(x = 1:58, y = X1))) + -->
<!--   geom_line(size = 1.2, colour = "indianred") + -->
<!--   xlab("Coastal section, west to east") + -->
<!--   ylab("Species richness") + -->
<!--   theme_linedraw() -->
<!-- ``` -->
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Submission Instructions
</div>
</div>
<div class="callout-body-container callout-body">
<p>The Lab 3 assignment is due at <strong>07:00 on Monday 12 August 2022</strong>.</p>
<p>Provide a <strong>neat and thoroughly annotated</strong> R file which can recreate all the graphs and all calculations. Written answers must be typed in the same file as comments.</p>
<p>Please label the R file as follows:</p>
<ul>
<li><code>BDC334_&lt;first_name&gt;_&lt;last_name&gt;_Lab_3.R</code></li>
</ul>
<p>(the <code>&lt;</code> and <code>&gt;</code> must be omitted as they are used in the example as field indicators only).</p>
<p>Submit your appropriately named R documents on iKamva when ready.</p>
<p>Failing to follow these instructions carefully, precisely, and thoroughly will cause you to lose marks, which could cause a significant drop in your score as formatting counts for 15% of the final mark (out of 100%).</p>
</div>
</div>


<!-- -->


</section>
</section>
</section>
</section>


<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-baselga2010partitioning" class="csl-entry" role="listitem">
Baselga A (2010) Partitioning the turnover and nestedness components of beta diversity. <span>Global Ecology and Biogeography</span> 19:134–143.
</div>
<div id="ref-baselga2022betapart" class="csl-entry" role="listitem">
Baselga A, Orme D, Villeger S, De Bortoli J, Leprieur F, Logez M (2022) <a href="https://CRAN.R-project.org/package=betapart"><span class="nocase">betapart: Partitioning Beta Diversity into Turnover and Nestedness Components</span></a>.
</div>
<div id="ref-bolton1986marine" class="csl-entry" role="listitem">
Bolton J (1986) Marine phytogeography of the <span>B</span>enguela upwelling region on the west coast of southern africa: A temperature dependent approach.
</div>
<div id="ref-bolton2002seaweed" class="csl-entry" role="listitem">
Bolton J, Stegenga H (2002) <span class="nocase">Seaweed species diversity in South Africa</span>. <span>South African Journal of Marine Science</span> 24:9–18.
</div>
<div id="ref-de2005guide" class="csl-entry" role="listitem">
De Clerck O, Bolton J, Anderson R, Coppejans E, Bolton J, Anderson R (2005) <span class="nocase">Guide to the seaweeds of KwaZulu-Natal</span>.
</div>
<div id="ref-jurasinski2009inventory" class="csl-entry" role="listitem">
Jurasinski G, Retzer V, Beierkuhnlein C (2009) Inventory, differentiation, and proportional diversity: A consistent terminology for quantifying species diversity. Oecologia 159:15–26.
</div>
<div id="ref-koleff2003measuring" class="csl-entry" role="listitem">
Koleff P, Gaston KJ, Lennon JJ (2003) Measuring beta diversity for presence–absence data. <span>Journal of Animal Ecology</span> 72:367–382.
</div>
<div id="ref-oksanen2022vegan" class="csl-entry" role="listitem">
Oksanen J, Simpson GL, Blanchet FG, Kindt R, Legendre P, Minchin PR, O’Hara RB, Solymos P, Stevens MHH, Szoecs E, Wagner H, Barbour M, Bedward M, Bolker B, Borcard D, Carvalho G, Chirico M, De Caceres M, Durand S, Evangelista HBA, FitzJohn R, Friendly M, Furneaux B, Hannigan G, Hill MO, Lahti L, McGlinn D, Ouellette M-H, Ribeiro Cunha E, Smith T, Stier A, Ter Braak CJF, Weedon J (2022) <a href="https://CRAN.R-project.org/package=vegan"><span class="nocase">vegan: Community Ecology Package</span></a>.
</div>
<div id="ref-smit2013coastal" class="csl-entry" role="listitem">
Smit AJ, Roberts M, Anderson RJ, Dufois F, Dudley SF, Bornman TG, Olbers J, Bolton JJ (2013) <span class="nocase">A coastal seawater temperature dataset for biogeographical studies: large biases between in situ and remotely-sensed data sets around the coast of South Africa</span>. <span>PLoS One</span> 8:e81944.
</div>
<div id="ref-smit2017seaweeds" class="csl-entry" role="listitem">
Smit AJ, Bolton JJ, Anderson RJ (2017) Seaweeds in two oceans: Beta-diversity. <span>Frontiers in Marine Science</span> 4:404.
</div>
<div id="ref-stegenga1997seaweeds" class="csl-entry" role="listitem">
Stegenga H, Bolton JJ, Anderson RJ (1997) <span class="nocase">Seaweeds of the South African west coast</span>.
</div>
<div id="ref-whittaker1960vegetation" class="csl-entry" role="listitem">
Whittaker RH (1960) <span class="nocase">Vegetation of the Siskiyou mountains, Oregon and California</span>. <span>Ecological Monographs</span> 30:279–338.
</div>
<div id="ref-whittaker1972evolution" class="csl-entry" role="listitem">
Whittaker RH (1972) Evolution and measurement of species diversity. Taxon 21:213–251.
</div>
</div></section><section class="quarto-appendix-contents" id="quarto-reuse"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a></div></div></section><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{smit,_a._j.2021,
  author = {Smit, A. J.,},
  title = {Lab 3. {Quantifying} {Biodiversity}},
  date = {2021-01-01},
  url = {http://tangledbank.netlify.app/BDC334/Lab-03-biodiversity.html},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-smit,_a._j.2021" class="csl-entry quarto-appendix-citeas" role="listitem">
Smit, A. J. (2021) Lab 3. Quantifying Biodiversity. <a href="http://tangledbank.netlify.app/BDC334/Lab-03-biodiversity.html">http://tangledbank.netlify.app/BDC334/Lab-03-biodiversity.html.
</a>
</div></div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("http:\/\/tangledbank\.netlify\.app");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
            // default icon
            link.classList.add("external");
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../BDC334/Lab-02b-env_dist.html" class="pagination-link" aria-label="Lab 2b. Environmental Distance">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Lab 2b. Environmental Distance</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../BDC334/Lab-04-biodiversity.html" class="pagination-link" aria-label="Lab 4. Species Distribution Patterns">
        <span class="nav-page-text">Lab 4. Species Distribution Patterns</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb40" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb40-1"><a href="#cb40-1"></a><span class="co">---</span></span>
<span id="cb40-2"><a href="#cb40-2"></a><span class="an">date:</span><span class="co"> "2021-01-01"</span></span>
<span id="cb40-3"><a href="#cb40-3"></a><span class="an">title:</span><span class="co"> "Lab 3. Quantifying Biodiversity"</span></span>
<span id="cb40-4"><a href="#cb40-4"></a><span class="co">---</span></span>
<span id="cb40-5"><a href="#cb40-5"></a></span>
<span id="cb40-6"><a href="#cb40-6"></a>::: callout-note</span>
<span id="cb40-7"><a href="#cb40-7"></a><span class="fu">## BCB743</span></span>
<span id="cb40-8"><a href="#cb40-8"></a></span>
<span id="cb40-9"><a href="#cb40-9"></a>**This material must be reviewed by BCB743 students in Week 1 of Quantitative Ecology.**</span>
<span id="cb40-10"><a href="#cb40-10"></a>:::</span>
<span id="cb40-11"><a href="#cb40-11"></a></span>
<span id="cb40-12"><a href="#cb40-12"></a>::: callout-tip</span>
<span id="cb40-13"><a href="#cb40-13"></a><span class="fu">## This Lab Accompanies the Following Lecture</span></span>
<span id="cb40-14"><a href="#cb40-14"></a></span>
<span id="cb40-15"><a href="#cb40-15"></a><span class="ss">- </span><span class="co">[</span><span class="ot">Lecture 4: Biodiversity Concepts</span><span class="co">](Lec-04-biodiversity.qmd)</span></span>
<span id="cb40-16"><a href="#cb40-16"></a>:::</span>
<span id="cb40-17"><a href="#cb40-17"></a></span>
<span id="cb40-18"><a href="#cb40-18"></a>::: callout-tip</span>
<span id="cb40-19"><a href="#cb40-19"></a><span class="fu">## Data For This Lab</span></span>
<span id="cb40-20"><a href="#cb40-20"></a></span>
<span id="cb40-21"><a href="#cb40-21"></a>The seaweed <span class="co">[</span><span class="ot">@smit2017seaweeds</span><span class="co">]</span> and toy data are at the links below:</span>
<span id="cb40-22"><a href="#cb40-22"></a></span>
<span id="cb40-23"><a href="#cb40-23"></a><span class="ss">- </span>The seaweed species data -- <span class="co">[</span><span class="ot">`SeaweedSpp.csv`</span><span class="co">](../data/seaweed/SeaweedSpp.csv)</span></span>
<span id="cb40-24"><a href="#cb40-24"></a><span class="ss">- </span>The seaweed environmental data -- <span class="co">[</span><span class="ot">`SeaweedEnv.csv`</span><span class="co">](../data/seaweed/SeaweedEnv.csv)</span></span>
<span id="cb40-25"><a href="#cb40-25"></a><span class="ss">- </span>The seaweed coastal sections -- <span class="co">[</span><span class="ot">`SeaweedSites.csv`</span><span class="co">](../data/seaweed/SeaweedSites.csv)</span></span>
<span id="cb40-26"><a href="#cb40-26"></a><span class="ss">- </span>The fictitious light data <span class="co">[</span><span class="ot">`light_levels.csv`</span><span class="co">](../data/light_levels.csv)</span></span>
<span id="cb40-27"><a href="#cb40-27"></a>:::</span>
<span id="cb40-28"><a href="#cb40-28"></a></span>
<span id="cb40-29"><a href="#cb40-29"></a></span>
<span id="cb40-30"><a href="#cb40-30"></a></span>
<span id="cb40-31"><a href="#cb40-31"></a><span class="at">&gt; ***Biodiversity** The variability among living organisms from all</span></span>
<span id="cb40-32"><a href="#cb40-32"></a><span class="at">&gt; sources including,* inter alia*, terrestrial, marine and other aquatic</span></span>
<span id="cb40-33"><a href="#cb40-33"></a><span class="at">&gt; ecosystems and the ecological complexes of which they are part; this</span></span>
<span id="cb40-34"><a href="#cb40-34"></a><span class="at">&gt; includes diversity within species, between species and of ecosystems.*</span></span>
<span id="cb40-35"><a href="#cb40-35"></a><span class="at">&gt;</span></span>
<span id="cb40-36"><a href="#cb40-36"></a><span class="at">&gt; --- International Union for the Conservation of Nature (IUCN), Convention on Biological Diversity</span></span>
<span id="cb40-37"><a href="#cb40-37"></a></span>
<span id="cb40-38"><a href="#cb40-38"></a>The IUCN definition considers a diversity of diversity concepts. This module looks at diversity only at the species level (species diversity). However, we can also approach macroecological problems from phylogenetic and functional (and other) diversity concepts of view. Functional and phylogenetic diversity ideas will be introduced in the <span class="co">[</span><span class="ot">BDC743 module Quantitative Ecology</span><span class="co">](../BCB743/BCB743_index.qmd)</span>.</span>
<span id="cb40-39"><a href="#cb40-39"></a></span>
<span id="cb40-40"><a href="#cb40-40"></a><span class="fu">## Preparation</span></span>
<span id="cb40-41"><a href="#cb40-41"></a></span>
<span id="cb40-42"><a href="#cb40-42"></a><span class="fu">### The South African Seaweed Data</span></span>
<span id="cb40-43"><a href="#cb40-43"></a></span>
<span id="cb40-44"><a href="#cb40-44"></a>In these examples, we will use the seaweed data of @smit2017seaweeds.</span>
<span id="cb40-45"><a href="#cb40-45"></a>Please make sure that you read this paper. An additional file describing the background to the data is available <span class="co">[</span><span class="ot">here</span><span class="co">](../docs/Smit_the_seaweed_data.pdf)</span> (@fig-seaweed-sections).</span>
<span id="cb40-46"><a href="#cb40-46"></a></span>
<span id="cb40-47"><a href="#cb40-47"></a><span class="al">![The coastal sections and associated seawater temperature profile associated with the study by @smit2017seaweeds.](../images/Seaweed_sections.png)</span>{#fig-seaweed-sections fig-align="center"}</span>
<span id="cb40-48"><a href="#cb40-48"></a></span>
<span id="cb40-49"><a href="#cb40-49"></a>One of the datasets, $Y$ (in the file <span class="in">`SeaweedSpp.csv`</span>),</span>
<span id="cb40-50"><a href="#cb40-50"></a>comprises updated distribution records of 847 macroalgal species within each of 58 × 50 km-long sections of the South African coast <span class="co">[</span><span class="ot">@bolton2002seaweed</span><span class="co">]</span>. The dataset captures *ca*. 90% of the known seaweed flora of South Africa, but excludes some very small and/or very rare species for which data are insufficient. The data are from verifiable literature sources and John Bolton and Rob Anderson's collections, assembled from information collected by teams of phycologists over three decades [@bolton1986marine; @stegenga1997seaweeds; @bolton2002seaweed; @de2005guide]. Another file, $E$ (in `env.csv`), is a dataset of *in situ* coastal seawater temperatures derived from daily</span>
<span id="cb40-51"><a href="#cb40-51"></a>measurements over 40 years <span class="co">[</span><span class="ot">@smit2013coastal</span><span class="co">]</span>.</span>
<span id="cb40-52"><a href="#cb40-52"></a></span>
<span id="cb40-53"><a href="#cb40-53"></a><span class="fu">### Setting Up the Analysis Environment</span></span>
<span id="cb40-54"><a href="#cb40-54"></a></span>
<span id="cb40-55"><a href="#cb40-55"></a>We will use **R**, so first, we must find, install and load various packages. Some packages will be available on CRAN and can be accessed and installed the usual way, but you will need to download others from <span class="co">[</span><span class="ot">R Forge</span><span class="co">](https://r-forge.r-project.org/R/?group_id=195)</span>.</span>
<span id="cb40-56"><a href="#cb40-56"></a></span>
<span id="cb40-59"><a href="#cb40-59"></a><span class="in">```{r}</span></span>
<span id="cb40-60"><a href="#cb40-60"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb40-61"><a href="#cb40-61"></a><span class="fu">library</span>(vegan)</span>
<span id="cb40-62"><a href="#cb40-62"></a><span class="fu">library</span>(betapart)</span>
<span id="cb40-63"><a href="#cb40-63"></a><span class="fu">library</span>(BiodiversityR) <span class="co"># this package may at times be problematic to install</span></span>
<span id="cb40-64"><a href="#cb40-64"></a><span class="in">```</span></span>
<span id="cb40-65"><a href="#cb40-65"></a></span>
<span id="cb40-66"><a href="#cb40-66"></a><span class="fu">### A Look at the Data</span></span>
<span id="cb40-67"><a href="#cb40-67"></a></span>
<span id="cb40-68"><a href="#cb40-68"></a>Let's load the data and see how it is structured:</span>
<span id="cb40-69"><a href="#cb40-69"></a></span>
<span id="cb40-72"><a href="#cb40-72"></a><span class="in">```{r}</span></span>
<span id="cb40-73"><a href="#cb40-73"></a>spp <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">'../data/seaweed/SeaweedSpp.csv'</span>)</span>
<span id="cb40-74"><a href="#cb40-74"></a>spp <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(spp, <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb40-75"><a href="#cb40-75"></a></span>
<span id="cb40-76"><a href="#cb40-76"></a><span class="co"># Lets look at the data:</span></span>
<span id="cb40-77"><a href="#cb40-77"></a><span class="fu">dim</span>(spp)</span>
<span id="cb40-78"><a href="#cb40-78"></a><span class="in">```</span></span>
<span id="cb40-79"><a href="#cb40-79"></a></span>
<span id="cb40-80"><a href="#cb40-80"></a>We see that our dataset has 58 rows and 847 columns. What is in the</span>
<span id="cb40-81"><a href="#cb40-81"></a>columns and rows? Start with the first five rows and five columns:</span>
<span id="cb40-82"><a href="#cb40-82"></a></span>
<span id="cb40-85"><a href="#cb40-85"></a><span class="in">```{r}</span></span>
<span id="cb40-86"><a href="#cb40-86"></a>spp[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb40-87"><a href="#cb40-87"></a><span class="in">```</span></span>
<span id="cb40-88"><a href="#cb40-88"></a></span>
<span id="cb40-89"><a href="#cb40-89"></a>Now the last five rows and five columns:</span>
<span id="cb40-90"><a href="#cb40-90"></a></span>
<span id="cb40-93"><a href="#cb40-93"></a><span class="in">```{r}</span></span>
<span id="cb40-94"><a href="#cb40-94"></a>spp[(<span class="fu">nrow</span>(spp) <span class="sc">-</span> <span class="dv">5</span>)<span class="sc">:</span><span class="fu">nrow</span>(spp), (<span class="fu">ncol</span>(spp) <span class="sc">-</span> <span class="dv">5</span>)<span class="sc">:</span><span class="fu">ncol</span>(spp)]</span>
<span id="cb40-95"><a href="#cb40-95"></a><span class="in">```</span></span>
<span id="cb40-96"><a href="#cb40-96"></a></span>
<span id="cb40-97"><a href="#cb40-97"></a>So, each row corresponds to a site (i.e. each of the coastal sections), and each column contains a species. We arrange the species alphabetically and use a six-letter code to identify them.</span>
<span id="cb40-98"><a href="#cb40-98"></a></span>
<span id="cb40-99"><a href="#cb40-99"></a><span class="fu">## Species Data</span></span>
<span id="cb40-100"><a href="#cb40-100"></a></span>
<span id="cb40-101"><a href="#cb40-101"></a>When ecologists talk about species diversity, they typically consider the characteristics of biological communities in a specific habitat, ecological community, or ecosystem. Species diversity considers three essential concepts about how species are distributed in space: their **richness**, **abundance**, and **evenness**. We can express each of these as biodiversity metrics that allow us to compare communities in space and time.</span>
<span id="cb40-102"><a href="#cb40-102"></a></span>
<span id="cb40-103"><a href="#cb40-103"></a>When ecologists talk about 'biodiversity', they might not necessarily be</span>
<span id="cb40-104"><a href="#cb40-104"></a>interested in *all* the plants and animals and things that are neither</span>
<span id="cb40-105"><a href="#cb40-105"></a>plant nor animal that occur at a particular place. Some ecologists are</span>
<span id="cb40-106"><a href="#cb40-106"></a>interested in ants and moths. Others might find fish more insightful.</span>
<span id="cb40-107"><a href="#cb40-107"></a>Some even like marine mammals! I prefer seaweed. The analysis of</span>
<span id="cb40-108"><a href="#cb40-108"></a>biodiversity data might often be constrained to some higher-level taxon,</span>
<span id="cb40-109"><a href="#cb40-109"></a>such as all angiosperms in a landscape, reptiles, etc. (but we sample all</span>
<span id="cb40-110"><a href="#cb40-110"></a>species in the higher-level taxon). Some ecological</span>
<span id="cb40-111"><a href="#cb40-111"></a>questions benefit from comparisons of diversity assessments among</span>
<span id="cb40-112"><a href="#cb40-112"></a>selected taxa (avifauna vs small mammals, for example), as this focus</span>
<span id="cb40-113"><a href="#cb40-113"></a>might address some particular ecological hypothesis. The bird vs small</span>
<span id="cb40-114"><a href="#cb40-114"></a>mammal comparison might reveal how barriers such as</span>
<span id="cb40-115"><a href="#cb40-115"></a>streams and rivers structure biodiversity patterns. In our examples, we</span>
<span id="cb40-116"><a href="#cb40-116"></a>will use such focused datasets.</span>
<span id="cb40-117"><a href="#cb40-117"></a></span>
<span id="cb40-118"><a href="#cb40-118"></a>Here we look at the various measures of biodiversity, viz. $\alpha$-, $\gamma$- and $\beta$-diversity. David Zelený, in his <span class="co">[</span><span class="ot">Analysis of community data in R</span><span class="co">](https://www.davidzeleny.net/anadat-r/doku.php/en:diversity_analysis)</span>, provides deeper analysis and compulsory reading. </span>
<span id="cb40-119"><a href="#cb40-119"></a></span>
<span id="cb40-120"><a href="#cb40-120"></a><span class="co">&lt;!--- # Topic 2: Measures of biodiversity ---&gt;</span></span>
<span id="cb40-121"><a href="#cb40-121"></a></span>
<span id="cb40-122"><a href="#cb40-122"></a><span class="fu">## Three Measures of Biodiversity: $\alpha$-, $\gamma$-, $\beta$-Diversity</span></span>
<span id="cb40-123"><a href="#cb40-123"></a></span>
<span id="cb40-124"><a href="#cb40-124"></a>@whittaker1972evolution coined three measures of biodiversity, and the concepts were ‘modernised’ by @jurasinski2009inventory. The concepts</span>
<span id="cb40-125"><a href="#cb40-125"></a>represent the measurement of biodiversity *across different spatial</span>
<span id="cb40-126"><a href="#cb40-126"></a>scales*. $\alpha$- and $\gamma$-diversity express the total number of species in an area. The first, $\alpha$-diversity, represents the number of species at the small (local) scale, such as, for example, within a sampling unit like a quadrat, transect, plot, or trawl. Alternatively, maybe the research question represents the local scale by several sampling units nesting within a small patch of landscape and defines the mean species richness within this patch as local. Multiples (sampling units or patches) are nested within a larger region (or ecosystem) and serve as replicates. The complete number of species across all of these replicates indicates the diversity at a larger scale---this is called $\gamma$-diversity. $\beta$-diversity refers to the change in species composition among samples (sites).</span>
<span id="cb40-127"><a href="#cb40-127"></a></span>
<span id="cb40-128"><a href="#cb40-128"></a>By now, you will have received a brief [Introduction to</span>
<span id="cb40-129"><a href="#cb40-129"></a>R](Lab-02a-r_rstudio.qmd), and we can proceed with looking at</span>
<span id="cb40-130"><a href="#cb40-130"></a>some of the measures of biodiversity. We will start by using data on the</span>
<span id="cb40-131"><a href="#cb40-131"></a>seaweeds of South Africa to demonstrate some ideas around diversity</span>
<span id="cb40-132"><a href="#cb40-132"></a>measures. The **vegan**^<span class="co">[</span><span class="ot">I am by no means an advocate for veganism.</span><span class="co">]</span> (for *vegetation analysis*) package <span class="co">[</span><span class="ot">@oksanen2022vegan</span><span class="co">]</span> offers [various</span>
<span id="cb40-133"><a href="#cb40-133"></a>functions](../docs/Oksanen_diversity-vegan.pdf) to calculate diversity</span>
<span id="cb40-134"><a href="#cb40-134"></a>indices. I will demonstrate some of these functions below.</span>
<span id="cb40-135"><a href="#cb40-135"></a></span>
<span id="cb40-136"><a href="#cb40-136"></a><span class="fu">### Alpha-Diversity</span></span>
<span id="cb40-137"><a href="#cb40-137"></a></span>
<span id="cb40-138"><a href="#cb40-138"></a>We can represent $\alpha$-diversity in three ways:</span>
<span id="cb40-139"><a href="#cb40-139"></a></span>
<span id="cb40-140"><a href="#cb40-140"></a><span class="ss">1. </span>as *species richness*, $S$;</span>
<span id="cb40-141"><a href="#cb40-141"></a><span class="ss">2. </span>as a *univariate diversity index*, such as the $\alpha$ parameter of <span class="co">[</span><span class="ot">Fisher’s log-series</span><span class="co">](04-biodiversity.html#species-abundance-distribution)</span>, Shannon diversity, $H'$, Simpson’s diversity, $\lambda$; or</span>
<span id="cb40-142"><a href="#cb40-142"></a><span class="ss">3. </span>Species *evenness*, e.g. Pielou’s evenness, $J$.</span>
<span id="cb40-143"><a href="#cb40-143"></a></span>
<span id="cb40-144"><a href="#cb40-144"></a>We will work through each in turn.</span>
<span id="cb40-145"><a href="#cb40-145"></a></span>
<span id="cb40-146"><a href="#cb40-146"></a><span class="fu">#### Species Richness, $S$</span></span>
<span id="cb40-147"><a href="#cb40-147"></a></span>
<span id="cb40-148"><a href="#cb40-148"></a>First, is species richness, which we denote by the symbol $S$. This is the simplest measure of $\alpha$-diversity, counting the number of species (or another taxonomic level) present in a given community or sample. It doesn't consider the abundance of species.</span>
<span id="cb40-149"><a href="#cb40-149"></a></span>
<span id="cb40-150"><a href="#cb40-150"></a>In the seaweed biodiversity data, I count the number of species within each of the sections. This is because we view each coastal section as the local scale (the smallest unit of sampling).</span>
<span id="cb40-151"><a href="#cb40-151"></a></span>
<span id="cb40-152"><a href="#cb40-152"></a>The preferred option for calculating species richness is the</span>
<span id="cb40-153"><a href="#cb40-153"></a><span class="in">`specnumber()`</span> function in **vegan**:</span>
<span id="cb40-154"><a href="#cb40-154"></a></span>
<span id="cb40-157"><a href="#cb40-157"></a><span class="in">```{r}</span></span>
<span id="cb40-158"><a href="#cb40-158"></a><span class="fu">specnumber</span>(spp, <span class="at">MARGIN =</span> <span class="dv">1</span>) <span class="co"># &lt;1&gt;</span></span>
<span id="cb40-159"><a href="#cb40-159"></a><span class="in">```</span></span>
<span id="cb40-160"><a href="#cb40-160"></a><span class="ss">1. </span>The <span class="in">`MARGIN = 1`</span> argument tells R to calculate the number of species within each row (site).</span>
<span id="cb40-161"><a href="#cb40-161"></a></span>
<span id="cb40-162"><a href="#cb40-162"></a>The data output is easier to understand if we display it as a <span class="in">`tibble()`</span>:</span>
<span id="cb40-163"><a href="#cb40-163"></a></span>
<span id="cb40-166"><a href="#cb40-166"></a><span class="in">```{r}</span></span>
<span id="cb40-167"><a href="#cb40-167"></a>spp_richness <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">section =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">58</span>,</span>
<span id="cb40-168"><a href="#cb40-168"></a>                       <span class="at">richness =</span> <span class="fu">specnumber</span>(spp, <span class="at">MARGIN =</span> <span class="dv">1</span>))</span>
<span id="cb40-169"><a href="#cb40-169"></a><span class="fu">head</span>(spp_richness)</span>
<span id="cb40-170"><a href="#cb40-170"></a><span class="in">```</span></span>
<span id="cb40-171"><a href="#cb40-171"></a></span>
<span id="cb40-172"><a href="#cb40-172"></a>The <span class="in">`diversityresult()`</span> function in the **BiodiversityR** package can do</span>
<span id="cb40-173"><a href="#cb40-173"></a>the same (sometimes this package is difficult to install due to various</span>
<span id="cb40-174"><a href="#cb40-174"></a>software dependencies that might be required for the package to load</span>
<span id="cb40-175"><a href="#cb40-175"></a>properly---do not be sad if this method does not work):</span>
<span id="cb40-176"><a href="#cb40-176"></a></span>
<span id="cb40-177"><a href="#cb40-177"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb40-178"><a href="#cb40-178"></a>spp_richness <span class="ot">&lt;-</span> <span class="fu">diversityresult</span>(spp, <span class="at">index =</span> <span class="st">'richness'</span>,</span>
<span id="cb40-179"><a href="#cb40-179"></a>                                <span class="at">method =</span> <span class="st">'each site'</span>)</span>
<span id="cb40-180"><a href="#cb40-180"></a><span class="fu">head</span>(spp_richness)</span>
<span id="cb40-181"><a href="#cb40-181"></a><span class="in">```</span></span>
<span id="cb40-182"><a href="#cb40-182"></a></span>
<span id="cb40-183"><a href="#cb40-183"></a>Now we make a plot seen in @fig-species-richness:</span>
<span id="cb40-184"><a href="#cb40-184"></a></span>
<span id="cb40-187"><a href="#cb40-187"></a><span class="in">```{r}</span></span>
<span id="cb40-188"><a href="#cb40-188"></a><span class="co">#| fig-cap: "The seaweed species richness, $S$, within each of the coastal sections along the shore of South Africa."</span></span>
<span id="cb40-189"><a href="#cb40-189"></a><span class="co">#| label: fig-species-richness</span></span>
<span id="cb40-190"><a href="#cb40-190"></a><span class="fu">ggplot</span>(<span class="at">data =</span> spp_richness, (<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">58</span>, <span class="at">y =</span> richness))) <span class="sc">+</span></span>
<span id="cb40-191"><a href="#cb40-191"></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.2</span>, <span class="at">colour =</span> <span class="st">"indianred"</span>) <span class="sc">+</span></span>
<span id="cb40-192"><a href="#cb40-192"></a>  <span class="fu">xlab</span>(<span class="st">"Coastal section, west to east"</span>) <span class="sc">+</span></span>
<span id="cb40-193"><a href="#cb40-193"></a>  <span class="fu">ylab</span>(<span class="st">"Species richness"</span>) <span class="sc">+</span></span>
<span id="cb40-194"><a href="#cb40-194"></a>  <span class="fu">theme_linedraw</span>()</span>
<span id="cb40-195"><a href="#cb40-195"></a><span class="in">```</span></span>
<span id="cb40-196"><a href="#cb40-196"></a></span>
<span id="cb40-197"><a href="#cb40-197"></a>In other instances, it makes more sense to calculate the mean species richness of all the sampling units (e.g. quadrats) taken inside the ecosystem of interest. How you calculate and present species richness depend on your research question and so you will have to decide based on your data and study.</span>
<span id="cb40-198"><a href="#cb40-198"></a></span>
<span id="cb40-199"><a href="#cb40-199"></a>In the seaweed study, the mean ± SD species richness across all of the 58 coastal sections is:</span>
<span id="cb40-200"><a href="#cb40-200"></a></span>
<span id="cb40-203"><a href="#cb40-203"></a><span class="in">```{r}</span></span>
<span id="cb40-204"><a href="#cb40-204"></a><span class="fu">round</span>(<span class="fu">mean</span>(spp_richness<span class="sc">$</span>richness), <span class="dv">2</span>)</span>
<span id="cb40-205"><a href="#cb40-205"></a><span class="fu">round</span>(<span class="fu">sd</span>(spp_richness<span class="sc">$</span>richness), <span class="dv">2</span>)</span>
<span id="cb40-206"><a href="#cb40-206"></a><span class="in">```</span></span>
<span id="cb40-207"><a href="#cb40-207"></a></span>
<span id="cb40-208"><a href="#cb40-208"></a><span class="fu">#### Univariate Diversity Indices</span></span>
<span id="cb40-209"><a href="#cb40-209"></a></span>
<span id="cb40-210"><a href="#cb40-210"></a>The second way we can express $\alpha$-diversity is to use one of the univariate diversity indices. The choice of which index to use should be informed by the extent to which one wants to emphasise richness or evenness. Species</span>
<span id="cb40-211"><a href="#cb40-211"></a>richness, $S$, does not consider evenness as it is all about richness (obviously). Simpson's $\lambda$ emphasises evenness a lot more. Shannon's $H'$ is somewhere in the middle.</span>
<span id="cb40-212"><a href="#cb40-212"></a></span>
<span id="cb40-213"><a href="#cb40-213"></a>**Shannon's $H'$** is sometimes called Shannon's diversity, the Shannon-Wiener index, the Shannon-Weaver index, or the Shannon entropy. This is a more nuanced measure that considers both species richness and evenness (how evenly individuals are distributed across different species).</span>
<span id="cb40-214"><a href="#cb40-214"></a></span>
<span id="cb40-215"><a href="#cb40-215"></a>It is calculated as:</span>
<span id="cb40-216"><a href="#cb40-216"></a></span>
<span id="cb40-217"><a href="#cb40-217"></a>$$H' = -\sum_{i=1}^{S} p_{i} \ln p_{i}$$ where $p_{i}$ is the proportion of individuals belonging to the $i$th species, and $S$ is the species richness.</span>
<span id="cb40-218"><a href="#cb40-218"></a></span>
<span id="cb40-219"><a href="#cb40-219"></a>**Simpson's $\lambda$**, or simply the Simpson index, is a measure that represents the probability that two individuals randomly selected from a sample will belong to the same species. It is calculated as:</span>
<span id="cb40-220"><a href="#cb40-220"></a></span>
<span id="cb40-221"><a href="#cb40-221"></a>$$\displaystyle \lambda = \sum_{i=1}^{S} p_{i}^{2}$$ where $S$ is the</span>
<span id="cb40-222"><a href="#cb40-222"></a>species richness and $p_{i}$ is the relative abundance of the $i$th</span>
<span id="cb40-223"><a href="#cb40-223"></a>species.</span>
<span id="cb40-224"><a href="#cb40-224"></a></span>
<span id="cb40-225"><a href="#cb40-225"></a>**Fisher's $\alpha$** estimates the $\alpha$ parameter of <span class="co">[</span><span class="ot">Fisher's logarithmic series</span><span class="co">](Lab-04-biodiversity.qmd#species-abundance-distribution)</span> (see functions <span class="in">`fisher.alpha()`</span> and <span class="in">`fisherfit()`</span>). The estimation is possible only for actual counts (i.e. integers) of individuals, so it will not work for per cent cover, biomass, and other measures that real numbers can express. It's especially useful for comparing the diversity of samples with different total abundances. We will get to this function later under <span class="co">[</span><span class="ot">Fisher's logarithmic series</span><span class="co">](Lab-04-biodiversity.qmd#species-abundance-distribution)</span>.</span>
<span id="cb40-226"><a href="#cb40-226"></a></span>
<span id="cb40-227"><a href="#cb40-227"></a>Except for Fisher's-$\alpha$, we cannot calculate these for the seaweed data, because, in order to do so, we require abundance data---the seaweed data are presence-absence only. Let us load a fictitious dataset of the diversity of three different communities of plants, with each community corresponding to a different light environment (dim, mid, and high light):</span>
<span id="cb40-228"><a href="#cb40-228"></a></span>
<span id="cb40-231"><a href="#cb40-231"></a><span class="in">```{r}</span></span>
<span id="cb40-232"><a href="#cb40-232"></a>light <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../data/light_levels.csv"</span>)</span>
<span id="cb40-233"><a href="#cb40-233"></a>light</span>
<span id="cb40-234"><a href="#cb40-234"></a><span class="in">```</span></span>
<span id="cb40-235"><a href="#cb40-235"></a></span>
<span id="cb40-236"><a href="#cb40-236"></a>We can see above that instead of having data with 1s and 0s for presence-absence, here we have some values that indicate the relative number of individuals belonging to each of the species in the three light environments. We calculate species richness (as before), and also the Shannon and Simpson indices using **vegan**'s <span class="in">`diversity()`</span> function:</span>
<span id="cb40-237"><a href="#cb40-237"></a></span>
<span id="cb40-240"><a href="#cb40-240"></a><span class="in">```{r}</span></span>
<span id="cb40-241"><a href="#cb40-241"></a>light_div <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb40-242"><a href="#cb40-242"></a>  <span class="at">site =</span> <span class="fu">c</span>(<span class="st">"low_light"</span>, <span class="st">"mid_light"</span>, <span class="st">"high_light"</span>),</span>
<span id="cb40-243"><a href="#cb40-243"></a>  <span class="at">richness =</span> <span class="fu">specnumber</span>(light[, <span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>], <span class="at">MARGIN =</span> <span class="dv">1</span>),</span>
<span id="cb40-244"><a href="#cb40-244"></a>  <span class="at">shannon =</span> <span class="fu">round</span>(<span class="fu">diversity</span>(light[, <span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>], <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">index =</span> <span class="st">"shannon"</span>), <span class="dv">2</span>),</span>
<span id="cb40-245"><a href="#cb40-245"></a>  <span class="at">simpson =</span> <span class="fu">round</span>(<span class="fu">diversity</span>(light[, <span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>], <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">index =</span> <span class="st">"simpson"</span>), <span class="dv">2</span>)</span>
<span id="cb40-246"><a href="#cb40-246"></a>)</span>
<span id="cb40-247"><a href="#cb40-247"></a>light_div</span>
<span id="cb40-248"><a href="#cb40-248"></a><span class="in">```</span></span>
<span id="cb40-249"><a href="#cb40-249"></a></span>
<span id="cb40-250"><a href="#cb40-250"></a><span class="co">&lt;!-- Hill numbers, $q$, (Hill, 1973) summarise $S$, Shannon's $H'$ and Simpson's $\lambda$. Higher $q$ puts less weight on rare species and weighs abundant ones more. Hill numbers can be used to draw diversity profiles, which allow for an elegant comparison of diversity among communities considering both richness and evenness. --&gt;</span></span>
<span id="cb40-251"><a href="#cb40-251"></a></span>
<span id="cb40-252"><a href="#cb40-252"></a><span class="co">&lt;!-- ```{r fig.height=3, fig.width=5} --&gt;</span></span>
<span id="cb40-253"><a href="#cb40-253"></a><span class="co">&lt;!-- data(BCI) --&gt;</span></span>
<span id="cb40-254"><a href="#cb40-254"></a><span class="co">&lt;!-- i &lt;- sample(nrow(BCI), 12) --&gt;</span></span>
<span id="cb40-255"><a href="#cb40-255"></a><span class="co">&lt;!-- mod &lt;- renyi(BCI[i,]) --&gt;</span></span>
<span id="cb40-256"><a href="#cb40-256"></a><span class="co">&lt;!-- plot(mod) --&gt;</span></span>
<span id="cb40-257"><a href="#cb40-257"></a><span class="co">&lt;!-- mod &lt;- renyiaccum(BCI[i,]) --&gt;</span></span>
<span id="cb40-258"><a href="#cb40-258"></a><span class="co">&lt;!-- plot(mod, as.table=TRUE, col = c(1, 2, 2)) --&gt;</span></span>
<span id="cb40-259"><a href="#cb40-259"></a><span class="co">&lt;!-- persp(mod) --&gt;</span></span>
<span id="cb40-260"><a href="#cb40-260"></a><span class="co">&lt;!-- ``` --&gt;</span></span>
<span id="cb40-261"><a href="#cb40-261"></a></span>
<span id="cb40-262"><a href="#cb40-262"></a>Evenness refers to the shape of a <span class="co">[</span><span class="ot">species abundance distribution</span><span class="co">](Lab-04-biodiversity.qmd#species-abundance-distribution)</span>, which suggests the relative abundance of different species.</span>
<span id="cb40-263"><a href="#cb40-263"></a></span>
<span id="cb40-264"><a href="#cb40-264"></a>One index for evenness is Pielou's evenness, $J$:</span>
<span id="cb40-265"><a href="#cb40-265"></a></span>
<span id="cb40-266"><a href="#cb40-266"></a>$$J = \frac{H^{\prime}} {log(S)}$$</span>
<span id="cb40-267"><a href="#cb40-267"></a></span>
<span id="cb40-268"><a href="#cb40-268"></a>where $H'$ is Shannon's diversity index, and $S$ the number of species</span>
<span id="cb40-269"><a href="#cb40-269"></a>(i.e. $S$).</span>
<span id="cb40-270"><a href="#cb40-270"></a></span>
<span id="cb40-271"><a href="#cb40-271"></a>To calculate Pielou's evenness index for the <span class="in">`light`</span> data, we can do</span>
<span id="cb40-272"><a href="#cb40-272"></a>this:</span>
<span id="cb40-273"><a href="#cb40-273"></a></span>
<span id="cb40-276"><a href="#cb40-276"></a><span class="in">```{r}</span></span>
<span id="cb40-277"><a href="#cb40-277"></a>H <span class="ot">&lt;-</span> <span class="fu">diversity</span>(light[, <span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>], <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">index =</span> <span class="st">"shannon"</span>)</span>
<span id="cb40-278"><a href="#cb40-278"></a></span>
<span id="cb40-279"><a href="#cb40-279"></a>J <span class="ot">&lt;-</span> H<span class="sc">/</span><span class="fu">log</span>(<span class="fu">specnumber</span>(light[, <span class="dv">2</span><span class="sc">:</span><span class="dv">7</span>]))</span>
<span id="cb40-280"><a href="#cb40-280"></a><span class="fu">round</span>(J, <span class="dv">2</span>)</span>
<span id="cb40-281"><a href="#cb40-281"></a><span class="in">```</span></span>
<span id="cb40-282"><a href="#cb40-282"></a></span>
<span id="cb40-283"><a href="#cb40-283"></a>**Berger-Parker Index** indicates the proportion of the community that the most abundant species represents. It is given by the formula: </span>
<span id="cb40-284"><a href="#cb40-284"></a></span>
<span id="cb40-285"><a href="#cb40-285"></a>$$d = \frac{N_{max}}{N}$$</span>
<span id="cb40-286"><a href="#cb40-286"></a>where $N_{max}$ is the number of individuals of the most common species and $N$ is the total number of individuals in the sample.</span>
<span id="cb40-287"><a href="#cb40-287"></a></span>
<span id="cb40-288"><a href="#cb40-288"></a>**Chao1 and ACE** are estimators often used to predict the total species richness in a community based on the number of rare species observed in samples.</span>
<span id="cb40-289"><a href="#cb40-289"></a></span>
<span id="cb40-290"><a href="#cb40-290"></a><span class="fu">### Gamma-Diversity</span></span>
<span id="cb40-291"><a href="#cb40-291"></a></span>
<span id="cb40-292"><a href="#cb40-292"></a>Returning to the seaweed data, $Y$, let us now look at $\gamma$-diversity---this would be the total number of species along the South African coastline in all 58 coastal sections. Since each column represents one species, and the dataset contains data collected at each of the 58 sites (the number of rows), we can do:</span>
<span id="cb40-293"><a href="#cb40-293"></a></span>
<span id="cb40-296"><a href="#cb40-296"></a><span class="in">```{r}</span></span>
<span id="cb40-297"><a href="#cb40-297"></a><span class="fu">ncol</span>(spp) <span class="co"># &lt;1&gt;</span></span>
<span id="cb40-298"><a href="#cb40-298"></a><span class="in">```</span></span>
<span id="cb40-299"><a href="#cb40-299"></a><span class="ss">1. </span>The number of columns gives the total number of species in this example.</span>
<span id="cb40-300"><a href="#cb40-300"></a></span>
<span id="cb40-301"><a href="#cb40-301"></a>We can also use:</span>
<span id="cb40-302"><a href="#cb40-302"></a></span>
<span id="cb40-303"><a href="#cb40-303"></a><span class="in">```{r, eval=TRUE}</span></span>
<span id="cb40-304"><a href="#cb40-304"></a><span class="fu">diversityresult</span>(spp, <span class="at">index =</span> <span class="st">'richness'</span>, <span class="at">method =</span> <span class="st">'pooled'</span>)</span>
<span id="cb40-305"><a href="#cb40-305"></a><span class="in">```</span></span>
<span id="cb40-306"><a href="#cb40-306"></a></span>
<span id="cb40-307"><a href="#cb40-307"></a><span class="co">&lt;!-- ```{r} --&gt;</span></span>
<span id="cb40-308"><a href="#cb40-308"></a></span>
<span id="cb40-309"><a href="#cb40-309"></a><span class="co">&lt;!-- specpool() --&gt;</span></span>
<span id="cb40-310"><a href="#cb40-310"></a></span>
<span id="cb40-311"><a href="#cb40-311"></a><span class="co">&lt;!-- ``` --&gt;</span></span>
<span id="cb40-312"><a href="#cb40-312"></a></span>
<span id="cb40-313"><a href="#cb40-313"></a>::: callout-important</span>
<span id="cb40-314"><a href="#cb40-314"></a><span class="fu">## Lab 3</span></span>
<span id="cb40-315"><a href="#cb40-315"></a>(To be reviewed by BCB743 student but not for marks)</span>
<span id="cb40-316"><a href="#cb40-316"></a></span>
<span id="cb40-317"><a href="#cb40-317"></a><span class="ss">1. </span>Why is there a difference between the two?</span>
<span id="cb40-318"><a href="#cb40-318"></a><span class="ss">2. </span>Which is correct?</span>
<span id="cb40-319"><a href="#cb40-319"></a>:::</span>
<span id="cb40-320"><a href="#cb40-320"></a></span>
<span id="cb40-321"><a href="#cb40-321"></a>Think before you calculate $\gamma$-diversity for your own data as it might not be as simple as here!</span>
<span id="cb40-322"><a href="#cb40-322"></a></span>
<span id="cb40-323"><a href="#cb40-323"></a><span class="fu">### Beta-Diversity</span></span>
<span id="cb40-324"><a href="#cb40-324"></a></span>
<span id="cb40-325"><a href="#cb40-325"></a><span class="fu">#### Whittaker's $\beta$-Diversity</span></span>
<span id="cb40-326"><a href="#cb40-326"></a></span>
<span id="cb40-327"><a href="#cb40-327"></a>The first measure of $\beta$-diversity comes from @whittaker1960vegetation and is called *true $\beta$-diversity*. In this instance, divide the $\gamma$-diversity for the region by the $\alpha$-diversity for a specific coastal section. We can calculate it all at once for the whole dataset and make a graph (@fig-true-beta):</span>
<span id="cb40-328"><a href="#cb40-328"></a></span>
<span id="cb40-331"><a href="#cb40-331"></a><span class="in">```{r}</span></span>
<span id="cb40-332"><a href="#cb40-332"></a><span class="co">#| fig-cap: "Whittaker’s true β-diversity shown in the seaweed data."</span></span>
<span id="cb40-333"><a href="#cb40-333"></a><span class="co">#| label: fig-true-beta</span></span>
<span id="cb40-334"><a href="#cb40-334"></a>true_beta <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb40-335"><a href="#cb40-335"></a>  <span class="at">beta =</span> <span class="fu">specnumber</span>(spp, <span class="at">MARGIN =</span> <span class="dv">1</span>) <span class="sc">/</span> <span class="fu">ncol</span>(spp),</span>
<span id="cb40-336"><a href="#cb40-336"></a>  <span class="at">section_no =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">58</span>)</span>
<span id="cb40-337"><a href="#cb40-337"></a>)</span>
<span id="cb40-338"><a href="#cb40-338"></a><span class="co"># true_beta</span></span>
<span id="cb40-339"><a href="#cb40-339"></a><span class="fu">ggplot</span>(<span class="at">data =</span> true_beta, (<span class="fu">aes</span>(<span class="at">x =</span> section_no, <span class="at">y =</span> beta))) <span class="sc">+</span></span>
<span id="cb40-340"><a href="#cb40-340"></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.2</span>, <span class="at">colour =</span> <span class="st">"indianred"</span>) <span class="sc">+</span></span>
<span id="cb40-341"><a href="#cb40-341"></a>  <span class="fu">xlab</span>(<span class="st">"Coastal section, west to east"</span>) <span class="sc">+</span></span>
<span id="cb40-342"><a href="#cb40-342"></a>  <span class="fu">ylab</span>(<span class="st">"True beta-diversity"</span>) <span class="sc">+</span></span>
<span id="cb40-343"><a href="#cb40-343"></a>  <span class="fu">theme_linedraw</span>()</span>
<span id="cb40-344"><a href="#cb40-344"></a><span class="in">```</span></span>
<span id="cb40-345"><a href="#cb40-345"></a></span>
<span id="cb40-346"><a href="#cb40-346"></a>The second measure of $\beta$-diversity is *absolute species turnover*, and to calculate this, we subtract $\alpha$-diversity for each  section from the region's $\gamma$-diversity (@fig-abs-beta):</span>
<span id="cb40-347"><a href="#cb40-347"></a></span>
<span id="cb40-350"><a href="#cb40-350"></a><span class="in">```{r}</span></span>
<span id="cb40-351"><a href="#cb40-351"></a><span class="co">#| fig-cap: "Whittaker’s absolute species turnover shown in action in the seaweed data."</span></span>
<span id="cb40-352"><a href="#cb40-352"></a><span class="co">#| label: fig-abs-beta</span></span>
<span id="cb40-353"><a href="#cb40-353"></a>abs_beta <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb40-354"><a href="#cb40-354"></a>  <span class="at">beta =</span> <span class="fu">ncol</span>(spp) <span class="sc">-</span> <span class="fu">specnumber</span>(spp, <span class="at">MARGIN =</span> <span class="dv">1</span>),</span>
<span id="cb40-355"><a href="#cb40-355"></a>  <span class="at">section_no =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">58</span>)</span>
<span id="cb40-356"><a href="#cb40-356"></a>)</span>
<span id="cb40-357"><a href="#cb40-357"></a><span class="co"># abs_beta</span></span>
<span id="cb40-358"><a href="#cb40-358"></a><span class="fu">ggplot</span>(<span class="at">data =</span> abs_beta, (<span class="fu">aes</span>(<span class="at">x =</span> section_no, <span class="at">y =</span> beta))) <span class="sc">+</span></span>
<span id="cb40-359"><a href="#cb40-359"></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.2</span>, <span class="at">colour =</span> <span class="st">"indianred"</span>) <span class="sc">+</span></span>
<span id="cb40-360"><a href="#cb40-360"></a>  <span class="fu">xlab</span>(<span class="st">"Coastal section, west to east"</span>) <span class="sc">+</span></span>
<span id="cb40-361"><a href="#cb40-361"></a>  <span class="fu">ylab</span>(<span class="st">"Absolute beta-diversity"</span>) <span class="sc">+</span></span>
<span id="cb40-362"><a href="#cb40-362"></a>  <span class="fu">theme_linedraw</span>()</span>
<span id="cb40-363"><a href="#cb40-363"></a><span class="in">```</span></span>
<span id="cb40-364"><a href="#cb40-364"></a></span>
<span id="cb40-365"><a href="#cb40-365"></a><span class="fu">#### Contemporary Definitions $\beta$-Diversity</span></span>
<span id="cb40-366"><a href="#cb40-366"></a></span>
<span id="cb40-367"><a href="#cb40-367"></a>Contemporary definitions of $\beta$-diversity rely on <span class="co">[</span><span class="ot">pairwise *dissimilarity indices*</span><span class="co">](Lec-04-biodiversity.qmd#sec-dissimilarity-indices)</span> such as Bray-Curtis, Jaccard, or Sørensen dissimilarities---see @koleff2003measuring for many more; also see <span class="in">`?vegdist`</span>. However, discussing pairwise dissimilarities with $\beta$-diversity makes more sense.</span>
<span id="cb40-368"><a href="#cb40-368"></a></span>
<span id="cb40-369"><a href="#cb40-369"></a><span class="fu">##### Dissimilarity indices</span></span>
<span id="cb40-370"><a href="#cb40-370"></a></span>
<span id="cb40-371"><a href="#cb40-371"></a>Dissimilarity indices are special cases of diversity indices that use pairwise comparisons between sampling units, habitats, or ecosystems.</span>
<span id="cb40-372"><a href="#cb40-372"></a></span>
<span id="cb40-373"><a href="#cb40-373"></a>Species dissimilarities result in pairwise matrices similar to the pairwise correlation or Euclidian distance matrices we have seen in <span class="co">[</span><span class="ot">Lab 1</span><span class="co">](Lec-01-introduction.qmd)</span>. In <span class="co">[</span><span class="ot">Lab 2b</span><span class="co">](Lec-04-biodiversity.qmd)</span> you will have also learned how to calculate these ecological distances in R. These dissimilarity indices are multivariate and compare between sites, sections, plots, etc., and must therefore not be confused with the univariate diversity indices.</span>
<span id="cb40-374"><a href="#cb40-374"></a></span>
<span id="cb40-375"><a href="#cb40-375"></a>We use the Bray-Curtis and Jaccard indices with abundance data and the Sørensen dissimilarity with presence-absence data. The seaweed dataset is a presence-absence dataset that necessitates using the Sørensen index. The interpretation of the resulting square (number of rows = number of columns) dissimilarity matrices is the same regardless of whether we calculate it for an abundance or presence-absence dataset. The values in the matrix range from 0 to 1. A 0 means that the pair of sites we compare is identical (all species in common) but 1 means they are completely different (no species in common). In the square dissimilarity matrix, the diagonal is 0, which essentially (and obviously) means that any site is identical to itself. Elsewhere the values will range from 0 to 1. Since this is a pairwise calculation (each site compared to every other site), our seaweed dataset will contain (58 × (58 - 1))/2 = 1653 values, each one ranging from 0 to 1.</span>
<span id="cb40-376"><a href="#cb40-376"></a></span>
<span id="cb40-377"><a href="#cb40-377"></a>The first step involves the species table, $Y$. First, we compute the Sørensen dissimilarity index, $\beta_{\text{sør}}$, to compare the dissimilarity of all pairs of coastal sections using presence-absence data. The dissimilarity in species composition between two sections is calculated from three parameters, *viz*., *b* and *c*, which represent the number of species unique to each of the two sites, and *a*, the number of species in common between them. It is given by:</span>
<span id="cb40-378"><a href="#cb40-378"></a></span>
<span id="cb40-379"><a href="#cb40-379"></a>$$\beta_\text{sør}=\frac{2a}{2a+b+c}$$</span>
<span id="cb40-380"><a href="#cb40-380"></a>Where $a$ is the number of species in common between two sites, and $b$ and $c$ are the number of species unique to each site. The Sørensen dissimilarity index ranges from 0 to 1, where 0 means that the pair of sites we compare is identical (all species in common) and 1 means they are completely different (no species in common).</span>
<span id="cb40-381"><a href="#cb40-381"></a></span>
<span id="cb40-382"><a href="#cb40-382"></a></span>
<span id="cb40-383"><a href="#cb40-383"></a>The **vegan** function <span class="in">`vegdist()`</span> provides access to the dissimilarity indices. We calculate the Sørensen dissimilarity index:</span>
<span id="cb40-384"><a href="#cb40-384"></a></span>
<span id="cb40-387"><a href="#cb40-387"></a><span class="in">```{r}</span></span>
<span id="cb40-388"><a href="#cb40-388"></a>sor <span class="ot">&lt;-</span> <span class="fu">vegdist</span>(spp, <span class="at">binary =</span> <span class="cn">TRUE</span>) <span class="co"># makes the lower triangle matrix</span></span>
<span id="cb40-389"><a href="#cb40-389"></a>sor_df <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">as.matrix</span>(sor), <span class="dv">4</span>)</span>
<span id="cb40-390"><a href="#cb40-390"></a><span class="fu">dim</span>(sor_df)</span>
<span id="cb40-391"><a href="#cb40-391"></a>sor_df[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>] <span class="co"># the first 10 rows and columns</span></span>
<span id="cb40-392"><a href="#cb40-392"></a><span class="in">```</span></span>
<span id="cb40-393"><a href="#cb40-393"></a></span>
<span id="cb40-394"><a href="#cb40-394"></a>What we see above is a square dissimilarity matrix. The most important characteristics of the matrix are:</span>
<span id="cb40-395"><a href="#cb40-395"></a></span>
<span id="cb40-396"><a href="#cb40-396"></a>i.  whereas the raw species data, $Y$, is rectangular (number rows ≠ number columns), the dissimilarity matrix is square (number rows = number columns);</span>
<span id="cb40-397"><a href="#cb40-397"></a>ii. the diagonal is filled with 0;</span>
<span id="cb40-398"><a href="#cb40-398"></a>iii. the matrix is symmetrical---it is comprised of symetrical upper and lower triangles.</span>
<span id="cb40-399"><a href="#cb40-399"></a></span>
<span id="cb40-400"><a href="#cb40-400"></a>Create a <span class="in">`data.frame`</span> suitable for plotting:</span>
<span id="cb40-401"><a href="#cb40-401"></a></span>
<span id="cb40-404"><a href="#cb40-404"></a><span class="in">```{r}</span></span>
<span id="cb40-405"><a href="#cb40-405"></a>sor_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">round</span>(<span class="fu">as.matrix</span>(sor), <span class="dv">4</span>))</span>
<span id="cb40-406"><a href="#cb40-406"></a><span class="in">```</span></span>
<span id="cb40-407"><a href="#cb40-407"></a></span>
<span id="cb40-408"><a href="#cb40-408"></a>::: callout-important</span>
<span id="cb40-409"><a href="#cb40-409"></a><span class="fu">## Lab 3</span></span>
<span id="cb40-410"><a href="#cb40-410"></a>(To be reviewed by BCB743 student but not for marks)</span>
<span id="cb40-411"><a href="#cb40-411"></a></span>
<span id="cb40-412"><a href="#cb40-412"></a>These questions concern matrices produced from species data using any of the indices available in <span class="in">`vegdist()`</span>:</span>
<span id="cb40-413"><a href="#cb40-413"></a></span>
<span id="cb40-414"><a href="#cb40-414"></a><span class="ss">3. </span>Why is the matrix square, and what determines the number of rows/columns?</span>
<span id="cb40-415"><a href="#cb40-415"></a><span class="ss">4. </span>What is the meaning of the diagonal?</span>
<span id="cb40-416"><a href="#cb40-416"></a><span class="ss">5. </span>What is the meaning of the non-diagonal elements?</span>
<span id="cb40-417"><a href="#cb40-417"></a><span class="ss">6. </span>Referring to the seaweed species data specifically, take the data in row 1 or column 1 and create a line graph showing these values as a function of the section number.</span>
<span id="cb40-418"><a href="#cb40-418"></a><span class="ss">7. </span>Provide a mechanistic (ecological) explanation for why this figure takes the shape that it does. Which community assembly process does this hint at?</span>
<span id="cb40-419"><a href="#cb40-419"></a><span class="co">&lt;!-- &gt; Answer Q4: --&gt;</span></span>
<span id="cb40-420"><a href="#cb40-420"></a></span>
<span id="cb40-421"><a href="#cb40-421"></a><span class="co">&lt;!-- ```{r} --&gt;</span></span>
<span id="cb40-422"><a href="#cb40-422"></a><span class="co">&lt;!-- ggplot(data = sor_df, (aes(x = 1:58, y = X1))) + --&gt;</span></span>
<span id="cb40-423"><a href="#cb40-423"></a><span class="co">&lt;!--   geom_line(size = 1.2, colour = "indianred") + --&gt;</span></span>
<span id="cb40-424"><a href="#cb40-424"></a><span class="co">&lt;!--   xlab("Coastal section, west to east") + --&gt;</span></span>
<span id="cb40-425"><a href="#cb40-425"></a><span class="co">&lt;!--   ylab("Total beta-diversity") + --&gt;</span></span>
<span id="cb40-426"><a href="#cb40-426"></a><span class="co">&lt;!--   theme_linedraw() --&gt;</span></span>
<span id="cb40-427"><a href="#cb40-427"></a><span class="co">&lt;!-- ``` --&gt;</span></span>
<span id="cb40-428"><a href="#cb40-428"></a>:::</span>
<span id="cb40-429"><a href="#cb40-429"></a></span>
<span id="cb40-430"><a href="#cb40-430"></a>There are different interpretations linked to $\beta$-diversity, each telling us something different about community formation processes. </span>
<span id="cb40-431"><a href="#cb40-431"></a></span>
<span id="cb40-432"><a href="#cb40-432"></a><span class="fu">##### Species turnover and nestedness-resultant $\beta$-diversity</span></span>
<span id="cb40-433"><a href="#cb40-433"></a></span>
<span id="cb40-434"><a href="#cb40-434"></a>There are two kinds of $\beta$-diversity: <span class="co">[</span><span class="ot">species turnover and nestedness-resultant $\beta$-diversity</span><span class="co">](Lec-04-biodiversity.qmd#sec-beta-diversity)</span>. The former is the result of species replacement between sites, whereas the latter is the result of species loss or gain between sites. The Sørensen dissimilarity index, $\beta_\text{sør}$, can be decomposed into these two components.</span>
<span id="cb40-435"><a href="#cb40-435"></a></span>
<span id="cb40-436"><a href="#cb40-436"></a>How do we calculate the turnover and nestedness-resultant components of $\beta$-diversity? The **betapart** package <span class="co">[</span><span class="ot">@baselga2022betapart</span><span class="co">]</span> comes to the rescue. We decompose the dissimilarity into the $\beta_\text{sim}$ and $\beta_\text{sne}$ components <span class="co">[</span><span class="ot">@baselga2010partitioning</span><span class="co">]</span> using the <span class="in">`betapart.core()`</span> and <span class="in">`betapart.pair()`</span> functions. The outcomes of this partitioning calculation are placed into the matrices $Y1$ and $Y2$. These data can then be analysed further---e.g. we can apply a principal components analysis (PCA) or another multivariate analysis on $Y$ to find the major patterns in the community data---we will do this in BCB743.</span>
<span id="cb40-437"><a href="#cb40-437"></a></span>
<span id="cb40-440"><a href="#cb40-440"></a><span class="in">```{r}</span></span>
<span id="cb40-441"><a href="#cb40-441"></a><span class="co"># Decompose total Sørensen dissimilarity into turnover and nestedness-resultant</span></span>
<span id="cb40-442"><a href="#cb40-442"></a><span class="co"># components:</span></span>
<span id="cb40-443"><a href="#cb40-443"></a>Y.core <span class="ot">&lt;-</span> <span class="fu">betapart.core</span>(spp)</span>
<span id="cb40-444"><a href="#cb40-444"></a>Y.pair <span class="ot">&lt;-</span> <span class="fu">beta.pair</span>(Y.core, <span class="at">index.family =</span> <span class="st">"sor"</span>)</span>
<span id="cb40-445"><a href="#cb40-445"></a></span>
<span id="cb40-446"><a href="#cb40-446"></a><span class="co"># Let Y1 be the turnover component (beta-sim):</span></span>
<span id="cb40-447"><a href="#cb40-447"></a>Y1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">round</span>(<span class="fu">as.matrix</span>(Y.pair<span class="sc">$</span>beta.sim), <span class="dv">3</span>))</span>
<span id="cb40-448"><a href="#cb40-448"></a></span>
<span id="cb40-449"><a href="#cb40-449"></a><span class="co"># Let Y2 be the nestedness-resultant component (beta-sne):</span></span>
<span id="cb40-450"><a href="#cb40-450"></a>Y2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">round</span>(<span class="fu">as.matrix</span>(Y.pair<span class="sc">$</span>beta.sne), <span class="dv">3</span>))</span>
<span id="cb40-451"><a href="#cb40-451"></a><span class="in">```</span></span>
<span id="cb40-452"><a href="#cb40-452"></a></span>
<span id="cb40-453"><a href="#cb40-453"></a>A portion of the turnover component matrix:</span>
<span id="cb40-456"><a href="#cb40-456"></a><span class="in">```{r}</span></span>
<span id="cb40-457"><a href="#cb40-457"></a>Y1[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span>
<span id="cb40-458"><a href="#cb40-458"></a><span class="in">```</span></span>
<span id="cb40-459"><a href="#cb40-459"></a></span>
<span id="cb40-460"><a href="#cb40-460"></a>A portion of the nestedness-resultant matrix:</span>
<span id="cb40-463"><a href="#cb40-463"></a><span class="in">```{r}</span></span>
<span id="cb40-464"><a href="#cb40-464"></a>Y2[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span>
<span id="cb40-465"><a href="#cb40-465"></a><span class="in">```</span></span>
<span id="cb40-466"><a href="#cb40-466"></a></span>
<span id="cb40-467"><a href="#cb40-467"></a>A portion of the nestedness-resultant matrix reformatted as a <span class="in">`tibble()`</span>^<span class="co">[</span><span class="ot">Note that the rows are no longer numbered in the tibble view, but it can easily be recreated by `seq(1:58)`.</span><span class="co">]</span>:</span>
<span id="cb40-470"><a href="#cb40-470"></a><span class="in">```{r}</span></span>
<span id="cb40-471"><a href="#cb40-471"></a>Y2_tib <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(Y2)</span>
<span id="cb40-472"><a href="#cb40-472"></a><span class="fu">head</span>(Y2_tib)</span>
<span id="cb40-473"><a href="#cb40-473"></a><span class="in">```</span></span>
<span id="cb40-474"><a href="#cb40-474"></a></span>
<span id="cb40-475"><a href="#cb40-475"></a></span>
<span id="cb40-476"><a href="#cb40-476"></a>::: callout-important</span>
<span id="cb40-477"><a href="#cb40-477"></a><span class="fu">## Lab 3 (continue)</span></span>
<span id="cb40-478"><a href="#cb40-478"></a>(To be reviewed by BCB743 student but not for marks)</span>
<span id="cb40-479"><a href="#cb40-479"></a></span>
<span id="cb40-480"><a href="#cb40-480"></a><span class="ss">8. </span>Plot species turnover as a function of Section number, and provide a mechanistic explanation for the pattern observed.</span>
<span id="cb40-481"><a href="#cb40-481"></a><span class="ss">9. </span>Based on an assessment of literature on the topic, provide a discussion of nestedness-resultant $\beta$-diversity. Use either a marine or terrestrial example to explain this mode of structuring biodiversity (i.e. assembly of species into a community).</span>
<span id="cb40-482"><a href="#cb40-482"></a><span class="co">&lt;!-- &gt; Answer Q8:--&gt;</span></span>
<span id="cb40-483"><a href="#cb40-483"></a></span>
<span id="cb40-484"><a href="#cb40-484"></a><span class="co">&lt;!-- ```{r} --&gt;</span></span>
<span id="cb40-485"><a href="#cb40-485"></a><span class="co">&lt;!-- # The turnover component --&gt;</span></span>
<span id="cb40-486"><a href="#cb40-486"></a><span class="co">&lt;!-- ggplot(data = Y1, (aes(x = 1:58, y = X1))) + --&gt;</span></span>
<span id="cb40-487"><a href="#cb40-487"></a><span class="co">&lt;!--   geom_line(size = 1.2, colour = "indianred") + --&gt;</span></span>
<span id="cb40-488"><a href="#cb40-488"></a><span class="co">&lt;!--   xlab("Coastal section, west to east") + --&gt;</span></span>
<span id="cb40-489"><a href="#cb40-489"></a><span class="co">&lt;!--   ylab("Species richness") + --&gt;</span></span>
<span id="cb40-490"><a href="#cb40-490"></a><span class="co">&lt;!--   theme_linedraw() --&gt;</span></span>
<span id="cb40-491"><a href="#cb40-491"></a><span class="co">&lt;!-- ``` --&gt;</span></span>
<span id="cb40-492"><a href="#cb40-492"></a>:::</span>
<span id="cb40-493"><a href="#cb40-493"></a></span>
<span id="cb40-494"><a href="#cb40-494"></a>:::{.callout-important}</span>
<span id="cb40-495"><a href="#cb40-495"></a><span class="fu">## Submission Instructions</span></span>
<span id="cb40-496"><a href="#cb40-496"></a></span>
<span id="cb40-497"><a href="#cb40-497"></a>The Lab 3 assignment is due at **07:00 on Monday 12 August 2022**.</span>
<span id="cb40-498"><a href="#cb40-498"></a></span>
<span id="cb40-499"><a href="#cb40-499"></a>Provide a **neat and thoroughly annotated** R file which can recreate all the</span>
<span id="cb40-500"><a href="#cb40-500"></a>graphs and all calculations. Written answers must be typed in the same file as comments.</span>
<span id="cb40-501"><a href="#cb40-501"></a></span>
<span id="cb40-502"><a href="#cb40-502"></a>Please label the R file as follows:</span>
<span id="cb40-503"><a href="#cb40-503"></a></span>
<span id="cb40-504"><a href="#cb40-504"></a><span class="ss">-   </span><span class="in">`BDC334_&lt;first_name&gt;_&lt;last_name&gt;_Lab_3.R`</span></span>
<span id="cb40-505"><a href="#cb40-505"></a></span>
<span id="cb40-506"><a href="#cb40-506"></a>(the <span class="in">`&lt;`</span> and <span class="in">`&gt;`</span> must be omitted as they are used in the example as</span>
<span id="cb40-507"><a href="#cb40-507"></a>field indicators only).</span>
<span id="cb40-508"><a href="#cb40-508"></a></span>
<span id="cb40-509"><a href="#cb40-509"></a>Submit your appropriately named R documents on iKamva when ready.</span>
<span id="cb40-510"><a href="#cb40-510"></a></span>
<span id="cb40-511"><a href="#cb40-511"></a>Failing to follow these instructions carefully, precisely, and</span>
<span id="cb40-512"><a href="#cb40-512"></a>thoroughly will cause you to lose marks, which could cause a significant</span>
<span id="cb40-513"><a href="#cb40-513"></a>drop in your score as formatting counts for 15% of the final mark (out</span>
<span id="cb40-514"><a href="#cb40-514"></a>of 100%).</span>
<span id="cb40-515"><a href="#cb40-515"></a>:::</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>© 2025, AJ Smit</p>
</div>   
    <div class="nav-footer-center">

<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BDC334/Lab-03-biodiversity.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
<p><span class="faux-block">Made with <i class="fa-brands fa-r-project" aria-label="r-project"></i> and <a href="https://quarto.org/">Quarto</a></span><br> <span class="faux-block"><a href="?var:github-url">View the source at <i class="fa-brands fa-github" aria-label="github"></i> GitHub</a></span></p>
</div>
  </div>
</footer>
<script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>