<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.4">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="AJ Smit">
<meta name="dcterms.date" content="2022-08-08">
<meta name="description" content="The Biodiversity and Conservation Biology R Pages serving BCB743.">

<title>BCB743: Quantitative Ecology – The Tangled Bank</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../BCB743/review.html" rel="next">
<link href="../images/favicon.ico" rel="icon">
<script src="../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-SCYH68M489"></script>

<script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-SCYH68M489', { 'anonymize_ip': true});
</script>

<script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"headline",
  "consent_type":"express",
  "palette":"dark",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  ,
"language":"en"
  });
});
</script> 
  
<style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<meta property="og:title" content="BCB743: Quantitative Ecology – The Tangled Bank">
<meta property="og:description" content="The Biodiversity and Conservation Biology R Pages serving BCB743.">
<meta property="og:image" content="http://tangledbank.netlify.app/BCB743/data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==">
<meta property="og:site_name" content="The Tangled Bank">
</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">The Tangled Bank</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-vignettes" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Vignettes</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-vignettes">    
        <li>
    <a class="dropdown-item" href="../vignettes/netCDF_dates.html">
 <span class="dropdown-text">Dates From netCDF Files: Two Approaches</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/prep_NOAA_OISST.html">
 <span class="dropdown-text">Downloading and Prep: ERDDAP</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_ERDDAP.html">
 <span class="dropdown-text">Retrieving Chlorophyll-a Data from ERDDAP Servers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/alt_method.html">
 <span class="dropdown-text">Downloading and Prep: Python + CDO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/gridded_data.html">
 <span class="dropdown-text">Detecting Events in Gridded Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/regridding.html">
 <span class="dropdown-text">Regridding gridded data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/MHW_MCS_horizonplots.html">
 <span class="dropdown-text">Displaying MHWs and MCSs as Horizon Plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/heatwaveR_issues.html">
 <span class="dropdown-text">heatwaveR Issues</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_sightings.html">
 <span class="dropdown-text">Plotting the Whale Sightings and Chlorophyll-<em>a</em> Concentrations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_localisation.html">
 <span class="dropdown-text">Spatial Localisation, Subsetting, and Aggregation of the Chlorophyll-<em>a</em> Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/elem_ts_methods.html">
 <span class="dropdown-text">Wavelet Analysis of Diatom Time Series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/buffer_data_extract.html">
 <span class="dropdown-text">Extracting gridded data within a buffer</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-high-performace-computing" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">High-Performace Computing</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-high-performace-computing">    
        <li>
    <a class="dropdown-item" href="../vignettes/README_Lengau.html">
 <span class="dropdown-text">Using Lengau</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/PBSPro_users.html">
 <span class="dropdown-text">PBSPro user and job management</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/README_tmux.html">
 <span class="dropdown-text">Using tmux</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../blog.html"> 
<span class="menu-text">Tangled Bank Blog</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-blogroll" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Blogroll</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-blogroll">    
        <li>
    <a class="dropdown-item" href="https://www.r-bloggers.com">
 <span class="dropdown-text">R-bloggers</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="../blog.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/ajsmit/tangled_bank">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://bugs.com">
 <span class="dropdown-text">Report a Bug</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BCB743/BCB743_index.html">BCB743 Quantitative Ecology</a></li><li class="breadcrumb-item"><a href="../BCB743/BCB743_index.html">Overview &amp; Introduction</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../images/hex_logo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">BCB UNDERGRADUATE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC223 Ecophysiology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/BDC223_FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC334 Biogeography &amp; Global Ecology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/BDC334_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BDC334: Biogeography &amp; Global Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/L01a-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1a: Setting the Scene</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/L01b-macroecology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1b: Macroecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/L02a-gradients.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2a: Ecological Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/L02b-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2b: Metrics of Environmental and Species Diversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1: Ecological Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/02a-r_rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2a: R &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/02b-env_dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2b: Environmental Distance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/03-biodiversity1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3: Quantifying Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/04-biodiversity2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4: Describing Biodiversity Patterns</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/Transboundary_systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transboundary systems</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">BCB HONOURS CORE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">BCB744 Intro R</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/BCB744_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744: Introduction to R, and Biostatistics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/01-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. R and RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/02-working-with-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Working with data and code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/03-workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. R workflows</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/04-graphics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Graphics with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/05-faceting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Faceting figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/06-brewing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Brewing colours</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/07-mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Mapping with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/08-mapping_style.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Mapping with style</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/09-mapping_rnaturalearth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Mapping with Natural Earth and the <strong>sf</strong> package</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/10-mapping_quakes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. The Fiji Earthquake data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/11-tidy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Tidy data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/12-tidier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Tidier data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/13-tidiest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Tidiest data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/14-recap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Recap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/15-functions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15. Functions by chapter</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/16-base_r.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16. Base R primer</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/17-dates.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">17. Dates</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">BCB744 Biostatistics</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/BCB744_scientific_philosophy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Biostatistics and the philosophy of scientific inquiry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/01-data-in-R.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Data classes and structures in R</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/02-summarise-and-describe.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Exploring with summaries and descriptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/03-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Exploring with figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/04-distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/05-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Statistical inference and hypothesis testing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/06-assumptions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Assumptions for parametric statistics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/07-t_tests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. <em>t</em>-tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/08-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Analysis of Variance (ANOVA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/09-regressions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Simple linear regressions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/10-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/11-decision_guide.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. A guide to selecting the right parametric test</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/12-glance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Non-parametric statistical tests at a glance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/13-confidence.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Confidence intervals</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/14-transformations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Data transformations</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">BCB HONOURS ELECTIVES</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true">
 <span class="menu-text">BCB743 Quantitative Ecology</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/BCB743_index.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Overview &amp; Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/review.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1-4: Review Concepts fo Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5: Correlations &amp; Associations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7: Intro to Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8a: Principal Components Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA_examples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8b: PCA: Additional Examples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA_SDG_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8c: PCA: WHO SDG Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/CA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9a: Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/DCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9b: Detrended Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCoA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10: Principal Coordinates Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/nMDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11a: Non-metric Multidimensional Scaling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/nMDS_diatoms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11b: nMDS: PERMANOVA (Diatoms) Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/unconstrained-summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12: Unconstrained Ordination Summary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/model_building.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13: Model Building</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/multiple_regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14a: Multiple Regression</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/deep_dive.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14b: Gradients Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/constrained_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15a: Distance-Based Redundancy Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/two_oceans_appendices.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15b: db-RDA: Seaweeds Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/cluster_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16: Cluster Analysis</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">SUPPORT</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="false">
 <span class="menu-text">Web Resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/general_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/ecology_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quantitative Ecology Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/spatial_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial R Resources</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#honours-coordinator" id="toc-honours-coordinator" class="nav-link active" data-scroll-target="#honours-coordinator"><span class="header-section-number">1</span> Honours Coordinator</a></li>
  <li><a href="#module-coordinator" id="toc-module-coordinator" class="nav-link" data-scroll-target="#module-coordinator"><span class="header-section-number">2</span> Module Coordinator</a></li>
  <li><a href="#instructors" id="toc-instructors" class="nav-link" data-scroll-target="#instructors"><span class="header-section-number">3</span> Instructors</a></li>
  <li><a href="#module-description" id="toc-module-description" class="nav-link" data-scroll-target="#module-description"><span class="header-section-number">4</span> Module Description</a></li>
  <li><a href="#module-content-and-framework" id="toc-module-content-and-framework" class="nav-link" data-scroll-target="#module-content-and-framework"><span class="header-section-number">5</span> Module Content and Framework</a></li>
  <li><a href="#outcomes-of-bcb743" id="toc-outcomes-of-bcb743" class="nav-link" data-scroll-target="#outcomes-of-bcb743"><span class="header-section-number">6</span> Outcomes of BCB743</a></li>
  <li><a href="#graduate-attributes" id="toc-graduate-attributes" class="nav-link" data-scroll-target="#graduate-attributes"><span class="header-section-number">7</span> Graduate Attributes</a></li>
  <li><a href="#data-and-reading-in-support-of-the-syllabus" id="toc-data-and-reading-in-support-of-the-syllabus" class="nav-link" data-scroll-target="#data-and-reading-in-support-of-the-syllabus"><span class="header-section-number">8</span> Data and Reading in Support of the Syllabus</a>
  <ul class="collapse">
  <li><a href="#reading" id="toc-reading" class="nav-link" data-scroll-target="#reading"><span class="header-section-number">8.1</span> Reading</a></li>
  <li><a href="#datasets-used-in-this-module" id="toc-datasets-used-in-this-module" class="nav-link" data-scroll-target="#datasets-used-in-this-module"><span class="header-section-number">8.2</span> Datasets Used in This Module</a></li>
  </ul></li>
  <li><a href="#prerequisites" id="toc-prerequisites" class="nav-link" data-scroll-target="#prerequisites"><span class="header-section-number">9</span> Prerequisites</a></li>
  <li><a href="#method-of-instruction" id="toc-method-of-instruction" class="nav-link" data-scroll-target="#method-of-instruction"><span class="header-section-number">10</span> Method of Instruction</a></li>
  <li><a href="#learning-colaboratively" id="toc-learning-colaboratively" class="nav-link" data-scroll-target="#learning-colaboratively"><span class="header-section-number">11</span> Learning Colaboratively</a></li>
  <li><a href="#reusing-code-found-elsewhere" id="toc-reusing-code-found-elsewhere" class="nav-link" data-scroll-target="#reusing-code-found-elsewhere"><span class="header-section-number">12</span> Reusing Code Found Elsewhere</a></li>
  <li><a href="#software" id="toc-software" class="nav-link" data-scroll-target="#software"><span class="header-section-number">13</span> Software</a></li>
  <li><a href="#computers" id="toc-computers" class="nav-link" data-scroll-target="#computers"><span class="header-section-number">14</span> Computers</a></li>
  <li><a href="#attendance" id="toc-attendance" class="nav-link" data-scroll-target="#attendance"><span class="header-section-number">15</span> Attendance</a></li>
  <li><a href="#sec-policy" id="toc-sec-policy" class="nav-link" data-scroll-target="#sec-policy"><span class="header-section-number">16</span> Assessment Policy</a>
  <ul class="collapse">
  <li><a href="#continuous-assessment" id="toc-continuous-assessment" class="nav-link" data-scroll-target="#continuous-assessment"><span class="header-section-number">16.1</span> Continuous Assessment</a></li>
  <li><a href="#task-g-final-assessment-exam" id="toc-task-g-final-assessment-exam" class="nav-link" data-scroll-target="#task-g-final-assessment-exam"><span class="header-section-number">16.2</span> Task G: Final Assessment (Exam)</a></li>
  <li><a href="#submission-of-assignments-and-exams" id="toc-submission-of-assignments-and-exams" class="nav-link" data-scroll-target="#submission-of-assignments-and-exams"><span class="header-section-number">16.3</span> Submission of Assignments and Exams</a></li>
  <li><a href="#late-submission-of-ca" id="toc-late-submission-of-ca" class="nav-link" data-scroll-target="#late-submission-of-ca"><span class="header-section-number">16.4</span> Late Submission of CA</a></li>
  </ul></li>
  <li><a href="#support" id="toc-support" class="nav-link" data-scroll-target="#support"><span class="header-section-number">17</span> Support</a>
  <ul class="collapse">
  <li><a href="#help-via-bcb744-and-bcb743-issues-on-github" id="toc-help-via-bcb744-and-bcb743-issues-on-github" class="nav-link" data-scroll-target="#help-via-bcb744-and-bcb743-issues-on-github"><span class="header-section-number">17.1</span> Help Via BCB744 and BCB743 Issues on GitHub</a></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references"><span class="header-section-number">18</span> References</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB743/BCB743_index.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BCB743/BCB743_index.html">BCB743 Quantitative Ecology</a></li><li class="breadcrumb-item"><a href="../BCB743/BCB743_index.html">Overview &amp; Introduction</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">BCB743: Quantitative Ecology</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>

<div>
  <div class="description">
    The Biodiversity and Conservation Biology R Pages serving BCB743.
  </div>
</div>

<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://tangledbank.netlify.app">AJ Smit</a> <a href="https://orcid.org/0000-0002-3799-6126" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uwc.ac.za">
            University of the Western Cape
            </a>
          </p>
      </div>
  </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">August 8, 2022</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<blockquote class="blockquote">
<p><em>“We have become, by the power of a glorious evolutionary accident called intelligence, the stewards of life’s continuity on earth. We did not ask for this role, but we cannot abjure it. We may not be suited to it, but here we are.”</em></p>
<p>— Stephen J. Gould</p>
</blockquote>
<blockquote class="blockquote">
<p><em>“Reports that say that something hasn’t happened are always interesting to me, because as we know, there are known knowns; there are things we know we know. We also know there are known unknowns; that is to say we know there are some things we do not know. But there are also unknown unknowns- the ones we don’t know we don’t know.”</em></p>
<p>— Donald Rumsfeld</p>
</blockquote>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/PhD_comics3.JPG" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p><strong>Welcome to the pages for BCB743 Quantitative Ecology. This page provides the syllabus and teaching policies for the module, and it serves is a starting point accessing all the theory, instruction, and data.</strong></p>
<section id="honours-coordinator" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Honours Coordinator</h1>
<p>Prof.&nbsp;Bryan Maritz—Room 4.105, Department of Biodiversity &amp; Conservation Biology</p>
</section>
<section id="module-coordinator" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Module Coordinator</h1>
<p>Prof.&nbsp;AJ Smit—Room 4.103, Department of Biodiversity &amp; Conservation Biology, ajsmit@uwc.ac.za</p>
</section>
<section id="instructors" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Instructors</h1>
<p>Isma-eel Jattiem—Department of Biodiversity &amp; Conservation Biology, 4035085@myuwc.ac.za</p>
<p>Zoë-Angelique Petersen—Department of Biodiversity &amp; Conservation Biology, 4042512@myuwc.ac.za</p>
</section>
<section id="module-description" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Module Description</h1>
<p>Quantitative ecology employs statistical and computational techniques to comprehend ecosystems. It aims to describe and quantify ecological processes, analyse and model complex multivariate ecological data, and make predictions about the structure and dynamics of ecosystems across various spatial and temporal scales. The multivariate statistical approaches taught in this module will equip you with the ability to interpret multidimensional data in a comprehensible two- or three-dimensional space.</p>
<p>In this course, you will cover the following topics:</p>
<ul>
<li><p><strong>Ecological Structure:</strong> You will explore the fundamental principles underlying the environmental structuring of ecosystems (ecosystem structure).</p></li>
<li><p><strong>Ecological Data Analysis:</strong> In this section, you will examine approaches to analyse ecological data, including hypothesis testing, regression analysis, and multivariate analysis.</p></li>
<li><p><strong>Multivariate Analyses:</strong> Here, you will learn how to utilize multivariate statistics to make sense of complex systems, predict ecological outcomes, and understand the underlying mechanisms that drive ecological processes.</p></li>
<li><p><strong>Spatial Ecology:</strong> You will acquire knowledge on how to analyse and model spatial patterns in ecological data, including the distribution of species and habitats across landscapes.</p></li>
<li><p><strong>Community Ecology:</strong> The theory covered in this section will prepare you to analyse and model the interactions between species within ecological communities, such as competition, predation, and mutualism.</p></li>
<li><p><strong>Ecosystem Ecology:</strong> You will learn how to model and analyse the flow of energy and nutrients through ecosystems, including the roles of producers, consumers, and decomposers in ecological processes.</p></li>
</ul>
<p>This module will provide you with the skills and tools necessary to analyse and model ecological data, community structure and the processes operating within them, and make predictions about the structure and dynamics of ecosystems. You will also learn how to communicate your findings effectively to a range of audiences, including scientists, policymakers, and the general public.</p>
<p>In BCB743, I will primarily focus on multivariate statistics. Multivariate methods play a crucial role in ecology as they enable us to analyse and interpret complex datasets involving multiple variables and ecosystems teeming with species. Ecosystems are characterised by their complexity and interconnectedness, with numerous factors typically influencing the distribution and abundance of species within an ecosystem. Multivariate statistical methods allow you to identify the underlying patterns and relationships among these variables and to explore how they interact to shape ecosystems.</p>
<p>Some of the most commonly employed multivariate statistical techniques in quantitative ecology include ordination methods, such as principal component analysis (PCA) and correspondence analysis (CA), which are utilised to reduce complexity and permit the visualisation of patterns of species distribution in ecosystems. We will also learn how to incorporate multiple regression into these multivariate analyses through the use of redundancy analysis (RDA) and canonical correspondence analysis (CCA), and examine non-metric multidimensional scaling (NMDS). All these methods extend our ability to explore species-environment relationships, assess community composition (structure), and identify influential environmental variables driving species distributions. We will also explore the application of multivariate statistics in addressing critical ecological issues, such as biodiversity monitoring, ecosystem functioning, and the impacts of anthropogenic disturbances on ecosystems.</p>
</section>
<section id="module-content-and-framework" class="level1 page-columns page-full" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Module Content and Framework</h1>
<p>These links point to online resources such as datasets and R scripts in support of the video and PDF lecture material. It is essential that you work through these examples and workflows. Here is the Syllabus:</p>
<div class="column-page-inset-right">
<table class="caption-top table">
<thead>
<tr class="header">
<th>Week</th>
<th>Class date</th>
<th>Lecture</th>
<th>Topic</th>
<th>Class</th>
<th>Slides</th>
<th>Reading</th>
<th>Tasks</th>
<th>Tasks due</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>W1</td>
<td>11 June</td>
<td>L0</td>
<td>Overview &amp; Introduction</td>
<td></td>
<td><a href="../slides/BCB743_01_intro.pdf">♦︎</a></td>
<td><a href="../docs/BES-Reproducible-Code-2017.pdf">▤</a> <a href="../docs/BES-Data-Guide-2017_web.pdf">▤</a></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>Lecture Set 1</td>
<td>Ecological and Earth Data</td>
<td></td>
<td></td>
<td></td>
<td><a href="../BCB743/assessments/Task_A1.html">◒</a> Task A1</td>
<td>1 July</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td></td>
<td><strong>RECAP OF BIODIVERSITY</strong></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>L1: Revision</td>
<td>Ecological Data</td>
<td><a href="../BDC334/01-introduction.html">★</a></td>
<td></td>
<td></td>
<td>Lab 1 Review</td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td></td>
<td><strong>DATA, MATRICES</strong></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>L2: Revision</td>
<td>Environmental Distance</td>
<td><a href="../BDC334/02b-env_dist.html">★</a></td>
<td><a href="../slides/BCB743_04_environmental_distance.pdf">♦︎</a></td>
<td></td>
<td>Lab 2 Review</td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>L3: Revision</td>
<td>Quantifying Biodiversity</td>
<td><a href="../BDC334/03-biodiversity1.html">★</a></td>
<td><a href="../slides/BCB743_02_biodiversity_1.pdf">♦︎</a></td>
<td><a href="../docs/Smit_et_al_2013.pdf">▤</a> <a href="../docs/Smit_et_al_2017.pdf">▤</a> <a href="../docs/Smit_the_seaweed_data.pdf">▤</a></td>
<td>Lab 3 Review</td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>L4: Revision</td>
<td>Describing Biodiversity Patterns</td>
<td><a href="../BDC334/04-biodiversity2.html">★</a></td>
<td><a href="../slides/BCB743_02_biodiversity_2.pdf">♦︎</a></td>
<td><a href="../docs/Shade_et_al_2018.pdf">▤</a></td>
<td>Lab 4 Review</td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>Lecture Set 2</td>
<td>Ecological Theories</td>
<td></td>
<td></td>
<td></td>
<td><a href="../BCB743/assessments/Task_A2.html">◒</a> Task A2</td>
<td>11 July</td>
</tr>
<tr class="even">
<td></td>
<td>13 June</td>
<td>L5</td>
<td>Correlations &amp; Associations</td>
<td><a href="../BCB743/correlations.html">★</a></td>
<td><a href="../slides/BCB743_06_correlations.pdf">♦︎</a></td>
<td></td>
<td><a href="../BCB743/assessments/Task_B.html">◒</a> Task B</td>
<td>17 June</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>L6: Self</td>
<td>Distance Metrics</td>
<td><a href="../BCB743/dis-metrics.html">★</a></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td><strong>UNCONSTRAINED ORDINATION</strong></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>W2</td>
<td>17 June</td>
<td>L7</td>
<td>Intro to Ordination</td>
<td></td>
<td><a href="../slides/BCB743_07_ordination.pdf">♦︎</a></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>L8</td>
<td>PCA</td>
<td><a href="../BCB743/PCA.html">★</a></td>
<td><a href="../slides/BCB743_08_PCA.pdf">♦︎</a></td>
<td></td>
<td><a href="../BCB743/assessments/Task_C.html">◒</a> Task C</td>
<td>24 June</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>L8: Self</td>
<td>PCA: Additional Examples</td>
<td><a href="../BCB743/PCA_examples.html">★</a></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>L8: Self</td>
<td>PCA: WHO SDG Example</td>
<td><a href="../BCB743/PCA_SDG_example.html">★</a></td>
<td></td>
<td></td>
<td><a href="../BCB743/assessments/Task_D.html">◒</a> Task D</td>
<td>11 July</td>
</tr>
<tr class="odd">
<td></td>
<td>20 June</td>
<td>L9a</td>
<td>CA</td>
<td><a href="../BCB743/CA.html">★</a></td>
<td><a href="../slides/BCB743_09_CA.pdf">♦︎</a></td>
<td></td>
<td><a href="../BCB743/assessments/Task_E.html">◒</a> Task E</td>
<td>24 June</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>L9b</td>
<td>DCA</td>
<td><a href="../BCB743/DCA.html">★</a></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>W3</td>
<td>24 June</td>
<td>L10</td>
<td>PCoA</td>
<td><a href="../BCB743/PCoA.html">★</a></td>
<td><a href="../slides/BCB743_10_PCoA.pdf">♦︎</a></td>
<td></td>
<td><a href="../BCB743/assessments/BCB743_intgrative_assignment.html">◒</a> Integrative Assignment</td>
<td>16 July</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>L11</td>
<td>nMDS</td>
<td><a href="../BCB743/nMDS.html">★</a></td>
<td><a href="../slides/BCB743_11_nMDS.pdf">♦︎</a></td>
<td></td>
<td><a href="../BCB743/assessments/Task_F.html">◒</a> Task F</td>
<td>1 July</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>L11: Self</td>
<td>nMDS: PERMANOVA (Diatoms) Example</td>
<td><a href="../BCB743/nMDS_diatoms.html">★</a></td>
<td></td>
<td><a href="../docs/Mayombo_et_al_2019.pdf">▤</a></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>L12: Self</td>
<td>Unconstrained Ordi. Summary</td>
<td><a href="../BCB743/unconstrained-summary.html">★</a></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>W4</td>
<td>1 July</td>
<td>Lecture Set 1</td>
<td>Ecological &amp; Earth Data</td>
<td></td>
<td></td>
<td></td>
<td>Present Lecture Set 1</td>
<td>1 July</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td><strong>REGRESSION ANALYSIS</strong></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td>4 July</td>
<td>L13</td>
<td>Model Building</td>
<td><a href="../BCB743/model_building.html">★</a></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>L14</td>
<td>Multiple Regression</td>
<td><a href="../BCB743/multiple_regression.html">★</a></td>
<td></td>
<td><a href="../docs/The-Biostatistics-Book-Ch-3-5.pdf">▤</a></td>
<td><a href="../BCB743/assessments/Task_G.html">◒</a> Task G (Final Assessment)</td>
<td><strong>16 July</strong></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>L14: Self</td>
<td>Gradients Example</td>
<td><a href="../BCB743/deep_dive.html">★</a></td>
<td></td>
<td><a href="../docs/Smit_et_al_2013.pdf">▤</a> <a href="../docs/Smit_et_al_2017.pdf">▤</a> <a href="../docs/Smit_the_seaweed_data.pdf">▤</a></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>Lx</td>
<td>Generalised Linear Models</td>
<td>TBA (2025)</td>
<td>ΤΒΑ (2025)</td>
<td>TBA (2025)</td>
<td>ΤΒΑ (2025)</td>
<td></td>
</tr>
<tr class="odd">
<td>W5</td>
<td>8 July</td>
<td>Lx</td>
<td>Generalised Additive Models</td>
<td>TBA (2025)</td>
<td>ΤΒΑ (2025)</td>
<td>TBA (2025)</td>
<td>ΤΒΑ (2025)</td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td><strong>CONSTRAINED ORDINATION</strong></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>L15</td>
<td>Distance-Based Redundancy Analysis</td>
<td><a href="../BCB743/constrained_ordination.html">★</a></td>
<td><a href="../slides/BCB743_12_constrained_ordination.pdf">♦︎</a></td>
<td><a href="../docs/Smit_et_al_2017.pdf">▤</a> <a href="../docs/Smit_the_seaweed_data.pdf">▤</a></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>L15: Self</td>
<td>db-RDA: Seaweeds Example</td>
<td><a href="Seaweed_in_Two_Oceans_v2/two_oceans_appendices.qmd">★</a></td>
<td></td>
<td><a href="../docs/Smit_et_al_2017.pdf">▤</a> <a href="../docs/Smit_the_seaweed_data.pdf">▤</a></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td></td>
<td><strong>CLUSTER ANALYSIS</strong></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>L16</td>
<td>Cluster Analysis</td>
<td><a href="../BCB743/cluster_analysis.html">★</a></td>
<td></td>
<td></td>
<td><a href="../BCB743/assessments/Task_D.html">◒</a> Task D (continue)</td>
<td>11 July</td>
</tr>
<tr class="odd">
<td></td>
<td>11 July</td>
<td>Lecture Set 2</td>
<td>Ecological Theories</td>
<td></td>
<td></td>
<td></td>
<td>Present Lecture Set 2</td>
<td>11 July</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>L17</td>
<td>Review</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
</div>
<p><strong>Core theoretical framework</strong> Ecological hypotheses underlying the processes of species assembly in space and time, including neutral and niche-based mechanisms, and historical events; overview of the currently known and understood distributional patterns of major groups of organisms at global, regional and local scales; consideration of sampling designs aimed at capturing these patterns and drivers so as to arrive at a processed based understanding of species assembly.</p>
<p><strong>Competence</strong> Data collection aimed at a quantitative test of the relevant hypotheses, above. The management and analysis of ecological data; reproducible and collaborative research; the use of R as a tool for the analysis multivariate ecological data; multivariate techniques such as nMDS, PCA, RDA and cluster analysis; graphical data summaries and visualisations.</p>
</section>
<section id="outcomes-of-bcb743" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Outcomes of BCB743</h1>
<p>By the end of this module, students will be able to:</p>
<ul>
<li>Understand the concepts of <span class="math inline">\(\alpha\)</span>-, <span class="math inline">\(\beta\)</span>- and <span class="math inline">\(\gamma\)</span>-diversity</li>
<li>Know and understand the current hypotheses that explain species assembly processes in space and time (e.g.&nbsp;neutral and niche mechanisms)</li>
<li>Collect ecological data at the appropriate scale, which would lend themselves to a quantitative analysis of points 1 and 2, above</li>
<li>Use the R software and associated packages to undertake the analyses required in point 3, above</li>
<li>Interpret the outcomes of the above analyses and use it to quantitatively characterise points 1 and 2, above</li>
<li>Communicate the findings by written and oral means</li>
</ul>
</section>
<section id="graduate-attributes" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> Graduate Attributes</h1>
<p>The <a href="../pages/graduate_attributes.html"><strong>graduate attributes</strong></a> resulting from completion of this modules alignment with the expectations of the workspace across diverse organisations and institutions where graduates typically find employment.</p>
</section>
<section id="data-and-reading-in-support-of-the-syllabus" class="level1" data-number="8">
<h1 data-number="8"><span class="header-section-number">8</span> Data and Reading in Support of the Syllabus</h1>
<p>In the table above there are links to several key papers to read in preparation of each week’s theory. It is essential that you read these papers.</p>
<p>Many other references are cited in each Chapter. These serve several functions in that they:</p>
<ul>
<li>Add additional theory relevant to some ecological concepts</li>
<li>Provide background to some of the datasets used in my examples</li>
<li>Discuss derivations of some equations used to calculate diversity concepts</li>
<li>Provide example walkthroughs of some of the computational aspects of the methods covered in the Labs</li>
<li>Collectively supplement the discussion about these concepts covered in the lectures</li>
</ul>
<p>Actively engaging with these reading materials will make to difference between a 60% average mark for the module, and a mark in excess of 80%.</p>
<section id="reading" class="level2" data-number="8.1">
<h2 data-number="8.1" class="anchored" data-anchor-id="reading"><span class="header-section-number">8.1</span> Reading</h2>
<p>You are <strong>expected to read additional material</strong> in support of the content covered in class and on this website.</p>
<p>A <strong>compulsory</strong> reference is <a href="http://adn.biol.umontreal.ca/~numericalecology/numecolR/">‘Numerical Ecology with R’</a> by Daniel Borcard, François Gillet and Pierre Legendre <span class="citation" data-cites="borcard2011numerical">(<a href="#ref-borcard2011numerical" role="doc-biblioref">Borcard et al. 2011</a>)</span>. Much of the class’ content and many of the examples (and code) that I use have been adapted from this source. There is also the excellent book by <span class="citation" data-cites="legendre2012numerical">Legendre and Legendre (<a href="#ref-legendre2012numerical" role="doc-biblioref">2012</a>)</span> called ‘Numerical Ecology’ which provides everything the former book has, but in greater detail and with less focus on R. Both should be considered a ‘gold standard’ reference for Quantitative Ecology.</p>
<p>A third <strong>highly recommended text</strong> is the book <a href="http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf">Tree Diversity Analysis</a> by Roeland Kindt and Richard Coe.</p>
<p>I can also recommend a these amazing websites with excellent content:</p>
<ul>
<li>David Zelený’s <a href="https://www.davidzeleny.net/anadat-r/doku.php">Analysis of Community Ecology Data in R</a></li>
<li>Mike Palmer’s <a href="http://ordination.okstate.edu/">Ordination Methods for Ecologists</a></li>
<li><a href="https://sites.google.com/site/mb3gustame/">GUide to STatistical Analysis in Microbial Ecology (GUSTA ME)</a></li>
</ul>
<p>Note that the URLs with links to additional reading that appear with the worked-through example code should <strong>not be seen as optional</strong>. They are there for a reason and should be consulted even though I might not necessarily refer to each of them in class. Use these materials liberally.</p>
<p>Should you want to download the source code for the BCB743 (and BCB744 website), you may find it on <a href="https://github.com/ajsmit/R_courses"><i class="fab fa-github"></i> GitHub</a>.</p>
</section>
<section id="datasets-used-in-this-module" class="level2" data-number="8.2">
<h2 data-number="8.2" class="anchored" data-anchor-id="datasets-used-in-this-module"><span class="header-section-number">8.2</span> Datasets Used in This Module</h2>
<p>Note that the links provided might not necessarily lead to the <strong>vegan</strong> help page.</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th>Dataset</th>
<th>Source</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td>Vegetation and Environment in Dutch Dune Meadows</td>
<td><a href="https://www.davidzeleny.net/anadat-r/doku.php/en:data:dune"><strong>vegan</strong></a></td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td>Oribatid Mite Data with Explanatory Variables</td>
<td><a href="http://adn.biol.umontreal.ca/~numericalecology/data/oribates.html"><strong>vegan</strong></a></td>
</tr>
<tr class="odd">
<td style="text-align: left;">3</td>
<td>The Doubs River Data</td>
<td><a href="https://www.davidzeleny.net/anadat-r/doku.php/en:data:doubs"><strong>Numerical Ecology with R</strong></a></td>
</tr>
<tr class="even">
<td style="text-align: left;">4</td>
<td>The Barro Colorado Island Tree Counts</td>
<td><a href="https://www.davidzeleny.net/anadat-r/doku.php/en:data:bci"><strong>vegan</strong></a></td>
</tr>
<tr class="odd">
<td style="text-align: left;">5</td>
<td>John Bolton, Rob Anderson, and Herre Stegenga’s Seaweed Data</td>
<td><a href="https://www.frontiersin.org/articles/10.3389/fmars.2017.00404/"><strong>Smit et al., 2017</strong></a></td>
</tr>
<tr class="even">
<td style="text-align: left;">6</td>
<td>Serge Mayombo’s Diatoms Data</td>
<td><a href="https://www.tandfonline.com/doi/abs/10.2989/1814232X.2019.1592778"><strong>Mayombo et al., 2019</strong></a></td>
</tr>
<tr class="odd">
<td style="text-align: left;">7</td>
<td>World Health Organization Sustainable Development Goals Data</td>
<td><a href="https://www.who.int/data"><strong>WHO</strong></a></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="prerequisites" class="level1" data-number="9">
<h1 data-number="9"><span class="header-section-number">9</span> Prerequisites</h1>
<p>You should have a moderate numerical literacy and have prior programming experience. Such experience will have been obtained in the <a href="../workshops/">BCB744</a> module, which is a module about doing statistics in R. If you have a reasonable experience in coding and statistical analysis you should find yourself well prepared. You should also thoroughly revise <a href="../BDC334/BDC334_index.html">BDC334</a> by the end of the first week of this module.</p>
</section>
<section id="method-of-instruction" class="level1" data-number="10">
<h1 data-number="10"><span class="header-section-number">10</span> Method of Instruction</h1>
<p>You are provided with reading material (lecture slides, code, website content) that you are expected to consume <strong>prior to the class</strong>. Classes will involve brief introductions to new concepts, and will be followed by working on exercises in class that cover those concepts. The workshop is designed to be as interactive as possible, so while you are working on exercises the tutor and I will circulate among you and engage with you to help you understand any material and the associated code you are uncomfortable with. Often this will result in discussions of novel applications and alternative approaches to the data analysis challenges you are required to solve. More challenging concepts might emerge during the assignments (typically these will be submitted the following day), and any such challenges will be dealt with in class prior to learning new concepts.</p>
<p>Although the module is theory-heavy, a large part of it is also about coding. It is up to you to take your coding skills to the next level and move beyond what I teach in class. Coding is a bit like learning a language, and as such programming is a skill that is best learned by doing.</p>
</section>
<section id="learning-colaboratively" class="level1" data-number="11">
<h1 data-number="11"><span class="header-section-number">11</span> Learning Colaboratively</h1>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Also read: How to learn
</div>
</div>
<div class="callout-body-container callout-body">
<p>Please refer to my <a href="../pages/How_to_learn.html">advice about how to learn</a>.</p>
</div>
</div>
<p>Discuss the BCB743 workshop activities with your peers as you work on them. Use also the WhatsApp group set up for the module for discussion purposes (I might assist via this medium if neccesary if your questions/comments have relevance to the whole class). A better option is to use <a href="../quantecol/#help-via-bcb744-and-bcb743-issues-on-github">GitHub Issues</a>. You will learn more in this module if you work with your friends than if you do not. Ask questions, answer questions, and share ideas liberally. Please identify your work partners by name on all assignments (if you decide to work in pairs).</p>
<p><strong>Cooperative learning is not a licence for plagiarism. Plagiarism is a serious offence and will be dealt with concisely. Consequences of cheating are severe—they range from a 0% for the assignment or exam up to dismissal from the course for a second offense.</strong></p>
</section>
<section id="reusing-code-found-elsewhere" class="level1" data-number="12">
<h1 data-number="12"><span class="header-section-number">12</span> Reusing Code Found Elsewhere</h1>
<p>A huge volume of code is available on the web and it can be adapted to solve your own problems. You may make use of any online resources (e.g.&nbsp;form <a href="https://stackoverflow.com/">StackOverflow</a>, a thoroughly-used source of discussion about <a href="https://stackoverflow.com/questions/tagged/r">R code</a>)—but you <strong>MUST</strong> clearly indicate (cite) that your solution relies on found code, regardless to what extent you have modified it to your own needs. Reused code that is discovered via a web search and which is not explicitly cited is plagiarism and it will be treated as such. On assignments you may not directly share code with your peers in this workshop.</p>
</section>
<section id="software" class="level1" data-number="13">
<h1 data-number="13"><span class="header-section-number">13</span> Software</h1>
<p>In this course we will rely entirely on <a href="https://cran.r-project.org/">R</a> running within the <a href="https://www.rstudio.com/">RStudio</a> IDE. The use of R was covered extensively in the <a href="http://localhost:4321/workshops/">BCB744</a> module where the <a href="http://localhost:4321/workshops/intro_r/chapters/02-rstudio/">installation process</a> was discussed. We will primarily use the <a href="https://cran.r-project.org/web/packages/vegan/index.html"><strong>vegan</strong></a> package, but some useful functions are also provided by the package <a href="https://github.com/cran/BiodiversityR"><strong>BiodiversityR</strong></a> (and <a href="http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf">here</a> and <a href="https://rpubs.com/Roeland-KINDT">here</a>). Various other R packages offer overlapping and additional methods, but <strong>vegan</strong> should accommodate &gt;90% of your Quantitative Ecology needs.</p>
</section>
<section id="computers" class="level1" data-number="14">
<h1 data-number="14"><span class="header-section-number">14</span> Computers</h1>
<p>You are encouraged to provide your own laptops and to install the necessary software before the module starts. Limited support can be provided if required. There are also computers with R and RStudio (and the neccesary add-on libraries) available in the 5th floor lab in the BCB Department.</p>
</section>
<section id="attendance" class="level1" data-number="15">
<h1 data-number="15"><span class="header-section-number">15</span> Attendance</h1>
<p>This worskhop-based, hands on course can only deliver acceptible outcomes if you attend all classes. The schedule is set and cannot be changed. Sometimes an occasional absence cannot be avoided. Please be curtious and notify myself or the tutor in advance of any absence. If you work with a partner in class, notify them too. Keep up with the reading assignments while you are away and we will all work with you to get you back up to speed on what you miss. If you do miss a class, however, the assignments must still be submitted on time (also see <a href="../quantecol/#late-submission-of-ca"><strong>Late submission of CA</strong></a>).</p>
<p>Since you may decide to work in collaboration with a peer on tasks and assignments, please keep this person informed at all times in case some emergency makes you unavailable for a period of time. Someone might depend on your input and contributions—do not leave someone in the lurch so that they cannot complete a task in your absence.</p>
<p>Sure, here is the improved and completed sentence and meaning at the end:</p>
<hr>
</section>
<section id="sec-policy" class="level1" data-number="16">
<h1 data-number="16"><span class="header-section-number">16</span> Assessment Policy</h1>
<p><strong>Continuous Assessments</strong> (CA) and a <strong>Final Assessment</strong> will provide a <strong>Final Mark</strong> for the module. They contribute 0.6 and 0.4, respectively. These modes of assessment meet our needs as far as <a href="../pages/assessment_theory.html">formative and summative assessments</a> are concerned. All assessments are open book, so consult your code and reading material if and when you need to.</p>
<section id="continuous-assessment" class="level2" data-number="16.1">
<h2 data-number="16.1" class="anchored" data-anchor-id="continuous-assessment"><span class="header-section-number">16.1</span> Continuous Assessment</h2>
<p>The Continuous Assessment is comprised of:</p>
<ul>
<li>Tasks A1 and A2, weighted 0.35 and 0.15, respectively, towards the CA</li>
<li>Lecture presentation, weighted 0.3 towards the CA</li>
<li>Tasks B to F, the average of which contributes 0.2 towards the CA</li>
</ul>
<p>For Tasks A1 and A2, please refer to the marking schedule as agreed on by the class.</p>
<p>When assessing Tasks B to F, we will pay attention to the following criteria:</p>
<ul>
<li>Presentation and formatting (10%), including:
<ul>
<li>Questions answered in order</li>
<li>Sectioning</li>
<li>General appearance</li>
<li>References (if required)</li>
</ul></li>
<li>Code formatting (10%), e.g.:
<ul>
<li>Application of <a href="http://adv-r.had.co.nz/Style.html">R code conventions</a>, e.g.&nbsp;spaces around <code>&lt;-</code>, after <code>#</code>, after <code>,</code>, etc.</li>
<li>New line for each <code>dplyr</code> function (lines end in <code>%&gt;%</code>)</li>
<li>New line for each <code>ggplot</code> layer (lines end in <code>+</code>)</li>
<li>Tidiness of code presentation in the HTML</li>
</ul></li>
<li>Code correctness in the context of the specified analysis (25%):
<ul>
<li>The code must faithfully execute the intended analysis as required by the research questions and/or hypotheses tests</li>
</ul></li>
<li>Figures (10%):
<ul>
<li>Sensible use of themes and colors</li>
<li>Publication quality (complete axis titles and labels, etc.)</li>
<li>Informative and complete titles, axis labels, legends, etc.</li>
</ul></li>
<li>Discussion (45%):
<ul>
<li>Here you will be assessed for integrating the results of your analyses within the correct theoretical framework</li>
</ul></li>
</ul>
</section>
<section id="task-g-final-assessment-exam" class="level2" data-number="16.2">
<h2 data-number="16.2" class="anchored" data-anchor-id="task-g-final-assessment-exam"><span class="header-section-number">16.2</span> Task G: Final Assessment (Exam)</h2>
<p>The Final Assessment starts after the Multiple Regression lecture on 4 July and you can do in the comfort of your home. It will involve the analysis of real world data and assess some more theoretical and philosophical aspects of Quantitative Ecology. A full mark breakdown is provided with <a href="../BCB743/assessments/Task_G.html">Task G</a>.</p>
</section>
<section id="submission-of-assignments-and-exams" class="level2" data-number="16.3">
<h2 data-number="16.3" class="anchored" data-anchor-id="submission-of-assignments-and-exams"><span class="header-section-number">16.3</span> Submission of Assignments and Exams</h2>
<p>A statement such as the one below accompanies every assignment—pay attention, as failing to observe this instruction may result in a loss of marks (i.e.&nbsp;if an assignment remains ungraded because the owner of the material cannot be identified):</p>
<p>Submit a R script wherein you provide answers to Questions 1–9 by no later than 8:00 tomorrow. Label the script as follows (e.g.): <strong>BCB743_AJ_Smit_Assignment_2.R</strong>.</p>
</section>
<section id="late-submission-of-ca" class="level2" data-number="16.4">
<h2 data-number="16.4" class="anchored" data-anchor-id="late-submission-of-ca"><span class="header-section-number">16.4</span> Late Submission of CA</h2>
<p>Late assignments will be penalised 10% per day and will not be accepted more than 48 hours late, unless evidence such as a doctor’s note, a death certificate, or another documented emergency can be provided. If you know in advance that a submission will be late, please discuss this and seek prior approval. This policy is based on the idea that in order to learn how to translate your human thoughts into computer language (coding) you should be working with them at multiple times each week—ideally daily. Time has been allocated in class for working on assignments and students are expected to continue to work on the assignments outside of class. Successfully completing (and passing) this module requires that you finish assignments based on what we have covered in class by the following class period. Work diligently from the onset so that even if something unexpected happens at the last minute you should already be close to done. This approach also allows rapid feedback to be provided to you, which can only be accomplished by returning assignments quickly and punctually.</p>
</section>
</section>
<section id="support" class="level1" data-number="17">
<h1 data-number="17"><span class="header-section-number">17</span> Support</h1>
<p>It’s expected that some tricky aspects of the module will take time to master, and the best way to master problematic material is to practice, practice some more, and then to ask questions. Trying for 10 minutes and then giving up is not good enough. I’ll be more sympathetic to your cause if you can demonstrate having tried for a full day before giving up and asking me. When you ask questions about some challenge, this is the way to do it—explain to me your numerous attempts at trying to solve the problem, and explain how these various attempts have failed. <em>I will not help you if you have not tried to help yourself first</em> (maybe with advice from friends). There will be time in class to do this, typically before we embark on a new topic. You are also encouraged to bring up related questions that arise in your own B.Sc. (Hons.) research project.</p>
<p>Should you require more time with me, find out when I am ‘free’ and set an appointment by sending me a calendar invitation. I am happy to have a personal meeting with you via Zoom, but I prefer face-to-face in my office.</p>
<section id="help-via-bcb744-and-bcb743-issues-on-github" class="level2" data-number="17.1">
<h2 data-number="17.1" class="anchored" data-anchor-id="help-via-bcb744-and-bcb743-issues-on-github"><span class="header-section-number">17.1</span> Help Via BCB744 and BCB743 Issues on GitHub</h2>
<p>All discussion for the BCB744 and BCB743 workshops will be held in the <a href="https://github.com/ajsmit/R_courses/issues">Issues</a> of <a href="https://github.com/ajsmit/R_courses">this repository</a>. Please post all content-related questions there, and use email only for personal matters. Note that this is a public repository, so be professional in your writing here (grammar, etc.).</p>
<p>To start a new thread, create a <strong>New issue</strong>. Tag your peers using their handle—<code>@ajsmit</code>, for example—to get their attention.</p>
<p>Once a question has been answered, the issue will be closed, so lots of good answers might end up in closed issues. Don’t forget to look there when looking for answers—you can use the <strong>Search</strong> feature on this repository to find answers that might have been offered by the same or similar problem experienced by someone else in the past.</p>
<p><strong>Guidelines for Posting Questions:</strong></p>
<ul>
<li>First search existing issues (open or closed) for answers. If the question has already been answered, you’re done! If there is an open issue, feel free to contribute to it. Or feel free to open a closed issue if you believe the answer is not satisfactory.</li>
<li>Give your issue an informative title.
<ul>
<li>Good: “Error: could not find function”ggplot””</li>
<li>Bad: “My code does not work!” Note that you can edit an issue’s title after it’s been posted.</li>
</ul></li>
<li>Format your questions nicely using markdown and code formatting. Preview your issue prior to posting.</li>
<li>As I explained above, your peers and I will more sympathetic to your cause if you can show <em>all the things you have tried as you, yourself, tried to fix the issue first</em>.</li>
<li>Include code and example data so the person trying to help you have something to work with (and which results in the error, perhaps)</li>
<li>Where appropriate, provide links to specific files, or even lines within them, in the body of your issue. This will help your peers understand your question. Note that only the teaching team will have access to private repos.</li>
<li>(Optional) Tag someone or some group of people. Start by typing their GitHub username prefixed with the @ symbol. Of course this supposes that each of you have a GitHub account and username.</li>
<li>Hit <strong>Submit new issue</strong> when you’re ready to post.</li>
</ul>
<!-- ## Function list

  You will encounter the following functions for the first time in this module:

  | Function              | Description                                        |
  |:----------------------|----------------------------------------------------|
  | `adonis2()`           | Permutational Multivariate Analysis of Variance Using Distance Matrices |
  | `anova.betadisper()`  | PERMDIST 2 for Multivariate homogeneity of groups dispersions (variances) |
  | `anova.cca ()`        | Permutation Test for CCA, RDA and CA of Principal Coordinates |
  | `anova.manyglm()`     | Analysis of Deviance for Multivariate Generalized Linear Model Fits for Abundance Data |
  | `beta.pair()`         | Incidence-based Pair-wise Dissimilarities |
  | `betadisper()`        | Multivariate homogeneity of groups dispersions (variances) |
  | `betapart.core()`     | Core Calculations of betapart |
  | `biplot()`            | Biplot of Multivariate Data |
  | `capscale()`          | [Partial] Distance-based Redundancy Analysis |
  | `cca()`               | [Partial] [Constrained] Correspondence Analysis and Redundancy Analysis |
  | `cleanplot.pc()`      | Numerical Ecology in R's PCA Plot |
  | `daisy()`             | Dissimilarity Matrix Calculation |
  | `decostand()`         | Standardization Methods for Community Ecology |
  | `diag()`              | Matrix Diagonals |
  | `diversity()`         | Ecological Diversity Indices |
  | `diversityresult()`   | Alternative Diversity Results |
  | `fviz_cluster()`      | Visualize Clustering Results |
  | `fviz_nbclust()`      | Determining and Visualizing the Optimal Number of Clusters |
  | `fviz_silhouette()`   | Visualize Silhouette Information from Clustering |
  | `geodist()`           | Convert Lon-Lat Coordinates into Vector or Matrix of Geodesic Distances in Metres |
  | `ggcorrplot()`        | Visualisation of a Correlation Matrix Using ggplot2 |
  | `goodness()`          | Diagnostic Tools for [Constrained] Ordination (CCA, RDA, DCA, CA, PCA) |
  | `imputePCA()`         | Impute Dataset with PCA |
  | `lowerTriangle()`     | Extract or Replace the Upper/Lower Triangular Portion of a Matrix |
  | `manyglm()`           | Fitting Generalized Linear Models for Multivariate Abundance Data |
  | `meanvar.plot()`      | Construct Mean-Variance Plots for Multivariate Abundance Data |
  | `metaMDS()`           | Nonmetric Multidimensional Scaling with Stable Solution from Random Starts, Axis Scaling and Species Scores |
  | `mvabund()`           | Multivariate Abundance Data Objects |
  | `ordiellipse()`       | Display Groups or Factor Levels in Ordination Diagrams |
  | `ordiplot()`          | Alternative Plot and Identify Functions for Ordination |
  | `ordispider()`        | Display Groups or Factor Levels in Ordination Diagrams |
  | `ordisurf()`          | Fit and Plot Smooth Surfaces of Variables on Ordination |
  | `pam()`               | Partitioning Around Medoids |
  | `permutest()`         | Permutation Test for CCA, RDA and CA of Principal Coordinates |
  | `rda()`               | [Partial] [Constrained] Correspondence Analysis and Redundancy Analysis |
  | `RsquareAdj()`        | Adjusted R-square |
  | `scores()`            | Get Species or Site Scores from an Ordination |
  | `specnumber()`        | Ecological Diversity Indices |
  | `stressplot()`        | Goodness of Fit and Shepard Plot for Nonmetric Multidimensional Scaling |
  | `t()`                 | Matrix Transpose |
  | `vegdist()`           | Dissimilarity Indices for Community Ecologists |
  | `vif.cca()`           | Diagnostic Tools for [Constrained] Ordination (CCA, RDA, DCA, CA, PCA) | -->
</section>
</section>
<section id="references" class="level1" data-number="18">
<h1 data-number="18"><span class="header-section-number">18</span> References</h1>
<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-borcard2011numerical" class="csl-entry" role="listitem">
Borcard D, Gillet F, Legendre P, others (2011) Numerical ecology with <span>R</span>. Springer
</div>
<div id="ref-legendre2012numerical" class="csl-entry" role="listitem">
Legendre P, Legendre L (2012) Numerical ecology. Elsevier
</div>
</div>


<!-- -->

</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-reuse"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a></div></div></section><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{smit2022,
  author = {Smit, AJ},
  title = {BCB743: {Quantitative} {Ecology}},
  date = {2022-08-08},
  url = {http://tangledbank.netlify.app/BCB743/BCB743_index.html},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-smit2022" class="csl-entry quarto-appendix-citeas" role="listitem">
Smit A (2022) BCB743: Quantitative Ecology. <a href="http://tangledbank.netlify.app/BCB743/BCB743_index.html">http://tangledbank.netlify.app/BCB743/BCB743_index.html.
</a>
</div></div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("http:\/\/tangledbank\.netlify\.app");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
          // default icon
          link.classList.add("external");
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
  </div>
  <div class="nav-page nav-page-next">
      <a href="../BCB743/review.html" class="pagination-link" aria-label="1-4: Review Concepts fo Biodiversity">
        <span class="nav-page-text">1-4: Review Concepts fo Biodiversity</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb1" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb1-1"><a href="#cb1-1"></a><span class="co">---</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="an">date:</span><span class="co"> "2022-08-08"</span></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="an">title:</span><span class="co"> "BCB743: Quantitative Ecology"</span></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="an">description:</span><span class="co"> The Biodiversity and Conservation Biology R Pages serving BCB743.</span></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="co">---</span></span>
<span id="cb1-6"><a href="#cb1-6"></a></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="at">&gt; *"We have become, by the power of a glorious evolutionary accident called intelligence, the stewards of life's continuity on earth. We did not ask for this role, but we cannot abjure it. We may not be suited to it, but here we are."*</span></span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="at">&gt;</span></span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="at">&gt; --- Stephen J. Gould</span></span>
<span id="cb1-10"><a href="#cb1-10"></a></span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="at">&gt; *"Reports that say that something hasn't happened are always interesting to me, because as we know, there are known knowns; there are things we know we know. We also know there are known unknowns; that is to say we know there are some things we do not know. But there are also unknown unknowns- the ones we don't know we don't know."*</span></span>
<span id="cb1-12"><a href="#cb1-12"></a><span class="at">&gt;</span></span>
<span id="cb1-13"><a href="#cb1-13"></a><span class="at">&gt; --- Donald Rumsfeld</span></span>
<span id="cb1-14"><a href="#cb1-14"></a></span>
<span id="cb1-15"><a href="#cb1-15"></a><span class="al">![](../images/PhD_comics3.JPG)</span>{fig-align="center"}</span>
<span id="cb1-16"><a href="#cb1-16"></a></span>
<span id="cb1-17"><a href="#cb1-17"></a>**Welcome to the pages for BCB743 Quantitative Ecology. This page provides the syllabus and teaching policies for the module, and it serves is a starting point accessing all the theory, instruction, and data.**</span>
<span id="cb1-18"><a href="#cb1-18"></a></span>
<span id="cb1-19"><a href="#cb1-19"></a><span class="fu"># Honours Coordinator</span></span>
<span id="cb1-20"><a href="#cb1-20"></a></span>
<span id="cb1-21"><a href="#cb1-21"></a>Prof. Bryan Maritz---Room 4.105, Department of Biodiversity &amp; Conservation Biology</span>
<span id="cb1-22"><a href="#cb1-22"></a></span>
<span id="cb1-23"><a href="#cb1-23"></a><span class="fu"># Module Coordinator</span></span>
<span id="cb1-24"><a href="#cb1-24"></a></span>
<span id="cb1-25"><a href="#cb1-25"></a>Prof. AJ Smit---Room 4.103, Department of Biodiversity &amp; Conservation Biology, ajsmit@uwc.ac.za</span>
<span id="cb1-26"><a href="#cb1-26"></a></span>
<span id="cb1-27"><a href="#cb1-27"></a><span class="fu"># Instructors</span></span>
<span id="cb1-28"><a href="#cb1-28"></a></span>
<span id="cb1-29"><a href="#cb1-29"></a>Isma-eel Jattiem---Department of Biodiversity &amp; Conservation Biology, 4035085@myuwc.ac.za</span>
<span id="cb1-30"><a href="#cb1-30"></a></span>
<span id="cb1-31"><a href="#cb1-31"></a>Zoë-Angelique Petersen---Department of Biodiversity &amp; Conservation Biology, 4042512@myuwc.ac.za</span>
<span id="cb1-32"><a href="#cb1-32"></a></span>
<span id="cb1-33"><a href="#cb1-33"></a><span class="fu"># Module Description</span></span>
<span id="cb1-34"><a href="#cb1-34"></a></span>
<span id="cb1-35"><a href="#cb1-35"></a>Quantitative ecology employs statistical and computational techniques to comprehend ecosystems. It aims to describe and quantify ecological processes, analyse and model complex multivariate ecological data, and make predictions about the structure and dynamics of ecosystems across various spatial and temporal scales. The multivariate statistical approaches taught in this module will equip you with the ability to interpret multidimensional data in a comprehensible two- or three-dimensional space.</span>
<span id="cb1-36"><a href="#cb1-36"></a></span>
<span id="cb1-37"><a href="#cb1-37"></a>In this course, you will cover the following topics:</span>
<span id="cb1-38"><a href="#cb1-38"></a></span>
<span id="cb1-39"><a href="#cb1-39"></a><span class="ss">- </span>**Ecological Structure:** You will explore the fundamental principles underlying the environmental structuring of ecosystems (ecosystem structure).</span>
<span id="cb1-40"><a href="#cb1-40"></a></span>
<span id="cb1-41"><a href="#cb1-41"></a><span class="ss">- </span>**Ecological Data Analysis:** In this section, you will examine approaches to analyse ecological data, including hypothesis testing, regression analysis, and multivariate analysis.</span>
<span id="cb1-42"><a href="#cb1-42"></a></span>
<span id="cb1-43"><a href="#cb1-43"></a><span class="ss">- </span>**Multivariate Analyses:** Here, you will learn how to utilize multivariate statistics to make sense of complex systems, predict ecological outcomes, and understand the underlying mechanisms that drive ecological processes.</span>
<span id="cb1-44"><a href="#cb1-44"></a></span>
<span id="cb1-45"><a href="#cb1-45"></a><span class="ss">- </span>**Spatial Ecology:** You will acquire knowledge on how to analyse and model spatial patterns in ecological data, including the distribution of species and habitats across landscapes.</span>
<span id="cb1-46"><a href="#cb1-46"></a></span>
<span id="cb1-47"><a href="#cb1-47"></a><span class="ss">- </span>**Community Ecology:** The theory covered in this section will prepare you to analyse and model the interactions between species within ecological communities, such as competition, predation, and mutualism.</span>
<span id="cb1-48"><a href="#cb1-48"></a></span>
<span id="cb1-49"><a href="#cb1-49"></a><span class="ss">- </span>**Ecosystem Ecology:** You will learn how to model and analyse the flow of energy and nutrients through ecosystems, including the roles of producers, consumers, and decomposers in ecological processes.</span>
<span id="cb1-50"><a href="#cb1-50"></a></span>
<span id="cb1-51"><a href="#cb1-51"></a>This module will provide you with the skills and tools necessary to analyse and model ecological data, community structure and the processes operating within them, and make predictions about the structure and dynamics of ecosystems. You will also learn how to communicate your findings effectively to a range of audiences, including scientists, policymakers, and the general public.</span>
<span id="cb1-52"><a href="#cb1-52"></a></span>
<span id="cb1-53"><a href="#cb1-53"></a>In BCB743, I will primarily focus on multivariate statistics. Multivariate methods play a crucial role in ecology as they enable us to analyse and interpret complex datasets involving multiple variables and ecosystems teeming with species. Ecosystems are characterised by their complexity and interconnectedness, with numerous factors typically influencing the distribution and abundance of species within an ecosystem. Multivariate statistical methods allow you to identify the underlying patterns and relationships among these variables and to explore how they interact to shape ecosystems.</span>
<span id="cb1-54"><a href="#cb1-54"></a></span>
<span id="cb1-55"><a href="#cb1-55"></a>Some of the most commonly employed multivariate statistical techniques in quantitative ecology include ordination methods, such as principal component analysis (PCA) and correspondence analysis (CA), which are utilised to reduce complexity and permit the visualisation of patterns of species distribution in ecosystems. We will also learn how to incorporate multiple regression into these multivariate analyses through the use of redundancy analysis (RDA) and canonical correspondence analysis (CCA), and examine non-metric multidimensional scaling (NMDS). All these methods extend our ability to explore species-environment relationships, assess community composition (structure), and identify influential environmental variables driving species distributions. We will also explore the application of multivariate statistics in addressing critical ecological issues, such as biodiversity monitoring, ecosystem functioning, and the impacts of anthropogenic disturbances on ecosystems.</span>
<span id="cb1-56"><a href="#cb1-56"></a></span>
<span id="cb1-57"><a href="#cb1-57"></a></span>
<span id="cb1-58"><a href="#cb1-58"></a></span>
<span id="cb1-59"><a href="#cb1-59"></a><span class="fu"># Module Content and Framework</span></span>
<span id="cb1-60"><a href="#cb1-60"></a></span>
<span id="cb1-61"><a href="#cb1-61"></a>These links point to online resources such as datasets and R scripts in support of the video and PDF lecture material. It is essential that you work through these examples and workflows. Here is the Syllabus:</span>
<span id="cb1-62"><a href="#cb1-62"></a></span>
<span id="cb1-63"><a href="#cb1-63"></a>:::{.column-page-inset-right}</span>
<span id="cb1-64"><a href="#cb1-64"></a>| Week | Class date       | Lecture       | Topic                              | Class                                                     | Slides                                                                         | Reading                                                                                                                                                        | Tasks                                                                       | Tasks due  |</span>
<span id="cb1-65"><a href="#cb1-65"></a>|------|------------------|---------------|------------------------------------|-----------------------------------------------------------|--------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------|------------|</span>
<span id="cb1-66"><a href="#cb1-66"></a>| W1   | 11 June          | L0            | Overview &amp; Introduction            |                                                           | <span class="co">[</span><span class="ot">♦︎</span><span class="co">](../slides/BCB743_01_intro.pdf)</span>                                             | <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/BES-Reproducible-Code-2017.pdf)</span> <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/BES-Data-Guide-2017_web.pdf)</span>                                                                           |                                                                             |            |</span>
<span id="cb1-67"><a href="#cb1-67"></a>|      |                  | Lecture Set 1 | Ecological and Earth Data          |                                                           |                                                                                |                                                                                                                                                                | <span class="co">[</span><span class="ot">◒</span><span class="co">](assessments/Task_A1.qmd)</span> Task A1                                        | 1 July     |</span>
<span id="cb1-68"><a href="#cb1-68"></a>|      |                  |               | **RECAP OF BIODIVERSITY**          |                                                           |                                                                                |                                                                                                                                                                |                                                                             |            |</span>
<span id="cb1-69"><a href="#cb1-69"></a>|      |                  | L1: Revision  | Ecological Data                    | <span class="co">[</span><span class="ot">★</span><span class="co">](../BDC334/01-introduction.qmd)</span>                        |                                                                                |                                                                                                                                                                | Lab 1 Review                                                                |            | </span>
<span id="cb1-70"><a href="#cb1-70"></a>|      |                  |               | **DATA, MATRICES**                 |                                                           |                                                                                |                                                                                                                                                                |                                                                             |            |</span>
<span id="cb1-71"><a href="#cb1-71"></a>|      |                  | L2: Revision  | Environmental Distance             | <span class="co">[</span><span class="ot">★</span><span class="co">](../BDC334/02b-env_dist.qmd)</span>                           | <span class="co">[</span><span class="ot">♦︎</span><span class="co">](../slides/BCB743_04_environmental_distance.pdf)</span>                            |                                                                                                                                                                | Lab 2 Review                                                                |            | </span>
<span id="cb1-72"><a href="#cb1-72"></a>|      |                  | L3: Revision  | Quantifying Biodiversity           | <span class="co">[</span><span class="ot">★</span><span class="co">](../BDC334/03-biodiversity1.qmd)</span>                       | <span class="co">[</span><span class="ot">♦︎</span><span class="co">](../slides/BCB743_02_biodiversity_1.pdf)</span>                                    | <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/Smit_et_al_2013.pdf)</span> <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/Smit_et_al_2017.pdf)</span> <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/Smit_the_seaweed_data.pdf)</span>                                                       | Lab 3 Review                                                                |            |</span>
<span id="cb1-73"><a href="#cb1-73"></a>|      |                  | L4: Revision  | Describing Biodiversity Patterns   | <span class="co">[</span><span class="ot">★</span><span class="co">](../BDC334/04-biodiversity2.qmd)</span>                       | <span class="co">[</span><span class="ot">♦︎</span><span class="co">](../slides/BCB743_02_biodiversity_2.pdf)</span>                                    | <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/Shade_et_al_2018.pdf)</span>                                                                                                                              | Lab 4 Review                                                                |            |</span>
<span id="cb1-74"><a href="#cb1-74"></a>|      |                  | Lecture Set 2 | Ecological Theories                |                                                           |                                                                                |                                                                                                                                                                | <span class="co">[</span><span class="ot">◒</span><span class="co">](assessments/Task_A2.qmd)</span> Task A2                                        | 11 July    | </span>
<span id="cb1-75"><a href="#cb1-75"></a>|      | 13 June          | L5            | Correlations &amp; Associations        | <span class="co">[</span><span class="ot">★</span><span class="co">](correlations.qmd)</span>                                     | <span class="co">[</span><span class="ot">♦︎</span><span class="co">](../slides/BCB743_06_correlations.pdf)</span>                                      |                                                                                                                                                                | <span class="co">[</span><span class="ot">◒</span><span class="co">](assessments/Task_B.qmd)</span> Task B                                          | 17 June    |</span>
<span id="cb1-76"><a href="#cb1-76"></a>|      |                  | L6: Self      | Distance Metrics                   | <span class="co">[</span><span class="ot">★</span><span class="co">](dis-metrics.qmd)</span>                                      |                                                                                |                                                                                                                                                                |                                                                             |            |</span>
<span id="cb1-77"><a href="#cb1-77"></a>|      |                  |               | **UNCONSTRAINED ORDINATION**       |                                                           |                                                                                |                                                                                                                                                                |                                                                             |            | </span>
<span id="cb1-78"><a href="#cb1-78"></a>| W2   | 17 June          | L7            | Intro to Ordination                |                                                           | <span class="co">[</span><span class="ot">♦︎</span><span class="co">](../slides/BCB743_07_ordination.pdf)</span>                                        |                                                                                                                                                                |                                                                             |            |</span>
<span id="cb1-79"><a href="#cb1-79"></a>|      |                  | L8            | PCA                                | <span class="co">[</span><span class="ot">★</span><span class="co">](PCA.qmd)</span>                                              | <span class="co">[</span><span class="ot">♦︎</span><span class="co">](../slides/BCB743_08_PCA.pdf)</span>                                               |                                                                                                                                                                | <span class="co">[</span><span class="ot">◒</span><span class="co">](assessments/Task_C.qmd)</span> Task C                                          | 24 June    |</span>
<span id="cb1-80"><a href="#cb1-80"></a>|      |                  | L8: Self      | PCA: Additional Examples           | <span class="co">[</span><span class="ot">★</span><span class="co">](PCA_examples.qmd)</span>                                     |                                                                                |                                                                                                                                                                |                                                                             |            | </span>
<span id="cb1-81"><a href="#cb1-81"></a>|      |                  | L8: Self      | PCA: WHO SDG Example               | <span class="co">[</span><span class="ot">★</span><span class="co">](PCA_SDG_example.qmd)</span>                                  |                                                                                |                                                                                                                                                                | <span class="co">[</span><span class="ot">◒</span><span class="co">](assessments/Task_D.qmd)</span> Task D                                          | 11 July    | </span>
<span id="cb1-82"><a href="#cb1-82"></a>|      | 20 June          | L9a           | CA                                 | <span class="co">[</span><span class="ot">★</span><span class="co">](CA.qmd)</span>                                               | <span class="co">[</span><span class="ot">♦︎</span><span class="co">](../slides/BCB743_09_CA.pdf)</span>                                                |                                                                                                                                                                | <span class="co">[</span><span class="ot">◒</span><span class="co">](assessments/Task_E.qmd)</span> Task E                                          | 24 June    |</span>
<span id="cb1-83"><a href="#cb1-83"></a>|      |                  | L9b           | DCA                                | <span class="co">[</span><span class="ot">★</span><span class="co">](DCA.qmd)</span>                                              |                                                                                |                                                                                                                                                                |                                                                             |            |</span>
<span id="cb1-84"><a href="#cb1-84"></a>| W3   | 24 June          | L10           | PCoA                               | <span class="co">[</span><span class="ot">★</span><span class="co">](PCoA.qmd)</span>                                             | <span class="co">[</span><span class="ot">♦︎</span><span class="co">](../slides/BCB743_10_PCoA.pdf)</span>                                              |                                                                                                                                                                | <span class="co">[</span><span class="ot">◒</span><span class="co">](assessments/BCB743_intgrative_assignment.qmd)</span> Integrative Assignment    | 16 July    |</span>
<span id="cb1-85"><a href="#cb1-85"></a>|      |                  | L11           | nMDS                               | <span class="co">[</span><span class="ot">★</span><span class="co">](nMDS.qmd)</span>                                             | <span class="co">[</span><span class="ot">♦︎</span><span class="co">](../slides/BCB743_11_nMDS.pdf)</span>                                              |                                                                                                                                                                | <span class="co">[</span><span class="ot">◒</span><span class="co">](assessments/Task_F.qmd)</span> Task F                                          | 1 July     |</span>
<span id="cb1-86"><a href="#cb1-86"></a>|      |                  | L11: Self     | nMDS: PERMANOVA (Diatoms) Example  | <span class="co">[</span><span class="ot">★</span><span class="co">](nMDS_diatoms.qmd)</span>                                     |                                                                                | <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/Mayombo_et_al_2019.pdf)</span>                                                                                                                            |                                                                             |            | </span>
<span id="cb1-87"><a href="#cb1-87"></a>|      |                  | L12: Self     | Unconstrained Ordi. Summary        | <span class="co">[</span><span class="ot">★</span><span class="co">](unconstrained-summary.qmd)</span>                            |                                                                                |                                                                                                                                                                |                                                                             |            | </span>
<span id="cb1-88"><a href="#cb1-88"></a>| W4   | 1 July           | Lecture Set 1 | Ecological &amp; Earth Data            |                                                           |                                                                                |                                                                                                                                                                | Present Lecture Set 1                                                       | 1 July     | </span>
<span id="cb1-89"><a href="#cb1-89"></a>|      |                  |               | **REGRESSION ANALYSIS**            |                                                           |                                                                                |                                                                                                                                                                |                                                                             |            | </span>
<span id="cb1-90"><a href="#cb1-90"></a>|      | 4 July           | L13           | Model Building                     | <span class="co">[</span><span class="ot">★</span><span class="co">](model_building.qmd)</span>                                   |                                                                                |                                                                                                                                                                |                                                                             |            |</span>
<span id="cb1-91"><a href="#cb1-91"></a>|      |                  | L14           | Multiple Regression                | <span class="co">[</span><span class="ot">★</span><span class="co">](multiple_regression.qmd)</span>                              |                                                                                | <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/The-Biostatistics-Book-Ch-3-5.pdf)</span>                                                                                                                 | <span class="co">[</span><span class="ot">◒</span><span class="co">](assessments/Task_G.qmd)</span> Task G (Final Assessment)                       | **16 July**|</span>
<span id="cb1-92"><a href="#cb1-92"></a>|      |                  | L14: Self     | Gradients Example                  | <span class="co">[</span><span class="ot">★</span><span class="co">](deep_dive.qmd)</span>                                        |                                                                                | <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/Smit_et_al_2013.pdf)</span> <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/Smit_et_al_2017.pdf)</span> <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/Smit_the_seaweed_data.pdf)</span>                                                       |                                                                             |            |</span>
<span id="cb1-93"><a href="#cb1-93"></a>|      |                  | Lx            | Generalised Linear Models          | TBA (2025)                                                | ΤΒΑ (2025)                                                                     | TBA (2025)                                                                                                                                                     | ΤΒΑ (2025)                                                                  |            |</span>
<span id="cb1-94"><a href="#cb1-94"></a>| W5   | 8 July           | Lx            | Generalised Additive Models        | TBA (2025)                                                | ΤΒΑ (2025)                                                                     | TBA (2025)                                                                                                                                                     | ΤΒΑ (2025)                                                                  |            |</span>
<span id="cb1-95"><a href="#cb1-95"></a>|      |                  |               | **CONSTRAINED ORDINATION**         |                                                           |                                                                                |                                                                                                                                                                |                                                                             |            | </span>
<span id="cb1-96"><a href="#cb1-96"></a>|      |                  | L15           | Distance-Based Redundancy Analysis | <span class="co">[</span><span class="ot">★</span><span class="co">](constrained_ordination.qmd)</span>                           | <span class="co">[</span><span class="ot">♦︎</span><span class="co">](../slides/BCB743_12_constrained_ordination.pdf)</span>                            | <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/Smit_et_al_2017.pdf)</span> <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/Smit_the_seaweed_data.pdf)</span>                                                                                        |                                                                             |            |</span>
<span id="cb1-97"><a href="#cb1-97"></a>|      |                  | L15: Self     | db-RDA: Seaweeds Example           | <span class="co">[</span><span class="ot">★</span><span class="co">](Seaweed_in_Two_Oceans_v2/two_oceans_appendices.qmd)</span>   |                                                                                | <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/Smit_et_al_2017.pdf)</span> <span class="co">[</span><span class="ot">▤</span><span class="co">](../docs/Smit_the_seaweed_data.pdf)</span>                                                                                        |                                                                             |            |</span>
<span id="cb1-98"><a href="#cb1-98"></a>|      |                  |               | **CLUSTER ANALYSIS**               |                                                           |                                                                                |                                                                                                                                                                |                                                                             |            | </span>
<span id="cb1-99"><a href="#cb1-99"></a>|      |                  | L16           | Cluster Analysis                   | <span class="co">[</span><span class="ot">★</span><span class="co">](cluster_analysis.qmd)</span>                                 |                                                                                |                                                                                                                                                                | <span class="co">[</span><span class="ot">◒</span><span class="co">](assessments/Task_D.qmd)</span> Task D (continue)                               | 11 July    |</span>
<span id="cb1-100"><a href="#cb1-100"></a>|      | 11 July          | Lecture Set 2 | Ecological Theories                |                                                           |                                                                                |                                                                                                                                                                | Present Lecture Set 2                                                       | 11 July    |</span>
<span id="cb1-101"><a href="#cb1-101"></a>|      |                  | L17           | Review                             |                                                           |                                                                                |                                                                                                                                                                |                                                                             |            |</span>
<span id="cb1-102"><a href="#cb1-102"></a>:::</span>
<span id="cb1-103"><a href="#cb1-103"></a></span>
<span id="cb1-104"><a href="#cb1-104"></a>**Core theoretical framework** Ecological hypotheses underlying the processes of species assembly in space and time, including neutral and niche-based mechanisms, and historical events; overview of the currently known and understood distributional patterns of major groups of organisms at global, regional and local scales; consideration of sampling designs aimed at capturing these patterns and drivers so as to arrive at a processed based understanding of species assembly. </span>
<span id="cb1-105"><a href="#cb1-105"></a></span>
<span id="cb1-106"><a href="#cb1-106"></a>**Competence** Data collection aimed at a quantitative test of the relevant hypotheses, above. The management and analysis of ecological data; reproducible and collaborative research; the use of R as a tool for the analysis multivariate ecological data; multivariate techniques such as nMDS, PCA, RDA and cluster analysis; graphical data summaries and visualisations.</span>
<span id="cb1-107"><a href="#cb1-107"></a></span>
<span id="cb1-108"><a href="#cb1-108"></a><span class="fu"># Outcomes of BCB743</span></span>
<span id="cb1-109"><a href="#cb1-109"></a></span>
<span id="cb1-110"><a href="#cb1-110"></a>By the end of this module, students will be able to:</span>
<span id="cb1-111"><a href="#cb1-111"></a></span>
<span id="cb1-112"><a href="#cb1-112"></a><span class="ss">- </span>Understand the concepts of $\alpha$-, $\beta$- and $\gamma$-diversity</span>
<span id="cb1-113"><a href="#cb1-113"></a><span class="ss">- </span>Know and understand the current hypotheses that explain species assembly processes in space and time (e.g. neutral and niche mechanisms)</span>
<span id="cb1-114"><a href="#cb1-114"></a><span class="ss">- </span>Collect ecological data at the appropriate scale, which would lend themselves to a quantitative analysis of points 1 and 2, above</span>
<span id="cb1-115"><a href="#cb1-115"></a><span class="ss">- </span>Use the R software and associated packages to undertake the analyses required in point 3, above</span>
<span id="cb1-116"><a href="#cb1-116"></a><span class="ss">- </span>Interpret the outcomes of the above analyses and use it to quantitatively characterise points 1 and 2, above</span>
<span id="cb1-117"><a href="#cb1-117"></a><span class="ss">- </span>Communicate the findings by written and oral means</span>
<span id="cb1-118"><a href="#cb1-118"></a></span>
<span id="cb1-119"><a href="#cb1-119"></a><span class="fu"># Graduate Attributes</span></span>
<span id="cb1-120"><a href="#cb1-120"></a></span>
<span id="cb1-121"><a href="#cb1-121"></a>The <span class="co">[</span><span class="ot">**graduate attributes**</span><span class="co">](../pages/graduate_attributes.qmd)</span> resulting from completion of this modules alignment with the expectations of the workspace across diverse organisations and institutions where graduates typically find employment.</span>
<span id="cb1-122"><a href="#cb1-122"></a></span>
<span id="cb1-123"><a href="#cb1-123"></a><span class="fu"># Data and Reading in Support of the Syllabus</span></span>
<span id="cb1-124"><a href="#cb1-124"></a>In the table above there are links to several key papers to read in preparation of each week's theory. It is essential that you read these papers.</span>
<span id="cb1-125"><a href="#cb1-125"></a></span>
<span id="cb1-126"><a href="#cb1-126"></a>Many other references are cited in each Chapter. These serve several functions in that they:</span>
<span id="cb1-127"><a href="#cb1-127"></a></span>
<span id="cb1-128"><a href="#cb1-128"></a><span class="ss">- </span>Add additional theory relevant to some ecological concepts</span>
<span id="cb1-129"><a href="#cb1-129"></a><span class="ss">- </span>Provide background to some of the datasets used in my examples</span>
<span id="cb1-130"><a href="#cb1-130"></a><span class="ss">- </span>Discuss derivations of some equations used to calculate diversity concepts</span>
<span id="cb1-131"><a href="#cb1-131"></a><span class="ss">- </span>Provide example walkthroughs of some of the computational aspects of the methods covered in the Labs</span>
<span id="cb1-132"><a href="#cb1-132"></a><span class="ss">- </span>Collectively supplement the discussion about these concepts covered in the lectures</span>
<span id="cb1-133"><a href="#cb1-133"></a></span>
<span id="cb1-134"><a href="#cb1-134"></a>Actively engaging with these reading materials will make to difference between a 60% average mark for the module, and a mark in excess of 80%.</span>
<span id="cb1-135"><a href="#cb1-135"></a></span>
<span id="cb1-136"><a href="#cb1-136"></a><span class="fu">## Reading</span></span>
<span id="cb1-137"><a href="#cb1-137"></a></span>
<span id="cb1-138"><a href="#cb1-138"></a>You are **expected to read additional material** in support of the</span>
<span id="cb1-139"><a href="#cb1-139"></a>content covered in class and on this website.</span>
<span id="cb1-140"><a href="#cb1-140"></a></span>
<span id="cb1-141"><a href="#cb1-141"></a>A **compulsory** reference is ['Numerical Ecology with</span>
<span id="cb1-142"><a href="#cb1-142"></a>R'](http://adn.biol.umontreal.ca/~numericalecology/numecolR/) by Daniel</span>
<span id="cb1-143"><a href="#cb1-143"></a>Borcard, François Gillet and Pierre Legendre <span class="co">[</span><span class="ot">@borcard2011numerical</span><span class="co">]</span>. Much of the class' content</span>
<span id="cb1-144"><a href="#cb1-144"></a>and many of the examples (and code) that I use have been adapted from</span>
<span id="cb1-145"><a href="#cb1-145"></a>this source. There is also the excellent book by @legendre2012numerical called 'Numerical Ecology' which provides everything the former book has, but in greater detail and with less focus on R. Both should be considered a 'gold standard' reference for Quantitative Ecology.</span>
<span id="cb1-146"><a href="#cb1-146"></a></span>
<span id="cb1-147"><a href="#cb1-147"></a>A third **highly recommended text** is the book [Tree Diversity</span>
<span id="cb1-148"><a href="#cb1-148"></a>Analysis](http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf)</span>
<span id="cb1-149"><a href="#cb1-149"></a>by Roeland Kindt and Richard Coe.</span>
<span id="cb1-150"><a href="#cb1-150"></a></span>
<span id="cb1-151"><a href="#cb1-151"></a>I can also recommend a these amazing websites with excellent content:</span>
<span id="cb1-152"><a href="#cb1-152"></a></span>
<span id="cb1-153"><a href="#cb1-153"></a><span class="ss">-   </span>David Zelený's <span class="co">[</span><span class="ot">Analysis of Community Ecology Data in R</span><span class="co">](https://www.davidzeleny.net/anadat-r/doku.php)</span></span>
<span id="cb1-154"><a href="#cb1-154"></a><span class="ss">-   </span>Mike Palmer's <span class="co">[</span><span class="ot">Ordination Methods for Ecologists</span><span class="co">](http://ordination.okstate.edu/)</span></span>
<span id="cb1-155"><a href="#cb1-155"></a><span class="ss">-   </span><span class="co">[</span><span class="ot">GUide to STatistical Analysis in Microbial Ecology (GUSTA ME)</span><span class="co">](https://sites.google.com/site/mb3gustame/)</span></span>
<span id="cb1-156"><a href="#cb1-156"></a></span>
<span id="cb1-157"><a href="#cb1-157"></a>Note that the URLs with links to additional reading that appear with the worked-through example code should **not be seen as optional**. They are there for a reason and should be consulted even though I might not necessarily refer to each of them in class. </span>
<span id="cb1-158"><a href="#cb1-158"></a>Use these materials liberally.</span>
<span id="cb1-159"><a href="#cb1-159"></a></span>
<span id="cb1-160"><a href="#cb1-160"></a>Should you want to download the source code for the BCB743 (and BCB744 website), you may find it on <span class="co">[</span><span class="ot">&lt;i class="fab fa-github"&gt;&lt;/i&gt; GitHub</span><span class="co">](https://github.com/ajsmit/R_courses)</span>.</span>
<span id="cb1-161"><a href="#cb1-161"></a></span>
<span id="cb1-162"><a href="#cb1-162"></a><span class="fu">## Datasets Used in This Module</span></span>
<span id="cb1-163"><a href="#cb1-163"></a></span>
<span id="cb1-164"><a href="#cb1-164"></a>Note that the links provided might not necessarily lead to the **vegan** help page.</span>
<span id="cb1-165"><a href="#cb1-165"></a></span>
<span id="cb1-166"><a href="#cb1-166"></a>|   | Dataset                                                      | Source                                                                                        |</span>
<span id="cb1-167"><a href="#cb1-167"></a>|:--|--------------------------------------------------------------|-----------------------------------------------------------------------------------------------|</span>
<span id="cb1-168"><a href="#cb1-168"></a>| 1 | Vegetation and Environment in Dutch Dune Meadows             | <span class="co">[</span><span class="ot">**vegan**</span><span class="co">](https://www.davidzeleny.net/anadat-r/doku.php/en:data:dune)</span>                       |</span>
<span id="cb1-169"><a href="#cb1-169"></a>| 2 | Oribatid Mite Data with Explanatory Variables                | <span class="co">[</span><span class="ot">**vegan**</span><span class="co">](http://adn.biol.umontreal.ca/~numericalecology/data/oribates.html)</span>                |</span>
<span id="cb1-170"><a href="#cb1-170"></a>| 3 | The Doubs River Data                                         | <span class="co">[</span><span class="ot">**Numerical Ecology with R**</span><span class="co">](https://www.davidzeleny.net/anadat-r/doku.php/en:data:doubs)</span>   |</span>
<span id="cb1-171"><a href="#cb1-171"></a>| 4 | The Barro Colorado Island Tree Counts                        | <span class="co">[</span><span class="ot">**vegan**</span><span class="co">](https://www.davidzeleny.net/anadat-r/doku.php/en:data:bci)</span>                        |</span>
<span id="cb1-172"><a href="#cb1-172"></a>| 5 | John Bolton, Rob Anderson, and Herre Stegenga's Seaweed Data | <span class="co">[</span><span class="ot">**Smit et al., 2017**</span><span class="co">](https://www.frontiersin.org/articles/10.3389/fmars.2017.00404/)</span>       |</span>
<span id="cb1-173"><a href="#cb1-173"></a>| 6 | Serge Mayombo's Diatoms Data                                 | <span class="co">[</span><span class="ot">**Mayombo et al., 2019**</span><span class="co">](https://www.tandfonline.com/doi/abs/10.2989/1814232X.2019.1592778)</span> |</span>
<span id="cb1-174"><a href="#cb1-174"></a>| 7 | World Health Organization Sustainable Development Goals Data | <span class="co">[</span><span class="ot">**WHO**</span><span class="co">](https://www.who.int/data)</span>                                                           |</span>
<span id="cb1-175"><a href="#cb1-175"></a></span>
<span id="cb1-176"><a href="#cb1-176"></a><span class="fu"># Prerequisites</span></span>
<span id="cb1-177"><a href="#cb1-177"></a></span>
<span id="cb1-178"><a href="#cb1-178"></a>You should have a moderate numerical literacy and have prior programming experience. Such experience will have been obtained in the <span class="co">[</span><span class="ot">BCB744</span><span class="co">](/workshops/)</span> module, which is a module about doing statistics in R. If you have a reasonable experience in coding and statistical analysis you should find yourself well prepared. You should also thoroughly revise  <span class="co">[</span><span class="ot">BDC334</span><span class="co">](../BDC334/BDC334_index.qmd)</span> by the end of the first week of this module.</span>
<span id="cb1-179"><a href="#cb1-179"></a></span>
<span id="cb1-180"><a href="#cb1-180"></a><span class="fu"># Method of Instruction</span></span>
<span id="cb1-181"><a href="#cb1-181"></a></span>
<span id="cb1-182"><a href="#cb1-182"></a>You are provided with reading material (lecture slides, code, website content) that you are expected to consume **prior to the class**. Classes will involve brief introductions to new concepts, and will be followed by working on exercises in class that cover those concepts. The workshop is designed to be as interactive as possible, so while you are working on exercises the tutor and I will circulate among you and engage with you to help you understand any material and the associated code you are uncomfortable with. Often this will result in discussions of novel applications and alternative approaches to the data analysis challenges you are required to solve. More challenging concepts might emerge during the assignments (typically these will be submitted the following day), and any such challenges will be dealt with in class prior to learning new concepts.</span>
<span id="cb1-183"><a href="#cb1-183"></a></span>
<span id="cb1-184"><a href="#cb1-184"></a>Although the module is theory-heavy, a large part of it is also about coding. It is up to you to take your coding skills to the next level and move beyond what I teach in class. Coding is a bit like learning a language, and as such programming is a skill that is best learned by doing.</span>
<span id="cb1-185"><a href="#cb1-185"></a></span>
<span id="cb1-186"><a href="#cb1-186"></a><span class="fu"># Learning Colaboratively</span></span>
<span id="cb1-187"><a href="#cb1-187"></a></span>
<span id="cb1-188"><a href="#cb1-188"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb1-189"><a href="#cb1-189"></a><span class="fu">## Also read: How to learn</span></span>
<span id="cb1-190"><a href="#cb1-190"></a>Please refer to my <span class="co">[</span><span class="ot">advice about how to learn</span><span class="co">](../pages/How_to_learn.qmd)</span>.</span>
<span id="cb1-191"><a href="#cb1-191"></a>:::</span>
<span id="cb1-192"><a href="#cb1-192"></a></span>
<span id="cb1-193"><a href="#cb1-193"></a>Discuss the BCB743 workshop activities with your peers as you work on them. Use also the WhatsApp group set up for the module for discussion purposes (I might assist via this medium if neccesary if your questions/comments have relevance to the whole class). A better option is to use <span class="co">[</span><span class="ot">GitHub Issues</span><span class="co">](/quantecol/#help-via-bcb744-and-bcb743-issues-on-github)</span>. You will learn more in this module if you work with your friends than if you do not. Ask questions, answer questions, and share ideas liberally. Please identify your work partners by name on all assignments (if you decide to work in pairs).</span>
<span id="cb1-194"><a href="#cb1-194"></a></span>
<span id="cb1-195"><a href="#cb1-195"></a>**Cooperative learning is not a licence for plagiarism. Plagiarism is a serious offence and will be dealt with concisely. Consequences of cheating are severe---they range from a 0% for the assignment or exam up to dismissal from the course for a second offense.** </span>
<span id="cb1-196"><a href="#cb1-196"></a></span>
<span id="cb1-197"><a href="#cb1-197"></a><span class="fu"># Reusing Code Found Elsewhere</span></span>
<span id="cb1-198"><a href="#cb1-198"></a></span>
<span id="cb1-199"><a href="#cb1-199"></a>A huge volume of code is available on the web and it can be adapted to solve your own problems. You may make use of any online resources (e.g. form <span class="co">[</span><span class="ot">StackOverflow</span><span class="co">](https://stackoverflow.com/)</span>, a thoroughly-used source of discussion about <span class="co">[</span><span class="ot">R code</span><span class="co">](https://stackoverflow.com/questions/tagged/r)</span>)---but you **MUST** clearly indicate (cite) that your solution relies on found code, regardless to what extent you have modified it to your own needs. Reused code that is discovered via a web search and which is not explicitly cited is plagiarism and it will be treated as such. On assignments you may not directly share code with your peers in this workshop.</span>
<span id="cb1-200"><a href="#cb1-200"></a></span>
<span id="cb1-201"><a href="#cb1-201"></a><span class="fu"># Software</span></span>
<span id="cb1-202"><a href="#cb1-202"></a></span>
<span id="cb1-203"><a href="#cb1-203"></a>In this course we will rely entirely on <span class="co">[</span><span class="ot">R</span><span class="co">](https://cran.r-project.org/)</span> running within the <span class="co">[</span><span class="ot">RStudio</span><span class="co">](https://www.rstudio.com/)</span> IDE. The use of R was covered extensively in the <span class="co">[</span><span class="ot">BCB744</span><span class="co">](http://localhost:4321/workshops/)</span> module where the <span class="co">[</span><span class="ot">installation process</span><span class="co">](http://localhost:4321/workshops/intro_r/chapters/02-rstudio/)</span> was discussed. We will primarily use the <span class="co">[</span><span class="ot">**vegan**</span><span class="co">](https://cran.r-project.org/web/packages/vegan/index.html)</span> package, but some useful functions are also provided by the package <span class="co">[</span><span class="ot">**BiodiversityR**</span><span class="co">](https://github.com/cran/BiodiversityR)</span> (and <span class="co">[</span><span class="ot">here</span><span class="co">](http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf)</span> and <span class="co">[</span><span class="ot">here</span><span class="co">](https://rpubs.com/Roeland-KINDT)</span>). Various other R packages offer overlapping and additional methods, but **vegan** should accommodate <span class="sc">\&gt;</span>90% of your Quantitative Ecology needs.</span>
<span id="cb1-204"><a href="#cb1-204"></a></span>
<span id="cb1-205"><a href="#cb1-205"></a><span class="fu"># Computers</span></span>
<span id="cb1-206"><a href="#cb1-206"></a></span>
<span id="cb1-207"><a href="#cb1-207"></a>You are encouraged to provide your own laptops and to install the necessary software before the module starts. Limited support can be provided if required. There are also computers with R and RStudio (and the neccesary add-on libraries) available in the 5th floor lab in the BCB Department.</span>
<span id="cb1-208"><a href="#cb1-208"></a></span>
<span id="cb1-209"><a href="#cb1-209"></a><span class="fu"># Attendance</span></span>
<span id="cb1-210"><a href="#cb1-210"></a></span>
<span id="cb1-211"><a href="#cb1-211"></a>This worskhop-based, hands on course can only deliver acceptible outcomes if you attend all classes. The schedule is set and cannot be changed. Sometimes an occasional absence cannot be avoided. Please be curtious and notify myself or the tutor in advance of any absence. If you work with a partner in class, notify them too. Keep up with the reading assignments while you are away and we will all work with you to get you back up to speed on what you miss. If you do miss a class, however, the assignments must still be submitted on time (also see <span class="co">[</span><span class="ot">**Late submission of CA**</span><span class="co">](/quantecol/#late-submission-of-ca)</span>).</span>
<span id="cb1-212"><a href="#cb1-212"></a></span>
<span id="cb1-213"><a href="#cb1-213"></a>Since you may decide to work in collaboration with a peer on tasks and assignments, please keep this person informed at all times in case some emergency makes you unavailable for a period of time. Someone might depend on your input and contributions---do not leave someone in the lurch so that they cannot complete a task in your absence. </span>
<span id="cb1-214"><a href="#cb1-214"></a></span>
<span id="cb1-215"><a href="#cb1-215"></a>Sure, here is the improved and completed sentence and meaning at the end:</span>
<span id="cb1-216"><a href="#cb1-216"></a></span>
<span id="cb1-217"><a href="#cb1-217"></a>---</span>
<span id="cb1-218"><a href="#cb1-218"></a></span>
<span id="cb1-219"><a href="#cb1-219"></a><span class="fu"># Assessment Policy {#sec-policy}</span></span>
<span id="cb1-220"><a href="#cb1-220"></a></span>
<span id="cb1-221"><a href="#cb1-221"></a>**Continuous Assessments** (CA) and a **Final Assessment** will provide a **Final Mark** for the module. They contribute 0.6 and 0.4, respectively. These modes of assessment meet our needs as far as <span class="co">[</span><span class="ot">formative and summative assessments</span><span class="co">](../pages/assessment_theory.qmd)</span> are concerned. All assessments are open book, so consult your code and reading material if and when you need to.</span>
<span id="cb1-222"><a href="#cb1-222"></a></span>
<span id="cb1-223"><a href="#cb1-223"></a><span class="fu">## Continuous Assessment</span></span>
<span id="cb1-224"><a href="#cb1-224"></a></span>
<span id="cb1-225"><a href="#cb1-225"></a>The Continuous Assessment is comprised of:</span>
<span id="cb1-226"><a href="#cb1-226"></a></span>
<span id="cb1-227"><a href="#cb1-227"></a><span class="ss">- </span>Tasks A1 and A2, weighted 0.35 and 0.15, respectively, towards the CA</span>
<span id="cb1-228"><a href="#cb1-228"></a><span class="ss">- </span>Lecture presentation, weighted 0.3 towards the CA</span>
<span id="cb1-229"><a href="#cb1-229"></a><span class="ss">- </span>Tasks B to F, the average of which contributes 0.2 towards the CA</span>
<span id="cb1-230"><a href="#cb1-230"></a></span>
<span id="cb1-231"><a href="#cb1-231"></a>For Tasks A1 and A2, please refer to the marking schedule as agreed on by the class.</span>
<span id="cb1-232"><a href="#cb1-232"></a></span>
<span id="cb1-233"><a href="#cb1-233"></a>When assessing Tasks B to F, we will pay attention to the following criteria:</span>
<span id="cb1-234"><a href="#cb1-234"></a></span>
<span id="cb1-235"><a href="#cb1-235"></a><span class="ss">- </span>Presentation and formatting (10%), including:</span>
<span id="cb1-236"><a href="#cb1-236"></a><span class="ss">    - </span>Questions answered in order</span>
<span id="cb1-237"><a href="#cb1-237"></a><span class="ss">    - </span>Sectioning</span>
<span id="cb1-238"><a href="#cb1-238"></a><span class="ss">    - </span>General appearance</span>
<span id="cb1-239"><a href="#cb1-239"></a><span class="ss">    - </span>References (if required)</span>
<span id="cb1-240"><a href="#cb1-240"></a><span class="ss">- </span>Code formatting (10%), e.g.:</span>
<span id="cb1-241"><a href="#cb1-241"></a><span class="ss">    - </span>Application of <span class="co">[</span><span class="ot">R code conventions</span><span class="co">](http://adv-r.had.co.nz/Style.html)</span>, e.g. spaces around <span class="in">`&lt;-`</span>, after <span class="in">`#`</span>, after <span class="in">`,`</span>, etc.</span>
<span id="cb1-242"><a href="#cb1-242"></a><span class="ss">    - </span>New line for each <span class="in">`dplyr`</span> function (lines end in <span class="in">`%&gt;%`</span>)</span>
<span id="cb1-243"><a href="#cb1-243"></a><span class="ss">    - </span>New line for each <span class="in">`ggplot`</span> layer (lines end in <span class="in">`+`</span>)</span>
<span id="cb1-244"><a href="#cb1-244"></a><span class="ss">    - </span>Tidiness of code presentation in the HTML</span>
<span id="cb1-245"><a href="#cb1-245"></a><span class="ss">- </span>Code correctness in the context of the specified analysis (25%):</span>
<span id="cb1-246"><a href="#cb1-246"></a><span class="ss">    - </span>The code must faithfully execute the intended analysis as required by the research questions and/or hypotheses tests</span>
<span id="cb1-247"><a href="#cb1-247"></a><span class="ss">- </span>Figures (10%):</span>
<span id="cb1-248"><a href="#cb1-248"></a><span class="ss">    - </span>Sensible use of themes and colors</span>
<span id="cb1-249"><a href="#cb1-249"></a><span class="ss">    - </span>Publication quality (complete axis titles and labels, etc.)</span>
<span id="cb1-250"><a href="#cb1-250"></a><span class="ss">    - </span>Informative and complete titles, axis labels, legends, etc.</span>
<span id="cb1-251"><a href="#cb1-251"></a><span class="ss">- </span>Discussion (45%):</span>
<span id="cb1-252"><a href="#cb1-252"></a><span class="ss">    - </span>Here you will be assessed for integrating the results of your analyses within the correct theoretical framework</span>
<span id="cb1-253"><a href="#cb1-253"></a></span>
<span id="cb1-254"><a href="#cb1-254"></a><span class="fu">## Task G: Final Assessment (Exam)</span></span>
<span id="cb1-255"><a href="#cb1-255"></a></span>
<span id="cb1-256"><a href="#cb1-256"></a>The Final Assessment starts after the Multiple Regression lecture on 4 July and you can do in the comfort of your home. It will involve the analysis of real world data and assess some more theoretical and philosophical aspects of Quantitative Ecology. A full mark breakdown is provided with <span class="co">[</span><span class="ot">Task G</span><span class="co">](assessments/Task_G.qmd)</span>.</span>
<span id="cb1-257"><a href="#cb1-257"></a></span>
<span id="cb1-258"><a href="#cb1-258"></a><span class="fu">## Submission of Assignments and Exams</span></span>
<span id="cb1-259"><a href="#cb1-259"></a></span>
<span id="cb1-260"><a href="#cb1-260"></a>A statement such as the one below accompanies every assignment---pay attention, as failing to observe this instruction may result in a loss of marks (i.e. if an assignment remains ungraded because the owner of the material cannot be identified): </span>
<span id="cb1-261"><a href="#cb1-261"></a></span>
<span id="cb1-262"><a href="#cb1-262"></a>Submit a R script wherein you provide answers to Questions 1--9 by no later than 8:00 tomorrow. Label the script as follows (e.g.): **BCB743_AJ_Smit_Assignment_2.R**. </span>
<span id="cb1-263"><a href="#cb1-263"></a></span>
<span id="cb1-264"><a href="#cb1-264"></a><span class="fu">## Late Submission of CA</span></span>
<span id="cb1-265"><a href="#cb1-265"></a></span>
<span id="cb1-266"><a href="#cb1-266"></a>Late assignments will be penalised 10% per day and will not be accepted more than 48 hours late, unless evidence such as a doctor's note, a death certificate, or another documented emergency can be provided. If you know in advance that a submission will be late, please discuss this and seek prior approval. This policy is based on the idea that in order to learn how to translate your human thoughts into computer language (coding) you should be working with them at multiple times each week---ideally daily. Time has been allocated in class for working on assignments and students are expected to continue to work on the assignments outside of class. Successfully completing (and passing) this module requires that you finish assignments based on what we have covered in class by the following class period. Work diligently from the onset so that even if something unexpected happens at the last minute you should already be close to done. This approach also allows rapid feedback to be provided to you, which can only be accomplished by returning assignments quickly and punctually.</span>
<span id="cb1-267"><a href="#cb1-267"></a></span>
<span id="cb1-268"><a href="#cb1-268"></a><span class="fu"># Support</span></span>
<span id="cb1-269"><a href="#cb1-269"></a></span>
<span id="cb1-270"><a href="#cb1-270"></a>It's expected that some tricky aspects of the module will take time to master, and the best way to master problematic material is to practice, practice some more, and then to ask questions. Trying for 10 minutes and then giving up is not good enough. I'll be more sympathetic to your cause if you can demonstrate having tried for a full day before giving up and asking me. When you ask questions about some challenge, this is the way to do it---explain to me your numerous attempts at trying to solve the problem, and explain how these various attempts have failed. *I will not help you if you have not tried to help yourself first* (maybe with advice from friends). There will be time in class to do this, typically before we embark on a new topic. You are also encouraged to bring up related questions that arise in your own B.Sc. (Hons.) research project.</span>
<span id="cb1-271"><a href="#cb1-271"></a></span>
<span id="cb1-272"><a href="#cb1-272"></a>Should you require more time with me, find out when I am 'free' and set an appointment by sending me a calendar invitation. I am happy to have a personal meeting with you via Zoom, but I prefer face-to-face in my office.</span>
<span id="cb1-273"><a href="#cb1-273"></a></span>
<span id="cb1-274"><a href="#cb1-274"></a><span class="fu">## Help Via BCB744 and BCB743 Issues on GitHub</span></span>
<span id="cb1-275"><a href="#cb1-275"></a></span>
<span id="cb1-276"><a href="#cb1-276"></a>All discussion for the BCB744 and BCB743 workshops will be held in the <span class="co">[</span><span class="ot">Issues</span><span class="co">](https://github.com/ajsmit/R_courses/issues)</span> of <span class="co">[</span><span class="ot">this repository</span><span class="co">](https://github.com/ajsmit/R_courses)</span>. Please post all content-related questions there, and use email only for personal matters. Note that this is a public repository, so be professional in your writing here (grammar, etc.).</span>
<span id="cb1-277"><a href="#cb1-277"></a></span>
<span id="cb1-278"><a href="#cb1-278"></a>To start a new thread, create a **New issue**. Tag your peers using their handle---<span class="in">`@ajsmit`</span>, for example---to get their attention.</span>
<span id="cb1-279"><a href="#cb1-279"></a></span>
<span id="cb1-280"><a href="#cb1-280"></a>Once a question has been answered, the issue will be closed, so lots of good answers might end up in closed issues. Don't forget to look there when looking for answers---you can use the **Search** feature on this repository to find answers that might have been offered by the same or similar problem experienced by someone else in the past.</span>
<span id="cb1-281"><a href="#cb1-281"></a></span>
<span id="cb1-282"><a href="#cb1-282"></a>**Guidelines for Posting Questions:**</span>
<span id="cb1-283"><a href="#cb1-283"></a></span>
<span id="cb1-284"><a href="#cb1-284"></a><span class="ss">-   </span>First search existing issues (open or closed) for answers. If the question has already been answered, you're done! If there is an open issue, feel free to contribute to it. Or feel free to open a closed issue if you believe the answer is not satisfactory.</span>
<span id="cb1-285"><a href="#cb1-285"></a><span class="ss">-   </span>Give your issue an informative title.</span>
<span id="cb1-286"><a href="#cb1-286"></a><span class="ss">    -   </span>Good: "Error: could not find function"ggplot""</span>
<span id="cb1-287"><a href="#cb1-287"></a><span class="ss">    -   </span>Bad: "My code does not work!" Note that you can edit an issue's</span>
<span id="cb1-288"><a href="#cb1-288"></a>        title after it's been posted.</span>
<span id="cb1-289"><a href="#cb1-289"></a><span class="ss">-   </span>Format your questions nicely using markdown and code formatting. Preview your issue prior to posting.</span>
<span id="cb1-290"><a href="#cb1-290"></a><span class="ss">-   </span>As I explained above, your peers and I will more sympathetic to your cause if you can show *all the things you have tried as you,</span>
<span id="cb1-291"><a href="#cb1-291"></a>    yourself, tried to fix the issue first*.</span>
<span id="cb1-292"><a href="#cb1-292"></a><span class="ss">-   </span>Include code and example data so the person trying to help you have something to work with (and which results in the error, perhaps)</span>
<span id="cb1-293"><a href="#cb1-293"></a><span class="ss">-   </span>Where appropriate, provide links to specific files, or even lines within them, in the body of your issue. This will help your peers</span>
<span id="cb1-294"><a href="#cb1-294"></a>    understand your question. Note that only the teaching team will have access to private repos.</span>
<span id="cb1-295"><a href="#cb1-295"></a><span class="ss">-   </span>(Optional) Tag someone or some group of people. Start by typing their GitHub username prefixed with the \@ symbol. Of course this</span>
<span id="cb1-296"><a href="#cb1-296"></a>    supposes that each of you have a GitHub account and username.</span>
<span id="cb1-297"><a href="#cb1-297"></a><span class="ss">-   </span>Hit **Submit new issue** when you're ready to post.</span>
<span id="cb1-298"><a href="#cb1-298"></a></span>
<span id="cb1-299"><a href="#cb1-299"></a></span>
<span id="cb1-300"><a href="#cb1-300"></a><span class="in">```{=html}</span></span>
<span id="cb1-301"><a href="#cb1-301"></a><span class="in">&lt;!-- ## Function list</span></span>
<span id="cb1-302"><a href="#cb1-302"></a></span>
<span id="cb1-303"><a href="#cb1-303"></a><span class="in">  You will encounter the following functions for the first time in this module:</span></span>
<span id="cb1-304"><a href="#cb1-304"></a></span>
<span id="cb1-305"><a href="#cb1-305"></a><span class="in">  | Function              | Description                                        |</span></span>
<span id="cb1-306"><a href="#cb1-306"></a><span class="in">  |:----------------------|----------------------------------------------------|</span></span>
<span id="cb1-307"><a href="#cb1-307"></a><span class="in">  | `adonis2()`           | Permutational Multivariate Analysis of Variance Using Distance Matrices |</span></span>
<span id="cb1-308"><a href="#cb1-308"></a><span class="in">  | `anova.betadisper()`  | PERMDIST 2 for Multivariate homogeneity of groups dispersions (variances) |</span></span>
<span id="cb1-309"><a href="#cb1-309"></a><span class="in">  | `anova.cca ()`        | Permutation Test for CCA, RDA and CA of Principal Coordinates |</span></span>
<span id="cb1-310"><a href="#cb1-310"></a><span class="in">  | `anova.manyglm()`     | Analysis of Deviance for Multivariate Generalized Linear Model Fits for Abundance Data |</span></span>
<span id="cb1-311"><a href="#cb1-311"></a><span class="in">  | `beta.pair()`         | Incidence-based Pair-wise Dissimilarities |</span></span>
<span id="cb1-312"><a href="#cb1-312"></a><span class="in">  | `betadisper()`        | Multivariate homogeneity of groups dispersions (variances) |</span></span>
<span id="cb1-313"><a href="#cb1-313"></a><span class="in">  | `betapart.core()`     | Core Calculations of betapart |</span></span>
<span id="cb1-314"><a href="#cb1-314"></a><span class="in">  | `biplot()`            | Biplot of Multivariate Data |</span></span>
<span id="cb1-315"><a href="#cb1-315"></a><span class="in">  | `capscale()`          | [Partial] Distance-based Redundancy Analysis |</span></span>
<span id="cb1-316"><a href="#cb1-316"></a><span class="in">  | `cca()`               | [Partial] [Constrained] Correspondence Analysis and Redundancy Analysis |</span></span>
<span id="cb1-317"><a href="#cb1-317"></a><span class="in">  | `cleanplot.pc()`      | Numerical Ecology in R's PCA Plot |</span></span>
<span id="cb1-318"><a href="#cb1-318"></a><span class="in">  | `daisy()`             | Dissimilarity Matrix Calculation |</span></span>
<span id="cb1-319"><a href="#cb1-319"></a><span class="in">  | `decostand()`         | Standardization Methods for Community Ecology |</span></span>
<span id="cb1-320"><a href="#cb1-320"></a><span class="in">  | `diag()`              | Matrix Diagonals |</span></span>
<span id="cb1-321"><a href="#cb1-321"></a><span class="in">  | `diversity()`         | Ecological Diversity Indices |</span></span>
<span id="cb1-322"><a href="#cb1-322"></a><span class="in">  | `diversityresult()`   | Alternative Diversity Results |</span></span>
<span id="cb1-323"><a href="#cb1-323"></a><span class="in">  | `fviz_cluster()`      | Visualize Clustering Results |</span></span>
<span id="cb1-324"><a href="#cb1-324"></a><span class="in">  | `fviz_nbclust()`      | Determining and Visualizing the Optimal Number of Clusters |</span></span>
<span id="cb1-325"><a href="#cb1-325"></a><span class="in">  | `fviz_silhouette()`   | Visualize Silhouette Information from Clustering |</span></span>
<span id="cb1-326"><a href="#cb1-326"></a><span class="in">  | `geodist()`           | Convert Lon-Lat Coordinates into Vector or Matrix of Geodesic Distances in Metres |</span></span>
<span id="cb1-327"><a href="#cb1-327"></a><span class="in">  | `ggcorrplot()`        | Visualisation of a Correlation Matrix Using ggplot2 |</span></span>
<span id="cb1-328"><a href="#cb1-328"></a><span class="in">  | `goodness()`          | Diagnostic Tools for [Constrained] Ordination (CCA, RDA, DCA, CA, PCA) |</span></span>
<span id="cb1-329"><a href="#cb1-329"></a><span class="in">  | `imputePCA()`         | Impute Dataset with PCA |</span></span>
<span id="cb1-330"><a href="#cb1-330"></a><span class="in">  | `lowerTriangle()`     | Extract or Replace the Upper/Lower Triangular Portion of a Matrix |</span></span>
<span id="cb1-331"><a href="#cb1-331"></a><span class="in">  | `manyglm()`           | Fitting Generalized Linear Models for Multivariate Abundance Data |</span></span>
<span id="cb1-332"><a href="#cb1-332"></a><span class="in">  | `meanvar.plot()`      | Construct Mean-Variance Plots for Multivariate Abundance Data |</span></span>
<span id="cb1-333"><a href="#cb1-333"></a><span class="in">  | `metaMDS()`           | Nonmetric Multidimensional Scaling with Stable Solution from Random Starts, Axis Scaling and Species Scores |</span></span>
<span id="cb1-334"><a href="#cb1-334"></a><span class="in">  | `mvabund()`           | Multivariate Abundance Data Objects |</span></span>
<span id="cb1-335"><a href="#cb1-335"></a><span class="in">  | `ordiellipse()`       | Display Groups or Factor Levels in Ordination Diagrams |</span></span>
<span id="cb1-336"><a href="#cb1-336"></a><span class="in">  | `ordiplot()`          | Alternative Plot and Identify Functions for Ordination |</span></span>
<span id="cb1-337"><a href="#cb1-337"></a><span class="in">  | `ordispider()`        | Display Groups or Factor Levels in Ordination Diagrams |</span></span>
<span id="cb1-338"><a href="#cb1-338"></a><span class="in">  | `ordisurf()`          | Fit and Plot Smooth Surfaces of Variables on Ordination |</span></span>
<span id="cb1-339"><a href="#cb1-339"></a><span class="in">  | `pam()`               | Partitioning Around Medoids |</span></span>
<span id="cb1-340"><a href="#cb1-340"></a><span class="in">  | `permutest()`         | Permutation Test for CCA, RDA and CA of Principal Coordinates |</span></span>
<span id="cb1-341"><a href="#cb1-341"></a><span class="in">  | `rda()`               | [Partial] [Constrained] Correspondence Analysis and Redundancy Analysis |</span></span>
<span id="cb1-342"><a href="#cb1-342"></a><span class="in">  | `RsquareAdj()`        | Adjusted R-square |</span></span>
<span id="cb1-343"><a href="#cb1-343"></a><span class="in">  | `scores()`            | Get Species or Site Scores from an Ordination |</span></span>
<span id="cb1-344"><a href="#cb1-344"></a><span class="in">  | `specnumber()`        | Ecological Diversity Indices |</span></span>
<span id="cb1-345"><a href="#cb1-345"></a><span class="in">  | `stressplot()`        | Goodness of Fit and Shepard Plot for Nonmetric Multidimensional Scaling |</span></span>
<span id="cb1-346"><a href="#cb1-346"></a><span class="in">  | `t()`                 | Matrix Transpose |</span></span>
<span id="cb1-347"><a href="#cb1-347"></a><span class="in">  | `vegdist()`           | Dissimilarity Indices for Community Ecologists |</span></span>
<span id="cb1-348"><a href="#cb1-348"></a><span class="in">  | `vif.cca()`           | Diagnostic Tools for [Constrained] Ordination (CCA, RDA, DCA, CA, PCA) | --&gt;</span></span>
<span id="cb1-349"><a href="#cb1-349"></a><span class="in">```</span></span>
<span id="cb1-350"><a href="#cb1-350"></a></span>
<span id="cb1-351"><a href="#cb1-351"></a><span class="fu"># References</span></span>
<span id="cb1-352"><a href="#cb1-352"></a></span>
<span id="cb1-353"><a href="#cb1-353"></a>::: {#refs}</span>
<span id="cb1-354"><a href="#cb1-354"></a>:::</span></code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>© 2024, AJ Smit</p>
</div>   
    <div class="nav-footer-center">

<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB743/BCB743_index.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
<p>This page is built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>




<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>