<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.510">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="AJ Smit">
<meta name="dcterms.date" content="2022-08-08">
<meta name="description" content="The Biodiversity and Conservation Biology R Pages serving BCB743.">

<title>The Tangled Bank - BCB743: Quantitative Ecology</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../BCB743/00-review.html" rel="next">
<link href="../images/favicon.ico" rel="icon">
<script src="../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    null
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-SCYH68M489"></script>

<script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-SCYH68M489', { 'anonymize_ip': true});
</script>

<script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"simple",
  "consent_type":"implied",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary","functionality","tracking","targeting"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  ,
"language":"en"
  });
});
</script> 
  
<style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>


</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">The Tangled Bank</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-vignettes" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Vignettes</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-vignettes">    
        <li>
    <a class="dropdown-item" href="../vignettes/netCDF_dates.html">
 <span class="dropdown-text">Dates From netCDF Files: Two Approaches</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/prep_NOAA_OISST.html">
 <span class="dropdown-text">Downloading and Prep: ERDDAP</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_ERDDAP.html">
 <span class="dropdown-text">Retrieving Chlorophyll-a Data from ERDDAP Servers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/alt_method.html">
 <span class="dropdown-text">Downloading and Prep: Python + CDO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/gridded_data.html">
 <span class="dropdown-text">Detecting Events in Gridded Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/regridding.html">
 <span class="dropdown-text">Regridding gridded data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/MHW_MCS_horizonplots.html">
 <span class="dropdown-text">Displaying MHWs and MCSs as Horizon Plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/heatwaveR_issues.html">
 <span class="dropdown-text">heatwaveR Issues</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_sightings.html">
 <span class="dropdown-text">Plotting the Whale Sightings and Chlorophyll-<em>a</em> Concentrations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_localisation.html">
 <span class="dropdown-text">Spatial Localisation, Subsetting, and Aggregation of the Chlorophyll-<em>a</em> Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/elem_ts_methods.html">
 <span class="dropdown-text">Wavelet Analysis of Diatom Time Series</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-high-performace-computing" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">High-Performace Computing</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-high-performace-computing">    
        <li>
    <a class="dropdown-item" href="../vignettes/README_Lengau.html">
 <span class="dropdown-text">Using Lengau</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/PBSPro_users.html">
 <span class="dropdown-text">PBSPro user and job management</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/README_tmux.html">
 <span class="dropdown-text">Using tmux</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../blog.html"> 
<span class="menu-text">Tangled Bank Blog</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-blogroll" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Blogroll</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-blogroll">    
        <li>
    <a class="dropdown-item" href="https://www.r-bloggers.com">
 <span class="dropdown-text">R-bloggers</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="../blog.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/ajsmit"> <i class="bi bi-twitter" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/ajsmit/tangled_bank">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://bugs.com">
 <span class="dropdown-text">Report a Bug</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BCB743/BCB743_index.html">BCB743</a></li><li class="breadcrumb-item"><a href="../BCB743/BCB743_index.html">BCB743: Quantitative Ecology</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-center sidebar-header">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../images/hex_logo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
 <span class="menu-text">‚òòÔ∏é</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">UNDERGRADUATE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
 <span class="menu-text">BDC223</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/BDC223_FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
 <span class="menu-text">BDC334</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/BDC334_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BDC334: Macroecology and Global Biogeography</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1: Ecological Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/02a-r_rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2a: R and RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/02b-env_dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2b: Environmental Distance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/03-biodiversity1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3: Quantifying Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/04-biodiversity2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4: Describing Biodiversity Patterns</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/Transboundary_systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transboundary systems</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">‚òò</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS CORE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false">
 <span class="menu-text">BCB744 (Intro R)</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/BCB744_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744: Introduction to R, and Biostatistics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/01-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. R and RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/02-working-with-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Working with data and code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/03-workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. R workflows</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/04-graphics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Graphics with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/05-faceting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Faceting figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/06-brewing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Brewing colours</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/07-mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Mapping with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/08-mapping_style.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Mapping with style</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/09-mapping_rnaturalearth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Mapping with Natural Earth and the <strong>sf</strong> package</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/10-mapping_quakes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. The Fiji Earthquake Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/11-tidy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Tidy data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/12-tidier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Tidier data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/13-tidiest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Tidiest data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/14-recap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Recap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/15-functions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15. Functions by chapter</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/16-base_r.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16. Base R primer</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/17-dates.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">17. Dates</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false">
 <span class="menu-text">BCB744 (Biostatistics)</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/01-data-in-R.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Data classes and structures in R</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/02-summarise-and-describe.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Exploring with summaries and descriptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/03-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Exploring with figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/04-distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/05-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Statistical inference and hypothesis testing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/05a-decision_guide.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5a. A quick guide to selecting the right test</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/06-assumptions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Assumptions for parametric statistics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/07-t_tests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. <em>t</em>-tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/08-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Analysis of Variance (ANOVA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/09-regressions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Simple linear regressions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/10-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/11-glance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Non-parametric statistical tests at a glance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/12-confidence.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Confidence intervals</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">‚òò</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS ELECTIVES</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true">
 <span class="menu-text">BCB743</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/BCB743_index.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">BCB743: Quantitative Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/00-review.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1-4. Review Biogeography and Global Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/05-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Correlations and Associations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/06-deep_dive.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Deep Dive into Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/07-ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/08-PCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8a. Principal Component Analysis (PCA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/08-PCA_examples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8b. PCA Additional Examples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/08-PCA_SDG.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8c. PCA of WHO SDGs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/09-CA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Correspondence Analysis (CA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/10-PCoA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Principal Coordinate Analysis (PCoA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/11-nMDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11a. non-Metric Multidimensional Scaling (nMDS)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/11-nMDS_diatoms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11b. nMDS of Mayombo‚Äôs Diatom Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/12-constrained_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Constrained Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/13-cluster_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Cluster Analysis</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">‚òòÔ∏é</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">SUPPORT</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
 <span class="menu-text">Web Resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/general_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/ecology_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quantitative Ecology Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/spatial_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial R Resources</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">‚òòÔ∏é</span>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#honours-coordinator" id="toc-honours-coordinator" class="nav-link active" data-scroll-target="#honours-coordinator">Honours Coordinator</a></li>
  <li><a href="#module-coordinator" id="toc-module-coordinator" class="nav-link" data-scroll-target="#module-coordinator">Module Coordinator</a></li>
  <li><a href="#instructors" id="toc-instructors" class="nav-link" data-scroll-target="#instructors">Instructors</a></li>
  <li><a href="#module-description" id="toc-module-description" class="nav-link" data-scroll-target="#module-description">Module description</a></li>
  <li><a href="#module-content-and-framework" id="toc-module-content-and-framework" class="nav-link" data-scroll-target="#module-content-and-framework">Module content and framework</a></li>
  <li><a href="#outcomes-of-bcb743" id="toc-outcomes-of-bcb743" class="nav-link" data-scroll-target="#outcomes-of-bcb743">Outcomes of BCB743</a></li>
  <li><a href="#graduate-attributes" id="toc-graduate-attributes" class="nav-link" data-scroll-target="#graduate-attributes">Graduate attributes</a></li>
  <li><a href="#data-and-reading-in-support-of-the-syllabus" id="toc-data-and-reading-in-support-of-the-syllabus" class="nav-link" data-scroll-target="#data-and-reading-in-support-of-the-syllabus">Data and reading in support of the syllabus</a>
  <ul class="collapse">
  <li><a href="#reading" id="toc-reading" class="nav-link" data-scroll-target="#reading">Reading</a></li>
  <li><a href="#datasets-used-in-this-module" id="toc-datasets-used-in-this-module" class="nav-link" data-scroll-target="#datasets-used-in-this-module">Datasets used in this module</a></li>
  </ul></li>
  <li><a href="#prerequisites" id="toc-prerequisites" class="nav-link" data-scroll-target="#prerequisites">Prerequisites</a></li>
  <li><a href="#method-of-instruction" id="toc-method-of-instruction" class="nav-link" data-scroll-target="#method-of-instruction">Method of instruction</a></li>
  <li><a href="#learning-colaboratively" id="toc-learning-colaboratively" class="nav-link" data-scroll-target="#learning-colaboratively">Learning colaboratively</a></li>
  <li><a href="#reusing-code-found-elsewhere" id="toc-reusing-code-found-elsewhere" class="nav-link" data-scroll-target="#reusing-code-found-elsewhere">Reusing code found elsewhere</a></li>
  <li><a href="#software" id="toc-software" class="nav-link" data-scroll-target="#software">Software</a></li>
  <li><a href="#computers" id="toc-computers" class="nav-link" data-scroll-target="#computers">Computers</a></li>
  <li><a href="#attendance" id="toc-attendance" class="nav-link" data-scroll-target="#attendance">Attendance</a></li>
  <li><a href="#sec-policy" id="toc-sec-policy" class="nav-link" data-scroll-target="#sec-policy">Assessment policy</a>
  <ul class="collapse">
  <li><a href="#continuous-assessment" id="toc-continuous-assessment" class="nav-link" data-scroll-target="#continuous-assessment">Continuous Assessment</a>
  <ul class="collapse">
  <li><a href="#tasks" id="toc-tasks" class="nav-link" data-scroll-target="#tasks">Tasks</a></li>
  <li><a href="#integrative-assessment" id="toc-integrative-assessment" class="nav-link" data-scroll-target="#integrative-assessment">Integrative Assessment</a></li>
  </ul></li>
  <li><a href="#final-assessment-exam" id="toc-final-assessment-exam" class="nav-link" data-scroll-target="#final-assessment-exam">Final Assessment (Exam)</a></li>
  <li><a href="#submission-of-assignments-and-exams" id="toc-submission-of-assignments-and-exams" class="nav-link" data-scroll-target="#submission-of-assignments-and-exams">Submission of assignments and exams</a></li>
  <li><a href="#late-submission-of-ca" id="toc-late-submission-of-ca" class="nav-link" data-scroll-target="#late-submission-of-ca">Late submission of CA</a></li>
  </ul></li>
  <li><a href="#support" id="toc-support" class="nav-link" data-scroll-target="#support">Support</a>
  <ul class="collapse">
  <li><a href="#help-via-bcb744-and-bcb743-issues-on-github" id="toc-help-via-bcb744-and-bcb743-issues-on-github" class="nav-link" data-scroll-target="#help-via-bcb744-and-bcb743-issues-on-github">Help via BCB744 and BCB743 Issues on GitHub</a></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB743/BCB743_index.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full column-body" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BCB743/BCB743_index.html">BCB743</a></li><li class="breadcrumb-item"><a href="../BCB743/BCB743_index.html">BCB743: Quantitative Ecology</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">BCB743: Quantitative Ecology</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>

<div>
  <div class="description">
    The Biodiversity and Conservation Biology R Pages serving BCB743.
  </div>
</div>

<div class="quarto-title-meta-author column-body">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://ajsmit.netlify.app">AJ Smit</a> <a href="https://orcid.org/0000-0002-3799-6126" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
    <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uwc.ac.za">
            University of the Western Cape
            </a>
          </p>
      </div>
    </div>

<div class="quarto-title-meta column-body">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">August 8, 2022</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<blockquote class="blockquote">
<p><em>‚ÄúWe have become, by the power of a glorious evolutionary accident called intelligence, the stewards of life‚Äôs continuity on earth. We did not ask for this role, but we cannot abjure it. We may not be suited to it, but here we are.‚Äù</em></p>
<p>‚Äî Stephen J. Gould</p>
</blockquote>
<blockquote class="blockquote">
<p><em>‚ÄúReports that say that something hasn‚Äôt happened are always interesting to me, because as we know, there are known knowns; there are things we know we know. We also know there are known unknowns; that is to say we know there are some things we do not know. But there are also unknown unknowns- the ones we don‚Äôt know we don‚Äôt know.‚Äù</em></p>
<p>‚Äî Donald Rumsfeld</p>
</blockquote>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../images/PhD_comics3.JPG" id="comic" class="img-fluid quarto-figure quarto-figure-center anchored figure-img"></p>
</figure>
</div>
<p><strong>Welcome to the pages for BCB743 Quantitative Ecology. This page provides the syllabus and teaching policies for the module, and it serves is a starting point accessing all the theory, instruction, and data.</strong></p>
<section id="honours-coordinator" class="level1">
<h1>Honours Coordinator</h1>
<p>Prof.&nbsp;Bryan Maritz‚ÄîRoom 4.105, Department of Biodiversity &amp; Conservation Biology</p>
</section>
<section id="module-coordinator" class="level1">
<h1>Module Coordinator</h1>
<p>Prof.&nbsp;AJ Smit‚ÄîRoom 4.103, Department of Biodiversity &amp; Conservation Biology, ajsmit@uwc.ac.za</p>
</section>
<section id="instructors" class="level1">
<h1>Instructors</h1>
<p>Jesse Phillips‚ÄîDepartment of Biodiversity &amp; Conservation Biology, jessephillips11.jp@gmail.com</p>
<p>Tom Spencer-Hicken‚ÄîDepartment of Biodiversity &amp; Conservation Biology, 3941863@myuwc.ac.za</p>
</section>
<section id="module-description" class="level1">
<h1>Module description</h1>
<p>Quantitative ecology is a field within the ecological sciences that uses statistical and computational approaches to understand ecosystems. It aims to describe and quantify ecological processes, analyse and model complex multivariate ecological data, and make predictions about the structure and dynamics of ecosystems across a range of spatial and temporal scales. The multivariate statistical approaches taught in this module will prepare the student to make sense of bewildering multidimensional datasets in a comprehensible two- or three-dimensional space.</p>
<p>Coursework on quantitative ecology can cover topics such as:</p>
<ul>
<li><p><strong>Ecological structure</strong> We will cover the basic principles underlying the environmental structuring of ecosystems (ecosystem structure).</p></li>
<li><p><strong>Ecological data analysis</strong> This involves studying statistical techniques to analyse ecological data, including hypothesis testing, regression analysis, and multivariate analysis.</p></li>
<li><p><strong>Multivariate analyses</strong> In this section, we learn how to use multivariate statistics to make sense of complex systems, predict ecological outcomes, and understand the underlying mechanisms that drive ecological processes.</p></li>
<li><p><strong>Spatial ecology</strong> Here we learn how to analyse and model spatial patterns in ecological data, including the distribution of species and habitats across landscapes.</p></li>
<li><p><strong>Population dynamics</strong> This is the theory about how to model and analyse the dynamics of populations, including birth and death rates, population growth, and interactions between species.</p></li>
<li><p><strong>Community ecology</strong> Theory in this section of the work deals with analysing and modelling the interactions between species within ecological communities, including competition, predation, and mutualism.</p></li>
<li><p><strong>Ecosystem ecology</strong> Here we will learn how to model and analyse the flow of energy and nutrients through ecosystems, including the role of producers, consumers, and decomposers in ecological processes.</p></li>
</ul>
<p>So, a course in quantitative ecology provides you with the skills and tools needed to analyse and model ecological data, communities, and make predictions about the structure and dynamics of ecosystems. It prepares students for careers in research, conservation, natural resource management, and environmental consulting.</p>
<p>In BCB743 I will focus on <strong>multivariate statistics</strong>. Multivariate methods play a central role in quantitative ecology as they enable researchers to analyse and interpret complex data sets that involve multiple variables and hordes of species. Ecological systems are characterised by their complexity, and there are typically many different factors that influence the distribution and abundance of species in an ecosystem. Multivariate statistical methods allow us to identify the underlying patterns and relationships among these variables and to explore how they interact to shape ecological communities.</p>
<p>Some of the most commonly used multivariate statistical techniques in quantitative ecology include ordination methods, such as principal component analysis (PCA) and correspondence analysis (CA), which are used to reduce complexity and permit the visualisation of patterns of species distribution in ecological communities. Cluster analysis, discriminant analysis, and canonical correlation analysis are other multivariate methods commonly used to analyse ecological data. These techniques can help us to identify groups of species or environmental factors that are associated with particular ecological conditions, as well as to assess the relative importance of different variables in shaping ecological communities.</p>
<p>In addition to providing a framework for data analysis, multivariate statistics also play an important role in developing ecological models. By incorporating multivariate data into models, we can gain a more complete understanding of the ecological processes that govern the behaviour of species and ecosystems. This, in turn, can inform management and conservation efforts aimed at preserving ecological diversity and resilience in the face of environmental change. Multivariate statistics are an essential tool for ecologists seeking to understand the complex relationships between species and their environment.</p>
</section>
<section id="module-content-and-framework" class="level1 page-columns page-full">
<h1>Module content and framework</h1>
<p>These links point to online resources such as datasets and R scripts in support of the video and PDF lecture material. It is essential that you work through these examples and workflows. Here is the Syllabus:</p>
<div class="column-page-inset-right">
<table class="table">
<thead>
<tr class="header">
<th>Week</th>
<th>Class date</th>
<th>Lecture</th>
<th>Topic</th>
<th>Class</th>
<th>Slides</th>
<th>Reading</th>
<th>Tasks</th>
<th>Tasks due</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>W1</td>
<td>20 June</td>
<td>L1</td>
<td>Introduction</td>
<td></td>
<td><a href="../slides/BCB743_01_intro.pdf">üò≥</a></td>
<td><a href="../docs/BES-Reproducible-Code-2017.pdf">ü•¥</a> <a href="../docs/BES-Data-Guide-2017_web.pdf">ü•¥</a></td>
<td><a href="../BCB743/assessments/Task_A.html">üò±</a></td>
<td>27 June</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td><strong>BIODIVERSITY</strong></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>revision</td>
<td>Ecological Data</td>
<td><a href="../BDC334/01-introduction.html">üò≠</a></td>
<td></td>
<td></td>
<td>Lab 1 Review</td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td><strong>DATA, MATRICES</strong></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>revision</td>
<td>Environmental Distance</td>
<td><a href="../BDC334/02b-env_dist.html">üò≠</a></td>
<td><a href="../slides/BCB743_04_environmental_distance.pdf">üò≥</a></td>
<td></td>
<td>Lab 2 Review</td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>revision</td>
<td>Quantifying Biodiversity</td>
<td><a href="../BDC334/03-biodiversity1.html">üò≠</a></td>
<td><a href="../slides/BCB743_02_biodiversity_1.pdf">üò≥</a></td>
<td><a href="../docs/Smit_et_al_2013.pdf">ü•¥</a> <a href="../docs/Smit_et_al_2017.pdf">ü•¥</a> <a href="../docs/Smit_the_seaweed_data.pdf">ü•¥</a></td>
<td>Lab 3 Review</td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>revision</td>
<td>Describing Biodiversity Patterns</td>
<td><a href="../BDC334/04-biodiversity2.html">üò≠</a></td>
<td><a href="../slides/BCB743_02_biodiversity_2.pdf">üò≥</a></td>
<td><a href="../docs/Shade_et_al_2018.pdf">ü•¥</a></td>
<td>Lab 4 Review</td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>revision</td>
<td>Ecological Theories</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>W1</td>
<td>22 June</td>
<td>L2</td>
<td>Correlations and Associations</td>
<td><a href="../BCB743/05-correlations.html">üò≠</a></td>
<td><a href="../slides/BCB743_06_correlations.pdf">üò≥</a></td>
<td></td>
<td><a href="../BCB743/assessments/Task_B.html">üò±</a></td>
<td>23 June</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>self</td>
<td>Deeper Into Gradients</td>
<td><a href="../BCB743/06-deep_dive.html">üò≠</a></td>
<td></td>
<td><a href="../docs/Smit_et_al_2013.pdf">ü•¥</a> <a href="../docs/Smit_et_al_2017.pdf">ü•¥</a> <a href="../docs/Smit_the_seaweed_data.pdf">ü•¥</a></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td></td>
<td><strong>ORDINATION</strong></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>W2</td>
<td>27 June</td>
<td>L3</td>
<td>Ordination</td>
<td></td>
<td><a href="../slides/BCB743_07_ordination.pdf">üò≥</a></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>L3</td>
<td>PCA (a)</td>
<td><a href="../BCB743/08-PCA.html">üò≠</a></td>
<td><a href="../slides/BCB743_08_PCA.pdf">üò≥</a></td>
<td></td>
<td><a href="../BCB743/assessments/Task_C.html">üò±</a></td>
<td>28 June</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>self</td>
<td>PCA (b) Additional Examples</td>
<td><a href="../BCB743/08-PCA_examples.html">üò≠</a></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>self</td>
<td>PCA (c) WHO SDGs</td>
<td><a href="../BCB743/08-PCA_SDG.html">üò≠</a></td>
<td></td>
<td></td>
<td><a href="../BCB743/assessments/Task_D.html">üò±</a></td>
<td>19 July</td>
</tr>
<tr class="even">
<td>W3</td>
<td>4 July</td>
<td>L4</td>
<td>CA</td>
<td><a href="../BCB743/09-CA.html">üò≠</a></td>
<td><a href="../slides/BCB743_09_CA.pdf">üò≥</a></td>
<td></td>
<td><a href="../BCB743/assessments/Task_E.html">üò±</a></td>
<td>5 July</td>
</tr>
<tr class="odd">
<td></td>
<td>6 July</td>
<td>L5</td>
<td>PCoA</td>
<td><a href="../BCB743/10-PCoA.html">üò≠</a></td>
<td><a href="../slides/BCB743_10_PCoA.pdf">üò≥</a></td>
<td></td>
<td><a href="../BCB743/assessments/BCB743_intgrative_assignment.html">üò±</a></td>
<td>7 July</td>
</tr>
<tr class="even">
<td>W4</td>
<td>11 July</td>
<td>L6</td>
<td>nMDS</td>
<td><a href="../BCB743/11-nMDS.html">üò≠</a></td>
<td><a href="../slides/BCB743_11_nMDS.pdf">üò≥</a></td>
<td></td>
<td><a href="../BCB743/assessments/Task_F.html">üò±</a></td>
<td>12 July</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>self</td>
<td>nMDS + PERMANOVA (diatoms)</td>
<td><a href="../BCB743/11-nMDS_diatoms.html">üò≠</a></td>
<td></td>
<td><a href="../docs/Mayombo_et_al_2019.pdf">ü•¥</a></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td>13 July</td>
<td>L7</td>
<td>CCA &amp; RDA</td>
<td><a href="../BCB743/12-constrained_ordination.html">üò≠</a></td>
<td><a href="../slides/BCB743_12_constrained_ordination.pdf">üò≥</a></td>
<td><a href="../docs/Smit_et_al_2017.pdf">ü•¥</a> <a href="../docs/Smit_the_seaweed_data.pdf">ü•¥</a></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td></td>
<td><strong>CLUSTER ANALYSIS</strong></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>W5</td>
<td>18 July</td>
<td>L8</td>
<td>Cluster Analysis</td>
<td><a href="../BCB743/13-cluster_analysis.html">üò≠</a></td>
<td></td>
<td></td>
<td><a href="../BCB743/assessments/Task_D.html">üò±</a></td>
<td>19 July</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td>L9</td>
<td>Review &amp; Integrative Assignment</td>
<td></td>
<td></td>
<td></td>
<td><a href="../BCB743/assessments/BCB743_intgrative_assignment.html">üò±</a></td>
<td>2 August</td>
</tr>
<tr class="even">
<td>W7</td>
<td>3‚Äì4 Aug</td>
<td></td>
<td><strong>EXAM</strong></td>
<td></td>
<td></td>
<td></td>
<td></td>
<td>4 August</td>
</tr>
</tbody>
</table>
</div>
<p><strong>Core theoretical framework</strong> Ecological hypotheses underlying the processes of species assembly in space and time, including neutral and niche-based mechanisms, and historical events; overview of the currently known and understood distributional patterns of major groups of organisms at global, regional and local scales; consideration of sampling designs aimed at capturing these patterns and drivers so as to arrive at a processed based understanding of species assembly.</p>
<p><strong>Competence</strong> Data collection aimed at a quantitative test of the relevant hypotheses, above. The management and analysis of ecological data; reproducible and collaborative research; the use of R as a tool for the analysis multivariate ecological data; multivariate techniques such as nMDS, PCA, RDA and cluster analysis; graphical data summaries and visualisations.</p>
</section>
<section id="outcomes-of-bcb743" class="level1">
<h1>Outcomes of BCB743</h1>
<p>By the end of this module, students will be able to:</p>
<ul>
<li>Understand the concepts of alpha, beta and gamma diversity</li>
<li>know and understand the current hypotheses that attempt to explain species assembly processes in space and time (e.g.&nbsp;neutral and niche mechanisms)</li>
<li>collect ecological data at the appropriate scale, which would lend themselves to a quantitative analysis of points 1 and 2, above</li>
<li>use the R software and associated packages to undertake the analyses required in point 3, above</li>
<li>interpret the outcomes of the above analyses and use it to quantitatively characterise points 1 and 2, above</li>
<li>communicate the findings by written and oral means</li>
</ul>
</section>
<section id="graduate-attributes" class="level1">
<h1>Graduate attributes</h1>
<p>The <a href="../pages/graduate_attributes.html"><strong>graduate attributes</strong></a> resulting from completion of this modules alignment with the expectations of the workspace across diverse organisations and institutions where graduates typically find employment.</p>
</section>
<section id="data-and-reading-in-support-of-the-syllabus" class="level1">
<h1>Data and reading in support of the syllabus</h1>
<p>In the table above there are links to several key papers to read in preparation of each week‚Äôs theory. It is essential that you read these papers.</p>
<p>Many other references are cited in each Chapter. These serve several functions in that they:</p>
<ul>
<li>Add additional theory relevant to some ecological concepts</li>
<li>Provide background to some of the datasets used in my examples</li>
<li>Discuss derivations of some equations used to calculate diversity concepts</li>
<li>Provide example walkthroughs of some of the computational aspects of the methods covered in the Labs</li>
<li>Collectively supplement the discussion about these concepts covered in the lectures</li>
</ul>
<p>Actively engaging with these reading materials will make to difference between a 60% average mark for the module, and a mark in excess of 80%.</p>
<section id="reading" class="level2">
<h2 class="anchored" data-anchor-id="reading">Reading</h2>
<p>You are <strong>expected to read additional material</strong> in support of the content covered in class and on this website.</p>
<p>A <strong>compulsory</strong> reference is <a href="http://adn.biol.umontreal.ca/~numericalecology/numecolR/">Numerical Ecology with R</a> by Daniel Borcard, Fran√ßois Gillet and Pierre Legendre. Much of the class‚Äô content and many of the examples (and code) that I use have been adapted from this source. It should be considered the ‚Äògold standard‚Äô reference for Quantitative Ecology.</p>
<p>A second <strong>highly recommended text</strong> is the book <a href="http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf">Tree Diversity Analysis</a> by Roeland Kindt and Richard Coe.</p>
<p>I can also recommend a few websites with excellent content, such as:</p>
<ul>
<li>David Zelen√Ω‚Äôs <a href="https://www.davidzeleny.net/anadat-r/doku.php">Analysis of Community Ecology Data in R</a></li>
<li>Mike Palmer‚Äôs <a href="http://ordination.okstate.edu/">Ordination Methods for Ecologists</a></li>
<li><a href="https://sites.google.com/site/mb3gustame/">GUide to STatistical Analysis in Microbial Ecology (GUSTA ME)</a></li>
</ul>
<p>Note that the URLs with links to additional reading that appear with the worked-through example code should <strong>not be seen as optional</strong>. They are there for a reason and should be consulted even though I might not necessarily refer to each of them in class.</p>
<p>Use these materials liberally.</p>
<p>Should you want to download the source code for the BCB743 (and BCB744 website), you may find it on <a href="https://github.com/ajsmit/R_courses"><i class="fab fa-github"></i> GitHub</a>.</p>
</section>
<section id="datasets-used-in-this-module" class="level2">
<h2 class="anchored" data-anchor-id="datasets-used-in-this-module">Datasets used in this module</h2>
<p>Note that the links provided might not necessarily lead to the <strong>vegan</strong> help page.</p>
<table class="table">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th>Dataset</th>
<th>Source</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td>Vegetation and Environment in Dutch Dune Meadows</td>
<td><a href="https://www.davidzeleny.net/anadat-r/doku.php/en:data:dune"><strong>vegan</strong></a></td>
</tr>
<tr class="even">
<td style="text-align: left;">2</td>
<td>Oribatid Mite Data with Explanatory Variables</td>
<td><a href="http://adn.biol.umontreal.ca/~numericalecology/data/oribates.html"><strong>vegan</strong></a></td>
</tr>
<tr class="odd">
<td style="text-align: left;">3</td>
<td>The Doubs River Data</td>
<td><a href="https://www.davidzeleny.net/anadat-r/doku.php/en:data:doubs"><strong>Numerical Ecology with R</strong></a></td>
</tr>
<tr class="even">
<td style="text-align: left;">4</td>
<td>The Barro Colorado Island Tree Counts</td>
<td><a href="https://www.davidzeleny.net/anadat-r/doku.php/en:data:bci"><strong>vegan</strong></a></td>
</tr>
<tr class="odd">
<td style="text-align: left;">5</td>
<td>John Bolton, Rob Anderson, and Herre Stegenga‚Äôs Seaweed Data</td>
<td><a href="https://www.frontiersin.org/articles/10.3389/fmars.2017.00404/"><strong>Smit et al., 2017</strong></a></td>
</tr>
<tr class="even">
<td style="text-align: left;">6</td>
<td>Serge Mayombo‚Äôs Diatoms Data</td>
<td><a href="https://www.tandfonline.com/doi/abs/10.2989/1814232X.2019.1592778"><strong>Mayombo et al., 2019</strong></a></td>
</tr>
<tr class="odd">
<td style="text-align: left;">7</td>
<td>World Health Organization Sustainable Development Goals Data</td>
<td><a href="https://www.who.int/data"><strong>WHO</strong></a></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="prerequisites" class="level1">
<h1>Prerequisites</h1>
<p>You should have a moderate numerical literacy and have prior programming experience. Such experience will have been obtained in the <a href="../workshops/">BCB744</a> module, which is a module about doing statistics in R. If you have a reasonable experience in coding and statistical analysis you should find yourself well prepared. You should also thoroughly revise <a href="../BDC334/BDC334_index.html">BDC334</a> by the end of the first week of this module.</p>
</section>
<section id="method-of-instruction" class="level1">
<h1>Method of instruction</h1>
<p>You are provided with reading material (lecture slides, code, website content) that you are expected to consume <strong>prior to the class</strong>. Classes will involve brief introductions to new concepts, and will be followed by working on exercises in class that cover those concepts. The workshop is designed to be as interactive as possible, so while you are working on exercises the tutor and I will circulate among you and engage with you to help you understand any material and the associated code you are uncomfortable with. Often this will result in discussions of novel applications and alternative approaches to the data analysis challenges you are required to solve. More challenging concepts might emerge during the assignments (typically these will be submitted the following day), and any such challenges will be dealt with in class prior to learning new concepts.</p>
<p>Although the module is theory-heavy, a large part of it is also about coding. It is up to you to take your coding skills to the next level and move beyond what I teach in class. Coding is a bit like learning a language, and as such programming is a skill that is best learned by doing.</p>
</section>
<section id="learning-colaboratively" class="level1">
<h1>Learning colaboratively</h1>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Also read: How to learn
</div>
</div>
<div class="callout-body-container callout-body">
<p>Please refer to my <a href="../pages/How_to_learn.html">advice about how to learn</a>.</p>
</div>
</div>
<p>Discuss the BCB743 workshop activities with your peers as you work on them. Use also the WhatsApp group set up for the module for discussion purposes (I might assist via this medium if neccesary if your questions/comments have relevance to the whole class). A better option is to use <a href="../quantecol/#help-via-bcb744-and-bcb743-issues-on-github">GitHub Issues</a>. You will learn more in this module if you work with your friends than if you do not. Ask questions, answer questions, and share ideas liberally. Please identify your work partners by name on all assignments (if you decide to work in pairs).</p>
<p><strong>Cooperative learning is not a licence for plagiarism. Plagiarism is a serious offence and will be dealt with concisely. Consequences of cheating are severe‚Äîthey range from a 0% for the assignment or exam up to dismissal from the course for a second offense.</strong></p>
</section>
<section id="reusing-code-found-elsewhere" class="level1">
<h1>Reusing code found elsewhere</h1>
<p>A huge volume of code is available on the web and it can be adapted to solve your own problems. You may make use of any online resources (e.g. form <a href="https://stackoverflow.com/">StackOverflow</a>, a thoroughly-used source of discussion about <a href="https://stackoverflow.com/questions/tagged/r">R code</a>)‚Äîbut you <strong>MUST</strong> clearly indicate (cite) that your solution relies on found code, regardless to what extent you have modified it to your own needs. Reused code that is discovered via a web search and which is not explicitly cited is plagiarism and it will be treated as such. On assignments you may not directly share code with your peers in this workshop.</p>
</section>
<section id="software" class="level1">
<h1>Software</h1>
<p>In this course we will rely entirely on <a href="https://cran.r-project.org/">R</a> running within the <a href="https://www.rstudio.com/">RStudio</a> IDE. The use of R was covered extensively in the <a href="http://localhost:4321/workshops/">BCB744</a> module where the <a href="http://localhost:4321/workshops/intro_r/chapters/02-rstudio/">installation process</a> was discussed. We will primarily use the <a href="https://cran.r-project.org/web/packages/vegan/index.html"><strong>vegan</strong></a> package, but some useful functions are also provided by the package <a href="https://github.com/cran/BiodiversityR"><strong>BiodiversityR</strong></a> (and <a href="http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf">here</a> and <a href="https://rpubs.com/Roeland-KINDT">here</a>). Various other R packages offer overlapping and additional methods, but <strong>vegan</strong> should accommodate &gt;90% of your Quantitative Ecology needs.</p>
</section>
<section id="computers" class="level1">
<h1>Computers</h1>
<p>You are encouraged to provide your own laptops and to install the necessary software before the module starts. Limited support can be provided if required. There are also computers with R and RStudio (and the neccesary add-on libraries) available in the 5th floor lab in the BCB Department.</p>
</section>
<section id="attendance" class="level1">
<h1>Attendance</h1>
<p>This workhop-based, hands on course can only deliver acceptible outcomes if you attend all classes. The schedule is set and cannot be changed. Sometimes an occasional absence cannot be avoided. Please be curtious and notify myself or the tutor in advance of any absence. If you work with a partner in class, notify them too. Keep up with the reading assignments while you are away and we will all work with you to get you back up to speed on what you miss. If you do miss a class, however, the assignments must still be submitted on time (also see <a href="../quantecol/#late-submission-of-ca"><strong>Late submission of CA</strong></a>).</p>
<p>Since you may decide to work in collaboration with a peer on tasks and assignments, please keep this person informed at all times in case some emergency makes you unavailable for a period of time. Someone might depend on your input and contributions‚Äîdo not leave someone in the lurch so that they cannot complete a task in your absence.</p>
</section>
<section id="sec-policy" class="level1">
<h1>Assessment policy</h1>
<p><strong>Continuous Assessments</strong> (CA) and a <strong>Final Assessment</strong> will provide a <strong>Final Mark</strong> for the module. These modes of assessment meet our needs as far as <a href="../pages/assessment_theory.html">formative and summative assessments</a> are concerned. The weighting of the CA and the Final Assessment is 0.6 and 0.4, respectively. All assessments are open book, so consult your code and reading material if and when you need to.</p>
<section id="continuous-assessment" class="level2">
<h2 class="anchored" data-anchor-id="continuous-assessment">Continuous Assessment</h2>
<p>Quantitative Ecology is about working with real-world datasets. To this end, a series of mini-projects (<strong>Tasks</strong>) involving real data are a required part of the BCB743. You may work alone or with a single partner on all projects (in which case you will share the same mark). All Tasks are due by the date and time indicated, and due care must be taken that they are submitted as instructed, i.e.&nbsp;paying attention to naming conventions and the format of the files submitted‚Äîtypically this will be in Quarto format (.qmd) and the knitted output (I prefer .html).</p>
<p>The Continuous Assessment is comprised of the material indicated in the Syllabus by the <i class="fa-balance-scale-right"></i> icon. There are seven smaller assessments (each weighted 0.1) and one slightly more demanding Integrative Assessment (weighted 0.3) at the end of the module.</p>
<section id="tasks" class="level3">
<h3 class="anchored" data-anchor-id="tasks">Tasks</h3>
<p>When assessing the tasks, we will pay attention to the following criteria:</p>
<ul>
<li>Content (10%):
<ul>
<li>Questions answered in order</li>
<li>Annotations (meta-data in file header, commnets about code, ideas, and approach)</li>
</ul></li>
<li>Code formatting and correctness (45%):
<ul>
<li>Application of <a href="http://adv-r.had.co.nz/Style.html">R code conventions</a>, e.g.&nbsp;spaces around <code>&lt;-</code>, after <code>#</code>, after <code>,</code>, etc.</li>
<li>New line for each <code>dplyr</code> function (lines end in <code>%&gt;%</code>) or <code>ggplot</code> layer (lines end in <code>+</code>)</li>
</ul></li>
<li>Figures (45%):
<ul>
<li>Sensible use of themes / colours</li>
<li>Publication quality</li>
<li>Informative and complete titles, axes labels, legends, etc.</li>
</ul></li>
</ul>
</section>
<section id="integrative-assessment" class="level3">
<h3 class="anchored" data-anchor-id="integrative-assessment">Integrative Assessment</h3>
<p>The above-mentioned Assessments will be graded with the following in expectations in mind:</p>
<ul>
<li>Content (20%):
<ul>
<li>Questions answered in order</li>
<li>A written explanation of approach included for each question</li>
<li>Appropriate formatting of text, for example, fonts not larger than necessary, headings used properly, etc. Be sensible and tasteful.</li>
</ul></li>
<li>Code formatting, structure, and correctness (50%):
<ul>
<li>Use <strong>Tidyverse</strong> code</li>
<li>No more than ~80 characters of code per line (pay particular attention to the comments)</li>
<li>Application of <a href="http://adv-r.had.co.nz/Style.html">R code conventions</a>, e.g.&nbsp;spaces around <code>&lt;-</code>, after <code>#</code>, after <code>,</code>, etc.</li>
<li>New line for each <code>dplyr</code> function (lines end in <code>%&gt;%</code>) or <code>ggplot</code> layer (lines end in <code>+</code>)</li>
<li>Proper indentation of pipes and <code>ggplot()</code> layers</li>
<li>All chunks labelled without spaces</li>
<li>No unwanted / commented out code left behind in the document</li>
</ul></li>
<li>Figures (30%):
<ul>
<li>Sensible use of themes / colours</li>
<li>Publication quality</li>
<li>Informative and complete titles, axes labels, legends, etc.</li>
<li>No redundant features or aesthetics</li>
</ul></li>
</ul>
<p>Random quizzes will not form part of the CA for BCB743.</p>
</section>
</section>
<section id="final-assessment-exam" class="level2">
<h2 class="anchored" data-anchor-id="final-assessment-exam">Final Assessment (Exam)</h2>
<p>The Final Assessment will be a 48-hour affair that you can do in the comfort of your home. It will involve the analysis of real world data, and it is expected that the submission includes 1) statements of aims, objectives, and hypotheses; 2) the full and detailed methods followed by analyses together with all code, 3) full reporting of results in a manner suited for peer reviewed publications; 4) graphical support highlighting the patterns observed (again with the code), and 5) a discussion if and when required. The allocation of marks to the various sections is:</p>
<ol type="1">
<li>Aims, objectives, and hypotheses: 5%</li>
<li>Methods and analyses: 45%</li>
<li>Results: 20%</li>
<li>Graphs: 15%</li>
<li>Dicsussion: 15%</li>
</ol>
</section>
<section id="submission-of-assignments-and-exams" class="level2">
<h2 class="anchored" data-anchor-id="submission-of-assignments-and-exams">Submission of assignments and exams</h2>
<p>A statement such as the one below accompanies every assignment‚Äîpay attention, as failing to observe this instruction may result in a loss of marks (i.e.&nbsp;if an assignment remains ungraded because the owner of the material cannot be identified):</p>
<p>Submit a R script wherein you provide answers to Questions 1‚Äì9 by no later than 8:00 tomorrow. Label the script as follows (e.g.): <strong>BCB743_AJ_Smit_Assignment_2.R</strong>.</p>
</section>
<section id="late-submission-of-ca" class="level2">
<h2 class="anchored" data-anchor-id="late-submission-of-ca">Late submission of CA</h2>
<p>Late assignments will be penalised 10% per day and will not be accepted more than 48 hours late, unless evidence such as a doctor‚Äôs note, a death certificate, or another documented emergency can be provided. If you know in advance that a submission will be late, please discuss this and seek prior approval. This policy is based on the idea that in order to learn how to translate your human thoughts into computer language (coding) you should be working with them at multiple times each week‚Äîideally daily. Time has been allocated in class for working on assignments and students are expected to continue to work on the assignments outside of class. Successfully completing (and passing) this module requires that you finish assignments based on what we have covered in class by the following class period. Work diligently from the onset so that even if something unexpected happens at the last minute you should already be close to done. This approach also allows rapid feedback to be provided to you, which can only be accomplished by returning assignments quickly and punctually.</p>
</section>
</section>
<section id="support" class="level1">
<h1>Support</h1>
<p>It‚Äôs expected that some tricky aspects of the module will take time to master, and the best way to master problematic material is to practice, practice some more, and then to ask questions. Trying for 10 minutes and then giving up is not good enough. I‚Äôll be more sympathetic to your cause if you can demonstrate having tried for a full day before giving up and asking me. When you ask questions about some challenge, this is the way to do it‚Äîexplain to me your numerous attempts at trying to solve the problem, and explain how these various attempts have failed. <em>I will not help you if you have not tried to help yourself first</em> (maybe with advice from friends). There will be time in class to do this, typically before we embark on a new topic. You are also encouraged to bring up related questions that arise in your own B.Sc. (Hons.) research project.</p>
<p>Should you require more time with me, find out when I am ‚Äòfree‚Äô and set an appointment by sending me a calendar invitation. I am happy to have a personal meeting with you via Zoom, but I prefer face-to-face in my office.</p>
<section id="help-via-bcb744-and-bcb743-issues-on-github" class="level2">
<h2 class="anchored" data-anchor-id="help-via-bcb744-and-bcb743-issues-on-github">Help via BCB744 and BCB743 Issues on GitHub</h2>
<p>All discussion for the BCB744 and BCB743 workshops will be held in the <a href="https://github.com/ajsmit/R_courses/issues">Issues</a> of <a href="https://github.com/ajsmit/R_courses">this repository</a>. Please post all content-related questions there, and use email only for personal matters. Note that this is a public repository, so be professional in your writing here (grammar, etc.).</p>
<p>To start a new thread, create a <strong>New issue</strong>. Tag your peers using their handle‚Äî<code>@ajsmit</code>, for example‚Äîto get their attention.</p>
<p>Once a question has been answered, the issue will be closed, so lots of good answers might end up in closed issues. Don‚Äôt forget to look there when looking for answers‚Äîyou can use the <strong>Search</strong> feature on this repository to find answers that might have been offered by the same or similar problem experienced by someone else in the past.</p>
<p><strong>Guidelines for posting questions:</strong></p>
<ul>
<li>First search existing issues (open or closed) for answers. If the question has already been answered, you‚Äôre done! If there is an open issue, feel free to contribute to it. Or feel free to open a closed issue if you believe the answer is not satisfactory.</li>
<li>Give your issue an informative title.
<ul>
<li>Good: ‚ÄúError: could not find function‚Äùggplot‚Äù‚Äù</li>
<li>Bad: ‚ÄúMy code does not work!‚Äù Note that you can edit an issue‚Äôs title after it‚Äôs been posted.</li>
</ul></li>
<li>Format your questions nicely using markdown and code formatting. Preview your issue prior to posting.</li>
<li>As I explained above, your peers and I will more sympathetic to your cause if you can show <em>all the things you have tried as you, yourself, tried to fix the issue first</em>.</li>
<li>Include code and example data so the person trying to help you have something to work with (and which results in the error, perhaps)</li>
<li>Where appropriate, provide links to specific files, or even lines within them, in the body of your issue. This will help your peers understand your question. Note that only the teaching team will have access to private repos.</li>
<li>(Optional) Tag someone or some group of people. Start by typing their GitHub username prefixed with the @ symbol. Of course this supposes that each of you have a GitHub account and username.</li>
<li>Hit <strong>Submit new issue</strong> when you‚Äôre ready to post.</li>
</ul>
<!-- ## Function list

  You will encounter the following functions for the first time in this module:

  | Function              | Description                                        |
  |:----------------------|----------------------------------------------------|
  | `adonis2()`           | Permutational Multivariate Analysis of Variance Using Distance Matrices |
  | `anova.betadisper()`  | PERMDIST 2 for Multivariate homogeneity of groups dispersions (variances) |
  | `anova.cca ()`        | Permutation Test for CCA, RDA and CA of Principal Coordinates |
  | `anova.manyglm()`     | Analysis of Deviance for Multivariate Generalized Linear Model Fits for Abundance Data |
  | `beta.pair()`         | Incidence-based Pair-wise Dissimilarities |
  | `betadisper()`        | Multivariate homogeneity of groups dispersions (variances) |
  | `betapart.core()`     | Core Calculations of betapart |
  | `biplot()`            | Biplot of Multivariate Data |
  | `capscale()`          | [Partial] Distance-based Redundancy Analysis |
  | `cca()`               | [Partial] [Constrained] Correspondence Analysis and Redundancy Analysis |
  | `cleanplot.pc()`      | Numerical Ecology in R's PCA Plot |
  | `daisy()`             | Dissimilarity Matrix Calculation |
  | `decostand()`         | Standardization Methods for Community Ecology |
  | `diag()`              | Matrix Diagonals |
  | `diversity()`         | Ecological Diversity Indices |
  | `diversityresult()`   | Alternative Diversity Results |
  | `fviz_cluster()`      | Visualize Clustering Results |
  | `fviz_nbclust()`      | Determining and Visualizing the Optimal Number of Clusters |
  | `fviz_silhouette()`   | Visualize Silhouette Information from Clustering |
  | `geodist()`           | Convert Lon-Lat Coordinates into Vector or Matrix of Geodesic Distances in Metres |
  | `ggcorrplot()`        | Visualization of a Correlation Matrix Using ggplot2 |
  | `goodness()`          | Diagnostic Tools for [Constrained] Ordination (CCA, RDA, DCA, CA, PCA) |
  | `imputePCA()`         | Impute Dataset with PCA |
  | `lowerTriangle()`     | Extract or Replace the Upper/Lower Triangular Portion of a Matrix |
  | `manyglm()`           | Fitting Generalized Linear Models for Multivariate Abundance Data |
  | `meanvar.plot()`      | Construct Mean-Variance Plots for Multivariate Abundance Data |
  | `metaMDS()`           | Nonmetric Multidimensional Scaling with Stable Solution from Random Starts, Axis Scaling and Species Scores |
  | `mvabund()`           | Multivariate Abundance Data Objects |
  | `ordiellipse()`       | Display Groups or Factor Levels in Ordination Diagrams |
  | `ordiplot()`          | Alternative Plot and Identify Functions for Ordination |
  | `ordispider()`        | Display Groups or Factor Levels in Ordination Diagrams |
  | `ordisurf()`          | Fit and Plot Smooth Surfaces of Variables on Ordination |
  | `pam()`               | Partitioning Around Medoids |
  | `permutest()`         | Permutation Test for CCA, RDA and CA of Principal Coordinates |
  | `rda()`               | [Partial] [Constrained] Correspondence Analysis and Redundancy Analysis |
  | `RsquareAdj()`        | Adjusted R-square |
  | `scores()`            | Get Species or Site Scores from an Ordination |
  | `specnumber()`        | Ecological Diversity Indices |
  | `stressplot()`        | Goodness of Fit and Shepard Plot for Nonmetric Multidimensional Scaling |
  | `t()`                 | Matrix Transpose |
  | `vegdist()`           | Dissimilarity Indices for Community Ecologists |
  | `vif.cca()`           | Diagnostic Tools for [Constrained] Ordination (CCA, RDA, DCA, CA, PCA) | -->
</section>
</section>
<section id="references" class="level1">
<h1>References</h1>
<div id="refs" role="list">

</div>


<!-- -->

</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div id="quarto-reuse" class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a></div></div></section><section class="quarto-appendix-contents"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{smit2022,
  author = {Smit, AJ},
  title = {BCB743: {Quantitative} {Ecology}},
  date = {2022-08-08},
  url = {https://tangledbank.netlify.app/BCB743/BCB743_index.html},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-smit2022" class="csl-entry quarto-appendix-citeas" role="listitem">
Smit A (2022) BCB743: Quantitative Ecology. <a href="https://tangledbank.netlify.app/BCB743/BCB743_index.html">https://tangledbank.netlify.app/BCB743/BCB743_index.html.
</a>
</div></div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "Óßã";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/tangledbank\.netlify\.app");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
          // default icon
          link.classList.add("external");
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    const typesetMath = (el) => {
      if (window.MathJax) {
        // MathJax Typeset
        window.MathJax.typeset([el]);
      } else if (window.katex) {
        // KaTeX Render
        var mathElements = el.getElementsByClassName("math");
        var macros = [];
        for (var i = 0; i < mathElements.length; i++) {
          var texText = mathElements[i].firstChild;
          if (mathElements[i].tagName == "SPAN") {
            window.katex.render(texText.data, mathElements[i], {
              displayMode: mathElements[i].classList.contains('display'),
              throwOnError: false,
              macros: macros,
              fleqn: false
            });
          }
        }
      }
    }
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        typesetMath(container);
        return container.innerHTML
      } else {
        typesetMath(note);
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      typesetMath(note);
      return note.innerHTML;
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<script src="https://utteranc.es/client.js" repo="ajsmit/tangled_bank" issue-term="pathname" theme="github-light" crossorigin="anonymous" async="">
</script>
<nav class="page-navigation column-body">
  <div class="nav-page nav-page-previous">
  </div>
  <div class="nav-page nav-page-next">
      <a href="../BCB743/00-review.html" class="pagination-link" aria-label="1-4. Review Biogeography and Global Ecology">
        <span class="nav-page-text">1-4. Review Biogeography and Global Ecology</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb1" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2022-08-08"</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "BCB743: Quantitative Ecology"</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> The Biodiversity and Conservation Biology R Pages serving BCB743.</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; *"We have become, by the power of a glorious evolutionary accident called intelligence, the stewards of life's continuity on earth. We did not ask for this role, but we cannot abjure it. We may not be suited to it, but here we are."*</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; --- Stephen J. Gould</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; *"Reports that say that something hasn't happened are always interesting to me, because as we know, there are known knowns; there are things we know we know. We also know there are known unknowns; that is to say we know there are some things we do not know. But there are also unknown unknowns- the ones we don't know we don't know."*</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; --- Donald Rumsfeld</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="al">![](../images/PhD_comics3.JPG)</span>{#comic fig-align="center"}</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>**Welcome to the pages for BCB743 Quantitative Ecology. This page provides the syllabus and teaching policies for the module, and it serves is a starting point accessing all the theory, instruction, and data.**</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu"># Honours Coordinator</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>Prof. Bryan Maritz---Room 4.105, Department of Biodiversity &amp; Conservation Biology</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu"># Module Coordinator</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>Prof. AJ Smit---Room 4.103, Department of Biodiversity &amp; Conservation Biology, ajsmit@uwc.ac.za</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="fu"># Instructors</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>Jesse Phillips---Department of Biodiversity &amp; Conservation Biology, jessephillips11.jp@gmail.com</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>Tom Spencer-Hicken---Department of Biodiversity &amp; Conservation Biology, 3941863@myuwc.ac.za</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="fu"># Module description</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>Quantitative ecology is a field within the ecological sciences that uses statistical and computational approaches to understand ecosystems. It aims to describe and quantify ecological processes, analyse and model complex multivariate ecological data, and make predictions about the structure and dynamics of ecosystems across a range of spatial and temporal scales. The multivariate statistical approaches taught in this module will prepare the student to make sense of bewildering multidimensional datasets in a comprehensible two- or three-dimensional space.</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>Coursework on quantitative ecology can cover topics such as:</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Ecological structure** We will cover the basic principles underlying the environmental structuring of ecosystems (ecosystem structure).</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Ecological data analysis** This involves studying statistical techniques to analyse ecological data, including hypothesis testing, regression analysis, and multivariate analysis.</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Multivariate analyses** In this section, we learn how to use multivariate statistics to make sense of complex systems, predict ecological outcomes, and understand the underlying mechanisms that drive ecological processes.</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Spatial ecology** Here we learn how to analyse and model spatial patterns in ecological data, including the distribution of species and habitats across landscapes.</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Population dynamics** This is the theory about how to model and analyse the dynamics of populations, including birth and death rates, population growth, and interactions between species.</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Community ecology** Theory in this section of the work deals with analysing and modelling the interactions between species within ecological communities, including competition, predation, and mutualism.</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Ecosystem ecology** Here we will learn how to model and analyse the flow of energy and nutrients through ecosystems, including the role of producers, consumers, and decomposers in ecological processes.</span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>So, a course in quantitative ecology provides you with the skills and tools needed to analyse and model ecological data, communities, and make predictions about the structure and dynamics of ecosystems. It prepares students for careers in research, conservation, natural resource management, and environmental consulting.</span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a>In BCB743 I will focus on **multivariate statistics**. Multivariate methods play a central role in quantitative ecology as they enable researchers to analyse and interpret complex data sets that involve multiple variables and hordes of species. Ecological systems are characterised by their complexity, and there are typically many different factors that influence the distribution and abundance of species in an ecosystem. Multivariate statistical methods allow us to identify the underlying patterns and relationships among these variables and to explore how they interact to shape ecological communities.</span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a>Some of the most commonly used multivariate statistical techniques in quantitative ecology include ordination methods, such as principal component analysis (PCA) and correspondence analysis (CA), which are used to reduce complexity and permit the visualisation of patterns of species distribution in ecological communities. Cluster analysis, discriminant analysis, and canonical correlation analysis are other multivariate methods commonly used to analyse ecological data. These techniques can help us to identify groups of species or environmental factors that are associated with particular ecological conditions, as well as to assess the relative importance of different variables in shaping ecological communities.</span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a>In addition to providing a framework for data analysis, multivariate statistics also play an important role in developing ecological models. By incorporating multivariate data into models, we can gain a more complete understanding of the ecological processes that govern the behaviour of species and ecosystems. This, in turn, can inform management and conservation efforts aimed at preserving ecological diversity and resilience in the face of environmental change. Multivariate statistics are an essential tool for ecologists seeking to understand the complex relationships between species and their environment.</span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a><span class="fu"># Module content and framework</span></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a>These links point to online resources such as datasets and R scripts in</span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a>support of the video and PDF lecture material. It is essential that you</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>work through these examples and workflows. Here is the Syllabus:</span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>:::{.column-page-inset-right}</span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a>| Week | Class date | Lecture  | Topic                            | Class                                                  | Slides                                                                         | Reading                                                                                                                                                        | Tasks                                                                       | Tasks due  |</span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a>|------|------------|----------|----------------------------------|--------------------------------------------------------|--------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------|------------|</span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a>| W1   | 20 June    | L1       | Introduction                     |                                                        | <span class="co">[</span><span class="ot">üò≥</span><span class="co">](../slides/BCB743_01_intro.pdf)</span>                  | <span class="co">[</span><span class="ot">ü•¥</span><span class="co">](../docs/BES-Reproducible-Code-2017.pdf)</span> <span class="co">[</span><span class="ot">ü•¥</span><span class="co">](../docs/BES-Data-Guide-2017_web.pdf)</span>                                       | <span class="co">[</span><span class="ot">üò±</span><span class="co">](assessments/Task_A.qmd)</span>                       | 27 June    |</span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a>|      |            |          | **BIODIVERSITY**                 |                                                        |                                                                                |                                                                                                                                                                |                                                                             |            |</span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a>|      |            | revision | Ecological Data                  | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](../BDC334/01-introduction.qmd)</span>                    |                                                                                |                                                                                                                                                                | Lab 1 Review                                                                |            | </span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a>|      |            |          | **DATA, MATRICES**               |                                                        |                                                                                |                                                                                                                                                                |                                                                             |            |</span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a>|      |            | revision | Environmental Distance           | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](../BDC334/02b-env_dist.qmd)</span>                        | <span class="co">[</span><span class="ot">üò≥</span><span class="co">](../slides/BCB743_04_environmental_distance.pdf)</span> |                                                                                                                                                                | Lab 2 Review                                                                |            | </span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a>|      |            | revision | Quantifying Biodiversity         | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](../BDC334/03-biodiversity1.qmd)</span>                    | <span class="co">[</span><span class="ot">üò≥</span><span class="co">](../slides/BCB743_02_biodiversity_1.pdf)</span>         | <span class="co">[</span><span class="ot">ü•¥</span><span class="co">](../docs/Smit_et_al_2013.pdf)</span> <span class="co">[</span><span class="ot">ü•¥</span><span class="co">](../docs/Smit_et_al_2017.pdf)</span> <span class="co">[</span><span class="ot">ü•¥</span><span class="co">](../docs/Smit_the_seaweed_data.pdf)</span> | Lab 3 Review                                                                |            |</span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a>|      |            | revision | Describing Biodiversity Patterns | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](../BDC334/04-biodiversity2.qmd)</span>                    | <span class="co">[</span><span class="ot">üò≥</span><span class="co">](../slides/BCB743_02_biodiversity_2.pdf)</span>         | <span class="co">[</span><span class="ot">ü•¥</span><span class="co">](../docs/Shade_et_al_2018.pdf)</span>                                                                                                            | Lab 4 Review                                                                |            |</span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a>|      |            | revision | Ecological Theories              |                                                        |                                                                                |                                                                                                                                                                |                                                                             |            | </span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a>| W1   | 22 June    | L2       | Correlations and Associations    | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](05-correlations.qmd)</span>                               | <span class="co">[</span><span class="ot">üò≥</span><span class="co">](../slides/BCB743_06_correlations.pdf)</span>           |                                                                                                                                                                | <span class="co">[</span><span class="ot">üò±</span><span class="co">](assessments/Task_B.qmd)</span>                       | 23 June    |</span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a>|      |            | self     | Deeper Into Gradients            | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](06-deep_dive.qmd)</span>                                  |                                                                                | <span class="co">[</span><span class="ot">ü•¥</span><span class="co">](../docs/Smit_et_al_2013.pdf)</span> <span class="co">[</span><span class="ot">ü•¥</span><span class="co">](../docs/Smit_et_al_2017.pdf)</span> <span class="co">[</span><span class="ot">ü•¥</span><span class="co">](../docs/Smit_the_seaweed_data.pdf)</span> |                                                                             |            |</span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a>|      |            |          | **ORDINATION**                   |                                                        |                                                                                |                                                                                                                                                                |                                                                             |            | </span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a>| W2   | 27 June    | L3       | Ordination                       |                                                        | <span class="co">[</span><span class="ot">üò≥</span><span class="co">](../slides/BCB743_07_ordination.pdf)</span>             |                                                                                                                                                                |                                                                             |            |</span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a>|      |            | L3       | PCA (a)                          | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](08-PCA.qmd)</span>                                        | <span class="co">[</span><span class="ot">üò≥</span><span class="co">](../slides/BCB743_08_PCA.pdf)</span>                    |                                                                                                                                                                | <span class="co">[</span><span class="ot">üò±</span><span class="co">](assessments/Task_C.qmd)</span>                       | 28 June    |</span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a>|      |            | self     | PCA (b) Additional Examples      | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](08-PCA_examples.qmd)</span>                               |                                                                                |                                                                                                                                                                |                                                                             |            | </span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a>|      |            | self     | PCA (c) WHO SDGs                 | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](08-PCA_SDG.qmd)</span>                                                       |                                                                                |                                                                                                                                                                | <span class="co">[</span><span class="ot">üò±</span><span class="co">](assessments/Task_D.qmd)</span>                       | 19 July    | </span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a>| W3   | 4 July     | L4       | CA                               | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](09-CA.qmd)</span>                                         | <span class="co">[</span><span class="ot">üò≥</span><span class="co">](../slides/BCB743_09_CA.pdf)</span>                     |                                                                                                                                                                | <span class="co">[</span><span class="ot">üò±</span><span class="co">](assessments/Task_E.qmd)</span>                       | 5 July     |</span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a>|      | 6 July     | L5       | PCoA                             | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](10-PCoA.qmd)</span>                                       | <span class="co">[</span><span class="ot">üò≥</span><span class="co">](../slides/BCB743_10_PCoA.pdf)</span>                   |                                                                                                                                                                | <span class="co">[</span><span class="ot">üò±</span><span class="co">](assessments/BCB743_intgrative_assignment.qmd)</span> | 7 July     |</span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a>| W4   | 11 July    | L6       | nMDS                             | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](11-nMDS.qmd)</span>                                       | <span class="co">[</span><span class="ot">üò≥</span><span class="co">](../slides/BCB743_11_nMDS.pdf)</span>                   |                                                                                                                                                                | <span class="co">[</span><span class="ot">üò±</span><span class="co">](assessments/Task_F.qmd)</span>                       | 12 July    |</span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a>|      |            | self     | nMDS + PERMANOVA (diatoms)       | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](11-nMDS_diatoms.qmd)</span>                               |                                                                                | <span class="co">[</span><span class="ot">ü•¥</span><span class="co">](../docs/Mayombo_et_al_2019.pdf)</span>                                                                                                          |                                                                             |            | </span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a>|      | 13 July    | L7       | CCA &amp; RDA                        | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](12-constrained_ordination.qmd)</span>                     | <span class="co">[</span><span class="ot">üò≥</span><span class="co">](../slides/BCB743_12_constrained_ordination.pdf)</span> | <span class="co">[</span><span class="ot">ü•¥</span><span class="co">](../docs/Smit_et_al_2017.pdf)</span> <span class="co">[</span><span class="ot">ü•¥</span><span class="co">](../docs/Smit_the_seaweed_data.pdf)</span>                                                    |                                                                             |            |</span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a>|      |            |          | **CLUSTER ANALYSIS**             |                                                        |                                                                                |                                                                                                                                                                |                                                                             |            | </span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>| W5   | 18 July    | L8       | Cluster Analysis                 | <span class="co">[</span><span class="ot">üò≠</span><span class="co">](13-cluster_analysis.qmd)</span>                          |                                                                                |                                                                                                                                                                | <span class="co">[</span><span class="ot">üò±</span><span class="co">](assessments/Task_D.qmd)</span>                       | 19 July    |</span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a>|      |            | L9       | Review &amp; Integrative Assignment  |                                                        |                                                                                |                                                                                                                                                                | <span class="co">[</span><span class="ot">üò±</span><span class="co">](assessments/BCB743_intgrative_assignment.qmd)</span> | 2 August   |</span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a>| W7   | 3--4 Aug   |          | **EXAM**                         |                                                        |                                                                                |                                                                                                                                                                |                                                                             | 4 August   |                                           </span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a>**Core theoretical framework** Ecological hypotheses underlying the</span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a>processes of species assembly in space and time, including neutral and</span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a>niche-based mechanisms, and historical events; overview of the currently</span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a>known and understood distributional patterns of major groups of</span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a>organisms at global, regional and local scales; consideration of</span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a>sampling designs aimed at capturing these patterns and drivers so as to</span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a>arrive at a processed based understanding of species assembly.</span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a>**Competence** Data collection aimed at a quantitative test of the</span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a>relevant hypotheses, above. The management and analysis of ecological</span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a>data; reproducible and collaborative research; the use of R as a tool</span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a>for the analysis multivariate ecological data; multivariate techniques</span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a>such as nMDS, PCA, RDA and cluster analysis; graphical data summaries</span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a>and visualisations.</span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a><span class="fu"># Outcomes of BCB743</span></span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a>By the end of this module, students will be able to:</span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Understand the concepts of alpha, beta and gamma diversity</span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>know and understand the current hypotheses that attempt to explain species assembly processes in space and time (e.g. neutral and niche mechanisms)</span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>collect ecological data at the appropriate scale, which would lend themselves to a quantitative analysis of points 1 and 2, above</span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>use the R software and associated packages to undertake the analyses required in point 3, above</span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>interpret the outcomes of the above analyses and use it to quantitatively characterise points 1 and 2, above</span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>communicate the findings by written and oral means</span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a><span class="fu"># Graduate attributes</span></span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a>The <span class="co">[</span><span class="ot">**graduate attributes**</span><span class="co">](../pages/graduate_attributes.qmd)</span> resulting from completion of this modules alignment with the expectations of the workspace across diverse organisations and institutions where graduates typically find employment.</span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data and reading in support of the syllabus</span></span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a>In the table above there are links to several key papers to read in preparation of each week's theory. It is essential that you read these papers.</span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a>Many other references are cited in each Chapter. These serve several functions in that they:</span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Add additional theory relevant to some ecological concepts</span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Provide background to some of the datasets used in my examples</span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Discuss derivations of some equations used to calculate diversity concepts</span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Provide example walkthroughs of some of the computational aspects of the methods covered in the Labs</span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Collectively supplement the discussion about these concepts covered in the lectures</span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a>Actively engaging with these reading materials will make to difference between a 60% average mark for the module, and a mark in excess of 80%.</span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a><span class="fu">## Reading</span></span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a>You are **expected to read additional material** in support of the</span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a>content covered in class and on this website.</span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a>A **compulsory** reference is [Numerical Ecology with</span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a>R](http://adn.biol.umontreal.ca/~numericalecology/numecolR/) by Daniel</span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a>Borcard, Fran√ßois Gillet and Pierre Legendre. Much of the class' content</span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a>and many of the examples (and code) that I use have been adapted from</span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a>this source. It should be considered the 'gold standard' reference for</span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a>Quantitative Ecology.</span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-151"><a href="#cb1-151" aria-hidden="true" tabindex="-1"></a>A second **highly recommended text** is the book [Tree Diversity</span>
<span id="cb1-152"><a href="#cb1-152" aria-hidden="true" tabindex="-1"></a>Analysis](http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf)</span>
<span id="cb1-153"><a href="#cb1-153" aria-hidden="true" tabindex="-1"></a>by Roeland Kindt and Richard Coe.</span>
<span id="cb1-154"><a href="#cb1-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-155"><a href="#cb1-155" aria-hidden="true" tabindex="-1"></a>I can also recommend a few websites with excellent content, such as:</span>
<span id="cb1-156"><a href="#cb1-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-157"><a href="#cb1-157" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>David Zelen√Ω's [Analysis of Community Ecology Data in</span>
<span id="cb1-158"><a href="#cb1-158" aria-hidden="true" tabindex="-1"></a>    R](https://www.davidzeleny.net/anadat-r/doku.php)</span>
<span id="cb1-159"><a href="#cb1-159" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Mike Palmer's [Ordination Methods for</span>
<span id="cb1-160"><a href="#cb1-160" aria-hidden="true" tabindex="-1"></a>    Ecologists](http://ordination.okstate.edu/)</span>
<span id="cb1-161"><a href="#cb1-161" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>[GUide to STatistical Analysis in Microbial Ecology (GUSTA</span>
<span id="cb1-162"><a href="#cb1-162" aria-hidden="true" tabindex="-1"></a>    ME)](https://sites.google.com/site/mb3gustame/)</span>
<span id="cb1-163"><a href="#cb1-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-164"><a href="#cb1-164" aria-hidden="true" tabindex="-1"></a>Note that the URLs with links to additional reading that appear with the</span>
<span id="cb1-165"><a href="#cb1-165" aria-hidden="true" tabindex="-1"></a>worked-through example code should **not be seen as optional**. They are</span>
<span id="cb1-166"><a href="#cb1-166" aria-hidden="true" tabindex="-1"></a>there for a reason and should be consulted even though I might not</span>
<span id="cb1-167"><a href="#cb1-167" aria-hidden="true" tabindex="-1"></a>necessarily refer to each of them in class.</span>
<span id="cb1-168"><a href="#cb1-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-169"><a href="#cb1-169" aria-hidden="true" tabindex="-1"></a>Use these materials liberally.</span>
<span id="cb1-170"><a href="#cb1-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-171"><a href="#cb1-171" aria-hidden="true" tabindex="-1"></a>Should you want to download the source code for the BCB743 (and BCB744</span>
<span id="cb1-172"><a href="#cb1-172" aria-hidden="true" tabindex="-1"></a>website), you may find it on [&lt;i class="fab fa-github"&gt;&lt;/i&gt;</span>
<span id="cb1-173"><a href="#cb1-173" aria-hidden="true" tabindex="-1"></a>GitHub](https://github.com/ajsmit/R_courses).</span>
<span id="cb1-174"><a href="#cb1-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-175"><a href="#cb1-175" aria-hidden="true" tabindex="-1"></a><span class="fu">## Datasets used in this module</span></span>
<span id="cb1-176"><a href="#cb1-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-177"><a href="#cb1-177" aria-hidden="true" tabindex="-1"></a>Note that the links provided might not necessarily lead to the **vegan**</span>
<span id="cb1-178"><a href="#cb1-178" aria-hidden="true" tabindex="-1"></a>help page.</span>
<span id="cb1-179"><a href="#cb1-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-180"><a href="#cb1-180" aria-hidden="true" tabindex="-1"></a>|   | Dataset                                                      | Source                                                                                        |</span>
<span id="cb1-181"><a href="#cb1-181" aria-hidden="true" tabindex="-1"></a>|:--|--------------------------------------------------------------|-----------------------------------------------------------------------------------------------|</span>
<span id="cb1-182"><a href="#cb1-182" aria-hidden="true" tabindex="-1"></a>| 1 | Vegetation and Environment in Dutch Dune Meadows             | <span class="co">[</span><span class="ot">**vegan**</span><span class="co">](https://www.davidzeleny.net/anadat-r/doku.php/en:data:dune)</span>                       |</span>
<span id="cb1-183"><a href="#cb1-183" aria-hidden="true" tabindex="-1"></a>| 2 | Oribatid Mite Data with Explanatory Variables                | <span class="co">[</span><span class="ot">**vegan**</span><span class="co">](http://adn.biol.umontreal.ca/~numericalecology/data/oribates.html)</span>                |</span>
<span id="cb1-184"><a href="#cb1-184" aria-hidden="true" tabindex="-1"></a>| 3 | The Doubs River Data                                         | <span class="co">[</span><span class="ot">**Numerical Ecology with R**</span><span class="co">](https://www.davidzeleny.net/anadat-r/doku.php/en:data:doubs)</span>   |</span>
<span id="cb1-185"><a href="#cb1-185" aria-hidden="true" tabindex="-1"></a>| 4 | The Barro Colorado Island Tree Counts                        | <span class="co">[</span><span class="ot">**vegan**</span><span class="co">](https://www.davidzeleny.net/anadat-r/doku.php/en:data:bci)</span>                        |</span>
<span id="cb1-186"><a href="#cb1-186" aria-hidden="true" tabindex="-1"></a>| 5 | John Bolton, Rob Anderson, and Herre Stegenga's Seaweed Data | <span class="co">[</span><span class="ot">**Smit et al., 2017**</span><span class="co">](https://www.frontiersin.org/articles/10.3389/fmars.2017.00404/)</span>       |</span>
<span id="cb1-187"><a href="#cb1-187" aria-hidden="true" tabindex="-1"></a>| 6 | Serge Mayombo's Diatoms Data                                 | <span class="co">[</span><span class="ot">**Mayombo et al., 2019**</span><span class="co">](https://www.tandfonline.com/doi/abs/10.2989/1814232X.2019.1592778)</span> |</span>
<span id="cb1-188"><a href="#cb1-188" aria-hidden="true" tabindex="-1"></a>| 7 | World Health Organization Sustainable Development Goals Data | <span class="co">[</span><span class="ot">**WHO**</span><span class="co">](https://www.who.int/data)</span>                                                           |</span>
<span id="cb1-189"><a href="#cb1-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-190"><a href="#cb1-190" aria-hidden="true" tabindex="-1"></a><span class="fu"># Prerequisites</span></span>
<span id="cb1-191"><a href="#cb1-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-192"><a href="#cb1-192" aria-hidden="true" tabindex="-1"></a>You should have a moderate numerical literacy and have prior programming</span>
<span id="cb1-193"><a href="#cb1-193" aria-hidden="true" tabindex="-1"></a>experience. Such experience will have been obtained in the</span>
<span id="cb1-194"><a href="#cb1-194" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">BCB744</span><span class="co">](/workshops/)</span> module, which is a module about doing statistics</span>
<span id="cb1-195"><a href="#cb1-195" aria-hidden="true" tabindex="-1"></a>in R. If you have a reasonable experience in coding and statistical</span>
<span id="cb1-196"><a href="#cb1-196" aria-hidden="true" tabindex="-1"></a>analysis you should find yourself well prepared. You should also thoroughly revise </span>
<span id="cb1-197"><a href="#cb1-197" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">BDC334</span><span class="co">](../BDC334/BDC334_index.qmd)</span> by the end of the first week of this module.</span>
<span id="cb1-198"><a href="#cb1-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-199"><a href="#cb1-199" aria-hidden="true" tabindex="-1"></a><span class="fu"># Method of instruction</span></span>
<span id="cb1-200"><a href="#cb1-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-201"><a href="#cb1-201" aria-hidden="true" tabindex="-1"></a>You are provided with reading material (lecture slides, code, website</span>
<span id="cb1-202"><a href="#cb1-202" aria-hidden="true" tabindex="-1"></a>content) that you are expected to consume **prior to the class**.</span>
<span id="cb1-203"><a href="#cb1-203" aria-hidden="true" tabindex="-1"></a>Classes will involve brief introductions to new concepts, and will be</span>
<span id="cb1-204"><a href="#cb1-204" aria-hidden="true" tabindex="-1"></a>followed by working on exercises in class that cover those concepts. The</span>
<span id="cb1-205"><a href="#cb1-205" aria-hidden="true" tabindex="-1"></a>workshop is designed to be as interactive as possible, so while you are</span>
<span id="cb1-206"><a href="#cb1-206" aria-hidden="true" tabindex="-1"></a>working on exercises the tutor and I will circulate among you and engage</span>
<span id="cb1-207"><a href="#cb1-207" aria-hidden="true" tabindex="-1"></a>with you to help you understand any material and the associated code you</span>
<span id="cb1-208"><a href="#cb1-208" aria-hidden="true" tabindex="-1"></a>are uncomfortable with. Often this will result in discussions of novel</span>
<span id="cb1-209"><a href="#cb1-209" aria-hidden="true" tabindex="-1"></a>applications and alternative approaches to the data analysis challenges</span>
<span id="cb1-210"><a href="#cb1-210" aria-hidden="true" tabindex="-1"></a>you are required to solve. More challenging concepts might emerge during</span>
<span id="cb1-211"><a href="#cb1-211" aria-hidden="true" tabindex="-1"></a>the assignments (typically these will be submitted the following day),</span>
<span id="cb1-212"><a href="#cb1-212" aria-hidden="true" tabindex="-1"></a>and any such challenges will be dealt with in class prior to learning</span>
<span id="cb1-213"><a href="#cb1-213" aria-hidden="true" tabindex="-1"></a>new concepts.</span>
<span id="cb1-214"><a href="#cb1-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-215"><a href="#cb1-215" aria-hidden="true" tabindex="-1"></a>Although the module is theory-heavy, a large part of it is also about</span>
<span id="cb1-216"><a href="#cb1-216" aria-hidden="true" tabindex="-1"></a>coding. It is up to you to take your coding skills to the next level and</span>
<span id="cb1-217"><a href="#cb1-217" aria-hidden="true" tabindex="-1"></a>move beyond what I teach in class. Coding is a bit like learning a</span>
<span id="cb1-218"><a href="#cb1-218" aria-hidden="true" tabindex="-1"></a>language, and as such programming is a skill that is best learned by</span>
<span id="cb1-219"><a href="#cb1-219" aria-hidden="true" tabindex="-1"></a>doing.</span>
<span id="cb1-220"><a href="#cb1-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-221"><a href="#cb1-221" aria-hidden="true" tabindex="-1"></a><span class="fu"># Learning colaboratively</span></span>
<span id="cb1-222"><a href="#cb1-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-223"><a href="#cb1-223" aria-hidden="true" tabindex="-1"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb1-224"><a href="#cb1-224" aria-hidden="true" tabindex="-1"></a><span class="fu">## Also read: How to learn</span></span>
<span id="cb1-225"><a href="#cb1-225" aria-hidden="true" tabindex="-1"></a>Please refer to my <span class="co">[</span><span class="ot">advice about how to learn</span><span class="co">](../pages/How_to_learn.qmd)</span>.</span>
<span id="cb1-226"><a href="#cb1-226" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb1-227"><a href="#cb1-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-228"><a href="#cb1-228" aria-hidden="true" tabindex="-1"></a>Discuss the BCB743 workshop activities with your peers as you work on</span>
<span id="cb1-229"><a href="#cb1-229" aria-hidden="true" tabindex="-1"></a>them. Use also the WhatsApp group set up for the module for discussion</span>
<span id="cb1-230"><a href="#cb1-230" aria-hidden="true" tabindex="-1"></a>purposes (I might assist via this medium if neccesary if your</span>
<span id="cb1-231"><a href="#cb1-231" aria-hidden="true" tabindex="-1"></a>questions/comments have relevance to the whole class). A better option</span>
<span id="cb1-232"><a href="#cb1-232" aria-hidden="true" tabindex="-1"></a>is to use [GitHub</span>
<span id="cb1-233"><a href="#cb1-233" aria-hidden="true" tabindex="-1"></a>Issues](/quantecol/#help-via-bcb744-and-bcb743-issues-on-github). You</span>
<span id="cb1-234"><a href="#cb1-234" aria-hidden="true" tabindex="-1"></a>will learn more in this module if you work with your friends than if you</span>
<span id="cb1-235"><a href="#cb1-235" aria-hidden="true" tabindex="-1"></a>do not. Ask questions, answer questions, and share ideas liberally.</span>
<span id="cb1-236"><a href="#cb1-236" aria-hidden="true" tabindex="-1"></a>Please identify your work partners by name on all assignments (if you</span>
<span id="cb1-237"><a href="#cb1-237" aria-hidden="true" tabindex="-1"></a>decide to work in pairs).</span>
<span id="cb1-238"><a href="#cb1-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-239"><a href="#cb1-239" aria-hidden="true" tabindex="-1"></a>**Cooperative learning is not a licence for plagiarism. Plagiarism is a</span>
<span id="cb1-240"><a href="#cb1-240" aria-hidden="true" tabindex="-1"></a>serious offence and will be dealt with concisely. Consequences of</span>
<span id="cb1-241"><a href="#cb1-241" aria-hidden="true" tabindex="-1"></a>cheating are severe---they range from a 0% for the assignment or exam up</span>
<span id="cb1-242"><a href="#cb1-242" aria-hidden="true" tabindex="-1"></a>to dismissal from the course for a second offense.**</span>
<span id="cb1-243"><a href="#cb1-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-244"><a href="#cb1-244" aria-hidden="true" tabindex="-1"></a><span class="fu"># Reusing code found elsewhere</span></span>
<span id="cb1-245"><a href="#cb1-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-246"><a href="#cb1-246" aria-hidden="true" tabindex="-1"></a>A huge volume of code is available on the web and it can be adapted to</span>
<span id="cb1-247"><a href="#cb1-247" aria-hidden="true" tabindex="-1"></a>solve your own problems. You may make use of any online resources (e.g.</span>
<span id="cb1-248"><a href="#cb1-248" aria-hidden="true" tabindex="-1"></a>form <span class="co">[</span><span class="ot">StackOverflow</span><span class="co">](https://stackoverflow.com/)</span>, a thoroughly-used</span>
<span id="cb1-249"><a href="#cb1-249" aria-hidden="true" tabindex="-1"></a>source of discussion about [R</span>
<span id="cb1-250"><a href="#cb1-250" aria-hidden="true" tabindex="-1"></a>code](https://stackoverflow.com/questions/tagged/r))---but you **MUST**</span>
<span id="cb1-251"><a href="#cb1-251" aria-hidden="true" tabindex="-1"></a>clearly indicate (cite) that your solution relies on found code,</span>
<span id="cb1-252"><a href="#cb1-252" aria-hidden="true" tabindex="-1"></a>regardless to what extent you have modified it to your own needs. Reused</span>
<span id="cb1-253"><a href="#cb1-253" aria-hidden="true" tabindex="-1"></a>code that is discovered via a web search and which is not explicitly</span>
<span id="cb1-254"><a href="#cb1-254" aria-hidden="true" tabindex="-1"></a>cited is plagiarism and it will be treated as such. On assignments you</span>
<span id="cb1-255"><a href="#cb1-255" aria-hidden="true" tabindex="-1"></a>may not directly share code with your peers in this workshop.</span>
<span id="cb1-256"><a href="#cb1-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-257"><a href="#cb1-257" aria-hidden="true" tabindex="-1"></a><span class="fu"># Software</span></span>
<span id="cb1-258"><a href="#cb1-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-259"><a href="#cb1-259" aria-hidden="true" tabindex="-1"></a>In this course we will rely entirely on <span class="co">[</span><span class="ot">R</span><span class="co">](https://cran.r-project.org/)</span></span>
<span id="cb1-260"><a href="#cb1-260" aria-hidden="true" tabindex="-1"></a>running within the <span class="co">[</span><span class="ot">RStudio</span><span class="co">](https://www.rstudio.com/)</span> IDE. The use of R</span>
<span id="cb1-261"><a href="#cb1-261" aria-hidden="true" tabindex="-1"></a>was covered extensively in the</span>
<span id="cb1-262"><a href="#cb1-262" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">BCB744</span><span class="co">](http://localhost:4321/workshops/)</span> module where the</span>
<span id="cb1-263"><a href="#cb1-263" aria-hidden="true" tabindex="-1"></a>[installation</span>
<span id="cb1-264"><a href="#cb1-264" aria-hidden="true" tabindex="-1"></a>process](http://localhost:4321/workshops/intro_r/chapters/02-rstudio/)</span>
<span id="cb1-265"><a href="#cb1-265" aria-hidden="true" tabindex="-1"></a>was discussed. We will primarily use the</span>
<span id="cb1-266"><a href="#cb1-266" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">**vegan**</span><span class="co">](https://cran.r-project.org/web/packages/vegan/index.html)</span></span>
<span id="cb1-267"><a href="#cb1-267" aria-hidden="true" tabindex="-1"></a>package, but some useful functions are also provided by the package</span>
<span id="cb1-268"><a href="#cb1-268" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">**BiodiversityR**</span><span class="co">](https://github.com/cran/BiodiversityR)</span> (and</span>
<span id="cb1-269"><a href="#cb1-269" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">here</span><span class="co">](http://apps.worldagroforestry.org/downloads/Publications/PDFS/b13695.pdf)</span></span>
<span id="cb1-270"><a href="#cb1-270" aria-hidden="true" tabindex="-1"></a>and <span class="co">[</span><span class="ot">here</span><span class="co">](https://rpubs.com/Roeland-KINDT)</span>). Various other R packages</span>
<span id="cb1-271"><a href="#cb1-271" aria-hidden="true" tabindex="-1"></a>offer overlapping and additional methods, but **vegan** should</span>
<span id="cb1-272"><a href="#cb1-272" aria-hidden="true" tabindex="-1"></a>accommodate <span class="sc">\&gt;</span>90% of your Quantitative Ecology needs.</span>
<span id="cb1-273"><a href="#cb1-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-274"><a href="#cb1-274" aria-hidden="true" tabindex="-1"></a><span class="fu"># Computers</span></span>
<span id="cb1-275"><a href="#cb1-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-276"><a href="#cb1-276" aria-hidden="true" tabindex="-1"></a>You are encouraged to provide your own laptops and to install the</span>
<span id="cb1-277"><a href="#cb1-277" aria-hidden="true" tabindex="-1"></a>necessary software before the module starts. Limited support can be</span>
<span id="cb1-278"><a href="#cb1-278" aria-hidden="true" tabindex="-1"></a>provided if required. There are also computers with R and RStudio (and</span>
<span id="cb1-279"><a href="#cb1-279" aria-hidden="true" tabindex="-1"></a>the neccesary add-on libraries) available in the 5th floor lab in the</span>
<span id="cb1-280"><a href="#cb1-280" aria-hidden="true" tabindex="-1"></a>BCB Department.</span>
<span id="cb1-281"><a href="#cb1-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-282"><a href="#cb1-282" aria-hidden="true" tabindex="-1"></a><span class="fu"># Attendance</span></span>
<span id="cb1-283"><a href="#cb1-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-284"><a href="#cb1-284" aria-hidden="true" tabindex="-1"></a>This workhop-based, hands on course can only deliver acceptible outcomes</span>
<span id="cb1-285"><a href="#cb1-285" aria-hidden="true" tabindex="-1"></a>if you attend all classes. The schedule is set and cannot be changed.</span>
<span id="cb1-286"><a href="#cb1-286" aria-hidden="true" tabindex="-1"></a>Sometimes an occasional absence cannot be avoided. Please be curtious</span>
<span id="cb1-287"><a href="#cb1-287" aria-hidden="true" tabindex="-1"></a>and notify myself or the tutor in advance of any absence. If you work</span>
<span id="cb1-288"><a href="#cb1-288" aria-hidden="true" tabindex="-1"></a>with a partner in class, notify them too. Keep up with the reading</span>
<span id="cb1-289"><a href="#cb1-289" aria-hidden="true" tabindex="-1"></a>assignments while you are away and we will all work with you to get you</span>
<span id="cb1-290"><a href="#cb1-290" aria-hidden="true" tabindex="-1"></a>back up to speed on what you miss. If you do miss a class, however, the</span>
<span id="cb1-291"><a href="#cb1-291" aria-hidden="true" tabindex="-1"></a>assignments must still be submitted on time (also see [**Late submission</span>
<span id="cb1-292"><a href="#cb1-292" aria-hidden="true" tabindex="-1"></a>of CA**](/quantecol/#late-submission-of-ca)).</span>
<span id="cb1-293"><a href="#cb1-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-294"><a href="#cb1-294" aria-hidden="true" tabindex="-1"></a>Since you may decide to work in collaboration with a peer on tasks and</span>
<span id="cb1-295"><a href="#cb1-295" aria-hidden="true" tabindex="-1"></a>assignments, please keep this person informed at all times in case some</span>
<span id="cb1-296"><a href="#cb1-296" aria-hidden="true" tabindex="-1"></a>emergency makes you unavailable for a period of time. Someone might</span>
<span id="cb1-297"><a href="#cb1-297" aria-hidden="true" tabindex="-1"></a>depend on your input and contributions---do not leave someone in the</span>
<span id="cb1-298"><a href="#cb1-298" aria-hidden="true" tabindex="-1"></a>lurch so that they cannot complete a task in your absence.</span>
<span id="cb1-299"><a href="#cb1-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-300"><a href="#cb1-300" aria-hidden="true" tabindex="-1"></a><span class="fu"># Assessment policy {#sec-policy}</span></span>
<span id="cb1-301"><a href="#cb1-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-302"><a href="#cb1-302" aria-hidden="true" tabindex="-1"></a>**Continuous Assessments** (CA) and a **Final Assessment** will provide a **Final Mark** for the module. These modes of assessment meet our needs as far as <span class="co">[</span><span class="ot">formative and summative assessments</span><span class="co">](../pages/assessment_theory.qmd)</span> are concerned. The weighting of the CA and the Final Assessment is 0.6 and 0.4,</span>
<span id="cb1-303"><a href="#cb1-303" aria-hidden="true" tabindex="-1"></a>respectively. All assessments are open book, so consult your code and</span>
<span id="cb1-304"><a href="#cb1-304" aria-hidden="true" tabindex="-1"></a>reading material if and when you need to.</span>
<span id="cb1-305"><a href="#cb1-305" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-306"><a href="#cb1-306" aria-hidden="true" tabindex="-1"></a><span class="fu">## Continuous Assessment</span></span>
<span id="cb1-307"><a href="#cb1-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-308"><a href="#cb1-308" aria-hidden="true" tabindex="-1"></a>Quantitative Ecology is about working with real-world datasets. To this</span>
<span id="cb1-309"><a href="#cb1-309" aria-hidden="true" tabindex="-1"></a>end, a series of mini-projects (**Tasks**) involving real data are a</span>
<span id="cb1-310"><a href="#cb1-310" aria-hidden="true" tabindex="-1"></a>required part of the BCB743. You may work alone or with a single partner</span>
<span id="cb1-311"><a href="#cb1-311" aria-hidden="true" tabindex="-1"></a>on all projects (in which case you will share the same mark). All</span>
<span id="cb1-312"><a href="#cb1-312" aria-hidden="true" tabindex="-1"></a>Tasks are due by the date and time indicated, and due care must be</span>
<span id="cb1-313"><a href="#cb1-313" aria-hidden="true" tabindex="-1"></a>taken that they are submitted as instructed, i.e. paying attention to</span>
<span id="cb1-314"><a href="#cb1-314" aria-hidden="true" tabindex="-1"></a>naming conventions and the format of the files submitted---typically</span>
<span id="cb1-315"><a href="#cb1-315" aria-hidden="true" tabindex="-1"></a>this will be in Quarto format (.qmd) and the knitted output (I prefer</span>
<span id="cb1-316"><a href="#cb1-316" aria-hidden="true" tabindex="-1"></a>.html).</span>
<span id="cb1-317"><a href="#cb1-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-318"><a href="#cb1-318" aria-hidden="true" tabindex="-1"></a>The Continuous Assessment is comprised of the material indicated in</span>
<span id="cb1-319"><a href="#cb1-319" aria-hidden="true" tabindex="-1"></a>the Syllabus by the &lt;i class="fa-balance-scale-right"&gt;&lt;/i&gt; icon.</span>
<span id="cb1-320"><a href="#cb1-320" aria-hidden="true" tabindex="-1"></a>There are seven smaller assessments (each weighted 0.1) and one slightly</span>
<span id="cb1-321"><a href="#cb1-321" aria-hidden="true" tabindex="-1"></a>more demanding Integrative Assessment (weighted 0.3) at the end of the</span>
<span id="cb1-322"><a href="#cb1-322" aria-hidden="true" tabindex="-1"></a>module.</span>
<span id="cb1-323"><a href="#cb1-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-324"><a href="#cb1-324" aria-hidden="true" tabindex="-1"></a><span class="fu">### Tasks</span></span>
<span id="cb1-325"><a href="#cb1-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-326"><a href="#cb1-326" aria-hidden="true" tabindex="-1"></a>When assessing the tasks, we will pay attention to the following criteria:</span>
<span id="cb1-327"><a href="#cb1-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-328"><a href="#cb1-328" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Content (10%):</span>
<span id="cb1-329"><a href="#cb1-329" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Questions answered in order</span>
<span id="cb1-330"><a href="#cb1-330" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Annotations (meta-data in file header, commnets about code, ideas, and approach)</span>
<span id="cb1-331"><a href="#cb1-331" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Code formatting and correctness (45%):</span>
<span id="cb1-332"><a href="#cb1-332" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Application of [R code</span>
<span id="cb1-333"><a href="#cb1-333" aria-hidden="true" tabindex="-1"></a>        conventions](http://adv-r.had.co.nz/Style.html), e.g. spaces</span>
<span id="cb1-334"><a href="#cb1-334" aria-hidden="true" tabindex="-1"></a>        around <span class="in">`&lt;-`</span>, after <span class="in">`#`</span>, after <span class="in">`,`</span>, etc.</span>
<span id="cb1-335"><a href="#cb1-335" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>New line for each <span class="in">`dplyr`</span> function (lines end in <span class="in">`%&gt;%`</span>) or</span>
<span id="cb1-336"><a href="#cb1-336" aria-hidden="true" tabindex="-1"></a>        <span class="in">`ggplot`</span> layer (lines end in <span class="in">`+`</span>)</span>
<span id="cb1-337"><a href="#cb1-337" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Figures (45%):</span>
<span id="cb1-338"><a href="#cb1-338" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Sensible use of themes / colours</span>
<span id="cb1-339"><a href="#cb1-339" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Publication quality</span>
<span id="cb1-340"><a href="#cb1-340" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Informative and complete titles, axes labels, legends, etc.</span>
<span id="cb1-341"><a href="#cb1-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-342"><a href="#cb1-342" aria-hidden="true" tabindex="-1"></a><span class="fu">### Integrative Assessment</span></span>
<span id="cb1-343"><a href="#cb1-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-344"><a href="#cb1-344" aria-hidden="true" tabindex="-1"></a>The above-mentioned Assessments will be graded with the following in expectations in mind:</span>
<span id="cb1-345"><a href="#cb1-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-346"><a href="#cb1-346" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Content (20%):</span>
<span id="cb1-347"><a href="#cb1-347" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Questions answered in order</span>
<span id="cb1-348"><a href="#cb1-348" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>A written explanation of approach included for each question</span>
<span id="cb1-349"><a href="#cb1-349" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Appropriate formatting of text, for example, fonts not larger</span>
<span id="cb1-350"><a href="#cb1-350" aria-hidden="true" tabindex="-1"></a>        than necessary, headings used properly, etc. Be sensible and</span>
<span id="cb1-351"><a href="#cb1-351" aria-hidden="true" tabindex="-1"></a>        tasteful.</span>
<span id="cb1-352"><a href="#cb1-352" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Code formatting, structure, and correctness  (50%):</span>
<span id="cb1-353"><a href="#cb1-353" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Use **Tidyverse** code</span>
<span id="cb1-354"><a href="#cb1-354" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>No more than \~80 characters of code per line (pay particular</span>
<span id="cb1-355"><a href="#cb1-355" aria-hidden="true" tabindex="-1"></a>        attention to the comments)</span>
<span id="cb1-356"><a href="#cb1-356" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Application of [R code</span>
<span id="cb1-357"><a href="#cb1-357" aria-hidden="true" tabindex="-1"></a>        conventions](http://adv-r.had.co.nz/Style.html), e.g. spaces</span>
<span id="cb1-358"><a href="#cb1-358" aria-hidden="true" tabindex="-1"></a>        around <span class="in">`&lt;-`</span>, after <span class="in">`#`</span>, after <span class="in">`,`</span>, etc.</span>
<span id="cb1-359"><a href="#cb1-359" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>New line for each <span class="in">`dplyr`</span> function (lines end in <span class="in">`%&gt;%`</span>) or</span>
<span id="cb1-360"><a href="#cb1-360" aria-hidden="true" tabindex="-1"></a>        <span class="in">`ggplot`</span> layer (lines end in <span class="in">`+`</span>)</span>
<span id="cb1-361"><a href="#cb1-361" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Proper indentation of pipes and <span class="in">`ggplot()`</span> layers</span>
<span id="cb1-362"><a href="#cb1-362" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>All chunks labelled without spaces</span>
<span id="cb1-363"><a href="#cb1-363" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>No unwanted / commented out code left behind in the document</span>
<span id="cb1-364"><a href="#cb1-364" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Figures (30%):</span>
<span id="cb1-365"><a href="#cb1-365" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Sensible use of themes / colours</span>
<span id="cb1-366"><a href="#cb1-366" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Publication quality</span>
<span id="cb1-367"><a href="#cb1-367" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Informative and complete titles, axes labels, legends, etc.</span>
<span id="cb1-368"><a href="#cb1-368" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>No redundant features or aesthetics</span>
<span id="cb1-369"><a href="#cb1-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-370"><a href="#cb1-370" aria-hidden="true" tabindex="-1"></a>Random quizzes will not form part of the CA for BCB743.</span>
<span id="cb1-371"><a href="#cb1-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-372"><a href="#cb1-372" aria-hidden="true" tabindex="-1"></a><span class="fu">## Final Assessment (Exam)</span></span>
<span id="cb1-373"><a href="#cb1-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-374"><a href="#cb1-374" aria-hidden="true" tabindex="-1"></a>The Final Assessment will be a 48-hour affair that you can do in the comfort of your</span>
<span id="cb1-375"><a href="#cb1-375" aria-hidden="true" tabindex="-1"></a>home. It will involve the analysis of real world data, and it is</span>
<span id="cb1-376"><a href="#cb1-376" aria-hidden="true" tabindex="-1"></a>expected that the submission includes 1) statements of aims, objectives,</span>
<span id="cb1-377"><a href="#cb1-377" aria-hidden="true" tabindex="-1"></a>and hypotheses; 2) the full and detailed methods followed by analyses</span>
<span id="cb1-378"><a href="#cb1-378" aria-hidden="true" tabindex="-1"></a>together with all code, 3) full reporting of results in a manner suited</span>
<span id="cb1-379"><a href="#cb1-379" aria-hidden="true" tabindex="-1"></a>for peer reviewed publications; 4) graphical support highlighting the</span>
<span id="cb1-380"><a href="#cb1-380" aria-hidden="true" tabindex="-1"></a>patterns observed (again with the code), and 5) a discussion if and when</span>
<span id="cb1-381"><a href="#cb1-381" aria-hidden="true" tabindex="-1"></a>required. The allocation of marks to the various sections is:</span>
<span id="cb1-382"><a href="#cb1-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-383"><a href="#cb1-383" aria-hidden="true" tabindex="-1"></a><span class="ss">1.  </span>Aims, objectives, and hypotheses: 5%</span>
<span id="cb1-384"><a href="#cb1-384" aria-hidden="true" tabindex="-1"></a><span class="ss">2.  </span>Methods and analyses: 45%</span>
<span id="cb1-385"><a href="#cb1-385" aria-hidden="true" tabindex="-1"></a><span class="ss">3.  </span>Results: 20%</span>
<span id="cb1-386"><a href="#cb1-386" aria-hidden="true" tabindex="-1"></a><span class="ss">4.  </span>Graphs: 15%</span>
<span id="cb1-387"><a href="#cb1-387" aria-hidden="true" tabindex="-1"></a><span class="ss">5.  </span>Dicsussion: 15%</span>
<span id="cb1-388"><a href="#cb1-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-389"><a href="#cb1-389" aria-hidden="true" tabindex="-1"></a><span class="fu">## Submission of assignments and exams</span></span>
<span id="cb1-390"><a href="#cb1-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-391"><a href="#cb1-391" aria-hidden="true" tabindex="-1"></a>A statement such as the one below accompanies every assignment---pay</span>
<span id="cb1-392"><a href="#cb1-392" aria-hidden="true" tabindex="-1"></a>attention, as failing to observe this instruction may result in a loss</span>
<span id="cb1-393"><a href="#cb1-393" aria-hidden="true" tabindex="-1"></a>of marks (i.e. if an assignment remains ungraded because the owner of</span>
<span id="cb1-394"><a href="#cb1-394" aria-hidden="true" tabindex="-1"></a>the material cannot be identified):</span>
<span id="cb1-395"><a href="#cb1-395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-396"><a href="#cb1-396" aria-hidden="true" tabindex="-1"></a>Submit a R script wherein you provide answers to Questions 1--9 by no</span>
<span id="cb1-397"><a href="#cb1-397" aria-hidden="true" tabindex="-1"></a>later than 8:00 tomorrow. Label the script as follows (e.g.):</span>
<span id="cb1-398"><a href="#cb1-398" aria-hidden="true" tabindex="-1"></a>**BCB743_AJ_Smit_Assignment_2.R**.</span>
<span id="cb1-399"><a href="#cb1-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-400"><a href="#cb1-400" aria-hidden="true" tabindex="-1"></a><span class="fu">## Late submission of CA</span></span>
<span id="cb1-401"><a href="#cb1-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-402"><a href="#cb1-402" aria-hidden="true" tabindex="-1"></a>Late assignments will be penalised 10% per day and will not be accepted</span>
<span id="cb1-403"><a href="#cb1-403" aria-hidden="true" tabindex="-1"></a>more than 48 hours late, unless evidence such as a doctor's note, a</span>
<span id="cb1-404"><a href="#cb1-404" aria-hidden="true" tabindex="-1"></a>death certificate, or another documented emergency can be provided. If</span>
<span id="cb1-405"><a href="#cb1-405" aria-hidden="true" tabindex="-1"></a>you know in advance that a submission will be late, please discuss this</span>
<span id="cb1-406"><a href="#cb1-406" aria-hidden="true" tabindex="-1"></a>and seek prior approval. This policy is based on the idea that in order</span>
<span id="cb1-407"><a href="#cb1-407" aria-hidden="true" tabindex="-1"></a>to learn how to translate your human thoughts into computer language</span>
<span id="cb1-408"><a href="#cb1-408" aria-hidden="true" tabindex="-1"></a>(coding) you should be working with them at multiple times each</span>
<span id="cb1-409"><a href="#cb1-409" aria-hidden="true" tabindex="-1"></a>week---ideally daily. Time has been allocated in class for working on</span>
<span id="cb1-410"><a href="#cb1-410" aria-hidden="true" tabindex="-1"></a>assignments and students are expected to continue to work on the</span>
<span id="cb1-411"><a href="#cb1-411" aria-hidden="true" tabindex="-1"></a>assignments outside of class. Successfully completing (and passing) this</span>
<span id="cb1-412"><a href="#cb1-412" aria-hidden="true" tabindex="-1"></a>module requires that you finish assignments based on what we have</span>
<span id="cb1-413"><a href="#cb1-413" aria-hidden="true" tabindex="-1"></a>covered in class by the following class period. Work diligently from the</span>
<span id="cb1-414"><a href="#cb1-414" aria-hidden="true" tabindex="-1"></a>onset so that even if something unexpected happens at the last minute</span>
<span id="cb1-415"><a href="#cb1-415" aria-hidden="true" tabindex="-1"></a>you should already be close to done. This approach also allows rapid</span>
<span id="cb1-416"><a href="#cb1-416" aria-hidden="true" tabindex="-1"></a>feedback to be provided to you, which can only be accomplished by</span>
<span id="cb1-417"><a href="#cb1-417" aria-hidden="true" tabindex="-1"></a>returning assignments quickly and punctually.</span>
<span id="cb1-418"><a href="#cb1-418" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-419"><a href="#cb1-419" aria-hidden="true" tabindex="-1"></a><span class="fu"># Support</span></span>
<span id="cb1-420"><a href="#cb1-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-421"><a href="#cb1-421" aria-hidden="true" tabindex="-1"></a>It's expected that some tricky aspects of the module will take time to</span>
<span id="cb1-422"><a href="#cb1-422" aria-hidden="true" tabindex="-1"></a>master, and the best way to master problematic material is to practice,</span>
<span id="cb1-423"><a href="#cb1-423" aria-hidden="true" tabindex="-1"></a>practice some more, and then to ask questions. Trying for 10 minutes and</span>
<span id="cb1-424"><a href="#cb1-424" aria-hidden="true" tabindex="-1"></a>then giving up is not good enough. I'll be more sympathetic to your</span>
<span id="cb1-425"><a href="#cb1-425" aria-hidden="true" tabindex="-1"></a>cause if you can demonstrate having tried for a full day before giving</span>
<span id="cb1-426"><a href="#cb1-426" aria-hidden="true" tabindex="-1"></a>up and asking me. When you ask questions about some challenge, this is</span>
<span id="cb1-427"><a href="#cb1-427" aria-hidden="true" tabindex="-1"></a>the way to do it---explain to me your numerous attempts at trying to</span>
<span id="cb1-428"><a href="#cb1-428" aria-hidden="true" tabindex="-1"></a>solve the problem, and explain how these various attempts have failed.</span>
<span id="cb1-429"><a href="#cb1-429" aria-hidden="true" tabindex="-1"></a>*I will not help you if you have not tried to help yourself first*</span>
<span id="cb1-430"><a href="#cb1-430" aria-hidden="true" tabindex="-1"></a>(maybe with advice from friends). There will be time in class to do</span>
<span id="cb1-431"><a href="#cb1-431" aria-hidden="true" tabindex="-1"></a>this, typically before we embark on a new topic. You are also encouraged</span>
<span id="cb1-432"><a href="#cb1-432" aria-hidden="true" tabindex="-1"></a>to bring up related questions that arise in your own B.Sc. (Hons.)</span>
<span id="cb1-433"><a href="#cb1-433" aria-hidden="true" tabindex="-1"></a>research project.</span>
<span id="cb1-434"><a href="#cb1-434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-435"><a href="#cb1-435" aria-hidden="true" tabindex="-1"></a>Should you require more time with me, find out when I am 'free' and set</span>
<span id="cb1-436"><a href="#cb1-436" aria-hidden="true" tabindex="-1"></a>an appointment by sending me a calendar invitation. I am happy to have a</span>
<span id="cb1-437"><a href="#cb1-437" aria-hidden="true" tabindex="-1"></a>personal meeting with you via Zoom, but I prefer face-to-face in my</span>
<span id="cb1-438"><a href="#cb1-438" aria-hidden="true" tabindex="-1"></a>office.</span>
<span id="cb1-439"><a href="#cb1-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-440"><a href="#cb1-440" aria-hidden="true" tabindex="-1"></a><span class="fu">## Help via BCB744 and BCB743 Issues on GitHub</span></span>
<span id="cb1-441"><a href="#cb1-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-442"><a href="#cb1-442" aria-hidden="true" tabindex="-1"></a>All discussion for the BCB744 and BCB743 workshops will be held in the</span>
<span id="cb1-443"><a href="#cb1-443" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">Issues</span><span class="co">](https://github.com/ajsmit/R_courses/issues)</span> of [this</span>
<span id="cb1-444"><a href="#cb1-444" aria-hidden="true" tabindex="-1"></a>repository](https://github.com/ajsmit/R_courses). Please post all</span>
<span id="cb1-445"><a href="#cb1-445" aria-hidden="true" tabindex="-1"></a>content-related questions there, and use email only for personal</span>
<span id="cb1-446"><a href="#cb1-446" aria-hidden="true" tabindex="-1"></a>matters. Note that this is a public repository, so be professional in</span>
<span id="cb1-447"><a href="#cb1-447" aria-hidden="true" tabindex="-1"></a>your writing here (grammar, etc.).</span>
<span id="cb1-448"><a href="#cb1-448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-449"><a href="#cb1-449" aria-hidden="true" tabindex="-1"></a>To start a new thread, create a **New issue**. Tag your peers using</span>
<span id="cb1-450"><a href="#cb1-450" aria-hidden="true" tabindex="-1"></a>their handle---<span class="in">`@ajsmit`</span>, for example---to get their attention.</span>
<span id="cb1-451"><a href="#cb1-451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-452"><a href="#cb1-452" aria-hidden="true" tabindex="-1"></a>Once a question has been answered, the issue will be closed, so lots of</span>
<span id="cb1-453"><a href="#cb1-453" aria-hidden="true" tabindex="-1"></a>good answers might end up in closed issues. Don't forget to look there</span>
<span id="cb1-454"><a href="#cb1-454" aria-hidden="true" tabindex="-1"></a>when looking for answers---you can use the **Search** feature on this</span>
<span id="cb1-455"><a href="#cb1-455" aria-hidden="true" tabindex="-1"></a>repository to find answers that might have been offered by the same or</span>
<span id="cb1-456"><a href="#cb1-456" aria-hidden="true" tabindex="-1"></a>similar problem experienced by someone else in the past.</span>
<span id="cb1-457"><a href="#cb1-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-458"><a href="#cb1-458" aria-hidden="true" tabindex="-1"></a>**Guidelines for posting questions:**</span>
<span id="cb1-459"><a href="#cb1-459" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-460"><a href="#cb1-460" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>First search existing issues (open or closed) for answers. If the</span>
<span id="cb1-461"><a href="#cb1-461" aria-hidden="true" tabindex="-1"></a>    question has already been answered, you're done! If there is an open</span>
<span id="cb1-462"><a href="#cb1-462" aria-hidden="true" tabindex="-1"></a>    issue, feel free to contribute to it. Or feel free to open a closed</span>
<span id="cb1-463"><a href="#cb1-463" aria-hidden="true" tabindex="-1"></a>    issue if you believe the answer is not satisfactory.</span>
<span id="cb1-464"><a href="#cb1-464" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Give your issue an informative title.</span>
<span id="cb1-465"><a href="#cb1-465" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Good: "Error: could not find function"ggplot""</span>
<span id="cb1-466"><a href="#cb1-466" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Bad: "My code does not work!" Note that you can edit an issue's</span>
<span id="cb1-467"><a href="#cb1-467" aria-hidden="true" tabindex="-1"></a>        title after it's been posted.</span>
<span id="cb1-468"><a href="#cb1-468" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Format your questions nicely using markdown and code formatting.</span>
<span id="cb1-469"><a href="#cb1-469" aria-hidden="true" tabindex="-1"></a>    Preview your issue prior to posting.</span>
<span id="cb1-470"><a href="#cb1-470" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>As I explained above, your peers and I will more sympathetic to your</span>
<span id="cb1-471"><a href="#cb1-471" aria-hidden="true" tabindex="-1"></a>    cause if you can show *all the things you have tried as you,</span>
<span id="cb1-472"><a href="#cb1-472" aria-hidden="true" tabindex="-1"></a>    yourself, tried to fix the issue first*.</span>
<span id="cb1-473"><a href="#cb1-473" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Include code and example data so the person trying to help you have</span>
<span id="cb1-474"><a href="#cb1-474" aria-hidden="true" tabindex="-1"></a>    something to work with (and which results in the error, perhaps)</span>
<span id="cb1-475"><a href="#cb1-475" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Where appropriate, provide links to specific files, or even lines</span>
<span id="cb1-476"><a href="#cb1-476" aria-hidden="true" tabindex="-1"></a>    within them, in the body of your issue. This will help your peers</span>
<span id="cb1-477"><a href="#cb1-477" aria-hidden="true" tabindex="-1"></a>    understand your question. Note that only the teaching team will have</span>
<span id="cb1-478"><a href="#cb1-478" aria-hidden="true" tabindex="-1"></a>    access to private repos.</span>
<span id="cb1-479"><a href="#cb1-479" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>(Optional) Tag someone or some group of people. Start by typing</span>
<span id="cb1-480"><a href="#cb1-480" aria-hidden="true" tabindex="-1"></a>    their GitHub username prefixed with the \@ symbol. Of course this</span>
<span id="cb1-481"><a href="#cb1-481" aria-hidden="true" tabindex="-1"></a>    supposes that each of you have a GitHub account and username.</span>
<span id="cb1-482"><a href="#cb1-482" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Hit **Submit new issue** when you're ready to post.</span>
<span id="cb1-483"><a href="#cb1-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-484"><a href="#cb1-484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-485"><a href="#cb1-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-486"><a href="#cb1-486" aria-hidden="true" tabindex="-1"></a><span class="in">```{=html}</span></span>
<span id="cb1-487"><a href="#cb1-487" aria-hidden="true" tabindex="-1"></a><span class="in">&lt;!-- ## Function list</span></span>
<span id="cb1-488"><a href="#cb1-488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-489"><a href="#cb1-489" aria-hidden="true" tabindex="-1"></a><span class="in">  You will encounter the following functions for the first time in this module:</span></span>
<span id="cb1-490"><a href="#cb1-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-491"><a href="#cb1-491" aria-hidden="true" tabindex="-1"></a><span class="in">  | Function              | Description                                        |</span></span>
<span id="cb1-492"><a href="#cb1-492" aria-hidden="true" tabindex="-1"></a><span class="in">  |:----------------------|----------------------------------------------------|</span></span>
<span id="cb1-493"><a href="#cb1-493" aria-hidden="true" tabindex="-1"></a><span class="in">  | `adonis2()`           | Permutational Multivariate Analysis of Variance Using Distance Matrices |</span></span>
<span id="cb1-494"><a href="#cb1-494" aria-hidden="true" tabindex="-1"></a><span class="in">  | `anova.betadisper()`  | PERMDIST 2 for Multivariate homogeneity of groups dispersions (variances) |</span></span>
<span id="cb1-495"><a href="#cb1-495" aria-hidden="true" tabindex="-1"></a><span class="in">  | `anova.cca ()`        | Permutation Test for CCA, RDA and CA of Principal Coordinates |</span></span>
<span id="cb1-496"><a href="#cb1-496" aria-hidden="true" tabindex="-1"></a><span class="in">  | `anova.manyglm()`     | Analysis of Deviance for Multivariate Generalized Linear Model Fits for Abundance Data |</span></span>
<span id="cb1-497"><a href="#cb1-497" aria-hidden="true" tabindex="-1"></a><span class="in">  | `beta.pair()`         | Incidence-based Pair-wise Dissimilarities |</span></span>
<span id="cb1-498"><a href="#cb1-498" aria-hidden="true" tabindex="-1"></a><span class="in">  | `betadisper()`        | Multivariate homogeneity of groups dispersions (variances) |</span></span>
<span id="cb1-499"><a href="#cb1-499" aria-hidden="true" tabindex="-1"></a><span class="in">  | `betapart.core()`     | Core Calculations of betapart |</span></span>
<span id="cb1-500"><a href="#cb1-500" aria-hidden="true" tabindex="-1"></a><span class="in">  | `biplot()`            | Biplot of Multivariate Data |</span></span>
<span id="cb1-501"><a href="#cb1-501" aria-hidden="true" tabindex="-1"></a><span class="in">  | `capscale()`          | [Partial] Distance-based Redundancy Analysis |</span></span>
<span id="cb1-502"><a href="#cb1-502" aria-hidden="true" tabindex="-1"></a><span class="in">  | `cca()`               | [Partial] [Constrained] Correspondence Analysis and Redundancy Analysis |</span></span>
<span id="cb1-503"><a href="#cb1-503" aria-hidden="true" tabindex="-1"></a><span class="in">  | `cleanplot.pc()`      | Numerical Ecology in R's PCA Plot |</span></span>
<span id="cb1-504"><a href="#cb1-504" aria-hidden="true" tabindex="-1"></a><span class="in">  | `daisy()`             | Dissimilarity Matrix Calculation |</span></span>
<span id="cb1-505"><a href="#cb1-505" aria-hidden="true" tabindex="-1"></a><span class="in">  | `decostand()`         | Standardization Methods for Community Ecology |</span></span>
<span id="cb1-506"><a href="#cb1-506" aria-hidden="true" tabindex="-1"></a><span class="in">  | `diag()`              | Matrix Diagonals |</span></span>
<span id="cb1-507"><a href="#cb1-507" aria-hidden="true" tabindex="-1"></a><span class="in">  | `diversity()`         | Ecological Diversity Indices |</span></span>
<span id="cb1-508"><a href="#cb1-508" aria-hidden="true" tabindex="-1"></a><span class="in">  | `diversityresult()`   | Alternative Diversity Results |</span></span>
<span id="cb1-509"><a href="#cb1-509" aria-hidden="true" tabindex="-1"></a><span class="in">  | `fviz_cluster()`      | Visualize Clustering Results |</span></span>
<span id="cb1-510"><a href="#cb1-510" aria-hidden="true" tabindex="-1"></a><span class="in">  | `fviz_nbclust()`      | Determining and Visualizing the Optimal Number of Clusters |</span></span>
<span id="cb1-511"><a href="#cb1-511" aria-hidden="true" tabindex="-1"></a><span class="in">  | `fviz_silhouette()`   | Visualize Silhouette Information from Clustering |</span></span>
<span id="cb1-512"><a href="#cb1-512" aria-hidden="true" tabindex="-1"></a><span class="in">  | `geodist()`           | Convert Lon-Lat Coordinates into Vector or Matrix of Geodesic Distances in Metres |</span></span>
<span id="cb1-513"><a href="#cb1-513" aria-hidden="true" tabindex="-1"></a><span class="in">  | `ggcorrplot()`        | Visualization of a Correlation Matrix Using ggplot2 |</span></span>
<span id="cb1-514"><a href="#cb1-514" aria-hidden="true" tabindex="-1"></a><span class="in">  | `goodness()`          | Diagnostic Tools for [Constrained] Ordination (CCA, RDA, DCA, CA, PCA) |</span></span>
<span id="cb1-515"><a href="#cb1-515" aria-hidden="true" tabindex="-1"></a><span class="in">  | `imputePCA()`         | Impute Dataset with PCA |</span></span>
<span id="cb1-516"><a href="#cb1-516" aria-hidden="true" tabindex="-1"></a><span class="in">  | `lowerTriangle()`     | Extract or Replace the Upper/Lower Triangular Portion of a Matrix |</span></span>
<span id="cb1-517"><a href="#cb1-517" aria-hidden="true" tabindex="-1"></a><span class="in">  | `manyglm()`           | Fitting Generalized Linear Models for Multivariate Abundance Data |</span></span>
<span id="cb1-518"><a href="#cb1-518" aria-hidden="true" tabindex="-1"></a><span class="in">  | `meanvar.plot()`      | Construct Mean-Variance Plots for Multivariate Abundance Data |</span></span>
<span id="cb1-519"><a href="#cb1-519" aria-hidden="true" tabindex="-1"></a><span class="in">  | `metaMDS()`           | Nonmetric Multidimensional Scaling with Stable Solution from Random Starts, Axis Scaling and Species Scores |</span></span>
<span id="cb1-520"><a href="#cb1-520" aria-hidden="true" tabindex="-1"></a><span class="in">  | `mvabund()`           | Multivariate Abundance Data Objects |</span></span>
<span id="cb1-521"><a href="#cb1-521" aria-hidden="true" tabindex="-1"></a><span class="in">  | `ordiellipse()`       | Display Groups or Factor Levels in Ordination Diagrams |</span></span>
<span id="cb1-522"><a href="#cb1-522" aria-hidden="true" tabindex="-1"></a><span class="in">  | `ordiplot()`          | Alternative Plot and Identify Functions for Ordination |</span></span>
<span id="cb1-523"><a href="#cb1-523" aria-hidden="true" tabindex="-1"></a><span class="in">  | `ordispider()`        | Display Groups or Factor Levels in Ordination Diagrams |</span></span>
<span id="cb1-524"><a href="#cb1-524" aria-hidden="true" tabindex="-1"></a><span class="in">  | `ordisurf()`          | Fit and Plot Smooth Surfaces of Variables on Ordination |</span></span>
<span id="cb1-525"><a href="#cb1-525" aria-hidden="true" tabindex="-1"></a><span class="in">  | `pam()`               | Partitioning Around Medoids |</span></span>
<span id="cb1-526"><a href="#cb1-526" aria-hidden="true" tabindex="-1"></a><span class="in">  | `permutest()`         | Permutation Test for CCA, RDA and CA of Principal Coordinates |</span></span>
<span id="cb1-527"><a href="#cb1-527" aria-hidden="true" tabindex="-1"></a><span class="in">  | `rda()`               | [Partial] [Constrained] Correspondence Analysis and Redundancy Analysis |</span></span>
<span id="cb1-528"><a href="#cb1-528" aria-hidden="true" tabindex="-1"></a><span class="in">  | `RsquareAdj()`        | Adjusted R-square |</span></span>
<span id="cb1-529"><a href="#cb1-529" aria-hidden="true" tabindex="-1"></a><span class="in">  | `scores()`            | Get Species or Site Scores from an Ordination |</span></span>
<span id="cb1-530"><a href="#cb1-530" aria-hidden="true" tabindex="-1"></a><span class="in">  | `specnumber()`        | Ecological Diversity Indices |</span></span>
<span id="cb1-531"><a href="#cb1-531" aria-hidden="true" tabindex="-1"></a><span class="in">  | `stressplot()`        | Goodness of Fit and Shepard Plot for Nonmetric Multidimensional Scaling |</span></span>
<span id="cb1-532"><a href="#cb1-532" aria-hidden="true" tabindex="-1"></a><span class="in">  | `t()`                 | Matrix Transpose |</span></span>
<span id="cb1-533"><a href="#cb1-533" aria-hidden="true" tabindex="-1"></a><span class="in">  | `vegdist()`           | Dissimilarity Indices for Community Ecologists |</span></span>
<span id="cb1-534"><a href="#cb1-534" aria-hidden="true" tabindex="-1"></a><span class="in">  | `vif.cca()`           | Diagnostic Tools for [Constrained] Ordination (CCA, RDA, DCA, CA, PCA) | --&gt;</span></span>
<span id="cb1-535"><a href="#cb1-535" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-536"><a href="#cb1-536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-537"><a href="#cb1-537" aria-hidden="true" tabindex="-1"></a><span class="fu"># References</span></span>
<span id="cb1-538"><a href="#cb1-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-539"><a href="#cb1-539" aria-hidden="true" tabindex="-1"></a>::: {#refs}</span>
<span id="cb1-540"><a href="#cb1-540" aria-hidden="true" tabindex="-1"></a>:::</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>¬© 2023, AJ Smit</p>
</div>   
    <div class="nav-footer-center">

<div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
<p>This page is built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>




<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>