<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Smit, A. J.">
<meta name="dcterms.date" content="2024-07-06">

<title>Ecological Model Building – The Tangled Bank</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../BCB743/deep_dive.html" rel="next">
<link href="../BCB743/cluster_analysis.html" rel="prev">
<link href="../images/favicon.ico" rel="icon">
<script src="../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-1b3e43c72e8be34557c75123b0b69e0d.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-517f42b09ea8b46c6e86e792f9c01f8c.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/quarto-contrib/fontawesome6-0.1.0/all.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/fontawesome6-0.1.0/latex-fontsize.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-SCYH68M489"></script>

<script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-SCYH68M489', { 'anonymize_ip': true});
</script>

<script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"headline",
  "consent_type":"express",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  ,
"language":"en"
  });
});
</script> 
  
<style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../styles.css">
<meta property="og:title" content="Ecological Model Building – The Tangled Bank">
<meta property="og:description" content="UWC’s Biodiversity &amp; Conservation Biology Department Pages Serving BDC334, BCB743, and BCB744.">
<meta property="og:image" content="http://tangledbank.netlify.app/BCB743/data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==">
<meta property="og:site_name" content="The Tangled Bank">
</head>

<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">The Tangled Bank</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-vignettes" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Vignettes</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-vignettes">    
        <li>
    <a class="dropdown-item" href="../vignettes/netCDF_dates.html">
 <span class="dropdown-text">Dates From netCDF Files: Two Approaches</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/prep_NOAA_OISST.html">
 <span class="dropdown-text">Downloading and Prep: ERDDAP</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_ERDDAP.html">
 <span class="dropdown-text">Retrieving Chlorophyll-a Data from ERDDAP Servers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/alt_method.html">
 <span class="dropdown-text">Downloading and Prep: Python + CDO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/gridded_data.html">
 <span class="dropdown-text">Detecting Events in Gridded Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/regridding.html">
 <span class="dropdown-text">Regridding gridded data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/MHW_MCS_horizonplots.html">
 <span class="dropdown-text">Displaying MHWs and MCSs as Horizon Plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/heatwaveR_issues.html">
 <span class="dropdown-text">heatwaveR Issues</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_sightings.html">
 <span class="dropdown-text">Plotting the Whale Sightings and Chlorophyll-<em>a</em> Concentrations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_localisation.html">
 <span class="dropdown-text">Spatial Localisation, Subsetting, and Aggregation of the Chlorophyll-<em>a</em> Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/elem_ts_methods.html">
 <span class="dropdown-text">Wavelet Analysis of Diatom Time Series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/buffer_data_extract.html">
 <span class="dropdown-text">Extracting gridded data within a buffer</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-high-performace-computing" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">High-Performace Computing</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-high-performace-computing">    
        <li>
    <a class="dropdown-item" href="../vignettes/README_Lengau.html">
 <span class="dropdown-text">Using Lengau</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/PBSPro_users.html">
 <span class="dropdown-text">PBSPro user and job management</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/README_tmux.html">
 <span class="dropdown-text">Using tmux</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../blog.html"> 
<span class="menu-text">Tangled Bank Blog</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-blogroll" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Blogroll</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-blogroll">    
        <li>
    <a class="dropdown-item" href="https://www.r-bloggers.com">
 <span class="dropdown-text">R-bloggers</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="../blog.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/ajsmit/tangled_bank">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://bugs.com">
 <span class="dropdown-text">Report a Bug</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BCB743/BCB743_index.html">BCB743: Quantitative Ecology</a></li><li class="breadcrumb-item"><a href="../BCB743/model_building.html">15: Model Building</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../images/hex_logo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">UNDERGRADUATE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC223: Plant Ecophysiology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/BDC223_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L00-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 0: Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L01-worldmapper.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: Worldmapper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L02-SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2: Surface Area to Volume (SA/V) Ratios in Biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L03-plant_stress.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3: Plant Stress</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L04-carbon_cycle.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Perturbations to the Carbon Cycle</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L05-light.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 5: Light</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L06a-pigments_photosynthesis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6a: Pigments and Photosynthesis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L06b-jassby_platt.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6b: PI Curves – Jassby and Platt</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L07-chromatic_adaptation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 7: Chromatic Adaptation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L08a-nutrient_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8a: Nutrient Uptake</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L08b-nutrients_michaelis_menten.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8b: Uptake Kinetics – Michaelis-Menten</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab1_SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1: Surface Area to Volume (SA/V) Ratios in Biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab2_misc_calcs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2: Miscellaneous Calculations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab3_PI_curves.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3: PI Curves – Jassby and Platt</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab4_nitrogen_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4: Uptake Kinetics – Michaelis-Menten</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/BDC223_FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC334: Biogeography &amp; Global Ecology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/BDC334_syllabus.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/BDC334-Lecture-Transcripts.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">BCB334 Lecture Transcript</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: Introductory Lecture</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-02-ecosystems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2. Overview of Ecosystems</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-03-gradients.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3. Ecological Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Biodiversity Concepts</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 5. Multivariate Data</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-06-unified-ecology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6: Unified Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1. Ecological Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-02a-r_rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2a. R &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-02b-env_dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2b. Environmental Distance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-03-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3. Quantifying Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4. Species Distribution Patterns</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/Transboundary_systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transboundary systems</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS CORE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">BCB744: Introduction to R, &amp; Biostatistics</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/BCB744_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/01-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. R &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/02-working-with-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Working With Data &amp; Code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/03-data-in-R.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Data Classes &amp; Structures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/04-workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. R Workflows</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/05-graphics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Graphics with <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/06-faceting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Faceting Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/07-brewing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Brewing Colours</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/08-mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Mapping With <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/09-mapping_style.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Mapping With Style</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/10-mapping_rnaturalearth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Mapping With Natural Earth</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/11-mapping_quakes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. The Fiji Earthquake Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/12-tidy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Tidy Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/13-tidier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Tidier Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/14-tidiest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Tidiest Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/15-recap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15. Recap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/BCB744_Intro_R_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Intro R Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/examples/BCB744_Intro_R_Example_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 Intro R Example 1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/examples/BCB744_Intro_R_Example_2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 Intro R Example 2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../self-assessment_v1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Self-Assessment Form</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/BCB744_Intro_R_Presentations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Presentations</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/BCB744_Intro_R_Test_2025.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Intro R Test</span></span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/01-scientific-philosophy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Philosophy of Science</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/02-summarise-and-describe.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Data Summaries &amp; Descriptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/03-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Statistical Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/04-distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data Distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/05-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Statistical Inference</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/06-assumptions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Assumptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/07-t_tests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. <em>t</em>-Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/08-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. ANOVA</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/09-regressions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Simple Linear Regressions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/10-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/11-decision_guide.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/12-glance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Non-Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/13-confidence.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Confidence Intervals</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/14-transformations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Data Transformations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/BCB744_Biostatistics_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Biostatistics Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assessments/examples/BCB744_BioStats_Example_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 BioStats 2025 Example 1</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS ELECTIVES</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">BCB743: Quantitative Ecology</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/BCB743_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/review.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1-4: Review Concepts fo Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5: Correlations &amp; Associations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7: Intro to Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8a: Principal Components Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA_examples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8b: PCA: Additional Examples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA_SDG_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8c: PCA: WHO SDG Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/CA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9a: Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/DCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9b: Detrended Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCoA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10: Principal Coordinates Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/nMDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11a: Non-metric Multidimensional Scaling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/nMDS_diatoms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11b: nMDS: PERMANOVA (Diatoms) Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/unconstrained-summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12: Unconstrained Ordination Summary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/constrained_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13a: Distance-Based Redundancy Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/two_oceans_appendices.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13b: db-RDA: Seaweeds Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/cluster_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14: Cluster Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/model_building.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">15: Model Building</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/deep_dive.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16a: Deep Dive Into Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/multiple_regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16b: Multiple Regression</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../AI4AI/AI4AI.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Artificial Intelligence for Academic Integrity (AI4AI)</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">SUPPORT</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Web Resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/general_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/ecology_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quantitative Ecology Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/spatial_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial R Resources</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#about-the-seaweed-data" id="toc-about-the-seaweed-data" class="nav-link active" data-scroll-target="#about-the-seaweed-data">About the Seaweed Data</a></li>
  <li><a href="#theoretical-understanding-of-environmental-drivers" id="toc-theoretical-understanding-of-environmental-drivers" class="nav-link" data-scroll-target="#theoretical-understanding-of-environmental-drivers">Theoretical Understanding of Environmental Drivers</a>
  <ul class="collapse">
  <li><a href="#new-concepts" id="toc-new-concepts" class="nav-link" data-scroll-target="#new-concepts">New Concepts</a></li>
  </ul></li>
  <li><a href="#identifying-spatial-gradients" id="toc-identifying-spatial-gradients" class="nav-link" data-scroll-target="#identifying-spatial-gradients">Identifying Spatial Gradients</a>
  <ul class="collapse">
  <li><a href="#new-concept" id="toc-new-concept" class="nav-link" data-scroll-target="#new-concept">New Concept</a></li>
  </ul></li>
  <li><a href="#assessing-environmental-gradients" id="toc-assessing-environmental-gradients" class="nav-link" data-scroll-target="#assessing-environmental-gradients">Assessing Environmental Gradients</a></li>
  <li><a href="#model-building-and-variable-selection" id="toc-model-building-and-variable-selection" class="nav-link" data-scroll-target="#model-building-and-variable-selection">Model Building and Variable Selection</a>
  <ul class="collapse">
  <li><a href="#new-concepts-1" id="toc-new-concepts-1" class="nav-link" data-scroll-target="#new-concepts-1">New Concepts</a></li>
  </ul></li>
  <li><a href="#reconciling-ecological-and-statistical-knowledge" id="toc-reconciling-ecological-and-statistical-knowledge" class="nav-link" data-scroll-target="#reconciling-ecological-and-statistical-knowledge">Reconciling Ecological and Statistical Knowledge</a>
  <ul class="collapse">
  <li><a href="#new-concepts-2" id="toc-new-concepts-2" class="nav-link" data-scroll-target="#new-concepts-2">New Concepts</a></li>
  </ul></li>
  <li><a href="#model-validation" id="toc-model-validation" class="nav-link" data-scroll-target="#model-validation">Model Validation</a>
  <ul class="collapse">
  <li><a href="#cross-validation" id="toc-cross-validation" class="nav-link" data-scroll-target="#cross-validation">Cross-Validation</a></li>
  <li><a href="#model-averaging-and-ensemble-methods" id="toc-model-averaging-and-ensemble-methods" class="nav-link" data-scroll-target="#model-averaging-and-ensemble-methods">Model Averaging and Ensemble Methods</a></li>
  </ul></li>
  <li><a href="#practical-steps-for-model-selection" id="toc-practical-steps-for-model-selection" class="nav-link" data-scroll-target="#practical-steps-for-model-selection">Practical Steps for Model Selection</a></li>
  <li><a href="#some-advanced-ideas" id="toc-some-advanced-ideas" class="nav-link" data-scroll-target="#some-advanced-ideas">Some Advanced Ideas</a>
  <ul class="collapse">
  <li><a href="#hierarchical-and-multi-scale-modeling" id="toc-hierarchical-and-multi-scale-modeling" class="nav-link" data-scroll-target="#hierarchical-and-multi-scale-modeling">Hierarchical and Multi-Scale Modeling</a></li>
  <li><a href="#multi-scale-environmental-predictors" id="toc-multi-scale-environmental-predictors" class="nav-link" data-scroll-target="#multi-scale-environmental-predictors">Multi-Scale Environmental Predictors</a></li>
  <li><a href="#advanced-methods-and-approaches" id="toc-advanced-methods-and-approaches" class="nav-link" data-scroll-target="#advanced-methods-and-approaches">Advanced Methods and Approaches</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB743/model_building.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="model_building.pdf"><i class="bi bi-file-pdf"></i>Typst</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BCB743/BCB743_index.html">BCB743: Quantitative Ecology</a></li><li class="breadcrumb-item"><a href="../BCB743/model_building.html">15: Model Building</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Ecological Model Building</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://tangledbank.netlify.app">Smit, A. J.</a> <a href="https://orcid.org/0000-0002-3799-6126" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uwc.ac.za">
            University of the Western Cape
            </a>
          </p>
      </div>
  </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 6, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<p>Developing ecological models, whether multiple regression models or constrained ordinations with numerous environmental predictors, requires a synthesis of biological and ecological theory (the substance of your undergraduate education) with defensible statistical analysis to achieve an integrated, acceptable outcome.</p>
<p>I shall guide you through the process of selecting the most appropriate model to explain ecological outcomes, using the dataset concerning seaweed species composition (<span class="math inline">\(\beta\)</span>-diversity) along the South African coastline as our exemple. You should be well-acquainted with this analysis by now. The principles we explore are broad applicability across various regions and scales.</p>
<section id="about-the-seaweed-data" class="level2">
<h2 class="anchored" data-anchor-id="about-the-seaweed-data">About the Seaweed Data</h2>
<p>In my examples (<a href="../BCB743/deep_dive.html">Gradients Example</a>, <a href="https://tangledbank.netlify.app/docs/The-Biostatistics-Book-Ch-3-5.pdf">Multiple Regression</a> and <a href="../BCB743/constrained_ordination.html">db-RDA</a>), I employ environmental variables derived from an extensive daily time series of seawater temperature. These data were transformed into various statistics that capture components of the thermal regime along the South African coastline:</p>
<ul>
<li>the annual mean climatology (<code>annMean</code>)</li>
<li>the climatological mean for February (warmest month) and August (coldest month) (<code>febMean</code> and <code>augMean</code>)</li>
<li>the climatological SD for those months (<code>febSD</code> and <code>augSD</code>)</li>
<li>the temperature range (from daily climatology) for February and August (<code>febRange</code> and <code>augRange</code>)</li>
</ul>
<p>Each variable was calculated as Euclidean distances between site pairs, thus establishing a foundation for <span class="math inline">\(\beta\)</span>-diversity assessment. To this end, I apply the Sørensen dissimilarity to the corresponding species data between those site pairs. Given that these summary statistics derive from the same initial dataset, issues of non-independence among predictor data become inevitable. Multicollinearity is virtually guaranteed (even theoretical considerations make this apparent). Nevertheless, these data retain utility for understanding whether seaweed flora composition responds primarily to mean annual temperature, minimum values, or maximum values. The analysis might further reveal which aspects of temperature variability along the coast exert greater influence in structuring species composition.</p>
<p>I have also incorporated geographical distance (recalculated as Euclidean distance) between site pairs along the coast, which thus corresponds to the same spatial grid as the <span class="math inline">\(\beta\)</span>-diversity and environmental distance data. Here lies a conceptual snare: does distance function as a genuine predictor, or merely serve as a convenient descriptor of the spatial domain across which species gradients unfold? The same question arises for <code>bio</code>, the bioregional classification of the seaweed flora established by Professor John Bolton.</p>
</section>
<section id="theoretical-understanding-of-environmental-drivers" class="level2">
<h2 class="anchored" data-anchor-id="theoretical-understanding-of-environmental-drivers">Theoretical Understanding of Environmental Drivers</h2>
<p>Begin by examining how specific environmental variables (<em>e.g.</em>, <code>dist</code>, <code>bio</code>, <code>augMean</code>, <code>febRange</code>, <code>febSD</code>, <code>augSD</code>, <code>annMean</code>) might influence seaweed community structure along the coast. Temperature is a fundamental driver, affecting biological processes and impacting species differentially. When selecting relevant predictor variables, consider how temperature metrics (such as the mean values, fluctuations, and ranges) influence reproductive timing, growth rates, and physiological tolerances. Develop your thinking from your ecological understanding of these principles. You’ll find that some variables may lack theoretical importance and warrant removal on conceptual grounds before modelling commences.</p>
<section id="new-concepts" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="new-concepts">New Concepts</h3>
<ol type="1">
<li>Where such data exist, consider incorporating <strong>trait-based approaches</strong> into your analysis. Functional traits of seaweeds (for example, thallus morphology, photosynthetic pigments, or reproductive strategies) may respond differentially to environmental gradients. This approach can illuminate mechanisms driving community composition beyond simple species presence/absence data. While I have not yet explored this avenue, I anticipate pursuing it as a future research project, probably through <strong>Fourth-corner analysis</strong> or <strong>RLQ ordination</strong>.</li>
</ol>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<strong>Fourth-corner analysis</strong>
</div>
</div>
<div class="callout-body-container callout-body">
<p>Fourth-corner analysis addresses questions about which species traits respond to which environmental gradients. The method derives its name from the “fourth corner” of a conceptual data matrix linking three data tables:</p>
<ul>
<li>R: <strong>Site-by-species abundance matrix</strong> (the species table with the community data)</li>
<li>L: <strong>Site-by-environment matrix</strong> (the temperature variables, distance, bioregions, <em>etc.</em>, in the environmental table)</li>
<li>Q: <strong>Species-by-traits matrix</strong> (morphological, physiological, or life-history characteristics)</li>
</ul>
<p>The “fourth corner” is the missing direct link between traits and environment, which the analysis infers through the observed community patterns. The method tests whether some trait-environment combinations occur more or less frequently than expected by chance, given the observed species distributions. The statistical underpinning relies on permutation tests to assess significance, typically using three different null models:</p>
<ul>
<li>Permuting sites (tests for environmental filtering)</li>
<li>Permuting species (tests for trait convergence)</li>
<li>Permuting both simultaneously (combines both ecological processes)</li>
</ul>
</div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<strong>RLQ ordination</strong>
</div>
</div>
<div class="callout-body-container callout-body">
<p>RLQ ordination extends the fourth-corner concept into multivariate space and provides an integrated view of trait-environment relationships. It does a double ordination that maximises the covariance between environmental variables (table <strong>R × L</strong>) and species traits (table <strong>L × Q</strong>), with the species composition table (<strong>L</strong>) bringing in the new concept.</p>
<p>Mathematically, RLQ finds linear combinations of environmental variables and linear combinations of traits that produce maximum correlation when projected through the species composition data. It reveals the main gradients along which traits and environmental conditions co-vary. The technique essentially performs a <strong>co-inertia analysis</strong> between two indirect ordinations:</p>
<ul>
<li>Environmental variables weighted by species abundances</li>
<li>Species traits weighted by their occurrence across sites</li>
</ul>
</div>
</div>
<ol start="2" type="1">
<li>Beyond environmental variables and spatial gradients, incorporating <strong>phylogenetic methods</strong> can yield deeper insights into factors influencing ecological outcomes. Phylogenetic approaches account for evolutionary relationships among species. It might reveal patterns and relationships that complement species composition and trait-based analyses, and can reveal whether community assembly is driven by environmental filtering (closely related species co-occurring) or competitive exclusion (distantly related species co-occurring). For marine seaweeds, phylogenetic structure can indicate evolutionary constraints on environmental tolerances, niche conservatism vs.&nbsp;adaptive radiation, and historical biogeographic processes.</li>
</ol>
</section>
</section>
<section id="identifying-spatial-gradients" class="level2">
<h2 class="anchored" data-anchor-id="identifying-spatial-gradients">Identifying Spatial Gradients</h2>
<p>Assess whether your variables exhibit strong spatial gradients or differences. Consider these examples:</p>
<ol type="1">
<li><p>The <strong>annual mean temperature (<code>annMean</code>)</strong> integrates data from warm and cold seasons to serve as an integrated predictor of global ecosystems. Regionally, it may function as a significant driver due to coastal temperature gradients, though potential collinearity with other variables requires examination.</p></li>
<li><p>The <strong>mean temperature of the warmest month (<code>febMean</code>)</strong> displays a clear gradient from the east coast to Cape Point, but remains relatively stable along the west coast. Variability in this region is captured by <code>febSD</code>.</p></li>
<li><p>The <strong>temperature range of the warmest month (<code>febRange</code>)</strong> differentiates the Benguela Current from the Agulhas Current. It exhibits both east-west and north-south gradients.</p></li>
<li><p>Looking at <strong>temperature variability</strong> such as during the coldest and warmest monsts, <code>augSD</code> and <code>febSD</code>, respectively, give geographically-linked explanations, maybe showing how to upwelling intensity or current stability affect species structural patterns.</p></li>
<li><p>Employ <strong>unconstrained ordinations</strong> with environmental vectors (<code>envfit()</code>) to guide selection of important structuring predictors.</p></li>
</ol>
<section id="new-concept" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="new-concept">New Concept</h3>
<ol start="6" type="1">
<li>Consider incorporating specific <strong>oceanographic features</strong> into your analysis. Ideas that come to mind include upwelling intensity, current velocity, or nutrient availability. These factors can markedly influence seaweed distribution and may enhance the explanatory power of your models. A more advanced approach would be to use <strong>oceanographic models</strong> to derive spatially explicit variables that capture the dynamic nature of coastal environments. For example, Langrangian models will allow one to track the directionality of water movement and its influence on species dispersal and connectivity.</li>
</ol>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<strong>Lagrangian Models</strong>
</div>
</div>
<div class="callout-body-container callout-body">
<p>Current patterns, upwelling dynamics, and water mass movements may be more relevant than Euclidean distance for understanding seaweed community patterns. Lagrangian oceanographic variables (tracking water movement) will outperform static Eulerian (as used in the present seaweed analysis) measures for marine species distributions.</p>
</div>
</div>
</section>
</section>
<section id="assessing-environmental-gradients" class="level2">
<h2 class="anchored" data-anchor-id="assessing-environmental-gradients">Assessing Environmental Gradients</h2>
<p>Environmental variables with strong spatial gradients likely exert the most significant impact on seaweeds, indicating plausible environmental filtering (niche mechanisms). To quantify these gradients:</p>
<ol type="1">
<li><p>Conduct <strong>multiple linear regressions</strong> using continuous predictor variables as functions of dist (distance between site pairs).</p></li>
<li><p>Create <strong>thematic maps</strong> (spatially implicit) of temperature variables and vary symbol size or colour intensity by magnitude. Maybe use GIS tools to interpolate values between sampling points for more comprehensive visualisation.</p></li>
<li><p>Assess <strong>spatial autocorrelation</strong> in your variables using techniques such as Moran’s I or Geary’s C. This helps identify the scale at which environmental factors operate and informs model structure.</p></li>
</ol>
</section>
<section id="model-building-and-variable-selection" class="level2">
<h2 class="anchored" data-anchor-id="model-building-and-variable-selection">Model Building and Variable Selection</h2>
<p>Throughout the model-building process, make informed decisions about variable selection based on both theoretical knowledge and data-driven approaches:</p>
<ol type="1">
<li><p>Select variables based on <strong>ecological understanding</strong> of species’ responses to environmental drivers, considering both direct and indirect effects.</p></li>
<li><p>Choose variables that reflect <strong>significant known environmental gradients</strong> influenced by factors such as ocean currents, coastal topography, and climate patterns.</p></li>
<li><p>Use <strong>unconstrained ordinations</strong> (<em>e.g.</em>, Principal Component Analysis, PCA) to explore the relationships between environmental variables and species composition. This can help identify key gradients and inform variable selection. Running such ordinations side-by-side on the species and environmental data can reveal how well the environmental variables explain the observed species patterns. This method becomes especially useful when you superimpose the environmental vectors onto the ordination plot using <code>envfit()</code> or a GAM or something similar.</p></li>
<li><p>Also use <strong>data-driven decision making</strong>. Employ statistical methods such as Variance Inflation Factors (VIFs) or forward selection (<em>e.g.</em>, <code>stepAIC()</code>) to address multicollinearity and refine model selection. Consider modern techniques like elastic net regression or random forests for variable importance ranking.</p></li>
</ol>
<section id="new-concepts-1" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="new-concepts-1">New Concepts</h3>
<ol start="5" type="1">
<li>Consider also spatial autocorrelation in the models. Beyond simply identifying spatial autocorrelation, one might develop explicit strategies for incorporating it into models. One such is <strong>Moran’s Eigenvector Maps (MEMs)</strong>, which can be used to account for spatial structure in species-environment relationships. MEMs are derived from distance matrices and can be included as predictors in regression models to capture spatial patterns.</li>
</ol>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<strong>Moran’s Eigenvector Maps (MEMs)</strong>
</div>
</div>
<div class="callout-body-container callout-body">
<p>Moran’s Eigenvector Maps (MEMs) can decompose spatial patterns at multiple scales. It allows one to partition variation into broad-scale environmental trends and fine-scale spatial processes. I have already done this for the seaweed analysis and you can read about it in the paper. For coastal seaweed data, it can reveal spatially explicit account for:</p>
<ul>
<li>Oceanographic connectivity between sites</li>
<li>Distance-decay relationships in species similarity</li>
<li>Scale-dependent environmental effects (local vs.&nbsp;regional processes)</li>
</ul>
</div>
</div>
<ol start="6" type="1">
<li>Explore <strong>model averaging</strong> techniques, such as Akaike weights or Bayesian Model Averaging, to account for model uncertainty. Model averaging yields more believable predictions and insights into the relative importance of different environmental variables.</li>
</ol>
</section>
</section>
<section id="reconciling-ecological-and-statistical-knowledge" class="level2">
<h2 class="anchored" data-anchor-id="reconciling-ecological-and-statistical-knowledge">Reconciling Ecological and Statistical Knowledge</h2>
<p>Achieving a model with optimal explanatory power requires you to reconcile your ecological knowledge with the statistical techniques at your disposal:</p>
<ol type="1">
<li><p><strong>Synthesise</strong> theoretical insights on environmental gradients and biological responses with statistical techniques to build defensible models.</p></li>
<li><p>Develop and <strong>test hypotheses</strong> using multiple regression models and consider both ecological relevance and statistical fit. Remain open to unexpected results that may challenge existing ideas.</p></li>
<li><p>Extend analysis to <strong>multivariate methods</strong> (for example constrained ordinations for gradient detection and attribution, and clustering for group identification) to uncover more complex ecological patterns.</p></li>
</ol>
<section id="new-concepts-2" class="level3 unnumbered">
<h3 class="unnumbered anchored" data-anchor-id="new-concepts-2">New Concepts</h3>
<ol start="4" type="1">
<li>Consider modern techniques like <strong>Joint Species Distribution Models (JSDMs)</strong> to simultaneously model multiple species and environmental factors.</li>
</ol>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<strong>Joint Species Distribution Models (JSDMs)</strong>
</div>
</div>
<div class="callout-body-container callout-body">
<p>Joint Species Distribution Models would be a great addition to the seaweed community-level analysis. JSDMs simultaneously fit distributions for multiple species and account for residual correlations among species after environmental effects are removed. For the seaweed dataset, JSDMs should be able to offer the following:</p>
<ul>
<li>Separate environmental filtering from potential biotic interactions</li>
<li>Improve predictions for rare species by borrowing strength from common species</li>
<li>Identify species associations that persist across environmental gradients</li>
<li>Provide more defensible estimates of environmental effects by accounting for community-level patterns</li>
</ul>
<p>The <strong>Hmsc</strong> R package provides hierarchical JSDMs that can handle various response types and spatial/temporal structures.</p>
</div>
</div>
<ol start="5" type="1">
<li>Begin identifying the most influential species using approaches such as <strong>multivariate abundance analysis with Generalised Linear Models (GLMs)</strong>, which offers a different approach to “Model-based Multivariate Analyses”.</li>
</ol>
</section>
</section>
<section id="model-validation" class="level2">
<h2 class="anchored" data-anchor-id="model-validation">Model Validation</h2>
<section id="cross-validation" class="level3">
<h3 class="anchored" data-anchor-id="cross-validation">Cross-Validation</h3>
<p>The purpose of model building is to develop models that can generalise well to new data (<em>i.e.</em>, they must be able to <strong>predict</strong>), and to do so, we need to carefully validate our model’s performance.</p>
<p>Traditional random cross-validation often inadequately represents the predictive challenges in ecological datasets. <strong>Spatial blocking cross-validation</strong> could be used when dealing with autocorrelated data. This involves partitioning data into spatially contiguous blocks rather than random subsets to better mimic real-world prediction scenarios where models must extrapolate to new locations or time periods.</p>
<p>For the seaweed data, something I might try is to implement <strong>leave-one-site-out cross-validation</strong> or <strong>geographic blocking</strong> that respects the spatial structure of the South African coastline. This should provide realistic estimates of model performance when predicting to unsampled coastal areas and help identify models that really capture generalisable ecological patterns versus those that exploit spatial autocorrelation.</p>
</section>
<section id="model-averaging-and-ensemble-methods" class="level3">
<h3 class="anchored" data-anchor-id="model-averaging-and-ensemble-methods">Model Averaging and Ensemble Methods</h3>
<p>Model averaging techniques takes our modelling beyond simply selecting single “best” models. One may use <strong>Akaike weight-based averaging</strong> or <strong>Bayesian Model Averaging</strong> to account for model uncertainty. These methods combine insights from multiple (different) models if they offer similar support from the data; averaging predictions across these models often provides more defensible forecasts than relying on any single model.</p>
<p>For seaweed community analysis, ensemble approaches combining multiple statistical techniques (GLMs, GAMs, machine learning methods) can capture different aspects of species-environment relationships. <strong>Boosted Regression Trees (BRTs)</strong> are particularly effective for ecological modelling as they handle non-linear relationships, variable interactions, and mixed data types while providing interpretable variable importance measures.</p>
</section>
</section>
<section id="practical-steps-for-model-selection" class="level2">
<h2 class="anchored" data-anchor-id="practical-steps-for-model-selection">Practical Steps for Model Selection</h2>
<ol type="1">
<li><p>Use ecological knowledge to select relevant environmental predictors, and consider both direct and indirect effects on seaweed physiology and ecology.</p></li>
<li><p>Evaluate the strength and pattern of spatial gradients using regression, mapping techniques, and spatial statistics.</p></li>
<li><p>Refine your models by address multicollinearity and other data issues using appropriate statistical methods. Consider interaction terms and non-linear relationships where ecologically justified.</p></li>
<li><p>Validate models using both ecological and statistical criteria to ensure parsimony and ecological meaning. Employ techniques like cross-validation or bootstrapping to assess model robustness.</p></li>
<li><p>Present findings accessibly to both ecologists and statisticians. This requires that you emphasise the biological significance of your models alongside their statistical performance.</p></li>
</ol>
</section>
<section id="some-advanced-ideas" class="level2">
<h2 class="anchored" data-anchor-id="some-advanced-ideas">Some Advanced Ideas</h2>
<p>(To be developed)</p>
<section id="hierarchical-and-multi-scale-modeling" class="level3">
<h3 class="anchored" data-anchor-id="hierarchical-and-multi-scale-modeling">Hierarchical and Multi-Scale Modeling</h3>
<p>Hierarchical Bayesian models may be used with ecological data with multiple sources of variation. For seaweed communities, implement hierarchical structures that account for:</p>
<ul>
<li>Site-level random effects (unmeasured local conditions)</li>
<li>Region-level variation (biogeographic differences)</li>
<li>Species-level random effects (unmeasured species traits)</li>
<li>Temporal variation (if data spans multiple years)</li>
</ul>
<p>It allows borrowing strength across hierarchical levels while maintaining appropriate uncertainty estimates.</p>
</section>
<section id="multi-scale-environmental-predictors" class="level3">
<h3 class="anchored" data-anchor-id="multi-scale-environmental-predictors">Multi-Scale Environmental Predictors</h3>
<p>One can build scale-explicit models that incorporate environmental predictors measured at multiple spatial and temporal scales. For coastal systems, this might include:</p>
<ul>
<li>Fine-scale habitat characteristics (substrate, local topography)</li>
<li>Intermediate-scale oceanographic features (upwelling cells, current boundaries)</li>
<li>Large-scale climate patterns (ENSO, Indian Ocean Dipole)</li>
</ul>
</section>
<section id="advanced-methods-and-approaches" class="level3">
<h3 class="anchored" data-anchor-id="advanced-methods-and-approaches">Advanced Methods and Approaches</h3>
<section id="null-models-and-randomisation-tests" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="null-models-and-randomisation-tests">Null Models and Randomisation Tests</h4>
<p>Null models provide baselines for interpreting ecological patterns. Implement randomisation tests to determine whether observed species associations, environmental relationships, or spatial patterns exceed those expected by chance alone. This is useful when dealing with sparse ecological datasets where spurious patterns can easily emerge.</p>
</section>
<section id="variable-transformation-and-standardisation" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="variable-transformation-and-standardisation">Variable Transformation and Standardisation</h4>
<p>Develop structures and principled methods around data transformation that maintain ecological interpretability. For species composition data, consider:</p>
<ul>
<li>Hellinger transformation for abundance data to reduce the influence of dominant species</li>
<li>Chord transformation for presence-absence data</li>
<li>Species-specific standardisation that accounts for different detection probabilities</li>
</ul>
</section>
<section id="model-interpretation-and-communication" class="level4 unnumbered">
<h4 class="unnumbered anchored" data-anchor-id="model-interpretation-and-communication">Model Interpretation and Communication</h4>
<p><strong><em>Uncertainty Quantification</em></strong></p>
<p>Stop relying on point estimates but favour comprehensive uncertainty assessment. For example, implement:</p>
<ul>
<li>Bootstrap confidence intervals for parameter estimates</li>
<li>Prediction intervals for spatial forecasts</li>
<li>Model selection uncertainty through information criteria weights</li>
<li>Structural uncertainty through ensemble approaches</li>
</ul>
<p><strong><em>Residual Analysis for Model Diagnostics</em></strong></p>
<p>Develop full residual analysis workflows specific to ecological data. For multivariate community data, examine:</p>
<ul>
<li>Spatial autocorrelation in residuals using Moran’s I</li>
<li>Temporal autocorrelation for time series data</li>
<li>Heteroscedasticity across environmental gradients</li>
<li>Outlier detection and influence analysis</li>
</ul>
<p><strong><em>Software</em></strong></p>
<p>There are many R packages that can do almost any conceivable analysis&gt; At the very least, know what they are and what they can do.</p>
<p><strong><em>Reproducible Workflows</em></strong></p>
<p>As always, emphasise reproducible research practices, especially as one starts to build on more advanced statistical methods involving larger teams of people. This includes:</p>
<ul>
<li>Version control such as GitHub for code and data</li>
<li>Documented model selection procedures</li>
<li>Apply standardised validation protocols</li>
<li>Clear reporting of model assumptions and limitations</li>
</ul>


<!-- -->

</section>
</section>
</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-reuse"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a></div></div></section><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{smit,_a._j.2024,
  author = {Smit, A. J.,},
  title = {Ecological {Model} {Building}},
  date = {2024-07-06},
  url = {http://tangledbank.netlify.app/BCB743/model_building.html},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-smit,_a._j.2024" class="csl-entry quarto-appendix-citeas" role="listitem">
Smit, A. J. (2024) Ecological Model Building. <a href="http://tangledbank.netlify.app/BCB743/model_building.html">http://tangledbank.netlify.app/BCB743/model_building.html.
</a>
</div></div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("http:\/\/tangledbank\.netlify\.app");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
            // default icon
            link.classList.add("external");
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../BCB743/cluster_analysis.html" class="pagination-link" aria-label="14: Cluster Analysis">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">14: Cluster Analysis</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../BCB743/deep_dive.html" class="pagination-link" aria-label="16a: Deep Dive Into Gradients">
        <span class="nav-page-text">16a: Deep Dive Into Gradients</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb1" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb1-1"><a href="#cb1-1"></a><span class="co">---</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="an">title:</span><span class="co"> "Ecological Model Building"</span></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="an">date:</span><span class="co"> "2024-07-06"</span></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="an">format:</span></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="co">  html: default</span></span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="co">  typst:</span></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="co">    toc: true</span></span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="co">    section-numbering: 1.1.a</span></span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="co">    columns: 1</span></span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="co">---</span></span>
<span id="cb1-11"><a href="#cb1-11"></a></span>
<span id="cb1-12"><a href="#cb1-12"></a>Developing ecological models, whether multiple regression models or constrained ordinations with numerous environmental predictors, requires a synthesis of biological and ecological theory (the substance of your undergraduate education) with defensible statistical analysis to achieve an integrated, acceptable outcome.</span>
<span id="cb1-13"><a href="#cb1-13"></a></span>
<span id="cb1-14"><a href="#cb1-14"></a>I shall guide you through the process of selecting the most appropriate model to explain ecological outcomes, using the dataset concerning seaweed species composition ($\beta$-diversity) along the South African coastline as our exemple. You should be well-acquainted with this analysis by now. The principles we explore are broad applicability across various regions and scales.</span>
<span id="cb1-15"><a href="#cb1-15"></a></span>
<span id="cb1-16"><a href="#cb1-16"></a><span class="fu">## About the Seaweed Data</span></span>
<span id="cb1-17"><a href="#cb1-17"></a></span>
<span id="cb1-18"><a href="#cb1-18"></a>In my examples (<span class="co">[</span><span class="ot">Gradients Example</span><span class="co">](deep_dive.qmd)</span>, <span class="co">[</span><span class="ot">Multiple Regression</span><span class="co">](https://tangledbank.netlify.app/docs/The-Biostatistics-Book-Ch-3-5.pdf)</span> and <span class="co">[</span><span class="ot">db-RDA</span><span class="co">](constrained_ordination.qmd)</span>), I employ environmental variables derived from an extensive daily time series of seawater temperature. These data were transformed into various statistics that capture components of the thermal regime along the South African coastline:</span>
<span id="cb1-19"><a href="#cb1-19"></a></span>
<span id="cb1-20"><a href="#cb1-20"></a><span class="ss">- </span>the annual mean climatology (<span class="in">`annMean`</span>)</span>
<span id="cb1-21"><a href="#cb1-21"></a><span class="ss">- </span>the climatological mean for February (warmest month) and August (coldest month) (<span class="in">`febMean`</span> and <span class="in">`augMean`</span>)</span>
<span id="cb1-22"><a href="#cb1-22"></a><span class="ss">- </span>the climatological SD for those months (<span class="in">`febSD`</span> and <span class="in">`augSD`</span>)</span>
<span id="cb1-23"><a href="#cb1-23"></a><span class="ss">- </span>the temperature range (from daily climatology) for February and August (<span class="in">`febRange`</span> and <span class="in">`augRange`</span>)</span>
<span id="cb1-24"><a href="#cb1-24"></a></span>
<span id="cb1-25"><a href="#cb1-25"></a>Each variable was calculated as Euclidean distances between site pairs, thus establishing a foundation for $\beta$-diversity assessment. To this end, I apply the Sørensen dissimilarity to the corresponding species data between those site pairs. Given that these summary statistics derive from the same initial dataset, issues of non-independence among predictor data become inevitable. Multicollinearity is virtually guaranteed (even theoretical considerations make this apparent). Nevertheless, these data retain utility for understanding whether seaweed flora composition responds primarily to mean annual temperature, minimum values, or maximum values. The analysis might further reveal which aspects of temperature variability along the coast exert greater influence in structuring species composition.</span>
<span id="cb1-26"><a href="#cb1-26"></a></span>
<span id="cb1-27"><a href="#cb1-27"></a>I have also incorporated geographical distance (recalculated as Euclidean distance) between site pairs along the coast, which thus corresponds to the same spatial grid as the $\beta$-diversity and environmental distance data. Here lies a conceptual snare: does distance function as a genuine predictor, or merely serve as a convenient descriptor of the spatial domain across which species gradients unfold? The same question arises for <span class="in">`bio`</span>, the bioregional classification of the seaweed flora established by Professor John Bolton.</span>
<span id="cb1-28"><a href="#cb1-28"></a></span>
<span id="cb1-29"><a href="#cb1-29"></a><span class="fu">## Theoretical Understanding of Environmental Drivers</span></span>
<span id="cb1-30"><a href="#cb1-30"></a></span>
<span id="cb1-31"><a href="#cb1-31"></a>Begin by examining how specific environmental variables (*e.g.*, <span class="in">`dist`</span>, <span class="in">`bio`</span>, <span class="in">`augMean`</span>, <span class="in">`febRange`</span>, <span class="in">`febSD`</span>, <span class="in">`augSD`</span>, <span class="in">`annMean`</span>) might influence seaweed community structure along the coast. Temperature is a fundamental driver, affecting biological processes and impacting species differentially. When selecting relevant predictor variables, consider how temperature metrics (such as the mean values, fluctuations, and ranges) influence reproductive timing, growth rates, and physiological tolerances. Develop your thinking from your ecological understanding of these principles. You'll find that some variables may lack theoretical importance and warrant removal on conceptual grounds before modelling commences.</span>
<span id="cb1-32"><a href="#cb1-32"></a></span>
<span id="cb1-33"><a href="#cb1-33"></a><span class="fu">### New Concepts {.unnumbered}</span></span>
<span id="cb1-34"><a href="#cb1-34"></a></span>
<span id="cb1-35"><a href="#cb1-35"></a><span class="ss">1. </span>Where such data exist, consider incorporating **trait-based approaches** into your analysis. Functional traits of seaweeds (for example, thallus morphology, photosynthetic pigments, or reproductive strategies) may respond differentially to environmental gradients. This approach can illuminate mechanisms driving community composition beyond simple species presence/absence data. While I have not yet explored this avenue, I anticipate pursuing it as a future research project, probably through **Fourth-corner analysis** or **RLQ ordination**.</span>
<span id="cb1-36"><a href="#cb1-36"></a></span>
<span id="cb1-37"><a href="#cb1-37"></a>::: callout-tip</span>
<span id="cb1-38"><a href="#cb1-38"></a><span class="fu">## **Fourth-corner analysis**</span></span>
<span id="cb1-39"><a href="#cb1-39"></a></span>
<span id="cb1-40"><a href="#cb1-40"></a>Fourth-corner analysis addresses questions about which species traits respond to which environmental gradients. The method derives its name from the "fourth corner" of a conceptual data matrix linking three data tables:</span>
<span id="cb1-41"><a href="#cb1-41"></a></span>
<span id="cb1-42"><a href="#cb1-42"></a><span class="ss">- </span>R: **Site-by-species abundance matrix** (the species table with the community data)</span>
<span id="cb1-43"><a href="#cb1-43"></a><span class="ss">- </span>L: **Site-by-environment matrix** (the temperature variables, distance, bioregions, *etc.*, in the environmental table)</span>
<span id="cb1-44"><a href="#cb1-44"></a><span class="ss">- </span>Q: **Species-by-traits matrix** (morphological, physiological, or life-history characteristics)</span>
<span id="cb1-45"><a href="#cb1-45"></a></span>
<span id="cb1-46"><a href="#cb1-46"></a>The "fourth corner" is the missing direct link between traits and environment, which the analysis infers through the observed community patterns. The method tests whether some trait-environment combinations occur more or less frequently than expected by chance, given the observed species distributions. The statistical underpinning relies on permutation tests to assess significance, typically using three different null models:</span>
<span id="cb1-47"><a href="#cb1-47"></a></span>
<span id="cb1-48"><a href="#cb1-48"></a><span class="ss">- </span>Permuting sites (tests for environmental filtering)</span>
<span id="cb1-49"><a href="#cb1-49"></a><span class="ss">- </span>Permuting species (tests for trait convergence)</span>
<span id="cb1-50"><a href="#cb1-50"></a><span class="ss">- </span>Permuting both simultaneously (combines both ecological processes)</span>
<span id="cb1-51"><a href="#cb1-51"></a>:::</span>
<span id="cb1-52"><a href="#cb1-52"></a></span>
<span id="cb1-53"><a href="#cb1-53"></a>::: callout-tip</span>
<span id="cb1-54"><a href="#cb1-54"></a><span class="fu">## **RLQ ordination**</span></span>
<span id="cb1-55"><a href="#cb1-55"></a></span>
<span id="cb1-56"><a href="#cb1-56"></a>RLQ ordination extends the fourth-corner concept into multivariate space and provides an integrated view of trait-environment relationships. It does a double ordination that maximises the covariance between environmental variables (table **R × L**) and species traits (table **L × Q**), with the species composition table (**L**) bringing in the new concept.</span>
<span id="cb1-57"><a href="#cb1-57"></a></span>
<span id="cb1-58"><a href="#cb1-58"></a>Mathematically, RLQ finds linear combinations of environmental variables and linear combinations of traits that produce maximum correlation when projected through the species composition data. It reveals the main gradients along which traits and environmental conditions co-vary. The technique essentially performs a **co-inertia analysis** between two indirect ordinations:</span>
<span id="cb1-59"><a href="#cb1-59"></a></span>
<span id="cb1-60"><a href="#cb1-60"></a><span class="ss">- </span>Environmental variables weighted by species abundances</span>
<span id="cb1-61"><a href="#cb1-61"></a><span class="ss">- </span>Species traits weighted by their occurrence across sites</span>
<span id="cb1-62"><a href="#cb1-62"></a>:::</span>
<span id="cb1-63"><a href="#cb1-63"></a></span>
<span id="cb1-64"><a href="#cb1-64"></a><span class="ss">2. </span>Beyond environmental variables and spatial gradients, incorporating **phylogenetic methods** can yield deeper insights into factors influencing ecological outcomes. Phylogenetic approaches account for evolutionary relationships among species. It might reveal patterns and relationships that complement species composition and trait-based analyses, and can reveal whether community assembly is driven by environmental filtering (closely related species co-occurring) or competitive exclusion (distantly related species co-occurring). For marine seaweeds, phylogenetic structure can indicate evolutionary constraints on environmental tolerances, niche conservatism vs. adaptive radiation, and historical biogeographic processes.</span>
<span id="cb1-65"><a href="#cb1-65"></a></span>
<span id="cb1-66"><a href="#cb1-66"></a><span class="fu">## Identifying Spatial Gradients</span></span>
<span id="cb1-67"><a href="#cb1-67"></a></span>
<span id="cb1-68"><a href="#cb1-68"></a>Assess whether your variables exhibit strong spatial gradients or differences. Consider these examples:</span>
<span id="cb1-69"><a href="#cb1-69"></a></span>
<span id="cb1-70"><a href="#cb1-70"></a><span class="ss">1. </span>The **annual mean temperature (`annMean`)** integrates data from warm and cold seasons to serve as an integrated predictor of global ecosystems. Regionally, it may function as a significant driver due to coastal temperature gradients, though potential collinearity with other variables requires examination.</span>
<span id="cb1-71"><a href="#cb1-71"></a></span>
<span id="cb1-72"><a href="#cb1-72"></a><span class="ss">2. </span>The **mean temperature of the warmest month (`febMean`)** displays a clear gradient from the east coast to Cape Point, but remains relatively stable along the west coast. Variability in this region is captured by <span class="in">`febSD`</span>.</span>
<span id="cb1-73"><a href="#cb1-73"></a></span>
<span id="cb1-74"><a href="#cb1-74"></a><span class="ss">3. </span>The **temperature range of the warmest month (`febRange`)** differentiates the Benguela Current from the Agulhas Current. It exhibits both east-west and north-south gradients.</span>
<span id="cb1-75"><a href="#cb1-75"></a></span>
<span id="cb1-76"><a href="#cb1-76"></a><span class="ss">4. </span>Looking at **temperature variability** such as during the coldest and warmest monsts, <span class="in">`augSD`</span> and <span class="in">`febSD`</span>, respectively, give geographically-linked explanations, maybe showing how to upwelling intensity or current stability affect species structural patterns.</span>
<span id="cb1-77"><a href="#cb1-77"></a></span>
<span id="cb1-78"><a href="#cb1-78"></a><span class="ss">5. </span>Employ **unconstrained ordinations** with environmental vectors (<span class="in">`envfit()`</span>) to guide selection of important structuring predictors.</span>
<span id="cb1-79"><a href="#cb1-79"></a></span>
<span id="cb1-80"><a href="#cb1-80"></a><span class="fu">### New Concept {.unnumbered}</span></span>
<span id="cb1-81"><a href="#cb1-81"></a></span>
<span id="cb1-82"><a href="#cb1-82"></a><span class="ss">6. </span>Consider incorporating specific **oceanographic features** into your analysis. Ideas that come to mind include upwelling intensity, current velocity, or nutrient availability. These factors can markedly influence seaweed distribution and may enhance the explanatory power of your models. A more advanced approach would be to use **oceanographic models** to derive spatially explicit variables that capture the dynamic nature of coastal environments. For example, Langrangian models will allow one to track the directionality of water movement and its influence on species dispersal and connectivity. </span>
<span id="cb1-83"><a href="#cb1-83"></a></span>
<span id="cb1-84"><a href="#cb1-84"></a>::: callout-tip</span>
<span id="cb1-85"><a href="#cb1-85"></a><span class="fu">## **Lagrangian Models**</span></span>
<span id="cb1-86"><a href="#cb1-86"></a></span>
<span id="cb1-87"><a href="#cb1-87"></a>Current patterns, upwelling dynamics, and water mass movements may be more relevant than Euclidean distance for understanding seaweed community patterns. Lagrangian oceanographic variables (tracking water movement) will outperform static Eulerian (as used in the present seaweed analysis) measures for marine species distributions.</span>
<span id="cb1-88"><a href="#cb1-88"></a>:::</span>
<span id="cb1-89"><a href="#cb1-89"></a></span>
<span id="cb1-90"><a href="#cb1-90"></a><span class="fu">## Assessing Environmental Gradients</span></span>
<span id="cb1-91"><a href="#cb1-91"></a></span>
<span id="cb1-92"><a href="#cb1-92"></a>Environmental variables with strong spatial gradients likely exert the most significant impact on seaweeds, indicating plausible environmental filtering (niche mechanisms). To quantify these gradients:</span>
<span id="cb1-93"><a href="#cb1-93"></a></span>
<span id="cb1-94"><a href="#cb1-94"></a><span class="ss">1. </span>Conduct **multiple linear regressions** using continuous predictor variables as functions of dist (distance between site pairs).</span>
<span id="cb1-95"><a href="#cb1-95"></a></span>
<span id="cb1-96"><a href="#cb1-96"></a><span class="ss">2. </span>Create **thematic maps** (spatially implicit) of temperature variables and vary symbol size or colour intensity by magnitude. Maybe use GIS tools to interpolate values between sampling points for more comprehensive visualisation.</span>
<span id="cb1-97"><a href="#cb1-97"></a></span>
<span id="cb1-98"><a href="#cb1-98"></a><span class="ss">3. </span>Assess **spatial autocorrelation** in your variables using techniques such as Moran's I or Geary's C. This helps identify the scale at which environmental factors operate and informs model structure.</span>
<span id="cb1-99"><a href="#cb1-99"></a></span>
<span id="cb1-100"><a href="#cb1-100"></a><span class="fu">## Model Building and Variable Selection</span></span>
<span id="cb1-101"><a href="#cb1-101"></a></span>
<span id="cb1-102"><a href="#cb1-102"></a>Throughout the model-building process, make informed decisions about variable selection based on both theoretical knowledge and data-driven approaches:</span>
<span id="cb1-103"><a href="#cb1-103"></a></span>
<span id="cb1-104"><a href="#cb1-104"></a><span class="ss">1. </span>Select variables based on **ecological understanding** of species' responses to environmental drivers, considering both direct and indirect effects.</span>
<span id="cb1-105"><a href="#cb1-105"></a></span>
<span id="cb1-106"><a href="#cb1-106"></a><span class="ss">2. </span>Choose variables that reflect **significant known environmental gradients** influenced by factors such as ocean currents, coastal topography, and climate patterns.</span>
<span id="cb1-107"><a href="#cb1-107"></a></span>
<span id="cb1-108"><a href="#cb1-108"></a><span class="ss">3. </span>Use **unconstrained ordinations** (*e.g.*, Principal Component Analysis, PCA) to explore the relationships between environmental variables and species composition. This can help identify key gradients and inform variable selection. Running such ordinations side-by-side on the species and environmental data can reveal how well the environmental variables explain the observed species patterns. This method becomes especially useful when you superimpose the environmental vectors onto the ordination plot using <span class="in">`envfit()`</span> or a GAM or something similar.</span>
<span id="cb1-109"><a href="#cb1-109"></a></span>
<span id="cb1-110"><a href="#cb1-110"></a><span class="ss">4. </span>Also use **data-driven decision making**. Employ statistical methods such as Variance Inflation Factors (VIFs) or forward selection (*e.g.*, <span class="in">`stepAIC()`</span>) to address multicollinearity and refine model selection. Consider modern techniques like elastic net regression or random forests for variable importance ranking.</span>
<span id="cb1-111"><a href="#cb1-111"></a></span>
<span id="cb1-112"><a href="#cb1-112"></a><span class="fu">### New Concepts {.unnumbered}</span></span>
<span id="cb1-113"><a href="#cb1-113"></a></span>
<span id="cb1-114"><a href="#cb1-114"></a><span class="ss">5. </span>Consider also spatial autocorrelation in the models. Beyond simply identifying spatial autocorrelation, one might develop explicit strategies for incorporating it into models. One such is **Moran's Eigenvector Maps (MEMs)**, which can be used to account for spatial structure in species-environment relationships. MEMs are derived from distance matrices and can be included as predictors in regression models to capture spatial patterns.</span>
<span id="cb1-115"><a href="#cb1-115"></a></span>
<span id="cb1-116"><a href="#cb1-116"></a>::: callout-tip</span>
<span id="cb1-117"><a href="#cb1-117"></a><span class="fu">## **Moran's Eigenvector Maps (MEMs)**</span></span>
<span id="cb1-118"><a href="#cb1-118"></a></span>
<span id="cb1-119"><a href="#cb1-119"></a>Moran’s Eigenvector Maps (MEMs) can decompose spatial patterns at multiple scales. It allows one to partition variation into broad-scale environmental trends and fine-scale spatial processes. I have already done this for the seaweed analysis and you can read about it in the paper. For coastal seaweed data, it can reveal spatially explicit account for:</span>
<span id="cb1-120"><a href="#cb1-120"></a></span>
<span id="cb1-121"><a href="#cb1-121"></a><span class="ss">- </span>Oceanographic connectivity between sites</span>
<span id="cb1-122"><a href="#cb1-122"></a><span class="ss">- </span>Distance-decay relationships in species similarity</span>
<span id="cb1-123"><a href="#cb1-123"></a><span class="ss">- </span>Scale-dependent environmental effects (local vs. regional processes)</span>
<span id="cb1-124"><a href="#cb1-124"></a>:::</span>
<span id="cb1-125"><a href="#cb1-125"></a></span>
<span id="cb1-126"><a href="#cb1-126"></a><span class="ss">6. </span>Explore **model averaging** techniques, such as Akaike weights or Bayesian Model Averaging, to account for model uncertainty. Model averaging yields more believable predictions and insights into the relative importance of different environmental variables.</span>
<span id="cb1-127"><a href="#cb1-127"></a></span>
<span id="cb1-128"><a href="#cb1-128"></a><span class="fu">## Reconciling Ecological and Statistical Knowledge</span></span>
<span id="cb1-129"><a href="#cb1-129"></a></span>
<span id="cb1-130"><a href="#cb1-130"></a>Achieving a model with optimal explanatory power requires you to reconcile your ecological knowledge with the statistical techniques at your disposal:</span>
<span id="cb1-131"><a href="#cb1-131"></a></span>
<span id="cb1-132"><a href="#cb1-132"></a><span class="ss">1. </span>**Synthesise** theoretical insights on environmental gradients and biological responses with statistical techniques to build defensible models.</span>
<span id="cb1-133"><a href="#cb1-133"></a></span>
<span id="cb1-134"><a href="#cb1-134"></a><span class="ss">2. </span>Develop and **test hypotheses** using multiple regression models and consider both ecological relevance and statistical fit. Remain open to unexpected results that may challenge existing ideas.</span>
<span id="cb1-135"><a href="#cb1-135"></a></span>
<span id="cb1-136"><a href="#cb1-136"></a><span class="ss">3. </span>Extend analysis to **multivariate methods** (for example constrained ordinations for gradient detection and attribution, and clustering for group identification) to uncover more complex ecological patterns. </span>
<span id="cb1-137"><a href="#cb1-137"></a></span>
<span id="cb1-138"><a href="#cb1-138"></a><span class="fu">### New Concepts {.unnumbered}</span></span>
<span id="cb1-139"><a href="#cb1-139"></a></span>
<span id="cb1-140"><a href="#cb1-140"></a><span class="ss">4. </span>Consider modern techniques like **Joint Species Distribution Models (JSDMs)** to simultaneously model multiple species and environmental factors.</span>
<span id="cb1-141"><a href="#cb1-141"></a></span>
<span id="cb1-142"><a href="#cb1-142"></a>::: callout-tip</span>
<span id="cb1-143"><a href="#cb1-143"></a><span class="fu">## **Joint Species Distribution Models (JSDMs)**</span></span>
<span id="cb1-144"><a href="#cb1-144"></a></span>
<span id="cb1-145"><a href="#cb1-145"></a>Joint Species Distribution Models would be a great addition to the seaweed community-level analysis. JSDMs simultaneously fit distributions for multiple species and account for residual correlations among species after environmental effects are removed. For the seaweed dataset, JSDMs should be able to offer the following:</span>
<span id="cb1-146"><a href="#cb1-146"></a></span>
<span id="cb1-147"><a href="#cb1-147"></a><span class="ss">- </span>Separate environmental filtering from potential biotic interactions</span>
<span id="cb1-148"><a href="#cb1-148"></a><span class="ss">- </span>Improve predictions for rare species by borrowing strength from common species</span>
<span id="cb1-149"><a href="#cb1-149"></a><span class="ss">- </span>Identify species associations that persist across environmental gradients</span>
<span id="cb1-150"><a href="#cb1-150"></a><span class="ss">- </span>Provide more defensible estimates of environmental effects by accounting for community-level patterns</span>
<span id="cb1-151"><a href="#cb1-151"></a></span>
<span id="cb1-152"><a href="#cb1-152"></a>The **Hmsc** R package provides hierarchical JSDMs that can handle various response types and spatial/temporal structures.</span>
<span id="cb1-153"><a href="#cb1-153"></a>:::</span>
<span id="cb1-154"><a href="#cb1-154"></a></span>
<span id="cb1-155"><a href="#cb1-155"></a><span class="ss">5. </span>Begin identifying the most influential species using approaches such as **multivariate abundance analysis with Generalised Linear Models (GLMs)**, which offers a different approach to "Model-based Multivariate Analyses".</span>
<span id="cb1-156"><a href="#cb1-156"></a></span>
<span id="cb1-157"><a href="#cb1-157"></a><span class="fu">## Model Validation</span></span>
<span id="cb1-158"><a href="#cb1-158"></a></span>
<span id="cb1-159"><a href="#cb1-159"></a><span class="fu">### Cross-Validation</span></span>
<span id="cb1-160"><a href="#cb1-160"></a></span>
<span id="cb1-161"><a href="#cb1-161"></a>The purpose of model building is to develop models that can generalise well to new data (*i.e.*, they must be able to **predict**), and to do so, we need to carefully validate our model's performance. </span>
<span id="cb1-162"><a href="#cb1-162"></a></span>
<span id="cb1-163"><a href="#cb1-163"></a>Traditional random cross-validation often inadequately represents the predictive challenges in ecological datasets. **Spatial blocking cross-validation** could be used when dealing with autocorrelated data. This involves partitioning data into spatially contiguous blocks rather than random subsets to better mimic real-world prediction scenarios where models must extrapolate to new locations or time periods.</span>
<span id="cb1-164"><a href="#cb1-164"></a></span>
<span id="cb1-165"><a href="#cb1-165"></a>For the seaweed data, something I might try is to implement **leave-one-site-out cross-validation** or **geographic blocking** that respects the spatial structure of the South African coastline. This should provide realistic estimates of model performance when predicting to unsampled coastal areas and help identify models that really capture generalisable ecological patterns versus those that exploit spatial autocorrelation.</span>
<span id="cb1-166"><a href="#cb1-166"></a></span>
<span id="cb1-167"><a href="#cb1-167"></a><span class="fu">### Model Averaging and Ensemble Methods</span></span>
<span id="cb1-168"><a href="#cb1-168"></a></span>
<span id="cb1-169"><a href="#cb1-169"></a>Model averaging techniques takes our modelling beyond simply selecting single “best” models. One may use **Akaike weight-based averaging** or **Bayesian Model Averaging** to account for model uncertainty. These methods combine insights from multiple (different) models if they offer similar support from the data; averaging predictions across these models often provides more defensible forecasts than relying on any single model.</span>
<span id="cb1-170"><a href="#cb1-170"></a></span>
<span id="cb1-171"><a href="#cb1-171"></a>For seaweed community analysis, ensemble approaches combining multiple statistical techniques (GLMs, GAMs, machine learning methods) can capture different aspects of species-environment relationships. **Boosted Regression Trees (BRTs)** are particularly effective for ecological modelling as they handle non-linear relationships, variable interactions, and mixed data types while providing interpretable variable importance measures.</span>
<span id="cb1-172"><a href="#cb1-172"></a></span>
<span id="cb1-173"><a href="#cb1-173"></a><span class="fu">## Practical Steps for Model Selection</span></span>
<span id="cb1-174"><a href="#cb1-174"></a></span>
<span id="cb1-175"><a href="#cb1-175"></a><span class="ss">1. </span>Use ecological knowledge to select relevant environmental predictors, and consider both direct and indirect effects on seaweed physiology and ecology.</span>
<span id="cb1-176"><a href="#cb1-176"></a></span>
<span id="cb1-177"><a href="#cb1-177"></a><span class="ss">2. </span>Evaluate the strength and pattern of spatial gradients using regression, mapping techniques, and spatial statistics.</span>
<span id="cb1-178"><a href="#cb1-178"></a></span>
<span id="cb1-179"><a href="#cb1-179"></a><span class="ss">3. </span>Refine your models by address multicollinearity and other data issues using appropriate statistical methods. Consider interaction terms and non-linear relationships where ecologically justified.</span>
<span id="cb1-180"><a href="#cb1-180"></a></span>
<span id="cb1-181"><a href="#cb1-181"></a><span class="ss">4. </span>Validate models using both ecological and statistical criteria to ensure parsimony and ecological meaning. Employ techniques like cross-validation or bootstrapping to assess model robustness.</span>
<span id="cb1-182"><a href="#cb1-182"></a></span>
<span id="cb1-183"><a href="#cb1-183"></a><span class="ss">5. </span>Present findings accessibly to both ecologists and statisticians. This requires that you emphasise the biological significance of your models alongside their statistical performance.</span>
<span id="cb1-184"><a href="#cb1-184"></a></span>
<span id="cb1-185"><a href="#cb1-185"></a><span class="fu">## Some Advanced Ideas</span></span>
<span id="cb1-186"><a href="#cb1-186"></a></span>
<span id="cb1-187"><a href="#cb1-187"></a>(To be developed)</span>
<span id="cb1-188"><a href="#cb1-188"></a></span>
<span id="cb1-189"><a href="#cb1-189"></a><span class="fu">### Hierarchical and Multi-Scale Modeling</span></span>
<span id="cb1-190"><a href="#cb1-190"></a></span>
<span id="cb1-191"><a href="#cb1-191"></a>Hierarchical Bayesian models may be used with ecological data with multiple sources of variation. For seaweed communities, implement hierarchical structures that account for:</span>
<span id="cb1-192"><a href="#cb1-192"></a></span>
<span id="cb1-193"><a href="#cb1-193"></a><span class="ss">- </span>Site-level random effects (unmeasured local conditions)</span>
<span id="cb1-194"><a href="#cb1-194"></a><span class="ss">- </span>Region-level variation (biogeographic differences)</span>
<span id="cb1-195"><a href="#cb1-195"></a><span class="ss">- </span>Species-level random effects (unmeasured species traits)</span>
<span id="cb1-196"><a href="#cb1-196"></a><span class="ss">- </span>Temporal variation (if data spans multiple years)</span>
<span id="cb1-197"><a href="#cb1-197"></a></span>
<span id="cb1-198"><a href="#cb1-198"></a>It allows borrowing strength across hierarchical levels while maintaining appropriate uncertainty estimates.</span>
<span id="cb1-199"><a href="#cb1-199"></a></span>
<span id="cb1-200"><a href="#cb1-200"></a><span class="fu">### Multi-Scale Environmental Predictors</span></span>
<span id="cb1-201"><a href="#cb1-201"></a></span>
<span id="cb1-202"><a href="#cb1-202"></a>One can build scale-explicit models that incorporate environmental predictors measured at multiple spatial and temporal scales. For coastal systems, this might include:</span>
<span id="cb1-203"><a href="#cb1-203"></a></span>
<span id="cb1-204"><a href="#cb1-204"></a><span class="ss">- </span>Fine-scale habitat characteristics (substrate, local topography)</span>
<span id="cb1-205"><a href="#cb1-205"></a><span class="ss">- </span>Intermediate-scale oceanographic features (upwelling cells, current boundaries)</span>
<span id="cb1-206"><a href="#cb1-206"></a><span class="ss">- </span>Large-scale climate patterns (ENSO, Indian Ocean Dipole)</span>
<span id="cb1-207"><a href="#cb1-207"></a></span>
<span id="cb1-208"><a href="#cb1-208"></a><span class="fu">### Advanced Methods and Approaches</span></span>
<span id="cb1-209"><a href="#cb1-209"></a></span>
<span id="cb1-210"><a href="#cb1-210"></a><span class="fu">#### Null Models and Randomisation Tests {.unnumbered}</span></span>
<span id="cb1-211"><a href="#cb1-211"></a></span>
<span id="cb1-212"><a href="#cb1-212"></a>Null models provide baselines for interpreting ecological patterns. Implement randomisation tests to determine whether observed species associations, environmental relationships, or spatial patterns exceed those expected by chance alone. This is useful when dealing with sparse ecological datasets where spurious patterns can easily emerge.</span>
<span id="cb1-213"><a href="#cb1-213"></a></span>
<span id="cb1-214"><a href="#cb1-214"></a><span class="fu">#### Variable Transformation and Standardisation {.unnumbered}</span></span>
<span id="cb1-215"><a href="#cb1-215"></a></span>
<span id="cb1-216"><a href="#cb1-216"></a>Develop structures and principled methods around data transformation that maintain ecological interpretability. For species composition data, consider:</span>
<span id="cb1-217"><a href="#cb1-217"></a></span>
<span id="cb1-218"><a href="#cb1-218"></a><span class="ss">- </span>Hellinger transformation for abundance data to reduce the influence of dominant species</span>
<span id="cb1-219"><a href="#cb1-219"></a><span class="ss">- </span>Chord transformation for presence-absence data</span>
<span id="cb1-220"><a href="#cb1-220"></a><span class="ss">- </span>Species-specific standardisation that accounts for different detection probabilities</span>
<span id="cb1-221"><a href="#cb1-221"></a></span>
<span id="cb1-222"><a href="#cb1-222"></a><span class="fu">#### Model Interpretation and Communication {.unnumbered}</span></span>
<span id="cb1-223"><a href="#cb1-223"></a></span>
<span id="cb1-224"><a href="#cb1-224"></a>***Uncertainty Quantification***</span>
<span id="cb1-225"><a href="#cb1-225"></a></span>
<span id="cb1-226"><a href="#cb1-226"></a>Stop relying on point estimates but favour comprehensive uncertainty assessment. For example, implement:</span>
<span id="cb1-227"><a href="#cb1-227"></a></span>
<span id="cb1-228"><a href="#cb1-228"></a><span class="ss">- </span>Bootstrap confidence intervals for parameter estimates</span>
<span id="cb1-229"><a href="#cb1-229"></a><span class="ss">- </span>Prediction intervals for spatial forecasts</span>
<span id="cb1-230"><a href="#cb1-230"></a><span class="ss">- </span>Model selection uncertainty through information criteria weights</span>
<span id="cb1-231"><a href="#cb1-231"></a><span class="ss">- </span>Structural uncertainty through ensemble approaches</span>
<span id="cb1-232"><a href="#cb1-232"></a></span>
<span id="cb1-233"><a href="#cb1-233"></a>***Residual Analysis for Model Diagnostics***</span>
<span id="cb1-234"><a href="#cb1-234"></a></span>
<span id="cb1-235"><a href="#cb1-235"></a>Develop full residual analysis workflows specific to ecological data. For multivariate community data, examine:</span>
<span id="cb1-236"><a href="#cb1-236"></a></span>
<span id="cb1-237"><a href="#cb1-237"></a><span class="ss">- </span>Spatial autocorrelation in residuals using Moran’s I</span>
<span id="cb1-238"><a href="#cb1-238"></a><span class="ss">- </span>Temporal autocorrelation for time series data</span>
<span id="cb1-239"><a href="#cb1-239"></a><span class="ss">- </span>Heteroscedasticity across environmental gradients</span>
<span id="cb1-240"><a href="#cb1-240"></a><span class="ss">- </span>Outlier detection and influence analysis</span>
<span id="cb1-241"><a href="#cb1-241"></a></span>
<span id="cb1-242"><a href="#cb1-242"></a>***Software***</span>
<span id="cb1-243"><a href="#cb1-243"></a></span>
<span id="cb1-244"><a href="#cb1-244"></a>There are many R packages that can do almost any conceivable analysis&gt; At the very least, know what they are and what they can do.</span>
<span id="cb1-245"><a href="#cb1-245"></a></span>
<span id="cb1-246"><a href="#cb1-246"></a>***Reproducible Workflows***</span>
<span id="cb1-247"><a href="#cb1-247"></a></span>
<span id="cb1-248"><a href="#cb1-248"></a>As always, emphasise reproducible research practices, especially as one starts to build on more advanced statistical methods involving larger teams of people. This includes:</span>
<span id="cb1-249"><a href="#cb1-249"></a></span>
<span id="cb1-250"><a href="#cb1-250"></a><span class="ss">- </span>Version control such as GitHub for code and data</span>
<span id="cb1-251"><a href="#cb1-251"></a><span class="ss">- </span>Documented model selection procedures</span>
<span id="cb1-252"><a href="#cb1-252"></a><span class="ss">- </span>Apply standardised validation protocols</span>
<span id="cb1-253"><a href="#cb1-253"></a><span class="ss">- </span>Clear reporting of model assumptions and limitations</span></code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>© 2025, AJ Smit</p>
</div>   
    <div class="nav-footer-center">

<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB743/model_building.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
<p><span class="faux-block">Made with <i class="fa-brands fa-r-project" aria-label="r-project"></i> and <a href="https://quarto.org/">Quarto</a></span><br> <span class="faux-block"><a href="?var:github-url">View the source at <i class="fa-brands fa-github" aria-label="github"></i> GitHub</a></span></p>
</div>
  </div>
</footer>




<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>