<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.26">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Smit, A. J.">
<meta name="dcterms.date" content="2021-01-01">

<title>Principal Component Analysis (PCA) – The Tangled Bank</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../BCB743/PCA_examples.html" rel="next">
<link href="../BCB743/ordination.html" rel="prev">
<link href="../images/favicon.ico" rel="icon">
<script src="../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-597958c53c93a607afca12fd375c57ed.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-37d9842ca13b32c9c58adfc4ed5d9451.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-SCYH68M489"></script>

<script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-SCYH68M489', { 'anonymize_ip': true});
</script>

<script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"headline",
  "consent_type":"express",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  ,
"language":"en"
  });
});
</script> 
  
<style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN" && texText && texText.data) {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../styles/styles.css">
<meta property="og:title" content="Principal Component Analysis (PCA) – The Tangled Bank">
<meta property="og:description" content="UWC’s Biodiversity &amp; Conservation Biology Department Pages For BDC223, BDC334, BCB743, and BCB744.">
<meta property="og:image" content="http://tangledbank.netlify.app/BCB743/data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==">
<meta property="og:site_name" content="The Tangled Bank">
</head>

<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">The Tangled Bank</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-vignettes" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Vignettes</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-vignettes">    
        <li>
    <a class="dropdown-item" href="../vignettes/netCDF_dates.html">
 <span class="dropdown-text">Dates From netCDF Files: Two Approaches</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/prep_NOAA_OISST.html">
 <span class="dropdown-text">Downloading and Prep: ERDDAP</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_ERDDAP.html">
 <span class="dropdown-text">Retrieving Chlorophyll-a Data from ERDDAP Servers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/alt_method.html">
 <span class="dropdown-text">Downloading and Prep: Python + CDO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/gridded_data.html">
 <span class="dropdown-text">Detecting Events in Gridded Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/regridding.html">
 <span class="dropdown-text">Regridding gridded data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/MHW_MCS_horizonplots.html">
 <span class="dropdown-text">Displaying MHWs and MCSs as Horizon Plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/heatwaveR_issues.html">
 <span class="dropdown-text">heatwaveR Issues</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_sightings.html">
 <span class="dropdown-text">Plotting the Whale Sightings and Chlorophyll-<em>a</em> Concentrations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/chl_localisation.html">
 <span class="dropdown-text">Spatial Localisation, Subsetting, and Aggregation of the Chlorophyll-<em>a</em> Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/elem_ts_methods.html">
 <span class="dropdown-text">Wavelet Analysis of Diatom Time Series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/buffer_data_extract.html">
 <span class="dropdown-text">Extracting gridded data within a buffer</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-high-performance-computing" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">High-Performance Computing</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-high-performance-computing">    
        <li>
    <a class="dropdown-item" href="../vignettes/README_Lengau.html">
 <span class="dropdown-text">Using Lengau</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/PBSPro_users.html">
 <span class="dropdown-text">PBSPro user and job management</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../vignettes/README_tmux.html">
 <span class="dropdown-text">Using tmux</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../blog.html"> 
<span class="menu-text">Tangled Bank Blog</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-blogroll" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Blogroll</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-blogroll">    
        <li>
    <a class="dropdown-item" href="https://www.r-bloggers.com">
 <span class="dropdown-text">R-bloggers</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item compact">
    <a class="nav-link" href="../blog.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">    
        <li>
    <a class="dropdown-item" href="https://github.com/ajsmit/tangled_bank">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BCB743/BCB743_index.html">BCB743: Quantitative Ecology</a></li><li class="breadcrumb-item"><a href="../BCB743/PCA.html">8a: Principal Components Analysis</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="../index.html" class="sidebar-logo-link">
      <img src="../images/my_tree_hex.png" alt="" class="sidebar-logo light-content py-0 d-lg-inline d-none">
      <img src="../images/my_tree_hex.png" alt="" class="sidebar-logo dark-content py-0 d-lg-inline d-none">
      </a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">UNDERGRADUATE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC223: Plant Ecophysiology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/BDC223_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L00-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introductory Lecture</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L01-worldmapper.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: the Limits to Life</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L02-SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2: Surface Area to Volume (SA:V) Ratio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L03-climate.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3: Earth’s Climate</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L04-carbon_cycle.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Climate Change (Carbon Cycle)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L05-light.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 5: Light</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L06-pigments_photosynthesis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6: Pigments &amp; Photosynthesis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L07-chromatic_adaptation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 7: Chromatic Adaptation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L08a-nutrient_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8a: Nutrient Uptake</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L08b-nutrients_michaelis_menten.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8b: Nutrient Uptake</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/L09-plant_stress.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3: Plant Stress</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab1_SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1: Surface Area to Volume (SA/V) Ratios in Biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab2_misc_calcs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2: Miscellaneous Calculations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab3_PI_curves.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3: PI Curves — Jassby &amp; Platt</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/Lab4_nitrogen_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4: Uptake Kinetics — Michaelis-Menten</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC223/BDC223_FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC334: Biogeography &amp; Global Ecology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/BDC334_syllabus.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/BDC334-Lecture-Transcripts.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">BCB334 Lecture Transcript</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: Introductory Lecture</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-02-ecosystems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2. Overview of Ecosystems</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-03-gradients.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3. Ecological Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Biodiversity Concepts</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 5: Multivariate Data</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lec-06-unified-ecology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6: Unified Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 7: Threats to Biodiversity</span>
  </li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 8: Nature's Contribution to People</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../docs/IKS_Conservation.pdf" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 9: Indigenous Knowledge Systems and Conservation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1. Ecological Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-02a-r_rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2a. Introduction to R &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-02b-env_dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2b. Environmental Distance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-03-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3. Quantifying Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/Lab-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4. Species Distribution Patterns</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BDC334/BDC334_AI_Tutor.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">BCB334 AI Tutor</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../pages/Transboundary_systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transboundary Systems</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS CORE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false">
 <span class="menu-text">BCB744: Introduction to R, &amp; Biostatistics</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/BCB744_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/01-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. R &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/02-working-with-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Working with Data &amp; Code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/03-quarto.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. R Markdown and Quarto</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/04-data-in-R.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data Classes &amp; Structures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/05-workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. R Workflows</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/06-graphics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Graphics with <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/07-faceting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Faceting Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/08-brewing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Brewing Colours</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/09-mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Mapping with <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/10-mapping_style.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Mapping with Style</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/11-mapping_rnaturalearth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Mapping with Natural Earth</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/12-mapping_quakes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. the Fiji Earthquake Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/13-tidy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Tidy Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/14-tidier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Tidier Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/15-tidiest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15. Tidiest Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/intro_r/16-recap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16. Recap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/assessments/BCB744_Intro_R_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Intro R Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/assessments/BCB744_Intro_R_Example_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 Intro R Example 1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/assessments/BCB744_Intro_R_Example_2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 Intro R Example 2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../self-assessment_v1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Self-Assessment Form</span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
 <span class="menu-text">1. Introduction to Statistics</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/02-summarise-and-describe.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Data Summaries &amp; Descriptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/03-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Statistical Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/04-distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data Distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/05-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Statistical Inference</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/06-assumptions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Assumptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/07-t_tests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. <em>t</em>-Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/08-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. ANOVA</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/09-regressions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Simple Linear Regressions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/10-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/11-decision_guide.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/12-glance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Non-Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/13-confidence.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Confidence Intervals</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/basic_stats/14-transformations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Data Transformations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/assessments/BCB744_Biostatistics_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Biostatistics Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB744/assessments/BCB744_BioStats_Example_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 BioStats 2025 Example 1</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS ELECTIVES</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">BCB743: Quantitative Ecology</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/BCB743_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/review.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1-4: Review Concepts fo Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5: Correlations &amp; Associations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7: Intro to Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">8a: Principal Components Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA_examples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8b: PCA: Additional Examples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCA_SDG_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8c: PCA: WHO SDG Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/CA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9a: Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/DCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9b: Detrended Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/PCoA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10: Principal Coordinates Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/nMDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11a: Non-metric Multidimensional Scaling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/nMDS_diatoms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11b: nMDS: PERMANOVA (Diatoms) Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/unconstrained-summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12: Unconstrained Ordination Summary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/constrained_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13a: Distance-Based Redundancy Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/two_oceans_appendices.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13b: db-RDA: Seaweeds Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/cluster_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14: Cluster Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/model_building.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15: Model Building</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/deep_dive.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16a: Deep Dive Into Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../BCB743/multiple_regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16b: Multiple Regression</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">AI &amp; POLICY</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../docs/Combined Faculty AI Policy 15082025.pdf" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Faculty of Natural Sciences AI Policy</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../AI/BCB_AI_Workshop_Sep_2025.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB AI Workshop</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">SUPPORT</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Web Resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/general_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/ecology_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quantitative Ecology Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../resources/spatial_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial R Resources</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#geometric-projection-of-high-dimensional-data" id="toc-geometric-projection-of-high-dimensional-data" class="nav-link active" data-scroll-target="#geometric-projection-of-high-dimensional-data"><span class="header-section-number">1</span> Geometric Projection of High-Dimensional Data</a></li>
  <li><a href="#sec-horseshoe-effect" id="toc-sec-horseshoe-effect" class="nav-link" data-scroll-target="#sec-horseshoe-effect"><span class="header-section-number">2</span> The Horseshoe Effect</a></li>
  <li><a href="#set-up-the-analysis-environment" id="toc-set-up-the-analysis-environment" class="nav-link" data-scroll-target="#set-up-the-analysis-environment"><span class="header-section-number">3</span> Set-up the Analysis Environment</a></li>
  <li><a href="#the-doubs-river-data" id="toc-the-doubs-river-data" class="nav-link" data-scroll-target="#the-doubs-river-data"><span class="header-section-number">4</span> The Doubs River Data</a></li>
  <li><a href="#first-do-a-correlation" id="toc-first-do-a-correlation" class="nav-link" data-scroll-target="#first-do-a-correlation"><span class="header-section-number">5</span> First Do a Correlation</a></li>
  <li><a href="#see-the-spatial-context" id="toc-see-the-spatial-context" class="nav-link" data-scroll-target="#see-the-spatial-context"><span class="header-section-number">6</span> See the Spatial Context</a></li>
  <li><a href="#do-the-pca" id="toc-do-the-pca" class="nav-link" data-scroll-target="#do-the-pca"><span class="header-section-number">7</span> Do the PCA</a></li>
  <li><a href="#terminology" id="toc-terminology" class="nav-link" data-scroll-target="#terminology"><span class="header-section-number">8</span> Terminology</a></li>
  <li><a href="#how-many-axes-to-retain" id="toc-how-many-axes-to-retain" class="nav-link" data-scroll-target="#how-many-axes-to-retain"><span class="header-section-number">9</span> How Many Axes to Retain?</a></li>
  <li><a href="#ordination-diagrams" id="toc-ordination-diagrams" class="nav-link" data-scroll-target="#ordination-diagrams"><span class="header-section-number">10</span> Ordination Diagrams</a></li>
  <li><a href="#scaling" id="toc-scaling" class="nav-link" data-scroll-target="#scaling"><span class="header-section-number">11</span> Scaling</a></li>
  <li><a href="#fitting-environmental-response-surfaces" id="toc-fitting-environmental-response-surfaces" class="nav-link" data-scroll-target="#fitting-environmental-response-surfaces"><span class="header-section-number">12</span> Fitting Environmental Response Surfaces</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB743/PCA.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../BCB743/BCB743_index.html">BCB743: Quantitative Ecology</a></li><li class="breadcrumb-item"><a href="../BCB743/PCA.html">8a: Principal Components Analysis</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Principal Component Analysis (PCA)</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://tangledbank.netlify.app">Smit, A. J.</a> <a href="https://orcid.org/0000-0002-3799-6126" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uwc.ac.za">
            University of the Western Cape
            </a>
          </p>
      </div>
  </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 1, 2021</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Tip</span><strong>Material Required for This Chapter</strong>
</div>
</div>
<div class="callout-body-container callout-body">
<table class="table">
<thead>
<tr class="header">
<th>Type</th>
<th>Name</th>
<th>Link</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Theory</strong></td>
<td>Numerical Ecology with R</td>
<td>See pages 117-132</td>
</tr>
<tr class="even">
<td><strong>Slides</strong></td>
<td>PCA lecture slides</td>
<td><a href="../slides/BCB743_08_PCA.pdf">💾 <code>BCB743_08_PCA.pdf</code></a></td>
</tr>
<tr class="odd">
<td><strong>Data</strong></td>
<td>The Doubs River data</td>
<td><a href="../data/NEwR-2ed_code_data/NeWR2-Data/Doubs.RData">💾 <code>Doubs.RData</code></a></td>
</tr>
<tr class="even">
<td><strong>R function</strong></td>
<td>A function for ordination plots</td>
<td><a href="../data/NEwR-2ed_code_data/NeWR2-Functions/cleanplot.pca.R">💾 <code>cleanplot.pca.R</code></a></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="callout callout-style-simple callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Important</span>Tasks to Complete in This Chapter
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li><a href="../BCB743/tasks/Task_C.html">Task C 1–5</a></li>
</ul>
</div>
</div>
<div class="callout callout-style-simple callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Important</span>PCA as a Foundation for Understanding Ordination
</div>
</div>
<div class="callout-body-container callout-body">
<p>Thoroughly understanding PCA is a prerequisite for understanding more complex ordination techniques. Mastered this chapter before moving on to CA, PCoA, RDA, and CCA. Much of the same terminology and concepts are used in these techniques.</p>
</div>
</div>
<section id="geometric-projection-of-high-dimensional-data" class="level2" data-number="1">
<h2 data-number="1" class="anchored" data-anchor-id="geometric-projection-of-high-dimensional-data"><span class="header-section-number">1</span> Geometric Projection of High-Dimensional Data</h2>
<p>Ordination refers to a suite of multivariate techniques that reduce a high-dimensional dataset into a lower-dimensional space, typically 2D or 3D, in such a way that any intrinsic structure in the data forms visually-discernible patterns (Pielou, 1984). In ecology, ordination techniques are used to describe relationships between community structure patterns and underlying environmental gradients. They allow us to determine the relative importance of different gradients and visualise species-environment relationships.</p>
<p>Principal Component Analysis (PCA) is one of the commonly used ordination techniques in ecology. It is a dimensionality reduction technique that transforms the original set of variables into a new set of uncorrelated variables called principal components. PCA is performed on a data matrix containing species abundances or environmental variables across multiple sites or samples.</p>
<p>Although PCA is often introduced as a method for reducing dimensionality, it is more accurately understood as a geometric re-expression of the data. The method does not discard variables arbitrarily; rather, it rotates the original coordinate system so that new, mutually orthogonal (independent) axes are aligned with directions of maximal variance, and then projects the data onto these axes. This rotation-projection viewpoint will become explicit later when we examine how PCA is computed in R, but it is worth keeping in mind from the outset that PCA reorganises variation by changing the basis in which the data are represented.</p>
<p>The PCA process involves calculating the <strong>eigenvectors</strong> and <strong>eigenvalues</strong> of the covariance or correlation matrix of the data. The eigenvectors represent the directions of maximum variance in the data, and the corresponding eigenvalues represent the amount of variance explained by each eigenvector. The new axes, called principal components, are linear combinations of the original variables, ordered by the amount of variance they explain.</p>
<p>PCA preserves the Euclidean distances between samples in the original high-dimensional space when projecting them onto the lower-dimensional ordination space. This property makes PCA more suitable for analysing environmental data, where Euclidean distances between samples are meaningful and interpretable. However, for species data, which is often in the form of counts or frequencies, Euclidean distances may not be an appropriate measure of dissimilarity between samples.</p>
</section>
<section id="sec-horseshoe-effect" class="level2" data-number="2">
<h2 data-number="2" class="anchored" data-anchor-id="sec-horseshoe-effect"><span class="header-section-number">2</span> The Horseshoe Effect</h2>
<p>The <strong>horseshoe effect</strong> (sometimes called the Guttman effect) is an artefact often seen with PCA when applied to species data, especially when using species abundance data for communities along environmental gradients. It distorts the data points in the ordination space. A less severe version of the horseshoe effect is called the ‘arch effect’ and is seen in <a href="../BCB743/CA.html">Correspondence Analysis (CA)</a>.</p>
<p>The horseshoe effect occurs because PCA assumes linear relationships between variables, while species data often exhibit unimodal responses to environmental gradients. The unimodal model was discussed in <a href="../BDC334/L02a-gradients.html#sec-unimodal">BDC334</a>. When species have unimodal distributions along a gradient, PCA tends to fold the ends of the gradient towards each other.</p>
<p>From a geometric perspective, the horseshoe effect points to a mismatch between the structure of the data and the space into which PCA forces it. Species responding unimodally to an underlying gradient occupy a curved configuration in multivariate space. PCA, however, seeks a linear subspace that maximises variance under Euclidean geometry. In doing so, it approximates this curved structure with straight axes, causing the ends of the gradient to bend back toward one another in the reduced ordination. The horseshoe, which is visible on the ordination diagram, is therefore not a peculiarity of the data, but a consequence of representing a non-linear manifold using a linear projection.</p>
<p>This distortion can lead to several issues:</p>
<ol type="1">
<li>The horseshoe shape can make it appear that sites at opposite ends of the gradient are more similar than they really are.</li>
<li>The folding of the gradient ends compresses the data and potentially obscures important ecological patterns.</li>
<li>The second PCA axis—the most affected axis—often does not represent a meaningful ecological gradient, making interpretation challenging.</li>
<li>Unlike the arch effect in CA, the horseshoe effect in PCA can lead to incorrect ordering of samples along the gradient.</li>
</ol>
<p>These distortions are informative, because they reveal the limits of PCA’s linear and Euclidean assumptions when confronted with non-linear ecological structure. To address these issues, we prefer to use <a href="../BCB743/nMDS.html">non-metric Multidimensional Scaling (nMDS)</a> or a distance-based method (like <a href="../BCB743/PCoA.html">Principal Coordinates Analysis, PCoA</a>) that are not constrained to linear projections. Alternatively, we may use techniques specifically designed to handle unimodal species responses, such as <a href="../BCB743/CA.html">Correspondence Analysis (CA)</a>, which uses <span class="math inline">\(\chi^2\)</span>-distance and not Euclidean distance, or its detrended version (<a href="../BCB743/DCA.html">Detrended Correspondence Analysis, DCA</a>), but these come with their own set of considerations.</p>
<p>Let us work through a PCA to demonstrate the ideas.</p>
</section>
<section id="set-up-the-analysis-environment" class="level2" data-number="3">
<h2 data-number="3" class="anchored" data-anchor-id="set-up-the-analysis-environment"><span class="header-section-number">3</span> Set-up the Analysis Environment</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(vegan)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(ggcorrplot) <span class="co"># for the correlations</span></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(ggpubr)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="the-doubs-river-data" class="level2" data-number="4">
<h2 data-number="4" class="anchored" data-anchor-id="the-doubs-river-data"><span class="header-section-number">4</span> The Doubs River Data</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="fu">load</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"NEwR-2ed_code_data"</span>, <span class="st">"NEwR2-Data"</span>, <span class="st">"Doubs.RData"</span>))</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="fu">head</span>(env)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>   dfs ele  slo  dis  pH har  pho  nit  amm  oxy bod
1  0.3 934 48.0 0.84 7.9  45 0.01 0.20 0.00 12.2 2.7
2  2.2 932  3.0 1.00 8.0  40 0.02 0.20 0.10 10.3 1.9
3 10.2 914  3.7 1.80 8.3  52 0.05 0.22 0.05 10.5 3.5
4 18.5 854  3.2 2.53 8.0  72 0.10 0.21 0.00 11.0 1.3
5 21.5 849  2.3 2.64 8.1  84 0.38 0.52 0.20  8.0 6.2
6 32.4 846  3.2 2.86 7.9  60 0.20 0.15 0.00 10.2 5.3</code></pre>
</div>
</div>
</section>
<section id="first-do-a-correlation" class="level2" data-number="5">
<h2 data-number="5" class="anchored" data-anchor-id="first-do-a-correlation"><span class="header-section-number">5</span> First Do a Correlation</h2>
<div class="cell" data-layout-align="center">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># computing a correlation matrix</span></span>
<span id="cb4-2"><a href="#cb4-2"></a>corr <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">cor</span>(env), <span class="dv">1</span>)</span>
<span id="cb4-3"><a href="#cb4-3"></a></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="co"># visualisation of the correlation matrix</span></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="fu">ggcorrplot</span>(corr, <span class="at">type =</span> <span class="st">'upper'</span>, <span class="at">outline.col =</span> <span class="st">"white"</span>,</span>
<span id="cb4-6"><a href="#cb4-6"></a>           <span class="at">colors =</span> <span class="fu">c</span>(<span class="st">"#1679a1"</span>, <span class="st">"white"</span>, <span class="st">"#f8766d"</span>),</span>
<span id="cb4-7"><a href="#cb4-7"></a>           <span class="at">lab =</span> <span class="cn">TRUE</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="PCA_files/figure-html/code-pca-correlation-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Pairwise correlations among the environmental variables included with the Doubs River study."><img src="PCA_files/figure-html/code-pca-correlation-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" alt="Pairwise correlations among the environmental variables included with the Doubs River study."></a></p>
</figure>
</div>
<figcaption>Pairwise correlations among the environmental variables included with the Doubs River study.</figcaption>
</figure>
</div>
</div>
</div>
<p>Some variables are very correlated, and they might be omitted from the subsequent analyses. We say that these variables are <strong>collinear</strong>. Collinear variables cannot be teased apart in terms of finding out which one is most influential in structuring the community. There are more advanced ways to search for collinear variables (e.g., Variance Inflation Factors, VIF) and in this way we can systematically exclude them from the PCA. See <span class="citation" data-cites="graham2003confronting">Graham (<a href="#ref-graham2003confronting" role="doc-biblioref">2003</a>)</span> for a discussion on collinearity.</p>
<p>Here we will proceed with all the variables — this is a delibirate decision for my approach here. In PCA, collinearity does not cause instability in estimated coefficients in the way it does in multiple regression, because PCA does not estimate regression parameters. Instead, correlated variables jointly determine the orientation of the ordination axes, i.e., strongly collinear variables tend to load in similar directions and contribute collectively to the same components. While this can complicate interpretation (by making it harder to disentangle the influence of individual variables) it does not invalidate the construction of the axes themselves. For teaching purposes, retaining collinear variables makes the structure of the data explicit and allows us to see how shared variance is redistributed across components. As we shall see later, in the constrained analyses where coefficients are interpreted directly, collinearity becomes a more serious inferential (statistical) issue and is addressed explicitly through diagnostics such as variance inflation factors and variable selection.</p>
</section>
<section id="see-the-spatial-context" class="level2" data-number="6">
<h2 data-number="6" class="anchored" data-anchor-id="see-the-spatial-context"><span class="header-section-number">6</span> See the Spatial Context</h2>
<p>The patterns in the data and the correlations between them will make more sense if we can visualise a spatial context. Thankfully spatial data are available:</p>
<div class="cell" data-layout-align="center">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="fu">head</span>(spa)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>       X      Y
1 85.678 20.000
2 84.955 20.100
3 92.301 23.796
4 91.280 26.431
5 92.005 29.163
6 95.954 36.315</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="fu">ggplot</span>(spa, <span class="fu">aes</span>(<span class="at">x =</span> X, <span class="at">y =</span> Y, <span class="at">label =</span> <span class="fu">rownames</span>(spa))) <span class="sc">+</span></span>
<span id="cb7-2"><a href="#cb7-2"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb7-3"><a href="#cb7-3"></a>  <span class="fu">geom_label</span>(<span class="at">vjust =</span> <span class="dv">0</span>, <span class="at">nudge_y =</span> <span class="fl">0.5</span>, <span class="at">check_overlap =</span> <span class="cn">TRUE</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-pca-map1" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-map1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="PCA_files/figure-html/fig-pca-map1-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;1: The spatial configuration of the Doubs River sites."><img src="PCA_files/figure-html/fig-pca-map1-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-map1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: The spatial configuration of the Doubs River sites.
</figcaption>
</figure>
</div>
</div>
</div>
<p>These site numbers correspond approximately to the ones in <span class="citation" data-cites="verneaux1973cours">Verneaux (<a href="#ref-verneaux1973cours" role="doc-biblioref">1973</a>)</span> but some of the numbers may have been shifted slightly in the example Doubs dataset used here compared to how they were originally numbered in Verneaux’s thesis and subsequent publication. This should not affect the interpretation. We can also scale the symbol size by the magnitude of the environmental variables. Lets look at two pairs of variables that are strongly correlated with one-another:</p>
<div class="cell" data-layout-align="center">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="co"># We scale the data first so as to better represent the full</span></span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="co"># magnitude of all variables with a common symbol size</span></span>
<span id="cb8-3"><a href="#cb8-3"></a>env_std <span class="ot">&lt;-</span> <span class="fu">decostand</span>(env, <span class="at">method =</span> <span class="st">"standardize"</span>)</span>
<span id="cb8-4"><a href="#cb8-4"></a></span>
<span id="cb8-5"><a href="#cb8-5"></a><span class="co"># positive correlations</span></span>
<span id="cb8-6"><a href="#cb8-6"></a>plt1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(spa, <span class="fu">aes</span>(<span class="at">x =</span> X, <span class="at">y =</span> Y, <span class="at">label =</span> <span class="fu">rownames</span>(spa))) <span class="sc">+</span></span>
<span id="cb8-7"><a href="#cb8-7"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">"red"</span>, <span class="fu">aes</span>(<span class="at">size =</span> env_std<span class="sc">$</span>amm, <span class="at">shape =</span> <span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb8-8"><a href="#cb8-8"></a>  <span class="fu">geom_text</span>(<span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">nudge_y =</span> <span class="fl">0.5</span>, <span class="at">check_overlap =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb8-9"><a href="#cb8-9"></a>  <span class="fu">labs</span>(<span class="at">size =</span> <span class="st">"Magnitude"</span>, <span class="at">title =</span> <span class="st">"Ammonium concentration"</span>)</span>
<span id="cb8-10"><a href="#cb8-10"></a></span>
<span id="cb8-11"><a href="#cb8-11"></a>plt2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(spa, <span class="fu">aes</span>(<span class="at">x =</span> X, <span class="at">y =</span> Y, <span class="at">label =</span> <span class="fu">rownames</span>(spa))) <span class="sc">+</span></span>
<span id="cb8-12"><a href="#cb8-12"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">"red"</span>, <span class="fu">aes</span>(<span class="at">size =</span> env_std<span class="sc">$</span>bod)) <span class="sc">+</span></span>
<span id="cb8-13"><a href="#cb8-13"></a>  <span class="fu">geom_text</span>(<span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">nudge_y =</span> <span class="fl">0.5</span>, <span class="at">check_overlap =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb8-14"><a href="#cb8-14"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Biological oxygen demand"</span>)</span>
<span id="cb8-15"><a href="#cb8-15"></a></span>
<span id="cb8-16"><a href="#cb8-16"></a><span class="co"># inverse correlations</span></span>
<span id="cb8-17"><a href="#cb8-17"></a>plt3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(spa, <span class="fu">aes</span>(<span class="at">x =</span> X, <span class="at">y =</span> Y, <span class="at">label =</span> <span class="fu">rownames</span>(spa))) <span class="sc">+</span></span>
<span id="cb8-18"><a href="#cb8-18"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">"blue"</span>, <span class="fu">aes</span>(<span class="at">size =</span> env_std<span class="sc">$</span>alt)) <span class="sc">+</span></span>
<span id="cb8-19"><a href="#cb8-19"></a>  <span class="fu">geom_text</span>(<span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">nudge_y =</span> <span class="fl">0.5</span>, <span class="at">check_overlap =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb8-20"><a href="#cb8-20"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Altitude"</span>)</span>
<span id="cb8-21"><a href="#cb8-21"></a></span>
<span id="cb8-22"><a href="#cb8-22"></a>plt4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(spa, <span class="fu">aes</span>(<span class="at">x =</span> X, <span class="at">y =</span> Y, <span class="at">label =</span> <span class="fu">rownames</span>(spa))) <span class="sc">+</span></span>
<span id="cb8-23"><a href="#cb8-23"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">"blue"</span>, <span class="fu">aes</span>(<span class="at">size =</span> env_std<span class="sc">$</span>flo)) <span class="sc">+</span></span>
<span id="cb8-24"><a href="#cb8-24"></a>  <span class="fu">geom_text</span>(<span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">nudge_y =</span> <span class="fl">0.5</span>, <span class="at">check_overlap =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb8-25"><a href="#cb8-25"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Flow rate"</span>)</span>
<span id="cb8-26"><a href="#cb8-26"></a></span>
<span id="cb8-27"><a href="#cb8-27"></a><span class="fu">ggarrange</span>(plt1, plt2, plt3, plt4, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb8-28"><a href="#cb8-28"></a>          <span class="at">common.legend =</span> <span class="cn">TRUE</span>, <span class="at">labels =</span> <span class="st">"AUTO"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-pca-map2" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-map2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="PCA_files/figure-html/fig-pca-map2-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;2: Four different representations of the site configuration (spatial context) of the Doubs River sampling layout. Symbols are scaled relative to A) ammonium concentration, B) BOD, C) Altitude, and D) Flow rate."><img src="PCA_files/figure-html/fig-pca-map2-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-map2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Four different representations of the site configuration (spatial context) of the Doubs River sampling layout. Symbols are scaled relative to A) ammonium concentration, B) BOD, C) Altitude, and D) Flow rate.
</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="do-the-pca" class="level2" data-number="7">
<h2 data-number="7" class="anchored" data-anchor-id="do-the-pca"><span class="header-section-number">7</span> Do the PCA</h2>
<p>We use the function <code>rda()</code> to do the PCA, but it can also be performed in base R with the functions <code>prcomp()</code> and <code>princomp()</code>. <code>rda()</code> is the same function that we will use later for a Redundancy Analysis, but when used without specifying constraints (as we do here) it amounts to simply doing a PCA. Typically we standardise environmental data to unit variance, but the PCA done by the <code>rda()</code> function accomplishes this step automagically when <code>scale = TRUE</code>. When applied to environmental data (as we typically do with a PCA) it works with correlations amongst the scaled variables. PCA preserves Euclidean distance and the relationships detected are linear, and for this reason it is not typically applied to species data without suitable transformations. In fact, in this module we will seldom apply a PCA to species data at all.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>env_pca <span class="ot">&lt;-</span> <span class="fu">rda</span>(env, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-2"><a href="#cb9-2"></a>env_pca</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call: rda(X = env, scale = TRUE)

              Inertia Rank
Total              11     
Unconstrained      11   11

Inertia is correlations

Eigenvalues for unconstrained axes:
  PC1   PC2   PC3   PC4   PC5   PC6   PC7   PC8   PC9  PC10  PC11 
5.969 2.164 1.065 0.739 0.400 0.336 0.173 0.108 0.024 0.017 0.006 </code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="co"># same ...</span></span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="co"># env_std &lt;- scale(env)</span></span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="co"># env_pca &lt;- rda(env_std, scale = FALSE)</span></span>
<span id="cb11-4"><a href="#cb11-4"></a><span class="co"># env_pca</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>In ordination we use the term <strong>inertia</strong> as a synonym for ‘variation’, but some PCA software (such as R’s <code>prcomp()</code> and <code>princomp()</code>) simply uses the term <code>sdev</code> for standard deviations. In PCA, when we use a correlation matrix (as we do here), the inertia is the sum of the diagonal values of the correlation matrix, which is simply the number of variables (11 in this example). When a PCA uses a covariance matrix the inertia is the sum of the variances of the variables.</p>
<p>You will also see in the output the mention of the term <strong>unconstrained</strong>. In a PCA the analysis is always unconstrained (i.e., not influenced by some <em>a priori</em> defined variables we hypothesise to explain the between site patterns in the multivariate data).</p>
<p>The section headed <strong>Eigenvalues for unconstrained axes</strong> shows the relative importance of the resultant reduced axes, and they can be used to determine the proportion of the total inertia (sum of the eigenvalues) captured by any one of the axes. They can be accessed with the function <code>eigenvals()</code> (the preferred function; see <code>?rda</code> for help), but an alternative method is given below. The first eigenvalue (the one associated with PC1) always explains the most variation (the largest fraction), and each subsequent one explains the largest proportion of the <strong>remaining</strong> variance. We say the axes are <strong>orthogonal</strong> and ranked in decreasing order of importance. The sum of all eigenvalues is the total inertia, so collectively they theoretically can explain all of the variation in the dataset (but clearly they <strong>should not</strong> be used to explain <strong>all</strong> the variance).</p>
<p>At this point it is important to distinguish between statistical adequacy and ecological sufficiency. A principal component that explains a large proportion of the variance is effective at compressing the data, but this does not by necessity imply that it represents the most influential ecological process. Variance explained is a measure of how much information is retained under dimensional reduction; it is not a measure of causal dominance or biological importance. Ecologically meaningful gradients may account for relatively little variance, while strong variance components may reflect broad-scale structure, measurement scale, or correlated responses rather than interpretable mechanisms. Here is where your knowledge as an ecologist will (or should) guide you.</p>
<p>To extract the first eigenvalue we can do:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="fu">round</span>(<span class="fu">eigenvals</span>(env_pca)[<span class="dv">1</span>], <span class="dv">3</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>  PC1 
5.969 </code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="co"># or</span></span>
<span id="cb14-2"><a href="#cb14-2"></a></span>
<span id="cb14-3"><a href="#cb14-3"></a><span class="fu">round</span>(env_pca<span class="sc">$</span>CA<span class="sc">$</span>eig[<span class="dv">1</span>], <span class="dv">3</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>  PC1 
5.969 </code></pre>
</div>
</div>
<p>The total inertia is:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="fu">sum</span>(<span class="fu">eigenvals</span>(env_pca))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 11</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a><span class="co"># or</span></span>
<span id="cb18-2"><a href="#cb18-2"></a></span>
<span id="cb18-3"><a href="#cb18-3"></a><span class="fu">sum</span>(env_pca<span class="sc">$</span>CA<span class="sc">$</span>eig)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 11</code></pre>
</div>
</div>
<p>So the proportion of variation explained by the first PC is:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="fu">round</span>(env_pca<span class="sc">$</span>CA<span class="sc">$</span>eig[<span class="dv">1</span>] <span class="sc">/</span> <span class="fu">sum</span>(env_pca<span class="sc">$</span>CA<span class="sc">$</span>eig) <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>) <span class="co"># result in %</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code> PC1 
54.3 </code></pre>
</div>
</div>
<p>We can show the same information as part of a more verbose summary. Here we see the pre-calculated <strong>Proportion Explained</strong> and <strong>Cumulative Proportion</strong> (it should be obvious what this is). There is also an assortment of other information, viz.&nbsp;<strong>Scaling 2 for species and site scores</strong>, <strong>Species scores</strong>, and <strong>Site scores</strong>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a><span class="fu">summary</span>(env_pca)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
rda(X = env, scale = TRUE) 

Partitioning of correlations:
              Inertia Proportion
Total              11          1
Unconstrained      11          1

Eigenvalues, and their contribution to the correlations 

Importance of components:
                         PC1    PC2     PC3     PC4     PC5     PC6     PC7
Eigenvalue            5.9687 2.1639 1.06517 0.73875 0.40019 0.33563 0.17263
Proportion Explained  0.5426 0.1967 0.09683 0.06716 0.03638 0.03051 0.01569
Cumulative Proportion 0.5426 0.7393 0.83616 0.90332 0.93970 0.97022 0.98591
                           PC8      PC9     PC10     PC11
Eigenvalue            0.108228 0.023701 0.017083 0.005983
Proportion Explained  0.009839 0.002155 0.001553 0.000544
Cumulative Proportion 0.995748 0.997903 0.999456 1.000000</code></pre>
</div>
</div>
<p>Newer versions of the ordinations’ <code>summary()</code> method no longer shows the species and site scores. To access these scores, we use the <code>scores()</code> function:</p>
<p>The species scores can be extracted from the ordination object by:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="co"># extract species scores for first three PCA axes:</span></span>
<span id="cb24-2"><a href="#cb24-2"></a><span class="fu">scores</span>(env_pca, <span class="at">display =</span> <span class="st">"species"</span>, <span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>            PC1        PC2         PC3
dfs  1.08657429  0.5342463 -0.27333233
ele -1.04396094 -0.6147742  0.20711738
slo -0.57702931 -0.4892811 -0.63490306
dis  0.95843365  0.6608150 -0.32456247
pH  -0.06364143  0.4629373  1.01316786
har  0.90118054  0.5849982  0.06448776
pho  1.05820606 -0.6013982  0.13865996
nit  1.15013343 -0.1004908 -0.05166769
amm  1.00678966 -0.6969288  0.14076555
oxy -0.97458581  0.4990886 -0.09017454
bod  0.97315389 -0.7148470  0.15145231
attr(,"const")
[1] 4.226177</code></pre>
</div>
</div>
<p>Similarly, for the site scores, do:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a><span class="co"># extract the site scores for axes 1 and 2</span></span>
<span id="cb26-2"><a href="#cb26-2"></a><span class="fu">scores</span>(env_pca, <span class="at">display =</span> <span class="st">"site"</span>, <span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>) <span class="sc">|&gt;</span> </span>
<span id="cb26-3"><a href="#cb26-3"></a>  <span class="fu">head</span>() <span class="co"># truncate to save space in this example</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>         PC1        PC2         PC3
1 -1.4127388 -1.4009788 -2.03483870
2 -1.0372473 -0.7795535  0.24400009
3 -0.9450700 -0.4676536  1.25042488
4 -0.8737116 -0.2698849  0.19304045
5 -0.4208759 -0.6694396  0.83190665
6 -0.7722358 -0.7206662 -0.07357441</code></pre>
</div>
</div>
<p>The <code>scores()</code> method also applies to other ordination methods.</p>
<div class="callout callout-style-simple callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Important</span>Question
</div>
</div>
<div class="callout-body-container callout-body">
<p>We are dealing with environmental variables here, so why do we call them “species scores”?</p>
</div>
</div>
</section>
<section id="terminology" class="level2" data-number="8">
<h2 data-number="8" class="anchored" data-anchor-id="terminology"><span class="header-section-number">8</span> Terminology</h2>
<p>The geometric ideas introduced earlier (rotation of axes and projection of data into a reduced space) are formalised in ordination methods through a singular-value decomposition (SVD). In <strong>vegan</strong>, ordination functions such as <code>rda()</code> for PCA/RDA and <code>cca()</code> for CA/CCA implement this decomposition directly.</p>
<p>Once the data matrix <span class="math inline">\(X\)</span> has been centred (and, where necessary, scaled or <span class="math inline">\(\chi^2\)</span>-transformed), the SVD computes:</p>
<p><span class="math display">\[X = U \,\Sigma \,V^{\mathsf T},\]</span></p>
<p>with orthonormal <em>left-singular vectors</em> <span class="math inline">\(U\)</span>, orthonormal <em>right-singular vectors</em> <span class="math inline">\(V\)</span>, and positive singular values on the diagonal of <span class="math inline">\(\Sigma\)</span>. From here, all the components of our ordination can be extracted. How do these relate to our data?</p>
<ul>
<li><strong>Eigenvectors</strong> are the directions in the original variable space along which the variance in the data is maximised. They relate to our original data in two ways:
<ul>
<li><span class="math inline">\(V\)</span> is the matrix of right singular vectors; they are the <em>eigenvectors of the columns</em> (species or variables). The quantities can be accessed by <code>env_pca$CA$v</code>.</li>
<li><span class="math inline">\(U\)</span> is the matrix of left singular vectors; <em>i.e.</em>, the <em>eigenvectors of the rows</em> (sites). The left singular vectors can be accessed by <code>env_pca$CA$u</code></li>
</ul></li>
<li><span class="math inline">\(\Sigma\)</span> is a diagonal matrix of singular values <span class="math inline">\(d\)</span>, and the <strong>eigenvalues</strong> are the square of these singular values, <span class="math inline">\(eig=d^ 2\)</span>. Eigenvalues represent the amount of variance explained by each corresponding eigenvector (or principal component), represented in PCA (and other unconstrained ordinations such as CA) by <code>env_pca$CA$eig</code>. Higher eigenvalues indicate that the corresponding principal component explains a larger portion of the total variance in the data. The sum of all eigenvalues is equal to the total variance in the data.</li>
</ul>
<p>Let us map these directly to our <code>env_pca</code> object, which I will assume was created with the <code>rda()</code> function. Looking inside the <code>env_pca</code> object, we will see a structure with a component named <span class="math inline">\(CA\)</span> or <span class="math inline">\(CCA\)</span>. This holds the results for the constrained/canonical analysis. For a simple PCA, all the results are in <span class="math inline">\(CA\)</span>. The eigenvectors (<code>env_pca$CA$eig</code>) define the direction of the new axes, they are not the final coordinates for ordination plots. <em>Projection</em> is the process of mapping the original data points onto the new reduced axes (principal components). What we plot are actually these <em>scores</em>.</p>
<ul>
<li><strong>Scores</strong> are the coordinates of the original sites and species projected onto the new ordination axes. They are calculated by scaling the eigenvectors. These scores are what we actually plot to create a biplot or triplot.
<ul>
<li>The <strong>Site Scores</strong> tell us where each site (sample) is located along the new ordination axes. Sites that are close together in the ordination plot have similar variable (or species) compositions. We obtain the site scores as <code>scores(env_pca, display = "site", ...)</code>.</li>
<li>The <strong>Species Scores</strong> (more accurately, <strong>variable scores</strong> in this context) describe how the columns of the data matrix, whatever they represent ecologically, relate to the ordination axes. Sometimes we call the species scores <em>loadings</em>. In a PCA biplot, these are represented by arrows (but in CA not, as that reduced space is no longer linear). An arrow pointing strongly towards PCA1 means that species or variable is a major contributor to that axis of variation. So, species scores represent the contributions of the original variables to the principal components. They indicate the strength and direction of the correlation between the original variables and the new principal components. Species scores are obtained with <code>scores(env_pca, display = "species", ...)</code>. Larger positive values indicate a stronger positive correlation, while larger negative values indicate a stronger negative correlation. It is worth considering the terminology “species scores”. The label “species scores” is structural since it inherited from a general ordination framework in which columns of the data matrix often correspond to species abundances. In PCA applied to environmental data, those columns contain variables, not taxa. The software does not “know” what the columns represent, so it treats them as abstract descriptors whose interpretation is supplied entirely by the analyst. Confusing function names with ontological claims may be a common source of error in multivariate analysis, and it is therefore important to read ordination output as a mathematical (or conceptual, if math isn’t your strong point) representation first, and only subsequently as an ecological one.</li>
</ul></li>
</ul>
<p>Here is the direct mapping:</p>
<table class="table">
<thead>
<tr class="header">
<th style="text-align: left;">Mathematical Concept</th>
<th style="text-align: left;"><code>vegan</code> Term</th>
<th style="text-align: left;">Where to Find It in the <code>rda</code> Object</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><strong>Eigenvalues</strong></td>
<td style="text-align: left;">Eigenvalues</td>
<td style="text-align: left;"><code>env_pca$CA$eig</code></td>
<td style="text-align: left;">The amount of variance explained by each axis (component). <code>summary(env_pca)</code> shows this clearly.</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong>Eigenvectors of Variables</strong></td>
<td style="text-align: left;">Species Scores</td>
<td style="text-align: left;"><code>env_pca$CA$v</code></td>
<td style="text-align: left;">The raw eigenvectors for the columns of the data (the variables). These are scaled to become the “species scores” that you plot as arrows.</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong>Eigenvectors of Sites</strong></td>
<td style="text-align: left;">Site Scores</td>
<td style="text-align: left;"><code>env_pca$CA$u</code></td>
<td style="text-align: left;">The raw eigenvectors for the rows of the data (the sites). These are scaled to become the “site scores” that you plot as points.</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong>Plottable Coordinates</strong></td>
<td style="text-align: left;">Scores</td>
<td style="text-align: left;"><code>scores(env_pca)</code></td>
<td style="text-align: left;">The function <code>scores()</code> extracts and properly scales the raw eigenvectors (<code>$u</code> and <code>$v</code>) to give the coordinates ready for plotting.</td>
</tr>
</tbody>
</table>
<p>When interpreting a PCA biplot (see below), the species scores (loadings) indicate the contributions of the original variables to the principal components, while the site scores represent the positions of the sites in the reduced-dimensional space defined by those principal components. The direction and length of the species score vectors (arrows) provide information about the relationships between the original variables and the principal components, while the positions of the site points reflect the similarities or differences between sites based on the environmental gradients represented by the principal components.</p>
<!-- It is customary to plot the Site and Species Scores as ordination diagrams called **biplots** for a few of the reduced axes. We will get to this below. -->
<!-- **Scaling 1** and **Scaling 2**, depending on what was specified in the `rda()` function call, are useful for whether one wants to interpret species (scaling 1) or variables (scaling 2). When calling Scaling 1, the distances between points plotted on the ordination diagram (i.e. the information in the rows, which in this case is sites) will retain their Euclidean distances, which allows for better interpretation of how sites relate to one-another. Calling Scaling 2 preserves more accurately the angles between variables (the columns, i.e. here we have environmental variables) with the consequence that in the biplot smaller angles between variable vectors will reflect stronger correlations. More on scaling below. -->
</section>
<section id="how-many-axes-to-retain" class="level2" data-number="9">
<h2 data-number="9" class="anchored" data-anchor-id="how-many-axes-to-retain"><span class="header-section-number">9</span> How Many Axes to Retain?</h2>
<p>The number of axes to retain is a difficult question to answer. The first few axes will always explain the most variation, but how do we know how many reduced axes are influential and should be kept? Commonly recommended is the <strong>broken stick</strong> method—keep the principal components whose eigenvalues are higher than corresponding random broken stick components:</p>
<div class="cell" data-layout-align="center">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a><span class="co"># make a scree plot using the vegan function:</span></span>
<span id="cb28-2"><a href="#cb28-2"></a><span class="fu">screeplot</span>(env_pca, <span class="at">bstick =</span> <span class="cn">TRUE</span>, <span class="at">type =</span> <span class="st">"lines"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="PCA_files/figure-html/code-pca-screeplot-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Scree plot of the Doubs River environmental data PCA."><img src="PCA_files/figure-html/code-pca-screeplot-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" alt="Scree plot of the Doubs River environmental data PCA."></a></p>
</figure>
</div>
<figcaption>Scree plot of the Doubs River environmental data PCA.</figcaption>
</figure>
</div>
</div>
</div>
<p>Or I can make a scree plot using <strong>ggplot2</strong>, which is more flexible:</p>
<div class="cell" data-layout-align="center">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb29"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a>scree_dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">eigenvalue =</span> <span class="fu">as.vector</span>(<span class="fu">eigenvals</span>(env_pca)),</span>
<span id="cb29-2"><a href="#cb29-2"></a>                        <span class="at">bstick =</span> <span class="fu">bstick</span>(env_pca))</span>
<span id="cb29-3"><a href="#cb29-3"></a>scree_dat<span class="sc">$</span>axis <span class="ot">&lt;-</span> <span class="fu">rownames</span>(scree_dat)</span>
<span id="cb29-4"><a href="#cb29-4"></a><span class="fu">rownames</span>(scree_dat) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb29-5"><a href="#cb29-5"></a>scree_dat <span class="ot">&lt;-</span> scree_dat <span class="sc">|&gt;</span> </span>
<span id="cb29-6"><a href="#cb29-6"></a>  <span class="fu">mutate</span>(<span class="at">axis =</span> <span class="fu">factor</span>(axis, <span class="at">levels =</span> <span class="fu">paste0</span>(<span class="fu">rep</span>(<span class="st">"PC"</span>, <span class="dv">11</span>), <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">11</span>))))</span>
<span id="cb29-7"><a href="#cb29-7"></a></span>
<span id="cb29-8"><a href="#cb29-8"></a><span class="fu">ggplot</span>(<span class="at">data =</span> scree_dat, <span class="fu">aes</span>(<span class="at">x =</span> axis, <span class="at">y =</span> eigenvalue)) <span class="sc">+</span></span>
<span id="cb29-9"><a href="#cb29-9"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb29-10"><a href="#cb29-10"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb29-11"><a href="#cb29-11"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> bstick), <span class="at">colour =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb29-12"><a href="#cb29-12"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> bstick, <span class="at">group =</span> <span class="dv">1</span>), <span class="at">colour =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb29-13"><a href="#cb29-13"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Principal component"</span>, <span class="at">y =</span> <span class="st">"Inertia"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-pca-scree-ggplot" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-scree-ggplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="PCA_files/figure-html/fig-pca-scree-ggplot-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Figure&nbsp;3: Scree plot of the Doubs River environmental data PCA made in ggplot2."><img src="PCA_files/figure-html/fig-pca-scree-ggplot-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-scree-ggplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Scree plot of the Doubs River environmental data PCA made in <strong>ggplot2</strong>.
</figcaption>
</figure>
</div>
</div>
</div>
<p>In the plot, above, the red line is the broken stick components and the black line the eigenvalues for the different PCs. The plot suggests retaining the first two PC axes, which together account for approximately 74% of the total inertia, indicating that they capture most of the large-scale variance in the dataset—but not necessarily all ecologically relevant structure. This distinction will be important in constrained ordination and model-based approaches, where axes are evaluated not only by variance accounted for, but by their relationship to explicitly hypothesised environmental drivers. See Numerical Ecology with R pp.&nbsp;121-122 for more information about how to decide how many PCs to retain.</p>
</section>
<section id="ordination-diagrams" class="level2" data-number="10">
<h2 data-number="10" class="anchored" data-anchor-id="ordination-diagrams"><span class="header-section-number">10</span> Ordination Diagrams</h2>
<!-- ### Plots along one-dimension -->
<!-- Now I will construct some primitive graphs of the Site and Species Scores to demonstrate how to interpret the eigenvectors associated with each eigenvalue. We will typically not do this kind of graphical display---for plots suitable for publication see the [Biplots](../BCB743/08-pca/#biplots) section, below. -->
<!-- The first thing we need to do is extract the Species and Site Scores in a manner that makes them convenient for plotting. To do this, we can apply the `scores()` function to the PCA object, `env_pca`, and assign the output to tidied dataframes. The `scores()` function can tidy the data to some extent, but I make it even tidier in subsequent steps by creating long format data (rather than wide) using the `pivot_longer()` function. Various other bits of code lines accomplish additional restructuring of the data to make datasets that are fully compliant for creating the kind of figure I have in mind: -->
<!-- ```{r} -->
<!-- # species scores first by setting 'display' to species -->
<!-- # we are interested in all axes (PC1 to PC11) so set the 'choices' argument -->
<!-- # appropriately (see `?scores`): -->
<!-- spp_sc <- scores(env_pca, display = "species", choices = seq(1:11), tidy = TRUE) -->
<!-- # now pivot longer to make the data even tidier: -->
<!-- spp_sc <- spp_sc |>  -->
<!--   select(-score) |> # remove column -->
<!--   pivot_longer(cols = PC1:PC11, # pivot -->
<!--                names_to = "PC_axis", -->
<!--                values_to = "score") |>  -->
<!--   group_by(PC_axis) |>  -->
<!--   mutate(rank = rank(abs(score)), # rank absolute scores -->
<!--          origin = 0, # create a column for start of arrows -->
<!--          PC_axis = factor(PC_axis, levels = paste0(rep("PC", 11), seq(1:11)))) |>  -->
<!--   # above, reorder the factor levels so PC axis plot in right order -->
<!--   filter(rank >= 10) |> # keep only 2 higheest ranked scores -->
<!--   ungroup() -->
<!-- head(spp_sc) -->
<!-- # now the site scores: -->
<!-- site_sc <- scores(env_pca, display = "sites", choices = seq(1:11), tidy = TRUE) -->
<!-- site_sc <- site_sc |>  -->
<!--   select(-score) |>  -->
<!--   pivot_longer(cols = PC1:PC11, -->
<!--                names_to = "PC_axis", -->
<!--                values_to = "score") |>  -->
<!--   mutate(label = as.numeric(label), -->
<!--          PC_axis = factor(PC_axis, levels = paste0(rep("PC", 11), seq(1:11)))) -->
<!-- head(site_sc) -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # var explained along PC1, PC2, and PC3 for adding to plot: -->
<!-- PC1_var <- round(env_pca$CA$eig[1] / sum(env_pca$CA$eig) * 100, 1) -->
<!-- PC2_var <- round(env_pca$CA$eig[2] / sum(env_pca$CA$eig) * 100, 1) -->
<!-- PC3_var <- round(env_pca$CA$eig[3] / sum(env_pca$CA$eig) * 100, 1) -->
<!-- ``` -->
<!-- Now we can assemble a plot, and in it focus on the first two PCs. It seems somewhat complex, but the code can easily be deciphered if you read through it 'layer-by-layer': -->
<!-- ```{r} -->
<!-- #| fig-width: 8 -->
<!-- #| fig-height: 6 -->
<!-- library(ggrepel) -->
<!-- #| fig.align: centre -->
<!-- #| fig.cap: "Ordination plot of the Doubs River environmental data showing only the influence of the first principal component." -->
<!-- #| label: fig-pca-linear -->
<!-- ggplot(data = site_sc, aes(x = PC_axis, y = score)) + -->
<!--   geom_hline(aes(yintercept = 0), linetype = "dashed") + -->
<!--   geom_jitter(shape = 19, width = 0.09, aes(colour = label)) + -->
<!--   scale_colour_viridis_c(name = "Site no.") + -->
<!--   geom_segment(data = spp_sc, aes(x = PC_axis, y = origin, -->
<!--                                   xend = PC_axis, yend = score), -->
<!--                lineend = "butt", -->
<!--                arrow = arrow(length = unit(3, "mm"), -->
<!--                              type = "open", -->
<!--                              angle = 30), -->
<!--                alpha = 0.8, size = 0.7, colour = "red") + -->
<!--   geom_text_repel(data = spp_sc, aes(label = label), size = 3.0, -->
<!--                   direction = "y", colour = "red") + -->
<!--   annotate(geom = "text", x = 1, y = -2.5, size = 3.0, colour = "red", -->
<!--            label = paste0(PC1_var, "% var. expl.")) + -->
<!--   annotate(geom = "text", x = 2, y = -2.5, size = 3.0, colour = "red", -->
<!--            label = paste0(PC2_var, "% var. expl.")) + -->
<!--   annotate(geom = "text", x = 3, y = -2.5, size = 3.0, colour = "red", -->
<!--            label = paste0(PC3_var, "% var. expl.")) + -->
<!--   coord_flip() + -->
<!--   labs(x = NULL, y = "Score") + -->
<!--   theme( -->
<!--     panel.grid.major.x = element_blank(), -->
<!--     panel.grid.minor.x = element_blank(), -->
<!--     panel.grid.major.y = element_line(colour = "pink", linetype = "dashed"), -->
<!--     legend.position = c(0.075, 0.75), -->
<!--     legend.box.background = element_rect(colour = "black") -->
<!--   ) -->
<!-- ``` -->
<!-- Although you will never see a graph like this one, examining it is nevertheless informative. What I did was: -->
<!-- * **plot all new PC axes on the vertical axis**; they represent the reduced, simplified ecological space; these PC axes are ranked from most important (PC1) to least important (PC11)---and each one's ability to explain some property of the environment is ranked by the magnitude of their eigenvalues -->
<!-- * PC1 explains 54.3% of the total variation in the environmental dataset -->
<!-- * PC2 explains an additional 19.7% of the remaining variance left over after accounting for the influence of PC1 -->
<!-- * the cumulative % variance explained by PC1 and PC2 is 74% -->
<!-- * for each PC axis I plot -->
<!-- * the **Site scores as coloured points** -->
<!-- * the colours indicate the sampled sites' numbers -->
<!-- * the points indicate the spread of the sites across linear Euclidean space with respect to the main environmental gradients represented by each PC axis -->
<!-- * the **Species scores as arrows** -->
<!-- * I only plot the top two most heavily loaded absolute eigenvectors -->
<!-- * the main environmental gradients are represented by the arrows -->
<!-- * the gradient represented is annotated by text giving the name of the environmental variables -->
<!-- * the location of the arrow heads is located in Euclidean space at the coordinates provided by the Species scores -->
<!-- * the longer the arrow, the more influence it has on causing the sites to spread out in Euclidean space -->
<!-- * the arrows point in the direction where the magnitude of the environmental variable is greater, and in the opposite direction the magnitude of the variable is less; for example, sites are spread out along PC1 primarily due to the influence of the variables nitrate and distance from source such that the sites further down the river (more yellow) tend to have a higher nitrate concentration and have a larger distance from source, and sites closer to the source (more blue) have a smaller distance from source and lower nitrate concentration. -->
<p>I provide some examples of ordination diagrams scattered throughout the course content (e.g., <a href="../quantecol/chapters/pca_examples/#a-ggplot-biplot">here</a>), but you may also refer to the step-by-step walk throughs provided by <a href="https://rpubs.com/Roeland-KINDT/694016">Roeland Kindt</a>. Also see David Zelený’s <a href="https://www.davidzeleny.net/anadat-r/doku.php/en:ordiagrams">excellent writing on the topic</a>.</p>
<p>Let us look at examples. In a PCA ordination diagram, following the tradition of scatter diagrams in Cartesian coordinate systems, objects are represented as points and variables are displayed as arrows. We first use the standard <strong>vegan</strong> <code>biplot()</code> function:</p>
<div class="cell" data-layout-align="center">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a>opar <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">no.readonly =</span> <span class="cn">TRUE</span>)</span>
<span id="cb30-2"><a href="#cb30-2"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb30-3"><a href="#cb30-3"></a><span class="fu">biplot</span>(env_pca, <span class="at">scaling =</span> <span class="dv">1</span>, <span class="at">main =</span> <span class="st">"PCA scaling 1"</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb30-4"><a href="#cb30-4"></a><span class="fu">biplot</span>(env_pca, <span class="at">scaling =</span> <span class="dv">2</span>, <span class="at">main =</span> <span class="st">"PCA scaling 2"</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="PCA_files/figure-html/code-pca-biplots-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6" title="Ordination plot of the Doubs River environmental data showing site scaling (left) and species scaling (right)."><img src="PCA_files/figure-html/code-pca-biplots-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" alt="Ordination plot of the Doubs River environmental data showing site scaling (left) and species scaling (right)."></a></p>
</figure>
</div>
<figcaption>Ordination plot of the Doubs River environmental data showing site scaling (left) and species scaling (right).</figcaption>
</figure>
</div>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb31"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a><span class="fu">par</span>(opar)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="scaling" class="level2" data-number="11">
<h2 data-number="11" class="anchored" data-anchor-id="scaling"><span class="header-section-number">11</span> Scaling</h2>
<p>The choice of scaling in a PCA biplot helps us to make the right inferences about our data. Each scaling answers a different question, and the interpretation of distances, angles, and vector lengths depends on which question is being asked.</p>
<p><strong>Scaling 1</strong> should be used when the primary question concerns how <em>sites relate to one another across environmental gradients</em>. Distances among objects (samples or sites—whatever is contained in the data table’s rows) in the biplot are approximations of their Euclidean distances in multidimensional space. Objects positioned further apart show a greater degree of environmental dissimilarity. The angles among descriptor vectors should not be interpreted as indicating the degree of correlation between the variables.</p>
<p><strong>Scaling 2</strong> should be used when the primary question concerns how <em>variables relate to the ordination axes and to one another</em>. This is the content of the data table’s columns. Distances among objects (samples or sites) in the biplot are not approximations of their Euclidean distances in multidimensional space. The angles among descriptor vectors can be interpreted as indicating the degree of correlation between the variables.</p>
<p>Now we create biplots using the <code>cleanplot.pca()</code> function that comes with the <em>Numerical Ecology with R</em> book. The figures are more or less the same, except the plot showing the Site scores with Scaling 1 adds a ‘circle of equilibrium contribution’ (see <em>Numerical Ecology with R</em>, p.&nbsp;125). The circle of equilibrium contribution is a visual aid drawn on a biplot to help assess the relative importance of species (or variables) in the ordination space. It is most useful in PCA biplots using scaling 1, where the focus is on species relationships. The circle is not a formal statistical test. It helps us to quickly identify the most important variables or species, but it does not directly indicate statistical significance.</p>
<p>We only assign importance to the arrows that extend beyond the radius of the circle (<a href="#fig-pca-cleanplot" class="quarto-xref">Figure&nbsp;4</a>):</p>
<div class="cell" data-layout-align="center">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb32"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a><span class="co"># we need to load the function first from its R file:</span></span>
<span id="cb32-2"><a href="#cb32-2"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"NEwR-2ed_code_data"</span>, <span class="st">"NEwR2-Functions"</span>, <span class="st">"cleanplot.pca.R"</span>))</span>
<span id="cb32-3"><a href="#cb32-3"></a><span class="fu">cleanplot.pca</span>(env_pca, <span class="at">scaling =</span> <span class="dv">1</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div id="fig-pca-cleanplot" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-pca-cleanplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="PCA_files/figure-html/fig-pca-cleanplot-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7" title="Figure&nbsp;4: Ordination plot of the Doubs River environmental data made with the cleanplot.pca() function."><img src="PCA_files/figure-html/fig-pca-cleanplot-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-pca-cleanplot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Ordination plot of the Doubs River environmental data made with the <code>cleanplot.pca()</code> function.
</figcaption>
</figure>
</div>
</div>
</div>
<p><strong>At this point it is essential that you refer to <em>Numerical Ecology with R</em> (pp.&nbsp;118 to 126) for help with interpreting the ordination diagrams.</strong></p>
</section>
<section id="fitting-environmental-response-surfaces" class="level2" data-number="12">
<h2 data-number="12" class="anchored" data-anchor-id="fitting-environmental-response-surfaces"><span class="header-section-number">12</span> Fitting Environmental Response Surfaces</h2>
<p>The <code>ordisurf()</code> function in <strong>vegan</strong> is used to visualise underlying environmental gradients on an ordination plot. This function fits a smooth surface (usually a generalised additive model GAM fitted to the site scores of PC axes of interest) to the ordination plot based on a specified environmental variable. It highlights how that variable changes across the ordination space so that we may interpret the spatial structure of the data in relation to the environmental gradient more easily.</p>
<p>For more about <code>ordisurf()</code>, see Gavin Simpson’s blog post <a href="https://fromthebottomoftheheap.net/2011/06/10/what-is-ordisurf-doing/">What is <code>ordifurf()</code> doing?</a>.</p>
<p>We plot the response surfaces for elevation and biological oxygen demand:</p>
<div class="cell" data-layout-align="center">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="annotated-cell-22"><pre class="sourceCode numberSource r code-annotation-code number-lines code-with-copy code-annotated"><code class="sourceCode r"><span id="annotated-cell-22-1"><a href="#annotated-cell-22-1"></a><span class="fu">biplot</span>(env_pca, <span class="at">type =</span> <span class="fu">c</span>(<span class="st">"text"</span>, <span class="st">"points"</span>), <span class="at">col =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"black"</span>))</span>
<a class="code-annotation-anchor" data-target-cell="annotated-cell-22" data-target-annotation="1" onclick="event.preventDefault();">1</a><span id="annotated-cell-22-2" class="code-annotation-target"><a href="#annotated-cell-22-2"></a><span class="fu">invisible</span>(<span class="fu">ordisurf</span>(env_pca <span class="sc">~</span> bod, env, <span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">col =</span> <span class="st">"turquoise"</span>, <span class="at">knots =</span> <span class="dv">1</span>))</span>
<span id="annotated-cell-22-3"><a href="#annotated-cell-22-3"></a><span class="fu">invisible</span>(<span class="fu">ordisurf</span>(env_pca <span class="sc">~</span> ele, env, <span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">col =</span> <span class="st">"salmon"</span>, <span class="at">knots =</span> <span class="dv">1</span>))</span><div class="code-annotation-gutter-bg"></div><div class="code-annotation-gutter"></div></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-annotation">
<dl class="code-annotation-container-grid">
<dt data-target-cell="annotated-cell-22" data-target-annotation="1">1</dt>
<dd>
<span data-code-cell="annotated-cell-22" data-code-lines="2" data-code-annotation="1"><code>invisible()</code> is used to suppress the output of the <code>ordisurf()</code> function; only the figure is returned.</span>
</dd>
</dl>
</div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><a href="PCA_files/figure-html/code-pca-ordisurf-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8" title="Ordination plot of the Doubs River environmental data fitted with a smooth response surface for elevation and biological oxygen demand."><img src="PCA_files/figure-html/code-pca-ordisurf-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" alt="Ordination plot of the Doubs River environmental data fitted with a smooth response surface for elevation and biological oxygen demand."></a></p>
</figure>
</div>
<figcaption>Ordination plot of the Doubs River environmental data fitted with a smooth response surface for elevation and biological oxygen demand.</figcaption>
</figure>
</div>
</div>
</div>
<p>PCA is best understood as an exploratory and diagnostic method rather than a catch-all solution for ecological ordination. Its strength lies in revealing broad structure in multivariate data, i.e., dominant gradients, major axes of covariation, and large-scale patterns that organise samples and variables in a reduced space. When applied to environmental data, PCA can provide a clear initial orientation to highlight correlated drivers, identify major contrasts among sites, and offer a coherent global summary of variation. Its limitations arise from the scope of a PCA: it is constrained to linear relationships under Euclidean geometry, and therefore cannot represent non-linear ecological responses without distortion them. PCA can be useful to establish a baseline against which alternative ordination methods can be evaluated, to guide you toward approaches such as CA, CCA, nMDS, or distance-based methods that are better aligned with specific response forms and inferential goals.</p>


<!-- -->


</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-graham2003confronting" class="csl-entry" role="listitem">
Graham MH (2003) Confronting multicollinearity in ecological multiple regression. Ecology 84:2809–2815.
</div>
<div id="ref-verneaux1973cours" class="csl-entry" role="listitem">
Verneaux J (1973) Cours d’eau de <span>F</span>ranche-<span>C</span>omt<span>é</span> (<span>M</span>assif du <span>J</span>ura). Recherches <span class="nocase">é</span>cologiques sur le r<span>é</span>seau hydrographique du <span>D</span>oubs.
</div>
</div></section><section class="quarto-appendix-contents" id="quarto-reuse"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a></div></div></section><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{smit,_a._j.2021,
  author = {Smit, A. J.,},
  title = {Principal {Component} {Analysis} {(PCA)}},
  date = {2021-01-01},
  url = {http://tangledbank.netlify.app/BCB743/PCA.html},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-smit,_a._j.2021" class="csl-entry quarto-appendix-citeas" role="listitem">
Smit, A. J. (2021) Principal Component Analysis (PCA). <a href="http://tangledbank.netlify.app/BCB743/PCA.html">http://tangledbank.netlify.app/BCB743/PCA.html.
</a>
</div></div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("http:\/\/tangledbank\.netlify\.app");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
            // default icon
            link.classList.add("external");
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../BCB743/ordination.html" class="pagination-link" aria-label="7: Intro to Ordination">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">7: Intro to Ordination</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../BCB743/PCA_examples.html" class="pagination-link" aria-label="8b: PCA: Additional Examples">
        <span class="nav-page-text">8b: PCA: Additional Examples</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb33" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb33-1"><a href="#cb33-1"></a><span class="co">---</span></span>
<span id="cb33-2"><a href="#cb33-2"></a><span class="an">date:</span><span class="co"> "2021-01-01"</span></span>
<span id="cb33-3"><a href="#cb33-3"></a><span class="an">title:</span><span class="co"> "Principal Component Analysis (PCA)"</span></span>
<span id="cb33-4"><a href="#cb33-4"></a><span class="co">---</span></span>
<span id="cb33-5"><a href="#cb33-5"></a></span>
<span id="cb33-6"><a href="#cb33-6"></a>::: callout-tip</span>
<span id="cb33-7"><a href="#cb33-7"></a><span class="fu">## **Material Required for This Chapter**</span></span>
<span id="cb33-8"><a href="#cb33-8"></a></span>
<span id="cb33-9"><a href="#cb33-9"></a><span class="pp">|</span> Type           <span class="pp">|</span> Name                                     <span class="pp">|</span> Link                                                                              <span class="pp">|</span></span>
<span id="cb33-10"><a href="#cb33-10"></a><span class="pp">|----------------|------------------------------------------|-----------------------------------------------------------------------------------|</span></span>
<span id="cb33-11"><a href="#cb33-11"></a><span class="pp">|</span> **Theory**     <span class="pp">|</span> Numerical Ecology with R                   <span class="pp">|</span> See pages 117-132                                                                 <span class="pp">|</span></span>
<span id="cb33-12"><a href="#cb33-12"></a><span class="pp">|</span> **Slides**     <span class="pp">|</span> PCA lecture slides                       <span class="pp">|</span> <span class="co">[</span><span class="ot">💾 `BCB743_08_PCA.pdf`</span><span class="co">](../slides/BCB743_08_PCA.pdf)</span>                             <span class="pp">|</span></span>
<span id="cb33-13"><a href="#cb33-13"></a><span class="pp">|</span> **Data**       <span class="pp">|</span> The Doubs River data                     <span class="pp">|</span> <span class="co">[</span><span class="ot">💾 `Doubs.RData`</span><span class="co">](../data/NEwR-2ed_code_data/NeWR2-Data/Doubs.RData)</span>             <span class="pp">|</span></span>
<span id="cb33-14"><a href="#cb33-14"></a><span class="pp">|</span> **R function** <span class="pp">|</span> A function for ordination plots          <span class="pp">|</span> <span class="co">[</span><span class="ot">💾 `cleanplot.pca.R`</span><span class="co">](../data/NEwR-2ed_code_data/NeWR2-Functions/cleanplot.pca.R)</span><span class="pp">|</span> </span>
<span id="cb33-15"><a href="#cb33-15"></a>:::</span>
<span id="cb33-16"><a href="#cb33-16"></a></span>
<span id="cb33-17"><a href="#cb33-17"></a>::: {.callout-important appearance="simple"}</span>
<span id="cb33-18"><a href="#cb33-18"></a><span class="fu">## Tasks to Complete in This Chapter</span></span>
<span id="cb33-19"><a href="#cb33-19"></a></span>
<span id="cb33-20"><a href="#cb33-20"></a><span class="ss">* </span><span class="co">[</span><span class="ot">Task C 1--5</span><span class="co">](tasks/Task_C.qmd)</span></span>
<span id="cb33-21"><a href="#cb33-21"></a>:::</span>
<span id="cb33-22"><a href="#cb33-22"></a></span>
<span id="cb33-23"><a href="#cb33-23"></a>::: {.callout-important appearance="simple"}</span>
<span id="cb33-24"><a href="#cb33-24"></a><span class="fu">## PCA as a Foundation for Understanding Ordination</span></span>
<span id="cb33-25"><a href="#cb33-25"></a></span>
<span id="cb33-26"><a href="#cb33-26"></a>Thoroughly understanding PCA is a prerequisite for understanding more complex ordination techniques. Mastered this chapter before moving on to CA, PCoA, RDA, and CCA. Much of the same terminology and concepts are used in these techniques.</span>
<span id="cb33-27"><a href="#cb33-27"></a>:::</span>
<span id="cb33-28"><a href="#cb33-28"></a></span>
<span id="cb33-29"><a href="#cb33-29"></a><span class="fu">## Geometric Projection of High-Dimensional Data</span></span>
<span id="cb33-30"><a href="#cb33-30"></a></span>
<span id="cb33-31"><a href="#cb33-31"></a>Ordination refers to a suite of multivariate techniques that reduce a high-dimensional dataset into a lower-dimensional space, typically 2D or 3D, in such a way that any intrinsic structure in the data forms visually-discernible patterns (Pielou, 1984). In ecology, ordination techniques are used to describe relationships between community structure patterns and underlying environmental gradients. They allow us to determine the relative importance of different gradients and visualise species-environment relationships.</span>
<span id="cb33-32"><a href="#cb33-32"></a></span>
<span id="cb33-33"><a href="#cb33-33"></a>Principal Component Analysis (PCA) is one of the commonly used ordination techniques in ecology. It is a dimensionality reduction technique that transforms the original set of variables into a new set of uncorrelated variables called principal components. PCA is performed on a data matrix containing species abundances or environmental variables across multiple sites or samples.</span>
<span id="cb33-34"><a href="#cb33-34"></a></span>
<span id="cb33-35"><a href="#cb33-35"></a>Although PCA is often introduced as a method for reducing dimensionality, it is more accurately understood as a geometric re-expression of the data. The method does not discard variables arbitrarily; rather, it rotates the original coordinate system so that new, mutually orthogonal (independent) axes are aligned with directions of maximal variance, and then projects the data onto these axes. This rotation-projection viewpoint will become explicit later when we examine how PCA is computed in R, but it is worth keeping in mind from the outset that PCA reorganises variation by changing the basis in which the data are represented.</span>
<span id="cb33-36"><a href="#cb33-36"></a></span>
<span id="cb33-37"><a href="#cb33-37"></a>The PCA process involves calculating the **eigenvectors** and **eigenvalues** of the covariance or correlation matrix of the data. The eigenvectors represent the directions of maximum variance in the data, and the corresponding eigenvalues represent the amount of variance explained by each eigenvector. The new axes, called principal components, are linear combinations of the original variables, ordered by the amount of variance they explain.</span>
<span id="cb33-38"><a href="#cb33-38"></a></span>
<span id="cb33-39"><a href="#cb33-39"></a>PCA preserves the Euclidean distances between samples in the original high-dimensional space when projecting them onto the lower-dimensional ordination space. This property makes PCA more suitable for analysing environmental data, where Euclidean distances between samples are meaningful and interpretable. However, for species data, which is often in the form of counts or frequencies, Euclidean distances may not be an appropriate measure of dissimilarity between samples.</span>
<span id="cb33-40"><a href="#cb33-40"></a></span>
<span id="cb33-41"><a href="#cb33-41"></a><span class="fu">## The Horseshoe Effect {#sec-horseshoe-effect}</span></span>
<span id="cb33-42"><a href="#cb33-42"></a></span>
<span id="cb33-43"><a href="#cb33-43"></a>The **horseshoe effect** (sometimes called the Guttman effect) is an artefact often seen with PCA when applied to species data, especially when using species abundance data for communities along environmental gradients. It distorts the data points in the ordination space. A less severe version of the horseshoe effect is called the ‘arch effect’ and is seen in <span class="co">[</span><span class="ot">Correspondence Analysis (CA)</span><span class="co">](CA.qmd)</span>.</span>
<span id="cb33-44"><a href="#cb33-44"></a></span>
<span id="cb33-45"><a href="#cb33-45"></a>The horseshoe effect occurs because PCA assumes linear relationships between variables, while species data often exhibit unimodal responses to environmental gradients. The unimodal model was discussed in <span class="co">[</span><span class="ot">BDC334</span><span class="co">](../BDC334/L02a-gradients.html#sec-unimodal)</span>. When species have unimodal distributions along a gradient, PCA tends to fold the ends of the gradient towards each other.</span>
<span id="cb33-46"><a href="#cb33-46"></a></span>
<span id="cb33-47"><a href="#cb33-47"></a>From a geometric perspective, the horseshoe effect points to a mismatch between the structure of the data and the space into which PCA forces it. Species responding unimodally to an underlying gradient occupy a curved configuration in multivariate space. PCA, however, seeks a linear subspace that maximises variance under Euclidean geometry. In doing so, it approximates this curved structure with straight axes, causing the ends of the gradient to bend back toward one another in the reduced ordination. The horseshoe, which is visible on the ordination diagram, is therefore not a peculiarity of the data, but a consequence of representing a non-linear manifold using a linear projection.</span>
<span id="cb33-48"><a href="#cb33-48"></a></span>
<span id="cb33-49"><a href="#cb33-49"></a>This distortion can lead to several issues:</span>
<span id="cb33-50"><a href="#cb33-50"></a></span>
<span id="cb33-51"><a href="#cb33-51"></a><span class="ss">1. </span>The horseshoe shape can make it appear that sites at opposite ends of the gradient are more similar than they really are.</span>
<span id="cb33-52"><a href="#cb33-52"></a><span class="ss">2. </span>The folding of the gradient ends compresses the data and potentially obscures important ecological patterns.</span>
<span id="cb33-53"><a href="#cb33-53"></a><span class="ss">3. </span>The second PCA axis---the most affected axis---often does not represent a meaningful ecological gradient, making interpretation challenging.</span>
<span id="cb33-54"><a href="#cb33-54"></a><span class="ss">4. </span>Unlike the arch effect in CA, the horseshoe effect in PCA can lead to incorrect ordering of samples along the gradient.</span>
<span id="cb33-55"><a href="#cb33-55"></a></span>
<span id="cb33-56"><a href="#cb33-56"></a>These distortions are informative, because they reveal the limits of PCA’s linear and Euclidean assumptions when confronted with non-linear ecological structure. To address these issues, we prefer to use <span class="co">[</span><span class="ot">non-metric Multidimensional Scaling (nMDS)</span><span class="co">](nMDS.qmd)</span> or a distance-based method (like <span class="co">[</span><span class="ot">Principal Coordinates Analysis, PCoA</span><span class="co">](PCoA.qmd)</span>) that are not constrained to linear projections. Alternatively, we may use techniques specifically designed to handle unimodal species responses, such as <span class="co">[</span><span class="ot">Correspondence Analysis (CA)</span><span class="co">](CA.qmd)</span>, which uses $\chi^2$-distance and not Euclidean distance, or its detrended version (<span class="co">[</span><span class="ot">Detrended Correspondence Analysis, DCA</span><span class="co">](DCA.qmd)</span>), but these come with their own set of considerations.</span>
<span id="cb33-57"><a href="#cb33-57"></a></span>
<span id="cb33-58"><a href="#cb33-58"></a>Let us work through a PCA to demonstrate the ideas.</span>
<span id="cb33-59"><a href="#cb33-59"></a></span>
<span id="cb33-60"><a href="#cb33-60"></a><span class="fu">## Set-up the Analysis Environment</span></span>
<span id="cb33-61"><a href="#cb33-61"></a></span>
<span id="cb33-62"><a href="#cb33-62"></a><span class="in">```{r code-library-tidyverse}</span></span>
<span id="cb33-63"><a href="#cb33-63"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb33-64"><a href="#cb33-64"></a><span class="fu">library</span>(vegan)</span>
<span id="cb33-65"><a href="#cb33-65"></a><span class="fu">library</span>(ggcorrplot) <span class="co"># for the correlations</span></span>
<span id="cb33-66"><a href="#cb33-66"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb33-67"><a href="#cb33-67"></a><span class="in">```</span></span>
<span id="cb33-68"><a href="#cb33-68"></a></span>
<span id="cb33-69"><a href="#cb33-69"></a><span class="fu">## The Doubs River Data</span></span>
<span id="cb33-70"><a href="#cb33-70"></a></span>
<span id="cb33-71"><a href="#cb33-71"></a><span class="in">```{r code-load-here-here-data}</span></span>
<span id="cb33-72"><a href="#cb33-72"></a><span class="fu">load</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"NEwR-2ed_code_data"</span>, <span class="st">"NEwR2-Data"</span>, <span class="st">"Doubs.RData"</span>))</span>
<span id="cb33-73"><a href="#cb33-73"></a><span class="fu">head</span>(env)</span>
<span id="cb33-74"><a href="#cb33-74"></a><span class="in">```</span></span>
<span id="cb33-75"><a href="#cb33-75"></a></span>
<span id="cb33-76"><a href="#cb33-76"></a><span class="fu">## First Do a Correlation</span></span>
<span id="cb33-77"><a href="#cb33-77"></a></span>
<span id="cb33-78"><a href="#cb33-78"></a><span class="in">```{r code-pca-correlation}</span></span>
<span id="cb33-79"><a href="#cb33-79"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb33-80"><a href="#cb33-80"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb33-81"><a href="#cb33-81"></a><span class="co">#| fig.align: center</span></span>
<span id="cb33-82"><a href="#cb33-82"></a><span class="co">#| fig.cap: "Pairwise correlations among the environmental variables included with the Doubs River study."</span></span>
<span id="cb33-83"><a href="#cb33-83"></a></span>
<span id="cb33-84"><a href="#cb33-84"></a><span class="co"># computing a correlation matrix</span></span>
<span id="cb33-85"><a href="#cb33-85"></a>corr <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">cor</span>(env), <span class="dv">1</span>)</span>
<span id="cb33-86"><a href="#cb33-86"></a></span>
<span id="cb33-87"><a href="#cb33-87"></a><span class="co"># visualisation of the correlation matrix</span></span>
<span id="cb33-88"><a href="#cb33-88"></a><span class="fu">ggcorrplot</span>(corr, <span class="at">type =</span> <span class="st">'upper'</span>, <span class="at">outline.col =</span> <span class="st">"white"</span>,</span>
<span id="cb33-89"><a href="#cb33-89"></a>           <span class="at">colors =</span> <span class="fu">c</span>(<span class="st">"#1679a1"</span>, <span class="st">"white"</span>, <span class="st">"#f8766d"</span>),</span>
<span id="cb33-90"><a href="#cb33-90"></a>           <span class="at">lab =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-91"><a href="#cb33-91"></a><span class="in">```</span></span>
<span id="cb33-92"><a href="#cb33-92"></a></span>
<span id="cb33-93"><a href="#cb33-93"></a>Some variables are very correlated, and they might be omitted from the subsequent analyses. We say that these variables are **collinear**. Collinear variables cannot be teased apart in terms of finding out which one is most influential in structuring the community. There are more advanced ways to search for collinear variables (e.g., Variance Inflation Factors, VIF) and in this way we can systematically exclude them from the PCA. See @graham2003confronting for a discussion on collinearity.</span>
<span id="cb33-94"><a href="#cb33-94"></a></span>
<span id="cb33-95"><a href="#cb33-95"></a>Here we will proceed with all the variables --- this is a delibirate decision for my approach here. In PCA, collinearity does not cause instability in estimated coefficients in the way it does in multiple regression, because PCA does not estimate regression parameters. Instead, correlated variables jointly determine the orientation of the ordination axes, i.e., strongly collinear variables tend to load in similar directions and contribute collectively to the same components. While this can complicate interpretation (by making it harder to disentangle the influence of individual variables) it does not invalidate the construction of the axes themselves. For teaching purposes, retaining collinear variables makes the structure of the data explicit and allows us to see how shared variance is redistributed across components. As we shall see later, in the constrained analyses where coefficients are interpreted directly, collinearity becomes a more serious inferential (statistical) issue and is addressed explicitly through diagnostics such as variance inflation factors and variable selection.</span>
<span id="cb33-96"><a href="#cb33-96"></a></span>
<span id="cb33-97"><a href="#cb33-97"></a><span class="fu">## See the Spatial Context</span></span>
<span id="cb33-98"><a href="#cb33-98"></a></span>
<span id="cb33-99"><a href="#cb33-99"></a>The patterns in the data and the correlations between them will make more sense if we can visualise a spatial context. Thankfully spatial data are available:</span>
<span id="cb33-100"><a href="#cb33-100"></a></span>
<span id="cb33-101"><a href="#cb33-101"></a><span class="in">```{r fig-pca-map1}</span></span>
<span id="cb33-102"><a href="#cb33-102"></a><span class="co">#| fig.align: center</span></span>
<span id="cb33-103"><a href="#cb33-103"></a><span class="co">#| fig.cap: "The spatial configuration of the Doubs River sites."</span></span>
<span id="cb33-104"><a href="#cb33-104"></a></span>
<span id="cb33-105"><a href="#cb33-105"></a><span class="fu">head</span>(spa)</span>
<span id="cb33-106"><a href="#cb33-106"></a><span class="fu">ggplot</span>(spa, <span class="fu">aes</span>(<span class="at">x =</span> X, <span class="at">y =</span> Y, <span class="at">label =</span> <span class="fu">rownames</span>(spa))) <span class="sc">+</span></span>
<span id="cb33-107"><a href="#cb33-107"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb33-108"><a href="#cb33-108"></a>  <span class="fu">geom_label</span>(<span class="at">vjust =</span> <span class="dv">0</span>, <span class="at">nudge_y =</span> <span class="fl">0.5</span>, <span class="at">check_overlap =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-109"><a href="#cb33-109"></a><span class="in">```</span></span>
<span id="cb33-110"><a href="#cb33-110"></a></span>
<span id="cb33-111"><a href="#cb33-111"></a>These site numbers correspond approximately to the ones in @verneaux1973cours but some of the numbers may have been shifted slightly in the example Doubs dataset used here compared to how they were originally numbered in Verneaux's thesis and subsequent publication. This should not affect the interpretation. We can also scale the symbol size by the magnitude of the environmental variables. Lets look at two pairs of variables that are strongly correlated with one-another:</span>
<span id="cb33-112"><a href="#cb33-112"></a></span>
<span id="cb33-113"><a href="#cb33-113"></a><span class="in">```{r fig-pca-map2}</span></span>
<span id="cb33-114"><a href="#cb33-114"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb33-115"><a href="#cb33-115"></a><span class="co">#| fig-height: 6</span></span>
<span id="cb33-116"><a href="#cb33-116"></a><span class="co">#| fig.align: center</span></span>
<span id="cb33-117"><a href="#cb33-117"></a><span class="co">#| fig.cap: "Four different representations of the site configuration (spatial context) of the Doubs River sampling layout. Symbols are scaled relative to A) ammonium concentration, B) BOD, C) Altitude, and D) Flow rate."</span></span>
<span id="cb33-118"><a href="#cb33-118"></a></span>
<span id="cb33-119"><a href="#cb33-119"></a><span class="co"># We scale the data first so as to better represent the full</span></span>
<span id="cb33-120"><a href="#cb33-120"></a><span class="co"># magnitude of all variables with a common symbol size</span></span>
<span id="cb33-121"><a href="#cb33-121"></a>env_std <span class="ot">&lt;-</span> <span class="fu">decostand</span>(env, <span class="at">method =</span> <span class="st">"standardize"</span>)</span>
<span id="cb33-122"><a href="#cb33-122"></a></span>
<span id="cb33-123"><a href="#cb33-123"></a><span class="co"># positive correlations</span></span>
<span id="cb33-124"><a href="#cb33-124"></a>plt1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(spa, <span class="fu">aes</span>(<span class="at">x =</span> X, <span class="at">y =</span> Y, <span class="at">label =</span> <span class="fu">rownames</span>(spa))) <span class="sc">+</span></span>
<span id="cb33-125"><a href="#cb33-125"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">"red"</span>, <span class="fu">aes</span>(<span class="at">size =</span> env_std<span class="sc">$</span>amm, <span class="at">shape =</span> <span class="dv">3</span>)) <span class="sc">+</span></span>
<span id="cb33-126"><a href="#cb33-126"></a>  <span class="fu">geom_text</span>(<span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">nudge_y =</span> <span class="fl">0.5</span>, <span class="at">check_overlap =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb33-127"><a href="#cb33-127"></a>  <span class="fu">labs</span>(<span class="at">size =</span> <span class="st">"Magnitude"</span>, <span class="at">title =</span> <span class="st">"Ammonium concentration"</span>)</span>
<span id="cb33-128"><a href="#cb33-128"></a></span>
<span id="cb33-129"><a href="#cb33-129"></a>plt2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(spa, <span class="fu">aes</span>(<span class="at">x =</span> X, <span class="at">y =</span> Y, <span class="at">label =</span> <span class="fu">rownames</span>(spa))) <span class="sc">+</span></span>
<span id="cb33-130"><a href="#cb33-130"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">"red"</span>, <span class="fu">aes</span>(<span class="at">size =</span> env_std<span class="sc">$</span>bod)) <span class="sc">+</span></span>
<span id="cb33-131"><a href="#cb33-131"></a>  <span class="fu">geom_text</span>(<span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">nudge_y =</span> <span class="fl">0.5</span>, <span class="at">check_overlap =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb33-132"><a href="#cb33-132"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Biological oxygen demand"</span>)</span>
<span id="cb33-133"><a href="#cb33-133"></a></span>
<span id="cb33-134"><a href="#cb33-134"></a><span class="co"># inverse correlations</span></span>
<span id="cb33-135"><a href="#cb33-135"></a>plt3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(spa, <span class="fu">aes</span>(<span class="at">x =</span> X, <span class="at">y =</span> Y, <span class="at">label =</span> <span class="fu">rownames</span>(spa))) <span class="sc">+</span></span>
<span id="cb33-136"><a href="#cb33-136"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">"blue"</span>, <span class="fu">aes</span>(<span class="at">size =</span> env_std<span class="sc">$</span>alt)) <span class="sc">+</span></span>
<span id="cb33-137"><a href="#cb33-137"></a>  <span class="fu">geom_text</span>(<span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">nudge_y =</span> <span class="fl">0.5</span>, <span class="at">check_overlap =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb33-138"><a href="#cb33-138"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Altitude"</span>)</span>
<span id="cb33-139"><a href="#cb33-139"></a></span>
<span id="cb33-140"><a href="#cb33-140"></a>plt4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(spa, <span class="fu">aes</span>(<span class="at">x =</span> X, <span class="at">y =</span> Y, <span class="at">label =</span> <span class="fu">rownames</span>(spa))) <span class="sc">+</span></span>
<span id="cb33-141"><a href="#cb33-141"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">"blue"</span>, <span class="fu">aes</span>(<span class="at">size =</span> env_std<span class="sc">$</span>flo)) <span class="sc">+</span></span>
<span id="cb33-142"><a href="#cb33-142"></a>  <span class="fu">geom_text</span>(<span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">nudge_y =</span> <span class="fl">0.5</span>, <span class="at">check_overlap =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb33-143"><a href="#cb33-143"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Flow rate"</span>)</span>
<span id="cb33-144"><a href="#cb33-144"></a></span>
<span id="cb33-145"><a href="#cb33-145"></a><span class="fu">ggarrange</span>(plt1, plt2, plt3, plt4, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">ncol =</span> <span class="dv">2</span>,</span>
<span id="cb33-146"><a href="#cb33-146"></a>          <span class="at">common.legend =</span> <span class="cn">TRUE</span>, <span class="at">labels =</span> <span class="st">"AUTO"</span>)</span>
<span id="cb33-147"><a href="#cb33-147"></a><span class="in">```</span></span>
<span id="cb33-148"><a href="#cb33-148"></a></span>
<span id="cb33-149"><a href="#cb33-149"></a><span class="fu">## Do the PCA</span></span>
<span id="cb33-150"><a href="#cb33-150"></a></span>
<span id="cb33-151"><a href="#cb33-151"></a>We use the function <span class="in">`rda()`</span> to do the PCA, but it can also be performed in base R with the functions <span class="in">`prcomp()`</span> and <span class="in">`princomp()`</span>. <span class="in">`rda()`</span> is the same function that we will use later for a Redundancy Analysis, but when used without specifying constraints (as we do here) it amounts to simply doing a PCA. Typically we standardise environmental data to unit variance, but the PCA done by the <span class="in">`rda()`</span> function accomplishes this step automagically when <span class="in">`scale = TRUE`</span>. When applied to environmental data (as we typically do with a PCA) it works with correlations amongst the scaled variables. PCA preserves Euclidean distance and the relationships detected are linear, and for this reason it is not typically applied to species data without suitable transformations. In fact, in this module we will seldom apply a PCA to species data at all.</span>
<span id="cb33-152"><a href="#cb33-152"></a></span>
<span id="cb33-153"><a href="#cb33-153"></a><span class="in">```{r code-env-pca-rda-env-scale}</span></span>
<span id="cb33-154"><a href="#cb33-154"></a>env_pca <span class="ot">&lt;-</span> <span class="fu">rda</span>(env, <span class="at">scale =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-155"><a href="#cb33-155"></a>env_pca</span>
<span id="cb33-156"><a href="#cb33-156"></a></span>
<span id="cb33-157"><a href="#cb33-157"></a><span class="co"># same ...</span></span>
<span id="cb33-158"><a href="#cb33-158"></a><span class="co"># env_std &lt;- scale(env)</span></span>
<span id="cb33-159"><a href="#cb33-159"></a><span class="co"># env_pca &lt;- rda(env_std, scale = FALSE)</span></span>
<span id="cb33-160"><a href="#cb33-160"></a><span class="co"># env_pca</span></span>
<span id="cb33-161"><a href="#cb33-161"></a><span class="in">```</span></span>
<span id="cb33-162"><a href="#cb33-162"></a></span>
<span id="cb33-163"><a href="#cb33-163"></a>In ordination we use the term **inertia** as a synonym for 'variation', but some PCA software (such as R's <span class="in">`prcomp()`</span> and <span class="in">`princomp()`</span>) simply uses the term <span class="in">`sdev`</span> for standard deviations. In PCA, when we use a correlation matrix (as we do here), the inertia is the sum of the diagonal values of the correlation matrix, which is simply the number of variables (11 in this example). When a PCA uses a covariance matrix the inertia is the sum of the variances of the variables.</span>
<span id="cb33-164"><a href="#cb33-164"></a></span>
<span id="cb33-165"><a href="#cb33-165"></a>You will also see in the output the mention of the term **unconstrained**. In a PCA the analysis is always unconstrained (i.e., not influenced by some *a priori* defined variables we hypothesise to explain the between site patterns in the multivariate data).</span>
<span id="cb33-166"><a href="#cb33-166"></a></span>
<span id="cb33-167"><a href="#cb33-167"></a>The section headed **Eigenvalues for unconstrained axes** shows the relative importance of the resultant reduced axes, and they can be used to determine the proportion of the total inertia (sum of the eigenvalues) captured by any one of the axes. They can be accessed with the function `eigenvals()` (the preferred function; see `?rda` for help), but an alternative method is given below. The first eigenvalue (the one associated with PC1) always explains the most variation (the largest fraction), and each subsequent one explains the largest proportion of the **remaining** variance. We say the axes are **orthogonal** and ranked in decreasing order of importance. The sum of all eigenvalues is the total inertia, so collectively they theoretically can explain all of the variation in the dataset (but clearly they **should not** be used to explain **all** the variance). </span>
<span id="cb33-168"><a href="#cb33-168"></a></span>
<span id="cb33-169"><a href="#cb33-169"></a>At this point it is important to distinguish between statistical adequacy and ecological sufficiency. A principal component that explains a large proportion of the variance is effective at compressing the data, but this does not by necessity imply that it represents the most influential ecological process. Variance explained is a measure of how much information is retained under dimensional reduction; it is not a measure of causal dominance or biological importance. Ecologically meaningful gradients may account for relatively little variance, while strong variance components may reflect broad-scale structure, measurement scale, or correlated responses rather than interpretable mechanisms. Here is where your knowledge as an ecologist will (or should) guide you.</span>
<span id="cb33-170"><a href="#cb33-170"></a></span>
<span id="cb33-171"><a href="#cb33-171"></a>To extract the first eigenvalue we can do:</span>
<span id="cb33-172"><a href="#cb33-172"></a></span>
<span id="cb33-173"><a href="#cb33-173"></a><span class="in">```{r code-round-eigenvals-env-pca}</span></span>
<span id="cb33-174"><a href="#cb33-174"></a><span class="fu">round</span>(<span class="fu">eigenvals</span>(env_pca)[<span class="dv">1</span>], <span class="dv">3</span>)</span>
<span id="cb33-175"><a href="#cb33-175"></a></span>
<span id="cb33-176"><a href="#cb33-176"></a><span class="co"># or</span></span>
<span id="cb33-177"><a href="#cb33-177"></a></span>
<span id="cb33-178"><a href="#cb33-178"></a><span class="fu">round</span>(env_pca<span class="sc">$</span>CA<span class="sc">$</span>eig[<span class="dv">1</span>], <span class="dv">3</span>)</span>
<span id="cb33-179"><a href="#cb33-179"></a><span class="in">```</span></span>
<span id="cb33-180"><a href="#cb33-180"></a></span>
<span id="cb33-181"><a href="#cb33-181"></a>The total inertia is:</span>
<span id="cb33-182"><a href="#cb33-182"></a></span>
<span id="cb33-183"><a href="#cb33-183"></a><span class="in">```{r code-sum-eigenvals-env-pca}</span></span>
<span id="cb33-184"><a href="#cb33-184"></a><span class="fu">sum</span>(<span class="fu">eigenvals</span>(env_pca))</span>
<span id="cb33-185"><a href="#cb33-185"></a></span>
<span id="cb33-186"><a href="#cb33-186"></a><span class="co"># or</span></span>
<span id="cb33-187"><a href="#cb33-187"></a></span>
<span id="cb33-188"><a href="#cb33-188"></a><span class="fu">sum</span>(env_pca<span class="sc">$</span>CA<span class="sc">$</span>eig)</span>
<span id="cb33-189"><a href="#cb33-189"></a><span class="in">```</span></span>
<span id="cb33-190"><a href="#cb33-190"></a></span>
<span id="cb33-191"><a href="#cb33-191"></a>So the proportion of variation explained by the first PC is:</span>
<span id="cb33-192"><a href="#cb33-192"></a></span>
<span id="cb33-193"><a href="#cb33-193"></a><span class="in">```{r code-round-env-pca-ca-eig}</span></span>
<span id="cb33-194"><a href="#cb33-194"></a><span class="fu">round</span>(env_pca<span class="sc">$</span>CA<span class="sc">$</span>eig[<span class="dv">1</span>] <span class="sc">/</span> <span class="fu">sum</span>(env_pca<span class="sc">$</span>CA<span class="sc">$</span>eig) <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>) <span class="co"># result in %</span></span>
<span id="cb33-195"><a href="#cb33-195"></a><span class="in">```</span></span>
<span id="cb33-196"><a href="#cb33-196"></a></span>
<span id="cb33-197"><a href="#cb33-197"></a>We can show the same information as part of a more verbose summary. Here we see the pre-calculated **Proportion Explained** and **Cumulative Proportion** (it should be obvious what this is). There is also an assortment of other information, viz. **Scaling 2 for species and site scores**, **Species scores**, and **Site scores**.</span>
<span id="cb33-198"><a href="#cb33-198"></a></span>
<span id="cb33-199"><a href="#cb33-199"></a><span class="in">```{r code-summary-env-pca}</span></span>
<span id="cb33-200"><a href="#cb33-200"></a><span class="fu">summary</span>(env_pca)</span>
<span id="cb33-201"><a href="#cb33-201"></a><span class="in">```</span></span>
<span id="cb33-202"><a href="#cb33-202"></a></span>
<span id="cb33-203"><a href="#cb33-203"></a>Newer versions of the ordinations' <span class="in">`summary()`</span> method no longer shows the species and site scores. To access these scores, we use the <span class="in">`scores()`</span> function:</span>
<span id="cb33-204"><a href="#cb33-204"></a></span>
<span id="cb33-205"><a href="#cb33-205"></a>The species scores can be extracted from the ordination object by:</span>
<span id="cb33-206"><a href="#cb33-206"></a></span>
<span id="cb33-207"><a href="#cb33-207"></a><span class="in">```{r code-scores-env-pca-display-species}</span></span>
<span id="cb33-208"><a href="#cb33-208"></a><span class="co"># extract species scores for first three PCA axes:</span></span>
<span id="cb33-209"><a href="#cb33-209"></a><span class="fu">scores</span>(env_pca, <span class="at">display =</span> <span class="st">"species"</span>, <span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)</span>
<span id="cb33-210"><a href="#cb33-210"></a><span class="in">```</span></span>
<span id="cb33-211"><a href="#cb33-211"></a>Similarly, for the site scores, do:</span>
<span id="cb33-212"><a href="#cb33-212"></a></span>
<span id="cb33-213"><a href="#cb33-213"></a><span class="in">```{r code-scores-env-pca-display-site}</span></span>
<span id="cb33-214"><a href="#cb33-214"></a><span class="co"># extract the site scores for axes 1 and 2</span></span>
<span id="cb33-215"><a href="#cb33-215"></a><span class="fu">scores</span>(env_pca, <span class="at">display =</span> <span class="st">"site"</span>, <span class="at">choices =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>) <span class="sc">|&gt;</span> </span>
<span id="cb33-216"><a href="#cb33-216"></a>  <span class="fu">head</span>() <span class="co"># truncate to save space in this example</span></span>
<span id="cb33-217"><a href="#cb33-217"></a><span class="in">```</span></span>
<span id="cb33-218"><a href="#cb33-218"></a></span>
<span id="cb33-219"><a href="#cb33-219"></a>The <span class="in">`scores()`</span> method also applies to other ordination methods.</span>
<span id="cb33-220"><a href="#cb33-220"></a></span>
<span id="cb33-221"><a href="#cb33-221"></a>::: {.callout-important appearance="simple"}</span>
<span id="cb33-222"><a href="#cb33-222"></a><span class="fu">## Question</span></span>
<span id="cb33-223"><a href="#cb33-223"></a></span>
<span id="cb33-224"><a href="#cb33-224"></a>We are dealing with environmental variables here, so why do we call them "species scores"?</span>
<span id="cb33-225"><a href="#cb33-225"></a>:::</span>
<span id="cb33-226"><a href="#cb33-226"></a></span>
<span id="cb33-227"><a href="#cb33-227"></a><span class="fu">## Terminology</span></span>
<span id="cb33-228"><a href="#cb33-228"></a></span>
<span id="cb33-229"><a href="#cb33-229"></a>The geometric ideas introduced earlier (rotation of axes and projection of data into a reduced space) are formalised in ordination methods through a singular-value decomposition (SVD). In **vegan**, ordination functions such as <span class="in">`rda()`</span> for PCA/RDA and <span class="in">`cca()`</span> for CA/CCA implement this decomposition directly.</span>
<span id="cb33-230"><a href="#cb33-230"></a></span>
<span id="cb33-231"><a href="#cb33-231"></a>Once the data matrix $X$ has been centred (and, where necessary, scaled or $\chi^2$-transformed), the SVD computes:</span>
<span id="cb33-232"><a href="#cb33-232"></a></span>
<span id="cb33-233"><a href="#cb33-233"></a>$$X = U \,\Sigma \,V^{\mathsf T},$$</span>
<span id="cb33-234"><a href="#cb33-234"></a></span>
<span id="cb33-235"><a href="#cb33-235"></a>with orthonormal *left-singular vectors* $U$, orthonormal *right-singular vectors* $V$, and positive singular values on the diagonal of $\Sigma$. From here, all the components of our ordination can be extracted. How do these relate to our data?</span>
<span id="cb33-236"><a href="#cb33-236"></a></span>
<span id="cb33-237"><a href="#cb33-237"></a><span class="ss">- </span>**Eigenvectors** are the directions in the original variable space along which the variance in the data is maximised. They relate to our original data in two ways:</span>
<span id="cb33-238"><a href="#cb33-238"></a><span class="ss">  - </span>$V$ is the matrix of right singular vectors; they are the *eigenvectors of the columns* (species or variables). The quantities can be accessed by <span class="in">`env_pca$CA$v`</span>.</span>
<span id="cb33-239"><a href="#cb33-239"></a><span class="ss">  - </span>$U$ is the matrix of left singular vectors; *i.e.*, the *eigenvectors of the rows* (sites). The left singular vectors can be accessed by <span class="in">`env_pca$CA$u`</span></span>
<span id="cb33-240"><a href="#cb33-240"></a><span class="ss">- </span>$\Sigma$ is a diagonal matrix of singular values $d$, and the **eigenvalues** are the square of these singular values, $eig=d^ 2$. Eigenvalues represent the amount of variance explained by each corresponding eigenvector (or principal component), represented in PCA (and other unconstrained ordinations such as CA) by <span class="in">`env_pca$CA$eig`</span>. Higher eigenvalues indicate that the corresponding principal component explains a larger portion of the total variance in the data. The sum of all eigenvalues is equal to the total variance in the data.</span>
<span id="cb33-241"><a href="#cb33-241"></a></span>
<span id="cb33-242"><a href="#cb33-242"></a>Let us map these directly to our <span class="in">`env_pca`</span> object, which I will assume was created with the <span class="in">`rda()`</span> function. Looking inside the <span class="in">`env_pca`</span> object, we will see a structure with a component named $CA$ or $CCA$. This holds the results for the constrained/canonical analysis. For a simple PCA, all the results are in $CA$. The eigenvectors (<span class="in">`env_pca$CA$eig`</span>) define the direction of the new axes, they are not the final coordinates for ordination plots. *Projection* is the process of mapping the original data points onto the new reduced axes (principal components). What we plot are actually these *scores*.</span>
<span id="cb33-243"><a href="#cb33-243"></a></span>
<span id="cb33-244"><a href="#cb33-244"></a><span class="ss">- </span>**Scores** are the coordinates of the original sites and species projected onto the new ordination axes. They are calculated by scaling the eigenvectors. These scores are what we actually plot to create a biplot or triplot.</span>
<span id="cb33-245"><a href="#cb33-245"></a><span class="ss">  - </span>The **Site Scores** tell us where each site (sample) is located along the new ordination axes. Sites that are close together in the ordination plot have similar variable (or species) compositions. We obtain the site scores as <span class="in">`scores(env_pca, display = "site", ...)`</span>.</span>
<span id="cb33-246"><a href="#cb33-246"></a><span class="ss">  - </span>The **Species Scores** (more accurately, **variable scores** in this context) describe how the columns of the data matrix, whatever they represent ecologically, relate to the ordination axes. Sometimes we call the species scores *loadings*. In a PCA biplot, these are represented by arrows (but in CA not, as that reduced space is no longer linear). An arrow pointing strongly towards PCA1 means that species or variable is a major contributor to that axis of variation. So, species scores represent the contributions of the original variables to the principal components. They indicate the strength and direction of the correlation between the original variables and the new principal components. Species scores are obtained with <span class="in">`scores(env_pca, display = "species", ...)`</span>. Larger positive values indicate a stronger positive correlation, while larger negative values indicate a stronger negative correlation. It is worth considering the terminology "species scores". The label “species scores” is structural since it inherited from a general ordination framework in which columns of the data matrix often correspond to species abundances. In PCA applied to environmental data, those columns contain variables, not taxa. The software does not “know” what the columns represent, so it treats them as abstract descriptors whose interpretation is supplied entirely by the analyst. Confusing function names with ontological claims may be a common source of error in multivariate analysis, and it is therefore important to read ordination output as a mathematical (or conceptual, if math isn't your strong point) representation first, and only subsequently as an ecological one.</span>
<span id="cb33-247"><a href="#cb33-247"></a></span>
<span id="cb33-248"><a href="#cb33-248"></a></span>
<span id="cb33-249"><a href="#cb33-249"></a>Here is the direct mapping:</span>
<span id="cb33-250"><a href="#cb33-250"></a></span>
<span id="cb33-251"><a href="#cb33-251"></a><span class="pp">|</span> Mathematical Concept <span class="pp">|</span> <span class="in">`vegan`</span> Term <span class="pp">|</span> Where to Find It in the <span class="in">`rda`</span> Object <span class="pp">|</span> Description <span class="pp">|</span></span>
<span id="cb33-252"><a href="#cb33-252"></a><span class="pp">| :---</span> <span class="pp">| :---</span> <span class="pp">| :---</span> <span class="pp">| :---</span> <span class="pp">|</span></span>
<span id="cb33-253"><a href="#cb33-253"></a><span class="pp">|</span> **Eigenvalues** <span class="pp">|</span> Eigenvalues <span class="pp">|</span> <span class="in">`env_pca$CA$eig`</span> <span class="pp">|</span> The amount of variance explained by each axis (component). <span class="in">`summary(env_pca)`</span> shows this clearly. <span class="pp">|</span></span>
<span id="cb33-254"><a href="#cb33-254"></a><span class="pp">|</span> **Eigenvectors of Variables** <span class="pp">|</span> Species Scores <span class="pp">|</span> <span class="in">`env_pca$CA$v`</span> <span class="pp">|</span> The raw eigenvectors for the columns of the data (the variables). These are scaled to become the "species scores" that you plot as arrows. <span class="pp">|</span></span>
<span id="cb33-255"><a href="#cb33-255"></a><span class="pp">|</span> **Eigenvectors of Sites** <span class="pp">|</span> Site Scores <span class="pp">|</span> <span class="in">`env_pca$CA$u`</span> <span class="pp">|</span> The raw eigenvectors for the rows of the data (the sites). These are scaled to become the "site scores" that you plot as points. <span class="pp">|</span></span>
<span id="cb33-256"><a href="#cb33-256"></a><span class="pp">|</span> **Plottable Coordinates** <span class="pp">|</span> Scores <span class="pp">|</span> <span class="in">`scores(env_pca)`</span> <span class="pp">|</span> The function <span class="in">`scores()`</span> extracts and properly scales the raw eigenvectors (<span class="in">`$u`</span> and <span class="in">`$v`</span>) to give the coordinates ready for plotting. <span class="pp">|</span></span>
<span id="cb33-257"><a href="#cb33-257"></a></span>
<span id="cb33-258"><a href="#cb33-258"></a>When interpreting a PCA biplot (see below), the species scores (loadings) indicate the contributions of the original variables to the principal components, while the site scores represent the positions of the sites in the reduced-dimensional space defined by those principal components. The direction and length of the species score vectors (arrows) provide information about the relationships between the original variables and the principal components, while the positions of the site points reflect the similarities or differences between sites based on the environmental gradients represented by the principal components.</span>
<span id="cb33-259"><a href="#cb33-259"></a></span>
<span id="cb33-260"><a href="#cb33-260"></a><span class="co">&lt;!-- It is customary to plot the Site and Species Scores as ordination diagrams called **biplots** for a few of the reduced axes. We will get to this below. --&gt;</span></span>
<span id="cb33-261"><a href="#cb33-261"></a></span>
<span id="cb33-262"><a href="#cb33-262"></a><span class="co">&lt;!-- **Scaling 1** and **Scaling 2**, depending on what was specified in the `rda()` function call, are useful for whether one wants to interpret species (scaling 1) or variables (scaling 2). When calling Scaling 1, the distances between points plotted on the ordination diagram (i.e. the information in the rows, which in this case is sites) will retain their Euclidean distances, which allows for better interpretation of how sites relate to one-another. Calling Scaling 2 preserves more accurately the angles between variables (the columns, i.e. here we have environmental variables) with the consequence that in the biplot smaller angles between variable vectors will reflect stronger correlations. More on scaling below. --&gt;</span></span>
<span id="cb33-263"><a href="#cb33-263"></a></span>
<span id="cb33-264"><a href="#cb33-264"></a><span class="fu">## How Many Axes to Retain?</span></span>
<span id="cb33-265"><a href="#cb33-265"></a></span>
<span id="cb33-266"><a href="#cb33-266"></a>The number of axes to retain is a difficult question to answer. The first few axes will always explain the most variation, but how do we know how many reduced axes are influential and should be kept? Commonly recommended is the **broken stick** method---keep the principal components whose eigenvalues are higher than corresponding random broken stick components:</span>
<span id="cb33-267"><a href="#cb33-267"></a></span>
<span id="cb33-268"><a href="#cb33-268"></a><span class="in">```{r code-pca-screeplot, fig.height=4, fig.width=6}</span></span>
<span id="cb33-269"><a href="#cb33-269"></a><span class="co">#| fig.align: center</span></span>
<span id="cb33-270"><a href="#cb33-270"></a><span class="co">#| fig.cap: "Scree plot of the Doubs River environmental data PCA."</span></span>
<span id="cb33-271"><a href="#cb33-271"></a></span>
<span id="cb33-272"><a href="#cb33-272"></a><span class="co"># make a scree plot using the vegan function:</span></span>
<span id="cb33-273"><a href="#cb33-273"></a><span class="fu">screeplot</span>(env_pca, <span class="at">bstick =</span> <span class="cn">TRUE</span>, <span class="at">type =</span> <span class="st">"lines"</span>)</span>
<span id="cb33-274"><a href="#cb33-274"></a><span class="in">```</span></span>
<span id="cb33-275"><a href="#cb33-275"></a></span>
<span id="cb33-276"><a href="#cb33-276"></a>Or I can make a scree plot using **ggplot2**, which is more flexible:</span>
<span id="cb33-277"><a href="#cb33-277"></a></span>
<span id="cb33-278"><a href="#cb33-278"></a><span class="in">```{r fig-pca-scree-ggplot, fig.height=3, fig.width=6}</span></span>
<span id="cb33-279"><a href="#cb33-279"></a><span class="co">#| fig.align: center</span></span>
<span id="cb33-280"><a href="#cb33-280"></a><span class="co">#| fig.cap: "Scree plot of the Doubs River environmental data PCA made in **ggplot2**."</span></span>
<span id="cb33-281"><a href="#cb33-281"></a></span>
<span id="cb33-282"><a href="#cb33-282"></a>scree_dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">eigenvalue =</span> <span class="fu">as.vector</span>(<span class="fu">eigenvals</span>(env_pca)),</span>
<span id="cb33-283"><a href="#cb33-283"></a>                        <span class="at">bstick =</span> <span class="fu">bstick</span>(env_pca))</span>
<span id="cb33-284"><a href="#cb33-284"></a>scree_dat<span class="sc">$</span>axis <span class="ot">&lt;-</span> <span class="fu">rownames</span>(scree_dat)</span>
<span id="cb33-285"><a href="#cb33-285"></a><span class="fu">rownames</span>(scree_dat) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb33-286"><a href="#cb33-286"></a>scree_dat <span class="ot">&lt;-</span> scree_dat <span class="sc">|&gt;</span> </span>
<span id="cb33-287"><a href="#cb33-287"></a>  <span class="fu">mutate</span>(<span class="at">axis =</span> <span class="fu">factor</span>(axis, <span class="at">levels =</span> <span class="fu">paste0</span>(<span class="fu">rep</span>(<span class="st">"PC"</span>, <span class="dv">11</span>), <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">11</span>))))</span>
<span id="cb33-288"><a href="#cb33-288"></a></span>
<span id="cb33-289"><a href="#cb33-289"></a><span class="fu">ggplot</span>(<span class="at">data =</span> scree_dat, <span class="fu">aes</span>(<span class="at">x =</span> axis, <span class="at">y =</span> eigenvalue)) <span class="sc">+</span></span>
<span id="cb33-290"><a href="#cb33-290"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb33-291"><a href="#cb33-291"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb33-292"><a href="#cb33-292"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">y =</span> bstick), <span class="at">colour =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb33-293"><a href="#cb33-293"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> bstick, <span class="at">group =</span> <span class="dv">1</span>), <span class="at">colour =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb33-294"><a href="#cb33-294"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Principal component"</span>, <span class="at">y =</span> <span class="st">"Inertia"</span>)</span>
<span id="cb33-295"><a href="#cb33-295"></a><span class="in">```</span></span>
<span id="cb33-296"><a href="#cb33-296"></a></span>
<span id="cb33-297"><a href="#cb33-297"></a>In the plot, above, the red line is the broken stick components and the black line the eigenvalues for the different PCs. The plot suggests retaining the first two PC axes, which together account for approximately 74% of the total inertia, indicating that they capture most of the large-scale variance in the dataset—but not necessarily all ecologically relevant structure. This distinction will be important in constrained ordination and model-based approaches, where axes are evaluated not only by variance accounted for, but by their relationship to explicitly hypothesised environmental drivers. See Numerical Ecology with R pp. 121-122 for more information about how to decide how many PCs to retain.</span>
<span id="cb33-298"><a href="#cb33-298"></a></span>
<span id="cb33-299"><a href="#cb33-299"></a><span class="fu">## Ordination Diagrams</span></span>
<span id="cb33-300"><a href="#cb33-300"></a></span>
<span id="cb33-301"><a href="#cb33-301"></a><span class="co">&lt;!-- </span><span class="al">###</span><span class="co"> Plots along one-dimension --&gt;</span></span>
<span id="cb33-302"><a href="#cb33-302"></a></span>
<span id="cb33-303"><a href="#cb33-303"></a><span class="co">&lt;!-- Now I will construct some primitive graphs of the Site and Species Scores to demonstrate how to interpret the eigenvectors associated with each eigenvalue. We will typically not do this kind of graphical display---for plots suitable for publication see the [Biplots](../BCB743/08-pca/#biplots) section, below. --&gt;</span></span>
<span id="cb33-304"><a href="#cb33-304"></a></span>
<span id="cb33-305"><a href="#cb33-305"></a><span class="co">&lt;!-- The first thing we need to do is extract the Species and Site Scores in a manner that makes them convenient for plotting. To do this, we can apply the `scores()` function to the PCA object, `env_pca`, and assign the output to tidied dataframes. The `scores()` function can tidy the data to some extent, but I make it even tidier in subsequent steps by creating long format data (rather than wide) using the `pivot_longer()` function. Various other bits of code lines accomplish additional restructuring of the data to make datasets that are fully compliant for creating the kind of figure I have in mind: --&gt;</span></span>
<span id="cb33-306"><a href="#cb33-306"></a></span>
<span id="cb33-307"><a href="#cb33-307"></a><span class="co">&lt;!-- ```{r} --&gt;</span></span>
<span id="cb33-308"><a href="#cb33-308"></a><span class="co">&lt;!-- # species scores first by setting 'display' to species --&gt;</span></span>
<span id="cb33-309"><a href="#cb33-309"></a><span class="co">&lt;!-- # we are interested in all axes (PC1 to PC11) so set the 'choices' argument --&gt;</span></span>
<span id="cb33-310"><a href="#cb33-310"></a><span class="co">&lt;!-- # appropriately (see `?scores`): --&gt;</span></span>
<span id="cb33-311"><a href="#cb33-311"></a><span class="co">&lt;!-- spp_sc &lt;- scores(env_pca, display = "species", choices = seq(1:11), tidy = TRUE) --&gt;</span></span>
<span id="cb33-312"><a href="#cb33-312"></a></span>
<span id="cb33-313"><a href="#cb33-313"></a><span class="co">&lt;!-- # now pivot longer to make the data even tidier: --&gt;</span></span>
<span id="cb33-314"><a href="#cb33-314"></a><span class="co">&lt;!-- spp_sc &lt;- spp_sc |&gt;  --&gt;</span></span>
<span id="cb33-315"><a href="#cb33-315"></a><span class="co">&lt;!--   select(-score) |&gt; # remove column --&gt;</span></span>
<span id="cb33-316"><a href="#cb33-316"></a><span class="co">&lt;!--   pivot_longer(cols = PC1:PC11, # pivot --&gt;</span></span>
<span id="cb33-317"><a href="#cb33-317"></a><span class="co">&lt;!--                names_to = "PC_axis", --&gt;</span></span>
<span id="cb33-318"><a href="#cb33-318"></a><span class="co">&lt;!--                values_to = "score") |&gt;  --&gt;</span></span>
<span id="cb33-319"><a href="#cb33-319"></a><span class="co">&lt;!--   group_by(PC_axis) |&gt;  --&gt;</span></span>
<span id="cb33-320"><a href="#cb33-320"></a><span class="co">&lt;!--   mutate(rank = rank(abs(score)), # rank absolute scores --&gt;</span></span>
<span id="cb33-321"><a href="#cb33-321"></a><span class="co">&lt;!--          origin = 0, # create a column for start of arrows --&gt;</span></span>
<span id="cb33-322"><a href="#cb33-322"></a><span class="co">&lt;!--          PC_axis = factor(PC_axis, levels = paste0(rep("PC", 11), seq(1:11)))) |&gt;  --&gt;</span></span>
<span id="cb33-323"><a href="#cb33-323"></a><span class="co">&lt;!--   # above, reorder the factor levels so PC axis plot in right order --&gt;</span></span>
<span id="cb33-324"><a href="#cb33-324"></a><span class="co">&lt;!--   filter(rank &gt;= 10) |&gt; # keep only 2 higheest ranked scores --&gt;</span></span>
<span id="cb33-325"><a href="#cb33-325"></a><span class="co">&lt;!--   ungroup() --&gt;</span></span>
<span id="cb33-326"><a href="#cb33-326"></a></span>
<span id="cb33-327"><a href="#cb33-327"></a><span class="co">&lt;!-- head(spp_sc) --&gt;</span></span>
<span id="cb33-328"><a href="#cb33-328"></a></span>
<span id="cb33-329"><a href="#cb33-329"></a><span class="co">&lt;!-- # now the site scores: --&gt;</span></span>
<span id="cb33-330"><a href="#cb33-330"></a><span class="co">&lt;!-- site_sc &lt;- scores(env_pca, display = "sites", choices = seq(1:11), tidy = TRUE) --&gt;</span></span>
<span id="cb33-331"><a href="#cb33-331"></a></span>
<span id="cb33-332"><a href="#cb33-332"></a><span class="co">&lt;!-- site_sc &lt;- site_sc |&gt;  --&gt;</span></span>
<span id="cb33-333"><a href="#cb33-333"></a><span class="co">&lt;!--   select(-score) |&gt;  --&gt;</span></span>
<span id="cb33-334"><a href="#cb33-334"></a><span class="co">&lt;!--   pivot_longer(cols = PC1:PC11, --&gt;</span></span>
<span id="cb33-335"><a href="#cb33-335"></a><span class="co">&lt;!--                names_to = "PC_axis", --&gt;</span></span>
<span id="cb33-336"><a href="#cb33-336"></a><span class="co">&lt;!--                values_to = "score") |&gt;  --&gt;</span></span>
<span id="cb33-337"><a href="#cb33-337"></a><span class="co">&lt;!--   mutate(label = as.numeric(label), --&gt;</span></span>
<span id="cb33-338"><a href="#cb33-338"></a><span class="co">&lt;!--          PC_axis = factor(PC_axis, levels = paste0(rep("PC", 11), seq(1:11)))) --&gt;</span></span>
<span id="cb33-339"><a href="#cb33-339"></a></span>
<span id="cb33-340"><a href="#cb33-340"></a><span class="co">&lt;!-- head(site_sc) --&gt;</span></span>
<span id="cb33-341"><a href="#cb33-341"></a><span class="co">&lt;!-- ``` --&gt;</span></span>
<span id="cb33-342"><a href="#cb33-342"></a></span>
<span id="cb33-343"><a href="#cb33-343"></a><span class="co">&lt;!-- ```{r} --&gt;</span></span>
<span id="cb33-344"><a href="#cb33-344"></a><span class="co">&lt;!-- # var explained along PC1, PC2, and PC3 for adding to plot: --&gt;</span></span>
<span id="cb33-345"><a href="#cb33-345"></a><span class="co">&lt;!-- PC1_var &lt;- round(env_pca$CA$eig[1] / sum(env_pca$CA$eig) * 100, 1) --&gt;</span></span>
<span id="cb33-346"><a href="#cb33-346"></a><span class="co">&lt;!-- PC2_var &lt;- round(env_pca$CA$eig[2] / sum(env_pca$CA$eig) * 100, 1) --&gt;</span></span>
<span id="cb33-347"><a href="#cb33-347"></a><span class="co">&lt;!-- PC3_var &lt;- round(env_pca$CA$eig[3] / sum(env_pca$CA$eig) * 100, 1) --&gt;</span></span>
<span id="cb33-348"><a href="#cb33-348"></a><span class="co">&lt;!-- ``` --&gt;</span></span>
<span id="cb33-349"><a href="#cb33-349"></a></span>
<span id="cb33-350"><a href="#cb33-350"></a><span class="co">&lt;!-- Now we can assemble a plot, and in it focus on the first two PCs. It seems somewhat complex, but the code can easily be deciphered if you read through it 'layer-by-layer': --&gt;</span></span>
<span id="cb33-351"><a href="#cb33-351"></a></span>
<span id="cb33-352"><a href="#cb33-352"></a><span class="co">&lt;!-- ```{r} --&gt;</span></span>
<span id="cb33-353"><a href="#cb33-353"></a><span class="co">&lt;!-- #| fig-width: 8 --&gt;</span></span>
<span id="cb33-354"><a href="#cb33-354"></a><span class="co">&lt;!-- #| fig-height: 6 --&gt;</span></span>
<span id="cb33-355"><a href="#cb33-355"></a><span class="co">&lt;!-- library(ggrepel) --&gt;</span></span>
<span id="cb33-356"><a href="#cb33-356"></a><span class="co">&lt;!-- #| fig.align: centre --&gt;</span></span>
<span id="cb33-357"><a href="#cb33-357"></a><span class="co">&lt;!-- #| fig.cap: "Ordination plot of the Doubs River environmental data showing only the influence of the first principal component." --&gt;</span></span>
<span id="cb33-358"><a href="#cb33-358"></a><span class="co">&lt;!-- #| label: fig-pca-linear --&gt;</span></span>
<span id="cb33-359"><a href="#cb33-359"></a></span>
<span id="cb33-360"><a href="#cb33-360"></a><span class="co">&lt;!-- ggplot(data = site_sc, aes(x = PC_axis, y = score)) + --&gt;</span></span>
<span id="cb33-361"><a href="#cb33-361"></a><span class="co">&lt;!--   geom_hline(aes(yintercept = 0), linetype = "dashed") + --&gt;</span></span>
<span id="cb33-362"><a href="#cb33-362"></a><span class="co">&lt;!--   geom_jitter(shape = 19, width = 0.09, aes(colour = label)) + --&gt;</span></span>
<span id="cb33-363"><a href="#cb33-363"></a><span class="co">&lt;!--   scale_colour_viridis_c(name = "Site no.") + --&gt;</span></span>
<span id="cb33-364"><a href="#cb33-364"></a><span class="co">&lt;!--   geom_segment(data = spp_sc, aes(x = PC_axis, y = origin, --&gt;</span></span>
<span id="cb33-365"><a href="#cb33-365"></a><span class="co">&lt;!--                                   xend = PC_axis, yend = score), --&gt;</span></span>
<span id="cb33-366"><a href="#cb33-366"></a><span class="co">&lt;!--                lineend = "butt", --&gt;</span></span>
<span id="cb33-367"><a href="#cb33-367"></a><span class="co">&lt;!--                arrow = arrow(length = unit(3, "mm"), --&gt;</span></span>
<span id="cb33-368"><a href="#cb33-368"></a><span class="co">&lt;!--                              type = "open", --&gt;</span></span>
<span id="cb33-369"><a href="#cb33-369"></a><span class="co">&lt;!--                              angle = 30), --&gt;</span></span>
<span id="cb33-370"><a href="#cb33-370"></a><span class="co">&lt;!--                alpha = 0.8, size = 0.7, colour = "red") + --&gt;</span></span>
<span id="cb33-371"><a href="#cb33-371"></a><span class="co">&lt;!--   geom_text_repel(data = spp_sc, aes(label = label), size = 3.0, --&gt;</span></span>
<span id="cb33-372"><a href="#cb33-372"></a><span class="co">&lt;!--                   direction = "y", colour = "red") + --&gt;</span></span>
<span id="cb33-373"><a href="#cb33-373"></a><span class="co">&lt;!--   annotate(geom = "text", x = 1, y = -2.5, size = 3.0, colour = "red", --&gt;</span></span>
<span id="cb33-374"><a href="#cb33-374"></a><span class="co">&lt;!--            label = paste0(PC1_var, "% var. expl.")) + --&gt;</span></span>
<span id="cb33-375"><a href="#cb33-375"></a><span class="co">&lt;!--   annotate(geom = "text", x = 2, y = -2.5, size = 3.0, colour = "red", --&gt;</span></span>
<span id="cb33-376"><a href="#cb33-376"></a><span class="co">&lt;!--            label = paste0(PC2_var, "% var. expl.")) + --&gt;</span></span>
<span id="cb33-377"><a href="#cb33-377"></a><span class="co">&lt;!--   annotate(geom = "text", x = 3, y = -2.5, size = 3.0, colour = "red", --&gt;</span></span>
<span id="cb33-378"><a href="#cb33-378"></a><span class="co">&lt;!--            label = paste0(PC3_var, "% var. expl.")) + --&gt;</span></span>
<span id="cb33-379"><a href="#cb33-379"></a><span class="co">&lt;!--   coord_flip() + --&gt;</span></span>
<span id="cb33-380"><a href="#cb33-380"></a><span class="co">&lt;!--   labs(x = NULL, y = "Score") + --&gt;</span></span>
<span id="cb33-381"><a href="#cb33-381"></a><span class="co">&lt;!--   theme( --&gt;</span></span>
<span id="cb33-382"><a href="#cb33-382"></a><span class="co">&lt;!--     panel.grid.major.x = element_blank(), --&gt;</span></span>
<span id="cb33-383"><a href="#cb33-383"></a><span class="co">&lt;!--     panel.grid.minor.x = element_blank(), --&gt;</span></span>
<span id="cb33-384"><a href="#cb33-384"></a><span class="co">&lt;!--     panel.grid.major.y = element_line(colour = "pink", linetype = "dashed"), --&gt;</span></span>
<span id="cb33-385"><a href="#cb33-385"></a><span class="co">&lt;!--     legend.position = c(0.075, 0.75), --&gt;</span></span>
<span id="cb33-386"><a href="#cb33-386"></a><span class="co">&lt;!--     legend.box.background = element_rect(colour = "black") --&gt;</span></span>
<span id="cb33-387"><a href="#cb33-387"></a><span class="co">&lt;!--   ) --&gt;</span></span>
<span id="cb33-388"><a href="#cb33-388"></a><span class="co">&lt;!-- ``` --&gt;</span></span>
<span id="cb33-389"><a href="#cb33-389"></a></span>
<span id="cb33-390"><a href="#cb33-390"></a><span class="co">&lt;!-- Although you will never see a graph like this one, examining it is nevertheless informative. What I did was: --&gt;</span></span>
<span id="cb33-391"><a href="#cb33-391"></a></span>
<span id="cb33-392"><a href="#cb33-392"></a><span class="co">&lt;!-- * **plot all new PC axes on the vertical axis**; they represent the reduced, simplified ecological space; these PC axes are ranked from most important (PC1) to least important (PC11)---and each one's ability to explain some property of the environment is ranked by the magnitude of their eigenvalues --&gt;</span></span>
<span id="cb33-393"><a href="#cb33-393"></a><span class="co">&lt;!-- * PC1 explains 54.3% of the total variation in the environmental dataset --&gt;</span></span>
<span id="cb33-394"><a href="#cb33-394"></a><span class="co">&lt;!-- * PC2 explains an additional 19.7% of the remaining variance left over after accounting for the influence of PC1 --&gt;</span></span>
<span id="cb33-395"><a href="#cb33-395"></a><span class="co">&lt;!-- * the cumulative % variance explained by PC1 and PC2 is 74% --&gt;</span></span>
<span id="cb33-396"><a href="#cb33-396"></a><span class="co">&lt;!-- * for each PC axis I plot --&gt;</span></span>
<span id="cb33-397"><a href="#cb33-397"></a><span class="co">&lt;!-- * the **Site scores as coloured points** --&gt;</span></span>
<span id="cb33-398"><a href="#cb33-398"></a><span class="co">&lt;!-- * the colours indicate the sampled sites' numbers --&gt;</span></span>
<span id="cb33-399"><a href="#cb33-399"></a><span class="co">&lt;!-- * the points indicate the spread of the sites across linear Euclidean space with respect to the main environmental gradients represented by each PC axis --&gt;</span></span>
<span id="cb33-400"><a href="#cb33-400"></a><span class="co">&lt;!-- * the **Species scores as arrows** --&gt;</span></span>
<span id="cb33-401"><a href="#cb33-401"></a><span class="co">&lt;!-- * I only plot the top two most heavily loaded absolute eigenvectors --&gt;</span></span>
<span id="cb33-402"><a href="#cb33-402"></a><span class="co">&lt;!-- * the main environmental gradients are represented by the arrows --&gt;</span></span>
<span id="cb33-403"><a href="#cb33-403"></a><span class="co">&lt;!-- * the gradient represented is annotated by text giving the name of the environmental variables --&gt;</span></span>
<span id="cb33-404"><a href="#cb33-404"></a><span class="co">&lt;!-- * the location of the arrow heads is located in Euclidean space at the coordinates provided by the Species scores --&gt;</span></span>
<span id="cb33-405"><a href="#cb33-405"></a><span class="co">&lt;!-- * the longer the arrow, the more influence it has on causing the sites to spread out in Euclidean space --&gt;</span></span>
<span id="cb33-406"><a href="#cb33-406"></a><span class="co">&lt;!-- * the arrows point in the direction where the magnitude of the environmental variable is greater, and in the opposite direction the magnitude of the variable is less; for example, sites are spread out along PC1 primarily due to the influence of the variables nitrate and distance from source such that the sites further down the river (more yellow) tend to have a higher nitrate concentration and have a larger distance from source, and sites closer to the source (more blue) have a smaller distance from source and lower nitrate concentration. --&gt;</span></span>
<span id="cb33-407"><a href="#cb33-407"></a></span>
<span id="cb33-408"><a href="#cb33-408"></a>I provide some examples of ordination diagrams scattered throughout the course content (e.g., <span class="co">[</span><span class="ot">here</span><span class="co">](/quantecol/chapters/pca_examples/#a-ggplot-biplot)</span>), but you may also refer to the step-by-step walk throughs provided by <span class="co">[</span><span class="ot">Roeland Kindt</span><span class="co">](https://rpubs.com/Roeland-KINDT/694016)</span>. Also see David Zelený's <span class="co">[</span><span class="ot">excellent writing on the topic</span><span class="co">](https://www.davidzeleny.net/anadat-r/doku.php/en:ordiagrams)</span>.</span>
<span id="cb33-409"><a href="#cb33-409"></a></span>
<span id="cb33-410"><a href="#cb33-410"></a>Let us look at examples. In a PCA ordination diagram, following the tradition of scatter diagrams in Cartesian coordinate systems, objects are represented as points and variables are displayed as arrows. We first use the standard **vegan** <span class="in">`biplot()`</span> function:</span>
<span id="cb33-411"><a href="#cb33-411"></a></span>
<span id="cb33-412"><a href="#cb33-412"></a><span class="in">```{r code-pca-biplots}</span></span>
<span id="cb33-413"><a href="#cb33-413"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb33-414"><a href="#cb33-414"></a><span class="co">#| fig-height: 5.625 </span></span>
<span id="cb33-415"><a href="#cb33-415"></a><span class="co">#| fig.align: center</span></span>
<span id="cb33-416"><a href="#cb33-416"></a><span class="co">#| fig.cap: "Ordination plot of the Doubs River environmental data showing site scaling (left) and species scaling (right)."</span></span>
<span id="cb33-417"><a href="#cb33-417"></a></span>
<span id="cb33-418"><a href="#cb33-418"></a>opar <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">no.readonly =</span> <span class="cn">TRUE</span>)</span>
<span id="cb33-419"><a href="#cb33-419"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb33-420"><a href="#cb33-420"></a><span class="fu">biplot</span>(env_pca, <span class="at">scaling =</span> <span class="dv">1</span>, <span class="at">main =</span> <span class="st">"PCA scaling 1"</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb33-421"><a href="#cb33-421"></a><span class="fu">biplot</span>(env_pca, <span class="at">scaling =</span> <span class="dv">2</span>, <span class="at">main =</span> <span class="st">"PCA scaling 2"</span>, <span class="at">choices =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb33-422"><a href="#cb33-422"></a><span class="fu">par</span>(opar)</span>
<span id="cb33-423"><a href="#cb33-423"></a><span class="in">```</span></span>
<span id="cb33-424"><a href="#cb33-424"></a></span>
<span id="cb33-425"><a href="#cb33-425"></a><span class="fu">## Scaling</span></span>
<span id="cb33-426"><a href="#cb33-426"></a></span>
<span id="cb33-427"><a href="#cb33-427"></a>The choice of scaling in a PCA biplot helps us to make the right inferences about our data. Each scaling answers a different question, and the interpretation of distances, angles, and vector lengths depends on which question is being asked.</span>
<span id="cb33-428"><a href="#cb33-428"></a></span>
<span id="cb33-429"><a href="#cb33-429"></a>**Scaling 1** should be used when the primary question concerns how *sites relate to one another across environmental gradients*. Distances among objects (samples or sites---whatever is contained in the data table's rows) in the biplot are approximations of their Euclidean distances in multidimensional space. Objects positioned further apart show a greater degree of environmental dissimilarity. The angles among descriptor vectors should not be interpreted as indicating the degree of correlation between the variables.</span>
<span id="cb33-430"><a href="#cb33-430"></a></span>
<span id="cb33-431"><a href="#cb33-431"></a>**Scaling 2** should be used when the primary question concerns how *variables relate to the ordination axes and to one another*. This is the content of the data table's columns. Distances among objects (samples or sites) in the biplot are not approximations of their Euclidean distances in multidimensional space. The angles among descriptor vectors can be interpreted as indicating the degree of correlation between the variables.</span>
<span id="cb33-432"><a href="#cb33-432"></a></span>
<span id="cb33-433"><a href="#cb33-433"></a>Now we create biplots using the <span class="in">`cleanplot.pca()`</span> function that comes with the *Numerical Ecology with R* book. The figures are more or less the same, except the plot showing the Site scores with Scaling 1 adds a 'circle of equilibrium contribution' (see *Numerical Ecology with R*, p. 125). The circle of equilibrium contribution is a visual aid drawn on a biplot to help assess the relative importance of species (or variables) in the ordination space. It is most useful in PCA biplots using scaling 1, where the focus is on species relationships. The circle is not a formal statistical test. It helps us to quickly identify the most important variables or species, but it does not directly indicate statistical significance.</span>
<span id="cb33-434"><a href="#cb33-434"></a></span>
<span id="cb33-435"><a href="#cb33-435"></a>We only assign importance to the arrows that extend beyond the radius of the circle (@fig-pca-cleanplot):</span>
<span id="cb33-436"><a href="#cb33-436"></a></span>
<span id="cb33-437"><a href="#cb33-437"></a><span class="in">```{r fig-pca-cleanplot}</span></span>
<span id="cb33-438"><a href="#cb33-438"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb33-439"><a href="#cb33-439"></a><span class="co">#| fig-height: 5.625 </span></span>
<span id="cb33-440"><a href="#cb33-440"></a><span class="co">#| fig.align: center</span></span>
<span id="cb33-441"><a href="#cb33-441"></a><span class="co">#| fig.cap: "Ordination plot of the Doubs River environmental data made with the `cleanplot.pca()` function."</span></span>
<span id="cb33-442"><a href="#cb33-442"></a></span>
<span id="cb33-443"><a href="#cb33-443"></a><span class="co"># we need to load the function first from its R file:</span></span>
<span id="cb33-444"><a href="#cb33-444"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"NEwR-2ed_code_data"</span>, <span class="st">"NEwR2-Functions"</span>, <span class="st">"cleanplot.pca.R"</span>))</span>
<span id="cb33-445"><a href="#cb33-445"></a><span class="fu">cleanplot.pca</span>(env_pca, <span class="at">scaling =</span> <span class="dv">1</span>)</span>
<span id="cb33-446"><a href="#cb33-446"></a><span class="in">```</span></span>
<span id="cb33-447"><a href="#cb33-447"></a></span>
<span id="cb33-448"><a href="#cb33-448"></a>**At this point it is essential that you refer to *Numerical Ecology with R* (pp. 118 to 126) for help with interpreting the ordination diagrams.**</span>
<span id="cb33-449"><a href="#cb33-449"></a></span>
<span id="cb33-450"><a href="#cb33-450"></a><span class="fu">## Fitting Environmental Response Surfaces</span></span>
<span id="cb33-451"><a href="#cb33-451"></a></span>
<span id="cb33-452"><a href="#cb33-452"></a>The <span class="in">`ordisurf()`</span> function in **vegan** is used to visualise underlying environmental gradients on an ordination plot. This function fits a smooth surface (usually a generalised additive model GAM fitted to the site scores of PC axes of interest) to the ordination plot based on a specified environmental variable. It highlights how that variable changes across the ordination space so that we may interpret the spatial structure of the data in relation to the environmental gradient more easily.</span>
<span id="cb33-453"><a href="#cb33-453"></a></span>
<span id="cb33-454"><a href="#cb33-454"></a>For more about <span class="in">`ordisurf()`</span>, see Gavin Simpson's blog post <span class="co">[</span><span class="ot">What is `ordifurf()` doing?</span><span class="co">](https://fromthebottomoftheheap.net/2011/06/10/what-is-ordisurf-doing/)</span>.</span>
<span id="cb33-455"><a href="#cb33-455"></a></span>
<span id="cb33-456"><a href="#cb33-456"></a>We plot the response surfaces for elevation and biological oxygen demand:</span>
<span id="cb33-457"><a href="#cb33-457"></a></span>
<span id="cb33-458"><a href="#cb33-458"></a><span class="in">```{r code-pca-ordisurf}</span></span>
<span id="cb33-459"><a href="#cb33-459"></a><span class="co">#| message: false</span></span>
<span id="cb33-460"><a href="#cb33-460"></a><span class="co">#| warning: false</span></span>
<span id="cb33-461"><a href="#cb33-461"></a><span class="co">#| fig-width: 10</span></span>
<span id="cb33-462"><a href="#cb33-462"></a><span class="co">#| fig-height: 5.625 </span></span>
<span id="cb33-463"><a href="#cb33-463"></a><span class="co">#| fig.align: center</span></span>
<span id="cb33-464"><a href="#cb33-464"></a><span class="co">#| fig.cap: "Ordination plot of the Doubs River environmental data fitted with a smooth response surface for elevation and biological oxygen demand."</span></span>
<span id="cb33-465"><a href="#cb33-465"></a></span>
<span id="cb33-466"><a href="#cb33-466"></a><span class="fu">biplot</span>(env_pca, <span class="at">type =</span> <span class="fu">c</span>(<span class="st">"text"</span>, <span class="st">"points"</span>), <span class="at">col =</span> <span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"black"</span>))</span>
<span id="cb33-467"><a href="#cb33-467"></a><span class="fu">invisible</span>(<span class="fu">ordisurf</span>(env_pca <span class="sc">~</span> bod, env, <span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">col =</span> <span class="st">"turquoise"</span>, <span class="at">knots =</span> <span class="dv">1</span>)) <span class="co"># &lt;1&gt;</span></span>
<span id="cb33-468"><a href="#cb33-468"></a><span class="fu">invisible</span>(<span class="fu">ordisurf</span>(env_pca <span class="sc">~</span> ele, env, <span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">col =</span> <span class="st">"salmon"</span>, <span class="at">knots =</span> <span class="dv">1</span>))</span>
<span id="cb33-469"><a href="#cb33-469"></a><span class="in">```</span></span>
<span id="cb33-470"><a href="#cb33-470"></a><span class="ss">1. </span><span class="in">`invisible()`</span> is used to suppress the output of the <span class="in">`ordisurf()`</span> function; only the figure is returned.</span>
<span id="cb33-471"><a href="#cb33-471"></a></span>
<span id="cb33-472"><a href="#cb33-472"></a>PCA is best understood as an exploratory and diagnostic method rather than a catch-all solution for ecological ordination. Its strength lies in revealing broad structure in multivariate data, i.e., dominant gradients, major axes of covariation, and large-scale patterns that organise samples and variables in a reduced space. When applied to environmental data, PCA can provide a clear initial orientation to highlight correlated drivers, identify major contrasts among sites, and offer a coherent global summary of variation. Its limitations arise from the scope of a PCA: it is constrained to linear relationships under Euclidean geometry, and therefore cannot represent non-linear ecological responses without distortion them. PCA can be useful to establish a baseline against which alternative ordination methods can be evaluated, to guide you toward approaches such as CA, CCA, nMDS, or distance-based methods that are better aligned with specific response forms and inferential goals.</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>© 2026, AJ Smit</p>
</div>   
    <div class="nav-footer-center">

<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB743/PCA.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>
<script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




<script src="../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>