<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.26">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Smit, A. J.">
<meta name="dcterms.date" content="2021-01-01">
<title>4. Data Distributions – The Tangled Bank</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../BCB744/basic_stats/05-inference.html" rel="next">
<link href="../../BCB744/basic_stats/03-visualise.html" rel="prev">
<link href="../../images/favicon.ico" rel="icon">
<script src="../../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-597958c53c93a607afca12fd375c57ed.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-37d9842ca13b32c9c58adfc4ed5d9451.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-SCYH68M489"></script><script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-SCYH68M489', { 'anonymize_ip': true});
</script><script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"headline",
  "consent_type":"express",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  ,
"language":"en"
  });
});
</script><style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script><script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script><script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN" && texText && texText.data) {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script><link rel="stylesheet" href="../../styles/styles.css">
<meta property="og:title" content="4. Data Distributions – The Tangled Bank">
<meta property="og:description" content="Getting familiar with data handling in R">
<meta property="og:image" content="http://tangledbank.netlify.app/BCB744/basic_stats/data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==">
<meta property="og:site_name" content="The Tangled Bank">
</head>
<body class="nav-sidebar floating nav-fixed slimcontent quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">The Tangled Bank</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-vignettes" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Vignettes</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-vignettes">
<li>
    <a class="dropdown-item" href="../../vignettes/netCDF_dates.html">
 <span class="dropdown-text">Dates From netCDF Files: Two Approaches</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/prep_NOAA_OISST.html">
 <span class="dropdown-text">Downloading and Prep: ERDDAP</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_ERDDAP.html">
 <span class="dropdown-text">Retrieving Chlorophyll-a Data from ERDDAP Servers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/alt_method.html">
 <span class="dropdown-text">Downloading and Prep: Python + CDO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/gridded_data.html">
 <span class="dropdown-text">Detecting Events in Gridded Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/regridding.html">
 <span class="dropdown-text">Regridding gridded data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/MHW_MCS_horizonplots.html">
 <span class="dropdown-text">Displaying MHWs and MCSs as Horizon Plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/heatwaveR_issues.html">
 <span class="dropdown-text">heatwaveR Issues</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_sightings.html">
 <span class="dropdown-text">Plotting the Whale Sightings and Chlorophyll-<em>a</em> Concentrations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_localisation.html">
 <span class="dropdown-text">Spatial Localisation, Subsetting, and Aggregation of the Chlorophyll-<em>a</em> Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/elem_ts_methods.html">
 <span class="dropdown-text">Wavelet Analysis of Diatom Time Series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/buffer_data_extract.html">
 <span class="dropdown-text">Extracting gridded data within a buffer</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-high-performance-computing" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">High-Performance Computing</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-high-performance-computing">
<li>
    <a class="dropdown-item" href="../../vignettes/README_Lengau.html">
 <span class="dropdown-text">Using Lengau</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/PBSPro_users.html">
 <span class="dropdown-text">PBSPro user and job management</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/README_tmux.html">
 <span class="dropdown-text">Using tmux</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item">
    <a class="nav-link" href="../../blog.html"> 
<span class="menu-text">Tangled Bank Blog</span></a>
  </li>  
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-blogroll" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Blogroll</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-blogroll">
<li>
    <a class="dropdown-item" href="https://www.r-bloggers.com">
 <span class="dropdown-text">R-bloggers</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item compact">
    <a class="nav-link" href="../../blog.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">
<li>
    <a class="dropdown-item" href="https://github.com/ajsmit/tangled_bank">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
    </ul>
</li>
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../BCB744/BCB744_index.html">BCB744: Introduction to R, &amp; Biostatistics</a></li><li class="breadcrumb-item"><a href="../../BCB744/basic_stats/04-distributions.html">4. Data Distributions</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="../../index.html" class="sidebar-logo-link">
      <img src="../../images/my_tree_hex.png" alt="" class="sidebar-logo light-content py-0 d-lg-inline d-none"><img src="../../images/my_tree_hex.png" alt="" class="sidebar-logo dark-content py-0 d-lg-inline d-none"></a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">UNDERGRADUATE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC223: Plant Ecophysiology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/BDC223_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L00-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introductory Lecture</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L01-worldmapper.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: The Limits to Life</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L02-SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2: Surface Area to Volume (SA:V) Ratio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L03-climate.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3: Earth’s Climate</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L04-carbon_cycle.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Climate Change (Carbon Cycle)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L05-light.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 5: Light</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L06-pigments_photosynthesis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6: Pigments &amp; Photosynthesis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L07-chromatic_adaptation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 7: Chromatic Adaptation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L08a-nutrient_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8a: Nutrient Uptake</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L08b-nutrients_michaelis_menten.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8b: Nutrient Uptake</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L09-plant_stress.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3: Plant Stress</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab1_SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1: Surface Area to Volume (SA/V) Ratios in Biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab2_misc_calcs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2: Miscellaneous Calculations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab3_PI_curves.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3: PI Curves -- Jassby &amp; Platt</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab4_nitrogen_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4: Uptake Kinetics – Michaelis-Menten</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/BDC223_FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC334: Biogeography &amp; Global Ecology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/BDC334_syllabus.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/BDC334-Lecture-Transcripts.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">BCB334 Lecture Transcript</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: Introductory Lecture</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-02-ecosystems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2. Overview of Ecosystems</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-03-gradients.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3. Ecological Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Biodiversity Concepts</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 5: Multivariate Data</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-06-unified-ecology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6: Unified Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 7: Threats to Biodiversity</span>
  </li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 8: Nature's Contribution to People</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/IKS_Conservation.pdf" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 9: Indigenous Knowledge Systems and Conservation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1. Ecological Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-02a-r_rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2a. Introduction `R` &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-02b-env_dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2b. Environmental Distance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-03-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3. Quantifying Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4. Species Distribution Patterns</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/BDC334_AI_Tutor.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">BCB334 AI Tutor</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pages/Transboundary_systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transboundary systems</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS CORE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">BCB744: Introduction to R, &amp; Biostatistics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/BCB744_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/01-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. R &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/02-working-with-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Working With Data &amp; Code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/03-data-in-R.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Data Classes &amp; Structures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/04-workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. R Workflows</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/05-graphics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Graphics with <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/06-faceting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Faceting Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/07-brewing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Brewing Colours</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/08-mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Mapping With <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/09-mapping_style.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Mapping With Style</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/10-mapping_rnaturalearth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Mapping With Natural Earth</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/11-mapping_quakes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. The Fiji Earthquake Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/12-tidy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Tidy Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/13-tidier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Tidier Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/14-tidiest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Tidiest Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/15-recap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15. Recap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/assessments/BCB744_Intro_R_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Intro R Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../self-assessment_v1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Self-Assessment Form</span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
 <span class="menu-text">1. Introduction to Statistics</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/02-summarise-and-describe.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Data Summaries &amp; Descriptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/03-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Statistical Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/04-distributions.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">4. Data Distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/05-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Statistical Inference</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/06-assumptions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Assumptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/07-t_tests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. <em>t</em>-Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/08-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. ANOVA</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/09-regressions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Simple Linear Regressions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/10-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/11-decision_guide.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/12-glance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Non-Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/13-confidence.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Confidence Intervals</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/14-transformations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Data Transformations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/assessments/BCB744_Biostatistics_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Biostatistics Self-Assessment</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS ELECTIVES</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">BCB743: Quantitative Ecology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/BCB743_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/review.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1-4: Review Concepts fo Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5: Correlations &amp; Associations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7: Intro to Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8a: Principal Components Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCA_examples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8b: PCA: Additional Examples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCA_SDG_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8c: PCA: WHO SDG Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/CA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9a: Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/DCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9b: Detrended Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCoA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10: Principal Coordinates Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/nMDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11a: Non-metric Multidimensional Scaling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/nMDS_diatoms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11b: nMDS: PERMANOVA (Diatoms) Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/unconstrained-summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12: Unconstrained Ordination Summary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/constrained_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13a: Distance-Based Redundancy Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/two_oceans_appendices.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13b: db-RDA: Seaweeds Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/cluster_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14: Cluster Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/model_building.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15: Model Building</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/deep_dive.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16a: Deep Dive Into Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/multiple_regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16b: Multiple Regression</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">AI &amp; POLICY</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/Combined Faculty AI Policy 15082025.pdf" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Faculty of Natural Sciences AI Policy</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../AI/BCB_AI_Workshop_Sep_2025.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB AI Workshop</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">SUPPORT</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Web Resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/general_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/ecology_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quantitative Ecology Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/spatial_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial R Resources</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> Introduction</a></li>
  <li><a href="#what-are-data-distributions" id="toc-what-are-data-distributions" class="nav-link" data-scroll-target="#what-are-data-distributions"><span class="header-section-number">2</span> What are data distributions?</a></li>
  <li>
<a href="#discrete-distributions" id="toc-discrete-distributions" class="nav-link" data-scroll-target="#discrete-distributions"><span class="header-section-number">3</span> Discrete distributions</a>
  <ul class="collapse">
<li><a href="#bernoulli-and-binomial-distributions" id="toc-bernoulli-and-binomial-distributions" class="nav-link" data-scroll-target="#bernoulli-and-binomial-distributions"><span class="header-section-number">3.1</span> Bernoulli and Binomial distributions</a></li>
  <li><a href="#negative-binomial-and-geometric-distributions" id="toc-negative-binomial-and-geometric-distributions" class="nav-link" data-scroll-target="#negative-binomial-and-geometric-distributions"><span class="header-section-number">3.2</span> Negative Binomial and Geometric distributions</a></li>
  <li><a href="#poisson-distribution" id="toc-poisson-distribution" class="nav-link" data-scroll-target="#poisson-distribution"><span class="header-section-number">3.3</span> Poisson distribution</a></li>
  <li><a href="#hypergeometric-distribution" id="toc-hypergeometric-distribution" class="nav-link" data-scroll-target="#hypergeometric-distribution"><span class="header-section-number">3.4</span> Hypergeometric distribution</a></li>
  </ul>
</li>
  <li>
<a href="#continuous-distributions" id="toc-continuous-distributions" class="nav-link" data-scroll-target="#continuous-distributions"><span class="header-section-number">4</span> Continuous distributions</a>
  <ul class="collapse">
<li><a href="#sec-normal" id="toc-sec-normal" class="nav-link" data-scroll-target="#sec-normal"><span class="header-section-number">4.1</span> Normal distribution</a></li>
  <li><a href="#uniform-distribution" id="toc-uniform-distribution" class="nav-link" data-scroll-target="#uniform-distribution"><span class="header-section-number">4.2</span> Uniform distribution</a></li>
  <li><a href="#student-t-distribution" id="toc-student-t-distribution" class="nav-link" data-scroll-target="#student-t-distribution"><span class="header-section-number">4.3</span> Student T distribution</a></li>
  <li><a href="#chi-squared-distribution" id="toc-chi-squared-distribution" class="nav-link" data-scroll-target="#chi-squared-distribution"><span class="header-section-number">4.4</span> Chi-squared distribution</a></li>
  <li><a href="#exponential-distribution" id="toc-exponential-distribution" class="nav-link" data-scroll-target="#exponential-distribution"><span class="header-section-number">4.5</span> Exponential distribution</a></li>
  <li><a href="#f-distribution" id="toc-f-distribution" class="nav-link" data-scroll-target="#f-distribution"><span class="header-section-number">4.6</span> F distribution</a></li>
  <li><a href="#gamma-distribution" id="toc-gamma-distribution" class="nav-link" data-scroll-target="#gamma-distribution"><span class="header-section-number">4.7</span> Gamma distribution</a></li>
  <li><a href="#beta-distribution" id="toc-beta-distribution" class="nav-link" data-scroll-target="#beta-distribution"><span class="header-section-number">4.8</span> Beta distribution</a></li>
  </ul>
</li>
  <li><a href="#finding-ones-data-distribution" id="toc-finding-ones-data-distribution" class="nav-link" data-scroll-target="#finding-ones-data-distribution"><span class="header-section-number">5</span> Finding one’s data distribution</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB744/basic_stats/04-distributions.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../BCB744/BCB744_index.html">BCB744: Introduction to R, &amp; Biostatistics</a></li><li class="breadcrumb-item"><a href="../../BCB744/basic_stats/04-distributions.html">4. Data Distributions</a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">4. Data Distributions</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
<p class="subtitle lead">Getting familiar with data handling in R</p>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://tangledbank.netlify.app">Smit, A. J.</a> <a href="https://orcid.org/0000-0002-3799-6126" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uwc.ac.za">
            University of the Western Cape
            </a>
          </p>
      </div>
  </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 1, 2021</p>
    </div>
  </div>
  
    
  </div>
  


</header><div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>In this chapter
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>The concept of data distributions</li>
</ul>
</div>
</div>
<div class="callout callout-style-simple callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Important</span>Tasks to complete in this chapter
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>None</li>
</ul>
</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="../../images/wahlberg_distribution.jpeg" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="../../images/wahlberg_distribution.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" width="600"></a></p>
</figure>
</div>
<div class="callout callout-style-simple callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Important</span>Software used in this chapter
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li><span class="my-highlight">All distribution generation and visualisations functions in this chapter are done using <code><a href="https://github.com/spsanderson/TidyDensity">library(TidyDensity)</a></code> and <code><a href="https://tidyverse.tidyverse.org">library(tidyverse)</a></code>.</span></li>
</ul>
</div>
</div>
<section id="introduction" class="level1" data-number="1"><h1 data-number="1">
<span class="header-section-number">1</span> Introduction</h1>
<p>A good grasp of data distributions is a prerequisite of any statistical analysis. It enables us to describe and summarise the underlying patterns, trends, and variations in our data. This allows for more robust predictions and inferences about natural processes, such as outcomes of experiments or the structure of biodiversity. Conclusions stemming from our application of inferential statistics are defensible only if we understand and can justify the distribution of our data. In this chapter we will learn about the most common data distributions encountered in ecological research, and how this knowledge will help us to effectively apply statistical methods to analyse and interpret them.</p>
</section><section id="what-are-data-distributions" class="level1" data-number="2"><h1 data-number="2">
<span class="header-section-number">2</span> What are data distributions?</h1>
<p>Data distributions are fundamental views through which we understand how values disperse across datasets. They are a mathematical underpinning that transforms our basic numerical observations (which we obtain as <em>samples</em> taken at random to represent a <em>population</em>) into interpretable patterns of <em>frequency</em>, <em>probability</em>, and <em>structural regularity</em>. Distributions show us the underlying “shape” of variation itself and reveal how values are clustered symmetrically (or not) around a central tendency, are skewed toward the extremes, or exhibit multiple peaks that could suggest distinct groupings within the broader dataset.</p>
<p>We need to understand distributions because any collection of measurements (e.g.&nbsp;heights of individuals, rates of nutrient uptake, or counts of animals moving past a point) exhibits characteristic shapes when we plot their frequency against their value. These shapes may be normal, exponential, uniform, bimodal, etc., and are specific to the processes that generated the data. As such, distributions are a statistical link between the the empirical observations and our claims about the underlying mechanisms. Our understanding of the data distribution and the process under scrutiny will therefore inform us of the statistical approaches most suited to extracting meaningful inference from them.</p>
</section><section id="discrete-distributions" class="level1 page-columns page-full" data-number="3"><h1 data-number="3">
<span class="header-section-number">3</span> Discrete distributions</h1>
<p>Discrete distributions form one of two fundamental branches in probability theory; the other is continuous distributions. This division rests on the nature of the sample space: discrete distributions assign probabilities to countable outcomes (integers, finite sets), while continuous distributions apply to uncountable intervals where individual point probabilities equal zero.</p>
<p><em>Discrete random variables</em> have a finite or countable number of possible values and are the foundation of discrete probability distributions. Two core mathematical constraints govern mapping these probability distributions to their discrete random values:</p>
<ul>
<li>First, each probability value must fall within the closed interval [0,1]; no outcome can possess negative probability or exceed certainty.</li>
<li>Second, the sum of all probabilities across the complete sample space must equal exactly 1.0; this ensures that the distribution accounts for all possible outcomes without omission or duplication.</li>
</ul>
<p>In practice, these constraints give us access to practical verification steps:</p>
<ul>
<li>When constructing or validating a discrete distribution, sum all probability values to confirm they equal unity (1).</li>
<li>Inspect each individual probability to ensure none violates the boundary conditions ([0,1]).</li>
</ul>
<p>Violations point to computational errors or incomplete specification of the sample space.</p>
<p>The expected value <span class="math inline">\(E(X)\)</span> of a discrete random variable <span class="math inline">\(X\)</span> represents the probability-weighted average of all possible outcomes: <span class="math inline">\(E(X) = \sum[x \cdot P(X=x)]\)</span>. Let’s use a standard, unbiased die as an example. For each possible value <span class="math inline">\(x\)</span>, multiply that value by its probability, then sum across all possibilities. If <span class="math inline">\(X\)</span> can take values <span class="math inline">\(\{1,2,3,4,5,6\}\)</span> each with probability <span class="math inline">\(1/6\)</span>, then:</p>
<p><span class="math display">\[E(X) = 1(1/6) + 2(1/6) + 3(1/6) + 4(1/6) + 5(1/6) + 6(1/6) = 3.5\]</span></p>
<p>Variance calculation follows a two-step process. First, compute the expected value of the squared outcomes:</p>
<p><span class="math display">\[E(X^2) = \sum[x^2 \cdot P(X=x)]\]</span></p>
<p>Then apply the computational formula:</p>
<p><span class="math display">\[\text{Var}(X) = E(X^2) - [E(X)]^2\]</span></p>
<p>For the die example: <span class="math inline">\(E(X^2) = 1^2(1/6) + 2^2(1/6) + ... + 6^2(1/6) = 91/6 \approx 15.17\)</span>, so <span class="math inline">\(\text{Var}(X) = 15.17 - (3.5)^2 = 15.17 - 12.25 = 2.92\)</span>.</p>
<p>Below I provide options to generate and visualise data belonging to several classes of discrete distributions. In <a href="../../BCB744/basic_stats/06-assumptions.html">Chapter 6</a> we will learn how to transform these data prior to performing the appropriate statistical analysis.</p>
<section id="bernoulli-and-binomial-distributions" class="level2 page-columns page-full" data-number="3.1"><h2 data-number="3.1" class="anchored" data-anchor-id="bernoulli-and-binomial-distributions">
<span class="header-section-number">3.1</span> Bernoulli and Binomial distributions</h2>
<p>The Bernoulli and Binomial distributions belong to what might be termed the “trial-based” subfamily of discrete distributions: they directly model outcomes of repeated experiments. Trial-based distributions require specifying both the number of trials and success probability.</p>
<p>Bernoulli and Binomial distributions are both discrete probability distributions that describe the <em>outcomes of binary events</em>. They are similar but there are also some key differences between the two. In real life examples encountered in ecology and biology we will probably mostly encounter the Binomial distributions. Let us consider each is more detail.</p>
<p><strong>Bernoulli distribution</strong> The Bernoulli distribution represents a single binary trial or experiment with only two possible outcomes: ‘success’ (usually represented as <code>1</code>) and ‘failure’ (usually represented as <code>0</code>). The probability of success is denoted by <span class="math inline">\(p\)</span>, while the probability of failure is <span class="math inline">\(1 - p\)</span>. A Bernoulli distribution is characterised by only one parameter, <span class="math inline">\(p\)</span>, which represents the probability of success for the single trial (<a href="#eq-bernoulli" class="quarto-xref">Equation&nbsp;1</a>) (<a href="#fig-bernoulli1" class="quarto-xref">Figure&nbsp;1</a>, <a href="#fig-bernoulli2" class="quarto-xref">Figure&nbsp;2</a>).</p>

<div class="no-row-height column-margin column-container"><div class="">
<p><strong>The Bernoulli distribution:</strong> <span id="eq-bernoulli"><span class="math display">\[
P(X=k) = \begin{cases}
  p, &amp; \text{if } k=1 \\
  1-p, &amp; \text{if } k=0
\end{cases}
\tag{1}\]</span></span></p>
<p>where <span class="math inline">\(X\)</span> is a random variable, <span class="math inline">\(k\)</span> is the outcome (1 for success and 0 for failure), and <span class="math inline">\(p\)</span> is the probability of success.</p>
</div></div><div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div id="fig-bernoulli1" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-bernoulli1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="04-distributions_files/figure-html/fig-bernoulli1-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;1: Bernoulli distribution with 10 trials and 20 simulations set at a probability of p = 0.2. This could be a heavily loaded coin that has a 20% chance of landing heads."><img src="04-distributions_files/figure-html/fig-bernoulli1-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-bernoulli1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Bernoulli distribution with 10 trials and 20 simulations set at a probability of p = 0.2. This could be a heavily loaded coin that has a 20% chance of landing heads.
</figcaption></figure>
</div>
</div>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div id="fig-bernoulli2" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-bernoulli2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="04-distributions_files/figure-html/fig-bernoulli2-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;2: Bernoulli distribution with 10 trials and 20 simulations set at a probability of p = 0.5. This represents an unbiased coin that has an equal chance of landing on head or tail."><img src="04-distributions_files/figure-html/fig-bernoulli2-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-bernoulli2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Bernoulli distribution with 10 trials and 20 simulations set at a probability of p = 0.5. This represents an unbiased coin that has an equal chance of landing on head or tail.
</figcaption></figure>
</div>
</div>
</div>
<p><strong>Binomial distribution</strong> The Binomial distribution represents the sum of outcomes in a fixed number of independent Bernoulli trials with the same probability of success, <span class="math inline">\(p\)</span>. It is characterised by two parameters, <span class="math inline">\(n\)</span> (the number of trials) and <span class="math inline">\(p\)</span> (the probability of success in each trial). The Binomial distribution describes the probability of obtaining a specific number of successes (<span class="math inline">\(k\)</span>) in <span class="math inline">\(n\)</span> trials (<a href="#eq-binomial" class="quarto-xref">Equation&nbsp;2</a>) (<a href="#fig-binomial2" class="quarto-xref">Figure&nbsp;3</a>).</p>

<!-- ```{r} -->
<!-- #| echo: false -->
<!-- #| fig.align: center -->
<!--  -->
<!--#| fig.cap: "Binomial distribution with 50 trials and 20 simulations at an *a priori* set probability of p = 0.75." -->
<!-- #| label: fig-binomial1 -->
<!-- binom1 <- tidy_binomial(.n = 40, .num_sims = 10, .prob = 0.75) -->
<!-- ggarrange( -->
<!--   tidy_autoplot(binom1, .plot_type = "density"), -->
<!--   tidy_autoplot(binom1, .plot_type = "quantile"), -->
<!--   tidy_autoplot(binom1, .plot_type = "probability"), -->
<!--   tidy_autoplot(binom1, .plot_type = "qq"), -->
<!--   ncol = 2, nrow = 2 -->
<!-- ) -->
<!-- ``` -->
<div class="no-row-height column-margin column-container"><div class="">
<p><strong>The Binomial distribution:</strong> <span id="eq-binomial"><span class="math display">\[P(X=k) = C(n,k) \cdot p^k \cdot (1-p)^{(n-k)} \tag{2}\]</span></span></p>
<p>where <span class="math inline">\(X\)</span> is a random variable, <span class="math inline">\(k\)</span> is the number of successes, <span class="math inline">\(n\)</span> is the total number of trials, <span class="math inline">\(p\)</span> is the probability of success in each trial, and <span class="math inline">\(C(n,k)\)</span> represents combinations.</p>
<p>In practice, determine <span class="math inline">\(n\)</span> (number of trials), <span class="math inline">\(p\)</span> (success probability), and <span class="math inline">\(k\)</span> (desired number of successes). Calculate the binomial coefficient <span class="math inline">\(C(n,k) = \frac{n!}{k!(n-k)!}\)</span>, then apply <a href="#eq-binomial" class="quarto-xref">Equation&nbsp;2</a>. If flipping a coin 10 times (<span class="math inline">\(n=10\)</span>, <span class="math inline">\(p=0.5\)</span>) and seeking exactly 3 heads (<span class="math inline">\(k=3\)</span>): <span class="math inline">\(C(10,3) = 120\)</span>, so <span class="math inline">\(P(X=3) = 120 \cdot (0.5)^3 \cdot (0.5)^7 = 120 \cdot (0.5)^{10} = 0.117\)</span>.</p>
</div></div><div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div id="fig-binomial2" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-binomial2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="04-distributions_files/figure-html/fig-binomial2-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Figure&nbsp;3: Ten simulations of a binomial distribution with 40 randomly generated points for each of 100 trials at an a priori set probability of p = 0.75."><img src="04-distributions_files/figure-html/fig-binomial2-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-binomial2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Ten simulations of a binomial distribution with 40 randomly generated points for each of 100 trials at an <em>a priori</em> set probability of p = 0.75.
</figcaption></figure>
</div>
</div>
</div>
<p>There are several examples of Binomial distributions in ecological and biological contexts. The Binomial distribution is relevant when studying the number of successes in a fixed number of independent trials, each with the same probability of success. A few examples of the Bernoulli distribution:</p>
<ul>
<li><p><strong>Seed germination</strong> Suppose we plant 100 seeds of a particular plant species and wants to know the probability of a certain number of seeds germinating. If the probability of germination for each seed is constant then we can model the number of germinated seeds by a Binomial distribution.</p></li>
<li><p><strong>Disease prevalence</strong> An epidemiologist studies the prevalence of a disease within a population. For a random sample of 500 individuals, and with a fixed probability of an individual having the disease, the number of infected individuals in the sample can be modeled using a Binomial distribution.</p></li>
<li><p><strong>Species occupancy</strong> We do an ecological assessment to determine the occupancy of bird species across 50 habitat patches. If the probability of the species occupying a patch is the same across all patches, the number of patches occupied by the species will follow a Binomial distribution.</p></li>
<li><p><strong>Allele inheritance</strong> We want to examine the inheritance of a specific trait following Mendelian inheritance patterns. If the probability of inheriting the dominant allele for a given gene is constant, the number of offspring with the dominant trait in a fixed number of offspring follows the Binomial distribution.</p></li>
</ul>
<p>Note that in these examples we assume a fixed probability and independence between trials and this is not always be true in real-world situations.</p>
</section><section id="negative-binomial-and-geometric-distributions" class="level2 page-columns page-full" data-number="3.2"><h2 data-number="3.2" class="anchored" data-anchor-id="negative-binomial-and-geometric-distributions">
<span class="header-section-number">3.2</span> Negative Binomial and Geometric distributions</h2>
<p>The Geometric and Negative Binomial distributions form a “waiting time” subfamily, focusing on the number of trials preceding specified success patterns. These waiting-time distributions focus on success probability and target achievement levels.</p>
<p><strong>Negative Binomial distribution</strong> A Negative Binomial random variable, <span class="math inline">\(X\)</span>, counts the number of successes in a sequence of independent Bernoulli trials with probability <span class="math inline">\(p\)</span> before <span class="math inline">\(r\)</span> failures occur. This distribution could for example be used to predict the number of heads that result from a series of coin tosses before three tails are observed (<a href="#eq-negbinomial" class="quarto-xref">Equation&nbsp;3</a>) (<a href="#fig-negbinomial" class="quarto-xref">Figure&nbsp;4</a>).</p>

<div class="no-row-height column-margin column-container"><div class="">
<p><strong>The Negative Binomial distribution:</strong> <span id="eq-negbinomial"><span class="math display">\[P(X=k) = \binom{k+r-1}{k} p^r (1-p)^k \tag{3}\]</span></span></p>
<p>The equation describes the probability mass function (PMF) of a Negative Binomial distribution, where <span class="math inline">\(X\)</span> is a random variable, <span class="math inline">\(k\)</span> is the number of failures, <span class="math inline">\(r\)</span> is the number of successes, and <span class="math inline">\(p\)</span> is the probability of success in each trial. The binomial coefficient is denoted by <span class="math inline">\(\binom{k+r-1}{k}\)</span>, which calculates the number of ways to arrange <span class="math inline">\(k\)</span> failures and <span class="math inline">\(r\)</span> successes such that the last trial is a success.</p>
</div></div><div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div id="fig-negbinomial" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-negbinomial-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="04-distributions_files/figure-html/fig-negbinomial-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Figure&nbsp;4: A negative binomial distribution with 50 trials and 10 simulations at an a priori expectation of p = 0.75."><img src="04-distributions_files/figure-html/fig-negbinomial-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-negbinomial-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: A negative binomial distribution with 50 trials and 10 simulations at an <em>a priori</em> expectation of p = 0.75.
</figcaption></figure>
</div>
</div>
</div>
<p><strong>Geometric distribution</strong> A geometric random variable, <span class="math inline">\(X\)</span>, represents the number of trials that are required to observe a single success. Each trial is independent and has success probability <span class="math inline">\(p\)</span>. As an example, the geometric distribution is useful to model the number of times a die must be tossed in order for a six to be observed (<a href="#eq-geometric" class="quarto-xref">Equation&nbsp;4</a>) (<a href="#fig-geomet" class="quarto-xref">Figure&nbsp;5</a>).</p>

<div class="no-row-height column-margin column-container"><div class="">
<p><strong>The Geometric distribution:</strong> <span id="eq-geometric"><span class="math display">\[P(X=k) = p (1-p)^k \tag{4}\]</span></span></p>
<p>The equation represents the PMF of a Geometric distribution, where <span class="math inline">\(X\)</span> is a random variable, <span class="math inline">\(k\)</span> is the number of failures before the first success, and <span class="math inline">\(p\)</span> is the probability of success in each trial. The Geometric distribution can be thought of as a special case of the Negative Binomial distribution with <span class="math inline">\(r = 1\)</span>, which models the number of failures before achieving a single success.</p>
</div></div><div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div id="fig-geomet" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-geomet-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="04-distributions_files/figure-html/fig-geomet-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6" title="Figure&nbsp;5: A geometric distribution with 50 trials and 10 simulations at an a priori expectation of p = 0.75."><img src="04-distributions_files/figure-html/fig-geomet-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-geomet-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: A geometric distribution with 50 trials and 10 simulations at an <em>a priori</em> expectation of p = 0.75.
</figcaption></figure>
</div>
</div>
</div>
</section><section id="poisson-distribution" class="level2 page-columns page-full" data-number="3.3"><h2 data-number="3.3" class="anchored" data-anchor-id="poisson-distribution">
<span class="header-section-number">3.3</span> Poisson distribution</h2>

<div class="no-row-height column-margin column-container"><div class="">
<p><strong>The Poisson distribution:</strong> <span id="eq-poisson"><span class="math display">\[P(X=k) = \frac{e^{-\lambda} \lambda^k}{k!} \tag{5}\]</span></span></p>
<p>The function represents the PMF of a Poisson distribution, where <span class="math inline">\(X\)</span> is a random variable, <span class="math inline">\(k\)</span> is the number of events or occurrences, and <span class="math inline">\(\lambda\)</span> (lambda) is the average rate of occurrences (events per unit of time or space). The constant <span class="math inline">\(e\)</span> is the base of the natural logarithm, and <span class="math inline">\(k!\)</span> is the factorial of <span class="math inline">\(k\)</span>. The Poisson distribution is commonly used to model the number of events occurring within a fixed interval of time or space when events occur independently and at a constant average rate.</p>
<p>For practical application, identify <span class="math inline">\(\lambda\)</span> (average rate) and <span class="math inline">\(k\)</span> (observed count). If a gerbil enters a nest 4 times per hour (<span class="math inline">\(\lambda=4\)</span>) and you want the probability of exactly 2 entries in one hour (<span class="math inline">\(k=2\)</span>): <span class="math inline">\(P(X=2) = \frac{4^2 \cdot e^{-4}}{2!} = \frac{16 \cdot 0.0183}{2} = 0.147\)</span>.</p>
</div></div><p>The Poisson distribution represents a “rate-based” subfamily, which is used to model count phenomena without explicit trial structure. They involve the average occurrence rates over specified intervals.</p>
<p>A Poisson random variable, <span class="math inline">\(X\)</span>, tallies the number of events occurring in a fixed interval of time or space, given that these events occur with an average rate <span class="math inline">\(\lambda\)</span>. Poisson distributions can be used to model events such as meteor showers and or number of people entering a shopping mall (<a href="#eq-poisson" class="quarto-xref">Equation&nbsp;5</a>) (<a href="#fig-poisson" class="quarto-xref">Figure&nbsp;6</a>).</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div id="fig-poisson" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-poisson-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="04-distributions_files/figure-html/fig-poisson-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7" title="Figure&nbsp;6: A Poisson distribution with 50 trials and 10 simulations at an a priori expectation of p = 0.75."><img src="04-distributions_files/figure-html/fig-poisson-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-poisson-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: A Poisson distribution with 50 trials and 10 simulations at an <em>a priori</em> expectation of p = 0.75.
</figcaption></figure>
</div>
</div>
</div>
</section><section id="hypergeometric-distribution" class="level2 page-columns page-full" data-number="3.4"><h2 data-number="3.4" class="anchored" data-anchor-id="hypergeometric-distribution">
<span class="header-section-number">3.4</span> Hypergeometric distribution</h2>

<div class="no-row-height column-margin column-container"><div class="">
<p><strong>The Hypergeometricd distribution:</strong> <span id="eq-hypergeometric"><span class="math display">\[P(X=k) = \frac{C(K,k) \cdot C(N-K,n-k)}{C(N,n)} \tag{6}\]</span></span> In the probability mass function, above, <span class="math inline">\(N\)</span> is population size, <span class="math inline">\(K\)</span> is number of success items in population, <span class="math inline">\(n\)</span> is sample size, and <span class="math inline">\(k\)</span> is observed successes in sample.</p>
<p>Consider drawing 5 cards from a deck without replacement, seeking exactly 2 hearts. Here <span class="math inline">\(N=52\)</span>, <span class="math inline">\(K=13\)</span> (hearts), <span class="math inline">\(n=5\)</span>, <span class="math inline">\(k=2\)</span>: <span class="math inline">\(P(X=2) = \frac{C(13,2) \cdot C(39,3)}{C(52,5)} = \frac{78 \cdot 9139}{2,598,960} = 0.274\)</span>.</p>
</div></div><p>The hypergeometric distribution occupies a distinct position within the discrete distribution taxonomy: we might term the “finite population sampling” subfamily.</p>
<p>The hypergeometric distribution models sampling without replacement from finite populations containing two types of items. Unlike binomial sampling, each draw changes the composition of remaining items.</p>
<p><strong>To do:</strong> insert figures.</p>
</section></section><section id="continuous-distributions" class="level1 page-columns page-full" data-number="4"><h1 data-number="4">
<span class="header-section-number">4</span> Continuous distributions</h1>
<section id="sec-normal" class="level2 page-columns page-full" data-number="4.1"><h2 data-number="4.1" class="anchored" data-anchor-id="sec-normal">
<span class="header-section-number">4.1</span> Normal distribution</h2>

<div class="no-row-height column-margin column-container"><div class="">
<p><strong>The Normal distribution:</strong> <span id="eq-normal"><span class="math display">\[f(x) = \frac{1}{\sigma\sqrt{2\pi}} e^{ -\frac{1}{2} \left(\frac{x-\mu}{\sigma}\right)^2 }
\tag{7}\]</span></span></p>
<p>where <span class="math inline">\(x\)</span> is a continuous random variable, <span class="math inline">\(\mu\)</span> (mu) is the mean, and <span class="math inline">\(\sigma\)</span> (sigma) is the standard deviation. The constant factor <span class="math inline">\(\frac{1}{\sigma\sqrt{2\pi}}\)</span> ensures that the Probability Density Function (PDF) integrates to 1, and the exponential term is responsible for the characteristic bell-shaped curve of the Normal distribution.</p>
</div><div class="">
<div id="fig-expectednormal" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-expectednormal-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../../images/Standard_deviation_diagram.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-8" title="Figure&nbsp;7: The idealised Normal distribution showing the proportion of data within 1, 2, and 3SD from the mean."><img src="../../images/Standard_deviation_diagram.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-expectednormal-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: The idealised Normal distribution showing the proportion of data within 1, 2, and 3SD from the mean.
</figcaption></figure>
</div>
</div></div>
<p>Another name for this kind of distribution is a Gaussian distribution. A random sample with a Gaussian distribution is normally distributed. These values are <em>independent and identically distributed</em> random variables (i.i.d.), and they have an expected mean given by <span class="math inline">\(\mu\)</span> (or <span class="math inline">\(\hat{x}\)</span> in Chapter 3.2.1) and a finite variance given by <span class="math inline">\(\sigma^{2}\)</span> (or <span class="math inline">\(S^{2}\)</span> in Chapter 3.3.1); if the number of samples drawn from a population is sufficiently large, the estimated mean and SD will be indistinguishable from the population (as per the central limit theorem). It is represented by <a href="#eq-normal" class="quarto-xref">Equation&nbsp;7</a> (<a href="#fig-normal" class="quarto-xref">Figure&nbsp;8</a>).</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div id="fig-normal" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-normal-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="04-distributions_files/figure-html/fig-normal-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-9" title="Figure&nbsp;8: Normal distribution with 40 trials and 5 simulations."><img src="04-distributions_files/figure-html/fig-normal-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-normal-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8: Normal distribution with 40 trials and 5 simulations.
</figcaption></figure>
</div>
</div>
</div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Central Limit Theorem
</div>
</div>
<div class="callout-body-container callout-body">
<p>The Central Limit Theorem (CLT) is a fundamental result in probability theory and statistics, which states that the distribution of the sum (or average) of a large number of independent, identically distributed (IID) random variables approaches a Normal distribution regardless of the shape of the original distribution. So, the CLT asserts that the Normal distribution is the limiting distribution for the sum or average of many random variables, as long as certain conditions are met.</p>
<p>The CLT provides a basis for making inferences about population parameters using sample statistics. For example, when dealing with large sample sizes, the sampling distribution of the sample mean is approximately normally distributed, even if the underlying population distribution is not normal. This allows us to apply inferential techniques based on the Normal distribution, such as hypothesis testing and constructing confidence intervals, to estimate population parameters using sample data.</p>
<p>Some conditions must be met for the CLT to be true:</p>
<ul>
<li>
<strong>The random variables must be independent</strong> The observations should not be influenced by one another.</li>
<li>
<strong>The random variables must be identically distributed</strong> They must come from the same population with the same mean and variance.</li>
<li>
<strong>The number of random variables (sample size) must be sufficiently large</strong> Although there is no strict rule for the sample size, a common rule of thumb is that the sample size should be at least 30 for the CLT to be a reasonable approximation.</li>
</ul>
</div>
</div>
</section><section id="uniform-distribution" class="level2" data-number="4.2"><h2 data-number="4.2" class="anchored" data-anchor-id="uniform-distribution">
<span class="header-section-number">4.2</span> Uniform distribution</h2>
<p>The continuous uniform distribution is sometime called a rectangular distribution. Simply, it states that all measurements of the same magnitude included with this distribution are equally probable. This is basically random numbers (<a href="#fig-unif" class="quarto-xref">Figure&nbsp;9</a>).</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div id="fig-unif" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-unif-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="04-distributions_files/figure-html/fig-unif-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-10" title="Figure&nbsp;9: Uniform distribution with 100 trials and 5 simulations."><img src="04-distributions_files/figure-html/fig-unif-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-unif-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;9: Uniform distribution with 100 trials and 5 simulations.
</figcaption></figure>
</div>
</div>
</div>
</section><section id="student-t-distribution" class="level2" data-number="4.3"><h2 data-number="4.3" class="anchored" data-anchor-id="student-t-distribution">
<span class="header-section-number">4.3</span> Student T distribution</h2>
<p>This is a continuous probability distribution that arises when estimating the mean of a normally distributed population in situations where the sample size is small and population standard deviation is unknown. It is used in the statistical significance testing between the means of different sets of samples, and not much so in the modelling of many kinds of experiments or observations (<a href="#fig-studt" class="quarto-xref">Figure&nbsp;10</a>).</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div id="fig-studt" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-studt-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="04-distributions_files/figure-html/fig-studt-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-11" title="Figure&nbsp;10: Uniform distribution with 100 trials and 5 simulations."><img src="04-distributions_files/figure-html/fig-studt-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-studt-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10: Uniform distribution with 100 trials and 5 simulations.
</figcaption></figure>
</div>
</div>
</div>
</section><section id="chi-squared-distribution" class="level2" data-number="4.4"><h2 data-number="4.4" class="anchored" data-anchor-id="chi-squared-distribution">
<span class="header-section-number">4.4</span> Chi-squared distribution</h2>
<p>Mostly used in hypothesis testing, but not to encapsulate the distribution of data drawn to represent natural phenomena (<a href="#fig-chi" class="quarto-xref">Figure&nbsp;11</a>).</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div id="fig-chi" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-chi-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="04-distributions_files/figure-html/fig-chi-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-12" title="Figure&nbsp;11: Chi distribution with 100 trials and 5 simulations."><img src="04-distributions_files/figure-html/fig-chi-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-chi-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;11: Chi distribution with 100 trials and 5 simulations.
</figcaption></figure>
</div>
</div>
</div>
</section><section id="exponential-distribution" class="level2" data-number="4.5"><h2 data-number="4.5" class="anchored" data-anchor-id="exponential-distribution">
<span class="header-section-number">4.5</span> Exponential distribution</h2>
<p>This is a probability distribution that describes the time between events in a Poisson point process, i.e., a process in which events occur continuously and independently at a constant average rate (<a href="#fig-exp" class="quarto-xref">Figure&nbsp;12</a>).</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div id="fig-exp" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-exp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="04-distributions_files/figure-html/fig-exp-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-13" title="Figure&nbsp;12: An exponential distribution with 100 trials and 5 simulations."><img src="04-distributions_files/figure-html/fig-exp-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-exp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12: An exponential distribution with 100 trials and 5 simulations.
</figcaption></figure>
</div>
</div>
</div>
</section><section id="f-distribution" class="level2" data-number="4.6"><h2 data-number="4.6" class="anchored" data-anchor-id="f-distribution">
<span class="header-section-number">4.6</span> F distribution</h2>
<p>This is a probability distribution that arises in the context of the analysis of variance (ANOVA) and regression analysis. It is used to compare the variances of two populations (<a href="#fig-f" class="quarto-xref">Figure&nbsp;13</a>).</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div id="fig-f" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-f-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="04-distributions_files/figure-html/fig-f-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-14" title="Figure&nbsp;13: An F distribution with 100 trials and 5 simulations."><img src="04-distributions_files/figure-html/fig-f-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-f-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13: An F distribution with 100 trials and 5 simulations.
</figcaption></figure>
</div>
</div>
</div>
</section><section id="gamma-distribution" class="level2" data-number="4.7"><h2 data-number="4.7" class="anchored" data-anchor-id="gamma-distribution">
<span class="header-section-number">4.7</span> Gamma distribution</h2>
<p>This is a two-parameter family of continuous probability distributions. It is used to model the time until an event occurs. It is a generalisation of the exponential distribution (<a href="#fig-gamma" class="quarto-xref">Figure&nbsp;14</a>).</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div id="fig-gamma" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-gamma-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="04-distributions_files/figure-html/fig-gamma-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-15" title="Figure&nbsp;14: A Gamma distribution with 100 trials and 5 simulations."><img src="04-distributions_files/figure-html/fig-gamma-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-gamma-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;14: A Gamma distribution with 100 trials and 5 simulations.
</figcaption></figure>
</div>
</div>
</div>
</section><section id="beta-distribution" class="level2" data-number="4.8"><h2 data-number="4.8" class="anchored" data-anchor-id="beta-distribution">
<span class="header-section-number">4.8</span> Beta distribution</h2>
<p>This is a family of continuous probability distributions defined on the interval [0, 1] parameterised by two positive shape parameters, typically denoted by α and β. It is used to model the behaviour of random variables limited to intervals of finite length in a wide variety of disciplines (<a href="#fig-beta" class="quarto-xref">Figure&nbsp;15</a>).</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div id="fig-beta" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-beta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="04-distributions_files/figure-html/fig-beta-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-16" title="Figure&nbsp;15: A Beta distribution with 100 trials and 5 simulations."><img src="04-distributions_files/figure-html/fig-beta-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-beta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;15: A Beta distribution with 100 trials and 5 simulations.
</figcaption></figure>
</div>
</div>
</div>
</section></section><section id="finding-ones-data-distribution" class="level1 page-columns page-full" data-number="5"><h1 data-number="5">
<span class="header-section-number">5</span> Finding one’s data distribution</h1>
<p>Data belonging to a sample will never exactly follow a specific distribution, even when the test for normality says it does—there will always be a small probability that they are non-normal and is in fact better described by some other distribution. In other words, data are only <em>compatible</em> with a certain distribution, and one can never answer the question “Does my data follow the distribution xy exactly?” as simply as providing a yes/no answer. So what now? How does one find one’s data distribution? We can use the <strong>Cullen and Frey graph</strong> function that lives in the <strong>fitdistrplus</strong> package. This graph tells us whether the skewness and kurtosis of our data are consistent with that of a particular distribution. We will demonstrate by generating various data distributions and testing them using the Cullen and Frey graph.</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(fitdistrplus)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(logspline)</span>
<span id="cb1-3"><a href="#cb1-3"></a></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="co"># Generate log-normal data</span></span>
<span id="cb1-5"><a href="#cb1-5"></a>y <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">37.50</span>,<span class="fl">46.79</span>,<span class="fl">48.30</span>,<span class="fl">46.04</span>,<span class="fl">43.40</span>,<span class="fl">39.25</span>,<span class="fl">38.49</span>,<span class="fl">49.51</span>,<span class="fl">40.38</span>,<span class="fl">36.98</span>,<span class="fl">40.00</span>,</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fl">38.49</span>,<span class="fl">37.74</span>,<span class="fl">47.92</span>,<span class="fl">44.53</span>,<span class="fl">44.91</span>,<span class="fl">44.91</span>,<span class="fl">40.00</span>,<span class="fl">41.51</span>,<span class="fl">47.92</span>,<span class="fl">36.98</span>,<span class="fl">43.40</span>,</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fl">42.26</span>,<span class="fl">41.89</span>,<span class="fl">38.87</span>,<span class="fl">43.02</span>,<span class="fl">39.25</span>,<span class="fl">40.38</span>,<span class="fl">42.64</span>,<span class="fl">36.98</span>,<span class="fl">44.15</span>,<span class="fl">44.91</span>,<span class="fl">43.40</span>,</span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="fl">49.81</span>,<span class="fl">38.87</span>,<span class="fl">40.00</span>,<span class="fl">52.45</span>,<span class="fl">53.13</span>,<span class="fl">47.92</span>,<span class="fl">52.45</span>,<span class="fl">44.91</span>,<span class="fl">29.54</span>,<span class="fl">27.13</span>,<span class="fl">35.60</span>,</span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="fl">45.34</span>,<span class="fl">43.37</span>,<span class="fl">54.15</span>,<span class="fl">42.77</span>,<span class="fl">42.88</span>,<span class="fl">44.26</span>,<span class="fl">27.14</span>,<span class="fl">39.31</span>,<span class="fl">24.80</span>,<span class="fl">16.62</span>,<span class="fl">30.30</span>,</span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="fl">36.39</span>,<span class="fl">28.60</span>,<span class="fl">28.53</span>,<span class="fl">35.84</span>,<span class="fl">31.10</span>,<span class="fl">34.55</span>,<span class="fl">52.65</span>,<span class="fl">48.81</span>,<span class="fl">43.42</span>,<span class="fl">52.49</span>,<span class="fl">38.00</span>,</span>
<span id="cb1-11"><a href="#cb1-11"></a><span class="fl">38.65</span>,<span class="fl">34.54</span>,<span class="fl">37.70</span>,<span class="fl">38.11</span>,<span class="fl">43.05</span>,<span class="fl">29.95</span>,<span class="fl">32.48</span>,<span class="fl">24.63</span>,<span class="fl">35.33</span>,<span class="fl">41.34</span>)</span>
<span id="cb1-12"><a href="#cb1-12"></a></span>
<span id="cb1-13"><a href="#cb1-13"></a><span class="fu">plot</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y)), <span class="at">y =</span> y)</span>
<span id="cb1-14"><a href="#cb1-14"></a><span class="fu">hist</span>(y)</span>
<span id="cb1-15"><a href="#cb1-15"></a><span class="fu">descdist</span>(y, <span class="at">discrete =</span> <span class="cn">FALSE</span>, <span class="at">boot =</span> <span class="dv">100</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell column-screen-inset-shaded quarto-layout-panel" data-layout-nrow="1" data-layout-align="center">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="04-distributions_files/figure-html/dist-plot1-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-17"><img src="04-distributions_files/figure-html/dist-plot1-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a></p>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="04-distributions_files/figure-html/dist-plot1-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-18"><img src="04-distributions_files/figure-html/dist-plot1-2.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a></p>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="04-distributions_files/figure-html/dist-plot1-3.png" class="lightbox" data-gallery="quarto-lightbox-gallery-19"><img src="04-distributions_files/figure-html/dist-plot1-3.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<pre><code>summary statistics
------
min:  16.62   max:  54.15 
median:  40.38 
mean:  40.28434 
estimated sd:  7.420034 
estimated skewness:  -0.551717 
estimated kurtosis:  3.565162 </code></pre>
</div>
</div>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># normally distributed data</span></span>
<span id="cb3-2"><a href="#cb3-2"></a>y <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>, <span class="dv">13</span>, <span class="dv">2</span>)</span>
<span id="cb3-3"><a href="#cb3-3"></a></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="fu">plot</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>), <span class="at">y =</span> y)</span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="fu">hist</span>(y)</span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="fu">descdist</span>(y, <span class="at">discrete =</span> <span class="cn">FALSE</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell column-screen-inset-shaded quarto-layout-panel" data-layout-nrow="1" data-layout-align="center">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="04-distributions_files/figure-html/dist-plot2-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-20"><img src="04-distributions_files/figure-html/dist-plot2-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a></p>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="04-distributions_files/figure-html/dist-plot2-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-21"><img src="04-distributions_files/figure-html/dist-plot2-2.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a></p>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="04-distributions_files/figure-html/dist-plot2-3.png" class="lightbox" data-gallery="quarto-lightbox-gallery-22"><img src="04-distributions_files/figure-html/dist-plot2-3.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<pre><code>summary statistics
------
min:  9.113063   max:  19.7655 
median:  13.26043 
mean:  13.30906 
estimated sd:  2.055262 
estimated skewness:  0.2559551 
estimated kurtosis:  2.832468 </code></pre>
</div>
</div>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="co"># uniformly distributed data</span></span>
<span id="cb5-2"><a href="#cb5-2"></a>y <span class="ot">&lt;-</span> <span class="fu">runif</span>(<span class="dv">100</span>)</span>
<span id="cb5-3"><a href="#cb5-3"></a></span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="fu">plot</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>), <span class="at">y =</span> y)</span>
<span id="cb5-5"><a href="#cb5-5"></a><span class="fu">hist</span>(y)</span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="fu">descdist</span>(y, <span class="at">discrete =</span> <span class="cn">FALSE</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell column-screen-inset-shaded quarto-layout-panel" data-layout-nrow="1" data-layout-align="center">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="04-distributions_files/figure-html/dist-plot3-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-23"><img src="04-distributions_files/figure-html/dist-plot3-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a></p>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="04-distributions_files/figure-html/dist-plot3-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-24"><img src="04-distributions_files/figure-html/dist-plot3-2.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a></p>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="04-distributions_files/figure-html/dist-plot3-3.png" class="lightbox" data-gallery="quarto-lightbox-gallery-25"><img src="04-distributions_files/figure-html/dist-plot3-3.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<pre><code>summary statistics
------
min:  0.0009542492   max:  0.9882974 
median:  0.5088009 
mean:  0.4940067 
estimated sd:  0.2761684 
estimated skewness:  -0.03710978 
estimated kurtosis:  1.898323 </code></pre>
</div>
</div>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co"># uniformly distributed data</span></span>
<span id="cb7-2"><a href="#cb7-2"></a>y <span class="ot">&lt;-</span> <span class="fu">rexp</span>(<span class="dv">100</span>, <span class="fl">0.7</span>)</span>
<span id="cb7-3"><a href="#cb7-3"></a></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="fu">plot</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>), <span class="at">y =</span> y)</span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="fu">hist</span>(y)</span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="fu">descdist</span>(y, <span class="at">discrete =</span> <span class="cn">FALSE</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell column-screen-inset-shaded quarto-layout-panel" data-layout-nrow="1" data-layout-align="center" height="10" width="6">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="04-distributions_files/figure-html/dist-plot4-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-26"><img src="04-distributions_files/figure-html/dist-plot4-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a></p>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="04-distributions_files/figure-html/dist-plot4-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-27"><img src="04-distributions_files/figure-html/dist-plot4-2.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a></p>
</figure>
</div>
</div>
<div class="quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="04-distributions_files/figure-html/dist-plot4-3.png" class="lightbox" data-gallery="quarto-lightbox-gallery-28"><img src="04-distributions_files/figure-html/dist-plot4-3.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></a></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout quarto-layout-cell" style="flex-basis: 25.0%;justify-content: center;">
<pre><code>summary statistics
------
min:  0.01627003   max:  8.223174 
median:  0.9430276 
mean:  1.573511 
estimated sd:  1.686057 
estimated skewness:  1.785293 
estimated kurtosis:  6.081988 </code></pre>
</div>
</div>
</div>
<p>There is also a whole bunch of other approaches to use to try and identify the data distribution. Let us start with the gold standard first: normal data. We will demonstrate some visualisation approaches. The one that you already know is a basic histogram; it tells us something about the distribution’s skewness, the tails, the mode(s) of the data, outliers, etc. Histograms can be compared to shapes associated with idealistic (simulated) distributions, as we will do here.</p>
<div class="cell" data-layout-align="center">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>y <span class="ot">&lt;-</span><span class="fu">rnorm</span>(<span class="at">n =</span> <span class="dv">200</span>, <span class="at">m =</span> <span class="dv">13</span>, <span class="at">sd =</span> <span class="dv">2</span>)</span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>))</span>
<span id="cb9-3"><a href="#cb9-3"></a><span class="co"># using some basic base graphics as ggplot2 is overkill;</span></span>
<span id="cb9-4"><a href="#cb9-4"></a><span class="co"># we can get a histogram using hist() statement</span></span>
<span id="cb9-5"><a href="#cb9-5"></a><span class="fu">hist</span>(y, <span class="at">main =</span> <span class="st">"Histogram of observed data"</span>)</span>
<span id="cb9-6"><a href="#cb9-6"></a><span class="fu">plot</span>(<span class="fu">density</span>(y), <span class="at">main =</span> <span class="st">"Density estimate of data"</span>)</span>
<span id="cb9-7"><a href="#cb9-7"></a><span class="fu">plot</span>(<span class="fu">ecdf</span>(y), <span class="at">main =</span> <span class="st">"Empirical cumulative distribution function"</span>)</span>
<span id="cb9-8"><a href="#cb9-8"></a><span class="co"># standardise the data</span></span>
<span id="cb9-9"><a href="#cb9-9"></a>z.norm <span class="ot">&lt;-</span> (y <span class="sc">-</span> <span class="fu">mean</span>(y)) <span class="sc">/</span> <span class="fu">sd</span>(y) </span>
<span id="cb9-10"><a href="#cb9-10"></a><span class="co"># make a qqplot</span></span>
<span id="cb9-11"><a href="#cb9-11"></a><span class="fu">qqnorm</span>(z.norm)</span>
<span id="cb9-12"><a href="#cb9-12"></a><span class="co"># add a 45-degree reference line</span></span>
<span id="cb9-13"><a href="#cb9-13"></a><span class="fu">abline</span>(<span class="dv">0</span>, <span class="dv">1</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="04-distributions_files/figure-html/dist-plot5-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-29"><img src="04-distributions_files/figure-html/dist-plot5-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></a></p>
</figure>
</div>
</div>
</div>
<p>Above we have also added a diagonal line to the qqplot. If the sampled data come from the population with the chosen distribution, the points should fall approximately along this reference line. The greater the departure from this reference line, the greater the evidence for the conclusion that the data set have come from a population with a different distribution.</p>
<div class="cell" data-layout-align="center">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="co"># curve(dnorm(100, m = 10, sd = 2), from = 0, to = 20, main = "Normal distribution")</span></span>
<span id="cb10-2"><a href="#cb10-2"></a><span class="co"># curve(dgamma(100, scale = 1.5, shape = 2), from = 0, to = 15, main = "Gamma distribution")</span></span>
<span id="cb10-3"><a href="#cb10-3"></a><span class="co"># curve(dweibull(100, scale = 2.5, shape = 1.5), from = 0, to = 15, main = "Weibull distribution")</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<!-- ## Exercises -->
<!-- ### Exercise 1 -->
<!-- Choose two different datasets and plot them as histograms with density curves overlayed. Label them with the distribution they appear to be and stitch them together with `ggarrange()`. -->


<!-- -->

</section><div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-reuse"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a></div></div></section><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{smit,_a._j.2021,
  author = {Smit, A. J.,},
  title = {4. {Data} {Distributions}},
  date = {2021-01-01},
  url = {http://tangledbank.netlify.app/BCB744/basic_stats/04-distributions.html},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-smit,_a._j.2021" class="csl-entry quarto-appendix-citeas" role="listitem">
Smit, A. J. (2021) 4. Data Distributions. <a href="http://tangledbank.netlify.app/BCB744/basic_stats/04-distributions.html">http://tangledbank.netlify.app/BCB744/basic_stats/04-distributions.html.
</a>
</div></div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("http:\/\/tangledbank\.netlify\.app");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
            // default icon
            link.classList.add("external");
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="../../BCB744/basic_stats/03-visualise.html" class="pagination-link" aria-label="3. Statistical Figures">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">3. Statistical Figures</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../BCB744/basic_stats/05-inference.html" class="pagination-link" aria-label="5. Statistical Inference">
        <span class="nav-page-text">5. Statistical Inference</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb11" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb11-1"><a href="#cb11-1"></a><span class="co">---</span></span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="an">date:</span><span class="co"> "2021-01-01"</span></span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="an">title:</span><span class="co"> "4. Data Distributions"</span></span>
<span id="cb11-4"><a href="#cb11-4"></a><span class="an">subtitle:</span><span class="co"> "Getting familiar with data handling in R"</span></span>
<span id="cb11-5"><a href="#cb11-5"></a><span class="an">execute:</span></span>
<span id="cb11-6"><a href="#cb11-6"></a><span class="co">  cache: FALSE</span></span>
<span id="cb11-7"><a href="#cb11-7"></a><span class="co">---</span></span>
<span id="cb11-8"><a href="#cb11-8"></a></span>
<span id="cb11-9"><a href="#cb11-9"></a><span class="in">```{r, echo=FALSE}</span></span>
<span id="cb11-10"><a href="#cb11-10"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(</span>
<span id="cb11-11"><a href="#cb11-11"></a>  <span class="at">echo =</span> <span class="cn">TRUE</span>,</span>
<span id="cb11-12"><a href="#cb11-12"></a>  <span class="at">eval =</span> <span class="cn">TRUE</span>,</span>
<span id="cb11-13"><a href="#cb11-13"></a>  <span class="at">warning =</span> <span class="cn">FALSE</span>,</span>
<span id="cb11-14"><a href="#cb11-14"></a>  <span class="at">message =</span> <span class="cn">FALSE</span>,</span>
<span id="cb11-15"><a href="#cb11-15"></a>  <span class="at">fig.width =</span> <span class="dv">8</span>,</span>
<span id="cb11-16"><a href="#cb11-16"></a>  <span class="at">fig.asp =</span> <span class="fl">0.65</span>,</span>
<span id="cb11-17"><a href="#cb11-17"></a>  <span class="at">out.width =</span> <span class="st">"100%"</span>,</span>
<span id="cb11-18"><a href="#cb11-18"></a>  <span class="at">fig.align =</span> <span class="st">"center"</span></span>
<span id="cb11-19"><a href="#cb11-19"></a>)</span>
<span id="cb11-20"><a href="#cb11-20"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb11-21"><a href="#cb11-21"></a><span class="fu">library</span>(TidyDensity)</span>
<span id="cb11-22"><a href="#cb11-22"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb11-23"><a href="#cb11-23"></a><span class="in">```</span></span>
<span id="cb11-24"><a href="#cb11-24"></a></span>
<span id="cb11-25"><a href="#cb11-25"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb11-26"><a href="#cb11-26"></a><span class="fu">## In this chapter</span></span>
<span id="cb11-27"><a href="#cb11-27"></a></span>
<span id="cb11-28"><a href="#cb11-28"></a><span class="ss">-   </span>The concept of data distributions</span>
<span id="cb11-29"><a href="#cb11-29"></a>:::</span>
<span id="cb11-30"><a href="#cb11-30"></a></span>
<span id="cb11-31"><a href="#cb11-31"></a>::: {.callout-important appearance="simple"}</span>
<span id="cb11-32"><a href="#cb11-32"></a><span class="fu">## Tasks to complete in this chapter</span></span>
<span id="cb11-33"><a href="#cb11-33"></a></span>
<span id="cb11-34"><a href="#cb11-34"></a><span class="ss">-   </span>None</span>
<span id="cb11-35"><a href="#cb11-35"></a>:::</span>
<span id="cb11-36"><a href="#cb11-36"></a></span>
<span id="cb11-37"><a href="#cb11-37"></a><span class="al">![](/images/wahlberg_distribution.jpeg)</span>{fig-align="center" width="600"}</span>
<span id="cb11-38"><a href="#cb11-38"></a></span>
<span id="cb11-39"><a href="#cb11-39"></a>::: {.callout-important appearance="simple"}</span>
<span id="cb11-40"><a href="#cb11-40"></a><span class="fu">## Software used in this chapter</span></span>
<span id="cb11-41"><a href="#cb11-41"></a></span>
<span id="cb11-42"><a href="#cb11-42"></a><span class="ss">-   </span><span class="co">[</span><span class="ot">All distribution generation and visualisations functions in this chapter are done using `library(TidyDensity)` and `library(tidyverse)`.</span><span class="co">]</span>{.my-highlight}</span>
<span id="cb11-43"><a href="#cb11-43"></a>:::</span>
<span id="cb11-44"><a href="#cb11-44"></a></span>
<span id="cb11-45"><a href="#cb11-45"></a><span class="fu"># Introduction</span></span>
<span id="cb11-46"><a href="#cb11-46"></a></span>
<span id="cb11-47"><a href="#cb11-47"></a>A good grasp of data distributions is a prerequisite of any statistical analysis. It enables us to describe and summarise the underlying patterns, trends, and variations in our data. This allows for more robust predictions and inferences about natural processes, such as outcomes of experiments or the structure of biodiversity. Conclusions stemming from our application of inferential statistics are defensible only if we understand and can justify the distribution of our data. In this chapter we will learn about the most common data distributions encountered in ecological research, and how this knowledge will help us to effectively apply statistical methods to analyse and interpret them.</span>
<span id="cb11-48"><a href="#cb11-48"></a></span>
<span id="cb11-49"><a href="#cb11-49"></a><span class="fu"># What are data distributions?</span></span>
<span id="cb11-50"><a href="#cb11-50"></a></span>
<span id="cb11-51"><a href="#cb11-51"></a>Data distributions are fundamental views through which we understand how values disperse across datasets. They are a mathematical underpinning that transforms our basic numerical observations (which we obtain as *samples* taken at random to represent a *population*) into interpretable patterns of *frequency*, *probability*, and *structural regularity*. Distributions show us the underlying "shape" of variation itself and reveal how values are clustered symmetrically (or not) around a central tendency, are skewed toward the extremes, or exhibit multiple peaks that could suggest distinct groupings within the broader dataset.</span>
<span id="cb11-52"><a href="#cb11-52"></a></span>
<span id="cb11-53"><a href="#cb11-53"></a>We need to understand distributions because any collection of measurements (e.g. heights of individuals, rates of nutrient uptake, or counts of animals moving past a point) exhibits characteristic shapes when we plot their frequency against their value. These shapes may be normal, exponential, uniform, bimodal, etc., and are specific to the processes that generated the data. As such, distributions are a statistical link between the the empirical observations and our claims about the underlying mechanisms. Our understanding of the data distribution and the process under scrutiny will therefore inform us of the statistical approaches most suited to extracting meaningful inference from them.</span>
<span id="cb11-54"><a href="#cb11-54"></a></span>
<span id="cb11-55"><a href="#cb11-55"></a><span class="fu"># Discrete distributions</span></span>
<span id="cb11-56"><a href="#cb11-56"></a></span>
<span id="cb11-57"><a href="#cb11-57"></a>Discrete distributions form one of two fundamental branches in probability theory; the other is continuous distributions. This division rests on the nature of the sample space: discrete distributions assign probabilities to countable outcomes (integers, finite sets), while continuous distributions apply to uncountable intervals where individual point probabilities equal zero.</span>
<span id="cb11-58"><a href="#cb11-58"></a></span>
<span id="cb11-59"><a href="#cb11-59"></a>*Discrete random variables* have a finite or countable number of possible values and are the foundation of discrete probability distributions. Two core mathematical constraints govern mapping these probability distributions to their discrete random values:</span>
<span id="cb11-60"><a href="#cb11-60"></a></span>
<span id="cb11-61"><a href="#cb11-61"></a><span class="ss">- </span>First, each probability value must fall within the closed interval <span class="co">[</span><span class="ot">0,1</span><span class="co">]</span>; no outcome can possess negative probability or exceed certainty.</span>
<span id="cb11-62"><a href="#cb11-62"></a><span class="ss">- </span>Second, the sum of all probabilities across the complete sample space must equal exactly 1.0; this ensures that the distribution accounts for all possible outcomes without omission or duplication.</span>
<span id="cb11-63"><a href="#cb11-63"></a></span>
<span id="cb11-64"><a href="#cb11-64"></a>In practice, these constraints give us access to practical verification steps: </span>
<span id="cb11-65"><a href="#cb11-65"></a></span>
<span id="cb11-66"><a href="#cb11-66"></a><span class="ss">- </span>When constructing or validating a discrete distribution, sum all probability values to confirm they equal unity (1).</span>
<span id="cb11-67"><a href="#cb11-67"></a><span class="ss">- </span>Inspect each individual probability to ensure none violates the boundary conditions (<span class="co">[</span><span class="ot">0,1</span><span class="co">]</span>).</span>
<span id="cb11-68"><a href="#cb11-68"></a></span>
<span id="cb11-69"><a href="#cb11-69"></a>Violations point to computational errors or incomplete specification of the sample space.</span>
<span id="cb11-70"><a href="#cb11-70"></a></span>
<span id="cb11-71"><a href="#cb11-71"></a>The expected value $E(X)$ of a discrete random variable $X$ represents the probability-weighted average of all possible outcomes: $E(X) = \sum<span class="co">[</span><span class="ot">x \cdot P(X=x)</span><span class="co">]</span>$. Let's use a standard, unbiased die as an example. For each possible value $x$, multiply that value by its probability, then sum across all possibilities. If $X$ can take values $<span class="sc">\{</span>1,2,3,4,5,6<span class="sc">\}</span>$ each with probability $1/6$, then:</span>
<span id="cb11-72"><a href="#cb11-72"></a></span>
<span id="cb11-73"><a href="#cb11-73"></a>$$E(X) = 1(1/6) + 2(1/6) + 3(1/6) + 4(1/6) + 5(1/6) + 6(1/6) = 3.5$$</span>
<span id="cb11-74"><a href="#cb11-74"></a></span>
<span id="cb11-75"><a href="#cb11-75"></a>Variance calculation follows a two-step process. First, compute the expected value of the squared outcomes:</span>
<span id="cb11-76"><a href="#cb11-76"></a></span>
<span id="cb11-77"><a href="#cb11-77"></a>$$E(X^2) = \sum<span class="co">[</span><span class="ot">x^2 \cdot P(X=x)</span><span class="co">]</span>$$</span>
<span id="cb11-78"><a href="#cb11-78"></a></span>
<span id="cb11-79"><a href="#cb11-79"></a>Then apply the computational formula:</span>
<span id="cb11-80"><a href="#cb11-80"></a></span>
<span id="cb11-81"><a href="#cb11-81"></a>$$\text{Var}(X) = E(X^2) - <span class="co">[</span><span class="ot">E(X)</span><span class="co">]</span>^2$$</span>
<span id="cb11-82"><a href="#cb11-82"></a></span>
<span id="cb11-83"><a href="#cb11-83"></a>For the die example: $E(X^2) = 1^2(1/6) + 2^2(1/6) + ... + 6^2(1/6) = 91/6 \approx 15.17$, so $\text{Var}(X) = 15.17 - (3.5)^2 = 15.17 - 12.25 = 2.92$.</span>
<span id="cb11-84"><a href="#cb11-84"></a></span>
<span id="cb11-85"><a href="#cb11-85"></a>Below I provide options to generate and visualise data belonging to several classes of discrete distributions. In <span class="co">[</span><span class="ot">Chapter 6</span><span class="co">](06-assumptions.qmd)</span> we will learn how to transform these data prior to performing the appropriate statistical analysis.</span>
<span id="cb11-86"><a href="#cb11-86"></a></span>
<span id="cb11-87"><a href="#cb11-87"></a><span class="fu">## Bernoulli and Binomial distributions</span></span>
<span id="cb11-88"><a href="#cb11-88"></a></span>
<span id="cb11-89"><a href="#cb11-89"></a>The Bernoulli and Binomial distributions belong to what might be termed the "trial-based" subfamily of discrete distributions: they directly model outcomes of repeated experiments. Trial-based distributions require specifying both the number of trials and success probability.</span>
<span id="cb11-90"><a href="#cb11-90"></a></span>
<span id="cb11-91"><a href="#cb11-91"></a>Bernoulli and Binomial distributions are both discrete probability distributions that describe the *outcomes of binary events*. They are similar but there are also some key differences between the two. In real life examples encountered in ecology and biology we will probably mostly encounter the Binomial distributions. Let us consider each is more detail.</span>
<span id="cb11-92"><a href="#cb11-92"></a></span>
<span id="cb11-93"><a href="#cb11-93"></a>**Bernoulli distribution** The Bernoulli distribution represents a single binary trial or experiment with only two possible outcomes: 'success' (usually represented as <span class="in">`1`</span>) and 'failure' (usually represented as <span class="in">`0`</span>). The probability of success is denoted by $p$, while the probability of failure is $1 - p$. A Bernoulli distribution is characterised by only one parameter, $p$, which represents the probability of success for the single trial (@eq-bernoulli) (@fig-bernoulli1, @fig-bernoulli2).</span>
<span id="cb11-94"><a href="#cb11-94"></a></span>
<span id="cb11-95"><a href="#cb11-95"></a>::: {.column-margin}</span>
<span id="cb11-96"><a href="#cb11-96"></a>**The Bernoulli distribution:**</span>
<span id="cb11-97"><a href="#cb11-97"></a>$$</span>
<span id="cb11-98"><a href="#cb11-98"></a>P(X=k) = \begin{cases} </span>
<span id="cb11-99"><a href="#cb11-99"></a>  p, &amp; \text{if } k=1 <span class="sc">\\</span></span>
<span id="cb11-100"><a href="#cb11-100"></a>  1-p, &amp; \text{if } k=0</span>
<span id="cb11-101"><a href="#cb11-101"></a>\end{cases}</span>
<span id="cb11-102"><a href="#cb11-102"></a>$$ {#eq-bernoulli}</span>
<span id="cb11-103"><a href="#cb11-103"></a></span>
<span id="cb11-104"><a href="#cb11-104"></a>where $X$ is a random variable, $k$ is the outcome (1 for success and 0 for failure), and $p$ is the probability of success.</span>
<span id="cb11-105"><a href="#cb11-105"></a>:::</span>
<span id="cb11-106"><a href="#cb11-106"></a></span>
<span id="cb11-109"><a href="#cb11-109"></a><span class="in">```{r}</span></span>
<span id="cb11-110"><a href="#cb11-110"></a><span class="co">#| echo: false</span></span>
<span id="cb11-111"><a href="#cb11-111"></a><span class="co">#| fig.align: center</span></span>
<span id="cb11-112"><a href="#cb11-112"></a><span class="co">#| fig.cap: "Bernoulli distribution with 10 trials and 20 simulations set at a probability of p = 0.2. This could be a heavily loaded coin that has a 20% chance of landing heads."</span></span>
<span id="cb11-113"><a href="#cb11-113"></a><span class="co">#| label: fig-bernoulli1</span></span>
<span id="cb11-114"><a href="#cb11-114"></a></span>
<span id="cb11-115"><a href="#cb11-115"></a>bern <span class="ot">&lt;-</span> <span class="fu">tidy_bernoulli</span>(<span class="at">.n =</span> <span class="dv">10</span>, <span class="at">.prob =</span> <span class="fl">0.2</span>, <span class="at">.num_sims =</span> <span class="dv">20</span>)</span>
<span id="cb11-116"><a href="#cb11-116"></a></span>
<span id="cb11-117"><a href="#cb11-117"></a><span class="fu">ggarrange</span>(</span>
<span id="cb11-118"><a href="#cb11-118"></a>  <span class="fu">tidy_autoplot</span>(bern, <span class="at">.plot_type =</span> <span class="st">"density"</span>),</span>
<span id="cb11-119"><a href="#cb11-119"></a>  <span class="fu">tidy_autoplot</span>(bern, <span class="at">.plot_type =</span> <span class="st">"quantile"</span>),</span>
<span id="cb11-120"><a href="#cb11-120"></a>  <span class="fu">tidy_autoplot</span>(bern, <span class="at">.plot_type =</span> <span class="st">"probability"</span>),</span>
<span id="cb11-121"><a href="#cb11-121"></a>  <span class="fu">tidy_autoplot</span>(bern, <span class="at">.plot_type =</span> <span class="st">"qq"</span>),</span>
<span id="cb11-122"><a href="#cb11-122"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb11-123"><a href="#cb11-123"></a>)</span>
<span id="cb11-124"><a href="#cb11-124"></a><span class="in">```</span></span>
<span id="cb11-125"><a href="#cb11-125"></a></span>
<span id="cb11-128"><a href="#cb11-128"></a><span class="in">```{r}</span></span>
<span id="cb11-129"><a href="#cb11-129"></a><span class="co">#| echo: false</span></span>
<span id="cb11-130"><a href="#cb11-130"></a><span class="co">#| fig.align: center</span></span>
<span id="cb11-131"><a href="#cb11-131"></a><span class="co">#| fig.cap: "Bernoulli distribution with 10 trials and 20 simulations set at a probability of p = 0.5. This represents an unbiased coin that has an equal chance of landing on head or tail."</span></span>
<span id="cb11-132"><a href="#cb11-132"></a><span class="co">#| label: fig-bernoulli2</span></span>
<span id="cb11-133"><a href="#cb11-133"></a></span>
<span id="cb11-134"><a href="#cb11-134"></a>bern <span class="ot">&lt;-</span> <span class="fu">tidy_bernoulli</span>(<span class="at">.n =</span> <span class="dv">50</span>, <span class="at">.prob =</span> <span class="fl">0.5</span>, <span class="at">.num_sims =</span> <span class="dv">20</span>)</span>
<span id="cb11-135"><a href="#cb11-135"></a></span>
<span id="cb11-136"><a href="#cb11-136"></a><span class="fu">ggarrange</span>(</span>
<span id="cb11-137"><a href="#cb11-137"></a>  <span class="fu">tidy_autoplot</span>(bern, <span class="at">.plot_type =</span> <span class="st">"density"</span>),</span>
<span id="cb11-138"><a href="#cb11-138"></a>  <span class="fu">tidy_autoplot</span>(bern, <span class="at">.plot_type =</span> <span class="st">"quantile"</span>),</span>
<span id="cb11-139"><a href="#cb11-139"></a>  <span class="fu">tidy_autoplot</span>(bern, <span class="at">.plot_type =</span> <span class="st">"probability"</span>),</span>
<span id="cb11-140"><a href="#cb11-140"></a>  <span class="fu">tidy_autoplot</span>(bern, <span class="at">.plot_type =</span> <span class="st">"qq"</span>),</span>
<span id="cb11-141"><a href="#cb11-141"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb11-142"><a href="#cb11-142"></a>)</span>
<span id="cb11-143"><a href="#cb11-143"></a><span class="in">```</span></span>
<span id="cb11-144"><a href="#cb11-144"></a></span>
<span id="cb11-145"><a href="#cb11-145"></a>**Binomial distribution** The Binomial distribution represents the sum of outcomes in a fixed number of independent Bernoulli trials with the same probability of success, $p$. It is characterised by two parameters, $n$ (the number of trials) and $p$ (the probability of success in each trial). The Binomial distribution describes the probability of obtaining a specific number of successes ($k$) in $n$ trials (@eq-binomial) (@fig-binomial2).</span>
<span id="cb11-146"><a href="#cb11-146"></a></span>
<span id="cb11-147"><a href="#cb11-147"></a>::: {.column-margin}</span>
<span id="cb11-148"><a href="#cb11-148"></a>**The Binomial distribution:**</span>
<span id="cb11-149"><a href="#cb11-149"></a>$$P(X=k) = C(n,k) \cdot p^k \cdot (1-p)^{(n-k)}$$ {#eq-binomial}</span>
<span id="cb11-150"><a href="#cb11-150"></a></span>
<span id="cb11-151"><a href="#cb11-151"></a>where $X$ is a random variable, $k$ is the number of successes, $n$ is the total number of trials, $p$ is the probability of success in each trial, and $C(n,k)$ represents combinations.</span>
<span id="cb11-152"><a href="#cb11-152"></a></span>
<span id="cb11-153"><a href="#cb11-153"></a>In practice, determine $n$ (number of trials), $p$ (success probability), and $k$ (desired number of successes). Calculate the binomial coefficient $C(n,k) = \frac{n!}{k!(n-k)!}$, then apply @eq-binomial. If flipping a coin 10 times ($n=10$, $p=0.5$) and seeking exactly 3 heads ($k=3$): $C(10,3) = 120$, so $P(X=3) = 120 \cdot (0.5)^3 \cdot (0.5)^7 = 120 \cdot (0.5)^{10} = 0.117$.</span>
<span id="cb11-154"><a href="#cb11-154"></a>:::</span>
<span id="cb11-155"><a href="#cb11-155"></a></span>
<span id="cb11-156"><a href="#cb11-156"></a><span class="co">&lt;!-- ```{r} --&gt;</span></span>
<span id="cb11-157"><a href="#cb11-157"></a><span class="co">&lt;!-- #| echo: false --&gt;</span></span>
<span id="cb11-158"><a href="#cb11-158"></a><span class="co">&lt;!-- #| fig.align: center --&gt;</span></span>
<span id="cb11-159"><a href="#cb11-159"></a><span class="co">&lt;!--  --&gt;</span></span>
<span id="cb11-160"><a href="#cb11-160"></a><span class="co">&lt;!--#| fig.cap: "Binomial distribution with 50 trials and 20 simulations at an *a priori* set probability of p = 0.75." --&gt;</span></span>
<span id="cb11-161"><a href="#cb11-161"></a><span class="co">&lt;!-- #| label: fig-binomial1 --&gt;</span></span>
<span id="cb11-162"><a href="#cb11-162"></a></span>
<span id="cb11-163"><a href="#cb11-163"></a><span class="co">&lt;!-- binom1 &lt;- tidy_binomial(.n = 40, .num_sims = 10, .prob = 0.75) --&gt;</span></span>
<span id="cb11-164"><a href="#cb11-164"></a></span>
<span id="cb11-165"><a href="#cb11-165"></a><span class="co">&lt;!-- ggarrange( --&gt;</span></span>
<span id="cb11-166"><a href="#cb11-166"></a><span class="co">&lt;!--   tidy_autoplot(binom1, .plot_type = "density"), --&gt;</span></span>
<span id="cb11-167"><a href="#cb11-167"></a><span class="co">&lt;!--   tidy_autoplot(binom1, .plot_type = "quantile"), --&gt;</span></span>
<span id="cb11-168"><a href="#cb11-168"></a><span class="co">&lt;!--   tidy_autoplot(binom1, .plot_type = "probability"), --&gt;</span></span>
<span id="cb11-169"><a href="#cb11-169"></a><span class="co">&lt;!--   tidy_autoplot(binom1, .plot_type = "qq"), --&gt;</span></span>
<span id="cb11-170"><a href="#cb11-170"></a><span class="co">&lt;!--   ncol = 2, nrow = 2 --&gt;</span></span>
<span id="cb11-171"><a href="#cb11-171"></a><span class="co">&lt;!-- ) --&gt;</span></span>
<span id="cb11-172"><a href="#cb11-172"></a><span class="co">&lt;!-- ``` --&gt;</span></span>
<span id="cb11-173"><a href="#cb11-173"></a></span>
<span id="cb11-176"><a href="#cb11-176"></a><span class="in">```{r}</span></span>
<span id="cb11-177"><a href="#cb11-177"></a><span class="co">#| echo: false</span></span>
<span id="cb11-178"><a href="#cb11-178"></a><span class="co">#| fig.align: center</span></span>
<span id="cb11-179"><a href="#cb11-179"></a><span class="co">#| fig.cap: "Ten simulations of a binomial distribution with 40 randomly generated points for each of 100 trials at an *a priori* set probability of p = 0.75."</span></span>
<span id="cb11-180"><a href="#cb11-180"></a><span class="co">#| label: fig-binomial2</span></span>
<span id="cb11-181"><a href="#cb11-181"></a></span>
<span id="cb11-182"><a href="#cb11-182"></a>binom2 <span class="ot">&lt;-</span> <span class="fu">tidy_binomial</span>(<span class="at">.n =</span> <span class="dv">40</span>, <span class="at">.num_sims =</span> <span class="dv">10</span>, <span class="at">.prob =</span> <span class="fl">0.75</span>, <span class="at">.size =</span> <span class="dv">100</span>)</span>
<span id="cb11-183"><a href="#cb11-183"></a></span>
<span id="cb11-184"><a href="#cb11-184"></a><span class="fu">ggarrange</span>(</span>
<span id="cb11-185"><a href="#cb11-185"></a>  <span class="fu">tidy_autoplot</span>(binom2, <span class="at">.plot_type =</span> <span class="st">"density"</span>),</span>
<span id="cb11-186"><a href="#cb11-186"></a>  <span class="fu">tidy_autoplot</span>(binom2, <span class="at">.plot_type =</span> <span class="st">"quantile"</span>),</span>
<span id="cb11-187"><a href="#cb11-187"></a>  <span class="fu">tidy_autoplot</span>(binom2, <span class="at">.plot_type =</span> <span class="st">"probability"</span>),</span>
<span id="cb11-188"><a href="#cb11-188"></a>  <span class="fu">tidy_autoplot</span>(binom2, <span class="at">.plot_type =</span> <span class="st">"qq"</span>),</span>
<span id="cb11-189"><a href="#cb11-189"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb11-190"><a href="#cb11-190"></a>)</span>
<span id="cb11-191"><a href="#cb11-191"></a><span class="in">```</span></span>
<span id="cb11-192"><a href="#cb11-192"></a></span>
<span id="cb11-193"><a href="#cb11-193"></a>There are several examples of Binomial distributions in ecological and biological contexts. The Binomial distribution is relevant when studying the number of successes in a fixed number of independent trials, each with the same probability of success. A few examples of the Bernoulli distribution:</span>
<span id="cb11-194"><a href="#cb11-194"></a></span>
<span id="cb11-195"><a href="#cb11-195"></a><span class="ss">-   </span>**Seed germination** Suppose we plant 100 seeds of a particular plant species and wants to know the probability of a certain number of seeds germinating. If the probability of germination for each seed is constant then we can model the number of germinated seeds by a Binomial distribution.</span>
<span id="cb11-196"><a href="#cb11-196"></a></span>
<span id="cb11-197"><a href="#cb11-197"></a><span class="ss">-   </span>**Disease prevalence** An epidemiologist studies the prevalence of a disease within a population. For a random sample of 500 individuals, and with a fixed probability of an individual having the disease, the number of infected individuals in the sample can be modeled using a Binomial distribution.</span>
<span id="cb11-198"><a href="#cb11-198"></a></span>
<span id="cb11-199"><a href="#cb11-199"></a><span class="ss">-   </span>**Species occupancy** We do an ecological assessment to determine the occupancy of bird species across 50 habitat patches. If the probability of the species occupying a patch is the same across all patches, the number of patches occupied by the species will follow a Binomial distribution.</span>
<span id="cb11-200"><a href="#cb11-200"></a></span>
<span id="cb11-201"><a href="#cb11-201"></a><span class="ss">-   </span>**Allele inheritance** We want to examine the inheritance of a specific trait following Mendelian inheritance patterns. If the probability of inheriting the dominant allele for a given gene is constant, the number of offspring with the dominant trait in a fixed number of offspring follows the Binomial distribution.</span>
<span id="cb11-202"><a href="#cb11-202"></a></span>
<span id="cb11-203"><a href="#cb11-203"></a>Note that in these examples we assume a fixed probability and independence between trials and this is not always be true in real-world situations.</span>
<span id="cb11-204"><a href="#cb11-204"></a></span>
<span id="cb11-205"><a href="#cb11-205"></a><span class="fu">## Negative Binomial and Geometric distributions</span></span>
<span id="cb11-206"><a href="#cb11-206"></a></span>
<span id="cb11-207"><a href="#cb11-207"></a>The Geometric and Negative Binomial distributions form a "waiting time" subfamily, focusing on the number of trials preceding specified success patterns. These waiting-time distributions focus on success probability and target achievement levels.</span>
<span id="cb11-208"><a href="#cb11-208"></a></span>
<span id="cb11-209"><a href="#cb11-209"></a>**Negative Binomial distribution** A Negative Binomial random variable, $X$, counts the number of successes in a sequence of independent Bernoulli trials with probability $p$ before $r$ failures occur. This distribution could for example be used to predict the number of heads that result from a series of coin tosses before three tails are observed (@eq-negbinomial) (@fig-negbinomial).</span>
<span id="cb11-210"><a href="#cb11-210"></a></span>
<span id="cb11-211"><a href="#cb11-211"></a>::: {.column-margin}</span>
<span id="cb11-212"><a href="#cb11-212"></a>**The Negative Binomial distribution:**</span>
<span id="cb11-213"><a href="#cb11-213"></a>$$P(X=k) = \binom{k+r-1}{k} p^r (1-p)^k$$ {#eq-negbinomial}</span>
<span id="cb11-214"><a href="#cb11-214"></a></span>
<span id="cb11-215"><a href="#cb11-215"></a>The equation describes the probability mass function (PMF) of a Negative Binomial distribution, where $X$ is a random variable, $k$ is the number of failures, $r$ is the number of successes, and $p$ is the probability of success in each trial. The binomial coefficient is denoted by $\binom{k+r-1}{k}$, which calculates the number of ways to arrange $k$ failures and $r$ successes such that the last trial is a success.</span>
<span id="cb11-216"><a href="#cb11-216"></a>:::</span>
<span id="cb11-217"><a href="#cb11-217"></a></span>
<span id="cb11-220"><a href="#cb11-220"></a><span class="in">```{r}</span></span>
<span id="cb11-221"><a href="#cb11-221"></a><span class="co">#| echo: false</span></span>
<span id="cb11-222"><a href="#cb11-222"></a><span class="co">#| fig.align: center</span></span>
<span id="cb11-223"><a href="#cb11-223"></a><span class="co">#| fig.cap: "A negative binomial distribution with 50 trials and 10 simulations at an *a priori* expectation of p = 0.75."</span></span>
<span id="cb11-224"><a href="#cb11-224"></a><span class="co">#| label: fig-negbinomial</span></span>
<span id="cb11-225"><a href="#cb11-225"></a></span>
<span id="cb11-226"><a href="#cb11-226"></a>negbinom <span class="ot">&lt;-</span> <span class="fu">tidy_negative_binomial</span>(<span class="at">.n =</span> <span class="dv">50</span>, <span class="at">.size =</span> <span class="dv">5</span>, <span class="at">.prob =</span> <span class="fl">0.7</span>, <span class="at">.num_sims =</span> <span class="dv">10</span>)</span>
<span id="cb11-227"><a href="#cb11-227"></a></span>
<span id="cb11-228"><a href="#cb11-228"></a><span class="fu">ggarrange</span>(</span>
<span id="cb11-229"><a href="#cb11-229"></a>  <span class="fu">tidy_autoplot</span>(negbinom, <span class="at">.plot_type =</span> <span class="st">"density"</span>),</span>
<span id="cb11-230"><a href="#cb11-230"></a>  <span class="fu">tidy_autoplot</span>(negbinom, <span class="at">.plot_type =</span> <span class="st">"quantile"</span>),</span>
<span id="cb11-231"><a href="#cb11-231"></a>  <span class="fu">tidy_autoplot</span>(negbinom, <span class="at">.plot_type =</span> <span class="st">"probability"</span>),</span>
<span id="cb11-232"><a href="#cb11-232"></a>  <span class="fu">tidy_autoplot</span>(negbinom, <span class="at">.plot_type =</span> <span class="st">"qq"</span>),</span>
<span id="cb11-233"><a href="#cb11-233"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb11-234"><a href="#cb11-234"></a>)</span>
<span id="cb11-235"><a href="#cb11-235"></a><span class="in">```</span></span>
<span id="cb11-236"><a href="#cb11-236"></a></span>
<span id="cb11-237"><a href="#cb11-237"></a></span>
<span id="cb11-238"><a href="#cb11-238"></a>**Geometric distribution** A geometric random variable, $X$, represents the number of trials that are required to observe a single success. Each trial is independent and has success probability $p$. As an example, the geometric distribution is useful to model the number of times a die must be tossed in order for a six to be observed (@eq-geometric) (@fig-geomet).</span>
<span id="cb11-239"><a href="#cb11-239"></a></span>
<span id="cb11-240"><a href="#cb11-240"></a>::: {.column-margin}</span>
<span id="cb11-241"><a href="#cb11-241"></a>**The Geometric distribution:**</span>
<span id="cb11-242"><a href="#cb11-242"></a>$$P(X=k) = p (1-p)^k$$ {#eq-geometric}</span>
<span id="cb11-243"><a href="#cb11-243"></a></span>
<span id="cb11-244"><a href="#cb11-244"></a>The equation represents the PMF of a Geometric distribution, where $X$ is a random variable, $k$ is the number of failures before the first success, and $p$ is the probability of success in each trial. The Geometric distribution can be thought of as a special case of the Negative Binomial distribution with $r = 1$, which models the number of failures before achieving a single success.</span>
<span id="cb11-245"><a href="#cb11-245"></a>:::</span>
<span id="cb11-246"><a href="#cb11-246"></a></span>
<span id="cb11-249"><a href="#cb11-249"></a><span class="in">```{r}</span></span>
<span id="cb11-250"><a href="#cb11-250"></a><span class="co">#| echo: false</span></span>
<span id="cb11-251"><a href="#cb11-251"></a><span class="co">#| fig.align: center</span></span>
<span id="cb11-252"><a href="#cb11-252"></a><span class="co">#| fig.cap: "A geometric distribution with 50 trials and 10 simulations at an *a priori* expectation of p = 0.75."</span></span>
<span id="cb11-253"><a href="#cb11-253"></a><span class="co">#| label: fig-geomet</span></span>
<span id="cb11-254"><a href="#cb11-254"></a></span>
<span id="cb11-255"><a href="#cb11-255"></a>geomet <span class="ot">&lt;-</span> <span class="fu">tidy_geometric</span>(<span class="at">.n =</span> <span class="dv">50</span>, <span class="at">.prob =</span> <span class="fl">0.7</span>, <span class="at">.num_sims =</span> <span class="dv">10</span>)</span>
<span id="cb11-256"><a href="#cb11-256"></a></span>
<span id="cb11-257"><a href="#cb11-257"></a><span class="fu">ggarrange</span>(</span>
<span id="cb11-258"><a href="#cb11-258"></a>  <span class="fu">tidy_autoplot</span>(geomet, <span class="at">.plot_type =</span> <span class="st">"density"</span>),</span>
<span id="cb11-259"><a href="#cb11-259"></a>  <span class="fu">tidy_autoplot</span>(geomet, <span class="at">.plot_type =</span> <span class="st">"quantile"</span>),</span>
<span id="cb11-260"><a href="#cb11-260"></a>  <span class="fu">tidy_autoplot</span>(geomet, <span class="at">.plot_type =</span> <span class="st">"probability"</span>),</span>
<span id="cb11-261"><a href="#cb11-261"></a>  <span class="fu">tidy_autoplot</span>(geomet, <span class="at">.plot_type =</span> <span class="st">"qq"</span>),</span>
<span id="cb11-262"><a href="#cb11-262"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb11-263"><a href="#cb11-263"></a>)</span>
<span id="cb11-264"><a href="#cb11-264"></a><span class="in">```</span></span>
<span id="cb11-265"><a href="#cb11-265"></a></span>
<span id="cb11-266"><a href="#cb11-266"></a><span class="fu">## Poisson distribution</span></span>
<span id="cb11-267"><a href="#cb11-267"></a></span>
<span id="cb11-268"><a href="#cb11-268"></a>::: {.column-margin}</span>
<span id="cb11-269"><a href="#cb11-269"></a>**The Poisson distribution:**</span>
<span id="cb11-270"><a href="#cb11-270"></a>$$P(X=k) = \frac{e^{-\lambda} \lambda^k}{k!}$$ {#eq-poisson}</span>
<span id="cb11-271"><a href="#cb11-271"></a></span>
<span id="cb11-272"><a href="#cb11-272"></a>The function represents the PMF of a Poisson distribution, where $X$ is a random variable, $k$ is the number of events or occurrences, and $\lambda$ (lambda) is the average rate of occurrences (events per unit of time or space). The constant $e$ is the base of the natural logarithm, and $k!$ is the factorial of $k$. The Poisson distribution is commonly used to model the number of events occurring within a fixed interval of time or space when events occur independently and at a constant average rate.</span>
<span id="cb11-273"><a href="#cb11-273"></a></span>
<span id="cb11-274"><a href="#cb11-274"></a>For practical application, identify $\lambda$ (average rate) and $k$ (observed count). If a gerbil enters a nest 4 times per hour ($\lambda=4$) and you want the probability of exactly 2 entries in one hour ($k=2$): $P(X=2) = \frac{4^2 \cdot e^{-4}}{2!} = \frac{16 \cdot 0.0183}{2} = 0.147$.</span>
<span id="cb11-275"><a href="#cb11-275"></a>:::</span>
<span id="cb11-276"><a href="#cb11-276"></a></span>
<span id="cb11-277"><a href="#cb11-277"></a>The Poisson distribution represents a "rate-based" subfamily, which is used to model count phenomena without explicit trial structure. They involve the average occurrence rates over specified intervals.</span>
<span id="cb11-278"><a href="#cb11-278"></a></span>
<span id="cb11-279"><a href="#cb11-279"></a>A Poisson random variable, $X$, tallies the number of events occurring in a fixed interval of time or space, given that these events occur with an average rate $\lambda$. Poisson distributions can be used to model events such as meteor showers and or number of people entering a shopping mall (@eq-poisson) (@fig-poisson).</span>
<span id="cb11-280"><a href="#cb11-280"></a></span>
<span id="cb11-281"><a href="#cb11-281"></a></span>
<span id="cb11-284"><a href="#cb11-284"></a><span class="in">```{r}</span></span>
<span id="cb11-285"><a href="#cb11-285"></a><span class="co">#| echo: false</span></span>
<span id="cb11-286"><a href="#cb11-286"></a><span class="co">#| fig.align: center</span></span>
<span id="cb11-287"><a href="#cb11-287"></a><span class="co">#| fig.cap: "A Poisson distribution with 50 trials and 10 simulations at an *a priori* expectation of p = 0.75."</span></span>
<span id="cb11-288"><a href="#cb11-288"></a><span class="co">#| label: fig-poisson</span></span>
<span id="cb11-289"><a href="#cb11-289"></a></span>
<span id="cb11-290"><a href="#cb11-290"></a>poisson <span class="ot">&lt;-</span> <span class="fu">tidy_poisson</span>(<span class="at">.n =</span> <span class="dv">50</span>, <span class="at">.lambda =</span> <span class="dv">2</span>, <span class="at">.num_sims =</span> <span class="dv">10</span>)</span>
<span id="cb11-291"><a href="#cb11-291"></a></span>
<span id="cb11-292"><a href="#cb11-292"></a><span class="fu">ggarrange</span>(</span>
<span id="cb11-293"><a href="#cb11-293"></a>  <span class="fu">tidy_autoplot</span>(poisson, <span class="at">.plot_type =</span> <span class="st">"density"</span>),</span>
<span id="cb11-294"><a href="#cb11-294"></a>  <span class="fu">tidy_autoplot</span>(poisson, <span class="at">.plot_type =</span> <span class="st">"quantile"</span>),</span>
<span id="cb11-295"><a href="#cb11-295"></a>  <span class="fu">tidy_autoplot</span>(poisson, <span class="at">.plot_type =</span> <span class="st">"probability"</span>),</span>
<span id="cb11-296"><a href="#cb11-296"></a>  <span class="fu">tidy_autoplot</span>(poisson, <span class="at">.plot_type =</span> <span class="st">"qq"</span>),</span>
<span id="cb11-297"><a href="#cb11-297"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb11-298"><a href="#cb11-298"></a>)</span>
<span id="cb11-299"><a href="#cb11-299"></a><span class="in">```</span></span>
<span id="cb11-300"><a href="#cb11-300"></a></span>
<span id="cb11-301"><a href="#cb11-301"></a><span class="fu">## Hypergeometric distribution</span></span>
<span id="cb11-302"><a href="#cb11-302"></a></span>
<span id="cb11-303"><a href="#cb11-303"></a>::: {.column-margin}</span>
<span id="cb11-304"><a href="#cb11-304"></a>**The Hypergeometricd distribution:**</span>
<span id="cb11-305"><a href="#cb11-305"></a>$$P(X=k) = \frac{C(K,k) \cdot C(N-K,n-k)}{C(N,n)}$$ {#eq-hypergeometric}</span>
<span id="cb11-306"><a href="#cb11-306"></a>In the probability mass function, above, $N$ is population size, $K$ is number of success items in population, $n$ is sample size, and $k$ is observed successes in sample.</span>
<span id="cb11-307"><a href="#cb11-307"></a></span>
<span id="cb11-308"><a href="#cb11-308"></a>Consider drawing 5 cards from a deck without replacement, seeking exactly 2 hearts. Here $N=52$, $K=13$ (hearts), $n=5$, $k=2$: $P(X=2) = \frac{C(13,2) \cdot C(39,3)}{C(52,5)} = \frac{78 \cdot 9139}{2,598,960} = 0.274$.</span>
<span id="cb11-309"><a href="#cb11-309"></a>:::</span>
<span id="cb11-310"><a href="#cb11-310"></a></span>
<span id="cb11-311"><a href="#cb11-311"></a>The hypergeometric distribution occupies a distinct position within the discrete distribution taxonomy: we might term the "finite population sampling" subfamily.</span>
<span id="cb11-312"><a href="#cb11-312"></a></span>
<span id="cb11-313"><a href="#cb11-313"></a>The hypergeometric distribution models sampling without replacement from finite populations containing two types of items. Unlike binomial sampling, each draw changes the composition of remaining items.</span>
<span id="cb11-314"><a href="#cb11-314"></a></span>
<span id="cb11-315"><a href="#cb11-315"></a></span>
<span id="cb11-316"><a href="#cb11-316"></a>**To do:** insert figures.</span>
<span id="cb11-317"><a href="#cb11-317"></a></span>
<span id="cb11-318"><a href="#cb11-318"></a><span class="fu"># Continuous distributions</span></span>
<span id="cb11-319"><a href="#cb11-319"></a></span>
<span id="cb11-320"><a href="#cb11-320"></a><span class="fu">## Normal distribution {#sec-normal}</span></span>
<span id="cb11-321"><a href="#cb11-321"></a></span>
<span id="cb11-322"><a href="#cb11-322"></a>::: {.column-margin}</span>
<span id="cb11-323"><a href="#cb11-323"></a>**The Normal distribution:**</span>
<span id="cb11-324"><a href="#cb11-324"></a>$$f(x) = \frac{1}{\sigma\sqrt{2\pi}} e^{ -\frac{1}{2} \left(\frac{x-\mu}{\sigma}\right)^2 }</span>
<span id="cb11-325"><a href="#cb11-325"></a>$$ {#eq-normal}</span>
<span id="cb11-326"><a href="#cb11-326"></a></span>
<span id="cb11-327"><a href="#cb11-327"></a>where $x$ is a continuous random variable, $\mu$ (mu) is the mean, and $\sigma$ (sigma) is the standard deviation. The constant factor $\frac{1}{\sigma\sqrt{2\pi}}$ ensures that the Probability Density Function (PDF) integrates to 1, and the exponential term is responsible for the characteristic bell-shaped curve of the Normal distribution.</span>
<span id="cb11-328"><a href="#cb11-328"></a>:::</span>
<span id="cb11-329"><a href="#cb11-329"></a></span>
<span id="cb11-330"><a href="#cb11-330"></a>::: {.column-margin}</span>
<span id="cb11-331"><a href="#cb11-331"></a><span class="al">![The idealised Normal distribution showing the proportion of data within 1, 2, and 3SD from the mean.](../../images/Standard_deviation_diagram.svg)</span>{#fig-expectednormal}</span>
<span id="cb11-332"><a href="#cb11-332"></a>:::</span>
<span id="cb11-333"><a href="#cb11-333"></a></span>
<span id="cb11-334"><a href="#cb11-334"></a>Another name for this kind of distribution is a Gaussian distribution. A random sample with a Gaussian distribution is normally distributed. These values are *independent and identically distributed* random variables (i.i.d.), and they have an expected mean given by $\mu$ (or $\hat{x}$ in Chapter 3.2.1) and a finite variance given by $\sigma^{2}$ (or $S^{2}$ in Chapter 3.3.1); if the number of samples drawn from a population is sufficiently large, the estimated mean and SD will be indistinguishable from the population (as per the central limit theorem). It is represented by @eq-normal (@fig-normal).</span>
<span id="cb11-335"><a href="#cb11-335"></a></span>
<span id="cb11-338"><a href="#cb11-338"></a><span class="in">```{r}</span></span>
<span id="cb11-339"><a href="#cb11-339"></a><span class="co">#| echo: false</span></span>
<span id="cb11-340"><a href="#cb11-340"></a><span class="co">#| fig.align: center</span></span>
<span id="cb11-341"><a href="#cb11-341"></a><span class="co">#| fig.cap: "Normal distribution with 40 trials and 5 simulations."</span></span>
<span id="cb11-342"><a href="#cb11-342"></a><span class="co">#| label: fig-normal</span></span>
<span id="cb11-343"><a href="#cb11-343"></a></span>
<span id="cb11-344"><a href="#cb11-344"></a>normnal <span class="ot">&lt;-</span> <span class="fu">tidy_normal</span>(<span class="at">.n =</span> <span class="dv">100</span>, <span class="at">.num_sims =</span> <span class="dv">5</span>)</span>
<span id="cb11-345"><a href="#cb11-345"></a></span>
<span id="cb11-346"><a href="#cb11-346"></a><span class="fu">ggarrange</span>(</span>
<span id="cb11-347"><a href="#cb11-347"></a>  <span class="fu">tidy_autoplot</span>(normnal, <span class="at">.plot_type =</span> <span class="st">"density"</span>),</span>
<span id="cb11-348"><a href="#cb11-348"></a>  <span class="fu">tidy_autoplot</span>(normnal, <span class="at">.plot_type =</span> <span class="st">"quantile"</span>),</span>
<span id="cb11-349"><a href="#cb11-349"></a>  <span class="fu">tidy_autoplot</span>(normnal, <span class="at">.plot_type =</span> <span class="st">"probability"</span>),</span>
<span id="cb11-350"><a href="#cb11-350"></a>  <span class="fu">tidy_autoplot</span>(normnal, <span class="at">.plot_type =</span> <span class="st">"qq"</span>),</span>
<span id="cb11-351"><a href="#cb11-351"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb11-352"><a href="#cb11-352"></a>)</span>
<span id="cb11-353"><a href="#cb11-353"></a><span class="in">```</span></span>
<span id="cb11-354"><a href="#cb11-354"></a></span>
<span id="cb11-355"><a href="#cb11-355"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb11-356"><a href="#cb11-356"></a><span class="fu">## Central Limit Theorem</span></span>
<span id="cb11-357"><a href="#cb11-357"></a></span>
<span id="cb11-358"><a href="#cb11-358"></a>The Central Limit Theorem (CLT) is a fundamental result in probability theory and statistics, which states that the distribution of the sum (or average) of a large number of independent, identically distributed (IID) random variables approaches a Normal distribution regardless of the shape of the original distribution. So, the CLT asserts that the Normal distribution is the limiting distribution for the sum or average of many random variables, as long as certain conditions are met.</span>
<span id="cb11-359"><a href="#cb11-359"></a></span>
<span id="cb11-360"><a href="#cb11-360"></a>The CLT provides a basis for making inferences about population parameters using sample statistics. For example, when dealing with large sample sizes, the sampling distribution of the sample mean is approximately normally distributed, even if the underlying population distribution is not normal. This allows us to apply inferential techniques based on the Normal distribution, such as hypothesis testing and constructing confidence intervals, to estimate population parameters using sample data.</span>
<span id="cb11-361"><a href="#cb11-361"></a></span>
<span id="cb11-362"><a href="#cb11-362"></a>Some conditions must be met for the CLT to be true:</span>
<span id="cb11-363"><a href="#cb11-363"></a></span>
<span id="cb11-364"><a href="#cb11-364"></a><span class="ss">-   </span>**The random variables must be independent** The observations should not be influenced by one another.</span>
<span id="cb11-365"><a href="#cb11-365"></a><span class="ss">-   </span>**The random variables must be identically distributed** They must come from the same population with the same mean and variance.</span>
<span id="cb11-366"><a href="#cb11-366"></a><span class="ss">-   </span>**The number of random variables (sample size) must be sufficiently large** Although there is no strict rule for the sample size, a common rule of thumb is that the sample size should be at least 30 for the CLT to be a reasonable approximation.</span>
<span id="cb11-367"><a href="#cb11-367"></a>:::</span>
<span id="cb11-368"><a href="#cb11-368"></a></span>
<span id="cb11-369"><a href="#cb11-369"></a><span class="fu">## Uniform distribution</span></span>
<span id="cb11-370"><a href="#cb11-370"></a></span>
<span id="cb11-371"><a href="#cb11-371"></a>The continuous uniform distribution is sometime called a rectangular distribution. Simply, it states that all measurements of the same magnitude included with this distribution are equally probable. This is basically random numbers (@fig-unif).</span>
<span id="cb11-372"><a href="#cb11-372"></a></span>
<span id="cb11-375"><a href="#cb11-375"></a><span class="in">```{r}</span></span>
<span id="cb11-376"><a href="#cb11-376"></a><span class="co">#| echo: false</span></span>
<span id="cb11-377"><a href="#cb11-377"></a><span class="co">#| fig.align: center</span></span>
<span id="cb11-378"><a href="#cb11-378"></a><span class="co">#| fig.cap: "Uniform distribution with 100 trials and 5 simulations."</span></span>
<span id="cb11-379"><a href="#cb11-379"></a><span class="co">#| label: fig-unif</span></span>
<span id="cb11-380"><a href="#cb11-380"></a></span>
<span id="cb11-381"><a href="#cb11-381"></a>unif <span class="ot">&lt;-</span> <span class="fu">tidy_uniform</span>(<span class="at">.n =</span> <span class="dv">100</span>, <span class="at">.min =</span> <span class="dv">0</span>, <span class="at">.max =</span> <span class="dv">5</span>, <span class="at">.num_sims =</span> <span class="dv">5</span>)</span>
<span id="cb11-382"><a href="#cb11-382"></a></span>
<span id="cb11-383"><a href="#cb11-383"></a><span class="fu">ggarrange</span>(</span>
<span id="cb11-384"><a href="#cb11-384"></a>  <span class="fu">tidy_autoplot</span>(unif, <span class="at">.plot_type =</span> <span class="st">"density"</span>),</span>
<span id="cb11-385"><a href="#cb11-385"></a>  <span class="fu">tidy_autoplot</span>(unif, <span class="at">.plot_type =</span> <span class="st">"quantile"</span>),</span>
<span id="cb11-386"><a href="#cb11-386"></a>  <span class="fu">tidy_autoplot</span>(unif, <span class="at">.plot_type =</span> <span class="st">"probability"</span>),</span>
<span id="cb11-387"><a href="#cb11-387"></a>  <span class="fu">tidy_autoplot</span>(unif, <span class="at">.plot_type =</span> <span class="st">"qq"</span>),</span>
<span id="cb11-388"><a href="#cb11-388"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb11-389"><a href="#cb11-389"></a>)</span>
<span id="cb11-390"><a href="#cb11-390"></a><span class="in">```</span></span>
<span id="cb11-391"><a href="#cb11-391"></a></span>
<span id="cb11-392"><a href="#cb11-392"></a><span class="fu">## Student T distribution</span></span>
<span id="cb11-393"><a href="#cb11-393"></a></span>
<span id="cb11-394"><a href="#cb11-394"></a>This is a continuous probability distribution that arises when estimating the mean of a normally distributed population in situations where the sample size is small and population standard deviation is unknown. It is used in the statistical significance testing between the means of different sets of samples, and not much so in the modelling of many kinds of experiments or observations (@fig-studt).</span>
<span id="cb11-395"><a href="#cb11-395"></a></span>
<span id="cb11-398"><a href="#cb11-398"></a><span class="in">```{r}</span></span>
<span id="cb11-399"><a href="#cb11-399"></a><span class="co">#| echo: false</span></span>
<span id="cb11-400"><a href="#cb11-400"></a><span class="co">#| fig.align: center</span></span>
<span id="cb11-401"><a href="#cb11-401"></a><span class="co">#| fig.cap: "Uniform distribution with 100 trials and 5 simulations."</span></span>
<span id="cb11-402"><a href="#cb11-402"></a><span class="co">#| label: fig-studt</span></span>
<span id="cb11-403"><a href="#cb11-403"></a></span>
<span id="cb11-404"><a href="#cb11-404"></a>studt <span class="ot">&lt;-</span> <span class="fu">tidy_t</span>(<span class="at">.n =</span> <span class="dv">100</span>, <span class="at">.df =</span> <span class="dv">1</span>, <span class="at">.ncp =</span> <span class="dv">0</span>, <span class="at">.num_sims =</span> <span class="dv">5</span>)</span>
<span id="cb11-405"><a href="#cb11-405"></a></span>
<span id="cb11-406"><a href="#cb11-406"></a><span class="fu">ggarrange</span>(</span>
<span id="cb11-407"><a href="#cb11-407"></a>  <span class="fu">tidy_autoplot</span>(studt, <span class="at">.plot_type =</span> <span class="st">"density"</span>),</span>
<span id="cb11-408"><a href="#cb11-408"></a>  <span class="fu">tidy_autoplot</span>(studt, <span class="at">.plot_type =</span> <span class="st">"quantile"</span>),</span>
<span id="cb11-409"><a href="#cb11-409"></a>  <span class="fu">tidy_autoplot</span>(studt, <span class="at">.plot_type =</span> <span class="st">"probability"</span>),</span>
<span id="cb11-410"><a href="#cb11-410"></a>  <span class="fu">tidy_autoplot</span>(studt, <span class="at">.plot_type =</span> <span class="st">"qq"</span>),</span>
<span id="cb11-411"><a href="#cb11-411"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb11-412"><a href="#cb11-412"></a>)</span>
<span id="cb11-413"><a href="#cb11-413"></a><span class="in">```</span></span>
<span id="cb11-414"><a href="#cb11-414"></a></span>
<span id="cb11-415"><a href="#cb11-415"></a><span class="fu">## Chi-squared distribution</span></span>
<span id="cb11-416"><a href="#cb11-416"></a></span>
<span id="cb11-417"><a href="#cb11-417"></a>Mostly used in hypothesis testing, but not to encapsulate the distribution of data drawn to represent natural phenomena (@fig-chi).</span>
<span id="cb11-418"><a href="#cb11-418"></a></span>
<span id="cb11-421"><a href="#cb11-421"></a><span class="in">```{r}</span></span>
<span id="cb11-422"><a href="#cb11-422"></a><span class="co">#| echo: false</span></span>
<span id="cb11-423"><a href="#cb11-423"></a><span class="co">#| fig.align: center</span></span>
<span id="cb11-424"><a href="#cb11-424"></a><span class="co">#| fig.cap: "Chi distribution with 100 trials and 5 simulations."</span></span>
<span id="cb11-425"><a href="#cb11-425"></a><span class="co">#| label: fig-chi</span></span>
<span id="cb11-426"><a href="#cb11-426"></a></span>
<span id="cb11-427"><a href="#cb11-427"></a>chi <span class="ot">&lt;-</span> <span class="fu">tidy_chisquare</span>(<span class="at">.n =</span> <span class="dv">100</span>, <span class="at">.df =</span> <span class="dv">1</span>, <span class="at">.ncp =</span> <span class="dv">0</span>, <span class="at">.num_sims =</span> <span class="dv">5</span>)</span>
<span id="cb11-428"><a href="#cb11-428"></a></span>
<span id="cb11-429"><a href="#cb11-429"></a><span class="fu">ggarrange</span>(</span>
<span id="cb11-430"><a href="#cb11-430"></a>  <span class="fu">tidy_autoplot</span>(chi, <span class="at">.plot_type =</span> <span class="st">"density"</span>),</span>
<span id="cb11-431"><a href="#cb11-431"></a>  <span class="fu">tidy_autoplot</span>(chi, <span class="at">.plot_type =</span> <span class="st">"quantile"</span>),</span>
<span id="cb11-432"><a href="#cb11-432"></a>  <span class="fu">tidy_autoplot</span>(chi, <span class="at">.plot_type =</span> <span class="st">"probability"</span>),</span>
<span id="cb11-433"><a href="#cb11-433"></a>  <span class="fu">tidy_autoplot</span>(chi, <span class="at">.plot_type =</span> <span class="st">"qq"</span>),</span>
<span id="cb11-434"><a href="#cb11-434"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb11-435"><a href="#cb11-435"></a>)</span>
<span id="cb11-436"><a href="#cb11-436"></a><span class="in">```</span></span>
<span id="cb11-437"><a href="#cb11-437"></a></span>
<span id="cb11-438"><a href="#cb11-438"></a><span class="fu">## Exponential distribution</span></span>
<span id="cb11-439"><a href="#cb11-439"></a></span>
<span id="cb11-440"><a href="#cb11-440"></a>This is a probability distribution that describes the time between events in a Poisson point process, i.e., a process in which events occur continuously and independently at a constant average rate (@fig-exp).</span>
<span id="cb11-441"><a href="#cb11-441"></a></span>
<span id="cb11-444"><a href="#cb11-444"></a><span class="in">```{r}</span></span>
<span id="cb11-445"><a href="#cb11-445"></a><span class="co">#| echo: false</span></span>
<span id="cb11-446"><a href="#cb11-446"></a><span class="co">#| fig.align: center</span></span>
<span id="cb11-447"><a href="#cb11-447"></a><span class="co">#| fig.cap: "An exponential distribution with 100 trials and 5 simulations."</span></span>
<span id="cb11-448"><a href="#cb11-448"></a><span class="co">#| label: fig-exp</span></span>
<span id="cb11-449"><a href="#cb11-449"></a></span>
<span id="cb11-450"><a href="#cb11-450"></a>exp <span class="ot">&lt;-</span> <span class="fu">tidy_exponential</span>(<span class="at">.n =</span> <span class="dv">100</span>, <span class="at">.rate =</span> <span class="dv">1</span>, <span class="at">.num_sims =</span> <span class="dv">5</span>)</span>
<span id="cb11-451"><a href="#cb11-451"></a></span>
<span id="cb11-452"><a href="#cb11-452"></a><span class="fu">ggarrange</span>(</span>
<span id="cb11-453"><a href="#cb11-453"></a>  <span class="fu">tidy_autoplot</span>(exp, <span class="at">.plot_type =</span> <span class="st">"density"</span>),</span>
<span id="cb11-454"><a href="#cb11-454"></a>  <span class="fu">tidy_autoplot</span>(exp, <span class="at">.plot_type =</span> <span class="st">"quantile"</span>),</span>
<span id="cb11-455"><a href="#cb11-455"></a>  <span class="fu">tidy_autoplot</span>(exp, <span class="at">.plot_type =</span> <span class="st">"probability"</span>),</span>
<span id="cb11-456"><a href="#cb11-456"></a>  <span class="fu">tidy_autoplot</span>(exp, <span class="at">.plot_type =</span> <span class="st">"qq"</span>),</span>
<span id="cb11-457"><a href="#cb11-457"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb11-458"><a href="#cb11-458"></a>)</span>
<span id="cb11-459"><a href="#cb11-459"></a><span class="in">```</span></span>
<span id="cb11-460"><a href="#cb11-460"></a></span>
<span id="cb11-461"><a href="#cb11-461"></a><span class="fu">## F distribution</span></span>
<span id="cb11-462"><a href="#cb11-462"></a></span>
<span id="cb11-463"><a href="#cb11-463"></a>This is a probability distribution that arises in the context of the analysis of variance (ANOVA) and regression analysis. It is used to compare the variances of two populations (@fig-f).</span>
<span id="cb11-464"><a href="#cb11-464"></a></span>
<span id="cb11-467"><a href="#cb11-467"></a><span class="in">```{r}</span></span>
<span id="cb11-468"><a href="#cb11-468"></a><span class="co">#| echo: false</span></span>
<span id="cb11-469"><a href="#cb11-469"></a><span class="co">#| fig.align: center</span></span>
<span id="cb11-470"><a href="#cb11-470"></a><span class="co">#| fig.cap: "An F distribution with 100 trials and 5 simulations."</span></span>
<span id="cb11-471"><a href="#cb11-471"></a><span class="co">#| label: fig-f</span></span>
<span id="cb11-472"><a href="#cb11-472"></a></span>
<span id="cb11-473"><a href="#cb11-473"></a>f <span class="ot">&lt;-</span> <span class="fu">tidy_f</span>(<span class="at">.n =</span> <span class="dv">100</span>, <span class="at">.df1 =</span> <span class="dv">1</span>, <span class="at">.df2 =</span> <span class="dv">1</span>, <span class="at">.ncp =</span> <span class="dv">5</span>, <span class="at">.num_sims =</span> <span class="dv">5</span>)</span>
<span id="cb11-474"><a href="#cb11-474"></a></span>
<span id="cb11-475"><a href="#cb11-475"></a><span class="fu">ggarrange</span>(</span>
<span id="cb11-476"><a href="#cb11-476"></a>  <span class="fu">tidy_autoplot</span>(f, <span class="at">.plot_type =</span> <span class="st">"density"</span>),</span>
<span id="cb11-477"><a href="#cb11-477"></a>  <span class="fu">tidy_autoplot</span>(f, <span class="at">.plot_type =</span> <span class="st">"quantile"</span>),</span>
<span id="cb11-478"><a href="#cb11-478"></a>  <span class="fu">tidy_autoplot</span>(f, <span class="at">.plot_type =</span> <span class="st">"probability"</span>),</span>
<span id="cb11-479"><a href="#cb11-479"></a>  <span class="fu">tidy_autoplot</span>(f, <span class="at">.plot_type =</span> <span class="st">"qq"</span>),</span>
<span id="cb11-480"><a href="#cb11-480"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb11-481"><a href="#cb11-481"></a>)</span>
<span id="cb11-482"><a href="#cb11-482"></a><span class="in">```</span></span>
<span id="cb11-483"><a href="#cb11-483"></a></span>
<span id="cb11-484"><a href="#cb11-484"></a><span class="fu">## Gamma distribution</span></span>
<span id="cb11-485"><a href="#cb11-485"></a></span>
<span id="cb11-486"><a href="#cb11-486"></a>This is a two-parameter family of continuous probability distributions. It is used to model the time until an event occurs. It is a generalisation of the exponential distribution (@fig-gamma).</span>
<span id="cb11-487"><a href="#cb11-487"></a></span>
<span id="cb11-490"><a href="#cb11-490"></a><span class="in">```{r}</span></span>
<span id="cb11-491"><a href="#cb11-491"></a><span class="co">#| echo: false</span></span>
<span id="cb11-492"><a href="#cb11-492"></a><span class="co">#| fig.align: center</span></span>
<span id="cb11-493"><a href="#cb11-493"></a><span class="co">#| fig.cap: "A Gamma distribution with 100 trials and 5 simulations."</span></span>
<span id="cb11-494"><a href="#cb11-494"></a><span class="co">#| label: fig-gamma</span></span>
<span id="cb11-495"><a href="#cb11-495"></a></span>
<span id="cb11-496"><a href="#cb11-496"></a>gamma <span class="ot">&lt;-</span> <span class="fu">tidy_gamma</span>(<span class="at">.n =</span> <span class="dv">100</span>, <span class="at">.shape =</span> <span class="dv">1</span>, <span class="at">.scale =</span> <span class="fl">0.4</span>, <span class="at">.num_sims =</span> <span class="dv">5</span>)</span>
<span id="cb11-497"><a href="#cb11-497"></a></span>
<span id="cb11-498"><a href="#cb11-498"></a><span class="fu">ggarrange</span>(</span>
<span id="cb11-499"><a href="#cb11-499"></a>  <span class="fu">tidy_autoplot</span>(gamma, <span class="at">.plot_type =</span> <span class="st">"density"</span>),</span>
<span id="cb11-500"><a href="#cb11-500"></a>  <span class="co"># tidy_autoplot(gamma, .plot_type = "quantile"),</span></span>
<span id="cb11-501"><a href="#cb11-501"></a>  <span class="fu">tidy_autoplot</span>(gamma, <span class="at">.plot_type =</span> <span class="st">"probability"</span>),</span>
<span id="cb11-502"><a href="#cb11-502"></a>  <span class="fu">tidy_autoplot</span>(gamma, <span class="at">.plot_type =</span> <span class="st">"qq"</span>),</span>
<span id="cb11-503"><a href="#cb11-503"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb11-504"><a href="#cb11-504"></a>)</span>
<span id="cb11-505"><a href="#cb11-505"></a><span class="in">```</span></span>
<span id="cb11-506"><a href="#cb11-506"></a></span>
<span id="cb11-507"><a href="#cb11-507"></a><span class="fu">## Beta distribution</span></span>
<span id="cb11-508"><a href="#cb11-508"></a></span>
<span id="cb11-509"><a href="#cb11-509"></a>This is a family of continuous probability distributions defined on the interval <span class="co">[</span><span class="ot">0, 1</span><span class="co">]</span> parameterised by two positive shape parameters, typically denoted by α and β. It is used to model the behaviour of random variables limited to intervals of finite length in a wide variety of disciplines (@fig-beta).</span>
<span id="cb11-510"><a href="#cb11-510"></a></span>
<span id="cb11-513"><a href="#cb11-513"></a><span class="in">```{r}</span></span>
<span id="cb11-514"><a href="#cb11-514"></a><span class="co">#| echo: false</span></span>
<span id="cb11-515"><a href="#cb11-515"></a><span class="co">#| fig.align: center</span></span>
<span id="cb11-516"><a href="#cb11-516"></a><span class="co">#| fig.cap: "A Beta distribution with 100 trials and 5 simulations."</span></span>
<span id="cb11-517"><a href="#cb11-517"></a><span class="co">#| label: fig-beta</span></span>
<span id="cb11-518"><a href="#cb11-518"></a></span>
<span id="cb11-519"><a href="#cb11-519"></a>beta <span class="ot">&lt;-</span> <span class="fu">tidy_beta</span>(<span class="at">.n =</span> <span class="dv">100</span>, <span class="at">.shape1 =</span> <span class="dv">1</span>, <span class="at">.shape2 =</span> <span class="dv">1</span>, <span class="at">.ncp =</span> <span class="dv">0</span>, <span class="at">.num_sims =</span> <span class="dv">5</span>)</span>
<span id="cb11-520"><a href="#cb11-520"></a></span>
<span id="cb11-521"><a href="#cb11-521"></a><span class="fu">ggarrange</span>(</span>
<span id="cb11-522"><a href="#cb11-522"></a>  <span class="fu">tidy_autoplot</span>(beta, <span class="at">.plot_type =</span> <span class="st">"density"</span>),</span>
<span id="cb11-523"><a href="#cb11-523"></a>  <span class="fu">tidy_autoplot</span>(beta, <span class="at">.plot_type =</span> <span class="st">"quantile"</span>),</span>
<span id="cb11-524"><a href="#cb11-524"></a>  <span class="fu">tidy_autoplot</span>(beta, <span class="at">.plot_type =</span> <span class="st">"probability"</span>),</span>
<span id="cb11-525"><a href="#cb11-525"></a>  <span class="fu">tidy_autoplot</span>(beta, <span class="at">.plot_type =</span> <span class="st">"qq"</span>),</span>
<span id="cb11-526"><a href="#cb11-526"></a>  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">nrow =</span> <span class="dv">2</span></span>
<span id="cb11-527"><a href="#cb11-527"></a>)</span>
<span id="cb11-528"><a href="#cb11-528"></a><span class="in">```</span></span>
<span id="cb11-529"><a href="#cb11-529"></a></span>
<span id="cb11-530"><a href="#cb11-530"></a><span class="fu"># Finding one's data distribution</span></span>
<span id="cb11-531"><a href="#cb11-531"></a></span>
<span id="cb11-532"><a href="#cb11-532"></a>Data belonging to a sample will never exactly follow a specific distribution, even when the test for normality says it does---there will always be a small probability that they are non-normal and is in fact better described by some other distribution. In other words, data are only *compatible* with a certain distribution, and one can never answer the question "Does my data follow the distribution xy exactly?" as simply as providing a yes/no answer. So what now? How does one find one's data distribution? We can use the **Cullen and Frey graph** function that lives in the **fitdistrplus** package. This graph tells us whether the skewness and kurtosis of our data are consistent with that of a particular distribution. We will demonstrate by generating various data distributions and testing them using the Cullen and Frey graph.</span>
<span id="cb11-533"><a href="#cb11-533"></a></span>
<span id="cb11-534"><a href="#cb11-534"></a><span class="in">```{r dist-plot1, message=FALSE}</span></span>
<span id="cb11-535"><a href="#cb11-535"></a><span class="co">#| column: screen-inset-shaded</span></span>
<span id="cb11-536"><a href="#cb11-536"></a><span class="co">#| layout-nrow: 1</span></span>
<span id="cb11-537"><a href="#cb11-537"></a></span>
<span id="cb11-538"><a href="#cb11-538"></a><span class="fu">library</span>(fitdistrplus)</span>
<span id="cb11-539"><a href="#cb11-539"></a><span class="fu">library</span>(logspline)</span>
<span id="cb11-540"><a href="#cb11-540"></a></span>
<span id="cb11-541"><a href="#cb11-541"></a><span class="co"># Generate log-normal data</span></span>
<span id="cb11-542"><a href="#cb11-542"></a>y <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">37.50</span>,<span class="fl">46.79</span>,<span class="fl">48.30</span>,<span class="fl">46.04</span>,<span class="fl">43.40</span>,<span class="fl">39.25</span>,<span class="fl">38.49</span>,<span class="fl">49.51</span>,<span class="fl">40.38</span>,<span class="fl">36.98</span>,<span class="fl">40.00</span>,</span>
<span id="cb11-543"><a href="#cb11-543"></a><span class="fl">38.49</span>,<span class="fl">37.74</span>,<span class="fl">47.92</span>,<span class="fl">44.53</span>,<span class="fl">44.91</span>,<span class="fl">44.91</span>,<span class="fl">40.00</span>,<span class="fl">41.51</span>,<span class="fl">47.92</span>,<span class="fl">36.98</span>,<span class="fl">43.40</span>,</span>
<span id="cb11-544"><a href="#cb11-544"></a><span class="fl">42.26</span>,<span class="fl">41.89</span>,<span class="fl">38.87</span>,<span class="fl">43.02</span>,<span class="fl">39.25</span>,<span class="fl">40.38</span>,<span class="fl">42.64</span>,<span class="fl">36.98</span>,<span class="fl">44.15</span>,<span class="fl">44.91</span>,<span class="fl">43.40</span>,</span>
<span id="cb11-545"><a href="#cb11-545"></a><span class="fl">49.81</span>,<span class="fl">38.87</span>,<span class="fl">40.00</span>,<span class="fl">52.45</span>,<span class="fl">53.13</span>,<span class="fl">47.92</span>,<span class="fl">52.45</span>,<span class="fl">44.91</span>,<span class="fl">29.54</span>,<span class="fl">27.13</span>,<span class="fl">35.60</span>,</span>
<span id="cb11-546"><a href="#cb11-546"></a><span class="fl">45.34</span>,<span class="fl">43.37</span>,<span class="fl">54.15</span>,<span class="fl">42.77</span>,<span class="fl">42.88</span>,<span class="fl">44.26</span>,<span class="fl">27.14</span>,<span class="fl">39.31</span>,<span class="fl">24.80</span>,<span class="fl">16.62</span>,<span class="fl">30.30</span>,</span>
<span id="cb11-547"><a href="#cb11-547"></a><span class="fl">36.39</span>,<span class="fl">28.60</span>,<span class="fl">28.53</span>,<span class="fl">35.84</span>,<span class="fl">31.10</span>,<span class="fl">34.55</span>,<span class="fl">52.65</span>,<span class="fl">48.81</span>,<span class="fl">43.42</span>,<span class="fl">52.49</span>,<span class="fl">38.00</span>,</span>
<span id="cb11-548"><a href="#cb11-548"></a><span class="fl">38.65</span>,<span class="fl">34.54</span>,<span class="fl">37.70</span>,<span class="fl">38.11</span>,<span class="fl">43.05</span>,<span class="fl">29.95</span>,<span class="fl">32.48</span>,<span class="fl">24.63</span>,<span class="fl">35.33</span>,<span class="fl">41.34</span>)</span>
<span id="cb11-549"><a href="#cb11-549"></a></span>
<span id="cb11-550"><a href="#cb11-550"></a><span class="fu">plot</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(y)), <span class="at">y =</span> y)</span>
<span id="cb11-551"><a href="#cb11-551"></a><span class="fu">hist</span>(y)</span>
<span id="cb11-552"><a href="#cb11-552"></a><span class="fu">descdist</span>(y, <span class="at">discrete =</span> <span class="cn">FALSE</span>, <span class="at">boot =</span> <span class="dv">100</span>)</span>
<span id="cb11-553"><a href="#cb11-553"></a><span class="in">```</span></span>
<span id="cb11-554"><a href="#cb11-554"></a></span>
<span id="cb11-555"><a href="#cb11-555"></a><span class="in">```{r dist-plot2}</span></span>
<span id="cb11-556"><a href="#cb11-556"></a><span class="co">#| column: screen-inset-shaded</span></span>
<span id="cb11-557"><a href="#cb11-557"></a><span class="co">#| layout-nrow: 1</span></span>
<span id="cb11-558"><a href="#cb11-558"></a></span>
<span id="cb11-559"><a href="#cb11-559"></a><span class="co"># normally distributed data</span></span>
<span id="cb11-560"><a href="#cb11-560"></a>y <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>, <span class="dv">13</span>, <span class="dv">2</span>)</span>
<span id="cb11-561"><a href="#cb11-561"></a></span>
<span id="cb11-562"><a href="#cb11-562"></a><span class="fu">plot</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>), <span class="at">y =</span> y)</span>
<span id="cb11-563"><a href="#cb11-563"></a><span class="fu">hist</span>(y)</span>
<span id="cb11-564"><a href="#cb11-564"></a><span class="fu">descdist</span>(y, <span class="at">discrete =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-565"><a href="#cb11-565"></a><span class="in">```</span></span>
<span id="cb11-566"><a href="#cb11-566"></a></span>
<span id="cb11-567"><a href="#cb11-567"></a><span class="in">```{r dist-plot3}</span></span>
<span id="cb11-568"><a href="#cb11-568"></a><span class="co">#| column: screen-inset-shaded</span></span>
<span id="cb11-569"><a href="#cb11-569"></a><span class="co">#| layout-nrow: 1</span></span>
<span id="cb11-570"><a href="#cb11-570"></a></span>
<span id="cb11-571"><a href="#cb11-571"></a><span class="co"># uniformly distributed data</span></span>
<span id="cb11-572"><a href="#cb11-572"></a>y <span class="ot">&lt;-</span> <span class="fu">runif</span>(<span class="dv">100</span>)</span>
<span id="cb11-573"><a href="#cb11-573"></a></span>
<span id="cb11-574"><a href="#cb11-574"></a><span class="fu">plot</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>), <span class="at">y =</span> y)</span>
<span id="cb11-575"><a href="#cb11-575"></a><span class="fu">hist</span>(y)</span>
<span id="cb11-576"><a href="#cb11-576"></a><span class="fu">descdist</span>(y, <span class="at">discrete =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-577"><a href="#cb11-577"></a><span class="in">```</span></span>
<span id="cb11-578"><a href="#cb11-578"></a></span>
<span id="cb11-579"><a href="#cb11-579"></a><span class="in">```{r dist-plot4, width = 6, height = 10}</span></span>
<span id="cb11-580"><a href="#cb11-580"></a><span class="co">#| column: screen-inset-shaded</span></span>
<span id="cb11-581"><a href="#cb11-581"></a><span class="co">#| layout-nrow: 1</span></span>
<span id="cb11-582"><a href="#cb11-582"></a></span>
<span id="cb11-583"><a href="#cb11-583"></a><span class="co"># uniformly distributed data</span></span>
<span id="cb11-584"><a href="#cb11-584"></a>y <span class="ot">&lt;-</span> <span class="fu">rexp</span>(<span class="dv">100</span>, <span class="fl">0.7</span>)</span>
<span id="cb11-585"><a href="#cb11-585"></a></span>
<span id="cb11-586"><a href="#cb11-586"></a><span class="fu">plot</span>(<span class="at">x =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>), <span class="at">y =</span> y)</span>
<span id="cb11-587"><a href="#cb11-587"></a><span class="fu">hist</span>(y)</span>
<span id="cb11-588"><a href="#cb11-588"></a><span class="fu">descdist</span>(y, <span class="at">discrete =</span> <span class="cn">FALSE</span>)</span>
<span id="cb11-589"><a href="#cb11-589"></a><span class="in">```</span></span>
<span id="cb11-590"><a href="#cb11-590"></a></span>
<span id="cb11-591"><a href="#cb11-591"></a>There is also a whole bunch of other approaches to use to try and identify the data distribution. Let us start with the gold standard first: normal data. We will demonstrate some visualisation approaches. The one that you already know is a basic histogram; it tells us something about the distribution's skewness, the tails, the mode(s) of the data, outliers, etc. Histograms can be compared to shapes associated with idealistic (simulated) distributions, as we will do here.</span>
<span id="cb11-592"><a href="#cb11-592"></a></span>
<span id="cb11-593"><a href="#cb11-593"></a><span class="in">```{r dist-plot5}</span></span>
<span id="cb11-594"><a href="#cb11-594"></a></span>
<span id="cb11-595"><a href="#cb11-595"></a>y <span class="ot">&lt;-</span><span class="fu">rnorm</span>(<span class="at">n =</span> <span class="dv">200</span>, <span class="at">m =</span> <span class="dv">13</span>, <span class="at">sd =</span> <span class="dv">2</span>)</span>
<span id="cb11-596"><a href="#cb11-596"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>))</span>
<span id="cb11-597"><a href="#cb11-597"></a><span class="co"># using some basic base graphics as ggplot2 is overkill;</span></span>
<span id="cb11-598"><a href="#cb11-598"></a><span class="co"># we can get a histogram using hist() statement</span></span>
<span id="cb11-599"><a href="#cb11-599"></a><span class="fu">hist</span>(y, <span class="at">main =</span> <span class="st">"Histogram of observed data"</span>)</span>
<span id="cb11-600"><a href="#cb11-600"></a><span class="fu">plot</span>(<span class="fu">density</span>(y), <span class="at">main =</span> <span class="st">"Density estimate of data"</span>)</span>
<span id="cb11-601"><a href="#cb11-601"></a><span class="fu">plot</span>(<span class="fu">ecdf</span>(y), <span class="at">main =</span> <span class="st">"Empirical cumulative distribution function"</span>)</span>
<span id="cb11-602"><a href="#cb11-602"></a><span class="co"># standardise the data</span></span>
<span id="cb11-603"><a href="#cb11-603"></a>z.norm <span class="ot">&lt;-</span> (y <span class="sc">-</span> <span class="fu">mean</span>(y)) <span class="sc">/</span> <span class="fu">sd</span>(y) </span>
<span id="cb11-604"><a href="#cb11-604"></a><span class="co"># make a qqplot</span></span>
<span id="cb11-605"><a href="#cb11-605"></a><span class="fu">qqnorm</span>(z.norm)</span>
<span id="cb11-606"><a href="#cb11-606"></a><span class="co"># add a 45-degree reference line</span></span>
<span id="cb11-607"><a href="#cb11-607"></a><span class="fu">abline</span>(<span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb11-608"><a href="#cb11-608"></a><span class="in">```</span></span>
<span id="cb11-609"><a href="#cb11-609"></a></span>
<span id="cb11-610"><a href="#cb11-610"></a>Above we have also added a diagonal line to the qqplot. If the sampled data come from the population with the chosen distribution, the points should fall approximately along this reference line. The greater the departure from this reference line, the greater the evidence for the conclusion that the data set have come from a population with a different distribution.</span>
<span id="cb11-611"><a href="#cb11-611"></a></span>
<span id="cb11-614"><a href="#cb11-614"></a><span class="in">```{r}</span></span>
<span id="cb11-615"><a href="#cb11-615"></a><span class="co"># curve(dnorm(100, m = 10, sd = 2), from = 0, to = 20, main = "Normal distribution")</span></span>
<span id="cb11-616"><a href="#cb11-616"></a><span class="co"># curve(dgamma(100, scale = 1.5, shape = 2), from = 0, to = 15, main = "Gamma distribution")</span></span>
<span id="cb11-617"><a href="#cb11-617"></a><span class="co"># curve(dweibull(100, scale = 2.5, shape = 1.5), from = 0, to = 15, main = "Weibull distribution")</span></span>
<span id="cb11-618"><a href="#cb11-618"></a><span class="in">```</span></span>
<span id="cb11-619"><a href="#cb11-619"></a></span>
<span id="cb11-620"><a href="#cb11-620"></a><span class="co">&lt;!-- ## Exercises --&gt;</span></span>
<span id="cb11-621"><a href="#cb11-621"></a></span>
<span id="cb11-622"><a href="#cb11-622"></a><span class="co">&lt;!-- </span><span class="al">###</span><span class="co"> Exercise 1 --&gt;</span></span>
<span id="cb11-623"><a href="#cb11-623"></a></span>
<span id="cb11-624"><a href="#cb11-624"></a><span class="co">&lt;!-- Choose two different datasets and plot them as histograms with density curves overlayed. Label them with the distribution they appear to be and stitch them together with `ggarrange()`. --&gt;</span></span>
</code></pre></div><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>© 2026, AJ Smit</p>
</div>   
    <div class="nav-footer-center">

<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB744/basic_stats/04-distributions.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>


<script src="../../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>