<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.33">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Smit, A. J.">
<meta name="dcterms.date" content="2021-01-01">
<title>8. ANOVA – The Tangled Bank</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../BCB744/basic_stats/09-regressions.html" rel="next">
<link href="../../BCB744/basic_stats/07-t_tests.html" rel="prev">
<link href="../../images/favicon.ico" rel="icon">
<script src="../../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-59898bd1c6b9d2bb783127feaa000c76.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-7116f089cfdb46b66375c744b53d3c7f.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-SCYH68M489"></script><script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-SCYH68M489', { 'anonymize_ip': true});
</script><script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"headline",
  "consent_type":"express",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  ,
"language":"en"
  });
});
</script><style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script><script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script><script src="../../site_libs/kePrint-0.0.1/kePrint.js"></script><link href="../../site_libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script><link rel="stylesheet" href="../../styles.css">
<meta property="og:title" content="8. ANOVA – The Tangled Bank">
<meta property="og:description" content="Comparing the means of more than two groups">
<meta property="og:image" content="http://tangledbank.netlify.app/BCB744/basic_stats/data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==">
<meta property="og:site_name" content="The Tangled Bank">
</head>
<body class="nav-sidebar floating nav-fixed slimcontent quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">The Tangled Bank</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-vignettes" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Vignettes</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-vignettes">
<li>
    <a class="dropdown-item" href="../../vignettes/netCDF_dates.html">
 <span class="dropdown-text">Dates From netCDF Files: Two Approaches</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/prep_NOAA_OISST.html">
 <span class="dropdown-text">Downloading and Prep: ERDDAP</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_ERDDAP.html">
 <span class="dropdown-text">Retrieving Chlorophyll-a Data from ERDDAP Servers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/alt_method.html">
 <span class="dropdown-text">Downloading and Prep: Python + CDO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/gridded_data.html">
 <span class="dropdown-text">Detecting Events in Gridded Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/regridding.html">
 <span class="dropdown-text">Regridding gridded data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/MHW_MCS_horizonplots.html">
 <span class="dropdown-text">Displaying MHWs and MCSs as Horizon Plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/heatwaveR_issues.html">
 <span class="dropdown-text">heatwaveR Issues</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_sightings.html">
 <span class="dropdown-text">Plotting the Whale Sightings and Chlorophyll-<em>a</em> Concentrations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_localisation.html">
 <span class="dropdown-text">Spatial Localisation, Subsetting, and Aggregation of the Chlorophyll-<em>a</em> Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/elem_ts_methods.html">
 <span class="dropdown-text">Wavelet Analysis of Diatom Time Series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/buffer_data_extract.html">
 <span class="dropdown-text">Extracting gridded data within a buffer</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-high-performace-computing" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">High-Performace Computing</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-high-performace-computing">
<li>
    <a class="dropdown-item" href="../../vignettes/README_Lengau.html">
 <span class="dropdown-text">Using Lengau</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/PBSPro_users.html">
 <span class="dropdown-text">PBSPro user and job management</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/README_tmux.html">
 <span class="dropdown-text">Using tmux</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item">
    <a class="nav-link" href="../../blog.html"> 
<span class="menu-text">Tangled Bank Blog</span></a>
  </li>  
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-blogroll" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Blogroll</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-blogroll">
<li>
    <a class="dropdown-item" href="https://www.r-bloggers.com">
 <span class="dropdown-text">R-bloggers</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item compact">
    <a class="nav-link" href="../../blog.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">
<li>
    <a class="dropdown-item" href="https://github.com/ajsmit/tangled_bank">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://bugs.com">
 <span class="dropdown-text">Report a Bug</span></a>
  </li>  
    </ul>
</li>
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../BCB744/BCB744_index.html">BCB744: Introduction to R, &amp; Biostatistics</a></li><li class="breadcrumb-item"><a href="../../BCB744/basic_stats/08-anova.html">8. ANOVA</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="../../index.html" class="sidebar-logo-link">
      <img src="../../images/my_tree_hex.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none"></a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">UNDERGRADUATE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC223: Plant Ecophysiology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/BDC223_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L00-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 0: Introduction</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L01-worldmapper.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: Worldmapper</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L02-SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2: Surface Area to Volume (SA/V) Ratios in Biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L03-plant_stress.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3: Plant Stress</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L04-carbon_cycle.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Perturbations to the Carbon Cycle</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L05-light.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 5: Light</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L06a-pigments_photosynthesis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6a: Pigments and Photosynthesis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L06b-jassby_platt.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6b: PI Curves – Jassby and Platt</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L07-chromatic_adaptation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 7: Chromatic Adaptation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L08a-nutrient_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8a: Nutrient Uptake</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L08b-nutrients_michaelis_menten.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8b: Uptake Kinetics – Michaelis-Menten</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab1_SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1: Surface Area to Volume (SA/V) Ratios in Biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab2_misc_calcs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2: Miscellaneous Calculations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab3_PI_curves.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3: PI Curves – Jassby and Platt</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab4_nitrogen_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4: Uptake Kinetics – Michaelis-Menten</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/BDC223_FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC334: Biogeography &amp; Global Ecology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/BDC334_syllabus.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/BDC334-Lecture-Transcripts.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">BCB334 Lecture Transcript</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: Introductory Lecture</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-02-ecosystems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2. Overview of Ecosystems</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-03-gradients.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3. Ecological Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Biodiversity Concepts</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 5: Multivariate Data</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-06-unified-ecology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6: Unified Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 7: Threats to Biodiversity</span>
  </li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 8: Nature's Contribution to People</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/IKS_Conservation.pdf" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 9: Indigenous Knowledge Systems and Conservation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1. Ecological Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-02a-r_rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2a. Introduction `R` &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-02b-env_dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2b. Environmental Distance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-03-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3. Quantifying Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4. Species Distribution Patterns</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/BDC334_AI_Tutor.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">BCB334 AI Tutor</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pages/Transboundary_systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transboundary systems</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS CORE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">BCB744: Introduction to R, &amp; Biostatistics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/BCB744_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/01-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. R &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/02-working-with-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Working With Data &amp; Code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/03-data-in-R.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Data Classes &amp; Structures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/04-workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. R Workflows</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/05-graphics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Graphics with <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/06-faceting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Faceting Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/07-brewing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Brewing Colours</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/08-mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Mapping With <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/09-mapping_style.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Mapping With Style</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/10-mapping_rnaturalearth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Mapping With Natural Earth</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/11-mapping_quakes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. The Fiji Earthquake Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/12-tidy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Tidy Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/13-tidier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Tidier Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/14-tidiest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Tidiest Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/15-recap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15. Recap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../assessments/BCB744_Intro_R_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Intro R Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../assessments/examples/BCB744_Intro_R_Example_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 Intro R Example 1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../assessments/examples/BCB744_Intro_R_Example_2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 Intro R Example 2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../self-assessment_v1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Self-Assessment Form</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../assessments/BCB744_Intro_R_Presentations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Presentations</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../assessments/BCB744_Intro_R_Test_2025.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Intro R Test</span></span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/01-scientific-philosophy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Philosophy of Science</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/02-summarise-and-describe.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Data Summaries &amp; Descriptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/03-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Statistical Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/04-distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data Distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/05-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Statistical Inference</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/06-assumptions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Assumptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/07-t_tests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. <em>t</em>-Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/08-anova.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">8. ANOVA</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/09-regressions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Simple Linear Regressions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/10-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/11-decision_guide.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/12-glance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Non-Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/13-confidence.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Confidence Intervals</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/14-transformations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Data Transformations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../assessments/BCB744_Biostatistics_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Biostatistics Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../assessments/examples/BCB744_BioStats_Example_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 BioStats 2025 Example 1</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS ELECTIVES</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">BCB743: Quantitative Ecology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/BCB743_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/review.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1-4: Review Concepts fo Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5: Correlations &amp; Associations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7: Intro to Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8a: Principal Components Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCA_examples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8b: PCA: Additional Examples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCA_SDG_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8c: PCA: WHO SDG Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/CA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9a: Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/DCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9b: Detrended Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCoA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10: Principal Coordinates Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/nMDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11a: Non-metric Multidimensional Scaling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/nMDS_diatoms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11b: nMDS: PERMANOVA (Diatoms) Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/unconstrained-summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12: Unconstrained Ordination Summary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/constrained_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13a: Distance-Based Redundancy Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/two_oceans_appendices.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13b: db-RDA: Seaweeds Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/cluster_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14: Cluster Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/model_building.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15: Model Building</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/deep_dive.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16a: Deep Dive Into Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/multiple_regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16b: Multiple Regression</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/Combined Faculty AI Policy 15082025.pdf" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Faculty of Natural Sciences AI Policy</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">SUPPORT</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Web Resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/general_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/ecology_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quantitative Ecology Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/spatial_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial R Resources</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#at-a-glance" id="toc-at-a-glance" class="nav-link active" data-scroll-target="#at-a-glance"><span class="header-section-number">1</span> At a glance</a></li>
  <li>
<a href="#introduction-to-anova" id="toc-introduction-to-anova" class="nav-link" data-scroll-target="#introduction-to-anova"><span class="header-section-number">2</span> Introduction to ANOVA</a>
  <ul class="collapse">
<li><a href="#remember-the-t-test" id="toc-remember-the-t-test" class="nav-link" data-scroll-target="#remember-the-t-test"><span class="header-section-number">2.1</span> Remember the <em>t</em>-test</a></li>
  <li><a href="#why-not-do-multiple-t-tests" id="toc-why-not-do-multiple-t-tests" class="nav-link" data-scroll-target="#why-not-do-multiple-t-tests"><span class="header-section-number">2.2</span> Why not do multiple <em>t</em>-tests?</a></li>
  </ul>
</li>
  <li>
<a href="#anova" id="toc-anova" class="nav-link" data-scroll-target="#anova"><span class="header-section-number">3</span> ANOVA</a>
  <ul class="collapse">
<li><a href="#single-factor" id="toc-single-factor" class="nav-link" data-scroll-target="#single-factor"><span class="header-section-number">3.1</span> Single factor</a></li>
  <li><a href="#multiple-factors" id="toc-multiple-factors" class="nav-link" data-scroll-target="#multiple-factors"><span class="header-section-number">3.2</span> Multiple factors</a></li>
  </ul>
</li>
  <li>
<a href="#sec-alt" id="toc-sec-alt" class="nav-link" data-scroll-target="#sec-alt"><span class="header-section-number">4</span> Alternatives to ANOVA</a>
  <ul class="collapse">
<li><a href="#wilcoxon-rank-sum-test" id="toc-wilcoxon-rank-sum-test" class="nav-link" data-scroll-target="#wilcoxon-rank-sum-test"><span class="header-section-number">4.1</span> Wilcoxon rank sum test</a></li>
  <li>
<a href="#kruskall-wallis-rank-sum-test" id="toc-kruskall-wallis-rank-sum-test" class="nav-link" data-scroll-target="#kruskall-wallis-rank-sum-test"><span class="header-section-number">4.2</span> Kruskall-Wallis rank sum test</a>
  <ul class="collapse">
<li><a href="#single-factor-1" id="toc-single-factor-1" class="nav-link" data-scroll-target="#single-factor-1"><span class="header-section-number">4.2.1</span> Single factor</a></li>
  <li><a href="#multiple-factors-1" id="toc-multiple-factors-1" class="nav-link" data-scroll-target="#multiple-factors-1"><span class="header-section-number">4.2.2</span> Multiple factors</a></li>
  </ul>
</li>
  <li><a href="#the-sa-time-data" id="toc-the-sa-time-data" class="nav-link" data-scroll-target="#the-sa-time-data"><span class="header-section-number">4.3</span> The SA time data</a></li>
  </ul>
</li>
  <li>
<a href="#examples" id="toc-examples" class="nav-link" data-scroll-target="#examples"><span class="header-section-number">5</span> Examples</a>
  <ul class="collapse">
<li><a href="#snakes" id="toc-snakes" class="nav-link" data-scroll-target="#snakes"><span class="header-section-number">5.1</span> Snakes!</a></li>
  </ul>
</li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB744/basic_stats/08-anova.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../BCB744/BCB744_index.html">BCB744: Introduction to R, &amp; Biostatistics</a></li><li class="breadcrumb-item"><a href="../../BCB744/basic_stats/08-anova.html">8. ANOVA</a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">8. ANOVA</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
<p class="subtitle lead">Comparing the means of more than two groups</p>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://tangledbank.netlify.app">Smit, A. J.</a> <a href="https://orcid.org/0000-0002-3799-6126" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uwc.ac.za">
            University of the Western Cape
            </a>
          </p>
      </div>
  </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 1, 2021</p>
    </div>
  </div>
  
    
  </div>
  


</header><p><a href="../../images/fancy_stats.jpg" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="../../images/fancy_stats.jpg" class="img-fluid" style="width:50.0%"></a></p>
<blockquote class="blockquote">
<p><em>“He uses statistics as a drunken man uses lamp posts—for support rather than for illumination.”</em></p>
<p>— Marissa Mayer</p>
</blockquote>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
In this Chapter
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Single factor ANOVA</li>
<li>Multiple factor ANOVA</li>
<li>Tukey HSD test</li>
<li>Wilcoxon rank sum test</li>
<li>Kruskal-Wallis rank sum test</li>
</ul>
</div>
</div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Cheatsheet
</div>
</div>
<div class="callout-body-container callout-body">
<p>Find here a <a href="../../docs/Methods_cheatsheet_v1.pdf">Cheatsheet</a> on statistical methods.</p>
</div>
</div>
<div class="callout callout-style-simple callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tasks to complete in this Chapter
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Task G</li>
</ul>
</div>
</div>
<section id="at-a-glance" class="level1 page-columns page-full" data-number="1"><h1 data-number="1">
<span class="header-section-number">1</span> At a glance</h1>
<p>Where the Student’s and Welch’s <em>t</em>-tests are used to test for the difference between the means of two groups, analysis of variance (ANOVA)<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a> is used to <em>compare the means of three or more groups</em>. The following expectations about your data should be met to conduct an ANOVA:</p>
<div class="no-row-height column-margin column-container"><div id="fn1"><p><sup>1</sup>&nbsp;Variations of the basic ANOVA include an analysis of covariance (ANCOVA) when, in addition to the categorical variable, your data also contains a <strong>continuous covariate</strong> (e.g., age, mass) that you want to control. Or, if your study involves multiple dependent variables, you may need to consider using <strong>multivariate analysis of variance (MANOVA)</strong> instead. ANOVAs that have multiple independent variables are called <strong>factorial ANOVAs</strong> (e.g.&nbsp;2-way ANOVA or 3-way ANOVA).</p></div></div><ul>
<li><p><strong>Continuous dependent variable</strong> The dependent variable should be measured on a <em>continuous scale</em> (e.g., height, weight, test scores).</p></li>
<li>
<p><strong>Categorical independent variable</strong> The independent variable should be <em>categorical with at least three levels</em> or groups (e.g., different treatments, age groups).</p>
<ul>
<li>
<strong>Independent groups</strong> The groups being compared should be independent of each other, meaning that the observations within each group should not affect the other group’s observations. In the case of repeated measures ANOVA, the groups are related (e.g., before-and-after measurements or multiple measurements on the same subjects).</li>
</ul>
</li>
</ul>
<p>Other assumptions to be aware of with regards to the dependent variable:</p>
<ul>
<li>Independence of data</li>
<li>Normally distributed data</li>
<li>Homogeneity of variances</li>
<li>We also encourage that the data are balanced</li>
</ul></section><section id="introduction-to-anova" class="level1 page-columns page-full" data-number="2"><h1 data-number="2">
<span class="header-section-number">2</span> Introduction to ANOVA</h1>
<p>Whole big books have been written about analysis of variance (ANOVA). Although there are many experimental designs that may be analysed with ANOVAs, biologists are taught to pay special attention to the design of experiments, and generally make sure that the experiments are fully <em>factorial</em> (in the case of two-way or higher ANOVAs) and balanced. For this reason we will focus in this introductory statistics course on one-way and factorial ANOVAs only.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Factorial designs
</div>
</div>
<div class="callout-body-container callout-body">
<p>A factorial experimental design involves studying the effects of two or more factors (also known as independent variables) on a response variable (dependent variable) simultaneously. The term ‘factorial’ comes from the fact that each level of one factor is combined with each level of the other factor(s), resulting in all possible combinations of factor levels.</p>
<p>In a factorial design, the <em>main effects</em> of each factor and the <em>interaction effects</em> between factors can be analysed. When we have two factors, this is typically done by applying a 2-way ANOVA, but higher-order ANOVAs are also available. The main effect of a factor is its individual effect on the response variable (ignoring the effect due to the other(s)), while the interaction effect occurs when the effect of one factor depends on the level of another factor.</p>
<p>The simplest factorial design is the 2 <span class="math inline">\(\times\)</span> 2 design, which involves two factors, each with two levels. For example, if you are studying the effect of temperature (high vs.&nbsp;low) and fertiliser concentration (high vs.&nbsp;low) on the yield of a crop, a 2 <span class="math inline">\(\times\)</span> 2 factorial design would involve four experimental conditions:</p>
<ul>
<li>high temperature and high fertiliser concentration</li>
<li>high temperature and low fertiliser concentration</li>
<li>low temperature and high fertiliser concentration</li>
<li>low temperature and low fertiliser concentration</li>
</ul>
<p>In the above example there are 4 possible combinations. Factorial designs can involve more than two factors and/or more than two levels per factor, leading to more complex experimental setups. For example, in a 3 <span class="math inline">\(\times\)</span> 4 there will be 12 combinations of the two factor levels, and in a 2 <span class="math inline">\(\times\)</span> 3 <span class="math inline">\(\times\)</span> 5 factorial experiment the three factors with their respective levels will result in 30 combinations.</p>
</div>
</div>
<p>As we have seen in <a href="../../BCB744/basic_stats/07-t_tests.html">Chapter 7</a> about <em>t</em>-tests, ANOVAs also require that some assumptions are met:</p>
<ul>
<li><p><strong>Normal distribution</strong> The data in each group should follow a normal distribution or be approximately normally distributed. If the assumption of normality is not met, as, for example, determined with the <code><a href="https://rdrr.io/r/stats/shapiro.test.html">shapiro.test()</a></code>, you may consider using a non-parametric alternative such as Kruskal-Wallis rank sum test, <code><a href="https://rdrr.io/r/stats/kruskal.test.html">kruskal.test()</a></code>. This assumption can be relaxed for large sample sizes due to the <a href="../../BCB744/basic_stats/04-distributions.html#sec-normal">Central Limit Theorem</a>.</p></li>
<li><p><strong>Homoscedasticity</strong> The variances of the groups should be approximately equal. This assumption can be tested using Levene’s test, <code><a href="https://rdrr.io/pkg/car/man/leveneTest.html">car::leveneTest()</a></code>, or Bartlett’s test, <code><a href="https://rdrr.io/r/stats/bartlett.test.html">bartlett.test()</a></code>. When the variances are different but the data are normally distributed, consider the ‘Welch ANOVA,’ <code><a href="https://rdrr.io/r/stats/oneway.test.html">oneway.test()</a></code>, instead of <code><a href="https://rdrr.io/r/stats/aov.html">aov()</a></code>.</p></li>
<li><p><strong>Random sampling</strong> The data should be obtained through random sampling or random assignment, ensuring that each observation has an equal chance of being included in the sample.</p></li>
<li><p><strong>Independent observations</strong> The observations within each group should be independent of each other.</p></li>
</ul>
<p>If some of the above assumptions are violated, then your course of action is to either use a <a href="@sec-alt">non-parametric test</a> (and <a href="../../docs/Methods_cheatsheet_v1.pdf">here</a>), transform the data (as in <a href="../../BCB744/basic_stats/06-assumptions.html">Chapter 6</a>) or use a generalised linear model if non-normal, or to use a linear mixed model when non-independence of data cannot be guaranteed. As I have already indicated, ANOVAs are also sensitive to the presence of outliers, so we need to ensure that outliers are not present. Outliers can be removed but if they are an important feature of the data, then a non-parametric test can be used.</p>
<p>Rather than talking about <em>t</em>-tests and ANOVAs as if they do different things, let us acknowledge that they ask a similar question. That question being, “are the means of these two or more things we want to compare different, or are they the same?” At this stage it is important to note that, as with <em>t</em>-tests, the independent variable is expected to be categorical (i.e.&nbsp;a factor denoting two or more different treatments or sampling conditions) and that the dependent variable must be continuous. You may perhaps be more familiar with this question when it is presented as a set of hypotheses as we saw in a <em>t</em>-test:</p>
<ul>
<li>
<strong><em>H</em><sub>0</sub></strong> Group A is not different from Group B.</li>
<li>
<strong><em>H</em><sub>a</sub></strong> Group A is different from Group B.</li>
</ul>
<p>In an ANOVA, hypotheses willl be more similar to these:</p>
<ul>
<li>
<strong><em>H</em><sub>0</sub></strong> There are no differences among Groups A, B, or C.</li>
<li>
<strong><em>H</em><sub>a</sub></strong> There are differences among Groups A, B, or C.</li>
</ul>
<p>This is a scientific question in the simplest sense. Often, for basic inquiries such as that posed above, we need to see if one group differs significantly from another. The way in which we accomplish this is by looking at the mean and variance within a set of data compared against another similar set.</p>
<section id="remember-the-t-test" class="level2" data-number="2.1"><h2 data-number="2.1" class="anchored" data-anchor-id="remember-the-t-test">
<span class="header-section-number">2.1</span> Remember the <em>t</em>-test</h2>
<p>As you already know, a <em>t</em>-test is used when we want to compare two different sample sets against one another. This is also known as a two-factor or two level test. When one wants to compare multiple (more than two) sample sets against one another an ANOVA is required (I’ll get there shortly). Remember how to perform a <em>t</em>-test in R: we will revisit this test using the <code>chicks</code> data, but only for Diets 1 and 2 from day 21.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># First grab the data</span></span>
<span id="cb1-2"><a href="#cb1-2"></a>chicks <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(ChickWeight)</span>
<span id="cb1-3"><a href="#cb1-3"></a></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="co"># Then subset out only the sample sets to be compared</span></span>
<span id="cb1-5"><a href="#cb1-5"></a>chicks_sub <span class="ot">&lt;-</span> chicks <span class="sc">%&gt;%</span> </span>
<span id="cb1-6"><a href="#cb1-6"></a>  <span class="fu">filter</span>(Diet <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), Time <span class="sc">==</span> <span class="dv">21</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Once we have filtered our data we may now perform the <em>t</em>-test.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="fu">t.test</span>(weight <span class="sc">~</span> Diet, <span class="at">data =</span> chicks_sub)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Welch Two Sample t-test

data:  weight by Diet
t = -1.2857, df = 15.325, p-value = 0.2176
alternative hypothesis: true difference in means between group 1 and group 2 is not equal to 0
95 percent confidence interval:
 -98.09263  24.19263
sample estimates:
mean in group 1 mean in group 2 
         177.75          214.70 </code></pre>
</div>
</div>
<p>As one may recall from <a href="../../BCB744/basic_stats/07-t_tests.html">Chapter 7</a>, whenever we want to give a formula to a function in R, we use the <code>~</code>. The formula used above, <code>weight ~ Diet</code>, reads in plain English as “weight as a function of diet”. This is perhaps easier to understand as “<em>Y</em> as a function of <em>X</em>.” This means that we are assuming whatever is to the left of the <code>~</code> is the dependant variable, and whatever is to the right is the independent variable. Did the Diet 1 and 2 produce significantly fatter birds?</p>
<p>One could also supplement the output by producing a graph (<a href="#fig-boxwhisker1" class="quarto-xref">Figure&nbsp;1</a>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="fu">library</span>(ggstatsplot)</span>
<span id="cb4-2"><a href="#cb4-2"></a></span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="do">## since the confidence intervals for the effect sizes are computed using</span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="do">## bootstrapping, important to set a seed for reproducibility</span></span>
<span id="cb4-5"><a href="#cb4-5"></a><span class="fu">set.seed</span>(<span class="dv">13</span>)</span>
<span id="cb4-6"><a href="#cb4-6"></a></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="do">## parametric t-test and box plot</span></span>
<span id="cb4-8"><a href="#cb4-8"></a><span class="fu">ggbetweenstats</span>(</span>
<span id="cb4-9"><a href="#cb4-9"></a>  <span class="at">data =</span> chicks_sub,</span>
<span id="cb4-10"><a href="#cb4-10"></a>  <span class="at">x =</span> Diet,</span>
<span id="cb4-11"><a href="#cb4-11"></a>  <span class="at">y =</span> weight,</span>
<span id="cb4-12"><a href="#cb4-12"></a>  <span class="at">xlab =</span> <span class="st">"Diet"</span>,</span>
<span id="cb4-13"><a href="#cb4-13"></a>  <span class="at">ylab =</span> <span class="st">"Chick mass (g)"</span>,</span>
<span id="cb4-14"><a href="#cb4-14"></a>  <span class="at">plot.type =</span> <span class="st">"box"</span>,</span>
<span id="cb4-15"><a href="#cb4-15"></a>  <span class="at">p.adjust.method =</span> <span class="st">"bonferroni"</span>,</span>
<span id="cb4-16"><a href="#cb4-16"></a>  <span class="at">pairwise.display =</span> <span class="st">"ns"</span>,</span>
<span id="cb4-17"><a href="#cb4-17"></a>  <span class="at">type =</span> <span class="st">"p"</span>,</span>
<span id="cb4-18"><a href="#cb4-18"></a>  <span class="at">results.subtitle =</span> <span class="cn">FALSE</span>,</span>
<span id="cb4-19"><a href="#cb4-19"></a>  <span class="at">conf.level =</span> <span class="fl">0.95</span>,</span>
<span id="cb4-20"><a href="#cb4-20"></a>  <span class="at">title =</span> <span class="st">"t-test"</span>,</span>
<span id="cb4-21"><a href="#cb4-21"></a>  <span class="at">ggtheme =</span> ggthemes<span class="sc">::</span><span class="fu">theme_fivethirtyeight</span>(),</span>
<span id="cb4-22"><a href="#cb4-22"></a>  <span class="at">package =</span> <span class="st">"basetheme"</span>,</span>
<span id="cb4-23"><a href="#cb4-23"></a>  <span class="at">palette =</span> <span class="st">"ink"</span></span>
<span id="cb4-24"><a href="#cb4-24"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-boxwhisker1" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-boxwhisker1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="08-anova_files/figure-html/fig-boxwhisker1-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;1: Box-and-whisker plot showing the differences in means between chicks reared to 21 days old and fed Diets 1 and 2"><img src="08-anova_files/figure-html/fig-boxwhisker1-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-boxwhisker1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Box-and-whisker plot showing the differences in means between chicks reared to 21 days old and fed Diets 1 and 2
</figcaption></figure>
</div>
</div>
</div>
<p>Notice above that we did not need to specify to use a <em>t</em>-test. The <code>ggbetweenstats()</code> function automatically determines if an independent samples <em>t</em>-test or a 1-way ANOVA is required based on whether there are two groups or three or more groups within the grouping (factor) variable.</p>
<p>That was a nice revision. But applied to the <code>chicks</code> data it seemed a bit silly, because you may ask, “What if I wanted to know if there are differences among the means computed at Day 1, Day 6, Day 10, and Day 21?” We should not use <em>t</em>-tests to do this (although we can). So now we can move on to the ANOVA.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Task G.1: Do it now!
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Why should we not just apply <em>t</em>-tests once per each of the pairs of comparisons we want to make?</li>
</ul>
</div>
</div>
</section><section id="why-not-do-multiple-t-tests" class="level2 page-columns page-full" data-number="2.2"><h2 data-number="2.2" class="anchored" data-anchor-id="why-not-do-multiple-t-tests">
<span class="header-section-number">2.2</span> Why not do multiple <em>t</em>-tests?</h2>
<p>In the <code>chicks</code> data we have four diets, not only two as in the <em>t</em>-test example just performed. Why not then simply do a <em>t</em>-test multiple times, once for each pair of diets given to the chickens? Multiple <em>t</em>-tests would be written as:</p>
<ul>
<li><span class="math inline">\(H_{0}: \mu_1 = \mu_2\)</span></li>
<li><span class="math inline">\(H_{0}: \mu_1 = \mu_3\)</span></li>
<li><span class="math inline">\(H_{0}: \mu_1 = \mu_4\)</span></li>
<li><span class="math inline">\(H_{0}: \mu_2 = \mu_3\)</span></li>
<li><span class="math inline">\(H_{0}: \mu_2 = \mu_4\)</span></li>
<li><span class="math inline">\(H_{0}: \mu_3 = \mu_4\)</span></li>
</ul>
<p>This would be invalid. The problem is that the chance of committing a Type I error increases as more multiple comparisons are done. So, the overall chance of rejecting the <em>H</em><sub>0</sub> increases. Why? If one sets <span class="math inline">\(\alpha=0.05\)</span> (the significance level below which the <em>H</em><sub>0</sub> is no longer accepted), one will still reject the <em>H</em><sub>0</sub> 5% of the time when it is in fact true (i.e.&nbsp;when there is no difference between the groups). When many pairwise comparisons are made, the probability of rejecting the <em>H</em><sub>0</sub> at least once is higher because we take this 5% risk each time we repeat a <em>t</em>-test. In the case of the chicken diets, we would have to perform six <em>t</em>-tests, and the error rate would increase to slightly less than <span class="math inline">\(6\times5\%\)</span>. See Table 1.</p>
<div id="kable-table-div" class="page-columns page-full">
<div class="cell page-columns page-full" data-tbl-cap="**Table 1.** Probability of committing a Type I error due to applying multiple *t*-tests to test for differences between *K* means. *α* from `0.2` to `0.0001` are shown.">
<div class="cell-output-display page-columns page-full">
<table class="lightable-classic table table-sm table-striped small" data-quarto-postprocess="true">

<thead><tr class="header">
<th style="text-align: right;" data-quarto-table-cell-role="th">K</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">0.2</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">0.1</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">0.05</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">0.02</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">0.01</th>
<th style="text-align: right;" data-quarto-table-cell-role="th">0.001</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">2</td>
<td style="text-align: right;">0.20</td>
<td style="text-align: right;">0.10</td>
<td style="text-align: right;">0.05</td>
<td style="text-align: right;">0.02</td>
<td style="text-align: right;">0.01</td>
<td style="text-align: right;">0.00</td>
</tr>
<tr class="even">
<td style="text-align: right;">3</td>
<td style="text-align: right;">0.49</td>
<td style="text-align: right;">0.27</td>
<td style="text-align: right;">0.14</td>
<td style="text-align: right;">0.06</td>
<td style="text-align: right;">0.03</td>
<td style="text-align: right;">0.00</td>
</tr>
<tr class="odd">
<td style="text-align: right;">4</td>
<td style="text-align: right;">0.74</td>
<td style="text-align: right;">0.47</td>
<td style="text-align: right;">0.26</td>
<td style="text-align: right;">0.11</td>
<td style="text-align: right;">0.06</td>
<td style="text-align: right;">0.01</td>
</tr>
<tr class="even">
<td style="text-align: right;">5</td>
<td style="text-align: right;">0.89</td>
<td style="text-align: right;">0.65</td>
<td style="text-align: right;">0.40</td>
<td style="text-align: right;">0.18</td>
<td style="text-align: right;">0.10</td>
<td style="text-align: right;">0.01</td>
</tr>
<tr class="odd">
<td style="text-align: right;">10</td>
<td style="text-align: right;">1.00</td>
<td style="text-align: right;">0.99</td>
<td style="text-align: right;">0.90</td>
<td style="text-align: right;">0.60</td>
<td style="text-align: right;">0.36</td>
<td style="text-align: right;">0.04</td>
</tr>
<tr class="even">
<td style="text-align: right;">20</td>
<td style="text-align: right;">1.00</td>
<td style="text-align: right;">1.00</td>
<td style="text-align: right;">1.00</td>
<td style="text-align: right;">0.98</td>
<td style="text-align: right;">0.85</td>
<td style="text-align: right;">0.17</td>
</tr>
<tr class="odd">
<td style="text-align: right;">100</td>
<td style="text-align: right;">1.00</td>
<td style="text-align: right;">1.00</td>
<td style="text-align: right;">1.00</td>
<td style="text-align: right;">1.00</td>
<td style="text-align: right;">1.00</td>
<td style="text-align: right;">0.99</td>
</tr>
</tbody>
</table>
<div class="quarto-table-caption margin-caption">
<strong>Table 1.</strong> Probability of committing a Type I error due to applying multiple <em>t</em>-tests to test for differences between <em>K</em> means. <em>α</em> from <code>0.2</code> to <code>0.0001</code> are shown.</div></div>
</div>
</div>
<p>If you insist in creating more work for yourself and do <em>t</em>-tests many times, one way to overcome the problem of committing Type I errors that stem from multiple comparisons is to apply a Bonferroni correction.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Bonferonni correction
</div>
</div>
<div class="callout-body-container callout-body">
<p>The Bonferroni correction is used to adjust the significance level of multiple hypothesis tests, such as multiple paired <em>t</em>-tests among many groups, in order to reduce the risk of false positives or Type I errors. It is named after the Italian mathematician Carlo Emilio Bonferroni.</p>
<p>The Bonferroni correction is based on the principle that when multiple hypothesis tests are performed, the probability of observing at least one significant result due to random chance increases. To correct for this, the significance level (usually 0.05) is divided by the number of tests being performed. This results in a more stringent significance level for each individual test and it so reduces the risk of committing a Type I error.</p>
<p>For example, if we conduct ten hypothesis tests, the significance level for each test after Bonferonni correction would become 0.05/10 = 0.005. The implication is that each individual test would need to have a <em>p</em>-value less than 0.005 to be considered significant at the overall significance level of 0.05.</p>
<p>On the downside, this method can be overly conservative and we may then increase the risk of Type II errors, which are false negatives. If you really cannot avoid multiple tests, then also assess one of the alternatives to Bonferonni’s method, viz: the false discovery rate (FDR) correction, the Holm-Bonferroni correction, Benjamini-Hochberg’s procedure, the Sidak correction, or some of the Bayesian approaches.</p>
</div>
</div>
<p>Or better still, we do an ANOVA that controls for these Type I errors so that it remains at 5%.</p>
</section></section><section id="anova" class="level1" data-number="3"><h1 data-number="3">
<span class="header-section-number">3</span> ANOVA</h1>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Hypothesis for an ANOVA
</div>
</div>
<div class="callout-body-container callout-body">
<p>If we have four groups whose means we want to compare, a suitable <em>H</em><sub>0</sub> would be:</p>
<p><span class="math inline">\(H_{0}: \mu_1 = \mu_2 = \mu_3 = \mu_4\)</span></p>
<p>and</p>
<p><span class="math inline">\(H_{a}\)</span>: the means of the four groups are not equal.</p>
<p>Here, <span class="math inline">\(\mu_1\)</span>, <span class="math inline">\(\mu_2\)</span>, <span class="math inline">\(\mu_3\)</span>, and <span class="math inline">\(\mu_4\)</span> are four population means. For the <em>H</em><sub>0</sub> to be rejected, all that is required is for one of the pairs of means to be different, not all of them.</p>
</div>
</div>
<section id="single-factor" class="level2" data-number="3.1"><h2 data-number="3.1" class="anchored" data-anchor-id="single-factor">
<span class="header-section-number">3.1</span> Single factor</h2>
<p>We continue with the chicken data. The <em>t</em>-test showed that Diets 1 and 2 resulted in the same chicken mass at Day 21. What about the other two diets? Our <em>H</em><sub>0</sub> is that, at Day 21, <span class="math inline">\(\mu_{1}=\mu_{2}=\mu_{3}=\mu_{4}\)</span>. Is there a statistical difference between chickens fed these four diets, or do we retain the <em>H</em><sub>0</sub>? The R function for an ANOVA is <code><a href="https://rdrr.io/r/stats/aov.html">aov()</a></code>. To look for significant differences between all four diets on the last day of sampling we use this one line of code:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>chicks.aov1 <span class="ot">&lt;-</span> <span class="fu">aov</span>(weight <span class="sc">~</span> Diet, <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">21</span>))</span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="fu">summary</span>(chicks.aov1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            Df Sum Sq Mean Sq F value  Pr(&gt;F)   
Diet         3  57164   19055   4.655 0.00686 **
Residuals   41 167839    4094                   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Task G.2: Do it now!
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="a">
<li>What does the outcome say about the chicken masses? Which ones are different from each other?</li>
<li>Devise a graphical display of this outcome.</li>
</ol>
</div>
</div>
<p>If this seems too easy to be true, it’s because we aren’t quite done yet. You could use your graphical display to eyeball where the significant differences are, or we can turn to a more ‘precise’ approach. The next step one could take is to run a Tukey HSD test on the results of the ANOVA by wrapping <code>tukeyHSD()</code> around <code><a href="https://rdrr.io/r/stats/aov.html">aov()</a></code>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="fu">TukeyHSD</span>(chicks.aov1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = weight ~ Diet, data = filter(chicks, Time == 21))

$Diet
         diff        lwr       upr     p adj
2-1  36.95000  -32.11064 106.01064 0.4868095
3-1  92.55000   23.48936 161.61064 0.0046959
4-1  60.80556  -10.57710 132.18821 0.1192661
3-2  55.60000  -21.01591 132.21591 0.2263918
4-2  23.85556  -54.85981 102.57092 0.8486781
4-3 -31.74444 -110.45981  46.97092 0.7036249</code></pre>
</div>
</div>
<p>The output of <code>tukeyHSD()</code> shows us that pairwise comparisons of all of the groups we are comparing. We can also display this as a very rough figure (<a href="#fig-tukeydiff" class="quarto-xref">Figure&nbsp;2</a>):</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="fu">plot</span>(<span class="fu">TukeyHSD</span>(chicks.aov1))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-tukeydiff" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-tukeydiff-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="08-anova_files/figure-html/fig-tukeydiff-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;2: A plot of the Tukey-HSD test showing the differences in means between chicks reared to 21 days old and fed four diets."><img src="08-anova_files/figure-html/fig-tukeydiff-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-tukeydiff-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: A plot of the Tukey-HSD test showing the differences in means between chicks reared to 21 days old and fed four diets.
</figcaption></figure>
</div>
</div>
</div>
<p>We may also produce a nicer looking graphical summary in the form of a box-and-whisker plot and/or a violin plot. Here I combine both (<a href="#fig-boxwhisker2" class="quarto-xref">Figure&nbsp;3</a>):</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="fu">set.seed</span>(<span class="dv">666</span>)</span>
<span id="cb10-2"><a href="#cb10-2"></a></span>
<span id="cb10-3"><a href="#cb10-3"></a><span class="do">## parametric t-test and box plot</span></span>
<span id="cb10-4"><a href="#cb10-4"></a><span class="fu">ggbetweenstats</span>(</span>
<span id="cb10-5"><a href="#cb10-5"></a>  <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">21</span>),</span>
<span id="cb10-6"><a href="#cb10-6"></a>  <span class="at">x =</span> Diet,</span>
<span id="cb10-7"><a href="#cb10-7"></a>  <span class="at">y =</span> weight,</span>
<span id="cb10-8"><a href="#cb10-8"></a>  <span class="at">xlab =</span> <span class="st">"Diet"</span>,</span>
<span id="cb10-9"><a href="#cb10-9"></a>  <span class="at">ylab =</span> <span class="st">"Chick mass (g)"</span>,</span>
<span id="cb10-10"><a href="#cb10-10"></a>  <span class="at">plot.type =</span> <span class="st">"box"</span>,</span>
<span id="cb10-11"><a href="#cb10-11"></a>  <span class="at">boxplot.args =</span> <span class="fu">list</span>(<span class="at">notch =</span> <span class="cn">TRUE</span>),</span>
<span id="cb10-12"><a href="#cb10-12"></a>  <span class="at">type =</span> <span class="st">"parametric"</span>,</span>
<span id="cb10-13"><a href="#cb10-13"></a>  <span class="at">results.subtitle =</span> <span class="cn">FALSE</span>,</span>
<span id="cb10-14"><a href="#cb10-14"></a>  <span class="at">pairwise.comparisons =</span> <span class="cn">TRUE</span>,</span>
<span id="cb10-15"><a href="#cb10-15"></a>  <span class="at">pairwise.display =</span> <span class="st">"s"</span>,</span>
<span id="cb10-16"><a href="#cb10-16"></a>  <span class="at">p.adjust.method =</span> <span class="st">"bonferroni"</span>,</span>
<span id="cb10-17"><a href="#cb10-17"></a>  <span class="at">conf.level =</span> <span class="fl">0.95</span>,</span>
<span id="cb10-18"><a href="#cb10-18"></a>  <span class="at">title =</span> <span class="st">"ANOVA"</span>,</span>
<span id="cb10-19"><a href="#cb10-19"></a>  <span class="at">ggtheme =</span> ggthemes<span class="sc">::</span><span class="fu">theme_fivethirtyeight</span>(),</span>
<span id="cb10-20"><a href="#cb10-20"></a>  <span class="at">package =</span> <span class="st">"basetheme"</span>,</span>
<span id="cb10-21"><a href="#cb10-21"></a>  <span class="at">palette =</span> <span class="st">"ink"</span></span>
<span id="cb10-22"><a href="#cb10-22"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-boxwhisker2" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-boxwhisker2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="08-anova_files/figure-html/fig-boxwhisker2-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Figure&nbsp;3: Box-and-whisker plot showing the differences in means between chicks reared to 21 days old and fed four diets. Shown is a notched box plot where the extent of the notches is 1.58 * IQR / sqrt(n). This is approximately equivalent to a 95% confidence interval andf may be used for comparing medians."><img src="08-anova_files/figure-html/fig-boxwhisker2-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-boxwhisker2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Box-and-whisker plot showing the differences in means between chicks reared to 21 days old and fed four diets. Shown is a notched box plot where the extent of the notches is <code>1.58 * IQR / sqrt(n)</code>. This is approximately equivalent to a 95% confidence interval andf may be used for comparing medians.
</figcaption></figure>
</div>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Task G.3: Do it now!
</div>
</div>
<div class="callout-body-container callout-body">
<p>Look at the help file for the <code><a href="https://rdrr.io/r/stats/TukeyHSD.html">TukeyHSD()</a></code> function to better understand what the output means.</p>
<ol type="a">
<li>How does one interpret the results? What does this tell us about the effect that that different diets has on the chicken weights at Day 21?</li>
<li>Figure out a way to plot the Tukey HSD outcomes in <strong>ggplot</strong>.</li>
<li>Why does the ANOVA return a significant result, but the Tukey test shows that not all of the groups are significantly different from one another?</li>
</ol>
</div>
</div>
<!-- ```{r} -->
<!-- # plot(TukeyHSD(chicks.aov)) -->
<!-- ``` -->
</section><section id="multiple-factors" class="level2" data-number="3.2"><h2 data-number="3.2" class="anchored" data-anchor-id="multiple-factors">
<span class="header-section-number">3.2</span> Multiple factors</h2>
<p>What if we have multiple grouping variables, and not just one? We would encounter this kind of situation in factorial designs. In the case of the chicken data, there is also time that seems to be having an effect.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Task G.4: Do it now!
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="a">
<li>How is time having an effect? <strong>(/3)</strong>
</li>
<li>What hypotheses can we construct around time? <strong>(/2)</strong>
</li>
</ol>
</div>
</div>
<p>Let us look at some variations around questions concerning time. We might ask, at a particular time step, are there differences amongst the effect due to diet on chicken mass? Let’s see when diets are starting to have an effect by examining the outcomes at times 0, 2, 10, and 21:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="co"># effect at time = 0</span></span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="fu">summary</span>(<span class="fu">aov</span>(weight <span class="sc">~</span> Diet, <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">0</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            Df Sum Sq Mean Sq F value Pr(&gt;F)
Diet         3   4.32   1.440   1.132  0.346
Residuals   46  58.50   1.272               </code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co"># effect at time = 2</span></span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="fu">summary</span>(<span class="fu">aov</span>(weight <span class="sc">~</span> Diet, <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">2</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            Df Sum Sq Mean Sq F value  Pr(&gt;F)   
Diet         3  158.4   52.81   4.781 0.00555 **
Residuals   46  508.1   11.05                   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="co"># effect at time = 10</span></span>
<span id="cb15-2"><a href="#cb15-2"></a><span class="fu">summary</span>(<span class="fu">aov</span>(weight <span class="sc">~</span> Diet, <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">10</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Diet         3   8314    2771    6.46 0.000989 ***
Residuals   45  19304     429                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="co"># effect at time = 21</span></span>
<span id="cb17-2"><a href="#cb17-2"></a><span class="fu">summary</span>(<span class="fu">aov</span>(weight <span class="sc">~</span> Diet, <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">21</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            Df Sum Sq Mean Sq F value  Pr(&gt;F)   
Diet         3  57164   19055   4.655 0.00686 **
Residuals   41 167839    4094                   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Task G.5: Do it now!
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="a">
<li>What do you conclude from the above series of ANOVAs? <strong>(/3)</strong>
</li>
<li>What problem is associated with running multiple tests in the way that we have done here? <strong>(/2)</strong>
</li>
</ol>
</div>
</div>
<p>Or we may ask, regardless of diet (i.e.&nbsp;disregarding the effect of diet by clumping all chickens together), is time having an effect?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>chicks.aov2 <span class="ot">&lt;-</span> <span class="fu">aov</span>(weight <span class="sc">~</span> <span class="fu">as.factor</span>(Time),</span>
<span id="cb19-2"><a href="#cb19-2"></a>                   <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">10</span>, <span class="dv">21</span>)))</span>
<span id="cb19-3"><a href="#cb19-3"></a><span class="fu">summary</span>(chicks.aov2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                 Df Sum Sq Mean Sq F value Pr(&gt;F)    
as.factor(Time)   3 939259  313086   234.8 &lt;2e-16 ***
Residuals       190 253352    1333                   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Task G.6: Do it now!
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="a">
<li>Write out the hypotheses for this ANOVA. <strong>(/2)</strong>
</li>
<li>What do you conclude from the above ANOVA. <strong>(/3)</strong>
</li>
</ol>
</div>
</div>
<p>Or, to save ourselves a lot of time and reduce the coding effort, we may simply run a two-way ANOVA and look at the effects of <code>Diet</code> and <code>Time</code> simultaneously. To specify the different factors we put them in our formula and separate them with a <code>+</code>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="fu">summary</span>(<span class="fu">aov</span>(weight <span class="sc">~</span> Diet <span class="sc">+</span> <span class="fu">as.factor</span>(Time),</span>
<span id="cb21-2"><a href="#cb21-2"></a>            <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">21</span>))))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                Df Sum Sq Mean Sq F value  Pr(&gt;F)    
Diet             3  39595   13198   5.987 0.00091 ***
as.factor(Time)  1 734353  734353 333.120 &lt; 2e-16 ***
Residuals       90 198402    2204                    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Task G.7: Do it now!
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="a">
<li>What question are we asking with the above line of code? <strong>(/3)</strong>
</li>
<li>What is the answer? <strong>(/2)</strong>
</li>
<li>Why did we wrap <code>Time</code> in <code><a href="https://rdrr.io/r/base/factor.html">as.factor()</a></code>? <strong>(/2)</strong>
</li>
</ol>
</div>
</div>
<p>It is also possible to look at what the interaction effect between grouping variables (i.e.&nbsp;in this case the effect of time on diet—does the effect of time depend on which diet we are looking at?), and not just within the individual grouping variables. To do this we replace the <code>+</code> in our formula with <code>*</code>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="fu">summary</span>(<span class="fu">aov</span>(weight <span class="sc">~</span> Diet <span class="sc">*</span> <span class="fu">as.factor</span>(Time),</span>
<span id="cb23-2"><a href="#cb23-2"></a>            <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">21</span>))))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                     Df Sum Sq Mean Sq F value   Pr(&gt;F)    
Diet                  3  40914   13638   6.968 0.000298 ***
as.factor(Time)       1 582221  582221 297.472  &lt; 2e-16 ***
Diet:as.factor(Time)  3  25530    8510   4.348 0.006684 ** 
Residuals            86 168322    1957                     
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Task G.8: Do it now!
</div>
</div>
<div class="callout-body-container callout-body">
<p>How do these results differ from the previous set? <strong>(/3)</strong></p>
</div>
</div>
<p>One may also run a post-hoc Tukey test on these results the same as for a single factor ANOVA:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a><span class="fu">TukeyHSD</span>(<span class="fu">aov</span>(weight <span class="sc">~</span> Diet <span class="sc">*</span> <span class="fu">as.factor</span>(Time),</span>
<span id="cb25-2"><a href="#cb25-2"></a>             <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">20</span>, <span class="dv">21</span>))))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = weight ~ Diet * as.factor(Time), data = filter(chicks, Time %in% c(20, 21)))

$Diet
         diff        lwr       upr     p adj
2-1  36.18030  -9.301330  81.66194 0.1663037
3-1  90.63030  45.148670 136.11194 0.0000075
4-1  62.25253  15.223937 109.28111 0.0045092
3-2  54.45000   3.696023 105.20398 0.0305957
4-2  26.07222 -26.072532  78.21698 0.5586643
4-3 -28.37778 -80.522532  23.76698 0.4863940

$`as.factor(Time)`
          diff       lwr      upr     p adj
21-20 8.088223 -17.44017 33.61661 0.5303164

$`Diet:as.factor(Time)`
                diff        lwr        upr     p adj
2:20-1:20  35.188235  -40.67378 111.050253 0.8347209
3:20-1:20  88.488235   12.62622 164.350253 0.0111136
4:20-1:20  63.477124  -14.99365 141.947897 0.2035951
1:21-1:20   7.338235  -58.96573  73.642198 0.9999703
2:21-1:20  44.288235  -31.57378 120.150253 0.6116081
3:21-1:20  99.888235   24.02622 175.750253 0.0023872
4:21-1:20  68.143791  -10.32698 146.614563 0.1371181
3:20-2:20  53.300000  -31.82987 138.429869 0.5234263
4:20-2:20  28.288889  -59.17374 115.751515 0.9723470
1:21-2:20 -27.850000 -104.58503  48.885027 0.9486212
2:21-2:20   9.100000  -76.02987  94.229869 0.9999766
3:21-2:20  64.700000  -20.42987 149.829869 0.2732059
4:21-2:20  32.955556  -54.50707 120.418182 0.9377007
4:20-3:20 -25.011111 -112.47374  62.451515 0.9862822
1:21-3:20 -81.150000 -157.88503  -4.414973 0.0305283
2:21-3:20 -44.200000 -129.32987  40.929869 0.7402877
3:21-3:20  11.400000  -73.72987  96.529869 0.9998919
4:21-3:20 -20.344444 -107.80707  67.118182 0.9960548
1:21-4:20 -56.138889 -135.45396  23.176184 0.3619622
2:21-4:20 -19.188889 -106.65152  68.273738 0.9972631
3:21-4:20  36.411111  -51.05152 123.873738 0.8984019
4:21-4:20   4.666667  -85.06809  94.401428 0.9999998
2:21-1:21  36.950000  -39.78503 113.685027 0.8067041
3:21-1:21  92.550000   15.81497 169.285027 0.0075185
4:21-1:21  60.805556  -18.50952 140.120628 0.2629945
3:21-2:21  55.600000  -29.52987 140.729869 0.4679025
4:21-2:21  23.855556  -63.60707 111.318182 0.9896157
4:21-3:21 -31.744444 -119.20707  55.718182 0.9486128</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Task G.9: Do it now!
</div>
</div>
<div class="callout-body-container callout-body">
<p>Yikes! That’s a massive amount of results. What does all of this mean, and why is it so verbose? <strong>(/5)</strong></p>
</div>
</div>
<!--- #### About interaction terms --->
<!-- AJS to insert stuff here -->
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Summary
</div>
</div>
<div class="callout-body-container callout-body">
<p>To summarise <em>t</em>-tests, single-factor (1-way) and multifactor (2- or 3-way, etc.) ANOVAs:</p>
<ol type="1">
<li><p>A <em>t</em>-test is applied to situations where one wants to compare the means of only <strong>two</strong> groups of a response variable within <strong>one categorical independent variable</strong> (we say a factor with two levels).</p></li>
<li><p>A 1-way ANOVA also looks at the means of a response variable belonging to <strong>one categorical independent variable</strong>, but the categorical response variable has <strong>more than two</strong> levels in it.</p></li>
<li><p>Following on from there, a 2-way ANOVA compares the means of response variables belonging to all the levels within <strong>two categorical independent variables</strong> (e.g.&nbsp;Factor 1 might have three levels, and Factor 2 five levels). In the simplest formulaton, it does so by looking at the <strong>main effects</strong>, which is the group differences between the three levels of Factor 1 and disregarding the contribution due to the group membership to Factor 2, and also the group differences amongst the levels of Factor 2 but disregarding the group membership of Factor 1. In addition to looking at the main effects, a 2-way ANOVA can also consider the <strong>interaction</strong> (or combined effect) of Factors 1 and 2 in influencing the means.</p></li>
</ol>
</div>
</div>
</section></section><section id="sec-alt" class="level1" data-number="4"><h1 data-number="4">
<span class="header-section-number">4</span> Alternatives to ANOVA</h1>
<p>In the first main section of this chapter we learned how to test hypotheses based on the comparisons of means between sets of data when we were able to meet our two base assumptions. These parametric tests are preferred over non-parametric tests because they are more robust. However, when we simply aren’t able to meet these assumptions we must not despair. Non-parametric tests are still useful. In this chapter we will learn how to run non-parametric tests for two sample and multiple sample datasets. To start, let’s load our libraries and <code>chicks</code> data if we have not already.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a><span class="co"># First activate libraries</span></span>
<span id="cb27-2"><a href="#cb27-2"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb27-3"><a href="#cb27-3"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb27-4"><a href="#cb27-4"></a></span>
<span id="cb27-5"><a href="#cb27-5"></a><span class="co"># Then load data</span></span>
<span id="cb27-6"><a href="#cb27-6"></a>chicks <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(ChickWeight)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>With our libraries and data loaded, let’s find a day in which at least one of our assumptions are violated.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a><span class="co"># Then check for failing assumptions</span></span>
<span id="cb28-2"><a href="#cb28-2"></a>chicks <span class="sc">%&gt;%</span> </span>
<span id="cb28-3"><a href="#cb28-3"></a>  <span class="fu">filter</span>(Time <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-4"><a href="#cb28-4"></a>  <span class="fu">group_by</span>(Diet) <span class="sc">%&gt;%</span> </span>
<span id="cb28-5"><a href="#cb28-5"></a>  <span class="fu">summarise</span>(<span class="at">norm_wt =</span> <span class="fu">as.numeric</span>(<span class="fu">shapiro.test</span>(weight)[<span class="dv">2</span>]),</span>
<span id="cb28-6"><a href="#cb28-6"></a>            <span class="at">var_wt =</span> <span class="fu">var</span>(weight))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4 × 3
  Diet  norm_wt var_wt
  &lt;fct&gt;   &lt;dbl&gt;  &lt;dbl&gt;
1 1     0.0138   0.989
2 2     0.138    2.23 
3 3     0.00527  1.07 
4 4     0.0739   1.11 </code></pre>
</div>
</div>
<section id="wilcoxon-rank-sum-test" class="level2" data-number="4.1"><h2 data-number="4.1" class="anchored" data-anchor-id="wilcoxon-rank-sum-test">
<span class="header-section-number">4.1</span> Wilcoxon rank sum test</h2>
<p>The non-parametric version of a <em>t</em>-test is a Wilcox rank sum test. To perform this test in R we may again use <code>compare_means()</code> and specify the test we want:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a><span class="fu">compare_means</span>(weight <span class="sc">~</span> Diet,</span>
<span id="cb30-2"><a href="#cb30-2"></a>              <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">0</span>,</span>
<span id="cb30-3"><a href="#cb30-3"></a>                            Diet <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)),</span>
<span id="cb30-4"><a href="#cb30-4"></a>              <span class="at">method =</span> <span class="st">"wilcox.test"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 8
  .y.    group1 group2     p p.adj p.format p.signif method  
  &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;   
1 weight 1      2      0.235  0.23 0.23     ns       Wilcoxon</code></pre>
</div>
</div>
<p>What do our results show?</p>
</section><section id="kruskall-wallis-rank-sum-test" class="level2" data-number="4.2"><h2 data-number="4.2" class="anchored" data-anchor-id="kruskall-wallis-rank-sum-test">
<span class="header-section-number">4.2</span> Kruskall-Wallis rank sum test</h2>
<section id="single-factor-1" class="level3" data-number="4.2.1"><h3 data-number="4.2.1" class="anchored" data-anchor-id="single-factor-1">
<span class="header-section-number">4.2.1</span> Single factor</h3>
<p>The non-parametric version of an ANOVA is a Kruskall-Wallis rank sum test. As you may have by now surmised, this may be done with <code>compare_means()</code> as seen below:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a><span class="fu">compare_means</span>(weight <span class="sc">~</span> Diet,</span>
<span id="cb32-2"><a href="#cb32-2"></a>              <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">0</span>),</span>
<span id="cb32-3"><a href="#cb32-3"></a>              <span class="at">method =</span> <span class="st">"kruskal.test"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 1 × 6
  .y.        p p.adj p.format p.signif method        
  &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;    &lt;chr&gt;         
1 weight 0.475  0.48 0.48     ns       Kruskal-Wallis</code></pre>
</div>
</div>
<p>As with the ANOVA, this first step with the Kruskall-Wallis test is not the last. We must again run a post-hoc test on our results. This time we will need to use <code><a href="https://rdrr.io/pkg/pgirmess/man/kruskalmc.html">pgirmess::kruskalmc()</a></code>, which means we will need to load a new library.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a><span class="fu">library</span>(pgirmess)</span>
<span id="cb34-2"><a href="#cb34-2"></a></span>
<span id="cb34-3"><a href="#cb34-3"></a><span class="fu">kruskalmc</span>(weight <span class="sc">~</span> Diet, <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">0</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Multiple comparison test after Kruskal-Wallis 
alpha: 0.05 
Comparisons
    obs.dif critical.dif stat.signif
1-2    6.95     14.89506       FALSE
1-3    6.90     14.89506       FALSE
1-4    4.15     14.89506       FALSE
2-3    0.05     17.19933       FALSE
2-4    2.80     17.19933       FALSE
3-4    2.75     17.19933       FALSE</code></pre>
</div>
</div>
<p>Let’s consult the help file for <code>kruskalmc()</code> to understand what this print-out means.</p>
</section><section id="multiple-factors-1" class="level3" data-number="4.2.2"><h3 data-number="4.2.2" class="anchored" data-anchor-id="multiple-factors-1">
<span class="header-section-number">4.2.2</span> Multiple factors</h3>
<p>The water becomes murky quickly when one wants to perform multiple factor non-parametric comparison of means tests. To that end, we will not cover the few existing methods here. Rather, one should avoid the necessity for these types of tests when designing an experiment.</p>
</section></section><section id="the-sa-time-data" class="level2" data-number="4.3"><h2 data-number="4.3" class="anchored" data-anchor-id="the-sa-time-data">
<span class="header-section-number">4.3</span> The SA time data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a>sa_time <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(<span class="fu">read_csv</span>(<span class="st">"../../data/snakes.csv"</span>,</span>
<span id="cb36-2"><a href="#cb36-2"></a>                              <span class="at">col_types =</span> <span class="fu">list</span>(<span class="fu">col_double</span>(),</span>
<span id="cb36-3"><a href="#cb36-3"></a>                                               <span class="fu">col_double</span>(),</span>
<span id="cb36-4"><a href="#cb36-4"></a>                                               <span class="fu">col_double</span>())))</span>
<span id="cb36-5"><a href="#cb36-5"></a>sa_time_long <span class="ot">&lt;-</span> sa_time <span class="sc">%&gt;%</span> </span>
<span id="cb36-6"><a href="#cb36-6"></a>  <span class="fu">gather</span>(<span class="at">key =</span> <span class="st">"term"</span>, <span class="at">value =</span> <span class="st">"minutes"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-7"><a href="#cb36-7"></a>  <span class="fu">filter</span>(minutes <span class="sc">&lt;</span> <span class="dv">300</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-8"><a href="#cb36-8"></a>  <span class="fu">mutate</span>(<span class="at">term =</span> <span class="fu">as.factor</span>(term))</span>
<span id="cb36-9"><a href="#cb36-9"></a></span>
<span id="cb36-10"><a href="#cb36-10"></a>my_comparisons <span class="ot">&lt;-</span> <span class="fu">list</span>( <span class="fu">c</span>(<span class="st">"now"</span>, <span class="st">"now_now"</span>),</span>
<span id="cb36-11"><a href="#cb36-11"></a>                        <span class="fu">c</span>(<span class="st">"now_now"</span>, <span class="st">"just_now"</span>),</span>
<span id="cb36-12"><a href="#cb36-12"></a>                        <span class="fu">c</span>(<span class="st">"now"</span>, <span class="st">"just_now"</span>) )</span>
<span id="cb36-13"><a href="#cb36-13"></a></span>
<span id="cb36-14"><a href="#cb36-14"></a><span class="fu">ggboxplot</span>(sa_time_long, <span class="at">x =</span> <span class="st">"term"</span>, <span class="at">y =</span> <span class="st">"minutes"</span>,</span>
<span id="cb36-15"><a href="#cb36-15"></a>          <span class="at">color =</span> <span class="st">"term"</span>, <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">"#00AFBB"</span>, <span class="st">"#E7B800"</span>, <span class="st">"#FC4E07"</span>),</span>
<span id="cb36-16"><a href="#cb36-16"></a>          <span class="at">add =</span> <span class="st">"jitter"</span>, <span class="at">shape =</span> <span class="st">"term"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-anova-plot6" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-anova-plot6-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="08-anova_files/figure-html/fig-anova-plot6-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Figure&nbsp;4: Time is not a limited resource in South Africa."><img src="08-anova_files/figure-html/fig-anova-plot6-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-anova-plot6-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Time is not a limited resource in South Africa.
</figcaption></figure>
</div>
</div>
</div>
</section></section><section id="examples" class="level1" data-number="5"><h1 data-number="5">
<span class="header-section-number">5</span> Examples</h1>
<section id="snakes" class="level2" data-number="5.1"><h2 data-number="5.1" class="anchored" data-anchor-id="snakes">
<span class="header-section-number">5.1</span> Snakes!</h2>
<p>These data could be analysed by a two-way ANOVA without replication, or a repeated measures ANOVA. Here I will analyse it by using a two-way ANOVA without replication.</p>
<p>Place and Abramson (2008) placed diamondback rattlesnakes (<em>Crotalus atrox</em>) in a ‘rattlebox,’ a box with a lid that would slide open and shut every 5 minutes. At first, the snake would rattle its tail each time the box opened. After a while, the snake would become habituated to the box opening and stop rattling its tail. They counted the number of box openings until a snake stopped rattling; fewer box openings means the snake was more quickly habituated. They repeated this experiment on each snake on four successive days, which is treated as an influential variable here. Place and Abramson (2008) used 10 snakes, but some of them never became habituated; to simplify this example, data from the six snakes that did become habituated on each day are used.</p>
<p>First, we read in the data, making sure to convert the column named <code>day</code> to a factor. Why? Because ANOVAs work with factor independent variables, while <code>day</code> as it is encoded by default is in fact a continuous variable.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>snakes <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../../data/snakes.csv"</span>)</span>
<span id="cb37-2"><a href="#cb37-2"></a>snakes<span class="sc">$</span>day <span class="ot">=</span> <span class="fu">as.factor</span>(snakes<span class="sc">$</span>day)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The first thing we do is to create some summaries of the data. Refer to the summary statistics Chapter.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1"></a>snakes.summary <span class="ot">&lt;-</span> snakes <span class="sc">%&gt;%</span> </span>
<span id="cb38-2"><a href="#cb38-2"></a>  <span class="fu">group_by</span>(day, snake) <span class="sc">%&gt;%</span> </span>
<span id="cb38-3"><a href="#cb38-3"></a>  <span class="fu">summarise</span>(<span class="at">mean_openings =</span> <span class="fu">mean</span>(openings),</span>
<span id="cb38-4"><a href="#cb38-4"></a>            <span class="at">sd_openings =</span> <span class="fu">sd</span>(openings)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-5"><a href="#cb38-5"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb38-6"><a href="#cb38-6"></a>snakes.summary</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 24 × 4
   day   snake mean_openings sd_openings
   &lt;fct&gt; &lt;chr&gt;         &lt;dbl&gt;       &lt;dbl&gt;
 1 1     D1               85          NA
 2 1     D11              40          NA
 3 1     D12              65          NA
 4 1     D3              107          NA
 5 1     D5               61          NA
 6 1     D8               22          NA
 7 2     D1               58          NA
 8 2     D11              45          NA
 9 2     D12              27          NA
10 2     D3               51          NA
# ℹ 14 more rows</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Task G.9: Do it now!
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Something seems… off. What’s going on here? Please explain this outcome.</li>
</ul>
</div>
</div>
<p>To fix this problem, let us ignore the grouping by both <code>snake</code> and <code>day</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1"></a>snakes.summary <span class="ot">&lt;-</span> snakes <span class="sc">%&gt;%</span> </span>
<span id="cb40-2"><a href="#cb40-2"></a>  <span class="fu">group_by</span>(day) <span class="sc">%&gt;%</span> </span>
<span id="cb40-3"><a href="#cb40-3"></a>  <span class="fu">summarise</span>(<span class="at">mean_openings =</span> <span class="fu">mean</span>(openings),</span>
<span id="cb40-4"><a href="#cb40-4"></a>            <span class="at">sd_openings =</span> <span class="fu">sd</span>(openings)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-5"><a href="#cb40-5"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb40-6"><a href="#cb40-6"></a>snakes.summary</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4 × 3
  day   mean_openings sd_openings
  &lt;fct&gt;         &lt;dbl&gt;       &lt;dbl&gt;
1 1              63.3        30.5
2 2              47          12.2
3 3              34.5        26.0
4 4              25.3        18.1</code></pre>
</div>
</div>
<p><code><a href="https://rdrr.io/pkg/Rmisc/man/summarySE.html">Rmisc::summarySE()</a></code> offers a convenience function if your feeling less frisky about calculating the summary statistics yourself:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1"></a><span class="fu">library</span>(Rmisc)</span>
<span id="cb42-2"><a href="#cb42-2"></a>snakes.summary2 <span class="ot">&lt;-</span> <span class="fu">summarySE</span>(<span class="at">data =</span> snakes,</span>
<span id="cb42-3"><a href="#cb42-3"></a>                             <span class="at">measurevar =</span> <span class="st">"openings"</span>,</span>
<span id="cb42-4"><a href="#cb42-4"></a>                             <span class="at">groupvars =</span> <span class="fu">c</span>(<span class="st">"day"</span>))</span>
<span id="cb42-5"><a href="#cb42-5"></a>snakes.summary2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  day N openings       sd        se       ci
1   1 6 63.33333 30.45434 12.432931 31.95987
2   2 6 47.00000 12.21475  4.986649 12.81859
3   3 6 34.50000 25.95958 10.597956 27.24291
4   4 6 25.33333 18.08498  7.383164 18.97903</code></pre>
</div>
</div>
<p>Now we turn to some visual data summaries (<a href="#fig-anova-plot" class="quarto-xref">Figure&nbsp;5</a>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> snakes, <span class="fu">aes</span>(<span class="at">x =</span> day, <span class="at">y =</span> openings)) <span class="sc">+</span></span>
<span id="cb44-2"><a href="#cb44-2"></a>  <span class="fu">geom_segment</span>(<span class="at">data =</span> snakes.summary2, <span class="fu">aes</span>(<span class="at">x =</span> day, <span class="at">xend =</span> day,</span>
<span id="cb44-3"><a href="#cb44-3"></a>                                           <span class="at">y =</span> openings <span class="sc">-</span> ci,</span>
<span id="cb44-4"><a href="#cb44-4"></a>                                           <span class="at">yend =</span> openings <span class="sc">+</span> ci,</span>
<span id="cb44-5"><a href="#cb44-5"></a>                                           <span class="at">colour =</span> day),</span>
<span id="cb44-6"><a href="#cb44-6"></a>              <span class="at">size =</span> <span class="fl">2.0</span>, <span class="at">linetype =</span> <span class="st">"solid"</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb44-7"><a href="#cb44-7"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> day), <span class="at">alpha =</span> <span class="fl">0.3</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span> </span>
<span id="cb44-8"><a href="#cb44-8"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb44-9"><a href="#cb44-9"></a>  <span class="fu">theme_pubclean</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-anova-plot" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-anova-plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="08-anova_files/figure-html/fig-anova-plot-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-6" title="Figure&nbsp;5: Boxplots showing the change in the snakes’ habituation to box opening over time."><img src="08-anova_files/figure-html/fig-anova-plot-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-anova-plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Boxplots showing the change in the snakes’ habituation to box opening over time.
</figcaption></figure>
</div>
</div>
</div>
<p>What are our null hypotheses?</p>
<ol type="1">
<li>
<strong><em>H</em><sub>0</sub></strong> There is no difference between snakes with respect to the number of openings at which they habituate.</li>
<li>
<strong><em>H</em><sub>0</sub></strong> There is no difference between days in terms of the number of openings at which the snakes habituate.</li>
</ol>
<p>Fit the ANOVA model to test these hypotheses:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1"></a>snakes.aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(openings <span class="sc">~</span> day <span class="sc">+</span> snake, <span class="at">data =</span> snakes)</span>
<span id="cb45-2"><a href="#cb45-2"></a><span class="fu">summary</span>(snakes.aov)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            Df Sum Sq Mean Sq F value Pr(&gt;F)  
day          3   4878  1625.9   3.320 0.0487 *
snake        5   3042   608.4   1.242 0.3382  
Residuals   15   7346   489.7                 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<p>Now we need to test of the assumptions hold true (i.e.&nbsp;erros are normally distributed and heteroscedastic) (<a href="#fig-anova-plot5" class="quarto-xref">Figure&nbsp;6</a>). Also, where are the differences (<a href="#fig-tukey" class="quarto-xref">Figure&nbsp;7</a>)?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb47-2"><a href="#cb47-2"></a><span class="co"># Checking assumptions...</span></span>
<span id="cb47-3"><a href="#cb47-3"></a><span class="co"># make a histogram of the residuals;</span></span>
<span id="cb47-4"><a href="#cb47-4"></a><span class="co"># they must be normal</span></span>
<span id="cb47-5"><a href="#cb47-5"></a>snakes.res <span class="ot">&lt;-</span> <span class="fu">residuals</span>(snakes.aov)</span>
<span id="cb47-6"><a href="#cb47-6"></a><span class="fu">hist</span>(snakes.res, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb47-7"><a href="#cb47-7"></a></span>
<span id="cb47-8"><a href="#cb47-8"></a><span class="co"># make a plot of residuals and the fitted values;</span></span>
<span id="cb47-9"><a href="#cb47-9"></a><span class="co"># # they must be normal and homoscedastic</span></span>
<span id="cb47-10"><a href="#cb47-10"></a><span class="fu">plot</span>(<span class="fu">fitted</span>(snakes.aov), <span class="fu">residuals</span>(snakes.aov), <span class="at">col =</span> <span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-anova-plot5" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-anova-plot5-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="08-anova_files/figure-html/fig-anova-plot5-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-7" title="Figure&nbsp;6: Exploring the assumptions visually."><img src="08-anova_files/figure-html/fig-anova-plot5-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-anova-plot5-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: Exploring the assumptions visually.
</figcaption></figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1"></a>snakes.tukey <span class="ot">&lt;-</span> <span class="fu">TukeyHSD</span>(snakes.aov, <span class="at">which =</span> <span class="st">"day"</span>, <span class="at">conf.level =</span> <span class="fl">0.90</span>)</span>
<span id="cb48-2"><a href="#cb48-2"></a><span class="fu">plot</span>(snakes.tukey, <span class="at">las =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="fig-tukey" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure"><div aria-describedby="fig-tukey-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="08-anova_files/figure-html/fig-tukey-1.svg" class="lightbox" data-gallery="quarto-lightbox-gallery-8" title="Figure&nbsp;7: Exploring the differences between days."><img src="08-anova_files/figure-html/fig-tukey-1.svg" class="img-fluid figure-img"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-tukey-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: Exploring the differences between days.
</figcaption></figure>
</div>
</div>
</div>


<!-- -->

</section></section><div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-reuse"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a></div></div></section><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{smit,_a._j.2021,
  author = {Smit, A. J.,},
  title = {8. {ANOVA}},
  date = {2021-01-01},
  url = {http://tangledbank.netlify.app/BCB744/basic_stats/08-anova.html},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-smit,_a._j.2021" class="csl-entry quarto-appendix-citeas" role="listitem">
Smit, A. J. (2021) 8. ANOVA. <a href="http://tangledbank.netlify.app/BCB744/basic_stats/08-anova.html">http://tangledbank.netlify.app/BCB744/basic_stats/08-anova.html.
</a>
</div></div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("http:\/\/tangledbank\.netlify\.app");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
            // default icon
            link.classList.add("external");
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="../../BCB744/basic_stats/07-t_tests.html" class="pagination-link" aria-label="7. *t*-Tests">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">7. <em>t</em>-Tests</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../BCB744/basic_stats/09-regressions.html" class="pagination-link" aria-label="9. Simple Linear Regressions">
        <span class="nav-page-text">9. Simple Linear Regressions</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb49" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb49-1"><a href="#cb49-1"></a><span class="co">---</span></span>
<span id="cb49-2"><a href="#cb49-2"></a><span class="an">date:</span><span class="co"> "2021-01-01"</span></span>
<span id="cb49-3"><a href="#cb49-3"></a><span class="an">title:</span><span class="co"> "8. ANOVA"</span></span>
<span id="cb49-4"><a href="#cb49-4"></a><span class="an">subtitle:</span><span class="co"> "Comparing the means of more than two groups"</span></span>
<span id="cb49-5"><a href="#cb49-5"></a><span class="an">reference-location:</span><span class="co"> margin</span></span>
<span id="cb49-6"><a href="#cb49-6"></a><span class="an">number-sections:</span><span class="co"> true</span></span>
<span id="cb49-7"><a href="#cb49-7"></a><span class="co">---</span></span>
<span id="cb49-8"><a href="#cb49-8"></a></span>
<span id="cb49-9"><a href="#cb49-9"></a><span class="al">![](../../images/fancy_stats.jpg)</span>{width=50%}   </span>
<span id="cb49-10"><a href="#cb49-10"></a></span>
<span id="cb49-11"><a href="#cb49-11"></a><span class="at">&gt; *"He uses statistics as a drunken man uses lamp posts---for support rather than for illumination."*</span></span>
<span id="cb49-12"><a href="#cb49-12"></a><span class="at">&gt;</span></span>
<span id="cb49-13"><a href="#cb49-13"></a><span class="at">&gt; --- Marissa Mayer</span></span>
<span id="cb49-14"><a href="#cb49-14"></a></span>
<span id="cb49-15"><a href="#cb49-15"></a><span class="in">```{r, echo=FALSE}</span></span>
<span id="cb49-16"><a href="#cb49-16"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(</span>
<span id="cb49-17"><a href="#cb49-17"></a>  <span class="at">echo =</span> <span class="cn">TRUE</span>,</span>
<span id="cb49-18"><a href="#cb49-18"></a>  <span class="at">eval =</span> <span class="cn">TRUE</span>,</span>
<span id="cb49-19"><a href="#cb49-19"></a>  <span class="at">warning =</span> <span class="cn">FALSE</span>,</span>
<span id="cb49-20"><a href="#cb49-20"></a>  <span class="at">message =</span> <span class="cn">FALSE</span>,</span>
<span id="cb49-21"><a href="#cb49-21"></a>  <span class="at">fig.width =</span> <span class="dv">6</span>,</span>
<span id="cb49-22"><a href="#cb49-22"></a>  <span class="at">fig.asp =</span> <span class="fl">0.65</span>,</span>
<span id="cb49-23"><a href="#cb49-23"></a>  <span class="at">out.width =</span> <span class="st">"100%"</span>,</span>
<span id="cb49-24"><a href="#cb49-24"></a>  <span class="at">fig.align =</span> <span class="st">"center"</span></span>
<span id="cb49-25"><a href="#cb49-25"></a>)</span>
<span id="cb49-26"><a href="#cb49-26"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb49-27"><a href="#cb49-27"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb49-28"><a href="#cb49-28"></a><span class="fu">library</span>(ggthemes)</span>
<span id="cb49-29"><a href="#cb49-29"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb49-30"><a href="#cb49-30"></a><span class="in">```</span></span>
<span id="cb49-31"><a href="#cb49-31"></a></span>
<span id="cb49-32"><a href="#cb49-32"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb49-33"><a href="#cb49-33"></a><span class="fu">## In this Chapter</span></span>
<span id="cb49-34"><a href="#cb49-34"></a></span>
<span id="cb49-35"><a href="#cb49-35"></a><span class="ss">-   </span>Single factor ANOVA</span>
<span id="cb49-36"><a href="#cb49-36"></a><span class="ss">-   </span>Multiple factor ANOVA</span>
<span id="cb49-37"><a href="#cb49-37"></a><span class="ss">-   </span>Tukey HSD test</span>
<span id="cb49-38"><a href="#cb49-38"></a><span class="ss">-   </span>Wilcoxon rank sum test</span>
<span id="cb49-39"><a href="#cb49-39"></a><span class="ss">-   </span>Kruskal-Wallis rank sum test</span>
<span id="cb49-40"><a href="#cb49-40"></a>:::</span>
<span id="cb49-41"><a href="#cb49-41"></a></span>
<span id="cb49-42"><a href="#cb49-42"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb49-43"><a href="#cb49-43"></a><span class="fu">## Cheatsheet</span></span>
<span id="cb49-44"><a href="#cb49-44"></a>Find here a <span class="co">[</span><span class="ot">Cheatsheet</span><span class="co">](../../docs/Methods_cheatsheet_v1.pdf)</span> on statistical methods.</span>
<span id="cb49-45"><a href="#cb49-45"></a>:::</span>
<span id="cb49-46"><a href="#cb49-46"></a></span>
<span id="cb49-47"><a href="#cb49-47"></a>::: {.callout-important appearance="simple"}</span>
<span id="cb49-48"><a href="#cb49-48"></a><span class="fu">## Tasks to complete in this Chapter</span></span>
<span id="cb49-49"><a href="#cb49-49"></a></span>
<span id="cb49-50"><a href="#cb49-50"></a><span class="ss">-   </span>Task G</span>
<span id="cb49-51"><a href="#cb49-51"></a>:::</span>
<span id="cb49-52"><a href="#cb49-52"></a></span>
<span id="cb49-53"><a href="#cb49-53"></a><span class="fu"># At a glance</span></span>
<span id="cb49-54"><a href="#cb49-54"></a></span>
<span id="cb49-55"><a href="#cb49-55"></a>Where the Student's and Welch's *t*-tests are used to test for the difference between the means of two groups, analysis of variance (ANOVA)[^1] is used to *compare the means of three or more groups*. The following expectations about your data should be met to conduct an ANOVA:</span>
<span id="cb49-56"><a href="#cb49-56"></a></span>
<span id="cb49-57"><a href="#cb49-57"></a><span class="ot">[^1]: </span>Variations of the basic ANOVA include an analysis of covariance (ANCOVA) when, in addition to the categorical variable, your data also contains a **continuous covariate** (e.g., age, mass) that you want to control. Or, if your study involves multiple dependent variables, you may need to consider using **multivariate analysis of variance (MANOVA)** instead. ANOVAs that have multiple independent variables are called **factorial ANOVAs** (e.g. 2-way ANOVA or 3-way ANOVA).</span>
<span id="cb49-58"><a href="#cb49-58"></a></span>
<span id="cb49-59"><a href="#cb49-59"></a></span>
<span id="cb49-60"><a href="#cb49-60"></a><span class="ss">-   </span>**Continuous dependent variable** The dependent variable should be measured on a *continuous scale* (e.g., height, weight, test scores).</span>
<span id="cb49-61"><a href="#cb49-61"></a></span>
<span id="cb49-62"><a href="#cb49-62"></a><span class="ss">-   </span>**Categorical independent variable** The independent variable should be *categorical with at least three levels* or groups (e.g., different treatments, age groups).</span>
<span id="cb49-63"><a href="#cb49-63"></a></span>
<span id="cb49-64"><a href="#cb49-64"></a><span class="ss">    - </span>**Independent groups** The groups being compared should be independent of each other, meaning that the observations within each group should not affect the other group's observations. In the case of repeated measures ANOVA, the groups are related (e.g., before-and-after measurements or multiple measurements on the same subjects).</span>
<span id="cb49-65"><a href="#cb49-65"></a></span>
<span id="cb49-66"><a href="#cb49-66"></a>Other assumptions to be aware of with regards to the dependent variable:</span>
<span id="cb49-67"><a href="#cb49-67"></a></span>
<span id="cb49-68"><a href="#cb49-68"></a><span class="ss">-   </span>Independence of data</span>
<span id="cb49-69"><a href="#cb49-69"></a><span class="ss">-   </span>Normally distributed data</span>
<span id="cb49-70"><a href="#cb49-70"></a><span class="ss">-   </span>Homogeneity of variances</span>
<span id="cb49-71"><a href="#cb49-71"></a><span class="ss">-   </span>We also encourage that the data are balanced</span>
<span id="cb49-72"><a href="#cb49-72"></a></span>
<span id="cb49-73"><a href="#cb49-73"></a><span class="fu"># Introduction to ANOVA</span></span>
<span id="cb49-74"><a href="#cb49-74"></a></span>
<span id="cb49-75"><a href="#cb49-75"></a>Whole big books have been written about analysis of variance (ANOVA). Although there are many experimental designs that may be analysed with ANOVAs, biologists are taught to pay special attention to the design of experiments, and generally make sure that the experiments are fully *factorial* (in the case of two-way or higher ANOVAs) and balanced. For this reason we will focus in this introductory statistics course on one-way and factorial ANOVAs only.</span>
<span id="cb49-76"><a href="#cb49-76"></a></span>
<span id="cb49-77"><a href="#cb49-77"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb49-78"><a href="#cb49-78"></a><span class="fu">## Factorial designs</span></span>
<span id="cb49-79"><a href="#cb49-79"></a>A factorial experimental design involves studying the effects of two or more factors (also known as independent variables) on a response variable (dependent variable) simultaneously. The term 'factorial' comes from the fact that each level of one factor is combined with each level of the other factor(s), resulting in all possible combinations of factor levels.</span>
<span id="cb49-80"><a href="#cb49-80"></a></span>
<span id="cb49-81"><a href="#cb49-81"></a>In a factorial design, the *main effects* of each factor and the *interaction effects* between factors can be analysed. When we have two factors, this is typically done by applying a 2-way ANOVA, but higher-order ANOVAs are also available. The main effect of a factor is its individual effect on the response variable (ignoring the effect due to the other(s)), while the interaction effect occurs when the effect of one factor depends on the level of another factor. </span>
<span id="cb49-82"><a href="#cb49-82"></a></span>
<span id="cb49-83"><a href="#cb49-83"></a>The simplest factorial design is the 2 $\times$ 2 design, which involves two factors, each with two levels. For example, if you are studying the effect of temperature (high vs. low) and fertiliser concentration (high vs. low) on the yield of a crop, a 2 $\times$ 2 factorial design would involve four experimental conditions:</span>
<span id="cb49-84"><a href="#cb49-84"></a></span>
<span id="cb49-85"><a href="#cb49-85"></a><span class="ss">* </span>high temperature and high fertiliser concentration</span>
<span id="cb49-86"><a href="#cb49-86"></a><span class="ss">* </span>high temperature and low fertiliser concentration</span>
<span id="cb49-87"><a href="#cb49-87"></a><span class="ss">* </span>low temperature and high fertiliser concentration</span>
<span id="cb49-88"><a href="#cb49-88"></a><span class="ss">* </span>low temperature and low fertiliser concentration</span>
<span id="cb49-89"><a href="#cb49-89"></a></span>
<span id="cb49-90"><a href="#cb49-90"></a>In the above example there are 4 possible combinations. Factorial designs can involve more than two factors and/or more than two levels per factor, leading to more complex experimental setups. For example, in a 3 $\times$ 4 there will be 12 combinations of the two factor levels, and in a 2 $\times$ 3 $\times$ 5 factorial experiment the three factors with their respective levels will result in 30 combinations.     </span>
<span id="cb49-91"><a href="#cb49-91"></a>:::</span>
<span id="cb49-92"><a href="#cb49-92"></a></span>
<span id="cb49-93"><a href="#cb49-93"></a>As we have seen in <span class="co">[</span><span class="ot">Chapter 7</span><span class="co">](07-t_tests.qmd)</span> about *t*-tests, ANOVAs also require that some assumptions are met:</span>
<span id="cb49-94"><a href="#cb49-94"></a></span>
<span id="cb49-95"><a href="#cb49-95"></a><span class="ss">-   </span>**Normal distribution** The data in each group should follow a normal distribution or be approximately normally distributed. If the assumption of normality is not met, as, for example, determined with the <span class="in">`shapiro.test()`</span>, you may consider using a non-parametric alternative such as Kruskal-Wallis rank sum test, <span class="in">`kruskal.test()`</span>. This assumption can be relaxed for large sample sizes due to the <span class="co">[</span><span class="ot">Central Limit Theorem</span><span class="co">](04-distributions.qmd#sec-normal)</span>.</span>
<span id="cb49-96"><a href="#cb49-96"></a></span>
<span id="cb49-97"><a href="#cb49-97"></a><span class="ss">-   </span>**Homoscedasticity** The variances of the groups should be approximately equal. This assumption can be tested using Levene's test, <span class="in">`car::leveneTest()`</span>, or Bartlett's test, <span class="in">`bartlett.test()`</span>. When the variances are different but the data are normally distributed, consider the 'Welch ANOVA,' <span class="in">`oneway.test()`</span>, instead of <span class="in">`aov()`</span>.</span>
<span id="cb49-98"><a href="#cb49-98"></a></span>
<span id="cb49-99"><a href="#cb49-99"></a><span class="ss">-   </span>**Random sampling** The data should be obtained through random sampling or random assignment, ensuring that each observation has an equal chance of being included in the sample.</span>
<span id="cb49-100"><a href="#cb49-100"></a></span>
<span id="cb49-101"><a href="#cb49-101"></a><span class="ss">-   </span>**Independent observations** The observations within each group should be independent of each other.</span>
<span id="cb49-102"><a href="#cb49-102"></a></span>
<span id="cb49-103"><a href="#cb49-103"></a>If some of the above assumptions are violated, then your course of action is to either use a <span class="co">[</span><span class="ot">non-parametric test</span><span class="co">](@sec-alt)</span> (and <span class="co">[</span><span class="ot">here</span><span class="co">](../../docs/Methods_cheatsheet_v1.pdf)</span>), transform the data (as in <span class="co">[</span><span class="ot">Chapter 6</span><span class="co">](06-assumptions.qmd)</span>) or use a generalised linear model if non-normal, or to use a linear mixed model when non-independence of data cannot be guaranteed. As I have already indicated, ANOVAs are also sensitive to the presence of outliers, so we need to ensure that outliers are not present. Outliers can be removed but if they are an important feature of the data, then a non-parametric test can be used.</span>
<span id="cb49-104"><a href="#cb49-104"></a></span>
<span id="cb49-105"><a href="#cb49-105"></a>Rather than talking about *t*-tests and ANOVAs as if they do different things, let us acknowledge that they ask a similar question. That question being, "are the means of these two or more things we want to compare different, or are they the same?" At this stage it is important to note that, as with *t*-tests, the independent variable is expected to be categorical (i.e. a factor denoting two or more different treatments or sampling conditions) and that the dependent variable must be continuous. You may perhaps be more familiar with this question when it is presented as a set of hypotheses as we saw in a *t*-test:</span>
<span id="cb49-106"><a href="#cb49-106"></a></span>
<span id="cb49-107"><a href="#cb49-107"></a><span class="ss">-   </span>***H*~0~** Group A is not different from Group B.</span>
<span id="cb49-108"><a href="#cb49-108"></a><span class="ss">-   </span>***H*~a~** Group A is different from Group B.</span>
<span id="cb49-109"><a href="#cb49-109"></a></span>
<span id="cb49-110"><a href="#cb49-110"></a>In an ANOVA, hypotheses willl be more similar to these:</span>
<span id="cb49-111"><a href="#cb49-111"></a></span>
<span id="cb49-112"><a href="#cb49-112"></a><span class="ss">-   </span>***H*~0~** There are no differences among Groups A, B, or C.</span>
<span id="cb49-113"><a href="#cb49-113"></a><span class="ss">-   </span>***H*~a~** There are differences among Groups A, B, or C.</span>
<span id="cb49-114"><a href="#cb49-114"></a></span>
<span id="cb49-115"><a href="#cb49-115"></a>This is a scientific question in the simplest sense. Often, for basic inquiries such as that posed above, we need to see if one group differs significantly from another. The way in which we accomplish this is by looking at the mean and variance within a set of data compared against another similar set. </span>
<span id="cb49-116"><a href="#cb49-116"></a></span>
<span id="cb49-117"><a href="#cb49-117"></a><span class="fu">## Remember the *t*-test</span></span>
<span id="cb49-118"><a href="#cb49-118"></a></span>
<span id="cb49-119"><a href="#cb49-119"></a>As you already know, a *t*-test is used when we want to compare two different sample sets against one another. This is also known as a two-factor or two level test. When one wants to compare multiple (more than two) sample sets against one another an ANOVA is required (I'll get there shortly). Remember how to perform a *t*-test in R: we will revisit this test using the <span class="in">`chicks`</span> data, but only for Diets 1 and 2 from day 21.</span>
<span id="cb49-120"><a href="#cb49-120"></a></span>
<span id="cb49-123"><a href="#cb49-123"></a><span class="in">```{r}</span></span>
<span id="cb49-124"><a href="#cb49-124"></a><span class="co"># First grab the data</span></span>
<span id="cb49-125"><a href="#cb49-125"></a>chicks <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(ChickWeight)</span>
<span id="cb49-126"><a href="#cb49-126"></a></span>
<span id="cb49-127"><a href="#cb49-127"></a><span class="co"># Then subset out only the sample sets to be compared</span></span>
<span id="cb49-128"><a href="#cb49-128"></a>chicks_sub <span class="ot">&lt;-</span> chicks <span class="sc">%&gt;%</span> </span>
<span id="cb49-129"><a href="#cb49-129"></a>  <span class="fu">filter</span>(Diet <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), Time <span class="sc">==</span> <span class="dv">21</span>)</span>
<span id="cb49-130"><a href="#cb49-130"></a><span class="in">```</span></span>
<span id="cb49-131"><a href="#cb49-131"></a></span>
<span id="cb49-132"><a href="#cb49-132"></a>Once we have filtered our data we may now perform the *t*-test. </span>
<span id="cb49-133"><a href="#cb49-133"></a></span>
<span id="cb49-136"><a href="#cb49-136"></a><span class="in">```{r}</span></span>
<span id="cb49-137"><a href="#cb49-137"></a><span class="fu">t.test</span>(weight <span class="sc">~</span> Diet, <span class="at">data =</span> chicks_sub)</span>
<span id="cb49-138"><a href="#cb49-138"></a><span class="in">```</span></span>
<span id="cb49-139"><a href="#cb49-139"></a></span>
<span id="cb49-140"><a href="#cb49-140"></a>As one may recall from <span class="co">[</span><span class="ot">Chapter 7</span><span class="co">](07-t_tests.qmd)</span>, whenever we want to give a formula to a function in R, we use the <span class="in">`~`</span>. The formula used above, <span class="in">`weight ~ Diet`</span>, reads in plain English as "weight as a function of diet". This is perhaps easier to understand as "*Y* as a function of *X*." This means that we are assuming whatever is to the left of the <span class="in">`~`</span> is the dependant variable, and whatever is to the right is the independent variable. Did the Diet 1 and 2 produce significantly fatter birds?</span>
<span id="cb49-141"><a href="#cb49-141"></a></span>
<span id="cb49-142"><a href="#cb49-142"></a>One could also supplement the output by producing a graph (@fig-boxwhisker1).</span>
<span id="cb49-143"><a href="#cb49-143"></a></span>
<span id="cb49-146"><a href="#cb49-146"></a><span class="in">```{r}</span></span>
<span id="cb49-147"><a href="#cb49-147"></a><span class="co">#| label: fig-boxwhisker1</span></span>
<span id="cb49-148"><a href="#cb49-148"></a><span class="co">#| fig-cap: "Box-and-whisker plot showing the differences in means between chicks reared to 21 days old and fed Diets 1 and 2"</span></span>
<span id="cb49-149"><a href="#cb49-149"></a></span>
<span id="cb49-150"><a href="#cb49-150"></a><span class="fu">library</span>(ggstatsplot)</span>
<span id="cb49-151"><a href="#cb49-151"></a></span>
<span id="cb49-152"><a href="#cb49-152"></a><span class="do">## since the confidence intervals for the effect sizes are computed using</span></span>
<span id="cb49-153"><a href="#cb49-153"></a><span class="do">## bootstrapping, important to set a seed for reproducibility</span></span>
<span id="cb49-154"><a href="#cb49-154"></a><span class="fu">set.seed</span>(<span class="dv">13</span>)</span>
<span id="cb49-155"><a href="#cb49-155"></a></span>
<span id="cb49-156"><a href="#cb49-156"></a><span class="do">## parametric t-test and box plot</span></span>
<span id="cb49-157"><a href="#cb49-157"></a><span class="fu">ggbetweenstats</span>(</span>
<span id="cb49-158"><a href="#cb49-158"></a>  <span class="at">data =</span> chicks_sub,</span>
<span id="cb49-159"><a href="#cb49-159"></a>  <span class="at">x =</span> Diet,</span>
<span id="cb49-160"><a href="#cb49-160"></a>  <span class="at">y =</span> weight,</span>
<span id="cb49-161"><a href="#cb49-161"></a>  <span class="at">xlab =</span> <span class="st">"Diet"</span>,</span>
<span id="cb49-162"><a href="#cb49-162"></a>  <span class="at">ylab =</span> <span class="st">"Chick mass (g)"</span>,</span>
<span id="cb49-163"><a href="#cb49-163"></a>  <span class="at">plot.type =</span> <span class="st">"box"</span>,</span>
<span id="cb49-164"><a href="#cb49-164"></a>  <span class="at">p.adjust.method =</span> <span class="st">"bonferroni"</span>,</span>
<span id="cb49-165"><a href="#cb49-165"></a>  <span class="at">pairwise.display =</span> <span class="st">"ns"</span>,</span>
<span id="cb49-166"><a href="#cb49-166"></a>  <span class="at">type =</span> <span class="st">"p"</span>,</span>
<span id="cb49-167"><a href="#cb49-167"></a>  <span class="at">results.subtitle =</span> <span class="cn">FALSE</span>,</span>
<span id="cb49-168"><a href="#cb49-168"></a>  <span class="at">conf.level =</span> <span class="fl">0.95</span>,</span>
<span id="cb49-169"><a href="#cb49-169"></a>  <span class="at">title =</span> <span class="st">"t-test"</span>,</span>
<span id="cb49-170"><a href="#cb49-170"></a>  <span class="at">ggtheme =</span> ggthemes<span class="sc">::</span><span class="fu">theme_fivethirtyeight</span>(),</span>
<span id="cb49-171"><a href="#cb49-171"></a>  <span class="at">package =</span> <span class="st">"basetheme"</span>,</span>
<span id="cb49-172"><a href="#cb49-172"></a>  <span class="at">palette =</span> <span class="st">"ink"</span></span>
<span id="cb49-173"><a href="#cb49-173"></a>)</span>
<span id="cb49-174"><a href="#cb49-174"></a><span class="in">```</span></span>
<span id="cb49-175"><a href="#cb49-175"></a></span>
<span id="cb49-176"><a href="#cb49-176"></a>Notice above that we did not need to specify to use a *t*-test. The `ggbetweenstats()` function automatically determines if an independent samples *t*-test or a 1-way ANOVA is required based on whether there are two groups or three or more groups within the grouping (factor) variable.</span>
<span id="cb49-177"><a href="#cb49-177"></a></span>
<span id="cb49-178"><a href="#cb49-178"></a>That was a nice revision. But applied to the <span class="in">`chicks`</span> data it seemed a bit silly, because you may ask, "What if I wanted to know if there are differences among the means computed at Day 1, Day 6, Day 10, and Day 21?" We should not use *t*-tests to do this (although we can). So now we can move on to the ANOVA.</span>
<span id="cb49-179"><a href="#cb49-179"></a></span>
<span id="cb49-180"><a href="#cb49-180"></a>:::: {.callout-important}</span>
<span id="cb49-181"><a href="#cb49-181"></a><span class="fu">## Task G.1: Do it now!</span></span>
<span id="cb49-182"><a href="#cb49-182"></a><span class="ss">- </span>Why should we not just apply *t*-tests once per each of the pairs of comparisons we want to make?</span>
<span id="cb49-183"><a href="#cb49-183"></a>::::</span>
<span id="cb49-184"><a href="#cb49-184"></a></span>
<span id="cb49-185"><a href="#cb49-185"></a><span class="fu">## Why not do multiple *t*-tests?</span></span>
<span id="cb49-186"><a href="#cb49-186"></a></span>
<span id="cb49-187"><a href="#cb49-187"></a>In the <span class="in">`chicks`</span> data we have four diets, not only two as in the *t*-test example just performed. Why not then simply do a *t*-test multiple times, once for each pair of diets given to the chickens? Multiple *t*-tests would be written as:</span>
<span id="cb49-188"><a href="#cb49-188"></a></span>
<span id="cb49-189"><a href="#cb49-189"></a><span class="ss">-   </span>$H_{0}: \mu_1 = \mu_2$</span>
<span id="cb49-190"><a href="#cb49-190"></a><span class="ss">-   </span>$H_{0}: \mu_1 = \mu_3$</span>
<span id="cb49-191"><a href="#cb49-191"></a><span class="ss">-   </span>$H_{0}: \mu_1 = \mu_4$</span>
<span id="cb49-192"><a href="#cb49-192"></a><span class="ss">-   </span>$H_{0}: \mu_2 = \mu_3$</span>
<span id="cb49-193"><a href="#cb49-193"></a><span class="ss">-   </span>$H_{0}: \mu_2 = \mu_4$</span>
<span id="cb49-194"><a href="#cb49-194"></a><span class="ss">-   </span>$H_{0}: \mu_3 = \mu_4$</span>
<span id="cb49-195"><a href="#cb49-195"></a></span>
<span id="cb49-196"><a href="#cb49-196"></a>This would be invalid. The problem is that the chance of committing a Type I error increases as more multiple comparisons are done. So, the overall chance of rejecting the *H*~0~ increases. Why? If one sets $\alpha=0.05$ (the significance level below which the *H*~0~ is no longer accepted), one will still reject the *H*~0~ 5% of the time when it is in fact true (i.e. when there is no difference between the groups). When many pairwise comparisons are made, the probability of rejecting the *H*~0~ at least once is higher because we take this 5% risk each time we repeat a *t*-test. In the case of the chicken diets, we would have to perform six *t*-tests, and the error rate would increase to slightly less than $6\times5\%$. See Table 1. </span>
<span id="cb49-199"><a href="#cb49-199"></a><span class="in">```{r}</span></span>
<span id="cb49-200"><a href="#cb49-200"></a><span class="co">#| echo: false</span></span>
<span id="cb49-201"><a href="#cb49-201"></a><span class="co"># Function to calculate the probability of Type I error</span></span>
<span id="cb49-202"><a href="#cb49-202"></a>type_one_error_probability <span class="ot">&lt;-</span> <span class="cf">function</span>(k, alpha) {</span>
<span id="cb49-203"><a href="#cb49-203"></a>  pairwise_comparisons <span class="ot">&lt;-</span> k <span class="sc">*</span> (k <span class="sc">-</span> <span class="dv">1</span>) <span class="sc">/</span> <span class="dv">2</span></span>
<span id="cb49-204"><a href="#cb49-204"></a>  <span class="fu">return</span>(<span class="dv">1</span> <span class="sc">-</span> (<span class="dv">1</span> <span class="sc">-</span> alpha)<span class="sc">^</span>pairwise_comparisons)</span>
<span id="cb49-205"><a href="#cb49-205"></a>}</span>
<span id="cb49-206"><a href="#cb49-206"></a></span>
<span id="cb49-207"><a href="#cb49-207"></a><span class="co"># Set the values of k and significance levels</span></span>
<span id="cb49-208"><a href="#cb49-208"></a>k_values <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>, <span class="dv">10</span>, <span class="dv">20</span>, <span class="dv">100</span>)</span>
<span id="cb49-209"><a href="#cb49-209"></a>alpha_values <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.20</span>, <span class="fl">0.10</span>, <span class="fl">0.05</span>, <span class="fl">0.02</span>, <span class="fl">0.01</span>, <span class="fl">0.001</span>)</span>
<span id="cb49-210"><a href="#cb49-210"></a></span>
<span id="cb49-211"><a href="#cb49-211"></a><span class="co"># Create an empty data frame to store the results</span></span>
<span id="cb49-212"><a href="#cb49-212"></a>results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">K =</span> <span class="fu">integer</span>(),</span>
<span id="cb49-213"><a href="#cb49-213"></a>                      <span class="at">Alpha =</span> <span class="fu">numeric</span>(),</span>
<span id="cb49-214"><a href="#cb49-214"></a>                      <span class="at">Type1ErrorProbability =</span> <span class="fu">numeric</span>())</span>
<span id="cb49-215"><a href="#cb49-215"></a></span>
<span id="cb49-216"><a href="#cb49-216"></a><span class="co"># Calculate the probabilities and store them in the data frame</span></span>
<span id="cb49-217"><a href="#cb49-217"></a><span class="cf">for</span> (k <span class="cf">in</span> k_values) {</span>
<span id="cb49-218"><a href="#cb49-218"></a>  <span class="cf">for</span> (alpha <span class="cf">in</span> alpha_values) {</span>
<span id="cb49-219"><a href="#cb49-219"></a>    type1_error_prob <span class="ot">&lt;-</span> <span class="fu">type_one_error_probability</span>(k, alpha)</span>
<span id="cb49-220"><a href="#cb49-220"></a>    results <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb49-221"><a href="#cb49-221"></a>      results, <span class="fu">data.frame</span>(<span class="at">K =</span> k,</span>
<span id="cb49-222"><a href="#cb49-222"></a>                          <span class="at">Alpha =</span> alpha,</span>
<span id="cb49-223"><a href="#cb49-223"></a>                          <span class="at">Type1ErrorProbability =</span> type1_error_prob))</span>
<span id="cb49-224"><a href="#cb49-224"></a>  }</span>
<span id="cb49-225"><a href="#cb49-225"></a>}</span>
<span id="cb49-226"><a href="#cb49-226"></a></span>
<span id="cb49-227"><a href="#cb49-227"></a>results_long <span class="ot">&lt;-</span> results <span class="sc">|&gt;</span> </span>
<span id="cb49-228"><a href="#cb49-228"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Alpha, <span class="at">values_from =</span> Type1ErrorProbability)</span>
<span id="cb49-229"><a href="#cb49-229"></a><span class="in">```</span></span>
<span id="cb49-230"><a href="#cb49-230"></a></span>
<span id="cb49-231"><a href="#cb49-231"></a>::: {#kable-table-div}</span>
<span id="cb49-234"><a href="#cb49-234"></a><span class="in">```{r}</span></span>
<span id="cb49-235"><a href="#cb49-235"></a><span class="co">#| tbl-cap: "**Table 1.** Probability of committing a Type I error due to applying multiple *t*-tests to test for differences between *K* means. *α* from `0.2` to `0.0001` are shown."</span></span>
<span id="cb49-236"><a href="#cb49-236"></a><span class="co">#| echo: false</span></span>
<span id="cb49-237"><a href="#cb49-237"></a>    <span class="fu">kbl</span>(results_long, <span class="at">digits =</span> <span class="dv">2</span>) <span class="sc">|&gt;</span></span>
<span id="cb49-238"><a href="#cb49-238"></a>      <span class="fu">kable_classic</span>(<span class="at">bootstrap_options =</span> <span class="st">"striped"</span>, <span class="at">full_width =</span> <span class="cn">FALSE</span>)</span>
<span id="cb49-239"><a href="#cb49-239"></a><span class="in">```</span></span>
<span id="cb49-240"><a href="#cb49-240"></a>:::</span>
<span id="cb49-241"><a href="#cb49-241"></a></span>
<span id="cb49-242"><a href="#cb49-242"></a>If you insist in creating more work for yourself and do *t*-tests many times, one way to overcome the problem of committing Type I errors that stem from multiple comparisons is to apply a Bonferroni correction.</span>
<span id="cb49-243"><a href="#cb49-243"></a></span>
<span id="cb49-244"><a href="#cb49-244"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb49-245"><a href="#cb49-245"></a><span class="fu">## Bonferonni correction</span></span>
<span id="cb49-246"><a href="#cb49-246"></a>The Bonferroni correction is used to adjust the significance level of multiple hypothesis tests, such as multiple paired *t*-tests among many groups, in order to reduce the risk of false positives or Type I errors. It is named after the Italian mathematician Carlo Emilio Bonferroni.</span>
<span id="cb49-247"><a href="#cb49-247"></a></span>
<span id="cb49-248"><a href="#cb49-248"></a>The Bonferroni correction is based on the principle that when multiple hypothesis tests are performed, the probability of observing at least one significant result due to random chance increases. To correct for this, the significance level (usually 0.05) is divided by the number of tests being performed. This results in a more stringent significance level for each individual test and it so reduces the risk of committing a Type I error.</span>
<span id="cb49-249"><a href="#cb49-249"></a></span>
<span id="cb49-250"><a href="#cb49-250"></a>For example, if we conduct ten hypothesis tests, the significance level for each test after Bonferonni correction would become 0.05/10 = 0.005. The implication is that each individual test would need to have a *p*-value less than 0.005 to be considered significant at the overall significance level of 0.05.</span>
<span id="cb49-251"><a href="#cb49-251"></a></span>
<span id="cb49-252"><a href="#cb49-252"></a>On the downside, this method can be overly conservative and we may then increase the risk of Type II errors, which are false negatives. If you really cannot avoid multiple tests, then also assess one of the alternatives to Bonferonni's method, viz: the false discovery rate (FDR) correction, the Holm-Bonferroni correction, Benjamini-Hochberg's procedure, the Sidak correction, or some of the Bayesian approaches.</span>
<span id="cb49-253"><a href="#cb49-253"></a>:::</span>
<span id="cb49-254"><a href="#cb49-254"></a></span>
<span id="cb49-255"><a href="#cb49-255"></a>Or better still, we do an ANOVA that controls for these Type I errors so that it remains at 5%.</span>
<span id="cb49-256"><a href="#cb49-256"></a></span>
<span id="cb49-257"><a href="#cb49-257"></a><span class="fu"># ANOVA</span></span>
<span id="cb49-258"><a href="#cb49-258"></a></span>
<span id="cb49-259"><a href="#cb49-259"></a>::: {.callout-note}</span>
<span id="cb49-260"><a href="#cb49-260"></a><span class="fu">## Hypothesis for an ANOVA</span></span>
<span id="cb49-261"><a href="#cb49-261"></a></span>
<span id="cb49-262"><a href="#cb49-262"></a>If we have four groups whose means we want to compare, a suitable *H*~0~ would be:</span>
<span id="cb49-263"><a href="#cb49-263"></a></span>
<span id="cb49-264"><a href="#cb49-264"></a>$H_{0}: \mu_1 = \mu_2 = \mu_3 = \mu_4$</span>
<span id="cb49-265"><a href="#cb49-265"></a></span>
<span id="cb49-266"><a href="#cb49-266"></a>and</span>
<span id="cb49-267"><a href="#cb49-267"></a></span>
<span id="cb49-268"><a href="#cb49-268"></a>$H_{a}$: the means of the four groups are not equal.</span>
<span id="cb49-269"><a href="#cb49-269"></a></span>
<span id="cb49-270"><a href="#cb49-270"></a>Here, $\mu_1$, $\mu_2$, $\mu_3$, and $\mu_4$ are four population means. For the *H*~0~ to be rejected, all that is required is for one of the pairs of means to be different, not all of them.</span>
<span id="cb49-271"><a href="#cb49-271"></a>:::</span>
<span id="cb49-272"><a href="#cb49-272"></a></span>
<span id="cb49-273"><a href="#cb49-273"></a><span class="fu">## Single factor</span></span>
<span id="cb49-274"><a href="#cb49-274"></a></span>
<span id="cb49-275"><a href="#cb49-275"></a>We continue with the chicken data. The *t*-test showed that Diets 1 and 2 resulted in the same chicken mass at Day 21. What about the other two diets? Our *H*~0~ is that, at Day 21, $\mu_{1}=\mu_{2}=\mu_{3}=\mu_{4}$. Is there a statistical difference between chickens fed these four diets, or do we retain the *H*~0~? The R function for an ANOVA is <span class="in">`aov()`</span>. To look for significant differences between all four diets on the last day of sampling we use this one line of code:</span>
<span id="cb49-276"><a href="#cb49-276"></a></span>
<span id="cb49-279"><a href="#cb49-279"></a><span class="in">```{r}</span></span>
<span id="cb49-280"><a href="#cb49-280"></a>chicks.aov1 <span class="ot">&lt;-</span> <span class="fu">aov</span>(weight <span class="sc">~</span> Diet, <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">21</span>))</span>
<span id="cb49-281"><a href="#cb49-281"></a><span class="fu">summary</span>(chicks.aov1)</span>
<span id="cb49-282"><a href="#cb49-282"></a><span class="in">```</span></span>
<span id="cb49-283"><a href="#cb49-283"></a></span>
<span id="cb49-284"><a href="#cb49-284"></a>:::: {.callout-important}</span>
<span id="cb49-285"><a href="#cb49-285"></a><span class="fu">## Task G.2: Do it now!</span></span>
<span id="cb49-286"><a href="#cb49-286"></a>a. What does the outcome say about the chicken masses? Which ones are different from each other?</span>
<span id="cb49-287"><a href="#cb49-287"></a>b. Devise a graphical display of this outcome.</span>
<span id="cb49-288"><a href="#cb49-288"></a>::::</span>
<span id="cb49-289"><a href="#cb49-289"></a></span>
<span id="cb49-290"><a href="#cb49-290"></a>If this seems too easy to be true, it's because we aren't quite done yet. You could use your graphical display to eyeball where the significant differences are, or we can turn to a more 'precise' approach. The next step one could take is to run a Tukey HSD test on the results of the ANOVA by wrapping <span class="in">`tukeyHSD()`</span> around <span class="in">`aov()`</span>:</span>
<span id="cb49-291"><a href="#cb49-291"></a></span>
<span id="cb49-294"><a href="#cb49-294"></a><span class="in">```{r}</span></span>
<span id="cb49-295"><a href="#cb49-295"></a><span class="fu">TukeyHSD</span>(chicks.aov1)</span>
<span id="cb49-296"><a href="#cb49-296"></a><span class="in">```</span></span>
<span id="cb49-297"><a href="#cb49-297"></a></span>
<span id="cb49-298"><a href="#cb49-298"></a>The output of <span class="in">`tukeyHSD()`</span> shows us that pairwise comparisons of all of the groups we are comparing. We can also display this as a very rough figure (@fig-tukeydiff):</span>
<span id="cb49-299"><a href="#cb49-299"></a></span>
<span id="cb49-302"><a href="#cb49-302"></a><span class="in">```{r}</span></span>
<span id="cb49-303"><a href="#cb49-303"></a><span class="co">#| label: fig-tukeydiff</span></span>
<span id="cb49-304"><a href="#cb49-304"></a><span class="co">#| fig-cap: "A plot of the Tukey-HSD test showing the differences in means between chicks reared to 21 days old and fed four diets."</span></span>
<span id="cb49-305"><a href="#cb49-305"></a></span>
<span id="cb49-306"><a href="#cb49-306"></a><span class="fu">plot</span>(<span class="fu">TukeyHSD</span>(chicks.aov1))</span>
<span id="cb49-307"><a href="#cb49-307"></a><span class="in">```</span></span>
<span id="cb49-308"><a href="#cb49-308"></a></span>
<span id="cb49-309"><a href="#cb49-309"></a>We may also produce a nicer looking graphical summary in the form of a box-and-whisker plot and/or a violin plot. Here I combine both (@fig-boxwhisker2):</span>
<span id="cb49-310"><a href="#cb49-310"></a></span>
<span id="cb49-313"><a href="#cb49-313"></a><span class="in">```{r}</span></span>
<span id="cb49-314"><a href="#cb49-314"></a><span class="co">#| label: fig-boxwhisker2</span></span>
<span id="cb49-315"><a href="#cb49-315"></a><span class="co">#| fig-cap: "Box-and-whisker plot showing the differences in means between chicks reared to 21 days old and fed four diets. Shown is a notched box plot where the extent of the notches is `1.58 * IQR / sqrt(n)`. This is approximately equivalent to a 95% confidence interval andf may be used for comparing medians."</span></span>
<span id="cb49-316"><a href="#cb49-316"></a></span>
<span id="cb49-317"><a href="#cb49-317"></a><span class="fu">set.seed</span>(<span class="dv">666</span>)</span>
<span id="cb49-318"><a href="#cb49-318"></a></span>
<span id="cb49-319"><a href="#cb49-319"></a><span class="do">## parametric t-test and box plot</span></span>
<span id="cb49-320"><a href="#cb49-320"></a><span class="fu">ggbetweenstats</span>(</span>
<span id="cb49-321"><a href="#cb49-321"></a>  <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">21</span>),</span>
<span id="cb49-322"><a href="#cb49-322"></a>  <span class="at">x =</span> Diet,</span>
<span id="cb49-323"><a href="#cb49-323"></a>  <span class="at">y =</span> weight,</span>
<span id="cb49-324"><a href="#cb49-324"></a>  <span class="at">xlab =</span> <span class="st">"Diet"</span>,</span>
<span id="cb49-325"><a href="#cb49-325"></a>  <span class="at">ylab =</span> <span class="st">"Chick mass (g)"</span>,</span>
<span id="cb49-326"><a href="#cb49-326"></a>  <span class="at">plot.type =</span> <span class="st">"box"</span>,</span>
<span id="cb49-327"><a href="#cb49-327"></a>  <span class="at">boxplot.args =</span> <span class="fu">list</span>(<span class="at">notch =</span> <span class="cn">TRUE</span>),</span>
<span id="cb49-328"><a href="#cb49-328"></a>  <span class="at">type =</span> <span class="st">"parametric"</span>,</span>
<span id="cb49-329"><a href="#cb49-329"></a>  <span class="at">results.subtitle =</span> <span class="cn">FALSE</span>,</span>
<span id="cb49-330"><a href="#cb49-330"></a>  <span class="at">pairwise.comparisons =</span> <span class="cn">TRUE</span>,</span>
<span id="cb49-331"><a href="#cb49-331"></a>  <span class="at">pairwise.display =</span> <span class="st">"s"</span>,</span>
<span id="cb49-332"><a href="#cb49-332"></a>  <span class="at">p.adjust.method =</span> <span class="st">"bonferroni"</span>,</span>
<span id="cb49-333"><a href="#cb49-333"></a>  <span class="at">conf.level =</span> <span class="fl">0.95</span>,</span>
<span id="cb49-334"><a href="#cb49-334"></a>  <span class="at">title =</span> <span class="st">"ANOVA"</span>,</span>
<span id="cb49-335"><a href="#cb49-335"></a>  <span class="at">ggtheme =</span> ggthemes<span class="sc">::</span><span class="fu">theme_fivethirtyeight</span>(),</span>
<span id="cb49-336"><a href="#cb49-336"></a>  <span class="at">package =</span> <span class="st">"basetheme"</span>,</span>
<span id="cb49-337"><a href="#cb49-337"></a>  <span class="at">palette =</span> <span class="st">"ink"</span></span>
<span id="cb49-338"><a href="#cb49-338"></a>)</span>
<span id="cb49-339"><a href="#cb49-339"></a><span class="in">```</span></span>
<span id="cb49-340"><a href="#cb49-340"></a></span>
<span id="cb49-341"><a href="#cb49-341"></a>:::: {.callout-important}</span>
<span id="cb49-342"><a href="#cb49-342"></a><span class="fu">## Task G.3: Do it now!</span></span>
<span id="cb49-343"><a href="#cb49-343"></a>Look at the help file for the <span class="in">`TukeyHSD()`</span> function to better understand what the output means.</span>
<span id="cb49-344"><a href="#cb49-344"></a></span>
<span id="cb49-345"><a href="#cb49-345"></a>a. How does one interpret the results? What does this tell us about the effect that that different diets has on the chicken weights at Day 21?</span>
<span id="cb49-346"><a href="#cb49-346"></a>b. Figure out a way to plot the Tukey HSD outcomes in **ggplot**.</span>
<span id="cb49-347"><a href="#cb49-347"></a>c. Why does the ANOVA return a significant result, but the Tukey test shows that not all of the groups are significantly different from one another?</span>
<span id="cb49-348"><a href="#cb49-348"></a>::::</span>
<span id="cb49-349"><a href="#cb49-349"></a></span>
<span id="cb49-350"><a href="#cb49-350"></a><span class="co">&lt;!-- ```{r} --&gt;</span></span>
<span id="cb49-351"><a href="#cb49-351"></a></span>
<span id="cb49-352"><a href="#cb49-352"></a><span class="co">&lt;!-- # plot(TukeyHSD(chicks.aov)) --&gt;</span></span>
<span id="cb49-353"><a href="#cb49-353"></a></span>
<span id="cb49-354"><a href="#cb49-354"></a><span class="co">&lt;!-- ``` --&gt;</span></span>
<span id="cb49-355"><a href="#cb49-355"></a></span>
<span id="cb49-356"><a href="#cb49-356"></a><span class="fu">## Multiple factors</span></span>
<span id="cb49-357"><a href="#cb49-357"></a></span>
<span id="cb49-358"><a href="#cb49-358"></a>What if we have multiple grouping variables, and not just one? We would encounter this kind of situation in factorial designs. In the case of the chicken data, there is also time that seems to be having an effect.</span>
<span id="cb49-359"><a href="#cb49-359"></a></span>
<span id="cb49-360"><a href="#cb49-360"></a>:::: {.callout-important}</span>
<span id="cb49-361"><a href="#cb49-361"></a><span class="fu">## Task G.4: Do it now!</span></span>
<span id="cb49-362"><a href="#cb49-362"></a>a. How is time having an effect? **(/3)**</span>
<span id="cb49-363"><a href="#cb49-363"></a>b. What hypotheses can we construct around time? **(/2)**</span>
<span id="cb49-364"><a href="#cb49-364"></a>::::</span>
<span id="cb49-365"><a href="#cb49-365"></a></span>
<span id="cb49-366"><a href="#cb49-366"></a>Let us look at some variations around questions concerning time. We might ask, at a particular time step, are there differences amongst the effect due to diet on chicken mass? Let's see when diets are starting to have an effect by examining the outcomes at times 0, 2, 10, and 21:</span>
<span id="cb49-367"><a href="#cb49-367"></a></span>
<span id="cb49-370"><a href="#cb49-370"></a><span class="in">```{r}</span></span>
<span id="cb49-371"><a href="#cb49-371"></a><span class="co"># effect at time = 0</span></span>
<span id="cb49-372"><a href="#cb49-372"></a><span class="fu">summary</span>(<span class="fu">aov</span>(weight <span class="sc">~</span> Diet, <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">0</span>)))</span>
<span id="cb49-373"><a href="#cb49-373"></a></span>
<span id="cb49-374"><a href="#cb49-374"></a><span class="co"># effect at time = 2</span></span>
<span id="cb49-375"><a href="#cb49-375"></a><span class="fu">summary</span>(<span class="fu">aov</span>(weight <span class="sc">~</span> Diet, <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">2</span>)))</span>
<span id="cb49-376"><a href="#cb49-376"></a></span>
<span id="cb49-377"><a href="#cb49-377"></a><span class="co"># effect at time = 10</span></span>
<span id="cb49-378"><a href="#cb49-378"></a><span class="fu">summary</span>(<span class="fu">aov</span>(weight <span class="sc">~</span> Diet, <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">10</span>)))</span>
<span id="cb49-379"><a href="#cb49-379"></a></span>
<span id="cb49-380"><a href="#cb49-380"></a><span class="co"># effect at time = 21</span></span>
<span id="cb49-381"><a href="#cb49-381"></a><span class="fu">summary</span>(<span class="fu">aov</span>(weight <span class="sc">~</span> Diet, <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">21</span>)))</span>
<span id="cb49-382"><a href="#cb49-382"></a><span class="in">```</span></span>
<span id="cb49-383"><a href="#cb49-383"></a></span>
<span id="cb49-384"><a href="#cb49-384"></a>:::: {.callout-important}</span>
<span id="cb49-385"><a href="#cb49-385"></a><span class="fu">## Task G.5: Do it now!</span></span>
<span id="cb49-386"><a href="#cb49-386"></a>a. What do you conclude from the above series of ANOVAs? **(/3)**</span>
<span id="cb49-387"><a href="#cb49-387"></a>b. What problem is associated with running multiple tests in the way that we have done here? **(/2)**</span>
<span id="cb49-388"><a href="#cb49-388"></a>::::</span>
<span id="cb49-389"><a href="#cb49-389"></a></span>
<span id="cb49-390"><a href="#cb49-390"></a>Or we may ask, regardless of diet (i.e. disregarding the effect of diet by clumping all chickens together), is time having an effect?</span>
<span id="cb49-391"><a href="#cb49-391"></a></span>
<span id="cb49-394"><a href="#cb49-394"></a><span class="in">```{r}</span></span>
<span id="cb49-395"><a href="#cb49-395"></a>chicks.aov2 <span class="ot">&lt;-</span> <span class="fu">aov</span>(weight <span class="sc">~</span> <span class="fu">as.factor</span>(Time),</span>
<span id="cb49-396"><a href="#cb49-396"></a>                   <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">10</span>, <span class="dv">21</span>)))</span>
<span id="cb49-397"><a href="#cb49-397"></a><span class="fu">summary</span>(chicks.aov2)</span>
<span id="cb49-398"><a href="#cb49-398"></a><span class="in">```</span></span>
<span id="cb49-399"><a href="#cb49-399"></a></span>
<span id="cb49-400"><a href="#cb49-400"></a>:::: {.callout-important}</span>
<span id="cb49-401"><a href="#cb49-401"></a><span class="fu">## Task G.6: Do it now!</span></span>
<span id="cb49-402"><a href="#cb49-402"></a>a. Write out the hypotheses for this ANOVA. **(/2)**</span>
<span id="cb49-403"><a href="#cb49-403"></a>b. What do you conclude from the above ANOVA. **(/3)**</span>
<span id="cb49-404"><a href="#cb49-404"></a>::::</span>
<span id="cb49-405"><a href="#cb49-405"></a></span>
<span id="cb49-406"><a href="#cb49-406"></a>Or, to save ourselves a lot of time and reduce the coding effort, we may simply run a two-way ANOVA and look at the effects of <span class="in">`Diet`</span> and <span class="in">`Time`</span> simultaneously. To specify the different factors we put them in our formula and separate them with a <span class="in">`+`</span>:</span>
<span id="cb49-407"><a href="#cb49-407"></a></span>
<span id="cb49-410"><a href="#cb49-410"></a><span class="in">```{r}</span></span>
<span id="cb49-411"><a href="#cb49-411"></a><span class="fu">summary</span>(<span class="fu">aov</span>(weight <span class="sc">~</span> Diet <span class="sc">+</span> <span class="fu">as.factor</span>(Time),</span>
<span id="cb49-412"><a href="#cb49-412"></a>            <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">21</span>))))</span>
<span id="cb49-413"><a href="#cb49-413"></a><span class="in">```</span></span>
<span id="cb49-414"><a href="#cb49-414"></a></span>
<span id="cb49-415"><a href="#cb49-415"></a>:::: {.callout-important}</span>
<span id="cb49-416"><a href="#cb49-416"></a><span class="fu">## Task G.7: Do it now!</span></span>
<span id="cb49-417"><a href="#cb49-417"></a>a. What question are we asking with the above line of code? **(/3)**</span>
<span id="cb49-418"><a href="#cb49-418"></a>b. What is the answer? **(/2)**</span>
<span id="cb49-419"><a href="#cb49-419"></a>c. Why did we wrap <span class="in">`Time`</span> in <span class="in">`as.factor()`</span>? **(/2)**</span>
<span id="cb49-420"><a href="#cb49-420"></a>::::</span>
<span id="cb49-421"><a href="#cb49-421"></a></span>
<span id="cb49-422"><a href="#cb49-422"></a>It is also possible to look at what the interaction effect between grouping variables (i.e. in this case the effect of time on diet---does the effect of time depend on which diet we are looking at?), and not just within the individual grouping variables. To do this we replace the <span class="in">`+`</span> in our formula with <span class="in">`*`</span>:</span>
<span id="cb49-423"><a href="#cb49-423"></a></span>
<span id="cb49-426"><a href="#cb49-426"></a><span class="in">```{r}</span></span>
<span id="cb49-427"><a href="#cb49-427"></a><span class="fu">summary</span>(<span class="fu">aov</span>(weight <span class="sc">~</span> Diet <span class="sc">*</span> <span class="fu">as.factor</span>(Time),</span>
<span id="cb49-428"><a href="#cb49-428"></a>            <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">4</span>, <span class="dv">21</span>))))</span>
<span id="cb49-429"><a href="#cb49-429"></a><span class="in">```</span></span>
<span id="cb49-430"><a href="#cb49-430"></a></span>
<span id="cb49-431"><a href="#cb49-431"></a>:::: {.callout-important}</span>
<span id="cb49-432"><a href="#cb49-432"></a><span class="fu">## Task G.8: Do it now! </span></span>
<span id="cb49-433"><a href="#cb49-433"></a>How do these results differ from the previous set? **(/3)**</span>
<span id="cb49-434"><a href="#cb49-434"></a>::::</span>
<span id="cb49-435"><a href="#cb49-435"></a></span>
<span id="cb49-436"><a href="#cb49-436"></a>One may also run a post-hoc Tukey test on these results the same as for a single factor ANOVA:</span>
<span id="cb49-437"><a href="#cb49-437"></a></span>
<span id="cb49-440"><a href="#cb49-440"></a><span class="in">```{r}</span></span>
<span id="cb49-441"><a href="#cb49-441"></a><span class="fu">TukeyHSD</span>(<span class="fu">aov</span>(weight <span class="sc">~</span> Diet <span class="sc">*</span> <span class="fu">as.factor</span>(Time),</span>
<span id="cb49-442"><a href="#cb49-442"></a>             <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">20</span>, <span class="dv">21</span>))))</span>
<span id="cb49-443"><a href="#cb49-443"></a><span class="in">```</span></span>
<span id="cb49-444"><a href="#cb49-444"></a></span>
<span id="cb49-445"><a href="#cb49-445"></a>:::: {.callout-important}</span>
<span id="cb49-446"><a href="#cb49-446"></a><span class="fu">## Task G.9: Do it now!</span></span>
<span id="cb49-447"><a href="#cb49-447"></a>Yikes! That's a massive amount of results. What does all of this mean, and why is it so verbose? **(/5)**</span>
<span id="cb49-448"><a href="#cb49-448"></a>::::</span>
<span id="cb49-449"><a href="#cb49-449"></a></span>
<span id="cb49-450"><a href="#cb49-450"></a><span class="co">&lt;!--- #### About interaction terms ---&gt;</span></span>
<span id="cb49-451"><a href="#cb49-451"></a></span>
<span id="cb49-452"><a href="#cb49-452"></a><span class="co">&lt;!-- AJS to insert stuff here --&gt;</span></span>
<span id="cb49-453"><a href="#cb49-453"></a></span>
<span id="cb49-454"><a href="#cb49-454"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb49-455"><a href="#cb49-455"></a></span>
<span id="cb49-456"><a href="#cb49-456"></a><span class="fu">## Summary</span></span>
<span id="cb49-457"><a href="#cb49-457"></a>To summarise *t*-tests, single-factor (1-way) and multifactor (2- or 3-way, etc.) ANOVAs:</span>
<span id="cb49-458"><a href="#cb49-458"></a></span>
<span id="cb49-459"><a href="#cb49-459"></a><span class="ss">1. </span>A *t*-test is applied to situations where one wants to compare the means of only **two** groups of a response variable within **one categorical independent variable** (we say a factor with two levels).</span>
<span id="cb49-460"><a href="#cb49-460"></a></span>
<span id="cb49-461"><a href="#cb49-461"></a><span class="ss">2. </span>A 1-way ANOVA also looks at the means of a response variable belonging to **one categorical independent variable**, but the categorical response variable has **more than two** levels in it.</span>
<span id="cb49-462"><a href="#cb49-462"></a></span>
<span id="cb49-463"><a href="#cb49-463"></a><span class="ss">3. </span>Following on from there, a 2-way ANOVA compares the means of response variables belonging to all the levels within **two categorical independent variables** (e.g. Factor 1 might have three levels, and Factor 2 five levels). In the simplest formulaton, it does so by looking at the **main effects**, which is the group differences between the three levels of Factor 1 and disregarding the contribution due to the group membership to Factor 2, and also the group differences amongst the levels of Factor 2 but disregarding the group membership of Factor 1. In addition to looking at the main effects, a 2-way ANOVA can also consider the **interaction** (or combined effect) of Factors 1 and 2 in influencing the means.</span>
<span id="cb49-464"><a href="#cb49-464"></a>:::</span>
<span id="cb49-465"><a href="#cb49-465"></a></span>
<span id="cb49-466"><a href="#cb49-466"></a><span class="fu"># Alternatives to ANOVA {#sec-alt}</span></span>
<span id="cb49-467"><a href="#cb49-467"></a></span>
<span id="cb49-468"><a href="#cb49-468"></a>In the first main section of this chapter we learned how to test hypotheses based on the comparisons of means between sets of data when we were able to meet our two base assumptions. These parametric tests are preferred over non-parametric tests because they are more robust. However, when we simply aren't able to meet these assumptions we must not despair. Non-parametric tests are still useful. In this chapter we will learn how to run non-parametric tests for two sample and multiple sample datasets. To start, let's load our libraries and <span class="in">`chicks`</span> data if we have not already.</span>
<span id="cb49-469"><a href="#cb49-469"></a></span>
<span id="cb49-472"><a href="#cb49-472"></a><span class="in">```{r}</span></span>
<span id="cb49-473"><a href="#cb49-473"></a><span class="co"># First activate libraries</span></span>
<span id="cb49-474"><a href="#cb49-474"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb49-475"><a href="#cb49-475"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb49-476"><a href="#cb49-476"></a></span>
<span id="cb49-477"><a href="#cb49-477"></a><span class="co"># Then load data</span></span>
<span id="cb49-478"><a href="#cb49-478"></a>chicks <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(ChickWeight)</span>
<span id="cb49-479"><a href="#cb49-479"></a><span class="in">```</span></span>
<span id="cb49-480"><a href="#cb49-480"></a></span>
<span id="cb49-481"><a href="#cb49-481"></a>With our libraries and data loaded, let's find a day in which at least one of our assumptions are violated.</span>
<span id="cb49-482"><a href="#cb49-482"></a></span>
<span id="cb49-485"><a href="#cb49-485"></a><span class="in">```{r}</span></span>
<span id="cb49-486"><a href="#cb49-486"></a><span class="co"># Then check for failing assumptions</span></span>
<span id="cb49-487"><a href="#cb49-487"></a>chicks <span class="sc">%&gt;%</span> </span>
<span id="cb49-488"><a href="#cb49-488"></a>  <span class="fu">filter</span>(Time <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-489"><a href="#cb49-489"></a>  <span class="fu">group_by</span>(Diet) <span class="sc">%&gt;%</span> </span>
<span id="cb49-490"><a href="#cb49-490"></a>  <span class="fu">summarise</span>(<span class="at">norm_wt =</span> <span class="fu">as.numeric</span>(<span class="fu">shapiro.test</span>(weight)[<span class="dv">2</span>]),</span>
<span id="cb49-491"><a href="#cb49-491"></a>            <span class="at">var_wt =</span> <span class="fu">var</span>(weight))</span>
<span id="cb49-492"><a href="#cb49-492"></a><span class="in">```</span></span>
<span id="cb49-493"><a href="#cb49-493"></a></span>
<span id="cb49-494"><a href="#cb49-494"></a><span class="fu">## Wilcoxon rank sum test</span></span>
<span id="cb49-495"><a href="#cb49-495"></a></span>
<span id="cb49-496"><a href="#cb49-496"></a>The non-parametric version of a *t*-test is a Wilcox rank sum test. To perform this test in R we may again use <span class="in">`compare_means()`</span> and specify the test we want:</span>
<span id="cb49-497"><a href="#cb49-497"></a></span>
<span id="cb49-500"><a href="#cb49-500"></a><span class="in">```{r}</span></span>
<span id="cb49-501"><a href="#cb49-501"></a><span class="fu">compare_means</span>(weight <span class="sc">~</span> Diet,</span>
<span id="cb49-502"><a href="#cb49-502"></a>              <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">0</span>,</span>
<span id="cb49-503"><a href="#cb49-503"></a>                            Diet <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>)),</span>
<span id="cb49-504"><a href="#cb49-504"></a>              <span class="at">method =</span> <span class="st">"wilcox.test"</span>)</span>
<span id="cb49-505"><a href="#cb49-505"></a><span class="in">```</span></span>
<span id="cb49-506"><a href="#cb49-506"></a></span>
<span id="cb49-507"><a href="#cb49-507"></a>What do our results show?</span>
<span id="cb49-508"><a href="#cb49-508"></a></span>
<span id="cb49-509"><a href="#cb49-509"></a><span class="fu">## Kruskall-Wallis rank sum test</span></span>
<span id="cb49-510"><a href="#cb49-510"></a></span>
<span id="cb49-511"><a href="#cb49-511"></a><span class="fu">### Single factor</span></span>
<span id="cb49-512"><a href="#cb49-512"></a></span>
<span id="cb49-513"><a href="#cb49-513"></a>The non-parametric version of an ANOVA is a Kruskall-Wallis rank sum test. As you may have by now surmised, this may be done with <span class="in">`compare_means()`</span> as seen below:</span>
<span id="cb49-514"><a href="#cb49-514"></a></span>
<span id="cb49-517"><a href="#cb49-517"></a><span class="in">```{r}</span></span>
<span id="cb49-518"><a href="#cb49-518"></a><span class="fu">compare_means</span>(weight <span class="sc">~</span> Diet,</span>
<span id="cb49-519"><a href="#cb49-519"></a>              <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">0</span>),</span>
<span id="cb49-520"><a href="#cb49-520"></a>              <span class="at">method =</span> <span class="st">"kruskal.test"</span>)</span>
<span id="cb49-521"><a href="#cb49-521"></a><span class="in">```</span></span>
<span id="cb49-522"><a href="#cb49-522"></a></span>
<span id="cb49-523"><a href="#cb49-523"></a>As with the ANOVA, this first step with the Kruskall-Wallis test is not the last. We must again run a post-hoc test on our results. This time we will need to use <span class="in">`pgirmess::kruskalmc()`</span>, which means we will need to load a new library.</span>
<span id="cb49-524"><a href="#cb49-524"></a></span>
<span id="cb49-527"><a href="#cb49-527"></a><span class="in">```{r}</span></span>
<span id="cb49-528"><a href="#cb49-528"></a><span class="fu">library</span>(pgirmess)</span>
<span id="cb49-529"><a href="#cb49-529"></a></span>
<span id="cb49-530"><a href="#cb49-530"></a><span class="fu">kruskalmc</span>(weight <span class="sc">~</span> Diet, <span class="at">data =</span> <span class="fu">filter</span>(chicks, Time <span class="sc">==</span> <span class="dv">0</span>))</span>
<span id="cb49-531"><a href="#cb49-531"></a><span class="in">```</span></span>
<span id="cb49-532"><a href="#cb49-532"></a></span>
<span id="cb49-533"><a href="#cb49-533"></a>Let's consult the help file for <span class="in">`kruskalmc()`</span> to understand what this print-out means.</span>
<span id="cb49-534"><a href="#cb49-534"></a></span>
<span id="cb49-535"><a href="#cb49-535"></a><span class="fu">### Multiple factors</span></span>
<span id="cb49-536"><a href="#cb49-536"></a></span>
<span id="cb49-537"><a href="#cb49-537"></a>The water becomes murky quickly when one wants to perform multiple factor non-parametric comparison of means tests. To that end, we will not cover the few existing methods here. Rather, one should avoid the necessity for these types of tests when designing an experiment.</span>
<span id="cb49-538"><a href="#cb49-538"></a></span>
<span id="cb49-539"><a href="#cb49-539"></a><span class="fu">## The SA time data</span></span>
<span id="cb49-540"><a href="#cb49-540"></a></span>
<span id="cb49-541"><a href="#cb49-541"></a><span class="in">```{r anova-plot7}</span></span>
<span id="cb49-542"><a href="#cb49-542"></a><span class="co">#| fig-cap: "Time is not a limited resource in South Africa."</span></span>
<span id="cb49-543"><a href="#cb49-543"></a><span class="co">#| label: fig-anova-plot6</span></span>
<span id="cb49-544"><a href="#cb49-544"></a></span>
<span id="cb49-545"><a href="#cb49-545"></a>sa_time <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(<span class="fu">read_csv</span>(<span class="st">"../../data/snakes.csv"</span>,</span>
<span id="cb49-546"><a href="#cb49-546"></a>                              <span class="at">col_types =</span> <span class="fu">list</span>(<span class="fu">col_double</span>(),</span>
<span id="cb49-547"><a href="#cb49-547"></a>                                               <span class="fu">col_double</span>(),</span>
<span id="cb49-548"><a href="#cb49-548"></a>                                               <span class="fu">col_double</span>())))</span>
<span id="cb49-549"><a href="#cb49-549"></a>sa_time_long <span class="ot">&lt;-</span> sa_time <span class="sc">%&gt;%</span> </span>
<span id="cb49-550"><a href="#cb49-550"></a>  <span class="fu">gather</span>(<span class="at">key =</span> <span class="st">"term"</span>, <span class="at">value =</span> <span class="st">"minutes"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-551"><a href="#cb49-551"></a>  <span class="fu">filter</span>(minutes <span class="sc">&lt;</span> <span class="dv">300</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb49-552"><a href="#cb49-552"></a>  <span class="fu">mutate</span>(<span class="at">term =</span> <span class="fu">as.factor</span>(term))</span>
<span id="cb49-553"><a href="#cb49-553"></a></span>
<span id="cb49-554"><a href="#cb49-554"></a>my_comparisons <span class="ot">&lt;-</span> <span class="fu">list</span>( <span class="fu">c</span>(<span class="st">"now"</span>, <span class="st">"now_now"</span>),</span>
<span id="cb49-555"><a href="#cb49-555"></a>                        <span class="fu">c</span>(<span class="st">"now_now"</span>, <span class="st">"just_now"</span>),</span>
<span id="cb49-556"><a href="#cb49-556"></a>                        <span class="fu">c</span>(<span class="st">"now"</span>, <span class="st">"just_now"</span>) )</span>
<span id="cb49-557"><a href="#cb49-557"></a></span>
<span id="cb49-558"><a href="#cb49-558"></a><span class="fu">ggboxplot</span>(sa_time_long, <span class="at">x =</span> <span class="st">"term"</span>, <span class="at">y =</span> <span class="st">"minutes"</span>,</span>
<span id="cb49-559"><a href="#cb49-559"></a>          <span class="at">color =</span> <span class="st">"term"</span>, <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">"#00AFBB"</span>, <span class="st">"#E7B800"</span>, <span class="st">"#FC4E07"</span>),</span>
<span id="cb49-560"><a href="#cb49-560"></a>          <span class="at">add =</span> <span class="st">"jitter"</span>, <span class="at">shape =</span> <span class="st">"term"</span>)</span>
<span id="cb49-561"><a href="#cb49-561"></a><span class="in">```</span></span>
<span id="cb49-562"><a href="#cb49-562"></a></span>
<span id="cb49-563"><a href="#cb49-563"></a><span class="fu"># Examples</span></span>
<span id="cb49-564"><a href="#cb49-564"></a></span>
<span id="cb49-565"><a href="#cb49-565"></a><span class="fu">## Snakes!</span></span>
<span id="cb49-566"><a href="#cb49-566"></a></span>
<span id="cb49-567"><a href="#cb49-567"></a>These data could be analysed by a two-way ANOVA without replication, or a repeated measures ANOVA. Here I will analyse it by using a two-way ANOVA without replication.</span>
<span id="cb49-568"><a href="#cb49-568"></a></span>
<span id="cb49-569"><a href="#cb49-569"></a>Place and Abramson (2008) placed diamondback rattlesnakes (*Crotalus atrox*) in a 'rattlebox,' a box with a lid that would slide open and shut every 5 minutes. At first, the snake would rattle its tail each time the box opened. After a while, the snake would become habituated to the box opening and stop rattling its tail. They counted the number of box openings until a snake stopped rattling; fewer box openings means the snake was more quickly habituated. They repeated this experiment on each snake on four successive days, which is treated as an influential variable here. Place and Abramson (2008) used 10 snakes, but some of them never became habituated; to simplify this example, data from the six snakes that did become habituated on each day are used.</span>
<span id="cb49-570"><a href="#cb49-570"></a></span>
<span id="cb49-571"><a href="#cb49-571"></a>First, we read in the data, making sure to convert the column named <span class="in">`day`</span> to a factor. Why? Because ANOVAs work with factor independent variables, while <span class="in">`day`</span> as it is encoded by default is in fact a continuous variable.</span>
<span id="cb49-572"><a href="#cb49-572"></a></span>
<span id="cb49-575"><a href="#cb49-575"></a><span class="in">```{r}</span></span>
<span id="cb49-576"><a href="#cb49-576"></a>snakes <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../../data/snakes.csv"</span>)</span>
<span id="cb49-577"><a href="#cb49-577"></a>snakes<span class="sc">$</span>day <span class="ot">=</span> <span class="fu">as.factor</span>(snakes<span class="sc">$</span>day)</span>
<span id="cb49-578"><a href="#cb49-578"></a><span class="in">```</span></span>
<span id="cb49-579"><a href="#cb49-579"></a></span>
<span id="cb49-580"><a href="#cb49-580"></a>The first thing we do is to create some summaries of the data. Refer to the summary statistics Chapter.</span>
<span id="cb49-581"><a href="#cb49-581"></a></span>
<span id="cb49-584"><a href="#cb49-584"></a><span class="in">```{r}</span></span>
<span id="cb49-585"><a href="#cb49-585"></a>snakes.summary <span class="ot">&lt;-</span> snakes <span class="sc">%&gt;%</span> </span>
<span id="cb49-586"><a href="#cb49-586"></a>  <span class="fu">group_by</span>(day, snake) <span class="sc">%&gt;%</span> </span>
<span id="cb49-587"><a href="#cb49-587"></a>  <span class="fu">summarise</span>(<span class="at">mean_openings =</span> <span class="fu">mean</span>(openings),</span>
<span id="cb49-588"><a href="#cb49-588"></a>            <span class="at">sd_openings =</span> <span class="fu">sd</span>(openings)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-589"><a href="#cb49-589"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb49-590"><a href="#cb49-590"></a>snakes.summary</span>
<span id="cb49-591"><a href="#cb49-591"></a><span class="in">```</span></span>
<span id="cb49-592"><a href="#cb49-592"></a></span>
<span id="cb49-593"><a href="#cb49-593"></a>:::: {.callout-important}</span>
<span id="cb49-594"><a href="#cb49-594"></a><span class="fu">## Task G.9: Do it now!</span></span>
<span id="cb49-595"><a href="#cb49-595"></a><span class="ss">- </span>Something seems... off. What's going on here? Please explain this outcome.</span>
<span id="cb49-596"><a href="#cb49-596"></a>::::</span>
<span id="cb49-597"><a href="#cb49-597"></a></span>
<span id="cb49-598"><a href="#cb49-598"></a>To fix this problem, let us ignore the grouping by both <span class="in">`snake`</span> and <span class="in">`day`</span>.</span>
<span id="cb49-599"><a href="#cb49-599"></a></span>
<span id="cb49-602"><a href="#cb49-602"></a><span class="in">```{r}</span></span>
<span id="cb49-603"><a href="#cb49-603"></a>snakes.summary <span class="ot">&lt;-</span> snakes <span class="sc">%&gt;%</span> </span>
<span id="cb49-604"><a href="#cb49-604"></a>  <span class="fu">group_by</span>(day) <span class="sc">%&gt;%</span> </span>
<span id="cb49-605"><a href="#cb49-605"></a>  <span class="fu">summarise</span>(<span class="at">mean_openings =</span> <span class="fu">mean</span>(openings),</span>
<span id="cb49-606"><a href="#cb49-606"></a>            <span class="at">sd_openings =</span> <span class="fu">sd</span>(openings)) <span class="sc">%&gt;%</span> </span>
<span id="cb49-607"><a href="#cb49-607"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb49-608"><a href="#cb49-608"></a>snakes.summary</span>
<span id="cb49-609"><a href="#cb49-609"></a><span class="in">```</span></span>
<span id="cb49-610"><a href="#cb49-610"></a></span>
<span id="cb49-611"><a href="#cb49-611"></a><span class="in">`Rmisc::summarySE()`</span> offers a convenience function if your feeling less frisky about calculating the summary statistics yourself:</span>
<span id="cb49-612"><a href="#cb49-612"></a></span>
<span id="cb49-615"><a href="#cb49-615"></a><span class="in">```{r}</span></span>
<span id="cb49-616"><a href="#cb49-616"></a><span class="fu">library</span>(Rmisc)</span>
<span id="cb49-617"><a href="#cb49-617"></a>snakes.summary2 <span class="ot">&lt;-</span> <span class="fu">summarySE</span>(<span class="at">data =</span> snakes,</span>
<span id="cb49-618"><a href="#cb49-618"></a>                             <span class="at">measurevar =</span> <span class="st">"openings"</span>,</span>
<span id="cb49-619"><a href="#cb49-619"></a>                             <span class="at">groupvars =</span> <span class="fu">c</span>(<span class="st">"day"</span>))</span>
<span id="cb49-620"><a href="#cb49-620"></a>snakes.summary2</span>
<span id="cb49-621"><a href="#cb49-621"></a><span class="in">```</span></span>
<span id="cb49-622"><a href="#cb49-622"></a></span>
<span id="cb49-623"><a href="#cb49-623"></a>Now we turn to some visual data summaries (@fig-anova-plot).</span>
<span id="cb49-624"><a href="#cb49-624"></a></span>
<span id="cb49-627"><a href="#cb49-627"></a><span class="in">```{r}</span></span>
<span id="cb49-628"><a href="#cb49-628"></a><span class="co">#| fig-cap: "Boxplots showing the change in the snakes' habituation to box opening over time."</span></span>
<span id="cb49-629"><a href="#cb49-629"></a><span class="co">#| label: fig-anova-plot</span></span>
<span id="cb49-630"><a href="#cb49-630"></a></span>
<span id="cb49-631"><a href="#cb49-631"></a><span class="fu">ggplot</span>(<span class="at">data =</span> snakes, <span class="fu">aes</span>(<span class="at">x =</span> day, <span class="at">y =</span> openings)) <span class="sc">+</span></span>
<span id="cb49-632"><a href="#cb49-632"></a>  <span class="fu">geom_segment</span>(<span class="at">data =</span> snakes.summary2, <span class="fu">aes</span>(<span class="at">x =</span> day, <span class="at">xend =</span> day,</span>
<span id="cb49-633"><a href="#cb49-633"></a>                                           <span class="at">y =</span> openings <span class="sc">-</span> ci,</span>
<span id="cb49-634"><a href="#cb49-634"></a>                                           <span class="at">yend =</span> openings <span class="sc">+</span> ci,</span>
<span id="cb49-635"><a href="#cb49-635"></a>                                           <span class="at">colour =</span> day),</span>
<span id="cb49-636"><a href="#cb49-636"></a>              <span class="at">size =</span> <span class="fl">2.0</span>, <span class="at">linetype =</span> <span class="st">"solid"</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb49-637"><a href="#cb49-637"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> day), <span class="at">alpha =</span> <span class="fl">0.3</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span> </span>
<span id="cb49-638"><a href="#cb49-638"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.05</span>) <span class="sc">+</span></span>
<span id="cb49-639"><a href="#cb49-639"></a>  <span class="fu">theme_pubclean</span>()</span>
<span id="cb49-640"><a href="#cb49-640"></a><span class="in">```</span></span>
<span id="cb49-641"><a href="#cb49-641"></a></span>
<span id="cb49-642"><a href="#cb49-642"></a>What are our null hypotheses?</span>
<span id="cb49-643"><a href="#cb49-643"></a></span>
<span id="cb49-644"><a href="#cb49-644"></a><span class="ss">1.  </span>***H*~0~** There is no difference between snakes with respect to the number of openings at which they habituate.</span>
<span id="cb49-645"><a href="#cb49-645"></a><span class="ss">2.  </span>***H*~0~** There is no difference between days in terms of the number of openings at which the snakes habituate.</span>
<span id="cb49-646"><a href="#cb49-646"></a></span>
<span id="cb49-647"><a href="#cb49-647"></a>Fit the ANOVA model to test these hypotheses:</span>
<span id="cb49-648"><a href="#cb49-648"></a></span>
<span id="cb49-651"><a href="#cb49-651"></a><span class="in">```{r}</span></span>
<span id="cb49-652"><a href="#cb49-652"></a>snakes.aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(openings <span class="sc">~</span> day <span class="sc">+</span> snake, <span class="at">data =</span> snakes)</span>
<span id="cb49-653"><a href="#cb49-653"></a><span class="fu">summary</span>(snakes.aov)</span>
<span id="cb49-654"><a href="#cb49-654"></a><span class="in">```</span></span>
<span id="cb49-655"><a href="#cb49-655"></a></span>
<span id="cb49-656"><a href="#cb49-656"></a>Now we need to test of the assumptions hold true (i.e. erros are normally distributed and heteroscedastic) (@fig-anova-plot5). Also, where are the differences (@fig-tukey)?</span>
<span id="cb49-657"><a href="#cb49-657"></a></span>
<span id="cb49-660"><a href="#cb49-660"></a><span class="in">```{r}</span></span>
<span id="cb49-661"><a href="#cb49-661"></a><span class="co">#| fig-cap: "Exploring the assumptions visually."</span></span>
<span id="cb49-662"><a href="#cb49-662"></a><span class="co">#| label: fig-anova-plot5</span></span>
<span id="cb49-663"><a href="#cb49-663"></a></span>
<span id="cb49-664"><a href="#cb49-664"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb49-665"><a href="#cb49-665"></a><span class="co"># Checking assumptions...</span></span>
<span id="cb49-666"><a href="#cb49-666"></a><span class="co"># make a histogram of the residuals;</span></span>
<span id="cb49-667"><a href="#cb49-667"></a><span class="co"># they must be normal</span></span>
<span id="cb49-668"><a href="#cb49-668"></a>snakes.res <span class="ot">&lt;-</span> <span class="fu">residuals</span>(snakes.aov)</span>
<span id="cb49-669"><a href="#cb49-669"></a><span class="fu">hist</span>(snakes.res, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb49-670"><a href="#cb49-670"></a></span>
<span id="cb49-671"><a href="#cb49-671"></a><span class="co"># make a plot of residuals and the fitted values;</span></span>
<span id="cb49-672"><a href="#cb49-672"></a><span class="co"># # they must be normal and homoscedastic</span></span>
<span id="cb49-673"><a href="#cb49-673"></a><span class="fu">plot</span>(<span class="fu">fitted</span>(snakes.aov), <span class="fu">residuals</span>(snakes.aov), <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb49-674"><a href="#cb49-674"></a><span class="in">```</span></span>
<span id="cb49-675"><a href="#cb49-675"></a></span>
<span id="cb49-678"><a href="#cb49-678"></a><span class="in">```{r}</span></span>
<span id="cb49-679"><a href="#cb49-679"></a><span class="co">#| fig-cap: "Exploring the differences between days."</span></span>
<span id="cb49-680"><a href="#cb49-680"></a><span class="co">#| label: fig-tukey</span></span>
<span id="cb49-681"><a href="#cb49-681"></a></span>
<span id="cb49-682"><a href="#cb49-682"></a>snakes.tukey <span class="ot">&lt;-</span> <span class="fu">TukeyHSD</span>(snakes.aov, <span class="at">which =</span> <span class="st">"day"</span>, <span class="at">conf.level =</span> <span class="fl">0.90</span>)</span>
<span id="cb49-683"><a href="#cb49-683"></a><span class="fu">plot</span>(snakes.tukey, <span class="at">las =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">"red"</span>)</span>
<span id="cb49-684"><a href="#cb49-684"></a><span class="in">```</span></span>
<span id="cb49-685"><a href="#cb49-685"></a></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>© 2025, AJ Smit</p>
</div>   
    <div class="nav-footer-center">

<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB744/basic_stats/08-anova.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>


<script src="../../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>