<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.361">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="AJ Smit">
<meta name="dcterms.date" content="2021-01-01">
<title>The Tangled Bank - 2. Working with data and code</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../BCB744/intro_r/03-workflow.html" rel="next">
<link href="../../BCB744/intro_r/01-RStudio.html" rel="prev">
<link href="../../images/favicon.ico" rel="icon">
<script src="../../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-SCYH68M489"></script><script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-SCYH68M489', { 'anonymize_ip': true});
</script><script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"simple",
  "consent_type":"implied",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary","functionality","tracking","targeting"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  });
});
</script><style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script><link rel="stylesheet" href="../../styles.css">
</head>
<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg navbar-dark "><div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">The Tangled Bank</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../../index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../about.html" rel="" target="">
 <span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-vignettes" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Vignettes</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-vignettes">
<li>
    <a class="dropdown-item" href="../../vignettes/chl_ERDDAP.html" rel="" target="">
 <span class="dropdown-text">Retrieving Chlorophyll-a Data from ERDDAP Servers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_sightings.html" rel="" target="">
 <span class="dropdown-text">Plotting the Whale Sightings and Chlorophyll-<em>a</em> Concentrations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_localisation.html" rel="" target="">
 <span class="dropdown-text">Spatial Localisation, Subsetting, and Aggregation of the Chlorophyll-<em>a</em> Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/prep_NOAA_OISST.html" rel="" target="">
 <span class="dropdown-text">Downloading and Prep: ERDDAP</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/MHW_basic_detection.html" rel="" target="">
 <span class="dropdown-text">Basic Detection and Visualisation of Events</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/gridded_data.html" rel="" target="">
 <span class="dropdown-text">Detecting Events in Gridded Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/alt_method.html" rel="" target="">
 <span class="dropdown-text">Downloading and Prep: Python + CDO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/elem_ts_methods.html" rel="" target="">
 <span class="dropdown-text">Wavelet Analysis of Diatom Time Series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/MHW_MCS_horizonplots.html" rel="" target="">
 <span class="dropdown-text">Displaying MHWs and MCSs as horizon plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/heatwaveR_issues.html" rel="" target="">
 <span class="dropdown-text">heatwaveR issues</span></a>
  </li>  
    </ul>
</li>
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/ajsmit" rel="" target=""><i class="bi bi-twitter" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">
<li>
    <a class="dropdown-item" href="https://github.com/ajsmit/tangled_bank" rel="" target="">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://bugs.com" rel="" target="">
 <span class="dropdown-text">Report a Bug</span></a>
  </li>  
    </ul>
</li>
</ul>
<div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../BCB744/BCB744_index.html">BCB744 (Intro R)</a></li><li class="breadcrumb-item"><a href="../../BCB744/intro_r/02-working-with-data.html">2. Working with data and code</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-center sidebar-header">
      <a href="../../index.html" class="sidebar-logo-link">
      <img src="../../images/hex_logo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none"></a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
 <span class="menu-text">☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">UNDERGRADUATE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
 <span class="menu-text">BDC223</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pages/BDC223_FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
 <span class="menu-text">BDC334</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/BDC334_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BDC334: Macroecology and Global Biogeography</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1: Ecological Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/02a-r_rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2a: R and RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/02b-env_dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2b: Environmental Distance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/03-biodiversity1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3: Quantifying Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/04-biodiversity2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4: Describing Biodiversity Patterns</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pages/Transboundary_systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transboundary systems</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS CORE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">BCB744 (Intro R)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/BCB744_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744: Introduction to R, and Biostatistics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/01-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. R and RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/02-working-with-data.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">2. Working with data and code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/03-workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. R workflows</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/04-graphics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Graphics with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/05-faceting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Faceting figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/06-brewing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Brewing colours</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/07-mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Mapping with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/08-mapping_style.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Mapping with style</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/09-mapping_rnaturalearth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Mapping with Natural Earth and the <strong>sf</strong> package</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/10-mapping_quakes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. The Fiji Earthquake Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/11-tidy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Tidy data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/12-tidier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Tidier data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/13-tidiest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Tidiest data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/14-recap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Recap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/15-functions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15. Functions by chapter</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/16-base_r.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16. Base R primer</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/17-dates.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">17. Dates</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false">
 <span class="menu-text">BCB744 (Biostatistics)</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/01-data-in-R.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Data classes and structures in R</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/02-summarise-and-describe.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Exploring with summaries and descriptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/03-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Exploring with figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/04-distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/05-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Statistical inference and hypothesis testing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/06-assumptions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Assumptions for parametric statistics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/07-t_tests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. <em>t</em>-tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/08-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Analysis of Variance (ANOVA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/09-regressions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Simple linear regressions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/10-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/11-glance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Non-parametric statistical tests at a glance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/12-confidence.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Confidence intervals</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS ELECTIVES</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">
 <span class="menu-text">BCB743</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/BCB743_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB743: Quantitative Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/00-review.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1-4. Review Biogeography and Global Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/05-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Correlations and Associations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/06-deep_dive.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Deep Dive into Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/07-ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/08-PCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8a. Principal Component Analysis (PCA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/08-PCA_examples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8b. PCA Additional Examples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/09-CA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Correspondence Analysis (CA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/10-PCoA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Principal Coordinate Analysis (PCoA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/11-nMDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11a. non-Metric Multidimensional Scaling (nMDS)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/11-nMDS_diatoms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11b. nMDS of Mayombo’s Diatom Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/12-constrained_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Constrained Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/13-cluster_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Cluster Analysis</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">SUPPORT</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
 <span class="menu-text">Web Resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/general_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/ecology_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quantitative Ecology Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/spatial_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial R Resources</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘︎</span>
  </li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> Introduction</a></li>
  <li>
<a href="#types-of-data-files" id="toc-types-of-data-files" class="nav-link" data-scroll-target="#types-of-data-files"><span class="header-section-number">2</span> Types of data files</a>
  <ul class="collapse">
<li><a href="#comma-separated-value-files" id="toc-comma-separated-value-files" class="nav-link" data-scroll-target="#comma-separated-value-files"><span class="header-section-number">2.1</span> Comma separated value files</a></li>
  <li><a href="#tab-separated-value-files" id="toc-tab-separated-value-files" class="nav-link" data-scroll-target="#tab-separated-value-files"><span class="header-section-number">2.2</span> Tab separated value files</a></li>
  <li><a href="#microsoft-excel-files" id="toc-microsoft-excel-files" class="nav-link" data-scroll-target="#microsoft-excel-files"><span class="header-section-number">2.3</span> Microsoft Excel files</a></li>
  <li><a href="#rdata-files" id="toc-rdata-files" class="nav-link" data-scroll-target="#rdata-files"><span class="header-section-number">2.4</span> Rdata files</a></li>
  <li><a href="#other-binary-files" id="toc-other-binary-files" class="nav-link" data-scroll-target="#other-binary-files"><span class="header-section-number">2.5</span> Other binary files</a></li>
  <li><a href="#netcdf-grib-and-hdf-files" id="toc-netcdf-grib-and-hdf-files" class="nav-link" data-scroll-target="#netcdf-grib-and-hdf-files"><span class="header-section-number">2.6</span> NetCDF, Grib, and HDF files</a></li>
  </ul>
</li>
  <li><a href="#file-extensions" id="toc-file-extensions" class="nav-link" data-scroll-target="#file-extensions"><span class="header-section-number">3</span> File extensions</a></li>
  <li><a href="#r-code-conventions" id="toc-r-code-conventions" class="nav-link" data-scroll-target="#r-code-conventions"><span class="header-section-number">4</span> R code conventions</a></li>
  <li><a href="#reproducible-research" id="toc-reproducible-research" class="nav-link" data-scroll-target="#reproducible-research"><span class="header-section-number">5</span> Reproducible research</a></li>
  </ul><div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB744/intro_r/02-working-with-data.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">2. Working with data and code</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
<p class="subtitle lead">Getting familiar with data handling in R</p>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://ajsmit.netlify.app">AJ Smit</a> <a href="https://orcid.org/0000-0002-3799-6126" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
    <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uwc.ac.za">
            University of the Western Cape
            </a>
          </p>
      </div>
    </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 1, 2021</p>
    </div>
  </div>
  
    
  </div>
  

</header><blockquote class="blockquote">
<p><em>“The plural of anecdote is not data.”</em></p>
<p>— Roger Brinner</p>
</blockquote>
<p>In this Chapter we will cover:</p>
<ul>
<li>Data file formats</li>
<li>The use of file extensions</li>
<li>Review some R code conventions</li>
<li>Introduce the concept of reproducible research</li>
</ul>
<section id="introduction" class="level1" data-number="1"><h1 data-number="1">
<span class="header-section-number">1</span> Introduction</h1>
<p>In today’s rapidly evolving world of biological and ecological research, understanding the basic principles of coding and data handling has become an indispensable skill for students. As the complexity and volume of data in these fields continue to grow, traditional manual methods of data analysis are increasingly insufficient to handle the demands of modern research. By acquiring coding and data handling skills, students in biology and ecology can unlock the full potential of computational tools and techniques, allowing them to analyse complex datasets, model integrated systems, and uncover hidden patterns that may not be apparent through conventional methods. Moreover, proficiency in coding and data handling empowers students to collaborate effectively with multidisciplinary teams, draw on data from various sources in different formats, and bridge the gap between theoretical knowledge and practical applications, and contribute to the advancement of scientific knowledge in a transparent, meaningful, and efficient manner.</p>
</section><section id="types-of-data-files" class="level1" data-number="2"><h1 data-number="2">
<span class="header-section-number">2</span> Types of data files</h1>
<p>R will read in many types of data, including spreadsheets, text files, binary files, and files from other statistical packages.</p>
<section id="comma-separated-value-files" class="level2" data-number="2.1"><h2 data-number="2.1" class="anchored" data-anchor-id="comma-separated-value-files">
<span class="header-section-number">2.1</span> Comma separated value files</h2>
<p>CSV stands for ‘Comma Separated Value’. A CSV file is a simple text file that stores data in a tabular format, with each row representing a record and each column representing a field of data. In a CSV file, each data value is separated by a comma (or sometimes another delimiter such as a semicolon or tab), and each row is terminated by a new line. But we should not assume that CSV files adhere to the <a href="https://r4ds.had.co.nz/tidy-data.html">tidy principles</a>!</p>
<p>CSV files are widely used in data analysis and can be opened and edited by most spreadsheet software, such as MS Excel and Google Sheets. Being comprised of plain text (ASCII), they are often used to import and export data between different applications or systems, as they provide a standardised format that can be easily parsed by software.</p>
<p>CSV files are easy to create and use, and they have the advantage of being lightweight and easy to read and write by both humans and machines. However, they can be limited in their ability to represent complex data structures or to handle large amounts of data efficiently. Additionally, if our data contain certain kinds of special characters, this can cause problems with parsing the file correctly.</p>
<p>We will most frequently use the functions <code><a href="https://rdrr.io/r/utils/read.table.html">read.csv()</a></code> or <code><a href="https://readr.tidyverse.org/reference/read_delim.html">readr::read_csv()</a></code> (and related forms) for reading in CSV data. We can write CSV files to disk with the <code><a href="https://rdrr.io/r/utils/write.table.html">write.csv()</a></code> or <code><a href="https://readr.tidyverse.org/reference/write_delim.html">readr::write_csv()</a></code> commands. For very large datasets that might take a long time to read in or save, <code><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html">data.table::fread()</a></code> or <code><a href="https://Rdatatable.gitlab.io/data.table/reference/fwrite.html">data.table::fwrite()</a></code> are faster alternatives to the aforementioned base R or <strong>tidyverse</strong> options. Even faster options are <code><a href="https://rdrr.io/pkg/feather/man/read_feather.html">feather::read_feather()</a></code> and <code><a href="https://rdrr.io/pkg/feather/man/read_feather.html">feather::write_feather()</a></code>; although <strong>feather</strong> saves tabular data, the format is not actually an ASCII CSV, however.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
ASCII files
</div>
</div>
<div class="callout-body-container callout-body">
<p>ASCII stands for “American Standard Code for Information Interchange”. An ASCII file is a plain text file that contains ASCII characters. ASCII is a character encoding standard that assigns a unique numeric code to each character, including letters, numbers, punctuation, and other symbols commonly used in the English language.</p>
<p>ASCII files are the most basic type of text file and are supported by virtually all operating systems and applications. We can create and edit ASCII files using any text editor, such as Notepad, TextEdit, or <a href="https://code.visualstudio.com/">VS Code</a>. ASCII files are typically used for storing and sharing simple text-based information, such as program source code, configuration files, and other types of data that do not require special formatting or rich media content.</p>
<p>ASCII files are limited in their ability to represent non-English characters or symbols that are not included in the ASCII character set. To handle these types of characters, other character encoding standards such as UTF-8 or Unicode are used. However, ASCII files remain an important and widely used format for storing and sharing simple text-based data.</p>
</div>
</div>
</section><section id="tab-separated-value-files" class="level2" data-number="2.2"><h2 data-number="2.2" class="anchored" data-anchor-id="tab-separated-value-files">
<span class="header-section-number">2.2</span> Tab separated value files</h2>
<p>The primary difference between a ‘tab-separated value’ (TSV) file and a ‘comma-separated values’ (CSV) file lies in the delimiter used to separate data fields. Both file formats are plain text ASCII files used to store data in a tabular format, but they employ different characters to distinguish individual fields within each row.</p>
<p>In a TSV file, the fields are separated by tab characters (represented as <code>\t</code> in many programming languages). This format is particularly useful when dealing with data that includes commas within the values, as it avoids potential conflicts and parsing issues.</p>
<p>We already saw that in a CSV file, the fields are separated by commas. CSV files are more common and widely supported than TSV files. However, they can present difficulties when the data itself contains commas, potentially causing confusion between actual field separators and commas within the data. To mitigate this issue, values containing commas are often enclosed in quotation marks.</p>
<p>Like CSV files, TSV can also be imported into and exported from spreadsheet software like Excel, or read and manipulated using programming languages like Python, R, and many others. The choice between TSV and CSV largely depends on the nature of the data and personal preferences, but it’s crucial to be aware of the delimiter used in order to accurately parse the files. The same functions that read or write CSV files in R can be used for TSV, but one has to set the arguments <code>sep = "\t"</code> or <code>delim = "\t"</code> for the functions <code><a href="https://rdrr.io/r/utils/read.table.html">read.csv()</a></code> and <code>read_csv()</code> respectively.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Missing values and CSV and TSV files
</div>
</div>
<div class="callout-body-container callout-body">
<p>Where we have missing data (blanks), the CSV format separates these by commas with empty field in-between. However, there can be problems with blanks if we read in a space-delimited format file. If we are having trouble reading in missing data as blanks, try replacing them in the spreadsheet with <code>NA</code>, the missing data code in R. In Excel, highlight the area of the spreadsheet that includes all the cells we need to fill with <code>NA</code>. Do an ‘Edit/Replace…’ and leave the ‘Find what:’ textbox blank and in the ‘Replace with:’ textbox enter <code>NA</code>. Once imported into R, the <code>NA</code> values will be recognised as missing data.</p>
</div>
</div>
</section><section id="microsoft-excel-files" class="level2" data-number="2.3"><h2 data-number="2.3" class="anchored" data-anchor-id="microsoft-excel-files">
<span class="header-section-number">2.3</span> Microsoft Excel files</h2>
<p>Microsoft Excel files are a type of file format that is used to store data in a tabular form, much like CSV files. However, Excel files are proprietary and are specifically designed to work with Excel software. Excel files can contain more advanced formatting features such as colours, fonts, and formulas, which make them a popular choice for people who like embellishments. But, as much as I dislike Excel as a software for data analysis, Excel files are definitely a good option for data entry.</p>
<p>Using MS Excel for data analysis can be a terrible idea for a number of reasons:</p>
<ul>
<li><p><strong>Compatibility</strong> Excel files may not be compatible with all data science tools and programming languages. For example, R cannot read Excel files directly.</p></li>
<li><p><strong>Data integrity</strong> Excel files can be prone to errors and inconsistencies in the data. For example, if a user changes a formula or formatting, it could affect the entire dataset. Also, it is possible for Excel to change the data types of certain columns, or to mix the class of data <em>within a column</em>, which can cause issues with data processing and analysis.</p></li>
<li><p><strong>File size</strong> Excel files can quickly become very large when dealing with large datasets, which can lead to performance issues and storage problems.</p></li>
<li><p><strong>Version control</strong> Excel files can make it difficult to keep track of changes and versions of the data, particularly when multiple people are working on the same file.</p></li>
</ul>
<p>In contrast, CSV files are a simple, lightweight, and widely supported file format that can be easily used with most data science tools and programming languages. CSV files are also less prone to errors and inconsistencies than Excel files, making them a more reliable choice for data science tasks.</p>
<p>So, while Excel files may be useful for certain tasks such as initial data entry, they are generally not recommended for use in data science due to their potential for errors (see box “Well-known Excel errors”), incompatibility, and other issues. I recommend exporting data deliberately to CSV files. This not only avoids complications, but also allows us to unambiguously identify the data we based our analysis on. This last statement should give us the hint that it is good practice to name our <code>.csv</code> slightly differently each time we export it from Excel, perhaps by appending a reference to the date it was exported. Also, for those of us who use commas in Excel as the decimal separator, or to separate 1000s, undo these features now.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Well-known Excel errors
</div>
</div>
<div class="callout-body-container callout-body">
<p>Excel is a widely used spreadsheet application, but it has been responsible for several serious errors in data analysis, science, and data science. Some of these errors include:</p>
<ul>
<li><p><strong>Gene name errors</strong> In 2016, researchers discovered that Excel automatically converted gene symbols to dates or floating-point numbers. For example, gene symbols like SEPT2 (Septin 2) were converted to “2-Sep” and gene symbols like MARCH1 (Membrane Associated Ring-CH-Type Finger 1) were converted to “1-Mar”. This led to errors and inconsistencies in genetic data, affecting nearly 20% of published papers in leading genomic journals.</p></li>
<li><p><strong>Reinhart-Rogoff controversy</strong> In 2010, economists Carmen Reinhart and Kenneth Rogoff published a paper arguing that high levels of public debt were associated with lower economic growth. Their findings influenced policy decisions worldwide. However, in 2013, other researchers found that Reinhart and Rogoff’s results were affected by an Excel spreadsheet error that excluded some data points, causing them to overstate the relationship between debt and growth.</p></li>
<li><p><strong>London Whale incident</strong> In 2012, JPMorgan Chase, a leading financial institution, suffered a trading loss of over $6 billion, partially due to an Excel error. The bank’s model for calculating the risk of their trades, implemented in Excel, used incorrect formulas that significantly underestimated the risk involved. The event, which became known as the “London Whale” incident, highlighted the potential consequences of relying on Excel for complex financial models.</p></li>
<li><p><strong>Truncation of large numbers</strong> Excel can handle only a limited number of digits for large numbers, truncating any value that exceeds this limit. This truncation has lead to a loss of precision and inaccurate calculations in scientific and data analysis contexts, where exact values were important.</p></li>
<li><p><strong>Issues with floating-point arithmetic</strong> Excel uses floating-point arithmetic, which can cause rounding errors and imprecise results when working with very large or very small numbers. These inaccuracies can lead to incorrect conclusions or predictions in data analysis and scientific research.</p></li>
</ul>
<p>These errors underscore the importance of using specialised tools and software, such as R or Python, for data analysis and scientific research. While Excel can be useful for basic tasks and smaller datasets, its limitations can lead to serious consequences in more complex situations.</p>
</div>
</div>
</section><section id="rdata-files" class="level2" data-number="2.4"><h2 data-number="2.4" class="anchored" data-anchor-id="rdata-files">
<span class="header-section-number">2.4</span> Rdata files</h2>
<p>Rdata files are a file format used by the R programming language to store data objects. These files can contain any type of R object, such as vectors, matrices, dataframes, lists, and more. Rdata files are binary files, which means they are not human-readable like text files such as CSV files. Binary R data files have a <code>.rda</code> or <code>.Rdata</code> file extension and can be created or read using the <code><a href="https://rdrr.io/r/base/save.html">save()</a></code> and <code><a href="https://rdrr.io/r/base/load.html">load()</a></code>, respectively, functions in R.</p>
<p>Rdata files are convenient for a number of reasons:</p>
<ul>
<li><p><strong>Efficient storage</strong> Rdata files can be more compact (they can be compressed) and efficient than other file formats, such as CSV files, because they are stored in a binary format. This means they take up less disk space and can be read and written to faster.</p></li>
<li><p><strong>Easy access to R objects</strong> Rdata files make it easy to save and load R objects, which can be useful for preserving data objects for future analysis or sharing them with others. This is especially useful for complex datasets or objects that would be difficult to recreate.</p></li>
<li><p><strong>Preserve metadata</strong> Rdata files can preserve metadata such as variable names, row and column names, and other attributes of R objects. This makes it easier to work with the data objects in the future without having to recreate this metadata.</p></li>
<li><p><strong>Convenient for reproducibility</strong> Rdata files can be used to save and load data objects as part of a reproducible research workflow. This can help ensure that data objects are preserved and can be easily accessed in the future, even if the data sources or code have changed.</p></li>
</ul>
<p>On the downside, they can only be used within R, making them a less than ideal proposition when you intend sharing your data with colleagues who sadly do not use R.</p>
</section><section id="other-binary-files" class="level2" data-number="2.5"><h2 data-number="2.5" class="anchored" data-anchor-id="other-binary-files">
<span class="header-section-number">2.5</span> Other binary files</h2>
<p>As a biostatistician, you may encounter several other binary data files in your work. Such binary data files may be software-specific and can be used to store large datasets or data objects that are not easily represented in a text format. For example, a binary data file might contain a large matrix or array of numeric data that would be difficult to store in a text file. Binary data files can also be used to store images, audio files, and other types of data that are not represented as text.</p>
<p>One common type of binary data file that you may encounter as a statistician is a SAS data file. SAS is a statistical software package that is widely used in data analysis, and SAS data files are a binary format used to store datasets in SAS. These files typically have a <code>.sas7bdat</code> file extension and contain metadata such as variable names and formats in addition to the data itself. Another type of binary data file you may encounter is a binary <code>.mat</code> data file, which is a file format used to store Matlab data.</p>
<p>When working with binary data files, it is important to be aware of the specific format of the file and the tools and software needed to read and manipulate the data. Some statistical software packages may have built-in functions for reading and writing certain types of binary data files, while others may require additional libraries or packages.</p>
</section><section id="netcdf-grib-and-hdf-files" class="level2" data-number="2.6"><h2 data-number="2.6" class="anchored" data-anchor-id="netcdf-grib-and-hdf-files">
<span class="header-section-number">2.6</span> NetCDF, Grib, and HDF files</h2>
<p>NetCDF, HDF, and GRIB are all file formats commonly used in the scientific and research communities to store and share large and complex datasets. While CSV files are a simple and widely used format, they can become impractical for large datasets with complex structures or metadata. Here’s a brief overview of each file format:</p>
<ul>
<li><p><strong>NetCDF (Network Common Data Form)</strong> is a binary file format that is designed for storing and sharing scientific data. It can store multidimensional arrays and metadata, such as variable names and units, in a self-describing format. NetCDF files are commonly used in fields such as atmospheric science, oceanography, and climate modelling.</p></li>
<li><p><strong>HDF (Hierarchical Data Format)</strong> is a file format that is designed to store and organise large and complex data structures. It can store a wide variety of data types, including multidimensional arrays, tables, and hierarchical data. HDF files are commonly used in fields such as remote sensing, astronomy, and engineering.</p></li>
<li><p><strong>GRIB (GRIdded Binary)</strong> is a binary file format used to store meteorological and oceanographic data. It can store gridded data, such as atmospheric or oceanic model output, in a compact and efficient binary format. GRIB files are commonly used by weather forecasting agencies and research organisations.</p></li>
</ul>
<p>Compared to CSV files, these file formats offer several benefits for storing and sharing complex datasets:</p>
<ul>
<li><p><strong>Support for multidimensional arrays</strong> These file formats can store and handle multidimensional arrays, which cannot be represented in a CSV file.</p></li>
<li><p><strong>Efficient storage</strong> Binary file formats can be more compact and efficient than text-based formats such as CSV files, which can save disk space and make it easier to share and transfer large datasets.</p></li>
<li><p><strong>Memory use efficiency</strong> NetCDF, GRIB, and HDF files are better for memory use efficiency compared to CSV files because they can store multidimensional arrays and metadata in a compact binary format, which can save disk space and memory when working with large and complex datasets. Also, they do not have to be read into memory all at once.</p></li>
<li><p><strong>Self-describing metadata</strong> These file formats can include metadata, such as variable names and units, which are self-describing and can be easily accessed and understood by other researchers and software.</p></li>
<li><p><strong>Support for compression</strong> Binary file formats can support compression, which can further reduce file size and make it easier to share and transfer large datasets.</p></li>
</ul>
<p>The various efficiencies mention above may be offset by them being quite challenging to work with, and as such novices might experience steep learning curves.</p>
</section></section><section id="file-extensions" class="level1" data-number="3"><h1 data-number="3">
<span class="header-section-number">3</span> File extensions</h1>
<p>File extensions are three or four character suffixes added to the end of a filename, usually preceded by a period (e.g., ‘.txt’, ‘.jpg’, ‘.pdf’, , ‘.csv’, ‘.txt’, ‘.xlsx’). These extensions indicate the format or type of the file, providing information about the content and structure of the data within the file. Some file extensions are particular to Windows (such as ‘.exe’) and others to MacOS (e.g.&nbsp;‘.dmg’) but many are transportable across operating systems. File extensions help both the operating system and applications recognise the file type and determine which program or application should be used to open, view, or edit the file.</p>
<p>The default setting for Windows computers is for file extensions to be hidden when files are viewed in the Windows Explorer. This is silly and irresponsible, and a frequent major source of frustration, annoyance, and irritability when I try to help a student with an issue on their computers. Displaying file extensions on your computer is essential for effective data science work because it allows us to quickly identify and manage various file types that we may encounter during our projects.</p>
<p>Some reasons why it’s important to display file extensions if we are to effective use one’s computer in scientific computing applications:</p>
<ul>
<li><p><strong>Differentiate file formats</strong> When analysing data, we often work with multiple file formats, such as CSV, TSV, Excel, JSON, and others. By displaying the file extensions we can easily differentiate between these formats, ensuring that we use the appropriate tools and methods to process and analyse the data.</p></li>
<li><p><strong>Prevent errors</strong> Knowing the exact file type helps us avoid errors when importing or exporting data, as different formats require specific handling techniques. For instance, using a CSV parser on a TSV file could lead to incorrect results.</p></li>
<li><p><strong>Improve file organisation</strong> When working on complex projects, it’s crucial that we maintain an organised file structure. Being able to see file extensions at a glance helps us keep track of various data files, scripts, and output files, making it easier to manage our project and ensure its reproducibility.</p></li>
<li><p><strong>Enhance security</strong> Displaying file extensions can also helps us identify potentially malicious files disguised as legitimate ones. For example, a file with an extension ‘.txt.exe’ might appear to be a harmless text file if the extension is hidden, but it is actually an executable file that could be harmful.</p></li>
<li><p><strong>Facilitate collaboration</strong> When sharing files with colleagues or collaborators, knowing the file format is essential for smooth communication and seamless collaboration. Displaying file extensions ensures that everyone is aware of the file types being used and can handle them accordingly.</p></li>
</ul>
<p>To display file extensions on your computer, we will need to adjust the settings in our operating system’s file explorer. The specific steps to do this vary depending on whether we’re using Windows, macOS, or a Linux-based system, but a quick online search will provide us with the necessary instructions. Since most people use Windows, I have included the instructions in the box “Displaying file extension in Windows”.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Displaying file extension in Windows
</div>
</div>
<div class="callout-body-container callout-body">
<p>Be a responsible adult and display the file extensions on your computers. To display file extensions in Windows Explorer on the latest version of Windows (assuming Windows 11), follow these steps:</p>
<ol type="1">
<li>Open a new File Explorer window by clicking on the File Explorer icon on the taskbar or pressing the ‘Windows key + E’ on your keyboard.</li>
<li>In the File Explorer window, click on the three horizontal dots in the upper-right corner to open the ‘More options’ menu.</li>
<li>From the ‘More options’ menu, click on ‘Options’ to open the ‘Folder Options’ window.</li>
<li>In the ‘Folder Options’ window, switch to the ‘View’ tab.</li>
<li>Under the ‘Advanced settings’ list, locate the option ‘Hide extensions for known file types’ and uncheck the box next to it.</li>
<li>Click ‘Apply’ to save your changes, and then click ‘OK’ to close the ‘Folder Options’ window.</li>
</ol>
<p>Now, the file extensions should be visible for all files in Windows Explorer. Remember that this setting applies to all folders on your computer. If you want to revert to hiding the file extensions, simply follow the same steps and check the box next to ‘Hide extensions for known file types’ in the ‘View’ tab of the ‘Folder Options’ window.</p>
</div>
</div>
</section><section id="r-code-conventions" class="level1" data-number="4"><h1 data-number="4">
<span class="header-section-number">4</span> R code conventions</h1>
<p>When writing code in support of a statistical data analysis in R, it’s important that we adhere to code conventions to ensure our code is tidy, reproducible, and unambiguous. Here are some general guidelines to follow:</p>
<ul>
<li><p><strong>Use a consistent coding style</strong> Adopt a consistent coding style throughout your codebase to make it easy for others to read and understand your code. R has a widely-accepted style guide called <a href="https://style.tidyverse.org/">The tidyverse style guide</a>, which you can follow. Also peruse <a href="http://adv-r.had.co.nz/Style.html">R Style Guide</a> or <a href="https://google.github.io/styleguide/Rguide.html">Google’s R Style Guide</a> with opinionated views of how to use spaces around <code>&lt;-</code>, after <code>#</code>, after <code>,</code>, and various other stylistics principles, with more guidelines you can adopt to if you want to produce well-structured, well-formatted, and legible code.</p></li>
<li><p><strong>Write meaningful variable and function names</strong> Choose clear, descriptive names for your variables and functions that reflect their purpose. I use lowercase letters and separate words with underscores (e.g., <code>average_temperature</code>), but other options are available—see “Naming Variables” below.</p></li>
<li><p><strong>Indentation and spacing</strong> Use consistent indentation (usually 2 or 4 spaces) to improve code readability. Place spaces around operators, commas, and after semicolons in control structures to make your code easier to read.</p></li>
<li><p><strong>Comment and document your code</strong> Provide comments throughout your code to explain the purpose of particular sections, especially when the code is complex or not self-explanatory. Use the <code>#</code> symbol to write single-line comments.</p></li>
<li><p><strong>Modularise your code</strong> Break your code into smaller, reusable functions that perform specific tasks. This makes your code more manageable and easier to understand.</p></li>
<li><p><strong>Splitting lines of code</strong> If you have long lines of code, then you can spread them over multiple lines. You just have to make sure that R knows something is coming, either by leaving a bracket open, or ending the line with an operator (<code>+</code> in <code>ggplot()</code> or the pipe operator <code>%&gt;%</code> in <strong>dplyr</strong> code.</p></li>
<li><p><strong>Use version control</strong> Utilise version control systems like <a href="https://github.com/">GitHub</a> to keep track of your code changes and collaborate with others. This practice ensures reproducibility and makes it easy to revert to a previous version if needed.</p></li>
<li><p><strong>Organise your project</strong> Keep your project files organised in a logical structure. A common approach is to have separate directories for data, scripts, and output. There is also the <a href="https://workflowr.github.io/workflowr/index.html"><strong>workflowr</strong></a> package designed specifically for magaging data analysis projects.</p></li>
<li><p><strong>Explicitly declare dependencies</strong> List and load all the required packages at the beginning of your script. This makes it clear which packages are needed and helps avoid potential conflicts.</p></li>
<li><p><strong>Be mindful of absolute vs.&nbsp;relative paths</strong> Use relative paths whenever possible, as this makes your code more portable and reproducible across different systems.</p></li>
<li><p><strong>Write tests</strong> Implement tests to check the correctness of your code and ensure it works as expected. This practice helps catch bugs early and makes your code more robust.</p></li>
</ul>
<p>It’s important to note that different programming languages and communities may have their own conventions, and it’s always a good idea to follow the established conventions to make your code more readable and understandable for others. But, whatever you decide to use, be consistent throughout your code. By following guidelines consistently, you will produce clean, well-organised code that is easy to understand and maintain, which is crucial for any data analysis project.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Naming variables
</div>
</div>
<div class="callout-body-container callout-body">
<p>R has pedantic requirements for naming variables. It is safest to not use spaces, special characters (e.g., commas, semicolons, any of the shift characters above the numbers), or function names (e.g., mean). In fact, I strongly suggest never to use spaces in variables, filenames, etc.</p>
<p>In programming, variable names are an essential aspect of writing readable and maintainable code. Good variable naming conventions can make code easier to understand and can even help prevent bugs by making it clear what the variable represents. Here are some commonly used naming conventions and their examples:</p>
<ul>
<li><p><strong>Pascal Case</strong> (also known as upper camel case) is a convention where the first letter of each word in a variable name is capitalised. This is often used in object-oriented programming to name classes and types. Examples of Pascal case are <code>SpeciesName</code>, <code>PetalColour</code>, and <code>FunctionalType</code>.</p></li>
<li><p><strong>Snake Case</strong> (also known as ‘snake_case’ or ‘lowercase with underscores’) is a convention where words in a variable name are separated by underscores and all letters are lowercase. This is commonly used in scripting languages and file naming conventions. Examples of Snake case are <code>species_name</code>, <code>petal_colour</code>, and <code>functional_type</code>.</p></li>
<li><p><strong>Camel Case</strong> (also known as ‘lower camel case’) is a convention where the first letter of each word in a variable name is capitalised, except for the first word. This is commonly used in object-oriented programming to name variables and methods. Examples of Camel case are <code>speciesName</code>, <code>petalColour</code>, and <code>functionalType</code>.</p></li>
<li><p><strong>Hungarian Notation</strong> is a naming convention where a prefix is added to the variable name to indicate its data type. This convention is less commonly used in modern programming languages. Examples of Hungarian Notation are <code>iSpeciesID</code> (where ‘i’ indicates an integer), <code>strPetalColour</code> (where ‘str’ indicates a string), and <code>fSeawaterTemperature</code> (where ‘f’ indicates a floating-point number).</p></li>
</ul>
</div>
</div>
</section><section id="reproducible-research" class="level1" data-number="5"><h1 data-number="5">
<span class="header-section-number">5</span> Reproducible research</h1>
<p>Reproducible research is a critical aspect of scientific integrity and advancement and it is increasingly being applied in the fields of biological and ecological sciences. These disciplines often involve complex analyses and data sets that require meticulous documentation with meta-data, and validation. Reproducible research ensures that results can be independently verified, repeated at a later date by our future selves and the reader audience of scientific publications, thereby improving transparency and minimising errors and biases. An example of some serious consequences stemming from lack of clear communication and transparency is presented in the box “The Mars Climate Orbiter mission in 1998”. Reproducible research practices foster collaboration, and promote the growth of scientific knowledge.</p>
<p>The R and RStudio ‘ecosystem’ offers a comprehensive suite of tools and utilities that facilitate the implementation of reproducible research practices. Key features of this suit of tools include:</p>
<ul>
<li><p><strong>Open-source platform</strong> R is an open-source programming language with a vast and continually growing library of packages that cater to various research needs. The open-source nature of R promotes transparency and allows us to inspect, reproduce, and modify code to suit our specific requirements.</p></li>
<li><p><strong>Version control</strong> Integrating R projects with version control systems like <a href="https://git-scm.com/">Git</a> and platforms like <a href="https://github.com/">GitHub</a> or <a href="https://about.gitlab.com/">GitLab</a> allows us to track changes in our code, share our work, and collaborate more effectively with our colleagues. This practice enables a clear record of code evolution and ensures the reproducibility of analyses across different stages of a project.</p></li>
<li><p><strong>Literate programming</strong> <a href="https://rmarkdown.rstudio.com/">R Markdown</a> and <a href="https://quarto.org/">Quarto</a>, powerful tools within the RStudio IDE, enable us to seamlessly integrate our code, results, and narrative text in a single document. This promotes transparency, documentation, and reproducibility by combining data analysis, visualisation, and interpretation in a human-readable format.</p></li>
<li><p><strong>Dynamic report generation</strong> R Markdown and Quarto documents can be converted into various output formats, such as HTML, PDF, or Word, facilitating the creation of dynamic, interactive reports and presentations. This feature ensures that any changes to our data or analyses are automatically updated in the final output, maintaining the consistency and accuracy of the results.</p></li>
<li><p><strong>Workflow management</strong> Tools like <a href="https://workflowr.github.io/workflowr/index.html"><strong>workflowr</strong></a> and <a href="https://docs.ropensci.org/targets/"><strong>targets</strong></a> provide us with robust and efficient workflow management solutions for large and complex projects. These packages streamline the process of organising, executing, and caching analytical pipelines, making it easier to reproduce and validate our results.</p></li>
<li><p><strong>Containerisation</strong> Although beyond the scope of BCB744, it is worth being aware of technologies such as <a href="https://www.docker.com/">Docker</a> that we can use in conjunction with R and RStudio to create isolated, reproducible environments with specific package versions and dependencies. This enhances portability and ensures that our analyses can be consistently reproduced across different systems.</p></li>
<li><p><strong>Data management</strong> R provides several packages for data import, manipulation, and storage, within the <a href="https://www.tidyverse.org/"><strong>tidyverse</strong></a> suite of packages. By employing these packages, we can create clean, well-structured data sets that are easier to analyse, understand, and share.</p></li>
<li><p><strong>Code testing and validation</strong> Packages like <a href="https://testthat.r-lib.org/"><strong>testthat</strong></a> enable us to develop and execute ‘unit tests’ for R functions, ensuring the reliability and accuracy of our code. This helps us to identify and resolve potential issues before they impact the overall reproducibility of our research findings.</p></li>
</ul>
<p>By using these resources, we can ensure the reliability, transparency, and validity of our research efforts, ultimately contributing to the advancement of our respective fields by richly reporting on the vast array of ‘lost’ knowledge and learning that takes place behind the scenes and never make it into peer reviewed publications’ Methods sections.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
The Mars Climate Orbiter mission in 1998
</div>
</div>
<div class="callout-body-container callout-body">
<p>One of the most famous examples of unclear communication is the Mars Climate Orbiter mission in 1998. NASA lost the spacecraft due to a navigation error caused by a unit conversion mistake. The error occurred because one team used metric units (newtons), while another team used imperial units (pound-force) for a crucial spacecraft operation parameter. The discrepancy in units led to incorrect trajectory calculations, causing the Mars Climate Orbiter to approach Mars at a dangerously low altitude and ultimately disintegrate in the Martian atmosphere.</p>
<p>This incident highlights the importance of clear communication, proper data handling, and rigorous verification processes in mission-critical systems, including space missions. The event emphasises the need for using appropriate tools, software, and methodologies to minimise the risk of errors in complex engineering projects.</p>
</div>
</div>


<!-- -->

</section><div id="quarto-appendix" class="default"><section class="quarto-appendix-contents"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div id="quarto-reuse" class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by/4.0/">https://creativecommons.org/licenses/by/4.0/</a></div></div></section><section class="quarto-appendix-contents"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{smit2021,
  author = {Smit, AJ},
  title = {2. {Working} with Data and Code},
  date = {2021-01-01},
  url = {https://tangledbank.netlify.app/BCB744/intro_r/02-working-with-data.html},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-smit2021" class="csl-entry quarto-appendix-citeas" role="listitem">
Smit A (2021) 2. Working with data and code. <a href="https://tangledbank.netlify.app/BCB744/intro_r/02-working-with-data.html">https://tangledbank.netlify.app/BCB744/intro_r/02-working-with-data.html.
</a>
</div></div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><script src="https://utteranc.es/client.js" repo="ajsmit/tangled_bank" issue-term="pathname" theme="github-light" crossorigin="anonymous" async="">
</script><nav class="page-navigation column-body"><div class="nav-page nav-page-previous">
      <a href="../../BCB744/intro_r/01-RStudio.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">1. R and RStudio</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../BCB744/intro_r/03-workflow.html" class="pagination-link">
        <span class="nav-page-text">3. R workflows</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb1" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "2021-01-01"</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "2. Working with data and code"</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "Getting familiar with data handling in R"</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; *"The plural of anecdote is not data."*</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; --- Roger Brinner</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="in">```{r prelim-opts, echo=FALSE}</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">comment =</span> <span class="st">"R&gt;"</span>, </span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">warning =</span> <span class="cn">FALSE</span>, </span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">message =</span> <span class="cn">FALSE</span>,</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">out.width =</span> <span class="st">'70%'</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>In this Chapter we will cover:</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Data file formats</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>The use of file extensions</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Review some R code conventions</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Introduce the concept of reproducible research</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="fu"># Introduction</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>In today's rapidly evolving world of biological and ecological research, understanding the basic principles of coding and data handling has become an indispensable skill for students. As the complexity and volume of data in these fields continue to grow, traditional manual methods of data analysis are increasingly insufficient to handle the demands of modern research. By acquiring coding and data handling skills, students in biology and ecology can unlock the full potential of computational tools and techniques, allowing them to analyse complex datasets, model integrated systems, and uncover hidden patterns that may not be apparent through conventional methods. Moreover, proficiency in coding and data handling empowers students to collaborate effectively with multidisciplinary teams, draw on data from various sources in different formats, and bridge the gap between theoretical knowledge and practical applications, and contribute to the advancement of scientific knowledge in a transparent, meaningful, and efficient manner.</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="fu"># Types of data files</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>R will read in many types of data, including spreadsheets, text files, binary files, and files from other statistical packages.</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="fu">## Comma separated value files</span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>CSV stands for 'Comma Separated Value'. A CSV file is a simple text file that stores data in a tabular format, with each row representing a record and each column representing a field of data. In a CSV file, each data value is separated by a comma (or sometimes another delimiter such as a semicolon or tab), and each row is terminated by a new line. But we should not assume that CSV files adhere to the <span class="co">[</span><span class="ot">tidy principles</span><span class="co">](https://r4ds.had.co.nz/tidy-data.html)</span>! </span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>CSV files are widely used in data analysis and can be opened and edited by most spreadsheet software, such as MS Excel and Google Sheets. Being comprised of plain text (ASCII), they are often used to import and export data between different applications or systems, as they provide a standardised format that can be easily parsed by software.</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>CSV files are easy to create and use, and they have the advantage of being lightweight and easy to read and write by both humans and machines. However, they can be limited in their ability to represent complex data structures or to handle large amounts of data efficiently. Additionally, if our data contain certain kinds of special characters, this can cause problems with parsing the file correctly.</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>We will most frequently use the functions <span class="in">`read.csv()`</span> or <span class="in">`readr::read_csv()`</span> (and related forms) for reading in CSV data. We can write CSV files to disk with the <span class="in">`write.csv()`</span> or <span class="in">`readr::write_csv()`</span> commands. For very large datasets that might take a long time to read in or save, <span class="in">`data.table::fread()`</span> or <span class="in">`data.table::fwrite()`</span> are faster alternatives to the aforementioned base R or **tidyverse** options. Even faster options are `feather::read_feather()` and `feather::write_feather()`; although **feather** saves tabular data, the format is not actually an ASCII CSV, however.</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a><span class="fu">## ASCII files</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>ASCII stands for "American Standard Code for Information Interchange". An ASCII file is a plain text file that contains ASCII characters. ASCII is a character encoding standard that assigns a unique numeric code to each character, including letters, numbers, punctuation, and other symbols commonly used in the English language.</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a>ASCII files are the most basic type of text file and are supported by virtually all operating systems and applications. We can create and edit ASCII files using any text editor, such as Notepad, TextEdit, or <span class="co">[</span><span class="ot">VS Code</span><span class="co">](https://code.visualstudio.com/)</span>. ASCII files are typically used for storing and sharing simple text-based information, such as program source code, configuration files, and other types of data that do not require special formatting or rich media content.</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a>ASCII files are limited in their ability to represent non-English characters or symbols that are not included in the ASCII character set. To handle these types of characters, other character encoding standards such as UTF-8 or Unicode are used. However, ASCII files remain an important and widely used format for storing and sharing simple text-based data.</span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="fu">## Tab separated value files</span></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>The primary difference between a 'tab-separated value' (TSV) file and a 'comma-separated values' (CSV) file lies in the delimiter used to separate data fields. Both file formats are plain text ASCII files used to store data in a tabular format, but they employ different characters to distinguish individual fields within each row.</span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a>In a TSV file, the fields are separated by tab characters (represented as <span class="in">`\t`</span> in many programming languages). This format is particularly useful when dealing with data that includes commas within the values, as it avoids potential conflicts and parsing issues.</span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a>We already saw that in a CSV file, the fields are separated by commas. CSV files are more common and widely supported than TSV files. However, they can present difficulties when the data itself contains commas, potentially causing confusion between actual field separators and commas within the data. To mitigate this issue, values containing commas are often enclosed in quotation marks.</span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a>Like CSV files, TSV can also be imported into and exported from spreadsheet software like Excel, or read and manipulated using programming languages like Python, R, and many others. The choice between TSV and CSV largely depends on the nature of the data and personal preferences, but it's crucial to be aware of the delimiter used in order to accurately parse the files. The same functions that read or write CSV files in R can be used for TSV, but one has to set the arguments <span class="in">`sep = "\t"`</span> or <span class="in">`delim = "\t"`</span> for the functions <span class="in">`read.csv()`</span> and <span class="in">`read_csv()`</span> respectively.</span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a><span class="fu">## Missing values and CSV and TSV files</span></span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>Where we have missing data (blanks), the CSV format separates these by commas with empty field in-between. However, there can be problems with blanks if we read in a space-delimited format file. If we are having trouble reading in missing data as blanks, try replacing them in the spreadsheet with <span class="in">`NA`</span>, the missing data code in R. In Excel, highlight the area of the spreadsheet that includes all the cells we need to fill with <span class="in">`NA`</span>. Do an 'Edit/Replace...' and leave the 'Find what:' textbox blank and in the 'Replace with:' textbox enter <span class="in">`NA`</span>. Once imported into R, the <span class="in">`NA`</span> values will be recognised as missing data.</span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a><span class="fu">## Microsoft Excel files</span></span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a>Microsoft Excel files are a type of file format that is used to store data in a tabular form, much like CSV files. However, Excel files are proprietary and are specifically designed to work with Excel software. Excel files can contain more advanced formatting features such as colours, fonts, and formulas, which make them a popular choice for people who like embellishments. But, as much as I dislike Excel as a software for data analysis, Excel files are definitely a good option for data entry.</span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a>Using MS Excel for data analysis can be a terrible idea for a number of reasons:</span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Compatibility** Excel files may not be compatible with all data science tools and programming languages. For example, R cannot read Excel files directly.</span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Data integrity** Excel files can be prone to errors and inconsistencies in the data. For example, if a user changes a formula or formatting, it could affect the entire dataset. Also, it is possible for Excel to change the data types of certain columns, or to mix the class of data *within a column*, which can cause issues with data processing and analysis.</span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**File size** Excel files can quickly become very large when dealing with large datasets, which can lead to performance issues and storage problems.</span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Version control** Excel files can make it difficult to keep track of changes and versions of the data, particularly when multiple people are working on the same file.</span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a>In contrast, CSV files are a simple, lightweight, and widely supported file format that can be easily used with most data science tools and programming languages. CSV files are also less prone to errors and inconsistencies than Excel files, making them a more reliable choice for data science tasks.</span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a>So, while Excel files may be useful for certain tasks such as initial data entry, they are generally not recommended for use in data science due to their potential for errors (see box "Well-known Excel errors"), incompatibility, and other issues. I recommend exporting data deliberately to CSV files. This not only avoids complications, but also allows us to unambiguously identify the data we based our analysis on. This last statement should give us the hint that it is good practice to name our <span class="in">`.csv`</span> slightly differently each time we export it from Excel, perhaps by appending a reference to the date it was exported. Also, for those of us who use commas in Excel as the decimal separator, or to separate 1000s, undo these features now.</span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a><span class="fu">## Well-known Excel errors</span></span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a>Excel is a widely used spreadsheet application, but it has been responsible for several serious errors in data analysis, science, and data science. Some of these errors include:</span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Gene name errors** In 2016, researchers discovered that Excel automatically converted gene symbols to dates or floating-point numbers. For example, gene symbols like SEPT2 (Septin 2) were converted to "2-Sep" and gene symbols like MARCH1 (Membrane Associated Ring-CH-Type Finger 1) were converted to "1-Mar". This led to errors and inconsistencies in genetic data, affecting nearly 20% of published papers in leading genomic journals.</span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Reinhart-Rogoff controversy** In 2010, economists Carmen Reinhart and Kenneth Rogoff published a paper arguing that high levels of public debt were associated with lower economic growth. Their findings influenced policy decisions worldwide. However, in 2013, other researchers found that Reinhart and Rogoff's results were affected by an Excel spreadsheet error that excluded some data points, causing them to overstate the relationship between debt and growth.</span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**London Whale incident** In 2012, JPMorgan Chase, a leading financial institution, suffered a trading loss of over $6 billion, partially due to an Excel error. The bank's model for calculating the risk of their trades, implemented in Excel, used incorrect formulas that significantly underestimated the risk involved. The event, which became known as the "London Whale" incident, highlighted the potential consequences of relying on Excel for complex financial models.</span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Truncation of large numbers** Excel can handle only a limited number of digits for large numbers, truncating any value that exceeds this limit. This truncation has lead to a loss of precision and inaccurate calculations in scientific and data analysis contexts, where exact values were important.</span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Issues with floating-point arithmetic** Excel uses floating-point arithmetic, which can cause rounding errors and imprecise results when working with very large or very small numbers. These inaccuracies can lead to incorrect conclusions or predictions in data analysis and scientific research.</span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a>These errors underscore the importance of using specialised tools and software, such as R or Python, for data analysis and scientific research. While Excel can be useful for basic tasks and smaller datasets, its limitations can lead to serious consequences in more complex situations.</span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a><span class="fu">## Rdata files</span></span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a>Rdata files are a file format used by the R programming language to store data objects. These files can contain any type of R object, such as vectors, matrices, dataframes, lists, and more. Rdata files are binary files, which means they are not human-readable like text files such as CSV files. Binary R data files have a <span class="in">`.rda`</span> or <span class="in">`.Rdata`</span> file extension and can be created or read using the <span class="in">`save()`</span> and <span class="in">`load()`</span>, respectively, functions in R. </span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a>Rdata files are convenient for a number of reasons:</span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Efficient storage** Rdata files can be more compact (they can be compressed) and efficient than other file formats, such as CSV files, because they are stored in a binary format. This means they take up less disk space and can be read and written to faster.</span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Easy access to R objects** Rdata files make it easy to save and load R objects, which can be useful for preserving data objects for future analysis or sharing them with others. This is especially useful for complex datasets or objects that would be difficult to recreate.</span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Preserve metadata** Rdata files can preserve metadata such as variable names, row and column names, and other attributes of R objects. This makes it easier to work with the data objects in the future without having to recreate this metadata.</span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Convenient for reproducibility** Rdata files can be used to save and load data objects as part of a reproducible research workflow. This can help ensure that data objects are preserved and can be easily accessed in the future, even if the data sources or code have changed.</span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a>On the downside, they can only be used within R, making them a less than ideal proposition when you intend sharing your data with colleagues who sadly do not use R.</span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a><span class="fu">## Other binary files</span></span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a>As a biostatistician, you may encounter several other binary data files in your work. Such binary data files may be software-specific and can be used to store large datasets or data objects that are not easily represented in a text format. For example, a binary data file might contain a large matrix or array of numeric data that would be difficult to store in a text file. Binary data files can also be used to store images, audio files, and other types of data that are not represented as text.</span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a>One common type of binary data file that you may encounter as a statistician is a SAS data file. SAS is a statistical software package that is widely used in data analysis, and SAS data files are a binary format used to store datasets in SAS. These files typically have a <span class="in">`.sas7bdat`</span> file extension and contain metadata such as variable names and formats in addition to the data itself. Another type of binary data file you may encounter is a binary <span class="in">`.mat`</span> data file, which is a file format used to store Matlab data. </span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a>When working with binary data files, it is important to be aware of the specific format of the file and the tools and software needed to read and manipulate the data. Some statistical software packages may have built-in functions for reading and writing certain types of binary data files, while others may require additional libraries or packages.</span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a><span class="fu">## NetCDF, Grib, and HDF files</span></span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a>NetCDF, HDF, and GRIB are all file formats commonly used in the scientific and research communities to store and share large and complex datasets. While CSV files are a simple and widely used format, they can become impractical for large datasets with complex structures or metadata. Here's a brief overview of each file format:</span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**NetCDF (Network Common Data Form)** is a binary file format that is designed for storing and sharing scientific data. It can store multidimensional arrays and metadata, such as variable names and units, in a self-describing format. NetCDF files are commonly used in fields such as atmospheric science, oceanography, and climate modelling.</span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**HDF (Hierarchical Data Format)** is a file format that is designed to store and organise large and complex data structures. It can store a wide variety of data types, including multidimensional arrays, tables, and hierarchical data. HDF files are commonly used in fields such as remote sensing, astronomy, and engineering.</span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**GRIB (GRIdded Binary)** is a binary file format used to store meteorological and oceanographic data. It can store gridded data, such as atmospheric or oceanic model output, in a compact and efficient binary format. GRIB files are commonly used by weather forecasting agencies and research organisations.</span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a>Compared to CSV files, these file formats offer several benefits for storing and sharing complex datasets:</span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Support for multidimensional arrays** These file formats can store and handle multidimensional arrays, which cannot be represented in a CSV file.</span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Efficient storage** Binary file formats can be more compact and efficient than text-based formats such as CSV files, which can save disk space and make it easier to share and transfer large datasets.</span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Memory use efficiency** NetCDF, GRIB, and HDF files are better for memory use efficiency compared to CSV files because they can store multidimensional arrays and metadata in a compact binary format, which can save disk space and memory when working with large and complex datasets. Also, they do not have to be read into memory all at once.</span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Self-describing metadata** These file formats can include metadata, such as variable names and units, which are self-describing and can be easily accessed and understood by other researchers and software.</span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Support for compression** Binary file formats can support compression, which can further reduce file size and make it easier to share and transfer large datasets.</span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a>The various efficiencies mention above may be offset by them being quite challenging to work with, and as such novices might experience steep learning curves.</span>
<span id="cb1-151"><a href="#cb1-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-152"><a href="#cb1-152" aria-hidden="true" tabindex="-1"></a><span class="fu"># File extensions</span></span>
<span id="cb1-153"><a href="#cb1-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-154"><a href="#cb1-154" aria-hidden="true" tabindex="-1"></a>File extensions are three or four character suffixes added to the end of a filename, usually preceded by a period (e.g., '.txt', '.jpg', '.pdf', , '.csv', '.txt', '.xlsx'). These extensions indicate the format or type of the file, providing information about the content and structure of the data within the file. Some file extensions are particular to Windows (such as '.exe') and others to MacOS (e.g. '.dmg') but many are transportable across operating systems. File extensions help both the operating system and applications recognise the file type and determine which program or application should be used to open, view, or edit the file.</span>
<span id="cb1-155"><a href="#cb1-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-156"><a href="#cb1-156" aria-hidden="true" tabindex="-1"></a>The default setting for Windows computers is for file extensions to be hidden when files are viewed in the Windows Explorer. This is silly and irresponsible, and a frequent major source of frustration, annoyance, and irritability when I try to help a student with an issue on their computers. Displaying file extensions on your computer is essential for effective data science work because it allows us to quickly identify and manage various file types that we may encounter during our projects. </span>
<span id="cb1-157"><a href="#cb1-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-158"><a href="#cb1-158" aria-hidden="true" tabindex="-1"></a>Some reasons why it's important to display file extensions if we are to effective use one's computer in scientific computing applications:</span>
<span id="cb1-159"><a href="#cb1-159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-160"><a href="#cb1-160" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Differentiate file formats** When analysing data, we often work with multiple file formats, such as CSV, TSV, Excel, JSON, and others. By displaying the file extensions we can easily differentiate between these formats, ensuring that we use the appropriate tools and methods to process and analyse the data.</span>
<span id="cb1-161"><a href="#cb1-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-162"><a href="#cb1-162" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Prevent errors** Knowing the exact file type helps us avoid errors when importing or exporting data, as different formats require specific handling techniques. For instance, using a CSV parser on a TSV file could lead to incorrect results.</span>
<span id="cb1-163"><a href="#cb1-163" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-164"><a href="#cb1-164" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Improve file organisation** When working on complex projects, it's crucial that we maintain an organised file structure. Being able to see file extensions at a glance helps us keep track of various data files, scripts, and output files, making it easier to manage our project and ensure its reproducibility.</span>
<span id="cb1-165"><a href="#cb1-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-166"><a href="#cb1-166" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Enhance security** Displaying file extensions can also helps us identify potentially malicious files disguised as legitimate ones. For example, a file with an extension '.txt.exe' might appear to be a harmless text file if the extension is hidden, but it is actually an executable file that could be harmful.</span>
<span id="cb1-167"><a href="#cb1-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-168"><a href="#cb1-168" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Facilitate collaboration** When sharing files with colleagues or collaborators, knowing the file format is essential for smooth communication and seamless collaboration. Displaying file extensions ensures that everyone is aware of the file types being used and can handle them accordingly.</span>
<span id="cb1-169"><a href="#cb1-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-170"><a href="#cb1-170" aria-hidden="true" tabindex="-1"></a>To display file extensions on your computer, we will need to adjust the settings in our operating system's file explorer. The specific steps to do this vary depending on whether we're using Windows, macOS, or a Linux-based system, but a quick online search will provide us with the necessary instructions. Since most people use Windows, I have included the instructions in the box "Displaying file extension in Windows".</span>
<span id="cb1-171"><a href="#cb1-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-172"><a href="#cb1-172" aria-hidden="true" tabindex="-1"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb1-173"><a href="#cb1-173" aria-hidden="true" tabindex="-1"></a><span class="fu">## Displaying file extension in Windows</span></span>
<span id="cb1-174"><a href="#cb1-174" aria-hidden="true" tabindex="-1"></a>Be a responsible adult and display the file extensions on your computers. To display file extensions in Windows Explorer on the latest version of Windows (assuming Windows 11), follow these steps:</span>
<span id="cb1-175"><a href="#cb1-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-176"><a href="#cb1-176" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Open a new File Explorer window by clicking on the File Explorer icon on the taskbar or pressing the 'Windows key + E' on your keyboard.</span>
<span id="cb1-177"><a href="#cb1-177" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>In the File Explorer window, click on the three horizontal dots in the upper-right corner to open the 'More options' menu.</span>
<span id="cb1-178"><a href="#cb1-178" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>From the 'More options' menu, click on 'Options' to open the 'Folder Options' window.</span>
<span id="cb1-179"><a href="#cb1-179" aria-hidden="true" tabindex="-1"></a><span class="ss">4. </span>In the 'Folder Options' window, switch to the 'View' tab.</span>
<span id="cb1-180"><a href="#cb1-180" aria-hidden="true" tabindex="-1"></a><span class="ss">5. </span>Under the 'Advanced settings' list, locate the option 'Hide extensions for known file types' and uncheck the box next to it.</span>
<span id="cb1-181"><a href="#cb1-181" aria-hidden="true" tabindex="-1"></a><span class="ss">6. </span>Click 'Apply' to save your changes, and then click 'OK' to close the 'Folder Options' window.</span>
<span id="cb1-182"><a href="#cb1-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-183"><a href="#cb1-183" aria-hidden="true" tabindex="-1"></a>Now, the file extensions should be visible for all files in Windows Explorer. Remember that this setting applies to all folders on your computer. If you want to revert to hiding the file extensions, simply follow the same steps and check the box next to 'Hide extensions for known file types' in the 'View' tab of the 'Folder Options' window.</span>
<span id="cb1-184"><a href="#cb1-184" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb1-185"><a href="#cb1-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-186"><a href="#cb1-186" aria-hidden="true" tabindex="-1"></a><span class="fu"># R code conventions</span></span>
<span id="cb1-187"><a href="#cb1-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-188"><a href="#cb1-188" aria-hidden="true" tabindex="-1"></a>When writing code in support of a statistical data analysis in R, it's important that we adhere to code conventions to ensure our code is tidy, reproducible, and unambiguous. Here are some general guidelines to follow:</span>
<span id="cb1-189"><a href="#cb1-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-190"><a href="#cb1-190" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Use a consistent coding style** Adopt a consistent coding style throughout your codebase to make it easy for others to read and understand your code. R has a widely-accepted style guide called <span class="co">[</span><span class="ot">The tidyverse style guide</span><span class="co">](https://style.tidyverse.org/)</span>, which you can follow. Also peruse <span class="co">[</span><span class="ot">R Style Guide</span><span class="co">](http://adv-r.had.co.nz/Style.html)</span> or <span class="co">[</span><span class="ot">Google's R Style Guide</span><span class="co">](https://google.github.io/styleguide/Rguide.html)</span> with opinionated views of how to use spaces around <span class="in">`&lt;-`</span>, after <span class="in">`#`</span>, after <span class="in">`,`</span>, and various other stylistics principles, with more guidelines you can adopt to if you want to produce well-structured, well-formatted, and legible code.</span>
<span id="cb1-191"><a href="#cb1-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-192"><a href="#cb1-192" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Write meaningful variable and function names** Choose clear, descriptive names for your variables and functions that reflect their purpose. I use lowercase letters and separate words with underscores (e.g., <span class="in">`average_temperature`</span>), but other options are available---see "Naming Variables" below.</span>
<span id="cb1-193"><a href="#cb1-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-194"><a href="#cb1-194" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Indentation and spacing** Use consistent indentation (usually 2 or 4 spaces) to improve code readability. Place spaces around operators, commas, and after semicolons in control structures to make your code easier to read.</span>
<span id="cb1-195"><a href="#cb1-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-196"><a href="#cb1-196" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Comment and document your code** Provide comments throughout your code to explain the purpose of particular sections, especially when the code is complex or not self-explanatory. Use the <span class="in">`#`</span> symbol to write single-line comments.</span>
<span id="cb1-197"><a href="#cb1-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-198"><a href="#cb1-198" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Modularise your code** Break your code into smaller, reusable functions that perform specific tasks. This makes your code more manageable and easier to understand.</span>
<span id="cb1-199"><a href="#cb1-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-200"><a href="#cb1-200" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Splitting lines of code** If you have long lines of code, then you can spread them over multiple lines. You just have to make sure that R knows something is coming, either by leaving a bracket open, or ending the line with an operator (`+` in `ggplot()` or the pipe operator `%&gt;%` in **dplyr** code.</span>
<span id="cb1-201"><a href="#cb1-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-202"><a href="#cb1-202" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Use version control** Utilise version control systems like <span class="co">[</span><span class="ot">GitHub</span><span class="co">](https://github.com/)</span> to keep track of your code changes and collaborate with others. This practice ensures reproducibility and makes it easy to revert to a previous version if needed.</span>
<span id="cb1-203"><a href="#cb1-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-204"><a href="#cb1-204" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Organise your project** Keep your project files organised in a logical structure. A common approach is to have separate directories for data, scripts, and output. There is also the [**workflowr**](https://workflowr.github.io/workflowr/index.html) package designed specifically for magaging data analysis projects.</span>
<span id="cb1-205"><a href="#cb1-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-206"><a href="#cb1-206" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Explicitly declare dependencies** List and load all the required packages at the beginning of your script. This makes it clear which packages are needed and helps avoid potential conflicts.</span>
<span id="cb1-207"><a href="#cb1-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-208"><a href="#cb1-208" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Be mindful of absolute vs. relative paths** Use relative paths whenever possible, as this makes your code more portable and reproducible across different systems.</span>
<span id="cb1-209"><a href="#cb1-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-210"><a href="#cb1-210" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Write tests** Implement tests to check the correctness of your code and ensure it works as expected. This practice helps catch bugs early and makes your code more robust.</span>
<span id="cb1-211"><a href="#cb1-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-212"><a href="#cb1-212" aria-hidden="true" tabindex="-1"></a>It's important to note that different programming languages and communities may have their own conventions, and it's always a good idea to follow the established conventions to make your code more readable and understandable for others. But, whatever you decide to use, be consistent throughout your code. By following guidelines consistently, you will produce clean, well-organised code that is easy to understand and maintain, which is crucial for any data analysis project. </span>
<span id="cb1-213"><a href="#cb1-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-214"><a href="#cb1-214" aria-hidden="true" tabindex="-1"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb1-215"><a href="#cb1-215" aria-hidden="true" tabindex="-1"></a><span class="fu">## Naming variables</span></span>
<span id="cb1-216"><a href="#cb1-216" aria-hidden="true" tabindex="-1"></a>R has pedantic requirements for naming variables. It is safest to not use spaces, special characters (e.g., commas, semicolons, any of the shift characters above the numbers), or function names (e.g., mean). In fact, I strongly suggest never to use spaces in variables, filenames, etc.</span>
<span id="cb1-217"><a href="#cb1-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-218"><a href="#cb1-218" aria-hidden="true" tabindex="-1"></a>In programming, variable names are an essential aspect of writing readable and maintainable code. Good variable naming conventions can make code easier to understand and can even help prevent bugs by making it clear what the variable represents. Here are some commonly used naming conventions and their examples:</span>
<span id="cb1-219"><a href="#cb1-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-220"><a href="#cb1-220" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Pascal Case** (also known as upper camel case) is a convention where the first letter of each word in a variable name is capitalised. This is often used in object-oriented programming to name classes and types. Examples of Pascal case are <span class="in">`SpeciesName`</span>, <span class="in">`PetalColour`</span>, and <span class="in">`FunctionalType`</span>.</span>
<span id="cb1-221"><a href="#cb1-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-222"><a href="#cb1-222" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Snake Case** (also known as 'snake_case' or 'lowercase with underscores') is a convention where words in a variable name are separated by underscores and all letters are lowercase. This is commonly used in scripting languages and file naming conventions. Examples of Snake case are <span class="in">`species_name`</span>, <span class="in">`petal_colour`</span>, and <span class="in">`functional_type`</span>.</span>
<span id="cb1-223"><a href="#cb1-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-224"><a href="#cb1-224" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Camel Case** (also known as 'lower camel case') is a convention where the first letter of each word in a variable name is capitalised, except for the first word. This is commonly used in object-oriented programming to name variables and methods. Examples of Camel case are <span class="in">`speciesName`</span>, <span class="in">`petalColour`</span>, and <span class="in">`functionalType`</span>.</span>
<span id="cb1-225"><a href="#cb1-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-226"><a href="#cb1-226" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Hungarian Notation** is a naming convention where a prefix is added to the variable name to indicate its data type. This convention is less commonly used in modern programming languages. Examples of Hungarian Notation are <span class="in">`iSpeciesID`</span> (where 'i' indicates an integer), <span class="in">`strPetalColour`</span> (where 'str' indicates a string), and <span class="in">`fSeawaterTemperature`</span> (where 'f' indicates a floating-point number).</span>
<span id="cb1-227"><a href="#cb1-227" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb1-228"><a href="#cb1-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-229"><a href="#cb1-229" aria-hidden="true" tabindex="-1"></a><span class="fu"># Reproducible research</span></span>
<span id="cb1-230"><a href="#cb1-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-231"><a href="#cb1-231" aria-hidden="true" tabindex="-1"></a>Reproducible research is a critical aspect of scientific integrity and advancement and it is increasingly being applied in the fields of biological and ecological sciences. These disciplines often involve complex analyses and data sets that require meticulous documentation with meta-data, and validation. Reproducible research ensures that results can be independently verified, repeated at a later date by our future selves and the reader audience of scientific publications, thereby improving transparency and minimising errors and biases. An example of some serious consequences stemming from lack of clear communication and transparency is presented in the box "The Mars Climate Orbiter mission in 1998". Reproducible research practices foster collaboration, and promote the growth of scientific knowledge.</span>
<span id="cb1-232"><a href="#cb1-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-233"><a href="#cb1-233" aria-hidden="true" tabindex="-1"></a>The R and RStudio 'ecosystem' offers a comprehensive suite of tools and utilities that facilitate the implementation of reproducible research practices. Key features of this suit of tools include:</span>
<span id="cb1-234"><a href="#cb1-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-235"><a href="#cb1-235" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Open-source platform** R is an open-source programming language with a vast and continually growing library of packages that cater to various research needs. The open-source nature of R promotes transparency and allows us to inspect, reproduce, and modify code to suit our specific requirements.</span>
<span id="cb1-236"><a href="#cb1-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-237"><a href="#cb1-237" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Version control** Integrating R projects with version control systems like <span class="co">[</span><span class="ot">Git</span><span class="co">](https://git-scm.com/)</span> and platforms like <span class="co">[</span><span class="ot">GitHub</span><span class="co">](https://github.com/)</span> or <span class="co">[</span><span class="ot">GitLab</span><span class="co">](https://about.gitlab.com/)</span> allows us to track changes in our code, share our work, and collaborate more effectively with our colleagues. This practice enables a clear record of code evolution and ensures the reproducibility of analyses across different stages of a project.</span>
<span id="cb1-238"><a href="#cb1-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-239"><a href="#cb1-239" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Literate programming** <span class="co">[</span><span class="ot">R Markdown</span><span class="co">](https://rmarkdown.rstudio.com/)</span> and <span class="co">[</span><span class="ot">Quarto</span><span class="co">](https://quarto.org/)</span>, powerful tools within the RStudio IDE, enable us to seamlessly integrate our code, results, and narrative text in a single document. This promotes transparency, documentation, and reproducibility by combining data analysis, visualisation, and interpretation in a human-readable format.</span>
<span id="cb1-240"><a href="#cb1-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-241"><a href="#cb1-241" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Dynamic report generation** R Markdown and Quarto documents can be converted into various output formats, such as HTML, PDF, or Word, facilitating the creation of dynamic, interactive reports and presentations. This feature ensures that any changes to our data or analyses are automatically updated in the final output, maintaining the consistency and accuracy of the results.</span>
<span id="cb1-242"><a href="#cb1-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-243"><a href="#cb1-243" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Workflow management** Tools like [**workflowr**](https://workflowr.github.io/workflowr/index.html) and [**targets**](https://docs.ropensci.org/targets/) provide us with robust and efficient workflow management solutions for large and complex projects. These packages streamline the process of organising, executing, and caching analytical pipelines, making it easier to reproduce and validate our results.</span>
<span id="cb1-244"><a href="#cb1-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-245"><a href="#cb1-245" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Containerisation** Although beyond the scope of BCB744, it is worth being aware of technologies such as <span class="co">[</span><span class="ot">Docker</span><span class="co">](https://www.docker.com/)</span> that we can use in conjunction with R and RStudio to create isolated, reproducible environments with specific package versions and dependencies. This enhances portability and ensures that our analyses can be consistently reproduced across different systems.</span>
<span id="cb1-246"><a href="#cb1-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-247"><a href="#cb1-247" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Data management** R provides several packages for data import, manipulation, and storage, within the [**tidyverse**](https://www.tidyverse.org/) suite of packages. By employing these packages, we can create clean, well-structured data sets that are easier to analyse, understand, and share.</span>
<span id="cb1-248"><a href="#cb1-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-249"><a href="#cb1-249" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Code testing and validation** Packages like [**testthat**](https://testthat.r-lib.org/) enable us to develop and execute 'unit tests' for R functions, ensuring the reliability and accuracy of our code. This helps us to identify and resolve potential issues before they impact the overall reproducibility of our research findings.</span>
<span id="cb1-250"><a href="#cb1-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-251"><a href="#cb1-251" aria-hidden="true" tabindex="-1"></a>By using these resources, we can ensure the reliability, transparency, and validity of our research efforts, ultimately contributing to the advancement of our respective fields by richly reporting on the vast array of 'lost' knowledge and learning that takes place behind the scenes and never make it into peer reviewed publications' Methods sections.</span>
<span id="cb1-252"><a href="#cb1-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-253"><a href="#cb1-253" aria-hidden="true" tabindex="-1"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb1-254"><a href="#cb1-254" aria-hidden="true" tabindex="-1"></a><span class="fu">## The Mars Climate Orbiter mission in 1998</span></span>
<span id="cb1-255"><a href="#cb1-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-256"><a href="#cb1-256" aria-hidden="true" tabindex="-1"></a>One of the most famous examples of unclear communication is the Mars Climate Orbiter mission in 1998. NASA lost the spacecraft due to a navigation error caused by a unit conversion mistake. The error occurred because one team used metric units (newtons), while another team used imperial units (pound-force) for a crucial spacecraft operation parameter. The discrepancy in units led to incorrect trajectory calculations, causing the Mars Climate Orbiter to approach Mars at a dangerously low altitude and ultimately disintegrate in the Martian atmosphere.</span>
<span id="cb1-257"><a href="#cb1-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-258"><a href="#cb1-258" aria-hidden="true" tabindex="-1"></a>This incident highlights the importance of clear communication, proper data handling, and rigorous verification processes in mission-critical systems, including space missions. The event emphasises the need for using appropriate tools, software, and methodologies to minimise the risk of errors in complex engineering projects.</span>
<span id="cb1-259"><a href="#cb1-259" aria-hidden="true" tabindex="-1"></a>:::</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">© 2023, AJ Smit</div>   
    <div class="nav-footer-center"><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">This page is built with <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>


<script src="../../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>