<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.26">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Smit, A. J.">
<meta name="dcterms.date" content="2021-01-01">
<title>2. Working with Data &amp; Code – The Tangled Bank</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../BCB744/intro_r/03-quarto.html" rel="next">
<link href="../../BCB744/intro_r/01-RStudio.html" rel="prev">
<link href="../../images/favicon.ico" rel="icon">
<script src="../../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-597958c53c93a607afca12fd375c57ed.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-37d9842ca13b32c9c58adfc4ed5d9451.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-SCYH68M489"></script><script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-SCYH68M489', { 'anonymize_ip': true});
</script><script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"headline",
  "consent_type":"express",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  ,
"language":"en"
  });
});
</script><style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script><script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script><link rel="stylesheet" href="../../styles/styles.css">
<meta property="og:title" content="2. Working with Data &amp; Code – The Tangled Bank">
<meta property="og:description" content="UWC’s Biodiversity &amp; Conservation Biology Department Pages For BDC223, BDC334, BCB743, and BCB744.">
<meta property="og:image" content="http://tangledbank.netlify.app/BCB744/intro_r/data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==">
<meta property="og:site_name" content="The Tangled Bank">
</head>
<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">The Tangled Bank</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-vignettes" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Vignettes</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-vignettes">
<li>
    <a class="dropdown-item" href="../../vignettes/netCDF_dates.html">
 <span class="dropdown-text">Dates From netCDF Files: Two Approaches</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/prep_NOAA_OISST.html">
 <span class="dropdown-text">Downloading and Prep: ERDDAP</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_ERDDAP.html">
 <span class="dropdown-text">Retrieving Chlorophyll-a Data from ERDDAP Servers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/alt_method.html">
 <span class="dropdown-text">Downloading and Prep: Python + CDO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/gridded_data.html">
 <span class="dropdown-text">Detecting Events in Gridded Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/regridding.html">
 <span class="dropdown-text">Regridding gridded data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/MHW_MCS_horizonplots.html">
 <span class="dropdown-text">Displaying MHWs and MCSs as Horizon Plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/heatwaveR_issues.html">
 <span class="dropdown-text">heatwaveR Issues</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_sightings.html">
 <span class="dropdown-text">Plotting the Whale Sightings and Chlorophyll-<em>a</em> Concentrations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_localisation.html">
 <span class="dropdown-text">Spatial Localisation, Subsetting, and Aggregation of the Chlorophyll-<em>a</em> Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/elem_ts_methods.html">
 <span class="dropdown-text">Wavelet Analysis of Diatom Time Series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/buffer_data_extract.html">
 <span class="dropdown-text">Extracting gridded data within a buffer</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-high-performance-computing" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">High-Performance Computing</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-high-performance-computing">
<li>
    <a class="dropdown-item" href="../../vignettes/README_Lengau.html">
 <span class="dropdown-text">Using Lengau</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/PBSPro_users.html">
 <span class="dropdown-text">PBSPro user and job management</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/README_tmux.html">
 <span class="dropdown-text">Using tmux</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item">
    <a class="nav-link" href="../../blog.html"> 
<span class="menu-text">Tangled Bank Blog</span></a>
  </li>  
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-blogroll" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Blogroll</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-blogroll">
<li>
    <a class="dropdown-item" href="https://www.r-bloggers.com">
 <span class="dropdown-text">R-bloggers</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item compact">
    <a class="nav-link" href="../../blog.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">
<li>
    <a class="dropdown-item" href="https://github.com/ajsmit/tangled_bank">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
    </ul>
</li>
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../BCB744/BCB744_index.html">BCB744: Introduction to R, &amp; Biostatistics</a></li><li class="breadcrumb-item"><a href="../../BCB744/intro_r/02-working-with-data.html">2. Working with Data &amp; Code</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="../../index.html" class="sidebar-logo-link">
      <img src="../../images/my_tree_hex.png" alt="" class="sidebar-logo light-content py-0 d-lg-inline d-none"><img src="../../images/my_tree_hex.png" alt="" class="sidebar-logo dark-content py-0 d-lg-inline d-none"></a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">UNDERGRADUATE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC223: Plant Ecophysiology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/BDC223_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L00-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introductory Lecture</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L01-worldmapper.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: the Limits to Life</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L02-SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2: Surface Area to Volume (SA:V) Ratio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L03-climate.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3: Earth’s Climate</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L04-carbon_cycle.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Climate Change (Carbon Cycle)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L05-light.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 5: Light</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L06-pigments_photosynthesis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6: Pigments &amp; Photosynthesis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L07-chromatic_adaptation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 7: Chromatic Adaptation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L08a-nutrient_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8a: Nutrient Uptake</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L08b-nutrients_michaelis_menten.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8b: Nutrient Uptake</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L09-plant_stress.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3: Plant Stress</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab1_SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1: Surface Area to Volume (SA/V) Ratios in Biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab2_misc_calcs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2: Miscellaneous Calculations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab3_PI_curves.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3: PI Curves — Jassby &amp; Platt</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab4_nitrogen_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4: Uptake Kinetics — Michaelis-Menten</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/BDC223_FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC334: Biogeography &amp; Global Ecology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/BDC334_syllabus.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/BDC334-Lecture-Transcripts.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">BCB334 Lecture Transcript</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: Introductory Lecture</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-02-ecosystems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2. Overview of Ecosystems</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-03-gradients.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3. Ecological Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Biodiversity Concepts</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 5: Multivariate Data</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-06-unified-ecology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6: Unified Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 7: Threats to Biodiversity</span>
  </li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 8: Nature's Contribution to People</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/IKS_Conservation.pdf" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 9: Indigenous Knowledge Systems and Conservation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1. Ecological Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-02a-r_rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2a. Introduction to R &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-02b-env_dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2b. Environmental Distance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-03-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3. Quantifying Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4. Species Distribution Patterns</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/BDC334_AI_Tutor.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">BCB334 AI Tutor</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pages/Transboundary_systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transboundary Systems</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS CORE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">BCB744: Introduction to R, &amp; Biostatistics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/BCB744_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/01-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. R &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/02-working-with-data.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">2. Working with Data &amp; Code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/03-quarto.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. R Markdown and Quarto</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/04-data-in-R.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data Classes &amp; Structures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/05-workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. R Workflows</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/06-graphics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Graphics with <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/07-faceting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Faceting Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/08-brewing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Brewing Colours</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/09-mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Mapping with <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/10-mapping_style.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Mapping with Style</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/11-mapping_rnaturalearth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Mapping with Natural Earth</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/12-mapping_quakes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. the Fiji Earthquake Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/13-tidy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Tidy Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/14-tidier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Tidier Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/15-tidiest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15. Tidiest Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/16-recap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16. Recap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/assessments/BCB744_Intro_R_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Intro R Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/assessments/BCB744_Intro_R_Example_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 Intro R Example 1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/assessments/BCB744_Intro_R_Example_2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 Intro R Example 2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../self-assessment_v1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Self-Assessment Form</span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
 <span class="menu-text">1. Introduction to Statistics</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/02-summarise-and-describe.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Data Summaries &amp; Descriptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/03-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Statistical Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/04-distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data Distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/05-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Statistical Inference</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/06-assumptions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Assumptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/07-t_tests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. <em>t</em>-Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/08-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. ANOVA</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/09-regressions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Simple Linear Regressions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/10-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/11-decision_guide.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/12-glance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Non-Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/13-confidence.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Confidence Intervals</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/14-transformations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Data Transformations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/assessments/BCB744_Biostatistics_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Biostatistics Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/assessments/BCB744_BioStats_Example_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 BioStats 2025 Example 1</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS ELECTIVES</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">BCB743: Quantitative Ecology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/BCB743_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/review.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1-4: Review Concepts fo Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5: Correlations &amp; Associations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7: Intro to Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8a: Principal Components Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCA_examples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8b: PCA: Additional Examples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCA_SDG_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8c: PCA: WHO SDG Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/CA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9a: Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/DCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9b: Detrended Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCoA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10: Principal Coordinates Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/nMDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11a: Non-metric Multidimensional Scaling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/nMDS_diatoms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11b: nMDS: PERMANOVA (Diatoms) Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/unconstrained-summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12: Unconstrained Ordination Summary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/constrained_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13a: Distance-Based Redundancy Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/two_oceans_appendices.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13b: db-RDA: Seaweeds Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/cluster_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14: Cluster Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/model_building.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15: Model Building</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/deep_dive.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16a: Deep Dive Into Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/multiple_regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16b: Multiple Regression</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">AI &amp; POLICY</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/Combined Faculty AI Policy 15082025.pdf" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Faculty of Natural Sciences AI Policy</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../AI/BCB_AI_Workshop_Sep_2025.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB AI Workshop</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">SUPPORT</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Web Resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/general_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/ecology_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quantitative Ecology Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/spatial_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial R Resources</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> Introduction</a></li>
  <li>
<a href="#navigating-the-file-system-and-the-working-directory" id="toc-navigating-the-file-system-and-the-working-directory" class="nav-link" data-scroll-target="#navigating-the-file-system-and-the-working-directory"><span class="header-section-number">2</span> Navigating the File System and the Working Directory</a>
  <ul class="collapse">
<li><a href="#the-working-directory" id="toc-the-working-directory" class="nav-link" data-scroll-target="#the-working-directory"><span class="header-section-number">2.1</span> The Working Directory</a></li>
  <li><a href="#rstudio-projects" id="toc-rstudio-projects" class="nav-link" data-scroll-target="#rstudio-projects"><span class="header-section-number">2.2</span> RStudio Projects</a></li>
  <li><a href="#exercises" id="toc-exercises" class="nav-link" data-scroll-target="#exercises"><span class="header-section-number">2.3</span> Exercises</a></li>
  </ul>
</li>
  <li>
<a href="#types-of-data-files" id="toc-types-of-data-files" class="nav-link" data-scroll-target="#types-of-data-files"><span class="header-section-number">3</span> Types of Data Files</a>
  <ul class="collapse">
<li><a href="#delimited-text-files-csv-and-tsv" id="toc-delimited-text-files-csv-and-tsv" class="nav-link" data-scroll-target="#delimited-text-files-csv-and-tsv"><span class="header-section-number">3.1</span> Delimited Text Files (CSV and TSV)</a></li>
  <li><a href="#microsoft-excel-files" id="toc-microsoft-excel-files" class="nav-link" data-scroll-target="#microsoft-excel-files"><span class="header-section-number">3.2</span> Microsoft Excel Files</a></li>
  <li><a href="#rdata-files" id="toc-rdata-files" class="nav-link" data-scroll-target="#rdata-files"><span class="header-section-number">3.3</span> Rdata Files</a></li>
  <li><a href="#sec-binary" id="toc-sec-binary" class="nav-link" data-scroll-target="#sec-binary"><span class="header-section-number">3.4</span> Other Binary Files</a></li>
  <li><a href="#sec-netcdf" id="toc-sec-netcdf" class="nav-link" data-scroll-target="#sec-netcdf"><span class="header-section-number">3.5</span> NetCDF, Grib, and HDF Files</a></li>
  <li><a href="#larger-than-memory-data" id="toc-larger-than-memory-data" class="nav-link" data-scroll-target="#larger-than-memory-data"><span class="header-section-number">3.6</span> Larger than Memory Data</a></li>
  </ul>
</li>
  <li><a href="#file-extensions" id="toc-file-extensions" class="nav-link" data-scroll-target="#file-extensions"><span class="header-section-number">4</span> File Extensions</a></li>
  <li><a href="#r-code-conventions" id="toc-r-code-conventions" class="nav-link" data-scroll-target="#r-code-conventions"><span class="header-section-number">5</span> R Code Conventions</a></li>
  <li>
<a href="#sec-reproducibility" id="toc-sec-reproducibility" class="nav-link" data-scroll-target="#sec-reproducibility"><span class="header-section-number">6</span> Reproducible Research</a>
  <ul class="collapse">
<li><a href="#required-practice-in-bcb744" id="toc-required-practice-in-bcb744" class="nav-link" data-scroll-target="#required-practice-in-bcb744"><span class="header-section-number">6.1</span> Required practice in BCB744</a></li>
  <li><a href="#a-look-to-the-future" id="toc-a-look-to-the-future" class="nav-link" data-scroll-target="#a-look-to-the-future"><span class="header-section-number">6.2</span> A look to the future</a></li>
  </ul>
</li>
  <li>
<a href="#exercise-from-excel-to-delimited-files-to-r" id="toc-exercise-from-excel-to-delimited-files-to-r" class="nav-link" data-scroll-target="#exercise-from-excel-to-delimited-files-to-r"><span class="header-section-number">7</span> Exercise: From Excel to Delimited Files to R</a>
  <ul class="collapse">
<li><a href="#part-a-create-a-project-folder-with-known-locations" id="toc-part-a-create-a-project-folder-with-known-locations" class="nav-link" data-scroll-target="#part-a-create-a-project-folder-with-known-locations"><span class="header-section-number">7.1</span> Part A — Create a Project folder with known locations</a></li>
  <li><a href="#part-b-create-a-small-dataset-in-excel" id="toc-part-b-create-a-small-dataset-in-excel" class="nav-link" data-scroll-target="#part-b-create-a-small-dataset-in-excel"><span class="header-section-number">7.2</span> Part B — Create a small dataset in Excel</a></li>
  <li><a href="#part-c-export-three-delimited-text-files-into-the-project" id="toc-part-c-export-three-delimited-text-files-into-the-project" class="nav-link" data-scroll-target="#part-c-export-three-delimited-text-files-into-the-project"><span class="header-section-number">7.3</span> Part C — Export three delimited text files into the project</a></li>
  <li><a href="#part-d-read-one-exported-file-from-a-new-r-script-in-rstudio" id="toc-part-d-read-one-exported-file-from-a-new-r-script-in-rstudio" class="nav-link" data-scroll-target="#part-d-read-one-exported-file-from-a-new-r-script-in-rstudio"><span class="header-section-number">7.4</span> Part D — Read one exported file from a new R script in RStudio</a></li>
  </ul>
</li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB744/intro_r/02-working-with-data.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../BCB744/BCB744_index.html">BCB744: Introduction to R, &amp; Biostatistics</a></li><li class="breadcrumb-item"><a href="../../BCB744/intro_r/02-working-with-data.html">2. Working with Data &amp; Code</a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">2. Working with Data &amp; Code</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://tangledbank.netlify.app">Smit, A. J.</a> <a href="https://orcid.org/0000-0002-3799-6126" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uwc.ac.za">
            University of the Western Cape
            </a>
          </p>
      </div>
  </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 1, 2021</p>
    </div>
  </div>
  
    
  </div>
  


</header><p><a href="../../images/PhD_comics2.PNG" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="../../images/PhD_comics2.PNG" class="img-fluid" style="width:100.0%"></a></p>
<blockquote class="blockquote">
<p><em>“The plural of anecdote is not data.”</em></p>
<p>— Roger Brinner</p>
</blockquote>
<p>In this Chapter we will cover:</p>
<ul>
<li>Data file formats</li>
<li>The use of file extensions</li>
<li>Review some R code conventions</li>
<li>Introduce the concept of reproducible research</li>
</ul>
<section id="introduction" class="level1" data-number="1"><h1 data-number="1">
<span class="header-section-number">1</span> Introduction</h1>
<p>In today’s rapidly evolving scientific fields, understanding the basic principles of coding (also called scripting or programming) and data handling has become indispensable. The complexity and volume of data continue to grow, and traditional manual data analysis methods are no longer sufficient to deal with the demands of research as practised today. Coding and data manipulation skills will unlock the potential of your computer and allow you to analyse datasets, fit models, and reveal hidden patterns that may not be apparent through conventional methods. Proficiency in coding and data handling will help you to collaborate effectively with multidisciplinary teams (as is increasingly the case today), access data from various data sources in different formats, cross the gap between theory and practical applications, and contribute to the advancement of scientific knowledge in a transparent and efficient way.</p>
<p>A useful way to understand the material in this chapter is to recognise that most failures in data analysis do not arise from advanced statistical methods, but from mismatches at the interface between how and where data are stored, how software interprets those representations, and how we interact with both. File systems organise data, file formats encode assumptions, software enforces expectations, and you, the budding scientist, bring good and bad habits into that system.</p>
<p>Each section in this chapter addresses a different point at which data can be misrepresented, misunderstood, or clarified, and the aim throughout is to make those points visible.</p>
<p>Throughout this discussion there are a few underlying assumptions which organise the practical guidance that follows.</p>
<ol type="1">
<li>Data have structure independent of the software used to store or analyse them.</li>
<li>Software necessarily imposes assumptions on that structure, whether explicitly or by default behaviour.</li>
<li>Reproducibility is as much a property of workflows and data representations <em>and</em> of intention or care.</li>
</ol>
<p>Below, I focus on how they play out in routine analytical choices. First — before we look at how data are stored — let’s look at <em>how</em> and <em>where</em> to find data and the outputs of your analyses once they reside on your computer.</p>
</section><section id="navigating-the-file-system-and-the-working-directory" class="level1" data-number="2"><h1 data-number="2">
<span class="header-section-number">2</span> Navigating the File System and the Working Directory</h1>
<p>Many analytical errors arise from confusion about where data and code actually live on a computer. Before software can read a file, the you must specify <em>where</em> that file is located. Where the file lives is not random, but poor understanding of how to navigate your computer’s file system makes it feel haphazard at best or like a black box at worst. You <em>should store files deliberately in precisely known locations</em>. This requires a working grasp of the file system: how files are organised, how locations are described, and how software interprets those descriptions.</p>
<p>So, before R can read data or write results, it must know where it is working. This sounds trivial but many failures you’ll encounter in the early days on your journey to becoming a biostatistician or data analyst trace back to uncertainty about file locations rather than to coding errors. This section introduces the minimal concepts required to reason correctly about files, directories, and paths, and explains how R and RStudio use these ideas in practice.</p>
<p>A <strong>file system</strong> is the hierarchical structure by which an operating system stores and organises files. At the top of this hierarchy sits a <em>root</em>, from which all folders (directories) and files descend. Every file therefore occupies a unique position within this structure, defined by the sequence of directories leading to it.</p>
<p>A <strong>directory</strong> (or <em>folder</em>) is simply a container that can hold files and other directories. When a directory sits inside another directory, it is a <strong>sub-directory</strong>. This nesting can continue indefinitely, producing a tree-like structure. Although Windows, macOS, and Linux differ in visual presentation, this underlying structure is shared across all three systems.</p>
<p>To refer to a file, we use a <strong>file path</strong>: a textual description of its location within the file system. You can observe these file paths within RStudio, near the top of the <strong>Console</strong>.</p>
<p>An <strong>absolute path</strong> specifies a file’s location starting from the root of the file system. Because it begins at the root, it uniquely identifies the file regardless of the current working context.</p>
<p><strong>Examples:</strong></p>
<p>Windows:</p>
<p><code>C:\Users\Student\Documents\BCB744\data\kelp.csv</code></p>
<p>macOS / Linux:</p>
<p><code>/home/student/BCB744/data/kelp.csv</code></p>
<p>Notice that Windows uses backslashes <code>\</code> and maxOS and Linux use forward slashes <code>/</code>. This is a very important distinction between Windows and Unix-like operating systems.</p>
<p>Absolute paths are unambiguous, but they are not very helpful when moving R (or other) scripts between different users. This is because they encode machine-specific details (usernames, drive letters) that differ between those users’ computers.</p>
<p>A <strong>relative path</strong> specifies a file’s location <em>relative to the current working directory</em>. It does not begin at the root, but instead assumes a starting point. Relative paths are shorter, portable, and central to reproducible workflows.</p>
<p>Assume the current working directory is:</p>
<ul>
<li><code>BCB744/</code></li>
</ul>
<p>Then:</p>
<ul>
<li>
<code>data/kelp.csv</code> refers to a file called <code>kelp.csv</code> inside the <code>data</code> sub-directory.</li>
<li>
<code>scripts/analysis.R</code> refers to an R file named <code>analysis.R</code> inside the <code>scripts</code> sub-directory.</li>
</ul>
<p>Two special symbols appear frequently in relative paths:</p>
<ul>
<li>
<code>.</code> refers to the current directory.</li>
<li>
<code>..</code> refers to the parent directory (one level up).</li>
</ul>
<p>Examples:</p>
<ul>
<li>
<code>../data/kelp.csv</code> means “go up one directory, then into <code>data</code>”.</li>
<li>
<code>../../raw/sst.nc</code> means “go up two levels, then into <code>raw</code>”.</li>
</ul>
<section id="the-working-directory" class="level2" data-number="2.1"><h2 data-number="2.1" class="anchored" data-anchor-id="the-working-directory">
<span class="header-section-number">2.1</span> The Working Directory</h2>
<p>The <strong>working directory</strong> is the directory that R treats as its current location. All relative paths are interpreted with respect to this directory.</p>
<p>You can think of the working directory as the answer to the question:</p>
<blockquote class="blockquote">
<p>“If I refer to a file without giving its full address, where will R look?”*</p>
</blockquote>
<p>In R, the working directory can be inspected with:</p>
<ul>
<li><code><a href="https://rdrr.io/r/base/getwd.html">getwd()</a></code></li>
</ul>
<p>and (technically) changed with:</p>
<ul>
<li><code>setwd("some/path")</code></li>
</ul>
<p>Beginners sometimes try to fix file-not-found errors by repeatedly calling <code><a href="https://rdrr.io/r/base/getwd.html">setwd()</a></code>. This usually makes things worse. Changing the working directory inside scripts creates hidden assumptions about where the code is run and breaks reproducibility.</p>
<p>The preferred solution is not to move the working directory around, but to <strong>control it once</strong>, deliberately.</p>
</section><section id="rstudio-projects" class="level2" data-number="2.2"><h2 data-number="2.2" class="anchored" data-anchor-id="rstudio-projects">
<span class="header-section-number">2.2</span> RStudio Projects</h2>
<p>An <strong>RStudio Project</strong> defines a directory as the root of an analysis and sets it as the working directory whenever the project is opened. This has three consequences that are foundational for this course:</p>
<ol type="1">
<li>
<code><a href="https://rdrr.io/r/base/getwd.html">getwd()</a></code> always returns the project root.</li>
<li>All scripts, data, and outputs can be referenced using relative paths.</li>
<li>Code runs the same way on different computers.</li>
</ol>
<p>A typical project structure might look like this:</p>
<ul>
<li><code>BCB744_project/</code></li>
</ul>
<p>With sub-folders specific to the different purposes:</p>
<ul>
<li><code>data/</code></li>
<li><code>scripts/</code></li>
<li><code>figures/</code></li>
<li><code>reports/</code></li>
</ul>
<p>If this directory is an RStudio Project, then a script inside <code>scripts/</code> can read data using:</p>
<ul>
<li><code>data/kelp.csv</code></li>
</ul>
<p>without calling <code><a href="https://rdrr.io/r/base/getwd.html">setwd()</a></code>, and without caring where the project lives on the machine.</p>
<p>This is why projects are treated as non-negotiable infrastructure in this course. They make file paths explicit, stable, and inspectable.</p>
<p>Software such as R interprets all file paths relative to its <strong>working directory</strong>—the directory it currently considers as “here”. Confusion about file paths is therefore almost always confusion about the working directory. When files cannot be found, the problem is rarely the file itself; it is the path used to describe it.</p>
<p>For this reason, analyses should be organised as <strong>projects</strong>, with a clear internal structure. A minimal example might look like this:</p>
<ul>
<li>
<p><code>BCB744/</code></p>
<ul>
<li>
<code>data/</code> … (raw or cleaned data)</li>
<li>
<code>scripts/</code> … (R scripts)</li>
<li>
<code>figures/</code> … (output plots)</li>
<li>
<code>reports/</code> … (Quarto or R Markdown documents)</li>
</ul>
</li>
</ul>
<p>When code is run from the project root, all file access can use relative paths, avoiding hard-coded references to specific machines.</p>
</section><section id="exercises" class="level2" data-number="2.3"><h2 data-number="2.3" class="anchored" data-anchor-id="exercises">
<span class="header-section-number">2.3</span> Exercises</h2>
<ol type="1">
<li><p>On your computer, create a directory called <code>filesystem_practice</code>. Inside it, create three sub-directories: <code>data</code>, <code>scripts</code>, and <code>output</code>. Place a small text file inside <code>data</code>.</p></li>
<li><p>Place a small text or CSV file inside <code>data</code>. From the R Console, run <code><a href="https://rdrr.io/r/base/getwd.html">getwd()</a></code> and confirm that it points to the project root.</p></li>
<li><p>From a script inside <code>scripts</code>, write the relative path needed to read the file in <code>data</code>. Then write the absolute path to the same file. Which one would still work on another computer?</p></li>
<li><p>Deliberately break a path by misspelling a directory name. Observe the error message produced by the software. Correct the path and rerun the command.</p></li>
<li><p>(Deliberately bad practice) Use setwd() to point R directly at the data directory. Now try running the script again from a different location. Explain why this approach fails as a general strategy.</p></li>
</ol>
<p>These exercises should help you to develop a form of spatial reasoning that underlies all your analyses 9and working with your computer in general — good housekeeping stems from this! If you cannot reliably locate files you also cannot reason clearly about data provenance, reproducibility, or control over your analysis.</p>
<p>Note: graphical file browsers hide structure behind icons and clicks. Code exposes structure explicitly. Learning to think in paths rather than windows is a necessary transition from consumer software use to scientific computing.</p>
<p>Now that we know how and where to locate your data, let’s take a look at how those data are represented.</p>
</section></section><section id="types-of-data-files" class="level1" data-number="3"><h1 data-number="3">
<span class="header-section-number">3</span> Types of Data Files</h1>
<p>The choice of data file format determines how information is stored and which assumptions software will make when reading it, and which errors are likely to remain invisible.</p>
<p>R will read in many types of data, including spreadsheets, text files, binary files, and files from other statistical packages.</p>
<p>It is useful to treat questions about which file type to use as answering recurring practical questions. At different stages of an analysis, we may need data to be easily readable by humans, reliably exchanged across software and operating systems, efficient to store or process at scale, or well annotated with metadata. No single format optimises all of these requirements simultaneously. The non-exhaustive list of file formats mentioned below offer different compromises among these constraints, and their suitability depends on which problem is being prioritised.</p>
<section id="delimited-text-files-csv-and-tsv" class="level2" data-number="3.1"><h2 data-number="3.1" class="anchored" data-anchor-id="delimited-text-files-csv-and-tsv">
<span class="header-section-number">3.1</span> Delimited Text Files (CSV and TSV)</h2>
<p>CSV and TSV files optimise human readability.</p>
<p>Delimited text files represent the simplest and most transparent way of storing tabular data. Both comma-separated value (CSV) files and tab-separated value (TSV) files encode tables as plain text, with rows separated by line breaks and columns separated by a designated delimiter. The distinction between them arises from how reliably the chosen delimiter can be distinguished from the data themselves.</p>
<p>CSV files use commas to separate fields (columns), which makes them widely supported and easy to exchange across software and platforms. However, this choice becomes ambiguous when data values themselves contain commas, such as free text, lists, or numbers formatted with thousands separators. TSV files address this ambiguity by using tab characters as delimiters, which are far less likely to occur naturally within data values. In this sense, TSV files trade a small loss in visual familiarity for greater robustness during parsing.</p>
<p>As such, CSV files are a simple and widely used format amongst biologists and ecologists, but they can become impractical for large datasets with complex structures or metadata.</p>
<p>We will most frequently use the functions <code><a href="https://rdrr.io/r/utils/read.table.html">read.csv()</a></code> and <code><a href="https://readr.tidyverse.org/reference/read_delim.html">readr::read_csv()</a></code> (and related forms) for reading in CSV data. We can write CSV files to disk with the <code><a href="https://rdrr.io/r/utils/write.table.html">write.csv()</a></code> and <code><a href="https://readr.tidyverse.org/reference/write_delim.html">readr::write_csv()</a></code> commands. For very large datasets that might take a long time to read in or save, <code><a href="https://rdrr.io/pkg/data.table/man/fread.html">data.table::fread()</a></code> and <code><a href="https://rdrr.io/pkg/data.table/man/fwrite.html">data.table::fwrite()</a></code> are faster alternatives to the aforementioned base R or <strong>tidyverse</strong> options. Even faster options are <code>feather::read_feather()</code> and <code>feather::write_feather()</code>; although <strong>feather</strong> saves tabular data, the format is not actually an ASCII CSV, however.</p>
<p>The same functions that read or write CSV files in R can be used for TSV, but one has to set the arguments <code>sep = "\t"</code> or <code>delim = "\t"</code> for the functions <code><a href="https://rdrr.io/r/utils/read.table.html">read.csv()</a></code> and <code>read_csv()</code> respectively.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>ASCII Files
</div>
</div>
<div class="callout-body-container callout-body">
<p>ASCII stands for “American Standard Code for Information Interchange”. An ASCII file is a plain text file that contains ASCII characters. ASCII is a character encoding standard that assigns a unique numeric code to each character, including letters, numbers, punctuation, and other symbols commonly used in the English language.</p>
<p>ASCII files are the most basic type of text file and are supported by virtually all operating systems and applications. We can create and edit ASCII files using any text editor, such as Notepad, TextEdit, or <a href="https://code.visualstudio.com/">VS Code</a>. ASCII files are typically used for storing, sharing simple text-based information such as program source code, configuration files, and other types of data that do not require special formatting or rich media content.</p>
<p>ASCII files are limited in their ability to represent non-English characters or symbols that are not included in the ASCII character set. To handle these types of characters, other character encoding standards such as UTF-8 and Unicode are used. However, ASCII files remain an important and widely used format for storing and sharing simple text-based data.</p>
</div>
</div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Missing Values and CSV and TSV Files
</div>
</div>
<div class="callout-body-container callout-body">
<p>Where we have missing data (blanks), the CSV format separates these by commas with empty field in-between. However, there can be problems with blanks if we read in a space-delimited format file. If we are having trouble reading in missing data as blanks, try replacing them in the spreadsheet with <code>NA</code>, the missing data code in R. In Excel, highlight the area of the spreadsheet that includes all the cells we need to fill with <code>NA</code>. Do an ‘Edit/Replace…’ and leave the ‘Find what:’ text box blank and in the ‘Replace with:’ text box enter <code>NA</code>. Once imported into R, the <code>NA</code> values will be recognised as missing data.</p>
</div>
</div>
</section><section id="microsoft-excel-files" class="level2" data-number="3.2"><h2 data-number="3.2" class="anchored" data-anchor-id="microsoft-excel-files">
<span class="header-section-number">3.2</span> Microsoft Excel Files</h2>
<p>Excel files optimise human convenience, engagement, and presentation.</p>
<p>However, Excel files also emphasise that software convenience can obscure underlying data transformations, which moves the analytical risk from computation to problems that might stem from user interaction.</p>
<p>Microsoft Excel files are a type of file format that is used to store data in a tabular form, much like CSV files. However, Excel files are proprietary and are specifically designed to work with Excel software. Excel files can contain more advanced formatting features such as colours, fonts, and formulas, which make them a popular choice for people who like embellishments.</p>
<p>These software behaviours impose constraints that are poorly aligned with statistical analysis, and it is from these constraints that the following limitations arise:</p>
<ul>
<li><p><strong>Compatibility</strong> Excel files may not be compatible with all data science tools and programming languages. For example, R cannot read Excel files directly.</p></li>
<li><p><strong>Data integrity</strong> Excel files can be prone to errors and inconsistencies in the data. For example, if a user changes a formula or formatting, it could affect the entire dataset. Also, it is possible for Excel to change the data types of certain columns, or to mix the class of data <em>within a column</em>, which can cause issues with data processing and analysis.</p></li>
<li><p><strong>File size</strong> Excel files can quickly become very large when dealing with large datasets, which can lead to performance issues, storage problems.</p></li>
<li><p><strong>Version control</strong> Excel files can make it difficult to keep track of changes and versions of the data, particularly when multiple people are working on the same file.</p></li>
</ul>
<p>In contrast, CSV files are a simple, lightweight, and widely supported file format that can be easily used with most data science tools and programming languages. CSV files are also less prone to errors and inconsistencies than Excel files, making them a more reliable choice for data science tasks.</p>
<p>For these reasons, Excel is best used as a tool for data entry rather than data analysis. Exporting deliberately to plain-text formats such as CSV fixes the data representation, reveals assumptions to inspection, and sets a clear boundary between data generation and analysis. Naming exported files explicitly (often with dates or version identifiers) further tightens provenance and reduces ambiguity about which data underlie a given result.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Well-known Excel Errors
</div>
</div>
<div class="callout-body-container callout-body">
<p>Excel is a widely used spreadsheet application, but it has been responsible for several serious errors in data analysis, science, and data science. Some of these errors include:</p>
<ul>
<li><p><strong>Gene name errors (2016):</strong> Accurate. Ziemann, Eren, and El-Osta published their findings in <em>Genome Biology</em>, demonstrating that Excel’s automatic conversion transformed gene symbols like SEPT2 and MARCH1 into dates. Their survey examined supplementary files from 3,597 published papers across 18 journals between 2005 and 2015, finding that roughly one-fifth contained such errors. This remains one of the most damaging documented cases of Excel undermining scientific reproducibility, and it directly affects biology students working with genomic data.</p></li>
<li><p>Another compelling case involves <strong>automatic data type conversion destroying scientific measurements</strong>. Excel converts identifiers that look like scientific notation (<em>e.g.</em>, “1E4” becomes 10,000) or treats leading zeros (<em>e.g.</em>, “0001”) in sample IDs as insignificant (<em>i.e.</em>, Excel displays “1”). Ecologists working with plot identifiers or anyone using structured identifiers encounters this pervasive, insidious problem.</p></li>
<li><p><strong>Truncation of large numbers</strong> Excel can handle only a limited number of digits for large numbers, truncating any value that exceeds this limit. This truncation has led to a loss of precision and inaccurate calculations in scientific and data analysis contexts where exact values were important.</p></li>
<li><p><strong>Issues with floating-point arithmetic</strong> Excel uses IEEE 754 double-precision floating-point representation, which produces rounding errors. The classic demonstration involves calculations like <code>(0.1 + 0.2) ≠ 0.3</code> in binary representation. For iterative scientific calculations or cumulative errors in modelling, this matters.</p></li>
<li><p>The <strong>UK COVID-19 testing data loss (October 2020)</strong> deserves attention. Public Health England used the legacy <code>.xls</code> format (limited to 65,536 rows) rather than the newer <code>.xlsx</code> (limited to 1,048,576 rows), resulting in 15,841 positive test results being lost when the file exceeded row limits. This directly affected contact tracing during a pandemic.</p></li>
<li><p>The <strong>date format ambiguity problem</strong> creates reproducibility nightmares across international collaborations. Excel interprets “01/02/2020” differently depending on regional settings (January 2nd vs.&nbsp;February 1st), and automatically converting text strings to dates corrupts datasets when shared between researchers using different locale settings. This affects any collaborative science.</p></li>
</ul>
</div>
</div>
</section><section id="rdata-files" class="level2" data-number="3.3"><h2 data-number="3.3" class="anchored" data-anchor-id="rdata-files">
<span class="header-section-number">3.3</span> Rdata Files</h2>
<p>Specialist files such as Rdata, binary files (<a href="#sec-binary" class="quarto-xref">Section&nbsp;3.4</a>), and NetCDF (<a href="#sec-netcdf" class="quarto-xref">Section&nbsp;3.5</a>) files were developed for optimasing scale (data size), metadata (data about data), and coupling (issues around speed and performance).</p>
<p>Rdata files are a file format used by the R programming language to store data objects. These files can contain any type of R object, such as vectors, matrices, dataframes, lists, and more. Rdata files are binary files, which means they are not human-readable like text files such as CSV files. Binary R data files have a <code>.rda</code> or <code>.Rdata</code> file extension and can be created or read using the <code><a href="https://rdrr.io/r/base/save.html">save()</a></code> and <code><a href="https://rdrr.io/r/base/load.html">load()</a></code>, respectively, functions in R.</p>
<p>Rdata files are convenient for a number of reasons:</p>
<ul>
<li><p><strong>Efficient storage</strong> Rdata files can be more compact (they can be compressed) and efficient than other file formats, such as CSV files, because they are stored in a binary format. This means they take up less disk space and can be read and written faster.</p></li>
<li><p><strong>Easy access to R objects</strong> Rdata files make it easy to save and load R objects, which can be useful for preserving data objects for future analysis, sharing them with others. This is especially useful for complex datasets or objects that would be difficult to recreate.</p></li>
<li><p><strong>Preserve metadata</strong> Rdata files can preserve metadata such as variable names, row names, column names, and other attributes of R objects. This makes it easier to work with the data objects in the future without having to recreate this metadata.</p></li>
<li><p><strong>Convenient for reproducibility</strong> Rdata files can be used to save and load data objects as part of a reproducible research workflow. This can help ensure that data objects are preserved and can be easily accessed in the future, even if the data sources or code have changed.</p></li>
</ul>
<p>On the downside, they can only be used within R, making them a less than ideal proposition when you intend to share your data with colleagues who sadly do not use R.</p>
</section><section id="sec-binary" class="level2" data-number="3.4"><h2 data-number="3.4" class="anchored" data-anchor-id="sec-binary">
<span class="header-section-number">3.4</span> Other Binary Files</h2>
<p>As a biostatistician, you may encounter several other binary data files in your work. Such binary data files may be software-specific and can be used to store large datasets or data objects that are not easily represented in a text format. For example, a binary data file might contain a large matrix or array of numeric data that would be difficult to store in a text file. Binary data files can also be used to store images, audio files, and other types of data that are not represented as text.</p>
<p>One common type of binary data file that you may encounter as a statistician is a SAS data file. SAS is a statistical software package that is widely used in data analysis, and SAS data files are a binary format used to store datasets in SAS. These files typically have a <code>.sas7bdat</code> file extension and contain metadata such as variable names and formats in addition to the data itself. Another type of binary data file you may encounter is a binary <code>.mat</code> data file, which is a file format used to store Matlab data.</p>
<p>When working with binary data files, it is important to be aware of the specific format of the file, the tools and software needed to read and manipulate the data. Some statistical software packages may have built-in functions for reading and writing certain types of binary data files, while others may require additional libraries or packages.</p>
</section><section id="sec-netcdf" class="level2" data-number="3.5"><h2 data-number="3.5" class="anchored" data-anchor-id="sec-netcdf">
<span class="header-section-number">3.5</span> NetCDF, Grib, and HDF Files</h2>
<p>NetCDF, HDF, and GRIB are file formats commonly used in the scientific and research communities to store and share large and complex datasets. These datasets are optimised for storing gridded (regular intervals in space and time) data that are interoperable amongst computing systems. Here is a brief overview of each file format:</p>
<ul>
<li><p><strong>NetCDF (Network Common Data Form)</strong> is a binary file format that is designed for storing and sharing scientific data. It can store multidimensional arrays and metadata, such as variable names and units, in a self-describing format. NetCDF files are commonly used in fields such as atmospheric science, oceanography, and climate modelling.</p></li>
<li><p>Like NetCDF files, <strong>HDF (Hierarchical Data Format)</strong> is a file format that is designed to store and organise large and complex data structures. It can store a wide variety of data types, including multidimensional arrays, tables, and hierarchical data. HDF files are commonly used in fields such as remote sensing, astronomy, and engineering.</p></li>
<li><p>Again, <strong>GRIB (GRIdded Binary)</strong> files are similar to NetCDF files. They are a binary file format used to store meteorological and oceanographic data. It can store gridded data, such as atmospheric and oceanic model output, in a compact and efficient binary format. GRIB files are commonly used by weather forecasting agencies and research organisations.</p></li>
</ul>
<p>Compared to CSV files, these file formats offer several benefits for storing, sharing complex datasets:</p>
<ul>
<li><p><strong>Support for multidimensional arrays</strong> These file formats can store and handle multidimensional arrays, which cannot be represented in a CSV file. However, they can be <em>exported</em> as CSV files, often after subsetting the data, but the resultant CSV files consume significant amounts of disk space.</p></li>
<li><p><strong>Efficient storage</strong> Binary file formats can be more compact and efficient than text-based formats such as CSV files, which can save disk space and make it easier to share and transfer large datasets.</p></li>
<li><p><strong>Memory use efficiency</strong> NetCDF, GRIB, and HDF files are better for memory use efficiency compared to CSV files because they can store multidimensional arrays and metadata in a compact binary format, which can save disk space and memory when working with large and complex datasets. Also, they do not have to be read into memory all at once.</p></li>
<li><p><strong>Self-describing metadata</strong> These file formats can include metadata, such as variable names and units, which are self-describing and can be easily accessed and understood by other researchers and software.</p></li>
<li><p><strong>Support for compression</strong> Binary file formats can support compression, which can further reduce file size and make it easier to share and transfer large datasets.</p></li>
</ul>
<p>The various efficiencies mentioned above may be offset by them being quite challenging to work with, and as such novices might experience steep learning curves.</p>
</section><section id="larger-than-memory-data" class="level2" data-number="3.6"><h2 data-number="3.6" class="anchored" data-anchor-id="larger-than-memory-data">
<span class="header-section-number">3.6</span> Larger than Memory Data</h2>
<p>Above we dealt with data that fit into your computer’s memory (RAM). However, there are many datasets that are too large to fit into memory, and as such, we need to use alternative methods to work with them. These methods include:</p>
<ul>
<li>Apache Arrow in the <code>arrow</code> package in R, which has support for the ‘feather’ file format, ‘parquet’ files</li>
<li>DuckDB in the <code>duckdb</code> package in R, which create a database on disk, can be queried using SQL</li>
</ul>
<p>I will develop vignettes for these in the future. We will not use these in this course, but it is important to be aware of them.</p>
</section></section><section id="file-extensions" class="level1" data-number="4"><h1 data-number="4">
<span class="header-section-number">4</span> File Extensions</h1>
<p>This section is about teaching you the value of knowing what to expect of a file, and how the data are represented inside it, by simply inspecting how it is named rather than relying on what software choose to display by default. File extensions are one of the simplest points at which this representation can be made either explicit or opaque, and they provide a real example of how small interface choices might affect analytical error.</p>
<p>File extensions are three or four character suffixes added to the end of a filename, usually preceded by a period (e.g., <code>.txt</code>, <code>.jpg</code>, <code>.pdf</code>, <code>.csv</code>, <code>.xlsx</code>). These extensions indicate the format or type of the file, providing information about the content and structure of the data within the file. Some file extensions are particular to Windows (such as <code>.exe</code>) and others to MacOS (<em>e.g.</em>, <code>.dmg</code>), but many are transportable across operating systems. File extensions help both the operating system and applications recognise the file type and determine which program or application should be used to open, view, or edit the file.</p>
<p>The default setting for Windows computers is for file extensions to be hidden when files are viewed in the Windows Explorer. This is silly and irresponsible, and a frequent major source of frustration, annoyance, and irritability when I try to help a student with an issue on their computers. Displaying file extensions on your computer is essential for effective data science work because it allows us to quickly identify, manage various file types that we may encounter during our projects.</p>
<p>Some reasons why it is important to display file extensions if we are to effectively use your computer in scientific computing applications:</p>
<ul>
<li><p><strong>Differentiate file formats</strong> When analysing data, you will often work with multiple file formats, such as CSV, TSV, Excel, JSON, and others. By displaying the file extensions you can easily differentiate between these formats so that you may use the most appropriate tools and methods to process and analyse your data.</p></li>
<li><p><strong>Prevent errors</strong> Knowing the exact file type helps you avoid errors when importing or exporting data, as different formats require specific handling techniques. For instance, using a CSV file reader on a TSV file could lead to unexpected outcomes.</p></li>
<li><p><strong>Improve file organisation</strong> When working on complex projects, it is important that you maintain an organised file structure. Being able to see file extensions at a glance helps you keep track of various data files, scripts, and output files, and it will be easier for you to manage your project, ensure its reproducibility.</p></li>
<li><p><strong>Enhance security</strong> Displaying file extensions can also help you identify potentially malicious files disguised as legitimate ones. For example, a file with an extension ‘.txt.exe’ might appear to be a harmless text file if the extension is hidden, but it is actually an executable file that could be harmful.</p></li>
<li><p><strong>Facilitate collaboration</strong> When sharing files with colleagues or collaborators, knowing the file format is essential for smooth communication, seamless collaboration. Displaying file extensions ensures that everyone is aware of the file types being used and can handle them accordingly.</p></li>
</ul>
<p>To display file extensions on your computer, you will need to adjust the settings in your operating system’s file explorer. The specific steps to do this vary depending on whether you are using Windows, MacOS, or a Linux-based system. A quick online search will provide you with the necessary instructions. Since most people use Windows, I have included the instructions for Windose user in the box, below, titled “Displaying File Extensions in Windows”.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Displaying File Extension in Windows
</div>
</div>
<div class="callout-body-container callout-body">
<p>Be a responsible adult and display the file extensions on your computer. To display file extensions in Windows Explorer on the latest version of Windows (assuming Windows 11), follow these steps:</p>
<ol type="1">
<li>Open a new File Explorer window by clicking on the File Explorer icon on the taskbar or pressing the ‘Windows key + E’ on your keyboard.</li>
<li>In the File Explorer window, click on the three horizontal dots in the upper-right corner to open the ‘More options’ menu.</li>
<li>From the ‘More options’ menu, click on ‘Options’ to open the ‘Folder Options’ window.</li>
<li>In the ‘Folder Options’ window, switch to the ‘View’ tab.</li>
<li>Under the ‘Advanced settings’ list, locate the option ‘Hide extensions for known file types’, uncheck the box next to it.</li>
<li>Click ‘Apply’ to save your changes, and then click ‘OK’ to close the ‘Folder Options’ window.</li>
</ol>
<p>Now, the file extensions should be visible for all files in Windows Explorer. Remember that this setting applies to all folders on your computer. If you want to revert to hiding the file extensions, simply follow the same steps, check the box next to ‘Hide extensions for known file types’ in the ‘View’ tab of the ‘Folder Options’ window.</p>
</div>
</div>
</section><section id="r-code-conventions" class="level1" data-number="5"><h1 data-number="5">
<span class="header-section-number">5</span> R Code Conventions</h1>
<p>When writing code in support of a statistical data analysis in R, the primary aim is of course to produce the correct output. Equally important, though, is to produce work that remains intelligible, inspectable, and usable long after it has been written. The conventions below should therefore serve as practical expressions of three main principles: <em>consistency</em>, <em>legibility</em> to others, and future <em>recoverability</em>. Good code is disciplined writing, and the practices below follow from that discipline.</p>
<p><strong>Consistency</strong> supports readability across an analysis. A consistent coding style allows both humans and software to detect structure without re-interpretation. Adopting an established style guide (such as the <a href="https://style.tidyverse.org/"><strong>tidyverse</strong> style guide</a>, the <a href="http://adv-r.had.co.nz/Style.html">R Style Guide</a>, or <a href="https://google.github.io/styleguide/Rguide.html">Google’s R Style Guide</a>) provides a stable baseline for decisions about spacing, indentation, line breaks, and operator placement. The specific guide matters less than applying one style coherently throughout a project. Inconsistent formatting increases cognitive load and obscures logical structure, even when the code itself is correct. You want to ensure your code is a pleasure to read by others — in this first instance, especially your instructors responsible for marking your reports.</p>
<p>Practices such as regular indentation, consistent use of spaces around operators and commas, and sensible line breaking all accomplish this goal. Long lines should be split in ways that preserve syntactic clarity, for example by leaving brackets open or ending a line with an operator (<code>+</code> in <strong>ggplot2</strong> and <code>|&gt;</code> elsewhere in the <strong>tidyverse</strong>) so that continuation is unambiguous to R and to the reader.</p>
<p><strong>Legibility to others</strong> concerns whether the intent of your analysis is visible without reconstruction, and without ambiguity. Code is read far more often than it is written, and this includes being read by your future self. Meaningful variable and function names make data transformations interpretable at a glance, and it reduces the need for explanatory comments (although these are still necessary). Where code is complex, comments should clarify <em>why</em> something is done, not merely restate <em>what</em> the code does.</p>
<p>Modularising code into small, well-named functions further supports legibility by isolating conceptual units of work. This makes analyses easier to follow, easier to debug (problems arise often!), and easier to extend. Code that can only be understood as a continuous script quickly becomes opaque once it grows beyond trivial size.</p>
<p><strong>Future recoverability</strong> addresses whether an analysis can be resumed, audited, or extended once its original context has faded. Version control systems such as Git, used via platforms like GitHub or GitLab, provide an explicit record of how code evolves and allow earlier states to be recovered without ambiguity. Organising projects into clear directory structures (for data, scripts, and outputs) serves the same purpose at the file-system level.</p>
<p>Recoverability also depends on making dependencies explicit. All required packages should be declared and loaded deliberately upfront so that analyses do not rely on accidental features of a particular session. Relative file paths should be used wherever possible to avoid embedding machine-specific assumptions. Where appropriate, tests can be written to verify that functions behave as expected, providing early warning when changes introduce unintended effects.</p>
<p>It is worth noting that conventions vary across programming languages and research communities. What matters is not universal agreement on style, but internal coherence. Once a set of conventions is chosen, they should be applied consistently throughout an analysis so that structure, intent, and provenance remain visible.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Naming Variables
</div>
</div>
<div class="callout-body-container callout-body">
<p>R imposes strict rules on variable naming, and violating them leads to errors that are often frustrating to diagnose. Variable names should never contain spaces or special characters, and should avoid masking existing function names (such as <code>mean</code>). As a general rule, spaces should be avoided not only in variable names, but also in file and directory names used in a project.</p>
<p>Several naming conventions are commonly encountered:</p>
<p><strong>Pascal case</strong> capitalises the first letter of each word and is often used for classes or types (e.g.&nbsp;<code>SpeciesName</code>, <code>PetalColour</code>, <code>FunctionalType</code>).</p>
<p><strong>Snake case</strong> separates words with underscores and uses lowercase letters, and is widely used in scripting and file naming (e.g.&nbsp;<code>species_name</code>, <code>petal_colour</code>, <code>functional_type</code>).</p>
<p><strong>Camel case</strong> capitalises words except for the first and is often used for variables or methods in object-oriented contexts (e.g.&nbsp;<code>speciesName</code>, <code>petalColour</code>, <code>functionalType</code>).</p>
<p><strong>Hungarian notation</strong> prefixes variable names with type indicators (e.g.&nbsp;<code>iSpeciesID</code>, <code>strPetalColour</code>), but is rarely used in modern R practice.</p>
<p>No naming convention is inherently superior. What matters is choosing one that supports readability and applying it consistently throughout a project.</p>
</div>
</div>
</section><section id="sec-reproducibility" class="level1" data-number="6"><h1 data-number="6">
<span class="header-section-number">6</span> Reproducible Research</h1>
<p>Reproducible research refers to the capacity for an analysis to be re-run, inspected, and verified using the same data and code, giving the same results. In this course, reproducibility is treated as a practical property of workflows: it depends on how data are represented, how code is written, and how analytical steps are organised over time. In advanced situations, such reprodicibility can recreate an entire scientific publication, or a thesis, from its initial data, code, and textual building blocks (see <a href="02-quarto.qmd">2. R Markdown and Quarto</a>). The aim here is to distinguish clearly between practices you are expected to implement now and tools you should recognise as part of the broader reproducibility landscape.</p>
<section id="required-practice-in-bcb744" class="level2" data-number="6.1"><h2 data-number="6.1" class="anchored" data-anchor-id="required-practice-in-bcb744">
<span class="header-section-number">6.1</span> Required practice in BCB744</h2>
<p>At this stage, reproducibility requires a small number of essential habits. These are the practices you are expected to adopt consistently and that will be evaluated in all assessments.</p>
<p>Analyses should be conducted using scripted code rather than manual interaction. Scripts provide an explicit record of analytical decisions and allow results to be regenerated without reliance on memory or interface state. Code should be organised within a coherent project structure, with separate locations for raw data, processed data, scripts, and outputs, so that analytical flow can be reconstructed with ease.</p>
<p>Version control forms part of this basic practice. Using Git (typically via platforms such as GitHub or GitLab) provides a transparent history of how analyses evolve and allows earlier states to be recovered if errors are introduced. This is not about public dissemination, but about maintaining an explicit record of change.</p>
<p>Literate programming tools such as R Markdown and Quarto are also part of required practice. These tools integrate code, output, and narrative text within a single document, making analytical reasoning visible alongside results. When data or code change, documents are regenerated rather than edited by hand, reducing the risk of divergence between analysis and interpretation.</p>
<p>Finally, reproducibility depends on making dependencies explicit. Required packages should be declared deliberately, relative file paths should be used to avoid machine-specific assumptions, and data files should be treated as fixed inputs rather than mutable artefacts. Together, these practices establish a workflow that can be re-run by others—or by your future self—without guesswork.</p>
</section><section id="a-look-to-the-future" class="level2" data-number="6.2"><h2 data-number="6.2" class="anchored" data-anchor-id="a-look-to-the-future">
<span class="header-section-number">6.2</span> A look to the future</h2>
<p>Beyond these core practices lies a wider R ecosystem of tools made to support reproducibility at larger scales or higher levels of complexity. You are not expected to use these tools in BCB744, but you should be aware of the problems they address and the contexts in which they become relevant.</p>
<p>Workflow management tools such as <strong>workflowr</strong> and <strong>targets</strong> formalise analytical pipelines by making dependencies between steps explicit and automating re-execution when inputs change. These tools are useful for large or long-running projects, but they introduce additional complexities that is unnecessary at this stage.</p>
<p>Containerisation technologies such as Docker provide isolated computational environments with fixed software versions and dependencies. They are widely used in collaborative or production settings where analyses must run identically across different machines. They are powerful, but sit beyond the scope of this course and should be seen as an extension of the principles already discussed, not as a prerequisite.</p>
<p>Testing frameworks such as <strong>testthat</strong> enable systematic verification of code behaviour, which becomes increasingly important as projects grow in size or are reused across contexts. At present, the emphasis is on writing code that is clear enough to inspect, rather than on formal test suites.</p>
<p>The purpose of introducing these tools here is not to encourage their immediate adoption, but to situate current practices within a broader trajectory. As projects scale, the same concerns (visibility, control, and recoverability) are addressed using increasingly formal mechanisms.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>The Mars Climate Orbiter Mission in 1998
</div>
</div>
<div class="callout-body-container callout-body">
<p>One of the most famous examples of unclear communication is the Mars Climate Orbiter mission in 1998. NASA lost the spacecraft due to a navigation error caused by a unit conversion mistake. The error occurred because one team used metric units (newtons), while another team used imperial units (pound-force) for a crucial spacecraft operation parameter. The discrepancy in units led to incorrect trajectory calculations, causing the Mars Climate Orbiter to approach Mars at a dangerously low altitude, ultimately disintegrate in the Martian atmosphere.</p>
<p>This incident highlights the importance of clear communication, proper data handling, and rigorous verification processes in mission-critical systems, including space missions. The event emphasises the need for using appropriate tools, software, and methodologies to minimise the risk of errors in complex engineering projects.</p>
</div>
</div>
</section></section><section id="exercise-from-excel-to-delimited-files-to-r" class="level1" data-number="7"><h1 data-number="7">
<span class="header-section-number">7</span> Exercise: From Excel to Delimited Files to R</h1>
<p><strong>Goal:</strong> You will (i) create a small dataset in Excel, (ii) export it to three delimited text formats (CSV, TSV, and semicolon-delimited), (iii) save them into a known location inside an R Project folder using disciplined file naming, and (iv) read <em>one</em> of the exported files from a new R script in RStudio.</p>
<section id="part-a-create-a-project-folder-with-known-locations" class="level2" data-number="7.1"><h2 data-number="7.1" class="anchored" data-anchor-id="part-a-create-a-project-folder-with-known-locations">
<span class="header-section-number">7.1</span> Part A — Create a Project folder with known locations</h2>
<ol type="1">
<li><p>In RStudio, create a new project: <strong>File → New Project → New Directory → New Project</strong>. Name it <code>bcb744_file_formats_exercise</code> and create it in a location you can find again (<em>e.g.</em>, Documents).</p></li>
<li><p>In the Files pane in RStudio (bottom right), confirm you can see the project folder and that it contains an <code>.Rproj</code> file.</p></li>
<li>
<p>Inside the project folder, create these subfolders:</p>
<ul>
<li><code>data-raw/</code></li>
<li><code>data/</code></li>
<li><code>scripts/</code></li>
</ul>
<p>(You can do this in the Files pane: <strong>New Folder</strong>.)</p>
</li>
<li>
<p>Decide on a file naming pattern you will follow for exports in this exercise (later, not now). Use this pattern exactly:</p>
<ul>
<li><code>survey_demo_YYYYMMDD_v01.csv</code></li>
<li><code>survey_demo_YYYYMMDD_v01.tsv</code></li>
<li><code>survey_demo_YYYYMMDD_v01.scsv</code></li>
</ul>
<p>Replace <code>YYYYMMDD</code> with today’s date. Keep <code>v01</code> as written.</p>
<p><em>Note:</em> <code>.scsv</code> here is just a human-visible label meaning “semicolon CSV”; it is still a plain text file.</p>
</li>
</ol></section><section id="part-b-create-a-small-dataset-in-excel" class="level2" data-number="7.2"><h2 data-number="7.2" class="anchored" data-anchor-id="part-b-create-a-small-dataset-in-excel">
<span class="header-section-number">7.2</span> Part B — Create a small dataset in Excel</h2>
<ol start="5" type="1">
<li><p>Open Excel and create a new blank workbook.</p></li>
<li>
<p>In row 1, enter these column names (exact spelling):</p>
<ul>
<li><code>id</code></li>
<li><code>species</code></li>
<li><code>site</code></li>
<li><code>temp_C</code></li>
<li><code>notes</code></li>
</ul>
</li>
<li>
<p>Enter at least <strong>10 rows</strong> of data. Include the following features on purpose:</p>
<ul>
<li>at least one missing value (leave one cell blank in <code>temp_C</code>)</li>
<li>at least one <code>notes</code> cell that contains a comma (<em>e.g.</em>, <code>juvenile, small</code>)</li>
<li>at least one <code>notes</code> cell that contains a semicolon (<em>e.g.</em>, <code>recheck; uncertain</code>)</li>
</ul>
</li>
<li>
<p>Save the Excel workbook into your project folder under <code>data-raw/</code> with this name:</p>
<ul>
<li><code>survey_demo_YYYYMMDD_source.xlsx</code></li>
</ul>
</li>
</ol></section><section id="part-c-export-three-delimited-text-files-into-the-project" class="level2" data-number="7.3"><h2 data-number="7.3" class="anchored" data-anchor-id="part-c-export-three-delimited-text-files-into-the-project">
<span class="header-section-number">7.3</span> Part C — Export three delimited text files into the project</h2>
<p>Your target location for <em>all exports</em> is:</p>
<ul>
<li><code>bcb744_file_formats_exercise/data/</code></li>
</ul>
<ol start="9" type="1">
<li>
<p>Export to <strong>CSV (comma delimited)</strong> using the naming scheme defined above:</p>
<ul>
<li>Use <strong>File → Save As</strong> (or <strong>Save a Copy</strong>) and choose <strong>CSV (Comma delimited)</strong>.</li>
<li>Save into the project’s <code>data/</code> folder as: <code>survey_demo_YYYYMMDD_v01.csv</code>
</li>
</ul>
</li>
<li><p>Export to <strong>TSV (tab delimited)</strong>:</p></li>
</ol>
<ul>
<li>Use <strong>Save As</strong> and choose <strong>Text (Tab delimited)</strong> (or equivalent).</li>
<li>Save into the project’s <code>data/</code> folder as: <code>survey_demo_YYYYMMDD_v01.tsv</code>
</li>
</ul>
<ol start="11" type="1">
<li>Export to <strong>semicolon-delimited</strong> (“semi-colon derived derivative”):</li>
</ol>
<ul>
<li><p>If your Excel offers a direct <strong>CSV (Semicolon delimited)</strong> option, use it and save as: <code>survey_demo_YYYYMMDD_v01.scsv</code></p></li>
<li>
<p>If it does <em>not</em> offer that option, do this instead:</p>
<ol type="1">
<li>Open the saved CSV file (<code>survey_demo_YYYYMMDD_v01.csv</code>) in a plain text editor (Notepad / TextEdit / VS Code).</li>
<li>Save a new copy into the same <code>data/</code> folder named <code>survey_demo_YYYYMMDD_v01.scsv</code>.</li>
<li>In that <code>.scsv</code> copy, replace field separators <strong>commas that function as delimiters</strong> with semicolons. Do not alter commas inside quoted text fields (your <code>notes</code> column will reveal why this matters).</li>
</ol>
</li>
</ul>
<ol start="12" type="1">
<li>In your file explorer (Windows Explorer / Finder), navigate to the project’s <code>data/</code> folder and confirm that these three files exist there:</li>
</ol>
<ul>
<li><code>survey_demo_YYYYMMDD_v01.csv</code></li>
<li><code>survey_demo_YYYYMMDD_v01.tsv</code></li>
<li><code>survey_demo_YYYYMMDD_v01.scsv</code></li>
</ul></section><section id="part-d-read-one-exported-file-from-a-new-r-script-in-rstudio" class="level2" data-number="7.4"><h2 data-number="7.4" class="anchored" data-anchor-id="part-d-read-one-exported-file-from-a-new-r-script-in-rstudio">
<span class="header-section-number">7.4</span> Part D — Read one exported file from a new R script in RStudio</h2>
<ol start="13" type="1">
<li><p>In RStudio, create a new script: <strong>File → New File → R Script</strong>.</p></li>
<li><p>Save the script into <code>scripts/</code> with this name:</p></li>
</ol>
<ul>
<li><code>01_read_delimited_files.R</code></li>
</ul>
<ol start="15" type="1">
<li>Copy and run the following lines <strong>one at a time</strong>, top to bottom.</li>
</ol>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># 1) Confirm you are working inside the project</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">getwd</span>()</span>
<span id="cb1-3"><a href="#cb1-3"></a></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="co"># 2) List the exported files (you should see your three exports)</span></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">list.files</span>(<span class="st">"data"</span>)</span>
<span id="cb1-6"><a href="#cb1-6"></a></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="co"># 3) Choose ONE file to read first (edit the filename to match your date)</span></span>
<span id="cb1-8"><a href="#cb1-8"></a>file_csv  <span class="ot">&lt;-</span> <span class="st">"data/survey_demo_YYYYMMDD_v01.csv"</span></span>
<span id="cb1-9"><a href="#cb1-9"></a>file_tsv  <span class="ot">&lt;-</span> <span class="st">"data/survey_demo_YYYYMMDD_v01.tsv"</span></span>
<span id="cb1-10"><a href="#cb1-10"></a>file_scsv <span class="ot">&lt;-</span> <span class="st">"data/survey_demo_YYYYMMDD_v01.scsv"</span></span>
<span id="cb1-11"><a href="#cb1-11"></a></span>
<span id="cb1-12"><a href="#cb1-12"></a><span class="co"># 4) Read one of them (start with TSV to avoid comma ambiguity in notes)</span></span>
<span id="cb1-13"><a href="#cb1-13"></a>dat <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(file_tsv, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-14"><a href="#cb1-14"></a></span>
<span id="cb1-15"><a href="#cb1-15"></a><span class="co"># 5) Inspect what you imported</span></span>
<span id="cb1-16"><a href="#cb1-16"></a><span class="fu">str</span>(dat)</span>
<span id="cb1-17"><a href="#cb1-17"></a><span class="fu">head</span>(dat, <span class="dv">3</span>)</span>
<span id="cb1-18"><a href="#cb1-18"></a></span>
<span id="cb1-19"><a href="#cb1-19"></a><span class="co"># 6) Basic checks: dimensions and missingness</span></span>
<span id="cb1-20"><a href="#cb1-20"></a><span class="fu">dim</span>(dat)</span>
<span id="cb1-21"><a href="#cb1-21"></a><span class="fu">colSums</span>(<span class="fu">is.na</span>(dat))</span>
<span id="cb1-22"><a href="#cb1-22"></a></span>
<span id="cb1-23"><a href="#cb1-23"></a><span class="co"># 7) Now read the comma CSV version and compare</span></span>
<span id="cb1-24"><a href="#cb1-24"></a>dat_csv <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(file_csv, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-25"><a href="#cb1-25"></a><span class="fu">identical</span>(<span class="fu">names</span>(dat), <span class="fu">names</span>(dat_csv))</span>
<span id="cb1-26"><a href="#cb1-26"></a></span>
<span id="cb1-27"><a href="#cb1-27"></a><span class="co"># 8) Read the semicolon-delimited version explicitly</span></span>
<span id="cb1-28"><a href="#cb1-28"></a>dat_scsv <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(file_scsv, <span class="at">sep =</span> <span class="st">";"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-29"><a href="#cb1-29"></a></span>
<span id="cb1-30"><a href="#cb1-30"></a><span class="co"># 9) Compare what happened to the notes column across imports</span></span>
<span id="cb1-31"><a href="#cb1-31"></a><span class="fu">head</span>(dat<span class="sc">$</span>notes, <span class="dv">3</span>)</span>
<span id="cb1-32"><a href="#cb1-32"></a><span class="fu">head</span>(dat_csv<span class="sc">$</span>notes, <span class="dv">3</span>)</span>
<span id="cb1-33"><a href="#cb1-33"></a><span class="fu">head</span>(dat_scsv<span class="sc">$</span>notes, <span class="dv">3</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<ol start="16" type="1">
<li>When you are done, answer these questions in a short paragraph in your notebook or as comments at the bottom of the script:</li>
</ol>
<ol type="a">
<li>Which file type imported with the least friction, and why?</li>
<li>Did any import alter the <code>notes</code> column in a way you did not expect? What does that suggest about delimiter choice?</li>
<li>Which steps in your workflow made it easy to locate the files again without clicking through folders?</li>
</ol>


<!-- -->

</section></section><div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-reuse"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a></div></div></section><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{smit,_a._j.2021,
  author = {Smit, A. J.,},
  title = {2. {Working} with {Data} \&amp; {Code}},
  date = {2021-01-01},
  url = {http://tangledbank.netlify.app/BCB744/intro_r/02-working-with-data.html},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-smit,_a._j.2021" class="csl-entry quarto-appendix-citeas" role="listitem">
Smit, A. J. (2021) 2. Working with Data &amp; Code. <a href="http://tangledbank.netlify.app/BCB744/intro_r/02-working-with-data.html">http://tangledbank.netlify.app/BCB744/intro_r/02-working-with-data.html.
</a>
</div></div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("http:\/\/tangledbank\.netlify\.app");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
            // default icon
            link.classList.add("external");
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="../../BCB744/intro_r/01-RStudio.html" class="pagination-link" aria-label="1. R &amp; RStudio">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">1. R &amp; RStudio</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../BCB744/intro_r/03-quarto.html" class="pagination-link" aria-label="3. R Markdown and Quarto">
        <span class="nav-page-text">3. R Markdown and Quarto</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb2" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb2-1"><a href="#cb2-1"></a><span class="co">---</span></span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="an">date:</span><span class="co"> "2021-01-01"</span></span>
<span id="cb2-3"><a href="#cb2-3"></a><span class="an">title:</span><span class="co"> "2. Working with Data &amp; Code"</span></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="co">---</span></span>
<span id="cb2-5"><a href="#cb2-5"></a></span>
<span id="cb2-6"><a href="#cb2-6"></a><span class="al">![](../../images/PhD_comics2.PNG)</span>{width=100%}</span>
<span id="cb2-7"><a href="#cb2-7"></a></span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="at">&gt; *"The plural of anecdote is not data."*</span></span>
<span id="cb2-9"><a href="#cb2-9"></a><span class="at">&gt;</span></span>
<span id="cb2-10"><a href="#cb2-10"></a><span class="at">&gt; — Roger Brinner</span></span>
<span id="cb2-11"><a href="#cb2-11"></a></span>
<span id="cb2-12"><a href="#cb2-12"></a><span class="in">```{r code-prelim-opts, echo=FALSE}</span></span>
<span id="cb2-13"><a href="#cb2-13"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(</span>
<span id="cb2-14"><a href="#cb2-14"></a>  <span class="at">comment =</span> <span class="st">"R&gt;"</span>, </span>
<span id="cb2-15"><a href="#cb2-15"></a>  <span class="at">warning =</span> <span class="cn">FALSE</span>, </span>
<span id="cb2-16"><a href="#cb2-16"></a>  <span class="at">message =</span> <span class="cn">FALSE</span>,</span>
<span id="cb2-17"><a href="#cb2-17"></a>  <span class="at">out.width =</span> <span class="st">'70%'</span></span>
<span id="cb2-18"><a href="#cb2-18"></a>)</span>
<span id="cb2-19"><a href="#cb2-19"></a><span class="in">```</span></span>
<span id="cb2-20"><a href="#cb2-20"></a></span>
<span id="cb2-21"><a href="#cb2-21"></a>In this Chapter we will cover:</span>
<span id="cb2-22"><a href="#cb2-22"></a></span>
<span id="cb2-23"><a href="#cb2-23"></a><span class="ss">* </span>Data file formats</span>
<span id="cb2-24"><a href="#cb2-24"></a><span class="ss">* </span>The use of file extensions</span>
<span id="cb2-25"><a href="#cb2-25"></a><span class="ss">* </span>Review some R code conventions</span>
<span id="cb2-26"><a href="#cb2-26"></a><span class="ss">* </span>Introduce the concept of reproducible research</span>
<span id="cb2-27"><a href="#cb2-27"></a></span>
<span id="cb2-28"><a href="#cb2-28"></a><span class="fu"># Introduction</span></span>
<span id="cb2-29"><a href="#cb2-29"></a></span>
<span id="cb2-30"><a href="#cb2-30"></a>In today's rapidly evolving scientific fields, understanding the basic principles of coding (also called scripting or programming) and data handling has become indispensable. The complexity and volume of data continue to grow, and traditional manual data analysis methods are no longer sufficient to deal with the demands of research as practised today. Coding and data manipulation skills will unlock the potential of your computer and allow you to analyse datasets, fit models, and reveal hidden patterns that may not be apparent through conventional methods. Proficiency in coding and data handling will help you to collaborate effectively with multidisciplinary teams (as is increasingly the case today), access data from various data sources in different formats, cross the gap between theory and practical applications, and contribute to the advancement of scientific knowledge in a transparent and efficient way.</span>
<span id="cb2-31"><a href="#cb2-31"></a></span>
<span id="cb2-32"><a href="#cb2-32"></a>A useful way to understand the material in this chapter is to recognise that most failures in data analysis do not arise from advanced statistical methods, but from mismatches at the interface between how and where data are stored, how software interprets those representations, and how we interact with both. File systems organise data, file formats encode assumptions, software enforces expectations, and you, the budding scientist, bring good and bad habits into that system. </span>
<span id="cb2-33"><a href="#cb2-33"></a></span>
<span id="cb2-34"><a href="#cb2-34"></a>Each section in this chapter addresses a different point at which data can be misrepresented, misunderstood, or clarified, and the aim throughout is to make those points visible.</span>
<span id="cb2-35"><a href="#cb2-35"></a></span>
<span id="cb2-36"><a href="#cb2-36"></a>Throughout this discussion there are a few underlying assumptions which organise the practical guidance that follows.</span>
<span id="cb2-37"><a href="#cb2-37"></a></span>
<span id="cb2-38"><a href="#cb2-38"></a><span class="ss">1. </span>Data have structure independent of the software used to store or analyse them.</span>
<span id="cb2-39"><a href="#cb2-39"></a><span class="ss">2. </span>Software necessarily imposes assumptions on that structure, whether explicitly or by default behaviour.</span>
<span id="cb2-40"><a href="#cb2-40"></a><span class="ss">3. </span>Reproducibility is as much a property of workflows and data representations *and* of intention or care.</span>
<span id="cb2-41"><a href="#cb2-41"></a></span>
<span id="cb2-42"><a href="#cb2-42"></a>Below, I focus on how they play out in routine analytical choices. First --- before we look at how data are stored --- let's look at *how* and *where* to find data and the outputs of your analyses once they reside on your computer.</span>
<span id="cb2-43"><a href="#cb2-43"></a></span>
<span id="cb2-44"><a href="#cb2-44"></a><span class="fu"># Navigating the File System and the Working Directory</span></span>
<span id="cb2-45"><a href="#cb2-45"></a></span>
<span id="cb2-46"><a href="#cb2-46"></a>Many analytical errors arise from confusion about where data and code actually live on a computer. Before software can read a file, the you must specify *where* that file is located. Where the file lives is not random, but poor understanding of how to navigate your computer's file system makes it feel haphazard at best or like a black box at worst. You *should store files deliberately in precisely known locations*. This requires a working grasp of the file system: how files are organised, how locations are described, and how software interprets those descriptions.</span>
<span id="cb2-47"><a href="#cb2-47"></a></span>
<span id="cb2-48"><a href="#cb2-48"></a>So, before R can read data or write results, it must know where it is working. This sounds trivial but many failures you'll encounter in the early days on your journey to becoming a biostatistician or data analyst trace back to uncertainty about file locations rather than to coding errors. This section introduces the minimal concepts required to reason correctly about files, directories, and paths, and explains how R and RStudio use these ideas in practice.</span>
<span id="cb2-49"><a href="#cb2-49"></a></span>
<span id="cb2-50"><a href="#cb2-50"></a>A **file system** is the hierarchical structure by which an operating system stores and organises files. At the top of this hierarchy sits a *root*, from which all folders (directories) and files descend. Every file therefore occupies a unique position within this structure, defined by the sequence of directories leading to it.</span>
<span id="cb2-51"><a href="#cb2-51"></a></span>
<span id="cb2-52"><a href="#cb2-52"></a>A **directory** (or *folder*) is simply a container that can hold files and other directories. When a directory sits inside another directory, it is a **sub-directory**. This nesting can continue indefinitely, producing a tree-like structure. Although Windows, macOS, and Linux differ in visual presentation, this underlying structure is shared across all three systems.</span>
<span id="cb2-53"><a href="#cb2-53"></a></span>
<span id="cb2-54"><a href="#cb2-54"></a>To refer to a file, we use a **file path**: a textual description of its location within the file system. You can observe these file paths within RStudio, near the top of the **Console**.</span>
<span id="cb2-55"><a href="#cb2-55"></a></span>
<span id="cb2-56"><a href="#cb2-56"></a>An **absolute path** specifies a file’s location starting from the root of the file system. Because it begins at the root, it uniquely identifies the file regardless of the current working context.</span>
<span id="cb2-57"><a href="#cb2-57"></a></span>
<span id="cb2-58"><a href="#cb2-58"></a>**Examples:**</span>
<span id="cb2-59"><a href="#cb2-59"></a></span>
<span id="cb2-60"><a href="#cb2-60"></a>Windows:</span>
<span id="cb2-61"><a href="#cb2-61"></a></span>
<span id="cb2-62"><a href="#cb2-62"></a><span class="in">`C:\Users\Student\Documents\BCB744\data\kelp.csv`</span></span>
<span id="cb2-63"><a href="#cb2-63"></a></span>
<span id="cb2-64"><a href="#cb2-64"></a>macOS / Linux:</span>
<span id="cb2-65"><a href="#cb2-65"></a></span>
<span id="cb2-66"><a href="#cb2-66"></a><span class="in">`/home/student/BCB744/data/kelp.csv`</span></span>
<span id="cb2-67"><a href="#cb2-67"></a></span>
<span id="cb2-68"><a href="#cb2-68"></a>Notice that Windows uses backslashes <span class="in">`\`</span> and maxOS and Linux use forward slashes <span class="in">`/`</span>. This is a very important distinction between Windows and Unix-like operating systems.</span>
<span id="cb2-69"><a href="#cb2-69"></a></span>
<span id="cb2-70"><a href="#cb2-70"></a>Absolute paths are unambiguous, but they are not very helpful when moving R (or other) scripts between different users. This is because they encode machine-specific details (usernames, drive letters) that differ between those users' computers.</span>
<span id="cb2-71"><a href="#cb2-71"></a></span>
<span id="cb2-72"><a href="#cb2-72"></a>A **relative path** specifies a file’s location *relative to the current working directory*. It does not begin at the root, but instead assumes a starting point. Relative paths are shorter, portable, and central to reproducible workflows.</span>
<span id="cb2-73"><a href="#cb2-73"></a></span>
<span id="cb2-74"><a href="#cb2-74"></a>Assume the current working directory is:</span>
<span id="cb2-75"><a href="#cb2-75"></a></span>
<span id="cb2-76"><a href="#cb2-76"></a><span class="ss">* </span><span class="in">`BCB744/`</span></span>
<span id="cb2-77"><a href="#cb2-77"></a></span>
<span id="cb2-78"><a href="#cb2-78"></a>Then:</span>
<span id="cb2-79"><a href="#cb2-79"></a></span>
<span id="cb2-80"><a href="#cb2-80"></a><span class="ss">* </span><span class="in">`data/kelp.csv`</span> refers to a file called <span class="in">`kelp.csv`</span> inside the <span class="in">`data`</span> sub-directory.</span>
<span id="cb2-81"><a href="#cb2-81"></a><span class="ss">* </span><span class="in">`scripts/analysis.R`</span> refers to an R file named <span class="in">`analysis.R`</span> inside the <span class="in">`scripts`</span> sub-directory.</span>
<span id="cb2-82"><a href="#cb2-82"></a></span>
<span id="cb2-83"><a href="#cb2-83"></a>Two special symbols appear frequently in relative paths:</span>
<span id="cb2-84"><a href="#cb2-84"></a></span>
<span id="cb2-85"><a href="#cb2-85"></a><span class="ss">* </span><span class="in">`.`</span> refers to the current directory.</span>
<span id="cb2-86"><a href="#cb2-86"></a><span class="ss">* </span><span class="in">`..`</span> refers to the parent directory (one level up).</span>
<span id="cb2-87"><a href="#cb2-87"></a></span>
<span id="cb2-88"><a href="#cb2-88"></a>Examples:</span>
<span id="cb2-89"><a href="#cb2-89"></a></span>
<span id="cb2-90"><a href="#cb2-90"></a><span class="ss">* </span><span class="in">`../data/kelp.csv`</span> means “go up one directory, then into <span class="in">`data`</span>”.</span>
<span id="cb2-91"><a href="#cb2-91"></a><span class="ss">* </span><span class="in">`../../raw/sst.nc`</span> means “go up two levels, then into <span class="in">`raw`</span>”.</span>
<span id="cb2-92"><a href="#cb2-92"></a></span>
<span id="cb2-93"><a href="#cb2-93"></a><span class="fu">## The Working Directory</span></span>
<span id="cb2-94"><a href="#cb2-94"></a></span>
<span id="cb2-95"><a href="#cb2-95"></a>The **working directory** is the directory that R treats as its current location. All relative paths are interpreted with respect to this directory.</span>
<span id="cb2-96"><a href="#cb2-96"></a></span>
<span id="cb2-97"><a href="#cb2-97"></a>You can think of the working directory as the answer to the question:</span>
<span id="cb2-98"><a href="#cb2-98"></a>  </span>
<span id="cb2-99"><a href="#cb2-99"></a><span class="at">&gt; “If I refer to a file without giving its full address, where will R look?”*</span></span>
<span id="cb2-100"><a href="#cb2-100"></a>  </span>
<span id="cb2-101"><a href="#cb2-101"></a>In R, the working directory can be inspected with:</span>
<span id="cb2-102"><a href="#cb2-102"></a>  </span>
<span id="cb2-103"><a href="#cb2-103"></a><span class="ss">* </span><span class="in">`getwd()`</span></span>
<span id="cb2-104"><a href="#cb2-104"></a></span>
<span id="cb2-105"><a href="#cb2-105"></a>and (technically) changed with:</span>
<span id="cb2-106"><a href="#cb2-106"></a>  </span>
<span id="cb2-107"><a href="#cb2-107"></a><span class="ss">* </span><span class="in">`setwd("some/path")`</span></span>
<span id="cb2-108"><a href="#cb2-108"></a></span>
<span id="cb2-109"><a href="#cb2-109"></a>Beginners sometimes try to fix file-not-found errors by repeatedly calling <span class="in">`setwd()`</span>. This usually makes things worse. Changing the working directory inside scripts creates hidden assumptions about where the code is run and breaks reproducibility.</span>
<span id="cb2-110"><a href="#cb2-110"></a></span>
<span id="cb2-111"><a href="#cb2-111"></a>The preferred solution is not to move the working directory around, but to **control it once**, deliberately.</span>
<span id="cb2-112"><a href="#cb2-112"></a></span>
<span id="cb2-113"><a href="#cb2-113"></a><span class="fu">## RStudio Projects</span></span>
<span id="cb2-114"><a href="#cb2-114"></a></span>
<span id="cb2-115"><a href="#cb2-115"></a>An **RStudio Project** defines a directory as the root of an analysis and sets it as the working directory whenever the project is opened. This has three consequences that are foundational for this course:</span>
<span id="cb2-116"><a href="#cb2-116"></a>  </span>
<span id="cb2-117"><a href="#cb2-117"></a><span class="ss">1. </span><span class="in">`getwd()`</span> always returns the project root.</span>
<span id="cb2-118"><a href="#cb2-118"></a><span class="ss">2. </span>All scripts, data, and outputs can be referenced using relative paths.</span>
<span id="cb2-119"><a href="#cb2-119"></a><span class="ss">3. </span>Code runs the same way on different computers.</span>
<span id="cb2-120"><a href="#cb2-120"></a></span>
<span id="cb2-121"><a href="#cb2-121"></a>A typical project structure might look like this:</span>
<span id="cb2-122"><a href="#cb2-122"></a>  </span>
<span id="cb2-123"><a href="#cb2-123"></a><span class="ss">* </span><span class="in">`BCB744_project/`</span></span>
<span id="cb2-124"><a href="#cb2-124"></a></span>
<span id="cb2-125"><a href="#cb2-125"></a>With sub-folders specific to the different purposes:</span>
<span id="cb2-126"><a href="#cb2-126"></a></span>
<span id="cb2-127"><a href="#cb2-127"></a><span class="ss">* </span><span class="in">`data/`</span></span>
<span id="cb2-128"><a href="#cb2-128"></a><span class="ss">* </span><span class="in">`scripts/`</span></span>
<span id="cb2-129"><a href="#cb2-129"></a><span class="ss">* </span><span class="in">`figures/`</span></span>
<span id="cb2-130"><a href="#cb2-130"></a><span class="ss">* </span><span class="in">`reports/`</span></span>
<span id="cb2-131"><a href="#cb2-131"></a></span>
<span id="cb2-132"><a href="#cb2-132"></a>If this directory is an RStudio Project, then a script inside <span class="in">`scripts/`</span> can read data using:</span>
<span id="cb2-133"><a href="#cb2-133"></a>  </span>
<span id="cb2-134"><a href="#cb2-134"></a><span class="ss">* </span><span class="in">`data/kelp.csv`</span></span>
<span id="cb2-135"><a href="#cb2-135"></a></span>
<span id="cb2-136"><a href="#cb2-136"></a>without calling <span class="in">`setwd()`</span>, and without caring where the project lives on the machine.</span>
<span id="cb2-137"><a href="#cb2-137"></a></span>
<span id="cb2-138"><a href="#cb2-138"></a>This is why projects are treated as non-negotiable infrastructure in this course. They make file paths explicit, stable, and inspectable.</span>
<span id="cb2-139"><a href="#cb2-139"></a></span>
<span id="cb2-140"><a href="#cb2-140"></a>Software such as R interprets all file paths relative to its **working directory**—the directory it currently considers as “here”. Confusion about file paths is therefore almost always confusion about the working directory. When files cannot be found, the problem is rarely the file itself; it is the path used to describe it.</span>
<span id="cb2-141"><a href="#cb2-141"></a></span>
<span id="cb2-142"><a href="#cb2-142"></a>For this reason, analyses should be organised as **projects**, with a clear internal structure. A minimal example might look like this:</span>
<span id="cb2-143"><a href="#cb2-143"></a></span>
<span id="cb2-144"><a href="#cb2-144"></a><span class="ss">* </span><span class="in">`BCB744/`</span></span>
<span id="cb2-145"><a href="#cb2-145"></a></span>
<span id="cb2-146"><a href="#cb2-146"></a><span class="ss">  * </span><span class="in">`data/`</span> ... (raw or cleaned data)</span>
<span id="cb2-147"><a href="#cb2-147"></a><span class="ss">  * </span><span class="in">`scripts/`</span> ... (R scripts)</span>
<span id="cb2-148"><a href="#cb2-148"></a><span class="ss">  * </span><span class="in">`figures/`</span> ... (output plots)</span>
<span id="cb2-149"><a href="#cb2-149"></a><span class="ss">  * </span><span class="in">`reports/`</span> ... (Quarto or R Markdown documents)</span>
<span id="cb2-150"><a href="#cb2-150"></a></span>
<span id="cb2-151"><a href="#cb2-151"></a>When code is run from the project root, all file access can use relative paths, avoiding hard-coded references to specific machines.</span>
<span id="cb2-152"><a href="#cb2-152"></a></span>
<span id="cb2-153"><a href="#cb2-153"></a><span class="fu">## Exercises</span></span>
<span id="cb2-154"><a href="#cb2-154"></a></span>
<span id="cb2-155"><a href="#cb2-155"></a><span class="ss">1. </span>On your computer, create a directory called <span class="in">`filesystem_practice`</span>. Inside it, create three sub-directories: <span class="in">`data`</span>, <span class="in">`scripts`</span>, and <span class="in">`output`</span>. Place a small text file inside <span class="in">`data`</span>.</span>
<span id="cb2-156"><a href="#cb2-156"></a></span>
<span id="cb2-157"><a href="#cb2-157"></a><span class="ss">2. </span>Place a small text or CSV file inside <span class="in">`data`</span>. From the R Console, run <span class="in">`getwd()`</span> and confirm that it points to the project root.</span>
<span id="cb2-158"><a href="#cb2-158"></a></span>
<span id="cb2-159"><a href="#cb2-159"></a><span class="ss">3. </span>From a script inside <span class="in">`scripts`</span>, write the relative path needed to read the file in <span class="in">`data`</span>. Then write the absolute path to the same file. Which one would still work on another computer?</span>
<span id="cb2-160"><a href="#cb2-160"></a></span>
<span id="cb2-161"><a href="#cb2-161"></a><span class="ss">4. </span>Deliberately break a path by misspelling a directory name. Observe the error message produced by the software. Correct the path and rerun the command.</span>
<span id="cb2-162"><a href="#cb2-162"></a></span>
<span id="cb2-163"><a href="#cb2-163"></a><span class="ss">5. </span>(Deliberately bad practice) Use setwd() to point R directly at the data directory. Now try running the script again from a different location. Explain why this approach fails as a general strategy.</span>
<span id="cb2-164"><a href="#cb2-164"></a></span>
<span id="cb2-165"><a href="#cb2-165"></a>These exercises should help you to develop a form of spatial reasoning that underlies all your analyses 9and working with your computer in general --- good housekeeping stems from this! If you cannot reliably locate files you also cannot reason clearly about data provenance, reproducibility, or control over your analysis.</span>
<span id="cb2-166"><a href="#cb2-166"></a></span>
<span id="cb2-167"><a href="#cb2-167"></a>Note: graphical file browsers hide structure behind icons and clicks. Code exposes structure explicitly. Learning to think in paths rather than windows is a necessary transition from consumer software use to scientific computing.</span>
<span id="cb2-168"><a href="#cb2-168"></a></span>
<span id="cb2-169"><a href="#cb2-169"></a>Now that we know how and where to locate your data, let's take a look at how those data are represented.</span>
<span id="cb2-170"><a href="#cb2-170"></a></span>
<span id="cb2-171"><a href="#cb2-171"></a><span class="fu"># Types of Data Files</span></span>
<span id="cb2-172"><a href="#cb2-172"></a></span>
<span id="cb2-173"><a href="#cb2-173"></a>The choice of data file format determines how information is stored and which assumptions software will make when reading it, and which errors are likely to remain invisible.</span>
<span id="cb2-174"><a href="#cb2-174"></a></span>
<span id="cb2-175"><a href="#cb2-175"></a>R will read in many types of data, including spreadsheets, text files, binary files, and files from other statistical packages.</span>
<span id="cb2-176"><a href="#cb2-176"></a></span>
<span id="cb2-177"><a href="#cb2-177"></a>It is useful to treat questions about which file type to use as answering recurring practical questions. At different stages of an analysis, we may need data to be easily readable by humans, reliably exchanged across software and operating systems, efficient to store or process at scale, or well annotated with metadata. No single format optimises all of these requirements simultaneously. The non-exhaustive list of file formats mentioned below offer different compromises among these constraints, and their suitability depends on which problem is being prioritised.</span>
<span id="cb2-178"><a href="#cb2-178"></a></span>
<span id="cb2-179"><a href="#cb2-179"></a><span class="fu">## Delimited Text Files (CSV and TSV)</span></span>
<span id="cb2-180"><a href="#cb2-180"></a></span>
<span id="cb2-181"><a href="#cb2-181"></a>CSV and TSV files optimise human readability.</span>
<span id="cb2-182"><a href="#cb2-182"></a></span>
<span id="cb2-183"><a href="#cb2-183"></a>Delimited text files represent the simplest and most transparent way of storing tabular data. Both comma-separated value (CSV) files and tab-separated value (TSV) files encode tables as plain text, with rows separated by line breaks and columns separated by a designated delimiter. The distinction between them arises from how reliably the chosen delimiter can be distinguished from the data themselves.</span>
<span id="cb2-184"><a href="#cb2-184"></a></span>
<span id="cb2-185"><a href="#cb2-185"></a>CSV files use commas to separate fields (columns), which makes them widely supported and easy to exchange across software and platforms. However, this choice becomes ambiguous when data values themselves contain commas, such as free text, lists, or numbers formatted with thousands separators. TSV files address this ambiguity by using tab characters as delimiters, which are far less likely to occur naturally within data values. In this sense, TSV files trade a small loss in visual familiarity for greater robustness during parsing.</span>
<span id="cb2-186"><a href="#cb2-186"></a></span>
<span id="cb2-187"><a href="#cb2-187"></a>As such, CSV files are a simple and widely used format amongst biologists and ecologists, but they can become impractical for large datasets with complex structures or metadata.</span>
<span id="cb2-188"><a href="#cb2-188"></a></span>
<span id="cb2-189"><a href="#cb2-189"></a>We will most frequently use the functions <span class="in">`read.csv()`</span> and <span class="in">`readr::read_csv()`</span> (and related forms) for reading in CSV data. We can write CSV files to disk with the <span class="in">`write.csv()`</span> and <span class="in">`readr::write_csv()`</span> commands. For very large datasets that might take a long time to read in or save, <span class="in">`data.table::fread()`</span> and <span class="in">`data.table::fwrite()`</span> are faster alternatives to the aforementioned base R or **tidyverse** options. Even faster options are `feather::read_feather()` and `feather::write_feather()`; although **feather** saves tabular data, the format is not actually an ASCII CSV, however.</span>
<span id="cb2-190"><a href="#cb2-190"></a></span>
<span id="cb2-191"><a href="#cb2-191"></a>The same functions that read or write CSV files in R can be used for TSV, but one has to set the arguments <span class="in">`sep = "\t"`</span> or <span class="in">`delim = "\t"`</span> for the functions <span class="in">`read.csv()`</span> and <span class="in">`read_csv()`</span> respectively.</span>
<span id="cb2-192"><a href="#cb2-192"></a></span>
<span id="cb2-193"><a href="#cb2-193"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb2-194"><a href="#cb2-194"></a><span class="fu">## ASCII Files</span></span>
<span id="cb2-195"><a href="#cb2-195"></a>ASCII stands for "American Standard Code for Information Interchange". An ASCII file is a plain text file that contains ASCII characters. ASCII is a character encoding standard that assigns a unique numeric code to each character, including letters, numbers, punctuation, and other symbols commonly used in the English language.</span>
<span id="cb2-196"><a href="#cb2-196"></a></span>
<span id="cb2-197"><a href="#cb2-197"></a>ASCII files are the most basic type of text file and are supported by virtually all operating systems and applications. We can create and edit ASCII files using any text editor, such as Notepad, TextEdit, or <span class="co">[</span><span class="ot">VS Code</span><span class="co">](https://code.visualstudio.com/)</span>. ASCII files are typically used for storing, sharing simple text-based information such as program source code, configuration files, and other types of data that do not require special formatting or rich media content.</span>
<span id="cb2-198"><a href="#cb2-198"></a></span>
<span id="cb2-199"><a href="#cb2-199"></a>ASCII files are limited in their ability to represent non-English characters or symbols that are not included in the ASCII character set. To handle these types of characters, other character encoding standards such as UTF-8 and Unicode are used. However, ASCII files remain an important and widely used format for storing and sharing simple text-based data.</span>
<span id="cb2-200"><a href="#cb2-200"></a>:::</span>
<span id="cb2-201"><a href="#cb2-201"></a></span>
<span id="cb2-202"><a href="#cb2-202"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb2-203"><a href="#cb2-203"></a><span class="fu">## Missing Values and CSV and TSV Files</span></span>
<span id="cb2-204"><a href="#cb2-204"></a>Where we have missing data (blanks), the CSV format separates these by commas with empty field in-between. However, there can be problems with blanks if we read in a space-delimited format file. If we are having trouble reading in missing data as blanks, try replacing them in the spreadsheet with <span class="in">`NA`</span>, the missing data code in R. In Excel, highlight the area of the spreadsheet that includes all the cells we need to fill with <span class="in">`NA`</span>. Do an 'Edit/Replace...' and leave the 'Find what:' text box blank and in the 'Replace with:' text box enter <span class="in">`NA`</span>. Once imported into R, the <span class="in">`NA`</span> values will be recognised as missing data.</span>
<span id="cb2-205"><a href="#cb2-205"></a>:::</span>
<span id="cb2-206"><a href="#cb2-206"></a></span>
<span id="cb2-207"><a href="#cb2-207"></a><span class="fu">## Microsoft Excel Files</span></span>
<span id="cb2-208"><a href="#cb2-208"></a></span>
<span id="cb2-209"><a href="#cb2-209"></a>Excel files optimise human convenience, engagement, and presentation.</span>
<span id="cb2-210"><a href="#cb2-210"></a></span>
<span id="cb2-211"><a href="#cb2-211"></a>However, Excel files also emphasise that software convenience can obscure underlying data transformations, which moves the analytical risk from computation to problems that might stem from user interaction.</span>
<span id="cb2-212"><a href="#cb2-212"></a></span>
<span id="cb2-213"><a href="#cb2-213"></a>Microsoft Excel files are a type of file format that is used to store data in a tabular form, much like CSV files. However, Excel files are proprietary and are specifically designed to work with Excel software. Excel files can contain more advanced formatting features such as colours, fonts, and formulas, which make them a popular choice for people who like embellishments. </span>
<span id="cb2-214"><a href="#cb2-214"></a></span>
<span id="cb2-215"><a href="#cb2-215"></a>These software behaviours impose constraints that are poorly aligned with statistical analysis, and it is from these constraints that the following limitations arise:</span>
<span id="cb2-216"><a href="#cb2-216"></a></span>
<span id="cb2-217"><a href="#cb2-217"></a><span class="ss">-   </span>**Compatibility** Excel files may not be compatible with all data science tools and programming languages. For example, R cannot read Excel files directly.</span>
<span id="cb2-218"><a href="#cb2-218"></a></span>
<span id="cb2-219"><a href="#cb2-219"></a><span class="ss">-   </span>**Data integrity** Excel files can be prone to errors and inconsistencies in the data. For example, if a user changes a formula or formatting, it could affect the entire dataset. Also, it is possible for Excel to change the data types of certain columns, or to mix the class of data *within a column*, which can cause issues with data processing and analysis.</span>
<span id="cb2-220"><a href="#cb2-220"></a></span>
<span id="cb2-221"><a href="#cb2-221"></a><span class="ss">-   </span>**File size** Excel files can quickly become very large when dealing with large datasets, which can lead to performance issues, storage problems.</span>
<span id="cb2-222"><a href="#cb2-222"></a></span>
<span id="cb2-223"><a href="#cb2-223"></a><span class="ss">-   </span>**Version control** Excel files can make it difficult to keep track of changes and versions of the data, particularly when multiple people are working on the same file.</span>
<span id="cb2-224"><a href="#cb2-224"></a></span>
<span id="cb2-225"><a href="#cb2-225"></a>In contrast, CSV files are a simple, lightweight, and widely supported file format that can be easily used with most data science tools and programming languages. CSV files are also less prone to errors and inconsistencies than Excel files, making them a more reliable choice for data science tasks.</span>
<span id="cb2-226"><a href="#cb2-226"></a></span>
<span id="cb2-227"><a href="#cb2-227"></a>For these reasons, Excel is best used as a tool for data entry rather than data analysis. Exporting deliberately to plain-text formats such as CSV fixes the data representation, reveals assumptions to inspection, and sets a clear boundary between data generation and analysis. Naming exported files explicitly (often with dates or version identifiers) further tightens provenance and reduces ambiguity about which data underlie a given result.</span>
<span id="cb2-228"><a href="#cb2-228"></a></span>
<span id="cb2-229"><a href="#cb2-229"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb2-230"><a href="#cb2-230"></a><span class="fu">## Well-known Excel Errors</span></span>
<span id="cb2-231"><a href="#cb2-231"></a>Excel is a widely used spreadsheet application, but it has been responsible for several serious errors in data analysis, science, and data science. Some of these errors include:</span>
<span id="cb2-232"><a href="#cb2-232"></a></span>
<span id="cb2-233"><a href="#cb2-233"></a><span class="ss">-   </span>**Gene name errors (2016):** Accurate. Ziemann, Eren, and El-Osta published their findings in *Genome Biology*, demonstrating that Excel's automatic conversion transformed gene symbols like SEPT2 and MARCH1 into dates. Their survey examined supplementary files from 3,597 published papers across 18 journals between 2005 and 2015, finding that roughly one-fifth contained such errors. This remains one of the most damaging documented cases of Excel undermining scientific reproducibility, and it directly affects biology students working with genomic data.</span>
<span id="cb2-234"><a href="#cb2-234"></a></span>
<span id="cb2-235"><a href="#cb2-235"></a><span class="ss">- </span>Another compelling case involves **automatic data type conversion destroying scientific measurements**. Excel converts identifiers that look like scientific notation (*e.g.*, "1E4" becomes 10,000) or treats leading zeros (*e.g.*, "0001") in sample IDs as insignificant (*i.e.*, Excel displays "1"). Ecologists working with plot identifiers or anyone using structured identifiers encounters this pervasive, insidious problem.</span>
<span id="cb2-236"><a href="#cb2-236"></a></span>
<span id="cb2-237"><a href="#cb2-237"></a><span class="ss">-   </span>**Truncation of large numbers** Excel can handle only a limited number of digits for large numbers, truncating any value that exceeds this limit. This truncation has led to a loss of precision and inaccurate calculations in scientific and data analysis contexts where exact values were important.</span>
<span id="cb2-238"><a href="#cb2-238"></a></span>
<span id="cb2-239"><a href="#cb2-239"></a><span class="ss">-   </span>**Issues with floating-point arithmetic** Excel uses IEEE 754 double-precision floating-point representation, which produces rounding errors. The classic demonstration involves calculations like <span class="in">`(0.1 + 0.2) ≠ 0.3`</span> in binary representation. For iterative scientific calculations or cumulative errors in modelling, this matters.</span>
<span id="cb2-240"><a href="#cb2-240"></a></span>
<span id="cb2-241"><a href="#cb2-241"></a><span class="ss">- </span>The **UK COVID-19 testing data loss (October 2020)** deserves attention. Public Health England used the legacy <span class="in">`.xls`</span> format (limited to 65,536 rows) rather than the newer <span class="in">`.xlsx`</span> (limited to 1,048,576 rows), resulting in 15,841 positive test results being lost when the file exceeded row limits. This directly affected contact tracing during a pandemic.</span>
<span id="cb2-242"><a href="#cb2-242"></a></span>
<span id="cb2-243"><a href="#cb2-243"></a><span class="ss">- </span>The **date format ambiguity problem** creates reproducibility nightmares across international collaborations. Excel interprets "01/02/2020" differently depending on regional settings (January 2nd vs. February 1st), and automatically converting text strings to dates corrupts datasets when shared between researchers using different locale settings. This affects any collaborative science.</span>
<span id="cb2-244"><a href="#cb2-244"></a>:::</span>
<span id="cb2-245"><a href="#cb2-245"></a></span>
<span id="cb2-246"><a href="#cb2-246"></a><span class="fu">## Rdata Files</span></span>
<span id="cb2-247"><a href="#cb2-247"></a>Specialist files such as Rdata, binary files (@sec-binary), and NetCDF (@sec-netcdf) files were developed for optimasing scale (data size), metadata (data about data), and coupling (issues around speed and performance).</span>
<span id="cb2-248"><a href="#cb2-248"></a></span>
<span id="cb2-249"><a href="#cb2-249"></a>Rdata files are a file format used by the R programming language to store data objects. These files can contain any type of R object, such as vectors, matrices, dataframes, lists, and more. Rdata files are binary files, which means they are not human-readable like text files such as CSV files. Binary R data files have a <span class="in">`.rda`</span> or <span class="in">`.Rdata`</span> file extension and can be created or read using the <span class="in">`save()`</span> and <span class="in">`load()`</span>, respectively, functions in R. </span>
<span id="cb2-250"><a href="#cb2-250"></a></span>
<span id="cb2-251"><a href="#cb2-251"></a>Rdata files are convenient for a number of reasons:</span>
<span id="cb2-252"><a href="#cb2-252"></a></span>
<span id="cb2-253"><a href="#cb2-253"></a><span class="ss">-   </span>**Efficient storage** Rdata files can be more compact (they can be compressed) and efficient than other file formats, such as CSV files, because they are stored in a binary format. This means they take up less disk space and can be read and written faster.</span>
<span id="cb2-254"><a href="#cb2-254"></a></span>
<span id="cb2-255"><a href="#cb2-255"></a><span class="ss">-   </span>**Easy access to R objects** Rdata files make it easy to save and load R objects, which can be useful for preserving data objects for future analysis, sharing them with others. This is especially useful for complex datasets or objects that would be difficult to recreate.</span>
<span id="cb2-256"><a href="#cb2-256"></a></span>
<span id="cb2-257"><a href="#cb2-257"></a><span class="ss">-   </span>**Preserve metadata** Rdata files can preserve metadata such as variable names, row names, column names, and other attributes of R objects. This makes it easier to work with the data objects in the future without having to recreate this metadata.</span>
<span id="cb2-258"><a href="#cb2-258"></a></span>
<span id="cb2-259"><a href="#cb2-259"></a><span class="ss">-   </span>**Convenient for reproducibility** Rdata files can be used to save and load data objects as part of a reproducible research workflow. This can help ensure that data objects are preserved and can be easily accessed in the future, even if the data sources or code have changed.</span>
<span id="cb2-260"><a href="#cb2-260"></a></span>
<span id="cb2-261"><a href="#cb2-261"></a>On the downside, they can only be used within R, making them a less than ideal proposition when you intend to share your data with colleagues who sadly do not use R.</span>
<span id="cb2-262"><a href="#cb2-262"></a></span>
<span id="cb2-263"><a href="#cb2-263"></a><span class="fu">## Other Binary Files {#sec-binary}</span></span>
<span id="cb2-264"><a href="#cb2-264"></a></span>
<span id="cb2-265"><a href="#cb2-265"></a>As a biostatistician, you may encounter several other binary data files in your work. Such binary data files may be software-specific and can be used to store large datasets or data objects that are not easily represented in a text format. For example, a binary data file might contain a large matrix or array of numeric data that would be difficult to store in a text file. Binary data files can also be used to store images, audio files, and other types of data that are not represented as text.</span>
<span id="cb2-266"><a href="#cb2-266"></a></span>
<span id="cb2-267"><a href="#cb2-267"></a>One common type of binary data file that you may encounter as a statistician is a SAS data file. SAS is a statistical software package that is widely used in data analysis, and SAS data files are a binary format used to store datasets in SAS. These files typically have a <span class="in">`.sas7bdat`</span> file extension and contain metadata such as variable names and formats in addition to the data itself. Another type of binary data file you may encounter is a binary <span class="in">`.mat`</span> data file, which is a file format used to store Matlab data. </span>
<span id="cb2-268"><a href="#cb2-268"></a></span>
<span id="cb2-269"><a href="#cb2-269"></a>When working with binary data files, it is important to be aware of the specific format of the file, the tools and software needed to read and manipulate the data. Some statistical software packages may have built-in functions for reading and writing certain types of binary data files, while others may require additional libraries or packages.</span>
<span id="cb2-270"><a href="#cb2-270"></a></span>
<span id="cb2-271"><a href="#cb2-271"></a><span class="fu">## NetCDF, Grib, and HDF Files {#sec-netcdf}</span></span>
<span id="cb2-272"><a href="#cb2-272"></a></span>
<span id="cb2-273"><a href="#cb2-273"></a>NetCDF, HDF, and GRIB are file formats commonly used in the scientific and research communities to store and share large and complex datasets. These datasets are optimised for storing gridded (regular intervals in space and time) data that are interoperable amongst computing systems. Here is a brief overview of each file format:</span>
<span id="cb2-274"><a href="#cb2-274"></a></span>
<span id="cb2-275"><a href="#cb2-275"></a><span class="ss">-   </span>**NetCDF (Network Common Data Form)** is a binary file format that is designed for storing and sharing scientific data. It can store multidimensional arrays and metadata, such as variable names and units, in a self-describing format. NetCDF files are commonly used in fields such as atmospheric science, oceanography, and climate modelling.</span>
<span id="cb2-276"><a href="#cb2-276"></a></span>
<span id="cb2-277"><a href="#cb2-277"></a><span class="ss">-   </span>Like NetCDF files, **HDF (Hierarchical Data Format)** is a file format that is designed to store and organise large and complex data structures. It can store a wide variety of data types, including multidimensional arrays, tables, and hierarchical data. HDF files are commonly used in fields such as remote sensing, astronomy, and engineering.</span>
<span id="cb2-278"><a href="#cb2-278"></a></span>
<span id="cb2-279"><a href="#cb2-279"></a><span class="ss">-   </span>Again, **GRIB (GRIdded Binary)** files are similar to NetCDF files. They are a binary file format used to store meteorological and oceanographic data. It can store gridded data, such as atmospheric and oceanic model output, in a compact and efficient binary format. GRIB files are commonly used by weather forecasting agencies and research organisations.</span>
<span id="cb2-280"><a href="#cb2-280"></a></span>
<span id="cb2-281"><a href="#cb2-281"></a>Compared to CSV files, these file formats offer several benefits for storing, sharing complex datasets:</span>
<span id="cb2-282"><a href="#cb2-282"></a></span>
<span id="cb2-283"><a href="#cb2-283"></a><span class="ss">-   </span>**Support for multidimensional arrays** These file formats can store and handle multidimensional arrays, which cannot be represented in a CSV file. However, they can be *exported* as CSV files, often after subsetting the data, but the resultant CSV files consume significant amounts of disk space.</span>
<span id="cb2-284"><a href="#cb2-284"></a></span>
<span id="cb2-285"><a href="#cb2-285"></a><span class="ss">-   </span>**Efficient storage** Binary file formats can be more compact and efficient than text-based formats such as CSV files, which can save disk space and make it easier to share and transfer large datasets.</span>
<span id="cb2-286"><a href="#cb2-286"></a></span>
<span id="cb2-287"><a href="#cb2-287"></a><span class="ss">-   </span>**Memory use efficiency** NetCDF, GRIB, and HDF files are better for memory use efficiency compared to CSV files because they can store multidimensional arrays and metadata in a compact binary format, which can save disk space and memory when working with large and complex datasets. Also, they do not have to be read into memory all at once.</span>
<span id="cb2-288"><a href="#cb2-288"></a></span>
<span id="cb2-289"><a href="#cb2-289"></a><span class="ss">-   </span>**Self-describing metadata** These file formats can include metadata, such as variable names and units, which are self-describing and can be easily accessed and understood by other researchers and software.</span>
<span id="cb2-290"><a href="#cb2-290"></a></span>
<span id="cb2-291"><a href="#cb2-291"></a><span class="ss">-   </span>**Support for compression** Binary file formats can support compression, which can further reduce file size and make it easier to share and transfer large datasets.</span>
<span id="cb2-292"><a href="#cb2-292"></a></span>
<span id="cb2-293"><a href="#cb2-293"></a>The various efficiencies mentioned above may be offset by them being quite challenging to work with, and as such novices might experience steep learning curves.</span>
<span id="cb2-294"><a href="#cb2-294"></a></span>
<span id="cb2-295"><a href="#cb2-295"></a><span class="fu">## Larger than Memory Data</span></span>
<span id="cb2-296"><a href="#cb2-296"></a></span>
<span id="cb2-297"><a href="#cb2-297"></a>Above we dealt with data that fit into your computer's memory (RAM). However, there are many datasets that are too large to fit into memory, and as such, we need to use alternative methods to work with them. These methods include:</span>
<span id="cb2-298"><a href="#cb2-298"></a></span>
<span id="cb2-299"><a href="#cb2-299"></a><span class="ss">* </span>Apache Arrow in the <span class="in">`arrow`</span> package in R, which has support for the 'feather' file format, 'parquet' files</span>
<span id="cb2-300"><a href="#cb2-300"></a><span class="ss">* </span>DuckDB in the <span class="in">`duckdb`</span> package in R, which create a database on disk, can be queried using SQL</span>
<span id="cb2-301"><a href="#cb2-301"></a></span>
<span id="cb2-302"><a href="#cb2-302"></a>I will develop vignettes for these in the future. We will not use these in this course, but it is important to be aware of them.</span>
<span id="cb2-303"><a href="#cb2-303"></a></span>
<span id="cb2-304"><a href="#cb2-304"></a><span class="fu"># File Extensions</span></span>
<span id="cb2-305"><a href="#cb2-305"></a></span>
<span id="cb2-306"><a href="#cb2-306"></a>This section is about teaching you the value of knowing what to expect of a file, and how the data are represented inside it, by simply inspecting how it is named rather than relying on what software choose to display by default. File extensions are one of the simplest points at which this representation can be made either explicit or opaque, and they provide a real example of how small interface choices might affect analytical error.</span>
<span id="cb2-307"><a href="#cb2-307"></a></span>
<span id="cb2-308"><a href="#cb2-308"></a>File extensions are three or four character suffixes added to the end of a filename, usually preceded by a period (e.g., <span class="in">`.txt`</span>, <span class="in">`.jpg`</span>, <span class="in">`.pdf`</span>, <span class="in">`.csv`</span>, <span class="in">`.xlsx`</span>). These extensions indicate the format or type of the file, providing information about the content and structure of the data within the file. Some file extensions are particular to Windows (such as <span class="in">`.exe`</span>) and others to MacOS (*e.g.*, <span class="in">`.dmg`</span>), but many are transportable across operating systems. File extensions help both the operating system and applications recognise the file type and determine which program or application should be used to open, view, or edit the file.</span>
<span id="cb2-309"><a href="#cb2-309"></a></span>
<span id="cb2-310"><a href="#cb2-310"></a>The default setting for Windows computers is for file extensions to be hidden when files are viewed in the Windows Explorer. This is silly and irresponsible, and a frequent major source of frustration, annoyance, and irritability when I try to help a student with an issue on their computers. Displaying file extensions on your computer is essential for effective data science work because it allows us to quickly identify, manage various file types that we may encounter during our projects.</span>
<span id="cb2-311"><a href="#cb2-311"></a></span>
<span id="cb2-312"><a href="#cb2-312"></a>Some reasons why it is important to display file extensions if we are to effectively use your computer in scientific computing applications:</span>
<span id="cb2-313"><a href="#cb2-313"></a></span>
<span id="cb2-314"><a href="#cb2-314"></a><span class="ss">-   </span>**Differentiate file formats** When analysing data, you will often work with multiple file formats, such as CSV, TSV, Excel, JSON, and others. By displaying the file extensions you can easily differentiate between these formats so that you may use the most appropriate tools and methods to process and analyse your data.</span>
<span id="cb2-315"><a href="#cb2-315"></a></span>
<span id="cb2-316"><a href="#cb2-316"></a><span class="ss">-   </span>**Prevent errors** Knowing the exact file type helps you avoid errors when importing or exporting data, as different formats require specific handling techniques. For instance, using a CSV file reader on a TSV file could lead to unexpected outcomes.</span>
<span id="cb2-317"><a href="#cb2-317"></a></span>
<span id="cb2-318"><a href="#cb2-318"></a><span class="ss">-   </span>**Improve file organisation** When working on complex projects, it is important that you maintain an organised file structure. Being able to see file extensions at a glance helps you keep track of various data files, scripts, and output files, and it will be easier for you to manage your project, ensure its reproducibility.</span>
<span id="cb2-319"><a href="#cb2-319"></a></span>
<span id="cb2-320"><a href="#cb2-320"></a><span class="ss">-   </span>**Enhance security** Displaying file extensions can also help you identify potentially malicious files disguised as legitimate ones. For example, a file with an extension '.txt.exe' might appear to be a harmless text file if the extension is hidden, but it is actually an executable file that could be harmful.</span>
<span id="cb2-321"><a href="#cb2-321"></a></span>
<span id="cb2-322"><a href="#cb2-322"></a><span class="ss">-   </span>**Facilitate collaboration** When sharing files with colleagues or collaborators, knowing the file format is essential for smooth communication, seamless collaboration. Displaying file extensions ensures that everyone is aware of the file types being used and can handle them accordingly.</span>
<span id="cb2-323"><a href="#cb2-323"></a></span>
<span id="cb2-324"><a href="#cb2-324"></a>To display file extensions on your computer, you will need to adjust the settings in your operating system's file explorer. The specific steps to do this vary depending on whether you are using Windows, MacOS, or a Linux-based system. A quick online search will provide you with the necessary instructions. Since most people use Windows, I have included the instructions for Windose user in the box, below, titled "Displaying File Extensions in Windows".</span>
<span id="cb2-325"><a href="#cb2-325"></a></span>
<span id="cb2-326"><a href="#cb2-326"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb2-327"><a href="#cb2-327"></a><span class="fu">## Displaying File Extension in Windows</span></span>
<span id="cb2-328"><a href="#cb2-328"></a>Be a responsible adult and display the file extensions on your computer. To display file extensions in Windows Explorer on the latest version of Windows (assuming Windows 11), follow these steps:</span>
<span id="cb2-329"><a href="#cb2-329"></a></span>
<span id="cb2-330"><a href="#cb2-330"></a><span class="ss">1. </span>Open a new File Explorer window by clicking on the File Explorer icon on the taskbar or pressing the 'Windows key + E' on your keyboard.</span>
<span id="cb2-331"><a href="#cb2-331"></a><span class="ss">2. </span>In the File Explorer window, click on the three horizontal dots in the upper-right corner to open the 'More options' menu.</span>
<span id="cb2-332"><a href="#cb2-332"></a><span class="ss">3. </span>From the 'More options' menu, click on 'Options' to open the 'Folder Options' window.</span>
<span id="cb2-333"><a href="#cb2-333"></a><span class="ss">4. </span>In the 'Folder Options' window, switch to the 'View' tab.</span>
<span id="cb2-334"><a href="#cb2-334"></a><span class="ss">5. </span>Under the 'Advanced settings' list, locate the option 'Hide extensions for known file types', uncheck the box next to it.</span>
<span id="cb2-335"><a href="#cb2-335"></a><span class="ss">6. </span>Click 'Apply' to save your changes, and then click 'OK' to close the 'Folder Options' window.</span>
<span id="cb2-336"><a href="#cb2-336"></a></span>
<span id="cb2-337"><a href="#cb2-337"></a>Now, the file extensions should be visible for all files in Windows Explorer. Remember that this setting applies to all folders on your computer. If you want to revert to hiding the file extensions, simply follow the same steps, check the box next to 'Hide extensions for known file types' in the 'View' tab of the 'Folder Options' window.</span>
<span id="cb2-338"><a href="#cb2-338"></a>:::</span>
<span id="cb2-339"><a href="#cb2-339"></a></span>
<span id="cb2-340"><a href="#cb2-340"></a><span class="fu"># R Code Conventions</span></span>
<span id="cb2-341"><a href="#cb2-341"></a></span>
<span id="cb2-342"><a href="#cb2-342"></a>When writing code in support of a statistical data analysis in R, the primary aim is of course to produce the correct output. Equally important, though, is to produce work that remains intelligible, inspectable, and usable long after it has been written. The conventions below should therefore serve as practical expressions of three main principles: *consistency*, *legibility* to others, and future *recoverability*. Good code is disciplined writing, and the practices below follow from that discipline.</span>
<span id="cb2-343"><a href="#cb2-343"></a></span>
<span id="cb2-344"><a href="#cb2-344"></a>**Consistency** supports readability across an analysis. A consistent coding style allows both humans and software to detect structure without re-interpretation. Adopting an established style guide (such as the [**tidyverse** style guide](https://style.tidyverse.org/), the <span class="co">[</span><span class="ot">R Style Guide</span><span class="co">](http://adv-r.had.co.nz/Style.html)</span>, or <span class="co">[</span><span class="ot">Google’s R Style Guide</span><span class="co">](https://google.github.io/styleguide/Rguide.html)</span>) provides a stable baseline for decisions about spacing, indentation, line breaks, and operator placement. The specific guide matters less than applying one style coherently throughout a project. Inconsistent formatting increases cognitive load and obscures logical structure, even when the code itself is correct. You want to ensure your code is a pleasure to read by others --- in this first instance, especially your instructors responsible for marking your reports.</span>
<span id="cb2-345"><a href="#cb2-345"></a></span>
<span id="cb2-346"><a href="#cb2-346"></a>Practices such as regular indentation, consistent use of spaces around operators and commas, and sensible line breaking all accomplish this goal. Long lines should be split in ways that preserve syntactic clarity, for example by leaving brackets open or ending a line with an operator (<span class="in">`+`</span> in **ggplot2** and `|&gt;` elsewhere in the **tidyverse**) so that continuation is unambiguous to R and to the reader.</span>
<span id="cb2-347"><a href="#cb2-347"></a></span>
<span id="cb2-348"><a href="#cb2-348"></a>**Legibility to others** concerns whether the intent of your analysis is visible without reconstruction, and without ambiguity. Code is read far more often than it is written, and this includes being read by your future self. Meaningful variable and function names make data transformations interpretable at a glance, and it reduces the need for explanatory comments (although these are still necessary). Where code is complex, comments should clarify *why* something is done, not merely restate *what* the code does.</span>
<span id="cb2-349"><a href="#cb2-349"></a></span>
<span id="cb2-350"><a href="#cb2-350"></a>Modularising code into small, well-named functions further supports legibility by isolating conceptual units of work. This makes analyses easier to follow, easier to debug (problems arise often!), and easier to extend. Code that can only be understood as a continuous script quickly becomes opaque once it grows beyond trivial size.</span>
<span id="cb2-351"><a href="#cb2-351"></a></span>
<span id="cb2-352"><a href="#cb2-352"></a>**Future recoverability** addresses whether an analysis can be resumed, audited, or extended once its original context has faded. Version control systems such as Git, used via platforms like GitHub or GitLab, provide an explicit record of how code evolves and allow earlier states to be recovered without ambiguity. Organising projects into clear directory structures (for data, scripts, and outputs) serves the same purpose at the file-system level.</span>
<span id="cb2-353"><a href="#cb2-353"></a></span>
<span id="cb2-354"><a href="#cb2-354"></a>Recoverability also depends on making dependencies explicit. All required packages should be declared and loaded deliberately upfront so that analyses do not rely on accidental features of a particular session. Relative file paths should be used wherever possible to avoid embedding machine-specific assumptions. Where appropriate, tests can be written to verify that functions behave as expected, providing early warning when changes introduce unintended effects.</span>
<span id="cb2-355"><a href="#cb2-355"></a></span>
<span id="cb2-356"><a href="#cb2-356"></a>It is worth noting that conventions vary across programming languages and research communities. What matters is not universal agreement on style, but internal coherence. Once a set of conventions is chosen, they should be applied consistently throughout an analysis so that structure, intent, and provenance remain visible.</span>
<span id="cb2-357"><a href="#cb2-357"></a></span>
<span id="cb2-358"><a href="#cb2-358"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb2-359"><a href="#cb2-359"></a><span class="fu">### Naming Variables</span></span>
<span id="cb2-360"><a href="#cb2-360"></a></span>
<span id="cb2-361"><a href="#cb2-361"></a>R imposes strict rules on variable naming, and violating them leads to errors that are often frustrating to diagnose. Variable names should never contain spaces or special characters, and should avoid masking existing function names (such as <span class="in">`mean`</span>). As a general rule, spaces should be avoided not only in variable names, but also in file and directory names used in a project.</span>
<span id="cb2-362"><a href="#cb2-362"></a></span>
<span id="cb2-363"><a href="#cb2-363"></a>Several naming conventions are commonly encountered:</span>
<span id="cb2-364"><a href="#cb2-364"></a></span>
<span id="cb2-365"><a href="#cb2-365"></a>**Pascal case** capitalises the first letter of each word and is often used for classes or types (e.g. <span class="in">`SpeciesName`</span>, <span class="in">`PetalColour`</span>, <span class="in">`FunctionalType`</span>).</span>
<span id="cb2-366"><a href="#cb2-366"></a></span>
<span id="cb2-367"><a href="#cb2-367"></a>**Snake case** separates words with underscores and uses lowercase letters, and is widely used in scripting and file naming (e.g. <span class="in">`species_name`</span>, <span class="in">`petal_colour`</span>, <span class="in">`functional_type`</span>).</span>
<span id="cb2-368"><a href="#cb2-368"></a></span>
<span id="cb2-369"><a href="#cb2-369"></a>**Camel case** capitalises words except for the first and is often used for variables or methods in object-oriented contexts (e.g. <span class="in">`speciesName`</span>, <span class="in">`petalColour`</span>, <span class="in">`functionalType`</span>).</span>
<span id="cb2-370"><a href="#cb2-370"></a></span>
<span id="cb2-371"><a href="#cb2-371"></a>**Hungarian notation** prefixes variable names with type indicators (e.g. <span class="in">`iSpeciesID`</span>, <span class="in">`strPetalColour`</span>), but is rarely used in modern R practice.</span>
<span id="cb2-372"><a href="#cb2-372"></a></span>
<span id="cb2-373"><a href="#cb2-373"></a>No naming convention is inherently superior. What matters is choosing one that supports readability and applying it consistently throughout a project.</span>
<span id="cb2-374"><a href="#cb2-374"></a>:::</span>
<span id="cb2-375"><a href="#cb2-375"></a></span>
<span id="cb2-376"><a href="#cb2-376"></a></span>
<span id="cb2-377"><a href="#cb2-377"></a><span class="fu"># Reproducible Research {#sec-reproducibility}</span></span>
<span id="cb2-378"><a href="#cb2-378"></a></span>
<span id="cb2-379"><a href="#cb2-379"></a>Reproducible research refers to the capacity for an analysis to be re-run, inspected, and verified using the same data and code, giving the same results. In this course, reproducibility is treated as a practical property of workflows: it depends on how data are represented, how code is written, and how analytical steps are organised over time. In advanced situations, such reprodicibility can recreate an entire scientific publication, or a thesis, from its initial data, code, and textual building blocks (see <span class="co">[</span><span class="ot">2. R Markdown and Quarto</span><span class="co">](02-quarto.qmd)</span>). The aim here is to distinguish clearly between practices you are expected to implement now and tools you should recognise as part of the broader reproducibility landscape.</span>
<span id="cb2-380"><a href="#cb2-380"></a></span>
<span id="cb2-381"><a href="#cb2-381"></a><span class="fu">## Required practice in BCB744</span></span>
<span id="cb2-382"><a href="#cb2-382"></a></span>
<span id="cb2-383"><a href="#cb2-383"></a>At this stage, reproducibility requires a small number of essential habits. These are the practices you are expected to adopt consistently and that will be evaluated in all assessments.</span>
<span id="cb2-384"><a href="#cb2-384"></a></span>
<span id="cb2-385"><a href="#cb2-385"></a>Analyses should be conducted using scripted code rather than manual interaction. Scripts provide an explicit record of analytical decisions and allow results to be regenerated without reliance on memory or interface state. Code should be organised within a coherent project structure, with separate locations for raw data, processed data, scripts, and outputs, so that analytical flow can be reconstructed with ease.</span>
<span id="cb2-386"><a href="#cb2-386"></a></span>
<span id="cb2-387"><a href="#cb2-387"></a>Version control forms part of this basic practice. Using Git (typically via platforms such as GitHub or GitLab) provides a transparent history of how analyses evolve and allows earlier states to be recovered if errors are introduced. This is not about public dissemination, but about maintaining an explicit record of change.</span>
<span id="cb2-388"><a href="#cb2-388"></a></span>
<span id="cb2-389"><a href="#cb2-389"></a>Literate programming tools such as R Markdown and Quarto are also part of required practice. These tools integrate code, output, and narrative text within a single document, making analytical reasoning visible alongside results. When data or code change, documents are regenerated rather than edited by hand, reducing the risk of divergence between analysis and interpretation.</span>
<span id="cb2-390"><a href="#cb2-390"></a></span>
<span id="cb2-391"><a href="#cb2-391"></a>Finally, reproducibility depends on making dependencies explicit. Required packages should be declared deliberately, relative file paths should be used to avoid machine-specific assumptions, and data files should be treated as fixed inputs rather than mutable artefacts. Together, these practices establish a workflow that can be re-run by others—or by your future self—without guesswork.</span>
<span id="cb2-392"><a href="#cb2-392"></a></span>
<span id="cb2-393"><a href="#cb2-393"></a><span class="fu">## A look to the future</span></span>
<span id="cb2-394"><a href="#cb2-394"></a></span>
<span id="cb2-395"><a href="#cb2-395"></a>Beyond these core practices lies a wider R ecosystem of tools made to support reproducibility at larger scales or higher levels of complexity. You are not expected to use these tools in BCB744, but you should be aware of the problems they address and the contexts in which they become relevant.</span>
<span id="cb2-396"><a href="#cb2-396"></a></span>
<span id="cb2-397"><a href="#cb2-397"></a>Workflow management tools such as **workflowr** and **targets** formalise analytical pipelines by making dependencies between steps explicit and automating re-execution when inputs change. These tools are useful for large or long-running projects, but they introduce additional complexities that is unnecessary at this stage.</span>
<span id="cb2-398"><a href="#cb2-398"></a></span>
<span id="cb2-399"><a href="#cb2-399"></a>Containerisation technologies such as Docker provide isolated computational environments with fixed software versions and dependencies. They are widely used in collaborative or production settings where analyses must run identically across different machines. They are powerful, but sit beyond the scope of this course and should be seen as an extension of the principles already discussed, not as a prerequisite.</span>
<span id="cb2-400"><a href="#cb2-400"></a></span>
<span id="cb2-401"><a href="#cb2-401"></a>Testing frameworks such as **testthat** enable systematic verification of code behaviour, which becomes increasingly important as projects grow in size or are reused across contexts. At present, the emphasis is on writing code that is clear enough to inspect, rather than on formal test suites.</span>
<span id="cb2-402"><a href="#cb2-402"></a></span>
<span id="cb2-403"><a href="#cb2-403"></a>The purpose of introducing these tools here is not to encourage their immediate adoption, but to situate current practices within a broader trajectory. As projects scale, the same concerns (visibility, control, and recoverability) are addressed using increasingly formal mechanisms.</span>
<span id="cb2-404"><a href="#cb2-404"></a></span>
<span id="cb2-405"><a href="#cb2-405"></a></span>
<span id="cb2-406"><a href="#cb2-406"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb2-407"><a href="#cb2-407"></a><span class="fu">## The Mars Climate Orbiter Mission in 1998</span></span>
<span id="cb2-408"><a href="#cb2-408"></a></span>
<span id="cb2-409"><a href="#cb2-409"></a>One of the most famous examples of unclear communication is the Mars Climate Orbiter mission in 1998. NASA lost the spacecraft due to a navigation error caused by a unit conversion mistake. The error occurred because one team used metric units (newtons), while another team used imperial units (pound-force) for a crucial spacecraft operation parameter. The discrepancy in units led to incorrect trajectory calculations, causing the Mars Climate Orbiter to approach Mars at a dangerously low altitude, ultimately disintegrate in the Martian atmosphere.</span>
<span id="cb2-410"><a href="#cb2-410"></a></span>
<span id="cb2-411"><a href="#cb2-411"></a>This incident highlights the importance of clear communication, proper data handling, and rigorous verification processes in mission-critical systems, including space missions. The event emphasises the need for using appropriate tools, software, and methodologies to minimise the risk of errors in complex engineering projects.</span>
<span id="cb2-412"><a href="#cb2-412"></a>:::</span>
<span id="cb2-413"><a href="#cb2-413"></a></span>
<span id="cb2-414"><a href="#cb2-414"></a><span class="fu"># Exercise: From Excel to Delimited Files to R</span></span>
<span id="cb2-415"><a href="#cb2-415"></a></span>
<span id="cb2-416"><a href="#cb2-416"></a>**Goal:** You will (i) create a small dataset in Excel, (ii) export it to three delimited text formats (CSV, TSV, and semicolon-delimited), (iii) save them into a known location inside an R Project folder using disciplined file naming, and (iv) read *one* of the exported files from a new R script in RStudio.</span>
<span id="cb2-417"><a href="#cb2-417"></a></span>
<span id="cb2-418"><a href="#cb2-418"></a><span class="fu">## Part A — Create a Project folder with known locations</span></span>
<span id="cb2-419"><a href="#cb2-419"></a></span>
<span id="cb2-420"><a href="#cb2-420"></a><span class="ss">1. </span>In RStudio, create a new project: **File → New Project → New Directory → New Project**.</span>
<span id="cb2-421"><a href="#cb2-421"></a>   Name it <span class="in">`bcb744_file_formats_exercise`</span> and create it in a location you can find again (*e.g.*, Documents).</span>
<span id="cb2-422"><a href="#cb2-422"></a></span>
<span id="cb2-423"><a href="#cb2-423"></a><span class="ss">2. </span>In the Files pane in RStudio (bottom right), confirm you can see the project folder and that it contains an <span class="in">`.Rproj`</span> file.</span>
<span id="cb2-424"><a href="#cb2-424"></a></span>
<span id="cb2-425"><a href="#cb2-425"></a><span class="ss">3. </span>Inside the project folder, create these subfolders:</span>
<span id="cb2-426"><a href="#cb2-426"></a></span>
<span id="cb2-427"><a href="#cb2-427"></a><span class="ss">   * </span><span class="in">`data-raw/`</span></span>
<span id="cb2-428"><a href="#cb2-428"></a><span class="ss">   * </span><span class="in">`data/`</span></span>
<span id="cb2-429"><a href="#cb2-429"></a><span class="ss">   * </span><span class="in">`scripts/`</span></span>
<span id="cb2-430"><a href="#cb2-430"></a></span>
<span id="cb2-431"><a href="#cb2-431"></a>   (You can do this in the Files pane: **New Folder**.)</span>
<span id="cb2-432"><a href="#cb2-432"></a></span>
<span id="cb2-433"><a href="#cb2-433"></a><span class="ss">4. </span>Decide on a file naming pattern you will follow for exports in this exercise (later, not now). Use this pattern exactly:</span>
<span id="cb2-434"><a href="#cb2-434"></a></span>
<span id="cb2-435"><a href="#cb2-435"></a><span class="ss">   * </span><span class="in">`survey_demo_YYYYMMDD_v01.csv`</span></span>
<span id="cb2-436"><a href="#cb2-436"></a><span class="ss">   * </span><span class="in">`survey_demo_YYYYMMDD_v01.tsv`</span></span>
<span id="cb2-437"><a href="#cb2-437"></a><span class="ss">   * </span><span class="in">`survey_demo_YYYYMMDD_v01.scsv`</span></span>
<span id="cb2-438"><a href="#cb2-438"></a></span>
<span id="cb2-439"><a href="#cb2-439"></a>   Replace <span class="in">`YYYYMMDD`</span> with today’s date. Keep <span class="in">`v01`</span> as written.</span>
<span id="cb2-440"><a href="#cb2-440"></a></span>
<span id="cb2-441"><a href="#cb2-441"></a>   *Note:* <span class="in">`.scsv`</span> here is just a human-visible label meaning “semicolon CSV”; it is still a plain text file.</span>
<span id="cb2-442"><a href="#cb2-442"></a></span>
<span id="cb2-443"><a href="#cb2-443"></a><span class="fu">## Part B — Create a small dataset in Excel</span></span>
<span id="cb2-444"><a href="#cb2-444"></a></span>
<span id="cb2-445"><a href="#cb2-445"></a><span class="ss">5. </span>Open Excel and create a new blank workbook.</span>
<span id="cb2-446"><a href="#cb2-446"></a></span>
<span id="cb2-447"><a href="#cb2-447"></a><span class="ss">6. </span>In row 1, enter these column names (exact spelling):</span>
<span id="cb2-448"><a href="#cb2-448"></a></span>
<span id="cb2-449"><a href="#cb2-449"></a><span class="ss">   * </span><span class="in">`id`</span></span>
<span id="cb2-450"><a href="#cb2-450"></a><span class="ss">   * </span><span class="in">`species`</span></span>
<span id="cb2-451"><a href="#cb2-451"></a><span class="ss">   * </span><span class="in">`site`</span></span>
<span id="cb2-452"><a href="#cb2-452"></a><span class="ss">   * </span><span class="in">`temp_C`</span></span>
<span id="cb2-453"><a href="#cb2-453"></a><span class="ss">   * </span><span class="in">`notes`</span></span>
<span id="cb2-454"><a href="#cb2-454"></a></span>
<span id="cb2-455"><a href="#cb2-455"></a><span class="ss">7. </span>Enter at least **10 rows** of data. Include the following features on purpose:</span>
<span id="cb2-456"><a href="#cb2-456"></a></span>
<span id="cb2-457"><a href="#cb2-457"></a><span class="ss">   * </span>at least one missing value (leave one cell blank in <span class="in">`temp_C`</span>)</span>
<span id="cb2-458"><a href="#cb2-458"></a><span class="ss">   * </span>at least one <span class="in">`notes`</span> cell that contains a comma (*e.g.*, <span class="in">`juvenile, small`</span>)</span>
<span id="cb2-459"><a href="#cb2-459"></a><span class="ss">   * </span>at least one <span class="in">`notes`</span> cell that contains a semicolon (*e.g.*, <span class="in">`recheck; uncertain`</span>)</span>
<span id="cb2-460"><a href="#cb2-460"></a></span>
<span id="cb2-461"><a href="#cb2-461"></a><span class="ss">8. </span>Save the Excel workbook into your project folder under <span class="in">`data-raw/`</span> with this name:</span>
<span id="cb2-462"><a href="#cb2-462"></a></span>
<span id="cb2-463"><a href="#cb2-463"></a><span class="ss">   * </span><span class="in">`survey_demo_YYYYMMDD_source.xlsx`</span></span>
<span id="cb2-464"><a href="#cb2-464"></a></span>
<span id="cb2-465"><a href="#cb2-465"></a><span class="fu">## Part C — Export three delimited text files into the project</span></span>
<span id="cb2-466"><a href="#cb2-466"></a></span>
<span id="cb2-467"><a href="#cb2-467"></a>Your target location for *all exports* is:</span>
<span id="cb2-468"><a href="#cb2-468"></a></span>
<span id="cb2-469"><a href="#cb2-469"></a><span class="ss">* </span><span class="in">`bcb744_file_formats_exercise/data/`</span></span>
<span id="cb2-470"><a href="#cb2-470"></a></span>
<span id="cb2-471"><a href="#cb2-471"></a><span class="ss">9. </span>Export to **CSV (comma delimited)** using the naming scheme defined above:</span>
<span id="cb2-472"><a href="#cb2-472"></a></span>
<span id="cb2-473"><a href="#cb2-473"></a><span class="ss">   * </span>Use **File → Save As** (or **Save a Copy**) and choose **CSV (Comma delimited)**.</span>
<span id="cb2-474"><a href="#cb2-474"></a><span class="ss">   * </span>Save into the project’s <span class="in">`data/`</span> folder as:</span>
<span id="cb2-475"><a href="#cb2-475"></a>     <span class="in">`survey_demo_YYYYMMDD_v01.csv`</span></span>
<span id="cb2-476"><a href="#cb2-476"></a></span>
<span id="cb2-477"><a href="#cb2-477"></a><span class="ss">10. </span>Export to **TSV (tab delimited)**:</span>
<span id="cb2-478"><a href="#cb2-478"></a></span>
<span id="cb2-479"><a href="#cb2-479"></a><span class="ss">* </span>Use **Save As** and choose **Text (Tab delimited)** (or equivalent).</span>
<span id="cb2-480"><a href="#cb2-480"></a><span class="ss">* </span>Save into the project’s <span class="in">`data/`</span> folder as:</span>
<span id="cb2-481"><a href="#cb2-481"></a>  <span class="in">`survey_demo_YYYYMMDD_v01.tsv`</span></span>
<span id="cb2-482"><a href="#cb2-482"></a></span>
<span id="cb2-483"><a href="#cb2-483"></a><span class="ss">11. </span>Export to **semicolon-delimited** (“semi-colon derived derivative”):</span>
<span id="cb2-484"><a href="#cb2-484"></a></span>
<span id="cb2-485"><a href="#cb2-485"></a><span class="ss">* </span>If your Excel offers a direct **CSV (Semicolon delimited)** option, use it and save as:</span>
<span id="cb2-486"><a href="#cb2-486"></a>  <span class="in">`survey_demo_YYYYMMDD_v01.scsv`</span></span>
<span id="cb2-487"><a href="#cb2-487"></a><span class="ss">* </span>If it does *not* offer that option, do this instead:</span>
<span id="cb2-488"><a href="#cb2-488"></a></span>
<span id="cb2-489"><a href="#cb2-489"></a><span class="ss">  1. </span>Open the saved CSV file (<span class="in">`survey_demo_YYYYMMDD_v01.csv`</span>) in a plain text editor (Notepad / TextEdit / VS Code).</span>
<span id="cb2-490"><a href="#cb2-490"></a><span class="ss">  2. </span>Save a new copy into the same <span class="in">`data/`</span> folder named <span class="in">`survey_demo_YYYYMMDD_v01.scsv`</span>.</span>
<span id="cb2-491"><a href="#cb2-491"></a><span class="ss">  3. </span>In that <span class="in">`.scsv`</span> copy, replace field separators **commas that function as delimiters** with semicolons.</span>
<span id="cb2-492"><a href="#cb2-492"></a>     Do not alter commas inside quoted text fields (your <span class="in">`notes`</span> column will reveal why this matters).</span>
<span id="cb2-493"><a href="#cb2-493"></a></span>
<span id="cb2-494"><a href="#cb2-494"></a><span class="ss">12. </span>In your file explorer (Windows Explorer / Finder), navigate to the project’s <span class="in">`data/`</span> folder and confirm that these three files exist there:</span>
<span id="cb2-495"><a href="#cb2-495"></a></span>
<span id="cb2-496"><a href="#cb2-496"></a><span class="ss">* </span><span class="in">`survey_demo_YYYYMMDD_v01.csv`</span></span>
<span id="cb2-497"><a href="#cb2-497"></a><span class="ss">* </span><span class="in">`survey_demo_YYYYMMDD_v01.tsv`</span></span>
<span id="cb2-498"><a href="#cb2-498"></a><span class="ss">* </span><span class="in">`survey_demo_YYYYMMDD_v01.scsv`</span></span>
<span id="cb2-499"><a href="#cb2-499"></a></span>
<span id="cb2-500"><a href="#cb2-500"></a><span class="fu">## Part D — Read one exported file from a new R script in RStudio</span></span>
<span id="cb2-501"><a href="#cb2-501"></a></span>
<span id="cb2-502"><a href="#cb2-502"></a><span class="ss">13. </span>In RStudio, create a new script: **File → New File → R Script**.</span>
<span id="cb2-503"><a href="#cb2-503"></a></span>
<span id="cb2-504"><a href="#cb2-504"></a><span class="ss">14. </span>Save the script into <span class="in">`scripts/`</span> with this name:</span>
<span id="cb2-505"><a href="#cb2-505"></a></span>
<span id="cb2-506"><a href="#cb2-506"></a><span class="ss">* </span><span class="in">`01_read_delimited_files.R`</span></span>
<span id="cb2-507"><a href="#cb2-507"></a></span>
<span id="cb2-508"><a href="#cb2-508"></a><span class="ss">15. </span>Copy and run the following lines **one at a time**, top to bottom.</span>
<span id="cb2-509"><a href="#cb2-509"></a></span>
<span id="cb2-510"><a href="#cb2-510"></a><span class="in">```{r eval=FALSE}</span></span>
<span id="cb2-511"><a href="#cb2-511"></a><span class="co"># 1) Confirm you are working inside the project</span></span>
<span id="cb2-512"><a href="#cb2-512"></a><span class="fu">getwd</span>()</span>
<span id="cb2-513"><a href="#cb2-513"></a></span>
<span id="cb2-514"><a href="#cb2-514"></a><span class="co"># 2) List the exported files (you should see your three exports)</span></span>
<span id="cb2-515"><a href="#cb2-515"></a><span class="fu">list.files</span>(<span class="st">"data"</span>)</span>
<span id="cb2-516"><a href="#cb2-516"></a></span>
<span id="cb2-517"><a href="#cb2-517"></a><span class="co"># 3) Choose ONE file to read first (edit the filename to match your date)</span></span>
<span id="cb2-518"><a href="#cb2-518"></a>file_csv  <span class="ot">&lt;-</span> <span class="st">"data/survey_demo_YYYYMMDD_v01.csv"</span></span>
<span id="cb2-519"><a href="#cb2-519"></a>file_tsv  <span class="ot">&lt;-</span> <span class="st">"data/survey_demo_YYYYMMDD_v01.tsv"</span></span>
<span id="cb2-520"><a href="#cb2-520"></a>file_scsv <span class="ot">&lt;-</span> <span class="st">"data/survey_demo_YYYYMMDD_v01.scsv"</span></span>
<span id="cb2-521"><a href="#cb2-521"></a></span>
<span id="cb2-522"><a href="#cb2-522"></a><span class="co"># 4) Read one of them (start with TSV to avoid comma ambiguity in notes)</span></span>
<span id="cb2-523"><a href="#cb2-523"></a>dat <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(file_tsv, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-524"><a href="#cb2-524"></a></span>
<span id="cb2-525"><a href="#cb2-525"></a><span class="co"># 5) Inspect what you imported</span></span>
<span id="cb2-526"><a href="#cb2-526"></a><span class="fu">str</span>(dat)</span>
<span id="cb2-527"><a href="#cb2-527"></a><span class="fu">head</span>(dat, <span class="dv">3</span>)</span>
<span id="cb2-528"><a href="#cb2-528"></a></span>
<span id="cb2-529"><a href="#cb2-529"></a><span class="co"># 6) Basic checks: dimensions and missingness</span></span>
<span id="cb2-530"><a href="#cb2-530"></a><span class="fu">dim</span>(dat)</span>
<span id="cb2-531"><a href="#cb2-531"></a><span class="fu">colSums</span>(<span class="fu">is.na</span>(dat))</span>
<span id="cb2-532"><a href="#cb2-532"></a></span>
<span id="cb2-533"><a href="#cb2-533"></a><span class="co"># 7) Now read the comma CSV version and compare</span></span>
<span id="cb2-534"><a href="#cb2-534"></a>dat_csv <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(file_csv, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-535"><a href="#cb2-535"></a><span class="fu">identical</span>(<span class="fu">names</span>(dat), <span class="fu">names</span>(dat_csv))</span>
<span id="cb2-536"><a href="#cb2-536"></a></span>
<span id="cb2-537"><a href="#cb2-537"></a><span class="co"># 8) Read the semicolon-delimited version explicitly</span></span>
<span id="cb2-538"><a href="#cb2-538"></a>dat_scsv <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(file_scsv, <span class="at">sep =</span> <span class="st">";"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb2-539"><a href="#cb2-539"></a></span>
<span id="cb2-540"><a href="#cb2-540"></a><span class="co"># 9) Compare what happened to the notes column across imports</span></span>
<span id="cb2-541"><a href="#cb2-541"></a><span class="fu">head</span>(dat<span class="sc">$</span>notes, <span class="dv">3</span>)</span>
<span id="cb2-542"><a href="#cb2-542"></a><span class="fu">head</span>(dat_csv<span class="sc">$</span>notes, <span class="dv">3</span>)</span>
<span id="cb2-543"><a href="#cb2-543"></a><span class="fu">head</span>(dat_scsv<span class="sc">$</span>notes, <span class="dv">3</span>)</span>
<span id="cb2-544"><a href="#cb2-544"></a><span class="in">```</span></span>
<span id="cb2-545"><a href="#cb2-545"></a></span>
<span id="cb2-546"><a href="#cb2-546"></a><span class="ss">16. </span>When you are done, answer these questions in a short paragraph in your notebook or as comments at the bottom of the script:</span>
<span id="cb2-547"><a href="#cb2-547"></a></span>
<span id="cb2-548"><a href="#cb2-548"></a>a) Which file type imported with the least friction, and why?</span>
<span id="cb2-549"><a href="#cb2-549"></a>b) Did any import alter the <span class="in">`notes`</span> column in a way you did not expect? What does that suggest about delimiter choice?</span>
<span id="cb2-550"><a href="#cb2-550"></a>c) Which steps in your workflow made it easy to locate the files again without clicking through folders?</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>© 2026, AJ Smit</p>
</div>   
    <div class="nav-footer-center">

<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB744/intro_r/02-working-with-data.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>


<script src="../../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>