<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.21">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Smit, A. J.">
<meta name="dcterms.date" content="2021-01-01">
<title>3. Data Classes &amp; Structures – The Tangled Bank</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../BCB744/intro_r/04-workflow.html" rel="next">
<link href="../../BCB744/intro_r/02-working-with-data.html" rel="prev">
<link href="../../images/favicon.ico" rel="icon">
<script src="../../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-f9d2e688c83043042fa051765c26f29b.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-b5324c11422050a6ca6038ce3f1a2d11.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-SCYH68M489"></script><script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-SCYH68M489', { 'anonymize_ip': true});
</script><script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"headline",
  "consent_type":"express",
  "palette":"light",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  ,
"language":"en"
  });
});
</script><style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script><script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script><script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN" && texText && texText.data) {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script><link rel="stylesheet" href="../../styles.css">
<meta property="og:title" content="3. Data Classes &amp; Structures – The Tangled Bank">
<meta property="og:description" content="Getting familiar with data classes and structures">
<meta property="og:image" content="http://tangledbank.netlify.app/BCB744/intro_r/data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==">
<meta property="og:site_name" content="The Tangled Bank">
</head>
<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../index.html" class="navbar-brand navbar-brand-logo">
    </a>
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">The Tangled Bank</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-vignettes" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Vignettes</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-vignettes">
<li>
    <a class="dropdown-item" href="../../vignettes/netCDF_dates.html">
 <span class="dropdown-text">Dates From netCDF Files: Two Approaches</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/prep_NOAA_OISST.html">
 <span class="dropdown-text">Downloading and Prep: ERDDAP</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_ERDDAP.html">
 <span class="dropdown-text">Retrieving Chlorophyll-a Data from ERDDAP Servers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/alt_method.html">
 <span class="dropdown-text">Downloading and Prep: Python + CDO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/gridded_data.html">
 <span class="dropdown-text">Detecting Events in Gridded Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/regridding.html">
 <span class="dropdown-text">Regridding gridded data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/MHW_MCS_horizonplots.html">
 <span class="dropdown-text">Displaying MHWs and MCSs as Horizon Plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/heatwaveR_issues.html">
 <span class="dropdown-text">heatwaveR Issues</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_sightings.html">
 <span class="dropdown-text">Plotting the Whale Sightings and Chlorophyll-<em>a</em> Concentrations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_localisation.html">
 <span class="dropdown-text">Spatial Localisation, Subsetting, and Aggregation of the Chlorophyll-<em>a</em> Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/elem_ts_methods.html">
 <span class="dropdown-text">Wavelet Analysis of Diatom Time Series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/buffer_data_extract.html">
 <span class="dropdown-text">Extracting gridded data within a buffer</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-high-performace-computing" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">High-Performace Computing</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-high-performace-computing">
<li>
    <a class="dropdown-item" href="../../vignettes/README_Lengau.html">
 <span class="dropdown-text">Using Lengau</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/PBSPro_users.html">
 <span class="dropdown-text">PBSPro user and job management</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/README_tmux.html">
 <span class="dropdown-text">Using tmux</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item">
    <a class="nav-link" href="../../blog.html"> 
<span class="menu-text">Tangled Bank Blog</span></a>
  </li>  
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-blogroll" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Blogroll</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-blogroll">
<li>
    <a class="dropdown-item" href="https://www.r-bloggers.com">
 <span class="dropdown-text">R-bloggers</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item compact">
    <a class="nav-link" href="../../blog.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="link" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">
<li>
    <a class="dropdown-item" href="https://github.com/ajsmit/tangled_bank">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://bugs.com">
 <span class="dropdown-text">Report a Bug</span></a>
  </li>  
    </ul>
</li>
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../BCB744/BCB744_index.html">BCB744: Introduction to R, &amp; Biostatistics</a></li><li class="breadcrumb-item"><a href="../../BCB744/intro_r/03-data-in-R.html">3. Data Classes &amp; Structures</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header sidebar-header-stacked">
      <a href="../../index.html" class="sidebar-logo-link">
      <img src="../../images/my_tree_hex.png" alt="" class="sidebar-logo light-content py-0 d-lg-inline d-none"><img src="../../images/my_tree_hex.png" alt="" class="sidebar-logo dark-content py-0 d-lg-inline d-none"></a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">UNDERGRADUATE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC223: Plant Ecophysiology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/BDC223_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L00-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Introductory Lecture</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L01-worldmapper.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: The Limits to Life</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L02-SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2: Surface Area to Volume (SA/V) Ratio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L03-plant_stress.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3: Plant Stress</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L04-carbon_cycle.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Perturbations to the Carbon Cycle</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L05-light.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 5: Light</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L06a-pigments_photosynthesis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6a: Pigments and Photosynthesis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L06b-jassby_platt.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6b: Pigments and Photosynthesis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L07-chromatic_adaptation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 7: Chromatic Adaptation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L08a-nutrient_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8a: Nutrient Uptake</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/L08b-nutrients_michaelis_menten.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 8b: Nutrient Uptake</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab1_SA_V.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1: Surface Area to Volume (SA/V) Ratios in Biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab2_misc_calcs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2: Miscellaneous Calculations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab3_PI_curves.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3: PI Curves – Jassby and Platt</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/Lab4_nitrogen_uptake.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4: Uptake Kinetics – Michaelis-Menten</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC223/BDC223_FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false">
 <span class="menu-text">BDC334: Biogeography &amp; Global Ecology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/BDC334_syllabus.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/BDC334-Lecture-Transcripts.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">BCB334 Lecture Transcript</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 1: Introductory Lecture</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-02-ecosystems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 2. Overview of Ecosystems</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-03-gradients.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 3. Ecological Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 4: Biodiversity Concepts</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 5: Multivariate Data</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lec-06-unified-ecology.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 6: Unified Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 7: Threats to Biodiversity</span>
  </li>
          <li class="sidebar-item">
 <span class="menu-text">Lecture 8: Nature's Contribution to People</span>
  </li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/IKS_Conservation.pdf" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lecture 9: Indigenous Knowledge Systems and Conservation</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1. Ecological Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-02a-r_rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2a. Introduction `R` &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-02b-env_dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2b. Environmental Distance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-03-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3. Quantifying Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/Lab-04-biodiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4. Species Distribution Patterns</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/BDC334_AI_Tutor.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">BCB334 AI Tutor</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pages/Transboundary_systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transboundary systems</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS CORE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">BCB744: Introduction to R, &amp; Biostatistics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/BCB744_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/01-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. R &amp; RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/02-working-with-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Working With Data &amp; Code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/03-data-in-R.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">3. Data Classes &amp; Structures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/04-workflow.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. R Workflows</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/05-graphics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Graphics with <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/06-faceting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Faceting Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/07-brewing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Brewing Colours</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/08-mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Mapping With <strong>ggplot2</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/09-mapping_style.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Mapping With Style</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/10-mapping_rnaturalearth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Mapping With Natural Earth</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/11-mapping_quakes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. The Fiji Earthquake Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/12-tidy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Tidy Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/13-tidier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Tidier Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/14-tidiest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Tidiest Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/15-recap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15. Recap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../assessments/BCB744_Intro_R_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Intro R Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../assessments/examples/BCB744_Intro_R_Example_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 Intro R Example 1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../assessments/examples/BCB744_Intro_R_Example_2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 Intro R Example 2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../self-assessment_v1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Self-Assessment Form</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../assessments/BCB744_Intro_R_Presentations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Presentations</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../assessments/BCB744_Intro_R_Test_2025.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Intro R Test</span></span></a>
  </div>
</li>
          <li class="px-0"><hr class="sidebar-divider hi "></li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/01-scientific-philosophy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Philosophy of Science</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/02-summarise-and-describe.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Data Summaries &amp; Descriptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/03-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Statistical Figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/04-distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data Distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/05-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Statistical Inference</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/06-assumptions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Assumptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/07-t_tests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. <em>t</em>-Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/08-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. ANOVA</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/09-regressions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Simple Linear Regressions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/10-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/11-decision_guide.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/12-glance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Non-Parametric Tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/13-confidence.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Confidence Intervals</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/14-transformations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Data Transformations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../assessments/BCB744_Biostatistics_Self-Assessment.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="my-highlight">Biostatistics Self-Assessment</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../assessments/examples/BCB744_BioStats_Example_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744 BioStats 2025 Example 1</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS ELECTIVES</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false">
 <span class="menu-text">BCB743: Quantitative Ecology</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/BCB743_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><strong>About</strong></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/review.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1-4: Review Concepts fo Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5: Correlations &amp; Associations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7: Intro to Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8a: Principal Components Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCA_examples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8b: PCA: Additional Examples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCA_SDG_example.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8c: PCA: WHO SDG Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/CA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9a: Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/DCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9b: Detrended Correspondence Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/PCoA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10: Principal Coordinates Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/nMDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11a: Non-metric Multidimensional Scaling</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/nMDS_diatoms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11b: nMDS: PERMANOVA (Diatoms) Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/unconstrained-summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12: Unconstrained Ordination Summary</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/constrained_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13a: Distance-Based Redundancy Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/two_oceans_appendices.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13b: db-RDA: Seaweeds Example</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/cluster_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14: Cluster Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/model_building.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15: Model Building</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/deep_dive.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16a: Deep Dive Into Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/multiple_regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16b: Multiple Regression</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">AI &amp; POLICY</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../docs/Combined Faculty AI Policy 15082025.pdf" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Faculty of Natural Sciences AI Policy</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../AI/BCB_AI_Workshop_Sep_2025.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB AI Workshop</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">SUPPORT</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false">
 <span class="menu-text">Web Resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/general_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/ecology_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quantitative Ecology Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/spatial_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial R Resources</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘☘☘︎</span>
  </li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a></li>
  <li>
<a href="#types-of-variables-common-in-biology-and-ecology" id="toc-types-of-variables-common-in-biology-and-ecology" class="nav-link" data-scroll-target="#types-of-variables-common-in-biology-and-ecology">Types of variables common in biology and ecology</a>
  <ul class="collapse">
<li>
<a href="#numeric-variables" id="toc-numeric-variables" class="nav-link" data-scroll-target="#numeric-variables">Numeric variables</a>
  <ul class="collapse">
<li><a href="#discrete-variables" id="toc-discrete-variables" class="nav-link" data-scroll-target="#discrete-variables">Discrete variables</a></li>
  <li><a href="#continuous-variables" id="toc-continuous-variables" class="nav-link" data-scroll-target="#continuous-variables">Continuous variables</a></li>
  </ul>
</li>
  <li><a href="#dates" id="toc-dates" class="nav-link" data-scroll-target="#dates">Dates</a></li>
  <li>
<a href="#character-data" id="toc-character-data" class="nav-link" data-scroll-target="#character-data">Character data</a>
  <ul class="collapse">
<li><a href="#nominal-variables" id="toc-nominal-variables" class="nav-link" data-scroll-target="#nominal-variables">Nominal variables</a></li>
  <li><a href="#ordinal-variables" id="toc-ordinal-variables" class="nav-link" data-scroll-target="#ordinal-variables">Ordinal variables</a></li>
  </ul>
</li>
  <li><a href="#binary-variables" id="toc-binary-variables" class="nav-link" data-scroll-target="#binary-variables">Binary variables</a></li>
  <li><a href="#missing-values" id="toc-missing-values" class="nav-link" data-scroll-target="#missing-values">Missing values</a></li>
  <li><a href="#complex-numbers" id="toc-complex-numbers" class="nav-link" data-scroll-target="#complex-numbers">Complex numbers</a></li>
  </ul>
</li>
  <li>
<a href="#data-classes-in-r" id="toc-data-classes-in-r" class="nav-link" data-scroll-target="#data-classes-in-r">Data classes in R</a>
  <ul class="collapse">
<li><a href="#numeric" id="toc-numeric" class="nav-link" data-scroll-target="#numeric"><code>numeric</code></a></li>
  <li><a href="#character" id="toc-character" class="nav-link" data-scroll-target="#character"><code>character</code></a></li>
  <li><a href="#factor" id="toc-factor" class="nav-link" data-scroll-target="#factor"><code>factor</code></a></li>
  <li><a href="#logical" id="toc-logical" class="nav-link" data-scroll-target="#logical"><code>logical</code></a></li>
  <li><a href="#date" id="toc-date" class="nav-link" data-scroll-target="#date"><code>date</code></a></li>
  <li><a href="#missing-values-na" id="toc-missing-values-na" class="nav-link" data-scroll-target="#missing-values-na">Missing values, <code>NA</code></a></li>
  </ul>
</li>
  <li>
<a href="#data-structures-in-r" id="toc-data-structures-in-r" class="nav-link" data-scroll-target="#data-structures-in-r">Data structures in R</a>
  <ul class="collapse">
<li><a href="#vector-array-and-matrix" id="toc-vector-array-and-matrix" class="nav-link" data-scroll-target="#vector-array-and-matrix"><code>vector</code>, <code>array</code>, and <code>matrix</code></a></li>
  <li><a href="#data.frame" id="toc-data.frame" class="nav-link" data-scroll-target="#data.frame"><code>data.frame</code></a></li>
  <li><a href="#tibble" id="toc-tibble" class="nav-link" data-scroll-target="#tibble"><code>tibble</code></a></li>
  <li><a href="#list" id="toc-list" class="nav-link" data-scroll-target="#list"><code>list</code></a></li>
  </ul>
</li>
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion">Conclusion</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB744/intro_r/03-data-in-R.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../BCB744/BCB744_index.html">BCB744: Introduction to R, &amp; Biostatistics</a></li><li class="breadcrumb-item"><a href="../../BCB744/intro_r/03-data-in-R.html">3. Data Classes &amp; Structures</a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">3. Data Classes &amp; Structures</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
<p class="subtitle lead">Getting familiar with data classes and structures</p>
</div>


<div class="quarto-title-meta-author">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://tangledbank.netlify.app">Smit, A. J.</a> <a href="https://orcid.org/0000-0002-3799-6126" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uwc.ac.za">
            University of the Western Cape
            </a>
          </p>
      </div>
  </div>

<div class="quarto-title-meta">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 1, 2021</p>
    </div>
  </div>
  
    
  </div>
  


</header><blockquote class="blockquote">
<p>“<em>That which can be destroyed by the truth should be.</em>”</p>
<p>— P.C. Hodgell</p>
</blockquote>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>In this Chapter
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Data as encountered by the statistician</li>
<li>Data classes in R</li>
<li>Data structures in R</li>
</ul>
</div>
</div>
<div class="callout callout-style-simple callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Important</span>Tasks to complete in this Chapter
</div>
</div>
<div class="callout-body-container callout-body">
<ul>
<li>Task A 1-4</li>
</ul>
</div>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="../../images/769_life_finds_a_way.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1"><img src="../../images/769_life_finds_a_way.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="300"></a></p>
</figure>
</div>
<section id="introduction" class="level1"><h1>Introduction</h1>
<p>In biostatistics we will encounter different data types, and comprehending <strong>data classes</strong>, <strong>data structures</strong>, and their <strong>statistical interpretations</strong> is important for several reasons:</p>
<ul>
<li><p><strong>Correctly identifying the data class</strong> To manipulate, analyse, or visualise data effectively, we must first identify the type of data we are working with. Since different data classes possess distinct properties, we must determine the class of our data to leverage the appropriate functions and operations.</p></li>
<li><p><strong>Efficient data processing</strong> As biologists using R, we may come across data structures that are not fully compatible with the available functions. For example, when dealing with extensive datasets, employing data structures intended for vectorised operations like arrays or matrices can significantly boost data processing speed. In contrast, other data representations may not perform well when managing large datasets. By becoming familiar with the different data classes and their attributes in R, we can make informed choices about which data structures and functions to use to accomplish their data processing and analysis tasks effectively.</p></li>
<li><p><strong>Data manipulation and analysis</strong> Different data classes in R have specific methods for manipulating and analysing data. For example, if we’re working with character strings, we can use string manipulation functions like <code><a href="https://rdrr.io/r/base/grep.html">gsub()</a></code> or <code><a href="https://rdrr.io/r/base/strsplit.html">strsplit()</a></code> which do not work on numerical data. If we’re working with dates and times, we need to use date and time functions like <code><a href="https://rdrr.io/r/base/as.Date.html">as.Date()</a></code> or <code><a href="https://lubridate.tidyverse.org/reference/ymd.html">lubridate::ymd()</a></code>. By understanding the different data classes in R, we can choose the appropriate functions for manipulating and analysing our data.</p></li>
<li><p><strong>Data visualisation</strong> Data can be visualised in numerous ways, such as using histograms for numeric data, bar graphs for categorical data, or scatter plots for representing the relationship between two numeric variables. However, the way we provide data to R is critical. By familiarising ourselves with the diverse data classes in R, we can select the most suitable visualisation methods for our data.</p></li>
</ul>
<p>Let us delve into how biologists classify data and the distinct terminology they employ to describe it. It’s essential to comprehend these subtleties before we investigate how R recognises data classes. We’ll scrutinise the data types that are ubiquitous in the fields of biology and ecology, and subsequently shift our attention to data classes and structures in R.</p>
</section><section id="types-of-variables-common-in-biology-and-ecology" class="level1"><h1>Types of variables common in biology and ecology</h1>
<p>We will most frequently encounter data arranged in columns of a data file—typically in MS Excel files or CSV files. A column is a variable, and one variable is comprised of one data type. So, when we refer to a variable, we expect that all the data within would be homogeneous—at least in as far as the data’s type.</p>
<p>The type of data that biologists and statisticians work with can influence the statistical techniques and methods they use to analyse and interpret the data. Let us discuss some of the different types of biological and ecological data we are likely to encounter.</p>
<section id="numeric-variables" class="level2"><h2 class="anchored" data-anchor-id="numeric-variables">Numeric variables</h2>
<p>Numeric data in the context of biostatistics refers to <strong>quantitative data</strong> that can be expressed in numerical form, typically obtained from field and laboratory measurements, or from field sampling campaigns. Examples of numeric data in biostatistics include the height and mass of a animals, concentrations of nutrients, laboratory test results such as respiration rates, or the number of limpets in a quadrat. Numeric data can be further categorised as <strong>discrete</strong> or <strong>continuous</strong>.</p>
<section id="discrete-variables" class="level3"><h3 class="anchored" data-anchor-id="discrete-variables">Discrete variables</h3>
<p>Discrete data are whole (integer) numbers that represent counts of items or events. Integer data usually answer the question, “how many?” For example, in the biological and Earth sciences, discrete data are commonly encountered in the form of counts or integers that represent the presence or absence of certain characteristics or events. For example, the number of individuals of some species in a population, the number of chromosomes in a cell, or the number of earthquakes occurring in a region within a given time frame. Other examples of discrete data in these sciences include the number of mutations in a gene, the number of cells in a tissue sample, or the number of species present in an ecosystem. These types of data are often analysed using statistical techniques such as frequency distributions, contingency tables, and chi-square tests.</p>
</section><section id="continuous-variables" class="level3"><h3 class="anchored" data-anchor-id="continuous-variables">Continuous variables</h3>
<p>Continuous data, on the other hand, are measured on a continuous scale. These usually represent measured quantities such as something’s heat content (temperature, measured in degrees Celsius) or distance (measured in metres or similar), etc. They can be rational numbers including integers and fractions, but typically they have an infinite number of ‘steps’ that depend on rounding (they can even be rounded to whole integers) or considerations such as measurement precision and accuracy. Often, continuous data have upper and lower bounds that depend on the characteristics of the phenomenon being studied or the measurement being taken.</p>
</section></section><section id="dates" class="level2"><h2 class="anchored" data-anchor-id="dates">Dates</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="../../images/Dates.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2"><img src="../../images/Dates.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="300"></a></p>
</figure>
</div>
<p>We often encounter date data when dealing with time-related data. For example, in ecological research, data collection may involve recording the date of a particular observation or sampling event, such as the date when a bird was sighted, or when water samples were taken from a stream. The purpose of using date (or time) data in biology and ecology is to enable us to understand and analyse temporal patterns and relationships in their response variables. This can include exploring seasonal trends, understanding the impact of environmental changes over time, or tracking the growth and development of organisms.</p>
<p>By analysing date data, we can gain insights into long-term trends and patterns that may not be apparent when looking at the data in aggregate. They can also use this information to make predictions about future trends, develop more effective management strategies, and identify potential areas for further research.</p>
</section><section id="character-data" class="level2"><h2 class="anchored" data-anchor-id="character-data">Character data</h2>
<p>Character data are used to describe qualitative variables or descriptive text that are not numerical in nature. Character data can be entered as descriptive character strings, and internally, they are translated into a vector of characters in R. They are often used to represent categorical variables, such as the type of plant species, the colour of a bird’s feathers, or the name of a some gene. Social scientists will sometimes use character data fields to record the names of people or places, or other descriptive information, such as a narrative that will later be subjected to, for example, a sentiment analysis. For convenience, I’ll call these data <strong>narrative style</strong> data to distinguish them from the <strong>qualitative</strong> data that are the main focus of the present discussion.</p>
<p>Since narrative style data are not directly amenable to statistical analsysis, in this module, we will mainly concern ourselves with qualitative data which are typically names of things, or categories of objects, classes of behaviours, properties, characteristics, and so on. Qualitative data typically refer to <em>non-numeric data</em> collected from observations, experimental treatment groups, or other sources. They tend to be textual and are often used to describe characteristics or properties of living organisms, ecosystems, or other biological phenomena. Examples may include the colour of flowers, the type of habitat where an animal is found, the behaviour of animals, or the presence or absence of certain traits or characteristics in a population.</p>
<p>Qualitative data can be further classified into <strong>nominal</strong> or <strong>ordinal</strong> data types. Ordinal and nominal data are both amenable to statistical interpretation.</p>
<section id="nominal-variables" class="level3"><h3 class="anchored" data-anchor-id="nominal-variables">Nominal variables</h3>
<p>Nominal data are used to describe qualitative variables that do not have any inherent order or ranking. Examples of nominal data in biology may include the type of plant or animal species, or the presence or absence of certain genetic traits. Another term for nominal data is <em>categorical data</em>. Because there are well-defined categories, the number of members belonging to each of the category can be counted. For example, there are three red flowers, 66 purple flowers, and 13 yellow flowers.</p>
</section><section id="ordinal-variables" class="level3"><h3 class="anchored" data-anchor-id="ordinal-variables">Ordinal variables</h3>
<p>Ordinal data refer to a type of data that can be used to describe qualitative categorical variables that have a natural order or ranking. It is used when we need to arrange things in a particular order, such as from worst to best or from least to most. However, the differences between the values cannot be measured or quantified exactly, making them somewhat subjective. Examples of ordinal data include the different stages of development of an organism or the performance of a species to different fertilisers. Ordinal data can be entered as descriptive character strings, and internally, they are translated into an ordered vector of integers in R. For example, we can use a scale of <code>1</code> for terrible, <code>2</code> for ‘so-so’, <code>3</code> for average, <code>4</code> for good, and <code>5</code> for brilliant.</p>
</section></section><section id="binary-variables" class="level2"><h2 class="anchored" data-anchor-id="binary-variables">Binary variables</h2>
<p>Life can be boiled down to a series of binary decisions: should I have pizza for dinner, <em>yes</em> or <em>no</em>? Should I go to bed early, <em>TRUE</em> or <em>FALSE</em>? Should I start that new series on Netflix, <em>accept</em> or <em>reject</em>? Am I <em>present</em> or <em>absent</em>? You get the gist… This kind of binary decision-making is known as ‘logical’, and in R they can only take on the values of <code>TRUE</code> or <code>FALSE</code> (remember to mind your case!). In the computing world, logical data are often represented by <code>1</code> for <code>TRUE</code> and <code>0</code> for <code>FALSE</code>. So basically, your life’s choices can be summarised as a string of <code>1</code>s and <code>0</code>s. Who knew it was that simple?</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="../../images/bad_choices.jpg" class="lightbox" data-gallery="quarto-lightbox-gallery-3"><img src="../../images/bad_choices.jpg" class="img-fluid quarto-figure quarto-figure-center figure-img" width="300"></a></p>
</figure>
</div>
<p>When it comes down to it, everything in life is either <code>black</code> or <code>white</code>, <code>right</code> or <code>wrong</code>, <code>good</code> or <code>bad</code>. It’s like a cosmic game of “Would You Rather?” — and we’re all just playing along.</p>
</section><section id="missing-values" class="level2"><h2 class="anchored" data-anchor-id="missing-values">Missing values</h2>
<p>It’s unfortunate to admit that one of the most reliable aspects of any biological dataset is the presence of missing data (the presence of something that’s missing?!). It is a stark reminder of the fragility of life. How can we say that something contains missing data? It seems counter intuitive, as if the data were never there in the first place. However, as we remember the principles of <em>tidy</em> data, we see that every observation must be documented in a row, and each column in that row must contain a value. This organisation allows us to create a matrix of data from multiple observations. Since the data are presented in a two-dimensional format, any missing values from an observation will leave a gaping hole in the matrix. We call these ‘missing values.’ It’s a somber reality that even the most meticulous collection of data can be marred by the loss of information.</p>
</section><section id="complex-numbers" class="level2"><h2 class="anchored" data-anchor-id="complex-numbers">Complex numbers</h2>
<blockquote class="blockquote">
<p><em>“And if you gaze long enough into an abyss, the abyss will gaze back into you.”</em></p>
<p>— Friedrich Nietzsche</p>
</blockquote>
<p>As we draw to a close on the topic of data types, we cling desperately to the threads of our sanity, hoping against hope that they remain tightly stitched. But let it be known, to those who dare enter further into the realm of data, that beneath the surface lie countless rocks, and around every corner lurk a legion of complex data types, waiting to ensnare the unwary. These shadows of information are as enigmatic as they are perilous, for they challenge the very essence of our understanding. It is not until the final chapter of our journey, when we confront the elusive art of modeling, that we will face these data demons head-on. But fear not, for we shall arm ourselves with the knowledge and techniques acquired on this treacherous path, and with each step forward, we shall move closer to mastering the darkness that awaits us.</p>
</section></section><section id="data-classes-in-r" class="level1"><h1>Data classes in R</h1>
<p>R is a powerful programming language and software environment for statistical computing and graphics. It offers a wide variety of data classes and types to represent different types of information.</p>
<p>The <strong>atomic modes</strong> are <code>logical</code>, <code>integer</code>, <code>numeric</code> (also sometimes called <code>double</code>), <code>complex</code>, <code>character</code> and <code>raw</code>. The <code>Date</code> class is a specialised form of the <code>numeric</code> class. Each atomic mode has its own properties and functions that can be used to manipulate objects of that mode. These atomic modes can be used to make an atomic data structure such as a <code>vector</code>, <code>array</code>, or <code>matrix</code>. This knowledge is also important when working with R tabular mixed data structures, such as a <code>data.frame</code>, <code>tibble</code>, or <code>list</code>. Please refer to Hadley Wickham’s overview of vectors in <a href="https://adv-r.hadley.nz/index.html">Advanced R, 2nd edition</a> for more insight presented in an informative yet concise way.</p>
<p>The <strong>data class</strong> can be determined with the <code><a href="https://rdrr.io/r/base/class.html">class()</a></code> or <code><a href="https://rdrr.io/r/utils/str.html">str()</a></code> commands. Here’s a brief overview of some of the main data types in R:</p>
<section id="numeric" class="level2"><h2 class="anchored" data-anchor-id="numeric"><code>numeric</code></h2>
<p>In R, the <code>numeric</code> data class represents either <strong>integers</strong> or <strong>floating point</strong> (decimal) values. Numerical data are quantitative in nature as they represent things that can be objectively counted, measured, or calculated—the <em>measured variables</em>.</p>
<p>Numeric data are one of the most common types of data used in statistical and mathematical analysis. In R, numeric data are represented by the class <code>numeric</code>, which includes both integers and floating-point numbers. Numeric data can be used in a variety of operations and calculations, including arithmetic operations, statistical analyses, and visualisations. One important feature of the <code>numeric</code> data class in R is that it supports vectorisation, which allows for efficient and concise operations on large sets of numeric data. Additionally, R provides a wide range of built-in functions for working with numeric data, including functions for calculating basic statistical measures such as mean, median, and standard deviation.</p>
<p>In R integer (discrete) data are called <code>int</code> or <code>&lt;int&gt;</code> while continuous data are denoted <code>num</code> or <code>&lt;dbl&gt;</code>.</p>
<p><strong>Example of integer data</strong> Suppose you have a dataset of the number of rats in different storm water drains in a neighbourhood. The number of rats is a discrete variable because it can only take on integer values (you can’t own a fraction of a rat).</p>
<p>Here’s how you could create a vector of this data in R:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># Create a vector of the number of pets owned by each household</span></span>
<span id="cb1-2"><a href="#cb1-2"></a>num_rats <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">1</span>, <span class="dv">4</span>, <span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">0</span>, <span class="dv">3</span>, <span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">4</span>, <span class="dv">2</span>, <span class="dv">0</span>)</span>
<span id="cb1-3"><a href="#cb1-3"></a>num_rats</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] 0 1 2 2 3 1 4 0 2 1 2 2 0 3 2 1 1 4 2 0</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="fu">class</span>(num_rats)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "numeric"</code></pre>
</div>
</div>
<p>In this example, the data are represented as a vector called <code>num_rats</code> of class <code>numeric</code> (as revealed by <code>class(num_rats)</code>). Each element of the vector represents the number of rats in one storm water drain. For example, the first element of the vector (<code>num_rats[1]</code>) is <code>0</code>, which means that the first drain in the dataset is free of rats. The fourth element of the vector (<code>num_rats[4]</code>) is <code>2</code>, indicating that the fourth drain in the dataset is occupied by <code>2</code> rats.</p>
<p>One can also explicitly create a vector of integer using the <code><a href="https://rdrr.io/r/base/integer.html">as.integer()</a></code> function:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>num_rats_int <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(num_rats)</span>
<span id="cb5-2"><a href="#cb5-2"></a>num_rats_int</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] 0 1 2 2 3 1 4 0 2 1 2 2 0 3 2 1 1 4 2 0</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="fu">class</span>(num_rats_int)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "integer"</code></pre>
</div>
</div>
<p>Above we <em>coerced</em> the class <code>numeric</code> data to class <code>integer</code>. But we can take floating point numeric and convert them to integers too with the <code><a href="https://rdrr.io/r/base/integer.html">as.integer()</a></code> function. As we see, the effect is that the whole part of the number is retained and the rest discarded:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>pies <span class="ot">&lt;-</span> pi <span class="sc">*</span> <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)</span>
<span id="cb9-2"><a href="#cb9-2"></a>pies</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  3.141593  6.283185  9.424778 12.566371 15.707963</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="fu">class</span>(pies)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "numeric"</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="fu">as.integer</span>(pies)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  3  6  9 12 15</code></pre>
</div>
</div>
<p>Effectively, what happened above is more-or-less equivalent to what the <code><a href="https://rdrr.io/r/base/Round.html">floor()</a></code> function would return:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="fu">floor</span>(pies)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  3  6  9 12 15</code></pre>
</div>
</div>
<p>Be careful when coercing floating point numbers to integers. If rounding is what you expect, this is not what you will get. For rounding, use <code><a href="https://rdrr.io/r/base/Round.html">round()</a></code> instead:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="fu">round</span>(pies, <span class="dv">0</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  3  6  9 13 16</code></pre>
</div>
</div>
<p><strong>Example of continuous data</strong> Here are some randomly generated temperature data assigned to an object called <code>temp_data</code>:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="co"># Generate a vector of 50 normally distributed temperature values</span></span>
<span id="cb19-2"><a href="#cb19-2"></a>temp_data <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">rnorm</span>(<span class="at">n =</span> <span class="dv">50</span>, <span class="at">mean =</span> <span class="dv">15</span>, <span class="at">sd =</span> <span class="dv">3</span>), <span class="dv">2</span>)</span>
<span id="cb19-3"><a href="#cb19-3"></a>temp_data</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] 14.18 20.21 15.95  7.26 10.20 17.08 21.95 13.82 16.02 15.23 14.68  9.82
[13] 16.26 13.63 12.76 15.45 15.81 15.34 13.43 20.26 11.77 20.24  9.84 16.22
[25] 12.53 13.88 14.55 19.23 15.01  8.66 12.60  6.28 17.44 15.21 17.49 16.56
[37] 14.73 14.20 14.76 10.37 17.08 13.90 17.46 16.69 16.17  9.93 13.76 12.45
[49] 15.75 19.20</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="fu">class</span>(temp_data)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "numeric"</code></pre>
</div>
</div>
</section><section id="character" class="level2"><h2 class="anchored" data-anchor-id="character"><code>character</code></h2>
<p>In R, the <code>character</code> data class represents textual data such as words, sentences, and paragraphs. Character data can be created using either single or double quotes, and it can include letters, numbers, and other special characters. In addition, character data can be concatenated using the <code><a href="https://rdrr.io/r/base/paste.html">paste()</a></code> function or other string manipulation functions.</p>
<p>One important feature of the character data class in R is its versatility in working with textual data. For instance, it can be used to store and manipulate text data, including text-based datasets, text-based files, and text-based visualisations. Additionally, R provides a wide range of built-in functions for working with character data, including functions for manipulating strings, searching for patterns, and formatting output. Overall, the character data class in R is a fundamental data type that is critical for working with textual data in a variety of contexts. You will most frequently use character values are often used to represent labels, names, or descriptions.</p>
</section><section id="factor" class="level2"><h2 class="anchored" data-anchor-id="factor"><code>factor</code></h2>
<p>In R, the <code>factor</code> data class is used to represent discrete categorical variables. Factors are often used in statistical analyses to represent class or group belonging. Factor values are categorical data, such as levels or categories of a variable. Factor variables are most commonly also character data, but they can be numeric too if coded correctly as factors. Factor values can be ordered (ordinal) or unordered (categorical or nominal).</p>
<p>Categorical variables take on a limited number of distinct values, often corresponding to different groups or levels. For example, a categorical variable might represent different colours, size classes, or species. Factors in R are represented as integers with corresponding character levels, where each level corresponds to a distinct category. The levels of a factor can be defined explicitly using the <code><a href="https://rdrr.io/r/base/factor.html">factor()</a></code> function or automatically using the <code><a href="https://rdrr.io/r/base/cut.html">cut()</a></code> function. One important feature of the factor data class in R is that it allows for efficient and effective data manipulation and analysis, particularly when working with large datasets. For instance, factors can be used in statistical analyses such as regression models or ANOVA, and they can also be used to create visualisations such as bar or pie graphs. The <code>factor</code> data class in R is a fundamental data type that is critical for representing and working with categorical variables in data analysis and visualisation.</p>
<p>The <code>factor</code> data class of data in an R <code>data.frame</code> structure (or in a <code>tibble</code>) is indicated by <code>Factor</code> or <code>&lt;fctr&gt;</code>. Ordered factors are denoted by columns named <code>Ord.factor</code> or <code>&lt;ord&gt;</code>.</p>
<p><strong>Nominal data</strong> One example of nominal factor data that ecologists might encounter is the type of vegetation in a particular area, such as ‘grassland’, ‘forest’, or ‘wetland’. Here’s an example of how to generate a vector of nominal data in R using the <code><a href="https://rdrr.io/r/base/sample.html">sample()</a></code> function:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="co"># Generate a vector of vegetation types</span></span>
<span id="cb23-2"><a href="#cb23-2"></a>vegetation <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="st">"grassland"</span>, <span class="st">"forest"</span>, <span class="st">"wetland"</span>), <span class="at">size =</span> <span class="dv">50</span>, <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb23-3"><a href="#cb23-3"></a></span>
<span id="cb23-4"><a href="#cb23-4"></a><span class="co"># View the vegetation data</span></span>
<span id="cb23-5"><a href="#cb23-5"></a>vegetation</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "grassland" "wetland"   "grassland" "forest"    "forest"    "grassland"
 [7] "forest"    "forest"    "forest"    "wetland"   "grassland" "wetland"  
[13] "grassland" "forest"    "wetland"   "forest"    "forest"    "grassland"
[19] "wetland"   "forest"    "grassland" "forest"    "wetland"   "forest"   
[25] "forest"    "forest"    "grassland" "forest"    "forest"    "forest"   
[31] "grassland" "grassland" "wetland"   "forest"    "wetland"   "wetland"  
[37] "grassland" "forest"    "wetland"   "forest"    "forest"    "forest"   
[43] "wetland"   "grassland" "forest"    "forest"    "forest"    "forest"   
[49] "forest"    "grassland"</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a><span class="fu">class</span>(vegetation)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "character"</code></pre>
</div>
</div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>The <code><a href="https://rdrr.io/r/base/sample.html">sample()</a></code> function
</div>
</div>
<div class="callout-body-container callout-body">
<p>Note that the <code><a href="https://rdrr.io/r/base/sample.html">sample()</a></code> function is not made specifically for nominal data; it can be used on any kind of data class.</p>
</div>
</div>
<p><strong>Ordinal data</strong> Here’s an example vector of ordinal data in R that could be encountered by ecologists:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a><span class="co"># Vector of ordinal data representing the successional stage of a forest</span></span>
<span id="cb27-2"><a href="#cb27-2"></a>succession <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Early Pioneer"</span>, <span class="st">"Late Pioneer"</span>,</span>
<span id="cb27-3"><a href="#cb27-3"></a>                <span class="st">"Young Forest"</span>, <span class="st">"Mature Forest"</span>,</span>
<span id="cb27-4"><a href="#cb27-4"></a>                <span class="st">"Old Growth"</span>)</span>
<span id="cb27-5"><a href="#cb27-5"></a>succession</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Early Pioneer" "Late Pioneer"  "Young Forest"  "Mature Forest"
[5] "Old Growth"   </code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb29"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a><span class="fu">class</span>(succession)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "character"</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb31"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a><span class="co"># Convert to ordered factor</span></span>
<span id="cb31-2"><a href="#cb31-2"></a>succession <span class="ot">&lt;-</span> <span class="fu">factor</span>(succession, <span class="at">ordered =</span> <span class="cn">TRUE</span>,</span>
<span id="cb31-3"><a href="#cb31-3"></a>                     <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Early Pioneer"</span>, <span class="st">"Late Pioneer"</span>,</span>
<span id="cb31-4"><a href="#cb31-4"></a>                                <span class="st">"Young Forest"</span>, <span class="st">"Mature Forest"</span>,</span>
<span id="cb31-5"><a href="#cb31-5"></a>                                <span class="st">"Old Growth"</span>))</span>
<span id="cb31-6"><a href="#cb31-6"></a>succession</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] Early Pioneer Late Pioneer  Young Forest  Mature Forest Old Growth   
5 Levels: Early Pioneer &lt; Late Pioneer &lt; Young Forest &lt; ... &lt; Old Growth</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb33"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a><span class="fu">class</span>(succession)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ordered" "factor" </code></pre>
</div>
</div>
<p>In this example, the successional stage of a forest is represented by an ordinal scale with five levels ranging from ‘Early Pioneer’ to ‘Old Growth’. The <code><a href="https://rdrr.io/r/base/factor.html">factor()</a></code> function is used to convert the vector to an ordered factor, with the <code>ordered</code> argument set to <code>TRUE</code> and the <code>levels</code> argument set to the same order as the original vector. This ensures that the levels are properly represented as an ordered factor.</p>
</section><section id="logical" class="level2"><h2 class="anchored" data-anchor-id="logical"><code>logical</code></h2>
<p>In R, the <code>logical</code> data class represents binary or Boolean data. Logical data are used to represent variables that can take on only two possible values, <code>TRUE</code> or <code>FALSE</code>. In addition to <code>TRUE</code> and <code>FALSE</code>, logical data can also take on the values of <code>NA</code> or <code>NULL</code>, which represent missing or undefined values.</p>
<p>Logical data can be created using logical operators such as <code>==</code>, <code>!=</code>, <code>&gt;</code>, <code>&lt;</code>, <code>&gt;=</code>, and <code>&lt;=</code>. Logical data are commonly used in R for data filtering and selection, conditional statements, and logical operations. For example, logical data can be used to filter a dataset to include only observations that meet certain criteria or to perform logical operations such as AND (<code>&amp;</code>) and OR (<code>|</code>). The logical data class in R is a fundamental data type that is critical for representing and working with binary or Boolean variables in data analysis and programming.</p>
<p><strong>Example logical (binary) data</strong> Here’s an example of generating a vector of binary or logical data in R, which represents the presence or absence of a particular species in different ecological sites:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb35"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a><span class="co"># Generate a vector of 1s and 0s to represent the presence</span></span>
<span id="cb35-2"><a href="#cb35-2"></a><span class="co"># or absence of a species in different ecological sites</span></span>
<span id="cb35-3"><a href="#cb35-3"></a>species_presence <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="dv">10</span>, <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb35-4"><a href="#cb35-4"></a>species_presence</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] 0 0 0 1 1 1 1 1 0 0</code></pre>
</div>
</div>
<p>We can also make a formal logical class data:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb37"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>species_presence_logi <span class="ot">&lt;-</span> <span class="fu">as.logical</span>(species_presence)</span>
<span id="cb37-2"><a href="#cb37-2"></a><span class="fu">class</span>(species_presence_logi)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "logical"</code></pre>
</div>
</div>
<p>In this example, we again use the <code><a href="https://rdrr.io/r/base/sample.html">sample()</a></code> function to randomly generate a vector of 10 values, each either 0 or 1, to represent the presence or absence of a species in 10 different ecological sites. However, it is often not necessary to coerce to class <code>logical</code>, as we see in the presence-absence datasets we will encounter in <a href="../../BCB743/BCB743_index.html">BCB743: Quantitative Ecology</a>.</p>
</section><section id="date" class="level2"><h2 class="anchored" data-anchor-id="date"><code>date</code></h2>
<p>In R, the <code>POSIXct</code>, <code>POSIXlt</code>, and <code>Date</code> classes are commonly used to represent date and time data. These classes each have unique characteristics that make them useful for different purposes.</p>
<p>The <code>POSIXct</code> class is a date/time class that represents dates and times as a numerical value, <em>typically</em> measured in seconds since January 1st, 1970. This class provides a high level of precision, with values accurate to the second. It is useful for performing calculations and data manipulation involving time, such as finding the difference between two dates or adding a certain number of seconds to a given time. An example of how to generate a <code>POSIXct</code> object in R is as follows:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb39"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a>my_time <span class="ot">&lt;-</span> <span class="fu">as.POSIXct</span>(<span class="st">"2022-03-10 12:34:56"</span>)</span>
<span id="cb39-2"><a href="#cb39-2"></a><span class="fu">class</span>(my_time)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "POSIXct" "POSIXt" </code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb41"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1"></a>my_time</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "2022-03-10 12:34:56 SAST"</code></pre>
</div>
</div>
<p>The <code>POSIXlt</code> class, on the other hand, <em>typically</em> represents dates and times in a more human-readable format. It stores date and time information as a list of named elements, including <code>year</code>, <code>month</code>, <code>day</code>, <code>hour</code>, <code>minute</code>, and <code>second.</code> This format is useful for displaying data in a more understandable way and for extracting specific components of a date or time. An example of how to generate a POSIXlt object in R is as follows:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb43"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1"></a>my_time <span class="ot">&lt;-</span> <span class="fu">as.POSIXlt</span>(<span class="st">"2022-03-10 12:34:56"</span>)</span>
<span id="cb43-2"><a href="#cb43-2"></a><span class="fu">class</span>(my_time)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "POSIXlt" "POSIXt" </code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb45"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1"></a>my_time</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "2022-03-10 12:34:56 SAST"</code></pre>
</div>
</div>
<p>The <code>Date</code> class is used to represent dates only, without any time information. Dates are <em>typically</em> stored as the number of days since January 1st, 1970. This class provides functions for performing arithmetic operations and comparisons between dates. It is useful for working with time-based data that is only concerned with the date component, such as daily sales or stock prices. An example of how to generate a Date object in R is as follows:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb47"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1"></a>my_date <span class="ot">&lt;-</span> <span class="fu">as.Date</span>(<span class="st">"2022-03-10"</span>)</span>
<span id="cb47-2"><a href="#cb47-2"></a><span class="fu">class</span>(my_date)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Date"</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb49"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1"></a>my_date</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "2022-03-10"</code></pre>
</div>
</div>
<p>To generate a vector of dates in R with daily intervals, we can use the <code><a href="https://rdrr.io/r/base/seq.html">seq()</a></code> function to create a sequence of dates, specifying the start and end dates and the time interval. Here’s an example:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb51"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1"></a><span class="co"># Generate a vector of dates from January 1, 2022 to December 31, 2022</span></span>
<span id="cb51-2"><a href="#cb51-2"></a>dates <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fu">as.Date</span>(<span class="st">"2022-01-01"</span>), <span class="fu">as.Date</span>(<span class="st">"2022-12-31"</span>), <span class="at">by =</span> <span class="st">"day"</span>)</span>
<span id="cb51-3"><a href="#cb51-3"></a></span>
<span id="cb51-4"><a href="#cb51-4"></a><span class="co"># View the first 10 dates in the vector</span></span>
<span id="cb51-5"><a href="#cb51-5"></a><span class="fu">head</span>(dates, <span class="dv">10</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "2022-01-01" "2022-01-02" "2022-01-03" "2022-01-04" "2022-01-05"
 [6] "2022-01-06" "2022-01-07" "2022-01-08" "2022-01-09" "2022-01-10"</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb53"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1"></a><span class="fu">class</span>(dates)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Date"</code></pre>
</div>
</div>
<p>Understanding the characteristics of these date and time classes in R is essential for effective data analysis and manipulation in fields where time-based data is a critical component.</p>
<p>Date and time data in R can be manipulated using various built-in functions and packages such as <strong>lubridate</strong> and <strong>chron</strong>. Additionally, date and time data can be visualised using different types of graphs such as time series plots, heatmaps, and Hovmöller diagrams. The date and time data classes in R are essential for working with temporal data and conducting time-related analyses in various biological and environmental datasets.</p>
</section><section id="missing-values-na" class="level2"><h2 class="anchored" data-anchor-id="missing-values-na">Missing values, <code>NA</code>
</h2>
<p>Missing values can be encountered in vectors of all data classes. To demonstrate some data that contains missing values, I will generate a data sequence containing 5% missing values. We can use the <code><a href="https://rdrr.io/r/stats/Normal.html">rnorm()</a></code> function to generate a sequence of random normal numbers and then randomly assign 5% of the values as missing using the <code><a href="https://rdrr.io/r/base/sample.html">sample()</a></code> function. The indices of the missing values are stored in missing_indices, and we use them to assign <code>NA</code> to the corresponding elements of the data sequence. Here’s some code to achieve this:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb55"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1"></a><span class="co"># Set the length of the sequence</span></span>
<span id="cb55-2"><a href="#cb55-2"></a>n <span class="ot">&lt;-</span> <span class="dv">100</span></span>
<span id="cb55-3"><a href="#cb55-3"></a></span>
<span id="cb55-4"><a href="#cb55-4"></a><span class="co"># Generate a sequence of random normal numbers with</span></span>
<span id="cb55-5"><a href="#cb55-5"></a><span class="co"># mean 0 and standard deviation 1</span></span>
<span id="cb55-6"><a href="#cb55-6"></a>data <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n, <span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>)</span>
<span id="cb55-7"><a href="#cb55-7"></a></span>
<span id="cb55-8"><a href="#cb55-8"></a><span class="co"># Randomly assign 5% of the values as missing</span></span>
<span id="cb55-9"><a href="#cb55-9"></a>missing_indices <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span>n, <span class="at">size =</span> <span class="fu">round</span>(<span class="fl">0.05</span><span class="sc">*</span>n))</span>
<span id="cb55-10"><a href="#cb55-10"></a>data[missing_indices] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb55-11"><a href="#cb55-11"></a><span class="fu">length</span>(data)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 100</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb57"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1"></a>data</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>  [1] -0.43893229  0.01974387 -1.33186453 -0.00604415 -0.27340465 -1.96578150
  [7]  0.80519266 -0.26910291 -1.18716061  0.14724895 -2.06307132  0.26743487
 [13]  1.24799667  0.94861069  0.34213769  1.00450813 -0.96998471 -1.08868347
 [19]  0.46311791 -0.85428187 -0.17499040  1.16796146  0.57246644  0.30265309
 [25] -1.29163397  0.58347529  0.84277752 -1.79853900  0.89689587  2.41061414
 [31]  0.94718411          NA  0.56382487  1.76054848 -1.22831988  2.06285169
 [37] -0.25518641  0.75181897  0.75635293 -2.05972401  2.51167272  0.79864421
 [43]          NA -0.13677150 -0.78365331          NA  0.47501091  0.57645090
 [49]  0.42552069  0.26871437 -0.31719160 -0.43283729  0.14214261  1.55236447
 [55] -0.50016935  1.73194340  1.14015057 -0.99746387  0.93999175  0.82959245
 [61] -0.79652555 -0.73916175 -1.04763073  1.46964595  0.12744893 -1.27652921
 [67]  0.04616807  0.59675766  0.58859599 -0.52205117 -0.01057440  0.14606629
 [73] -0.25300451  0.10965145          NA -1.28487686  1.81380033  0.87177567
 [79]  1.61475708  0.42301453  1.24437290  0.13025747  1.05024325 -0.31187800
 [85] -1.59583920  1.51996711 -1.48988572 -0.05826837 -0.50812472          NA
 [91]  0.51904718  0.42054858 -0.18533010  0.80120952  1.74426119  0.25357084
 [97]  0.64195957  2.29072398 -0.80525184  0.19655255</code></pre>
</div>
</div>
<p>To remove all <code>NA</code>s from the vector of data we can use <code><a href="https://rdrr.io/r/stats/na.fail.html">na.omit()</a></code>:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb59"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1"></a>data_sans_na <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(data)</span>
<span id="cb59-2"><a href="#cb59-2"></a><span class="fu">length</span>(data_sans_na)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 95</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb61"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1"></a>data_sans_na</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] -0.43893229  0.01974387 -1.33186453 -0.00604415 -0.27340465 -1.96578150
 [7]  0.80519266 -0.26910291 -1.18716061  0.14724895 -2.06307132  0.26743487
[13]  1.24799667  0.94861069  0.34213769  1.00450813 -0.96998471 -1.08868347
[19]  0.46311791 -0.85428187 -0.17499040  1.16796146  0.57246644  0.30265309
[25] -1.29163397  0.58347529  0.84277752 -1.79853900  0.89689587  2.41061414
[31]  0.94718411  0.56382487  1.76054848 -1.22831988  2.06285169 -0.25518641
[37]  0.75181897  0.75635293 -2.05972401  2.51167272  0.79864421 -0.13677150
[43] -0.78365331  0.47501091  0.57645090  0.42552069  0.26871437 -0.31719160
[49] -0.43283729  0.14214261  1.55236447 -0.50016935  1.73194340  1.14015057
[55] -0.99746387  0.93999175  0.82959245 -0.79652555 -0.73916175 -1.04763073
[61]  1.46964595  0.12744893 -1.27652921  0.04616807  0.59675766  0.58859599
[67] -0.52205117 -0.01057440  0.14606629 -0.25300451  0.10965145 -1.28487686
[73]  1.81380033  0.87177567  1.61475708  0.42301453  1.24437290  0.13025747
[79]  1.05024325 -0.31187800 -1.59583920  1.51996711 -1.48988572 -0.05826837
[85] -0.50812472  0.51904718  0.42054858 -0.18533010  0.80120952  1.74426119
[91]  0.25357084  0.64195957  2.29072398 -0.80525184  0.19655255
attr(,"na.action")
[1] 32 43 46 75 90
attr(,"class")
[1] "omit"</code></pre>
</div>
</div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Dealing with <code>NA</code>s in functions
</div>
</div>
<div class="callout-body-container callout-body">
<p>Many functions have specific arguments to deal with <code>NA</code>s in data. See for example the <code>na.rm = TRUE</code> argument given to <code><a href="https://rdrr.io/r/base/mean.html">mean()</a></code>, <code><a href="https://rdrr.io/r/stats/median.html">median()</a></code>, <code><a href="https://rdrr.io/r/base/Extremes.html">min()</a></code>, <code><a href="https://rdrr.io/r/stats/lm.html">lm()</a></code>, etc.</p>
</div>
</div>
</section></section><section id="data-structures-in-r" class="level1"><h1>Data structures in R</h1>
<p>Data structures can be viewed as the ‘containers’ that hold the different data classes. The kind of data structure can also be revealed by the <code><a href="https://rdrr.io/r/base/class.html">class()</a></code> command.</p>
<section id="vector-array-and-matrix" class="level2"><h2 class="anchored" data-anchor-id="vector-array-and-matrix">
<code>vector</code>, <code>array</code>, and <code>matrix</code>
</h2>
<p><strong>Vectors</strong> In R, a vector is a one-dimensional array-like data structure that can hold a sequence of values of the same atomic mode, such as <code>numeric</code>, <code>character</code>, <code>logical</code> values, or <code>Date</code> and times. A vector can be created using the <code><a href="https://rdrr.io/r/base/c.html">c()</a></code> function, which stands for ‘combine’ or ‘concatenate,’ and is used to combine a sequence of values into a vector. Vectors can also be created by using the <code><a href="https://rdrr.io/r/base/seq.html">seq()</a></code> function to generate a sequence of numbers, or the <code><a href="https://rdrr.io/r/base/rep.html">rep()</a></code> function to repeat a value or sequence of values. Here is an example of a numeric vector:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb63"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1"></a><span class="co"># create a numeric vector</span></span>
<span id="cb63-2"><a href="#cb63-2"></a>my_vector <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>)</span>
<span id="cb63-3"><a href="#cb63-3"></a></span>
<span id="cb63-4"><a href="#cb63-4"></a><span class="co"># coerce to vector</span></span>
<span id="cb63-5"><a href="#cb63-5"></a>my_vector <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>))</span>
<span id="cb63-6"><a href="#cb63-6"></a><span class="fu">class</span>(my_vector) <span class="co"># but it doesn't change the class from numeric</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "numeric"</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb65"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1"></a><span class="co"># print the vector</span></span>
<span id="cb65-2"><a href="#cb65-2"></a>my_vector</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1 2 3 4 5</code></pre>
</div>
</div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Coercion to vector
</div>
</div>
<div class="callout-body-container callout-body">
<p>The behaviour is such that the output of coercion to vector is that one the atomic modes (the basic data types) is returned.</p>
</div>
</div>
<p>One of the advantages of using vectors in R is that many of the built-in functions and operations work on vectors, allowing us to easily manipulate and analyse large amounts of data. Additionally, R provides many functions specifically designed for working with vectors, such as <code><a href="https://rdrr.io/r/base/mean.html">mean()</a></code>, <code><a href="https://rdrr.io/r/stats/median.html">median()</a></code>, <code><a href="https://rdrr.io/r/base/sum.html">sum()</a></code>, <code><a href="https://rdrr.io/r/base/Extremes.html">min()</a></code>, <code><a href="https://rdrr.io/r/base/Extremes.html">max()</a></code>, and many others.</p>
<p><strong>Matrices</strong> A matrix (again, this terminology may be different for other languages), on the other hand, is a special case of an array that has two dimensions (rows and columns). It is also a multi-dimensional data structure that can hold elements of the <em>same data type</em>, but it is specifically designed for handling data in a tabular format. A matrix can be created using the <code><a href="https://rdrr.io/r/base/matrix.html">matrix()</a></code> function in R.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb67"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1"></a><span class="co"># create a numeric matrix</span></span>
<span id="cb67-2"><a href="#cb67-2"></a>my_matrix <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb67-3"><a href="#cb67-3"></a></span>
<span id="cb67-4"><a href="#cb67-4"></a><span class="co"># print the matrix</span></span>
<span id="cb67-5"><a href="#cb67-5"></a>my_matrix</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>     [,1] [,2] [,3]
[1,]    1    3    5
[2,]    2    4    6</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb69"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1"></a><span class="fu">class</span>(my_matrix)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "matrix" "array" </code></pre>
</div>
</div>
<p>We can query the size or dimensions of the matrix as follows:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb71"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1"></a><span class="fu">dim</span>(my_matrix)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2 3</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb73"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1"></a><span class="fu">ncol</span>(my_matrix)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb75"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1"></a><span class="fu">nrow</span>(my_matrix)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2</code></pre>
</div>
</div>
<p><strong>Coercion of matrices to vectors</strong> A matrix can be coerced to a vector:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb77"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1"></a><span class="fu">as.vector</span>(my_matrix)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1 2 3 4 5 6</code></pre>
</div>
</div>
<p><strong>Arrays</strong> In R (as opposed to in python or some other languages), an array specifically refers to a multi-dimensional data structure that can hold elements of the <em>same data type</em>. It can have any number of dimensions (1, 2, 3, etc.), and its dimensions can be named. An array can be created using the <code><a href="https://rdrr.io/r/base/array.html">array()</a></code> function in R.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb79"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1"></a><span class="co"># create a 2-dimensional array</span></span>
<span id="cb79-2"><a href="#cb79-2"></a>my_array <span class="ot">&lt;-</span> <span class="fu">array</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">27</span>, <span class="at">dim =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">3</span>))</span>
<span id="cb79-3"><a href="#cb79-3"></a></span>
<span id="cb79-4"><a href="#cb79-4"></a><span class="co"># print the array</span></span>
<span id="cb79-5"><a href="#cb79-5"></a>my_array</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>, , 1

     [,1] [,2] [,3]
[1,]    1    4    7
[2,]    2    5    8
[3,]    3    6    9

, , 2

     [,1] [,2] [,3]
[1,]   10   13   16
[2,]   11   14   17
[3,]   12   15   18

, , 3

     [,1] [,2] [,3]
[1,]   19   22   25
[2,]   20   23   26
[3,]   21   24   27</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb81"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1"></a><span class="fu">class</span>(my_array)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "array"</code></pre>
</div>
</div>
<p>We can figure something out about the size or dimensions of the array:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb83"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1"></a><span class="fu">dim</span>(my_array)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3 3 3</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb85"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1"></a><span class="fu">ncol</span>(my_array)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb87"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1"></a><span class="fu">nrow</span>(my_array)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3</code></pre>
</div>
</div>
<p><strong>Coercion of arrays to vectors</strong> The array can be coerced to a vector:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb89"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1"></a><span class="fu">as.vector</span>(my_array)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1]  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25
[26] 26 27</code></pre>
</div>
</div>
<p>The key difference between vectors, arrays, and a matrices in R is their dimensions. A vector has one dimension, an array can have any number of dimensions, while a matrix is limited to two dimensions. Additionally, a matrix is often used to store data in a tabular format, while an array is used to store multi-dimensional data in general. A commonly encountered kind of matrix is seen in multivariate statistics is a distance or dissimilarity matrix.</p>
<p><em>In R, vectors, arrays, and matrices share a common characteristic: they do not have row or column names.</em> Therefore, to refer to any element, row, or column, one must use their corresponding index. How?</p>
<p><strong>Accessing elements, rows, columns, and matrices</strong> In R, the square bracket notation is used to access elements, rows, columns, or matrices in arrays. The notation takes the form of <code>[i, j, k, ...]</code>, where <code>i</code>, <code>j</code>, <code>k</code>, and so on, represent the indices of the rows, columns, or matrices to be accessed.</p>
<p>Suppose we have the following array:</p>
<!-- # my_array <- array(data = 1:60, dim = c(5, 4, 3)) -->
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb91"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1"></a>my_array <span class="ot">&lt;-</span> <span class="fu">array</span>(<span class="at">data =</span> <span class="fu">round</span>(<span class="fu">rnorm</span>(<span class="at">n =</span> <span class="dv">60</span>, <span class="at">mean =</span> <span class="dv">13</span>, <span class="at">sd =</span> <span class="dv">2</span>), <span class="dv">1</span>),</span>
<span id="cb91-2"><a href="#cb91-2"></a>                  <span class="at">dim =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">3</span>))</span>
<span id="cb91-3"><a href="#cb91-3"></a>my_array</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>, , 1

     [,1] [,2] [,3] [,4]
[1,] 11.0 12.1 12.2 13.3
[2,] 13.6 11.9 11.9 10.8
[3,] 10.3 13.1 11.7 10.9
[4,] 15.2 10.2 17.2 12.3
[5,] 14.8 11.8 13.7 12.1

, , 2

     [,1] [,2] [,3] [,4]
[1,] 11.3 14.5 14.7 13.8
[2,] 12.4 13.5 15.0 12.1
[3,] 14.8  9.9 13.3 12.1
[4,] 14.4 15.7 14.8 12.3
[5,] 12.7 10.0 14.3  7.3

, , 3

     [,1] [,2] [,3] [,4]
[1,]  8.9 15.1 14.2 12.3
[2,] 15.4 13.1 12.7 12.0
[3,] 15.0 11.8 11.2 13.1
[4,] 13.5 10.7 12.0 15.5
[5,] 15.4 12.7 11.8 10.3</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb93"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1"></a><span class="fu">dim</span>(my_array)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 5 4 3</code></pre>
</div>
</div>
<p>This creates a <span class="math inline">\(5\times4\times3\)</span> array with values from <code>1</code> to <code>60</code>.</p>
<p>When working with multidimensional arrays, it is possible to omit some of the indices in the square bracket notation. This results in a subset of the array, which can be thought of as a lower-dimensional array obtained by fixing the omitted dimensions. For example, consider a 3-dimensional array <code>my_array</code> above with dimensions <code>dim(my_array) = c(5,4,3)</code>. If we use the notation <code>my_array[1,,]</code>, we would obtain a 2-dimensional array with dimensions <code>dim(my_array[1,,]) = c(4,3)</code> obtained by fixing the first index at 1:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb95"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1"></a><span class="fu">dim</span>(my_array[<span class="dv">1</span>,,])</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 4 3</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb97"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1"></a>my_array[<span class="dv">1</span>,,]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>     [,1] [,2] [,3]
[1,] 11.0 11.3  8.9
[2,] 12.1 14.5 15.1
[3,] 12.2 14.7 14.2
[4,] 13.3 13.8 12.3</code></pre>
</div>
</div>
<p>Here are some more examples of how to use square brackets notation with arrays in R:</p>
<p>To access a <strong>single element</strong> in the array, use the notation <code>[i, j, k]</code>, where <code>i</code>, <code>j</code>, and <code>k</code> are the indices along each of the three dimensions, which in combination, uniquely identifies each element. Below we return the element in the second row, third column, and first matrix:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb99"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1"></a>my_array[<span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">1</span>]  </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 11.9</code></pre>
</div>
</div>
<p>To access a single row in the array, use the notation <code>[i, , ]</code>, where <code>i</code> is the index of the row. This will return the second rows and all of the columns of the first matrix:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb101"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1"></a>my_array[<span class="dv">2</span>,,<span class="dv">1</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 13.6 11.9 11.9 10.8</code></pre>
</div>
</div>
<p>To access a single column in the array, use the notation <code>[ , j, ]</code>, where <code>j</code> is the index of the column. Here we will return all the elements in the row of column two and matrix three:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb103"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1"></a>my_array[ , <span class="dv">2</span>, <span class="dv">3</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 15.1 13.1 11.8 10.7 12.7</code></pre>
</div>
</div>
<p>To access a single matrix in the array, use the notation <code>[ , , k]</code>, where <code>k</code> is the index of the matrix:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb105"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1"></a>my_array[ , , <span class="dv">2</span>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>     [,1] [,2] [,3] [,4]
[1,] 11.3 14.5 14.7 13.8
[2,] 12.4 13.5 15.0 12.1
[3,] 14.8  9.9 13.3 12.1
[4,] 14.4 15.7 14.8 12.3
[5,] 12.7 10.0 14.3  7.3</code></pre>
</div>
</div>
<p>To obtain a subset of the array, use the notation <code>[i, j, k]</code> with <code>i</code>, <code>j</code>, or <code>k</code> omitted to obtain a lower-dimensional array:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb107"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1"></a>my_array[<span class="dv">1</span>, , ]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>     [,1] [,2] [,3]
[1,] 11.0 11.3  8.9
[2,] 12.1 14.5 15.1
[3,] 12.2 14.7 14.2
[4,] 13.3 13.8 12.3</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb109"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1"></a>my_array[ , <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>, ]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>, , 1

     [,1] [,2]
[1,] 12.1 12.2
[2,] 11.9 11.9
[3,] 13.1 11.7
[4,] 10.2 17.2
[5,] 11.8 13.7

, , 2

     [,1] [,2]
[1,] 14.5 14.7
[2,] 13.5 15.0
[3,]  9.9 13.3
[4,] 15.7 14.8
[5,] 10.0 14.3

, , 3

     [,1] [,2]
[1,] 15.1 14.2
[2,] 13.1 12.7
[3,] 11.8 11.2
[4,] 10.7 12.0
[5,] 12.7 11.8</code></pre>
</div>
</div>
</section><section id="data.frame" class="level2"><h2 class="anchored" data-anchor-id="data.frame"><code>data.frame</code></h2>
<p>A dataframe is perhaps the most commonly-used ‘container’ for data in R because they are so convenient and serve many purposes. A dataframe is <em>not a data class</em>—more correctly, it is a form of tabular data (like a table in MS Excel), with each vector (a variable or column) comprising the table sharing the same length. What makes a dataframe versatile is that its variables can be any combination of the atomic data types. It may even include <em>list columns</em> (we will not cover list columns in this module). Applying the <code><a href="https://rdrr.io/r/base/class.html">class()</a></code> function to a dataframe shows that it blongs to class <code>data.frame</code>.</p>
<p>Here’s an example of an R <code>data.frame</code> with <code>Date</code>, <code>numeric</code>, and <code>categorical</code> data classes:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb111"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1"></a><span class="co"># Create a vector of dates</span></span>
<span id="cb111-2"><a href="#cb111-2"></a>dates <span class="ot">&lt;-</span> <span class="fu">as.Date</span>(<span class="fu">c</span>(<span class="st">"2022-01-01"</span>, <span class="st">"2022-01-02"</span>, <span class="st">"2022-01-03"</span>,</span>
<span id="cb111-3"><a href="#cb111-3"></a>                   <span class="st">"2022-01-04"</span>, <span class="st">"2022-01-05"</span>))</span>
<span id="cb111-4"><a href="#cb111-4"></a></span>
<span id="cb111-5"><a href="#cb111-5"></a><span class="co"># Create a vector of numeric data</span></span>
<span id="cb111-6"><a href="#cb111-6"></a>numeric_data <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="at">n =</span> <span class="dv">5</span>, <span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>)</span>
<span id="cb111-7"><a href="#cb111-7"></a></span>
<span id="cb111-8"><a href="#cb111-8"></a><span class="co"># Create a vector of categorical data</span></span>
<span id="cb111-9"><a href="#cb111-9"></a>categorical_data <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"C"</span>, <span class="st">"A"</span>, <span class="st">"B"</span>)</span>
<span id="cb111-10"><a href="#cb111-10"></a></span>
<span id="cb111-11"><a href="#cb111-11"></a><span class="co"># Combine the vectors into a data.frame</span></span>
<span id="cb111-12"><a href="#cb111-12"></a>my_dataframe <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">dates =</span> dates,</span>
<span id="cb111-13"><a href="#cb111-13"></a>                           <span class="at">numeric_data =</span> numeric_data,</span>
<span id="cb111-14"><a href="#cb111-14"></a>                           <span class="at">categorical_data =</span> categorical_data)</span>
<span id="cb111-15"><a href="#cb111-15"></a></span>
<span id="cb111-16"><a href="#cb111-16"></a><span class="co"># Print the dataframe</span></span>
<span id="cb111-17"><a href="#cb111-17"></a>my_dataframe</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>       dates numeric_data categorical_data
1 2022-01-01  -0.14354983                A
2 2022-01-02   0.47971805                B
3 2022-01-03  -0.59996086                C
4 2022-01-04   0.06288558                A
5 2022-01-05  -1.01865179                B</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb113"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1"></a><span class="fu">class</span>(my_dataframe)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "data.frame"</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb115"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1"></a><span class="fu">str</span>(my_dataframe)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>'data.frame':   5 obs. of  3 variables:
 $ dates           : Date, format: "2022-01-01" "2022-01-02" ...
 $ numeric_data    : num  -0.1435 0.4797 -0.6 0.0629 -1.0187
 $ categorical_data: chr  "A" "B" "C" "A" ...</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb117"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1"></a><span class="fu">summary</span>(my_dataframe)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>     dates             numeric_data      categorical_data  
 Min.   :2022-01-01   Min.   :-1.01865   Length:5          
 1st Qu.:2022-01-02   1st Qu.:-0.59996   Class :character  
 Median :2022-01-03   Median :-0.14355   Mode  :character  
 Mean   :2022-01-03   Mean   :-0.24391                     
 3rd Qu.:2022-01-04   3rd Qu.: 0.06289                     
 Max.   :2022-01-05   Max.   : 0.47972                     </code></pre>
</div>
</div>
<p>Dataframes may also have row names:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb119"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1"></a><span class="fu">rownames</span>(my_dataframe) <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">rep</span>(<span class="st">"row"</span>, <span class="dv">5</span>), <span class="at">seq =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)</span>
<span id="cb119-2"><a href="#cb119-2"></a>my_dataframe</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>           dates numeric_data categorical_data
row 1 2022-01-01  -0.14354983                A
row 2 2022-01-02   0.47971805                B
row 3 2022-01-03  -0.59996086                C
row 4 2022-01-04   0.06288558                A
row 5 2022-01-05  -1.01865179                B</code></pre>
</div>
</div>
<p>Typically we will create a dataframe by reading in data from a <code>.csv</code> file, but it is useful to be able to construct one from scratch.</p>
</section><section id="tibble" class="level2"><h2 class="anchored" data-anchor-id="tibble"><code>tibble</code></h2>
<p>In R, a dataframe and a tibble are both data structures used to store tabular data. Although tibbles are also dataframes, but they differ subtly in several ways.</p>
<ul>
<li><p>A tibble is a relatively new addition to the R language and forms part of the <strong>tidyverse</strong> suite of packages. They are designed to be more user-friendly than traditional data frames and have several additional features, such as more informative error messages, stricter data input and output rules, and better handling of <code>NA</code>.</p></li>
<li><p>Unlike a dataframe, a tibble never automatically converts strings to factors or changes column names, which can help avoid unexpected behavior when working with the data.</p></li>
<li><p>A tibble does not have row names.</p></li>
<li><p>A tibble has a slightly different and more compact printing method than a dataframe, which makes them easier to read and work with.</p></li>
<li><p>Finally, a tibble has better performance than dataframes for many tasks, especially when working with large datasets.</p></li>
</ul>
<p>While a dataframe is a core data structure in R, a tibble provides additional functionality and are becoming increasingly popular among R users, particularly those working with <strong>tidyverse</strong> packages. Applying the <code><a href="https://rdrr.io/r/base/class.html">class()</a></code> function to a tibble revelas that it belongs to the classes <code>tbl_df</code>, <code>tbl</code> and <code>data.frame</code>.</p>
<p>We can convert our dataframe <code>my_dataframe</code> to a tibble, and present the output with the <code><a href="https://rdrr.io/r/base/print.html">print()</a></code> function that applies nicely to tibbles:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb121"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1"></a><span class="fu">library</span>(tidyverse) <span class="co"># we need to load the tidyverse package</span></span>
<span id="cb121-2"><a href="#cb121-2"></a>my_tibble <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(my_dataframe)</span>
<span id="cb121-3"><a href="#cb121-3"></a><span class="fu">class</span>(my_tibble)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "tbl_df"     "tbl"        "data.frame"</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb123"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1"></a><span class="fu">print</span>(my_tibble)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 5 × 3
  dates      numeric_data categorical_data
  &lt;date&gt;            &lt;dbl&gt; &lt;chr&gt;           
1 2022-01-01      -0.144  A               
2 2022-01-02       0.480  B               
3 2022-01-03      -0.600  C               
4 2022-01-04       0.0629 A               
5 2022-01-05      -1.02   B               </code></pre>
</div>
</div>
<p>This very simple tibble looks identical to a dataframe, but as we start using more complex sets of data you’ll learn to appreciate the small convenience that tibbles offer.</p>
</section><section id="list" class="level2"><h2 class="anchored" data-anchor-id="list"><code>list</code></h2>
<p>This is also not actually a data class, but rather another way of representing a collection of objects of different types, all the way from numerical vectors to dataframes. Lists are useful for storing complex data structures and can also be accessed using indexing.</p>
<p>As an example, we create another dataframe:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb125"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1"></a>dates <span class="ot">&lt;-</span> <span class="fu">as.Date</span>(<span class="fu">c</span>(<span class="st">"2022-01-01"</span>, <span class="st">"2022-01-02"</span>, <span class="st">"2022-01-03"</span>,</span>
<span id="cb125-2"><a href="#cb125-2"></a>                   <span class="st">"2022-01-04"</span>, <span class="st">"2022-01-05"</span>))</span>
<span id="cb125-3"><a href="#cb125-3"></a></span>
<span id="cb125-4"><a href="#cb125-4"></a><span class="co"># Create a vector of numeric data</span></span>
<span id="cb125-5"><a href="#cb125-5"></a>numeric_data <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="at">n =</span> <span class="dv">5</span>, <span class="at">mean =</span> <span class="dv">1</span>, <span class="at">sd =</span> <span class="dv">1</span>)</span>
<span id="cb125-6"><a href="#cb125-6"></a></span>
<span id="cb125-7"><a href="#cb125-7"></a><span class="co"># Create a vector of categorical data</span></span>
<span id="cb125-8"><a href="#cb125-8"></a>categorical_data <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"C"</span>, <span class="st">"D"</span>, <span class="st">"D"</span>, <span class="st">"F"</span>, <span class="st">"A"</span>)</span>
<span id="cb125-9"><a href="#cb125-9"></a></span>
<span id="cb125-10"><a href="#cb125-10"></a><span class="co"># Combine the vectors into a data.frame</span></span>
<span id="cb125-11"><a href="#cb125-11"></a>my_other_dataframe <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">dates =</span> dates,</span>
<span id="cb125-12"><a href="#cb125-12"></a>                                  <span class="at">numeric_data =</span> numeric_data,</span>
<span id="cb125-13"><a href="#cb125-13"></a>                                  <span class="at">categorical_data =</span> categorical_data)</span>
<span id="cb125-14"><a href="#cb125-14"></a></span>
<span id="cb125-15"><a href="#cb125-15"></a>my_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">A =</span> my_dataframe,</span>
<span id="cb125-16"><a href="#cb125-16"></a>                <span class="at">B =</span> my_other_dataframe)</span>
<span id="cb125-17"><a href="#cb125-17"></a>my_list</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>$A
           dates numeric_data categorical_data
row 1 2022-01-01  -0.14354983                A
row 2 2022-01-02   0.47971805                B
row 3 2022-01-03  -0.59996086                C
row 4 2022-01-04   0.06288558                A
row 5 2022-01-05  -1.01865179                B

$B
       dates numeric_data categorical_data
1 2022-01-01    3.4982586                C
2 2022-01-02    0.1419428                D
3 2022-01-03    0.5939201                D
4 2022-01-04    1.2093531                F
5 2022-01-05    1.5751606                A</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb127"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1"></a><span class="fu">class</span>(my_list)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "list"</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb129"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1"></a><span class="fu">str</span>(my_list)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>List of 2
 $ A:'data.frame':  5 obs. of  3 variables:
  ..$ dates           : Date[1:5], format: "2022-01-01" "2022-01-02" ...
  ..$ numeric_data    : num [1:5] -0.1435 0.4797 -0.6 0.0629 -1.0187
  ..$ categorical_data: chr [1:5] "A" "B" "C" "A" ...
 $ B:'data.frame':  5 obs. of  3 variables:
  ..$ dates           : Date[1:5], format: "2022-01-01" "2022-01-02" ...
  ..$ numeric_data    : num [1:5] 3.498 0.142 0.594 1.209 1.575
  ..$ categorical_data: chr [1:5] "C" "D" "D" "F" ...</code></pre>
</div>
</div>
<p>We can access one of the dataframes is the list as follows:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb131"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1"></a>my_list[[<span class="dv">2</span>]]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>       dates numeric_data categorical_data
1 2022-01-01    3.4982586                C
2 2022-01-02    0.1419428                D
3 2022-01-03    0.5939201                D
4 2022-01-04    1.2093531                F
5 2022-01-05    1.5751606                A</code></pre>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb133"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1"></a>my_list[[<span class="st">"A"</span>]]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>           dates numeric_data categorical_data
row 1 2022-01-01  -0.14354983                A
row 2 2022-01-02   0.47971805                B
row 3 2022-01-03  -0.59996086                C
row 4 2022-01-04   0.06288558                A
row 5 2022-01-05  -1.01865179                B</code></pre>
</div>
</div>
<p>To access a variable within one of the elements of the list we can do something like:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb135"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1"></a>my_list[[<span class="st">"B"</span>]]<span class="sc">$</span>numeric_data</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3.4982586 0.1419428 0.5939201 1.2093531 1.5751606</code></pre>
</div>
</div>
</section></section><section id="conclusion" class="level1"><h1>Conclusion</h1>
<p>These are the bread and butter data classes and structures in R. Other data classes and structures also exist, but these may be particular to certain packages. We will encounter some of them in <a href="../../BCB743/BCB743_index.html">BCB743: Quantitative Ecology</a>.</p>


<!-- -->

</section><div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-reuse"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a></div></div></section><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{smit,_a._j.2021,
  author = {Smit, A. J.,},
  title = {3. {Data} {Classes} \&amp; {Structures}},
  date = {2021-01-01},
  url = {http://tangledbank.netlify.app/BCB744/intro_r/03-data-in-R.html},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-smit,_a._j.2021" class="csl-entry quarto-appendix-citeas" role="listitem">
Smit, A. J. (2021) 3. Data Classes &amp; Structures. <a href="http://tangledbank.netlify.app/BCB744/intro_r/03-data-in-R.html">http://tangledbank.netlify.app/BCB744/intro_r/03-data-in-R.html.
</a>
</div></div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("http:\/\/tangledbank\.netlify\.app");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
            // default icon
            link.classList.add("external");
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="../../BCB744/intro_r/02-working-with-data.html" class="pagination-link" aria-label="2. Working With Data &amp; Code">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">2. Working With Data &amp; Code</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../BCB744/intro_r/04-workflow.html" class="pagination-link" aria-label="4. R Workflows">
        <span class="nav-page-text">4. R Workflows</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb137" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb137-1"><a href="#cb137-1"></a><span class="co">---</span></span>
<span id="cb137-2"><a href="#cb137-2"></a><span class="an">date:</span><span class="co"> "2021-01-01"</span></span>
<span id="cb137-3"><a href="#cb137-3"></a><span class="an">title:</span><span class="co"> "3. Data Classes &amp; Structures"</span></span>
<span id="cb137-4"><a href="#cb137-4"></a><span class="an">subtitle:</span><span class="co"> "Getting familiar with data classes and structures"</span></span>
<span id="cb137-5"><a href="#cb137-5"></a><span class="co">---</span></span>
<span id="cb137-6"><a href="#cb137-6"></a></span>
<span id="cb137-7"><a href="#cb137-7"></a><span class="at">&gt; "*That which can be destroyed by the truth should be.*"</span></span>
<span id="cb137-8"><a href="#cb137-8"></a><span class="at">&gt;</span></span>
<span id="cb137-9"><a href="#cb137-9"></a><span class="at">&gt; --- P.C. Hodgell</span></span>
<span id="cb137-10"><a href="#cb137-10"></a></span>
<span id="cb137-11"><a href="#cb137-11"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb137-12"><a href="#cb137-12"></a><span class="fu">## In this Chapter</span></span>
<span id="cb137-13"><a href="#cb137-13"></a></span>
<span id="cb137-14"><a href="#cb137-14"></a><span class="ss">* </span>Data as encountered by the statistician</span>
<span id="cb137-15"><a href="#cb137-15"></a><span class="ss">* </span>Data classes in R</span>
<span id="cb137-16"><a href="#cb137-16"></a><span class="ss">* </span>Data structures in R</span>
<span id="cb137-17"><a href="#cb137-17"></a>:::</span>
<span id="cb137-18"><a href="#cb137-18"></a></span>
<span id="cb137-19"><a href="#cb137-19"></a>::: {.callout-important appearance="simple"}</span>
<span id="cb137-20"><a href="#cb137-20"></a><span class="fu">## Tasks to complete in this Chapter</span></span>
<span id="cb137-21"><a href="#cb137-21"></a></span>
<span id="cb137-22"><a href="#cb137-22"></a><span class="ss">* </span>Task A 1-4</span>
<span id="cb137-23"><a href="#cb137-23"></a>:::</span>
<span id="cb137-24"><a href="#cb137-24"></a></span>
<span id="cb137-25"><a href="#cb137-25"></a><span class="al">![](/images/769_life_finds_a_way.png)</span>{fig-align="center" width="300"}</span>
<span id="cb137-26"><a href="#cb137-26"></a></span>
<span id="cb137-27"><a href="#cb137-27"></a><span class="fu"># Introduction</span></span>
<span id="cb137-28"><a href="#cb137-28"></a></span>
<span id="cb137-29"><a href="#cb137-29"></a>In biostatistics we will encounter different data types, and comprehending **data classes**, **data structures**, and their **statistical interpretations** is important for several reasons:</span>
<span id="cb137-30"><a href="#cb137-30"></a></span>
<span id="cb137-31"><a href="#cb137-31"></a><span class="ss">-   </span>**Correctly identifying the data class** To manipulate, analyse, or visualise data effectively, we must first identify the type of data we are working with. Since different data classes possess distinct properties, we must determine the class of our data to leverage the appropriate functions and operations.</span>
<span id="cb137-32"><a href="#cb137-32"></a></span>
<span id="cb137-33"><a href="#cb137-33"></a><span class="ss">-   </span>**Efficient data processing** As biologists using R, we may come across data structures that are not fully compatible with the available functions. For example, when dealing with extensive datasets, employing data structures intended for vectorised operations like arrays or matrices can significantly boost data processing speed. In contrast, other data representations may not perform well when managing large datasets. By becoming familiar with the different data classes and their attributes in R, we can make informed choices about which data structures and functions to use to accomplish their data processing and analysis tasks effectively.</span>
<span id="cb137-34"><a href="#cb137-34"></a></span>
<span id="cb137-35"><a href="#cb137-35"></a><span class="ss">-   </span>**Data manipulation and analysis** Different data classes in R have specific methods for manipulating and analysing data. For example, if we're working with character strings, we can use string manipulation functions like <span class="in">`gsub()`</span> or <span class="in">`strsplit()`</span> which do not work on numerical data. If we're working with dates and times, we need to use date and time functions like <span class="in">`as.Date()`</span> or <span class="in">`lubridate::ymd()`</span>. By understanding the different data classes in R, we can choose the appropriate functions for manipulating and analysing our data.</span>
<span id="cb137-36"><a href="#cb137-36"></a></span>
<span id="cb137-37"><a href="#cb137-37"></a><span class="ss">-   </span>**Data visualisation** Data can be visualised in numerous ways, such as using histograms for numeric data, bar graphs for categorical data, or scatter plots for representing the relationship between two numeric variables. However, the way we provide data to R is critical. By familiarising ourselves with the diverse data classes in R, we can select the most suitable visualisation methods for our data.</span>
<span id="cb137-38"><a href="#cb137-38"></a></span>
<span id="cb137-39"><a href="#cb137-39"></a>Let us delve into how biologists classify data and the distinct terminology they employ to describe it. It's essential to comprehend these subtleties before we investigate how R recognises data classes. We'll scrutinise the data types that are ubiquitous in the fields of biology and ecology, and subsequently shift our attention to data classes and structures in R.</span>
<span id="cb137-40"><a href="#cb137-40"></a></span>
<span id="cb137-41"><a href="#cb137-41"></a><span class="fu"># Types of variables common in biology and ecology</span></span>
<span id="cb137-42"><a href="#cb137-42"></a></span>
<span id="cb137-43"><a href="#cb137-43"></a>We will most frequently encounter data arranged in columns of a data file---typically in MS Excel files or CSV files. A column is a variable, and one variable is comprised of one data type. So, when we refer to a variable, we expect that all the data within would be homogeneous---at least in as far as the data's type.</span>
<span id="cb137-44"><a href="#cb137-44"></a></span>
<span id="cb137-45"><a href="#cb137-45"></a>The type of data that biologists and statisticians work with can influence the statistical techniques and methods they use to analyse and interpret the data. Let us discuss some of the different types of biological and ecological data we are likely to encounter.</span>
<span id="cb137-46"><a href="#cb137-46"></a></span>
<span id="cb137-47"><a href="#cb137-47"></a><span class="fu">## Numeric variables</span></span>
<span id="cb137-48"><a href="#cb137-48"></a></span>
<span id="cb137-49"><a href="#cb137-49"></a>Numeric data in the context of biostatistics refers to **quantitative data** that can be expressed in numerical form, typically obtained from field and laboratory measurements, or from field sampling campaigns. Examples of numeric data in biostatistics include the height and mass of a animals, concentrations of nutrients, laboratory test results such as respiration rates, or the number of limpets in a quadrat. Numeric data can be further categorised as **discrete** or **continuous**.</span>
<span id="cb137-50"><a href="#cb137-50"></a></span>
<span id="cb137-51"><a href="#cb137-51"></a><span class="fu">### Discrete variables</span></span>
<span id="cb137-52"><a href="#cb137-52"></a></span>
<span id="cb137-53"><a href="#cb137-53"></a>Discrete data are whole (integer) numbers that represent counts of items or events. Integer data usually answer the question, "how many?" For example, in the biological and Earth sciences, discrete data are commonly encountered in the form of counts or integers that represent the presence or absence of certain characteristics or events. For example, the number of individuals of some species in a population, the number of chromosomes in a cell, or the number of earthquakes occurring in a region within a given time frame. Other examples of discrete data in these sciences include the number of mutations in a gene, the number of cells in a tissue sample, or the number of species present in an ecosystem. These types of data are often analysed using statistical techniques such as frequency distributions, contingency tables, and chi-square tests.</span>
<span id="cb137-54"><a href="#cb137-54"></a></span>
<span id="cb137-55"><a href="#cb137-55"></a><span class="fu">### Continuous variables</span></span>
<span id="cb137-56"><a href="#cb137-56"></a></span>
<span id="cb137-57"><a href="#cb137-57"></a>Continuous data, on the other hand, are measured on a continuous scale. These usually represent measured quantities such as something's heat content (temperature, measured in degrees Celsius) or distance (measured in metres or similar), etc. They can be rational numbers including integers and fractions, but typically they have an infinite number of 'steps' that depend on rounding (they can even be rounded to whole integers) or considerations such as measurement precision and accuracy. Often, continuous data have upper and lower bounds that depend on the characteristics of the phenomenon being studied or the measurement being taken.</span>
<span id="cb137-58"><a href="#cb137-58"></a></span>
<span id="cb137-59"><a href="#cb137-59"></a><span class="fu">## Dates</span></span>
<span id="cb137-60"><a href="#cb137-60"></a></span>
<span id="cb137-61"><a href="#cb137-61"></a><span class="al">![](/images/Dates.png)</span>{fig-align="center" width="300"}</span>
<span id="cb137-62"><a href="#cb137-62"></a></span>
<span id="cb137-63"><a href="#cb137-63"></a>We often encounter date data when dealing with time-related data. For example, in ecological research, data collection may involve recording the date of a particular observation or sampling event, such as the date when a bird was sighted, or when water samples were taken from a stream. The purpose of using date (or time) data in biology and ecology is to enable us to understand and analyse temporal patterns and relationships in their response variables. This can include exploring seasonal trends, understanding the impact of environmental changes over time, or tracking the growth and development of organisms.</span>
<span id="cb137-64"><a href="#cb137-64"></a></span>
<span id="cb137-65"><a href="#cb137-65"></a>By analysing date data, we can gain insights into long-term trends and patterns that may not be apparent when looking at the data in aggregate. They can also use this information to make predictions about future trends, develop more effective management strategies, and identify potential areas for further research.</span>
<span id="cb137-66"><a href="#cb137-66"></a></span>
<span id="cb137-67"><a href="#cb137-67"></a><span class="fu">## Character data</span></span>
<span id="cb137-68"><a href="#cb137-68"></a></span>
<span id="cb137-69"><a href="#cb137-69"></a>Character data are used to describe qualitative variables or descriptive text that are not numerical in nature. Character data can be entered as descriptive character strings, and internally, they are translated into a vector of characters in R. They are often used to represent categorical variables, such as the type of plant species, the colour of a bird's feathers, or the name of a some gene. Social scientists will sometimes use character data fields to record the names of people or places, or other descriptive information, such as a narrative that will later be subjected to, for example, a sentiment analysis. For convenience, I'll call these data **narrative style** data to distinguish them from the **qualitative** data that are the main focus of the present discussion.</span>
<span id="cb137-70"><a href="#cb137-70"></a></span>
<span id="cb137-71"><a href="#cb137-71"></a>Since narrative style data are not directly amenable to statistical analsysis, in this module, we will mainly concern ourselves with qualitative data which are typically names of things, or categories of objects, classes of behaviours, properties, characteristics, and so on. Qualitative data typically refer to *non-numeric data* collected from observations, experimental treatment groups, or other sources. They tend to be textual and are often used to describe characteristics or properties of living organisms, ecosystems, or other biological phenomena. Examples may include the colour of flowers, the type of habitat where an animal is found, the behaviour of animals, or the presence or absence of certain traits or characteristics in a population.</span>
<span id="cb137-72"><a href="#cb137-72"></a></span>
<span id="cb137-73"><a href="#cb137-73"></a>Qualitative data can be further classified into **nominal** or **ordinal** data types. Ordinal and nominal data are both amenable to statistical interpretation.</span>
<span id="cb137-74"><a href="#cb137-74"></a></span>
<span id="cb137-75"><a href="#cb137-75"></a><span class="fu">### Nominal variables</span></span>
<span id="cb137-76"><a href="#cb137-76"></a></span>
<span id="cb137-77"><a href="#cb137-77"></a>Nominal data are used to describe qualitative variables that do not have any inherent order or ranking. Examples of nominal data in biology may include the type of plant or animal species, or the presence or absence of certain genetic traits. Another term for nominal data is *categorical data*. Because there are well-defined categories, the number of members belonging to each of the category can be counted. For example, there are three red flowers, 66 purple flowers, and 13 yellow flowers. </span>
<span id="cb137-78"><a href="#cb137-78"></a></span>
<span id="cb137-79"><a href="#cb137-79"></a><span class="fu">### Ordinal variables</span></span>
<span id="cb137-80"><a href="#cb137-80"></a></span>
<span id="cb137-81"><a href="#cb137-81"></a>Ordinal data refer to a type of data that can be used to describe qualitative categorical variables that have a natural order or ranking. It is used when we need to arrange things in a particular order, such as from worst to best or from least to most. However, the differences between the values cannot be measured or quantified exactly, making them somewhat subjective. Examples of ordinal data include the different stages of development of an organism or the performance of a species to different fertilisers. Ordinal data can be entered as descriptive character strings, and internally, they are translated into an ordered vector of integers in R. For example, we can use a scale of <span class="in">`1`</span> for terrible, <span class="in">`2`</span> for 'so-so', <span class="in">`3`</span> for average, <span class="in">`4`</span> for good, and <span class="in">`5`</span> for brilliant.</span>
<span id="cb137-82"><a href="#cb137-82"></a></span>
<span id="cb137-83"><a href="#cb137-83"></a><span class="fu">## Binary variables</span></span>
<span id="cb137-84"><a href="#cb137-84"></a></span>
<span id="cb137-85"><a href="#cb137-85"></a>Life can be boiled down to a series of binary decisions: should I have pizza for dinner, *yes* or *no*? Should I go to bed early, *TRUE* or *FALSE*? Should I start that new series on Netflix, *accept* or *reject*? Am I *present* or *absent*? You get the gist... This kind of binary decision-making is known as 'logical', and in R they can only take on the values of <span class="in">`TRUE`</span> or <span class="in">`FALSE`</span> (remember to mind your case!). In the computing world, logical data are often represented by <span class="in">`1`</span> for <span class="in">`TRUE`</span> and <span class="in">`0`</span> for <span class="in">`FALSE`</span>. So basically, your life's choices can be summarised as a string of <span class="in">`1`</span>s and <span class="in">`0`</span>s. Who knew it was that simple?</span>
<span id="cb137-86"><a href="#cb137-86"></a></span>
<span id="cb137-87"><a href="#cb137-87"></a><span class="al">![](../../images/bad_choices.jpg)</span>{fig-align="center" width="300"}</span>
<span id="cb137-88"><a href="#cb137-88"></a></span>
<span id="cb137-89"><a href="#cb137-89"></a>When it comes down to it, everything in life is either <span class="in">`black`</span> or <span class="in">`white`</span>, <span class="in">`right`</span> or <span class="in">`wrong`</span>, <span class="in">`good`</span> or <span class="in">`bad`</span>. It's like a cosmic game of "Would You Rather?" --- and we're all just playing along.</span>
<span id="cb137-90"><a href="#cb137-90"></a></span>
<span id="cb137-91"><a href="#cb137-91"></a><span class="fu">## Missing values</span></span>
<span id="cb137-92"><a href="#cb137-92"></a></span>
<span id="cb137-93"><a href="#cb137-93"></a>It's unfortunate to admit that one of the most reliable aspects of any biological dataset is the presence of missing data (the presence of something that's missing?!). It is a stark reminder of the fragility of life. How can we say that something contains missing data? It seems counter intuitive, as if the data were never there in the first place. However, as we remember the principles of *tidy* data, we see that every observation must be documented in a row, and each column in that row must contain a value. This organisation allows us to create a matrix of data from multiple observations. Since the data are presented in a two-dimensional format, any missing values from an observation will leave a gaping hole in the matrix. We call these 'missing values.' It's a somber reality that even the most meticulous collection of data can be marred by the loss of information.</span>
<span id="cb137-94"><a href="#cb137-94"></a></span>
<span id="cb137-95"><a href="#cb137-95"></a><span class="fu">## Complex numbers</span></span>
<span id="cb137-96"><a href="#cb137-96"></a></span>
<span id="cb137-97"><a href="#cb137-97"></a><span class="at">&gt; *"And if you gaze long enough into an abyss, the abyss will gaze back into you."*</span></span>
<span id="cb137-98"><a href="#cb137-98"></a><span class="at">&gt;</span></span>
<span id="cb137-99"><a href="#cb137-99"></a><span class="at">&gt; --- Friedrich Nietzsche</span></span>
<span id="cb137-100"><a href="#cb137-100"></a></span>
<span id="cb137-101"><a href="#cb137-101"></a>As we draw to a close on the topic of data types, we cling desperately to the threads of our sanity, hoping against hope that they remain tightly stitched. But let it be known, to those who dare enter further into the realm of data, that beneath the surface lie countless rocks, and around every corner lurk a legion of complex data types, waiting to ensnare the unwary. These shadows of information are as enigmatic as they are perilous, for they challenge the very essence of our understanding. It is not until the final chapter of our journey, when we confront the elusive art of modeling, that we will face these data demons head-on. But fear not, for we shall arm ourselves with the knowledge and techniques acquired on this treacherous path, and with each step forward, we shall move closer to mastering the darkness that awaits us.</span>
<span id="cb137-102"><a href="#cb137-102"></a></span>
<span id="cb137-103"><a href="#cb137-103"></a><span class="fu"># Data classes in R</span></span>
<span id="cb137-104"><a href="#cb137-104"></a></span>
<span id="cb137-105"><a href="#cb137-105"></a>R is a powerful programming language and software environment for statistical computing and graphics. It offers a wide variety of data classes and types to represent different types of information.</span>
<span id="cb137-106"><a href="#cb137-106"></a></span>
<span id="cb137-107"><a href="#cb137-107"></a>The **atomic modes** are <span class="in">`logical`</span>, <span class="in">`integer`</span>, <span class="in">`numeric`</span> (also sometimes called <span class="in">`double`</span>), <span class="in">`complex`</span>, <span class="in">`character`</span> and <span class="in">`raw`</span>. The <span class="in">`Date`</span> class is a specialised form of the <span class="in">`numeric`</span> class. Each atomic mode has its own properties and functions that can be used to manipulate objects of that mode. These atomic modes can be used to make an atomic data structure such as a <span class="in">`vector`</span>, <span class="in">`array`</span>, or <span class="in">`matrix`</span>. This knowledge is also important when working with R tabular mixed data structures, such as a <span class="in">`data.frame`</span>, <span class="in">`tibble`</span>, or <span class="in">`list`</span>. Please refer to Hadley Wickham's overview of vectors in <span class="co">[</span><span class="ot">Advanced R, 2nd edition</span><span class="co">](https://adv-r.hadley.nz/index.html)</span> for more insight presented in an informative yet concise way.</span>
<span id="cb137-108"><a href="#cb137-108"></a></span>
<span id="cb137-109"><a href="#cb137-109"></a>The **data class** can be determined with the <span class="in">`class()`</span> or <span class="in">`str()`</span> commands. Here's a brief overview of some of the main data types in R:</span>
<span id="cb137-110"><a href="#cb137-110"></a></span>
<span id="cb137-111"><a href="#cb137-111"></a><span class="fu">## `numeric`</span></span>
<span id="cb137-112"><a href="#cb137-112"></a></span>
<span id="cb137-113"><a href="#cb137-113"></a>In R, the <span class="in">`numeric`</span> data class represents either **integers** or **floating point** (decimal) values. Numerical data are quantitative in nature as they represent things that can be objectively counted, measured, or calculated---the *measured variables*.</span>
<span id="cb137-114"><a href="#cb137-114"></a></span>
<span id="cb137-115"><a href="#cb137-115"></a>Numeric data are one of the most common types of data used in statistical and mathematical analysis. In R, numeric data are represented by the class <span class="in">`numeric`</span>, which includes both integers and floating-point numbers. Numeric data can be used in a variety of operations and calculations, including arithmetic operations, statistical analyses, and visualisations. One important feature of the <span class="in">`numeric`</span> data class in R is that it supports vectorisation, which allows for efficient and concise operations on large sets of numeric data. Additionally, R provides a wide range of built-in functions for working with numeric data, including functions for calculating basic statistical measures such as mean, median, and standard deviation.</span>
<span id="cb137-116"><a href="#cb137-116"></a></span>
<span id="cb137-117"><a href="#cb137-117"></a>In R integer (discrete) data are called <span class="in">`int`</span> or <span class="in">`&lt;int&gt;`</span> while continuous data are denoted <span class="in">`num`</span> or <span class="in">`&lt;dbl&gt;`</span>.</span>
<span id="cb137-118"><a href="#cb137-118"></a></span>
<span id="cb137-119"><a href="#cb137-119"></a>**Example of integer data** Suppose you have a dataset of the number of rats in different storm water drains in a neighbourhood. The number of rats is a discrete variable because it can only take on integer values (you can't own a fraction of a rat).</span>
<span id="cb137-120"><a href="#cb137-120"></a></span>
<span id="cb137-121"><a href="#cb137-121"></a>Here's how you could create a vector of this data in R:</span>
<span id="cb137-122"><a href="#cb137-122"></a></span>
<span id="cb137-125"><a href="#cb137-125"></a><span class="in">```{r}</span></span>
<span id="cb137-126"><a href="#cb137-126"></a><span class="co"># Create a vector of the number of pets owned by each household</span></span>
<span id="cb137-127"><a href="#cb137-127"></a>num_rats <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">1</span>, <span class="dv">4</span>, <span class="dv">0</span>, <span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">0</span>, <span class="dv">3</span>, <span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">4</span>, <span class="dv">2</span>, <span class="dv">0</span>)</span>
<span id="cb137-128"><a href="#cb137-128"></a>num_rats</span>
<span id="cb137-129"><a href="#cb137-129"></a><span class="fu">class</span>(num_rats)</span>
<span id="cb137-130"><a href="#cb137-130"></a><span class="in">```</span></span>
<span id="cb137-131"><a href="#cb137-131"></a></span>
<span id="cb137-132"><a href="#cb137-132"></a>In this example, the data are represented as a vector called <span class="in">`num_rats`</span> of class <span class="in">`numeric`</span> (as revealed by <span class="in">`class(num_rats)`</span>). Each element of the vector represents the number of rats in one storm water drain. For example, the first element of the vector (<span class="in">`num_rats[1]`</span>) is <span class="in">`0`</span>, which means that the first drain in the dataset is free of rats. The fourth element of the vector (<span class="in">`num_rats[4]`</span>) is <span class="in">`2`</span>, indicating that the fourth drain in the dataset is occupied by <span class="in">`2`</span> rats.</span>
<span id="cb137-133"><a href="#cb137-133"></a></span>
<span id="cb137-134"><a href="#cb137-134"></a>One can also explicitly create a vector of integer using the <span class="in">`as.integer()`</span> function:</span>
<span id="cb137-135"><a href="#cb137-135"></a></span>
<span id="cb137-138"><a href="#cb137-138"></a><span class="in">```{r}</span></span>
<span id="cb137-139"><a href="#cb137-139"></a>num_rats_int <span class="ot">&lt;-</span> <span class="fu">as.integer</span>(num_rats)</span>
<span id="cb137-140"><a href="#cb137-140"></a>num_rats_int</span>
<span id="cb137-141"><a href="#cb137-141"></a><span class="fu">class</span>(num_rats_int)</span>
<span id="cb137-142"><a href="#cb137-142"></a><span class="in">```</span></span>
<span id="cb137-143"><a href="#cb137-143"></a></span>
<span id="cb137-144"><a href="#cb137-144"></a>Above we *coerced* the class <span class="in">`numeric`</span> data to class <span class="in">`integer`</span>. But we can take floating point numeric and convert them to integers too with the <span class="in">`as.integer()`</span> function. As we see, the effect is that the whole part of the number is retained and the rest discarded:</span>
<span id="cb137-145"><a href="#cb137-145"></a></span>
<span id="cb137-148"><a href="#cb137-148"></a><span class="in">```{r}</span></span>
<span id="cb137-149"><a href="#cb137-149"></a>pies <span class="ot">&lt;-</span> pi <span class="sc">*</span> <span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)</span>
<span id="cb137-150"><a href="#cb137-150"></a>pies</span>
<span id="cb137-151"><a href="#cb137-151"></a><span class="fu">class</span>(pies)</span>
<span id="cb137-152"><a href="#cb137-152"></a><span class="fu">as.integer</span>(pies)</span>
<span id="cb137-153"><a href="#cb137-153"></a><span class="in">```</span></span>
<span id="cb137-154"><a href="#cb137-154"></a></span>
<span id="cb137-155"><a href="#cb137-155"></a>Effectively, what happened above is more-or-less equivalent to what the <span class="in">`floor()`</span> function would return:</span>
<span id="cb137-156"><a href="#cb137-156"></a></span>
<span id="cb137-159"><a href="#cb137-159"></a><span class="in">```{r}</span></span>
<span id="cb137-160"><a href="#cb137-160"></a><span class="fu">floor</span>(pies)</span>
<span id="cb137-161"><a href="#cb137-161"></a><span class="in">```</span></span>
<span id="cb137-162"><a href="#cb137-162"></a></span>
<span id="cb137-163"><a href="#cb137-163"></a>Be careful when coercing floating point numbers to integers. If rounding is what you expect, this is not what you will get. For rounding, use <span class="in">`round()`</span> instead:</span>
<span id="cb137-164"><a href="#cb137-164"></a></span>
<span id="cb137-167"><a href="#cb137-167"></a><span class="in">```{r}</span></span>
<span id="cb137-168"><a href="#cb137-168"></a><span class="fu">round</span>(pies, <span class="dv">0</span>)</span>
<span id="cb137-169"><a href="#cb137-169"></a><span class="in">```</span></span>
<span id="cb137-170"><a href="#cb137-170"></a></span>
<span id="cb137-171"><a href="#cb137-171"></a>**Example of continuous data** Here are some randomly generated temperature data assigned to an object called <span class="in">`temp_data`</span>:</span>
<span id="cb137-172"><a href="#cb137-172"></a></span>
<span id="cb137-175"><a href="#cb137-175"></a><span class="in">```{r}</span></span>
<span id="cb137-176"><a href="#cb137-176"></a><span class="co"># Generate a vector of 50 normally distributed temperature values</span></span>
<span id="cb137-177"><a href="#cb137-177"></a>temp_data <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">rnorm</span>(<span class="at">n =</span> <span class="dv">50</span>, <span class="at">mean =</span> <span class="dv">15</span>, <span class="at">sd =</span> <span class="dv">3</span>), <span class="dv">2</span>)</span>
<span id="cb137-178"><a href="#cb137-178"></a>temp_data</span>
<span id="cb137-179"><a href="#cb137-179"></a><span class="fu">class</span>(temp_data)</span>
<span id="cb137-180"><a href="#cb137-180"></a><span class="in">```</span></span>
<span id="cb137-181"><a href="#cb137-181"></a></span>
<span id="cb137-182"><a href="#cb137-182"></a><span class="fu">## `character`</span></span>
<span id="cb137-183"><a href="#cb137-183"></a></span>
<span id="cb137-184"><a href="#cb137-184"></a>In R, the <span class="in">`character`</span> data class represents textual data such as words, sentences, and paragraphs. Character data can be created using either single or double quotes, and it can include letters, numbers, and other special characters. In addition, character data can be concatenated using the <span class="in">`paste()`</span> function or other string manipulation functions.</span>
<span id="cb137-185"><a href="#cb137-185"></a></span>
<span id="cb137-186"><a href="#cb137-186"></a>One important feature of the character data class in R is its versatility in working with textual data. For instance, it can be used to store and manipulate text data, including text-based datasets, text-based files, and text-based visualisations. Additionally, R provides a wide range of built-in functions for working with character data, including functions for manipulating strings, searching for patterns, and formatting output. Overall, the character data class in R is a fundamental data type that is critical for working with textual data in a variety of contexts. You will most frequently use character values are often used to represent labels, names, or descriptions.</span>
<span id="cb137-187"><a href="#cb137-187"></a></span>
<span id="cb137-188"><a href="#cb137-188"></a><span class="fu">## `factor`</span></span>
<span id="cb137-189"><a href="#cb137-189"></a></span>
<span id="cb137-190"><a href="#cb137-190"></a>In R, the <span class="in">`factor`</span> data class is used to represent discrete categorical variables. Factors are often used in statistical analyses to represent class or group belonging. Factor values are categorical data, such as levels or categories of a variable. Factor variables are most commonly also character data, but they can be numeric too if coded correctly as factors. Factor values can be ordered (ordinal) or unordered (categorical or nominal).</span>
<span id="cb137-191"><a href="#cb137-191"></a></span>
<span id="cb137-192"><a href="#cb137-192"></a>Categorical variables take on a limited number of distinct values, often corresponding to different groups or levels. For example, a categorical variable might represent different colours, size classes, or species. Factors in R are represented as integers with corresponding character levels, where each level corresponds to a distinct category. The levels of a factor can be defined explicitly using the <span class="in">`factor()`</span> function or automatically using the <span class="in">`cut()`</span> function. One important feature of the factor data class in R is that it allows for efficient and effective data manipulation and analysis, particularly when working with large datasets. For instance, factors can be used in statistical analyses such as regression models or ANOVA, and they can also be used to create visualisations such as bar or pie graphs. The <span class="in">`factor`</span> data class in R is a fundamental data type that is critical for representing and working with categorical variables in data analysis and visualisation.</span>
<span id="cb137-193"><a href="#cb137-193"></a></span>
<span id="cb137-194"><a href="#cb137-194"></a>The <span class="in">`factor`</span> data class of data in an R <span class="in">`data.frame`</span> structure (or in a <span class="in">`tibble`</span>) is indicated by <span class="in">`Factor`</span> or <span class="in">`&lt;fctr&gt;`</span>. Ordered factors are denoted by columns named <span class="in">`Ord.factor`</span> or <span class="in">`&lt;ord&gt;`</span>.</span>
<span id="cb137-195"><a href="#cb137-195"></a></span>
<span id="cb137-196"><a href="#cb137-196"></a>**Nominal data** One example of nominal factor data that ecologists might encounter is the type of vegetation in a particular area, such as 'grassland', 'forest', or 'wetland'. Here's an example of how to generate a vector of nominal data in R using the <span class="in">`sample()`</span> function:</span>
<span id="cb137-197"><a href="#cb137-197"></a></span>
<span id="cb137-200"><a href="#cb137-200"></a><span class="in">```{r}</span></span>
<span id="cb137-201"><a href="#cb137-201"></a><span class="co"># Generate a vector of vegetation types</span></span>
<span id="cb137-202"><a href="#cb137-202"></a>vegetation <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="st">"grassland"</span>, <span class="st">"forest"</span>, <span class="st">"wetland"</span>), <span class="at">size =</span> <span class="dv">50</span>, <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb137-203"><a href="#cb137-203"></a></span>
<span id="cb137-204"><a href="#cb137-204"></a><span class="co"># View the vegetation data</span></span>
<span id="cb137-205"><a href="#cb137-205"></a>vegetation</span>
<span id="cb137-206"><a href="#cb137-206"></a><span class="fu">class</span>(vegetation)</span>
<span id="cb137-207"><a href="#cb137-207"></a><span class="in">```</span></span>
<span id="cb137-208"><a href="#cb137-208"></a></span>
<span id="cb137-209"><a href="#cb137-209"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb137-210"><a href="#cb137-210"></a><span class="fu">## The `sample()` function</span></span>
<span id="cb137-211"><a href="#cb137-211"></a>Note that the <span class="in">`sample()`</span> function is not made specifically for nominal data; it can be used on any kind of data class.</span>
<span id="cb137-212"><a href="#cb137-212"></a>:::</span>
<span id="cb137-213"><a href="#cb137-213"></a></span>
<span id="cb137-214"><a href="#cb137-214"></a>**Ordinal data** Here's an example vector of ordinal data in R that could be encountered by ecologists:</span>
<span id="cb137-215"><a href="#cb137-215"></a></span>
<span id="cb137-218"><a href="#cb137-218"></a><span class="in">```{r}</span></span>
<span id="cb137-219"><a href="#cb137-219"></a><span class="co"># Vector of ordinal data representing the successional stage of a forest</span></span>
<span id="cb137-220"><a href="#cb137-220"></a>succession <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Early Pioneer"</span>, <span class="st">"Late Pioneer"</span>,</span>
<span id="cb137-221"><a href="#cb137-221"></a>                <span class="st">"Young Forest"</span>, <span class="st">"Mature Forest"</span>,</span>
<span id="cb137-222"><a href="#cb137-222"></a>                <span class="st">"Old Growth"</span>)</span>
<span id="cb137-223"><a href="#cb137-223"></a>succession</span>
<span id="cb137-224"><a href="#cb137-224"></a><span class="fu">class</span>(succession)</span>
<span id="cb137-225"><a href="#cb137-225"></a></span>
<span id="cb137-226"><a href="#cb137-226"></a><span class="co"># Convert to ordered factor</span></span>
<span id="cb137-227"><a href="#cb137-227"></a>succession <span class="ot">&lt;-</span> <span class="fu">factor</span>(succession, <span class="at">ordered =</span> <span class="cn">TRUE</span>,</span>
<span id="cb137-228"><a href="#cb137-228"></a>                     <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Early Pioneer"</span>, <span class="st">"Late Pioneer"</span>,</span>
<span id="cb137-229"><a href="#cb137-229"></a>                                <span class="st">"Young Forest"</span>, <span class="st">"Mature Forest"</span>,</span>
<span id="cb137-230"><a href="#cb137-230"></a>                                <span class="st">"Old Growth"</span>))</span>
<span id="cb137-231"><a href="#cb137-231"></a>succession</span>
<span id="cb137-232"><a href="#cb137-232"></a><span class="fu">class</span>(succession)</span>
<span id="cb137-233"><a href="#cb137-233"></a><span class="in">```</span></span>
<span id="cb137-234"><a href="#cb137-234"></a></span>
<span id="cb137-235"><a href="#cb137-235"></a>In this example, the successional stage of a forest is represented by an ordinal scale with five levels ranging from 'Early Pioneer' to 'Old Growth'. The <span class="in">`factor()`</span> function is used to convert the vector to an ordered factor, with the <span class="in">`ordered`</span> argument set to <span class="in">`TRUE`</span> and the <span class="in">`levels`</span> argument set to the same order as the original vector. This ensures that the levels are properly represented as an ordered factor. </span>
<span id="cb137-236"><a href="#cb137-236"></a></span>
<span id="cb137-237"><a href="#cb137-237"></a><span class="fu">## `logical`</span></span>
<span id="cb137-238"><a href="#cb137-238"></a></span>
<span id="cb137-239"><a href="#cb137-239"></a>In R, the <span class="in">`logical`</span> data class represents binary or Boolean data. Logical data are used to represent variables that can take on only two possible values, <span class="in">`TRUE`</span> or <span class="in">`FALSE`</span>. In addition to <span class="in">`TRUE`</span> and <span class="in">`FALSE`</span>, logical data can also take on the values of <span class="in">`NA`</span> or <span class="in">`NULL`</span>, which represent missing or undefined values.</span>
<span id="cb137-240"><a href="#cb137-240"></a></span>
<span id="cb137-241"><a href="#cb137-241"></a>Logical data can be created using logical operators such as <span class="in">`==`</span>, <span class="in">`!=`</span>, <span class="in">`&gt;`</span>, <span class="in">`&lt;`</span>, <span class="in">`&gt;=`</span>, and <span class="in">`&lt;=`</span>. Logical data are commonly used in R for data filtering and selection, conditional statements, and logical operations. For example, logical data can be used to filter a dataset to include only observations that meet certain criteria or to perform logical operations such as AND (<span class="in">`&amp;`</span>) and OR (<span class="in">`|`</span>). The logical data class in R is a fundamental data type that is critical for representing and working with binary or Boolean variables in data analysis and programming.</span>
<span id="cb137-242"><a href="#cb137-242"></a></span>
<span id="cb137-243"><a href="#cb137-243"></a>**Example logical (binary) data** Here's an example of generating a vector of binary or logical data in R, which represents the presence or absence of a particular species in different ecological sites:</span>
<span id="cb137-244"><a href="#cb137-244"></a></span>
<span id="cb137-247"><a href="#cb137-247"></a><span class="in">```{r}</span></span>
<span id="cb137-248"><a href="#cb137-248"></a><span class="co"># Generate a vector of 1s and 0s to represent the presence</span></span>
<span id="cb137-249"><a href="#cb137-249"></a><span class="co"># or absence of a species in different ecological sites</span></span>
<span id="cb137-250"><a href="#cb137-250"></a>species_presence <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="dv">10</span>, <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb137-251"><a href="#cb137-251"></a>species_presence</span>
<span id="cb137-252"><a href="#cb137-252"></a><span class="in">```</span></span>
<span id="cb137-253"><a href="#cb137-253"></a></span>
<span id="cb137-254"><a href="#cb137-254"></a>We can also make a formal logical class data:</span>
<span id="cb137-255"><a href="#cb137-255"></a></span>
<span id="cb137-258"><a href="#cb137-258"></a><span class="in">```{r}</span></span>
<span id="cb137-259"><a href="#cb137-259"></a>species_presence_logi <span class="ot">&lt;-</span> <span class="fu">as.logical</span>(species_presence)</span>
<span id="cb137-260"><a href="#cb137-260"></a><span class="fu">class</span>(species_presence_logi)</span>
<span id="cb137-261"><a href="#cb137-261"></a><span class="in">```</span></span>
<span id="cb137-262"><a href="#cb137-262"></a></span>
<span id="cb137-263"><a href="#cb137-263"></a>In this example, we again    use the <span class="in">`sample()`</span> function to randomly generate a vector of 10 values, each either 0 or 1, to represent the presence or absence of a species in 10 different ecological sites. However, it is often not necessary to coerce to class <span class="in">`logical`</span>, as we see in the presence-absence datasets we will encounter in <span class="co">[</span><span class="ot">BCB743: Quantitative Ecology</span><span class="co">](../../BCB743/BCB743_index.html)</span>.</span>
<span id="cb137-264"><a href="#cb137-264"></a></span>
<span id="cb137-265"><a href="#cb137-265"></a><span class="fu">## `date`</span></span>
<span id="cb137-266"><a href="#cb137-266"></a></span>
<span id="cb137-267"><a href="#cb137-267"></a>In R, the <span class="in">`POSIXct`</span>, <span class="in">`POSIXlt`</span>, and <span class="in">`Date`</span> classes are commonly used to represent date and time data. These classes each have unique characteristics that make them useful for different purposes.</span>
<span id="cb137-268"><a href="#cb137-268"></a></span>
<span id="cb137-269"><a href="#cb137-269"></a>The <span class="in">`POSIXct`</span> class is a date/time class that represents dates and times as a numerical value, *typically* measured in seconds since January 1st, 1970. This class provides a high level of precision, with values accurate to the second. It is useful for performing calculations and data manipulation involving time, such as finding the difference between two dates or adding a certain number of seconds to a given time. An example of how to generate a <span class="in">`POSIXct`</span> object in R is as follows:</span>
<span id="cb137-270"><a href="#cb137-270"></a></span>
<span id="cb137-273"><a href="#cb137-273"></a><span class="in">```{r}</span></span>
<span id="cb137-274"><a href="#cb137-274"></a>my_time <span class="ot">&lt;-</span> <span class="fu">as.POSIXct</span>(<span class="st">"2022-03-10 12:34:56"</span>)</span>
<span id="cb137-275"><a href="#cb137-275"></a><span class="fu">class</span>(my_time)</span>
<span id="cb137-276"><a href="#cb137-276"></a>my_time</span>
<span id="cb137-277"><a href="#cb137-277"></a><span class="in">```</span></span>
<span id="cb137-278"><a href="#cb137-278"></a></span>
<span id="cb137-279"><a href="#cb137-279"></a>The <span class="in">`POSIXlt`</span> class, on the other hand, *typically* represents dates and times in a more human-readable format. It stores date and time information as a list of named elements, including <span class="in">`year`</span>, <span class="in">`month`</span>, <span class="in">`day`</span>, <span class="in">`hour`</span>, <span class="in">`minute`</span>, and <span class="in">`second.`</span> This format is useful for displaying data in a more understandable way and for extracting specific components of a date or time. An example of how to generate a POSIXlt object in R is as follows:</span>
<span id="cb137-280"><a href="#cb137-280"></a></span>
<span id="cb137-283"><a href="#cb137-283"></a><span class="in">```{r}</span></span>
<span id="cb137-284"><a href="#cb137-284"></a>my_time <span class="ot">&lt;-</span> <span class="fu">as.POSIXlt</span>(<span class="st">"2022-03-10 12:34:56"</span>)</span>
<span id="cb137-285"><a href="#cb137-285"></a><span class="fu">class</span>(my_time)</span>
<span id="cb137-286"><a href="#cb137-286"></a>my_time</span>
<span id="cb137-287"><a href="#cb137-287"></a><span class="in">```</span></span>
<span id="cb137-288"><a href="#cb137-288"></a></span>
<span id="cb137-289"><a href="#cb137-289"></a>The <span class="in">`Date`</span> class is used to represent dates only, without any time information. Dates are *typically* stored as the number of days since January 1st, 1970. This class provides functions for performing arithmetic operations and comparisons between dates. It is useful for working with time-based data that is only concerned with the date component, such as daily sales or stock prices. An example of how to generate a Date object in R is as follows:</span>
<span id="cb137-290"><a href="#cb137-290"></a></span>
<span id="cb137-293"><a href="#cb137-293"></a><span class="in">```{r}</span></span>
<span id="cb137-294"><a href="#cb137-294"></a>my_date <span class="ot">&lt;-</span> <span class="fu">as.Date</span>(<span class="st">"2022-03-10"</span>)</span>
<span id="cb137-295"><a href="#cb137-295"></a><span class="fu">class</span>(my_date)</span>
<span id="cb137-296"><a href="#cb137-296"></a>my_date</span>
<span id="cb137-297"><a href="#cb137-297"></a><span class="in">```</span></span>
<span id="cb137-298"><a href="#cb137-298"></a></span>
<span id="cb137-299"><a href="#cb137-299"></a>To generate a vector of dates in R with daily intervals, we can use the <span class="in">`seq()`</span> function to create a sequence of dates, specifying the start and end dates and the time interval. Here's an example:</span>
<span id="cb137-300"><a href="#cb137-300"></a></span>
<span id="cb137-303"><a href="#cb137-303"></a><span class="in">```{r}</span></span>
<span id="cb137-304"><a href="#cb137-304"></a><span class="co"># Generate a vector of dates from January 1, 2022 to December 31, 2022</span></span>
<span id="cb137-305"><a href="#cb137-305"></a>dates <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fu">as.Date</span>(<span class="st">"2022-01-01"</span>), <span class="fu">as.Date</span>(<span class="st">"2022-12-31"</span>), <span class="at">by =</span> <span class="st">"day"</span>)</span>
<span id="cb137-306"><a href="#cb137-306"></a></span>
<span id="cb137-307"><a href="#cb137-307"></a><span class="co"># View the first 10 dates in the vector</span></span>
<span id="cb137-308"><a href="#cb137-308"></a><span class="fu">head</span>(dates, <span class="dv">10</span>)</span>
<span id="cb137-309"><a href="#cb137-309"></a><span class="fu">class</span>(dates)</span>
<span id="cb137-310"><a href="#cb137-310"></a><span class="in">```</span></span>
<span id="cb137-311"><a href="#cb137-311"></a></span>
<span id="cb137-312"><a href="#cb137-312"></a>Understanding the characteristics of these date and time classes in R is essential for effective data analysis and manipulation in fields where time-based data is a critical component.</span>
<span id="cb137-313"><a href="#cb137-313"></a></span>
<span id="cb137-314"><a href="#cb137-314"></a>Date and time data in R can be manipulated using various built-in functions and packages such as **lubridate** and **chron**. Additionally, date and time data can be visualised using different types of graphs such as time series plots, heatmaps, and Hovmöller diagrams. The date and time data classes in R are essential for working with temporal data and conducting time-related analyses in various biological and environmental datasets.</span>
<span id="cb137-315"><a href="#cb137-315"></a></span>
<span id="cb137-316"><a href="#cb137-316"></a><span class="fu">## Missing values, `NA`</span></span>
<span id="cb137-317"><a href="#cb137-317"></a></span>
<span id="cb137-318"><a href="#cb137-318"></a>Missing values can be encountered in vectors of all data classes. To demonstrate some data that contains missing values, I will generate a data sequence containing 5% missing values. We can use the <span class="in">`rnorm()`</span> function to generate a sequence of random normal numbers and then randomly assign 5% of the values as missing using the <span class="in">`sample()`</span> function. The indices of the missing values are stored in missing_indices, and we use them to assign <span class="in">`NA`</span> to the corresponding elements of the data sequence. Here's some code to achieve this:</span>
<span id="cb137-319"><a href="#cb137-319"></a></span>
<span id="cb137-322"><a href="#cb137-322"></a><span class="in">```{r}</span></span>
<span id="cb137-323"><a href="#cb137-323"></a><span class="co"># Set the length of the sequence</span></span>
<span id="cb137-324"><a href="#cb137-324"></a>n <span class="ot">&lt;-</span> <span class="dv">100</span></span>
<span id="cb137-325"><a href="#cb137-325"></a></span>
<span id="cb137-326"><a href="#cb137-326"></a><span class="co"># Generate a sequence of random normal numbers with</span></span>
<span id="cb137-327"><a href="#cb137-327"></a><span class="co"># mean 0 and standard deviation 1</span></span>
<span id="cb137-328"><a href="#cb137-328"></a>data <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(n, <span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>)</span>
<span id="cb137-329"><a href="#cb137-329"></a></span>
<span id="cb137-330"><a href="#cb137-330"></a><span class="co"># Randomly assign 5% of the values as missing</span></span>
<span id="cb137-331"><a href="#cb137-331"></a>missing_indices <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span>n, <span class="at">size =</span> <span class="fu">round</span>(<span class="fl">0.05</span><span class="sc">*</span>n))</span>
<span id="cb137-332"><a href="#cb137-332"></a>data[missing_indices] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb137-333"><a href="#cb137-333"></a><span class="fu">length</span>(data)</span>
<span id="cb137-334"><a href="#cb137-334"></a>data</span>
<span id="cb137-335"><a href="#cb137-335"></a><span class="in">```</span></span>
<span id="cb137-336"><a href="#cb137-336"></a></span>
<span id="cb137-337"><a href="#cb137-337"></a>To remove all <span class="in">`NA`</span>s from the vector of data we can use <span class="in">`na.omit()`</span>:</span>
<span id="cb137-338"><a href="#cb137-338"></a></span>
<span id="cb137-341"><a href="#cb137-341"></a><span class="in">```{r}</span></span>
<span id="cb137-342"><a href="#cb137-342"></a>data_sans_na <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(data)</span>
<span id="cb137-343"><a href="#cb137-343"></a><span class="fu">length</span>(data_sans_na)</span>
<span id="cb137-344"><a href="#cb137-344"></a>data_sans_na</span>
<span id="cb137-345"><a href="#cb137-345"></a><span class="in">```</span></span>
<span id="cb137-346"><a href="#cb137-346"></a></span>
<span id="cb137-347"><a href="#cb137-347"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb137-348"><a href="#cb137-348"></a><span class="fu">## Dealing with `NA`s in functions</span></span>
<span id="cb137-349"><a href="#cb137-349"></a>Many functions have specific arguments to deal with <span class="in">`NA`</span>s in data. See for example the <span class="in">`na.rm = TRUE`</span> argument given to <span class="in">`mean()`</span>, <span class="in">`median()`</span>, <span class="in">`min()`</span>, <span class="in">`lm()`</span>, etc.</span>
<span id="cb137-350"><a href="#cb137-350"></a>:::</span>
<span id="cb137-351"><a href="#cb137-351"></a></span>
<span id="cb137-352"><a href="#cb137-352"></a><span class="fu"># Data structures in R</span></span>
<span id="cb137-353"><a href="#cb137-353"></a></span>
<span id="cb137-354"><a href="#cb137-354"></a>Data structures can be viewed as the 'containers' that hold the different data classes. The kind of data structure can also be revealed by the <span class="in">`class()`</span> command.</span>
<span id="cb137-355"><a href="#cb137-355"></a></span>
<span id="cb137-356"><a href="#cb137-356"></a><span class="fu">## `vector`, `array`, and `matrix`</span></span>
<span id="cb137-357"><a href="#cb137-357"></a></span>
<span id="cb137-358"><a href="#cb137-358"></a>**Vectors** In R, a vector is a one-dimensional array-like data structure that can hold a sequence of values of the same atomic mode, such as <span class="in">`numeric`</span>, <span class="in">`character`</span>, <span class="in">`logical`</span> values, or <span class="in">`Date`</span> and times. A vector can be created using the <span class="in">`c()`</span> function, which stands for 'combine' or 'concatenate,' and is used to combine a sequence of values into a vector. Vectors can also be created by using the <span class="in">`seq()`</span> function to generate a sequence of numbers, or the <span class="in">`rep()`</span> function to repeat a value or sequence of values. Here is an example of a numeric vector:</span>
<span id="cb137-359"><a href="#cb137-359"></a></span>
<span id="cb137-362"><a href="#cb137-362"></a><span class="in">```{r}</span></span>
<span id="cb137-363"><a href="#cb137-363"></a><span class="co"># create a numeric vector</span></span>
<span id="cb137-364"><a href="#cb137-364"></a>my_vector <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>)</span>
<span id="cb137-365"><a href="#cb137-365"></a></span>
<span id="cb137-366"><a href="#cb137-366"></a><span class="co"># coerce to vector</span></span>
<span id="cb137-367"><a href="#cb137-367"></a>my_vector <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>))</span>
<span id="cb137-368"><a href="#cb137-368"></a><span class="fu">class</span>(my_vector) <span class="co"># but it doesn't change the class from numeric</span></span>
<span id="cb137-369"><a href="#cb137-369"></a></span>
<span id="cb137-370"><a href="#cb137-370"></a><span class="co"># print the vector</span></span>
<span id="cb137-371"><a href="#cb137-371"></a>my_vector</span>
<span id="cb137-372"><a href="#cb137-372"></a><span class="in">```</span></span>
<span id="cb137-373"><a href="#cb137-373"></a></span>
<span id="cb137-374"><a href="#cb137-374"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb137-375"><a href="#cb137-375"></a><span class="fu">## Coercion to vector</span></span>
<span id="cb137-376"><a href="#cb137-376"></a>The behaviour is such that the output of coercion to vector is that one the atomic modes (the basic data types) is returned.</span>
<span id="cb137-377"><a href="#cb137-377"></a>:::</span>
<span id="cb137-378"><a href="#cb137-378"></a></span>
<span id="cb137-379"><a href="#cb137-379"></a>One of the advantages of using vectors in R is that many of the built-in functions and operations work on vectors, allowing us to easily manipulate and analyse large amounts of data. Additionally, R provides many functions specifically designed for working with vectors, such as <span class="in">`mean()`</span>, <span class="in">`median()`</span>, <span class="in">`sum()`</span>, <span class="in">`min()`</span>, <span class="in">`max()`</span>, and many others.</span>
<span id="cb137-380"><a href="#cb137-380"></a></span>
<span id="cb137-381"><a href="#cb137-381"></a>**Matrices** A matrix (again, this terminology may be different for other languages), on the other hand, is a special case of an array that has two dimensions (rows and columns). It is also a multi-dimensional data structure that can hold elements of the *same data type*, but it is specifically designed for handling data in a tabular format. A matrix can be created using the <span class="in">`matrix()`</span> function in R.</span>
<span id="cb137-382"><a href="#cb137-382"></a></span>
<span id="cb137-385"><a href="#cb137-385"></a><span class="in">```{r}</span></span>
<span id="cb137-386"><a href="#cb137-386"></a><span class="co"># create a numeric matrix</span></span>
<span id="cb137-387"><a href="#cb137-387"></a>my_matrix <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb137-388"><a href="#cb137-388"></a></span>
<span id="cb137-389"><a href="#cb137-389"></a><span class="co"># print the matrix</span></span>
<span id="cb137-390"><a href="#cb137-390"></a>my_matrix</span>
<span id="cb137-391"><a href="#cb137-391"></a><span class="fu">class</span>(my_matrix)</span>
<span id="cb137-392"><a href="#cb137-392"></a><span class="in">```</span></span>
<span id="cb137-393"><a href="#cb137-393"></a></span>
<span id="cb137-394"><a href="#cb137-394"></a>We can query the size or dimensions of the matrix as follows:</span>
<span id="cb137-395"><a href="#cb137-395"></a></span>
<span id="cb137-398"><a href="#cb137-398"></a><span class="in">```{r}</span></span>
<span id="cb137-399"><a href="#cb137-399"></a><span class="fu">dim</span>(my_matrix)</span>
<span id="cb137-400"><a href="#cb137-400"></a><span class="fu">ncol</span>(my_matrix)</span>
<span id="cb137-401"><a href="#cb137-401"></a><span class="fu">nrow</span>(my_matrix)</span>
<span id="cb137-402"><a href="#cb137-402"></a><span class="in">```</span></span>
<span id="cb137-403"><a href="#cb137-403"></a></span>
<span id="cb137-404"><a href="#cb137-404"></a>**Coercion of matrices to vectors** A matrix can be coerced to a vector:</span>
<span id="cb137-405"><a href="#cb137-405"></a></span>
<span id="cb137-408"><a href="#cb137-408"></a><span class="in">```{r}</span></span>
<span id="cb137-409"><a href="#cb137-409"></a><span class="fu">as.vector</span>(my_matrix)</span>
<span id="cb137-410"><a href="#cb137-410"></a><span class="in">```</span></span>
<span id="cb137-411"><a href="#cb137-411"></a></span>
<span id="cb137-412"><a href="#cb137-412"></a>**Arrays** In R (as opposed to in python or some other languages), an array specifically refers to a multi-dimensional data structure that can hold elements of the *same data type*. It can have any number of dimensions (1, 2, 3, etc.), and its dimensions can be named. An array can be created using the <span class="in">`array()`</span> function in R.</span>
<span id="cb137-413"><a href="#cb137-413"></a></span>
<span id="cb137-416"><a href="#cb137-416"></a><span class="in">```{r}</span></span>
<span id="cb137-417"><a href="#cb137-417"></a><span class="co"># create a 2-dimensional array</span></span>
<span id="cb137-418"><a href="#cb137-418"></a>my_array <span class="ot">&lt;-</span> <span class="fu">array</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">27</span>, <span class="at">dim =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">3</span>))</span>
<span id="cb137-419"><a href="#cb137-419"></a></span>
<span id="cb137-420"><a href="#cb137-420"></a><span class="co"># print the array</span></span>
<span id="cb137-421"><a href="#cb137-421"></a>my_array</span>
<span id="cb137-422"><a href="#cb137-422"></a><span class="fu">class</span>(my_array)</span>
<span id="cb137-423"><a href="#cb137-423"></a><span class="in">```</span></span>
<span id="cb137-424"><a href="#cb137-424"></a></span>
<span id="cb137-425"><a href="#cb137-425"></a>We can figure something out about the size or dimensions of the array:</span>
<span id="cb137-426"><a href="#cb137-426"></a></span>
<span id="cb137-429"><a href="#cb137-429"></a><span class="in">```{r}</span></span>
<span id="cb137-430"><a href="#cb137-430"></a><span class="fu">dim</span>(my_array)</span>
<span id="cb137-431"><a href="#cb137-431"></a><span class="fu">ncol</span>(my_array)</span>
<span id="cb137-432"><a href="#cb137-432"></a><span class="fu">nrow</span>(my_array)</span>
<span id="cb137-433"><a href="#cb137-433"></a><span class="in">```</span></span>
<span id="cb137-434"><a href="#cb137-434"></a></span>
<span id="cb137-435"><a href="#cb137-435"></a>**Coercion of arrays to vectors** The array can be coerced to a vector:</span>
<span id="cb137-436"><a href="#cb137-436"></a></span>
<span id="cb137-439"><a href="#cb137-439"></a><span class="in">```{r}</span></span>
<span id="cb137-440"><a href="#cb137-440"></a><span class="fu">as.vector</span>(my_array)</span>
<span id="cb137-441"><a href="#cb137-441"></a><span class="in">```</span></span>
<span id="cb137-442"><a href="#cb137-442"></a></span>
<span id="cb137-443"><a href="#cb137-443"></a>The key difference between vectors, arrays, and a matrices in R is their dimensions. A vector has one dimension, an array can have any number of dimensions, while a matrix is limited to two dimensions. Additionally, a matrix is often used to store data in a tabular format, while an array is used to store multi-dimensional data in general. A commonly encountered kind of matrix is seen in multivariate statistics is a distance or dissimilarity matrix.</span>
<span id="cb137-444"><a href="#cb137-444"></a></span>
<span id="cb137-445"><a href="#cb137-445"></a>*In R, vectors, arrays, and matrices share a common characteristic: they do not have row or column names.* Therefore, to refer to any element, row, or column, one must use their corresponding index. How?</span>
<span id="cb137-446"><a href="#cb137-446"></a></span>
<span id="cb137-447"><a href="#cb137-447"></a>**Accessing elements, rows, columns, and matrices** In R, the square bracket notation is used to access elements, rows, columns, or matrices in arrays. The notation takes the form of <span class="in">`[i, j, k, ...]`</span>, where <span class="in">`i`</span>, <span class="in">`j`</span>, <span class="in">`k`</span>, and so on, represent the indices of the rows, columns, or matrices to be accessed.</span>
<span id="cb137-448"><a href="#cb137-448"></a></span>
<span id="cb137-449"><a href="#cb137-449"></a>Suppose we have the following array:</span>
<span id="cb137-450"><a href="#cb137-450"></a></span>
<span id="cb137-451"><a href="#cb137-451"></a><span class="co">&lt;!-- # my_array &lt;- array(data = 1:60, dim = c(5, 4, 3)) --&gt;</span></span>
<span id="cb137-452"><a href="#cb137-452"></a></span>
<span id="cb137-455"><a href="#cb137-455"></a><span class="in">```{r}</span></span>
<span id="cb137-456"><a href="#cb137-456"></a>my_array <span class="ot">&lt;-</span> <span class="fu">array</span>(<span class="at">data =</span> <span class="fu">round</span>(<span class="fu">rnorm</span>(<span class="at">n =</span> <span class="dv">60</span>, <span class="at">mean =</span> <span class="dv">13</span>, <span class="at">sd =</span> <span class="dv">2</span>), <span class="dv">1</span>),</span>
<span id="cb137-457"><a href="#cb137-457"></a>                  <span class="at">dim =</span> <span class="fu">c</span>(<span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">3</span>))</span>
<span id="cb137-458"><a href="#cb137-458"></a>my_array</span>
<span id="cb137-459"><a href="#cb137-459"></a><span class="fu">dim</span>(my_array)</span>
<span id="cb137-460"><a href="#cb137-460"></a><span class="in">```</span></span>
<span id="cb137-461"><a href="#cb137-461"></a></span>
<span id="cb137-462"><a href="#cb137-462"></a>This creates a $5\times4\times3$ array with values from <span class="in">`1`</span> to <span class="in">`60`</span>.</span>
<span id="cb137-463"><a href="#cb137-463"></a></span>
<span id="cb137-464"><a href="#cb137-464"></a>When working with multidimensional arrays, it is possible to omit some of the indices in the square bracket notation. This results in a subset of the array, which can be thought of as a lower-dimensional array obtained by fixing the omitted dimensions. For example, consider a 3-dimensional array <span class="in">`my_array`</span> above with dimensions <span class="in">`dim(my_array) = c(5,4,3)`</span>. If we use the notation <span class="in">`my_array[1,,]`</span>, we would obtain a 2-dimensional array with dimensions <span class="in">`dim(my_array[1,,]) = c(4,3)`</span> obtained by fixing the first index at 1:</span>
<span id="cb137-465"><a href="#cb137-465"></a></span>
<span id="cb137-468"><a href="#cb137-468"></a><span class="in">```{r}</span></span>
<span id="cb137-469"><a href="#cb137-469"></a><span class="fu">dim</span>(my_array[<span class="dv">1</span>,,])</span>
<span id="cb137-470"><a href="#cb137-470"></a>my_array[<span class="dv">1</span>,,]</span>
<span id="cb137-471"><a href="#cb137-471"></a><span class="in">```</span></span>
<span id="cb137-472"><a href="#cb137-472"></a></span>
<span id="cb137-473"><a href="#cb137-473"></a>Here are some more examples of how to use square brackets notation with arrays in R:</span>
<span id="cb137-474"><a href="#cb137-474"></a></span>
<span id="cb137-475"><a href="#cb137-475"></a>To access a **single element** in the array, use the notation <span class="in">`[i, j, k]`</span>, where <span class="in">`i`</span>, <span class="in">`j`</span>, and <span class="in">`k`</span> are the indices along each of the three dimensions, which in combination, uniquely identifies each element. Below we return the element in the second row, third column, and first matrix:</span>
<span id="cb137-476"><a href="#cb137-476"></a></span>
<span id="cb137-479"><a href="#cb137-479"></a><span class="in">```{r}</span></span>
<span id="cb137-480"><a href="#cb137-480"></a>my_array[<span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">1</span>]  </span>
<span id="cb137-481"><a href="#cb137-481"></a><span class="in">```</span></span>
<span id="cb137-482"><a href="#cb137-482"></a></span>
<span id="cb137-483"><a href="#cb137-483"></a>To access a single row in the array, use the notation <span class="in">`[i, , ]`</span>, where <span class="in">`i`</span> is the index of the row. This will return the second rows and all of the columns of the first matrix:</span>
<span id="cb137-484"><a href="#cb137-484"></a></span>
<span id="cb137-487"><a href="#cb137-487"></a><span class="in">```{r}</span></span>
<span id="cb137-488"><a href="#cb137-488"></a>my_array[<span class="dv">2</span>,,<span class="dv">1</span>]</span>
<span id="cb137-489"><a href="#cb137-489"></a><span class="in">```</span></span>
<span id="cb137-490"><a href="#cb137-490"></a></span>
<span id="cb137-491"><a href="#cb137-491"></a>To access a single column in the array, use the notation <span class="in">`[ , j, ]`</span>, where <span class="in">`j`</span> is the index of the column. Here we will return all the elements in the row of column two and matrix three:</span>
<span id="cb137-492"><a href="#cb137-492"></a></span>
<span id="cb137-495"><a href="#cb137-495"></a><span class="in">```{r}</span></span>
<span id="cb137-496"><a href="#cb137-496"></a>my_array[ , <span class="dv">2</span>, <span class="dv">3</span>]</span>
<span id="cb137-497"><a href="#cb137-497"></a><span class="in">```</span></span>
<span id="cb137-498"><a href="#cb137-498"></a></span>
<span id="cb137-499"><a href="#cb137-499"></a>To access a single matrix in the array, use the notation <span class="in">`[ , , k]`</span>, where <span class="in">`k`</span> is the index of the matrix:</span>
<span id="cb137-500"><a href="#cb137-500"></a></span>
<span id="cb137-503"><a href="#cb137-503"></a><span class="in">```{r}</span></span>
<span id="cb137-504"><a href="#cb137-504"></a>my_array[ , , <span class="dv">2</span>]</span>
<span id="cb137-505"><a href="#cb137-505"></a><span class="in">```</span></span>
<span id="cb137-506"><a href="#cb137-506"></a></span>
<span id="cb137-507"><a href="#cb137-507"></a>To obtain a subset of the array, use the notation <span class="in">`[i, j, k]`</span> with <span class="in">`i`</span>, <span class="in">`j`</span>, or <span class="in">`k`</span> omitted to obtain a lower-dimensional array:</span>
<span id="cb137-508"><a href="#cb137-508"></a></span>
<span id="cb137-511"><a href="#cb137-511"></a><span class="in">```{r}</span></span>
<span id="cb137-512"><a href="#cb137-512"></a>my_array[<span class="dv">1</span>, , ]</span>
<span id="cb137-513"><a href="#cb137-513"></a>my_array[ , <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>, ]</span>
<span id="cb137-514"><a href="#cb137-514"></a><span class="in">```</span></span>
<span id="cb137-515"><a href="#cb137-515"></a></span>
<span id="cb137-516"><a href="#cb137-516"></a><span class="fu">## `data.frame`</span></span>
<span id="cb137-517"><a href="#cb137-517"></a></span>
<span id="cb137-518"><a href="#cb137-518"></a>A dataframe is perhaps the most commonly-used 'container' for data in R because they are so convenient and serve many purposes. A dataframe is *not a data class*---more correctly, it is a form of tabular data (like a table in MS Excel), with each vector (a variable or column) comprising the table sharing the same length. What makes a dataframe versatile is that its variables can be any combination of the atomic data types. It may even include *list columns* (we will not cover list columns in this module). Applying the <span class="in">`class()`</span> function to a dataframe shows that it blongs to class <span class="in">`data.frame`</span>.</span>
<span id="cb137-519"><a href="#cb137-519"></a></span>
<span id="cb137-520"><a href="#cb137-520"></a>Here's an example of an R <span class="in">`data.frame`</span> with <span class="in">`Date`</span>, <span class="in">`numeric`</span>, and <span class="in">`categorical`</span> data classes:</span>
<span id="cb137-521"><a href="#cb137-521"></a></span>
<span id="cb137-524"><a href="#cb137-524"></a><span class="in">```{r}</span></span>
<span id="cb137-525"><a href="#cb137-525"></a><span class="co"># Create a vector of dates</span></span>
<span id="cb137-526"><a href="#cb137-526"></a>dates <span class="ot">&lt;-</span> <span class="fu">as.Date</span>(<span class="fu">c</span>(<span class="st">"2022-01-01"</span>, <span class="st">"2022-01-02"</span>, <span class="st">"2022-01-03"</span>,</span>
<span id="cb137-527"><a href="#cb137-527"></a>                   <span class="st">"2022-01-04"</span>, <span class="st">"2022-01-05"</span>))</span>
<span id="cb137-528"><a href="#cb137-528"></a></span>
<span id="cb137-529"><a href="#cb137-529"></a><span class="co"># Create a vector of numeric data</span></span>
<span id="cb137-530"><a href="#cb137-530"></a>numeric_data <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="at">n =</span> <span class="dv">5</span>, <span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>)</span>
<span id="cb137-531"><a href="#cb137-531"></a></span>
<span id="cb137-532"><a href="#cb137-532"></a><span class="co"># Create a vector of categorical data</span></span>
<span id="cb137-533"><a href="#cb137-533"></a>categorical_data <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"C"</span>, <span class="st">"A"</span>, <span class="st">"B"</span>)</span>
<span id="cb137-534"><a href="#cb137-534"></a></span>
<span id="cb137-535"><a href="#cb137-535"></a><span class="co"># Combine the vectors into a data.frame</span></span>
<span id="cb137-536"><a href="#cb137-536"></a>my_dataframe <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">dates =</span> dates,</span>
<span id="cb137-537"><a href="#cb137-537"></a>                           <span class="at">numeric_data =</span> numeric_data,</span>
<span id="cb137-538"><a href="#cb137-538"></a>                           <span class="at">categorical_data =</span> categorical_data)</span>
<span id="cb137-539"><a href="#cb137-539"></a></span>
<span id="cb137-540"><a href="#cb137-540"></a><span class="co"># Print the dataframe</span></span>
<span id="cb137-541"><a href="#cb137-541"></a>my_dataframe</span>
<span id="cb137-542"><a href="#cb137-542"></a><span class="fu">class</span>(my_dataframe)</span>
<span id="cb137-543"><a href="#cb137-543"></a><span class="fu">str</span>(my_dataframe)</span>
<span id="cb137-544"><a href="#cb137-544"></a><span class="fu">summary</span>(my_dataframe)</span>
<span id="cb137-545"><a href="#cb137-545"></a><span class="in">```</span></span>
<span id="cb137-546"><a href="#cb137-546"></a></span>
<span id="cb137-547"><a href="#cb137-547"></a>Dataframes may also have row names:</span>
<span id="cb137-548"><a href="#cb137-548"></a></span>
<span id="cb137-551"><a href="#cb137-551"></a><span class="in">```{r}</span></span>
<span id="cb137-552"><a href="#cb137-552"></a><span class="fu">rownames</span>(my_dataframe) <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">rep</span>(<span class="st">"row"</span>, <span class="dv">5</span>), <span class="at">seq =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)</span>
<span id="cb137-553"><a href="#cb137-553"></a>my_dataframe</span>
<span id="cb137-554"><a href="#cb137-554"></a><span class="in">```</span></span>
<span id="cb137-555"><a href="#cb137-555"></a></span>
<span id="cb137-556"><a href="#cb137-556"></a>Typically we will create a dataframe by reading in data from a <span class="in">`.csv`</span> file, but it is useful to be able to construct one from scratch.</span>
<span id="cb137-557"><a href="#cb137-557"></a></span>
<span id="cb137-558"><a href="#cb137-558"></a><span class="fu">## `tibble`</span></span>
<span id="cb137-559"><a href="#cb137-559"></a></span>
<span id="cb137-560"><a href="#cb137-560"></a>In R, a dataframe and a tibble are both data structures used to store tabular data. Although tibbles are also dataframes, but they differ subtly in several ways.</span>
<span id="cb137-561"><a href="#cb137-561"></a></span>
<span id="cb137-562"><a href="#cb137-562"></a><span class="ss">* </span>A tibble is a relatively new addition to the R language and forms part of the **tidyverse** suite of packages. They are designed to be more user-friendly than traditional data frames and have several additional features, such as more informative error messages, stricter data input and output rules, and better handling of <span class="in">`NA`</span>.</span>
<span id="cb137-563"><a href="#cb137-563"></a></span>
<span id="cb137-564"><a href="#cb137-564"></a><span class="ss">* </span>Unlike a dataframe, a tibble never automatically converts strings to factors or changes column names, which can help avoid unexpected behavior when working with the data.</span>
<span id="cb137-565"><a href="#cb137-565"></a></span>
<span id="cb137-566"><a href="#cb137-566"></a><span class="ss">* </span>A tibble does not have row names.</span>
<span id="cb137-567"><a href="#cb137-567"></a></span>
<span id="cb137-568"><a href="#cb137-568"></a><span class="ss">* </span>A tibble has a slightly different and more compact printing method than a dataframe, which makes them easier to read and work with.</span>
<span id="cb137-569"><a href="#cb137-569"></a></span>
<span id="cb137-570"><a href="#cb137-570"></a><span class="ss">* </span>Finally, a tibble has better performance than dataframes for many tasks, especially when working with large datasets.</span>
<span id="cb137-571"><a href="#cb137-571"></a></span>
<span id="cb137-572"><a href="#cb137-572"></a>While a dataframe is a core data structure in R, a tibble provides additional functionality and are becoming increasingly popular among R users, particularly those working with **tidyverse** packages. Applying the <span class="in">`class()`</span> function to a tibble revelas that it belongs to the classes <span class="in">`tbl_df`</span>, <span class="in">`tbl`</span> and  <span class="in">`data.frame`</span>.</span>
<span id="cb137-573"><a href="#cb137-573"></a></span>
<span id="cb137-574"><a href="#cb137-574"></a>We can convert our dataframe <span class="in">`my_dataframe`</span> to a tibble, and present the output with the     <span class="in">`print()`</span> function that applies nicely to tibbles:</span>
<span id="cb137-575"><a href="#cb137-575"></a></span>
<span id="cb137-578"><a href="#cb137-578"></a><span class="in">```{r}</span></span>
<span id="cb137-579"><a href="#cb137-579"></a><span class="co">#| warning: false</span></span>
<span id="cb137-580"><a href="#cb137-580"></a><span class="fu">library</span>(tidyverse) <span class="co"># we need to load the tidyverse package</span></span>
<span id="cb137-581"><a href="#cb137-581"></a>my_tibble <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(my_dataframe)</span>
<span id="cb137-582"><a href="#cb137-582"></a><span class="fu">class</span>(my_tibble)</span>
<span id="cb137-583"><a href="#cb137-583"></a><span class="fu">print</span>(my_tibble)</span>
<span id="cb137-584"><a href="#cb137-584"></a><span class="in">```</span></span>
<span id="cb137-585"><a href="#cb137-585"></a></span>
<span id="cb137-586"><a href="#cb137-586"></a>This very simple tibble looks identical to a dataframe, but as we start using more complex sets of data you'll learn to appreciate the small convenience that tibbles offer.</span>
<span id="cb137-587"><a href="#cb137-587"></a></span>
<span id="cb137-588"><a href="#cb137-588"></a><span class="fu">## `list`</span></span>
<span id="cb137-589"><a href="#cb137-589"></a></span>
<span id="cb137-590"><a href="#cb137-590"></a>This is also not actually a data class, but rather another way of representing a collection of objects of different types, all the way from numerical vectors to dataframes. Lists are useful for storing complex data structures and can also be accessed using indexing.</span>
<span id="cb137-591"><a href="#cb137-591"></a></span>
<span id="cb137-592"><a href="#cb137-592"></a>As an example, we create another dataframe:</span>
<span id="cb137-593"><a href="#cb137-593"></a></span>
<span id="cb137-596"><a href="#cb137-596"></a><span class="in">```{r}</span></span>
<span id="cb137-597"><a href="#cb137-597"></a>dates <span class="ot">&lt;-</span> <span class="fu">as.Date</span>(<span class="fu">c</span>(<span class="st">"2022-01-01"</span>, <span class="st">"2022-01-02"</span>, <span class="st">"2022-01-03"</span>,</span>
<span id="cb137-598"><a href="#cb137-598"></a>                   <span class="st">"2022-01-04"</span>, <span class="st">"2022-01-05"</span>))</span>
<span id="cb137-599"><a href="#cb137-599"></a></span>
<span id="cb137-600"><a href="#cb137-600"></a><span class="co"># Create a vector of numeric data</span></span>
<span id="cb137-601"><a href="#cb137-601"></a>numeric_data <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="at">n =</span> <span class="dv">5</span>, <span class="at">mean =</span> <span class="dv">1</span>, <span class="at">sd =</span> <span class="dv">1</span>)</span>
<span id="cb137-602"><a href="#cb137-602"></a></span>
<span id="cb137-603"><a href="#cb137-603"></a><span class="co"># Create a vector of categorical data</span></span>
<span id="cb137-604"><a href="#cb137-604"></a>categorical_data <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"C"</span>, <span class="st">"D"</span>, <span class="st">"D"</span>, <span class="st">"F"</span>, <span class="st">"A"</span>)</span>
<span id="cb137-605"><a href="#cb137-605"></a></span>
<span id="cb137-606"><a href="#cb137-606"></a><span class="co"># Combine the vectors into a data.frame</span></span>
<span id="cb137-607"><a href="#cb137-607"></a>my_other_dataframe <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">dates =</span> dates,</span>
<span id="cb137-608"><a href="#cb137-608"></a>                                  <span class="at">numeric_data =</span> numeric_data,</span>
<span id="cb137-609"><a href="#cb137-609"></a>                                  <span class="at">categorical_data =</span> categorical_data)</span>
<span id="cb137-610"><a href="#cb137-610"></a></span>
<span id="cb137-611"><a href="#cb137-611"></a>my_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">A =</span> my_dataframe,</span>
<span id="cb137-612"><a href="#cb137-612"></a>                <span class="at">B =</span> my_other_dataframe)</span>
<span id="cb137-613"><a href="#cb137-613"></a>my_list</span>
<span id="cb137-614"><a href="#cb137-614"></a><span class="fu">class</span>(my_list)</span>
<span id="cb137-615"><a href="#cb137-615"></a><span class="fu">str</span>(my_list)</span>
<span id="cb137-616"><a href="#cb137-616"></a><span class="in">```</span></span>
<span id="cb137-617"><a href="#cb137-617"></a></span>
<span id="cb137-618"><a href="#cb137-618"></a>We can access one of the dataframes is the list as follows:</span>
<span id="cb137-619"><a href="#cb137-619"></a></span>
<span id="cb137-622"><a href="#cb137-622"></a><span class="in">```{r}</span></span>
<span id="cb137-623"><a href="#cb137-623"></a>my_list[[<span class="dv">2</span>]]</span>
<span id="cb137-624"><a href="#cb137-624"></a>my_list[[<span class="st">"A"</span>]]</span>
<span id="cb137-625"><a href="#cb137-625"></a><span class="in">```</span></span>
<span id="cb137-626"><a href="#cb137-626"></a></span>
<span id="cb137-627"><a href="#cb137-627"></a>To access a variable within one of the elements of the list we can do something like:</span>
<span id="cb137-628"><a href="#cb137-628"></a></span>
<span id="cb137-631"><a href="#cb137-631"></a><span class="in">```{r}</span></span>
<span id="cb137-632"><a href="#cb137-632"></a>my_list[[<span class="st">"B"</span>]]<span class="sc">$</span>numeric_data</span>
<span id="cb137-633"><a href="#cb137-633"></a><span class="in">```</span></span>
<span id="cb137-634"><a href="#cb137-634"></a></span>
<span id="cb137-635"><a href="#cb137-635"></a><span class="fu"># Conclusion</span></span>
<span id="cb137-636"><a href="#cb137-636"></a></span>
<span id="cb137-637"><a href="#cb137-637"></a>These are the bread and butter data classes and structures in R. Other data classes and structures also exist, but these may be particular to certain packages. We will encounter some of them in <span class="co">[</span><span class="ot">BCB743: Quantitative Ecology</span><span class="co">](../../BCB743/BCB743_index.qmd)</span>. </span>
</code></pre></div><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>© 2025, AJ Smit</p>
</div>   
    <div class="nav-footer-center">

<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB744/intro_r/03-data-in-R.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>


<script src="../../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>