<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.554">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="AJ Smit">
<meta name="dcterms.date" content="2021-01-01">
<title>The Tangled Bank - 3. R workflows</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../BCB744/intro_r/04-graphics.html" rel="next">
<link href="../../BCB744/intro_r/02-working-with-data.html" rel="prev">
<link href="../../images/favicon.ico" rel="icon">
<script src="../../site_libs/cookie-consent/cookie-consent.js"></script>
<link href="../../site_libs/cookie-consent/cookie-consent.css" rel="stylesheet">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script async="" src="https://www.googletagmanager.com/gtag/js?id=G-SCYH68M489"></script><script type="text/plain" cookie-consent="tracking">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-SCYH68M489', { 'anonymize_ip': true});
</script><script type="text/javascript" charset="UTF-8">
document.addEventListener('DOMContentLoaded', function () {
cookieconsent.run({
  "notice_banner_type":"headline",
  "consent_type":"express",
  "palette":"dark",
  "language":"en",
  "page_load_consent_levels":["strictly-necessary"],
  "notice_banner_reject_button_hide":false,
  "preferences_center_close_button_hide":false,
  "website_name":""
  ,
"language":"en"
  });
});
</script><style>html{ scroll-behavior: smooth; }</style>
<script async="" src="https://hypothes.is/embed.js"></script><script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script><script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>
<meta property="og:title" content="The Tangled Bank - 3. R workflows">
<meta property="og:description" content="Getting familiar with R and RStudio">
<meta property="og:image" content="https://tangledbank.netlify.app/BCB744/intro_r/data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg==">
<meta property="og:site_name" content="The Tangled Bank">
</head>
<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">The Tangled Bank</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-vignettes" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Vignettes</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-vignettes">
<li>
    <a class="dropdown-item" href="../../vignettes/netCDF_dates.html">
 <span class="dropdown-text">Dates From netCDF Files: Two Approaches</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/prep_NOAA_OISST.html">
 <span class="dropdown-text">Downloading and Prep: ERDDAP</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_ERDDAP.html">
 <span class="dropdown-text">Retrieving Chlorophyll-a Data from ERDDAP Servers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/alt_method.html">
 <span class="dropdown-text">Downloading and Prep: Python + CDO</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/gridded_data.html">
 <span class="dropdown-text">Detecting Events in Gridded Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/regridding.html">
 <span class="dropdown-text">Regridding gridded data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/MHW_MCS_horizonplots.html">
 <span class="dropdown-text">Displaying MHWs and MCSs as Horizon Plots</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/heatwaveR_issues.html">
 <span class="dropdown-text">heatwaveR Issues</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_sightings.html">
 <span class="dropdown-text">Plotting the Whale Sightings and Chlorophyll-<em>a</em> Concentrations</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/chl_localisation.html">
 <span class="dropdown-text">Spatial Localisation, Subsetting, and Aggregation of the Chlorophyll-<em>a</em> Data</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/elem_ts_methods.html">
 <span class="dropdown-text">Wavelet Analysis of Diatom Time Series</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/buffer_data_extract.html">
 <span class="dropdown-text">Extracting gridded data within a buffer</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-high-performace-computing" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">High-Performace Computing</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-high-performace-computing">
<li>
    <a class="dropdown-item" href="../../vignettes/README_Lengau.html">
 <span class="dropdown-text">Using Lengau</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/PBSPro_users.html">
 <span class="dropdown-text">PBSPro user and job management</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../vignettes/README_tmux.html">
 <span class="dropdown-text">Using tmux</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item">
    <a class="nav-link" href="../../blog.html"> 
<span class="menu-text">Tangled Bank Blog</span></a>
  </li>  
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-blogroll" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Blogroll</span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-blogroll">
<li>
    <a class="dropdown-item" href="https://www.r-bloggers.com">
 <span class="dropdown-text">R-bloggers</span></a>
  </li>  
    </ul>
</li>
  <li class="nav-item compact">
    <a class="nav-link" href="../../blog.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bi-github" role="button" data-bs-toggle="dropdown" aria-expanded="false">
      <i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span>
    </a>
    <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="nav-menu-bi-github">
<li>
    <a class="dropdown-item" href="https://github.com/ajsmit/tangled_bank">
 <span class="dropdown-text">Source Code</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://bugs.com">
 <span class="dropdown-text">Report a Bug</span></a>
  </li>  
    </ul>
</li>
</ul>
</div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../BCB744/BCB744_index.html">BCB744 (Intro R)</a></li><li class="breadcrumb-item"><a href="../../BCB744/intro_r/03-workflow.html">3. R workflows</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-full page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto"><div class="pt-lg-2 mt-2 text-end sidebar-header">
      <a href="../../index.html" class="sidebar-logo-link">
      <img src="../../images/hex_logo.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none"></a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
 <span class="menu-text">☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">UNDERGRADUATE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false">
 <span class="menu-text">BDC223</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pages/BDC223_FAQ.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">FAQ</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false">
 <span class="menu-text">BDC334</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/BDC334_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BDC334: Macroecology and Global Biogeography</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/01-introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 1: Ecological Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/02a-r_rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2a: R and RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/02b-env_dist.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 2b: Environmental Distance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/03-biodiversity1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 3: Quantifying Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BDC334/04-biodiversity2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Lab 4: Describing Biodiversity Patterns</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../pages/Transboundary_systems.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Transboundary systems</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS CORE</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">BCB744 (Intro R)</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/BCB744_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB744: Introduction to R, and Biostatistics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/01-RStudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. R and RStudio</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/02-working-with-data.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Working with data and code</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/03-workflow.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">3. R workflows</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/04-graphics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Graphics with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/05-faceting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Faceting figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/06-brewing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Brewing colours</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/07-mapping.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Mapping with ggplot2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/08-mapping_style.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Mapping with style</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/09-mapping_rnaturalearth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Mapping with Natural Earth and the <strong>sf</strong> package</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/10-mapping_quakes.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. The Fiji Earthquake data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/11-tidy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. Tidy data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/12-tidier.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Tidier data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/13-tidiest.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Tidiest data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/14-recap.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Recap</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/15-functions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15. Functions by chapter</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/16-base_r.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16. Base R primer</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/intro_r/17-dates.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">17. Dates</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false">
 <span class="menu-text">BCB744 (Biostatistics)</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/BCB744_scientific_philosophy.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Biostatistics and the philosophy of scientific inquiry</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/01-data-in-R.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Data classes and structures in R</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/02-summarise-and-describe.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Exploring with summaries and descriptions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/03-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Exploring with figures</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/04-distributions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. Data distributions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/05-inference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Statistical inference and hypothesis testing</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/06-assumptions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Assumptions for parametric statistics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/07-t_tests.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. <em>t</em>-tests</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/08-anova.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8. Analysis of Variance (ANOVA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/09-regressions.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Simple linear regressions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/10-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Correlations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/11-decision_guide.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11. A guide to selecting the right parametric test</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/12-glance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12. Non-parametric statistical tests at a glance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/13-confidence.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Confidence intervals</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB744/basic_stats/14-transformations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Data transformations</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">HONOURS ELECTIVES</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false">
 <span class="menu-text">BCB743 (Quantitative Ecology)</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/BCB743_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">BCB743: Quantitative Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/00-review.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1-4. Review Biogeography and Global Ecology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/05-correlations.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. Correlations and Associations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/06-deep_dive.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">6. Deep Dive into Gradients</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/dis-metrics.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Useful distance and dissimilarities metrics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/07-ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">7. Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/08-PCA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8a. Principal Component Analysis (PCA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/08-PCA_examples.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8b. PCA Additional Examples</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/08-PCA_SDG.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">8c. PCA of WHO SDGs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/09-CA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">9. Correspondence Analysis (CA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/10-PCoA.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10. Principal Coordinate Analysis (PCoA)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/11-nMDS.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11a. non-Metric Multidimensional Scaling (nMDS)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/11-nMDS_diatoms.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11b. nMDS of Mayombo’s Diatom Data</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/unconstrained-summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Summary: Unconstrained Ordinations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/12a-constrained_ordination.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12a. Constrained Ordination</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/Seaweed_in_Two_Oceans_v2/12b-two_oceans_appendices.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12b. Seaweeds in Two Oceans: Beta-diversity (Appendices)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/13-cluster_analysis.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13. Cluster Analysis</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../BCB743/14-multiple_regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">14. Multiple Linear Regression (MLR)</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘︎</span>
  </li>
        <li class="sidebar-item">
 <span class="menu-text">SUPPORT</span>
  </li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false">
 <span class="menu-text">Web Resources</span></a>
          <a class="sidebar-item-toggle text-start collapsed" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-6" aria-expanded="false" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-6" class="collapse list-unstyled sidebar-section depth1 ">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/general_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/ecology_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Quantitative Ecology Resources</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../resources/spatial_resources_web.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Spatial R Resources</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item">
 <span class="menu-text">☘︎</span>
  </li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#r-scripts" id="toc-r-scripts" class="nav-link active" data-scroll-target="#r-scripts">R Scripts</a></li>
  <li>
<a href="#reading-data-into-r" id="toc-reading-data-into-r" class="nav-link" data-scroll-target="#reading-data-into-r">Reading data into R</a>
  <ul class="collapse">
<li><a href="#preparing-data-for-r" id="toc-preparing-data-for-r" class="nav-link" data-scroll-target="#preparing-data-for-r">Preparing data for R</a></li>
  <li><a href="#converting-data" id="toc-converting-data" class="nav-link" data-scroll-target="#converting-data">Converting data</a></li>
  <li><a href="#importing-data" id="toc-importing-data" class="nav-link" data-scroll-target="#importing-data">Importing data</a></li>
  <li><a href="#loading-a-file" id="toc-loading-a-file" class="nav-link" data-scroll-target="#loading-a-file">Loading a file</a></li>
  </ul>
</li>
  <li>
<a href="#working-with-data" id="toc-working-with-data" class="nav-link" data-scroll-target="#working-with-data">Working with data</a>
  <ul class="collapse">
<li><a href="#examine-your-data" id="toc-examine-your-data" class="nav-link" data-scroll-target="#examine-your-data">Examine your data</a></li>
  <li><a href="#tidyverse-sneak-peek" id="toc-tidyverse-sneak-peek" class="nav-link" data-scroll-target="#tidyverse-sneak-peek"><strong>Tidyverse</strong> sneak peek</a></li>
  <li><a href="#subsetting" id="toc-subsetting" class="nav-link" data-scroll-target="#subsetting">Subsetting</a></li>
  <li><a href="#basic-stats" id="toc-basic-stats" class="nav-link" data-scroll-target="#basic-stats">Basic stats</a></li>
  </ul>
</li>
  <li>
<a href="#summary-of-all-variables-in-a-dataframe" id="toc-summary-of-all-variables-in-a-dataframe" class="nav-link" data-scroll-target="#summary-of-all-variables-in-a-dataframe">Summary of all variables in a dataframe</a>
  <ul class="collapse">
<li><a href="#summary-statistics-by-variable" id="toc-summary-statistics-by-variable" class="nav-link" data-scroll-target="#summary-statistics-by-variable">Summary statistics by variable</a></li>
  <li><a href="#more-complex-calculations" id="toc-more-complex-calculations" class="nav-link" data-scroll-target="#more-complex-calculations">More complex calculations</a></li>
  </ul>
</li>
  <li><a href="#saving-data" id="toc-saving-data" class="nav-link" data-scroll-target="#saving-data">Saving data</a></li>
  <li><a href="#visualisations" id="toc-visualisations" class="nav-link" data-scroll-target="#visualisations">Visualisations</a></li>
  <li><a href="#clearing-the-memory" id="toc-clearing-the-memory" class="nav-link" data-scroll-target="#clearing-the-memory">Clearing the memory</a></li>
  <li><a href="#working-directories" id="toc-working-directories" class="nav-link" data-scroll-target="#working-directories">Working directories</a></li>
  <li><a href="#help" id="toc-help" class="nav-link" data-scroll-target="#help">Help</a></li>
  <li><a href="#other-data" id="toc-other-data" class="nav-link" data-scroll-target="#other-data">Other data</a></li>
  <li><a href="#r-markdown" id="toc-r-markdown" class="nav-link" data-scroll-target="#r-markdown">R Markdown</a></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info">Session info</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB744/intro_r/03-workflow.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content column-body" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../BCB744/BCB744_index.html">BCB744 (Intro R)</a></li><li class="breadcrumb-item"><a href="../../BCB744/intro_r/03-workflow.html">3. R workflows</a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">3. R workflows</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
<p class="subtitle lead">Getting familiar with R and RStudio</p>
</div>


<div class="quarto-title-meta-author column-body">
  <div class="quarto-title-meta-heading">Author</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author"><a href="https://ajsmit.netlify.app">AJ Smit</a> <a href="https://orcid.org/0000-0002-3799-6126" class="quarto-title-author-orcid"> <img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA2ZpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bWxuczp4bXA9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC8iIHhtcE1NOk9yaWdpbmFsRG9jdW1lbnRJRD0ieG1wLmRpZDo1N0NEMjA4MDI1MjA2ODExOTk0QzkzNTEzRjZEQTg1NyIgeG1wTU06RG9jdW1lbnRJRD0ieG1wLmRpZDozM0NDOEJGNEZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wTU06SW5zdGFuY2VJRD0ieG1wLmlpZDozM0NDOEJGM0ZGNTcxMUUxODdBOEVCODg2RjdCQ0QwOSIgeG1wOkNyZWF0b3JUb29sPSJBZG9iZSBQaG90b3Nob3AgQ1M1IE1hY2ludG9zaCI+IDx4bXBNTTpEZXJpdmVkRnJvbSBzdFJlZjppbnN0YW5jZUlEPSJ4bXAuaWlkOkZDN0YxMTc0MDcyMDY4MTE5NUZFRDc5MUM2MUUwNEREIiBzdFJlZjpkb2N1bWVudElEPSJ4bXAuZGlkOjU3Q0QyMDgwMjUyMDY4MTE5OTRDOTM1MTNGNkRBODU3Ii8+IDwvcmRmOkRlc2NyaXB0aW9uPiA8L3JkZjpSREY+IDwveDp4bXBtZXRhPiA8P3hwYWNrZXQgZW5kPSJyIj8+84NovQAAAR1JREFUeNpiZEADy85ZJgCpeCB2QJM6AMQLo4yOL0AWZETSqACk1gOxAQN+cAGIA4EGPQBxmJA0nwdpjjQ8xqArmczw5tMHXAaALDgP1QMxAGqzAAPxQACqh4ER6uf5MBlkm0X4EGayMfMw/Pr7Bd2gRBZogMFBrv01hisv5jLsv9nLAPIOMnjy8RDDyYctyAbFM2EJbRQw+aAWw/LzVgx7b+cwCHKqMhjJFCBLOzAR6+lXX84xnHjYyqAo5IUizkRCwIENQQckGSDGY4TVgAPEaraQr2a4/24bSuoExcJCfAEJihXkWDj3ZAKy9EJGaEo8T0QSxkjSwORsCAuDQCD+QILmD1A9kECEZgxDaEZhICIzGcIyEyOl2RkgwAAhkmC+eAm0TAAAAABJRU5ErkJggg=="></a></p>
  </div>
  <div class="quarto-title-meta-contents">
        <p class="affiliation">
            <a href="https://uwc.ac.za">
            University of the Western Cape
            </a>
          </p>
      </div>
  </div>

<div class="quarto-title-meta column-body">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">January 1, 2021</p>
    </div>
  </div>
  
    
  </div>
  


</header><p><img src="../../images/Skulls_meme.jpg" class="img-fluid" style="width:60.0%"></p>
<blockquote class="blockquote">
<p>“<em>A dream doesn’t become reality through magic; it takes sweat, determination and hard work.</em>”</p>
<p>— Colin Powell</p>
</blockquote>
<blockquote class="blockquote">
<p>“<em>Choose a job you love, and you will never have to work a day in your life.</em>”</p>
<p>— Confucius</p>
</blockquote>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Exercise in frustration
</div>
</div>
<div class="callout-body-container callout-body">
<p>This exercise requires that you analyse the data provided in <code>data/SACTN_SAWS.csv</code> within <code>&lt;that which shall not be named&gt;</code>.</p>
<p>This dataset contains monthly seawater temperature time series for a selection of sites around the coast of South Africa.</p>
<ol type="1">
<li>How many sites are there?</li>
</ol>
<p>For each of the sites there are monthly mean temperatures for every year the 1970s to the 2000s.</p>
<ol start="2" type="1">
<li><p>What is the earliest and latest date for which temperature measurements are available, and which sites do they belong to?</p></li>
<li><p>In this exercise, please create, <strong>for each site</strong>, a <em>monthly climatology</em> using all available data in the time series (see box below).</p></li>
<li><p>Plot the monthly climatology for each site as a line graph showing the mean temperature for each month.</p></li>
<li><p>Optionally, you could also calculate the standard deviation of the monthly mean temperatures to represent the variability in temperature within each month. This information can be used to determine the range of typical temperatures for each month.</p></li>
</ol>
</div>
</div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
WMO Guidelines for calculating climatologies
</div>
</div>
<div class="callout-body-container callout-body">
<p>The World Meteorological Organization (WMO) provides guidelines for the calculation of climatologies, including monthly and annual climatologies. The WMO guidelines are based on standard meteorological practices and are designed to ensure consistent and accurate calculation of climatological data.</p>
<p>A monthly climatology is a study of the average temperature of seawater (or any environmental variable that is expected to vary over time in a cyclical seasonal manner) over a specified period of time, typically over a year, and broken down by month. Climatologies provide insight into the patterns and trends of weather phenomena and how they vary from one month to another and from year to year.</p>
<p>By analysing long-term data, a monthly climatology can determine the average seawater temperature for each month, the range of temperatures that are typical for each month, as well as finding deviations from expected patterns. This information can be used for a variety of purposes, such as predicting weather patterns, studying the effects of climate change, or understanding the impacts of ocean temperature on marine life.</p>
<p>In essence, a monthly climatology provides a comprehensive and organised overview of the variability and patterns in seawater temperature over a given period of time.</p>
<p>The steps for calculating a monthly climatology according to the WMO guidelines are as follows:</p>
<ul>
<li><p><strong>Obtain long-term meteorological observations:</strong> The WMO recommends using at least 30 years of data, although longer periods are preferred, to ensure the climatology is representative of the long-term average conditions.</p></li>
<li><p><strong>Group the observations into months:</strong> The observations should be grouped into months, such as January, February, etc.</p></li>
<li><p><strong>Calculate the mean value for each month:</strong> For each month over the observation period, the mean value should be calculated as the sum of all the observations for that month divided by the number of observations.</p></li>
<li><p><strong>Calculate the mean climatology for each month:</strong> The mean climatology for each month is calculated by taking the mean of the monthly mean values for each month over the period of record.</p></li>
<li><p><strong>Quality control and homogeneity adjustment:</strong> The WMO recommends checking the quality and consistency of the data, and adjusting the data if necessary to correct for inhomogeneities or biases.</p></li>
</ul>
<p>The steps for calculating an annual climatology are similar, with the observations being grouped into years rather than months. The WMO guidelines also provide recommendations for the calculation of other types of climatologies, such as seasonal and daily climatologies.</p>
<p>The WMO guidelines are not mandatory, but they are widely used and respected in the meteorological community as the standard for calculating climatologies.</p>
</div>
</div>
<section id="r-scripts" class="level1"><h1>R Scripts</h1>
<p>The first step for any project in R is to create a new script. You do this by clicking on the ‘New Document’ button (in the top left and selecting ‘R Script’). This creates an unnamed file in the <strong>Source Editor</strong> pane. Best to save it first of all so you do not lose what you do. ‘File’/‘Save As’/ and the Working Directory should come up. Type in <code>Day_1</code> as the file name and click ‘save.’ R will automatically add a <code>.R</code> extension.</p>
<p>It is recommended that you start your script with some basic information for you to refer back to later. Start with a comment line (the line begins with a <code>#</code>) that tells you the name of the script, something about the script, who created it, and the date it was created. In the source editor enter to following lines and save the file again:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># Day_1.R</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="co"># Reads in some data about Laminaria collected along the Cape Peninsula</span></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="co"># Do various data manipulations, analyses and graphs</span></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="co"># &lt;your_name&gt;</span></span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="co"># &lt;curent_date&gt;</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Remember that anything appearing after the <code>#</code> is not executed by R as script and is a comment.</p>
<p>It is recommend that for each day of the workshop you start a new script (in the <strong>Source Editor</strong>), type in the code as you go along, and only execute the required lines. That way you will have a record of what you have done.</p>
<p>Below, you will learn how to import the file <code>laminaria.csv</code> into R, assign it to a dataframe named <code>laminaria</code>, and spend a while looking it over. These data reflect results of a sampling campaign on one of the species of kelps (<em>Laminaria pallida</em>) in the Western Cape designed to find the morphometric properties of populations at different sites. We visited 13 different locations along the Cape Peninsula (<code>site</code>), and at each site, collected <em>ca.</em> 13 specimens of the largest kelps we could find. We then brought the kelps back to the shore and measured/calculated nine morphometric properties of the plants (e.g.&nbsp;the mass of the fronds (<code>blade_weight</code>), the frond length (<code>blade_length</code>), <em>etc.</em>).</p>
</section><section id="reading-data-into-r" class="level1"><h1>Reading data into R</h1>
<p>You will now see how easy it is to read data into R. R will read in many types of data, including spreadsheets, text files, binary files and files from other statistical packages and software.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Full stops
</div>
</div>
<div class="callout-body-container callout-body">
<p>Unfortunately in South Africa you are taught from a young age to use commas (<code>,</code>) instead of full stops (<code>.</code>) for decimal places. This simply will not do when you are working with a computer. You must always use a full stop for a decimal place and never insert commas anywhere into any numbers.</p>
</div>
</div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Commas
</div>
</div>
<div class="callout-body-container callout-body">
<p>R generally thinks that commas mean the user is telling the computer to separate values. So if you think you are typing a big number like 2,300 you may actually end up with two numbers. Never use commas with numbers.</p>
</div>
</div>
<section id="preparing-data-for-r" class="level2"><h2 class="anchored" data-anchor-id="preparing-data-for-r">Preparing data for R</h2>
<p>Importing data can actually take longer than the statistical analysis itself! In order to avoid as much frustration as possible it is important to remember that for R to be able to analyse your data they need to be in a consistent format, with each variable in a column and each sample in a row. The format within each variable (column) needs to be consistent and is commonly one of the following types: a continuous numeric variable (e.g., fish length (m): <code>0.133</code>, <code>0.145</code>); a factor or categorical variable (e.g., Month: <code>Jan</code>, <code>Feb</code> or <code>1</code>, <code>2</code>, <code>…</code>, <code>12</code>); a nominal variable (e.g., algal colour: <code>red</code>, <code>green</code>, <code>brown</code>); or a logical variable (i.e., <code>TRUE</code> or <code>FALSE</code>). You can also use other more specific formats such as dates and times, and more general text formats.</p>
<p>You will learn more about working with data in R — specifically, you will teach you about the <em>tidyverse</em> principles and the distinction between <em>long</em> and <em>wide</em> format data in more detail on Day 4. For most of our work in R you require our data to be in the long format, but Excel users (poor things!) are more familiar with data stored in the wide format. For now let’s bring some data into R and not worry too much about the data being tidy.</p>
</section><section id="converting-data" class="level2"><h2 class="anchored" data-anchor-id="converting-data">Converting data</h2>
<p>Before you can read in the <em>Laminaria</em> dataset provided for the following exercises, you need to convert the Excel file supplied into a <code>.csv</code> file. Open ‘laminaria.xlsx’ in Excel, then select ‘Save As’ from the File menu. In the ‘Format’ drop-down menu, select the option called ‘Comma Separated Values’, then hit ‘Save’. You’ll get a warning that formatting will be removed and that only one sheet will be exported; simply ‘Continue’. Your working directory should now contain a file called <code>laminaria.csv</code>.</p>
</section><section id="importing-data" class="level2"><h2 class="anchored" data-anchor-id="importing-data">Importing data</h2>
<p>The easiest way to import data into R is by changing your working directory to be the same as the file path where the file(s) are you want to load. A file path is effectively an address. In most operating systems, if you open the folder where your files are you may click on the navigation bar and it will show you the complete file path. Many people develop the nasty habit of squirrelling away their files within folders within folders within folders within folders… within folders within folders. Please don’t do that.</p>
<p>The concept of file paths is either one that you are familiar with, or you’ve never heard of before. There tends to be little middle ground. Happily, RStudio allows us to circumvent this issue. You do this by using the <code>Intro_R_Workshop.Rproj</code> that you may find in the files downloaded for this workshop. If you have not already switched to the <code>Intro_R_Workshop.Rproj</code> as outlined in Chapter 2, click on the project button in the top right corner your RStudio window. Then navigate to where you saved <code>Intro_R_Workshop.Rproj</code> and select it. Notice that your RStudio has changed a bit and all of the objects you may have previously created in your environment have been removed and any tabs in the source editor pane have been closed. That is fine for now, but it may mean you need to re-open the <code>Day_1.R</code> script you just created.</p>
<p>Once you have the working directory set, either by doing it manually with <code><a href="https://rdrr.io/r/base/getwd.html">setwd()</a></code> or by loading a project, R will now know where to look for the files you want to read. The function <code>read_csv()</code> is the most convenient way to read in raw data. There are several other ways to read in data, but for the purposes of this workshop we’ll stick to this one, for now. To find out what it does, you will go to its help entry in the usual way (i.e.&nbsp;<code>?read_csv</code>).</p>
<p>All R Help items are in the same format. A short <em>Description</em> (of what it does), <em>Usage</em>, <em>Arguments</em> (the different inputs it requires), <em>Details</em> (of what it does), <em>Value</em> (what it returns) and <em>Examples</em>. Arguments (the parameters that are passed to the function) are the lifeblood of any function, as this is how you provide information to R. You do not need to specify all arguments, as most have appropriate default values for your requirements, and others might not be needed for your particular case.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Data formats
</div>
</div>
<div class="callout-body-container callout-body">
<p>R has pedantic requirements for naming variables. It is safest to not use spaces, special characters (e.g., commas, semicolons, any of the shift characters above the numbers), or function names (e.g., mean). One can use ‘camelCase’, such as <code>myFirstVariable</code>, or simply separate the ‘parts’ of the variable name using an underscore such as in <code>my_first_variable</code>. Always make sure to use meaningful names; eventually you will learn to find a balance between meaningfulness and something short that’s easy enough to retype repeatedly (although R’s ability to use tab completion helps with not having to type long names to often).</p>
</div>
</div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Import
</div>
</div>
<div class="callout-body-container callout-body">
<p><code>read_csv()</code> is simply a ‘wrapper’ (i.e., a command that modifies) a more basic command called <code>read_delim()</code>, which itself allows you to read in many types of files besides <code>.csv</code>. To find out more, type <code>?read_delim()</code>.</p>
</div>
</div>
</section><section id="loading-a-file" class="level2"><h2 class="anchored" data-anchor-id="loading-a-file">Loading a file</h2>
<p>To load the <code>laminaria.csv</code> file you created, and assign it to an object name in R, you will use the <code>read_csv()</code> function from the <strong>tidyverse</strong> package, so let’s make sure it is activated.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Depending on the version of Excel you are using, or perhaps the settings within it, the <code>laminaria.csv</code> file you created may be corrupted in different ways. Generally Excel likes to replace the <code>,</code> between columns in our .csv files with <code>;</code>. This may seem like a triviality but sadly it is not. Lucky for use, the <strong>tidyverse</strong> knows about this problem and they have made a plan. Please open your <code>laminaria.csv</code> file and look at which character is being used to separate columns. If it is <code>,</code> then you will load the data with <code>read_csv()</code>. If the columns are separated with <code>;</code> you will use <code>read_csv2()</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a><span class="co"># Run this if 'laminaria.csv` has columns separated by ','</span></span>
<span id="cb3-2"><a href="#cb3-2"></a>laminaria <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../../data/laminaria.csv"</span>)</span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="co"># Run this if 'laminaria.csv` has columns separated by ';'</span></span>
<span id="cb3-4"><a href="#cb3-4"></a>laminaria <span class="ot">&lt;-</span> <span class="fu">read_csv2</span>(<span class="st">"../../data/laminaria.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>If one clicks on the newly created <code>laminaria</code> object in the <strong>Environment</strong> pane it will open a new panel that shows the information as a spreadsheet. To go back to your script click the appropriate tab in the <strong>Source Editor</strong> pane. With these data loaded you may now perform analyses on them.</p>
<p>At any point when working in R, you can see exactly what objects are in memory in several ways. First, you can look at the <strong>Environment</strong> tab in RStudio, then <strong>Workspace Browser</strong>. Alternatively you can type either of the following:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="fu">ls</span>()</span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="co"># or</span></span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="fu">objects</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>You can delete an object from memory by specifying the <code><a href="https://rdrr.io/r/base/rm.html">rm()</a></code> function with the name of the object:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="fu">rm</span>(laminaria)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This will of course delete our variable, so you will import it in again using whichever of the following two lines of code matched our Excel situation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>laminaria <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../../data/laminaria.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Managing variables
</div>
</div>
<div class="callout-body-container callout-body">
<p>It is good practice to remove variables from memory that you are not using, especially if they are large.</p>
</div>
</div>
</section></section><section id="working-with-data" class="level1"><h1>Working with data</h1>
<section id="examine-your-data" class="level2"><h2 class="anchored" data-anchor-id="examine-your-data">Examine your data</h2>
<p>Once the data are in R, you need to check there are no glaring errors. It is useful to call up the first few lines of the dataframe using the function <code><a href="https://rdrr.io/r/utils/head.html">head()</a></code>. Try it yourself by typing:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="fu">head</span>(laminaria)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R&gt; # A tibble: 6 × 12
R&gt;   region site        Ind blade_weight blade_length blade_thickness stipe_mass
R&gt;   &lt;chr&gt;  &lt;chr&gt;     &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;           &lt;dbl&gt;      &lt;dbl&gt;
R&gt; 1 WC     Kommetjie     2         1.9           160            2          1.5 
R&gt; 2 WC     Kommetjie     3         1.5           120            1.4        2.25
R&gt; 3 WC     Kommetjie     4         0.55          110            1.5        1.15
R&gt; 4 WC     Kommetjie     5         1             159            1.5        2.6 
R&gt; 5 WC     Kommetjie     6         2.3           149            2         NA   
R&gt; 6 WC     Kommetjie     7         1.6           107            1.75       2.9 
R&gt; # ℹ 5 more variables: stipe_length &lt;dbl&gt;, stipe_diameter &lt;dbl&gt;, digits &lt;dbl&gt;,
R&gt; #   thallus_mass &lt;dbl&gt;, total_length &lt;dbl&gt;</code></pre>
</div>
</div>
<p>This lists the first six lines of each of the variables in the dataframe as a table. You can similarly retrieve the last six lines of a dataframe by an identical call to the function <code><a href="https://rdrr.io/r/utils/head.html">tail()</a></code>. Of course, this works better when you have fewer than 10 or so variables (columns); for larger data sets, things can get a little messy. If you want more or fewer rows in your head or tail, tell R how many rows it is you want by adding this information to your function call. Try typing:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="fu">head</span>(laminaria, <span class="at">n =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R&gt; # A tibble: 3 × 12
R&gt;   region site        Ind blade_weight blade_length blade_thickness stipe_mass
R&gt;   &lt;chr&gt;  &lt;chr&gt;     &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;           &lt;dbl&gt;      &lt;dbl&gt;
R&gt; 1 WC     Kommetjie     2         1.9           160             2         1.5 
R&gt; 2 WC     Kommetjie     3         1.5           120             1.4       2.25
R&gt; 3 WC     Kommetjie     4         0.55          110             1.5       1.15
R&gt; # ℹ 5 more variables: stipe_length &lt;dbl&gt;, stipe_diameter &lt;dbl&gt;, digits &lt;dbl&gt;,
R&gt; #   thallus_mass &lt;dbl&gt;, total_length &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="fu">tail</span>(laminaria, <span class="at">n =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R&gt; # A tibble: 2 × 12
R&gt;   region site         Ind blade_weight blade_length blade_thickness stipe_mass
R&gt;   &lt;chr&gt;  &lt;chr&gt;      &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;           &lt;dbl&gt;      &lt;dbl&gt;
R&gt; 1 WC     Rocky Bank    12          2.1          194             1.4       3.75
R&gt; 2 WC     Rocky Bank    13          1.3          160             1.9       2.45
R&gt; # ℹ 5 more variables: stipe_length &lt;dbl&gt;, stipe_diameter &lt;dbl&gt;, digits &lt;dbl&gt;,
R&gt; #   thallus_mass &lt;dbl&gt;, total_length &lt;dbl&gt;</code></pre>
</div>
</div>
<p>You can also check the structure of your data by using the <code>glimpse()</code> function:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="fu">glimpse</span>(laminaria)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R&gt; Rows: 140
R&gt; Columns: 12
R&gt; $ region          &lt;chr&gt; "WC", "WC", "WC", "WC", "WC", "WC", "WC", "WC", "WC", …
R&gt; $ site            &lt;chr&gt; "Kommetjie", "Kommetjie", "Kommetjie", "Kommetjie", "K…
R&gt; $ Ind             &lt;dbl&gt; 2, 3, 4, 5, 6, 7, 8, 10, 11, 1, 3, 4, 5, 6, 7, 8, 9, 1…
R&gt; $ blade_weight    &lt;dbl&gt; 1.90, 1.50, 0.55, 1.00, 2.30, 1.60, 0.65, 0.95, 2.30, …
R&gt; $ blade_length    &lt;dbl&gt; 160, 120, 110, 159, 149, 107, 104, 111, 178, 145, 146,…
R&gt; $ blade_thickness &lt;dbl&gt; 2.00, 1.40, 1.50, 1.50, 2.00, 1.75, 2.00, 1.25, 2.50, …
R&gt; $ stipe_mass      &lt;dbl&gt; 1.50, 2.25, 1.15, 2.60, NA, 2.90, 0.75, 1.60, 4.20, 0.…
R&gt; $ stipe_length    &lt;dbl&gt; 120, 149, 97, 167, 146, 161, 110, 136, 176, 82, 118, 1…
R&gt; $ stipe_diameter  &lt;dbl&gt; 56.0, 68.5, 69.0, 60.0, 73.0, 63.0, 51.0, 56.0, 76.0, …
R&gt; $ digits          &lt;dbl&gt; 12, 12, 13, 8, 15, 17, 11, 11, 8, 19, 20, 23, 20, 24, …
R&gt; $ thallus_mass    &lt;dbl&gt; 3000, 3750, 1700, 3600, 5100, 4500, 1400, 2550, 6500, …
R&gt; $ total_length    &lt;dbl&gt; 256, 269, 207, 326, 295, 268, 214, 247, 354, 227, 264,…</code></pre>
</div>
</div>
<p>This very handy function lists the variables in your dataframe by name, tells you what sorts of data are contained in each variable (e.g., continuous number, discrete factor) and provides an indication of the actual contents of each.</p>
<p>If you wanted only the names of the variables (columns) in the dataframe, you could use:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="fu">names</span>(laminaria)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R&gt;  [1] "region"          "site"            "Ind"             "blade_weight"   
R&gt;  [5] "blade_length"    "blade_thickness" "stipe_mass"      "stipe_length"   
R&gt;  [9] "stipe_diameter"  "digits"          "thallus_mass"    "total_length"</code></pre>
</div>
</div>
<p>Another option, but by no means the only one remaining, is to install a library called <strong>skimr</strong> and to use thew <code>skim()</code> function:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="fu">library</span>(skimr)</span>
<span id="cb17-2"><a href="#cb17-2"></a><span class="fu">skim</span>(iris) <span class="co"># using built-in `iris` data</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<caption>Data summary</caption>
<tbody>
<tr class="odd">
<td style="text-align: left;">Name</td>
<td style="text-align: left;">iris</td>
</tr>
<tr class="even">
<td style="text-align: left;">Number of rows</td>
<td style="text-align: left;">150</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Number of columns</td>
<td style="text-align: left;">5</td>
</tr>
<tr class="even">
<td style="text-align: left;">_______________________</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Column type frequency:</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: left;">factor</td>
<td style="text-align: left;">1</td>
</tr>
<tr class="odd">
<td style="text-align: left;">numeric</td>
<td style="text-align: left;">4</td>
</tr>
<tr class="even">
<td style="text-align: left;">________________________</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Group variables</td>
<td style="text-align: left;">None</td>
</tr>
</tbody>
</table>
<p><strong>Variable type: factor</strong></p>
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 17%">
<col style="width: 12%">
<col style="width: 17%">
<col style="width: 9%">
<col style="width: 11%">
<col style="width: 32%">
</colgroup>
<thead><tr class="header">
<th style="text-align: left;">skim_variable</th>
<th style="text-align: right;">n_missing</th>
<th style="text-align: right;">complete_rate</th>
<th style="text-align: left;">ordered</th>
<th style="text-align: right;">n_unique</th>
<th style="text-align: left;">top_counts</th>
</tr></thead>
<tbody><tr class="odd">
<td style="text-align: left;">Species</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: left;">FALSE</td>
<td style="text-align: right;">3</td>
<td style="text-align: left;">set: 50, ver: 50, vir: 50</td>
</tr></tbody>
</table>
<p><strong>Variable type: numeric</strong></p>
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 18%">
<col style="width: 13%">
<col style="width: 18%">
<col style="width: 6%">
<col style="width: 6%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 6%">
<col style="width: 5%">
<col style="width: 6%">
<col style="width: 7%">
</colgroup>
<thead><tr class="header">
<th style="text-align: left;">skim_variable</th>
<th style="text-align: right;">n_missing</th>
<th style="text-align: right;">complete_rate</th>
<th style="text-align: right;">mean</th>
<th style="text-align: right;">sd</th>
<th style="text-align: right;">p0</th>
<th style="text-align: right;">p25</th>
<th style="text-align: right;">p50</th>
<th style="text-align: right;">p75</th>
<th style="text-align: right;">p100</th>
<th style="text-align: left;">hist</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Sepal.Length</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5.84</td>
<td style="text-align: right;">0.83</td>
<td style="text-align: right;">4.3</td>
<td style="text-align: right;">5.1</td>
<td style="text-align: right;">5.80</td>
<td style="text-align: right;">6.4</td>
<td style="text-align: right;">7.9</td>
<td style="text-align: left;">▆▇▇▅▂</td>
</tr>
<tr class="even">
<td style="text-align: left;">Sepal.Width</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3.06</td>
<td style="text-align: right;">0.44</td>
<td style="text-align: right;">2.0</td>
<td style="text-align: right;">2.8</td>
<td style="text-align: right;">3.00</td>
<td style="text-align: right;">3.3</td>
<td style="text-align: right;">4.4</td>
<td style="text-align: left;">▁▆▇▂▁</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Petal.Length</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3.76</td>
<td style="text-align: right;">1.77</td>
<td style="text-align: right;">1.0</td>
<td style="text-align: right;">1.6</td>
<td style="text-align: right;">4.35</td>
<td style="text-align: right;">5.1</td>
<td style="text-align: right;">6.9</td>
<td style="text-align: left;">▇▁▆▇▂</td>
</tr>
<tr class="even">
<td style="text-align: left;">Petal.Width</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1.20</td>
<td style="text-align: right;">0.76</td>
<td style="text-align: right;">0.1</td>
<td style="text-align: right;">0.3</td>
<td style="text-align: right;">1.30</td>
<td style="text-align: right;">1.8</td>
<td style="text-align: right;">2.5</td>
<td style="text-align: left;">▇▁▇▅▃</td>
</tr>
</tbody>
</table>
</div>
</div>
</section><section id="tidyverse-sneak-peek" class="level2"><h2 class="anchored" data-anchor-id="tidyverse-sneak-peek">
<strong>Tidyverse</strong> sneak peek</h2>
<p>Before you begin to manipulate our data further I need to briefly introduce you to the <a href="https://www.tidyverse.org/"><strong>tidyverse</strong></a>. And no introduction can be complete within learning about the <em>pipe</em> command, <code>%&gt;%</code>. You may type this by pushing the following keys together: <strong>ctrl-shift-m</strong>. The pipe (<code>%&gt;%</code>, or <code>|&gt;</code> if you selected to use the native pipe operator under ‘Global Options’) allows you to perform calculations sequentially, which helps us to avoid making errors.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
The pipe operator
</div>
</div>
<div class="callout-body-container callout-body">
<p>The <strong>pipe operator</strong> allows you to take the output of one function and pass it directly as the input to the next function. This creates a more intuitive and readable way to string together a series of data operations. Instead of nesting functions inside one another, which can quickly become confusing and hard to read, the pipe operator lets you lay out your data processing steps sequentially. This makes your code cleaner and easier to understand, as it clearly outlines the workflow from start to finish, almost like a step-by-step recipe for your data analysis.</p>
</div>
</div>
<p>The pipe works best in tandem with the following five common functions:</p>
<ul>
<li>Arrange observations (rows) with <code>arrange()</code>
</li>
<li>Filter observations (rows) with <code><a href="https://rdrr.io/r/stats/filter.html">filter()</a></code>
</li>
<li>Select variables (columns) with <code>select()</code>
</li>
<li>Create new variables (columns) with <code>mutate()</code>
</li>
<li>Summarise variables (columns) with <code>summarise()</code>
</li>
<li>Group observations (rows) with <code>group_by()</code>
</li>
</ul>
<p>You will cover these functions in more detail on Day 4. For now you will ease ourselves into the code with some simple examples.</p>
</section><section id="subsetting" class="level2"><h2 class="anchored" data-anchor-id="subsetting">Subsetting</h2>
<p>Now let’s have a look at specific parts of the data. You will likely need to do this in almost every script you write. If you want to refer to a variable, you specify the dataframe then the column name within the <code>select()</code> function. In your script type:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>laminaria <span class="sc">%&gt;%</span> <span class="co"># Tell R which dataframe you are using</span></span>
<span id="cb18-2"><a href="#cb18-2"></a>  <span class="fu">select</span>(site, total_length) <span class="co"># Select only specific columns</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>If you want to only select values from specific columns you insert one more line of code.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>laminaria <span class="sc">%&gt;%</span> </span>
<span id="cb19-2"><a href="#cb19-2"></a>  <span class="fu">select</span>(site, total_length) <span class="sc">%&gt;%</span> <span class="co"># Select specific columns first</span></span>
<span id="cb19-3"><a href="#cb19-3"></a>  <span class="fu">slice</span>(<span class="dv">56</span><span class="sc">:</span><span class="dv">78</span>)</span>
<span id="cb19-4"><a href="#cb19-4"></a><span class="co"># what does the '56:78' do? Change some numbers and run the code again. What happens?</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>If you wanted to select only the rows of data belonging to the Kommetjie site, you could type:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>laminaria <span class="sc">%&gt;%</span></span>
<span id="cb20-2"><a href="#cb20-2"></a>  <span class="fu">filter</span>(site <span class="sc">==</span> <span class="st">"Kommetjie"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The function <code><a href="https://rdrr.io/r/stats/filter.html">filter()</a></code> has two arguments: the first is a dataframe (we specify <code>laminaria</code> in the previous line and the pipe supplies this for us) and the second is an expression that relates to which rows of a particular variable you want to include. Here you include all rows for Kommetjie and you find that in the variable site. It returns a subset that is actually a dataframe itself; it is in the same form as the original dataframe. You could assign that subset of the full dataframe to a new dataframe if you wanted to.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a>lam_kom <span class="ot">&lt;-</span> laminaria <span class="sc">%&gt;%</span> </span>
<span id="cb21-2"><a href="#cb21-2"></a>  <span class="fu">filter</span>(site <span class="sc">==</span> <span class="st">"Kommetjie"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Task B
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="1">
<li>In the script you have started, create a new named dataframe containing only kelps from two of the sites. Check that the new dataframe has the correct values in it. What purpose can the naming of a newly-created dataframe serve?</li>
</ol>
</div>
</div>
</section><section id="basic-stats" class="level2"><h2 class="anchored" data-anchor-id="basic-stats">Basic stats</h2>
<p>Straight out of the box it is possible in R to perform a broad range of statistical calculations on a dataframe. If you wanted to know how many samples you have at Kommetjie, you simply type the following:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a>laminaria <span class="sc">%&gt;%</span> <span class="co"># Tell R which dataset to use</span></span>
<span id="cb22-2"><a href="#cb22-2"></a>  <span class="fu">filter</span>(site <span class="sc">==</span> <span class="st">"Kommetjie"</span>) <span class="sc">%&gt;%</span> <span class="co"># Filter out only records from Kommetjie</span></span>
<span id="cb22-3"><a href="#cb22-3"></a>  <span class="fu">nrow</span>() <span class="co"># Count the number of remaining rows</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Or, if you want to select only the row with the greatest total length:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a>laminaria <span class="sc">%&gt;%</span> <span class="co"># Tell R which dataset to use</span></span>
<span id="cb23-2"><a href="#cb23-2"></a>  <span class="fu">filter</span>(total_length <span class="sc">==</span> <span class="fu">max</span>(total_length)) <span class="co"># Select row with max total length</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Task B
</div>
</div>
<div class="callout-body-container callout-body">
<ol start="2" type="1">
<li>Using pipes, subset the <em>Laminaria</em> data to include regions where the blade thickness is thicker than 5 cm and retain only the columns site, region, blade weight and blade thickness. Now exit RStudio. Pretend it is three days later and revisit your analysis. Calculate the number of entries at Kommetjie and find the row with the greatest length. Do this now.</li>
</ol>
</div>
</div>
<p>Imagine doing this daily as our analysis grows in complexity. It will very soon become quite repetitive if each day you had to retype all these lines of code. And now, six weeks into the research and attendant statistical analysis, you discover that there were some mistakes and some of the raw data were incorrect. Now everything would have to be repeated by retyping it at the command prompt. Or worse still (and bad for repetitive strain injury) doing all of it in SPSS and remembering which buttons to click and then re-clicking them. A pain. Let’s avoid that altogether and do it the right way by writing an R script to automate and annotate all of this.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Dealing with missing data
</div>
</div>
<div class="callout-body-container callout-body">
<p>The <code>.csv</code> file format is usually the most robust for reading data into R. Where you have missing data (blanks), the <code>.csv</code> format separates these by commas. However, there can be problems with blanks if you read in a space-delimited format file. If you are having trouble reading in missing data as blanks, try replacing them in your spreadsheet with <code>NA</code>, the missing data code in R. In Excel, highlight the area of the spreadsheet that includes all the cells you need to fill with <code>NA</code>. Do an Edit/Replace… and leave the ‘Find what:’ textbox blank and in the ‘Replace with:’ textbox enter <code>NA</code>, the missing value code. Once imported into R, the <code>NA</code> values will be recognised as missing data.</p>
</div>
</div>
<p>So far you have calculated the mean and standard deviation of some data in the <em>Laminaria</em> data set. If you have not, please append those lines of code to the end of your script. You can run individual lines of code by highlighting them and pressing <strong>ctrl-Enter</strong> (<strong>cmd-Enter</strong> on a Mac). Do this.</p>
<p>Your file will now look similar to this one, but of course you will have added your own notes and comments as you went along:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="co"># Day_1.R</span></span>
<span id="cb24-2"><a href="#cb24-2"></a><span class="co"># Reads in some data about Laminaria collected along the Cape Peninsula</span></span>
<span id="cb24-3"><a href="#cb24-3"></a><span class="co"># do various data manipulations, analyses and graphs</span></span>
<span id="cb24-4"><a href="#cb24-4"></a><span class="co"># AJ Smit</span></span>
<span id="cb24-5"><a href="#cb24-5"></a><span class="co"># 9 January 2020</span></span>
<span id="cb24-6"><a href="#cb24-6"></a></span>
<span id="cb24-7"><a href="#cb24-7"></a><span class="co"># Find the current working directory (it will be correct if a project was</span></span>
<span id="cb24-8"><a href="#cb24-8"></a><span class="co"># created as instructed earlier)</span></span>
<span id="cb24-9"><a href="#cb24-9"></a><span class="fu">getwd</span>()</span>
<span id="cb24-10"><a href="#cb24-10"></a></span>
<span id="cb24-11"><a href="#cb24-11"></a><span class="co"># If the directory is wrong because you chose not to use an Rworkspace (project),</span></span>
<span id="cb24-12"><a href="#cb24-12"></a><span class="co"># set your directory manually to where the script will be saved and where the data</span></span>
<span id="cb24-13"><a href="#cb24-13"></a><span class="co"># are located</span></span>
<span id="cb24-14"><a href="#cb24-14"></a><span class="co"># setwd("&lt;insert_path_here&gt;")</span></span>
<span id="cb24-15"><a href="#cb24-15"></a></span>
<span id="cb24-16"><a href="#cb24-16"></a><span class="co"># Load libraries</span></span>
<span id="cb24-17"><a href="#cb24-17"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb24-18"><a href="#cb24-18"></a></span>
<span id="cb24-19"><a href="#cb24-19"></a><span class="co"># Load the data</span></span>
<span id="cb24-20"><a href="#cb24-20"></a>laminaria <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"../../data/laminaria.csv"</span>)</span>
<span id="cb24-21"><a href="#cb24-21"></a></span>
<span id="cb24-22"><a href="#cb24-22"></a><span class="co"># Examine the data</span></span>
<span id="cb24-23"><a href="#cb24-23"></a><span class="fu">head</span>(laminaria, <span class="dv">5</span>) <span class="co"># First five lines</span></span>
<span id="cb24-24"><a href="#cb24-24"></a><span class="fu">tail</span>(laminaria, <span class="dv">2</span>) <span class="co"># Last two lines</span></span>
<span id="cb24-25"><a href="#cb24-25"></a><span class="fu">glimpse</span>(laminaria) <span class="co"># A more thorough summary</span></span>
<span id="cb24-26"><a href="#cb24-26"></a><span class="fu">names</span>(laminaria) <span class="co"># THe names of the columns</span></span>
<span id="cb24-27"><a href="#cb24-27"></a></span>
<span id="cb24-28"><a href="#cb24-28"></a><span class="co"># Subsetting data</span></span>
<span id="cb24-29"><a href="#cb24-29"></a>laminaria <span class="sc">%&gt;%</span> <span class="co"># Tell R which dataframe to use</span></span>
<span id="cb24-30"><a href="#cb24-30"></a>  <span class="fu">select</span>(site, total_length) <span class="sc">%&gt;%</span> <span class="co"># Select specific columns</span></span>
<span id="cb24-31"><a href="#cb24-31"></a>  <span class="fu">slice</span>(<span class="dv">56</span><span class="sc">:</span><span class="dv">78</span>) <span class="co"># Select specific rows</span></span>
<span id="cb24-32"><a href="#cb24-32"></a></span>
<span id="cb24-33"><a href="#cb24-33"></a><span class="co"># How many data points do you have at Kommetjie?</span></span>
<span id="cb24-34"><a href="#cb24-34"></a>laminaria <span class="sc">%&gt;%</span></span>
<span id="cb24-35"><a href="#cb24-35"></a>  <span class="fu">filter</span>(site <span class="sc">==</span> <span class="st">"Kommetjie"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-36"><a href="#cb24-36"></a>  <span class="fu">nrow</span>()</span>
<span id="cb24-37"><a href="#cb24-37"></a></span>
<span id="cb24-38"><a href="#cb24-38"></a><span class="co"># The row with the greatest length</span></span>
<span id="cb24-39"><a href="#cb24-39"></a>laminaria <span class="sc">%&gt;%</span> <span class="co"># Tell R which dataset to use</span></span>
<span id="cb24-40"><a href="#cb24-40"></a>  <span class="fu">filter</span>(total_length <span class="sc">==</span> <span class="fu">max</span>(total_length)) <span class="co"># Select row with max total length</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Making sure all the latest edits in your R script have been saved, close your R session. Pretend this is now 2019 and you need to revisit the analysis. Open the file you created in 2017 in RStudio. All you need to do now is highlight the file’s entire contents and hit <strong>ctrl-Enter</strong>.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Stick with <code>.csv</code> files
</div>
</div>
<div class="callout-body-container callout-body">
<p>There are packages in R to read in Excel spreadsheets (e.g., .xlsx), but remember there are likely to be problems reading in formulae, graphs, macros and multiple worksheets. You recommend exporting data deliberately to <code>.csv</code> files (which are also commonly used in other programs). This not only avoids complications, but also allows you to unambiguously identify the data you based your analysis on. This last statement should give you the hint that it is good practice to name your <code>.csv</code> slightly differently each time you export it from Excel, perhaps by appending a reference to the date it was exported.</p>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Remember…
</div>
</div>
<div class="callout-body-container callout-body">
<p>Friends don’t let friends use Excel.</p>
</div>
</div>
</section></section><section id="summary-of-all-variables-in-a-dataframe" class="level1"><h1>Summary of all variables in a dataframe</h1>
<p>Import the data into a dataframe called <code>laminaria</code> once more (if it isn’t already in your <strong>Environment</strong>), and check that it is in order. Once we’re happy that the data have imported correctly, and that you know what the variables are called and what sorts of data they contain, you can dig a little deeper. Try typing:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a><span class="fu">summary</span>(laminaria)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The output is quite informative. It tabulates variables by name, and for each provides summary statistics. For continuous variables, the name, minimum, maximum, first, second (median) and third quartiles, and the mean are provided. For factors (categorical variables), a list of the levels of the factor and the count of each level are given. In either case, the last line of the table indicates how many NAs are contained in the variable. The function <code><a href="https://rdrr.io/r/base/summary.html">summary()</a></code> is useful to remember as it can be applied to many different R objects (e.g., variables, dataframes, models, arrays, <em>etc.</em>) and will give you a summary of that object. You will use it liberally throughout the workshop.</p>
<section id="summary-statistics-by-variable" class="level2"><h2 class="anchored" data-anchor-id="summary-statistics-by-variable">Summary statistics by variable</h2>
<p>This is all very convenient, but you may want to ask R specifically for just the mean of a particular variable. In this case, you simply need to tell R which summary statistic you are interested in, and to specify the variable to apply it to using <code>summarise()</code>. Try typing:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a>laminaria <span class="sc">%&gt;%</span> <span class="co"># Chose the dataframe</span></span>
<span id="cb26-2"><a href="#cb26-2"></a>  <span class="fu">summarise</span>(<span class="at">avg_bld_wdt =</span> <span class="fu">mean</span>(blade_length)) <span class="co"># Calculate mean blade length</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Or, if you wanted to know the mean and standard deviation for the total lengths of all the plants across all sites, do:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a>laminaria <span class="sc">%&gt;%</span> <span class="co"># Tell R that you want to use the 'laminaria' dataframe</span></span>
<span id="cb27-2"><a href="#cb27-2"></a>  <span class="fu">summarise</span>(<span class="at">avg_stp_ln =</span> <span class="fu">mean</span>(total_length), <span class="co"># Create a summary of the mean of the total lengths</span></span>
<span id="cb27-3"><a href="#cb27-3"></a>            <span class="at">sd_stp_ln =</span> <span class="fu">sd</span>(total_length)) <span class="co"># Create a summary of the sd of the total lengths</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Of course, the mean and standard deviation are not the only summary statistic that R can calculate. Try <code><a href="https://rdrr.io/r/base/Extremes.html">max()</a></code>, <code><a href="https://rdrr.io/r/base/Extremes.html">min()</a></code>, <code><a href="https://rdrr.io/r/stats/median.html">median()</a></code>, <code><a href="https://rdrr.io/r/base/range.html">range()</a></code>, <code><a href="https://rdrr.io/r/stats/sd.html">sd()</a></code> and <code><a href="https://rdrr.io/r/stats/cor.html">var()</a></code>. Do they return the values you expected? Now try:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a>laminaria <span class="sc">%&gt;%</span> </span>
<span id="cb28-2"><a href="#cb28-2"></a>  <span class="fu">summarise</span>(<span class="at">avg_stp_ms =</span> <span class="fu">mean</span>(stipe_mass))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The answer probably isn’t what you would expect. Why not? Sometimes, you need to tell R how you want it to deal with missing data. In this case, you have NAs in the named variable, and R takes the cautious approach of giving you the answer of <code>NA</code>, meaning that there are missing values here. This may not seem useful, but as the programmer, you can tell R to respond differently, and it will. Simply append an argument to your function call, and you will get a different response. Type:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a>laminaria <span class="sc">%&gt;%</span> </span>
<span id="cb29-2"><a href="#cb29-2"></a>  <span class="fu">summarise</span>(<span class="at">avg_stp_ms =</span> <span class="fu">mean</span>(stipe_mass, <span class="at">na.rm =</span> T))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The <code>na.rm</code> argument tells R to remove (or more correctly ‘strip’) NAs from the data string before calculating the mean. It now returns the correct answer. Although needing to deal explicitly with missing values in this way can be a bit painful, it does make you more aware of missing data, what the analyses in R are doing, and makes you decide explicitly how you will treat missing data.</p>
</section><section id="more-complex-calculations" class="level2"><h2 class="anchored" data-anchor-id="more-complex-calculations">More complex calculations</h2>
<p>Let’s say you want to calculate something that is not standard in R, say the standard error of the mean for a variable, rather than just the corresponding standard deviation. How can this be done?</p>
<p>The trick is to remember that R is a calculator, so you can use it to do maths, even complex maths (which you won’t do). The formula for standard error is:</p>
<p><span class="math display">\[se = \frac{var}{\sqrt{n}}\]</span></p>
<p>You know that the variance is given by <code><a href="https://rdrr.io/r/stats/cor.html">var()</a></code>, so all you need to do is figure out how to get <code>n</code> and calculate a square root. The simplest way to determine the number of elements in a variable is a call to the function <code><a href="https://rdrr.io/r/base/nrow.html">nrow()</a></code>, as you saw previously. You may therefore calculate standard error with one chunk of code, step by step, using the pipe. Furthermore, by using <code>group_by()</code> you may calculate the standard error for all sites in one go.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a>laminaria <span class="sc">%&gt;%</span> <span class="co"># Select 'laminaria'</span></span>
<span id="cb30-2"><a href="#cb30-2"></a>  <span class="fu">group_by</span>(site) <span class="sc">%&gt;%</span> <span class="co"># Group the dataframe by site</span></span>
<span id="cb30-3"><a href="#cb30-3"></a>  <span class="fu">summarise</span>(<span class="at">var_bl =</span> <span class="fu">var</span>(blade_length), <span class="co"># Calculate variance</span></span>
<span id="cb30-4"><a href="#cb30-4"></a>            <span class="at">n_bl =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span>  <span class="co"># Count number of values</span></span>
<span id="cb30-5"><a href="#cb30-5"></a>  <span class="fu">mutate</span>(<span class="at">se_bl =</span> var_bl <span class="sc">/</span> <span class="fu">sqrt</span>(n_bl)) <span class="co"># Calculate se</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>When calculating the mean, you specified that R should strip the NAs, using the argument <code>na.rm = TRUE</code>. In the example above, you didn’t have NAs in the variable of interest. What happens if you <em>do</em>?</p>
<p>Unfortunately, the call to the function <code><a href="https://rdrr.io/r/base/nrow.html">nrow()</a></code> has no arguments telling R how to treat NAs; instead, they are simply treated as elements of the variable and are therefore counted. The easiest way to resolve this problem is to strip out NAs in advance of any calculations. Try typing:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a>laminaria <span class="sc">%&gt;%</span> </span>
<span id="cb31-2"><a href="#cb31-2"></a>  <span class="fu">select</span>(stipe_mass) <span class="sc">%&gt;%</span> </span>
<span id="cb31-3"><a href="#cb31-3"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>then:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a>laminaria <span class="sc">%&gt;%</span> </span>
<span id="cb32-2"><a href="#cb32-2"></a>  <span class="fu">select</span>(stipe_mass) <span class="sc">%&gt;%</span> </span>
<span id="cb32-3"><a href="#cb32-3"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-4"><a href="#cb32-4"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>You will notice that the function <code><a href="https://rdrr.io/r/stats/na.fail.html">na.omit()</a></code> removes NAs from the variable that is specified as its argument.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Task B
</div>
</div>
<div class="callout-body-container callout-body">
<ol start="3" type="1">
<li><p>Using this new information, <em>calculate</em> the mean stipe mass and the corresponding standard error.</p></li>
<li><p>Create a new data frame from the <em>Laminaria</em> dataset that meets the following criteria: contains only the site column and a new column called total_length_half containing values that are half of the total_length. In this total_length_half column, there are no NAs and all values are less than 100. Hint: think about how the commands should be ordered to produce this data frame!</p></li>
<li><p>Use <code>group_by()</code> and <code>summarise()</code> to find the mean, min, and max blade_length for each site. Also add the number of observations (hint: see <code>?n</code>).</p></li>
<li><p>What was the heaviest stipe measured in each site? Return the columns <code>site</code>, <code>region</code>, and <code>stipe_length</code>.</p></li>
</ol>
</div>
</div>
</section></section><section id="saving-data" class="level1"><h1>Saving data</h1>
<p>A major advantage of R over many other statistics packages is that you can generate exactly the same answers time and time again by simply re-running saved code. However, there are times when you will want to output data to a file that can be read by a spreadsheet program such as Excel (but try not to… please). The simplest general format is .csv (comma-separated values). This format is easily read by Excel, and also by many other software programs. To output a .csv type:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a><span class="fu">write_csv</span>(site_by_region, <span class="at">path =</span> <span class="st">"../../data/kelp_summary.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The first argument is simply the name of an object in R, in this case our table (a data object of class <em>table</em>) of counts by region and site (other sorts of data are available, so play around to see what can be done). The second argument is the name of the file you want to write to. This file will always be written to your working directory, unless otherwise specified by including a different path in the file name. Remember that file names need to be within quotation marks. The resultant file can sadly be opened in Excel.</p>
</section><section id="visualisations" class="level1"><h1>Visualisations</h1>
<p>R has powerful and flexible graphics capabilities. In this Workshop you will not use the traditional graphics (i.e.&nbsp;<em>base graphics</em> in the graphics package automatically loaded in R). You will instead use a package called <strong><code>ggplot2</code></strong> that has the ability for extensive customisation (see the examples at the beginning of tomorrow’s section), so it will cover most of the graphs that you will want to produce. You will spend the next two days working on our <strong><code>ggplot2</code></strong> skills. Here is a quick example of a <strong><code>ggplot2</code></strong> graphic made from two of the kelp variables to show the relationship between them:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> laminaria, <span class="fu">aes</span>(<span class="at">x =</span> stipe_mass, <span class="at">y =</span> stipe_length)) <span class="sc">+</span></span>
<span id="cb34-2"><a href="#cb34-2"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">colour =</span> <span class="st">"salmon"</span>, <span class="at">fill =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb34-3"><a href="#cb34-3"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Stipe mass (kg)"</span>, <span class="at">y =</span> <span class="st">"Stipe length (cm)"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03-workflow_files/figure-html/workflow-1-1.svg" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</section><section id="clearing-the-memory" class="level1"><h1>Clearing the memory</h1>
<p>You will be left with many objects after working through these examples. Note that in RStudio when you quit it can save the Environment if you choose, and so it can retain the objects in memory when you start RStudio again. The choice to save the objects resulting from an R Session until next time can be selected in the Global Options menu (‘Tools’ &gt; ‘Global Options’ &gt; ‘General’ &gt; ‘Save workspace to .RData on exit’). Personally, you never save objects as it is preferable to start on a clean slate when one opens RStudio. Either way, to avoid long load times and clogged memory, it is good practice to clear the objects in memory every now and then unless you can think of a compelling reason not to. This may be done by clicking on the broom icon at the top of the <strong>Environment</strong> pane.</p>
<p>Of course, you could remove an individual object by placing only its name within the brackets of <code><a href="https://rdrr.io/r/base/rm.html">rm()</a></code>. Do not use this line of code carelessly in the middle of your script; doing so will mean that you have to go back and regenerate the objects you accidentally removed — this is more of a nuisance than a train smash, especially for long, complicated scripts, as you will have (I hope!) saved the R script from which the objects in memory can be regenerated at any time.</p>
</section><section id="working-directories" class="level1"><h1>Working directories</h1>
<p>At the beginning of this session you glossed over this topic by setting the working directory via RStudio’s project functionality. This concept is however critically important to understand so you must now cover it in more detail. The current working directory, where R will read and write files, is displayed by RStudio within the title region of the Console. There are a number of ways to change the current working directory:</p>
<ol type="1">
<li><p>Select ‘Session’/‘Set Working Directory’ and then choose from the four options for how to set your working directory depending on your preference</p></li>
<li><p>From within the <strong>Files</strong> pane, navigate to the directory you want to set as the working directory and then select ‘More’/‘Set As Working Directory’ menu item (navigation within the Files pane alone will not change the working directory)</p></li>
<li><p>Use <code><a href="https://rdrr.io/r/base/getwd.html">setwd()</a></code>, providing the name of your desired working directory as a character string</p></li>
</ol>
<p>In the <strong>Files</strong> tab, use the directory structure to navigate to the Intro R Workshop directory… this will differ from person to person. Then under ‘More’, select the small upside down (drill-down) triangle and select ‘Set As Working Directory’. This means that whenever you read or write a file it will always be working in that directory. This gives us the code for setting the directory (below is the code that I would enter in the <strong>Console</strong> on my computer):</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a><span class="fu">setwd</span>(<span class="st">"~/Intro_R_Workshop"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>It will be different for you, but copy it into your script and make a note for future reference.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Working directories
</div>
</div>
<div class="callout-body-container callout-body">
<p>For Windows users, if you copy from a file path the slashes will be the wrong way around and must be changed!</p>
</div>
</div>
<p>You can check that R got this right by typing into the <strong>Console</strong>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a><span class="fu">getwd</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Organising R projects
</div>
</div>
<div class="callout-body-container callout-body">
<p>For every R project, set up a separate directory that includes the scripts, data files and outputs.</p>
</div>
</div>
</section><section id="help" class="level1"><h1>Help</h1>
<p>The help files in R are not readily clear. It requires a bit of work to understand them well, but it is effort worth spending. There is method however to what appears to be madness. Soon you’ll grasp how sensible they really are. The figure below shows the beginning of a help file for a function in R. Please type <code>?read.table()</code> in your console now to bring up this help file in your RStudio GUI.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="../../images/R_helpfiles.png" class="img-fluid figure-img" style="width:60.0%"></p>
<figcaption>R help files</figcaption></figure>
</div>
<p>The first thing you see at the top of the help file in small font is the name of the function, and the package it comes from in curly braces. After this, in very large text, is a very short description of what the function is used for. After this is the ‘Description’ section, which gives a sentence or two more fully explaining the use(s) of the function. The ‘Usage’ then shows all of the arguments that may be given to the function, and what their default settings are. When you write a function in our script you do <em>not</em> need to include all of the possible arguments. The help file shows us all of them so that you know what our options are. In some cases a help file will show the usage of several different functions together. This is done, as is the case here, if these functions forma a sort of ‘family’ and share many common purposes. The ‘Arguments’ section gives a long explanation for what each individual argument may do. The Arguments section here is particularly verbose. Up next is the ‘Details’ section that gives a more in depth description of what the function does. The ‘Value’ section tells us what sort of output you may expect from the function. Some of the more well documented functions, such as this one, will have additional sections that are not a requirement for function documentation. In this case the ‘Memory usage’ and ‘Note’ sections are not things one should always expect to see in help files. Also not always present is a ‘References’ section. Should there be actual published documentation for the function, or the function has been used in a publication for some other purpose, these references tend to be listed here. There are many functions in the <strong><code>vegan</code></strong> package that have been used in dozens of publications. If there is additional reading relevant to the function in question, the authors may also have included a ‘See also’ section, but this is not standard. Lastly, any well documented function should end with an ‘Examples’ section. The code is this section is designed to be able to be copy-pasted directly from the help file into the users R script or console and run <em>as is</em>. It is perhaps a bad habit, but when I am looking up a help file for a function, I tend to look first at the Examples section. And only if I can’t solve my problem with the examples do I actually read the documentation.</p>
</section><section id="other-data" class="level1"><h1>Other data</h1>
<p>Many of the R packages that can be installed come with additional datasets that are available for you to use. They can easily be loaded into the workspace, but the trick is finding them first. This <a href="https://ajsmit.github.io/BCB744/R_datasets.html">presentation</a> shows how to go about doing this, and it focuses in on a few of interesting ones you can use to practice your R skills on. Please explore them – in fact, many exercises in the workshop will require that you find some of you ‘own’ datasets and use them to demonstrate your understanding of important concepts.</p>
</section><section id="r-markdown" class="level1"><h1>R Markdown</h1>
<p>You are now familiar with some basic R commands and can load data and perhaps even make a simple graphic. For this module, I want you to write your homework assignments in R Markdown instead of MS Word. RStudio (the company) has a helpful <a href="http://rmarkdown.rstudio.com/lesson-1.html">tutorial</a> and a short video about R Markdown. After you have watched the video and read the material, go through these short lessons:</p>
<ul>
<li><p><a href="https://rmarkdown.rstudio.com/lesson-2.html">How it Works</a></p></li>
<li><p><a href="https://rmarkdown.rstudio.com/lesson-3.html">Code Chunks</a></p></li>
<li><p><a href="https://rmarkdown.rstudio.com/lesson-4.html">Inline Code</a></p></li>
<li><p><a href="https://rmarkdown.rstudio.com/lesson-8.html">Markdown Basics</a></p></li>
<li><p><a href="https://rmarkdown.rstudio.com/lesson-9.html">Output Formats</a></p></li>
</ul>
<p>For basic understanding of R Markdown, familiarise yourself with the above material. For more depth, please also consult the additional heading items in the menu of the left of these pages.</p>
<p>You’ll need to install a couple more R packages before embarking on the R Markdown tutorials. Use RStudio’s <strong>Packages</strong> panel to install <strong>rmarkdown</strong> and <strong>viridis</strong>. Alternatively, paste these commands in the RStudio Console: <code>install.packages("rmarkdown")</code> and <code>install.packages("viridis")</code>. You can also type <code>install.packages(c("rmarkdown", "viridis"))</code> to install both at the same time.</p>
<p>Don’t worry if you don’t completely understand R Markdown! We’ll go over it at the beginning of class tomorrow if you have any questions.</p>
<p>In this module I encourage you to set the R Markdown output to either html (preferred) or MS Word. Getting it to compile to PDF takes a bit more effort, and the installation of LaTeX – this opens a can of worms, and except for the most geeky persons amongst you (an admirable attribute!), I won’t actively encourage going down this path (for now).</p>
<p>Speaking of .html output: when you create a new R Markdown document (in the menu, go to ‘File’ &gt; ‘New File’ &gt; ‘R Markdown’) and select the ‘From Template’ option in the resultant dialog box, you’ll be offered some interesting and useful templates for your ‘knitted’ final output. Play around with all or some of these, and to spice things up, submit your various assignments using different templates from day to day.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Task B
</div>
</div>
<div class="callout-body-container callout-body">
<ol start="7" type="1">
<li>Recreate the <strong>Exercise in frustration</strong> in R. Repeat all the step, but you are welcome to omit making figures of the montly climatologies.</li>
</ol>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Submission instructions
</div>
</div>
<div class="callout-body-container callout-body">
<p>Submit a R script wherein you provide answers to the Task questions by no later than 8:00 tomorrow.</p>
<p>Provide a <strong>neat and thoroughly annotated and labelled</strong> Rmarkdown file which outlines the graphs and all calculations (as necessary).</p>
<p>Please label the Rmarkdown and resulting HTML files as follows:</p>
<ul>
<li><p><code>BCB744_&lt;first_name&gt;_&lt;last_name&gt;_Task_B.Rmd</code>, and</p></li>
<li><p><code>BCB744_&lt;first_name&gt;_&lt;last_name&gt;_Task_B.html</code></p></li>
</ul>
<p>(the <code>&lt;</code> and <code>&gt;</code> must be omitted as they are used in the example as field indicators only).</p>
<p>Failing to follow these instructions carefully, precisely, and thoroughly will cause you to lose marks, which could cause a significant drop in your score as formatting counts for 15% of the final mark (out of 100%).</p>
<p>Submit your Tasks on iKamva when ready.</p>
</div>
</div>
</section><section id="session-info" class="level1"><h1>Session info</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a><span class="fu">installed.packages</span>()[<span class="fu">names</span>(<span class="fu">sessionInfo</span>()<span class="sc">$</span>otherPkgs), <span class="st">"Version"</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R&gt;     skimr lubridate   forcats   stringr     dplyr     purrr     readr     tidyr 
R&gt;   "2.1.5"   "1.9.3"   "1.0.0"   "1.5.1"   "1.1.4"   "1.0.2"   "2.1.5"   "1.3.1" 
R&gt;    tibble   ggplot2 tidyverse 
R&gt;   "3.2.1"   "3.4.4"   "2.0.0"</code></pre>
</div>
</div>


<!-- -->

</section><div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" id="quarto-reuse"><h2 class="anchored quarto-appendix-heading">Reuse</h2><div class="quarto-appendix-contents"><div><a rel="license" href="https://creativecommons.org/licenses/by/4.0/">CC BY 4.0</a></div></div></section><section class="quarto-appendix-contents" id="quarto-citation"><h2 class="anchored quarto-appendix-heading">Citation</h2><div><div class="quarto-appendix-secondary-label">BibTeX citation:</div><pre class="sourceCode code-with-copy quarto-appendix-bibtex"><code class="sourceCode bibtex">@online{smit2021,
  author = {Smit, AJ},
  title = {3. {R} Workflows},
  date = {2021-01-01},
  url = {https://tangledbank.netlify.app/BCB744/intro_r/03-workflow.html},
  langid = {en}
}
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre><div class="quarto-appendix-secondary-label">For attribution, please cite this work as:</div><div id="ref-smit2021" class="csl-entry quarto-appendix-citeas" role="listitem">
Smit A (2021) 3. R workflows. <a href="https://tangledbank.netlify.app/BCB744/intro_r/03-workflow.html">https://tangledbank.netlify.app/BCB744/intro_r/03-workflow.html.
</a>
</div></div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/tangledbank\.netlify\.app");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
          // default icon
          link.classList.add("external");
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation column-body"><div class="nav-page nav-page-previous">
      <a href="../../BCB744/intro_r/02-working-with-data.html" class="pagination-link" aria-label="2. Working with data and code">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">2. Working with data and code</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../BCB744/intro_r/04-graphics.html" class="pagination-link" aria-label="4. Graphics with ggplot2">
        <span class="nav-page-text">4. Graphics with ggplot2</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb39" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb39-1"><a href="#cb39-1"></a><span class="co">---</span></span>
<span id="cb39-2"><a href="#cb39-2"></a><span class="an">date:</span><span class="co"> "2021-01-01"</span></span>
<span id="cb39-3"><a href="#cb39-3"></a><span class="an">title:</span><span class="co"> "3. R workflows"</span></span>
<span id="cb39-4"><a href="#cb39-4"></a><span class="an">subtitle:</span><span class="co"> "Getting familiar with R and RStudio"</span></span>
<span id="cb39-5"><a href="#cb39-5"></a><span class="co">---</span></span>
<span id="cb39-6"><a href="#cb39-6"></a></span>
<span id="cb39-7"><a href="#cb39-7"></a><span class="al">![](../../images/Skulls_meme.jpg)</span>{width=60%}</span>
<span id="cb39-8"><a href="#cb39-8"></a></span>
<span id="cb39-9"><a href="#cb39-9"></a><span class="at">&gt; "*A dream doesn't become reality through magic; it takes sweat, determination and hard work.*"</span></span>
<span id="cb39-10"><a href="#cb39-10"></a><span class="at">&gt;</span></span>
<span id="cb39-11"><a href="#cb39-11"></a><span class="at">&gt; --- Colin Powell</span></span>
<span id="cb39-12"><a href="#cb39-12"></a></span>
<span id="cb39-13"><a href="#cb39-13"></a><span class="at">&gt; "*Choose a job you love, and you will never have to work a day in your life.*"</span></span>
<span id="cb39-14"><a href="#cb39-14"></a><span class="at">&gt;</span></span>
<span id="cb39-15"><a href="#cb39-15"></a><span class="at">&gt; --- Confucius</span></span>
<span id="cb39-16"><a href="#cb39-16"></a></span>
<span id="cb39-17"><a href="#cb39-17"></a></span>
<span id="cb39-18"><a href="#cb39-18"></a><span class="in">```{r workflow-opts, echo=FALSE}</span></span>
<span id="cb39-19"><a href="#cb39-19"></a><span class="in">knitr::opts_chunk$set(</span></span>
<span id="cb39-20"><a href="#cb39-20"></a><span class="in">  comment = "R&gt;", </span></span>
<span id="cb39-21"><a href="#cb39-21"></a><span class="in">  warning = FALSE, </span></span>
<span id="cb39-22"><a href="#cb39-22"></a><span class="in">  message = FALSE</span></span>
<span id="cb39-23"><a href="#cb39-23"></a><span class="in">)</span></span>
<span id="cb39-24"><a href="#cb39-24"></a><span class="in">```</span></span>
<span id="cb39-25"><a href="#cb39-25"></a></span>
<span id="cb39-26"><a href="#cb39-26"></a>::: {.callout-important}</span>
<span id="cb39-27"><a href="#cb39-27"></a><span class="fu">## Exercise in frustration</span></span>
<span id="cb39-28"><a href="#cb39-28"></a></span>
<span id="cb39-29"><a href="#cb39-29"></a>This exercise requires that you analyse the data provided in <span class="in">`data/SACTN_SAWS.csv`</span> within <span class="in">`&lt;that which shall not be named&gt;`</span>.</span>
<span id="cb39-30"><a href="#cb39-30"></a></span>
<span id="cb39-31"><a href="#cb39-31"></a>This dataset contains monthly seawater temperature time series for a selection of sites around the coast of South Africa.</span>
<span id="cb39-32"><a href="#cb39-32"></a></span>
<span id="cb39-33"><a href="#cb39-33"></a><span class="ss">1. </span>How many sites are there?</span>
<span id="cb39-34"><a href="#cb39-34"></a></span>
<span id="cb39-35"><a href="#cb39-35"></a>For each of the sites there are monthly mean temperatures for every year the 1970s to the 2000s.</span>
<span id="cb39-36"><a href="#cb39-36"></a></span>
<span id="cb39-37"><a href="#cb39-37"></a><span class="ss">2. </span>What is the earliest and latest date for which temperature measurements are available, and which sites do they belong to?</span>
<span id="cb39-38"><a href="#cb39-38"></a></span>
<span id="cb39-39"><a href="#cb39-39"></a><span class="ss">3. </span>In this exercise, please create, **for each site**, a *monthly climatology* using all available data in the time series (see box below).</span>
<span id="cb39-40"><a href="#cb39-40"></a></span>
<span id="cb39-41"><a href="#cb39-41"></a><span class="ss">4. </span>Plot the monthly climatology for each site as a line graph showing the mean temperature for each month.</span>
<span id="cb39-42"><a href="#cb39-42"></a></span>
<span id="cb39-43"><a href="#cb39-43"></a><span class="ss">5. </span>Optionally, you could also calculate the standard deviation of the monthly mean temperatures to represent the variability in temperature within each month. This information can be used to determine the range of typical temperatures for each month.</span>
<span id="cb39-44"><a href="#cb39-44"></a>:::</span>
<span id="cb39-45"><a href="#cb39-45"></a></span>
<span id="cb39-46"><a href="#cb39-46"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb39-47"><a href="#cb39-47"></a><span class="fu">## WMO Guidelines for calculating climatologies</span></span>
<span id="cb39-48"><a href="#cb39-48"></a></span>
<span id="cb39-49"><a href="#cb39-49"></a>The World Meteorological Organization (WMO) provides guidelines for the calculation of climatologies, including monthly and annual climatologies. The WMO guidelines are based on standard meteorological practices and are designed to ensure consistent and accurate calculation of climatological data.</span>
<span id="cb39-50"><a href="#cb39-50"></a></span>
<span id="cb39-51"><a href="#cb39-51"></a>A monthly climatology is a study of the average temperature of seawater (or any environmental variable that is expected to vary over time in a cyclical seasonal manner) over a specified period of time, typically over a year, and broken down by month. Climatologies provide insight into the patterns and trends of weather phenomena and how they vary from one month to another and from year to year. </span>
<span id="cb39-52"><a href="#cb39-52"></a></span>
<span id="cb39-53"><a href="#cb39-53"></a>By analysing long-term data, a monthly climatology can determine the average seawater temperature for each month, the range of temperatures that are typical for each month, as well as finding deviations from expected patterns. This information can be used for a variety of purposes, such as predicting weather patterns, studying the effects of climate change, or understanding the impacts of ocean temperature on marine life.</span>
<span id="cb39-54"><a href="#cb39-54"></a></span>
<span id="cb39-55"><a href="#cb39-55"></a>In essence, a monthly climatology provides a comprehensive and organised overview of the variability and patterns in seawater temperature over a given period of time.</span>
<span id="cb39-56"><a href="#cb39-56"></a></span>
<span id="cb39-57"><a href="#cb39-57"></a>The steps for calculating a monthly climatology according to the WMO guidelines are as follows:</span>
<span id="cb39-58"><a href="#cb39-58"></a></span>
<span id="cb39-59"><a href="#cb39-59"></a><span class="ss">- </span>**Obtain long-term meteorological observations:** The WMO recommends using at least 30 years of data, although longer periods are preferred, to ensure the climatology is representative of the long-term average conditions.</span>
<span id="cb39-60"><a href="#cb39-60"></a></span>
<span id="cb39-61"><a href="#cb39-61"></a><span class="ss">- </span>**Group the observations into months:** The observations should be grouped into months, such as January, February, etc.</span>
<span id="cb39-62"><a href="#cb39-62"></a></span>
<span id="cb39-63"><a href="#cb39-63"></a><span class="ss">- </span>**Calculate the mean value for each month:** For each month over the observation period, the mean value should be calculated as the sum of all the observations for that month divided by the number of observations.</span>
<span id="cb39-64"><a href="#cb39-64"></a></span>
<span id="cb39-65"><a href="#cb39-65"></a><span class="ss">- </span>**Calculate the mean climatology for each month:** The mean climatology for each month is calculated by taking the mean of the monthly mean values for each month over the period of record.</span>
<span id="cb39-66"><a href="#cb39-66"></a></span>
<span id="cb39-67"><a href="#cb39-67"></a><span class="ss">- </span>**Quality control and homogeneity adjustment:** The WMO recommends checking the quality and consistency of the data, and adjusting the data if necessary to correct for inhomogeneities or biases.</span>
<span id="cb39-68"><a href="#cb39-68"></a></span>
<span id="cb39-69"><a href="#cb39-69"></a>The steps for calculating an annual climatology are similar, with the observations being grouped into years rather than months. The WMO guidelines also provide recommendations for the calculation of other types of climatologies, such as seasonal and daily climatologies.</span>
<span id="cb39-70"><a href="#cb39-70"></a></span>
<span id="cb39-71"><a href="#cb39-71"></a>The WMO guidelines are not mandatory, but they are widely used and respected in the meteorological community as the standard for calculating climatologies.</span>
<span id="cb39-72"><a href="#cb39-72"></a>:::</span>
<span id="cb39-73"><a href="#cb39-73"></a></span>
<span id="cb39-74"><a href="#cb39-74"></a><span class="fu"># R Scripts</span></span>
<span id="cb39-75"><a href="#cb39-75"></a></span>
<span id="cb39-76"><a href="#cb39-76"></a>The first step for any project in R is to create a new script. You do this by clicking on the 'New Document' button (in the top left and selecting 'R Script'). This creates an unnamed file in the **Source Editor** pane. Best to save it first of all so you do not lose what you do. 'File'/'Save As'/ and the Working Directory should come up. Type in <span class="in">`Day_1`</span> as the file name and click 'save.' R will automatically add a <span class="in">`.R`</span> extension.</span>
<span id="cb39-77"><a href="#cb39-77"></a></span>
<span id="cb39-78"><a href="#cb39-78"></a>It is recommended that you start your script with some basic information for you to refer back to later. Start with a comment line (the line begins with a <span class="in">`#`</span>) that tells you the name of the script, something about the script, who created it, and the date it was created. In the source editor enter to following lines and save the file again:</span>
<span id="cb39-79"><a href="#cb39-79"></a></span>
<span id="cb39-80"><a href="#cb39-80"></a><span class="in">```{r workflow-header, eval=FALSE}</span></span>
<span id="cb39-81"><a href="#cb39-81"></a><span class="in"># Day_1.R</span></span>
<span id="cb39-82"><a href="#cb39-82"></a><span class="in"># Reads in some data about Laminaria collected along the Cape Peninsula</span></span>
<span id="cb39-83"><a href="#cb39-83"></a><span class="in"># Do various data manipulations, analyses and graphs</span></span>
<span id="cb39-84"><a href="#cb39-84"></a><span class="in"># &lt;your_name&gt;</span></span>
<span id="cb39-85"><a href="#cb39-85"></a><span class="in"># &lt;curent_date&gt;</span></span>
<span id="cb39-86"><a href="#cb39-86"></a><span class="in">```</span></span>
<span id="cb39-87"><a href="#cb39-87"></a></span>
<span id="cb39-88"><a href="#cb39-88"></a>Remember that anything appearing after the <span class="in">`#`</span> is not executed by R as script and is a comment.</span>
<span id="cb39-89"><a href="#cb39-89"></a></span>
<span id="cb39-90"><a href="#cb39-90"></a>It is recommend that for each day of the workshop you start a new script (in the **Source Editor**), type in the code as you go along, and only execute the required lines. That way you will have a record of what you have done.</span>
<span id="cb39-91"><a href="#cb39-91"></a></span>
<span id="cb39-92"><a href="#cb39-92"></a>Below, you will learn how to import the file <span class="in">`laminaria.csv`</span> into R, assign it to a dataframe named <span class="in">`laminaria`</span>, and spend a while looking it over. These data reflect results of a sampling campaign on one of the species of kelps (*Laminaria pallida*) in the Western Cape designed to find the morphometric properties of populations at different sites. We visited 13 different locations along the Cape Peninsula (`site`), and at each site, collected *ca.* 13 specimens of the largest kelps we could find. We then brought the kelps back to the shore and measured/calculated nine morphometric properties of the plants (e.g. the mass of the fronds (`blade_weight`), the frond length (`blade_length`), *etc.*).</span>
<span id="cb39-93"><a href="#cb39-93"></a></span>
<span id="cb39-94"><a href="#cb39-94"></a><span class="fu"># Reading data into R</span></span>
<span id="cb39-95"><a href="#cb39-95"></a></span>
<span id="cb39-96"><a href="#cb39-96"></a>You will now see how easy it is to read data into R. R will read in many types of data, including spreadsheets, text files, binary files and files from other statistical packages and software.</span>
<span id="cb39-97"><a href="#cb39-97"></a></span>
<span id="cb39-98"><a href="#cb39-98"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb39-99"><a href="#cb39-99"></a><span class="fu">## Full stops</span></span>
<span id="cb39-100"><a href="#cb39-100"></a>Unfortunately in South Africa you are taught from a young age to use commas (<span class="in">`,`</span>) instead of full stops (<span class="in">`.`</span>) for decimal places. This simply will not do when you are working with a computer. You must always use a full stop for a decimal place and never insert commas anywhere into any numbers.</span>
<span id="cb39-101"><a href="#cb39-101"></a>:::</span>
<span id="cb39-102"><a href="#cb39-102"></a></span>
<span id="cb39-103"><a href="#cb39-103"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb39-104"><a href="#cb39-104"></a><span class="fu">## Commas</span></span>
<span id="cb39-105"><a href="#cb39-105"></a>R generally thinks that commas mean the user is telling the computer to separate values. So if you think you are typing a big number like 2,300 you may actually end up with two numbers. Never use commas with numbers.</span>
<span id="cb39-106"><a href="#cb39-106"></a>:::</span>
<span id="cb39-107"><a href="#cb39-107"></a></span>
<span id="cb39-108"><a href="#cb39-108"></a><span class="fu">## Preparing data for R</span></span>
<span id="cb39-109"><a href="#cb39-109"></a></span>
<span id="cb39-110"><a href="#cb39-110"></a>Importing data can actually take longer than the statistical analysis itself! In order to avoid as much frustration as possible it is important to remember that for R to be able to analyse your data they need to be in a consistent format, with each variable in a column and each sample in a row. The format within each variable (column) needs to be consistent and is commonly one of the following types: a continuous numeric variable (e.g., fish length (m): <span class="in">`0.133`</span>, <span class="in">`0.145`</span>); a factor or categorical variable (e.g., Month: <span class="in">`Jan`</span>, <span class="in">`Feb`</span> or <span class="in">`1`</span>, <span class="in">`2`</span>, <span class="in">`…`</span>, <span class="in">`12`</span>); a nominal variable (e.g., algal colour: <span class="in">`red`</span>, <span class="in">`green`</span>, <span class="in">`brown`</span>); or a logical variable (i.e., <span class="in">`TRUE`</span> or <span class="in">`FALSE`</span>). You can also use other more specific formats such as dates and times, and more general text formats.</span>
<span id="cb39-111"><a href="#cb39-111"></a></span>
<span id="cb39-112"><a href="#cb39-112"></a>You will learn more about working with data in R --- specifically, you will teach you about the *tidyverse* principles and the distinction between *long* and *wide* format data in more detail on Day 4. For most of our work in R you require our data to be in the long format, but Excel users (poor things!) are more familiar with data stored in the wide format. For now let's bring some data into R and not worry too much about the data being tidy.</span>
<span id="cb39-113"><a href="#cb39-113"></a></span>
<span id="cb39-114"><a href="#cb39-114"></a><span class="fu">## Converting data</span></span>
<span id="cb39-115"><a href="#cb39-115"></a></span>
<span id="cb39-116"><a href="#cb39-116"></a>Before you can read in the *Laminaria* dataset provided for the following exercises, you need to convert the Excel file supplied into a <span class="in">`.csv`</span> file. Open 'laminaria.xlsx' in Excel, then select 'Save As' from the File menu. In the 'Format' drop-down menu, select the option called 'Comma Separated Values', then hit 'Save'. You'll get a warning that formatting will be removed and that only one sheet will be exported; simply 'Continue'. Your working directory should now contain a file called <span class="in">`laminaria.csv`</span>.</span>
<span id="cb39-117"><a href="#cb39-117"></a></span>
<span id="cb39-118"><a href="#cb39-118"></a><span class="fu">## Importing data</span></span>
<span id="cb39-119"><a href="#cb39-119"></a></span>
<span id="cb39-120"><a href="#cb39-120"></a>The easiest way to import data into R is by changing your working directory to be the same as the file path where the file(s) are you want to load. A file path is effectively an address. In most operating systems, if you open the folder where your files are you may click on the navigation bar and it will show you the complete file path. Many people develop the nasty habit of squirrelling away their files within folders within folders within folders within folders... within folders within folders. Please don't do that.</span>
<span id="cb39-121"><a href="#cb39-121"></a></span>
<span id="cb39-122"><a href="#cb39-122"></a>The concept of file paths is either one that you are familiar with, or you've never heard of before. There tends to be little middle ground. Happily, RStudio allows us to circumvent this issue. You do this by using the <span class="in">`Intro_R_Workshop.Rproj`</span> that you may find in the files downloaded for this workshop. If you have not already switched to the <span class="in">`Intro_R_Workshop.Rproj`</span> as outlined in Chapter 2, click on the project button in the top right corner your RStudio window. Then navigate to where you saved <span class="in">`Intro_R_Workshop.Rproj`</span> and select it. Notice that your RStudio has changed a bit and all of the objects you may have previously created in your environment have been removed and any tabs in the source editor pane have been closed. That is fine for now, but it may mean you need to re-open the <span class="in">`Day_1.R`</span> script you just created.</span>
<span id="cb39-123"><a href="#cb39-123"></a></span>
<span id="cb39-124"><a href="#cb39-124"></a>Once you have the working directory set, either by doing it manually with <span class="in">`setwd()`</span> or by loading a project, R will now know where to look for the files you want to read. The function <span class="in">`read_csv()`</span> is the most convenient way to read in raw data. There are several other ways to read in data, but for the purposes of this workshop we'll stick to this one, for now. To find out what it does, you will go to its help entry in the usual way (i.e. <span class="in">`?read_csv`</span>).</span>
<span id="cb39-125"><a href="#cb39-125"></a></span>
<span id="cb39-126"><a href="#cb39-126"></a>All R Help items are in the same format. A short *Description* (of what it does), *Usage*, *Arguments* (the different inputs it requires), *Details* (of what it does), *Value* (what it returns) and *Examples*. Arguments (the parameters that are passed to the function) are the lifeblood of any function, as this is how you provide information to R. You do not need to specify all arguments, as most have appropriate default values for your requirements, and others might not be needed for your particular case.</span>
<span id="cb39-127"><a href="#cb39-127"></a></span>
<span id="cb39-128"><a href="#cb39-128"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb39-129"><a href="#cb39-129"></a><span class="fu">## Data formats</span></span>
<span id="cb39-130"><a href="#cb39-130"></a>R has pedantic requirements for naming variables. It is safest to not use spaces, special characters (e.g., commas, semicolons, any of the shift characters above the numbers), or function names (e.g., mean). One can use 'camelCase', such as <span class="in">`myFirstVariable`</span>, or simply separate the 'parts' of the variable name using an underscore such as in <span class="in">`my_first_variable`</span>. Always make sure to use meaningful names; eventually you will learn to find a balance between meaningfulness and something short that's easy enough to retype repeatedly (although R's ability to use tab completion helps with not having to type long names to often).</span>
<span id="cb39-131"><a href="#cb39-131"></a>:::</span>
<span id="cb39-132"><a href="#cb39-132"></a></span>
<span id="cb39-133"><a href="#cb39-133"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb39-134"><a href="#cb39-134"></a><span class="fu">## Import</span></span>
<span id="cb39-135"><a href="#cb39-135"></a><span class="in">`read_csv()`</span> is simply a 'wrapper' (i.e., a command that modifies) a more basic command called <span class="in">`read_delim()`</span>, which itself allows you to read in many types of files besides <span class="in">`.csv`</span>. To find out more, type <span class="in">`?read_delim()`</span>.</span>
<span id="cb39-136"><a href="#cb39-136"></a>:::</span>
<span id="cb39-137"><a href="#cb39-137"></a></span>
<span id="cb39-138"><a href="#cb39-138"></a><span class="fu">## Loading a file</span></span>
<span id="cb39-139"><a href="#cb39-139"></a></span>
<span id="cb39-140"><a href="#cb39-140"></a>To load the <span class="in">`laminaria.csv`</span> file you created, and assign it to an object name in R, you will use the <span class="in">`read_csv()`</span> function from the **tidyverse** package, so let's make sure it is activated.</span>
<span id="cb39-141"><a href="#cb39-141"></a></span>
<span id="cb39-142"><a href="#cb39-142"></a><span class="in">```{r workflow-library}</span></span>
<span id="cb39-143"><a href="#cb39-143"></a><span class="in">library(tidyverse)</span></span>
<span id="cb39-144"><a href="#cb39-144"></a><span class="in">```</span></span>
<span id="cb39-145"><a href="#cb39-145"></a></span>
<span id="cb39-146"><a href="#cb39-146"></a>Depending on the version of Excel you are using, or perhaps the settings within it, the <span class="in">`laminaria.csv`</span> file you created may be corrupted in different ways. Generally Excel likes to replace the <span class="in">`,`</span> between columns in our .csv files with <span class="in">`;`</span>. This may seem like a triviality but sadly it is not. Lucky for use, the **tidyverse** knows about this problem and they have made a plan. Please open your <span class="in">`laminaria.csv`</span> file and look at which character is being used to separate columns. If it is <span class="in">`,`</span> then you will load the data with <span class="in">`read_csv()`</span>. If the columns are separated with <span class="in">`;`</span> you will use <span class="in">`read_csv2()`</span>.</span>
<span id="cb39-147"><a href="#cb39-147"></a></span>
<span id="cb39-148"><a href="#cb39-148"></a><span class="in">```{r workflow-load-lam, eval=FALSE}</span></span>
<span id="cb39-149"><a href="#cb39-149"></a><span class="in"># Run this if 'laminaria.csv` has columns separated by ','</span></span>
<span id="cb39-150"><a href="#cb39-150"></a><span class="in">laminaria &lt;- read_csv("../../data/laminaria.csv")</span></span>
<span id="cb39-151"><a href="#cb39-151"></a><span class="in"># Run this if 'laminaria.csv` has columns separated by ';'</span></span>
<span id="cb39-152"><a href="#cb39-152"></a><span class="in">laminaria &lt;- read_csv2("../../data/laminaria.csv")</span></span>
<span id="cb39-153"><a href="#cb39-153"></a><span class="in">```</span></span>
<span id="cb39-154"><a href="#cb39-154"></a></span>
<span id="cb39-155"><a href="#cb39-155"></a><span class="in">```{r workflow-load-lam-ghost, echo=FALSE}</span></span>
<span id="cb39-156"><a href="#cb39-156"></a><span class="in">laminaria &lt;- read_csv("../../data/laminaria.csv")</span></span>
<span id="cb39-157"><a href="#cb39-157"></a><span class="in">```</span></span>
<span id="cb39-158"><a href="#cb39-158"></a></span>
<span id="cb39-159"><a href="#cb39-159"></a>If one clicks on the newly created <span class="in">`laminaria`</span> object in the **Environment** pane it will open a new panel that shows the information as a spreadsheet. To go back to your script click the appropriate tab in the **Source Editor** pane. With these data loaded you may now perform analyses on them.</span>
<span id="cb39-160"><a href="#cb39-160"></a></span>
<span id="cb39-161"><a href="#cb39-161"></a>At any point when working in R, you can see exactly what objects are in memory in several ways. First, you can look at the **Environment** tab in RStudio, then **Workspace Browser**. Alternatively you can type either of the following:</span>
<span id="cb39-162"><a href="#cb39-162"></a></span>
<span id="cb39-163"><a href="#cb39-163"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-164"><a href="#cb39-164"></a><span class="in">ls()</span></span>
<span id="cb39-165"><a href="#cb39-165"></a><span class="in"># or</span></span>
<span id="cb39-166"><a href="#cb39-166"></a><span class="in">objects()</span></span>
<span id="cb39-167"><a href="#cb39-167"></a><span class="in">```</span></span>
<span id="cb39-168"><a href="#cb39-168"></a></span>
<span id="cb39-169"><a href="#cb39-169"></a>You can delete an object from memory by specifying the <span class="in">`rm()`</span> function with the name of the object:</span>
<span id="cb39-170"><a href="#cb39-170"></a></span>
<span id="cb39-171"><a href="#cb39-171"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-172"><a href="#cb39-172"></a><span class="in">rm(laminaria)</span></span>
<span id="cb39-173"><a href="#cb39-173"></a><span class="in">```</span></span>
<span id="cb39-174"><a href="#cb39-174"></a></span>
<span id="cb39-175"><a href="#cb39-175"></a>This will of course delete our variable, so you will import it in again using whichever of the following two lines of code matched our Excel situation.</span>
<span id="cb39-176"><a href="#cb39-176"></a></span>
<span id="cb39-177"><a href="#cb39-177"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-178"><a href="#cb39-178"></a><span class="in">laminaria &lt;- read.csv("../../data/laminaria.csv")</span></span>
<span id="cb39-179"><a href="#cb39-179"></a><span class="in">```</span></span>
<span id="cb39-180"><a href="#cb39-180"></a></span>
<span id="cb39-181"><a href="#cb39-181"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb39-182"><a href="#cb39-182"></a><span class="fu">## Managing variables</span></span>
<span id="cb39-183"><a href="#cb39-183"></a>It is good practice to remove variables from memory that you are not using, especially if they are large.</span>
<span id="cb39-184"><a href="#cb39-184"></a>:::</span>
<span id="cb39-185"><a href="#cb39-185"></a></span>
<span id="cb39-186"><a href="#cb39-186"></a><span class="fu"># Working with data</span></span>
<span id="cb39-187"><a href="#cb39-187"></a></span>
<span id="cb39-188"><a href="#cb39-188"></a><span class="fu">## Examine your data</span></span>
<span id="cb39-189"><a href="#cb39-189"></a></span>
<span id="cb39-190"><a href="#cb39-190"></a>Once the data are in R, you need to check there are no glaring errors. It is useful to call up the first few lines of the dataframe using the function <span class="in">`head()`</span>. Try it yourself by typing:</span>
<span id="cb39-191"><a href="#cb39-191"></a></span>
<span id="cb39-192"><a href="#cb39-192"></a><span class="in">```{r, eval=TRUE}</span></span>
<span id="cb39-193"><a href="#cb39-193"></a><span class="in">head(laminaria)</span></span>
<span id="cb39-194"><a href="#cb39-194"></a><span class="in">```</span></span>
<span id="cb39-195"><a href="#cb39-195"></a></span>
<span id="cb39-196"><a href="#cb39-196"></a>This lists the first six lines of each of the variables in the dataframe as a table. You can similarly retrieve the last six lines of a dataframe by an identical call to the function <span class="in">`tail()`</span>. Of course, this works better when you have fewer than 10 or so variables (columns); for larger data sets, things can get a little messy. If you want more or fewer rows in your head or tail, tell R how many rows it is you want by adding this information to your function call. Try typing:</span>
<span id="cb39-197"><a href="#cb39-197"></a></span>
<span id="cb39-198"><a href="#cb39-198"></a><span class="in">```{r, eval=TRUE}</span></span>
<span id="cb39-199"><a href="#cb39-199"></a><span class="in">head(laminaria, n = 3)</span></span>
<span id="cb39-200"><a href="#cb39-200"></a><span class="in">tail(laminaria, n = 2)</span></span>
<span id="cb39-201"><a href="#cb39-201"></a><span class="in">```</span></span>
<span id="cb39-202"><a href="#cb39-202"></a></span>
<span id="cb39-203"><a href="#cb39-203"></a>You can also check the structure of your data by using the <span class="in">`glimpse()`</span> function:</span>
<span id="cb39-204"><a href="#cb39-204"></a></span>
<span id="cb39-205"><a href="#cb39-205"></a><span class="in">```{r, eval=TRUE}</span></span>
<span id="cb39-206"><a href="#cb39-206"></a><span class="in">glimpse(laminaria)</span></span>
<span id="cb39-207"><a href="#cb39-207"></a><span class="in">```</span></span>
<span id="cb39-208"><a href="#cb39-208"></a></span>
<span id="cb39-209"><a href="#cb39-209"></a>This very handy function lists the variables in your dataframe by name, tells you what sorts of data are contained in each variable (e.g., continuous number, discrete factor) and provides an indication of the actual contents of each.</span>
<span id="cb39-210"><a href="#cb39-210"></a></span>
<span id="cb39-211"><a href="#cb39-211"></a>If you wanted only the names of the variables (columns) in the dataframe, you could use:</span>
<span id="cb39-212"><a href="#cb39-212"></a></span>
<span id="cb39-213"><a href="#cb39-213"></a><span class="in">```{r, eval=TRUE}</span></span>
<span id="cb39-214"><a href="#cb39-214"></a><span class="in">names(laminaria)</span></span>
<span id="cb39-215"><a href="#cb39-215"></a><span class="in">```</span></span>
<span id="cb39-216"><a href="#cb39-216"></a></span>
<span id="cb39-217"><a href="#cb39-217"></a>Another option, but by no means the only one remaining, is to install a library called **skimr** and to use thew <span class="in">`skim()`</span> function:</span>
<span id="cb39-218"><a href="#cb39-218"></a></span>
<span id="cb39-221"><a href="#cb39-221"></a><span class="in">```{r}</span></span>
<span id="cb39-222"><a href="#cb39-222"></a><span class="fu">library</span>(skimr)</span>
<span id="cb39-223"><a href="#cb39-223"></a><span class="fu">skim</span>(iris) <span class="co"># using built-in `iris` data</span></span>
<span id="cb39-224"><a href="#cb39-224"></a><span class="in">```</span></span>
<span id="cb39-225"><a href="#cb39-225"></a></span>
<span id="cb39-226"><a href="#cb39-226"></a><span class="fu">## **Tidyverse** sneak peek</span></span>
<span id="cb39-227"><a href="#cb39-227"></a></span>
<span id="cb39-228"><a href="#cb39-228"></a>Before you begin to manipulate our data further I need to briefly introduce you to the <span class="co">[</span><span class="ot">**tidyverse**</span><span class="co">](https://www.tidyverse.org/)</span>. And no introduction can be complete within learning about the *pipe* command, `%&gt;%`. You may type this by pushing the following keys together: **ctrl-shift-m**. The pipe (<span class="in">`%&gt;%`</span>, or <span class="in">`|&gt;`</span> if you selected to use the native pipe operator under 'Global Options') allows you to perform calculations sequentially, which helps us to avoid making errors.</span>
<span id="cb39-229"><a href="#cb39-229"></a></span>
<span id="cb39-230"><a href="#cb39-230"></a>:::{.callout-note appearance="simple"}</span>
<span id="cb39-231"><a href="#cb39-231"></a><span class="fu">## The pipe operator</span></span>
<span id="cb39-232"><a href="#cb39-232"></a></span>
<span id="cb39-233"><a href="#cb39-233"></a>The **pipe operator** allows you to take the output of one function and pass it directly as the input to the next function. This creates a more intuitive and readable way to string together a series of data operations. Instead of nesting functions inside one another, which can quickly become confusing and hard to read, the pipe operator lets you lay out your data processing steps sequentially. This makes your code cleaner and easier to understand, as it clearly outlines the workflow from start to finish, almost like a step-by-step recipe for your data analysis.</span>
<span id="cb39-234"><a href="#cb39-234"></a>:::</span>
<span id="cb39-235"><a href="#cb39-235"></a></span>
<span id="cb39-236"><a href="#cb39-236"></a>The pipe works best in tandem with the following five common functions:</span>
<span id="cb39-237"><a href="#cb39-237"></a></span>
<span id="cb39-238"><a href="#cb39-238"></a><span class="ss">-   </span>Arrange observations (rows) with <span class="in">`arrange()`</span></span>
<span id="cb39-239"><a href="#cb39-239"></a><span class="ss">-   </span>Filter observations (rows) with <span class="in">`filter()`</span></span>
<span id="cb39-240"><a href="#cb39-240"></a><span class="ss">-   </span>Select variables (columns) with <span class="in">`select()`</span></span>
<span id="cb39-241"><a href="#cb39-241"></a><span class="ss">-   </span>Create new variables (columns) with <span class="in">`mutate()`</span></span>
<span id="cb39-242"><a href="#cb39-242"></a><span class="ss">-   </span>Summarise variables (columns) with <span class="in">`summarise()`</span></span>
<span id="cb39-243"><a href="#cb39-243"></a><span class="ss">-   </span>Group observations (rows) with <span class="in">`group_by()`</span></span>
<span id="cb39-244"><a href="#cb39-244"></a></span>
<span id="cb39-245"><a href="#cb39-245"></a>You will cover these functions in more detail on Day 4. For now you will ease ourselves into the code with some simple examples.</span>
<span id="cb39-246"><a href="#cb39-246"></a></span>
<span id="cb39-247"><a href="#cb39-247"></a><span class="fu">## Subsetting</span></span>
<span id="cb39-248"><a href="#cb39-248"></a></span>
<span id="cb39-249"><a href="#cb39-249"></a>Now let's have a look at specific parts of the data. You will likely need to do this in almost every script you write. If you want to refer to a variable, you specify the dataframe then the column name within the <span class="in">`select()`</span> function. In your script type:</span>
<span id="cb39-250"><a href="#cb39-250"></a></span>
<span id="cb39-251"><a href="#cb39-251"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-252"><a href="#cb39-252"></a><span class="in">laminaria %&gt;% # Tell R which dataframe you are using</span></span>
<span id="cb39-253"><a href="#cb39-253"></a><span class="in">  select(site, total_length) # Select only specific columns</span></span>
<span id="cb39-254"><a href="#cb39-254"></a><span class="in">```</span></span>
<span id="cb39-255"><a href="#cb39-255"></a></span>
<span id="cb39-256"><a href="#cb39-256"></a>If you want to only select values from specific columns you insert one more line of code.</span>
<span id="cb39-257"><a href="#cb39-257"></a></span>
<span id="cb39-258"><a href="#cb39-258"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-259"><a href="#cb39-259"></a><span class="in">laminaria %&gt;% </span></span>
<span id="cb39-260"><a href="#cb39-260"></a><span class="in">  select(site, total_length) %&gt;% # Select specific columns first</span></span>
<span id="cb39-261"><a href="#cb39-261"></a><span class="in">  slice(56:78)</span></span>
<span id="cb39-262"><a href="#cb39-262"></a><span class="in"># what does the '56:78' do? Change some numbers and run the code again. What happens?</span></span>
<span id="cb39-263"><a href="#cb39-263"></a><span class="in">```</span></span>
<span id="cb39-264"><a href="#cb39-264"></a></span>
<span id="cb39-265"><a href="#cb39-265"></a>If you wanted to select only the rows of data belonging to the Kommetjie site, you could type:</span>
<span id="cb39-266"><a href="#cb39-266"></a></span>
<span id="cb39-267"><a href="#cb39-267"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-268"><a href="#cb39-268"></a><span class="in">laminaria %&gt;%</span></span>
<span id="cb39-269"><a href="#cb39-269"></a><span class="in">  filter(site == "Kommetjie")</span></span>
<span id="cb39-270"><a href="#cb39-270"></a><span class="in">```</span></span>
<span id="cb39-271"><a href="#cb39-271"></a></span>
<span id="cb39-272"><a href="#cb39-272"></a>The function <span class="in">`filter()`</span> has two arguments: the first is a dataframe (we specify <span class="in">`laminaria`</span> in the previous line and the pipe supplies this for us) and the second is an expression that relates to which rows of a particular variable you want to include. Here you include all rows for Kommetjie and you find that in the variable site. It returns a subset that is actually a dataframe itself; it is in the same form as the original dataframe. You could assign that subset of the full dataframe to a new dataframe if you wanted to.</span>
<span id="cb39-273"><a href="#cb39-273"></a></span>
<span id="cb39-274"><a href="#cb39-274"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-275"><a href="#cb39-275"></a><span class="in">lam_kom &lt;- laminaria %&gt;% </span></span>
<span id="cb39-276"><a href="#cb39-276"></a><span class="in">  filter(site == "Kommetjie")</span></span>
<span id="cb39-277"><a href="#cb39-277"></a><span class="in">```</span></span>
<span id="cb39-278"><a href="#cb39-278"></a></span>
<span id="cb39-279"><a href="#cb39-279"></a>::: callout-important</span>
<span id="cb39-280"><a href="#cb39-280"></a><span class="fu">## Task B</span></span>
<span id="cb39-281"><a href="#cb39-281"></a><span class="ss">1. </span>In the script you have started, create a new named dataframe containing only kelps from two of the sites. Check that the new dataframe has the correct values in it. What purpose can the naming of a newly-created dataframe serve?</span>
<span id="cb39-282"><a href="#cb39-282"></a>:::</span>
<span id="cb39-283"><a href="#cb39-283"></a></span>
<span id="cb39-284"><a href="#cb39-284"></a><span class="fu">## Basic stats</span></span>
<span id="cb39-285"><a href="#cb39-285"></a></span>
<span id="cb39-286"><a href="#cb39-286"></a>Straight out of the box it is possible in R to perform a broad range of statistical calculations on a dataframe. If you wanted to know how many samples you have at Kommetjie, you simply type the following:</span>
<span id="cb39-287"><a href="#cb39-287"></a></span>
<span id="cb39-288"><a href="#cb39-288"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-289"><a href="#cb39-289"></a><span class="in">laminaria %&gt;% # Tell R which dataset to use</span></span>
<span id="cb39-290"><a href="#cb39-290"></a><span class="in">  filter(site == "Kommetjie") %&gt;% # Filter out only records from Kommetjie</span></span>
<span id="cb39-291"><a href="#cb39-291"></a><span class="in">  nrow() # Count the number of remaining rows</span></span>
<span id="cb39-292"><a href="#cb39-292"></a><span class="in">```</span></span>
<span id="cb39-293"><a href="#cb39-293"></a></span>
<span id="cb39-294"><a href="#cb39-294"></a>Or, if you want to select only the row with the greatest total length:</span>
<span id="cb39-295"><a href="#cb39-295"></a></span>
<span id="cb39-296"><a href="#cb39-296"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-297"><a href="#cb39-297"></a><span class="in">laminaria %&gt;% # Tell R which dataset to use</span></span>
<span id="cb39-298"><a href="#cb39-298"></a><span class="in">  filter(total_length == max(total_length)) # Select row with max total length</span></span>
<span id="cb39-299"><a href="#cb39-299"></a><span class="in">```</span></span>
<span id="cb39-300"><a href="#cb39-300"></a></span>
<span id="cb39-301"><a href="#cb39-301"></a>::: callout-important</span>
<span id="cb39-302"><a href="#cb39-302"></a><span class="fu">## Task B</span></span>
<span id="cb39-303"><a href="#cb39-303"></a><span class="ss">2. </span>Using pipes, subset the *Laminaria* data to include regions where the blade thickness is thicker than 5 cm and retain only the columns site, region, blade weight and blade thickness. Now exit RStudio. Pretend it is three days later and revisit your analysis. Calculate the number of entries at Kommetjie and find the row with the greatest length. Do this now.</span>
<span id="cb39-304"><a href="#cb39-304"></a>:::</span>
<span id="cb39-305"><a href="#cb39-305"></a></span>
<span id="cb39-306"><a href="#cb39-306"></a>Imagine doing this daily as our analysis grows in complexity. It will very soon become quite repetitive if each day you had to retype all these lines of code. And now, six weeks into the research and attendant statistical analysis, you discover that there were some mistakes and some of the raw data were incorrect. Now everything would have to be repeated by retyping it at the command prompt. Or worse still (and bad for repetitive strain injury) doing all of it in SPSS and remembering which buttons to click and then re-clicking them. A pain. Let's avoid that altogether and do it the right way by writing an R script to automate and annotate all of this.</span>
<span id="cb39-307"><a href="#cb39-307"></a></span>
<span id="cb39-308"><a href="#cb39-308"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb39-309"><a href="#cb39-309"></a><span class="fu">## Dealing with missing data</span></span>
<span id="cb39-310"><a href="#cb39-310"></a>The <span class="in">`.csv`</span> file format is usually the most robust for reading data into R. Where you have missing data (blanks), the <span class="in">`.csv`</span> format separates these by commas. However, there can be problems with blanks if you read in a space-delimited format file. If you are having trouble reading in missing data as blanks, try replacing them in your spreadsheet with <span class="in">`NA`</span>, the missing data code in R. In Excel, highlight the area of the spreadsheet that includes all the cells you need to fill with <span class="in">`NA`</span>. Do an Edit/Replace... and leave the 'Find what:' textbox blank and in the 'Replace with:' textbox enter <span class="in">`NA`</span>, the missing value code. Once imported into R, the <span class="in">`NA`</span> values will be recognised as missing data.</span>
<span id="cb39-311"><a href="#cb39-311"></a>:::</span>
<span id="cb39-312"><a href="#cb39-312"></a></span>
<span id="cb39-313"><a href="#cb39-313"></a>So far you have calculated the mean and standard deviation of some data in the *Laminaria* data set. If you have not, please append those lines of code to the end of your script. You can run individual lines of code by highlighting them and pressing **ctrl-Enter** (**cmd-Enter** on a Mac). Do this.</span>
<span id="cb39-314"><a href="#cb39-314"></a></span>
<span id="cb39-315"><a href="#cb39-315"></a>Your file will now look similar to this one, but of course you will have added your own notes and comments as you went along:</span>
<span id="cb39-316"><a href="#cb39-316"></a></span>
<span id="cb39-317"><a href="#cb39-317"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-318"><a href="#cb39-318"></a><span class="in"># Day_1.R</span></span>
<span id="cb39-319"><a href="#cb39-319"></a><span class="in"># Reads in some data about Laminaria collected along the Cape Peninsula</span></span>
<span id="cb39-320"><a href="#cb39-320"></a><span class="in"># do various data manipulations, analyses and graphs</span></span>
<span id="cb39-321"><a href="#cb39-321"></a><span class="in"># AJ Smit</span></span>
<span id="cb39-322"><a href="#cb39-322"></a><span class="in"># 9 January 2020</span></span>
<span id="cb39-323"><a href="#cb39-323"></a></span>
<span id="cb39-324"><a href="#cb39-324"></a><span class="in"># Find the current working directory (it will be correct if a project was</span></span>
<span id="cb39-325"><a href="#cb39-325"></a><span class="in"># created as instructed earlier)</span></span>
<span id="cb39-326"><a href="#cb39-326"></a><span class="in">getwd()</span></span>
<span id="cb39-327"><a href="#cb39-327"></a></span>
<span id="cb39-328"><a href="#cb39-328"></a><span class="in"># If the directory is wrong because you chose not to use an Rworkspace (project),</span></span>
<span id="cb39-329"><a href="#cb39-329"></a><span class="in"># set your directory manually to where the script will be saved and where the data</span></span>
<span id="cb39-330"><a href="#cb39-330"></a><span class="in"># are located</span></span>
<span id="cb39-331"><a href="#cb39-331"></a><span class="in"># setwd("&lt;insert_path_here&gt;")</span></span>
<span id="cb39-332"><a href="#cb39-332"></a></span>
<span id="cb39-333"><a href="#cb39-333"></a><span class="in"># Load libraries</span></span>
<span id="cb39-334"><a href="#cb39-334"></a><span class="in">library(tidyverse)</span></span>
<span id="cb39-335"><a href="#cb39-335"></a></span>
<span id="cb39-336"><a href="#cb39-336"></a><span class="in"># Load the data</span></span>
<span id="cb39-337"><a href="#cb39-337"></a><span class="in">laminaria &lt;- read_csv("../../data/laminaria.csv")</span></span>
<span id="cb39-338"><a href="#cb39-338"></a></span>
<span id="cb39-339"><a href="#cb39-339"></a><span class="in"># Examine the data</span></span>
<span id="cb39-340"><a href="#cb39-340"></a><span class="in">head(laminaria, 5) # First five lines</span></span>
<span id="cb39-341"><a href="#cb39-341"></a><span class="in">tail(laminaria, 2) # Last two lines</span></span>
<span id="cb39-342"><a href="#cb39-342"></a><span class="in">glimpse(laminaria) # A more thorough summary</span></span>
<span id="cb39-343"><a href="#cb39-343"></a><span class="in">names(laminaria) # THe names of the columns</span></span>
<span id="cb39-344"><a href="#cb39-344"></a></span>
<span id="cb39-345"><a href="#cb39-345"></a><span class="in"># Subsetting data</span></span>
<span id="cb39-346"><a href="#cb39-346"></a><span class="in">laminaria %&gt;% # Tell R which dataframe to use</span></span>
<span id="cb39-347"><a href="#cb39-347"></a><span class="in">  select(site, total_length) %&gt;% # Select specific columns</span></span>
<span id="cb39-348"><a href="#cb39-348"></a><span class="in">  slice(56:78) # Select specific rows</span></span>
<span id="cb39-349"><a href="#cb39-349"></a></span>
<span id="cb39-350"><a href="#cb39-350"></a><span class="in"># How many data points do you have at Kommetjie?</span></span>
<span id="cb39-351"><a href="#cb39-351"></a><span class="in">laminaria %&gt;%</span></span>
<span id="cb39-352"><a href="#cb39-352"></a><span class="in">  filter(site == "Kommetjie") %&gt;%</span></span>
<span id="cb39-353"><a href="#cb39-353"></a><span class="in">  nrow()</span></span>
<span id="cb39-354"><a href="#cb39-354"></a></span>
<span id="cb39-355"><a href="#cb39-355"></a><span class="in"># The row with the greatest length</span></span>
<span id="cb39-356"><a href="#cb39-356"></a><span class="in">laminaria %&gt;% # Tell R which dataset to use</span></span>
<span id="cb39-357"><a href="#cb39-357"></a><span class="in">  filter(total_length == max(total_length)) # Select row with max total length</span></span>
<span id="cb39-358"><a href="#cb39-358"></a><span class="in">```</span></span>
<span id="cb39-359"><a href="#cb39-359"></a></span>
<span id="cb39-360"><a href="#cb39-360"></a>Making sure all the latest edits in your R script have been saved, close your R session. Pretend this is now 2019 and you need to revisit the analysis. Open the file you created in 2017 in RStudio. All you need to do now is highlight the file's entire contents and hit **ctrl-Enter**.</span>
<span id="cb39-361"><a href="#cb39-361"></a></span>
<span id="cb39-362"><a href="#cb39-362"></a>::: {.callout-note}</span>
<span id="cb39-363"><a href="#cb39-363"></a><span class="fu">## Stick with `.csv` files</span></span>
<span id="cb39-364"><a href="#cb39-364"></a>There are packages in R to read in Excel spreadsheets (e.g., .xlsx), but remember there are likely to be problems reading in formulae, graphs, macros and multiple worksheets. You recommend exporting data deliberately to <span class="in">`.csv`</span> files (which are also commonly used in other programs). This not only avoids complications, but also allows you to unambiguously identify the data you based your analysis on. This last statement should give you the hint that it is good practice to name your <span class="in">`.csv`</span> slightly differently each time you export it from Excel, perhaps by appending a reference to the date it was exported.</span>
<span id="cb39-365"><a href="#cb39-365"></a>:::</span>
<span id="cb39-366"><a href="#cb39-366"></a></span>
<span id="cb39-367"><a href="#cb39-367"></a>::: {.callout-note}</span>
<span id="cb39-368"><a href="#cb39-368"></a><span class="fu">## Remember...</span></span>
<span id="cb39-369"><a href="#cb39-369"></a>Friends don't let friends use Excel.</span>
<span id="cb39-370"><a href="#cb39-370"></a>:::</span>
<span id="cb39-371"><a href="#cb39-371"></a></span>
<span id="cb39-372"><a href="#cb39-372"></a><span class="fu"># Summary of all variables in a dataframe</span></span>
<span id="cb39-373"><a href="#cb39-373"></a></span>
<span id="cb39-374"><a href="#cb39-374"></a>Import the data into a dataframe called <span class="in">`laminaria`</span> once more (if it isn't already in your **Environment**), and check that it is in order. Once we're happy that the data have imported correctly, and that you know what the variables are called and what sorts of data they contain, you can dig a little deeper. Try typing:</span>
<span id="cb39-375"><a href="#cb39-375"></a></span>
<span id="cb39-376"><a href="#cb39-376"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-377"><a href="#cb39-377"></a><span class="in">summary(laminaria)</span></span>
<span id="cb39-378"><a href="#cb39-378"></a><span class="in">```</span></span>
<span id="cb39-379"><a href="#cb39-379"></a></span>
<span id="cb39-380"><a href="#cb39-380"></a>The output is quite informative. It tabulates variables by name, and for each provides summary statistics. For continuous variables, the name, minimum, maximum, first, second (median) and third quartiles, and the mean are provided. For factors (categorical variables), a list of the levels of the factor and the count of each level are given. In either case, the last line of the table indicates how many NAs are contained in the variable. The function <span class="in">`summary()`</span> is useful to remember as it can be applied to many different R objects (e.g., variables, dataframes, models, arrays, *etc.*) and will give you a summary of that object. You will use it liberally throughout the workshop.</span>
<span id="cb39-381"><a href="#cb39-381"></a></span>
<span id="cb39-382"><a href="#cb39-382"></a><span class="fu">## Summary statistics by variable</span></span>
<span id="cb39-383"><a href="#cb39-383"></a></span>
<span id="cb39-384"><a href="#cb39-384"></a>This is all very convenient, but you may want to ask R specifically for just the mean of a particular variable. In this case, you simply need to tell R which summary statistic you are interested in, and to specify the variable to apply it to using <span class="in">`summarise()`</span>. Try typing:</span>
<span id="cb39-385"><a href="#cb39-385"></a></span>
<span id="cb39-386"><a href="#cb39-386"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-387"><a href="#cb39-387"></a><span class="in">laminaria %&gt;% # Chose the dataframe</span></span>
<span id="cb39-388"><a href="#cb39-388"></a><span class="in">  summarise(avg_bld_wdt = mean(blade_length)) # Calculate mean blade length</span></span>
<span id="cb39-389"><a href="#cb39-389"></a><span class="in">```</span></span>
<span id="cb39-390"><a href="#cb39-390"></a></span>
<span id="cb39-391"><a href="#cb39-391"></a>Or, if you wanted to know the mean and standard deviation for the total lengths of all the plants across all sites, do:</span>
<span id="cb39-392"><a href="#cb39-392"></a></span>
<span id="cb39-393"><a href="#cb39-393"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-394"><a href="#cb39-394"></a><span class="in">laminaria %&gt;% # Tell R that you want to use the 'laminaria' dataframe</span></span>
<span id="cb39-395"><a href="#cb39-395"></a><span class="in">  summarise(avg_stp_ln = mean(total_length), # Create a summary of the mean of the total lengths</span></span>
<span id="cb39-396"><a href="#cb39-396"></a><span class="in">            sd_stp_ln = sd(total_length)) # Create a summary of the sd of the total lengths</span></span>
<span id="cb39-397"><a href="#cb39-397"></a><span class="in">```</span></span>
<span id="cb39-398"><a href="#cb39-398"></a></span>
<span id="cb39-399"><a href="#cb39-399"></a>Of course, the mean and standard deviation are not the only summary statistic that R can calculate. Try <span class="in">`max()`</span>, <span class="in">`min()`</span>, <span class="in">`median()`</span>, <span class="in">`range()`</span>, <span class="in">`sd()`</span> and <span class="in">`var()`</span>. Do they return the values you expected? Now try:</span>
<span id="cb39-400"><a href="#cb39-400"></a></span>
<span id="cb39-401"><a href="#cb39-401"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-402"><a href="#cb39-402"></a><span class="in">laminaria %&gt;% </span></span>
<span id="cb39-403"><a href="#cb39-403"></a><span class="in">  summarise(avg_stp_ms = mean(stipe_mass))</span></span>
<span id="cb39-404"><a href="#cb39-404"></a><span class="in">```</span></span>
<span id="cb39-405"><a href="#cb39-405"></a></span>
<span id="cb39-406"><a href="#cb39-406"></a>The answer probably isn't what you would expect. Why not? Sometimes, you need to tell R how you want it to deal with missing data. In this case, you have NAs in the named variable, and R takes the cautious approach of giving you the answer of <span class="in">`NA`</span>, meaning that there are missing values here. This may not seem useful, but as the programmer, you can tell R to respond differently, and it will. Simply append an argument to your function call, and you will get a different response. Type:</span>
<span id="cb39-407"><a href="#cb39-407"></a></span>
<span id="cb39-408"><a href="#cb39-408"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-409"><a href="#cb39-409"></a><span class="in">laminaria %&gt;% </span></span>
<span id="cb39-410"><a href="#cb39-410"></a><span class="in">  summarise(avg_stp_ms = mean(stipe_mass, na.rm = T))</span></span>
<span id="cb39-411"><a href="#cb39-411"></a><span class="in">```</span></span>
<span id="cb39-412"><a href="#cb39-412"></a></span>
<span id="cb39-413"><a href="#cb39-413"></a>The <span class="in">`na.rm`</span> argument tells R to remove (or more correctly 'strip') NAs from the data string before calculating the mean. It now returns the correct answer. Although needing to deal explicitly with missing values in this way can be a bit painful, it does make you more aware of missing data, what the analyses in R are doing, and makes you decide explicitly how you will treat missing data.</span>
<span id="cb39-414"><a href="#cb39-414"></a></span>
<span id="cb39-415"><a href="#cb39-415"></a><span class="fu">## More complex calculations</span></span>
<span id="cb39-416"><a href="#cb39-416"></a></span>
<span id="cb39-417"><a href="#cb39-417"></a>Let's say you want to calculate something that is not standard in R, say the standard error of the mean for a variable, rather than just the corresponding standard deviation. How can this be done?</span>
<span id="cb39-418"><a href="#cb39-418"></a></span>
<span id="cb39-419"><a href="#cb39-419"></a>The trick is to remember that R is a calculator, so you can use it to do maths, even complex maths (which you won't do). The formula for standard error is:</span>
<span id="cb39-420"><a href="#cb39-420"></a></span>
<span id="cb39-421"><a href="#cb39-421"></a>$$se = \frac{var}{\sqrt{n}}$$</span>
<span id="cb39-422"><a href="#cb39-422"></a></span>
<span id="cb39-423"><a href="#cb39-423"></a>You know that the variance is given by <span class="in">`var()`</span>, so all you need to do is figure out how to get <span class="in">`n`</span> and calculate a square root. The simplest way to determine the number of elements in a variable is a call to the function <span class="in">`nrow()`</span>, as you saw previously. You may therefore calculate standard error with one chunk of code, step by step, using the pipe. Furthermore, by using <span class="in">`group_by()`</span> you may calculate the standard error for all sites in one go.</span>
<span id="cb39-424"><a href="#cb39-424"></a></span>
<span id="cb39-425"><a href="#cb39-425"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-426"><a href="#cb39-426"></a><span class="in">laminaria %&gt;% # Select 'laminaria'</span></span>
<span id="cb39-427"><a href="#cb39-427"></a><span class="in">  group_by(site) %&gt;% # Group the dataframe by site</span></span>
<span id="cb39-428"><a href="#cb39-428"></a><span class="in">  summarise(var_bl = var(blade_length), # Calculate variance</span></span>
<span id="cb39-429"><a href="#cb39-429"></a><span class="in">            n_bl = n()) %&gt;%  # Count number of values</span></span>
<span id="cb39-430"><a href="#cb39-430"></a><span class="in">  mutate(se_bl = var_bl / sqrt(n_bl)) # Calculate se</span></span>
<span id="cb39-431"><a href="#cb39-431"></a><span class="in">```</span></span>
<span id="cb39-432"><a href="#cb39-432"></a></span>
<span id="cb39-433"><a href="#cb39-433"></a>When calculating the mean, you specified that R should strip the NAs, using the argument <span class="in">`na.rm = TRUE`</span>. In the example above, you didn't have NAs in the variable of interest. What happens if you *do*?</span>
<span id="cb39-434"><a href="#cb39-434"></a></span>
<span id="cb39-435"><a href="#cb39-435"></a>Unfortunately, the call to the function <span class="in">`nrow()`</span> has no arguments telling R how to treat NAs; instead, they are simply treated as elements of the variable and are therefore counted. The easiest way to resolve this problem is to strip out NAs in advance of any calculations. Try typing:</span>
<span id="cb39-436"><a href="#cb39-436"></a></span>
<span id="cb39-437"><a href="#cb39-437"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-438"><a href="#cb39-438"></a><span class="in">laminaria %&gt;% </span></span>
<span id="cb39-439"><a href="#cb39-439"></a><span class="in">  select(stipe_mass) %&gt;% </span></span>
<span id="cb39-440"><a href="#cb39-440"></a><span class="in">  summarise(n = n())</span></span>
<span id="cb39-441"><a href="#cb39-441"></a><span class="in">```</span></span>
<span id="cb39-442"><a href="#cb39-442"></a></span>
<span id="cb39-443"><a href="#cb39-443"></a>then:</span>
<span id="cb39-444"><a href="#cb39-444"></a></span>
<span id="cb39-445"><a href="#cb39-445"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-446"><a href="#cb39-446"></a><span class="in">laminaria %&gt;% </span></span>
<span id="cb39-447"><a href="#cb39-447"></a><span class="in">  select(stipe_mass) %&gt;% </span></span>
<span id="cb39-448"><a href="#cb39-448"></a><span class="in">  na.omit() %&gt;% </span></span>
<span id="cb39-449"><a href="#cb39-449"></a><span class="in">  summarise(n = n())</span></span>
<span id="cb39-450"><a href="#cb39-450"></a><span class="in">```</span></span>
<span id="cb39-451"><a href="#cb39-451"></a></span>
<span id="cb39-452"><a href="#cb39-452"></a>You will notice that the function <span class="in">`na.omit()`</span> removes NAs from the variable that is specified as its argument.</span>
<span id="cb39-453"><a href="#cb39-453"></a></span>
<span id="cb39-454"><a href="#cb39-454"></a>::: callout-important</span>
<span id="cb39-455"><a href="#cb39-455"></a><span class="fu">## Task B</span></span>
<span id="cb39-456"><a href="#cb39-456"></a><span class="ss">3. </span>Using this new information, *calculate* the mean stipe mass and the corresponding standard error.</span>
<span id="cb39-457"><a href="#cb39-457"></a></span>
<span id="cb39-458"><a href="#cb39-458"></a><span class="ss">4. </span>Create a new data frame from the *Laminaria* dataset that meets the following criteria: contains only the site column and a new column called total_length_half containing values that are half of the total_length. In this total_length_half column, there are no NAs and all values are less than 100. Hint: think about how the commands should be ordered to produce this data frame!</span>
<span id="cb39-459"><a href="#cb39-459"></a></span>
<span id="cb39-460"><a href="#cb39-460"></a><span class="ss">5. </span>Use <span class="in">`group_by()`</span> and <span class="in">`summarise()`</span> to find the mean, min, and max blade_length for each site. Also add the number of observations (hint: see <span class="in">`?n`</span>).</span>
<span id="cb39-461"><a href="#cb39-461"></a></span>
<span id="cb39-462"><a href="#cb39-462"></a><span class="ss">6. </span>What was the heaviest stipe measured in each site? Return the columns <span class="in">`site`</span>, <span class="in">`region`</span>, and <span class="in">`stipe_length`</span>.</span>
<span id="cb39-463"><a href="#cb39-463"></a>:::</span>
<span id="cb39-464"><a href="#cb39-464"></a></span>
<span id="cb39-465"><a href="#cb39-465"></a><span class="fu"># Saving data</span></span>
<span id="cb39-466"><a href="#cb39-466"></a></span>
<span id="cb39-467"><a href="#cb39-467"></a>A major advantage of R over many other statistics packages is that you can generate exactly the same answers time and time again by simply re-running saved code. However, there are times when you will want to output data to a file that can be read by a spreadsheet program such as Excel (but try not to... please). The simplest general format is .csv (comma-separated values). This format is easily read by Excel, and also by many other software programs. To output a .csv type:</span>
<span id="cb39-468"><a href="#cb39-468"></a></span>
<span id="cb39-469"><a href="#cb39-469"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-470"><a href="#cb39-470"></a><span class="in">write_csv(site_by_region, path = "../../data/kelp_summary.csv")</span></span>
<span id="cb39-471"><a href="#cb39-471"></a><span class="in">```</span></span>
<span id="cb39-472"><a href="#cb39-472"></a></span>
<span id="cb39-473"><a href="#cb39-473"></a>The first argument is simply the name of an object in R, in this case our table (a data object of class *table*) of counts by region and site (other sorts of data are available, so play around to see what can be done). The second argument is the name of the file you want to write to. This file will always be written to your working directory, unless otherwise specified by including a different path in the file name. Remember that file names need to be within quotation marks. The resultant file can sadly be opened in Excel.</span>
<span id="cb39-474"><a href="#cb39-474"></a></span>
<span id="cb39-475"><a href="#cb39-475"></a><span class="fu"># Visualisations</span></span>
<span id="cb39-476"><a href="#cb39-476"></a></span>
<span id="cb39-477"><a href="#cb39-477"></a>R has powerful and flexible graphics capabilities. In this Workshop you will not use the traditional graphics (i.e. *base graphics* in the graphics package automatically loaded in R). You will instead use a package called **`ggplot2`** that has the ability for extensive customisation (see the examples at the beginning of tomorrow's section), so it will cover most of the graphs that you will want to produce. You will spend the next two days working on our **`ggplot2`** skills. Here is a quick example of a **`ggplot2`** graphic made from two of the kelp variables to show the relationship between them:</span>
<span id="cb39-478"><a href="#cb39-478"></a></span>
<span id="cb39-479"><a href="#cb39-479"></a><span class="in">```{r workflow-1, warning=FALSE}</span></span>
<span id="cb39-480"><a href="#cb39-480"></a><span class="in">ggplot(data = laminaria, aes(x = stipe_mass, y = stipe_length)) +</span></span>
<span id="cb39-481"><a href="#cb39-481"></a><span class="in">  geom_point(shape = 21, colour = "salmon", fill = "white") +</span></span>
<span id="cb39-482"><a href="#cb39-482"></a><span class="in">  labs(x = "Stipe mass (kg)", y = "Stipe length (cm)")</span></span>
<span id="cb39-483"><a href="#cb39-483"></a><span class="in">```</span></span>
<span id="cb39-484"><a href="#cb39-484"></a></span>
<span id="cb39-485"><a href="#cb39-485"></a><span class="fu"># Clearing the memory</span></span>
<span id="cb39-486"><a href="#cb39-486"></a></span>
<span id="cb39-487"><a href="#cb39-487"></a>You will be left with many objects after working through these examples. Note that in RStudio when you quit it can save the Environment if you choose, and so it can retain the objects in memory when you start RStudio again. The choice to save the objects resulting from an R Session until next time can be selected in the Global Options menu ('Tools' <span class="sc">\&gt;</span> 'Global Options' <span class="sc">\&gt;</span> 'General' <span class="sc">\&gt;</span> 'Save workspace to .RData on exit'). Personally, you never save objects as it is preferable to start on a clean slate when one opens RStudio. Either way, to avoid long load times and clogged memory, it is good practice to clear the objects in memory every now and then unless you can think of a compelling reason not to. This may be done by clicking on the broom icon at the top of the **Environment** pane.</span>
<span id="cb39-488"><a href="#cb39-488"></a></span>
<span id="cb39-489"><a href="#cb39-489"></a>Of course, you could remove an individual object by placing only its name within the brackets of <span class="in">`rm()`</span>. Do not use this line of code carelessly in the middle of your script; doing so will mean that you have to go back and regenerate the objects you accidentally removed --- this is more of a nuisance than a train smash, especially for long, complicated scripts, as you will have (I hope!) saved the R script from which the objects in memory can be regenerated at any time.</span>
<span id="cb39-490"><a href="#cb39-490"></a></span>
<span id="cb39-491"><a href="#cb39-491"></a><span class="fu"># Working directories</span></span>
<span id="cb39-492"><a href="#cb39-492"></a></span>
<span id="cb39-493"><a href="#cb39-493"></a>At the beginning of this session you glossed over this topic by setting the working directory via RStudio's project functionality. This concept is however critically important to understand so you must now cover it in more detail. The current working directory, where R will read and write files, is displayed by RStudio within the title region of the Console. There are a number of ways to change the current working directory:</span>
<span id="cb39-494"><a href="#cb39-494"></a></span>
<span id="cb39-495"><a href="#cb39-495"></a><span class="ss">1.  </span>Select 'Session'/'Set Working Directory' and then choose from the four options for how to set your working directory depending on your preference</span>
<span id="cb39-496"><a href="#cb39-496"></a></span>
<span id="cb39-497"><a href="#cb39-497"></a><span class="ss">2.  </span>From within the **Files** pane, navigate to the directory you want to set as the working directory and then select 'More'/'Set As Working Directory' menu item (navigation within the Files pane alone will not change the working directory)</span>
<span id="cb39-498"><a href="#cb39-498"></a></span>
<span id="cb39-499"><a href="#cb39-499"></a><span class="ss">3.  </span>Use <span class="in">`setwd()`</span>, providing the name of your desired working directory as a character string</span>
<span id="cb39-500"><a href="#cb39-500"></a></span>
<span id="cb39-501"><a href="#cb39-501"></a>In the **Files** tab, use the directory structure to navigate to the Intro R Workshop directory... this will differ from person to person. Then under 'More', select the small upside down (drill-down) triangle and select 'Set As Working Directory'. This means that whenever you read or write a file it will always be working in that directory. This gives us the code for setting the directory (below is the code that I would enter in the **Console** on my computer):</span>
<span id="cb39-502"><a href="#cb39-502"></a></span>
<span id="cb39-503"><a href="#cb39-503"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-504"><a href="#cb39-504"></a><span class="in">setwd("~/Intro_R_Workshop")</span></span>
<span id="cb39-505"><a href="#cb39-505"></a><span class="in">```</span></span>
<span id="cb39-506"><a href="#cb39-506"></a></span>
<span id="cb39-507"><a href="#cb39-507"></a>It will be different for you, but copy it into your script and make a note for future reference.</span>
<span id="cb39-508"><a href="#cb39-508"></a></span>
<span id="cb39-509"><a href="#cb39-509"></a>::: {.callout-note}</span>
<span id="cb39-510"><a href="#cb39-510"></a><span class="fu">## Working directories</span></span>
<span id="cb39-511"><a href="#cb39-511"></a>For Windows users, if you copy from a file path the slashes will be the wrong way around and must be changed!</span>
<span id="cb39-512"><a href="#cb39-512"></a>:::</span>
<span id="cb39-513"><a href="#cb39-513"></a></span>
<span id="cb39-514"><a href="#cb39-514"></a>You can check that R got this right by typing into the **Console**:</span>
<span id="cb39-515"><a href="#cb39-515"></a></span>
<span id="cb39-516"><a href="#cb39-516"></a><span class="in">```{r, eval=FALSE}</span></span>
<span id="cb39-517"><a href="#cb39-517"></a><span class="in">getwd()</span></span>
<span id="cb39-518"><a href="#cb39-518"></a><span class="in">```</span></span>
<span id="cb39-519"><a href="#cb39-519"></a></span>
<span id="cb39-520"><a href="#cb39-520"></a>::: {.callout-note appearance="simple"}</span>
<span id="cb39-521"><a href="#cb39-521"></a><span class="fu">## Organising R projects</span></span>
<span id="cb39-522"><a href="#cb39-522"></a>For every R project, set up a separate directory that includes the scripts, data files and outputs.</span>
<span id="cb39-523"><a href="#cb39-523"></a>:::</span>
<span id="cb39-524"><a href="#cb39-524"></a></span>
<span id="cb39-525"><a href="#cb39-525"></a><span class="fu"># Help</span></span>
<span id="cb39-526"><a href="#cb39-526"></a></span>
<span id="cb39-527"><a href="#cb39-527"></a>The help files in R are not readily clear. It requires a bit of work to understand them well, but it is effort worth spending. There is method however to what appears to be madness. Soon you'll grasp how sensible they really are. The figure below shows the beginning of a help file for a function in R. Please type <span class="in">`?read.table()`</span> in your console now to bring up this help file in your RStudio GUI.</span>
<span id="cb39-528"><a href="#cb39-528"></a></span>
<span id="cb39-529"><a href="#cb39-529"></a><span class="al">![R help files](../../images/R_helpfiles.png)</span>{width=60%}</span>
<span id="cb39-530"><a href="#cb39-530"></a></span>
<span id="cb39-531"><a href="#cb39-531"></a>The first thing you see at the top of the help file in small font is the name of the function, and the package it comes from in curly braces. After this, in very large text, is a very short description of what the function is used for. After this is the 'Description' section, which gives a sentence or two more fully explaining the use(s) of the function. The 'Usage' then shows all of the arguments that may be given to the function, and what their default settings are. When you write a function in our script you do *not* need to include all of the possible arguments. The help file shows us all of them so that you know what our options are. In some cases a help file will show the usage of several different functions together. This is done, as is the case here, if these functions forma a sort of 'family' and share many common purposes. The 'Arguments' section gives a long explanation for what each individual argument may do. The Arguments section here is particularly verbose. Up next is the 'Details' section that gives a more in depth description of what the function does. The 'Value' section tells us what sort of output you may expect from the function. Some of the more well documented functions, such as this one, will have additional sections that are not a requirement for function documentation. In this case the 'Memory usage' and 'Note' sections are not things one should always expect to see in help files. Also not always present is a 'References' section. Should there be actual published documentation for the function, or the function has been used in a publication for some other purpose, these references tend to be listed here. There are many functions in the **`vegan`** package that have been used in dozens of publications. If there is additional reading relevant to the function in question, the authors may also have included a 'See also' section, but this is not standard. Lastly, any well documented function should end with an 'Examples' section. The code is this section is designed to be able to be copy-pasted directly from the help file into the users R script or console and run *as is*. It is perhaps a bad habit, but when I am looking up a help file for a function, I tend to look first at the Examples section. And only if I can't solve my problem with the examples do I actually read the documentation.</span>
<span id="cb39-532"><a href="#cb39-532"></a></span>
<span id="cb39-533"><a href="#cb39-533"></a><span class="fu"># Other data</span></span>
<span id="cb39-534"><a href="#cb39-534"></a></span>
<span id="cb39-535"><a href="#cb39-535"></a>Many of the R packages that can be installed come with additional datasets that are available for you to use. They can easily be loaded into the workspace, but the trick is finding them first. This <span class="co">[</span><span class="ot">presentation</span><span class="co">](https://ajsmit.github.io/BCB744/R_datasets.html)</span> shows how to go about doing this, and it focuses in on a few of interesting ones you can use to practice your R skills on. Please explore them -- in fact, many exercises in the workshop will require that you find some of you 'own' datasets and use them to demonstrate your understanding of important concepts.</span>
<span id="cb39-536"><a href="#cb39-536"></a></span>
<span id="cb39-537"><a href="#cb39-537"></a><span class="fu"># R Markdown</span></span>
<span id="cb39-538"><a href="#cb39-538"></a></span>
<span id="cb39-539"><a href="#cb39-539"></a>You are now familiar with some basic R commands and can load data and perhaps even make a simple graphic. For this module, I want you to write your homework assignments in R Markdown instead of MS Word. RStudio (the company) has a helpful <span class="co">[</span><span class="ot">tutorial</span><span class="co">](http://rmarkdown.rstudio.com/lesson-1.html)</span> and a short video about R Markdown. After you have watched the video and read the material, go through these short lessons:</span>
<span id="cb39-540"><a href="#cb39-540"></a></span>
<span id="cb39-541"><a href="#cb39-541"></a><span class="ss">* </span><span class="co">[</span><span class="ot">How it Works</span><span class="co">](https://rmarkdown.rstudio.com/lesson-2.html)</span></span>
<span id="cb39-542"><a href="#cb39-542"></a></span>
<span id="cb39-543"><a href="#cb39-543"></a><span class="ss">* </span><span class="co">[</span><span class="ot">Code Chunks</span><span class="co">](https://rmarkdown.rstudio.com/lesson-3.html)</span></span>
<span id="cb39-544"><a href="#cb39-544"></a></span>
<span id="cb39-545"><a href="#cb39-545"></a><span class="ss">* </span><span class="co">[</span><span class="ot">Inline Code</span><span class="co">](https://rmarkdown.rstudio.com/lesson-4.html)</span></span>
<span id="cb39-546"><a href="#cb39-546"></a></span>
<span id="cb39-547"><a href="#cb39-547"></a><span class="ss">* </span><span class="co">[</span><span class="ot">Markdown Basics</span><span class="co">](https://rmarkdown.rstudio.com/lesson-8.html)</span></span>
<span id="cb39-548"><a href="#cb39-548"></a></span>
<span id="cb39-549"><a href="#cb39-549"></a><span class="ss">* </span><span class="co">[</span><span class="ot">Output Formats</span><span class="co">](https://rmarkdown.rstudio.com/lesson-9.html)</span></span>
<span id="cb39-550"><a href="#cb39-550"></a></span>
<span id="cb39-551"><a href="#cb39-551"></a>For basic understanding of R Markdown, familiarise yourself with the above material. For more depth, please also consult the additional heading items in the menu of the left of these pages.</span>
<span id="cb39-552"><a href="#cb39-552"></a></span>
<span id="cb39-553"><a href="#cb39-553"></a>You’ll need to install a couple more R packages before embarking on the R Markdown tutorials. Use RStudio’s **Packages** panel to install **rmarkdown** and **viridis**. Alternatively, paste these commands in the RStudio Console: <span class="in">`install.packages("rmarkdown")`</span> and <span class="in">`install.packages("viridis")`</span>. You can also type <span class="in">`install.packages(c("rmarkdown", "viridis"))`</span> to install both at the same time.</span>
<span id="cb39-554"><a href="#cb39-554"></a></span>
<span id="cb39-555"><a href="#cb39-555"></a>Don’t worry if you don’t completely understand R Markdown! We’ll go over it at the beginning of class tomorrow if you have any questions.</span>
<span id="cb39-556"><a href="#cb39-556"></a></span>
<span id="cb39-557"><a href="#cb39-557"></a>In this module I encourage you to set the R Markdown output to either html (preferred) or MS Word. Getting it to compile to PDF takes a bit more effort, and the installation of LaTeX -- this opens a can of worms, and except for the most geeky persons amongst you (an admirable attribute!), I won't actively encourage going down this path (for now).</span>
<span id="cb39-558"><a href="#cb39-558"></a></span>
<span id="cb39-559"><a href="#cb39-559"></a>Speaking of .html output: when you create a new R Markdown document (in the menu, go to 'File' &gt; 'New File' &gt; 'R Markdown') and select the 'From Template' option in the resultant dialog box, you'll be offered some interesting and useful templates for your 'knitted' final output. Play around with all or some of these, and to spice things up, submit your various assignments using different templates from day to day.</span>
<span id="cb39-560"><a href="#cb39-560"></a></span>
<span id="cb39-561"><a href="#cb39-561"></a>::: {.callout-important}</span>
<span id="cb39-562"><a href="#cb39-562"></a><span class="fu">## Task B</span></span>
<span id="cb39-563"><a href="#cb39-563"></a><span class="ss">7. </span>Recreate the **Exercise in frustration** in R. Repeat all the step, but you are welcome to omit making figures of the montly climatologies.</span>
<span id="cb39-564"><a href="#cb39-564"></a>:::</span>
<span id="cb39-565"><a href="#cb39-565"></a></span>
<span id="cb39-566"><a href="#cb39-566"></a>::: {.callout-important}</span>
<span id="cb39-567"><a href="#cb39-567"></a><span class="fu">## Submission instructions</span></span>
<span id="cb39-568"><a href="#cb39-568"></a></span>
<span id="cb39-569"><a href="#cb39-569"></a>Submit a R script wherein you provide answers to the Task questions by no later than</span>
<span id="cb39-570"><a href="#cb39-570"></a>8:00 tomorrow.</span>
<span id="cb39-571"><a href="#cb39-571"></a></span>
<span id="cb39-572"><a href="#cb39-572"></a>Provide a **neat and thoroughly annotated and labelled** Rmarkdown file which</span>
<span id="cb39-573"><a href="#cb39-573"></a>outlines the graphs and all calculations (as necessary).</span>
<span id="cb39-574"><a href="#cb39-574"></a></span>
<span id="cb39-575"><a href="#cb39-575"></a>Please label the Rmarkdown and resulting HTML files as follows:</span>
<span id="cb39-576"><a href="#cb39-576"></a></span>
<span id="cb39-577"><a href="#cb39-577"></a><span class="ss">-   </span><span class="in">`BCB744_&lt;first_name&gt;_&lt;last_name&gt;_Task_B.Rmd`</span>, and</span>
<span id="cb39-578"><a href="#cb39-578"></a></span>
<span id="cb39-579"><a href="#cb39-579"></a><span class="ss">-   </span><span class="in">`BCB744_&lt;first_name&gt;_&lt;last_name&gt;_Task_B.html`</span></span>
<span id="cb39-580"><a href="#cb39-580"></a></span>
<span id="cb39-581"><a href="#cb39-581"></a>(the <span class="in">`&lt;`</span> and <span class="in">`&gt;`</span> must be omitted as they are used in the example as</span>
<span id="cb39-582"><a href="#cb39-582"></a>field indicators only).</span>
<span id="cb39-583"><a href="#cb39-583"></a></span>
<span id="cb39-584"><a href="#cb39-584"></a>Failing to follow these instructions carefully, precisely, and</span>
<span id="cb39-585"><a href="#cb39-585"></a>thoroughly will cause you to lose marks, which could cause a significant</span>
<span id="cb39-586"><a href="#cb39-586"></a>drop in your score as formatting counts for 15% of the final mark (out</span>
<span id="cb39-587"><a href="#cb39-587"></a>of 100%).</span>
<span id="cb39-588"><a href="#cb39-588"></a></span>
<span id="cb39-589"><a href="#cb39-589"></a>Submit your Tasks on iKamva when ready.</span>
<span id="cb39-590"><a href="#cb39-590"></a>:::</span>
<span id="cb39-591"><a href="#cb39-591"></a></span>
<span id="cb39-592"><a href="#cb39-592"></a><span class="fu"># Session info</span></span>
<span id="cb39-593"><a href="#cb39-593"></a></span>
<span id="cb39-594"><a href="#cb39-594"></a><span class="in">```{r, echo=TRUE, include=TRUE}</span></span>
<span id="cb39-595"><a href="#cb39-595"></a><span class="in">installed.packages()[names(sessionInfo()$otherPkgs), "Version"]</span></span>
<span id="cb39-596"><a href="#cb39-596"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>© 2024, AJ Smit</p>
</div>   
    <div class="nav-footer-center">

<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/ajsmit/tangled_bank/edit/main/BCB744/intro_r/03-workflow.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/ajsmit/tangled_bank/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div><div class="cookie-consent-footer"><a href="#" id="open_preferences_center">Cookie Preferences</a></div></div>
    <div class="nav-footer-right">
<p>This page is built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>


<script src="../../site_libs/quarto-html/zenscroll-min.js"></script>
</body></html>